{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "HIV-1 IIIB Virus https://aidsreagent.org/reagentdetail.cfm?t=viruses&id=607\n",
    "\n",
    "# Alignment \n",
    "Firstly, HIV genome indices created by changing sequence provided by Judd to `.fa` format. It didn't work as we needed gtf file! \n",
    "\n",
    "Ok! I'm switching to the [HIV-1 clone pIIIB from USA, complete genome\n",
    "](https://www.ncbi.nlm.nih.gov/nucleotide/EU541617.1?report=genbank&log$=nuclalign&blast_rank=1&RID=8B8ETWT201R) which is highly matched with Judd's clone (using Blast alignment). Then, I donwloaded `.fa` and `.gff3` format for downstream analysis. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# %%bash \n",
    "# # make hiv genome indices \n",
    "# bowtie2-build HIV/EU541617.1/EU541617.1.fasta HIV/EU541617.1/index\n",
    "# # Aligning to hiv genome \n",
    "# for f in fastq_unmapped/*; do\n",
    "# \to=${f/.fastq/.bam};\n",
    "#  \to=${o/fastq_unmapped/bam};\n",
    "#  \tbowtie2 --sensitive -N 1 -x HIV/EU541617.1/index -U $f | samtools sort -o $o;\n",
    "# done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Apr 03 03:33:26 ..... started STAR run\n",
      "Apr 03 03:33:26 ... starting to generate Genome files\n",
      "Apr 03 03:33:26 ... starting to sort Suffix Array. This may take a long time...\n",
      "Apr 03 03:33:26 ... sorting Suffix Array chunks and saving them to disk...\n",
      "Apr 03 03:33:26 ... loading chunks from disk, packing SA...\n",
      "Apr 03 03:33:26 ... finished generating suffix array\n",
      "Apr 03 03:33:26 ... generating Suffix Array index\n",
      "Apr 03 03:33:26 ... completed Suffix Array index\n",
      "Apr 03 03:33:26 ..... processing annotations GTF\n",
      "Apr 03 03:33:26 ... writing Genome to disk ...\n",
      "Apr 03 03:33:26 ... writing Suffix Array to disk ...\n",
      "Apr 03 03:33:26 ... writing SAindex to disk\n",
      "Apr 03 03:33:26 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# use STAR \n",
    "STAR --runThreadN 16 \\\n",
    "--runMode genomeGenerate \\\n",
    "--genomeDir HIV/EU541617.1/ \\\n",
    "--genomeFastaFiles HIV/EU541617.1/EU541617.1.fasta \\\n",
    "--sjdbGTFfile HIV/EU541617.1/EU541617.1.gff3 \\\n",
    "--sjdbOverhang 99 --genomeSAindexNbases 8 --sjdbGTFfeatureExon CDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.METTL14.input.fastq --outFileNamePrefix bam/s23.hiv.METTL14.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.METTL14.m6A.fastq --outFileNamePrefix bam/s23.hiv.METTL14.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.METTL3.input.fastq --outFileNamePrefix bam/s23.hiv.METTL3.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.METTL3.m6A.fastq --outFileNamePrefix bam/s23.hiv.METTL3.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.NT.input.fastq --outFileNamePrefix bam/s23.hiv.NT.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.NT.m6A.fastq --outFileNamePrefix bam/s23.hiv.NT.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.Virma.input.fastq --outFileNamePrefix bam/s23.hiv.Virma.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.Virma.m6A.fastq --outFileNamePrefix bam/s23.hiv.Virma.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.WTAP.input.fastq --outFileNamePrefix bam/s23.hiv.WTAP.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.WTAP.m6A.fastq --outFileNamePrefix bam/s23.hiv.WTAP.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.METTL14.input.fastq --outFileNamePrefix bam/s24.hiv.METTL14.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.METTL14.m6A.fastq --outFileNamePrefix bam/s24.hiv.METTL14.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.METTL3.input.fastq --outFileNamePrefix bam/s24.hiv.METTL3.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.METTL3.m6A.fastq --outFileNamePrefix bam/s24.hiv.METTL3.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.NT.input.fastq --outFileNamePrefix bam/s24.hiv.NT.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.NT.m6A.fastq --outFileNamePrefix bam/s24.hiv.NT.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.Virma.input.fastq --outFileNamePrefix bam/s24.hiv.Virma.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.Virma.m6A.fastq --outFileNamePrefix bam/s24.hiv.Virma.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.WTAP.input.fastq --outFileNamePrefix bam/s24.hiv.WTAP.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.WTAP.m6A.fastq --outFileNamePrefix bam/s24.hiv.WTAP.m6A.bam --limitBAMsortRAM 1000000000\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# STAR --genomeLoad LoadAndExit --genomeDir HIV/EU541617.1/\n",
    "for f in fastq_unmapped/*; do\n",
    "\to=${f/.fastq/.bam};\n",
    "\to=${o/fastq_unmapped/bam};\n",
    "\techo STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn $f --outFileNamePrefix $o --limitBAMsortRAM 1000000000;\n",
    "# \tSTAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn $f --outFileNamePrefix $o --limitBAMsortRAM 1000000000;\n",
    "done\n",
    "# STAR --genomeLoad Remove --genomeDir HIV/EU541617.1/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'bam/s23.hiv.METTL14.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.METTL14.input.bam'\n",
      "renamed 'bam/s23.hiv.METTL14.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.METTL14.m6A.bam'\n",
      "renamed 'bam/s23.hiv.METTL3.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.METTL3.input.bam'\n",
      "renamed 'bam/s23.hiv.METTL3.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.METTL3.m6A.bam'\n",
      "renamed 'bam/s23.hiv.NT.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.NT.input.bam'\n",
      "renamed 'bam/s23.hiv.NT.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.NT.m6A.bam'\n",
      "renamed 'bam/s23.hiv.Virma.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.Virma.input.bam'\n",
      "renamed 'bam/s23.hiv.Virma.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.Virma.m6A.bam'\n",
      "renamed 'bam/s23.hiv.WTAP.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.WTAP.input.bam'\n",
      "renamed 'bam/s23.hiv.WTAP.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.WTAP.m6A.bam'\n",
      "renamed 'bam/s24.hiv.METTL14.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.METTL14.input.bam'\n",
      "renamed 'bam/s24.hiv.METTL14.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.METTL14.m6A.bam'\n",
      "renamed 'bam/s24.hiv.METTL3.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.METTL3.input.bam'\n",
      "renamed 'bam/s24.hiv.METTL3.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.METTL3.m6A.bam'\n",
      "renamed 'bam/s24.hiv.NT.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.NT.input.bam'\n",
      "renamed 'bam/s24.hiv.NT.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.NT.m6A.bam'\n",
      "renamed 'bam/s24.hiv.Virma.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.Virma.input.bam'\n",
      "renamed 'bam/s24.hiv.Virma.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.Virma.m6A.bam'\n",
      "renamed 'bam/s24.hiv.WTAP.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.WTAP.input.bam'\n",
      "renamed 'bam/s24.hiv.WTAP.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.WTAP.m6A.bam'\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in bam/*Aligned.sortedByCoord.out.bam; do o=${f/Aligned.sortedByCoord.out.bam/}; mv -v $f $o; done "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All `bam`  files (mapped to human genome or hiv genome) renamed in this format: \n",
    "> `<sample>.<species>.<treatment>.<input/m6A>.bam`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HIV\n",
    "\n",
    "### Peak calling for HIV genome using exomePeak"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(exomePeak)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Stage: index bam file s23.hiv.NT.m6A.bam\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Divide transcriptome into chr-gene-batch sections ...\"\n",
      "[1] \"Stage: index bam file s24.hiv.NT.m6A.bam\"\n",
      "[1] \"Stage: index bam file s23.hiv.NT.input.bam\"\n",
      "[1] \"Stage: index bam file s24.hiv.NT.input.bam\"\n",
      "[1] \"Get Reads Count ...\"\n",
      "[1] \"This step may take a few hours ...\"\n",
      "[1] \"100 %\"\n",
      "[1] \"Get all the peaks ...\"\n",
      "[1] \"Get the consistent peaks ...\"\n",
      "[1] \"---------------------------------\"\n",
      "[1] \"The bam files used:\"\n",
      "[1] \"2 IP replicate(s)\"\n",
      "[1] \"2 Input replicate(s)\"\n",
      "[1] \"---------------------------------\"\n",
      "[1] \"Peak calling result: \"\n",
      "[1] \"4 peaks detected on merged data.\"\n",
      "[1] \"Please check 'peak.bed/xls' under ../exomepeak//hiv.control\"\n",
      "[1] \"4 consistent peaks detected on every replicates. (Recommended list)\"\n",
      "[1] \"Please check 'con_peak.bed/xls' under ../exomepeak//hiv.control\"\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "GENE_ANNO_GTF=\"/rumi/shams/abe/People/Judd/HIV/EU541617.1/EU541617.1.gff3\"\n",
    "txdb = makeTxDbFromGFF(GENE_ANNO_GTF)\n",
    "\n",
    "setwd(\"~/People/Judd/bam\")\n",
    "res <- exomepeak(\n",
    "    TXDB = txdb,\n",
    "    IP_BAM=list.files(pattern='hiv.NT.m6A.bam$'),\n",
    "    INPUT_BAM=list.files(pattern='hiv.NT.input.bam$'),\n",
    "    OUTPUT_DIR='../exomepeak/',\n",
    "    EXPERIMENT_NAME='hiv.control'\n",
    ")\n",
    "saveRDS(res, '../exomepeak/hiv.control/results.rds')\n",
    "setwd(\"~/People/Judd\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# chr\tchromStart\tchromEnd\tname\tscore\tstrand\tthickStart\tthickEnd\titemRgb\tblockCount\tblockSizes\tblockStarts\r\n",
      "EU541617.1\t5519\t7966\ttat\t9.99999999999999e-283\t+\t5519\t7966\t0\t2\t71,46,\t0,2401\r\n",
      "EU541617.1\t5514\t8195\trev\t0\t+\t5514\t8195\t0\t2\t76,275,\t0,2406\r\n",
      "EU541617.1\t7828\t8337\tenv\t0\t+\t7828\t8337\t0\t1\t509,\t0\r\n",
      "EU541617.1\t8338\t8959\tnef\t0\t+\t8338\t8959\t0\t1\t621,\t0\r\n"
     ]
    }
   ],
   "source": [
    "cat exomepeak/hiv.control/peak.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot _Guitar_ \n",
    "Now, let's draw metagene plots for control samples:\n",
    "(using seprate env: `Guitar`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# conda environments:\n",
      "#\n",
      "base                     /rumi/shams/abe/anaconda3\n",
      "ACTIONet                 /rumi/shams/abe/anaconda3/envs/ACTIONet\n",
      "Keras                    /rumi/shams/abe/anaconda3/envs/Keras\n",
      "RADAR                    /rumi/shams/abe/anaconda3/envs/RADAR\n",
      "bedtools                 /rumi/shams/abe/anaconda3/envs/bedtools\n",
      "bowtie2                  /rumi/shams/abe/anaconda3/envs/bowtie2\n",
      "cellrangerenv            /rumi/shams/abe/anaconda3/envs/cellrangerenv\n",
      "cgat                     /rumi/shams/abe/anaconda3/envs/cgat\n",
      "chipseq                  /rumi/shams/abe/anaconda3/envs/chipseq\n",
      "deseq                 *  /rumi/shams/abe/anaconda3/envs/deseq\n",
      "deseq_1.20               /rumi/shams/abe/anaconda3/envs/deseq_1.20\n",
      "guitar                   /rumi/shams/abe/anaconda3/envs/guitar\n",
      "htseq                    /rumi/shams/abe/anaconda3/envs/htseq\n",
      "intermine                /rumi/shams/abe/anaconda3/envs/intermine\n",
      "meRIPseq                 /rumi/shams/abe/anaconda3/envs/meRIPseq\n",
      "motif                    /rumi/shams/abe/anaconda3/envs/motif\n",
      "piranha                  /rumi/shams/abe/anaconda3/envs/piranha\n",
      "scanpy                   /rumi/shams/abe/anaconda3/envs/scanpy\n",
      "snakemake-tutorial       /rumi/shams/abe/anaconda3/envs/snakemake-tutorial\n",
      "yikai                    /rumi/shams/abe/anaconda3/envs/yikai\n",
      "yikai_prime              /rumi/shams/abe/anaconda3/envs/yikai_prime\n",
      "\n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "conda env list "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Guitar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"There are 9 transcripts of 9 genes in the genome.\"\n",
      "[1] \"total 9 transcripts extracted ...\"\n",
      "[1] \"total 9 transcripts left after ambiguity filter ...\"\n",
      "[1] \"total 9 transcripts left after check chromosome validity ...\"\n",
      "[1] \"total 0 mRNAs left after component length filter ...\"\n",
      "[1] \"total 0 ncRNAs left after ncRNA length filter ...\"\n",
      "[1] \"generate components for all tx\"\n",
      "[1] \"generate components for mRNA\"\n",
      "[1] \"generate components for lncRNA\"\n",
      "[1] \"generate chiped transcriptome\"\n",
      "[1] \"generate coverage checking ranges for tx\"\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "GENE_ANNO_GTF=\"/rumi/shams/abe/People/Judd/HIV/EU541617.1/EU541617.1.gff3\"\n",
    "txdb = makeTxDbFromGFF(GENE_ANNO_GTF)\n",
    "guitarTxdb <- makeGuitarTxdb(txdb = txdb, txPrimaryOnly = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GRanges object with 29 ranges and 2 metadata columns:\n",
      "        seqnames    ranges strand |      txType componentType\n",
      "           <Rle> <IRanges>  <Rle> | <character>   <character>\n",
      "  gag EU541617.1  -664-335      + |          tx      promoter\n",
      "  pol EU541617.1  631-1630      + |          tx      promoter\n",
      "  vif EU541617.1 3587-4586      + |          tx      promoter\n",
      "  vpr EU541617.1 4105-5104      + |          tx      promoter\n",
      "  tat EU541617.1 4376-5375      + |          tx      promoter\n",
      "  ...        ...       ...    ... .         ...           ...\n",
      "  tat EU541617.1 7967-8966      + |          tx          tail\n",
      "  rev EU541617.1 8196-9195      + |          tx          tail\n",
      "  vpu EU541617.1 5853-6852      + |          tx          tail\n",
      "  env EU541617.1 8338-9337      + |          tx          tail\n",
      "  nef EU541617.1 8960-9959      + |          tx          tail\n",
      "  -------\n",
      "  seqinfo: 1 sequence from an unspecified genome; no seqlengths\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "guitarTxdb$tx$txComponentGRange\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%R\n",
    "# guitarTxdb\n",
    "# GuitarPlot(txTxdb = guitarTxdb,stBedFiles = \"exomepeak/hiv.control/peak.bed\"\n",
    "#            ,headOrtail = FALSE,enableCI = FALSE\n",
    "#            ,pltTxType = c(\"mrna\")\n",
    "#            ,miscOutFilePrefix = \"hiv.control\"\n",
    "#            )\n",
    "# GuitarPlot(txTxdb = guitarTxdb,stBedFiles = list(\"exomepeak/control/peak.bed\"),miscOutFilePrefix = \"hiv.control\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !magick -density 300 plots/control_mrna_test.pdf -flatten -quality 90 plots/control_mrna_test.png"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Motif analysis "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "cp exomepeak/control/motifs_of_interest.txt exomepeak/hiv.control/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fri Apr  3 07:04:03 UTC 2020\n",
      "RNA, Processing peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt\n",
      "RNA, Step 2: skip seed optimization, use -motiffile_rna option instead.\n",
      "RNA, Step 3: evaluation of motif significances.\n",
      "Processing motif [AG]GAC.\n",
      "MI=0.13792539\n",
      "Shuffle rank=10000\n",
      "Z-score=0.000\n",
      "Sigificance category= NOT-SIGNIF\n",
      "Robustness (jn_f=3)=0/10\n",
      "\n",
      "Processing motif [AGT][AG]AC[ACT].\n",
      "MI=0.00000000\n",
      "Shuffle rank=10000\n",
      "Z-score=0.000\n",
      "Sigificance category= NOT-SIGNIF\n",
      "Robustness (jn_f=3)=0/10\n",
      "\n",
      "RNA, Step 3.5: creating binary expression profiles.\n",
      "RNA, Step 4: discovery of distance constraints.\n",
      "Read 2 motifs.\n",
      "Read 8 (genes) x 2 (conditions) matrix.\n",
      "Read 8 (genes) x 1 (conditions) matrix.\n",
      "Reading sequences and building motif profiles ...Done.\n",
      "Average sequence length is 400.\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AG]GAC, present in 7 genes.\n",
      "       Average distance: MI=0.0000, rank=10000/10000, Z=0.0000, mbins=2\n",
      "  Orientation bias {5'): MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "  Orientation bias (3'): MI=1.0000, rank=301/10000, Z=4.8784\n",
      "       Copy number (1+): 7 genes, MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "       Copy number (2+): 7 genes, MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "       Copy number (3+): 7 genes, MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "       Copy number (4+): 7 genes, MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "       Copy number (5+): 5 genes, MI=0.0488, rank=10000/10000, Z=-0.4125\n",
      "       Copy number (6+): 5 genes, MI=0.0488, rank=10000/10000, Z=-0.4159\n",
      "       Copy number (7+): 4 genes, MI=0.0000, rank=10000/10000, Z=-0.6513\n",
      "       Copy number (8+): 4 genes, MI=0.0000, rank=10000/10000, Z=-0.6386\n",
      "       Copy number (9+): 4 genes, MI=0.0000, rank=10000/10000, Z=-0.6409\n",
      "       Copy number (10+): 1 genes, MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT][AG]AC[ACT], present in 8 genes.\n",
      "       Average distance: MI=0.0000, rank=10000/10000, Z=0.0000, mbins=2\n",
      "  Orientation bias {5'): MI=0.0000, rank=10000/10000, Z=0.0000\n",
      "  Orientation bias (3'): MI=0.0000, rank=10000/10000, Z=0.0000\n",
      "       Copy number (1+): 8 genes, MI=0.0000, rank=10000/10000, Z=0.0000\n",
      "       Copy number (2+): 7 genes, MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "       Copy number (3+): 7 genes, MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "       Copy number (4+): 7 genes, MI=0.1379, rank=10000/10000, Z=0.0000\n",
      "       Copy number (5+): 6 genes, MI=0.0000, rank=10000/10000, Z=-0.8597\n",
      "       Copy number (6+): 6 genes, MI=0.0000, rank=10000/10000, Z=-0.8534\n",
      "       Copy number (7+): 6 genes, MI=0.0000, rank=10000/10000, Z=-0.8581\n",
      "       Copy number (8+): 6 genes, MI=0.0000, rank=10000/10000, Z=-0.8452\n",
      "       Copy number (9+): 4 genes, MI=0.1887, rank=4927/10000, Z=0.4040\n",
      "       Copy number (10+): 4 genes, MI=0.1887, rank=4823/10000, Z=0.3941\n",
      "\n",
      "RNA, Step 5.5: conservation.\n",
      "Setting -rna to 1.\n",
      "[AG]GAC\t98.12\n",
      "[AGT][AG]AC[ACT]\t21.42\n",
      "RNA, Step 5: summarize information.\n",
      "Reading RNA motifs mi_dist output ...Done.\n",
      "Parsing RNA motif file ... Done.\n",
      "Reading mi_signif output file for RNA motifs ...!!!! 1005992!!!! -3463040Done.\n",
      "Got 0 motifs\n",
      "Writing good columns to peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.signifcolumns ...Done.\n",
      "Outputing summary ... Done.\n",
      "HI!HELLO!peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.binary created.\n",
      "RNA, Step 5.3: generate profiles for motifs in summary file.\n",
      "Setting -rna to 1.\n",
      "RNA, Step 5.3 bis: Generate profiles for /flash/hani/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa.\n",
      "Setting -rna to 1.\n",
      "RNA, Determining target gene sets.\n",
      "RNA, Drawing position histograms.\n",
      "RNA, Step 6: cluster motifs.\n",
      "Using positive correlation.\n",
      "No motifs in summary file.\n",
      "RNA, Step 5.6: generate genome-wide GO enrichments for motifs in summary file.\n",
      "RNA, draw motif maps for co-localizing pairs\n",
      "RNA, Step 6.8: identifying motifs that match micrornas.\n",
      "RNA, Step 7: draw matrix figure.\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.clusters does not exist, ignoring.\n",
      "Now doing the graphical display.\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.clusters does not exist, ignoring.\n",
      "Now doing the graphical display.\n",
      "RNA, Step 7b: draw density figure.\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.clusters does not exist, ignoring.\n",
      "Now doing the graphical display.\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.clusters does not exist, ignoring.\n",
      "Now doing the graphical display.\n",
      "Parsing distance report file ...  Done.\n",
      "RNA, Step 10: motif report.\n",
      "Fri Apr  3 07:04:05 UTC 2020\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No motifs.\n",
      "Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /flash/hani/bin/FIRE-1.1/SCRIPTS/draw_position_histogram.pl line 134.\n",
      "Table.pm: cannot open file \"peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.clusters\" ..\n",
      "Table.pm: cannot open file \"peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.mimatrix\" ..\n",
      "No motifs in @MOTIFS ...\n",
      "No motifs in @MOTIFS ...\n",
      "No motifs in @MOTIFS ...\n",
      "No motifs in @MOTIFS ...\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix does not exists. at /flash/hani/bin/FIRE-1.1/SCRIPTS/Sets.pm line 165.\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix does not exists. at /flash/hani/bin/FIRE-1.1/SCRIPTS/Sets.pm line 165.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "cd exomepeak/hiv.control/\n",
    "bedtools getfasta -name -s -fi /rumi/shams/abe/People/Judd/HIV/EU541617.1/EU541617.1.fasta -bed peak.bed -split -fo peak.fa\n",
    "# prepare fasta file for FIRE\n",
    "perl /flash/hani/bin/Tools/prep_seqs_for_teiser_run.pl peak.fa peaks\n",
    "# run FIRE --doskipdiscovery=1\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --doskipdiscovery=1 --motiffile_rna=motifs_of_interest.txt --oribiasonly=0\n",
    "mv peaks_teiser.txt_FIRE/ motifs_of_interest_FIRE\n",
    "# run FIRE --doskipdiscovery=0\n",
    "# perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --oribiasonly=0\n",
    "\n",
    "cd ../../"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running a custom pipeline \n",
    "\n",
    "Here we're using https://github.com/gold-lab/bed12ToGTF to make gtf file from control exomepeak result.\n",
    "\n",
    "https://metacpan.org/pod/Bio::FeatureIO::gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "# cat exomepeak/hiv.control/peak.bed | sort -k4 | \\\n",
    "# # rename features peak by peak\n",
    "# awk -F \"\\t\" '{OFS=FS}NR>1{$4=$4\"_peak\"}; cnt[$4]++{$4=$4\"_\"cnt[$4]} 1' | \\\n",
    "# # bed12 to gtf\n",
    "# perl ../../Workflows/bed12ToGTF/bed12ToGTF.1.pl | \\\n",
    "# awk '{gsub(/\\\"|\\;/,\"\")}1''{gsub(/transcript_id/,\"gene_id\")}1''{gsub(/ exon_number [[:digit:]]/,\"\")}1' > exomepeak/hiv.control/peak.gtf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using `bamToBed` to count reads which are present in control peak cordinates (found by `exomepeak`). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP' 'NT')\n",
    "for enz in \"${StringArray[@]}\"; \n",
    "    do for f in bam/*hiv.$enz*.bam; \n",
    "        do out=${f/.bam/.fc}; out=${out/bam/peakcount};             \n",
    "            bamToBed -i $f | intersectBed -s -wo -a - -b exomepeak/hiv.control/peak.bed | cut -f10 | sort | uniq -c | awk '{ print $2 \"\\t\" $1}' > $out\n",
    "    done;\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "peakcount/s23.hiv.METTL14.input.fc  peakcount/s24.hiv.METTL14.input.fc\r\n",
      "peakcount/s23.hiv.METTL14.m6A.fc    peakcount/s24.hiv.METTL14.m6A.fc\r\n",
      "peakcount/s23.hiv.METTL3.input.fc   peakcount/s24.hiv.METTL3.input.fc\r\n",
      "peakcount/s23.hiv.METTL3.m6A.fc     peakcount/s24.hiv.METTL3.m6A.fc\r\n",
      "peakcount/s23.hiv.NT.input.fc       peakcount/s24.hiv.NT.input.fc\r\n",
      "peakcount/s23.hiv.NT.m6A.fc         peakcount/s24.hiv.NT.m6A.fc\r\n",
      "peakcount/s23.hiv.Virma.input.fc    peakcount/s24.hiv.Virma.input.fc\r\n",
      "peakcount/s23.hiv.Virma.m6A.fc      peakcount/s24.hiv.Virma.m6A.fc\r\n",
      "peakcount/s23.hiv.WTAP.input.fc     peakcount/s24.hiv.WTAP.input.fc\r\n",
      "peakcount/s23.hiv.WTAP.m6A.fc       peakcount/s24.hiv.WTAP.m6A.fc\r\n"
     ]
    }
   ],
   "source": [
    "ls peakcount/*hiv*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Error in library(DESeq2) : there is no package called ‘DESeq2’\n",
      "Calls: <Anonymous> ... <Anonymous> -> <Anonymous> -> withVisible -> library\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Error in library(DESeq2) : there is no package called ‘DESeq2’\n",
      "Calls: <Anonymous> ... <Anonymous> -> <Anonymous> -> withVisible -> library\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "library(DESeq2)\n",
    "library(ggplot2)\n",
    "library(tidyverse)\n",
    "library(pheatmap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     cond\n",
      "s23.METTL14.input METTL14\n",
      "s23.METTL14.m6A   METTL14\n",
      "s23.METTL3.input   METTL3\n",
      "s23.METTL3.m6A     METTL3\n",
      "s23.NT.input           NT\n",
      "s23.NT.m6A             NT\n",
      "s23.Virma.input     Virma\n",
      "s23.Virma.m6A       Virma\n",
      "s23.WTAP.input       WTAP\n",
      "s23.WTAP.m6A         WTAP\n",
      "s24.METTL14.input METTL14\n",
      "s24.METTL14.m6A   METTL14\n",
      "s24.METTL3.input   METTL3\n",
      "s24.METTL3.m6A     METTL3\n",
      "s24.NT.input           NT\n",
      "s24.NT.m6A             NT\n",
      "s24.Virma.input     Virma\n",
      "s24.Virma.m6A       Virma\n",
      "s24.WTAP.input       WTAP\n",
      "s24.WTAP.m6A         WTAP\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "files <- list.files(path = \"peakcount/\", pattern=\"hiv\") \n",
    "reps <- c(rep('s23',10),rep('s24',10))\n",
    "ip <- rep(c('input','m6A'),10)\n",
    "treatment <- rep(unlist(lapply(c('METTL14','METTL3','NT','Virma','WTAP'), rep, 2)),2)\n",
    "conds <- treatment\n",
    "all_conds <- paste(reps,treatment, ip, sep='.')\n",
    "datalist <- lapply(files, function(x){read.table(file=paste('peakcount/', x, sep = \"\") ,header=FALSE)})\n",
    "m <- Reduce(function(...) merge(..., by=1, all = TRUE), datalist)\n",
    "rownames(m) <- m[,1]\n",
    "m <- m[,-1]\n",
    "# m[is.na(m)]<-0\n",
    "# m <- round(m)\n",
    "colnames(m) <- all_conds\n",
    "colData <- data.frame(cond = conds, row.names=colnames(m))\n",
    "colData"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "comparing m6A/input for each knockdown to control\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     cond\n",
      "s23.METTL14.input METTL14\n",
      "s23.METTL14.m6A   METTL14\n",
      "s24.METTL14.input METTL14\n",
      "s24.METTL14.m6A   METTL14\n",
      "s23.NT.input           NT\n",
      "s23.NT.m6A             NT\n",
      "s24.NT.input           NT\n",
      "s24.NT.m6A             NT\n",
      "                   cond\n",
      "s23.METTL3.input METTL3\n",
      "s23.METTL3.m6A   METTL3\n",
      "s24.METTL3.input METTL3\n",
      "s24.METTL3.m6A   METTL3\n",
      "s23.NT.input         NT\n",
      "s23.NT.m6A           NT\n",
      "s24.NT.input         NT\n",
      "s24.NT.m6A           NT\n",
      "                 cond\n",
      "s23.Virma.input Virma\n",
      "s23.Virma.m6A   Virma\n",
      "s24.Virma.input Virma\n",
      "s24.Virma.m6A   Virma\n",
      "s23.NT.input       NT\n",
      "s23.NT.m6A         NT\n",
      "s24.NT.input       NT\n",
      "s24.NT.m6A         NT\n",
      "               cond\n",
      "s23.WTAP.input WTAP\n",
      "s23.WTAP.m6A   WTAP\n",
      "s24.WTAP.input WTAP\n",
      "s24.WTAP.m6A   WTAP\n",
      "s23.NT.input     NT\n",
      "s23.NT.m6A       NT\n",
      "s24.NT.input     NT\n",
      "s24.NT.m6A       NT\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "options(repr.plot.width=4, repr.plot.height=4)\n",
    "all_vsd = list()\n",
    "for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "    icond = c(grep(treat,all_conds), grep('NT',all_conds) )\n",
    "    mm <- m[,icond]\n",
    "    colData <- data.frame(cond= conds[icond], row.names=colnames(mm))\n",
    "    print (colData)\n",
    "    dds <- DESeqDataSetFromMatrix(countData = mm, colData = colData, design = ~ cond)\n",
    "    dds$cond <- relevel(dds$cond, ref='NT')\n",
    "#     dds <- DESeq(dds)\n",
    "#     cds <- estimateDispersions(dds)\n",
    "#     vsd <- varianceStabilizingTransformation(cds)\n",
    "#     # Plot heatmap\n",
    "    \n",
    "#     vsd_mat <- assay(vsd)\n",
    "#     df = data.frame(vsd_mat)\n",
    "#     colnames(df) = c(\n",
    "#         's23.treat.input','s23.treat.m6A','s24.treat.input','s24.treat.m6A',\n",
    "#         's23.NT.input','s23.NT.m6A','s24.NT.input','s24.NT.m6A'\n",
    "#     )\n",
    "#     rownames(df) = paste(treat,rownames(df),sep='_')\n",
    "#     all_vsd[[treat]] = df\n",
    "#     # save scales to normalize peak visulization with GViz \n",
    "#     dds <- estimateSizeFactors(dds)\n",
    "#     scale_df <- data.frame('scale' = 1/sizeFactors(dds))\n",
    "#     write.table(scale_df, paste(\"bigwig/hiv/\",treat,\"/scale.txt\", sep=''), sep=\"\\t\", quote=FALSE, col.names=NA)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Error in pheatmap(mat = mat, border_color = NA, show_colnames = TRUE,  : \n",
      "  could not find function \"pheatmap\"\n",
      "Calls: <Anonymous> -> <Anonymous> -> <Anonymous> -> withVisible\n",
      "\n",
      "R[write to console]: In addition: \n",
      "R[write to console]: There were 50 or more warnings (use warnings() to see the first 50)\n",
      "R[write to console]: \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Error in pheatmap(mat = mat, border_color = NA, show_colnames = TRUE,  : \n",
      "  could not find function \"pheatmap\"\n",
      "Calls: <Anonymous> -> <Anonymous> -> <Anonymous> -> withVisible\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "mat <-do.call('rbind',all_vsd)\n",
    "# Data frame with column annotations.\n",
    "mat_col <- data.frame(Treatment = c(rep('Treated',4),rep('NT',4)))\n",
    "mat_row <- data.frame(Peak = c(rep('METTL14', 6), rep('METTL3', 6), rep('Virma', 6), rep('WTAP', 6) ))\n",
    "rownames(mat_col) <- colnames(mat)\n",
    "rownames(mat_row) <- rownames(mat)\n",
    "\n",
    "h = pheatmap(\n",
    "    mat               = mat,\n",
    "    border_color      = NA,\n",
    "    show_colnames     = TRUE,\n",
    "    show_rownames     = FALSE,\n",
    "    annotation_col    = mat_col,\n",
    "    annotation_row    = mat_row,\n",
    "    fontsize          = 10,\n",
    "    cluster_rows = FALSE, cluster_cols=FALSE,cellwidth = 25, cellheight = 15,\n",
    "  main              = \"HIV Peak Heatmap\"\n",
    ")\n",
    "# ggsave(\"plots/hiv.peak_Heatmaps.png\", plot = h, device = 'png', dpi = 300)\n",
    "# ggsave(\"plots/hiv.peak_Heatmaps.pdf\", plot = h, device = 'pdf', dpi = 300)\n",
    "h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run piranha for HIV bam files "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 577,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "### m6A bam files ###\n",
    "# Merge .bam files:\n",
    "# samtools merge -f -@ 12 piranha/hiv.m6A.bam bam/*hiv*m6A.bam\n",
    "# sort the merged file >>> it's not making any change\n",
    "# samtools sort -@ 12 piranha/hiv.m6A.bam > piranha/hiv.m6A.srt.bam\n",
    "# bam to bed\n",
    "bedtools bamtobed -split -bed12 -i piranha/hiv.m6A.srt.bam > piranha/hiv.m6A.srt.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 579,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 0 files and 12 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "### input bam files ###\n",
    "# Merge .bam files:\n",
    "samtools merge -f -@ 12 piranha/hiv.input.bam bam/*hiv*input.bam\n",
    "# sort the merged file >>> it's not making any change\n",
    "samtools sort -@ 12 piranha/hiv.input.bam > piranha/hiv.input.srt.bam\n",
    "# bam to bet\n",
    "bedtools bamtobed -split -bed12 -i piranha/hiv.input.srt.bam > piranha/hiv.input.srt.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 583,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "### run Piranha ###\n",
    "# Piranha -s -i 50 piranha/hiv.m6A.srt.bed piranha/hiv.input.srt.bed -o piranha/hiv.piranha.txt\n",
    "# make new ref file\n",
    "awk '{printf(\"%s\\t%d\\t%d\\t%d\\t1\\t%s\\n\",$1,$2,$3, NR,$4)}' piranha/hiv.piranha.txt > piranha/final_piranha.bed\n",
    "awk 'NR>1{printf(\"%s\\t%s\\t%d\\t%d\\t%s\\n\",$4-1,$1,$2+1,$3,$6)}' piranha/final_piranha.bed > piranha/final_piranha.saf\n",
    "### first line of piranha/final_piranha.saf removed manually "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 584,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "# cat exomepeak/control/peak.bed | sort -k4 | \\\n",
    "# # rename features peak by peak\n",
    "# awk -F \"\\t\" '{OFS=FS}NR>1{$4=$4\"_peak\"}; cnt[$4]++{$4=$4\"_\"cnt[$4]} 1' | \\\n",
    "# # bed12 to gtf\n",
    "# perl ../../Workflows/bed12ToGTF/bed12ToGTF.1.pl | \\\n",
    "# awk '{gsub(/\\\"|\\;/,\"\")}1''{gsub(/transcript_id/,\"gene_id\")}1''{gsub(/ exon_number [[:digit:]]/,\"\")}1' > exomepeak/control/peak.gtf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using `featureCounts` to count reads which are present in control peak cordinates (found by `piranha`). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 580,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# %%bash\n",
    "# declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "# for enz in \"${StringArray[@]}\"; \n",
    "#     do for f in bam/*human.$enz*.bam; \n",
    "#         do out=${f/.bam/.fc}; out=${out/bam/peakcount}; \n",
    "#             featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/control/peak.gtf -o $out $f; \n",
    "#     done;\n",
    "# done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 586,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.METTL14.m6A.bam                    ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.METTL14.m6A.piranha.fc        ||\n",
      "||                 Summary : piranha/fc/s23.hiv.METTL14.m6A.piranha.fc.su ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.METTL14.m6A.bam...                            ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 4894526                                                   ||\n",
      "||    Successfully assigned reads : 100426 (2.1%)                             ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.METT  ||\n",
      "|| L14.m6A.piranha.fc.summary\"                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.METTL3.m6A.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.METTL3.m6A.piranha.fc         ||\n",
      "||                 Summary : piranha/fc/s23.hiv.METTL3.m6A.piranha.fc.sum ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.METTL3.m6A.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 10022948                                                  ||\n",
      "||    Successfully assigned reads : 60620 (0.6%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.METT  ||\n",
      "|| L3.m6A.piranha.fc.summary\"                                                 ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.NT.m6A.bam                         ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.NT.m6A.piranha.fc             ||\n",
      "||                 Summary : piranha/fc/s23.hiv.NT.m6A.piranha.fc.summary     ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.NT.m6A.bam...                                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 5220214                                                   ||\n",
      "||    Successfully assigned reads : 96966 (1.9%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.NT.m  ||\n",
      "|| 6A.piranha.fc.summary\"                                                     ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.Virma.m6A.bam                      ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.Virma.m6A.piranha.fc          ||\n",
      "||                 Summary : piranha/fc/s23.hiv.Virma.m6A.piranha.fc.summary  ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.Virma.m6A.bam...                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 4256800                                                   ||\n",
      "||    Successfully assigned reads : 98203 (2.3%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.Virm  ||\n",
      "|| a.m6A.piranha.fc.summary\"                                                  ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.WTAP.m6A.bam                       ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.WTAP.m6A.piranha.fc           ||\n",
      "||                 Summary : piranha/fc/s23.hiv.WTAP.m6A.piranha.fc.summary   ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.WTAP.m6A.bam...                               ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 4964136                                                   ||\n",
      "||    Successfully assigned reads : 82108 (1.7%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.WTAP  ||\n",
      "|| .m6A.piranha.fc.summary\"                                                   ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.METTL14.m6A.bam                    ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.METTL14.m6A.piranha.fc        ||\n",
      "||                 Summary : piranha/fc/s24.hiv.METTL14.m6A.piranha.fc.su ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.METTL14.m6A.bam...                            ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 10197444                                                  ||\n",
      "||    Successfully assigned reads : 62766 (0.6%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.METT  ||\n",
      "|| L14.m6A.piranha.fc.summary\"                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.METTL3.m6A.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.METTL3.m6A.piranha.fc         ||\n",
      "||                 Summary : piranha/fc/s24.hiv.METTL3.m6A.piranha.fc.sum ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.METTL3.m6A.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 6241746                                                   ||\n",
      "||    Successfully assigned reads : 103128 (1.7%)                             ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.METT  ||\n",
      "|| L3.m6A.piranha.fc.summary\"                                                 ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.NT.m6A.bam                         ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.NT.m6A.piranha.fc             ||\n",
      "||                 Summary : piranha/fc/s24.hiv.NT.m6A.piranha.fc.summary     ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.NT.m6A.bam...                                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 2751185                                                   ||\n",
      "||    Successfully assigned reads : 79308 (2.9%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.NT.m  ||\n",
      "|| 6A.piranha.fc.summary\"                                                     ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.Virma.m6A.bam                      ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.Virma.m6A.piranha.fc          ||\n",
      "||                 Summary : piranha/fc/s24.hiv.Virma.m6A.piranha.fc.summary  ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.Virma.m6A.bam...                              ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 3883624                                                   ||\n",
      "||    Successfully assigned reads : 74097 (1.9%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.Virm  ||\n",
      "|| a.m6A.piranha.fc.summary\"                                                  ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.WTAP.m6A.bam                       ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.WTAP.m6A.piranha.fc           ||\n",
      "||                 Summary : piranha/fc/s24.hiv.WTAP.m6A.piranha.fc.summary   ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.WTAP.m6A.bam...                               ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7907298                                                   ||\n",
      "||    Successfully assigned reads : 75612 (1.0%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.WTAP  ||\n",
      "|| .m6A.piranha.fc.summary\"                                                   ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.METTL14.input.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.METTL14.input.piranha.fc      ||\n",
      "||                 Summary : piranha/fc/s23.hiv.METTL14.input.piranha.fc. ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.METTL14.input.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 688648                                                    ||\n",
      "||    Successfully assigned reads : 17514 (2.5%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.METT  ||\n",
      "|| L14.input.piranha.fc.summary\"                                              ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.METTL3.input.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.METTL3.input.piranha.fc       ||\n",
      "||                 Summary : piranha/fc/s23.hiv.METTL3.input.piranha.fc.s ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.METTL3.input.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 705564                                                    ||\n",
      "||    Successfully assigned reads : 19274 (2.7%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.METT  ||\n",
      "|| L3.input.piranha.fc.summary\"                                               ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.NT.input.bam                       ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.NT.input.piranha.fc           ||\n",
      "||                 Summary : piranha/fc/s23.hiv.NT.input.piranha.fc.summary   ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.NT.input.bam...                               ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 577656                                                    ||\n",
      "||    Successfully assigned reads : 10916 (1.9%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.NT.i  ||\n",
      "|| nput.piranha.fc.summary\"                                                   ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.Virma.input.bam                    ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.Virma.input.piranha.fc        ||\n",
      "||                 Summary : piranha/fc/s23.hiv.Virma.input.piranha.fc.su ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.Virma.input.bam...                            ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 783163                                                    ||\n",
      "||    Successfully assigned reads : 21889 (2.8%)                              ||\n",
      "||    Running time : 0.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.Virm  ||\n",
      "|| a.input.piranha.fc.summary\"                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.hiv.WTAP.input.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s23.hiv.WTAP.input.piranha.fc         ||\n",
      "||                 Summary : piranha/fc/s23.hiv.WTAP.input.piranha.fc.sum ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.hiv.WTAP.input.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 728720                                                    ||\n",
      "||    Successfully assigned reads : 22966 (3.2%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s23.hiv.WTAP  ||\n",
      "|| .input.piranha.fc.summary\"                                                 ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.METTL14.input.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.METTL14.input.piranha.fc      ||\n",
      "||                 Summary : piranha/fc/s24.hiv.METTL14.input.piranha.fc. ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.METTL14.input.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 735265                                                    ||\n",
      "||    Successfully assigned reads : 19591 (2.7%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.METT  ||\n",
      "|| L14.input.piranha.fc.summary\"                                              ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.METTL3.input.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.METTL3.input.piranha.fc       ||\n",
      "||                 Summary : piranha/fc/s24.hiv.METTL3.input.piranha.fc.s ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.METTL3.input.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 648737                                                    ||\n",
      "||    Successfully assigned reads : 18147 (2.8%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.METT  ||\n",
      "|| L3.input.piranha.fc.summary\"                                               ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.NT.input.bam                       ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.NT.input.piranha.fc           ||\n",
      "||                 Summary : piranha/fc/s24.hiv.NT.input.piranha.fc.summary   ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.NT.input.bam...                               ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 680748                                                    ||\n",
      "||    Successfully assigned reads : 15206 (2.2%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.NT.i  ||\n",
      "|| nput.piranha.fc.summary\"                                                   ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.Virma.input.bam                    ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.Virma.input.piranha.fc        ||\n",
      "||                 Summary : piranha/fc/s24.hiv.Virma.input.piranha.fc.su ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.Virma.input.bam...                            ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 593259                                                    ||\n",
      "||    Successfully assigned reads : 14699 (2.5%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.Virm  ||\n",
      "|| a.input.piranha.fc.summary\"                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.hiv.WTAP.input.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : piranha/fc/s24.hiv.WTAP.input.piranha.fc         ||\n",
      "||                 Summary : piranha/fc/s24.hiv.WTAP.input.piranha.fc.sum ... ||\n",
      "||              Annotation : piranha/final_piranha.saf (SAF)                  ||\n",
      "||      Dir for temp files : piranha/fc                                       ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file piranha/final_piranha.saf ...                         ||\n",
      "||    Features : 6                                                            ||\n",
      "||    Meta-features : 6                                                       ||\n",
      "||    Chromosomes/contigs : 1                                                 ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.hiv.WTAP.input.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 724259                                                    ||\n",
      "||    Successfully assigned reads : 15955 (2.2%)                              ||\n",
      "||    Running time : 0.00 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"piranha/fc/s24.hiv.WTAP  ||\n",
      "|| .input.piranha.fc.summary\"                                                 ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# make new reletive counts\n",
    "for f in bam/*hiv*.m6A.bam; do\n",
    "    out=${f/.bam/.piranha.fc};\n",
    "    out=${out/bam/piranha\\/fc};\n",
    "    featureCounts -T 12 -O -F SAF -a piranha/final_piranha.saf -o $out $f;\n",
    "done\n",
    "\n",
    "for f in bam/*hiv*.input.bam; do\n",
    "    out=${f/.bam/.piranha.fc};\n",
    "    out=${out/bam/piranha\\/fc};\n",
    "    featureCounts -T 12 -O -F SAF -a piranha/final_piranha.saf -o $out $f;\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(DESeq2)\n",
    "library(ggplot2)\n",
    "library(tidyverse)\n",
    "library(pheatmap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%R\n",
    "# files <- list.files(path = \"piranha/fc\", pattern=\"fc$\") \n",
    "# reps <- c(rep('s23',10),rep('s24',10))\n",
    "# ip <- rep(c('input','m6A'),10)\n",
    "# enzymes <- rep(unlist(lapply(c('METTL14','METTL3','NT','Virma','WTAP'), rep, 2)),2)\n",
    "# conds <- ip\n",
    "# all_conds <- paste(\n",
    "#     reps,\n",
    "#     enzymes, \n",
    "#     ip, \n",
    "#     sep='.')\n",
    "# datalist <- lapply(files, function(x){read.table(file=paste('piranha/fc/', x, sep = \"\") ,header=TRUE,colClasses=c(NA, \"NULL\", \"NULL\", \"NULL\", \"NULL\", \"NULL\", NA),check.names=F)})\n",
    "# m <- Reduce(function(...) merge(..., by=1, all = TRUE), datalist)\n",
    "# rownames(m) <- paste ('hiv',m[,1],sep='-')\n",
    "# m <- m[,-1]\n",
    "# m[is.na(m)]<-0\n",
    "# m <- round(m)\n",
    "# colnames(m) <- all_conds\n",
    "# colData <- data.frame(cond = conds, row.names=colnames(m))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "comparing m6A/input for each knockdown to control\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# %%R\n",
    "# options(repr.plot.width=4, repr.plot.height=4)\n",
    "# all_vsd = list()\n",
    "# for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "#     icond = c(grep(treat,all_conds), grep('NT',all_conds) )\n",
    "#     mm <- m[,icond]\n",
    "#     colData <- data.frame(cond= conds[icond], row.names=colnames(mm))\n",
    "#     dds <- DESeqDataSetFromMatrix(countData = mm, colData = colData, design = ~ cond)\n",
    "#     dds$cond <- relevel(dds$cond, ref='input')\n",
    "#     dds <- DESeq(dds)\n",
    "#     cds <- estimateDispersions(dds)\n",
    "#     vsd <- varianceStabilizingTransformation(cds)\n",
    "#     # Plot heatmap\n",
    "    \n",
    "#     vsd_mat <- assay(vsd)\n",
    "#     df = data.frame(vsd_mat)\n",
    "#     colnames(df) = c(\n",
    "#         's23.treat.input','s23.treat.m6A','s24.treat.input','s24.treat.m6A',\n",
    "#         's23.NT.input','s23.NT.m6A','s24.NT.input','s24.NT.m6A'\n",
    "#     )\n",
    "#     rownames(df) = paste(treat,rownames(df),sep='_')\n",
    "#     all_vsd[[treat]] = df\n",
    "#     # save scales to normalize peak visulization with GViz \n",
    "#     dds <- estimateSizeFactors(dds)\n",
    "#     scale_df <- data.frame('scale' = 1/sizeFactors(dds))\n",
    "#     write.table(scale_df, paste(\"bigwig/hiv/\",treat,\"/scale.txt\", sep=''), sep=\"\\t\", quote=FALSE, col.names=NA)\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# %%R\n",
    "\n",
    "# mat <-do.call('rbind',all_vsd)\n",
    "# # Data frame with column annotations.\n",
    "# mat_col <- data.frame(Treatment = c(rep('Treated',4),rep('NT',4)))\n",
    "# mat_row <- data.frame(Peak = c(rep('METTL14', 6), rep('METTL3', 6), rep('Virma', 6), rep('WTAP', 6) ))\n",
    "# rownames(mat_col) <- colnames(mat)\n",
    "# rownames(mat_row) <- rownames(mat)\n",
    "\n",
    "# h = pheatmap(\n",
    "#     mat               = mat,\n",
    "#     border_color      = NA,\n",
    "#     show_colnames     = TRUE,\n",
    "#     show_rownames     = FALSE,\n",
    "#     annotation_col    = mat_col,\n",
    "#     annotation_row    = mat_row,\n",
    "#     fontsize          = 10,\n",
    "#     cluster_rows = FALSE, cluster_cols=FALSE,cellwidth = 25, cellheight = 15,\n",
    "#   main              = \"HIV Peak Heatmap\"\n",
    "# )\n",
    "# ggsave(\"plots/hiv.peak_Heatmaps.png\", plot = h, device = 'png', dpi = 300)\n",
    "# ggsave(\"plots/hiv.peak_Heatmaps.pdf\", plot = h, device = 'pdf', dpi = 300)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot peak cordinates "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "making `bigwig` file from `bam` files to visualize genome coverage of peak coordinates of hiv genome. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bedtools genomecov -bga -ibam bam/s23.hiv.METTL14.input.bam bedgraph/hiv/METTL14/s23.hiv.METTL14.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.METTL14.m6A.bam bedgraph/hiv/METTL14/s23.hiv.METTL14.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.METTL14.input.bam bedgraph/hiv/METTL14/s24.hiv.METTL14.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.METTL14.m6A.bam bedgraph/hiv/METTL14/s24.hiv.METTL14.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.METTL3.input.bam bedgraph/hiv/METTL3/s23.hiv.METTL3.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.METTL3.m6A.bam bedgraph/hiv/METTL3/s23.hiv.METTL3.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.METTL3.input.bam bedgraph/hiv/METTL3/s24.hiv.METTL3.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.METTL3.m6A.bam bedgraph/hiv/METTL3/s24.hiv.METTL3.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.Virma.input.bam bedgraph/hiv/Virma/s23.hiv.Virma.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.Virma.m6A.bam bedgraph/hiv/Virma/s23.hiv.Virma.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.Virma.input.bam bedgraph/hiv/Virma/s24.hiv.Virma.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.Virma.m6A.bam bedgraph/hiv/Virma/s24.hiv.Virma.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.WTAP.input.bam bedgraph/hiv/WTAP/s23.hiv.WTAP.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.WTAP.m6A.bam bedgraph/hiv/WTAP/s23.hiv.WTAP.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.WTAP.input.bam bedgraph/hiv/WTAP/s24.hiv.WTAP.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.WTAP.m6A.bam bedgraph/hiv/WTAP/s24.hiv.WTAP.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.NT.input.bam bedgraph/hiv/NT/s23.hiv.NT.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s23.hiv.NT.m6A.bam bedgraph/hiv/NT/s23.hiv.NT.m6A.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.NT.input.bam bedgraph/hiv/NT/s24.hiv.NT.input.bg\n",
      "bedtools genomecov -bga -ibam bam/s24.hiv.NT.m6A.bam bedgraph/hiv/NT/s24.hiv.NT.m6A.bg\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP' 'NT')\n",
    "# mkdir bedgraph/hiv/\n",
    "# for treat in \"${StringArray[@]}\"; do mkdir bedgraph/hiv/$treat; done\n",
    "\n",
    "# normalize \n",
    "# # bigwigs \n",
    "# for treat in \"${StringArray[@]}\"; do\n",
    "#     awk '!(NR<=1){gsub(/s2[3-4]./,substr($1, 0, 4)\"hiv.\");print \"bamCoverage --scaleFactor \"$2\" -b bam/\"$1\".bam -o bigwig/'$treat'/\"$1\".bw\"}' bigwig/$treat/scale.txt # | bash\n",
    "# done\n",
    "\n",
    "# bedgraphs using bedtools \n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "    for f in bam/*hiv*$treat*bam; do \n",
    "        base=`basename \"$f\"`\n",
    "        o=${base/.bam/.bg}\n",
    "        bedtools genomecov -bga -ibam $f > bedgraph/hiv/$treat/$o;\n",
    "    done;\n",
    "#     awk '!(NR<=1){gsub(/s2[3-4]./,substr($1, 0, 4)\"hiv.\");print \"bedtools genomecov -bga -scale \"$2\" -ibam bam/\"$1\".bam > bedgraph/hiv/'$treat'/\"$1\".bg\"}' bigwig/hiv/$treat/scale.txt | bash\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 1374\r\n",
      "-rw-r--r-- 1 abe users 205633 Apr  2 11:30 s23.hiv.NT.input.bg\r\n",
      "-rw-r--r-- 1 abe users  64008 Apr  2 11:30 s23.hiv.NT.m6A.bg\r\n",
      "-rw-r--r-- 1 abe users 226791 Apr  2 11:30 s23.hiv.Virma.input.bg\r\n",
      "-rw-r--r-- 1 abe users  69301 Apr  2 11:30 s23.hiv.Virma.m6A.bg\r\n",
      "-rw-r--r-- 1 abe users 215101 Apr  2 11:30 s24.hiv.NT.input.bg\r\n",
      "-rw-r--r-- 1 abe users  61545 Apr  2 11:30 s24.hiv.NT.m6A.bg\r\n",
      "-rw-r--r-- 1 abe users 215995 Apr  2 11:30 s24.hiv.Virma.input.bg\r\n",
      "-rw-r--r-- 1 abe users  56603 Apr  2 11:30 s24.hiv.Virma.m6A.bg\r\n"
     ]
    }
   ],
   "source": [
    "ls -l bedgraph/hiv/Virma/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(Gviz)\n",
    "library(Rsamtools)\n",
    "library(data.table)\n",
    "library(Biostrings)\n",
    "library(rtracklayer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# hiv_dna <- readDNAStringSet(\"exomepeak/hiv.control/peak.fa\")\n",
    "# names(hiv_dna) <- rep('chr1', length(hiv_dna))\n",
    "# # export(hiv_dna, \"HIV/HIV.fa.2bit\")\n",
    "# # hiv <- TwoBitFile(\"HIV/HIV.fa.2bit\")\n",
    "\n",
    "# read HIV genome \n",
    "hiv_peaks <- read.table(file=\"exomepeak/hiv.control/peak.bed\", sep=\"\\t\")[,c(2,3)]\n",
    "colnames(hiv_peaks) <- c('start', 'end')\n",
    "hiv_gr <- GRanges(seqnames=\"chr1\", range=IRanges(start=hiv_peaks$start, end=hiv_peaks$end),strand='*')\n",
    "names(hiv_gr) <- paste('peak', c(1:dim(hiv_peaks)[1]), sep='-')\n",
    "genome(hiv_gr) <- 'HIV'; gen<- 'HIV'\n",
    "chr <- as.character(unique(seqnames(hiv_gr)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GRanges object with 4 ranges and 0 metadata columns:\n",
      "         seqnames    ranges strand\n",
      "            <Rle> <IRanges>  <Rle>\n",
      "  peak-1     chr1 5519-7966      *\n",
      "  peak-2     chr1 5514-8195      *\n",
      "  peak-3     chr1 7828-8337      *\n",
      "  peak-4     chr1 8338-8959      *\n",
      "  -------\n",
      "  seqinfo: 1 sequence from HIV genome; no seqlengths\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "hiv_gr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "using normalized `bedgraph` and making `Gviz` tracks "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"bedgraph/hiv/METTL14/s23.hiv.METTL14.input.bg\"\n",
      "[2] \"bedgraph/hiv/METTL14/s23.hiv.METTL14.m6A.bg\"  \n",
      "[3] \"bedgraph/hiv/METTL14/s24.hiv.METTL14.input.bg\"\n",
      "[4] \"bedgraph/hiv/METTL14/s24.hiv.METTL14.m6A.bg\"  \n",
      "[1] \"bedgraph/hiv/METTL14/s23.hiv.NT.input.bg\"\n",
      "[2] \"bedgraph/hiv/METTL14/s23.hiv.NT.m6A.bg\"  \n",
      "[3] \"bedgraph/hiv/METTL14/s24.hiv.NT.input.bg\"\n",
      "[4] \"bedgraph/hiv/METTL14/s24.hiv.NT.m6A.bg\"  \n",
      "[1] \"bedgraph/hiv/METTL3/s23.hiv.METTL3.input.bg\"\n",
      "[2] \"bedgraph/hiv/METTL3/s23.hiv.METTL3.m6A.bg\"  \n",
      "[3] \"bedgraph/hiv/METTL3/s24.hiv.METTL3.input.bg\"\n",
      "[4] \"bedgraph/hiv/METTL3/s24.hiv.METTL3.m6A.bg\"  \n",
      "[1] \"bedgraph/hiv/METTL3/s23.hiv.NT.input.bg\"\n",
      "[2] \"bedgraph/hiv/METTL3/s23.hiv.NT.m6A.bg\"  \n",
      "[3] \"bedgraph/hiv/METTL3/s24.hiv.NT.input.bg\"\n",
      "[4] \"bedgraph/hiv/METTL3/s24.hiv.NT.m6A.bg\"  \n",
      "[1] \"bedgraph/hiv/Virma/s23.hiv.Virma.input.bg\"\n",
      "[2] \"bedgraph/hiv/Virma/s23.hiv.Virma.m6A.bg\"  \n",
      "[3] \"bedgraph/hiv/Virma/s24.hiv.Virma.input.bg\"\n",
      "[4] \"bedgraph/hiv/Virma/s24.hiv.Virma.m6A.bg\"  \n",
      "[1] \"bedgraph/hiv/Virma/s23.hiv.NT.input.bg\"\n",
      "[2] \"bedgraph/hiv/Virma/s23.hiv.NT.m6A.bg\"  \n",
      "[3] \"bedgraph/hiv/Virma/s24.hiv.NT.input.bg\"\n",
      "[4] \"bedgraph/hiv/Virma/s24.hiv.NT.m6A.bg\"  \n",
      "[1] \"bedgraph/hiv/WTAP/s23.hiv.WTAP.input.bg\"\n",
      "[2] \"bedgraph/hiv/WTAP/s23.hiv.WTAP.m6A.bg\"  \n",
      "[3] \"bedgraph/hiv/WTAP/s24.hiv.WTAP.input.bg\"\n",
      "[4] \"bedgraph/hiv/WTAP/s24.hiv.WTAP.m6A.bg\"  \n",
      "[1] \"bedgraph/hiv/WTAP/s23.hiv.NT.input.bg\"\n",
      "[2] \"bedgraph/hiv/WTAP/s23.hiv.NT.m6A.bg\"  \n",
      "[3] \"bedgraph/hiv/WTAP/s24.hiv.NT.input.bg\"\n",
      "[4] \"bedgraph/hiv/WTAP/s24.hiv.NT.m6A.bg\"  \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "    reps <- c(rep('s23',2),rep('s24',2))\n",
    "    ip <- rep(c('input','m6A'),2)\n",
    "    conds <- paste(reps,treat, sep='.')\n",
    "    # absolute path to bedgraph files  \n",
    "    bw_files.treated <- paste('bedgraph/hiv/',treat,'/',reps,'.hiv.',treat,'.',ip,'.bg', sep='')\n",
    "    bw_files.nontreated <- paste('bedgraph/hiv/',treat,'/',reps,'.hiv.NT.',ip,'.bg', sep='')\n",
    "    print (bw_files.treated)\n",
    "    print (bw_files.nontreated)\n",
    "    for (i in c(2,4)){\n",
    "        ### read bedgraph files\n",
    "        #-- treated tracks --#\n",
    "        # m6A track           \n",
    "        bw_ip.treated <- import.bedGraph(bw_files.treated[i]) #,as=\"GRanges\")\n",
    "        genome(bw_ip.treated) <- 'HIV'\n",
    "        seqlevels(bw_ip.treated) <- 'chr1'\n",
    "        # input track\n",
    "        bw_in.treated <- import.bedGraph(bw_files.treated[i-1]) #,as=\"GRanges\")\n",
    "        genome(bw_in.treated) <- 'HIV'\n",
    "        seqlevels(bw_in.treated) <- 'chr1'\n",
    "        # Overlay input and m6A\n",
    "        dtrack_ip.treated <- DataTrack(bw_ip.treated,chomosome=\"chr1\", name='Treated')\n",
    "        dtrack_in.treated <- DataTrack(bw_in.treated,chomosome=\"chr1\", groups = factor('input',levels = c('input','m6A')) )\n",
    "        displayPars(dtrack_ip.treated) <- list(groups = factor('m6A',levels = c('input','m6A')), legend = TRUE)\n",
    "        assign(paste('treated', conds[i],sep='.'),\n",
    "               OverlayTrack(trackList=list(dtrack_ip.treated,dtrack_in.treated)))\n",
    "        #-- non-treated tracks --#\n",
    "        # m6A track\n",
    "        bw_ip.nontreated <- import.bedGraph(bw_files.nontreated[i]) # ,as=\"GRanges\")\n",
    "        genome(bw_ip.nontreated) <- 'HIV'\n",
    "        seqlevels(bw_ip.nontreated) <- 'chr1'\n",
    "        # input track\n",
    "        bw_in.nontreated <- import.bedGraph(bw_files.nontreated[i-1]) # ,as=\"GRanges\")\n",
    "        genome(bw_in.nontreated) <- 'HIV'\n",
    "        seqlevels(bw_in.nontreated) <- 'chr1'\n",
    "        # Overlay input and m6A\n",
    "        dtrack_ip.nontreated <- DataTrack(bw_ip.nontreated,chomosome=\"chr1\", name='Non-treated')\n",
    "        dtrack_in.nontreated <- DataTrack(bw_in.nontreated,chomosome=\"chr1\", groups = factor('input',levels = c('input','m6A')) )\n",
    "        displayPars(dtrack_ip.nontreated) <- list(groups = factor('m6A',levels = c('input','m6A')), legend = TRUE, name = 'Non treated')\n",
    "        assign(paste('nontreated',conds[i],sep='.'), \n",
    "               OverlayTrack(trackList=list(dtrack_ip.nontreated,dtrack_in.nontreated),\n",
    "                            background.panel = \"#FFFEDB\", background.title = \"darkblue\"))\n",
    "    }\n",
    "}\n",
    "\n",
    "grtrack <- GeneRegionTrack(hiv_gr, genome = gen, chromosome = chr, name = \"Gene Model\")\n",
    "gtrack <- GenomeAxisTrack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "peak_ploter <- function(treat,ymax, saveplot = '' ){\n",
    "    for (i in 1:4){\n",
    "        if (saveplot == 'pdf') pdf(paste(\"plots/hiv.\",treat,'.',names(hiv_gr)[i],\".pdf\",sep=''), \n",
    "                                   width=7, height=7, paper = \"a4r\")\n",
    "        if (saveplot == 'png') png(paste(\"plots/hiv.\",treat,'.',names(hiv_gr)[i],\".png\",sep=''))\n",
    "        plotTracks(\n",
    "            list (gtrack, grtrack,\n",
    "                  get (paste('treated.s23', treat, sep ='.')),get(paste('treated.s24', treat, sep ='.')),\n",
    "                  get (paste('nontreated.s23', treat,sep ='.')),get(paste('nontreated.s24', treat,sep ='.'))\n",
    "                 ), type=\"l\",from=start(hiv_gr)[i], to=end(hiv_gr)[i] , ylim= c(-100,ymax[i]),\n",
    "            main = paste(treat, names(hiv_gr)[i]), add53=T,chromosome = \"chr1\",cex=1)\n",
    "        if (saveplot != '') dev.off()\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Error in plotTracks(list(gtrack, grtrack, get(paste(\"treated.s23\", treat,  : \n",
      "  could not find function \"plotTracks\"\n",
      "Calls: <Anonymous> ... <Anonymous> -> <Anonymous> -> withVisible -> peak_ploter\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Error in plotTracks(list(gtrack, grtrack, get(paste(\"treated.s23\", treat,  : \n",
      "  could not find function \"plotTracks\"\n",
      "Calls: <Anonymous> ... <Anonymous> -> <Anonymous> -> withVisible -> peak_ploter\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "treat = 'METTL14'\n",
    "y = c(1e3,6e3,1.5e4,1e4)\n",
    "p = peak_ploter (treat,ymax)\n",
    "p = peak_ploter (treat,ymax)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "treat = 'METTL14'\n",
    "y = c(1e3,6e3,1.5e4,1e4)\n",
    "p = peak_ploter (treat,ymax, saveplot='pdf')\n",
    "p = peak_ploter (treat,ymax, saveplot='png')\n",
    "\n",
    "treat = 'METTL3'\n",
    "y = c(1e3,6e3,1.5e4,1e4)\n",
    "p = peak_ploter (treat,ymax, saveplot='pdf')\n",
    "p = peak_ploter (treat,ymax, saveplot='png')\n",
    "\n",
    "treat = 'Virma'\n",
    "y = c(1e3,6e3,1.5e4,1e4)\n",
    "p = peak_ploter (treat,ymax, saveplot='pdf')\n",
    "p = peak_ploter (treat,ymax, saveplot='png')\n",
    "\n",
    "treat = 'WTAP'\n",
    "y = c(1e3,6e3,1.5e4,1e4)\n",
    "p = peak_ploter (treat,ymax, saveplot='pdf')\n",
    "p = peak_ploter (treat,ymax, saveplot='png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R version 3.5.1 (2018-07-02)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 18.04.2 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /rumi/shams/abe/anaconda3/envs/deseq/lib/R/lib/libRlapack.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      " [1] grid      parallel  stats4    tools     stats     graphics  grDevices\n",
      " [8] utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      " [1] rtracklayer_1.40.6          data.table_1.11.6          \n",
      " [3] Rsamtools_1.32.3            Biostrings_2.48.0          \n",
      " [5] XVector_0.20.0              Gviz_1.24.0                \n",
      " [7] pheatmap_1.0.12             forcats_0.4.0              \n",
      " [9] stringr_1.4.0               dplyr_0.8.3                \n",
      "[11] purrr_0.3.3                 readr_1.3.1                \n",
      "[13] tidyr_1.0.0                 tibble_2.1.3               \n",
      "[15] tidyverse_1.2.1             ggplot2_3.2.1              \n",
      "[17] DESeq2_1.20.0               SummarizedExperiment_1.10.1\n",
      "[19] DelayedArray_0.6.6          BiocParallel_1.14.2        \n",
      "[21] matrixStats_0.55.0          Biobase_2.40.0             \n",
      "[23] GenomicRanges_1.32.7        GenomeInfoDb_1.16.0        \n",
      "[25] IRanges_2.14.12             S4Vectors_0.18.3           \n",
      "[27] BiocGenerics_0.26.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      " [1] colorspace_1.4-1         biovizBase_1.28.2        htmlTable_1.13.2        \n",
      " [4] base64enc_0.1-3          dichromat_2.0-0          rstudioapi_0.10         \n",
      " [7] bit64_0.9-7              AnnotationDbi_1.42.1     lubridate_1.7.4         \n",
      "[10] xml2_1.2.2               splines_3.5.1            geneplotter_1.58.0      \n",
      "[13] knitr_1.25               zeallot_0.1.0            Formula_1.2-3           \n",
      "[16] jsonlite_1.6             broom_0.5.0              annotate_1.58.0         \n",
      "[19] cluster_2.1.0            compiler_3.5.1           httr_1.4.1              \n",
      "[22] backports_1.1.5          assertthat_0.2.1         Matrix_1.2-17           \n",
      "[25] lazyeval_0.2.2           cli_1.1.0                acepack_1.4.1           \n",
      "[28] htmltools_0.4.0          prettyunits_1.0.2        gtable_0.3.0            \n",
      "[31] glue_1.3.1               GenomeInfoDbData_1.1.0   Rcpp_1.0.3              \n",
      "[34] cellranger_1.1.0         vctrs_0.2.0              nlme_3.1-141            \n",
      "[37] xfun_0.10                rvest_0.3.5              lifecycle_0.1.0         \n",
      "[40] ensembldb_2.4.1          XML_3.98-1.20            zlibbioc_1.26.0         \n",
      "[43] scales_1.0.0             BSgenome_1.48.0          VariantAnnotation_1.26.1\n",
      "[46] hms_0.5.2                ProtGenerics_1.12.0      AnnotationFilter_1.4.0  \n",
      "[49] RColorBrewer_1.1-2       curl_4.2                 memoise_1.1.0           \n",
      "[52] gridExtra_2.3            biomaRt_2.36.1           rpart_4.1-15            \n",
      "[55] latticeExtra_0.6-28      stringi_1.4.3            RSQLite_2.1.2           \n",
      "[58] genefilter_1.62.0        checkmate_1.9.4          GenomicFeatures_1.32.3  \n",
      "[61] rlang_0.4.1              pkgconfig_2.0.3          bitops_1.0-6            \n",
      "[64] lattice_0.20-38          GenomicAlignments_1.16.0 htmlwidgets_1.5.1       \n",
      "[67] bit_1.1-14               tidyselect_0.2.5         magrittr_1.5            \n",
      "[70] R6_2.4.0                 Hmisc_4.2-0              DBI_1.0.0               \n",
      "[73] pillar_1.4.2             haven_2.2.0              foreign_0.8-72          \n",
      "[76] withr_2.1.2              survival_2.44-1.1        RCurl_1.95-4.12         \n",
      "[79] nnet_7.3-12              modelr_0.1.5             crayon_1.3.4            \n",
      "[82] progress_1.2.2           locfit_1.5-9.1           readxl_1.3.1            \n",
      "[85] blob_1.2.0               digest_0.6.22            xtable_1.8-4            \n",
      "[88] munsell_0.5.0           \n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
