{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Protocol used by Albertas to make this data: https://benchling.com/anavickas/f/lib_VNejBHt6-protocols/prt_2Bkfh557-meripseq/edit\n",
    "\n",
    "HIV-1 IIIB Virus\n",
    "\n",
    "\n",
    "https://aidsreagent.org/reagentdetail.cfm?t=viruses&id=607\n",
    "\n",
    "We are starting from `fastq` files as raw data: "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Trimming "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cd ~/People/Judd/fastq\n",
    "for f in *_L002_R1_001.fastq.gz; do  \n",
    "    out=${f/\\_S[1-9]*/.trim.fastq.gz}; \n",
    "    cutadapt -j 12 -q 15 -m 20 -a NNNAGATCGGAAGAGCACACGTCTGAACTCCAGTCA -o ../trim/$out $f; \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alignment\n",
    "\n",
    "Unmmaped reads for each `fastq` file saved for later alignment to HIV genome \n",
    "\n",
    "Human genome: `gencode.v28.annotation.gtf`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "STAR --genomeLoad LoadAndExit --genomeDir /rumi/shams/abe/genomes/hg38/\n",
    "for f in trim/*.fastq.gz;do\n",
    "    out=${f/.trim.fastq.gz/_};\n",
    "    out=${out/trim/bam};\n",
    "    STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --runThreadN 16 --genomeDir /rumi/shams/abe/genomes/hg38/ --readFilesIn $f --outFileNamePrefix $out --outReadsUnmapped Fastx;\n",
    "done\n",
    "STAR --genomeLoad Remove --genomeDir /rumi/shams/abe/genomes/hg38/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Firstly, HIV genome indices created by changing sequence provided by Judd to `.fa` format. It didn't work as we needed gtf file! \n",
    "\n",
    "Ok! I'm switching to the [HIV-1 clone pIIIB from USA, complete genome\n",
    "](https://www.ncbi.nlm.nih.gov/nucleotide/EU541617.1?report=genbank&log$=nuclalign&blast_rank=1&RID=8B8ETWT201R) which is highly matched with Judd's clone (using Blast alignment). Then, I donwloaded `.fa` and `.gff3` format for downstream analysis. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash \n",
    "# # make hiv genome indices \n",
    "# bowtie2-build HIV/EU541617.1/EU541617.1.fasta HIV/EU541617.1/index\n",
    "# # Aligning to hiv genome \n",
    "# for f in fastq_unmapped/*; do\n",
    "# \to=${f/.fastq/.bam};\n",
    "#  \to=${o/fastq_unmapped/bam};\n",
    "#  \tbowtie2 --sensitive -N 1 -x HIV/EU541617.1/index -U $f | samtools sort -o $o;\n",
    "# done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Apr 03 03:33:26 ..... started STAR run\n",
      "Apr 03 03:33:26 ... starting to generate Genome files\n",
      "Apr 03 03:33:26 ... starting to sort Suffix Array. This may take a long time...\n",
      "Apr 03 03:33:26 ... sorting Suffix Array chunks and saving them to disk...\n",
      "Apr 03 03:33:26 ... loading chunks from disk, packing SA...\n",
      "Apr 03 03:33:26 ... finished generating suffix array\n",
      "Apr 03 03:33:26 ... generating Suffix Array index\n",
      "Apr 03 03:33:26 ... completed Suffix Array index\n",
      "Apr 03 03:33:26 ..... processing annotations GTF\n",
      "Apr 03 03:33:26 ... writing Genome to disk ...\n",
      "Apr 03 03:33:26 ... writing Suffix Array to disk ...\n",
      "Apr 03 03:33:26 ... writing SAindex to disk\n",
      "Apr 03 03:33:26 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# use STAR \n",
    "STAR --runThreadN 16 \\\n",
    "--runMode genomeGenerate \\\n",
    "--genomeDir HIV/EU541617.1/ \\\n",
    "--genomeFastaFiles HIV/EU541617.1/EU541617.1.fasta \\\n",
    "--sjdbGTFfile HIV/EU541617.1/EU541617.1.gff3 \\\n",
    "--sjdbOverhang 99 --genomeSAindexNbases 8 --sjdbGTFfeatureExon CDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.METTL14.input.fastq --outFileNamePrefix bam/s23.hiv.METTL14.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.METTL14.m6A.fastq --outFileNamePrefix bam/s23.hiv.METTL14.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.METTL3.input.fastq --outFileNamePrefix bam/s23.hiv.METTL3.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.METTL3.m6A.fastq --outFileNamePrefix bam/s23.hiv.METTL3.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.NT.input.fastq --outFileNamePrefix bam/s23.hiv.NT.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.NT.m6A.fastq --outFileNamePrefix bam/s23.hiv.NT.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.Virma.input.fastq --outFileNamePrefix bam/s23.hiv.Virma.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.Virma.m6A.fastq --outFileNamePrefix bam/s23.hiv.Virma.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.WTAP.input.fastq --outFileNamePrefix bam/s23.hiv.WTAP.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s23.hiv.WTAP.m6A.fastq --outFileNamePrefix bam/s23.hiv.WTAP.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.METTL14.input.fastq --outFileNamePrefix bam/s24.hiv.METTL14.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.METTL14.m6A.fastq --outFileNamePrefix bam/s24.hiv.METTL14.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.METTL3.input.fastq --outFileNamePrefix bam/s24.hiv.METTL3.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.METTL3.m6A.fastq --outFileNamePrefix bam/s24.hiv.METTL3.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.NT.input.fastq --outFileNamePrefix bam/s24.hiv.NT.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.NT.m6A.fastq --outFileNamePrefix bam/s24.hiv.NT.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.Virma.input.fastq --outFileNamePrefix bam/s24.hiv.Virma.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.Virma.m6A.fastq --outFileNamePrefix bam/s24.hiv.Virma.m6A.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.WTAP.input.fastq --outFileNamePrefix bam/s24.hiv.WTAP.input.bam --limitBAMsortRAM 1000000000\n",
      "STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn fastq_unmapped/s24.hiv.WTAP.m6A.fastq --outFileNamePrefix bam/s24.hiv.WTAP.m6A.bam --limitBAMsortRAM 1000000000\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# STAR --genomeLoad LoadAndExit --genomeDir HIV/EU541617.1/\n",
    "for f in fastq_unmapped/*; do\n",
    "\to=${f/.fastq/.bam};\n",
    "\to=${o/fastq_unmapped/bam};\n",
    "\techo STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn $f --outFileNamePrefix $o --limitBAMsortRAM 1000000000;\n",
    "# \tSTAR --outSAMtype BAM SortedByCoordinate --readFilesCommand cat --runThreadN 16 --genomeDir HIV/EU541617.1/ --readFilesIn $f --outFileNamePrefix $o --limitBAMsortRAM 1000000000;\n",
    "done\n",
    "# STAR --genomeLoad Remove --genomeDir HIV/EU541617.1/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'bam/s23.hiv.METTL14.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.METTL14.input.bam'\n",
      "renamed 'bam/s23.hiv.METTL14.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.METTL14.m6A.bam'\n",
      "renamed 'bam/s23.hiv.METTL3.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.METTL3.input.bam'\n",
      "renamed 'bam/s23.hiv.METTL3.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.METTL3.m6A.bam'\n",
      "renamed 'bam/s23.hiv.NT.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.NT.input.bam'\n",
      "renamed 'bam/s23.hiv.NT.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.NT.m6A.bam'\n",
      "renamed 'bam/s23.hiv.Virma.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.Virma.input.bam'\n",
      "renamed 'bam/s23.hiv.Virma.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.Virma.m6A.bam'\n",
      "renamed 'bam/s23.hiv.WTAP.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.WTAP.input.bam'\n",
      "renamed 'bam/s23.hiv.WTAP.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s23.hiv.WTAP.m6A.bam'\n",
      "renamed 'bam/s24.hiv.METTL14.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.METTL14.input.bam'\n",
      "renamed 'bam/s24.hiv.METTL14.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.METTL14.m6A.bam'\n",
      "renamed 'bam/s24.hiv.METTL3.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.METTL3.input.bam'\n",
      "renamed 'bam/s24.hiv.METTL3.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.METTL3.m6A.bam'\n",
      "renamed 'bam/s24.hiv.NT.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.NT.input.bam'\n",
      "renamed 'bam/s24.hiv.NT.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.NT.m6A.bam'\n",
      "renamed 'bam/s24.hiv.Virma.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.Virma.input.bam'\n",
      "renamed 'bam/s24.hiv.Virma.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.Virma.m6A.bam'\n",
      "renamed 'bam/s24.hiv.WTAP.input.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.WTAP.input.bam'\n",
      "renamed 'bam/s24.hiv.WTAP.m6A.bamAligned.sortedByCoord.out.bam' -> 'bam/s24.hiv.WTAP.m6A.bam'\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in bam/*Aligned.sortedByCoord.out.bam; do o=${f/Aligned.sortedByCoord.out.bam/}; mv -v $f $o; done "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All `bam`  files (mapped to human genome or hiv genome) renamed in this format: \n",
    "> `<sample>.<species>.<treatment>.<input/m6A>.bam`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Id transfer \n",
    "Read gene names from `gtf` file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import os\n",
    "import feather as ft\n",
    "\n",
    "\n",
    "def gtf_id_transfer(gtf):\n",
    "    # awk code to extract gene ID and Name \n",
    "    gtf_raw = os.popen(\"cat \" + gtf + \" | awk 'BEGIN{FS=\\\"\\t\\\"}{split($9,a,\\\";\\\"); if($3~\\\"gene\\\") print a[1]\\\"\\t\\\"a[3]}' | sed 's/gene_id \\\"//' | sed 's/gene_name \\\"//' | sed 's/\\\"//g' | sed 's/ //g'\").read().split('\\n')\n",
    "    # data frame \n",
    "    gtf_df = pd.DataFrame(\n",
    "        # col: Names \n",
    "        data=[i.split('\\t') for i in gtf_raw[:len(gtf_raw)-1]],\n",
    "#         # index: ids \n",
    "#         index=[i.split('\\t')[0] for i in gtf_raw[:len(gtf_raw)-1]],\n",
    "        columns=['ID','Name'], dtype=str)\n",
    "    return gtf_df\n",
    "\n",
    "gtf = '/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf'\n",
    "gtf_df = gtf_id_transfer(gtf)\n",
    "\n",
    "# pass data between R and Python using feather-format\n",
    "ft.write_dataframe(gtf_df, 'gtf_df.ft')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(feather)\n",
    "gtf_df = read_feather('gtf_df.ft') %>% remove_rownames %>% column_to_rownames(var=\"ID\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peakcalling "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Control\n",
    "\n",
    "Comparing to RADAR package, exomepeak also claculate coverage for only non threted samples. We used different conda `environment` for exomepeak (and it takes long time to run!!). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(exomePeak)\n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')\n",
    "setwd(\"/rumi/shams/abe/People/Judd/bam\")\n",
    "print (txdb)\n",
    "res <- exomepeak(\n",
    "    TXDB=txdb,\n",
    "    IP_BAM=list.files(pattern='human.NT.m6A.bam$'),\n",
    "    INPUT_BAM=list.files(pattern='human.NT.input.bam$'),\n",
    "    OUTPUT_DIR='../exomepeak',\n",
    "    EXPERIMENT_NAME='control'\n",
    ")\n",
    "saveRDS(res, paste('../exomepeak','control', 'results.rds', sep='/'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot _Guitar_ \n",
    "Now, let's draw metagene plots for control samples:\n",
    "(using seprate env: `Guitar`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Guitar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"20200325183907\"\n",
      "[1] \"There are 203835 transcripts of 58381 genes in the genome.\"\n",
      "[1] \"total 203835 transcripts extracted ...\"\n",
      "[1] \"total 94682 transcripts left after ambiguity filter ...\"\n",
      "[1] \"total 94682 transcripts left after check chromosome validity ...\"\n",
      "[1] \"total 11625 mRNAs left after component length filter ...\"\n",
      "[1] \"total 56222 ncRNAs left after ncRNA length filter ...\"\n",
      "[1] \"generate components for all tx\"\n",
      "[1] \"generate components for mRNA\"\n",
      "[1] \"generate components for lncRNA\"\n",
      "[1] \"generate chiped transcriptome\"\n",
      "[1] \"generate coverage checking ranges for tx\"\n",
      "[1] \"generate coverage checking ranges for mrna\"\n",
      "[1] \"generate coverage checking ranges for ncrna\"\n",
      "[1] \"20200325184342\"\n",
      "[1] \"import BED file exomepeak/control/peak.bed\"\n",
      "[1] \"sample 10 points for Group1\"\n",
      "[1] \"start figure plotting for tx ...\"\n",
      "[1] \"start figure plotting for mrna ...\"\n",
      "[1] \"start figure plotting for ncrna ...\"\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "GuitarPlot(txTxdb = txdb,stBedFiles = list(\"exomepeak/control/peak.bed\"),miscOutFilePrefix = \"control\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !magick -density 300 plots/control_mrna_test.pdf -flatten -quality 90 plots/control_mrna_test.png"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Motif analysis "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cd exomepeak/control/\n",
    "bedtools getfasta -name -s -fi /rumi/shams/genomes/hg38/hg38.fa -bed peak.bed -split -fo peak.fa\n",
    "# prepare fasta file for FIRE\n",
    "perl /flash/hani/bin/Tools/prep_seqs_for_teiser_run.pl peak.fa peaks\n",
    "# run FIRE --doskipdiscovery=1\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --doskipdiscovery=1 --motiffile_rna=motifs_of_interest.txt --oribiasonly=0\n",
    "mv peaks_teiser.txt_FIRE/ motifs_of_interest_FIRE\n",
    "# run FIRE --doskipdiscovery=0\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --oribiasonly=0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running a custom pipeline \n",
    "\n",
    "Here we're using https://github.com/gold-lab/bed12ToGTF to make gtf file from control exomepeak result.\n",
    "\n",
    "https://metacpan.org/pod/Bio::FeatureIO::gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat exomepeak/control/peak.bed | sort -k4 | \\\n",
    "# rename features peak by peak\n",
    "awk -F \"\\t\" '{OFS=FS}NR>1{$4=$4\"_peak\"}; cnt[$4]++{$4=$4\"_\"cnt[$4]} 1' | \\\n",
    "# bed12 to gtf\n",
    "perl ../../Workflows/bed12ToGTF/bed12ToGTF.1.pl | \\\n",
    "awk '{gsub(/\\\"|\\;/,\"\")}1''{gsub(/transcript_id/,\"gene_id\")}1''{gsub(/ exon_number [[:digit:]]/,\"\")}1' > exomepeak/control/peak.gtf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using `featureCounts` to count reads which are present in control peak cordinates (found by `exomepeak`). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.METTL14.input.bam                ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.METTL14.input.fc             ||\n",
      "||                 Summary : peakcount/s23.human.METTL14.input.fc.summary     ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.METTL14.input.bam...                        ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 38750994                                                  ||\n",
      "||    Successfully assigned reads : 939351 (2.4%)                             ||\n",
      "||    Running time : 0.84 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.MET  ||\n",
      "|| TL14.input.fc.summary\"                                                     ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.METTL14.m6A.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.METTL14.m6A.fc               ||\n",
      "||                 Summary : peakcount/s23.human.METTL14.m6A.fc.summary       ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.METTL14.m6A.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 17154692                                                  ||\n",
      "||    Successfully assigned reads : 9044401 (52.7%)                           ||\n",
      "||    Running time : 0.37 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.MET  ||\n",
      "|| TL14.m6A.fc.summary\"                                                       ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.METTL14.input.bam                ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.METTL14.input.fc             ||\n",
      "||                 Summary : peakcount/s24.human.METTL14.input.fc.summary     ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.METTL14.input.bam...                        ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 64012763                                                  ||\n",
      "||    Successfully assigned reads : 822388 (1.3%)                             ||\n",
      "||    Running time : 1.40 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.MET  ||\n",
      "|| TL14.input.fc.summary\"                                                     ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.METTL14.m6A.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.METTL14.m6A.fc               ||\n",
      "||                 Summary : peakcount/s24.human.METTL14.m6A.fc.summary       ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.METTL14.m6A.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8506966                                                   ||\n",
      "||    Successfully assigned reads : 4543360 (53.4%)                           ||\n",
      "||    Running time : 0.19 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.MET  ||\n",
      "|| TL14.m6A.fc.summary\"                                                       ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.METTL3.input.bam                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.METTL3.input.fc              ||\n",
      "||                 Summary : peakcount/s23.human.METTL3.input.fc.summary      ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.METTL3.input.bam...                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 45742294                                                  ||\n",
      "||    Successfully assigned reads : 1149406 (2.5%)                            ||\n",
      "||    Running time : 1.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.MET  ||\n",
      "|| TL3.input.fc.summary\"                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.METTL3.m6A.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.METTL3.m6A.fc                ||\n",
      "||                 Summary : peakcount/s23.human.METTL3.m6A.fc.summary        ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.METTL3.m6A.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 12180232                                                  ||\n",
      "||    Successfully assigned reads : 5997991 (49.2%)                           ||\n",
      "||    Running time : 0.27 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.MET  ||\n",
      "|| TL3.m6A.fc.summary\"                                                        ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.METTL3.input.bam                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.METTL3.input.fc              ||\n",
      "||                 Summary : peakcount/s24.human.METTL3.input.fc.summary      ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.METTL3.input.bam...                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 47661894                                                  ||\n",
      "||    Successfully assigned reads : 714579 (1.5%)                             ||\n",
      "||    Running time : 1.05 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.MET  ||\n",
      "|| TL3.input.fc.summary\"                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.METTL3.m6A.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.METTL3.m6A.fc                ||\n",
      "||                 Summary : peakcount/s24.human.METTL3.m6A.fc.summary        ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.METTL3.m6A.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 11789267                                                  ||\n",
      "||    Successfully assigned reads : 5806586 (49.3%)                           ||\n",
      "||    Running time : 0.25 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.MET  ||\n",
      "|| TL3.m6A.fc.summary\"                                                        ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.Virma.input.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.Virma.input.fc               ||\n",
      "||                 Summary : peakcount/s23.human.Virma.input.fc.summary       ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.Virma.input.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 43594515                                                  ||\n",
      "||    Successfully assigned reads : 860140 (2.0%)                             ||\n",
      "||    Running time : 0.97 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.Vir  ||\n",
      "|| ma.input.fc.summary\"                                                       ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.Virma.m6A.bam                    ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.Virma.m6A.fc                 ||\n",
      "||                 Summary : peakcount/s23.human.Virma.m6A.fc.summary         ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.Virma.m6A.bam...                            ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 15882391                                                  ||\n",
      "||    Successfully assigned reads : 8113156 (51.1%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.Vir  ||\n",
      "|| ma.m6A.fc.summary\"                                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.Virma.input.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.Virma.input.fc               ||\n",
      "||                 Summary : peakcount/s24.human.Virma.input.fc.summary       ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.Virma.input.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 56471415                                                  ||\n",
      "||    Successfully assigned reads : 689932 (1.2%)                             ||\n",
      "||    Running time : 1.24 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.Vir  ||\n",
      "|| ma.input.fc.summary\"                                                       ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.Virma.m6A.bam                    ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.Virma.m6A.fc                 ||\n",
      "||                 Summary : peakcount/s24.human.Virma.m6A.fc.summary         ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.Virma.m6A.bam...                            ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 12393273                                                  ||\n",
      "||    Successfully assigned reads : 6371965 (51.4%)                           ||\n",
      "||    Running time : 0.27 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.Vir  ||\n",
      "|| ma.m6A.fc.summary\"                                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.WTAP.input.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.WTAP.input.fc                ||\n",
      "||                 Summary : peakcount/s23.human.WTAP.input.fc.summary        ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.WTAP.input.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 37223063                                                  ||\n",
      "||    Successfully assigned reads : 1025554 (2.8%)                            ||\n",
      "||    Running time : 0.83 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.WTA  ||\n",
      "|| P.input.fc.summary\"                                                        ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.WTAP.m6A.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.WTAP.m6A.fc                  ||\n",
      "||                 Summary : peakcount/s23.human.WTAP.m6A.fc.summary          ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.WTAP.m6A.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 14386845                                                  ||\n",
      "||    Successfully assigned reads : 7796629 (54.2%)                           ||\n",
      "||    Running time : 0.30 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.WTA  ||\n",
      "|| P.m6A.fc.summary\"                                                          ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.WTAP.input.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.WTAP.input.fc                ||\n",
      "||                 Summary : peakcount/s24.human.WTAP.input.fc.summary        ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.WTAP.input.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 75448172                                                  ||\n",
      "||    Successfully assigned reads : 867570 (1.1%)                             ||\n",
      "||    Running time : 1.64 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.WTA  ||\n",
      "|| P.input.fc.summary\"                                                        ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.WTAP.m6A.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.WTAP.m6A.fc                  ||\n",
      "||                 Summary : peakcount/s24.human.WTAP.m6A.fc.summary          ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.WTAP.m6A.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 14438949                                                  ||\n",
      "||    Successfully assigned reads : 7201731 (49.9%)                           ||\n",
      "||    Running time : 0.31 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.WTA  ||\n",
      "|| P.m6A.fc.summary\"                                                          ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for enz in \"${StringArray[@]}\"; \n",
    "    do for f in bam/*human.$enz*.bam; \n",
    "        do out=${f/.bam/.fc}; out=${out/bam/peakcount}; \n",
    "            featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/control/peak.gtf -o $out $f; \n",
    "    done;\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.NT.input.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.NT.input.fc                  ||\n",
      "||                 Summary : peakcount/s23.human.NT.input.fc.summary          ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.NT.input.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 41946113                                                  ||\n",
      "||    Successfully assigned reads : 601429 (1.4%)                             ||\n",
      "||    Running time : 1.46 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.NT.  ||\n",
      "|| input.fc.summary\"                                                          ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.NT.m6A.bam                       ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.NT.m6A.fc                    ||\n",
      "||                 Summary : peakcount/s23.human.NT.m6A.fc.summary            ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.NT.m6A.bam...                               ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 15481162                                                  ||\n",
      "||    Successfully assigned reads : 8231587 (53.2%)                           ||\n",
      "||    Running time : 0.52 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.NT.  ||\n",
      "|| m6A.fc.summary\"                                                            ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.NT.input.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.NT.input.fc                  ||\n",
      "||                 Summary : peakcount/s24.human.NT.input.fc.summary          ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.NT.input.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 39882640                                                  ||\n",
      "||    Successfully assigned reads : 823906 (2.1%)                             ||\n",
      "||    Running time : 1.25 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.NT.  ||\n",
      "|| input.fc.summary\"                                                          ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.NT.m6A.bam                       ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.NT.m6A.fc                    ||\n",
      "||                 Summary : peakcount/s24.human.NT.m6A.fc.summary            ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.NT.m6A.bam...                               ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 13525429                                                  ||\n",
      "||    Successfully assigned reads : 7258122 (53.7%)                           ||\n",
      "||    Running time : 0.50 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.NT.  ||\n",
      "|| m6A.fc.summary\"                                                            ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in bam/*human.NT*.bam; do \n",
    "    out=${f/.bam/.fc}; out=${out/bam/peakcount}; \n",
    "    featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/control/peak.gtf -o $out $f; \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(DESeq2)\n",
    "library(tidyverse)\n",
    "library(ggplot2)\n",
    "require(ggrepel)\n",
    "library(patchwork)\n",
    "library(feather)\n",
    "gtf_df = read_feather('gtf_df.ft') %>% remove_rownames %>% column_to_rownames(var=\"ID\")\n",
    "\n",
    "pca = list(); vol = list(); hist = list()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "using `DESeq2` to find differential methylation peaks: \n",
    "\n",
    "### Contrast design: $\\frac{m6A^{treated}}{input^{treated}}$ vs.$\\frac{m6A^{nontreated}}{input^{nontreated}}$"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n",
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n",
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n",
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "all_dds = list()\n",
    "all_res = list()\n",
    "for (treat in c('METTL14','METTL3','Virma','WTAP'\n",
    "               )){\n",
    "    treat.input <- list.files(path = 'peakcount',pattern=paste('human.',treat,\".input.fc$\",sep = ''))\n",
    "    treat.m6A <- list.files(path = 'peakcount',pattern=paste('human.',treat,\".m6A.fc$\",sep = ''))\n",
    "    \n",
    "    n_treat.input <- list.files(path = 'peakcount',pattern=paste(\"human.NT.input.fc$\",sep = ''))\n",
    "    n_treat.m6A <- list.files(path = 'peakcount',pattern=paste(\"human.NT.m6A.fc$\",sep = ''))\n",
    "    \n",
    "    files = c(treat.input, treat.m6A, n_treat.input, n_treat.m6A)\n",
    "    # meta\n",
    "    reps <- rep(c('s23','s24'),2)\n",
    "    ip <- c(rep('input',2),rep('m6A',2))\n",
    "    conds <- c(paste(treat,ip, sep='.'), paste('NT',ip, sep='.'))\n",
    "    all_conds <- c(paste(reps,treat,ip, sep='.'), paste(reps,'NT',ip, sep='.'))\n",
    "    datalist <- lapply(files, function(x){read.table(file=paste('peakcount/', x, sep = \"\"),skip=1,sep='\\t',header=TRUE,colClasses=c(NA, \"NULL\", \"NULL\", \"NULL\", \"NULL\", \"NULL\", NA),check.names=F)})\n",
    "    m <- Reduce(function(...) merge(..., by=1, all = TRUE), datalist)\n",
    "    rownames(m) <- m[,1]\n",
    "    m <- m[,-1]\n",
    "    m[is.na(m)]<-0\n",
    "    m <- round(m)\n",
    "    colnames(m) <- all_conds\n",
    "    colData <- data.frame(cond = conds, rep = reps, row.names=all_conds)\n",
    "    dds <- DESeqDataSetFromMatrix(countData = m, colData = colData, design = ~ cond)\n",
    "    dds$cond <- relevel(dds$cond, ref=\"NT.input\")\n",
    "    dds <- DESeq(dds)\n",
    "    all_dds[[treat]] = dds\n",
    "    # normalized counts \n",
    "    ncu <- counts(dds, normalized=TRUE)\n",
    "    # contrast design: (treat_m6A - treat_input) - (NT_m6A - NT_input)\n",
    "    res <- results(dds, \n",
    "                   contrast=list(\n",
    "                       c(paste(\"cond_\",treat,\".m6A_vs_NT.input\",sep=\"\")), \n",
    "                       c(paste(\"cond_\",treat,\".input_vs_NT.input\",sep=\"\"),\n",
    "                         \"cond_NT.m6A_vs_NT.input\")), \n",
    "                   listValues=c(1,-1))    \n",
    "    all_res[[treat]] = res\n",
    "    # idtransfer\n",
    "    res <- as.data.frame(res)\n",
    "    peak_ids = strsplit(rownames(m), '_',)\n",
    "    ids = unlist(lapply(peak_ids, '[[', 1))\n",
    "    res$name = gtf_df[ids,]\n",
    "    \n",
    "    # save files \n",
    "    write.table(res, paste('deseq/',treat,'_peak_m6A_vs_input.txt',sep=''), sep=\"\\t\", quote=FALSE, col.names=NA)\n",
    "    # Raw tabels \n",
    "    write.table(ncu, file=paste('deseq/',treat,'_peak_count_table_DE2norm.txt',sep=''), sep=\"\\t\", quote=FALSE, col.names=all_conds) \n",
    "    write.table(m, file=paste('deseq/',treat,'_peak_count_table_raw.txt',sep=''), sep=\"\\t\", quote=FALSE, col.names=all_conds)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plots \n",
    "\n",
    "### PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NULL\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# ma = list()\n",
    "# for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "#     res = all_res[[treat]]\n",
    "# #     ma[[treat]] = plotMA(res)\n",
    "# }    \n",
    "# p = ma$METTL14 #+ ma$METTL3 + ma$Virma + ma$WTAP\n",
    "# ggsave(\"plots/peak_plotMA.png\", plot = p, device = 'png', dpi = 300)\n",
    "# ggsave(\"plots/peak_plotMA.pdf\", plot = p, device = 'pdf', dpi = 300)\n",
    "# p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////isF19AAAgAElEQVR4nOydB5wTVf7AudM7z794VuROLGcXBVEpogJLR0CKgnqceBZUrKCggggqCFKkI4oIAiJ9lw7bssn23nfSNsmmzrb03vP+781MdpPs7G4SdpPdXH4fyEze/t6b3y/fmTevvz4gLjEtfaJtQFy6V+KAY1zigGNc4oBjXOKAY1zigGNc4oBjXOKAY1zigGNc4oBjXIICPL3PT/BT97c+cniK5Oc+ASKfvpDQPDPhhj5GMtLzfZK7y+hulE5dheGkq1sHXnvLLEF0rQ1CggN8x9Pwc98dyLu5FVBU8P+MsRUV5cXkscJBeX3ku40U4H2TeyfgdlwtP1BMeeoFvO9YQfKoh6JrbRASHOCFN4gASFjWevtCeX2W77ElPIcELLlT0jsBd+AqdWgNB3l9VBG2L2QJDvDit1YDad/C4AF7xh5w9k7AHbjaBrDmnUc8kTYwVAkSMOtBsG5+DfL6T1dBqQSdAN46FfRWwO27GgA46ao+D0uiZGbwEiRgz11FA1MJr1+ugWIFHQPm/0PRawG372oAYH1N2oRRrijZGbQECRh8Oeqfrpqgs+jD6Obv8+f/dLGxEZCOXG37Dtb+KSPC9oUswQLm9FkCggesg7d+ZZ898i42NgLSkattATf3yYywfSFLsICBykZ6TdQdGkAbwES4QVNxoE9+hQWF9NYsun1XvYBJV986W3jm6fvNUbQ1KAkaMJSaltr/N6ANYCI8+TBxqEAhvRcwaMdVL2DS1dfu/Osd8+uiZGbwEm+qjHGJA45xiQOOcYkDjnGJA45xiQOOcYkDjnGJA45xCQbw+p2kbNu6k0620Ybu3EwbumM7bfAW+mD6pMO0Y5m1U0+zvWZs3hHCJba2YyZt6I4ttMHb6S8Ynh3rzoUGeCd1tBho/2ym7RN103eFu2y0wTo7bbCJNtQakh1K78lQI200Xzmspk7U9L1E9PYY6dsr6ZUdGtpgu4M2OCQ7TF479m5sDYwD9pU4YDqJA44DbpU44DhgH4kDjgPu1I444E4lDrhje+KAfSQOOA7YR+KA44A7tSMOuFOJA+7YnjhgH4kDjgP2kTjgOOBO7YgD7lS2WkjRay10ojfThZpVtMomA22wxkgbrKMNNdDboaO3Q+m9chCAD+GUssoUgj1a+mD6UCP972Kk/wFUtD61k7ROT53sDvUJ9pBiNnjoxOSmC3WpaJWdVtpgnY0+adpQS0h2KL0nQQD+3Wu02hmCPUb6YPpQu4Y22GanDQ7JDpOZOvklnkW3I/+jWXQccMf2xAH7SBxwHLCPxAHHAXdqRxxwpxIH3LE9ccA+EgccB+wjccBxwJ3aEQfcqcQBd2xPHLCPxAHHAftIHHAccKd2xAF3KnHAHdsTB+wjccBxwD4SI4Dt/540djd3DjrNHzd6TGbJ02Om65X9JkzgADDxUxiaOXbMAdo04oA7t6MHAD6yAgA9AVj1mFRjKmowgU2blcS6ec3Pj/YAc4I58k+w56fVq6nfJtYBt7raTYCTxgsAIAAfWE1l0Vu3KW+fsNgGfvrjqxyQPHPOHBFtGt0IuGwbqN5CnsY64FZXuwmwe8Nj96UTgNfuIwHXD9Y6jeDLzWCqkfMR+G2glTGbNo1uBHzxDDC8SZ7GOuBWV7uvkFUx0PcJNo0qQoFl/6kfMGXKg+7TbwD7QNo0uhEwe40n9yV4/G3Zsk0GUrRqA51oaUP1KvpgHW2wmj5YQxuqC8kOpTdWu4ApV/Fly5ZLKWWVPgR71PTBraF8laHukdJZ8EzymEDVnK2ddsRgaDAYfli26TeDYflFwVP6oom0aYRkh8b7C+wKqpCVuObgh/CQl5S0xUqKQWulE72FLtSiolU2G2mDNfTBOtrQ0OxQeq/cfiGLdFWXlLSxnlJWmUOwR6fvRDn52Ykjz1X2GzdumTVz3Ljxab/fMG7c1svDxkxveEZmtaa/bd0yYUwRbRrh2fFTkKXovAPkMdazaNDi6v9SPdi1cuU2amGpWAfc6mq3Ay5t6jGAfSTWAbdKdwPOu6EoDthXYgwwdsvBnpNF+0gccMf2BAtYftfWHvQO9pE44I7tCRKw8uGVIA7YX2IJsPnpd9EhDthXejxgop+orA8TeO5aRvYMcftPeaaC6C7yVbborbNeINKMA/aVHg+Y6CcqG/o+yBuxjOwZ4s4BOdOI7iIfZWOtlHeW9DwO2Fd6PGCin6hs7jT34h+WkT1DELBuEEDdRV5lj0lXj+M4lXR0ATtLzqQ2tokYi4DpPQ0ZMNFPVDZ3Q8ZzF5aRPUMQ8PmXiO4ir7JOJCsW4FId+S2agJuX3Dp40uh+z54NiBh7gNvzNJxCVsXAsrmiJ9dcWEb2DMF38Ow6orvIq8zGcUkdW+0mv0UT8Ihf9ehQ+vaP/hFjD3B7noYMGLeD5kFlc8FiwYVlO44B8DWT6BV0zTzdqqyW43Wtl44mYFebE1JiD3B7noYMOGP05GeSIWAALiwb1QRA1kIC8LEbJ0zY5VV2qtmaVtOiXMiyCWgixh5g0I6nMd+blH7XAyD/lcCIsQiY3tOYBzxMngDAw4ERYxEwvacxD/hJkABAm9FBsQiY3tNuAWw3O31Cowt4VGMCuDQqMGIsAqb3tDsAG3l1bB/vogs4Z1C/0bcUBEaMRcD0nnYHYAzHZdrW0CiXolXnzihBoGx3kGLUOejEaKcLtalolW1m2mCthTbYQBtqorfDQG+H0nviX4qm9fT3RkpZbQvBHj19MBXaIJGKlK2hFjWtsoX+BwjJDoOROvk5xLbomARMK+EB1jWb2lfGj1ViDdbW0OgCvoaQQLdjMYum9zSsLNpdXMPT0oSTyqfHenzLWNHubHA6zSfWBUaMRcD0noYF2FItxDnu9pR/neMf2gO6C+cFBsQiYELaeBoW4J+yhTi7XcCb3vYPjT5gfHhgSKwCbutpOIDrrqus5NJZRCp/+bl/aHQB33LLLTf/MzEwYiwCpvc0HMCfvGPUOWnCKeWF36NPq1pNEYwuYK1WS1PUjEXA9J6GAdhwQ1WH9eCXfoIfTo5CjpHmRROwdxJTYMTYA9yep2EA3jqlw4YO0zPb1B5gEeE4j3zMown4mmv+V6pJ7XkaOmDXvSkdAfZgjx0Tmokn+GIeERr9QlZbiT3A7UnogBMf8nQImHN3cp0JvYM1GTfmo9A4YF/p+YCf3dtxW7T2hswaqgp19KZKEG3Ami+nT5gwITBiLAKm9zRkwKU3mzsG7P5zQ0sVefcAcbQBz/5+yLl5nwVGjEXA9J62As4dP25Mpvmp69MB8dFqz7w5ssyxY393zZ88php+Q9ONEOAFmeD3YQAMGjGh77gJNcTiV+VojoPmzz5NIF/d3xhlwIPBKADmB0aMRcD0nrYAFgyuB6YiV+PidEB8eO1xG6agVa0AyH8ZZL0GFNfhgAS8fz34YKLF9AQAQ6zkFAdQjuY4CG/2Sd7zzjADBdipZut8rxwhwEPBaA14LjBiLAKm97QF8I/fkEeCLfxwzJ8xraFg7Hunlt/xdvLMOS9JFf8G55YC4Q6khABzZoHZW7OZCwnAxBQHBFg3qOEN3/RdL06ykYDVMlzo612EAM9T/vxgwquBEWMRML2nLYC/2ksevYCPrAIXFxVMNQHldLSqVdaLrteH3SumEkaAPQPNr+Zt/P4gAZiY4oAAozkO/qavsZCA2Tgu832EI1eKLklrk2YsAga0nrb7BH97FjQ8V7CEAHwaPpSPJC4AFc8jDZeKW4xIPXdwq/WVmXwEmJzigN7Bs+va2ENl0RqxgufbzhIhwPQRYxEwfcqt7+AhTcBa2gr4j9UgeVHBUgJwwzMeznMnlwHZs0hTK8V5CnhcPSQPzHgEIMDkFAf0BPuIakwCLH5tW+jQHPuG23/K0zs1FnIhUpDRRwvA+MW0BnV5Z8Ob2TQRYxEwvac+peiJkydmgll3PrGe+HC8+sI0nAIMdk4aV22fO3UUC2lqFHgtApz2Fyv4EP4RAianOPgDNglx6SArSCcBzwH2B6iFSAGYM0QLmqeNorW+qwGrdj5533dS8py3cuUp8iwWAft62upqGG3RVo64rLntVc1Wf2WtHMcf8gGsewxQZbGLO5/Sgp8OfplDd8FueAfXvP5n4mj/3OINikXAoNVTH1fD6S50GFVKS8Cl3ZhIoPFTNvNxcesT3H/K3VsAWRbzzLRBwFPV1R/RXbDLATvPvXjrW8RZ5eqN36EiQvrBg1uojaV7+A7g7djRzg7grZ56XVUfPLg2nB3AzaJKTOEfpGaLRBVq74XxsnqTRYk3DtZYMl4zqvZ+VzHbYhheW3f7xIkPmI5ttwxvIE9p0u7qHcA/6T/+d+o2YC6w1n8Cj8fXrdtkJEWrMdKJljbUoKIP1tEGq/UhJK0LyQ4ldaL3A+zjqdfVxnXrVsooZZUh+Euoyri15f4mKStra0vU1BcFt5ajQCeDlUbJ42rVm6fLZhmNows3HTAal19aMXbi35/bdECvX36JLu0Q7DBqvcE/tgu45a0EKxEbAFhINrPRZ9H5qDEPrTbSWIyWHGn9w7w5MlgyVKtQix4IlJ6SRft46uNqWGOy7OVCnBugr+GyjS3zg73rN8AaMvhtdMLHaKGWhMWAKosB8JR2VJPdQZwGSpdn0U3nz1MlBsOnHsOH5CktYNVjqDEPrTayvh4tOeINR415qGSolifQtcH3FMA+nvq4Gt7MBrysBtdY/MNc7hZlY62stmdMAAcg6Y6ZU289Q56nr1xeSZ7RAj6wmjrZutGDlhyhGvM+Ro15qGSoPoFa9NoY0VMA+3ra6mqYU1eMKol/s4W/slnrc6tHF/AzsOjLeygwIi3gtfvIY/3geg9acoRqzAOopohKhuqdqEWvjRE9BTC9p+HOTWLjLYusdKYcXcBE4+xjgRE7eoJNo4rMHrTkCNWYRwBO2g2eUh9CLXptjHDZuOSyYbAuyL1twoRPuP2fG5kGWCP9XuMdO9IVgOk9DRewWoYL2ryQeiLgr44YNRvXBEakfwc/jhrz0GojZidacoRqzCMAfzthyg3TOahFr40REPBwYtkwBBg19cAP0UjA1/i8xjtzpCsA03saLmCnmt329uyJgPtQ4h+xnVI0asxDq41sOYqWHKEa8wjAUJ5Sq1CLXhsjIOBXiGXDfACXjERZ9NZtwTrSFYDpPY35Gf700rUtWdxXiGXDEOBbExLWcvsn/D0bAoavcRqTadPoSWOykPQawFyvDUL/iKEB3k+24ncAmFg2zOcJPrwa6DTkRjOBJtOmceWA2/M0xgFPmpsodxiK1w654B8xNMBnbypHhw4AE8uG+QD2DNOqnz8dvCNXDrg9T7sUsJ52QktUs+jUl/r1+cvj3wV2joSYRR+6jQc6BoyWDaNK0W8iyjs27kev8WAd6YIsuh1PuxKwqZSnorEo2u9gB41Nob6Dv7u3vhf0JtF52pWAMQFeR6NPAraZAh7vCAFuKCpqaBsx5ELWp4M1nQBmBQRHHDC9p10I2MMW4GKaYh0B2MAXs/09jghgXkLfBx/oO7Y2MGLIgD2vJ1g7BPzNAwGGRxhwe55eAWC30eAfR18hxFw2tTqgCZMAjAUMuYvUarMHYL7h2PtUYMTQq0n2KduVtbQZFwH459sDB6NFerXZdjwNH7BHJZZgXgM8VuSlxuJ2YwoFxz8zRoCdIkws8XciIoDvIQ93BkYMox7sqZFiPDpoCPCZmysCgyMMuD1Pwwds5+O4gCpfeFSCWg2h7IChQv9HGAHGhBkXG/0XfYgI4OHkfT0iMGIYgF08OUZXxECAmTdmtQmOMOD2PA0fsIuD41zqWbUKiNnAJjR7VK5g+ycFAbt4uAI1XXuMrcN6IgKY80zfBx/sO4oXGDGclixVXUEBXYXAZau6uc3KCREH3J6nV/AONmGYN31rjRiNAIDKpsp8eYBnELAHU+BcCNQgVAi8cSJUipbl5srbRgwHsLOOVVfbxg6PSYv330ejHvFSNL2nV1KKdjRTy357VGWpRXqkTDzWJo02cDFSu1qFoqhbt+rohW3RNRgub/MIa0R1ZbTDQ3vt/GAkBGBHbhZDTHy11eJSVLIyAXstBFwhl/pWKIhStMcGebrr+QqJtywdScB3BwbscJFi0rvoxOSkC22qkvF1AWHOGpmEFxhIiNZKm7SRNtRMb4eR1g6H0nvSFnAbT39vppTV9hDsMaDgRmZJyXnCCzNHJsM0RqfR5awSiYr4Mhnb1qpsU8PfoYnLUTuqeMU5KgcVbLW1TRfZ4aANprfDaKJO9rQDuICUfoFub6WWLAlp522TqL6pzR/K+GW5TXTaEd4BvD1PD4W/A3hDckFBoon4jhcX8zjseqhsam7WVtaJynwsM6mht+y6ugoV/KgyeYONpsBUw7QDSXs7gPe5j5CrAt3uuu5Cy1lGOUanHuEsuj1PryCL9tRdPN9EfHXV8CoqYDma6v40q9W+zqEs2iLEcY4F5t+tC+RFJIu+myx2DAiM2HWAbQKZuJaOZYQBt+fpFTVVOqg4pjpccgmWkwPGeLjIzVcQYI+Kz9MDLZfdalFEAC/JJQ4fB0bsOsAujqzu9zZLgIKIA27P0y5pizaLcLyQzTYEKGukuFjvLWTZUcHa7dPW0ftK0W4VMTLeuyUnGlnH7T914jv8nH+y2o6I7/WlaDOGeX9ej4rHMbndfsouNSaU4wp19LsL6SQ8wLXEyPiWLTlFI4kO/s0L3LcOa9tK3dsBoxovpyWO0414kcqkURoZzq6U801ewG1MjRDgq5Dc/LzYP2J4gHeR42q9W3KWkIAdd4Gxj7QdER9xwPSehg0Y1n3x2paubo+aD4tYSNmMkWutqKVSuZwY/050Nqg4yuj0B2/ZqtP98PWhSf4RwwO8gmyxIrfkJEbWIcCu+2wLrms7Ij7igOk9DRuwG5NKMfRKtai1LmAoFyh4TqjsZuM4H3GXpzCLzcQrFwHWyGtTWf47RkQI8Gj08Th4wj/ilTzB1JacxMg64gm+G/x2lbHNiPiIA6b3NPx3sFNHrCls4+FSlb0yq4BDAHbycFxkAW4HT8grreSq3CRgtSztVFqy38s4QoAflgEgeQgM9Y8Y5juYGBm/4xjwEFtyeoZpiXfwu6DhunNtRsRHHDC9p1dcijZIYOVWJ2HnXCSzaFWdGLNinHoujheJcQlVijZhp08V5NX7phEhwMdue2H2bSdMX/lHDLcUjUbGj1JgnAvvkiPrYCl68iL4e0+6s82I+IgDpvc0RMAuI3HNFsAOrZaHy1QmIS6uJJU9JqNLpcAlktrafHL/JATYUX7ht+wMv3pkpErR9YlJ9SBQrqgerJb7zctCHf6se9poR74UTetpaIBdmFiEIngBG84xLsrVGqdHi2F2QtlpcpBLYqka6hvIcVjEO1gmKWZV+yGNWDXJQPPjhQnYZUJjGdS4X68SAuz4e5utXaNQTaLzNDTApjoc57pbARcVlRcX+Spbi1LOa4CBL2HXS/FarRG9pRFgOTOTHTBiN0KA+cP/+ten2vz64QFuwupqtUDH4Es4PoNWiDFZs9oMgY44YHpPQwNshXUj1LLuBVxSUF5ciMJVKiuhjOeWl+c5VGVye+v8UtThX8YqSAlo74gQ4LG/ul17xgdGDA+wVAzLG+ZRH1mNdrfJ6G2VIwD/9HygdsQB03sa4jtYW11Yb6NK0TCCMT/tnB7VjRQ4GqdjAjJmdn6BCu3wruawy8g1ACBgR2F1lSAgqYitNgu6bHtZuQjHq1+caEdtd7AsSREmAIuvCxz+HoXVZkEXbC+rlMo5DgRYw0N9C04iE0bd/Gj8nQk07Dm+V87GRRyFqgyr0TsRRAhYXZ3HKgm4QIQAD+EDwB0SGDE8wEo1F3t9BPpR0dhCAZVNk+OiH2R49GqtT1t7xAHTexpqKRrWfupMdq0ZNWS1rMPixmRydD+bAFvC4Wp1lZnnzuZwoCaPpyEAs3FhhSTgFo8Q4PO3PDfl1kuBEcMtRbs+f4T4wVBDLY9ywGUzqDTORZ8bRbjU5zUUccD0nob6BKPRc/X8sgZqHCUlDo3GQSjD3Flk8pTWZJQWMyWiYhzn2xFgjQQ7U1LuP5o2UqXohqTTbWd0hAt41x0S8tyItUyAd2lrcZn60mCNAr6hW7UjX4qm9TRUwHZ1tTCZyy9zqXh8Ddp6Q6uH1ST0h0al2wQcqmpYK1ZLsos4eWXN8HGvJQC7dcnZ5ZlN/ilFfN8k9ecrPqNKmWECTu3H8X7xtKi4mqWQrOVangAX+1SPo7pvUqurobdk6Xj5RWXlDmU1fCLtXIWwEJerIK6zOcw6E/DYmiV4ra6KeXHj9mS9urZW7SE7/AsK87Ma/dKJ/L5Jbg+oXk+GhwlY2aYWAsWl4ymkajBpv1Gt94kX1X2TWl0NHbBageWyGlALJQaMYryWIcc5HtCUU15+QefGBBf5uFyD1Z0/lHI2A9jQe5cAXLnr09VRABwoRSfgB7+oaBu17/CVbRBNatnMukalzbFxjkrJGjd2dPL48X3Hjj/ef+LEjwj9upFjRot0I65PdhAfHPS1pv/kp0uIrw5GQsIBh8Mk6jd+fHWwdgS1QTRy1VxUtCPkDaLVfEmVxKASSaWVDiMm5Z+RiBocjsbU4uIkg0YgrThfXSDhSn/cc/z3w/WEfRa1Q5m2asWG/SebfK8RuR3AD3hP5CveQCsd/DB//kYtKWqVlk7U2uLbJgw7rS3uw9Se+ai4X0LCcebwZyaL64f1PaMV3pqQUASVNOLCgkZ0VDZp4CH/+qrc+zlanKHVDmrUFs9Uada9gdJSqbV7PlHx3zujJT6a0dfimdrkScTX+mfrkZKGP43eDjrRKL1/DgTc4inlqnT+/MV1lLJK046rbZJX1CmU6uaCimKFVouLROzyGnTByhNHxUr26cwza06fOJrGYX2y9qUtZ0rrCkVKjbKR8c3r015Ysvhwic9FNPQXVNKGtuOq2hu8syPA97WeiqhVyDvNoqkBG4/OBelLiXUZGkxg4xpiuxLlLFLJJMIVGDwaK2oxNC/ogZJti8lX4BAr4M7R2d13U8ntPdq60LoJfUWrKw8iviaTi+dZxbdPWNx2InmoIzp8PPW6Gv6IDpOViKET43VEedLtBvy0s3vOpB3Nzykpq2alHrmQd0Igl2kcGt2pb18f/uwTG8/t8Ukuclm01214waYvyNNgAKMBG6/M4FGAoWxZSzBRUiz0UrLU3FQnrZNr7W42/sUa0icKMLifQFY65iGeD2D4lcslN7eAX38jF8+zaozgy7araoUNuNXVcAG71SX5NRqX1ekzG8V4npm3r4J5PDONbwFY0eYd2XtZmXy1Q6PcsXDk6GlPffr1EZ9SZuQA76GO2IqVXwZbiqYGbLyS/SYE3H/KlHq0wqGIYOKkWDjYEiEaKCxPzU6qlLGdlsJFC8mqk/cJ/hf4eToy7Nwc3yfY9OPkYnJzC/j1NLl4Hipklf2Hxg4669oH7PXUx9VwARtErFI2u1zIbqrDRehXcmubs85dSDxVwj59/HKtG+guHv9pW9qljEs6h0Z9acOEmfPnfrz+BxcwowEgSHr2oDtqwMYrYOxv1BNsGlWgojhRLIxNBgRAmpazv0YsMjl42KO5TrTvBQV46/tIywbsB2erXF7Aajc7cW7yC8Rl4FdiOwx3kxVWYHeuorGDzrqIDLrTSrLKWWVCXKHS4MQUNK1Emnos80KtVV/CFwvMpkahjJ+ZVsDMdTg0+sw/3nrtxRmfrUgnBoCQl+zxgNGAjVfA+UEkYNfM024SsJFioeeJMLPBCpqENb/syrikt9Xi55+ZNDGTBNx/0sTFROKscQlj8qXTvHthJI99Zkx+6mz0F2J/DLR4XsNQ6/kRY2e0HX0aRcA2TjWjKl+Ky/mnT+WgjEmNl32/JaNYq8xLSsa4tdya40fXfHwoOz8fVpM8FS++/PyHb6w7mO0ySHFpkRHZ3eMBowEbrwAPBfjYjRPGbyCYMCgWGI7XFkv4RkNp8vZfd3NUdqVI6DOkgagHG9Ual3//sQmoRC39FC3S04bNNpRiBoPGYsP4JYUnLh3PhfY2VR48feZ0zeXTP6Ulbkg+sTt5x5frty66wFVBwK7UD96a+u4X+zIL3Va+NKOyDl20ZwOmCw2Y2QABV/JgAcSFn2SknKjM49doLa3tWwRgSyWX32DS8urYLSVkEzHmJTDpHgbYViEkKgjAY9eVJaWeKrYBTW3ZluzcVGY5Y9PpM8t/+mzboZ8+2PHFinN1LgjY/s3sEY+NXvHVGfgLGWWVUtQy0vsBWzBenRiXq5U1ey4fu5gmVMjVptYWagJwQ2Zpxqm6imafJaR6xcB3Uw25pL+7uVReuu/4hXKXh4PjrEuXTxWXslbvWfLeonmzt2///vW3VxedV1odGse8Z/51z6S5J4jGO0stWX3s9YCBx+lWcTALW1xzoaxcwWWeY+RJRcUNlF8IcH12OSOXmVvkU3noFYCdJXwJcrYxozA1PY3FMwIPzK+qa2tdjaWJjMStn7/+xqr331+1ZmMRi6nAbBp89MDbb3vohcPkJXQYhjKs3g8YiQeNU1NUNdt1Fy78cYmVfSiFh5F/QYB1HHZuTsqJMnZrer0CsFnIUaJigrCsnJmZXYhcMrPLSiTlZo9FxWX9tv71qa9+uHH32u2XSrgCjlmTOnTgLdc/M+ELv1FZvRmwzeajoKlCvd7y3Xs2peX/9OvuQh51H9uBRy8qlB87ymRW9i7ATk6tAJYDrVhGasnvWYWlPDQOz9MskQlQYiZpwZEZ/124aNWqr7/5eddRZq5N88eIof1vGPrCogqbpfVl1IsBq2sFPotyuszQKdXJtXu3Jl7Y9u2OFflkMASsFTDSysqrspJ9Okp7A2CLsFZY09BchkuTCjKr05ki5Lg7PyuzUgfMooIm0/nNyz5e8PWczza+vmzprxyj5o95T/Yf8OyMn2sqRRDXnXwAACAASURBVDUqb7d/7wXs4PlOywJOeOrMO3148+7kxE/XbNyT1fIEs3klJbxcfvr5qlanegNgF5vHPSeXXpLiAoNbmnqhDj2WBhHGynA1/br7UlXO8VfnvPT+otn/nf/iu2+tzDVrmEsmDHx4yAclOTIRo7yWKnH0XsBoeg7f7rYTlVmPurpYaWz4Zd+iJavL87/ZuHpHlkpj8hCA1ZxsJruikCuRtk6N7w2AgV0u4eCK3AoR5jLnZxZdQKVj1EGcVbl+/ZaNWzd/8O+3nv/vawtenjTz7ZeWFZs1wrefHPDE2BH7LqUcOVDiHdbSawF7dOISrsaG1aKx/kAvFiVllRXl/bj8251p5/e/MfPLbw8xK/UEYAfv5wO/5ZTguBhvWdazVwAGRmNhVdGFqkILUBy4lJiC7k/z2eT8HP4v32z4auvapa++NG7olBHTn3305U/3lJk1xe8P69v35gGf73hx2Sfn5VQp09JbAevEuFAF1ApYBYblkCKs+Hhi5vHzny3dvX7nz198uvSND74uyq5CgD2Kk9tXrkxMrij5g9OyrGcvAdwg4CSxcYkRVP9yMpGFXquYTFhdJNj+4aqvfz2x6q2Zz78w9l+DH376xVfXp5k1h6fd97err7n+noT/bt60J7sRXcmOVTfQXrHnAyb7zyBghRpguPjssk/2p+w4u3Xpii92rXvrpadffn1temY9Alx/evXCMVOXbirnZbEU3pd2LwGswHEmGxeaQSHz3EERKkTDJ/pSedr3Kz48fe70dx99Me3pIYPvH/TQYzNPp2rEuwbdcM3VV197/Z0Tn190RiFDv5NaLhG12VgrZDuiAlhP9J/Z2AK2zVMjFJzcsmrZuoOZqT+t2vnNK68kPD3nwy++3XVE4tSZKg7+MPGhSc+tqpFl5ki9y3r2dMAOs9Nh8xgbROwLaTUaj6fs4tFTVbDO5Mk8cfFg9sHjp/btPfPj9vXjpj46+KF/XH/9rfctOGpkfzkA8r366r/dPuKpd3LJhko1LpHQrt3R8wF7DGrUTeiyuoAz//LOuSu+X3b86PG0Mz/+tmD8O3NnzXj5tc/Wbsiu11kqVn49bui05944K+SdrfHezT0csJlbV1Rdi+lNsioeehI9TNbRY1idGTgqk9Mysy9s2bPyxKUfv/h63uwXn7jr+muu+dvtgzdaKmf9DfG9+qb+zy5Ycb6aeIJNvNpq2r0Oej5gHzFIsk58u3Lxq6fSUnPTdix586WRwweNXThn4vrl+SU6u+Td6U8Mmfva5vJarU3FxchRlk1+Qy+90mMAq3EFk4PXNZnRKG818KhObPp2PyawABfz+23r0wu2bjux6cNJL72WcM/Up26++a9X/+XaR/8tKhxK8P3rbbcMX/BZ0lk5kVnZNPQ7t/RwwHoMaxm/7zE3CvMv7frms/mffPbh5tfmTZswbNgDjw0ceP/Q2RtSmnX28g+HPXzfjA/3/FpsxNPYYuKXNlfiYprtsnoOYIUig4sAW3jSWvgiqj136cj+XLXHrUxNvvxjtiC9Oumzt//92MDb+989oC+Btf/MZNZ9xNlf/vHAtEU/JGaRK3ZFt5q0xUSKTmOiE52RLtSoMqkrBYIS8o9qpaymhpG5Zsdr//1i0guvv/PCR4+PmXbvvXfd9vd7h888LDWq1X88fsutAx5csGb9yaL0tP072A1isVzEFghK2yatp7dDSxtqVFInhiAAH5RTyipanwIvoSkpzyqulKm0JnUDvIy24mLKqUKYhLwi6Ujal/uOMzJPrP9w0pDb+v79xmuv/guRL09LyX2EAPy3fg9/8Pa3J7OZSnFhE/TJQHvB4Ozwhuqok/Z3AKeXcJ9gc513Spaay7sgx0Wq6qxFH7zz8qufvzV9wb9nT33q/lv+efd9/335nB3o+J/0u+2hQUNnvvLjBW7hT/u2JNYymJKSSr8JTF7pMU8w8Dg9TrunZQkHXfmFCszhcrEVFQd/27l3++l87tH3x4+5u+/1N9x2A3zzXnPNrVMqy++++ab77+t3wz+mvPbtlxtPlRRkZhfbovsEf69QKFBW28xTKIjx+A0wACVmhsd69MM2wRP081nhEUcaSoWsRu/CpLU1MMRjKxOWHMkvKzVzyvYveX3CJ1+8tWvtJys/eHH8uGmvffX15v0FVu3Fz+eOGZTw74+/2FOacWbnr+f2VeakZ2YXZmugh/UwDTS51gSPaD6RVQhPkI8W7+XQ9Q3QDhs8KsjrKxSopOZQyKoVqBFNrZAFAXgfplAQTeEcaauHxBSmRsJDU6uHzfAENWnYFQq+AN1aKhigsttcOiEvuVBQWpDz02/7U9O4v635dOR9jz4yfvywZ0Y+eN/Qt8RFH4yfPDnhuanjF3x6YPO2AzWJpWXMGoVY4vWw5RdGHuJS9At7PYRH0kMT6aGe8BAK6WGtAL3PnArF5hABf5yUlIS8xP5ISiKG0V2GAWgpMxk8XkKAGfAEjUtsgEdiW+2spMSDbODUyxJhiNNQwT++edOObGAq+OPTpcvfPnaelb1315I33l2684/DX61YsbNRl7th4awXpr657nBW1t7DS1ZfPMZn/rZ98x9MlNgFmAZaPVQMjykI8Fl4IkLuw+N5pMGCJzxoRzM8JqFfOw8e0UIvuqTEA0mofFqSlBgE4G2/JyUx0MnxU0lJMniUwHQuo4B0eCKEP2w9PJ5DAZnwBE0MUCUlHTnqAh5XLgyogvWFpiOH9+9kyrkHvvho7Y7EYkb6LzOmTJm+aNGq9198ZtTYxcoLu5ZMmPT8a98sX31s04crd53Iyi1I/j3xxEl024hgGqko9YtHTyaK4bEG2kFsv5cB/8JHoJGHCHAOPNbAAC0KQEXSoqSjR8vQPZmU9On60ABfcSnaJC+vkBM7jdnrK1TO+pLc9LRCbuWF40nJl1OSLzMrzDr+2UPHjv3y8/7LhVVio7KoBK8o5OMGmj2jQU/Kokkxmt1OrV5PFvxdJo1B3ayQyqpzMnKLJA21xdmXT5348Ydfj/7y69bNP6WLc88f2rRrx5mLl5mXLqRWyBR8Zi7f4vLPot0Wm80Rsh3RqyY5m5UaI/LfaTapXea6giwmq6gom8VkZqSlX2ayKkw6jAFJJ6eksYr5gkYLrDV7zCartZcANhr19Ro92YHvcTRpzLIGEVYrEmD1bHZNbhb0MyUt9XLK5cvJjIyivPTkC2cvpmewmNlFORypUFHLbzS4bH6AXVabzRiyHVGsBzsMajXqCHZYmxs1TbzyvMycgjzoN4PBYKYzWYXNzSwGZJ3GZLKyKgT1FtTB5IYFSHokPQ6wwWZoajLAC7mdaC27pqamxkapuIbLl5Tyy7Lh3czKSM9IZ6alZzAY+SxWWloadJbBYuUV1qq0MoVYDEvAFj+jTVZLL3mCyb85TUarzejxOA31uEpeWZjJysvOYjGYGRkQcGoyowZnMRgQcAoT3tZcpVZHRHP39JYsSowGi6WpWW2DXCwmt6PJZNM3aeW4UCwX1NSWZWVmZhBZVXpqRgqDlclkpGWkM+CXDFZOmUipFEglzRqbUe93RbfDGbodUQDsMqGVwIDTYlJabRa93mpQNKtFVdmsnDx4JxNZdEZ6OiuDJU5PS02FN3Y6qyBL2iD3WtpLAJsdJpvN5gYO+GkxNMjqNdp6dbNQUl/LrywqhE7BJzc9DT24rPSMjMvp0NmUFJhf5Zaw6+vl9c1NepvRQLvbcE8HbLTZ0Bqr8GDSOXRandrc0NwoExcXFRXmZrBY6cjr9PQURkYJIz09OZmRwmJlFwubZU32jhzpeYBh6jqVzumw2qz6BgFfVqdrUFpkDfUcSU05zJsyWOhfCrx/UxlMZir0OCUtJR2+jwowgZhf2+hwWPQ6Wut7OGAPJGtB683Ag8ujMxh0VpOiob68tKggq5DFYKWzYK4Mb3BY4khNh28oeGTllYllTY3UCjy9B7BVC71zW7Q2Q7NIjEuVKoVFoWqQNHJhUSOdKGBkpCcz4AcLck5jpKYhp1n5FbxyTl2dB7jM9Esc9HDA8N1rJspLJjSY3aLRqe0Gg6WeU1GWk5PHRBkXvK9T01MyUpmQLyMDPsasPL5C3qjQ9TrA6Pa1G4xGtVIqEXOVeq2j2dokqyllMaFfDCYDvYNSYU7NgI8xzKtT01IZGdkFxdwKSb3Y1nuH7LjdPn+0Ge1uvdpUz8UK8nKZTOKRTWWkwdJVBnwBpzOYML9m5WGNjY0N1C5xvQewq1mttDYZtI31zbi4qUmp17ubTXoxpyyXyYIljVTkG3EfZxCvpQwEmpWXV8SplOGoQ6m3Am4jmkZtnbS4oLg0PzOTlc5gpKYmpzOIGzoVFjxgKTqvRq1t1lMseg9g4LG7XAZdk1ytNjXazE0NOr3OoG4Q8UuYDBYTPrNpqbA4CW/jVOgrg/CZlZmTw8flsgZ9LMxN8oZqmlSNdYLC8tKKQlh4zkglsmdGGnSamZqanpHGLOKqlSYr5W4vAoyuYNQ3NOm0FrPF2mzUaRs8dg2HW5YH/YQVfli0SkeM05Lh4wsLWQxY5c8v4EpxcrulGAHs1kkVzXI5l18tEtfkMzPgqwnmWPD1BF/GGTCzhlWHwjq90uzsXYPuvD+s066zGHQOt8tp0Bh0SpfHqBbDPJrJSkMVX+hdSmoKfBOlprNghgWfbFhNqqNKz7EB2GMy1IsVkka0QHYzNycDMk2Djy2DlYrubwYsdmTkVAlVTS0TgnsZYNR/SN6bria1Run2mC31NeV5uSz0KoKlDZhVQSdTMhipyTCLZrIKMZHe6HE5YyaL9hhNuKhR2dxssigbagqZKMdKgfny5TRY+khNv5zCYBVU1mlaawy9DnCLuAwmAxqUpxBUFhfmsNKZacQDnAZvaljQYqQzMzKYJRjWrLPaLFZY04gNwMBhapJpNBaroUmj5ZcWoia8dFgDTk2D/qag8jSzQqr18an3AqaUnXqFQlBems9ErbKohMVMz4A1f/hOhhk3JqrXW63NVtRWEAXA575eRW1q0IWFLKcFFi6dwKHXG5p5/GJWJjM9BdWA4XuYkcJgZBQKdL5NdhEC3OJqlwMGDqNSJeJUF7NgxQiWpS/DMhYrBRW2mMwSocRittnUFltUnmDV557a78nTrqwmketFe3QavbJe21CNmt4ZZNss9JpRxPf/rSIDuNXVrgfsNhrVTfyqirJMFnyEkyFiVKxkwKpTiVSr8xitNoPR5IrGOzj3EAALATESZTu1sXQ7O4C3s/O2ilbZbiH/ajHabNrmBgxmXrCEyUxNQe0debUq/7QMtGl02Q7g/q46FIq9newATm+PoZ2NwYlPp8MC72V1HZYLy1aooIVqhCnpWeVYvcHmclit5B7foe0ATm9HpzuA+0nKKQDegcfl48atV5GiVKrohD60vWA/FfiC4hZlkO14aRmYpLH5CpLuRLm5XcCUq4Jx4xYKw7pw5/Yom6TcigJYC4R5FWrwKKrhyeubaaMFccHOftvtoTzBoDuyaK94PMBtU5cQnWkXNIHLJUUqi251teuz6BbxuJ0OZQ3q+GcxciU2d4Cxkc+i1Z97+Fe2XjStsotuhoYlNYORQZdKZAC3utqNgAnxuE8z09IsNIZGoxS9cmXXl6LpAUd7/+AWV7sbMIilDaJ7EeAWiQOmkzjgOOBWiQOOA/aROOCIAH56aEzISPp7y1cOR9vGLpKdrS4FA9igJ2X/l/rgpe7pEJT1b14OQfnQFyEoS4d7zyyde2r36k7AQrjEd7tCUM6ZE4KyfnJVCMrrd3jPfFoUggHslSNfh6CsfDoEZfB2ZgjKJ77qXKdFNMNDscMrEyUhKG/cG4Jy+Uuh2DFFFILylj00gaEAbgjlavbiEJQBm/7NRC+NwhCUHUWd67SVMmvnOi0iwUNQNlSFYkd5ENlOi0jlNIGhAI5LL5SgAQuXrVjR5NtL3JG4Pn8FTbMNTjkUxZASDsnkMKN1m6dd5WrQgHVWkLPbt5e4I/GojzNAsMohKIaWcEgmhxmt2zztKldDyaJz9/j2MXUsyLiglYNWDDXhkEwOM1q3edo1roYA2LS4oaWXOCjjglYOWjHUhEMyOcxo3eZp17gaHOBL3xwAjpXFwd1QSLlnPMFBm9wqPcjTrnE16CfYsxGtv+LTS9y5cUErB60YWsIhmhxmtG7ztGtcDRpw7ssrVx7y7SXuUDa888EvQSuHoBhSwqGZHGa0bvO0i1yN14NjXOKAY1zigGNc4oBjXOKAY1y6E/A1AV0y6hcGDHheAcAt/QcMWBCgdBWsyq265+4Hljl64z3Xgz2NJGBlksP98YvQ7Zq2StDtV6c2A9P32lgA3IM87WbAlx4fPF4AwImHh227igg7O8zrNvUnqJT08Kg1V4Hav1G7NfbZNOTes/BXGPrYbA38tuE/w862pJA35snhGd1ocpjSgz3tXsANt/LBLyNAfX8Z2Eq47Z60Hrp9+913n6f+RCiJwIarwIlHvSbtAWV3wacAgO+/hN/Ogrq7vCmoBzcA2V2dj62KtPRgT7sX8OkpALj+akiaDYCOcPuD2S7qvqb+BJWSZsCX1lXg+CNek2BWdq0D/Dx61JDp8JsdgP9zUCmc65eQkPCAoBttDk96sKfdCzjpOcK3xBcotz+agYbHEm5Tf0JKcwAwwIzrGm/GBf9fZy15UAPOTfB+o1I4M7obzb0C6cGedncWzQV7ngL1t8nAFuj2hzOJ4c+E29SfkNJdWvAH/Ou/pzYC01ot5WhqAgDzWtymUlDekgJAQTeaHKb0YE+7uZB1ccjAcTCfOfbgM+tvBlV9YK2hpehB/QkqJY767zfQbfuKuwfcv9JBOeqa99y8pS1uUymA/GcHPhDSsMTISA/2NDJ1EhMAh6ZHOYXISI/zNDKAvxr6+IRQxtx2RwqRkR7naS9sVYhLKBIHHOMSBxzjEgcc4xIHHOMSBxzjEgcc4xIHHOMSBxzjEgcc4xIa4IRt3WRGXLpLggE89T/E4Y0JYH1q91oTddlyD/xI7LMffg5a1ocS+c9/3oH+OL1Pn/8bdim6BoYswQA+9n9oYSrL3w+1hPS8QTNdJCV95AAsvuMNADR/YlRUVMwYCz8cT382FP1x+ivckjf+you2jaFJMICtNxyEn8euMxJZ9NBvp/bdNXTTrGsfq8ke1Pc1yPrAkP+7cxn9Ql69TVzXHwXgya33AXD+L2hZsddnwQ/BLbZ/sOFxOlo1/PpdUTYxRAnqHbxgIvyY/hr5Dh56U4bHOLTfb+yZj4/KzvvHzwDsTZOy7o2Rt/Pk94DhWuP19eBzYhkoAvCqd8BHywAJGNyyJar2hSxBAc76swI0X51OAf4Ahgx9D4CCPgwAlswjVfb30NFSocraR0HKSDDpBBiJkBKAPfcwQM4dbgKwdVOfwmibGJoEBdhzzyawY4CbAozyqKE/AVDXpxmATZMAYL/88D9u/Fc3GxohyfqT6qvPwOqPLH8hSlMIcNZtLuC5A97e0/901Z/+b3O0LQxRgqsmrXoMDEd3NAH4V0B+iPtoAfhhAnAM+KgS3zegW82MmNiuOTvmHMgYwvyzDn1FgBf0ueqqq/4EX1DTX64R0e7T3ZMlOMCCPqf6oGIGLWAeGt67OkYAg9EfXasC5ms+fYL4BgFbbzhcU1OTeJ2JfAf3NgmyoePpG4mKAi1g/bUXQdU/YgXwir4D4efwvouJbxDwsVvQkr6e2w/FNOA9fYiqPi1gcPKeO0ZvjhXAKX3QQkSf9kkivkHA014nzhZOiGnAcemtEgcc4xIHHOMSBxzjEgcc4xIHHOMSBxzjEgcc4xIMYFxBilymoBM5bahCSh9Mry2jD24nNEw72m5fGyjrllHyxbIQJCTl7ky6RXxWbwkG8KIkUo4fSaKTY4l0oYm/0yqfOkEb/Ad98FHa0NDsOOQ9CWLnsxGHKeXfT4Vgz9FjISifPEwbfOIkbXB4dnwa39quHRnxv7m1XRxwx/bEAftIHHAcsI/EAccBd2pHHHCnEgfcsT1xwD4SBxwH7CNxwHHAndoRB9ypxAF3bE8csI/EAccB+0gccBxwp3bEAXcqccAd29PrAW8xkaLTmOhEZ6QLNapolY162mA1fbCWNlRPbwe9slFJnRiCASynlFW0PrVzCQ19MH2ogf530Rtog0OyQ6ujTn6MP8HtyP/oExwH3LE9ccA+EgccB+wjccBxwJ3aERHA3P5TpyzSew+VzwEwkct7OmGMlDz1UY4D9pXeA3gOAFvf9R7AB4lHPgN6N/h9GXnqI3HAvtKrALvv9h6AZugwA1Lee5Q89ZE4YF/pVYDB/TbqAMDUhVC5dMxDPPLUR+KAfaVXAXb/y3sA6fNHi5Dy+bnkqY/EAftKrwK89X3vwfFUQ+FsANPK/C956iNxwL7SewD3nzp5scF7+GEzAG8kJ4+bOElEnvooxwH7Sq8BTK/cQBMaB+wrvRzwE+vbhsYB+0ovByz8x69tQuOAfaWXAwZVNycGhsYB+0pvBwyYN2YFhMYB+0qvBwzO3FzuHxoH7Cu9HzDY/U/cYvUJ7TGAPT+tXk39NrEOuNXV7uguPJEvqvWB2mMAl20D1dTq9bEOuNXV7gBsEeI4tzW5HgP44hlgeBMeVQrFdhcpJr2LToxOulCHilbZbqEN1lppgw20oeaQ7FB6T9oFTLnqUChGNFPKansI9hiMHSmbuDJZtaMl1KamVbbaaIPVDtpgejuMJupkTzCA2Ws8uS/B4/Jx49arSFEqVXRCH9pecEjKIQV3otzcLmDKVcG4ccOEYV24Y3uEBfnSzs0M5YKdpbE9qEJW4pqDH5JnsZ5Ft7raPUN2XL7rN/WYLBpK3gHyGPOAW1z9XxqT5Vq5chtVvI91wK2u/i8B9pFYB9wqccB0EgccB9wqccBxwD4SBxwH3KkdEQGcP37cmExq2Luy34QJHNnTCc9K6ZTjgH2ltwBWPVYPTEXUsHcl2rEU1n1/XUanHAfsK70F8IHV6EgNe1fePmEx8nLXUTplh4Z42s1PXZ8OPAumztGTwQjwvDkykNFHK0NTXsKxIwjAzpIzqY1tIsYiYHpPwwW8dh86UsPenUbw5WZQOuYBHig8bAlUdgiJp93VuDgdXHgPHPqODLY73IYp8DhniNZFTHkJw45OATcvuXXwpNH9nj0bEDH2ALfnaRc8wcQ2u2X/gR8n54LckTd+VEFcvGXXUsePq8kTCHj7j6BikmP+jGkNBWM+PLX8jrfBxZ1PaQEx5SUMOzoFPOJXIr8offtH/4ixB7g9T8N+Bz/eBKyl1LB3eJmdq6CbKf+Ff+Mu6zd0u9rE5qko6xxf7SNPIOCMl8G2oUdWgYuLCqY4gHI68My0QcDklJfQ7egUsKvNCSmxB7g9T8MvRU+cPDGTGvbOGDF2hoY1buJ4csqK9ej4vuvkQg716/s+wWD1jCXTvz0LGp4rWEwATtoNiCf4/Nxw7AimkGUT0ESMPcCgHU+7qR4sTGPmZ8rJc4eQeNpJwABs2/fHapC8qOATAvC3E6bc8LwNTXkJx44gAKff9QDIfyUwYiwCpve0uxo6Ghn5LIxMEpai0dMOZt35xHrTxEmLXI5XX5iGU4ChPKVloSkv4dgRBOBh8gQAHg6MGIuA6T3tLsBWTCRnk73C0a0HPwkSABgYGDEWAdN72m1NlRoum7p6dAGPakwAl0YFRoxFwPSedl9bdMuojugCzhnUb/QtBYERYxEwvafdBNiqbx0ZHeWmStW5M0oQKLEImN7T7gFs4kn5LV96Ylt0fK3Kji/R2VqVcoFA4E04ymtVXkNIoNux+ATTe9o9T7BWissOeG2O7hPsdDrNJ9YFRoxFwPSedtOwWTVb/cJK6ksPyKLnBQbEImBC2njafaXomr9TfVfRB4wPDwyJVcBtPe3GITuvLiaP0QV8yy233PzPNpPSYxEwvaddDtiEYdRVxX3riGN0AWu1WpqiZiwCpve0qwG7ODjOpbqC33+TOEQTsJWSwIixB7g9T7sCsNvublG21+J4LeVx/fUcdIgm4Guu+V+pJrXnaRcAtmG1mMHsIpU5hRIJ5jXrC6KTN/qFrLYSe4Dbky4ArFbgnBIR5kTK5TflGgwtKWlvLgRxwP7SOwEzObgEmmgS/nOvn953UwBYwND8PgyAQSMm9B03oWbipzCY23/KMxURAqz5cvqECRMCI8YiYHpPuwCwlS04K8Hr4LXEDwQ0pBj754D9azUfTLSYngBgiBU0Pz/aQ4zVy5kWIcCzvx9ybt5ngRFjETC9p11RyHJZjWyeyg30Qz4IVMysA5wZmtlbs5kLCcA//fFVDgFYNyhCgAeDUQDMD4wYi4DpPe2iahIaJGsdM9d3Yj+gjPM8jL+at/H7gwTgqUbORwTg8y9FCPBQMFoDnguMGIuA6T0NA7CnnXrwG5N9S1MeDQcjK2VTdm+1vjKTjwDXD5gy5UE3egfProsQ4HnKnx9MeDUwYiwCpvc0ZMAOFUflpgWcZ/JVNotwHCPOvh6UB2Y8AhDgHcfgVyY5ISJypeiStDZpxiJgQOtpyIDVclyqC6It2iDBcTZh3+W/WMGH0+1q9WPWUU0AZC2MKGD6iLEImD7l0AHj558d/WxK4FQyJPNmnX56zHQ9MZkUODGJQEsEE/VgN6ZQsP0uEanOhjezaSLGImB6T0MGbCx6ONvalBU4lQxezTDF1GACmzYTk0kBcBqpaWgEYDsfxwV+DaURAqza+eR931FTF3krV54iz2IRsK+nra6GXsjau8JOFLL8p5KN/fjU8jvegH/eus07mdQr1BMsj84TDKXm9T+TV/y8ZeZjLAIGrZ76uBo6YDRxlALsM5VsKgDK6VC5frAWTSb9wWxsSY9sqoTvYP8fIlKAnedevPUt4qxy9cbvyB7M2ATc6qmPq6EDRhNHKcA+U8mWkIBNo4qQTtlcoRjzJhjdtuhP+o//ndJiLrDWfwKCWqsy4X2V16KpGQAAIABJREFUqqDfhGGJqoIZ8GTi+HfFKsbwpyeJu2tNxi5Yq9LHU6+rYa1VWTuIp1IwVKqFSejb2u17lqlOvJvygUrFn6Rseu6QSiVVqb5/C8PK5J2YGewFg0mj/bUqv2tdN6RkAwAL3cGsNiuePsrpwl501T6FPuE/1+Z3XAqtasMmGuWestqsj6deV8NbbTZ3wqRxya6Zdz6+Tj9h4sd2639mT5XlLXG5GqcZjtw4fvyOtOEJz+fIZFyTQ6dHS8dGebXZpvPnm8kzw6ceQ3CLkW4//FUOqsyVjESf3v0Z3aqt22iUe0wW3eqpj6vdNCbLjHG0HqyuDk0Cj24WnXTHzKm3niHP01curyTPOgE82cD5CHD7J/w92wsY7c/oZg/W0igHAl6QCVD32cDhRPcZQP1nRPdZR450BWBfT1td7VrAdhVb6wFmnRV4PMAmwHG+PdqAn4ElI95DgRE7Blx/O2pRhVwPr255gv8FH4uRRXTagYD3rweo++wxO2p8J/vPiO6zjhzpCsD0nnYtYEyBC02GWikP/d3BIxd+jy5gonH2scCIHQPecdADvmZCKp5hWgrw1veBa8YhWiMCAXNmAdR99iYJmOg/I7rPOnKkKwDTe9q1gNk4LtWrcVxBpGpgEwMsowv4qyNGzcY1gRE7BjxK4gFZCxHXHRsRYHKhimM3jpmwi0Y7ELBnoBl1n/1MAib6z4jus44c6QrA9J6GAditEAroh4qYtCIpx66R4VId8d1D2BddwH0o8Y/YrfXg5w6i7rNSAjDZf0Z0n1Em06bRFYDpPQ0DsCyr/FKd20lzbZPHqLMR81Z66orvXgkdMCygBgNYgloHhqDuMx0BmOw/I4pplMm0afSoMVnC8vLcUoynahuB3vhoAuZ6bRD6Rwwd8NIn9cEAPjcSgDSi+4wETPafRQBwe56GAViSW55cqoAv22CUQXQBT5qbKHcYitcOueAfMXTA7pfHmoMAXHedN/OKcD24PU/DAOyUCHnFOC7T0Sm72iYT1Sw69aV+ff7y+HfNwF/CeAfbp8xsolX2A+z5u3e5oEg3dLTjaZil6CaeCGsLyAR0XG4b7tF+BztofqVwClmGoa/TKvsXsp72LhYZhd4kOk/DrSY5LC5HGw9MTlj15QWCiy7ghqIimm3JwypFN93/DV2wP+B3vR3kEQdM72n49WAdr1YVcH0T6tavDaw1RBMwL6Hvgw/0HVsbGDG8alLZ7alabRuz/QHvepk6iTDg9jwNG7CLGzhEAymrBUKVRtZEFjQ8dmKaYTQBjzgATXDsfSowYpj14MYKmQxzBwT7A870LkUW6dVm2/G0iwF7rBZVchFThr7ZJdUcZHc0Ad9DHu4MjBgmYFstTQ7lD1h9FTWWIsKA2/P0SrJofpssGn1IigtyUZ+GsyQnn8OOcm/ScPK+HhEYMUzALjaOswMN9AVs1VvvqCJPIwy4PU+voC2appCFPupTzpxLVapwWUVheSYWZcCcZ/o++GDfUTwQIOE2VVpVqjZTrH0Am3lSfib1NcKA2/O06/uD3WkH87Gy2qzitMu5F6KdRQMgy82Vt43YTW3RahzHvT9oxEvR9J52Q4e/WoFjGLucxckrJ64f7XownXQTYLQ+mNfb2JkfHKhsrynMLKgqYEspVwnALltg4TOSgO8ODOgmwC41W+XdqyI6gO8ODOgGwG68UlLLF9d4U0SALezawJaviAAuIKVfYMTYW6uyPU+7Zq3KppKSJlLZ2FQvzM9K5pW2/hGtVSkTCHh4gE+RWKuyz32EXBXoduyNi27P0yt5gm0GqsqH1k5CLZRQWScUHb/0e2aNT4s0eoJVOC7TuvR6n4tE5Am+myx2DAiMGHuA2/P0CgBbudJa0i4HNfEIKrNx3pGs/HS1BcO83YlkFi0o01ZLpT7NQBEBvCSXOHwcGDH2ALfn6RUA1ii8dQKPSixB6Exodik38zKr2IMpcL4VEMM+yEJWA593Ue7bDNSbStH2f08au5v7IgJMbNNH7MXnmj95TDUAEz+xAWKuJdg46Xl1a5zeX4rWS3A5FdttNFDrZDlUFQV1fL3pAotbZ1Dx0CR/cvIZF5elCnFOy35ovQrwkRXQXQIwuSknsRdf/ssg6zXQ/PwoKyDmWkrHg8QvW+P0fsBunzqBr7LDqMLOFRZmlakluFBKAfawcby6DPNJLkKAr0Jy8/Ni/4ghAk4aLwCAAHxgtfeKRxX/BueWgp/+WMEExEyto+uAzmdPjIgDpvf0iqpJgZdHyh6bw8VVpF8oLsI4gqIjewUesqHDxOb6N3hECPCWrTrdD18fmuQfMUTA7g2P3ZdOACY35ST24nO9PuxeMZhqrHkfEIB37wKeR1vjRBwwvaddXQ92Y7U8JZ+7e8vpNYcO/Lz/dC5DR7VkeQJaOiIEeDT6eBw84R8x9EJWxUD/J/j83MQFoOL5+gFTJj8APVt0Vr1/LdA92xoh4oDpPb0SwO7A1ikTcBrrcBxrvHT8wqk1l1K3/ZJd3gI4UCIE+GEZAJKHwFD/iMEDtqvVFoDbQfMg8h1MbNNH7MV3chmQPbvjGHB9BfPo9/7Ac8aCpOWtaUQcML2nVwDYjgnY/jUEvYpXwsNxti1r3+mTq5JPLz+45qTA7QXs0asMPgZHCPCx216YfdsJ01f+EYMGDKsDCo4zY/TkZ5K5t42Z8BW5KSexF5997tRRrFFNwMVYCGYNGPQlvmrSdJ9tbSIOmN7TKwCsVlBTVFqkXorLBHxOkzGH8cfxs3uWbsrMyrG3dDboxLjIx49IlaLrE5Pq20QMGjCaYCWk2nM6qgcHOAeiUYqm9fSKAEslrYBNaq0bl+O40iji1J0TiET15777fOfFMgcB2KVrMPj2pYEIVpMMND9e0ICNla1ri3QEOCB7AlGpJtF5eiVZdF7qxZaRwhY+LlVr2HU8JZuRU1tdwS4t+uXbj9Ycxqmmyqqcy8KoPMH84X/961NtNtYNFrD60T80LWuL9OiWrPY8vQLANp5Yzm3pHJPCd6/Jabar5cxytkBRVb15zYJFW4tMBGBPeV5JcaFGrY/8O3jsr27XnvGBEYMEbBk138fiHg6Y3tOwATut6lKujOe0GNWY2g4stfAJRso6qTCjoLFMwf7htXnTDhS6ScDsfSePJVX7IY3YarMg7O1lXbOn+7bk9HDA9J6GC9jEZZ+vzCrS6AS1lyQKDAapNS6k7FJxlI21yWX735//8be7cjEzBOxRFZ3c8nux3K9IFiHAQ/gAcIcERgwKsOetp/wu28MB03saLmAM5+dzxZUeNi7I5eFct4+yx8XFazevXjxr/vdfZ/I5LofGwWOXnc9i49EAfP6W56bceikwYlCAP3vUv5LQwwHTexo+4DomT64CmEKaxpOq/JTdHFx4asved99Zv+T47xU2h8bFEWZfOolx/dq/IlWKbkg63XZGRzCAN98VMIatc8C+PCJfiqb1NOwsupxT3KRxAhvGqVdJa7QeX2UDp4qxe887X3/y7fmzGaihQ5t7/FJNnkTndltbXmqR3zdJ/fmKz6hSZjCA3+IGhHYKWMfGWu/hqO6b1OpqmIANGK+G6tD3AI0UF+v9lO0lxclb1694/+OTl3FUyMLk7Bx2DiZpxoRcr1Lk901ye0D1ejL8Cjr8PXpY4/cNbgFs4+N4y5ZC0d03qdXVcKePimGtyPuN7V11xavs4Z9NS9z84efzXl/OdiHAbJxbUFbMxgul3lXCo9QfXHSCPPoB5vafOmWRHqQvBODYN0TffgVa14roxad69QkhARtFuNSvdb0FMFoDnddivi9gLRvzKkWsP5hyNVzAda2kgF6kqDX5KjceTfrljVmj5z7/zn6BBwGWXGD8wrgglkmiAviA90S+4g2U7f4wf/5GLSlqFfwonqnVrntDe+ZNrXb/sl+XaLWywpna5Ekq/ntntGkvaC+9QilrkbJWzOFyC7U+otF4EyuoqsgraVHWtGjUV3A5BV4lLZ1oVLTB7SgrvX8OBNziKeWqdP78YXVt7en8EiqlqLQIb7mgQkyeqxsKCqVarWD/2m2vvzFp4mOjZq+7VAftqc/4bePO7HShuKmgtLDBG4v+gkra0HbsUHuDd3YE+L7WUxHaEbUiPX2rjRSDFn5UvWCzWe6yXX7bZvt95fGxbJutZLat6VGb7aPLtrqXbYmfUMpWFfpUVdfxG20+YjF6z8wq9al+NdQXjalFQ1UrFldYyXO9jU6MOtpgvZUu1Kr0XjkQsI+nhKuG9PThDV4jLPSXoA3V6a1Gc9vgxswKPltnyrx0eNOHi14dNe7FZTvKqsxmteLQN+99cr4a/hBWQ8tlTDQJhGqH3kCd/BQMYJhlNH1Bnvpn0eRShWQWTfTtV5DrWi2GWTTRq08KkUXbsDKBxi/n9K8mbbiDWhPUJ4t2sKUib67erVl0C+BWV7uyw99ZnJbxG2Yynzhz5Pjpb/4zcebc936qrPE4NNLlixYs3H5W7Wdu5LLoPdQRW7HyS7pSNLVUYe4bABzcAFDffgW5rhUETPTqU8oEYLXCuwqYVwLqwd7as+872KE3eX3vVsBeT31c7SLALpPFAyyV588fOqLXl54/dK7g3OdzXkmY++UpFXwHC7968Ymx69h8k2/ps+cMuqOWKlQ+6QTvJhN9+xXkskcQMNGrTykTgFU4LvcfwdAC2KEUlKicnjfJ9q/eP+iOEofNA1yYWKgBzsLsy7+xKtRlGRkpMtHOV2aO/WAXwwIBS1cMevqh+b+ey+L4rNbagwCTSxWC38ZM/BgQffsk4Fl3PrGe6NWnlAnAJm4d2x9dC2CMU5RaLgeuF4gW7FgBrOPVYm4TLFRz3ECVd7aQVdeQy2Llc2s2vDNz2px3j5sgYPUnjz362OCFq8vx2tbZtT0HcFvpqB7sMPsPJaX6g40aA4eTnZRTrPGYn0GvvxgB7OHguMhsERBVfJOzkCNlF6ZniLDi3V+9NH30i8sTq9wOjerjxx/855B33zxZVdbauttbAaNYao0PZAKwQagQSPkXTqSeZjW4dKgyRgJ2BLrT6wCzEWCg4aBGuiasnJ/PkXBYhaILP897+v6HJn97Itvp0Mh3T737tpHjX1zyc3KNUUX9OL0KsBuVMbyAHVxcjrX+jQCMhqs06/jlmaeZDPL3h4DdalFNQYnGr92rpwO2EnOOIGA3dQ8buMRSHW74316O886zLx79+YLw1HdvPvHgfTc/NHVzOnyCpa+MuL/fE3MXrP35cnqhQkb+Tr0JsBurE6pbAKMXErfVTAKwVopL1FaVaNPQ/2SPIpZ6L+43buj7Hz4y7F8vFM/xWfK9hwP2YDjOt6HOBjaPqvU4qXeNR1eWJ03+I2Pjyj0rGT8sm3fv/f379Ru35pzRoRFNGfD324dN2rx795kTWTycQ0TsTYA15TwZ12VrJL85uAriCXaZ9RY3CdipkVTVY+Wp0n2DH1iTrEYLRRe/oNmXdPCXlFnbj840F85wUku+93DAaPGkOhPRXYhyZj9NoexidU5axq5tq744/cHqlRNvv63ffaOXnscdmoIHH7jtptETFv+SwSrNZstUTrunVwF2lmcXY5iKW6B3EkZbiNeMubQitRxzIcAeTCrFmnF2Ce/jD5+YverzwQTgpjW/frvz6O1JiRMrs6apqCXfezhgoBJLoU9GtLa72Cd1h8Wlk+HcQklF2vovFq1euvSlaf8Z+dAdjzx/OLPOoWGN+ee11w+asvTzoyVSbp1Gy4UF794E2FzHzbzYJJRjRRyeBu2ci/oO9OVZB4pZ8EdwGdQNWB3zci7Oy6t8/MDsIZ8vHYgA3zbu8ZWbEo8/m3ogAcuazjn3UgeO9BzAbqPeid7BOr6o0tBSlDRxhZiBJ69qsKnL0vZ+9/HWTQteGTF42MhHEt7ObXZoKof3v/Wme59+Z36JDNO6AUbcHL0IsJ2Hy2rMInnV6QxmlQ31GikwgAkLjmWz6ozw+ZZzj5wpziyr5qdsu2ngjTeufvtR4glWsQtyOTPf2b99UFHWrePJJd+dTW2WYULScwCTAgtZNjUm4lipnjCITKbR1aXx2DaDkHXpveXfzpn83NDb+90zbOFvZocmd8RNNz344PCZH7CIho5eBxiY1Gq7R8XPPVlenmwid84FmIJ98USlyq0vz63NTvv5jFCgb65e+EXiZ4Of+4N8Byt2b/l0zz/Wfv/1M0U5M0i37Gw2v+062z0SMCD23jDipVJ4N6sUuLCw6nhapkitYeSmfvbOxPHTnnrkxuv7PzRkE3yCD91/W98bHpk4bWEqhGrRNnIEMJsPBOzQap2h2hHRerDHLC3JyS21AgdbItQCC8ZrcjiApVaacjHzwLmkirM1/PR7P1675ssZlYMIwIKN65aMfPS1BZ8/W11ILfmukwpwjCbtngfYY2+W4/IGXm0BBkvFVjY/T8pmXi7DGs7nXjr4+6p5H43rd+ONV//tultWHtZrVlx39dVX3zX+368VmIGFJxepbG0LWS5MJkPDIXouYPjDYkIBWsnAaUAKHuIN1VRUUZ51tCDr0OELfFby968u/OJwrhSNhdDZC9579bmp016dteECq5xy1iAWKOgG/vQgwKjpGRj1WG1VGQ/TSKQshhCWO9wOlYKdf4KVI2Kyjn6275d3xk//xw3XXn3NNf96/Vup/FnI9+r/e2DIl8dZjiaemBwLEgDYirbRcvRswCqXXkeVO4jtR6AYCg4fP19wKSvt3JaT6eeO/LBp4/KlvxFNHTr1kXnPPPGvhLc/+/4MX2rwJlGG0Q3C7DmAyVWijQ1SHFc6PTaeQiQhe8JsbM65AmlVzZ6Nvy1f+u5rMyf98xpI9Zrrpn+C855EgK++deD8JbvyC7KzucQ1AwA7ObiC7e7hgNEnuWmBpk7GQQ0ApeePndlwfPPKbxa+t2zvyl0fzH9t3sfbLzfC+LqGsk2j7rnrvrmbfmbgQoP31qC/YI8B7EZNzxYIWIbjyF2LWuuN57I61XhlxqkFG1a9vX3tlGE3X00+tytMYuIJvvq6u6cu3FdRXXywlChIBr6DLeQ6nz0dsJbctABtAdZkdrsKsi/s3XJg3epZHyz4aNs3y+aMmjrx5Xm/cmCmpmsUrB/4xPAhi9YVlGL1pRJqhaEeXk1CgIUQsA4TcCyBurrMlMOMo98teevDbe+/ePdfCKp9H/jOJH+VOP3LnZMXXq7KSMmroXuCvdJzATsdELC+EBPy4LOIKfCKqjrMws64eOTYto1LP1j06rx3X345YfLYpx4f+fqpXBvQWXgrH7x9wJNLmTxxPTcdU5Cra/dwwEBPrBJtNLutSN+v68yVm33k5Lb1W/+/vSuBbuo603MOadM003SmaZNpOCecUuiwNAMToFMSCG0pTSlNQknTKckkbZOQNklLckgoWSiEJCbQHjJlcliCAzGnBANhsUFGy9tky7It77ItL7IWW7b25Wl52mW9+a8WkONn0HMsI/noP8e+V/f9uu//3nfff/97dd+72156a9PTK+bemrxtv70/YFyXIPvmr659rOIEfvrjzsQblQuM4Lizt2uAHWmVtVZ3R9DbhLoaYLTn7dJePnP4qZee+/OG9Y+uWzNvwerv3rNgzsPP7+/2ud2fPDbnm7Pm/VZULRT2UbUDPdGwi47mO8HJqefUr0mBbm3GM/7+lurjf9v7UdmWD3a9+NLKu5J37T9//0DYuD6Rv+nO3+5UXFJj5VUKdPPfUILfTy0T99JBLvEExhTRXTqt3O9VtpJCXbKkRafrdjLasmNvvnf00JEdO7esXbFg/tw75sz+13t/ta+3xaXf/IM5t972zQf2VTVhbd3nFA6mVdPfwn3C7O0ACdhTGX82BJtSyg4/5ync3KWeRGLS9bdILWCEtcXo0neRJ068/2nZtsc3rF+5fPZXkwTftnSPt2thkuBvbd7TRKkVlU1aI3zZx3BWPRE7gsFrLrrjkP2pjaXH2QGcGbvztk9tGKwPR5Xa3qbUztbeji5bODJUVlX6qrym9m87X3lm3cxvL/j6zG/8y3c3PKe30sY/LLzjK7fdteQPZ3Sd8g57GFVh6KY5TzgZO4CPQ/B1dgD3cZZ6k8XGjlpZp9odszY0tx0l5McOlDYrSw5uWbFw6aLbb73pC1+46eZbvvRv77tOf/mWr3wR4q373z1vctvknYZBdFJ+O4Bz25HNDuCcwt9Fj3RperTgwNxXt12JRwN2m+3CmbLd5y/j7RffevvR5d9btmTpT2bft/5PRNBt+fXDjyxeec/SFw42WsIhxmsL9AwburgpySMXnZDUW3baiLJjA0O0v5U6gx+kGs6QWFXLiY9+v3j+f963cMmCf58/b97CH81+zVUyZ8kDa2betvzFQ73MCBvt6kv49Rvqol8WCBJvJ+j5VCCoRgU4FKBnzYYhxdCFrYYMmj22QnoZadSdP1/eBykNBQIUejQLKsobvBb5hR2lZeU1yh78ned+9sPHNv/mmS2rnnzibYdbvf+NZ//noe///F2R5Wzl6XId7XXJys6elaHKxFAHep/GIKToPWrBKsgMQMYMqQhp1EKmH+xwoNMhnhWQdkPqEVw6KUD9ZJvgUhYELzslENSgzPmLAoER0iGoB0cFUsjo4cJaIBWiAjlk0LY8ToHg7DlEg7z0+IGjut7A0KGdO14tK6/9R9mR3aWn3n1y9cpFix/c+PTvNqxds2b1Iy/T+zY9sfKRn9276e0P5NJj5y/JwjBeEAkqKtCjcANQKYVqJyCDNmtRgR1iVCCDAg2kdoQQEdwAaWItDCpAraMV7ECPlQQEglf38CP4Ha028WunpUOrTTxDOAgF6C7yQTqACB6CDBoP+SFN/Exg1PYnFhwFoUCLRjwWTU+L3muLDUmE4lqNvLG18sjupzbt3bPr3e27dpw00C0VRz/Y9dbTu04KBlobKNJI+9wmVVfHMKpMD3UgcjyQonXUwW7IoElqBlI90hiGDA12BNDpUIEZUjROC2v7m7Xo8lu1/dncwa0phO3q5Cm9qVMmEHrgwmYg1CaWbwPCzsQzVsPiOjnm8LMOYdWHJxtqyxsEF6tqG2rKXn954+Yd+/fvfmXrm1t3lZRZ8YP/99qfn391/weldTJhTfsQcm4adW8fk0JoyLzCBnUmQkRfAiHomiBFVziUvsJWsAN5q6hWu5cnwZ9vTVZS4ozfYLf74kbNQGd3f89gRwMlrroorDx/8fyFyhq9XYYLRRVnTh4rpzp0AwadjvHY3C5Pnk90pMTrjaIrjPYgiQUDUYtR2ag3uJzablW3qrHlYuknlyTCqgvlx/7+ieBydX+d6PKnR05eqLhEUIqOYRuazPWH/MHCGiaxYwiOBkN+M22jNWpNW4e6v7dRTlEESRIETuCkVDNEYRiOExhF1iqUg+ZBG/TDNO1PUzISjWWcJN8Ipp3B1CL2Ea+bNrvdANTkcPao9Z0t7Q3V1QCUEmMYSRHwT1lLSHCJBJNQFCVtVlvgi+hpHqbQCY75A16X02nTKTsUKl2Lol5GAb0UJsYIAiO6NIREjAlxjJBQzZ3Kfo2bjYTgZkg/JO8LBjIuQJ4RHLfSoUAUvawhHnV4XCZf0O52mN1WpV6nbquXEtCISQRPjAsBrEIGTVmCAVSCouTdepczGA6GAv6CJ9hqHXI7rE6zeqBNqahXSKspEigmcREhkVCNnYQIxwE5tHXwW8ND3lg84ovEU0BgEBHKwJRnBLNAsH+EHWH8THDQYnf6Q0GPOxJz9GuU7b3yGrhTKYygAJ4Eh38UatQSSkLgIrK6RqU3mYb8EV+0wGay2DEER9yGIYNt0KQdGGxWyevqSKCXoiQSEifEEinWKBZDq5aQIpysVeoHhrxM5lx0zJ/XBHs8vijqhIJBmqFNriCTmLEMGjUGXbtUBkiRk8JEIqBYAo0aiJagexgcmLyt36AbNEcKbqqSzSA42TuFtaZhNW20W/r7+1R1MjkppTASGjKGicQYKa3GxdADi3HoqBo6ejU6g2/Ujw1RJvNJrXwjODkOjvqgCwqF/NFYwtR4xKE3DjfIZRSFQ+crAqgEtGAMgg0MUUzAzVzT3KJV6zRWb7xwCY75mMROzNGBocEBt8vrtug76uvl0hpKikNHhLBCa8YIEvwzeGiClDZ1KXv1Rteo95iMlvwkOOahXS5/IPVkJIQNDs+QvlVRQ6FQA/pgCLPEEgnqkIW4WAR+GpfKm9oMAxojepqyYAlmQqFAwkiLzWbwe31uo7Gzvam+XgaOi0LRBoZ8MwZdMQZ55LbqGpvbB4yBgiN4hAmGmCvRPowaHGb7oEpRVyOngFRSgpEEeCghgcgm0W1MkvI6tdNN2z0BJlbQBCe6pJFAIBSJxMK0caC7vbUZxgg4CTcusEvAnxhFleDHSEzWVK9q7rAwvti4QPKTYDbCZPxcGAsErS67RaNuqa+mcBKHm5aAexi/DCEHJobWjIPnlrf2+71M2AsRBoyDRzjsz3+CY34/k3k85rLq+zu7VbVSAo2DUQ+MHDX4LIKSQCRNyBoaevpUFjoxMCwogke9pj/uNw8NG/UDXQoFRQJCNAQGkCQuForQOElEUFKyWaXz2P0MEBwIhCEIH3vS/CeYHcmcomDjIVvAa3doB3pqwUUDchKaNQQcJIorMYhGCJmiUantNVkTW5gUFsEZEvN5nRaL1Wzv7WgCXwU9L0aiGBr6XgQUBRxSWY1K4wmHQkF/MMgEg4FRI4WJ2JEPu48G7GavN+B26VXyRMNGw2BwXxggl0hEYkJCyNq7evTo2sTHBVIABEdCjMfksDkcVlVTHQwIMTTKF+M4QARfjZo1Kalu67CEoK8OhqORkfD0IDjqoB32UDDksRuUtSQhxRKjJBgOA3IcAk0YHMvb1Xob43KExwdSAARHgwGn2Wh02Ae1rfUy8FWECPVAEGAlQg4M9ce1XT02d5RJdmFh6MF9BemiR0nMTjsc0OW/dOvWAAAGjElEQVQE3cY+mVyKZmcJGD+IgFshhNJiCSlt7dDbPD7GVwiL7lLCQXAcbSjqdntps3mwTwneipTA/SsGhIBWLJIQYpKkOnpdvkh6niDCxqc4yKrc8ZfUpgaT6aKt5lAkGPJHfD1NDfU1MPonIeIA8OCgJZiIpBp7dGb3SDwVUU4RwVegTt5rlOLxOBO1uxhv0GU1tTdCW4ZASwxx1mUxasmXCYJqbNc4rvzGMPXDJMfWuHp3MjuJBMcjAfjzRdm4bVClbKqDwQJWBWElIYQ+GIYOSp3RkQF1agi+CnWSN8aKJSbg4qFhXVezlBJBnIG8FYGJSZyqa1H2ub2RtM1TT3DtcZb9fTI7iQRnvKvSadHpdYpakgIfLUazHCKSaDO7vZlvbZkagq9CzcEO4EhCDrOjV46hHw1hBCxCs5VUh2oo88UzU0+w6FOW3QTpqZKSv/qSQrt8XEJzlnod3MXuVMbjMtFu57BKISXQDw0EgKc6hqx26/WrdvOyw54+4bgEp6BaSkqWGVLKDi+PU7i4i6+Woq54SFVPocl2CZkYD/a76FGn8Hg46+BlB50uHm8H8FGSbtZYWdm+QFI8dIBLPH6uUr+DU5nxjvropfWDMiIxXymp01ghpBx12M1Zh5eXHfb0ma93BzvLypYZU8oOhquuceyhuYszS/2Mx6nRKHEKomihpKp+8LO2+nycdTg4MY1jh9uTyhzIhmDn1njf539f9BgZ+0r/kaipEsfEQtPoKZGETI2Lvgo1Ry46KRAlh4cxoUQi9Yz1xzdgLrpy+/bJj6LH2bOBloo5tacqik5DzSnBSSEFFq7igv2x4bOSL5tycMsUEHxjdx/lliLB17anSHCGFAkuEpwhRYKnhODny5JS+mEZl5R+zFX68UFO5WMfcRYfPspZfISz9CNedhxInzkbgtNGHzzGw54jpTyUjx7iLua+ABOz4wWeBO8tScqzvyzJXnbcz0O5ZN1mHsqb+Nixc3k69/eR6yI9lNZd9TqPU2z8DQ/lLT/hoVzyg9d4KD/+VDqn4EdwWj7ZwUPZvpyHMvuslIfy6Tevr3NFXMv42JGWHw/wUN57hIdy62N87HhQy0N532GOQj4Ei7gqGE/o3/FQZt9p4aEsOcBD2fMkHzvS8vzY3cHHl+MXeCj3buNjxx+HeSj/4yxHIR+Ci1KAUiR4mkvWBGu2vfGGNXMZwLUktvW/0WPT2SnzUeRVMS+TJ/i1nCGdLKhZE+wOsrIDmcsAriVx5ymczVaZhyK/inmZPMGv5QzpZEHl46JrD2cuA7i2IOOyVs5akW/FvEye4NdyhnRyoPIgmHnJfGUZQFbGZa2ctSLfinmZPMGv5Qzp5EDNjuCqnR+zke2N2TUopJwfd3DWJl+VPEI6OVCzvoPjezF21DKA6xuXtXLWivwq5mnyBL+WM6STAzVrgmt/tX378cxlANeUPZte+DBrZR6KvCrmZ/IEv5YzpJMEtTgOnuZSJHiaS5HgaS5Fgqe5FAme5pJLgm/+zLZWzl/MnPnzYZa9/c6ZM5/5jNIMGMr95Vuz5m6LFGKby2OkU0mw/Vxk5E8bAHbnWCWA/cRaG8vspqcDwXmENMcEVy2+50f9LHt63tL/nZEoq1iahp06BErn5q14ewar/pItZdJfF82ugKuw5D/Wu+DTnseXVlypQf7AvcuIHJo8QcljpLkl2Pz1PvbD77GmOw3s+wnYI2veA9h3zZp1MXUooaRl98xgTy9Mm3SYbbkb7gKW3f06fKpgdXena3DeY2YNd3MvzLyRksdIc0vw+QdZNvZF77n1LOtOwH5hfSzVrlOHQOncQ9BpzWBPLUibBK7slgh7aOWKRevgU5hlvxxJ1VD5jVWrVs3tz6HNE5M8Rppbgs/9NIHt7C9SsP/4EFr/nICdOoSUHmVZLzium9OOC/5uDTZ9x8VWrk5/StVwYWUOzf0cksdIc+2ie9jD/8Wa7jCw+wD2iw8n1rcnYKcOIaW7afYEHP31WgvLvEungIpXsezGK7BTNdhvF7FsfQ5NnqDkMdIcB1mCRfN/CH6m/Dv3vfc1VvlPMGq4EnqkDoHS2RVP7QTY4TdmzZyzPZICGtv4042vXIGdqoGtu3/+XF7rTqdG8hjp1IxJGJY9vu4G1zA1kndIp4bgN5csXs1nCXcuapgayTukBTirUBQ+UiR4mkuR4GkuRYKnuRQJnuZSJHiaS5HgaS5Fgqe5FAme5lIkeJpLkeBpLv8PQBpICdlXFgEAAAAASUVORK5CYII=\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "    res = all_res[[treat]]\n",
    "    # significant thr \n",
    "    pvalue_lim = 0.05 #; log2FoldChange_lim = 5\n",
    "    res$sig <- as.factor(\n",
    "        res$pvalue < pvalue_lim\n",
    "#         | (abs(res$log2FoldChange) > log2FoldChange_lim & res$pvalue < pvalue_lim)\n",
    "    )\n",
    "    relevel(res$sig, ref=TRUE)\n",
    "#     ## Volcano plot\n",
    "    vol[[treat]] = res %>% ggplot(aes(x=log2FoldChange, y=-log10(pvalue),colour=sig, fill=sig)) +\n",
    "            geom_point(aes(color = sig),alpha = 1/10) +\n",
    "            ggtitle(treat) +\n",
    "            xlim(c(-20,20)) +\n",
    "            ylim(c(0,11)) +\n",
    "            geom_hline(yintercept=-log10(pvalue_lim), linetype=\"dashed\", alpha = 4/10) +\n",
    "#             geom_vline(xintercept=log2FoldChange_lim, linetype=\"dashed\", alpha = 4/10) +\n",
    "#             geom_vline(xintercept=(-1)*log2FoldChange_lim, linetype=\"dashed\", alpha = 4/10) +\n",
    "            scale_color_manual(values = c(\"grey\", \"black\")) +\n",
    "            theme_bw() + theme(legend.position=\"none\") +\n",
    "            geom_text_repel(\n",
    "                data = subset(res[order(res$pvalue),], sig == TRUE)[1:5,],\n",
    "                aes(label = name),\n",
    "                size = 3,\n",
    "                box.padding = unit(0.35, \"lines\"),\n",
    "                point.padding = unit(0.3, \"lines\")\n",
    "            )\n",
    "}\n",
    "v = vol$METTL14 + vol$METTL3 + vol$Virma + vol$WTAP\n",
    "ggsave(\"plots/peak_Volcanos.png\", plot = v, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/peak_Volcanos.pdf\", plot = v, device = 'pdf', dpi = 300)\n",
    "v"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%R\n",
    "# library(Gviz)\n",
    "# library(Biostrings)\n",
    "# library(rtracklayer)\n",
    "# library(GenomicFeatures )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%R\n",
    "# txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')\n",
    "\n",
    "# %R bg_ip.treated \n",
    "\n",
    "# grtrack <- GeneRegionTrack(txdb, chromosome = chr, name = \"Gene Model\")\n",
    "# gtrack <- GenomeAxisTrack()\n",
    "\n",
    "# # %%R \n",
    "# # call_peaks('chr1')\n",
    "# Tx <- transcripts(edb, filter = GeneNameFilter(\"BCL2L11\"))\n",
    "\n",
    "# # %%R\n",
    "# peak_ploter <- function(treat,ymax, mins = rep(50,6), plus = rep(50,6), saveplot = '' ){\n",
    "#     for (i in 1:6){\n",
    "#         if (saveplot == 'pdf') pdf(paste(\"plots/hiv.\",treat,'.',names(hiv_gr)[i],\".pdf\",sep=''), width=7, height=7, paper = \"a4r\")\n",
    "#         if (saveplot == 'png') png(paste(\"plots/hiv.\",treat,'.',names(hiv_gr)[i],\".png\",sep=''))\n",
    "#         plotTracks(\n",
    "#             list (gtrack, grtrack,\n",
    "#                   get (paste('treated.s23', treat, sep ='.')),get(paste('treated.s24', treat, sep ='.')),\n",
    "#                   get (paste('nontreated.s23', treat,sep ='.')),get(paste('nontreated.s24', treat,sep ='.'))\n",
    "#                  ), \n",
    "#             type=\"l\",\n",
    "#             from=start(hiv_gr)[i] - mins[i], to=end(hiv_gr)[i] + plus[i] , ylim= c(-100,ymax[i]), \n",
    "#             main = paste(treat, names(hiv_gr)[i]),\n",
    "#             add53=T,chromosome = \"chr1\",cex=1)\n",
    "#         if (saveplot != '') dev.off() \n",
    "#     }\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[Wilcoxon signed rank test](http://www.sthda.com/english/wiki/one-sample-wilcoxon-signed-rank-test-in-r) amd t.test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"METTL14\"\n",
      "\n",
      "\tWilcoxon signed rank test with continuity correction\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "V = 13440000, p-value = 5.852e-05\n",
      "alternative hypothesis: true location is less than 0\n",
      "\n",
      "\n",
      "\tOne Sample t-test\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "t = -12.325, df = 7526, p-value < 2.2e-16\n",
      "alternative hypothesis: true mean is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.1625452 -0.1179335\n",
      "sample estimates:\n",
      " mean of x \n",
      "-0.1402393 \n",
      "\n",
      "[1] \"METTL3\"\n",
      "\n",
      "\tWilcoxon signed rank test with continuity correction\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "V = 13260000, p-value = 8.452e-07\n",
      "alternative hypothesis: true location is less than 0\n",
      "\n",
      "\n",
      "\tOne Sample t-test\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "t = -13.059, df = 7525, p-value < 2.2e-16\n",
      "alternative hypothesis: true mean is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.2367021 -0.1749132\n",
      "sample estimates:\n",
      " mean of x \n",
      "-0.2058077 \n",
      "\n",
      "[1] \"Virma\"\n",
      "\n",
      "\tWilcoxon signed rank test with continuity correction\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "V = 13526000, p-value = 0.0003431\n",
      "alternative hypothesis: true location is less than 0\n",
      "\n",
      "\n",
      "\tOne Sample t-test\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "t = -11.271, df = 7526, p-value < 2.2e-16\n",
      "alternative hypothesis: true mean is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.1591423 -0.1119880\n",
      "sample estimates:\n",
      " mean of x \n",
      "-0.1355652 \n",
      "\n",
      "[1] \"WTAP\"\n",
      "\n",
      "\tWilcoxon signed rank test with continuity correction\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "V = 13032000, p-value = 9.081e-10\n",
      "alternative hypothesis: true location is less than 0\n",
      "\n",
      "\n",
      "\tOne Sample t-test\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "t = -13.518, df = 7526, p-value < 2.2e-16\n",
      "alternative hypothesis: true mean is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.2191196 -0.1636185\n",
      "sample estimates:\n",
      " mean of x \n",
      "-0.1913691 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAC+lBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+RkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///+7+53vAAAgAElEQVR4nO2dC3hU1b3oR+dI5NHruUmI0EByqRJMIQk1CS8D4dEc4aY8tBaJxaK1LQdfbbVFrRwOIODVqtWeEo9Y31dDiUlQTkHqFRSUSC3aWN5YCI/0JCGTZN57Hv/vu2u/ZvZMsh9rz16ZzJr1+2T267/X/u/9c+/J7LX3WjZgUI0t2QkwyMIEUw4TTDlMMOUwwZTDBFMOE0w5TDDlMMGUwwRTjknB1bbfoc/uK23n0CjPFlsc56pXCpEN866yOcWVvmP7ozVJM4xjVvCY6ehz6xhe8C2HEZ3o38LZhw//5VNxeJiTBL+x4f9Igrf+CxM88JgVvPKq0wCVq3nBK+WZKxYrh5H5H4mCz4w9wwQPPGYF//SH6+DsiIPGBYdnvxRgggce04I/KICNy1t4wZfZEZ+DjuCnFwATnARMCw7nNRfuFgQvbUF4QVvw8VHnmeBkYFowPFwxOthi+BL9Gn+e2y6/LaFkGfiYF3zE9gAYF9yNzvLPbbXnEkqWgY95wdDpEwULP5PaoI9gYX5v1+GXbB8f9vBz2CU6CSQgGNESudGxFvoIFub/8TVhcJifwwQnAXarknKYYMphgimHCaYcJphymGDKYYIphwmmnEQFZ3j5/9RYF1KbUXVUs8wYglPaoxPHpoybeiJmtN+B/76c0Xdqpy7nYnGSBhNUhBElUcHNYS3B9oDKjPdu0SizT4EvPBAdr3gRXpwdM9rv4BcLnXBSO3U5F4uTNJigIowo5gTX3g3Ntk9h1fPSGbxvWnHJXtg5uWjuSTg65gysvVUIe8g2q9JzYNb15e+Dd2nxtxaLM/glS7ejTa9ZfkODELf+ZwCdmS74fmnxki7+2DmyAAIZAOK60DMyCC6/EHpuBAfciDbFaL8D75V/19sHOReLkzSYoCKMLOYEn5wAm6ZvhoJTouD2kQch6GjLPg7PTwF4derugh4xDp0Ll4raoDXPt2MxQHf05Mg5hzb9BnSO+W9+6uyoADyLLqgdAJseVhw7aV2AokNQ/Zyw4v4C9FHQrBjtd/C3rEcmTfkv7Z1QnMEWJmkwQUUYWUxeovPOzdv77dZ86Tu4oYqf9/aN6JtoSC/AiqF/lsLQoWoaWVlZOf7kmfxV21yRY8fZOLRpJ8CiJmF6bhNc/yHAlpkVJdWKYyeti74MG+Cv54VI4cCMb1aM9jv43PY0HLxK+wRRCLYwSYMJKsLIYlLwHbWTYOKWO2ME188XBfvLRu+UwtChapgpjjqb7rnGLR+78BXo/wMbutAt2CFMv3pzyzUAhwq6oGkeX6YzE8CdEVkXZu6SN9wavbRJo/0Oui5Df/IU7dHcB4VgC5M0mGDr4L5Ew2v5K+En+a9LgjtyDkKgqy37KNROBbh/5Zdj0cXtvQsAmR3QkYX2+xM47wTfqLP8DIHJh9Cmn4QT2e1CnDt7xQaA3ZUANcKxg1EXoC5DXhddLNtg/ylxzRlbYWulWLo02u9gwe/h6FXnNfeBz4VAkgYTlAbEMSn4gm0b1Nkuyj+TPppSWLQX3i0pnHMSGoq9sLUiABPR3yYbJpR4Pr6hcPz3YHdx0cQN4gx+/fWP8cdu2gQUw8fBXZefRad/zfyaB8Vjt61s2Ub094u4Lnw2A+TvYDhSPrb8mLiWNNrv4O+V107crr0PfC4EkjSYoDQgTpJudLSXcBib/lEjwVTUSYkk9UjWnax3ThnfdPC3JDPRICWS1IHdqqQcJphymGDKYYIphwmmHFXBwV/e+icA760PP3wYoOnf1lyUPxkpharg8KW3eMEP8+Odvwyf2CR9CjhUcXnVl+ngMb2mz2V2zW636Y0Gus2u6ewxu2YvZyzOiGAAQfBtq59xwv5XAFZKnwDHm5sd3Wp4ONVFevhMr8l5zK7Z4zW9Uegxu6bbaXZNV9BYnHHBYSc0/Q52/QHgx9InwJPLl3sCaoTCqov0CJleMxxMwkbB9JpB0+kGDR5d44IR3Q/GncH8NbtDDZdPdZEeHtNr+nvNrtnpMr3R8CWza/Y6zK7ZHTAWh3GJDsP+X8OlX4aPb5Y+mWAJXnCNCOaag0Tw4z+++3loeeCRR/8b/f386KMX5U8mWCQlBXsiH9owwSkqWKiBLmGCjZCCgkOBWegvsI4CfcHqvxDS62eSJBhzzaT9TFpnt9nt9swNTLARUlAwwEP6bgXYJTolL9E8nNer/qICE6wgJQU35A0bPny4MCpWNsRUM0QrG5jgFBVc+lVkVKxsUFYzKCobmOAUFbxMOcELVt6kZLcqO5RKU1LwpkZ/rGBlNYNc2fAfq1a5ODWCIdVFeiSwZtD0qrhrSkrRGMRMYW0zgLlChEDYWJyqYLGxdp0z+PCePd29avg41UV6+E2vGfCaXtWHGS8pRWPgVE7h4HFhrhDBHTIWpyo4Bl6wspqBVTZ0pPwlWolQ2RBTzcAqG1JecIaA+kktwwSnqOBAIOCu28gEq5PiggVqmGB1KBB8oVxfsEcVLqi+TIeA6TWDfrNrejnMFSSlaAy8kanoTEP4fZgbjeALGYtTFZyVlZU5Wue1WnYG9zmDMU/kJJ7B/EO7+nqZ4JQVDBDzI5kJ7kOKCz5ePmTIVGX7YSqvrqSh4D4uU1Lw7BdCwdq5SsH9v7rCBKeo4CL+o1ApuJ9XVzrPn7/UpYbbr7pID6/pNTmX2TUdbqORsS7RDOjuZ6YhXD2mckX0Bo3FqQouOQ5wVPlUZX+vrjw0Z443rAqoLyLGQGwz1qW00T4zSWNwT1UF78iaf2P2zphZ7NUVETou0dBW/3ZM62vs1RUZOgR/4ADo2qeYwV5dkaFD8KQgAFcEuqST4Jr+6OhPsFHPSRRczH8Ugi5McIoKnnYI4MBUfcHqf6Cn2s8kj25IvxK7+vuZFF2mTRJ/Ju3LqarK+UhfsFMVX0B9mQ5+02sGvGbXdOlvtF+JaD641Jdp43WbTdcTMhanKhjaGxo7QJ+0uET3ay/FL9FGYYI1BOt6HrSC06WyQUsbzYLpr2wwYNa4YHXPg1UwjbcqjStNA8FyZcNbGzc6vWpwQdVFegRMrxni+p1t1p4BUOngM7EFIS/Ob3ZH/WFjcYmdwXtefrnXpYY/oLpID870mkGf2TXdftMbBbfZNX0es2t6Q8bizAlO78qGPqR+Q2h9iFY2qPPB7zEKtIrfHUzCRu/rGfhtnngMexWr24t+61GLCzTCz99JwkanG7kjZDGf3Yq9ChNsljQVzBhkMMGUY6ngmJZ5Bg7p4e2BJQn9G5jaT0sFK1vmGUDEh7cHlgHfSTC5nxZfoqPtegwg4sPbA8uA7ySY3E8CguVbmgOG+PD2wDLgOwkm99M6wTvXvpSMM1jYLP/w9sCSjDMYzOwngTNYcUtzYBAf3h5YBnwnweR+WitY2TLPwCE9vD2wDPROgsn9ZL+DKYcJphwmmHKYYMphgimHXsEZcV0TXLopN/c75wGyrs7NvSsuyA7ArRmXP341R90BoW1/osQL7qjnQvfdjAS39A1Cgr+/oB1cmxzUHRDa9icKcrdzctHckwB115U9YxfmNZbJgqVFKKj+uor1djhxZbu4mu2Jkm80It+lxUu60NTjt5U1Rko4MOv68veTtkPmoFlwW/ZxeH4KXLy6FZ4WBIeqNiPBX8/P3yEtEoJOw+N2qJsorWarhc/y0PkOsOlhNNUIX+XJJVwqaoPWPF8S98kENAt++0aA4JDe+iUA3YLgu5cEpTNYWoSC6heir2c7vPVNaTUbumgP5WDLzIqSajTlBxjGSSU0jaysrBx/Un2TgxGaBdfPFyxuv0kSfO9CvkcKQbC0iA/6LkAvukRnyJdo9G+491BBFzTNk6ekEhpmJmtfEoBmwW3ZR6F2KlzMaYWnkOB7Fgk9jgiCpUV8UJ4DXkdLly34B7gec0hKd1cC1EQESyV0ZO0C+CSpO4UPzYLh3ZLCOeiK+mbBjM2Z8IUN/T6K/JElLUJB2yt+sBYJ9j+Sn3vto5ykNFgzv+bBiGCpBPj4hsLx30vuXmFDr+AoLoBXqpNcQtJIB8G/Kp0873SSS0ga6SA4rWGCKYcJphwmmHKYYMphgimHCaYcJphymGDKYYIphwmmHJOCq238a27dV9rOoVGeLbY4zlWL72Y9XTg0a3GKVZLThFnBY6ajz61jeMG3HEZ0on8LZx8+/JdPxeFhThK89c1P/lgxwbqEGXiYFbzyqtMAlat5wZG3KFcsVg6j8+GArTOBFBmJYFbwT3+4Ds6OOGhIcNePvxmOL4AxQJgW/EEBbFzewgu+zI74HNQE19tt152xIFOGKUwLDuc1F+4WBC9tQfAPkPcvuKflvXkVQQtSZZjBtGB4uGJ0sMXYd7DjslR7XJwezAs+YnsADAput+1NIEVGIpgXDJ0+UbDwM4nv8TBOsDC/94eNBxumX+u2KF0GLgkIRrREbnSshT6Chfl/vH3skDHLv7IiVYYZ2K1KymGCKYcJphwmmHKYYMphgimHCaYcJphymGDKSVRwhrdPe0UK1oXUZlQd1SwzhuCU9ujEsSnjpp6IGdUaaCHnYnGS1iVoCYkKbg5rCbYHVGa8d4tGmX0KfOGB6HjFi/Di7JhRrYEWci4WJ2ldgpZgTnDt3dBs+xRWPS+dwfumFZfsldqeOjrmDKwVO3B6yDar0iM2LuVdWvytxeIMfsnS7WjTa5bf0CDErf8ZQGemS26cKsPryAIIZEQapuoZGQSX0MAGnBvBATeiTTGqNdDaBzkXi5O0LkFrMCf45ATYNH0zFJwSBbePPAhBh9z21KtTdxdI/fqhc0FqXGrHYoDu6MmRcw5t+g3oHCM0cH12VACevTPSOFXk2EUapio6BNXPCSvuL0AfBc2KUa2B5k4ozmALk7QwQUsweYnOOzdv77db86Xv4IYqfp7c9hSsGPpnKQwdKqlxqTP5q7a5IseOs3Fo006ARU3C9NwmuP5DkBunihy7SMNUVQ3w1/NCpHBgxjcrRrUGmvugEGxhkhYmaAkmBd9ROwkmbrkzRrDc9pS/bPROKQwdKrlxKWfTPde45WMXvgL9f2BDF7oFO4TpV29uuQZAbpwqw+vMBHBnRBummrlL3nBr9NImjWoNNPdBIdjCJC1M0BJMCn4tfyX8JP91SXBHzkEIdMltT92/8sux6OL23gWAzA65canzTvCNOsvPEJh8CG36STiR3S7EubNXbACQG6dCZY66AHUZ0Yapctpg/ylxzRlbYWulWLo0qjXQgs+FQJLWJWgJJgVfsG2DOttF+WfSR1MKi/ZKbU81FHtha0UAJqK/TTZMKPGIjUvtLi6auEGcwa+//jH+2E2bgGL4OLjr8rPo9Jcap0IFbitbtjEj0jDVZzNA/g6GI+Vjy4+Ja0mjWgMt+FwIJGldgpaQpBsd7SU4LTP/qJFgKuqkRJJ6JOtO1junjG86+FuSmWiQEknqwG5VUg4TTDlMMOUwwZSjKljszVvqdFrs7joJnV4zEkVVsNibt9jptNjdtaLTa4/HhYPHhxXuJlq6y4tXup9k6W4ypRsRLHYVK3Y6LXaWq+gyt7u3AweXDyu8y4kV3uvHCu90Y4X3BLDCOzxY0Y4gkdKNCxY7nRa7u5Y7vb6/tNSvc11gDBp0BIPQ6XTsGezu6el2duLg8mGFO1xY4U4OK/ySByu8J4AV3unFiu4OESnduGCx02mxu2tFp9fsEq1K6lyihd68pU6nxe6uo51eM8GqDG7BnsiHNkywKoNbsFBBWcIEK6BJcCgwKxAIdBQwwQpoErzObrPb7Zkb9AX7/B4cuCBWuJfDCvfjle7BLD2EV3oAK9oXJlK6imCAh/TdsjNYm8F8BvNwXq/6c+xMsD6DW3BD3rDhw4czwQp4wTUCxuIHt+DSaHM4Ym1STD1StDaJCVZlcAteFh0Va5OU9UiK2iQmWJXBLXhTo6IagResvAst34ve8/LLLrwaOn8AKxyzhs4XxAp3cxjBNQqMrYFTOtrVMFa40dJVBYuNtSsEK+uR5NqktzZudPN/ixmHC2KF+zBLD2GFewMYsUrB1pfu9frDWOFGS1cVHIPaGZxOl2ilYGNrDO5LdB/BynqkdKxNqolHd43BLThDQBwXapNi6pHSsDaJNsGBQMBdt1H9pJZhglUZ3IIFaphgnj5uaRF8oVxfsNvrxMEXwAp3+7DCvUGscJffWFz/gnVXM1i6hDuEFW60dFXBWVlZmaO36wvucXXh4PZjhXe7scJdHFa4w2Msrn/Buqt5sZLpCWGFGy1dVbDD4XCCAdglWpXBfonu7WWCRagUfLx8yJCpyo4GVV5doVtw/2qpEDz7hVCwdq5ScP+vrjDBqgxuwUX8R6FScDq+uqIlWFfz4BZcchzgqPKpyv5eXVm3aJE3FMQhFMYLJ1p6UDdcX3AipccQAqxwo6WrCt6RNf/G7J0xs/q+utJ65EiP24GDx48Vjlm6m8MK7/bqRegL1ljZh5VMbwgr3GjpqoKhrf7tmMa50u3VFX23qX2J/gC579qnmJFur67QLngSunpzRaALE6zK4BZczH8Ugi5MsCqDW/C0QwAHpqalYINqdTUPbsH7cqqqcj7SF0zhqysmBPdbzuB+dQXaGxo7QB+qzmATalP2DDYKE8wEKxm0gk27VbecyoJpqk2yQC11glO/NskyqxqaU1iwfC/68J49Tk8vDj4OK9zlxQr3BrSWkpOqAlbu7hBWeK/fWJg5wXJt0n+sWuUJcjgEQ1jhHGbpYZKlBzBLx9tVQqUndgZDyl6iBdL6BXBNFLVJXQ6stD7djRXeiVf65+9ghXd0Y0Uf2Y5Xeg9W9Kk3iJRuTrCiNgmTN9eYWs0g/3UfydI/vp1k6UcWkiydZ0AaBGeC1aBE8D9OkSy9k2j3Q91fkizd9TnJ0nlYk/6UQ15wTGsuJJCe3SYD0W4MiGYuQV6wsjUXIojPbpOBYNpANnOZgbhER9uCIIL47DYZCKYNZDOXGSjB8m0wAojPbpOBYNpANnMZwoJ3rn2J7BksbIB/dpsMZM9gIJi5zECdwYrbYFYjPrtNBoJpA9nMZQZAsLI1FxJIz26TgVzaQDhzCfY7mHKYYMphgimHCaYcJphy6BWcEdc1waWbcnO/cx4g6+rc3LviguwA3Jpx+eNXc9QdENr2J0q84I56LnTfzUhwS98gJPj7C9rBtclB3QGhbX+iIHc7JxfNPQlQd13ZM3ZhXmOZLFhahILqr6tYb4cTV7aLq9meKPlGI/JdWrykC009fltZY6SEA7OuL38/aTtkDpoFt2Ufh+enwMWrW+FpQXCoajMS/PX8/B3SIiHoNDxuh7qJ0mq2WvgsD53vAJseRlON8FWeXMKlojZozfMlcZ9MQLPgt28ECA7prV8C0C0IvntJUDqDpUUoqH4h+nq2w1vflFazoYv2UA62zKwoqUZTfoBhnFRC08jKysrxJ9U3ORihWXD9fMHi9pskwfcu5HukEARLi/ig7wL0okt0hnyJRv+Gew8VdEHTPHlKKqFhZrL2JQFoFtyWfRRqp8LFnFZ4Cgm+Z5HQ44ggWFrEB+U54HW0dNmCf4DrMYekdHclQE1EsFRCR9YugE+SulP40CwY3i0pnIOuqG8WzNicCV/Y0O+jyB9Z0iIUtL3iB2uRYP8j+bnXPspJSoM182sejAiWSoCPbygc/73k7hU29AqO4gJ4pTrJJSSNdBD8q9LJ804nuYSkkQ6C0xommHKYYMphgimHCaYcJphymGDKYYIphwmmHCaYchIVXPmMJWkwSGFO8ILbhMEd82DzbiuzYViOOcFvDuMbU/P8j1cic1LsQZb0wZxg71Uvo883hzuFS3Tpvy8Y8VzpE4uHFrd8OGnE7cj1SyXDxq7mdIthkMfkd/Bd30Yf1beL38Gl//P9sLN05O//tmhyxYcHRm0B+M/3zn7wDfbtPBgwKXjf5eeh/Z/2SILvRnNK/xXgE9ufAB6QehB/caZFKTISwaTg8Lgn4De5IUnwc2hO6e8AvrK1AzxRBfC3pdeN+uf/ZWmiDHOY/Zm0phjKV4Mk+AUQP/5ucwA8OQ+43Hs/v7A118I0GWYxK/ik7Q+2v4GK4GP8w8XrmODBgOkbHdP/uZQf9Cu4Z+i78MUoJngwYFpwre03/KBfwbBt3JiZv2aCBwPsXjTlMMGUwwRTDhNMOUww5TDBlMMEUw4TTDlMMOUkKjjD26e9IgXrQmozqo5qlhlDcEp7dOLYlHFTT8SMxg4W5ObO+Au/9HGbTlfmci4WJ6mXYOyAOIkKbg5rCbYHVGa8d4tGmX0KfOGB6HjFi/Di7JjR2MEZgP8sRgtb5o3XESznYnGSegnGDohjTnDt3dBs+xRWPS+dwfumFZfsldqeOjrmDKy9VQh7yDar0iM2LuVdWvytxeIMfsnS7WjTa5bf0CDErf8ZQGemS26cKsPryAIIZEQapuoZGQSX0MAGnBvBATeiTTEaOxCyKwHgbjg+QVuwnIvFSeol2DddspgTfHICbJq+GQpOiYLbRx6EoENue+rVqbsLesQ4dC5IjUvtWAzQHT05cs6hTb8BnWOE9rDPjgrAs3dGGqeKHLtIw1RFh6D6OWHF/QXoo6BZMRo7ALg1O/cLgH/7NegIVp7BFiapl2B8uqQxeYnOOzdv77db86Xv4IYqfp7c9hSsGPpnKQwdKqlxqTP5q7a5IseOs3Fo006ARU3C9NwmuP5DkBunihy7SMNUVQ3w1/NCpHBgxjcrRmMH6CP82xvhs4oQjmALk9RLsE+6hDEp+I7aSTBxy50xguW2p/xlo3dKYehQyY1LOZvuucYtH7vwFej/Axu60C3YIUy/enPLNQBy41QZXmcmgDsj2jDVzF3yhlujlzZpNHbAL+Au63hmVH7+P436g+Y+KARbmKRegn3TJYtJwa/lr4Sf5L8uCe7IOQiBLrntqftXfjkWXdzeuwCQ2SE3LnXeCb5RZ/kZApMPoU0/CSey24U4d/aKDQBy41SozFEXoC4j2jBVThvsl/o/nLEVtlaKpUujMQPHlxDekhvmI/XOYD4XAknqJBg3II5JwRds26DOdlH+mfTRlMKivVLbUw3FXthaEYCJ6G+TDRNKPGLjUruLiyZuEGfw669/jD920yagGD4O7rr8LDr9pcapUIHbypZtzIg0TPXZDJC/g+FI+djyY+Ja0mjM4OKUUblzxe7i9ATzuRBIUifBuAFxknSjo70Ep2XmHzUSTEWdlEhSj2TdyXrnlPFNB39LMhMNUiJJHditSsphgimHCaYcJphyVAWLHXdLfVSL/WAT7Q2bQQZVwWLH3WIf1WI/2GR7w2aQQeMSLQgW+qgWe9FV9KXb2RFHT3f8HG0CmPEevPAeDi++Gzf9Hrx43PQDePF90jcuWOyjWuwHW+4N+8nlyz2BOILB+DnahDHjQ3jhwTBmPGXpGxcMQh/VsWfw4T17unvj8Ljj52gT9ODF+/HCPUG8eDdl6WNcooU+qsV+sBW9YbNLdEdHDY/B+EF5iRY67pb6qBb7wY72hs0EUyBYCyY4RQV7Ih9MsBYpK1iogS5hgvVIUcGhwCz0J3ZHgb5gTzx+X59ZmoQw4wN44f4QXrwPN32/xyMINhg/0OmrCF5nt9nt9swN7AzWI0XPYICH9N0ywR0pLBiA83rVX1RggiVSVnBD3rDhw4cLo2JtUkw9UrQ2iQlOUcGlX0VGxdokZT2SojaJCU5RwcuUE7xg5V1o+V5065EjXY44XM74OdoEMeN9eOGuAF6800T6gmCD8QOdvqrgTY3+WMHKeiS5NmndokWeYByhUPwcbcK48XjhIdx4g+kIUmvE9KVRQwx0+qqCbQI6Z3AaX6JrpOtyyl6iY+AFK+uRWG0SXYKF2qSYeiRWm5TygjME1E9qGSY4RQUHAgF33UYmWI2UFyxQoy+4Kx5nb59ZmgSdePFevHBnADPeYPqi4K6ugFMel2boMNDpawq+UK4v2BmPx9NnliYhzHg/Xrg3iBdvNH3Rp5i+QrDuepjpexJNX1VwVlZW5ujt+oLZJVoeN3ahHjSXaP6uiL5eJjhlBQPEPHKp+uoKE5yigo+XDxky9aRScP+vrqSf4JoYnykrePYLoWDtXKVg9uqKCCWCi/iPQsWM/l5deWvjRqc3Dr8/fo42Icz4AF64P4QZr5uOUrCYfswMHQY6fVXBJccBjsY9Vdnn1ZU9L7/c64rD542fo00IM57DC/cGMeN101EKFtOPmaEDbvohzPj49FUF78iaf2P2TsUM9uqKBCWXaGirfzum9TX26kqs21QX/IEDoGsf6MIE9ztblUEjeFIQgCtigmOgSTDfknbMX9FMMF2Cpx0CODBVX3Dfly2ofHWlj0MRxasrsbNVGSSvrgDsy6mqytFpwzONzmAVwSl8BkN7Q2MH6MMEp6pgg1AvWEWtpmANzSkiOC1qkzTV0i04PWqTEhXcr+XUECzfi3b39HTG09t3liYBzHgvXnhvwGikAZ99ENM3Ekc8fZE+QswJlmuT7i8t9Rlfi5FcTJzBQPUl2hB0NGUYj1ZtksOBl9KxC3jxTrzwC8fx4nHTP044/WN48X3SNydYUZuUMMsPWlRQ/3x4J9Hi4faPiRa//w7rykpSg+BMsBYUCH7rLNHi/67d51nC1JFN/8w268piTfpTDhNMOUkQHNN+Dxmk5/VJQbbzCmuTT4JgZfs9hBCf1ycF4c4rrE0+KZfoaOsfhBCf1ycF0dStTj5pguUbn0QQn9cnBdHUrU5+oAXvXPsS6TNY2AT/vD4pCJ/BYGnySTuDFTc+rUd8Xp8URFO3OvlkCFa230MG6Xl9UpBM3erk2e9gymGCKYcJphwmmHLoFZwR13L9pZtyc79zHiDr6tzcu+KC7ADcmnH541dz1B0Q2vYnSrzgjnoudN/NSHBL3yAk+PsL2sG1yUHdAaFtf6IgdzsnF809CVB3XdkzdmFeY5ksWFqEguqvq1hvhxNXtour2Z4o+UYj8lKRcqgAAAwTSURBVF1avKQLTT1+W1ljpIQDs64vfz9pO2QOmgW3ZR+H56fAxatb4WlBcKhqMxL89fz8HdIiIeg0PG6HuonSarZa+CwPne8Amx5GU43wVZ5cwqWiNmjNS7EnSmkW/PaNAMEhvfVLALoFwXcvCUpnsLQIBdUvRF/Pdnjrm9JqNnTRHsrBlpkVJdVoyg8wjJNKaBpZWVk5/qT6JgcjNAuuny9Y3H6TJPjehXyHBYJgaREf9F2AXnSJzpAv0ejfcO+hgi5omidPSSU0zEzWviQAzYLbso9C7VS4mNMKTyHB9ywSOqQQBEuL+KA8B7yOli5b8A9wPeaQlO6uBKiJCJZK6MjaBfBJUncKH5oFw7slhXPQFfXNghmbM+ELG/p9FPkjS1qEgrZX/GAtEux/JD/32kc5SWmwZn7NgxHBUgnw8Q2F47+X3L3Chl7BUVwAr1QnuYSkkQ6Cf1U6ed7pJJeQNNJBcFrDBFMOE0w5TDDlMMGUwwRTDhNMOUww5TDBlMMEUw4TTDlMMOWYE/zUOPSx3fYi+py02iZxDrZc/hs0p9pmG1a2U6cExgBhTvAhZBN+OuYOgK7L/nT48OGFs9EHB9N/UYoWVt969NAdQ45ZmyfDJOYEB7/2fwGuf/oagB1XuNH0isX83JNZvlF/Q4JXAnBfe87KLBmmMfkd/C//Cr1DnV+7CL+czk+Kgtf8GO5dLQqGrKcsS5GRCCYFPzYRdk2DqjqYtpqfFASHx/0JPhoT4gV7n7CRbemMYRSTgvdd1vmrX8C6ez1XCH9NCYL35QQhPGYPVF9mv2wYudevGViYFOzLaJzVBO+X/L/Lu/lJQfBdNrvdftntUL205XTAwhQZiWD2d/DMe4d2gjvj598SpnjB3qtea2lp2T7cVU22/QoGFmYFPzKC7z+rfMRPhSle8JtZQTQW/vorTPBgwqzgXTa+IaGf2+qFKV7w/14hjK6cxwQPJtitSsphgimHCaYcJphymGDKYYIphwmmHCaYchIVnOHt016RgnUhtRlVRzXLjCE4pT06cWzKuKknYka1BlrIuVicpHUJWkKigpvDWoLt8ZUO8oz3btEos0+BLzwQHa94EV6cHTOqNdBCzsXiJK1L0BLMCa69G5ptn8Kq56UzeN+04pK9UttTR8ecgbW3CmEP2WZVesTGpbxLi7+1WJzBL1m6HW16zfIbGoS49T8D6Mx0yY1TZXgdWQCBjEjDVD0jg+ASGtiAcyM44Ea0KUa1Blr7IOdicZLWJWgN5gSfnACbpm+GglOi4PaRByHokNueenXq7oIeMQ6dC1LjUjsWA3RHT46cc2jTb0DnGKFh5LOjAvDsnZHGqSLHLtIwVdEhqBafAdpfgD4KmhWjWgPNnVCcwRYmaWGClmDyEp13bt7eb7fmS9/BDVX8PLntKVgx9M9SGDpUUuNSZ/JXbXNFjh1n49CmnQCLmoTpuU1w/YcgN04VOXaRhqmqGuCv54VI4cCMb1aMag0090Eh2MIkLUzQEkwKvqN2EkzccmeMYLntKX/ZaPmhWXSo5MalnE33XOOWj134CvT/gQ1d6BbsEKZfvbnlGgC5caoMrzMTwJ0RbZhq5i55w63RS5s0qjXQ3AeFYAuTtDBBSzAp+LX8lfCT/NclwR05ByHQJbc9df/KL8eii9t7FwAyO+TGpc47wTfqLD9DYPIhtOkn4UR2uxDnzl6xAUBunAqVOeoC1GVEG6bKaYP9p8Q1Z2yFrZVi6dKo1kALPhcCSVqXoCWYFHzBtg3qbBfln0kfTSks2iu1PdVQ7IWtFQGYiP422TChxCM2LrW7uGjiBnEGv/76x/hjN20CiuHj4K7Lz6LTX2qcChW4rWzZxoxIw1SfzQD5OxiOlI8tPyauJY1qDbTgcyGQpHUJWkKSbnS0l+C0zPyjRoKpqJMSSeqRrDtZ75wyvungb0lmokFKJKkDu1VJOUww5TDBlMMEUw4TTDmqgsV+uqXeqMUur8l2fM0ggqpgsZ9usTdqsctrRcfXnR0G6QSjkSIevHDjiZgpPdRFsvRgN7nSjQgWO4EVe6MWO8yVu81tevZZp8cgXjAaKRLACwcvydLDZEv3kSvduGCxN2qxy2u542sm2JrSB4VgEHqjjj2D2SXaktIHySVa6I1a7PJa0fE1E5x46UkXLPTTLfVGLXZ5He34mglOvPSkCPZEPrRhghMvPSmChRroEiZYhDrBocCsQCDQUaAv2O0yiBuMRopweOHGEzFTethDsvQQwdJVBK+z882oZG7QF3yp0yCXwGikiAcv3HgiZkoPO0iWHuwmV7qKYICH9N0KsEt04qUn6a9ozutVf1GBCbaw9KQIbsgbNnz4cGFUrGyIqWaIVjakjeAaASKlJ0Vw6VeRUbGyQVnNkGaVDTVRCJSeJMHLlBO8YOVNyjS7VUml4E2N/ljBymoGubLhoTlzvGGjgOFIM5AsXSGYSPkkc1cVLDbdrnMGd54/f6nLIA4wGinixQsHB7nSFYIJlN7VFezFCscqXVVwDLxgZTVDmlU21MRjaemDpbIhppohvSobqBScIaB+UsvQLbiPWnoEBwIBd91GJphawQI1TDDVgi+U6ws2/HBQSj6TpSLYotJlkvJMVlZWVubo7ewMpvYMdjgcTn29THAipcsk6RLd28sEUyz4ePmQIVNPKmaovLpCrWAVt4Ysp4Lg2S+EgrVzlYLT7NUV2gUX8R+FSsH9vLpyoL6+x2kQFxiNFOHwwsFlcem6ghMqPYaQByscq3RVwSXHAY4qn6rs79WV369e7fIZxA9GI0WCeOGWl64rOKHSYwhzWOFYpasK3pE1/8bsuK5/0+rVFdov0dBW/3ZM62vp8+qKrloqBH/gAOjap5iRPq+upIfgSUEArgh0oUqwQbVUCC7mPwpBFybYcOmqJEXwtEPoV9BUfcG9RnEaDxXw44WD04LSsQSLWJF7yG1B7iqoCt6XU1WV85G+4G6HQbrBaKSIFy/ceCL9l27CrYAVuQedieWuhapgaG9o7AB9Uv8SbVatiBW5J/2ZLCY47QTTVNmQmOB+Nae8YEoqG6xQS6fg1LxVaZ1PPcspL1iubHhy+XJPwChgOFIghBcOpPWZwGju4SC5I5PYGXy8udnRbZAeMBop4sMLhx6SpYd7SZYecpIr3Zxguisb+oHCF8C1iVY2GKZnDuYKeMzpIVn6knMkS//BlyRLj0C4vejuMqLFlxIVPO8sydK/+wXJ0iMQFuy8nWjxy136MeZZRbRjsl8OROeyrMV36mGCKYek4JiGekggPbVNCKI9GJBNXQFJwcqGeoggPrVNCIJ5A+HUlZC9REeb+SCC+NQ2IQjmDYRTV0JesHyHkwDiU9uEIJg3EE5dCTHBO9e+RPYMFjbAP7VNCLJnMJBMXQn5M1hxh9NqxKe2CUEwbyCcuhKigpUN9ZBAemqbEOTyBtKpK2C/gymHCaYcJphymGDKYYIph17BGXFdE1y6KTf3O+cBsq7Ozb0rLsgOwK0Zlz9+NUfdAaFtf6LEC+6o50L33YwEt/QNQoK/v6AdXJsc1B0Q2vYnCnK3c3LR3JMAddeVPWMX5jWWyYKlRSio/rqK9XY4cWW7uJrtiZJvNCLfpcVLutDU47eVNUZKODDr+vL3k7ZD5qBZcFv2cXh+Cly8uhWeFgSHqjYjwV/Pz98hLRKCTsPjdqibKK1mq4XP8tD5DrDpYTTVCF/lySVcKmqD1jyfxjYHITQLfvtGgOCQ3volAN2C4LuXBKUzWFqEguoXoq9nO7z1TWk1G7poD+Vgy8yKkmo05QcYxkklNI2srKwcf1J9k4MRmgXXzxcsbr9JEnzvQr5HCkGwtIgP+i5AL7pEZ8iXaPRvuPdQQRc0zZOnpBIaZiZrXxKAZsFt2UehdipczGmFp5DgexYJPY4IgqVFfFCeA15HS5ct+Ae4HnNISndXAtREBEsldGTtAvgkqTuFD82C4d2SwjnoivpmwYzNmfCFDf0+ivyRJS1CQdsrfrAWCfY/kp977aOcpDRYM7/mwYhgqQT4+IbC8d9L7l5hQ6/gKC6AV6qTXELSSAfBvyqdPO90kktIGukgOK1hgimHCaYcJphymGDKYYIphwmmHCaYcphgymGCKYcJppz/D16WCzSnfu9uAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "    res <- all_res[[treat]]\n",
    "    d <- res$log2FoldChange[!is.na(res$log2FoldChange)]\n",
    "    w <- wilcox.test(res$log2FoldChange,mu = 0, alternative = \"less\")\n",
    "    \n",
    "    t <- t.test(res$log2FoldChange, mu=0)\n",
    "\n",
    "    print (treat)\n",
    "    print (w)\n",
    "    print (t)\n",
    "    \n",
    "    hist[[treat]] = ggplot(res, aes(x=log2FoldChange)) +\n",
    "            ggtitle(\n",
    "                treat, \n",
    "                sprintf(\"wilcox.test (p.value): %.5f \\t t.test (p.value):%.5f\", w$p.value, t$p.value)\n",
    "            ) +\n",
    "            ylim(c(-10,1500)) +\n",
    "            geom_histogram(binwidth=0.3)\n",
    "}\n",
    "h = hist$METTL14 / hist$METTL3 / hist$Virma / hist$WTAP\n",
    "ggsave(\"plots/peak_Histograms.png\", plot = h, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/peak_Histograms.pdf\", plot = h, device = 'pdf', dpi = 300)\n",
    "h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pairwise correlation of each treatments \n",
    "http://www.sthda.com/english/wiki/correlation-matrix-a-quick-start-guide-to-analyze-format-and-visualize-a-correlation-matrix-using-r-software"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Hmisc)\n",
    "library(corrplot)\n",
    "library(pheatmap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "METTL14  METTL3   Virma    WTAP \n",
      "      2       3       2       2 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "all_fc = data.frame(\n",
    "    METTL14 = all_res$METTL14$log2FoldChange,\n",
    "    METTL3 = all_res$METTL3$log2FoldChange,\n",
    "    Virma = all_res$Virma$log2FoldChange,\n",
    "    WTAP = all_res$WTAP$log2FoldChange,\n",
    "    row.names = rownames(all_res[[1]])\n",
    ")\n",
    "colSums(is.na(all_fc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "png \n",
      "  2 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "res2 <- rcorr(as.matrix(all_fc))\n",
    "# # Extract the correlation coefficients\n",
    "# res2$r\n",
    "# # Extract p-values\n",
    "# res2$P\n",
    "\n",
    "pdf(\"plots/Corrplot.pdf\", width=7, height=7)\n",
    "c = corrplot(res2$r, type=\"upper\", order=\"hclust\", method=\"pie\",\n",
    "         p.mat = res2$P, sig.level = 0.01, insig = \"blank\")\n",
    "dev.off()\n",
    "png(\"plots/Corrplot.png\")\n",
    "c = corrplot(res2$r, type=\"upper\", order=\"hclust\", method=\"pie\",\n",
    "         p.mat = res2$P, sig.level = 0.01, insig = \"blank\")\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nOydZ3wUVdfAz53Zmi0pu+lt0xMSCCQkhB46KFIFCSJNUcQCdtQHFFF5rNgeuygBFAFFEFGUGuk1lAChhIT03raXmfdD8oYAybbszmyS+//Nh83snXvPZHbP3nvuKYimacBgMBiM60GwLQAGg8Fg2gYraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraAwGg3FRsILGYDAYFwUraBdDoYDycraFwGAwLgGHbQG6N2PH3nmmtBQyMoDHg7/+YkMgDAbjQmAFzSpZWZCcDPfee+vMoUMwYACIxezJhMFgXAVE0zTbMnRj8vPhqacAIfj4YwgLAwDw84PsbPDzY1syDAbDPtgGzSoKBfz+O8yfD2PHwooVoNWyLRAGg3EhsIJ2ASZNgtOnQa2G3r2hsZFtaTAYjKuATRyuxMWLkJUFc+aAUMi2KBgMhn2wgsZgMBgXBZs4XA+jEZYuZVsIDCP8/Td8/jlUVoJWC2+9BQ8/DD/9xLZMGBcCz6BdD60WhELAz6XL8+GH8N57IBIBScL48aBWA0XBunXw/vuwaBHbwmFcAuwHzSoLF7Zx0mRiXA4MG3z2GWzdChER4OcHKSkwYwYAwIgR8MYbWEFjmsAmDlb57jtQKoHDufPAdAeKiyE2Fry9wcsLBg5sPjliBOTlsSoWxoXAuoBV4uNhxgwYP/62k1otfPklSwJhGEQuh7o68PCABQvAw6P5ZH09uLmxKhbGhcAzaFaZNw90ujtPcjjw6qtsSINhlkmToLAQAODtt0EiaT65YwekprIoFMalwJuEGIwr0dgIBAEiEdtyYFwCPIN2PQwGmDuXbSEwLCEQwBNPsC0ExlXAM2jXA7vZdWfw08e0As+gMRgMxkXBChqDwWBcFOxmxyrr17dx0mBgXA4XQKOBK1egqurW6n7kSFYFcj746WMsgW3QrJKQ0O5bFy4wKAfbHDgA06eDUAglJRAYCPn5kJAA58+zLZaT6eZPv7oaXnkFzp6FmBhYtQoCAtgWyBXBChrjAvTpA4sWwYIFoFBAfj789hscOwarVrEtFsaZZGTAlSswfTrs3AlcLuzezbZArghW0Kyyfj3MnAlEt98JkEigrg5IEoKDm2M3xo2DP/9kWywn082fvlwOx45BRAQ0NoK3NzQ2ApfLtkwuR3f9cLgIDz0Eej3bQrgA7u5QXQ0A4OsLFy6ARtMt8lF086dfXQ2hoQAAEgl4eEBNDdsCuSJ4kxDjAsydC3v2QEYGPP00DB4MQiHMnMm2TBjn8+OPzQsIrRa2bAF39+bzs2axKJRLgU0crIJQswHubrq8D0N7XL0KDQ2QnMy2HM6nmz/9br5Hah1YQbMKQiCTtf1WVRWzomAYBz99jCWwgmYVhECjAYGAbTkwbICffpsYjTglegv4H8E2FMW2BC6ASgXffgtnzoBSeevkli3sCcQU+Om35uxZyMyEDRugrIxtUVwFrKDZRq/HCdphxgyorYUpU7rdvwI/fQAoL4cff4S1a+HsWRg8GD75hG2BXAisoNkGr3ABYN8+KC/vjkmQu/PT1+lg+3ZYuxZ27YKICJg5EwoKYNMm8PNjWzIXAitottm3r/vu47eQnAw3bpjb1u+qdOen7+cHCMHcubBiRbPTzuefsy2Ty4EVNNs89FDb57vVPv769TBjBgwbdtvs6ckn2ROIKbrz009MhH//hWPHIDISQkNBLmdbIFcEe3GwCt7Hb+K552DNGhg8+DaD7MaN7AnECPjp5+dDZiZkZkJBAYweDQcOwMWLEBLCtlguBFbQrIK/ok2IxXDhAigUbMvBLN386bdORXLwIKxdC5s2AUnCvHnwwQdsC+cqYAXNKt38K9pCdDQcOdJu1EZXpZs//btvX6OB336DH36AXbvYE8u1wAqaVbTa7vv9bM3nn8Nvv8GLL95mg+7ye4bd/Ol3898n68AKmlXGj4etW5v38VeuhMWLQSoFvR5SUyE7m23hGKRNz6ouH63QzZ9+N09FYh1YQbNK60mEXA7Z2RAU1O3qOhsM0Ls3XLgACLEtCrN086ePU5FYAXazw7ANlws8HhgMwOOxLQqGWYqKsInDPFhBY1yAhx+GqVPh6aebgxea6PI2aAzGElhBs81PPzWb4XQ62LoVPD27Y13nN98EuCtqo8vboAE/fYwFsA2aVXDO8u5MN3/6P/wAo0fjYt7mwQoag8GwQVQUXLsGEREwZAgMHQpDhkBYGNsyuRxYQbOKGW+q3r0ZlIMlTp6E0FDw9oaTJ9t4t29fxgVilm7+9AGgtBT+/ReysuDff+HCBQgIaFbWjz7KtmSuAlbQrIIQ8Hhtu5dptYxLwzgKBSxfDvPntx3knZ/PsDhM082f/h3U1MA338CHH0JFRbfwMrQOvEnIKoMGQXExPPggzJ4NUVFsS8M4+fng5wc5ObBtGyQmsi0N43Tzpw8AJhNkZ0NWFmRlweHDEBAA06fDoEFsi+VC4Bk02+TlQWYmrF8P3t4wezY88AB4ebEtE4Ns2wbr1sGOHRAbC3PmwMyZ4OvLtkwM0p2f/rhxcOYMxMTAoEEweDAMGABSKdsyuRxYQbsMBw/Cd9/Bxo2wfz/068e2NMxSWwubNkFmJpw4AaNHw5w5MGEC8Plsi8Ug3fDpR0WBwQDDhsHgwTB4cDddQ1gCK2jX4PhxWLcOtmyB5GT47LNul3izhbw8WLcOvvkGVCqorWVbGqbotk+/9SZhZSUMHNisrPv0YVsyVwHboFklPx82bIB160Ashtmz4dw58PZmWyb20Gjg+HE4dgwqKmDAALalcT746fv7w/TpMH06wP9vEr71Ft4kbA2eQbMKQUBUFMyaBXFxd751//1sCMQGNA3//guZmbB5c7MpdvbsbjGLxE+/9SbhwYPQ0ADJyTBkCPz3v2xL5ipgBc0q3TyWDACWL4d166CmBqZPhzlzutcOfjd/+uPGweHDYDBAv37NgSr9+4NQyLZYrgVW0BhWGTsW5syBSZPwN7PbsWoVDBkCKSk4i6EZsILGYDAYF4VgWwAMBoPpUkyePNnDw+N+R2wkYAWNwWAwjmTx4sXff/+9Q7rCChqDwWAcSXp6ukQicUhX2A/6FiaTqTtb5CmKIoju+4PdzW+fpmnU3WpCtoIkyda3f+NmcXstaZp243P5fJ5EIuFwnK4/sYK+xYkTJxz1u9fpoCiqpqZGLpezLQg7GAyGhoYGWXs1TLs6Go1Gq9V6enqyLQg76HQ6X1/fwMDApj8zf942d+4cJGj7v0HrlP7e0oH9kp988skhQ4Y4WzasoG/B5/Pj4+PZloIdjEbjsWPHuu3t19fXX758udvefnFxcXl5ebe9/crKSp1O1/JnfaMKuYeS/sltNqYbiqJ7+GzatJ4Z2bCCxmAwmFsQBIEAIdS2vYtGiCCYswVhBY3BYDC30OmNgKDtQgoAAEijN5rvYcyYMadPn1apVEFBQb///nufDuR+wgoag8FgbiEVCwERiGhbNyKC9JSKzPewa9cuRwmDFTQGg8HcokGlBYSgPTsGQrX1asaEwQqaBQwG48Gjpw4cPnb2XE5RUVF9fa1Oq9NqVARB8gVCgVAok3uHBAf1Teo9fEj/Pj3vSnXWyTHRdJVSX9qgLW3Q1mkMOqNJpTdlbf/54qHdfIHATST29fMPCQpMTOgxMKV3RHAg2/I6GBNF12kN1Sp9jcag1BkNJlpnMm357n9Xsk/weXyxVOrnHxAcFNg7IT4tKTHAt6slIDVSdJVKV6nUV6l0jTqj3kTrjKYfP/lvecE1Hpcn9fD08fVVhIYk9Uron5ToK2ehvoxQwEUIEQTZ5rs0QYhFzJWSwAqaIUwmauPWnet+2nTm5InK6lokcEd8d+BLEdcNuCGIxzHVnUdu3gTXj9bprudrjl29sPnPQ/SyN2m9KigwYNDgoY/NnzV0QGctdG2i6bwq1aXyxiuVSiNF+4j5/lKBr4Qf5ysR8Ug+h7jwY1HI6FnuAQptQ52yofpkRcWeDduWv7Oaaqz2CwweNmz47OmT4yND2b4POzFRdFG99madurheSwN4CrleblxvEU/h6SbgEDwOkVlwaejDLwol7qraamVddW55ycENvyx7+x1DY11waNioUaNmTpkQEtBZi4EZTNSNGvX1alVhnYZESC7iycW8QHehmM8RcAg+h/zqRs6gRatomlLX1ajqqk8VF/9zbEPlzRWgVYZFRI0ZM/bByeP9fRhygjQaKQAE7WwSAiCjibloCaygnc4ff+9/+72Pz2efUoGIkAYhj74cL+7dzRAiEcEFkodIHvAkrddXZQbNpt1nfv51u5RHp48YteLV53r1iGFM/g5SUKs+dKM6r0oV5S2O85WMjvUVcNr46JME4grFPJGUJ5JK/UNav6WqLj94/ujPM+dzDKp7J05+/rF5wf4+TInfUYrrtRfKGyqV+hAPYYinsH+oF6ettTOBkFAsdZO4u0ncvUPCW79VXVxw9OTBtfdn8ICaNn36wjkz5Z4eTInfIWga8mpUp4vqatT6SLk4zkcyOtqHbOv2SQIJxFIAEEo8ZMHh0DOlpYvqoht/njz4+dffuLsJZmRkLJqTIRG5OVVsI0Uh1K4XB0LIaDQ5VYDWYAXtLGiaXvHuZ5//73/VSiPhFYWCRpBgr3cOV0h4RoBnhIoy/X7o6rb+IxXBAatWvvbA5HscKrIjoWk4WVi750qlr5Q/MEyWkRRst2uSSOYblT4xKn2iQaM6dXxPv1HjoyIi3vzPS4NTejtSYodC0XROeePZkgY/CT/RX+onEdjdlSwwVBYYmjbxQXVj/cm9O1KGjuqTlPTGyy8kxEQ6UGDHYjBRJwrrskvqw7zchoTLfSX22gQQkgWHy4LDUybNVtZU/p2189NPBwwekv7mq89HhAQ5VORbSIQCQAQi2zZxIIKUiOx/mrbSfWNbncryVZ94BUSsXP1tnbQ3GToESfzBbu3cGoIkPMM44aMKTQEzH1kcGJW4ebvD9osdBQ1wOL/m7d25JQ3ap4ZEzEsNjfYWO8RxlCsURQ6dMPqVLyTJYx968vm0MZOOZOc4omNHQtH0meL69aeLNAbT9MSAkVHeHdHOrXGTuKdNfPCxj3+U9Ro0be6Cex6YfTkv3yE9OxAjRWflVX1zrIBE8GiaYlysr/3a+XbEXt4pk+Y8+N46o1/0iAnTpsx/vKC4zCE930GDRocACITaPBCCBrXOci8OAs+gHcw/+w/PnregQoVI3zSS46xfWiTwJEMGV2jrZ8xZ+GZMxLbNmYrgACeNZROFtZqfzhTG+EieGxbVpinDIcgjE9IXv1N949K0uY+mJvf5bvUqT6lLxOgX1WsO5FVHy8Uz+wS1acpwCBF90iL6pN04d+LeqRmjR474cOV/hALm5nRmuFKp3H21MiXYY2F/BeGczB4IERGpQyNSh14/mTVo9L0ZGTPeXvosh9P2bNc+JEI+IATtbBICIiRueAbdCdEbDGOmzhkzYXqVIIYMTAWnaecWkMCdVAy7WEWGx/VeuuJ9Zw9nHiNFbzxT9Ov5knmpoRMT/J2nnVuQhcWNfPGTGlFQYtrg77fscPZw5tGbqD8vl58urp8Y75cS7OE87dxCWK+URz74oYIWJqYN/uvAYWcPZx6V3rThdGFOWcPclJCUYE8naefWRPQdkvH2mpP5FfH9Bh06fc6BPat1BmQWtc7gwOHMgxW0Yzhz/nJgeMKeY5c5EaOQgNE9HCQJ5ESMee/TbxNS0uvqG5kcuoUKpe79vVdDPISLh0R4i5lzQgIARdqogUtWr/zw04lzFuoNzH1zWlOh1P18tjhKLp7Qw0/MY25VihCRcu/0+19d/czSV59Y+hpbuRjza9RrTxQMUMgm9wxw4zpyMmsegsNJnTw3feGymfMW/OfdjxzVLQIEgACR7RwEk2n/sIJ2AJ98vS65/5A6URzhHe8YW7OtEFwyeMDlKiIsNvHw8TMMD366qO7bI/lzUkMGhLGTDY4vlg558u1KrqzngOF5RSUMj36utGHvtaqJ8f6RcgsBZk5CKved/dZX16tVA8fcV1Vbx/DoWXlVWXlVc1JCw7yc61zRHp4BoQ+s/OafY2dHTM5QqTUd71DA5yJEECTZ5oEIQsBrwwvLSWAF3VGeffXtJS+8zAkfhYQsONW3hvBQKD16Dxo+ZvO2vxgbdN/VysM3ap4fHuUvZdkMGj18SuR9D6ePvvfE+UuMDXoov6a4Xjs9MUDKZ3M7ByGUPvOx2JFTBo8ce6Ow3VzGjoWmYXtOqcZgeqhviIjH3MT5bgiSHDb/eXF0cr/hY8qrajrYm1qrBwTQNI9u61Bq9I6Q2iqwgu4QMx9Z8vEX33HCRgDJ6Lq+PZDAg6MY/sCDcz9f8xMDw227UJpfq140KIxHusQHyTuyZ8qC1+6dOuOfQyecPRZNw99XKo0UPS7WhwGTqzVEpwwe9djLo8dPPHv5qrPHMlH0xuwimRtvTIyvS9w8QNyQcb0mzR88amx+cWlH+pG4NbvZtX0QhIeEuQr0LvG96qTMXfTiz79uJ0OGAGJz+nAnXDdO+MgnFz/79dqfnTrO9gulWoNpbmqoi6inJiS+wcOeeX/Ww4/uP3rKqQP9c7XCQ8gZGu5aOf4DonpMeO7tqdNnXLqW57xRaBp+Plsc4y0eyJJRqz1CeqYMnPfC6PGTikrL7e6kUaMDhAAR7R1MutlhBW0nS19/d91PW8iQoe2nJWQPkscJG77wiad3/L3fSSPsv1ZV3qib3ifI9W4ehO5eQ554c/pD887lOktJHcyvEXDJ1GBXLEHiHRx23zMrJ97/QGl5pZOG2HGxLMRDmBTkigGNvhE9+j349Ijxk+oblfb1wOVyEEJEOyCEeM6vdNUCVtD28NXan9/98GMyZLAraucmSD4ZOnTmrNkXLl9zeN9niurOl9Y/nBbqqjcPIplf/0denXD/9Ioax2+anS1tUOqMQ1xs8tgaX0XU8PnPjZ08Tatz/Fxv77VKPocY5MK3HxjXO+GejNFTHjBRlB2XkwRh3ouDYNCghxW0zVy5lv/4E09xQoe268ruGiCeWOOZOHzMeMc6n1Wr9H9eLn+sf5hLWTbuxjM4MnLc7HumP+TYbitVussVjaOjXT3JnKJncsyQezIWPOnYbvOqVUV1mlExrp4LJSpthDAoZtFLr9lxrU5vAgRm/KC1lhL2OxCsoG2Doqj00ePJgH4MxKF0HOTmXUN5jp3sMCVloug1xwrmpoTynB+H0nGCeg/Ui32WrHjHUR3qTdRfuRVjY1xlV9A8SWMmF1XXf/rDBkd1qNQZ/7xcfn9iYCe4eYC0qfP3HTyy+Y+/bb1QKuIjRCCS0/ZBkB5ivEnoqjww/+kKrQCJXH0C1QIhjztw6Nj3P211SG+/nitJU3gFuHeCH6cm+ty/8OdNW46eveiQ3v65Ujkg1MtdwJwbbAeZ8NR/Pnj/Q0flrPjlXMmEeH8mQ1E6BELjnn79paVLaxtsi95qUOsAEKB2DoBapdY5ErcBVtA2cOpszi+//EL4JLAtiG0Qgf2eeeaZjpsjS+q1hXWawS7mt2AeRBD95r4495HHOh5lV1SvoQEiZOxEo9gHly8c++jzcxc+0fGuzpXWe4v5wR7MTR47jlDqmXjf7PlPPWfTVQK+2U1CghAx+AuNFbQNTM2YSwb0bT+Tt4uCuG5Kjv9Djz7TwX5+Ol2YkeSsHI/OwyMwnB8U/fb/1nSkE4qm91+vTo/oTD9OTSh69lVS5CbbV/qt0Rmpg3nVI6M6zcKxhZiBoy7kXjtwzAafS6OJBoTaSWZHIIRMFHMh9Z1M17DI599vLCyvQ26d7zMKAIQ85pdffskrKLK7hxM3axVeItbDBe2j531zv/r8U2UH4oDPFNfH+YiZzLPhQEbPX/La8uWUXS4NTey/XjU4XN4pNh7uJn3es0uef8n69iYT3VxRpc0DCIMRK2jX4/XXVxB+rpsh3gKIILzjH3niefuupgH2XKkcF9dZSy5xBG6B/cYue+9T+y6naPpiRWPvAHfHSsUYEi/vwISULzI32ne51mjKr1El+EsdKxVjyILCaJHH1r/2WNleIuQhRBAEp80DEaSUwZqEWEFbxXfrf6lWUYjnEkmH7YPwCN2/f1+xXRFW2UV1Ud4iN1bzLXSQyPSJv/z8o8YuQ/y50oYePpI2azV1FgZMnfPZp5/Yd+2hGzUDFLJOfPMA/ac/uvLtVVY2VmoNgAAI1OaBAOrVOBeHi7HizbcJ73i2peggiJBFP/Xi63ZcuftK5ejYzjp9boLk8rx7Dfrg2/W2XkjTcL6sIbHTTp+bcJO4y8JiN+2w2RJtMFFXKpUJfp11+tyEu2+gFrj/nrAq0aObgIMQQgTZ5gEkIRHynC1wC1hBWyYn91pRaTnDWZ6dAeERvmfXTlv9Gcoate5CroTVbG0OISp94rofvrf1qpt1miB3IQMJ+J1Nv/sy/vfll7ZedbG8McFP2hncvi3Q594ZH3z6uTUttXoKEAKCaPtAhFbPXNFYrKAts/zNDwhP1y3QaQMEqaRF6zdvt+mig3nVA8NYzqTqEARSLz0pzL5kW6a3nPKGeN9ObNpqwTskvLi0rLKm1qarzhTXJwZ07ulzE8EJfU+eOGFVVC1C5r047k7wsGPHjvDwcIVCkZmZecdbq1evjo6OjoyMfOYZe9yosIK2zL49fxPuwWxL4RgIz7CPP/vK+vY0DZfKG+O6hIYCAEXaqE++vfMrZAaDia7VGHyYrRHjPHoMHPnNhk3Wt2/UGQFA2nkCc8wT2jtt/dY/LDYTckmzyZII4e3OPEajcfHixfv27Tt58uSyZcsaGhpa3iorK/vkk0/Onj17+fLlrKyskydP2iozVtAWyM7JrVcbgegin1Ek8r2aa0Ml7Jt1aoWnW6eIbLaGwN4Ds/b+Y337wjq1wpOdQiHOIGHImN+327B+yq1Q9ugqv80AEN1/xJZff7PYTK03msnWD027iK04fvx4dHR0aGioXC5PT0/fs+eWuwhN0yaTSafTGQwGmqZlMpv96LGCtsDXazYgSSDbUjgSpZH77xFr/fYvlTXG+XWdryiHJ9CYUFlltZXtC+o0oZ6dKXbOPFKZT2lZmdForQk1r1oVwVIdL2fgGx6Xe/G8xWYiAc9cLg5EkJQmLy9Pq20O+C4pKQkMbFYRQUFBxcW3Ktr4+/svWbIkJCQkICBg4sSJYWFhtsqMFbQF9u7bhyT+bEvhSJA44Aer17mXKxpjfLqOggYAvx59t/y528rGJQ1af0mnjM1pj6CYnnsPHbWycbVaL3NjzmPB6SDE9/S9dP2G+VYqrRGgXRs0pWu8dObYO++8c/r06ab2NE23LiPb+nV9ff22bdvy8vIKCwt37Nhx6pTNRSSwgrZAWUkR4nduF6s7QG7yw0ePWdlYbTB1Af+N1nhHJhywTkMZTBSXQJ3a/fluguMSrVTQNV1MOwMAgH90z10HDplvw+EghBAiiDYP0s2zZ/+RX3311YABA5raBwYGtsyai4uL/f1vzeeysrKCg4PlcrlYLB4+fPixY9Z+71rACtoc9Q2NDWpHJlN2BZDAvbyk0JqWdRqDJ4Mun8zgpYi9dP6sNS2r1QaZqKvdfkBU/Llsq9yBK5V6X0kX2R1twS+yx8lTp8234RCEmWR2CAF5e8L+1NTU3NzcmzdvVldX79u3b8SIEQCwdetWmqaDgoKOHz+uVCp1Ol1WVlZ0dLStAjtTQSckAEFAQcGtM4MGAUKQnd38buubHj8etFpz/5U7jvXrISEB1t8Vd/Dxx9CnD3C5sHBhGyLl54ObG4wda+UdZB05gfhdwcfodlBdo9oab+jSBq2/tKt9RXluYrXSqmJINWq9V5f7fZIFhtwsyLemZaVKJ2cwppkZvILC8q5ZqDGka7LRt698NLrbjPgcDmf16tXp6elJSUkrVqxwd3cHgIyMDJPJ1KdPnxkzZiQmJsbHxw8YMGDkyJG2CuzkGXRU1C0dmpcHxcXAa/WJX7MGNJrmY+tWEAhu/SmTwc6dza9VqjZOzpzZ9ohBQfDGG/DAA22/+9RTkJxsvfhHjmcjfpeywDaBOMLLVy3X6ytr0Pp1LQtsE1oKrPGHrVHrvdy6iPdOCwRBqrVWRSpXKnXe4q72+yTykFVVWMh2IBZw27dwIAIhj7vWVRMmTMjLyysoKJg7d27TGa1Wy+FwAOCNN964fv36tWvXPvjgAzsEdrKCnjUL1q1rfp2ZCbNm3ebjzeWCQNB8cLkAcOvP1u+6ubVxkmhH8qlT4b77wKOtqL/ffgOE4N57rRf/RkEBcLuOl9UtuMIcK2oV1moMnl3OCgkAfIlnQYnlnCRKfVezvzdBkKTJZNmRo0FnlAq64O1rjRay+jV70bU7g+5KuTji4kAshuPHAQDWr4eHHFwgzgZUKnjpJfjoI5suqqmuRp2htJWtII6guNRylQ2t0STgdsFdCr7EvbzCcsVrnZHqpAk2zSMQS6usqKWrN1I8BqujMol5+x6fQ6J2wrwJAhACIYNZw5z/AGbPhnXr4OBB8PaGO2zkjz8OcnnzsXKlc8V44w2YORPCw226SKVSA9EFJxFAcGusCPnVGSgBpxNnsGsPnlBUVWtZQxlMXVNDCUSSqlrLT58G6DIBSq3h8gXmqwtRQJuL9CaQ/Xm1bcf52icjA3r1AqUSZs++862334bJk5tfS525F3fxIvz2G0qAhUUAACAASURBVJy1au++NTq9HhBTGgoRd8f4O20spFJbrqumNVJ8pqaQFEUBY3dPEBqd5cLMBoritmdJczQEQRj1Ha1JZu1YJMFkXWproWlmPv+IIDRavVDQ7srYYKIAzH0Ymxowg/M/f97ekJwMGze2sXHn6QlBQc2HUxV0VhYUFYFCAX5+8NZbsG8fKBTWXMfkDILwSUBiP8aG41gxN+QQzFX3uXHjhljG2O0jwgrvZgIhqsOVDK1kypQpJ//cwsxYAOBqvt0RERHVxfkMDYaQ+dsX83lmCqogxGhyEkYmCJ99Bvv3g9ddGdEMBtBqmw+9XXb31j007csbjaDVgskEJhNotWA0AgDMmQPXr0N2NmRnw+LFMHAgWOcxLhKJgGJqrsFkqUPK6OlpOXuqgENorQ4L7iCnb5RxBAztxxq1Gi8PyxMCHknomZorTZs2TZV7QqdWMTCWXqP2lLqWb1K/fv1KLtu8wLUPg1YjlYjNNFDpjAigPQ2NANVrmYuNYEQpKBTQr18b5+fPB6Gw+UhNtafn1j0sWAAA8PrrIBTCl1/Ct9+CUAj/+Q8AgFAIfn7Nh1gMPB74WpV+XiqVgom5HVvGoE0GucxyBlEBl9QamNBQNTU1fBFz/uYGjdLL3bUUNEmSjz322NFtGxgYS6Ns9JZ5WmzGIZCRqfVT//79S3LPMTMWTVGEWcuVkE+AmWSjBBLzmZtBO9MGfeFCGyf/P8NI2++2UFVl+WSbPbz5Jrz5prmely6FpUvNNWhFQEAAbbjoYstBR2BQR0YoLLZy45EqRoyV58+f9wxhLuO2uq5KEWw5AZaAS2gMlDtTXjwzZ8788ssheu1DPIFz0zMZdFqRm+XFipBLqvUmZjztevToUXXzOgMDAYCQa2FXSW+kEZizh+uYWlYCDvU2T0KPaNrAxKqTYWiDOrlXD4vNfCX8skYmdq4uXLjgEcScgjbqtTJ3y2t8TyG3VsPc+onD4Tz88MPHd9hZ2tV6BNZt/MrceFUqpvYtCSI+SK7Xqp09kMlo4PIsuPbT0FTUu91ZNHM1vbGCNk/6wFTQ1rMtheORunHFIstzKD+JoLzBsrNHx7lw4YJncAQDAwEATVFivlXRN15uvBpmM7HMnj27+vQ+p7pzaBrrpe5WFW/zFvOqVMz9PvXt27fsqg2Zyu2jtuRmiMJCzk8hlzBfUKVr+UF3ZnrERNIGp/+qM41J7yayao/ITypgZgadm5sr9QthYCAAaCgtCAmz6sdA5sarZjBmDAC4XO7s2bNP7NzsvCFKrl3q0bOXNS29xfxKpmbQ0LxPmO3sUSryLvfslWi+jVpvYeNBbYWPpqPACtoC3nIvMHWphHa0uiouPsGalhwCGSjK2a5mNE3nlNQSHIY2XqrzL/fu09ualhI+p0HLtL/w/Pnzs3dvNxmd9ZEruZqTktTHmpa+Yn4pI+unJtLS0kpyze5LOYLy65cGplrIxiPikwhBe7k4EIHEXc3NrjPTO6kvrbKct6ETQanKJ44fZ2VjhadbQY1z1xCFhYUMekBD2aVTk8dam1RMwucw6VMFAHw+/9lH557661cn9X/j7IkJo4db05JLEgDAmB+Lt7e3HDl9tVp69cLowWnm26j1JjCXiqOpJhZDYAVtgWmTJ1DKUralcCS0snz2AxOtbBzrK7lY3uhUeS5cuODBlAEaAOqL8wb1tbDIbSHUU3izTuNUee7m0UcfPfXXLyaj47UAZTLROmWgn4+V7RVeonwn/zy3Jjw8vL682HI7e9E21nu5SwR8CzlUSQJQ+wBCJFPxpYAVtEXmZEyiVRVsS+E4jBo/H7mHFV7ATcT6SC45WUGfP3/eM5ghF47awmuxcfHI6gBRhafbDQY1VBNCofCpeQ9m77ahuquVXM8+mtJ/sPXto+SiK5VW5c52CGlpacXODFe5dnz/iNGW147NCfvb3SREXA6DAcaMjdRJ4XI4sbFxtKaGbUEcA1V/c/z48da3d+ORPJKoceZeWU5OjkcQQzPomyf2zpg2xfr2HkKuUmdkbJnfwuOPP358x88U5WB/25ysXXNnTrO+fYin281aNWPx7mlpaaXODFe5cmTvglmWb1/XlI+0/breTTYQZsAK2jJPPL6AqrGcPblTQNXmvbZ0sU2X9Fd4Hcl34u9Tfn6+WM5IWV6aLjl/dP60CTZdFOMtzmVwFtmESCRaOGvaub1/OLBPg05be/Pa8AFtxfS2AwKIlIuvVTEUCpCYmFhxI9dJnavragTIFBuusNhSLOCYS9hPgLsQbxK6Eo/PyxBDIzh6OsM8tLY2PEwR6G9VmHsLfYI8sovrnTSJMhqNV6vUzKQxK805kTZgEN9SnMId9HC+Fb5NnnjiiaPbfqRph03ezx/4695JNqwemkgK8jhVZDk1q0PgcrmR3hKTwSnLtQt7t2e0V4bpdpQ6o5npMwA0MujbgxW0ZQiCGD1uPFXLUCiq86Aqcpa98pKtV3EIFOMjPlvslICdq1evMuYBfWnXxteet231AABCLinhc5h0OGtCKpXOmzbh/IG/HNXhyZ2bX1i0wNar5CKe3khVMxWxkpSUVHb9ksO7pYzGywd3PT1/ljWN+RwSAWq/4hXBtxQs7kCwgraKLz96i6q5CkwZ45wBbVB7S8g5MybZce3YWN9dl53ia8iYC0fNzSsB3p5J8TaXVQaAAaFeR29aznDvcJYsWXL410yHTKKvHM9KTU3x95Hbce2wKO+svLZy4ziBfv36OcMMfSnrz8lTpojcrEpyQtM0QmYcOQCHerscMk+PsWPHUbWd2BJNlZ35z6uv2HetmM8J9hTmlDU4ViRozsLBhII++8s377yx3L5rPYRckkAVSubC6ppwd3efNXHcpUN7O9gPTdMHNn771jJrc4TdQYiHsFZjYMYfPC0tzeF5R01Gw5mdG1e+9IyV7Y0UBe37QQOAwVJVQweCFbS1/PDlaqo6l7n00A6F1tb5uvOeXGB/TcjxPfy2nS91eP7+CxcuMOBjV375dGiAz9DUJLt7GKTw2nedoVlka5555pmDW743X0PPIuf27Rw0aGBESLDdPYyM8t55iYlwraCgoMZqBw+U/eemB2ZkeHm4W9lexOcgAiGy3UMiZK4MHlbQ1uIj93zqySdN5QylFXcsppLja7/9siM9SAXc1FDPf3Id7BJeUVEhkFrOTdwRKJPx9KbPv/vkvY504uXG85MIchjfLZTJZPePSb9yPMvuHnRq1eFf165euawjYoR4unFJghmf6H5x4coayyV9rURVW5V/9J+3li6x4ZLmSML2LBx4k9BV+fi/y70Fhk7nE03VXE0fPGjUsIEd7Gd4lPeZ4joH7hep1eq8eqd/1nN2ZD44c2ZUqP3zxyYGhHqdKqrTMri8beK5557L+vk7uy/f9e37L734ovXzx/YYF+u7+2olAyn809LSHJi8f8/X/33v3XctRg+2pqmSfbuh3gi54Wx2LsvWzT+Zio52Ipc7Wt8o1hZu/fHbjndFIPRgcvCaYwUmB22WXrx40SPQtjrrtlJ945I6/8J7rz7X8a64JBoSLvvTOZulZvD19b1vaL+rpw7Zce3VU4e4evWiORkdF0PEIwcqvH6/WNbxrszjwHCVC3u3x4SHTho9zKarDCbafKi33sTcNiFW0LbRPyXx+eeWmEqOsy2IddAm082DP67PdJeaK8JmPSGebsnBHr+dK3FIb8524TBoVMfWvvfbxnWkFRVyrUHh6SYX8U4y5RfcwosvvvjvRpt/YhtrKrPWfbY50/7Z9x0kBrjTNJ1d4twM6cnJyWXXLna8n5ri/Ct7f9vwxUe2Xthk8TeTLIm+y49jx44d4eHhCoUiMzPzjrcKCwuHDx/u4+MTFxen1drsrIkVtM28u2JpYkwoVe2skCcHYio6+vSTT9wzaogD+xwe5V2h1Gc7wi3aqTuENEX9+8Xy/765MibMkX7WgxSy69Wq4npG3aIDAgJG9Uu8cdaGaYFRr/v5rec+/+xTHyvKD1rPfT38jhXUONWhRSgUhki5lKlDi1SdWvnnJ6/9tG6tm9DmkmUCLokQAIHaOe4smmU0GhcvXrxv376TJ08uW7asoeE2Z6f58+fPnDmzoqLi77//5nBs3l3ECtoejuzd7sOppxudmHmr41Dl5wal9vpo1X8c3vPDaaH/5FZ0fMsoJyfHI9BCeQu7ObH+wwnjRs+fbltgt0UQggk9/PZer2TY6+6ll1468NPXVjamaeqX91558olFowb3d6wYXJKY0Sdoy7mSWo0Tve569uxZmX/F7stNRsPv7774zlsrkxNi7bhcbaDAXCqO5l3EFo4fPx4dHR0aGiqXy9PT0/fs2dPy1o0bN/Lz8x955BEACA4OxgqaIfg8XvbxA1L1NZdNdEdV50b5u+3/Y5MzOueRxOMDw34+U1RY26FUnKdvlHEElitv2UH2L19H+kg+W2mn37d5hFxyUrz/risVTKaKDgkJGdwruiDntMWWNE1v+2jF8EH9lzxsv1elGdwF3Pt7Bfx4ulDptLTIHdknpClqxwdLH3t47syJ1iY9vwMRj0AEIkiizQOANjRUnTp1SqVqTlFSUlISGNhcgzgoKKi4+Na87dq1a0FBQVOmTOnTp8/rr79uhzBYQduJj1x25N894oYcWun0bRNboaouh0gN2Ud2W59X01bEfM6igeFrTxTYnc6/urqaL+6oa0GbZP/ylR/XsD3T2vmmHUj4nHExvtsvljl1InkHL7/88oEfLdwUTVPbP17RJy7yg9dfdp4kPmL++B5+G04VNjqn+JPdCtpkNGx/74Wp4+95YeE8u0fXGCkA1J4N2qhuyD13+uuvv754sdlQTtN06y9a69dGo/Ho0aOrVq06cuTInj17/vrL5sB9rKDtJzpCcf70EQ/tVar+Jtuy3IIqy4714+WeOWxrViBbkYl4Tw2J2HC68EKpPRGGzjBA0xR1bO274Z6CvzaucWzPdyMX8e7r4ff7xbISptJ0hIeHp0YHF7VfF8po0P/81vODknt98c4bzhYm1NNtUkLA+lOFzigsGxERUVdq83dKr1FtfWvx7Aemvvni0x0ZnUSEmVBvvsQzKX3sV199lZKS0tQ+MDCwZdZcXFzs738rNWNQUFBMTExMTIxAIBg9evT58+dtFQYr6A4RFOB3+dyJAG4NVWHzv97xUCZTQVZ6atz54/u4XCaCndwF3GeGRu66XH7A9ii78+fPOzbIW69W7lv9/Jj+Sb9++4kDuzWDh4A7tWfAvutVjOUjffnll/dv+KLNtxqrK39Y+sjs6VPeW25zPiz78JXwp/cO3HS2+Hq1g/ORIoT6RARpG23YiK4rK9r82uPLXnzu5SdsTgh1BxwOIAQkgdo8CAJxb3cKSk1Nzc3NvXnzZnV19b59+0aMGAEAW7dupWk6ISGBoqiioiKTyfTvv//GxcXZKgxW0B1F5ulx4+KJ0Wlxpvx9YGI6XUMLtLbOeH3Xs08u2L1tg/MsG3cj5JJL0iNr1fr/HczTGGzYeXfsDLrqes7ud59+benzH73OkHpqQsQjH0gMLKzT7LhUzkAQR0xMTGKId+n1y3ecv376yKY3nvri4w+ffcz+pb0dyNx4D6eGHrtZ++flcsfm9U9NTbXeypF76O/dH7+6ZcPaOfc7YE9Yb6SQ2Vwc2ts/5xwOZ/Xq1enp6UlJSStWrHB3dweAjIwMk8mEEPr4449Hjx4dFxcXHx9vU62M5s47fj8YkiR3/pL56TfrFz/zLOmbiKQdDVqzDZqmqi6609Xb9+wc2M/+dBN2QyI0qWdAdnH9B/uu3p8YGOsrseaqK1euBA10gDahjIbz23/QFece3v1HZEhQxzu0FQ6BRkZ5Xyhr/Pls8YhIbz+JDUFrdvDKK6/MfPLFjOXN7r16reafNR8R6rpD+/7xkXk5deg24XOIjD5Bh/Or1xwvmBjv7y12zO2npaX9tebX8L4WCnRpG+v3fve+v4fb6cMHpGKRQ4YWcTkEAg7R9iyHJJDkruXphAkTJky47behxeV52LBhLdZqO8AzaIfx1IJZBVfO9/CmTfn7aT1Da15aVWG8/td96b1LblxiRTu30DvQfcnQyKMFNR8duGaxRBZN0xdL6giyo/OD0pzju95aOKxP7IVDe1nRzi0k+Ekm9PA7VVS341K5TSsJW4mPj4/xFlUUXAOAnIP/fPf8nEmjhh7441dWtHMTCGCgQjYpIeCPS+W/55TpHRENn5qaWnq1XWs7ANA0dWHPtu1vPfXMY3N3bV7vKO0MAGqDCdqN8waE7nSzcyp4Bu1IggN8zx7dvX7T9iXPPl9r4JPeCcC1KgWtHdDaWqrsbFiw388HdycnxjtpFJsQ8zlzU0NzK5RfHr4R6yMZFeMj4bf9ASssLBTJbCvscgdVeRfP/vpNTITi+N4/QwMZqZhlCQmfc2+c7/Vq1ZbzJVEycZ9Adz7HKROgV1555f6HFwFN9+2bfPrgfpmnU5xhbEUu4s1NCckuqf/2WH7vQI+UYA9uBwI4pVJpgICmaQqhuzqh6esnso7+smbMmDEbj/0rdnOwpyaXRKj90t0EQk56rG2CFbTjmTV9wqzpE9764IvVH62u1ZKELAYJHTm7oRtLqKpLoYG+H67/etK9Ix3Ys0OI8RG/PDLmTFHd/w7mKbzchkV6+9616j9//rx9Qd40RRVlH8zdvSU8JOiXH75M7dXDESI7kgiZKMzL7VKFcvO5klBPYWKAu7SdXyk7MFH01SrVRVo+LePBeVPHR4eFOqpnR9E7wD3BT3qqqO6bYwXxvpLkYA8xz87bj42NrSnKlwXfytZi0Gkv//vnub9/HTho0OG/dwQH+DlI6ttomiiT7WzkEMxm7McK2lm8+tzjrz73+DfrNr/3wUfXrh0n3EOQRyji2r8QozU1VH0B3VjSLy1t1Q+b0welOlBax4IAkoI8koI8csoatp4rqdcaUkI8+wZ7SgXNnzebdwhpuup6TsGJvXXXsgcMHf7plg09IhXOkNwhEAjF+0p6+EiuVat2X600mKgevpIouVhg78yLouniem1OeWOVSh8pF01O8J/Z5wnHyuxAOATqF+KZEuxxvrRhy9kSAkFigHusj8TWiWe/fv225pyVBYdTRmPB+eO5B/9WleXfN/n+77P2ODZ+/Q4MJrppk7BNEICBwWRJWEE7lwUPTVvw0LTyiqqV73++bdvW4vxS5CZDIh8klCG+1MLFNEVr62hNNa0so3UNsbFxcx5ftGThHD7fuQ7ODiTeTxrvJ9UYTCdu1v5wvECpM4Z6ucX6SE6eOevR30KKNZNeV1d0verGpYrLZ4x1ZbE9+7w4/4GMCWs4JHPJHjsCQhAlF0XJRRqD6WJ54+8Xy/QmKkAqCPYQyt147gKueV8brcFUrTaUNGhv1qk1BipAKugd4O7sHUgHQiCUGOCeGODeoDVml9StP11IUXSop5vCy81bzPcQcs17Gqn0xuC43tkfPHLlyF6kU6YOGPzh8heGD2BiUuLGJwmEOO2YOEgCifnMfQKxgmYCXx/5Z+8u/+zd5UajcfO2v7bu2HXh/LnykjO1jWpAJOLwgGilc0062qgDoOWe0uDQsKSkwQ9On5Q+0HXnyxYRcskhEfIhEXKKpgtq1LmVSsor+MyalWq9icMX8MXuHL4QIURy+UadRqdq0CnraYrylIgiYmJHJyXd/+LDPWOYKIvlJIRcMjnIIznIg6LpkgZtUb02t0JZrzXQACSBBByCTxJckjBStN5E6U2UwURRNAi5pJeQ6ycV3BPrK2SwSqnDkQo4Q8LlQ8LlRoouqFUX1KrPFNfVaQwAwCUJPocQcEgeSRgoSmuk9EZKb6IAQMQj5V5By5cvvzd9QJC/U0wZ7aHVm5oCVdp+GyG1Dm8SdlE4HE7G1PEZU2+5QzY0KotLygpLyurqGkkO4S3z8vP1DgsJ5HK5LMrpJAiEwmSiMJlo7OfvNJ1pVKlLKyqqahvq6pUkiXxkMpmHJMjfl2hn/tKpIRAKchcGud/aNzZStMFE6YyUzkRxCcQjCR6H4DkoOaqrwSFQhEwUIbtl5TOYKL2J0hkpjYHicwg+h+CTBK+1JSRuCvNy8jgIIWjvIZCoOaM/M2AFzTJSiVgaExkX4/S6fK6JROQmCVNEOyulnavDIRCHIDv1BLkjcEmCSxIiF7PYmShACBHt+EEjhEwMFtXBChqDwWBuQSO6KR10myAEHazhaxNYQWMwGMwt+CSJELTnB00SSMDgigcraAwGg7mFzmhCcHd4zC2cGil6B1hB34KiqMbGRralYAeTyURRVHk50xVRXQS1Wm00Grvt7Tc2Nup0um57+w0NDfxWZb8FHJIggCTbzcXhhmfQrCCXyysqXLRCirOhKIqm6aKiIrYFYQej0WgwGLrt7ev1er1e321v32g0yuXylj+b4lDas0ETAHoGdwmxgr5FaKjLBc4yBkVRGo2mV69ebAvCDvX19ZcvX05OTmZbEHYoLi4uLy9PSmIz2RaLVFZW6nStEgUjRCBEtOMH3ZS2nyHJsILGYDCY1vCI5oT9bb5LENCRJFC2ghU0BoPB3EJP0U0Vr9p8FwFySD5VK8EKGoPBYG4h5BLmE/aLeHiTEIPBYNhAa6AQtG+Dxm52mK7N1RuFO//ed/J0dm7ulYrKCnVjvV6vQwBNXwqBUCSWuIeEhvRKSBjQL/ne0elCQadJ4WYNSp2xQqmvUumqVPoGrbEpO9K2bz7MPX0MAERiiczbJyQ0JD4uNiWxZ1pSLy6nS31J67WGSqW+WqWvVOmUOpPORBlM1E/vLy+6fpkgCJFYKvf1Cw0J7tmjR0qfXim94tsLuXYeXAIBgvaSwSACONgGjelilFZUfvzF2j/+2FlYkNeopcBNhgQeiCcBbjjyEgAiTEVHCXksEngoKWOlXn3jYu3+0zs+/no9ra2VeUiSUtJmz5yeMeWeTppESW0wXatS5VWrKpU6MZ/rI+Z5i/m9A9ylAk5TdqQ/3yse/9JqodRDr1E1VpU3Vlf8febqht92VhZc83SXDhw8dMbUSUNSerN9H3bSqDM23X61Su8h5PqI+TIRL9nTQyrgcEmCRxIbqm5Oe+sHANCqGpXVFQ2VpduOnPvu51+rCm94y7yGDhv+4LTJfRNiGRKXAAIhsp1QFRIY/cnAChrjRPQGw6qPvvr++7U3C4sJaSCSBqGA4RYMeAQH8aXAlyJxcyGresq492zp7n9feHzRoqSU/q+9/OywQf2cL7sDMFF0bqUyu7jeYKJifMQDFF4+lmqq8oQiWXC4LDhc0Tut6Yxeo8o7ffjp/6xUlhYMHz322ccfiY1QOF10R2AwURfLG8+VNCAEMT6S9Ai5zFJiJIFIIhBJ5CER4cmDms5oGutvnD4096kX9HUV99w3Yclj8xVOrnBmMgG0n7AfEDBQvr0FrKAxTqGqpvaxJa/88+cfKo6M8IzkRPa0vy+Cg6TBpDRYS9OHLhePvG+Gr5fkP68uXTR/puPkdTA6I3W0oDa3ojHGR3xPnK+H0P7ksTyhKHbgqNiBo0wG/dXjB+6bOdfH02PZyy+MHTrQgQI7FrXedDi/5kaNKt5POrmnv7gDRb+EEvceQ+/pMfQeg06be3j38PumhitCV7zy0sC+zlpP8Js2Cc1FEjK3jOuUC0aMK9OoVE96cKFPSOy2Axe0gcNIvz6IL3FM1wghaRCpGFYpiHnqpZU+oXHfrv/FMT07Dr2J2nO1MvNkoYeQ80iaYmiEvCPauTUklxc7cNQDr3/Rc/IjL735fsrwcfuPnnJIzw5EYzDtvFS+Mbso0F3wSJpigMKrI9q5NVy+IGHY+JlvfReaPmXBMy8NGT/11PlLDun5DvRGqilhfzuAxsCcmx1W0BhH8s7H3wRFxO3IOseJHEd4RQFyikMS4knIoP614p6PPf1STJ9Bl67kOWMUO7hY3vj98Zs+Yv4jaaGJAe5OMlbKQyLue/bt5BlPLnz2pQmzHi6vqnHKMDZCA5wuqlt3sjBSLpqfGhrnK3GSqdY/KmHKKx9FjcmYPnfBrCeebVCqHNs/l0QIAdnOQSBgsqo3VtAYx1BYUh6ZOPCVle9rfAcR8liz6cAcA+KJydAheUppfFL/p5eudPZw5lHqjOtPFeXXqOemhPT0lzKwjSQPDr9/2afCyOTkwcO/+3mr8wc0R63G8MPxm3Va4/x+odHeYgZG9I+Mz1j5tVLk3ztt8K+79jmwZ4oC4v+jve8+ECAGi3pjBY1xBOs2bY/vnVqgFJPBA4Fk1CsOif04EWP/t2ZjZK8BVTV1TA7dwo0a9frTRUPCZffE+TI5vQKAqH7DMlZ+/fE3ayc+tECr1zM5dAuXyhs3ZxffE+c7PFLeXnyHU0AoYdj4ia9+8sqKt+Y8/aLJQTmMaKABgCBQewd1V8L+HTt2hIeHKxSKzMzMuztUq9WhoaFLliyxQxisoDEdZc7jL8x5ZJHWdyCSBrIjAUGSgf3ylW6RPfpkHWHaLHvgetXRgpo5fYNDPIWWWzsBvkgy4blVtHdY8uAReYWMZqSjadh5qfxyhXJeaogvSxXHRR6y+5d9WqKBfiPHVVTXdrxDHkkihMh2DgIB7/a68kajcfHixfv27Tt58uSyZcsaGhru6HDVqlX9+tnpd4QVNMZ+aJoeOm7a+i07OOEjgSNgVxjCPUTllZQ+atzPW3cyMyJNw/acMoOJntEniPW6gr1GTe438+lR90w86Zyts7sxUvSms8VebtzJPf2ZzB/UJv2mzIsdPWPQyLHXCm52sCsDZUIICKLtA92VbvT48ePR0dGhoaFyuTw9PX3Pnj2t383Ly8vJybnnnnvsEwYraIydmExUz34jDp3NI4P6M2BxtgbEd+coRmTMfvh/a35y9lgmit50tthHzB8Z7c10rFs7BEQnjHt21eQHHtxz+Lizx9IZqR9PF8X5StJCvZw9lpUoeqcNCte6swAAIABJREFUfeTlMeMnZ1+62pF+uCRBIEQSRJsHTVENVWWnTp2qr69val9SUhIY2Lx2DAoKKi4ubt3biy++uGrVKruFcYnvFabTQdN0n0GjLxcrCd9EtmW5Ha6QEzbiqcXPrtngRA88moZfzpXEeIvTQj2dN4odePgF37/sk3mPPn7wZLbzRjFS9Mbs4rRQz17+UueNYge+4TFjlrw1Zdr0nCvX7O6kKVdde052WlXDlUs5mzdvvnateQiaplunvmv9eteuXSEhITExMXYLgxU0xh6G35dxMb+G8ElgW5C2IHmcsOGPPLZo5+4sJ42w83J5kIewd6C7k/rvCCIP2aQX35s9d/7lvAJn9E/T8Ou5kqRAd2a8NWzFKyB01JNvTLx/RmlFlX09EAiaqnq3eUg9ZWlDR/73v/9tKe8QGBjYMmsuLi72978V6Hj06NFNmzYpFIrnn39+zZo1b7zxhs3C2HcPmO7Mo0v+k3XkNOHnwqkhSD4ZOnTGzFm51/Id3ndWXjWfJAYoXGVpfzdSb/+Ri5ZPnDq9pv7ODauOs/NyeYinW08Xmzu3xjs0sv+DT42ZdL/OLrcWDkIEApJAbR4EAu7tQYapqam5ubk3b96srq7et2/fiBEjAGDr1q00Tb/22mtFRUX5+fnvv//+/Pnzly9fbqswWEFjbOPXHf98u2YNGTyAbUEsgHhijUev9NHjjUZHJofMr1HfrNWMiPZ2YJ/OwCcsJnHCnCmz5ju225yyBoOJcjXDzt2E9koNThn+4EJ7PNsMFNWUiqPNAwD0ptvc7DgczurVq9PT05OSklasWOHu7g4AGRkZJpMDPnhYQWNsoKKqZt7Dj5Ahg11kV9A8SORTqReNf2CeozpU6U1/5VZM6envIruC5olOG6bjS5e//6mjOqxRGw7n19wb5+uoDp1Kn7HTrhQUf/3jFlsv5HFIAkF7bnYkQoK7XN0nTJiQl5dXUFAwd+7cpjNarZbTKk/s3LlzP/roIzvuohN8zTCuw8j7HlC5RSCuiG1BrIXwjv979/6ffv3DIb1tu1A6LtbXjcGCGh1k2Jxn1q7NvOCIUHiaht8ulE5KYN+jznrGPrHsjZVvlpRX2nSVwURBO9Pn/59B41wcGNdjzYYtObnXCQ8F24LYBhmU9sSTTxsMxg72c7VSKRVwQlmKRrEPgsMZ9djLsx95rONdnSqqC/dy87aULtWl4AlFQx96avbjT9t0FYHAjB900+TaSQK3IQxjI2E6NUaj6fkXlhIBqWwLYjOIJ25AXgsWv9yRTgwmeu+1qpFRrm56vhu/iDiuLOCLdZs60onGYDpdVDcoXOYoqRgjrM+AstrGrbv2Wn8JAQgBIto5ECAmo9mxgsZYxcJnlzWAB+K6sS2IPRDePTIz11ZUVtvdw+H86tQQTwHb4YL2MXjmovfffUffgTXEnqtVw6O8Gc2z4TiGz3v2P8tscJ8w0jQAINTuwWTCfqygMZbRGwybN24gvOPYFsReEEnIYh95+iX7rtabqMsVysQAV/R6tgaBWKpIHfbeF9/Zd7lSZ6xQaiPlnWbj4Q6k3v6SoMjMX363sj2fJJAZNzsCmPydxgoaY5lnXnlTxfUBwjGJ51mB8Azf8fv2mrp6O649cbMuJdijc04fm0m+d8b3335jX763QzdqBoV1PuNGa/pNmffe++9b2dhA0e3PnhECpDPiTUKMK/Hj+nWEzP5wVZcAEYRH+PPL/mvrdRRN55Q1JLpk0KD18IQi37jkNZt+s/VCnZEqrNNEuWTQoPVIZD6kRPb3v8esacxByFyypLsCVZwKVtAYC6zfvL3BwAXSQrlP14fwitz262Zbr7pWpYqQi5jcuHcSSeOmffPNN7ZedaGsoWcAE/UHnE3SPTNWf/qZNS0poBGYy9l/VzpoJ4IVNMYCH378OeEVxbYUjoDk1WngTxsTdGQX13de63Nr3H0DaxpVhaXlNl11vrShp5/rRnVbT0BMz5yLF9UareWmNCAEBLR9IACsoDGugk5vyM4+jUQ+bAviGAjPsPc+/sL69jojpdSb5KJOv3poImbAyC/W2pCItU5j4JFEJwrMMU940sAfNm+z2IxDNuXiaPsgEPCwiQPjImzY8jty63zOv+2BJAHZp6wyRDaRV63qvN4LdxMzYORfO20IqrxSqYzzdVBFdhcgZsCorb9ZVtBGE5j1skMGCm8SYlyDjZt+JSQsFbJyBoisUxmvWp2HM69aHSHrlK7fbeIm9ayqrdVodVa2v1GjDvPqOrfvHRqZf91yLn+SADM2aISAS+Cq3hjXIPvMKSTuHMlxrIQQ+/2wwdoEOiUNWn9pZ4rttkhgbOKuA4esaUkDNGiNHsJO7Ft5N0J5wJmcXPNtqPaz9SOEECATg0ZorKAx7UJRVFVNXad2f74bJPLel3XQmpY6IyXgEJ3a/fluAqJ77T9slZGnRqWXdRXjewv+0T33HDxqvk3TA28vYX+TmYMxsILuPCgUcPIkkwMePn4G8bqOCbIJJJTlXbMwh2qiUqXzFnc5DRUZd/6sVdWwKlU6n653+1E9Tp0+bb4NYT5hP4FIBnPtYgXteowZA3dUXti2Dfz8YPlyCA1lUpCsIycQvyu4WN0GwdFprfC1AqhS6uWizpS8zRrcfQNLiwqtaVmp6oK3LwsMy7NkhqYsWTCMeJOwWzN3Lqxbd5uz5bp18OCDMH8+eN/lUGHsaBZNM5w9lwNdT0ED1Ku01myUVXXFNT5ChEpv1WemWqXvMv6FLQilHnU1FnJmcQmEoN1s/SQAj8GM2FhBux6TJkFNDfz7b/OftbWwYwfMnn2bicPPD955B5KSoGdP8PODzz+H6GgQi+G556CwEIYOBbEY7rsPVKpb3TZNwCUS6NMHDhywRpDSsjLE7VJbZE0gjvBqXr7FZkq9USrgWGzW6SBI0mDF73qjrmvevtZocYJMI9TuPiEgxGAqDqygXRChEKZPh8zM5j83bYLYWEhMvLNZVhYcOQIXLwIA/PorHD4M587BmjXwwAPw6adQVgY1NfDtt7fax8XB8eNQVweLFsH06aDRWBSkvq4OyK62yAUAIPmFRaUWW+mMFL/zVA+xHoFIWlldY7GZ3kgx6U/GJLRZI0ZTVW8StX00bRUyRtd8AJ2eOXNg8+ZmHZqZCXPmtNFm8WLg85t3lJcuBbkcwsNh8GAYPhx69QKxGKZOhTNnbrXPyABfXyBJWLAAhEK4csWiFBqNGsgu5cLRDMmtqbWc1k5vpHh3VZ/rAvDdRNVW3D4Ao+4KjMHlCzRmA76bK6q0dzD7b+mCn7+uwKBB4O0N27ZBXh6cOAEzZ7bRxt//1ms/v+YXbm63vVYqb7XJzISUFAgOBoUCSkqgqsqiFAaDAQFTH0aCZLAQLWpQqiw2MlAUY/X3SJI0GQ3MjIUIUtfhAmCOBSFEOaIGtlVjEYTOrIWHotstSNh00NjEgYHZsyEzE9atg9GjwbetUBGbfsevXoXFi2HNGigshPx8CAqyJuMLQSDGPPLJgBQmPUb4PMsrAxIhE1O1M/r27Vt0ySrvN4dAuph3d1JSUknuWcaGM3/7JEEg1LaPXZObHU43igGYPRt274ZvvmnbvmErDQ3g5gZRUQAAv/8ON25Yc5Gbmwgo15pqOQbKIJd5WmzF45B6pvaDRo4cmZ99hJmxdGqlzN21nHNGjx6dd/owM2MZtBqxm7n4dRNNmZk+A+CSVxgAUChgwABQqWDCBAf0lpwMGRnQpw+MGweHD0NCgjUXiURiMOkdMLqrYTL4+cgttuKTSGdXCRI76N27d3meVeEzHUevVsq8PCw245CEwcSQJhoyZMjNCwwFYdE0TZjd/CQQAgQEgdo+EGIyurQLutF0Hfbvv+3P/Pxbr8vK2n69ceOt1wsXwsKFt/58/31oqfqzapU14/v5+9P5111rMewIaIM6OjLcYjM3HqnWGz0ZSUaBEBqUGFNXVuThF+TssQw6rcjsFLIJNy6pNhjdGdklFggEPQJlmsZ6ocTpqbcFljZ+EQ2o/alryzyaGfAMGtMuMVERYLC8mdb5oE1eHpbX+F5uvCoVQxt3ADBq1KgbjFg5hNbVPJW5catVzK2fRowYkZ9tIUtGxzHqtQKhBdf+pooqZrIl3b19s2PHjvDwcIVCkdniHQsAADdu3Bg+fHhgYGBcXNxff/1lh8BYQWPaZWBaX1pbx7YUjoamvNytqrDnLeZXqazNzNlxRo0adeOM0+2w6voaT5ll8w4AyMX8KgYV9JgxY/JOW5VmryPUFBeEKiwsnjjmNwkR4txu4jAajYsXL963b9/JkyeXLVvW0NDQ8hZN02+//XZxcfGWLVvmzp1L2R4jjhU0pl1GDOlPa+0pg+3K0Lr6gGCrUprIRTwmNVRwcLCnqZF2sg9X2bWL8T3vCnpqC28Rr1LJ3O0nJCRUFlhO1txByq5fSuxt4fZNtLlIQgRgvH0Kffz48ejo6NDQULlcnp6evmfPnpa3wsPD09LSACA+Ph4AVCqb16NYQWPaReQmdBcLGHX7dD60uqpfaqo1LSV8Tr2GORMHACQnJ5ddu+jUIcquX0xLTbampY+YX660KquUoxjdP6mqMM+pQ5Rdy/m/9u47PopqewD4mdne0ns2hZBGTyihQ0gAaQYsgEH0ASp2wfdDfIqo4FP0iWJ9+tSHUgRFeaCiUg0EKdKRlmBI35BedzfbZu7vj0QIsL3MLni+n/kj2b1z5yzlZHLn3nNHDbXxt8+jKCsrCRmTsaK0ePfu3U1NTR3tq6qqoqM7N7VQKpUqlerGPnft2pWYmKhQOFwbEhM0siYppSdpt1Fc5uZC1NX33XOnnY2DZMJGLXd3kWPHji056dlh6PIzx3LGZdrTkkdTPJrWGTlaPwIA48aN8/QoR/WlC2OGDLLehhD4s+qGmUOrVZdcKtq9e3dFRcWf7QnVZVECdcMChfLy8qeeeuqzrnUX7IYJGlkzedIE0lbl7SjcSc43jBo60M7GCUHS4gatR+PpKjMzs+z3I57rnzEaBMQUFhxkZ/v4QGlpk+2aLe4ybty4Uk/+fNI0N4aHhvL5dj0jtbQhYUBg0Jix419//fW+fft2tIyOjr5y16xSqSK7LvEFaG5uzsnJ+eCDD1JTU52IGRM0suaxB2azt1CCJtr6+IQU+9t3D5FdrFPbbucmfn5+8QEio85TOfHS8QPDRo+xv30Stx8/JCQkUswyRk/9ynLx0O4JkybbbEZRFA1gqdwoDRR97T1yRkZGYWFheXl5Q0NDXl5ednY2AGzZsoUQYjAY7rzzzqeffrrjRSdggkbWhAYHKKMjiYG7/6UexTaXPjB3tv3t/cUCA8NqOfw1f9SoUeXnjnuo88KDux6YfY/97aP8xdVtOi4Xzo0cObLinI0dT5x28be8h+6dbkdDG5/3uml2fD5/1apVmZmZ/fv3X7Zsmb+/PwDk5uYyDLN79+78/PwlS5YolUqlUlnadSmDfTBBIxvm/O1vpLHI21G4A2FlbNNj88xVnrKsT6Tf2cutttu5ybhx40o8s+jZ0K7RNVwe1r+vQ2clh8r/4PAmety4ccWemWvYWl/tLxXHRkXYbNmxdTdNWzuuk5OTU1xcXFZWNmfOnI5XdDodn8+fNGmSyWSq/FN8fLyjYWOCRjY8//dH2bbKW2AuB9taOWDwcD7fsdWzvSP8znCYoDMyMqou/u6Jns/m/Xjn9JmOnpUW5X9Cxd1Uy2HDhlVeOGm7neNO7/zf3D+zp3UsIVaL2VEc7niFCRrZIhYJx46bwLaUeTsQV7H1F1a9sczRs0R8OlwhKmnk6FEhn88fkBSrbrJdDNYhhLBn9ny36NF5jp4YIBHwaaqmjaMFOwKBoH93pabZzROHTAZd5ckDD826y94TLNeDpsDMHbTnYIJGtn36/utsfYG3o3AJ0dT2TE1O6+XAE8IrRiYE7y/mbq7h2LFj3b7oufDg7uyx2cEBzlS6GN09eH8Jdx/fE4M8p3dumTHrXqHArroiNFBU597eZg/grkg6Jmhkjzhl1Ijhw9nmUm8H4jym+tQn77/l3Ln+YoGfmF/O1YQzt6/5JixzZOvaV59/xrnTIxRivYnlrC6H29d8G/W6wn3blix81M72rNUBDgAgtp4iuhEmaGSXDZ9/yNadBcLdfAY3Iq0V/dP6Dhvc3+keshJDdl2s5WY6Q3Jyslxba8+OCnY6tWPzlCm3R0eEOd3DuOTQny7UuCse6xITE+XtdW78+L/974t58x+2XgO6Kx4NFGWh1uifFUfdFZtNmKCRXZSR4XPnzmNrzng7EMexDFN9etO6T1zpw08sSA6VH61ocldQ1vXo0aOu/JJbutK2NhXkff/mS8+60kmYXBQsE56vaXNLSDb169evpsQ91bGbqytqzh157omH7D/F5uNw63vOuhcmaGSvT999LYCnvunKJzHVJ5944omEOFfrLA/rFnS6qqVNz8UWM24sPbr7kzde+ec/ZbZqbNqUnRSaX9yg52SLmfHjx7tnlIOQHR+9+tGH7/Mcea5HdTwJtPyQ8MbF3J6DCRrZi6bpb7/+kqk8eBNNuSOaGmWg4L03XnS9Kx5FTe4Rsfn3Kg4GOtxVHPnC/h3K0MBZUye63pWIT2cnhmw9e9n1rmzKysoqPf2b6/0c/f7LkcOGZg6xd2X/FRRl5SEhpztYYIJGDsgcPuixh+czVUe9HYh9TO1M1dF9u35wV3/R/uLkUPneS26eA3ejkJCQSBHj4j7fzdUVZ37a8PV//+2uqJJC5YESAQfjPP7+/t1cXvJeXXS+6vSv/1n5qsNnEgCrDwm5hAkaOeaDlct6dotgGy56OxBbCGsq279m9afxMVFu7HVofFBNm77Q84vrhg4dqipwfqNrvVb9/VvPrV+zWiGz9+GYPcYmh56rbqto9viEltGjR5edcf4+QN1Yt/3D5Vu+Wi+wrzRSVxQNFFi8faYpiodDHMiX/bb3xxC60ber3BGmfP9Tjz9y34yp7u2XAri7b+TBksayJs8uXRk7dqzT04EZk/G7fy3+57JlGf3s2h3YfjRFzUiL3l5QU6v27NKV8ePHlzg7DG1o12x5Y9F/P/k4MS7GidM7FxJaQAEw+JAQ+TKpRHzqt30K7UWiqfN2LOYxlYfvypnwzooXPNG5gEfPTIveUVjr0SQ1YsQI5xY9E5b54e0lD/xt9v13TXF7VAAgFfCm94vecuZyi86DuxkMHDjwctE5J040GXSbX1348pLnbhs11LlLU3THckHzc+w6FhlyBhM0ckZEeOih/F/kbeeIutp2a04RpuLguFEZm7740HPXkAp5M/pFbz1zuarFU3uOiMXiHlFBOrVjZUAYo2HLG8/cOXn8s4894KHAACBAIpjWO/KrkyrPbQnG4/GG9kpsrXPsmaShXfvtPxcsfOKxuTPvcP7axGolDgAux6IxQSMnpSTGnz1xKFB/yYfKdLAMU7ZvRs747ZvXePpSARLBrP7Kny7U/FHvqY3Px4wZ49BkBr1W/c0rT87Jnb580ZMeCumKcIVoRlr05t+ryj02Hu3oZDtNc8OmZY+/sHjRk3MdK1h4I8pKLQ4KKFxJiG4K0ZHhhWeORota2BqPFGBzCDFqTcU7n3nqkY2r3+PminIR/76BMYdKG38r98jEBofWfDeqSr9a+vBLzy1e7HhFJOcESgT39lfuKqz93TPV/hxK0NVF579Z/sRH7741Z3qOi9ftrIhk+cBaHOimEejvV3LuyMSRfZmSPWDidI/RrkhrpbT20FfrVr/+8mIuryvi07MHxGj0po0nK9vdXde/X79+tSV2zZY5t/fH3R++vHXThvvu9Mi4syVyEf9vg2Irmto3/15lZNw8Oz4mJiaIVRPW1p8qIcd++PLw+nfytv8wMXO469dlCVje8YoCoLi7f8YEjVxH0/QPX6/++L03xdW/eqGgEmNgKg7GyzUXzx6fMc0NKzIcRVOQlRQ6KCZwzbEK99a2pyhqZFpqc3WllTaa5sbv3lxMLl88eWDvgN7O7HrnIj5NTe4Znhwq//xoeZm760kNHDiw6g9rjwpbalSblj0eKTCe/DUvyak5GzeiKQooiw8JKZrLhYSYoJGbPHT/jEsFpzMS5EzJHq6WgxO2sUhcnf/is0/+cfpAeFgIJxc1LzFENmdgzMU6zZcnKpvb3Ta9wcooB8swx3/86ocVC154+omtaz+RSV1dzO2KPpF+uenKYxVNm3+v0hjcthreymQ7k0F/6JvPdr73wodvrfj83TcEAsf2YbCCALH1kJA7mKCR24QGBx7c8/3/Nn4erCtgKg95dCdDtqXcVLQ9My2m7OLZl5553HMXsp9YwJvcMzyze/CWM5d/vFDTqnNDnho7duyNRTkIy5zb++PaxfclBfDPHj0w8/bbXL+Q6xQi/l19o9Kj/TeeUO2+WKc1uGHAJzMz88blKozJeGrHt+v/MWdYj7hzRw+MGzHE9QtdiwLKCk5ztNt+7CDUYerErKkl5z7+fOOyV16tvayjQ1IoWbjbemeNbFMx21g0cuSoT3cdSu4e77ae3STaXzI3I/ZSg2bLmaoAiWBQTGCUv9jp3jrHYQlLUTQAtLe1nPnl+/P7fpowcdKpA3khgQHuC9w9EoJl3YJlBTVtG09WRvqJB8YEhMlFTvcmlUpTwv31mjaRTAEA6qb633dtKTu6N+euuzccPeAnl7kv8Ks6bpMtTXbmeJodJmjkEY/MzX1kbu7WH/csX/HmqVM/0opoKiCOEgc62R3LsG0q0lLqLyKTpkz91/ItUS5UN+ZA92BZ92BZZXP70Yrm2gu6lDBFz3BFiEzoRFf9+/cvO31E29pUcGAnrddMn5n7v7cPyb06oGEdBdAjXNEjXFHcoN1f3NDcbuwZrkgNVwRK7NrQ5DpZWVn5v+7k8fkFB3ZJeXDvffc9+Z/XRUJn/iTtRqwMZVDA6WNCTNDIg6ZNzp42OVvbrnvzg/9+/fUmVfkJNSOgpGGUJJiSBABt7X8sMWiIrhG0DaymNshP0n/Q0GcWLB+XOYyz4F2nDJAoAyQmlhTWqvddqm/QGIJkwvhAaZhCFCITSgUWy0QwLGnQGuo1hormdlHK0OOfvz92wsSX//Ne3x7JXMbvooRgaUKw1MCw52vadl+sbWo3hspE8UHSULkwRCoUW/74JpbUa/T1GkN5U7s+bmDld/8cN3HSmxv+m9wtnoOwqc5dvc2naIrmdJADEzTyOKlE/NIzj3eMFB84fPzrrT8dO3a8rPSUVqNpUbcDUBRPCBQFAIRlgDUCQKCfLDgkLGVgj9Ejc/+WOy00OMjLn8EFfJrqFaHoFaEAgEatoaypvaCmrV5j0BgYHg0sASGPFvJohhCDiWUJAQAeTQVJhSEyYWqYfOyj97z5eK63P4TzhDw6Lco/LcofAGrV+vKm9rOX2+o0ep2R5dEUS0jHxzexxMCwHdXw+TQVLBOGykR9o/wmpGYtnZnFacR/VrMzi+OHhJigEaeGDxkwfMiArq+wLFtTW9euN1EUBAf4+fkpvBUbB4KkwiDp9b+eGxiWYQkASCzfVN4awuSiG4ekOz4+RVFivm/MWfjzGaGlN7mMBRM08jKapiMj3PcU8SYk5NFwi2dma3zt43ekYIuznTmdBo3T7BBC6FrWl3rbtG3btoSEhPj4+LVr17oYCd5BI4RQFxRQFFjaxZCiwfottMlkWrBgQV5enkwmGzBgwLRp0/z8/JyOBe+gEULoGq6sJDxy5EhycnJcXFxISEhmZuaePXtciQTvoFEno9HY1OTx7eZ8k1qtNhqNFRUV3g7EO5qamnQ63V/242s0GrH46mKizhFoC/fJep2uqKho9+7dffr0CQ838+ykqqoqOjq642ulUqlSqVyJDRM0AgCgaTosLKytrc3bgXiH0WgEgL/szyedTmcymf6yH99gMAivX/licSmKVqtVqVTHjx+PiIgwm6AJIV2Tu4vPFDFBo04xMe4pBnYz0ul0RqMxNdUL1eB8gUqlqqmp6du3r7cD8Y66ujq9/prdyyiwWJU/KCgoc/Toxc8+a6m36OjoK3fNKpVqwIABllraA8egEUKoC0KAECCs+QNslL3OyMgoLCwsLy9vaGjIy8vLzs52JRa8g0YIoa4IAAFiIRETYmUABAD4fP6qVasyMzMZhlm2bJm/v78roWCCRgihaxECrIUEzbLWEzQA5OTk5OS4uvNWB0zQCCHURcctMrGchTnc8woTNELc0RuMW3/ak3/g0Jkz58rLy1qbG4EQiuY11VeDUO4vlwCAWCINj4hMSU0d1D99+rSJccpIb0ftNgxLVC26WrW+Tq2vU+vbDAyfohhCPl3yWGNzkxBYiubJ5IromJhePXum9e1926hhQf5eqc1CLCZoK4nbAzBBI+Rxu/Yd+venaw4d2F9b10BJgihJECX2p4TdIax3x7o0SpvHUw7T8EUAoGGM9U3qs3nnNv98YPGSl/wkvOTUPnffNe3Jh2ZLxM4Xv/eiqlZdQU1bcYPWyLJKf0mYXJQSJh+ZECwXdeafbyhd9vLPOr7WqVtb66rOq0r3rfvfcy8uE1Js+qDBd03Nueu2MTweV5MaCAFLm9USFu+gEboVlJRX/eOlFbt2/NSiA8o/llak8wPsqDTPE1C8wI7NDWgALSEnq+qOv/reP557Pi0t/dlFC2dOm+Dx0N2hVWc6qWo+e7k1VC7qHeE3LD7ISg3oK8RyP7HcL6xbasqICQDAMibV+RNvfrruH4ufGTB46NNPPDIsvY+HA7eZgLFgP0I3s8PHTj/y1DO/nzlDByXSocN4VrcmsIGiKFkYTxYGAGdqa2c98OQTT9ELnlrwwqLH3Bauu9VrDL/8UdeqMw2MCZg/NF7gwp0vzePH9MmI6ZNBCFt59ti8p56RUMyzzyy6x6M7MRIWiOU7aA7hPGiE3OlsQVGfwWOHZU8+V8fnJ06kg5KsbxwekpLaAAAgAElEQVTjEEoWxosd2eLf/6U3Pw6KSlz10Rfu6tldmtqNX56o3Ha+ekhc0IND4tKi/V3Jzl1RFB3TJ2PKMysH3f/M6/9e3WvIqB/27HdLzzdgAa7MhrZwcAUTNELuYTAaZ857su/AEQUNAn7CeEoe4akr8SW8qIHqoEGLlq7o1mvQid8veOpCjmBY8ssfdV+drBydEDxnUGxsoKd2TfQLixr76NLhD77wzMuvZd+RW3m5xs0XIJS1hSq25kG7FyZohNxgT/7hyPiem3ce4SdOoBSczLsQSHgxwytNEQOHjZm/8AUurmiZqkX30cESiYD3yNBuygAuNrT1C4ua8sybwf3HDs667b0vNri17z8Xqji1ktC9MEEj5Kr/W/LauMl3tPr1pUN7AcXp/ylKGspPvO2/G7/v3ntwTW0Dl5e+4lhF8/fnLs/qrxwaH8TpdiMAsf0G373sk/9+tWXiPXO07Tr3dHplHrTF8Q3ucjQmaIScZzAa+4+c+M5/1vK7j6fELi3qdR7F40UNLG/3T+k7cO+vR7i8soklG05U1qr1Dw+Nv3GvRW4IxNJxj71ER6Wmj8gqKq90Q48UBYQAa+EgwGXaxASNkJNaWtsSew/+vbSFFzMMKC9vq0f5KTVBA7ImTPnymx+4uaLexK49Wt4zXDGpRzjN8Z3zDXpkTsmY/fS4STmHTp5xtS9idYiDsBZXgXsAJmiEnFFT25DUJ6PKGEiH+EqRUkrkx++Wfd+8+R989qWnr6U2mP77W9mIhOC0aC/93nCD8O49xi18/e5Z9/1y6JirfVl/SMjhDyNM0Ag5rKW1re+gEU18Je0f5+1YrsWX8LtlP/X0os83bPbcRfQmdt3Risk9w5ND5Z67ihP8wqKmLH7rvnkPHjh+2oVuWAIAwFo6CE6zQ8hnmUxM34zMBl4U5eeTWxzwhPz4rAfmP7bjlwOe6J5hyZfHK8YkhcYFSj3Rv4sUweGTF705+29zC0vKneyCUADEko4W7ovXBkzQCDlmxPg7VWq+z907d8UX8eJGTb8n91Kp+7cZ3HRK1V8ZkBrmW/fOXfmFRo586Pkpd05v02idOZ+igLDAMhYOFh8SIuSjlr76ztHTF+jQXt4OxAZKqGgP6DMyezLr1idah8sa/cR83xl3tiSsW2qvSbOnzLzfmZM7FnkTC8eVBpzABI2QvQ4fO/3qitd5yiHeDsQulCy8Vie6Y/Yj7uqwuk13UtUyIdXMTqk+KHHwGA1P+ur7nzp8Zsc0O8JYPDicsoIJGiG7sCw79e5cXvRgoG+aEmN0WO8ftv24bede17tiWLL5dFVuupJHe3lGnf1G3r/wo48/vlBU4thpndPsLB1/PkTkBCZohOzy6P+9WK8XUdJgbwfiEIqnHDrngfmuD3TsL2noF+0fIHFb4ScO8ATCzHnPzHn0SYfPtLGSkDuYoBGyraqm9tNPP+WF9fV2IA6jRH5NJtmC5/7pSietOtO5y63D4oPcFRVnIpJ662jJ6k1bHTyPAMuaP3AeNEK+Zsb9j9KhvYD28nJB5/DC+qxZ/VmbWuN0Dz8X1EzpFeH15YLOGTH7qddefZW1/863YxzD4lNCy/vJegAmaIRsKK2oOnjgAB0Q7+1AnEXzteLoR//uZMW75najRm/yzVnP9pD4B4ak9H9v9Xp7T7A5zY7DeliYoBGy4YHHF9GhPb0dhUvooOQftm7W6Q1OnPvLH3WZiSFuD4lLA6f97d8ffmhv685ZdJbvoInJY5FeDxM0QtaoNe15v+zx6WUp9qB5Gl7IC6+ucvQ8rYGpUxsSgmWeCIozYrmfLLLbtz/tsas1oazWgybA4SA0JmiErHn5jXdp/1gup756CB2csn7dWkfPOqlq6a/09WUp9kibdM/7H31kV1PKerEkLNiPkM/YuGEjFdjd21G4g0BS16Q+dbbAoZN+r2rpE+nnoYi4FBzTvbiktKml1XbTjttka/Ogzefobdu2JSQkxMfHr117zQ/CkpKSrKys6OjoHj16bN++3aGwMUEjZFFxuepybT0luFmfj12HCoh/5V/v2d++XmPwE/PFgpty7sqNEgdnfbrxf7bbUTwgVqbZsWYn85hMpgULFuTl5R07dmzp0qWtrVd/EhBCXnvtNZVK9e23386ZM8ehOemYoBGy6INP1tJ+Sm9H4Ta0X8yv+Xn2ty+oaesVcSvcPndIGDT6xx9/tN2ONdmoB80abzzpyJEjycnJcXFxISEhmZmZe/ZcHe9OSEgYMmQIAPTq1QsANBoH5jtigkbIoh07dlCKKG9H4T48YV1DS7M9v+YDAEBRg6Z7yM39eLCrwKi4oqIi29WcO2bRWRji0Gjbz1y4+M0331RWXrO9VlVVVXR0dMfXSqVSpVLd2PGuXbsSExMVCoX9MWOCRsgiVXkpJQ70dhTuRMlC7dwTixDQGhiF6KYpPGKPwOi4Y2cuWG9DCAFCCGHNHkaTsb29vampSavVXncW1eVJMnXDU+Xy8vKnnnrqs88+cyhgTNAImVdSXtWm4/SRPQcoaej23XvtaVmr1ofJRR4Oh2tRKWk//5Jvo1HnQ0LzQxz+cmlGet/58+cnJycDwMqVKxMTExMTE6Ojo6/cNatUqsjIyK5dNjc35+TkfPDBB6mpjm2QhgkaIfO2794Hkpuv+oR1lCS44MJZe1rWqfXhilstQYfGJ585a2tXWZoHhADDmD/Yax4SLlq0qKioqKioKCMjo7CwsLy8vKGhIS8vLzs7GwC2bNlCCDEYDHfeeefTTz/d8aJDMEEjDvXuDTQNZWVXXxkxAigKTp3qfJeirh5TpoBOd80r1o/166F3b1h/w4reJ56A2FgQiyEhAd5+2/5gjx4/RYluhSnAXVFCWWN9nT0ta9X6ULnQ0/FwLDgm4dLFizYaMYy1UnaEAGNmJSGfz1+1alVmZmb//v2XLVvm7+8PALm5uQzD7N69Oz8/f8mSJUqlUqlUlpaW2h/wLTXAhG4CSUmwfj0sWQIAUFwMKhUIu2SB1ashN7fzax4PBAJob+/8VqmEdetgzBgAAJYFmr7+RaEQXn/dzBVnzoRFi8DfHwoLYdo06N0bxo+3J9LCixcpkQPPc24WTa12zSKoUxv6Rd1qP5+EUnlbm61npBRNCCGs+W1TCLFYiyMnJycnJ6frKzqdDgAmTZpkMjm5OhzvoBG3Zs+Gdes6v167FmbPvmaRnkAAYnHnIRAAwNVvu74rlZp5kbbwj3nkSIiPh8BA6N4dFAooKrIz0oaGehBInPuUPo2i9HYU5VAbTArxLXgDpzPa2rCKMNYKcYDF3O0JmKARt3r0ALkcjhwBAFi/Hu67j4uLPv88hIVBRATQNEyfbudJ6rZWinerDcICAMUTV6gu22xmYFgB7xbMDxRF25ppR6wtVOFwOxXABI284P77Yd06+PVXCA2F5ORr3nr0UQgJ6TxeecVtV3zuOTh+HL78EqZMAbtnoRr0uptodysH0PzG5hbbzTgtTM8dnkCo0+uttaB4QAhhGfMHYSkOy43eiv/+kI/LzYW+fUGthvtv2HT5tdfgjjs6v/Zz3xo2hQIUCrjnHti3D956q3ME3BaGYbir/CuQWByicTuKalNr7Wjm+Uj+RNM0yzA0j4s15TyBUK3VSTrGx8whzJ+7ept/GwhjZiWhh2CCRpwLDYUBA+Crr+Ctt65/KzAQlJ5cWk0IFBfb2ZamaSCEmzp2vGgOdwonRCr2raGb5OTkJlVJcGwiB9diTEaZxNrHp3g8YFmzUzUAAFiG4nGXNnGIA3nDBx/A3r0QdMMsY6MRdLrOw+BMdflrejAaoa0N3nsPSkqgsRE2b4Z168DuuagiscRs1YWbHmsMC7Nj61ti8SbS7dLS0mpLCrm5FmM0WLl9Bui4QbZWi4OYuPtXgQkaeUN8PAwebOb1efNAIuk8MjKc6blrDw89BDQNP/8MAwdCdDQsXQqvvw6zZtnZk0Qq4/KXWc4QxhgRZnuHFAGPNjEcLaRMT0+vLbaxAttdCMvcuA67Kwooa7t6A0e/VHXAIQ7EobPm1rDpdNbevaK+3vaLZnv4+We7YrtBaFhEaZkGhLdOtaBOhJFJbRdQlYt4bXpTkJSLtSp9+vSpL/uDgwsBgO3qqRRFCEssFQVlcUcVhHxA94R4YnR+J2yfFaCwq7x1iExUr3FqlMlxEokkRs7FE0KjTiu19cOpc5qzxe1ULA9PewAmaITMG9Q/jeiavR2FmxGjJijYrh1gQ+XCWrXV6WhuFRsb21pne3a2ixorSxKSUqy3oXh8IKzFWhwMC3zuVsBjgkbIvPFZI6G9ydtRuFt7U3dbGapDmFxU28Zdgk5PT68tdmw7LifUlxX17tXLehtiMtgYgzZx98eCCRoh81KTusmEnC4b4wCrrRufPcaeluEKUTXXCdrjzwmrCk7dNmakjUY0DwhrcaEKa7EWhydggkbIosjoGKK3d/+RmwLR1N5/z1R7WtIUJebTGgNHdSfS0tLqSjx+B91QcWlY/742GrHE2jS7joXgXMEEjZBF2VnZpK3K21G4D2sM9peFBttb5Lp7iOxSA0ePSYOCgnRtdixAd0FbfXVsbAxtc8Uma7I2xEEIECyWhJAPWPDoPLa1wttRuA3bWjlk+Cj726eEyS9Ut3kunusMTI5tb/PgU9lLR/ZOmDDRdju+kBCWMIz5g2WBJ/BckNfBBI2QRSmJcaGBcjDpbDe9GZDmsn8sfNT+9hEKcb3WYORquUpaWlpdsQfXExYd3vPw7Bm225kMVlcSssSADwkR8g133TWDbbrk7SjcgdEHKwTDhwxw6KTeEYqzXN1Ee/Q5YUt1ZURoSHiIHcM7NB9YsFhulBDg4x00Qr5h+ZKn2eZSb0fhBmxj0R132XH/eK3+yoDjFRxNBk9PT6/12HPCU9u/fvjhh+xqypqs1+LAhSoI+YqQoIAhQ4eR1kpvB+IawrJNJf9a9qyj5ylEfIWIX9Hcbrupy2JiYto8s1bFqGtvKjrzt7tut/cEQghhzR6dOZormKARsuGT9//F1J3zdhQuYZtKptx+u7+f3Ilzs5ND84rs2mfWdb1iQo1694/4n/xx45wHHqTtLHJE86ytJGRZ4KRudWcsnF0JoZtU79Sk9LR+N/FNNGHZhgufvf8v584OkQl5FFXVwsWT0j59+tSX2dp120EGrbryxL5/PG7f+AaA7WqiRiw3ipAv2fjFR0zNaSAczWdwL7b+Qm7urLBQO2pAWzAhNfyHc9Uc/GLviQXfh77+eMHCp/l8e297KZ6AEJYwJvMHy2AtDoR8S3JC3IwZM9m6894OxHFGrcJUu/oDJ2+fOwTLhN2CpccqPF6ZJC0tzb0JurGy2FBbtuCB2Q6cYzIAsfqQEGtxIORr1v3nbT/ScNOt/GYqD7377iqRyNWbvqyk0MOlTWq9ZycwpKSkNKlK3dUbyzC7P371vx996NhpFA0EgCXmD4L1oBHyPQIB/5uvvmQqDtxEAx1sfcHIYUPunznN9a74NDW1d+TGk5WsJ+cw8Hi87iEylnHPWuoj33529113Duzb07HTaAqABcJYOFigMUEj5HuyRw15cN4c5vJxbwdiF9LeGEDqtm9Z664OYwMliSHyvUXmtrZxn9TU1CZViev9lP/+m67qj7deXOzwmSZbtTgsPCTctm1bQkJCfHz82rVm/sy1Wm1cXNzChQsdigUTNEIO+OTdFSkxwTfB2kKTjqk8uHv7NrHInRt4ZyaGlDe1F9Sq3djnddzynLC5uuLY1x9u+2aD9e0HzRMIgRBrtTiEZv5ITSbTggUL8vLyjh07tnTp0tbW64fCVqxYMdjsPpxWYYJGyDFH838OJrWkzePbfziPNZpK89as/iytt121+e1HAdw7QLmvqL60Uevenq9IS0tzcT1he1vzz6ue37BubYQ9C7tvZNBZu30GQgxmZhweOXIkOTk5Li4uJCQkMzNzz549Xd8tLi4+d+7cpEmTHI0FEzRCjpFJJSd/26fQFBINR8s3HMMyptK9K/65/L4ZOZ7oXsCj7x2o3Ha+usYz5fxd3EBWp275fsXC/7z/7pC03k52weMDIZZqcai1+rOXyr/55pvy8vKuJ1VVVUVHR3d8rVQqVSpV13cXL168YsUKJ2LBBI2Qw6Iiwg7m75G3nSPqam/Hci3WaCrZ89wzTz+7wN51GU6QC/n3DojZdErliSXgrmwg297S9N1rC95+4/UpYx2oqnod0oFlzR4sy7brdE1NTQaD4bqzug6ndP16x44dsbGxKSnO/DaDCRohZ6QmdTt74lCg/pLvlFIiRq2pePeKV1569YW/e/pagRLBnIzY789Ve2I8Oi4urrXW4X0Smi+Xb33tyX+/8/b0yWNduTphTHAlTd9AIRYO6pE4f/78xMREAFi5cmViYmJiYmJ0dPSVu2aVShUZGXmlw8OHD2/atCk+Pn7RokWrV69evny5/cFggkbISdGR4X+cPR4j1bDVpwC8vHsh0dZLag58te5zj947d6UQ8R8cHHewpGF/cYN7P7wTzwlLTx785YMXtn694XYX7p070EIRYQlhWPMHy1JC8ZXGixYtKioqKioqysjIKCwsLC8vb2hoyMvLy87OBoAtW7YQQl566aXKysrS0tKVK1fOmzfvxRdfdCAYFz8MQn9l/n7y4rOH7xg/xFS8G4xclHwzi607F6QrOpD/y4xpE7i8rohPzx0cZ2LZNUfLte7bvdCh54QsYzq44cPy/O8O790zyNEpz2Y71Os7Nx40dxBCWJ2Zv2g+n79q1arMzMz+/fsvW7bM398fAHJzcxnX5nRjgkbIJRRFfbPm32s+/UBSe5Bt5Hr6HdG3mi7tHDekR1XxmT49kji+OgBQAGMSQ0d0C/7st9Izl92zzNL+DWTrSi9+8+JDo/ol/7Z7W1hwoFuuTvH5hBCWYcwewBJKYL5gf05OTnFxcVlZ2Zw5czpe0el0fD7/SoM5c+a88847DgXDt90EIWTLfTOm3j4hK2fG3F8P7eRFDqQkTk3wcghrZGrOBAm0azZvmDTO1d/rXZQYIps/tNvOwpqjFU1Te0UGy1xaWW7PBrI6devBjR9CS+22TRv6ufUnE2FMQAiwFoZtCCEsFuxH6GYT4KfI3/7t7m3fRrBlTFk+0XlsIxLWxNadE1/e99S86bXlhV7Pzh3EfDqnV+TE1PCtZy9vOqWq1xhsn2OZlQ1k9Vr1oa8++nnFk/NnTjuRv8u92RmgY66zxYeEhBDgcKk/3kEj5E5ZI4dUXDz5zdafn3luaUXJcTo4hfKLdld5HaJvYxsvytnme2bNfuf1pRKx2PY53Ir0Ez8wOK68qX3buWoBnx4cG9g9WObEar60tLSzxQVx/YZ0fbGxsvjUz1+3VRbNn//w39e/z6M9cn9J8/iEEGIyn4YJQ4DPXdrEBI2Q+02fNnH6tIn7D59Yuvz1/PwfKWkYHRBHSUOBcianEIOatJSzrRWx0RHzn31s8ZMP2l/d2CtiAyVzMmJr2vTHKpp+vlDTPUTWK0IREyCxd08TgPT09Lwt+R0JurGypPjovkvH9iXExz378PyZU8Z7MnZgjQYgQCxOyzG/ktBDMEEj5Ckjh/Tf+9MmhmE+XfftmnUbCs7nt2pNlDSEkgaDOIASyCzma5OO6FtJeyPR1hN9S3x83MTcSUv+77GoiDBuP4FLwhWiyT0jCIGievXpqpZt52skArpbkDRMLg6TC4OkQp65snAEoLnd6B+bXLD/udriCy21qpSU1NsnT37o/WV+chkHYdMCIRCWWJh9QViWEnD3iwsmaIQ8i8fjPTJn5iNzZgJAWWXVxs0/Hjz02x9/XKyrqWZYpqVNC9BRg5h0TKYOkEsUAYFxCfH90ydNvi1r7OihzlT88RkUBUmh8qRQOQBoDUxZk7ZGrT9f09qoMZoIoaCzeGdnEVMKKAB/sSBMEbz0pZdHZfTvk9Kd44BZg4EQIBYeEhKWEAN3BfsxQSPEnThl1D8WPARcrSXxNVIhr0e4oke4wp7G2UnTPR2PeTQNhFhM0IRwWQ8aEzRCCHVB0R21OMy/SwjQ3D0AwASNEEJdGI2dBfvNIoTFIQ6EEPIOkYgQljWZf0jIMiwtk3AWCyZohBDqQt/eUYrDPAKMzlObFdwIEzRCAABarbaqyuESl7eG5uZmrVZ7/vx5bwfiHXq9PjCwSx0P+s+C/WYRQuFCFYS4JBQKIyMjiSf3q/ZlQqGQz2HS8UHX/dUTQghjcZodl5Vl/9J/Kwh1oGm6a4X1v5qWlhYej5eQkODtQLyjrq5Or7/63I+YTECAEAtLvQkhFnb19gRM0AghdBUlEhOrKwlpMT4kRAghb2B17dan2TFaDWfBYIJGCKGrKIGAEGAtj0FTQhFnwWCCRgihq4iJsVGw38RdwX5M0AghjhT8UbLx2++PnjhVdLGwob6WZdkWtQ4IQ3TNlDgAaL6/TEhRtJ9/QLfuSen9+t6VM3H44HSuoyRMx74p5t8kBFi37b5oEyZohJAH6Q2Gdz5au2nz/y5dvNCmYylpKCUJBrGSCkuFPxOQqehnfuJEAOgY3FWbdKrC5v2nf3j3o9XE0JaUnDxl0uTn/v5wSJB7dh20jhJ07upt/m2WdN3V29NwyyuEkEf8uHPfoNGTJUHK51e8/3s1TxuZyeuWTYf3pfyiKaHc2pl8MSWPoEN68OJG8ZMmF7eHvfv5/8Lieib1G/b+p+s9HTar13bcQVs4CNvO3UpCTNAIITf7aPXGyIQ+OTPnnqoi/KQpvMgBlCzU6X2/KLE/Hdabnzix1BCxcMlrgZEJi154zWShVobrKJGUsEAYYv5gWVrE3TQ7TNAIIbf5esvP4fE9n1i8rF7SgxefSckj3Ng5JQ7gRQ/RhA5757/fhigTX37jfTd2fgXR66xtGUuA5XDLK0zQCCE3qKiqSRs2ftbcRxtlvXgxw2wMYriCJ6Qj0rRhw19562NlUr9fD59w9wVoAoSwrNkDCOuuLYDtCoWzKyGEblUrP1zdq9+gczWE1y2LEtq1YYqreEJe9OAaXrdR4ybf98j/ubGOCsXnAbGYoAm3xZIwQSOEnGc0mrKmzHx26T91kaMovxiOr05Jgvjdb9vw3S+xqf0rqmrc0idr0AMBYImlg9W1mz1x27ZtCQkJ8fHxa9euve6tioqKrKyssLCwHj166HQOjJBggkYIOanyck1Mcr/8U2W8uEzgCb0TBEXzItKrmbDe6Rl5+39zvT9aLCUsYRnW7EFYQknNbC5uMpkWLFiQl5d37NixpUuXtra2dn133rx5s2bNqq2t3blzp0OFAzFBI4SccepsQe/0IQ38WDq0p7djAUoRrQ3JyJ6Ys37T9y52xbRrrD4kJKzaTC2OI0eOJCcnx8XFhYSEZGZm7tmz58pbJSUlpaWlDz74IADExMRggkYIedbhY6dHZ92mCehHyX2lTCslVPC7Zd//wMMfrt7oUj88PgAAw5o91HrjydLyTz75pKioqOtZVVVV0dHRHV8rlUqVSnXlraKiIqVSeeedd6anp7/88ssOBYMrCRFCjvn9fOFtk3K0wQMocYC3Y7kWX8zvlv3kgr8rpJL775nmZCcU1fGM0MLbRCwSBQYGCoXXDOkQQijq6uyOrl+bTKbDhw+fOnUqLi5u3LhxQ4YMmTBhgp2x4B00QsgBl2vqRmdN0Aal+1x27sAT8uOz5jz48K69B53rgBgNneVGzR1SHi81Imz69OmxsbEAsHLlysTExMTExOjo6Ct3zSqVquv+D0qlMiUlJSUlRSwWjx8//syZM/YHgwkaIWQvhmEGDh+rliVRkiBvx2IZX8SLHXn3jNxyVbUTZ3c+JDSxZg/CEFp6dYr3okWLioqKioqKMjIyCgsLy8vLGxoa8vLysrOzAWDLli2EkN69e7MsW1lZyTDM/v37e/To4cBHceIDIIT+miZNn1vTLqLDlN4OxAZK5Kf17zl8zITywpNdRxvswWjbrdbrB0bdduPrfD5/1apVmZmZDMMsW7bM398fAHJzc9VqNZ/Pf/fdd8ePH28ymSZPnjxlyhT7g8EEjRCyy6drN+3+JZ/XLcvbgdiFkkderq7JfWDBV6vfc+xEoQBYi5vGAktoC9XscnJycnJyur5yZcrzmDFjnNs0HYc4EEK2NTQ1/9+iZ3gxQ70diAPoiH6bvt3s8ORohiHE8kQ7IITDetCYoBFCtk275wGttBvwuSvk5g4UL3pw7n1zHTqHMAyAxWWEwAJr5G5HFUzQCCEb9h44euDQITowwduBOIwSB9S1855/5W37T6FFEmAJYVjzB8vyJFhuFCHkM+Y89BgvapC3o3ASL7zfh++/qzcY7GzPtGsJgKV6/UCA0XC3qzcmaISQNT/u2ldxuZ6SBHs7EGfxhBp+6P8tec3O5rRYQgiwrMWDlkg9Gu81wXB2JYTQzWjB//2DDu/n7ShcQof2WLf2c8bSNoPXYnU6qzteEcaRcnQuwgSNELLo3MVLxaUVPr0sxR60QA1+Kz9YbVdjCgixPMTBEnBwYrUrMEEjhCx6dumrdHCSt6NwAzok9aOPP7KnJSUQEmsPCQnNF3g62iswQSOELPp17x7aL9bbUbgBJVSUqy7bs/ib1enAYikOIACMFnf1Rgh52/fb89pMQqB53g7EPWj/uFdXfmCzGU8itTLNDhiWp+BkTy8AwASNELLkszUbKP9b4fa5A+Ufu3PHDpvNGI3mzyWD5u+gTa1manF4CCZohJB5Rw4doH2mHr/rKIGsrLzcZLKxDpDi84EAYYj5gwVagGPQCCGv0huMtQ1NXttp0DMoceDPe/bbaETThADLELMHYQhQ3KVNTNAIITN+2rXPR0vyu4CShnz/0y7rbVi9HsDyU0LAedAI/WWtWAGTJl39Ninp+m+/+AIoysyxfn1nm8xMCAyEjizToXfvzjahoTBrFjQ02BPIvl8P3fTTn29ASauCVDcAAAy5SURBVIN/P33aehueTG5lV2+WJXx8SIjQX9SoUXDgADAMAEB1NRiNcOLE1W+LimD8eGhv7zyCg+Gnnzq/njULAKC0FPbvB4qC76/d3HrNGjAa4eBBKCiA55+3J5Az5y5QIj83fzpvo0R+FRXl1tswajUAALF4GFtauYgVADBBI+RbBg0CoxFOnQIAyM+HMWMgJeXqt927Q1QUiMWdBwAIBJ1f0zQAwNq1MGQIzJkDa9Zc0y1NA58PSUkwcybYuoXsUFZaCkLubhU5QgvatTZKHVEiIWHND0CzDCEs4YnNF+z3BEzQCPkSoRAGD4b8fACA/HwYORJGjLj67ahRNk5fuxbuvRfuvRd27ICamuvfVath+3ZITrYnkNbmRuomq/5slxZ1u/UGxMQAWC7YTwhhsGA/Qn9Zo0d3ZuT9+2HkSBg58uq3o0dbO/HXX6GsDGbMgAEDoHt32LDh6ltPPAERERASAgCwYoU9UZiMxltmicq1KMZqhmVNJiCWC/YTLNiP0F/ZqFHw66/Q1AR1dZCUBMOGwcGD0NQEZ8/auINeswbGj+/MwrNmXTPK8eqrcPo0tLZCXh5ER9sVBncVgbhF0+06vZX3+VKZ1V29Wb6Mu3KjuGksQj5m6FBoaYFPPoHhwwEA/PwgKgo++QSioqBbN4tntbfDpk3AMBARAQCg10NzM5w+Df36AQD4+0N4OCfRO4OShnB3LaDb1Bq55SRr0miAALG8rbcRVxIi9NclkcDAgfD22zByZOcrI0bA22/buH3euhV4PDh/Hk6dglOn4MIFGDkS1q51PgwLCcoTuNyuhQBRyK3dAvMkEkKAMOYPlgWeTMZZtJigEfI9o0dDbS2MGNH57ciRUFtre3xj7lyIjYWIiM7jiSfgyy/B1spmi27VIQ7WJLeaYRmdnlgu2A8EmHbuFqrgEAdCvmfFimse5c2YATNmmGlWX3/16+3br3/3yllnzzoRgkAoBJa5RZ8T2mZpiIN0FEziCt5BI4TMCAwKJUbudkfljL/cxtxBWtRRsN9SsSRCi0XchAqYoBFCZsXGxoLhlkvQjFEqk9toomkHa/OgwdSmNnvitm3bEhIS4uPj194w9L9q1ark5OTExMSnn37aoXgxQSOEzEjr15fomrwdhZsRXVN8t+7W2/DltqbZ+ZlZYGkymRYsWJCXl3fs2LGlS5e2tl5dDl5dXf3ee++dPn26oKAgPz//2LFj9geMCRohZMboEUNuyQTdt28f621MbRprywgt3EEfOXIkOTk5Li4uJCQkMzNzz549Vy9KCMMwer3eaDQSQoKDg+0PGBM0QsiM28YMJ3ruqgJxg2jqpk+dZL0NxeMDAWDNHxqGOVZZ/sYbbxQUFHQ9q6qqKvrPFUBKpVKlUl15KzIycuHChbGxsVFRUVOnTu1mZTL7DTBBI4TM4PN5keGhYOJuShkHiL4la9QQG40EPACw9IyQBgiQyRISEvz9/a/pmRCKujozsevXLS0t3333XXFxcUVFxbZt244fP25/wJigEULmDRsxkm2r8nYUbkP0rd272xiABgCmvd3ihoQERBTdTRE4ffr0yMhIAFi5cmViYmJiYmJ0dPSVu2aVStXxbof8/PyYmJiQkBC5XJ6VlfXbb7/ZHzMmaISQeU8+8gBpsVE9+SZCWsqmTZ1qsxnfT8ECMRHzB0OIMPDqvfOiRYuKioqKiooyMjIKCwvLy8sbGhry8vKys7MBYMuWLYQQpVJ55MgRtVqt1+vz8/OT7asm2BmME58TIfRXMGpIf38JpWaNQHO3TarnsK2Vixc8ZLOZsbWNEMsr3QkYms0MzfP5/FWrVmVmZjIMs2zZso4BkNzcXLVanZ6efs899/Tr14+iqKlTp44dO9b+mDFBI4QsGjdh0uY9p+igJG8H4iqia0pOTAwNtr2JFy0SEQBLBUlZAEsLVXJycnJycrq+ovtz98Lly5cvX77coYA7g3HiHITQX8RrLz7LNl7ydhRuwNYXPvXkY/a0JAxLACyU4iAsIYRlPR3tFZigEUIWdY9X9undi2hu2Jzl5sLo/QX6x+bl2tXWZLQ8yw4IAGMwejreKzBBI4Ss+feq15maM96OwiVM3flHHn3UzsZCuZwFYCwcLIDAj7u9dDFBI4SsGT5kQEpSAlFf9nYgzjLp/EjzK8/bWwRDr9YAAMsS8weArrnZk+FeAxM0QsiGjV98zFSfAkubjPg25vKJJS+8wOPZWzeVL5EQAoyFg2VBoLBRbsmNMEEjhGxI6506ZfJktqHAdlMfQ7T1sWGKRU88YP8pJp2eEGJlDNrEYcF+TNAIIdu+/uJDubGaGMxX2vRRLMOofvvf1w7u+0WszuIAYnG7Qg/ABI0Qsk0iFq39YjVTceAmGuhgLh97/PHH0/v0cOgsnlTCguUhDgI8qY2S/26ECRohZJecCWNm3TODuexArR8vYptLU2KC33/jRUdPNLZpbTXgbh8DTNAIIXut/2RV9yg/tqnY24HYQNob/fRlB37Z5sS5Aj8pS8BKLQ6BHz4kRAj5pBO/7gwm1b48644Y1NLGE3t2bAv0d2bCskHdbn2higHvoBFCvkkuk544vE+uLiSaOm/HYgYxapmyfd9u2ujo0PMVFJ8mACyxcABQfO52OscEjRByTHRk+KH8PfK2c0Rd7e1YrkEMbUzpL5s2rh2XOdzpTmgen1gpxEGA4nNXYw4TNELIYalJ3c6eOBRoKPad8WiirZfU/rZj29a7b7/NlX5M7XqwfAdNCBjUNp4iuhEmaISQM6Ijw4vOHo/3MzCXT3h97h3bVOynPn9w/55xmcNc7EroJ2eJxS2vGELEQf62e3ETTNAIISf5KWR/nD5w79QxpuJdxMjdfeU1WBNTcaB3lLCk4FTvVDfUrda1tgEAsXAAQLu5gv0eggkaIeQ8iqLW/mfVxjWfyuoOsw0XOb46aasyXdr+3MKHTx3c6eemEhk8kcjaQhUAvoWC/Z6ACRoh5KqZd0xUlVzIHhBvKtpONLUcXJEY2piyvUkB2qJzJ/75wt/d2jWx8pCQJYRlcKk3QuimIpNKdmxZf2jv9jhRA1PyC9HWe+hCxKBmKg8Fqs+s+c8754/lJ8Qp3du/yWCwPg/apNe794pWYIJGCLnN4AF9i84c2r71y+6yJtOlnWxTMRBL2/s5jLRVMaV7Q3UXPvzXi7VlBbOn59g+x3EiPwULYCLmD4aAODDAE9c1CxM0QsjNxmUOLzix/8zRfZOHJsmq85ny/WxLObBO7RRFWKKuZlRHTH/8mNFNsn3L+qpLZx6Za9fmVc7RtaoBgBALB4C2kbuC/birN0LII3qldP9u46cA8MOOvR99tub4kUN1ja2UJJCSBFNif0ooB765snCMgRjURN9K2htIe0OgTNSzT797c5c+dN/d9hfdd4VAKiIEWAsTBwkhQrmUgzA6YIJGCHnW7bdl3n5bJgBotO1bf9qT/+uhc+cvlJdfVNc3AwXN6s769wFyMRAQS2XRypjU1LRhQzLumDw2IiyE42hNeiMBwoL5BM0CmPQGzoLBBI0Q4ohMKrn37in33j3F24FYwzIsIcBamKlBCLAsy1kwmKARQugqYUfBfgvvsgBCGXdDHPiQECGErtJrtGDlISGB9pY2m53ccccdAQEBd999t4vBYIJGCKGrhHKZtYL9QMT+CpudLFiw4PPPP3c9GEzQCCF0VecdtOVaHDo7qtllZmYqFLbzuE04Bo3QXx1FUS0tLeXl5d4OxDs0Gk3XZCrxU1SD/gQxXxGpHgxBjdVvvPHG5MmTe/fu7enYMEEj9Fcnl8vj4+O9HYXXKBQKmUx25dtxd099pbHRUmO9QS8NC45OiA8NDeUgNkzQCP3V0TQdGBjo7Sh8hSLAf97ihU6cuHLlyo8//hgAioqK3BUMjkEjhJAbLFq0qKioyI3ZGQAo4u2tEBBC6BZz2223nThxQqPRBAUF/fDDD+np6c71gwkaIYR8FA5xIISQj8IEjRBCPgoTNEII+ShM0Agh5KMwQSOEkI/CBI0QQj4KEzRCCPkoTNAIIeSjMEEjhJCPwgSNEEI+ChM0Qgj5KEzQCCHkozBBI4SQj8IEjRBCPgoTNEII+ShM0Agh5KMwQSOEkI/CBI0QQj4KEzRCCPkoTNAIIeSjMEEjhJCPwgSNEEI+ChM0Qgj5KEzQCCHkozBBI4SQj8IEjRBCPgoTNEII+ShM0Agh5KMwQSOEkI/CBI0QQj4KEzRCCPkoTNAIIeSjMEEjhJCPwgSNEEI+ChM0Qgj5KEzQCCHkozBBI4SQj8IEjRBCPgoTNEII+ShM0Agh5KMwQSOEkI/CBI0QQj4KEzRCCPkoTNAIIeSj/h/AuIOFZGlOtQAAAABJRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "corrplot(\n",
    "    res2$r, type=\"upper\", \n",
    "    order=\"hclust\", method=\"pie\",\n",
    "    p.mat = res2$P, sig.level = 0.01, insig = \"blank\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nOzdfVxT1/048HOTEAgSeRJEBAmBhidFJW2xNpUqdmJbRcFOhmhrtW62U+vQttNJZa2dD1T8Tl2d1u++L93P2OmUirW1KmpFFFeoDxQhBhqEVKWEBwMh5On+/rhdliUhBEg4N+TzfvFHcnLuzScKHy7nnvM5BEmSCAAAAP0wcAcAAADAOhbuAOhOLBb/7//+L+4oQL8lJyd/+OGHuKMAYFAgQfehurr6/fffF4lEuAMB/UCSZGpqKu4oABgsGOIAAACaggQNAAA0BQkaAABoChI0AADQFNwkBL06evRoQ0MD7igGgiTJ+/fvb9u2DXcgAzRt2rRnnnkGdxQAP0jQoFd/+ctf8vLycEcxQOHh4aNHj8YdxUDU1NR8/vnnkKABggQNbGAymTNnzsQdhdvhcrlNTU24owC0AGPQAABAU5CgAQCApiBBAwAATUGCBgAAmoKbhACY02q1mzdv1uv1WN79wYMHEonkvffew/LuCKHs7OzExERc7w5MQYIGwNzjx4+vX7++fft2LO+u1+s7Ozt9fX2xvPuXX35ZXl4OCZomIEEDYIW/v79QKMQdBQY1NTUqlQp3FOBnMAYNAAA0BQkaAABoChI0AADQFCRoAACgKUjQAABAU5CgAQCApiBBAwAATUGCBgAAmoIEDQAANAUJGgAAaAoSNAAA0BQkaAAAoClI0AAAQFOQoAEAgKYgQQMAAE1BggYAAJqCBA0AADQFCRoAAGgKEjQAANAUJGgAAKApSNAAAEBTkKABAICmIEEDAABNQYIGAACaggQNAAA0BQkaAABoChI0AADQFCRoAACgKUjQAABAU5CgAQCApiBBAwAATUGCBgAAmoIEDQAANAUJGgAAaAoSNAAA0BQkaAAAoCkW7gD6plKpGhoacL37iBEj2tra7t69i+Xdu7q6vL29CYLA8u5RUVG4PjhCqKura8SIEbjeOiwsDNdn1+v1Go2Gw+FgeXetVkuSJK7PThAEn89ns9lY3p2GCJIkccfQB4lEcub691oPLu5AMOB23p/i0eCB9LgDweCud2T8E164o8CguUXX1dweye3GHQgG9zs5ic+/EhoaijsQu9RMjrXxaux3NYN/Cxe4gkYIadm+Kq8g3FFg4NPVFMVUeBM63IFg8INnVFy0OyZoBqOn47E21q8TdyAYqHRM3CH0A8PD6dHCGDQAgI50Oh1BELm5udTTgoKCzZs363Q6Ly8v6lUGgxEdHc3n89PT09va2qhuJ0+eJAiipqbGeBKqv5HBYHj66ad5PF5ERMS6detoPoQACRoAQFMcDuf48eMtLS1WX2Wz2VKptK6ujsvlFhYWUo1isVgkEh09erS3czIYjFOnTslkstra2rKysrNnzw44PAabaeNrwKf9r7dwyFkAAMDh2Gz28uXLjcnXKoIgUlNTpVIpQqizs/Pq1asHDx60kaARQiEhIQghg8FgMBgGEx7Dg2njazBnNsI/Bk2S5K1bt/T6Xu+DKRQKOsQJABh6q1atSkxMfOedd3rroNFoiouLk5KSEEJFRUVpaWkCgSAgIKCyspJqtCohIaG+vj4nJ2fWrFlOidtB8Ce+Bw8eLFq0aM6cOb114HK57KhnhjIkAABN+Pn55eTk/PnPf7acd9jT08Pj8RgMhkgkevvttxFCYrGYepCVlSUWi20k6O+//769vT0jI6O8vHzKlCkDi81R4xg24E/QJElOnjx569atvXWQSCSfVzYNZUgAAPr43e9+JxQKX3/9dbN2T09PmUxmfKpQKEpKSqqqqgiC0Ov1BEFs377dxmn9/PxmzZp1+vTpASdopvMTNIxBAwBobdSoUZmZmQcPHrTd7fjx40uWLGloaJDJZI2NjZGRkaWlpZbdmpubqYVv7e3tn3/+eVxcnFOCdhBI0AAAulu/fn1zc7PtPmKxeP78+canmZmZR44cQQj19PSE/duOHTva29vnzJkTGho6ceLE559/Pjs7e8BRucVNQgAAsMRisdrb26nHY8aMUavV1GPqAYvFMrZQLl26ZPp09erV1APLmc63b992SITDZwxar9c/fvzY6ksdHR19zhVnGHQMndp2n+Gq08DWE+74h45eTyo7BzULykWp1YYePUOpcceLJ40BT9kZ2hqib4KdO3cePXo0ICDA8qWenh6NRmPjWKVS6dX9yEPd6rTo6ItJ6r4dmcAgaL3YyUm6ulmXKtzyg3eRSO39uMUdCwZ1aliju12mCIn9V9CNjY1Lly6tqanx9PRcu3btb3/7WzsPHKIE3dPTs23btpkzZ1q+JJfL3333XRvHcrlcpVdoF3uU06KjrzEd381IZnp7ueNlxamrHtNnjccdBQb19x6p628/Fd6FOxAMKpu8cZXxGwBmfwaa8/Lypk2b1tzcnJSUNGPGjPj4eHuOcse/nQEAYCiFh4dPmzYNIRQcHBwTE/Pjjz/aeSAkaAAATRUUFAgEAoFAkJiYWFpaalkgyWqLWXWkxsbGmTNnhoWFRUVF7dmzB5kUWoqOjk5MTLxy5crAwhtALQ6JRCKRSJKTk+19i4FFBgAATlVZWXngwIGKigqJRHL+/Plx48YhawWSrJZMMpOXl9fU1HTt2rWtW7dWV1cbj5JKpR988MF77703sAj7O82uvb39lVde2b9/P5drb3V7SNAAADp6+PBhUFCQj48PQig4OJhK0BTTAkm9tRjZHl5QqVT+/v7O+gwm1Gp1enr6mjVrZs+ebf9R+KfycDic6urqF154obcOPB4v6vmsoQwJAIDd9OnT8/PzY2NjZ86cuWDBgunTpxtfMi2Q1FuLJdPhBY1GExsbq1ar29raLly4MLAI7Z/FodfrFy5cmJaWZrlg3Tb8CZqqO2Wjg0QiOfGvxiGLBwBABxwO5/r16zdu3CgpKVm8ePGGDRtWrFhhWSDJssUq0+EFnU7HZrOpov6lpaWLFi2qrq5mMvu96sT+WhyXL18+depURUXF3r17EUK7d+82XfRoA/4EDQAAVhEEkZycnJycHB8fv2vXrhUrVpgVSNLpdJYtluexMbwgEolaWlrkcrnpEIrDzZgxY2Bbt0CCBgDQUU1NTVdXl1AoJEmyvLw8IiJiYOexPbxQUVGh0+moEv79NQR7EkKCBgDQkV6vz83NlclkTCYzPj5+//79dh5IVUeiHq9Zs0YoFJoNL8yZM4caGEEIcTicw4cPs9kDWbc5fGpxAABAvyQkJJjVP0L/rpRkZFkyicViWQ4m2NNCTy6QoJVKpZ/qBz9VA+5AMGAg/bGzWsIta3EYSOY/j1zHHQUGBr0BGbxqmj1xB4KBzkD4xahwR2Evul9Ba7XaO3fu2PO76MGDBzZ2HbSNy+Xeah55X+kyK/Qd6Bfjml9ZMI7Dcfr3AQ19feaHXzw3AncUGEgbNI81XklJgbgDweD27VZvb2/cUdiL7mPQJSUl77///sSJE/vsefPmTZFINOA30pOE1uCOa2pIErHZDE9Pd0zQDAbhyXbHKlEsJsFiue9/Ou4Q6GVQCdpgMMyfP992LTrKhx9+GBQUNJj3AgC4m9bW1rVr1547d87LyysmJubjjz8WCAQeHh5jx46lOly4cCEqKsrHx8dsJBohdPLkyYyMjLt378bGxiKEqLnPfD7fYDAIBIJDhw4FBwcPNj7nX0G742UpAMAlZGdnh4aG3r9/v76+ftOmTfX19QghT0/Ppn+LiYnp7VixWCwSiY4ePWpsMVbtCAkJKSgocEB8HixbX44ACRoAQEeVlZVSqXTLli0sFgshNHXq1JdfftnOYzs7O69evXrw4EHTBE0hCCIlJaWpqcnB4ToH5lkcmzdvtvzbxAyXy0VIMDTxAABooqqqKikpicEwv4g0TmHm8XiW8/AoRUVFaWlpAoGAqiRhWqNDr9d/9dVXmZmZDgiR5jcJB4kkyS+++GLfvn22uykUioYbHUMTEgCAPqzOEDNb222VWCym6nJkZWWJxWIqQVMFkuRyeXx8fHp6ugPiG94JGiHE5XKFQqHtPhKJBN24NTTxAABoIiEhIT8/32AwWF5E26ZQKEpKSqqqqgiC0Ov1BEFs374dIUQVSOro6MjMzNy2bVteXp5zAnckGIMGANCRUCiMiorasGEDtal0WVlZcXGxPQceP358yZIlDQ0NMpmssbExMjKytLTU+Kqvr+/u3bv37NmjUg16RQzcJAQAuC2xWCyXy8PDwyMiIrZs2SIQWL8XRRXfoOzYsUMsFpsW88zMzDxy5Ihp/7i4uNTU1IMHDw42Pg+mra//dvHixZiYmMjIyI0bN9r/Di6w1BsA4J4CAwMPHz5s1thn8Y3169ebPl29erXlgWKx2JGB9oUkyeXLlxcVFcXFxU2dOvWll16aOnWqPQe6QILWaDSR3K4gTh+TPYYlT6ahrKyZyXTH5VWPOw0Xr3fhjgKDx0pSS+o6O7W4A8Ggra0nONhKQWeasvsm4c2bNwMCAiZMmIAQWrx48YkTJ4ZPgkYIefmOJFj2brM4nBCq+yMDQ1jOv1lMQwrZrdHtbbijwIDUePeM9AsZ6Y5XJFrVACv2YEHYPdAsl8uNqx/Dw8PLysrsPNAFEjSbze7x8O1ij8IdCAYj1fII/mgvjjsWNvuxsjLWux13FBgwCKT0D4zju8DPpsOp1CS1LGWYMR2EMRgM9h84dP8WlhsPkiTZ1eWOf8MCAOwxbGpxhIWFGdcuNjU1GePv0xDN4nj55ZdJkqyw0N3dPTQBAABcDu1rcdg7i2PixImtra23b9/WarV///vf582bZ+c7DNEV9KRJkyZNmmTWSJLk119/PTQBAABcC1WL48yZM9RCFequmtU9YS1RtTguXrw4d+7czZs3m75E1eI4d+6cE0LuFYPBOHDgQGZmplqtXrRokf21l2k63KNQKIyrORUKBULueJcMAHfmCrU4+pE/U1NT79271993oGmCLiwsbGhooEZquFwukzcFd0QAgKEGtThomqBJkly5ciX1R41EIjnxr0bcEQEAhhTU4kCw1BsAQE+uUIujH0u9BwYSNACApmhfi8PpxZJoOsQBAADDphbHgGFO0Fqttq3NynJe6o8aI4ZBxzK448pXAqGuTrVO14+lR8OGnkRKnTteQKgNzB4Nqexyx/90rdbKXUH6Gt43CQmCGDVq1C9/+UvLl+rr65977jnqsVKpHKlv4WoeD210tMAkddXfVffzHskwodazSjvG4I4Cgy69h0Gnf9zpjgla2UUGj3OZxWvE8E7QCKGTJ09abd+4ceOoUT8X3+ByuTq/cC03ZAjjogtOw7UZM8Z4e7vjheTpQy2zA9xx9o6k2085OuypBHf8T6+4q+NwOLijoBF3/CYAAAAHYDj9Ctot/3gGANCeTCaLjo42Pl23bl1BQYFOp2MwGNHR0Xw+Pz09va2tTafTEQRhnMVRW1uLEMrJyQkKCho/frxzQ2QwbX055B0cchYAABgaxppHXC63sLAQWSuf9MYbb5w5cwZ3pA4ACRoA4HoIgkhNTZVKpVZfTUlJ8ff3d3oQzr+CpukYdGho6Lvvvuvl5YUQ4vF4E19ajDsiAACNaDSa4uJiqsiGPeWTnGJAWbixsXHp0qU1NTWenp5r16797W9/a6MzTRP0W2+99dZbb1GPJRLJ19UP8cYDABhiBGG+FSfVQqVjBoMhEomoikj2lE+ilby8vGnTpjU3NyclJc2YMSM+Pr63nj8n6AULFnR0dPT3bRQKhWkdPwAAcJTAwMDW1lbjU4VCQW26apaO7awQ7RQDuoIODw8PDw9HCAUHB8fExPz44499J2iFQnHx4sX+vtOXX355+/btAYQIAAC2+fj48Pn8oqKiefPmPXjw4Pz58xs3bsQd1H8b3ECzRCKRSCTJycm23mEwbwAAAM5z6NChnTt3RkdHp6amfvTRR6az7mzLyMh47rnnamtrw8LCHFAUyQna29tfeeWV/fv3c7lcG91oOgYNAADx8fHffPONaQuLxbIslmS5Y+yJEyecHhxCyO4iDIWFhXv37kUIXb58eezYsWq1Oj09fc2aNbNnz7Z9oAskaKVS6fHTPY8W6/NphjeC1B879oPFzRK3YNB4/f3hE7ijwEBPIqTU3f0B39AqPjo98hsz6DLNQ8buIY61a9euXbuWeqzX6xcuXJiWlvb666/3eaALJGgul8tPnBw4NgJ3IBjcvvhl4v/uZXe7zres49RtWP/CC+NwR4FBXV1Hy6krCT/V4A4Eg5pRAm/vl3FH4VyXL18+depURUUFdU29e/du0+rVZlwgQSOEGEwWi83GHQUerJ4eVk8P7igwYBCEp6c7bhbMYjJYiGTrtbgDwYBJ6nGH0B/EQL4/Z8yYYXWvRavgJiEAgKZaW1tfffXV0NBQPp8/e/bs6upq1P86G9u3b4+Pj6fWVVgea9bS2Ng4c+bMsLCwqKioPXv29HFqqMUBAHBb2dnZoaGh9+/fr6+v37RpU319PepPnQ2SJA0Gw1/+8pfy8nJqPMHyWMuWvLy8pqama9eubd26lfqVgBEkaAAAHVVWVkql0i1btrBYLITQ1KlTX375ZWStzsaBAwcmTJgwfvz4/Px8hNDDhw/j4uLeeuut2bNnL1y48Mcff5wzZw61LaHlsWYt4eHh06ZNQyarSGyF6A61OJqbm3fu3GmjA5fLnTJrzpDFAwCgg6qqqqSkJEZfU9lqa2t37Njxr3/9y8vL6/nnn3/22WfHjx9/7969oqIiqrJdSEjIAAp02LOKZAjqQeNP0HV1dY8ePbJRMUShUAxlPAAAmrDnZlppaenLL7/s6+uLEMrKyvrmm2/Gjx8fFRVFZeeBsXMVyRDAn6ARQmPGjBEKhb29KpFIGh+7446xALizhISE/Px8g8HQ50W0pcHsm2X/KhLYUQUA4KaEQmFUVNSGDRs0Gg1CqKysrLi42LKbSCT64osvOjo6enp6jh49mpKSMpg37dcqEsRg2PpyBEjQAACaEovFcrk8PDw8IiJiy5YtAoEAWdTZiImJyc3NfeaZZyZNmpSWlpaammrjhJY1OsxaqFUke/fupTbQ6m1X6yFDiyEOAACwFBgYePjwYbNGyzobK1asWLFihfFpSEjIzZs3jU8fPvxPNXnLYy1b7F9F4to3CW/dulVeXt5nt/r6+ra2NhsdNBrNI5m04yd3rNmv6+mRTktl6NxxUZlKqb10qQl3FBg87ujR+IzuCvPEHQgGj718wzHWd+4vl07Qn376aWBgYGho6OBP5ec3wj8oYPDncTntD5v4T3A8kAfuQDCo8mD7BY3BHQUGWkO7B6NpLLMTdyAYkIyB39wblpw7xLFgwYI+V2Reu3bN6ti/EZvNHhUSHMZ3x2JJ0qqaqJ4Wb8Idr6DveXpECcbijgIDBoOh/r5ToJbjDgQDFcOTWpbiImAWBwDALel0OmrbaKstCoXitddeCw0NjY2NfeaZZ6i7eQaD4emnn+bxeBEREevWrevHaPKAGEiWja8+jjUYpkyZIhKJbHeDBA0AcD1ZWVmhoaENDQ01NTVisfjevXsIIQaDcerUKZlMVltbW1ZWdvbsWdxh9uqvf/0rn8/vsxskaACAi/n2229lMtmHH37o4eGBEOLxeO+88w71UkhICELIYDAYDAZnh2FATBtfNg5sbm7+7LPPqAJ7ttFiuOfBgwcVFRW9vapQKPxCg4YyHgAAnVVXV0+ePLm3FYYJCQn19fU5OTmzZs1yahh9jmP0Jjc398MPP2Qy+x7Cxp+go6KiRo8efezYsd46cLncF9JfHMqQAAA0Zxxfzs3N/eKLL0iSrK2tpVq+//779vb2jIyM8vLyKVOm4IvRuosXLzIYDJFIdP369T4740/QwcHBW7dutdFBIpG0a7qGLB4AAM3Fx8d/8MEHVJmOjz/+OD8/n8fjmXbw8/ObNWvW6dOnnZqgDaS9szhMN40tKys7f/48j8fr6elpa2ubO3fuqVOnejsQxqABAC7mySefHDdu3MaNG6kyHe3t7VR7c3NzQ0MD1fL555/HxcU5NQwDYtn4Mu25du1aqVQqlUrHjh27ceNGuVwuk8lOnjz55JNP2sjOCBI0AIC2enp6wv5tx44dpi999tlnVJkOgUCwcOHCffv2IYTa29vnzJkTGho6ceLE559/Pjs7G1PgDoN/iAMAACyxWCzLicxq9c+Vh0eNGnXo0CGzVwUCwe3bt4ciOIRQf4Y4rJoyZUppaantPq6RoDXqHlWnOw5DkwayE7H1JIE7EAz0ekOnsht3FBj0qDUaxFIyvPruOuxoCdfISJRBJmh7uMA/h1KpvF93/9GPP+EOBANNT8+3/vEMd8zPSPPgx+/E9bijwKCLyTF4BHZ4u2OC7vLwHtPtjr+Ve+MCCZrL5UbGJ4RGRuEOBIPLRf+cMbnTLX9UUfGJnpkt/8IdBQbSEWGPCc6kthrcgWBw2++JwWyGMsTIgc6Dtt+g3sDLy+vTTz89f/681VclEsmvfvWrwZwfAABoy/ZyQYcYVIKePn06tQTeqlWrVo0cOXIw5wcAuC2dTsdms/l8vlarHTly5LJly1avXs1gMHQ6nY+Pj/FuYUFBQUtLy9atWxsbG6dNm1ZRUREQENDW1iYUCi9fvjx27NgpU6Y0NzeTJPnKK6/s2LGDJEmzFoKg7xgiTLMDANAUm82WSqUNDQ3FxcVisXjLli02OoeHh69cufK9995DCL333nsrV64MDw+3LJ/kwIJKg6lmZydI0AAAuuPxeJ988snu3bttd1u7du3169d37dp17dq1t99+m2q0LJ/kqIJKBpJp42swZzbCdpPwypUrubm5vr6+ffbk8Xivr357CEICANDWxIkT29raFAqFr69vT0+PcW3348ePjRsSenh47NixIy0traSkhCp0R7EsnzRkBZUGCVuCVigU2dnZxt9yNkgkEkWPO24pAgAwIknSuG7F09NTJpNRj6kxaGO3L7/8csyYMVVVVdOnTzc2WpZPckhBJUeNY9gAQxwAABdw69atgICAwMBAG31u3rx57ty569evf/zxx83NzaYvGcsn2WjprwHXg7YfJGgAAN3V1dW9+eabq1atstGHJMmVK1fu2rVr3Lhxubm569evR9bKJw1xQaVBcoGFKgAA90SNNWu1Wn9//2XLlq1Zs8ZG5wMHDowbN+6FF15ACL355ptPP/10aWlpcHDwggULWlpaPDw8Fi1alJ2dfe/ePbOWAYc3BEMckKABAHRktVgS1W6cBI0QWrduHfVgxYoVxruFTCbTuEmTWfkkBxZUGvBUjZ9++mnZsmU3btzw9vb+5z//OXny5N56ukCCViqV1bdu1VS446pfnU537DKNp9E7k8Fj5GdjUnFHgYEBESTBuOcTjjsQDPQEM0ilwh2F0/3mN7+ZMmXK559/3tnZqdXamgHhAgmay+UGCyaMDHHH79f6snMv/njey6DBHQgGF0OTX/Bvwh0FBnVq38eckUljh3+esnT7gbe3tzfuKOxFDuhO4MOHD0tLSz/77DOCILhcru3OLpCgEUIMJovpwcYdBRYEm9R5kjrcYWDAIJAnw+kbM9MQiyBZDOTJsvLX/bDHJFzpU2t1A4lWKpWOGzfu1Vdf/e6776ZMmbJ79+4RI0b01hlmcQAA6Ein0zEYjOjoaD6fn5aWRk2bMzZGR0cnJiZeuXKFaiQIYunSpdSB3d3dXC43KyuLevrTTz/NnTs3JCSEz+d/9913uD6OkU6nq6ysXLlyZVVVFUmS27dvt9HZuQn68ePHbb3o7Owc5DpLAMDwRtXiqKurCwkJKSgoMG2USqUffPABVXkDITRy5MjKykpqi8LTp0/z+XzjSagB3wcPHty6dSsiIsKB4Wn1Bhtfpj0LCwupXypyuTwsLCwkJEQkEjEYjIyMjJs3b9p4CycOcUydOnXXrl29vSqXy4VCofPeHQAwPBAEkZKScu7cObN2lUrl7+9v7POLX/zi7Nmzc+bM+eyzzxYuXEhN1ejXgG9/mWVhG9auXbt27Vrj06CgoNu3bycmJp47d278+PE2DnRigv7Vr35lox50UVGRTCa7d+/ewYMHbZ+Hy+VOmvGio6MDALgGvV7/1VdfZWZmUk81Gk1sbKxarW5ra7tw4YKxW1ZW1s6dO59//vnGxsacnBwqQfdrwHfI7Nu3LycnR61WT5gw4W9/+5uNnphvElKX96+88oqNPgqFAipxAOCGqFwsl8vj4+PT09OpRjabXVNTgxAqLS1dtGhRdXU11S4UCqurq8Vi8dy5c41noAZ8CwsLp06dumzZsu3bt+fn5zsqPPuvoM1MmTLFzrnY+GdxhIeH2x7rkEgk9xTuuGMsAG6OysUdHR2ZmZnbtm3Ly8szfVUkErW0tMjl8tDQUKrlxRdf/P3vf19eXl5VVUW1GAd8EUIZGRmffvqpA8PT6Z0+5wRmcQAAaM3X13f37t179uxR/fcaloqKCp1ORxV3przxxhsffPBBdHS0sSU6Opoa8EUI9TngS0P4r6ABAMC2uLi41NTUgwcPrly50lgMmsPhHD58mM1m63Q/LxTg8Xhvvvmm2bH2D/j214CHOOwHCRoAQEdmNTfEYjH1wLJAB4vFam9vN22ZN2/evHnzqMf2D/j2l1YHCRohjUbTer+us+UR7kAw0Gt6yn3imKQedyAYKHUel9rH4I4Cg8c6trbHo0vj4DlhLqFVxQzWueO62d64QIJGCHWSbNLAwR0FBkxEhEwO9mC60vpXR2m72xPCdMubw3qkVmtGa9pwx4GBhvTDHUI/wBAHQgix2Ww0IsAwMqTvrsMOU9EQHar19sIdBw51NYZYL3dMUgyCfKxnxjJ+wh0IBiq9B4vlAkmJoh3eszgkEsmdO3c6OzsxxgAAoK3W1tZXX301NDSUz+fPnj3bOOX55MmTBEFQs6EpBQUFAoFAIBAkJiaWlpaa1vFIT09va2tDCOXk5AQFBZlO5Lh48WJMTAtgZ/0AACAASURBVExkZOTGjRuH+KPZCVuCfvLJJyMiIurr67u63PLPWABAX7Kzs0NDQ+/fv19fX79p06b6+nqqXSwWi0Sio0ePUk8rKysPHDhQUVEhkUjOnz8/btw4ZFLHg8vlFhYWIoTeeOONM2fOGE9OkuTy5cuPHz9+7969c+fOlZWV9Tc8nd5g48sBnx9jgg4LC3v33XfT09NHjx6NKwYAAG1VVlZKpdItW7ZQgx5Tp059+eWXEUKdnZ1Xr149ePCgMUE/fPgwKCjIx8cHIRQcHEwlaApBEKmpqVKpFCGUkpJirN2BELp582ZAQMCECRNYLNbixYtPnDjR3wjtL5Y0YLRbqFJfX1/x34z7qwMA3EdVVVVSUhKDYZ6jioqK0tLSBAJBQEBAZWUlQmj69OlarTY2Nvatt966ePGiaWeNRlNcXBwfH295frlcPnbsWOpxeHi4XC53zucYFNqNx+fk5EybNs20hcvlBiSm4IoHAICL1T0JxWLx22+/jRDKysoSi8VJSUkcDuf69es3btwoKSlZvHjxhg0bVqxYQa1nYTAYIpGI6m/j5AMrfeyOszg8PT23bt1q2iKRSC7UNuOKBwCARUJCQn5+vsFgML2IVigUJSUlVVVVBEHo9XqCILZv304QBEEQycnJycnJ8fHxu3btWrFihaenp+0/vsPCwpqaft5TrampyXg1bb8Bz+LYt2/f//zP/yCEYmJiDh8+bKMOKu2GOAAAACEkFAqjoqI2bNhAleEvKysrLi4+fvz4kiVLGhoaZDJZY2NjZGRkaWlpTU0NtYc3SZLl5eV2VuWfOHFia2vr7du3tVrt3//+d+PKQ2dra2vbtGlTWVnZ3bt3uVyu7XrLkKABADQlFovlcnl4eHhERMSWLVsEAoFYLJ4/f76xQ2Zm5pEjR/R6fW5uLo/Hi46OvnPnzp/+9CerZ8vIyHjuuedqa2vDwsIOHjzIYDAOHDiQmZnJ5/NnzJhBVbzrF63OYOOrt6NIkiRJUq1W6/V6tVptLMVnFe2GOAAAgBIYGHj48GHTlkuXLpk+Xb16tdV2hJBpHQ+K5TyN1NTUe/fuDTi8gc2lCwgI2Lp16xNPPDFixIhnnnnml7/8pY3O+BN0d3c3NY2cYn20XqdBmu6hi4lGyE41cv6tCDrSk6RS74E7CgzUBlYPSShJd9zGXksycYfgdJ2dnX/7299qampCQkKys7P37dv3m9/8prfOmBN0VFTU/v37z549a2yx/IWmVCqZjx8wVa1DGxo9GLQ3buoIl9qL3lHULK9Skt93v2FHZSD0LPSY7Ys7EAw6NazR3S5zKWb/LI7CwsK9e/cihC5fvnz79m0/P7+wsDCE0Lx587766iv6JuikpCSzvSCnT59u1ofL5aJRPNLXHQuboXul0yMfeXu44yX0F3WBL453xzIAtY/Yj9sNT4U8xh0IBpWPuByOy5RFs7/cqOmmse3t7d99911LS0tgYODXX38dExNj40C4SQgAAEMnISFhzZo1U6ZMEQgEXV1dxmF0qyBBAwBoyrJYkrEKUkRExIQJE3bt2kXdtTIYDE8//TSPx4uIiFi3bh21CGXfvn1xcXFxcXHz5s1TKpUOD0+nM9j4snHgu+++K5VK7927d+zYMRuToBEkaAAAbVktlkRVQWpoaCguLhaLxVu2bEEIMRiMU6dOyWSy2trasrKys2fP9mu68cBotQYbXw55C/yzOAAAwBJVLOnMmTPUSsKpU6cihHQm+63weLxPPvkkLS1t06ZNCCFq91iDwUBdU/drujFt0S5Bjx49+oUXXjBt4fF4T817DVM4AAA8eiuWZGrixIltbW0KhSIwMBAhlJCQUF9fn5OTM2vWLIIg7J9uPDBandP3oqNdgjaWEDSSSCSX7rnj7hIAuDmrxZLMOpj2+f7779vb2zMyMsrLy8ePH2//dOOB0TloHMMGGIMGANBRQkJCZWWl7Tpzt27dCggIoC6fKX5+frNmzTp9+vSVK1eo6cYsFmvevHkDqMdPB5CgAQB0ZLVYkmmHurq6N998c9WqVQih5ubmhoYGhFB7e/vnn38eFxc3btw4aroxSZJ9TjcemIHV4ugX2g1xAAAAhSr9HB4e7uXlNX78+J07dyKEqELPWq3W399/2bJla9asQQi1t7cvWLCgpaXFw8Nj0aJF2dnZBEFQ040Jgpg0aZLt6cYDo9O63xg0AABQLIsloV4GpgUCwe3bt80a33333XfffddZwQ0JF0jQSqUSPaghHkpwB4KDQXe8OpDAHQUWBkQcvu6O9SioAg81rd64A8FAZyD8VCrcUdjLUeMYNrhAguZyuc9Gq/jBWtyBYHDiX94pNZe89D24A8Hg+IQs3dhE3FFgwFQ+mujflDTR1gKz4er2913e3i7zm0kLQxwUFpP09HDHim4EQh4GLdvgjr+cCIKBmO5YbpQkmCwWw5PtjjfwbU56dkfw7wEAoKmCggKBQCAQCBITEy9cuBAWFhYWFubn58flcqnHUqmUKs3B5/PT09OpyvIKheK1114LDQ2NjY195plnTp48SZ0tJycnKCho/PjxjgpvwLU47AcJGgBAR5WVlQcOHKioqJBIJOfPn3/iiSeampqampo2b97861//mnrM4/Go0hx1dXVcLrewsBAhlJWVFRoa2tDQUFNTIxaLjSXm33jjjTNnzjgwwgHX4rD8VXHx4sWYmJjIyMiNGzea9oQEDQCgo4cPHwYFBfn4+CCEgoODx40bZ6MzQRCpqalSqfTbb7+VyWQffvihh4cHQojH473zzjtUn5SUFH9//yGIvE9mvypIkly+fPnx48fv3bt37tw50zU1zh2D1uv1mzdv1moHNYTK5XIXTHPHGyYAuLPp06fn5+fHxsbOnDlzwYIFllt5mNJoNMXFxUlJSdXV1ZMnT7ZdwcNRdAOtxZGSkiKVSo1Pb968GRAQMGHCBITQ4sWLT5w4QVWGQs5O0CqV6vLly9TfHQOmUCgQqnFUSAAAl8DhcK5fv37jxo2SkpLFixdv2LDhzTfftOxGrVthMBgikejtt98+ceKEcaJ0bm7uF198QZJkbW2tMyJ0VE1RuVw+duxY6nF4ePjQXUEjhPz8/IRC4WDOIJFIyEeQoAFwOwRBJCcnJycnx8fH79q1y2qC9vT0lMlkxqfx8fEffPCBwWBgMBgff/xxfn4+j8cbsoAHxnTpjVntERiDBgDQUU1NTUVFBUKIJMny8vKIiAh7jnryySfHjRu3ceNGqoJHe3u78yK0vxZHYWFhdHR0dHS0XC63PE9YWFhTUxP1uKmpyXg1jSBBAwDoSa/X5+bm8ni86OjoO3fu/OlPf7LzwM8++0wul4eHhwsEgoULF+7bt49qz8jIeO6552pra8PCwhyywYpOq7fxZdpz7dq1UqlUKpWaJl+jiRMntra23r59W6vV/v3vf583b57xJddYqAIAcDcJCQmXLl2ybH/77beNj1ksllqtNuswatSoQ4cOWR544sQJhwY4cBkZGdeuXWtpaQkLC8vPz1+2bNmBAwcyMzPVavWiRYtEIpGxpwskaI1GI5V7yFuZuAPBoFtLVI5OZBqcvqKUhghtN/PB97ijwIDQqBq61Z1d7vif3tauDQ7V9d2PHgZci8PyV0VqaqpxvrYpF0jQCCFGpYT500PcUWBACJ8N5XM8GE6vyUJDCrXXOIFdw47DTOujZlT77aiGR7gDwUDtFYQ7hH5w1HJBG1wgQbPZ7IDmH0Okd3EHgkHDhCejRnZ6s9wxQVe3eYZHP4E7CgwIBgN9r3yiqxF3IBioGJ4slgskpSFj/d9CrVaXlpYO/uwqlaqzs3Pw5wEAuKHW1ta1a9eeO3fOy8srJibm448/FggEHh4exlttFy5ciIqK8vHxsRyJPnnyZEZGxt27d2NjYxFCOp2OzWbz+XyDwSAQCA4dOhQcHDzI8L7e/tIgz9An6wn6u+++27JlS1pa2iDP3tPTo3Kd6q4AAFrJzs6ePHny/fv3WSxWWVlZfX29QCDw9PQ0TkpDCOl01sesxWKxSCQ6evTo5s2bqRaqagdJkkuXLi0oKNi+ffsQfIRB6vWviWeeeWbwmxEolcpvv/12kCcBALihyspKqVR65swZat02tfq5t3RsprOz8+rVqxcvXpw7d64xQVMIgkhJSTl37pwTQnY8nMM9JEnevHnT9q69CCGFQuEyFbwBAA5SVVWVlJRkWVWDWtuNEOLxeFbn4SGEioqK0tLSBAJBQEBAZWVlUlKS8SW9Xv/VV19lZmY6K26Hwpmg6+vrX3vttdmzZ9vuxuVyU4YmIAAAnVjdftBsbbdV1G6zCKGsrCyxWEwlaI1GExsbK5fL4+Pj09PTnRCv42G+gn766ae3bt1qu5tEImn+yHw7SADA8JaQkJCfn09V1ejXgQqFoqSkpKqqiiAIvV5PEAQ13Mxms2tqajo6OjIzM7dt25aXl+ecwB0JlnoDAOhIKBRGRUVt2LCBqqpRVlZWXFxsz4HHjx9fsmRJQ0ODTCZrbGyMjIw0nZPm6+u7e/fuPXv2uMT8BUjQAACaEovFVFWNiIiILVu2CAQCq916enrC/m3Hjh1isXj+/PnGVzMzM48cOWLaPy4uLjU11SHlOJwN5oQDAGgqMDDw8OHDZo1mU55ZLJbZUPX69etNn65evdryQLFY7MhAncbpCVqr1VI7OVrq6OiwehPAkobt1e0z0qFxuQaSQXRqWfq+JroMSwa9XuWWq5w06h5EMDuZHNyBYKAl3LHkjg3OTdBeXl4sFuuXv/yl1VdVKhW1b5htSqWyeUKiMtYdV/1qvUf8S+nNIHDHgYOqu7vi6jXcUWCgU6vYQRFdPHesQ6JUodDubtxR0IhzE7SHh4eNcX2pVLpt27Y+T8LlcmNb7vHaGhwamms4HTNrxkSmtyfuOHA4csM74slpuKPAoF0uC9Z891S8O94fqqjRczju+KdDb9zxmwAAAFwCJGgAAB3JZLLo6Gjj03Xr1hUUFOh0OgaDER0dzefz09PT29radDodQRDGWRzU/rA5OTlBQUHjx4/HF75jQIIGALgSquZRXV0dl8stLCxECFHlkygxMTEIoTfeeOPMmTO4I3UASNAAANdDEERqaqpUKrX6akpKir+//xCH5Aw450H7+PhUVFS88MILtrvxeLy3QmHyDQDgPzQaTXFxMVVkw57ySS4KZ4IOCQmprKzss5tEIlH9v51DEA8AgD4Iwnx6KdVCpWMGgyESiaiKSPaUT3JRsJIQAEBHgYGBra2txqcKhWLChAnIIh3bWSHaRcEYNACAjnx8fPh8flFREULowYMH58+ff/bZZ3EHNdQgQQMAaOrQoUM7d+6Mjo5OTU396KOPTGfd2ZaRkfHcc8/V1taGhYW5RFGk3sAQBwCApuLj47/55hvTFhaLZVksyXLH2BMnTjg9uCHhAglaqVTeGpN4M8Tl55wPgJbJPlZKumUpDqQl22pLTuGOAgODQd+M9Hd/0OMOBAOdAfmFukCZ5iHjAgmay+Umj+7gj3DHwmYnmsJGJP6CyXbHYhzkrZMv8KzXQRze6to55dpENMbeP+eHE+LRD97esAXpf7hAgkYIsQjSk+mOJTcJRDJYLAar75p/w4+BQbjnfzqTIAkGk2C64386IuCu2H+Bfw4AAE21tra++uqroaGhfD5/9uzZ1dXVlpU3dDqdl5eX5bEnT54kCKKmpoZ6alrEIy0trbm5eWg/ygBBggYA0FR2dnZoaOj9+/fr6+s3bdpUX1+PrFXesEosFotEoqNHjxpbjEU8QkJCCgoKhuIDDBokaAAAHVVWVkql0i1btrBYLITQ1KlTX375ZTuP7ezsvHr16sGDB00TNIUgiJSUlKamJgeH6xy0G4PesWOHQqEwbeFyuQsS3HMiAwDuq6qqKikpicEwv4i0p/JGUVFRWlqaQCAICAiorKykSnZQ9Hr9V199lZmZ6ay4HYp2Cfqf//zn3r17TVsUCgVSfYsrHgAALlb3LLWn8oZYLKbKdGRlZYnFYipBazSa2NhYuVweHx+fnp7uhHgdj3YJmsPhCIVC0xaJREJWQ4IGwL0kJCTk5+cbDAbLi2jbFApFSUlJVVUVQRB6vZ4giO3btyOE2Gx2TU1NR0dHZmbmtm3b8vLynBO4I8EYNACAjoRCYVRU1IYNGzQaDUKorKzMxganpo4fP75kyZKGhgaZTNbY2BgZGVlaWmp81dfXd/fu3Xv27FGpXGBFDCRoAABNicViuVweHh4eERGxZcsWgUBgtVtPT49x4t2OHTvEYvH8+fONr2ZmZh45csS0f1xcXGpqqkvU6KDdEAcAAFACAwMPHz5s1mhZi8NsqHr9+vWmT1evXm15oFgsdmSgTuOUBH3nzp1r164N7FjTCrAUjUYjfewjV1mZiz7sdesZ+ro77rm8iqlhXmr0wx0FBh1qDz1qJjXmBYDcgUGt1OnccY17b5ySoP/v//6Pw+GMGzfOUSf09UGjuY46mStp6CbYI/0Jhjv+oUO01Y0mH+OOAgOS8Db4RYwOd9iPjwtpkctxh0AvzvrJz8jIMJ17aD/LPz3YbHaor+6JUe54QXFTzuEEhTHZ7vjXg6HxWqx3O+4oMGAQiPQPCI+2Pt46vPV0d1PLUgDFHf92BgDQH1V2Izc3l3paUFCwefNmY0mNiIiICRMm7Nq1y2AwIIRkMplpOf9169YVFBRYFu5AFjU6jG9k2o2q9E/hcDjULD0s4JcVAICmOBzO8ePHf//7348aNcrYSJXUQAjJZLKFCxcqlcpNmzb1dgaqcIdpi7FGx+bNm210o56SJMnj8UznhAwxuIIGANAUm81evnx5YWGh1Vd5PN4nn3yye/du+09oo0aHVaWlpcHBwU888YT9b+FYtLuC7u7urqioMG1RKBQRTFzhAABwWrVqVWJi4jvvvGP11YkTJ7a1tZlV7zFlVrijtxodvdX3OHLkSHZ2tsM+TP/RLkFnZmYeO3bMtIXL5UZM4eCKBwCAkZ+fX05Ozp///GcOx0oSIEmSmgRNEOb11KgWs8IdVmt0WHaj6HS6EydO3Lx501GfZQBol6DNJpkjqhZH42kswQAAsPvd734nFApff/11y5du3boVEBAQGBjo6elpuoRCoVBMmDDBrLPVGh2Wmd3o7Nmz48ePHzNmjKM+yADAGDQAgNZGjRqVmZlpuTK7rq7uzTffXLVqFULIx8eHz+cXFRUhhB48eHD+/Plnn33WrL/tGh2WsI9vIEjQAAD6W79+vXGTKmq8eOzYsenp6VlZWRs3bqTaDx06tHPnzujo6NTU1I8++sh01h2lzxodplQqFR3KRtNuiAMAABBCLBarvf3nlUpjxowxVtKwWiQaIRQfH//NN9+YncG0/oZZdX9jjQ6zbhRvb28b9x6HjGskaLWOoVS748W+ARE6dTdpcMfNrUkSKXWu8f3pWGoDU6vVqjo7cQeCgU6jwR0CvbjAD4BSqaxv9W7q8sYdCAY9OuTbdIXBsH7JMLwpWZyrZBTuKDDo0qPW+oa6+49wB4IBS6+ODgvFHQWNuECC5nK5iVFIEOaOSUp8Ac0QEt5e7vjXwxdX9S8+rccdBQa1jYyzslFdXB7uQDDwVt63Op3ObbnjTz4AALgESNAAAJoqKCgQCAQCgSAxMbG0tNRYKYnP56enp7e1tVlt8fL6r+qPjY2NM2fODAsLi4qK2rNnD0LIeFR0dHRiYuKVK1cwfb6+QYIGANBRZWXlgQMHKioqJBLJ+fPnqfryVKWkuro6LpdL1eiwbLGUl5fX1NR07dq1rVu3VldXG4+SSqUffPDBe++9N5Sfq18gQQMA6Ojhw4dBQUE+Pj4IoeDgYNMNQAiCSE1NpWra9dZiFB4ePm3aNOokMTExP/74o+mrKpXK39/fWZ9h0Gh0k/Cjjz76+uuvPTw8zNp5PN66N57CEhIAAJfp06fn5+fHxsbOnDlzwYIF06dPN76k0WiKi4tNtwSxbLEkkUgkEklycjLVPzY2Vq1Wt7W1XbhwwXmfYpBolKCbmpo++eSTuLg4s3aJREK2X8IREQAAGw6Hc/369Rs3bpSUlCxevHjDhg0rVqyglhEyGAyRSESVPbJssaq9vf2VV17Zv38/l8vV6XRsNpuq2V9aWrpo0aLq6momk441M2mUoAEAwBRBEMnJycnJyfHx8bt27VqxYoVZ2TmdTmfZYnketVqdnp6+Zs2a2bNnm70kEolaWlrkcrkD91B1oJ8TNEEQv/71r42tjx49gtmIAACMampqurq6hEIhSZLl5eUREREDO49er1+4cGFaWprVengVFRU6nS4kJGRwwTrLzwn6H//4x+PH/9lB+bvvvrt27Rr1WKlUlpeX9+ukTU1NVn+PAQCAnfR6fW5urkwmYzKZ8fHx+/fvt/PAnp6esLAw6vGaNWuEQuGpU6cqKir27t2LENq9e/ecOXOMFfo5HM7hw4fZbLZzPsRg/ZygR40aZbrr16NHj4wRnz17dv/+/f3aoruhoaG7u9uBUQIA3E1CQoJZeSOEkFlVI8s6RywWy7Kakj0t9NT3GDRJkunp6W+99Zb9J83NzeVyuYOI6r8olcpbd9D173utqz2MaXXo2EWSQK7xzeRYBpI4dM4d75HoDcjL8KNn10PcgWBAkAaVyrxMqDtzgR8ALpebPK6LH2BeD9AdnLjjn+Yr4zDccbzofMe4F0LcsWBQXafPvxhJnqHYNirFSPOw3tvbHcui9cYFEjRCiMUgPVnueBVJIJJN6D0Z7lhulCCQJ9MdPziTIAkmk2CZLwhwCwxYOvdfsCXovLy8R4/+6/qotLT0tddewxQOAICOTp48mZGRcffu3djYWISQTqfz8fGhxp2//vrrlStXnjt3zsPDY+nSpTU1NZ6enmvXrv3tb3+LO2qHwZagS0pKDh06ZNqiUqkcOHINABgGxGKxSCQ6evTo5s2bTdsvXrz4m9/85ssvv+Tz+Y2NjXl5edOmTWtubk5KSpoxY0Z8fDymeB0MW4L28PDg8/mmLZCdAQCmOjs7r169evHixblz55om6KtXr77++uunT5+OiYlBCIWHh4eHhyOTahuQoJ2oqKjo+vXrxqdcLnfBFFg1A4DbKSoqSktLEwgEAQEBlZWV1GRfrVY7Z86c8+fPJyQkmPU3rbYxPNAxQR87dmzx4sVBQUHUU4VCgVAV3pAAAENPLBZT5TWysrLEYjGVoFks1nPPPXfw4EGzxRmm1TbwhOsEdEzQCKHExMTQ0J+3JpNIJGQjJGgA3ItCoSgpKamqqiIIQq/XEwSxfft2hBBBEEePHv3FL37xxz/+MS8vj+pso9qGS4NJLQAAOjp+/PiSJUsaGhpkMlljY2NkZGRpaSn1EofDOXXq1LFjx/7617+ivqptuDRI0AAAOhKLxfPnzzc+zczMPHLkiPGpv7//V199tW3bthMnTly+fPnUqVN79+4NCwsLCws7efIkjnidgqZDHAAAN2dWiGP16tXUA2PxjbFjx9bX11OPXaW2Rn/RK0H/4x//GDNmTH19vWkxPI1GI33kKe9wx4VV3TrGtcchTMIdF9QptaxLj4JwR4FBh4alYzaTPe5Y20DfrdTpoBbHf9AoQa9YseLGjRtWX/L10I721AxxPHTQgDxGR47wYA7PqwPbWu9qR+vbcUeBAUn6tLO92L6BuAPBQEvqcYdALzRK0JMmTZo0aRJC6PLlyyzWfwJjs9mhIzRP+KvwhYbNzWZudCjp7Yk7DhzqavSx7FbcUWDAQOjHEb6ewXTc4MPZDBq16c8+gJuEAACaKigoEAgEAoEgMTGxtLRUp9MxGIzo6Gg+n5+ent7W1ma1xcvLy/QkjY2NM2fODAsLi4qK2rNnD0LIeFR0dHRiYuKVK1cwfb6+QYIGANBRZWXlgQMHKioqJBLJ+fPnqT0D2Wy2VCqtq6vjcrmFhYVWWyzl5eU1NTVdu3Zt69at1dXVxqOkUukHH3zw3nvvDeXn6hdI0AAAOnr48GFQUJCPjw9CKDg42HRTV4IgUlNTpVKpjRaj8PDwadOmIZNKHaavqlQqf39/Z32GQaPpcM/t27cfPHhAPVYoFAPcLRIA4LKmT5+en58fGxs7c+bMBQsWTJ8+3fiSRqMpLi42Xept2WLJtFKHRqOJjY1Vq9VtbW0XLlxw3qcYJDom6FdeecV0CiSXy42Y7I5z7ABwZxwO5/r16zdu3CgpKVm8ePGGDRtWrFhBbfbKYDBEIhFVpsOyxSrTSh06nY7NZtfU1CCESktLFy1aVF1dzWQyh+6z2Y2OCXrevHnz5s0zPpVIJKS0CGM8AAAsCIJITk5OTk6Oj4/ftWvXihUrPD09ZTKZsYNOp7NssTyPjUodIpGopaVFLpebDqHQh7MStFwu9/Pzs9FBo3HHec0AADvV1NR0dXUJhUKSJMvLyyMiBjjSabtSR0VFhU6nCwkJGVywzuKUBJ2SknL69OnTp0/b6PPwoTtuWgwAsJNer8/NzZXJZEwmMz4+fv/+/XYe2NPTExYWRj1es2aNUCg8depURUXF3r17EUK7d++eM2cONTCCEOJwOIcPH2az2c75EIPllAQ9d+7cuXPn2u5jOuQPAABmEhISzMpxIJNCHBQWi2XZYlmXw54WeqLjGLQltZ6h1NBxCN/ZDAh1diO9Wy5/1ZNIaaDpdY1TqUmmQa81qN1x6axBb2UE2Z25QIJWKpX1P3k1KtwxQffomDduaQjCNX7bO5ZKRV7uCMAdBQYqtjdJKHT6HtyB4KDq6u7uxh0EjbhAguZyuRN8FE9wOnAHgsFnzVHTw1u9Pdyxmt2pR+wU2VXcUWDwg39EQ9xTY2Im4A4Eg4f3vudwYAPS/4CVhAAAQFOQoAEANNXa2vrqq6+Ghoby+fzZs2dXV1frdDqCIML+rba21rI6EuXkyZMEQVCrUZBJgSQ+n5+Wltbc3Dy0H2WAIEEDAGgqftDsjgAAIABJREFUOzs7NDT0/v379fX1mzZtovZP8fT0bPq3mJiY3o4Vi8Uikejo0aPGFmNZpZCQkIKCgqH4AIMGCRoAQEeVlZVSqXTLli1UheipU6e+/PLLdh7b2dl59erVgwcPmiZoCkEQKSkpTU1NDg7XOWh0k3D+/PmdnZ2W7TweL/eFsKGPBwCAUVVVVVJSEoNhfhFpXGPC4/EsJ0pTioqK0tLSBAJBQEBAZWWlaRElvV7/1VdfZWZmOituh6JRgm5vb7948aJlu0QiMdwUD308AAC8rC4nMSu+YZVYLKYKJ2VlZYnFYipBUxXs5HJ5fHx8enq6E+J1PBjiAADQUUJCQmVlpcHQ7zmmCoWipKRk+fLlPB5vx44dn332GZXoqQp2TU1NXC5327ZtTgjZ8SBBAwDoSCgURkVFbdiwgSqsVlZWVlxcbM+Bx48fX7JkSUNDg0wma2xsjIyMLC0tNb7q6+u7e/fuPXv2qFQusFYTEjQAgKbEYrFcLg8PD4+IiNiyZYtAILDajaqORNmxY4dYLJ4/f77x1czMzCNHjpj2j4uLS01NPXjwoHOjdwQajUEDAICpwMDAw4cPmzX2WR1p/fr1pk9Xr15teaBY7Bq3tVwgQSuVylsdo8s7gnEHgoEWMY/fCyZwh4GFgWMoinsJdxQYGBDD8INEcb8OdyAYkHq9Knw07ihoxAUSNJfLfVIp4akf4A4Eg1MBU2ePkHEY7lji65w68gV/15is6lh1at/vfZ7xi4zFHQgGHfel3t7euKOgERdI0AghFqn3JN0xSREkySb0noQ71htlEMiT4Y5VolgESTBZDJY7llolLGY9uzlsCbqlpeXXv/61aUufcxsBAG6ltbV17dq1586d8/LyiomJ+fjjjwUCgYeHx9ixY6kOFy5ciIqK8vHxMRuYRgidPHkyIyPj7t27sbGxCCFqo1g+n28wGAQCwaFDh4KDXWDUFFuCPn/+fFdXl2lLVVUVrmAAADSUnZ09efLk+/fvs1issrKy+vp6gUBA1eIw9rG6SywyqcWxefNmqoWqxUGS5NKlSwsKCrZv3z4EH2GQsCXo0aPNbwXQdlswAMDQo2pxnDlzhlrtPXXqVNR7OjZD1eK4ePHi3LlzjQmaQtXiOHfunBNCdjx6jUFXVFQcO3bMrJHL5c63Uk0QADCcQS0ORLcEXVpaOmrUKLP9ZBUKBbrzBa6QAAC4QC0OeiVohFBUVJRQKDRtkUgkuju4wgEA4JGQkJCfn28wGCwvom2janFUVVURBKHX6wmCoIabqVocHR0dmZmZ27Zty8vLc07gjgSTWgAAdAS1OBAkaAAAbUEtDtoNcQAAAAVqcdAoQatUquvXr/v5+Zm1azSae15jf/QIwBIVXmoG+5p6LBO544I6pcHjUvsY3FFg8FjHVmke6Hq6cQeCgbarQ6fj4Y6CRmiUoN9//33L35YUX7I7mOwY4njooBEFjRb4edDof2notFX3jOH04I4CBzVDxfEeGRiEOw4MOkl3rGpgA41+9F988cV79+6NHDnSrJ3NZo/Rt0dr3LFY0i1PXvRYD28vd6xnV1erj/V5jDsKDAgCdfj6+4fzcQeCga5HTW0RCyhwkxAAQFOtra2vvvpqaGgon8+fPXt2dXW1TqdjMBjR0dF8Pj8tLa25uRkhpNPpCIIw3if8/e9/b+wWHR2dmJh45coVqpuNRj6fn56e3tbWhvkz/zfrv6wCAgKKi4u//vprhFBbW9u8efOGNioAALBei8NqSQ3LAh1UN4TQ559//t577129ehX9uxyH1UaSJBcvXlxYWPjHP/4Rz6e1xnqCjomJuXPn58Uhx44do35NAQDAkLFdi8P+khoqlcrf37/PRoIgUlNT6Vajg3bDPXV1dRUVFaYtCoUiHFc0AABMeqvFQTErqWEs0IEQysvLW7JkCbWwW61Wt7W1XbhwgXrJaqPxpeLiYtOqHXRArwQtEomOHTtmVi+Jy+WG++CKCACAjdVaHFZLapgV6KCGOGpqahBCpaWlixYtqq6uRv9e7W3WSCV3BoMhEomoCh70Qa8ELRQKzQpxIIQkEon+y0+wxAMAwKW3Whz9LakhEolaWlrkcnloaKjVRnuqL+ECszgAAHRkuxaH/SU1KioqdDpdSEhIn400RK8raAAAMKKqhoaHh3t5eY0fP37nzp2mrxpLaqxcuZIqx0G1v/TSS3v37jWOSnM4nMOHD7PZbJ1OZ7VxqD9Vf0CCBgDQVJ+1OIwlNSxHqy1bLKt2UI2W+xnSh2skaDXhoWS4464qJCI6u5HeHUtxIAOJlDrX+P50rB49U6/VaFRdfXcddvQ6Le4Q6MUFfgCUSmW9f6TcYxzuQDDo6WSfqvRDhDsu9R7x+PElhTfuKDBQsT2Rslb18DbuQDDQGTy7o93xJ703LpCguVxuYmj3E6Po+2eI8xz9zl8Vnkh4eOIOBAPvR5IU2VXcUWDwg3+E5olxT3o/wh0IBpWqIA6HgzsKGoFZHAAAQFOQoAEAdESVQMrNzaWeFhQUbN682awuUm1t7UsvvWS8VfjNN98IhUK9Xn/y5EmCIKg1KS4NEjQAgKY4HM7x48dbWlpMG6m6SJSYmJg9e/b84Q9/6Ojo0Ol0q1at2rdvH5PJFIvFIpHo6NGjuCJ3FEjQAACaYrPZy5cvLywstNEnMjJy2bJlf/jDH3bt2iUSiZ566qnOzs6rV68ePHhwGCRoOt4kvHnz5vLly421png83rqsOLwhAQCwWLVqVWJi4jvvvGNsMS424fF4ly5dQgitW7dOKBSqVCqqzlpRUVFaWppAIAgICKisrKRb/aN+oWOCbmtrmzNnzvvvv089lUgkZONpvCEBALDw8/PLycn585//bJzdYVk6g81mL126tKmpidrRlFp/iBDKysoSi8WQoAEAwFl+97vfCYXC119/3UYfBoNB1VRSKBQlJSVVVVUEQej1eoIgtm/fTrjsSgIYgwYA0NqoUaMyMzMPHjxoT+fjx48vWbKkoaFBJpM1NjZGRkaWlpY6O0LngQQNAKC79evX27mvk1gsnj9/vvFpZmbmkSNHnBaX08EQBwCAjlgsVnt7O/V4zJgxxpJGVmsbGQvtU7cNjVavXu3EEJ0PQ4K+c+dObm6u1b0SKG1tbbGxscanSqXy1g8jrje4Y1kGrZ5huHcVIVcdQRuMNo5fUdxLuKPAwIAYqJtZ022+jZ470JGEf1/1nd0KhgQtk8mmTZv21ltv9dbhypUrlZWVxqdcLjc5Vh81Rj8k0dHLP6+yRySkMNnuWIvDcFX8guEu7igwqGMGPQ4YnTS6E3cgGNz+ycfb2x0vxXqDZ4iDw+FY7rNrxOVyze66spjI08P5YdEPgRCDxWKw3PTDeyJaF1N3EhZpYDGQJ9Mdi8wyiV7/sHZPcJMQAEBHMpksOjra+HTdunUFBQU6nY7BYERHR/P5/PT09La2NsvqHAihnJycoKCg8ePH4wvfMSBBAwBcCZvNlkqldXV1XC6XWgVuVp0DIfTGG2+cOXMGd6QOAAkaAOB6CIJITU2VSqVWX01JSbExiOpC6DjNjiCI8+fPd3d3U0+5XO6CGcPh3xoA4Cgajaa4uJhaxm1ZnWPYoGOCnjp16q5du4xPFQoFQtZ/TwIAhivL9dlUC5WOGQyGSCSipj9bVucYNuiYoNlstlAoND6VSCSkAhI0AO4lMDCwtbXV+FShUEyYMAFZpGOdbjhP9YExaAAAHfn4+PD5/KKiIoTQgwcPzp8//+yzz+IOaqhBggYA0NShQ4d27twZHR2dmpr60Ucfmc66sy0jI+O5556rra0NCwuzs8oSPdFxiAMAABBC8fHx33zzjWkLi8Uyq8Vh2YIQOnHihNODGxIOS9BHjhzp7LRrcert27d9fX3tP7NGo6lrZMgV7liPoluD9HV3EOGOf+gwCa9LTAHuKDB4THC0jz27tH64A8GgVe0RPKzHlPvLYQk6Pz/fuP+uwzEqJR4tj5x0cjojJj/TyfAmCSbuQDAIYpEhXm75s6ox9HCZIaPd8YpE+5M7XovY4LAEHRoaumLFCnt6FhcXSyQS+8/MZrMDfnowpt7ld1AfgB8SklTc0YjljsWSvFq/i/Vqwx0FBgyCVI70iQt3x+pgqh7EYsG463/A7ysAAE0VFBQIBAKBQJCYmEhtjNLa2vrqq6+Ghoby+fzZs2dXV1e/9NJLYrGY6v/NN98IhcKenh4vLy/T8xgMhqeffprH40VERKxbt85GrWO6gV9WAAA6qqysPHDgQEVFBZfLbW5upu4EZmdnT548+f79+ywWq6ysrL6+fs+ePTNnznzxxRdHjBixatWqTz/9lMk0HxJkMBinTp0KCQlRq9UzZsw4e/ZsWloajs/Ub5CgAQB09PDhw6CgIB8fH4RQcHAwQqiyslIqlZ45c4baH3bq1KlUz2XLlv3hD3+IiIgQiURPPfWU1aUrISEhCCGDwWAwuFIdV3ol6J9++un+/ftmjQqFYgSWaAAA+EyfPj0/Pz82NnbmzJkLFiyYPn16VVVVUlISlZ1NrVu3TigUqlSqiooKGydMSEior6/PycmZNWuWMwN3JHol6D/96U/t7e3Ub0sjLpf7PKZ4AAC4cDic69ev37hxo6SkZPHixRs2bPDx8bE6fMxms5cuXdrU1OTnZ2tu4vfff9/e3p6RkVFeXj5lyhSnBe5I9ErQJEmuXr160qRJpo0SieSnbVW4QgIA4EIQRHJycnJycnx8/K5duwoKCvLz8w0Gg+VFNIPBsGy05OfnN2vWrNOnT7tKgoZZHAAAOqqpqaGGLEiSLC8vj4iIEAqFUVFRGzZs0Gg0CKGysrLi4mJ7TtXc3NzQ0IAQam9v//zzz+Pi4pwauQPR6woaAAAoer0+NzdXJpMxmcz4+Pj9+/cjhMRi8dtvvx0eHu7l5TV+/PidO3daPbanpycsLIx6vGbNmvT09AULFrS0tHh4eCxatCg7O3voPsbgQIIGANBRQkKCZfX9wMDAw4cPW3amCkNTWCzW/2/v7oOaOvM9gD8nCQkBs4C8iAIlAgWEir1SR8kGGcW1vNiyii9A2e6u2s7OdigjRad1Xa+2y2ztsiMVan0pO3Nra7KzuFqcddAqbmsQ7BRGKy+ajRgKLIpBQDCEcJLcP87tubnJAbmV5Dnh+X3+Ck8Ome+j+PPwnPP8jvNS9XfffTfTAd0BT4EeGxsbHOTYJMb85sIxLvYe85W5OBQf2SiKosdtHnVj0Eyx2GwjFhIfZ26yisYn0IgRdw4cJmgSN7hPAUOBlsvllZWVX331lfNbOp1u8+bNDoMjIyO98tgHwQvcko5fJqQ+PxnWU9O4+jH7mGihZigUdwoMHiOx1WQdvjeBOwgGo1ZBSPQY7hQ8gqFAL168+MKFC5xv7dixw7nRnUwmi7j+7bxbba6PxjvXCreHpq4TSbyffOisM3r5vzKFt3GnwEBrDXokkb3gQ2J3sBZjsFQqxZ2CR0g8NQMAAI8ABRoAwEc0TVMUxTYxLi8v37dvHzMY/oPbt287N0uyWCynT5+mKOrWLY9vgQkFGgDAU1KptKamxmAw2A9KJJKeH8TFxVVVVe3Zs2d4eJim6aKioiNHjgiFQpVKpVQq1Wo1ruQzBQo0AICnxGLx9u3bDx48OMUxCxcuZJolVVRUMM2SRkdHGxoaqqurZ0GB5td90GFhYcXFxWKx2H5QLpcX4goEAMCqqKgoKSlp165d7Mj4+LhcLkcIyeVy5kZph2ZJZ86cycjIiI2NnTt3bktLy9KlS/FEnwn8KtClpaWlpaUOg1qttu8/38GSBwCAl7+/f2Fh4aFDh9i7OyQSiV6vtz/GoVkSs9sQIZSXl6dSqWZ5gRYIBF988UV3d/fUh92/T+JdQQAAVyspKUlOTt66desUx7DNkgYGBurr61tbWymKslgsFEV98MEHFOWp+1+eXKCzsrKYXyimxrnxBAAAnlJQUFBubm51dfXUNZpRU1Pz6quvHj16lPkyLS1No9Gkpqa6OKOrPLlAS6XS5OTkJx7m8BAwAACYKTt37jx8+PB0jlSpVG+//Tb7ZW5u7smTJ2dzgQYAAPcTiURDQ0PM6/nz5zPPJEQIsS/ssc2SHPorvfnmmy6M6HoeUKBHRkb+pVx9JyUNdxAMLN7eXU1fIo9dQXsaQqvPZ9b/wJ0CAwui0Jjg1lgA7iAY0DYqwEhkm6hJeECBlslkKUuo6PkkFqlTDba7PolWAYlN3Z616eOWr8SdAgNDj963X7M0jMQ69V2fj4+PD+4UPOIBBRohJBIiCYk1ClEI2QRCm8Az/ppmFmUViP7vHfGEEApFIgGSiDgevjfrCSkSZz0F2EkIAOCphw8f/vKXv1ywYEFUVFRmZmZ7e7tzLw6apjnvUHBox0HTtEAgiImJiYqKysjI6O/vd+9UfiQo0AAAniooKFiwYMH333/f2dn5+9//vrOzEzn14pjse53bcYjFYp1Od+fOndDQ0PLycndM4KlBgQYA8FFLS4tOpysrKxOJRAghhUKxbt26aX7vFO04KIpKS0vr6emZ4biu4abFzT/84Q+jo6M/7ntlMtnG1SRe0QaAZK2trUuXLhU4PU7IuReHsynacVgslrq6utzcXFflnlFuKtC1tbUff/zxj/vegYEBhHQzmwcAwH/Oz35FXL04nHG24zCbzfHx8b29vQkJCTk5OS7IO/PcVKB9fX2nsx2Rk1artQ1AgQaALImJifv377darc4n0VPjbMeBEBKLxbdu3RoeHs7NzT1w4MDevXtdE3wmwRo0AICPkpOTo6Ojd+/ebTabEUJXr149e/bsdL6RacfR1dWl1+u7u7sXLlyo0WjYd/38/CorK6uqqoyesCMGCjQAgKdUKlVvb29ERERkZGRZWVlsbCznYePj4+yNd3/6059UKtX69evZd5l2HPbHL1q0KD09vbq62rXpZwKJOyAAAB4hMDDwxIkTDoMOvThEIpHDUvXOnTvtv2Tbcdh/I/sYQ57zgAJtNpvvdAt6B0jc6j1mRn4jd20Uib/ojAvGdM2NuFNgMDY64mXyemyW4Q6CwUOjMISmcafgEQ8o0AghaWCYX4gf7hQYCB7oHo5LLDYSC3TYTx7E/2QYdwoM7tECk7dvaCSJ/XsnesZxR+AXDyjQYrE4ZH6APHo+7iAYdHynf2j2oxGJjUi8vXoWRVhwp8BAQKFHyGvRsyT2DDIaLcy2FMAg8dQMAOARysvLY2NjY2Njk5KSNBqNfT+NnJycwcFBzhGH1hzd3d1r1qwJDw+Pjo6uqqpCdn05YmJikpKSrly5gml+TwYFGgDARy0tLcePH29ubtZqtRcvXnzmmWeQXT8NmUx28OBBzhFne/fu7enpaWxsfP/999vb29nv0ul07733nv0TWPhmxn6b0Gq1P/vZzyZ794k7fwAAwN69e/eCg4PnzJmDEAoJCUEI0T9cP6QoKj09/csvv2QPdh5hRUREREREMB8SFxf373//2/52PaPRGBDA304SM1age3t7p3h31apVnOMmk6mtrW3qTx4YGAgJJ3E9DgCSrVq1av/+/fHx8WvWrNm4caN9DTGbzWfPnrXvsOE84kyr1Wq12uXLl6Mftn2bTKbBwcFLly65bhZPCfN6/MWLFw8ePLhs2bIpjpHJZNnhJD7vCgCSSaXSpqamb775pr6+/he/+MXu3btff/11plOSQCBQKpVMtw3nEU5DQ0ObNm06duyYTCajaZrZ9o0Q0mg0r7zySnt7u1AodN/cpg1zgbbZbFlZWW+99dYUx2i1WiPtGd21AQAziKKo5cuXL1++PCEhoaKi4vXXX3folETTtPOI8+eYTKacnJzi4uLMzEyHt5RKpcFg6O3tZda4+QYuEgIA+OjWrVvNzc0IIZvNdu3atcjIyB/3ORaLZcuWLRkZGVu3bnV+t7m5mabp0NDQp8rqMnDLIQCAjywWy1tvvaXX64VCYUJCwrFjx6b5jUxrDuZ1cXFxcnJybW1tc3PzRx99hBCqrKx86aWX2KbSUqn0xIkTYr4+/RIKNACAjxITE5378Ts34nhiaw7E1Veas9M0D3lGgR43TYyOjOFOgYHNZhOjCQHyjB+mmWWx2kY8oB/kzDOZ0TiyjIyS2JJiYoLEH/UpeECBHhkZ+f72vwx6Le4gGJhNVKitz2YjslHUOLrS6gE/nzPusUlgsRkfGX7kI+I82ohJELKAxFOxyXjAPwCZTJYUbY0Ns+IOgoGqXnjtcdiEzQP+mmbcouC2rBdIbJ1zu1f0yChY9qwZdxAMmu94SaVS3Cl4BO7iAAAAnnLTqZlEIuHcCP7gwYMVK1a4JwMAwIPQNO3l5VVSUvLnP/8ZIVReXj46Orpnzx4vL6+wsDDmmEuXLpWUlBQWFubn5yOEvv766x07dnzzzTe1tbUbNmzo6OiIj4/HOYen5qYCXVdXxzl+9uxZrZbExWUAwBNJpdKampp33nknKCiIHZRIJD09PeyXVVVVa9asycrK8vX1LSoq+uSTT4RCoUqlUiqVarV63759GHLPHFjiAADwlFgs3r59+2Q96hgLFy7ctm3bnj17KioqlErlsmXLRkdHGxoaqqur1Wq126K6CH+vPh05cuTzzz/39vaWy+Wl25NxxwEAYFBUVJSUlLRr1y52hN1jIpfLmRulS0tLk5OTjUYjs/PwzJkzGRkZsbGxc+fObWlpmbqDEs/xt0D39fWVlZWtXLlSq9XaButxxwEAYODv719YWHjo0CH27g6H5hsIIbFY/Otf/7qnp8ff3x8hpFKpmK5JeXl5KpUKCjQAALhKSUlJcnIyZycNlkAgEAgECKGBgYH6+vrW1laKoiwWC0VRH3zwAUV56k4CWIMGAPBaUFBQbm5udXX1dA6uqal59dVXu7q69Hp9d3f3woULNRqNqxO6DhRoAADf7dy5s79/Wj2HVSrV+vXr2S9zc3NPnjzpslwuB0scAAA+EolEQ0NDzOv58+ezTZEcuiMx2Fb9Dv2V3nzzTRdGdD3MBVoikRw9epTzLum7d+8uXrwYITQyMnKjVdDUTuLJ/gSNnve+g5CnrqA9jQfD1KeXSdz1a7EiZKM6evj4gA9Xoy3IP5LIFlmTwFyg165de+3aNc63Dhw4wDwpUiaTLX9OGB1GYpE6dZm+2hdutpL4b3VzTOfaqR6FNmvpetEjo2BpPIlnJN/prD4+8ADS/4V/iWOyR+pKJBL2tUhIScQkFmgKIdoqoG0kFmgBRUm8SGw+KRIikYjQH3gBif8rTQX+PAAA/HX69GmKopgHvCKEaJqmKIp9iml5efm+fftomvb29mZGLly4EB0d3dnZyQ7SNC0QCGJiYmJiYpKSkq5cucIcWVhYGBwc/Nxzz7l9Tv8PUKABAPzFdtVgR5gGHQaDwfngy5cv/+Y3vzl37lxUVJT9uFgs1ul0Op3uvffee/vtt5nB11577dy5cy4N//SgQAMAeIqzq8ZkDToaGhq2bt169uzZuLi4yT7QaDSya6ppaWmTra/yB/416MlQFHX06NFz587JZLKNL87HHQcA4G6TddVwbtAxMTHx0ksvXbx4MTEx0flzzGZzfHy8yWQaHBy8dOmSm9LPBP4W6KKiImbH/cDAAEI9TzocADDbTNZVw7lBh0gkSk1Nra6u5uy8IRaLmVVsjUbzyiuvtLe3C4WeceGdvwU6MDAwMDAQIaTVam2PoEADQBbOrhrsuw4NOiiKUqvVa9eufffdd/fu3TvZZyqVSoPB0Nvb+8wzz7h8AjMB1qABAHw0dVcN5wYdUqm0trb2b3/729GjRyf7zObmZpqmQ0NDXRt95kCBBgDw0RO7ajg36AgICKirqztw4MDf//53+3GmhbRcLi8sLDxx4oRYLEYIbdiwITU19fbt2+Hh4dPsxOR+/F3iAACQbLKuGlM36AgLC+vs7LQfFIlENhvHjieHIs5PnlGg+wwWi5XEk/0JGoV6j9JWEjeVmcyoowt3CBz6Bymj2dqhx50Dh4EhtAB3Bl6hOP9v4RWj0djVReS/VIQeP37s4+Pjue3Gn8bjx499fX1xp8DAYrGYzWb2/gSiUBQVFRXFLEEA5BEFGgAAyETiugEAAHgEKNDT8vDhw+PHj69evRp3EHegJ4E7l8sRO3FnfX19p06dKikpwR2EdJ5xkRCX0dHR2tpalUrV1NSUnZ1tv7V0FgsKCnIeHB4envWrYcROHCFktVpv3rx59erVhoaGq1evms3mlJQUhUKBOxfpoEBzO3PmjFqtrq+vX7169bZt22pqauz7U89u7G1MCKGWlpa//vWv58+ff/755zFGcg9iJ44Q8vf3DwoKys7Ozs7OLisri4yMxJ0IIAQXCSdDUdSSJUv+8pe/cG7tn/W6urqOHz/+j3/8Y9GiRXl5eRkZGYRcWCd24r/73e+ampqGhoYWL16sUCgUCkVCQoIA+ufjBgWaW0dHh0qlUqvVvr6+eXl5W7ZskcvluEO5z0cffbR37973338/Ly9PJpPhjuM+xE6cYbFYbt682djY2NjY2NbWFhwczPm8UOA28D8kt0WLFr377rtarfb48eP9/f2pqakpKSkffvgh7lxu8sYbbzQ1Nd2/fz89PX3Tpk01NTVjY2O4Q7kDsRNnDA8P9/X19fX13b9/38vLKywsDHci0sEZ9LTYbLavv/5apVIdOXIEdxaXO3PmjP2X7e3tarVar9c/evQIVyT3IHbiCKFf/epXN2/e9Pf3X7FihUKhSElJmTt3Lu5QAC4STsL+ehFDqVQuWbIESxg3s396BSMhISEhIQFLGHciduIIocbGxuHh4cjIyICAgICAADL3cPIQnEFzi4+Pdxg5efJkQUEB+/BKAGYZg8HA3GPX0NDQ2tqakJCgUCjKy8tx5yIaFGjgaLKtGSLRLP99i9iJOzObzc3NzZcvX969ezfuLESDAs3t3r17DiMe1OT7KVEUxd7+YPk6AAAFEUlEQVT0PT4+zrweHx+f9T8qxE58MuHh4T098DAjnIg7NZimjIwM5sXY2NidO3eeffbZjo4OvJHcJiwsjP1nKZfLmSdDhoeH48zkFsROHPAWFGhu169fZ1/fvn370KFDGMMA4GoVFRXOg6Ojo+5PAuxBgX6yuLi4zs5Om81GZl9mQALOpQyLxeL+JMAeFGhuDmvQf/zjH8kp0NnZ2exr9gHJmzdvxhTHfYidOEKI824N5/sOgZvBRUJubIsck8mk0+ni4+NbW1vxRsKL2OtFxE4ckT13noAzaG72a9A3btyANWgwu61ataqgoCA3N9d+A6FGo8EYCSAo0NORlJTU2NiIO4X7EHu9iNiJI4R27dp1+vTp4uLi9PT0/Pz8l19+ec6cOUQ1COMnKNDcSktLmRdWq/X69euJiYl487gTsdeLiJ04QigzMzMzM/P8+fNFRUVqtXrnzp2pqan5+fk5OTm4oxEN1qC5ffbZZ8wLk8l04cKFN954Iy0tDW8kvIhdjiRq4uzd31qtdtu2bRqNBuoDXlCgn8xisfz0pz9tamrCHQSn4ODgBw8e4E6BAVETj4iI2LVrl1qtNhgMW7ZsKSgocG5KA9wJlji4sW0ZrFbrjRs3+vv78ebBzmAw4I6AByETr6ysPHXqFEVRer3+ww8/fOGFF3AnAghBgZ4Me3lEKBRGRkYeO3YMaxz8iO3dTsjE29ra9u/fv3LlSkJu9vcUUKC5kbPs6IzYmxmInThCqKqqCnFdESWwkx+vwJ8+t7Gxsdra2oCAgLVr1+LO4m7E3sxA7MQRQnPmzGFesG38EEJms9lqteILBeAi4SSysrLkcvndu3cVCsVvf/vbgoKC8+fP4w6FE1E3M9gjbeLsjRwIoZiYGJ1OhzUO6eChsdzmzJlz+PDhL774oq6uLjAwkJArRQAIBIKJiQmEkM1mg9Nn7GCJg5uPj09/f39ISIhIJDKZTOPj47gTuQ/ng0X++c9/uj2IuxE7cXtKpTI/P3/dunX19fWLFy/GHYd0UKC56fX6mJgYhULR3t6uUCh27NiBO5H7eHl5sauQ9kwmk/vDuBOxE7d3+PDhioqKCxcuxMTEEPVjz0+wBs2N3ZYikUiioqL8/Pzw5nEn0lZdWcROHPAWnEFzW7FiBe4IAADSwUVCAADgKSjQwFFxcTHuCHgQO3HAW7AGDcD/0Gq1fn5+8+bNsx+0Wq0CAZzHADzgJw842rBhg8PIV199VVhYiCWMO23cuHFsbMx+5N69e3A1AmAEFwmBo8bGRof9Yz4+PiRspDQYDGyTrMTExLa2ttDQ0Pv372MNBYgGBRo4MhgMP//5zx0GHX7xn5W8vLyGhob8/f2HhoY6OzuNRqNQKMQdChANCjRwNG/ePDIfYZ6fn5+VlZWTk3P+/PnCwsL09HSBQLBp0ybcuQC54CIhcFRdXb1t2zbcKTCw2Wyff/55R0dHdnZ2SkpKXV0dRVEvvvgitEgGuECBBtzKysoUCsWqVatwBwGAXFCgAbewsLD09HSEUHl5eUhICO44AJAIbrMD3Ly8vD799NPXXntt06ZNR44cgc6TALgfnEEDR0yjqPXr158+fRohZLFYKisru7q6GhsbcUcDgCxQoIEjpVKJEPr2228dHu2s0WgwJQKAUFCgAbeXX365trYWdwoAiAYFGgAAeAouEgIAAE9BgQYAAJ6CAg0AADwFBRoAAHgKCjQAAPAUFGgAAOCp/wYdUD4qVcFbmwAAAABJRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# mostVar Calculate the top n most variable genes in a matrix of gene expression data\n",
    "# https://rdrr.io/github/abc-igmm/transcripTools/man/mostVar.html\n",
    "mostVar <- function(data, n, i_want_most_var = TRUE) {\n",
    "  data.var <- apply(data, 1, stats::var)\n",
    "  data[order(data.var, decreasing = i_want_most_var)[1:n],] \n",
    "}\n",
    "# scale - Z-Score\n",
    "fc <- data.frame(apply(all_fc,2,scale, center=TRUE, scale=TRUE), row.names=rownames(all_fc))\n",
    "\n",
    "# idtransfer\n",
    "peak_ids = strsplit(rownames(all_fc), '_',)\n",
    "ids = unlist(lapply(peak_ids, '[[', 1))\n",
    "\n",
    "# filter most variable genes\n",
    "fc = mostVar(fc,30)\n",
    "# Plot heatmap\n",
    "h1 <- pheatmap(fc, labels_row = gtf_df[ids,])\n",
    "ggsave('plots/peak_mostVar_Heatmap.png', plot = h1, device = 'png', dpi = 300)\n",
    "ggsave('plots/peak_mostVar_Heatmap.pdf', plot = h1, device = 'pdf', dpi = 300)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## iPAGE\n",
    "Enrichment analysis "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7509 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7509\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "683 hits go_terms_num = 17689\n",
      "cat 1329, mi = 0.00307518, regulation of inflammatory response\n",
      "cat 1899, mi = 0.00302082, negative regulation of osteoblast differentiation\n",
      "cat 658, mi = 0.00302004, Notch signaling pathway\n",
      "cat 3519, mi = 0.00298433, SH3/SH2 adaptor activity\n",
      "cat 1308, mi = 0.00288274, spindle pole\n",
      "cat 327, mi = 0.00277679, positive regulation of nitric oxide biosynthetic process\n",
      "cat 84, mi = 0.00275510, animal organ regeneration\n",
      "cat 69, mi = 0.00272475, response to hypoxia\n",
      "cat 1490, mi = 0.00268103, mitotic spindle pole\n",
      "cat 1115, mi = 0.00259360, inositol phosphate metabolic process\n",
      "cat 2542, mi = 0.00257366, negative regulation of proteasomal ubiquitin-dependent protein catabolic process\n",
      "cat 1532, mi = 0.00256420, RNA polymerase II distal enhancer sequence-specific DNA binding\n",
      "cat 3586, mi = 0.00250557, clathrin-coated pit\n",
      "cat 1863, mi = 0.00248940, nucleosome assembly\n",
      "cat 738, mi = 0.00248545, circadian rhythm\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1329, mi = 0.003, pvalue = 0.000, regulation of inflammatory response\n",
      "Passed: cat 1899, mi = 0.003, pvalue = 0.001, negative regulation of osteoblast differentiation\n",
      "Passed: cat 3519, mi = 0.003, pvalue = 0.001, SH3/SH2 adaptor activity\n",
      "Passed: cat 1308, mi = 0.003, pvalue = 0.001, spindle pole\n",
      "Passed: cat 1115, mi = 0.003, pvalue = 0.004, inositol phosphate metabolic process\n",
      "Passed: cat 2542, mi = 0.003, pvalue = 0.004, negative regulation of proteasomal ubiquitin-dependent protein catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 3586, mi = 0.003, pvalue = 0.006, clathrin-coated pit\n",
      "Didn't pass (cutoff=0.0050): cat 1863, mi = 0.002, pvalue = 0.009, nucleosome assembly\n",
      "Didn't pass (cutoff=0.0050): cat 738, mi = 0.002, pvalue = 0.008, circadian rhythm\n",
      "Didn't pass (cutoff=0.0050): cat 1141, mi = 0.002, pvalue = 0.010, fatty acid metabolic process\n",
      "Passed: cat 6349, mi = 0.002, pvalue = 0.000, glycosaminoglycan catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 6968, mi = 0.002, pvalue = 0.010, histone methylation\n",
      "Didn't pass (cutoff=0.0050): cat 1940, mi = 0.002, pvalue = 0.011, regulation of RNA splicing\n",
      "Didn't pass (cutoff=0.0050): cat 1968, mi = 0.002, pvalue = 0.010, secretory granule membrane\n",
      "Didn't pass (cutoff=0.0050): cat 498, mi = 0.002, pvalue = 0.005, limb development\n",
      "Passed: cat 5060, mi = 0.002, pvalue = 0.000, keratin filament binding\n",
      "Didn't pass (cutoff=0.0050): cat 2893, mi = 0.002, pvalue = 0.017, mitochondrion organization\n",
      "Didn't pass (cutoff=0.0050): cat 4166, mi = 0.002, pvalue = 0.005, S100 protein binding\n",
      "Didn't pass (cutoff=0.0050): cat 4968, mi = 0.002, pvalue = 0.007, structural molecule activity conferring elasticity\n",
      "Didn't pass (cutoff=0.0050): cat 5629, mi = 0.002, pvalue = 0.008, membrane protein ectodomain proteolysis\n",
      "Didn't pass (cutoff=0.0050): cat 2144, mi = 0.002, pvalue = 0.022, histone H4 acetylation\n",
      "Didn't pass (cutoff=0.0050): cat 3022, mi = 0.002, pvalue = 0.014, endodeoxyribonuclease activity\n",
      "Didn't pass (cutoff=0.0050): cat 541, mi = 0.002, pvalue = 0.022, lysine-acetylated histone binding\n",
      "Didn't pass (cutoff=0.0050): cat 9495, mi = 0.002, pvalue = 0.012, regulation of voltage-gated calcium channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 74, mi = 0.002, pvalue = 0.012, response to organonitrogen compound\n",
      "Didn't pass (cutoff=0.0050): cat 720, mi = 0.002, pvalue = 0.021, histone H3 acetylation\n",
      "Didn't pass (cutoff=0.0050): cat 1404, mi = 0.002, pvalue = 0.007, sphingolipid biosynthetic process\n",
      "Passed: cat 2879, mi = 0.002, pvalue = 0.004, cortical cytoskeleton\n",
      "Didn't pass (cutoff=0.0050): cat 725, mi = 0.002, pvalue = 0.019, transcriptional repressor activity, RNA polymerase II transcription factor binding\n",
      "Didn't pass (cutoff=0.0050): cat 1912, mi = 0.002, pvalue = 0.028, retrograde transport, endosome to Golgi\n",
      "Didn't pass (cutoff=0.0050): cat 2708, mi = 0.002, pvalue = 0.029, nucleotide-excision repair\n",
      "Didn't pass (cutoff=0.0050): cat 1905, mi = 0.002, pvalue = 0.027, Cul3-RING ubiquitin ligase complex\n",
      "Didn't pass (cutoff=0.0050): cat 558, mi = 0.002, pvalue = 0.024, tumor necrosis factor receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 824, mi = 0.002, pvalue = 0.027, recycling endosome membrane\n",
      "Didn't pass (cutoff=0.0050): cat 4082, mi = 0.002, pvalue = 0.011, positive regulation of transcription from RNA polymerase II promoter in response to stress\n",
      "Didn't pass (cutoff=0.0050): cat 5546, mi = 0.002, pvalue = 0.025, behavioral fear response\n",
      "Didn't pass (cutoff=0.0050): cat 1371, mi = 0.002, pvalue = 0.018, positive regulation of smooth muscle cell migration\n",
      "Didn't pass (cutoff=0.0050): cat 3588, mi = 0.002, pvalue = 0.015, extracellular vesicle\n",
      "Passed: cat 1986, mi = 0.002, pvalue = 0.003, ER to Golgi transport vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1690, mi = 0.002, pvalue = 0.036, response to radiation\n",
      "Passed: cat 2509, mi = 0.002, pvalue = 0.005, snRNA processing\n",
      "Didn't pass (cutoff=0.0050): cat 3347, mi = 0.002, pvalue = 0.021, nucleotide-excision repair, DNA incision, 3'-to lesion\n",
      "Didn't pass (cutoff=0.0050): cat 175, mi = 0.002, pvalue = 0.032, protein-DNA complex\n",
      "Didn't pass (cutoff=0.0050): cat 4854, mi = 0.002, pvalue = 0.030, intrinsic apoptotic signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 2675, mi = 0.002, pvalue = 0.013, positive regulation of cellular protein metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 2274, mi = 0.002, pvalue = 0.022, innervation\n",
      "Didn't pass (cutoff=0.0050): cat 1719, mi = 0.002, pvalue = 0.043, cellular response to oxidative stress\n",
      "Didn't pass (cutoff=0.0050): cat 3355, mi = 0.002, pvalue = 0.028, response to exogenous dsRNA\n",
      "Didn't pass (cutoff=0.0050): cat 1660, mi = 0.002, pvalue = 0.033, blood circulation\n",
      "Didn't pass (cutoff=0.0050): cat 3447, mi = 0.002, pvalue = 0.041, positive regulation of protein kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1062, mi = 0.002, pvalue = 0.043, single-stranded DNA binding\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.002, pvalue = 0.036, cerebellum development\n",
      "Didn't pass (cutoff=0.0050): cat 5323, mi = 0.002, pvalue = 0.042, cortical actin cytoskeleton\n",
      "Didn't pass (cutoff=0.0050): cat 2249, mi = 0.002, pvalue = 0.047, ligand-dependent nuclear receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 1071, mi = 0.002, pvalue = 0.014, negative regulation of G2/M transition of mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 1235, mi = 0.002, pvalue = 0.042, transcription cofactor activity\n",
      "Passed: cat 4198, mi = 0.002, pvalue = 0.003, lymph node development\n",
      "Didn't pass (cutoff=0.0050): cat 1684, mi = 0.002, pvalue = 0.057, costamere\n",
      "Didn't pass (cutoff=0.0050): cat 475, mi = 0.002, pvalue = 0.053, odontogenesis of dentin-containing tooth\n",
      "Didn't pass (cutoff=0.0050): cat 2308, mi = 0.002, pvalue = 0.029, mRNA cleavage\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.002, pvalue = 0.058, negative regulation of translation\n",
      "Didn't pass (cutoff=0.0050): cat 2337, mi = 0.002, pvalue = 0.049, RNA processing\n",
      "Didn't pass (cutoff=0.0050): cat 322, mi = 0.002, pvalue = 0.028, defense response to protozoan\n",
      "Didn't pass (cutoff=0.0050): cat 5082, mi = 0.002, pvalue = 0.035, histone H2A acetylation\n",
      "Didn't pass (cutoff=0.0050): cat 1097, mi = 0.002, pvalue = 0.011, neuropeptide signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 836, mi = 0.002, pvalue = 0.066, translational initiation\n",
      "Didn't pass (cutoff=0.0050): cat 3774, mi = 0.002, pvalue = 0.069, DNA duplex unwinding\n",
      "Passed: cat 8958, mi = 0.002, pvalue = 0.001, T-helper 1 type immune response\n",
      "Didn't pass (cutoff=0.0050): cat 1469, mi = 0.002, pvalue = 0.068, viral entry into host cell\n",
      "Didn't pass (cutoff=0.0050): cat 3921, mi = 0.002, pvalue = 0.064, androgen receptor signaling pathway\n",
      "Passed: cat 2109, mi = 0.002, pvalue = 0.003, low-density lipoprotein particle clearance\n",
      "Passed: cat 2320, mi = 0.002, pvalue = 0.003, NAD binding\n",
      "Passed: cat 10259, mi = 0.002, pvalue = 0.003, ATPase inhibitor activity\n",
      "Didn't pass (cutoff=0.0050): cat 3890, mi = 0.002, pvalue = 0.012, pre-miRNA processing\n",
      "Didn't pass (cutoff=0.0050): cat 1943, mi = 0.002, pvalue = 0.068, phosphatidylinositol-3-phosphate binding\n",
      "Didn't pass (cutoff=0.0050): cat 1224, mi = 0.002, pvalue = 0.047, protein catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 5394, mi = 0.002, pvalue = 0.033, membrane fusion\n",
      "Didn't pass (cutoff=0.0050): cat 2611, mi = 0.002, pvalue = 0.052, positive regulation of axon extension\n",
      "Didn't pass (cutoff=0.0050): cat 1218, mi = 0.002, pvalue = 0.076, regulation of macroautophagy\n",
      "Didn't pass (cutoff=0.0050): cat 5694, mi = 0.002, pvalue = 0.048, nuclear outer membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1673, mi = 0.002, pvalue = 0.077, protein complex scaffold\n",
      "Didn't pass (cutoff=0.0050): cat 6314, mi = 0.002, pvalue = 0.057, tRNA methylation\n",
      "Didn't pass (cutoff=0.0050): cat 1585, mi = 0.002, pvalue = 0.070, RNA polymerase II core promoter sequence-specific DNA binding\n",
      "Didn't pass (cutoff=0.0050): cat 4972, mi = 0.002, pvalue = 0.057, positive regulation of dendrite morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 5703, mi = 0.002, pvalue = 0.057, embryonic hemopoiesis\n",
      "Didn't pass (cutoff=0.0050): cat 897, mi = 0.002, pvalue = 0.078, protein serine/threonine/tyrosine kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 12281, mi = 0.002, pvalue = 0.033, cellular response to laminar fluid shear stress\n",
      "Didn't pass (cutoff=0.0050): cat 7280, mi = 0.002, pvalue = 0.090, protein acetylation\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.002, pvalue = 0.034, hydrogen ion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 127, mi = 0.002, pvalue = 0.035, nuclear envelope organization\n",
      "Didn't pass (cutoff=0.0050): cat 669, mi = 0.002, pvalue = 0.033, cell maturation\n",
      "Didn't pass (cutoff=0.0050): cat 3500, mi = 0.002, pvalue = 0.037, transcription factor activity, RNA polymerase II core promoter proximal region sequence-specific binding\n",
      "Didn't pass (cutoff=0.0050): cat 6468, mi = 0.002, pvalue = 0.035, regulation of chromosome segregation\n",
      "Total number of categories evaluated = 161.\n",
      "Number of categories that passed the tests = 16\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 14, 15, 0\n",
      "best merge is 12, 14, 0.0143683639402026\n",
      "best merge is 9, 12, 0.0443987577482371\n",
      "best merge is 8, 13, 0.0543899765385343\n",
      "best merge is 7, 9, 0.0604019206521377\n",
      "best merge is 2, 10, 0.391258945848023\n",
      "best merge is 5, 7, 0.440983086424123\n",
      "best merge is 3, 4, 0.542883392011971\n",
      "best merge is 1, 11, 0.606476562326226\n",
      "best merge is 0, 8, 0.658605627726167\n",
      "best merge is 3, 6, 0.830782858455503\n",
      "best merge is 2, 3, 0.965376457990528\n",
      "order classes\n",
      "3 - 1.62560146013732\n",
      "2 - 2.42590624930491\n",
      "2 - 1.56503578339209\n",
      "3 - 2.6126804253902\n",
      "6 - 0.386437943089523\n",
      "5 7 9 12 14 15\n",
      "1 11\n",
      "2 10\n",
      "3 4 6\n",
      "0 8 13\n",
      "Done.Start drawing\n",
      "5.81\t-10.06\n",
      "glycosaminoglycan catabolic process, GO:0006027\n",
      "inositol phosphate metabolic process, GO:0043647\n",
      "negative regulation of osteoblast differentiation, GO:0045668\n",
      "cortical cytoskeleton, GO:0030863\n",
      "low-density lipoprotein particle clearance, GO:0034383\n",
      "lymph node development, GO:0048535\n",
      "regulation of inflammatory response, GO:0050727\n",
      "SH3/SH2 adaptor activity, GO:0005070\n",
      "snRNA processing, GO:0016180\n",
      "negative regulation of proteasomal ubiquitin-dependent protein catabolic process, GO:0032435\n",
      "keratin filament binding, GO:1990254\n",
      "ER to Golgi transport vesicle membrane, GO:0012507\n",
      "T-helper 1 type immune response, GO:0042088\n",
      "NAD binding, GO:0051287\n",
      "ATPase inhibitor activity, GO:0042030\n",
      "spindle pole, GO:0000922\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 14, 15, 0\n",
      "best merge is 12, 14, 0.0143683639402026\n",
      "best merge is 9, 12, 0.0443987577482371\n",
      "best merge is 8, 13, 0.0543899765385343\n",
      "best merge is 7, 9, 0.0604019206521377\n",
      "best merge is 2, 10, 0.391258945848023\n",
      "best merge is 5, 7, 0.440983086424123\n",
      "best merge is 3, 4, 0.542883392011971\n",
      "best merge is 1, 11, 0.606476562326226\n",
      "best merge is 0, 8, 0.658605627726167\n",
      "best merge is 3, 6, 0.830782858455503\n",
      "best merge is 2, 3, 0.965376457990528\n",
      "order classes\n",
      "3 - 1.62560146013732\n",
      "2 - 2.42590624930491\n",
      "2 - 1.56503578339209\n",
      "3 - 2.6126804253902\n",
      "6 - 0.386437943089523\n",
      "5 7 9 12 14 15\n",
      "1 11\n",
      "2 10\n",
      "3 4 6\n",
      "0 8 13\n",
      "Done.-8.28922495274102\t42\t0\n",
      "4.16257088846881\t42\t23.421550094518\n",
      "5.27410207939509\t42\t24.9035916824197\n",
      "5.89603024574669\t42\t25.6446124763705\n",
      "6.33270321361059\t42\t26.1474480151229\n",
      "6.66351606805293\t42\t26.5179584120983\n",
      "6.95463137996219\t42\t26.8090737240076\n",
      "7.27221172022684\t42\t27.1001890359168\n",
      "7.66918714555766\t42\t27.4442344045369\n",
      "8.26465028355388\t42\t27.8941398865785\n",
      "15.3175803402646\t42\t28.6351606805293\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6496 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6496\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "591 hits go_terms_num = 181\n",
      "cat 13, mi = 0.00174331, BRF2_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00132465, PPARGC1A_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00125542, BRF1_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00117208, SMARCA4_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00113277, BDP1_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00098976, POU5F1_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00075718, XRCC4_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00071002, POLR3G_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.002, pvalue = 0.118, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.561, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.482, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.539, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.531, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.001, pvalue = 0.573, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass: cat 167, mi = 0.001, pvalue = 1.000, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.964, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.26\t-9.79\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_encode_tf/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 326\n",
      "cat 79, mi = 0.00324991, chr16q12\n",
      "cat 165, mi = 0.00297276, chr2q21\n",
      "cat 139, mi = 0.00271351, chr20q13\n",
      "cat 146, mi = 0.00270319, chr22q12\n",
      "cat 280, mi = 0.00253005, chr8p11\n",
      "cat 93, mi = 0.00238213, chr17q23\n",
      "cat 116, mi = 0.00208603, chr1p32\n",
      "cat 292, mi = 0.00206299, chr8q24\n",
      "cat 236, mi = 0.00206114, chr5q31\n",
      "cat 60, mi = 0.00201706, chr14q24\n",
      "cat 51, mi = 0.00199679, chr13q33\n",
      "cat 205, mi = 0.00196533, chr4p16\n",
      "cat 256, mi = 0.00189786, chr6q22\n",
      "cat 191, mi = 0.00186367, chr3q21\n",
      "cat 143, mi = 0.00185244, chr21q22\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 79, mi = 0.003, pvalue = 0.001, chr16q12\n",
      "Passed: cat 165, mi = 0.003, pvalue = 0.003, chr2q21\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.003, pvalue = 0.006, chr20q13\n",
      "Didn't pass (cutoff=0.0050): cat 146, mi = 0.003, pvalue = 0.010, chr22q12\n",
      "Didn't pass (cutoff=0.0050): cat 280, mi = 0.003, pvalue = 0.014, chr8p11\n",
      "Didn't pass (cutoff=0.0050): cat 93, mi = 0.002, pvalue = 0.028, chr17q23\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.002, pvalue = 0.059, chr1p32\n",
      "Didn't pass (cutoff=0.0050): cat 292, mi = 0.002, pvalue = 0.046, chr8q24\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.051, chr5q31\n",
      "Didn't pass (cutoff=0.0050): cat 60, mi = 0.002, pvalue = 0.067, chr14q24\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.002, pvalue = 0.028, chr13q33\n",
      "Didn't pass (cutoff=0.0050): cat 205, mi = 0.002, pvalue = 0.063, chr4p16\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.002, pvalue = 0.115, chr6q22\n",
      "Didn't pass (cutoff=0.0050): cat 191, mi = 0.002, pvalue = 0.109, chr3q21\n",
      "Didn't pass (cutoff=0.0050): cat 143, mi = 0.002, pvalue = 0.088, chr21q22\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.002, pvalue = 0.138, chr18p11\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.002, pvalue = 0.121, chr10q26\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.002, pvalue = 0.137, chr13q14\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.002, pvalue = 0.090, chr5q13\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.002, pvalue = 0.097, chr11q24\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.002, pvalue = 0.096, chr11q25\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.002, pvalue = 0.138, chr14q11\n",
      "Total number of categories evaluated = 12.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "chr2q21\n",
      "chr16q12\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_c1/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4738\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t4738\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 4738\n",
      "cat 1943, mi = 0.00539882, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "cat 89, mi = 0.00406525, AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "cat 865, mi = 0.00401068, DIRMEIER_LMP1_RESPONSE_EARLY\n",
      "cat 1326, mi = 0.00399804, HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "cat 1371, mi = 0.00350794, HOOI_ST7_TARGETS_DN\n",
      "cat 2039, mi = 0.00344669, LIN_NPAS4_TARGETS_DN\n",
      "cat 2031, mi = 0.00338780, LINDSTEDT_DENDRITIC_CELL_MATURATION_C\n",
      "cat 1083, mi = 0.00337885, GAUSSMANN_MLL_AF4_FUSION_TARGETS_A_UP\n",
      "cat 1631, mi = 0.00335702, KEGG_ERBB_SIGNALING_PATHWAY\n",
      "cat 126, mi = 0.00332761, BASSO_CD40_SIGNALING_UP\n",
      "cat 1756, mi = 0.00332682, KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY\n",
      "cat 3986, mi = 0.00330020, SIMBULAN_UV_RESPONSE_NORMAL_DN\n",
      "cat 1718, mi = 0.00328628, KEGG_PROSTATE_CANCER\n",
      "cat 2786, mi = 0.00328176, PID_FCER1_PATHWAY\n",
      "cat 1945, mi = 0.00327164, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1943, mi = 0.005, pvalue = 0.000, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "Passed: cat 89, mi = 0.004, pvalue = 0.000, AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "Passed: cat 1371, mi = 0.004, pvalue = 0.001, HOOI_ST7_TARGETS_DN\n",
      "Passed: cat 2039, mi = 0.003, pvalue = 0.001, LIN_NPAS4_TARGETS_DN\n",
      "Passed: cat 2031, mi = 0.003, pvalue = 0.001, LINDSTEDT_DENDRITIC_CELL_MATURATION_C\n",
      "Passed: cat 1631, mi = 0.003, pvalue = 0.001, KEGG_ERBB_SIGNALING_PATHWAY\n",
      "Passed: cat 1078, mi = 0.003, pvalue = 0.002, GARGALOVIC_RESPONSE_TO_OXIDIZED_PHOSPHOLIPIDS_YELLOW_UP\n",
      "Passed: cat 27, mi = 0.003, pvalue = 0.004, ACOSTA_PROLIFERATION_INDEPENDENT_MYC_TARGETS_DN\n",
      "Passed: cat 1941, mi = 0.003, pvalue = 0.003, LEE_CALORIE_RESTRICTION_NEOCORTEX_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2950, mi = 0.003, pvalue = 0.006, PUIFFE_INVASION_INHIBITED_BY_ASCITES_UP\n",
      "Passed: cat 821, mi = 0.003, pvalue = 0.004, DAZARD_UV_RESPONSE_CLUSTER_G1\n",
      "Passed: cat 797, mi = 0.003, pvalue = 0.004, DARWICHE_SKIN_TUMOR_PROMOTER_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4405, mi = 0.003, pvalue = 0.007, WATANABE_RECTAL_CANCER_RADIOTHERAPY_RESPONSIVE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1214, mi = 0.003, pvalue = 0.005, GROSS_HYPOXIA_VIA_ELK3_AND_HIF1A_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4586, mi = 0.003, pvalue = 0.007, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_17\n",
      "Didn't pass (cutoff=0.0050): cat 1203, mi = 0.003, pvalue = 0.007, GRANDVAUX_IRF3_TARGETS_UP\n",
      "Passed: cat 3812, mi = 0.003, pvalue = 0.004, SASAI_RESISTANCE_TO_NEOPLASTIC_TRANSFROMATION\n",
      "Didn't pass (cutoff=0.0050): cat 2361, mi = 0.003, pvalue = 0.010, MOHANKUMAR_TLX1_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 970, mi = 0.003, pvalue = 0.011, FIGUEROA_AML_METHYLATION_CLUSTER_3_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2749, mi = 0.003, pvalue = 0.011, PID_BMP_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 406, mi = 0.003, pvalue = 0.009, BIOCARTA_VDR_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 1977, mi = 0.003, pvalue = 0.016, LEIN_CEREBELLUM_MARKERS\n",
      "Didn't pass (cutoff=0.0050): cat 781, mi = 0.003, pvalue = 0.013, DAIRKEE_TERT_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 225, mi = 0.003, pvalue = 0.009, BIOCARTA_CARM1_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 3821, mi = 0.003, pvalue = 0.013, SASSON_RESPONSE_TO_GONADOTROPHINS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.003, pvalue = 0.015, BASSO_B_LYMPHOCYTE_NETWORK\n",
      "Didn't pass (cutoff=0.0050): cat 395, mi = 0.002, pvalue = 0.016, BIOCARTA_TGFB_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 4240, mi = 0.002, pvalue = 0.019, TURASHVILI_BREAST_NORMAL_DUCTAL_VS_LOBULAR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2542, mi = 0.002, pvalue = 0.018, NIKOLSKY_BREAST_CANCER_17Q11_Q21_AMPLICON\n",
      "Didn't pass (cutoff=0.0050): cat 2617, mi = 0.002, pvalue = 0.020, OSADA_ASCL1_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 905, mi = 0.002, pvalue = 0.024, DUTTA_APOPTOSIS_VIA_NFKB\n",
      "Passed: cat 158, mi = 0.002, pvalue = 0.004, BERENJENO_TRANSFORMED_BY_RHOA_FOREVER_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1474, mi = 0.002, pvalue = 0.027, IWANAGA_E2F1_TARGETS_INDUCED_BY_SERUM\n",
      "Didn't pass (cutoff=0.0050): cat 1065, mi = 0.002, pvalue = 0.025, GARGALOVIC_RESPONSE_TO_OXIDIZED_PHOSPHOLIPIDS_GREEN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.002, pvalue = 0.029, ALONSO_METASTASIS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 187, mi = 0.002, pvalue = 0.022, BILBAN_B_CLL_LPL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4019, mi = 0.002, pvalue = 0.024, SNIJDERS_AMPLIFIED_IN_HEAD_AND_NECK_TUMORS\n",
      "Didn't pass (cutoff=0.0050): cat 777, mi = 0.002, pvalue = 0.032, DACOSTA_UV_RESPONSE_VIA_ERCC3_XPCS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4577, mi = 0.002, pvalue = 0.022, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_0\n",
      "Didn't pass (cutoff=0.0050): cat 4382, mi = 0.002, pvalue = 0.025, WANG_RESPONSE_TO_ANDROGEN_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2231, mi = 0.002, pvalue = 0.010, MATZUK_SPERMATID_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 4542, mi = 0.002, pvalue = 0.025, YAGI_AML_RELAPSE_PROGNOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 1438, mi = 0.002, pvalue = 0.033, IKEDA_MIR133_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1208, mi = 0.002, pvalue = 0.030, GREGORY_SYNTHETIC_LETHAL_WITH_IMATINIB\n",
      "Didn't pass (cutoff=0.0050): cat 4400, mi = 0.002, pvalue = 0.033, WARTERS_IR_RESPONSE_5GY\n",
      "Didn't pass (cutoff=0.0050): cat 3755, mi = 0.002, pvalue = 0.031, ROVERSI_GLIOMA_COPY_NUMBER_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4602, mi = 0.002, pvalue = 0.007, YIH_RESPONSE_TO_ARSENITE_C3\n",
      "Didn't pass (cutoff=0.0050): cat 1472, mi = 0.002, pvalue = 0.036, IWANAGA_CARCINOGENESIS_BY_KRAS_PTEN_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2794, mi = 0.002, pvalue = 0.018, PID_HDAC_CLASSII_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 4622, mi = 0.002, pvalue = 0.021, ZEMBUTSU_SENSITIVITY_TO_CYCLOPHOSPHAMIDE\n",
      "Didn't pass (cutoff=0.0050): cat 3462, mi = 0.002, pvalue = 0.034, REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 1035, mi = 0.002, pvalue = 0.005, FUKUSHIMA_TNFSF11_TARGETS\n",
      "Total number of categories evaluated = 126.\n",
      "Number of categories that passed the tests = 13\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 10, 12, 0.0950501170871971\n",
      "best merge is 1, 8, 0.188065333903958\n",
      "best merge is 7, 10, 0.202961349568326\n",
      "best merge is 6, 7, 0.342754262022997\n",
      "best merge is 4, 5, 0.364047391957158\n",
      "best merge is 0, 9, 0.386487080672689\n",
      "best merge is 1, 4, 0.497559916446469\n",
      "best merge is 0, 1, 0.626920368557117\n",
      "best merge is 3, 11, 0.711291723129224\n",
      "order classes\n",
      "6 - 0.956246587686075\n",
      "4 - 0.621251963431467\n",
      "6 7 10 12\n",
      "2\n",
      "3\n",
      "0 9 1 8 4 5\n",
      "11\n",
      "Done.Start drawing\n",
      "5.81\t-10.06\n",
      "LINDSTEDT_DENDRITIC_CELL_MATURATION_C\n",
      "KEGG_ERBB_SIGNALING_PATHWAY\n",
      "GARGALOVIC_RESPONSE_TO_OXIDIZED_PHOSPHOLIPIDS_YELLOW_UP\n",
      "ACOSTA_PROLIFERATION_INDEPENDENT_MYC_TARGETS_DN\n",
      "DARWICHE_SKIN_TUMOR_PROMOTER_UP\n",
      "BERENJENO_TRANSFORMED_BY_RHOA_FOREVER_DN\n",
      "DAZARD_UV_RESPONSE_CLUSTER_G1\n",
      "LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "LEE_CALORIE_RESTRICTION_NEOCORTEX_DN\n",
      "HOOI_ST7_TARGETS_DN\n",
      "LIN_NPAS4_TARGETS_DN\n",
      "SASAI_RESISTANCE_TO_NEOPLASTIC_TRANSFROMATION\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 10, 12, 0.0950501170871971\n",
      "best merge is 1, 8, 0.188065333903958\n",
      "best merge is 7, 10, 0.202961349568326\n",
      "best merge is 6, 7, 0.342754262022997\n",
      "best merge is 4, 5, 0.364047391957158\n",
      "best merge is 0, 9, 0.386487080672689\n",
      "best merge is 1, 4, 0.497559916446469\n",
      "best merge is 0, 1, 0.626920368557117\n",
      "best merge is 3, 11, 0.711291723129224\n",
      "order classes\n",
      "6 - 0.956246587686075\n",
      "4 - 0.621251963431467\n",
      "6 7 10 12\n",
      "2\n",
      "3\n",
      "0 9 1 8 4 5\n",
      "11\n",
      "Done.-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_c2/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 836\n",
      "cat 391, mi = 0.00372232, GGGCATT_MIR365\n",
      "cat 550, mi = 0.00278432, NKX25_01\n",
      "cat 726, mi = 0.00267560, TCTGATA_MIR361\n",
      "cat 42, mi = 0.00261415, ACTGCAG_MIR173P\n",
      "cat 50, mi = 0.00257768, AGCATTA_MIR155\n",
      "cat 381, mi = 0.00253557, GGCAGTG_MIR3243P\n",
      "cat 382, mi = 0.00240625, GGCCAGT_MIR193A_MIR193B\n",
      "cat 377, mi = 0.00240281, GGCACAT_MIR455\n",
      "cat 612, mi = 0.00239857, RACTNNRTTTNC_UNKNOWN\n",
      "cat 7, mi = 0.00234875, AACATTC_MIR4093P\n",
      "cat 803, mi = 0.00234288, WWTAAGGC_UNKNOWN\n",
      "cat 59, mi = 0.00229298, AGGGCCA_MIR328\n",
      "cat 26, mi = 0.00228199, ACACTCC_MIR122A\n",
      "cat 464, mi = 0.00226374, HOX13_01\n",
      "cat 449, mi = 0.00225498, HMEF2_Q6\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 391, mi = 0.004, pvalue = 0.000, GGGCATT_MIR365\n",
      "Didn't pass (cutoff=0.0050): cat 726, mi = 0.003, pvalue = 0.010, TCTGATA_MIR361\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.003, pvalue = 0.013, ACTGCAG_MIR173P\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.002, pvalue = 0.025, GGCACAT_MIR455\n",
      "Didn't pass (cutoff=0.0050): cat 612, mi = 0.002, pvalue = 0.022, RACTNNRTTTNC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.002, pvalue = 0.023, AACATTC_MIR4093P\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.002, pvalue = 0.035, ACACTCC_MIR122A\n",
      "Didn't pass (cutoff=0.0050): cat 464, mi = 0.002, pvalue = 0.038, HOX13_01\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.035, HMEF2_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 421, mi = 0.002, pvalue = 0.032, GTGACTT_MIR224\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.002, pvalue = 0.031, ATACCTC_MIR202\n",
      "Didn't pass (cutoff=0.0050): cat 162, mi = 0.002, pvalue = 0.055, CAGTCAC_MIR134\n",
      "Didn't pass (cutoff=0.0050): cat 379, mi = 0.002, pvalue = 0.051, GGCAGAC_MIR346\n",
      "Didn't pass (cutoff=0.0050): cat 662, mi = 0.002, pvalue = 0.058, SRF_01\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.002, pvalue = 0.056, AGCGCAG_MIR191\n",
      "Didn't pass (cutoff=0.0050): cat 222, mi = 0.002, pvalue = 0.038, CREBP1_01\n",
      "Didn't pass (cutoff=0.0050): cat 243, mi = 0.002, pvalue = 0.057, CTCNANGTGNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 393, mi = 0.002, pvalue = 0.061, GGGGCCC_MIR296\n",
      "Didn't pass (cutoff=0.0050): cat 588, mi = 0.002, pvalue = 0.064, PAX8_01\n",
      "Didn't pass (cutoff=0.0050): cat 414, mi = 0.002, pvalue = 0.052, GTCAACC_MIR3805P\n",
      "Didn't pass (cutoff=0.0050): cat 515, mi = 0.002, pvalue = 0.070, MYAATNNNNNNNGGC_UNKNOWN\n",
      "Total number of categories evaluated = 23.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "GGGCATT_MIR365\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_c3/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 858\n",
      "cat 321, mi = 0.00359784, MODULE_183\n",
      "cat 561, mi = 0.00331848, MODULE_480\n",
      "cat 526, mi = 0.00279330, MODULE_427\n",
      "cat 703, mi = 0.00267358, MORF_CUL1\n",
      "cat 668, mi = 0.00242762, MORF_ANP32B\n",
      "cat 443, mi = 0.00239642, MODULE_322\n",
      "cat 456, mi = 0.00239285, MODULE_334\n",
      "cat 840, mi = 0.00237650, MORF_TERF2IP\n",
      "cat 823, mi = 0.00237520, MORF_RFC1\n",
      "cat 9, mi = 0.00233030, GCM_APEX1\n",
      "cat 229, mi = 0.00232037, GNF2_TYK2\n",
      "cat 372, mi = 0.00230941, MODULE_239\n",
      "cat 121, mi = 0.00229439, GNF2_CENPF\n",
      "cat 712, mi = 0.00229059, MORF_DNMT1\n",
      "cat 281, mi = 0.00225792, MODULE_145\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 321, mi = 0.004, pvalue = 0.000, MODULE_183\n",
      "Passed: cat 561, mi = 0.003, pvalue = 0.001, MODULE_480\n",
      "Didn't pass (cutoff=0.0050): cat 703, mi = 0.003, pvalue = 0.012, MORF_CUL1\n",
      "Didn't pass (cutoff=0.0050): cat 443, mi = 0.002, pvalue = 0.022, MODULE_322\n",
      "Didn't pass (cutoff=0.0050): cat 840, mi = 0.002, pvalue = 0.023, MORF_TERF2IP\n",
      "Didn't pass (cutoff=0.0050): cat 823, mi = 0.002, pvalue = 0.022, MORF_RFC1\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.002, pvalue = 0.032, GNF2_TYK2\n",
      "Didn't pass (cutoff=0.0050): cat 372, mi = 0.002, pvalue = 0.024, MODULE_239\n",
      "Didn't pass (cutoff=0.0050): cat 121, mi = 0.002, pvalue = 0.037, GNF2_CENPF\n",
      "Didn't pass (cutoff=0.0050): cat 281, mi = 0.002, pvalue = 0.033, MODULE_145\n",
      "Didn't pass (cutoff=0.0050): cat 426, mi = 0.002, pvalue = 0.025, MODULE_301\n",
      "Didn't pass (cutoff=0.0050): cat 113, mi = 0.002, pvalue = 0.048, GNF2_CDC2\n",
      "Didn't pass (cutoff=0.0050): cat 423, mi = 0.002, pvalue = 0.006, MODULE_299\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.002, pvalue = 0.055, GCM_DDX11\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.002, pvalue = 0.051, GNF2_CASP8\n",
      "Didn't pass (cutoff=0.0050): cat 650, mi = 0.002, pvalue = 0.035, MODULE_86\n",
      "Didn't pass (cutoff=0.0050): cat 294, mi = 0.002, pvalue = 0.024, MODULE_157\n",
      "Didn't pass (cutoff=0.0050): cat 809, mi = 0.002, pvalue = 0.063, MORF_RAB6A\n",
      "Didn't pass (cutoff=0.0050): cat 416, mi = 0.002, pvalue = 0.073, MODULE_291\n",
      "Didn't pass (cutoff=0.0050): cat 598, mi = 0.002, pvalue = 0.064, MODULE_537\n",
      "Didn't pass (cutoff=0.0050): cat 57, mi = 0.002, pvalue = 0.059, GCM_NUMA1\n",
      "Didn't pass (cutoff=0.0050): cat 219, mi = 0.002, pvalue = 0.057, GNF2_TAL1\n",
      "Total number of categories evaluated = 26.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "MODULE_183\n",
      "MODULE_480\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_c4/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 5917\n",
      "cat 3913, mi = 0.00380333, GO_PROTEIN_ACETYLATION\n",
      "cat 2676, mi = 0.00371216, GO_NEGATIVE_REGULATION_OF_OSSIFICATION\n",
      "cat 5372, mi = 0.00359205, GO_SH3_SH2_ADAPTOR_ACTIVITY\n",
      "cat 4409, mi = 0.00357409, GO_REGULATION_OF_EPITHELIAL_CELL_DIFFERENTIATION\n",
      "cat 4251, mi = 0.00353303, GO_REGULATION_OF_CATION_CHANNEL_ACTIVITY\n",
      "cat 3915, mi = 0.00342881, GO_PROTEIN_ACYLATION\n",
      "cat 462, mi = 0.00341999, GO_CARDIOCYTE_DIFFERENTIATION\n",
      "cat 1071, mi = 0.00338838, GO_DNA_DIRECTED_RNA_POLYMERASE_II_HOLOENZYME\n",
      "cat 5273, mi = 0.00335509, GO_RNA_POLYMERASE_COMPLEX\n",
      "cat 2925, mi = 0.00324355, GO_NUCLEAR_SPECK\n",
      "cat 822, mi = 0.00315031, GO_COATED_PIT\n",
      "cat 5041, mi = 0.00306919, GO_RESPONSE_TO_AMINO_ACID\n",
      "cat 4225, mi = 0.00306368, GO_REGULATION_OF_CALCIUM_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "cat 640, mi = 0.00300616, GO_CELL_CHEMOTAXIS\n",
      "cat 352, mi = 0.00297070, GO_BONE_DEVELOPMENT\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 3913, mi = 0.004, pvalue = 0.000, GO_PROTEIN_ACETYLATION\n",
      "Passed: cat 2676, mi = 0.004, pvalue = 0.000, GO_NEGATIVE_REGULATION_OF_OSSIFICATION\n",
      "Passed: cat 5372, mi = 0.004, pvalue = 0.000, GO_SH3_SH2_ADAPTOR_ACTIVITY\n",
      "Passed: cat 4251, mi = 0.004, pvalue = 0.000, GO_REGULATION_OF_CATION_CHANNEL_ACTIVITY\n",
      "Passed: cat 2925, mi = 0.003, pvalue = 0.001, GO_NUCLEAR_SPECK\n",
      "Passed: cat 822, mi = 0.003, pvalue = 0.002, GO_COATED_PIT\n",
      "Passed: cat 5041, mi = 0.003, pvalue = 0.003, GO_RESPONSE_TO_AMINO_ACID\n",
      "Didn't pass (cutoff=0.0050): cat 2398, mi = 0.003, pvalue = 0.008, GO_NCRNA_TRANSCRIPTION\n",
      "Passed: cat 1671, mi = 0.003, pvalue = 0.003, GO_HORMONE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2576, mi = 0.003, pvalue = 0.008, GO_NEGATIVE_REGULATION_OF_IMMUNE_RESPONSE\n",
      "Passed: cat 5757, mi = 0.003, pvalue = 0.003, GO_TRNA_METHYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3024, mi = 0.003, pvalue = 0.009, GO_ORGANELLE_ENVELOPE_LUMEN\n",
      "Didn't pass (cutoff=0.0050): cat 737, mi = 0.003, pvalue = 0.012, GO_CHAPERONE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 4582, mi = 0.003, pvalue = 0.006, GO_REGULATION_OF_LIPID_STORAGE\n",
      "Didn't pass (cutoff=0.0050): cat 2781, mi = 0.003, pvalue = 0.011, GO_NEGATIVE_REGULATION_OF_TUMOR_NECROSIS_FACTOR_SUPERFAMILY_CYTOKINE_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 4348, mi = 0.003, pvalue = 0.011, GO_REGULATION_OF_CYTOKINESIS\n",
      "Didn't pass (cutoff=0.0050): cat 109, mi = 0.003, pvalue = 0.010, GO_ALCOHOL_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3048, mi = 0.003, pvalue = 0.011, GO_ORGANIC_HYDROXY_COMPOUND_CATABOLIC_PROCESS\n",
      "Passed: cat 1767, mi = 0.003, pvalue = 0.001, GO_INTEGRATOR_COMPLEX\n",
      "Passed: cat 4230, mi = 0.003, pvalue = 0.004, GO_REGULATION_OF_CAMP_MEDIATED_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 5323, mi = 0.002, pvalue = 0.017, GO_SCF_UBIQUITIN_LIGASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 3559, mi = 0.002, pvalue = 0.015, GO_POSITIVE_REGULATION_OF_INTERFERON_BETA_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 4332, mi = 0.002, pvalue = 0.019, GO_REGULATION_OF_CHROMOSOME_SEGREGATION\n",
      "Passed: cat 3458, mi = 0.002, pvalue = 0.004, GO_POSITIVE_REGULATION_OF_CYTOKINESIS\n",
      "Didn't pass (cutoff=0.0050): cat 1754, mi = 0.002, pvalue = 0.021, GO_INOSITOL_PHOSPHATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4835, mi = 0.002, pvalue = 0.017, GO_REGULATION_OF_RESPONSE_TO_CYTOKINE_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 4045, mi = 0.002, pvalue = 0.027, GO_PROTEIN_TRANSMEMBRANE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 2196, mi = 0.002, pvalue = 0.029, GO_MIDBRAIN_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 1253, mi = 0.002, pvalue = 0.021, GO_EPITHELIAL_CELL_MORPHOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 2041, mi = 0.002, pvalue = 0.019, GO_MAINTENANCE_OF_PROTEIN_LOCALIZATION_IN_ORGANELLE\n",
      "Didn't pass (cutoff=0.0050): cat 891, mi = 0.002, pvalue = 0.031, GO_CUL3_RING_UBIQUITIN_LIGASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 1208, mi = 0.002, pvalue = 0.027, GO_ENDORIBONUCLEASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 2679, mi = 0.002, pvalue = 0.012, GO_NEGATIVE_REGULATION_OF_OXIDATIVE_STRESS_INDUCED_INTRINSIC_APOPTOTIC_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 879, mi = 0.002, pvalue = 0.025, GO_CORTICAL_ACTIN_CYTOSKELETON\n",
      "Didn't pass (cutoff=0.0050): cat 3202, mi = 0.002, pvalue = 0.022, GO_PHOSPHATASE_INHIBITOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1326, mi = 0.002, pvalue = 0.019, GO_EXODEOXYRIBONUCLEASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5214, mi = 0.002, pvalue = 0.042, GO_RETROGRADE_TRANSPORT_ENDOSOME_TO_GOLGI\n",
      "Didn't pass (cutoff=0.0050): cat 1065, mi = 0.002, pvalue = 0.044, GO_DNA_DEPENDENT_ATPASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2792, mi = 0.002, pvalue = 0.039, GO_NEGATIVE_REGULATION_OF_VIRAL_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2112, mi = 0.002, pvalue = 0.031, GO_MEMBRANE_PROTEIN_ECTODOMAIN_PROTEOLYSIS\n",
      "Didn't pass (cutoff=0.0050): cat 1192, mi = 0.002, pvalue = 0.043, GO_ENDONUCLEASE_ACTIVITY_ACTIVE_WITH_EITHER_RIBO_OR_DEOXYRIBONUCLEIC_ACIDS_AND_PRODUCING_5_PHOSPHOMONOESTERS\n",
      "Didn't pass (cutoff=0.0050): cat 5785, mi = 0.002, pvalue = 0.041, GO_T_CELL_PROLIFERATION\n",
      "Didn't pass (cutoff=0.0050): cat 4865, mi = 0.002, pvalue = 0.039, GO_REGULATION_OF_SISTER_CHROMATID_SEGREGATION\n",
      "Didn't pass (cutoff=0.0050): cat 820, mi = 0.002, pvalue = 0.044, GO_CLUSTER_OF_ACTIN_BASED_CELL_PROJECTIONS\n",
      "Total number of categories evaluated = 139.\n",
      "Number of categories that passed the tests = 12\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 8, 9, 0.227752461373296\n",
      "best merge is 3, 6, 0.255840837749657\n",
      "best merge is 2, 4, 0.319069802087905\n",
      "best merge is 0, 2, 0.515933996468983\n",
      "best merge is 7, 10, 0.582176382185643\n",
      "best merge is 3, 5, 0.613281198089772\n",
      "best merge is 1, 7, 0.773559138652121\n",
      "best merge is 0, 3, 0.890813262752557\n",
      "order classes\n",
      "3 - 1.20083359557855\n",
      "3 - 1.89270636399101\n",
      "3 - 1.31769176349262\n",
      "2 - 0.683257384119888\n",
      "8 9\n",
      "1 7 10\n",
      "3 6 5\n",
      "0 2 4\n",
      "11\n",
      "Done.Start drawing\n",
      "5.81\t-10.06\n",
      "GO_PROTEIN_ACETYLATION\n",
      "GO_NEGATIVE_REGULATION_OF_OSSIFICATION\n",
      "GO_COATED_PIT\n",
      "GO_NUCLEAR_SPECK\n",
      "GO_REGULATION_OF_CAMP_MEDIATED_SIGNALING\n",
      "GO_REGULATION_OF_CATION_CHANNEL_ACTIVITY\n",
      "GO_HORMONE_BINDING\n",
      "GO_RESPONSE_TO_AMINO_ACID\n",
      "GO_TRNA_METHYLTRANSFERASE_ACTIVITY\n",
      "GO_INTEGRATOR_COMPLEX\n",
      "GO_SH3_SH2_ADAPTOR_ACTIVITY\n",
      "GO_POSITIVE_REGULATION_OF_CYTOKINESIS\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 8, 9, 0.227752461373296\n",
      "best merge is 3, 6, 0.255840837749657\n",
      "best merge is 2, 4, 0.319069802087905\n",
      "best merge is 0, 2, 0.515933996468983\n",
      "best merge is 7, 10, 0.582176382185643\n",
      "best merge is 3, 5, 0.613281198089772\n",
      "best merge is 1, 7, 0.773559138652121\n",
      "best merge is 0, 3, 0.890813262752557\n",
      "order classes\n",
      "3 - 1.20083359557855\n",
      "3 - 1.89270636399101\n",
      "3 - 1.31769176349262\n",
      "2 - 0.683257384119888\n",
      "8 9\n",
      "1 7 10\n",
      "3 6 5\n",
      "0 2 4\n",
      "11\n",
      "Done.-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_c5/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 189\n",
      "cat 138, mi = 0.00369224, PRC1_BMI_UP.V1_DN\n",
      "cat 185, mi = 0.00310159, WNT_UP.V1_DN\n",
      "cat 98, mi = 0.00293279, KRAS.DF.V1_UP\n",
      "cat 81, mi = 0.00282861, JAK2_DN.V1_DN\n",
      "cat 37, mi = 0.00282524, CSR_LATE_UP.V1_DN\n",
      "cat 99, mi = 0.00273994, KRAS.KIDNEY_UP.V1_DN\n",
      "cat 168, mi = 0.00271044, SNF5_DN.V1_UP\n",
      "cat 105, mi = 0.00261937, KRAS.PROSTATE_UP.V1_DN\n",
      "cat 32, mi = 0.00231937, CRX_DN.V1_UP\n",
      "cat 186, mi = 0.00228924, WNT_UP.V1_UP\n",
      "cat 103, mi = 0.00228413, KRAS.LUNG_UP.V1_DN\n",
      "cat 78, mi = 0.00227822, IL21_UP.V1_UP\n",
      "cat 104, mi = 0.00226482, KRAS.LUNG_UP.V1_UP\n",
      "cat 142, mi = 0.00211172, PRC2_EZH2_UP.V1_DN\n",
      "cat 163, mi = 0.00210024, RPS14_DN.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 138, mi = 0.004, pvalue = 0.000, PRC1_BMI_UP.V1_DN\n",
      "Passed: cat 185, mi = 0.003, pvalue = 0.002, WNT_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 81, mi = 0.003, pvalue = 0.006, JAK2_DN.V1_DN\n",
      "Passed: cat 37, mi = 0.003, pvalue = 0.004, CSR_LATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.003, pvalue = 0.008, KRAS.KIDNEY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 168, mi = 0.003, pvalue = 0.008, SNF5_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 105, mi = 0.003, pvalue = 0.013, KRAS.PROSTATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.002, pvalue = 0.034, CRX_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 186, mi = 0.002, pvalue = 0.033, WNT_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 103, mi = 0.002, pvalue = 0.037, KRAS.LUNG_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 104, mi = 0.002, pvalue = 0.042, KRAS.LUNG_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 53, mi = 0.002, pvalue = 0.046, EIF4E_DN\n",
      "Didn't pass (cutoff=0.0050): cat 61, mi = 0.002, pvalue = 0.062, ESC_V6.5_UP_EARLY.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 120, mi = 0.002, pvalue = 0.057, MYC_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 183, mi = 0.002, pvalue = 0.064, VEGF_A_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.002, pvalue = 0.090, ERB2_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.083, CRX_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.093, RB_P107_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.002, pvalue = 0.122, PDGF_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 82, mi = 0.002, pvalue = 0.120, JAK2_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 77, mi = 0.002, pvalue = 0.141, IL21_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 153, mi = 0.002, pvalue = 0.098, RAPA_EARLY_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 70, mi = 0.002, pvalue = 0.157, GLI1_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.002, pvalue = 0.159, RB_P130_DN.V1_UP\n",
      "Total number of categories evaluated = 26.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.928156170898188\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "5.81\t-10.06\n",
      "PRC1_BMI_UP.V1_DN\n",
      "CSR_LATE_UP.V1_DN\n",
      "WNT_UP.V1_DN\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.928156170898188\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_c6/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 4872\n",
      "cat 2226, mi = 0.00546523, GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "cat 65, mi = 0.00526414, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 4673, mi = 0.00520939, GSE7852_TREG_VS_TCONV_LN_DN\n",
      "cat 4587, mi = 0.00431960, GSE7460_TCONV_VS_TREG_LN_UP\n",
      "cat 4544, mi = 0.00407107, GSE7219_UNSTIM_VS_LPS_AND_ANTI_CD40_STIM_DC_DN\n",
      "cat 2708, mi = 0.00399801, GSE29617_CTRL_VS_DAY3_TIV_FLU_VACCINE_PBMC_2008_DN\n",
      "cat 597, mi = 0.00396885, GSE15330_HSC_VS_PRO_BCELL_UP\n",
      "cat 2496, mi = 0.00385302, GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "cat 4432, mi = 0.00384159, GSE5679_PPARG_LIGAND_ROSIGLITAZONE_VS_RARA_AGONIST_AM580_TREATED_DC_DN\n",
      "cat 206, mi = 0.00380300, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "cat 53, mi = 0.00372730, GSE10325_BCELL_VS_MYELOID_UP\n",
      "cat 4670, mi = 0.00371119, GSE7852_TREG_VS_TCONV_DN\n",
      "cat 2815, mi = 0.00370454, GSE3039_ALPHABETA_CD8_TCELL_VS_B2_BCELL_UP\n",
      "cat 59, mi = 0.00369326, GSE10325_CD4_TCELL_VS_MYELOID_UP\n",
      "cat 1653, mi = 0.00368230, GSE21670_STAT3_KO_VS_WT_CD4_TCELL_TGFB_IL6_TREATED_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2226, mi = 0.005, pvalue = 0.000, GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "Passed: cat 65, mi = 0.005, pvalue = 0.000, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "Passed: cat 4544, mi = 0.004, pvalue = 0.000, GSE7219_UNSTIM_VS_LPS_AND_ANTI_CD40_STIM_DC_DN\n",
      "Passed: cat 2708, mi = 0.004, pvalue = 0.000, GSE29617_CTRL_VS_DAY3_TIV_FLU_VACCINE_PBMC_2008_DN\n",
      "Passed: cat 2496, mi = 0.004, pvalue = 0.000, GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Passed: cat 4432, mi = 0.004, pvalue = 0.000, GSE5679_PPARG_LIGAND_ROSIGLITAZONE_VS_RARA_AGONIST_AM580_TREATED_DC_DN\n",
      "Passed: cat 3991, mi = 0.004, pvalue = 0.000, GSE42021_TCONV_PLN_VS_CD24HI_TCONV_THYMUS_UP\n",
      "Passed: cat 4697, mi = 0.004, pvalue = 0.000, GSE8678_IL7R_LOW_VS_HIGH_EFF_CD8_TCELL_UP\n",
      "Passed: cat 1562, mi = 0.003, pvalue = 0.001, GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_DN\n",
      "Passed: cat 4539, mi = 0.003, pvalue = 0.001, GSE7218_IGM_VS_IGG_SIGNAL_THGOUGH_ANTIGEN_BCELL_UP\n",
      "Passed: cat 4147, mi = 0.003, pvalue = 0.001, GSE45365_CTRL_VS_MCMV_INFECTION_NK_CELL_UP\n",
      "Passed: cat 4850, mi = 0.003, pvalue = 0.001, GSE9988_LOW_LPS_VS_VEHICLE_TREATED_MONOCYTE_DN\n",
      "Passed: cat 1685, mi = 0.003, pvalue = 0.001, GSE21774_CD62L_POS_CD56_DIM_VS_CD62L_NEG_CD56_DIM_NK_CELL_UP\n",
      "Passed: cat 4223, mi = 0.003, pvalue = 0.003, GSE46242_TH1_VS_ANERGIC_TH1_CD4_TCELL_UP\n",
      "Passed: cat 4605, mi = 0.003, pvalue = 0.003, GSE7509_UNSTIM_VS_FCGRIIB_STIM_DC_UP\n",
      "Passed: cat 1092, mi = 0.003, pvalue = 0.002, GSE17721_PAM3CSK4_VS_CPG_8H_BMDC_DN\n",
      "Passed: cat 431, mi = 0.003, pvalue = 0.002, GSE14308_TH17_VS_NATURAL_TREG_UP\n",
      "Passed: cat 4268, mi = 0.003, pvalue = 0.002, GSE46606_UNSTIM_VS_CD40L_IL2_IL5_DAY3_STIMULATED_BCELL_DN\n",
      "Passed: cat 1469, mi = 0.003, pvalue = 0.005, GSE20366_CD103_POS_VS_NEG_TREG_KLRG1NEG_UP\n",
      "Passed: cat 129, mi = 0.003, pvalue = 0.003, GSE11924_TFH_VS_TH1_CD4_TCELL_UP\n",
      "Passed: cat 135, mi = 0.003, pvalue = 0.003, GSE11924_TH1_VS_TH2_CD4_TCELL_UP\n",
      "Passed: cat 289, mi = 0.003, pvalue = 0.004, GSE13411_IGM_VS_SWITCHED_MEMORY_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4138, mi = 0.003, pvalue = 0.007, GSE45365_BCELL_VS_CD8_TCELL_DN\n",
      "Passed: cat 3212, mi = 0.003, pvalue = 0.005, GSE360_T_GONDII_VS_B_MALAYI_LOW_DOSE_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3772, mi = 0.003, pvalue = 0.008, GSE40274_EOS_VS_FOXP3_AND_EOS_TRANSDUCED_ACTIVATED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1448, mi = 0.003, pvalue = 0.005, GSE20152_SPHK1_KO_VS_HTNFA_OVEREXPRESS_ANKLE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 505, mi = 0.003, pvalue = 0.005, GSE1448_CTRL_VS_ANTI_VBETA5_DP_THYMOCYTE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2869, mi = 0.003, pvalue = 0.008, GSE31082_DN_VS_DP_THYMOCYTE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 201, mi = 0.003, pvalue = 0.005, GSE12198_NK_VS_NK_ACT_EXPANSION_SYSTEM_DERIVED_NK_CELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4414, mi = 0.003, pvalue = 0.008, GSE5589_WT_VS_IL10_KO_LPS_STIM_MACROPHAGE_45MIN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 651, mi = 0.003, pvalue = 0.008, GSE1566_WT_VS_EZH2_KO_LN_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 985, mi = 0.003, pvalue = 0.007, GSE17721_CTRL_VS_PAM3CSK4_6H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1514, mi = 0.003, pvalue = 0.009, GSE20727_CTRL_VS_DNFB_ALLERGEN_TREATED_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 560, mi = 0.003, pvalue = 0.011, GSE14908_ATOPIC_VS_NONATOPIC_PATIENT_HDM_STIM_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4425, mi = 0.003, pvalue = 0.007, GSE5589_WT_VS_IL6_KO_LPS_STIM_MACROPHAGE_45MIN_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3312, mi = 0.003, pvalue = 0.015, GSE3720_LPS_VS_PMA_STIM_VD2_GAMMADELTA_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1847, mi = 0.003, pvalue = 0.012, GSE22611_UNSTIM_VS_2H_MDP_STIM_NOD2_TRANSDUCED_HEK293T_CELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4272, mi = 0.003, pvalue = 0.017, GSE4748_CTRL_VS_CYANOBACTERIUM_LPSLIKE_STIM_DC_3H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4010, mi = 0.002, pvalue = 0.016, GSE42088_UNINF_VS_LEISHMANIA_INF_DC_24H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.014, GSE15139_GMCSF_TREATED_VS_UNTREATED_NEUTROPHILS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3824, mi = 0.002, pvalue = 0.015, GSE40666_UNTREATED_VS_IFNA_STIM_CD8_TCELL_90MIN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3916, mi = 0.002, pvalue = 0.015, GSE41867_DAY6_VS_DAY8_LCMV_ARMSTRONG_EFFECTOR_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1323, mi = 0.002, pvalue = 0.025, GSE19198_1H_VS_6H_IL21_TREATED_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 984, mi = 0.002, pvalue = 0.016, GSE17721_CTRL_VS_PAM3CSK4_6H_BMDC_DN\n",
      "Total number of categories evaluated = 238.\n",
      "Number of categories that passed the tests = 23\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 5, 0.120969208430764\n",
      "best merge is 0, 1, 0.160861094835309\n",
      "best merge is 6, 11, 0.20265898878909\n",
      "best merge is 9, 15, 0.24066481155955\n",
      "best merge is 0, 6, 0.322810321876976\n",
      "best merge is 3, 20, 0.357886466315949\n",
      "best merge is 12, 16, 0.37714518627478\n",
      "best merge is 4, 10, 0.39250115124526\n",
      "best merge is 2, 17, 0.447681211453734\n",
      "best merge is 14, 21, 0.45412907724918\n",
      "best merge is 12, 13, 0.549256958060896\n",
      "best merge is 3, 22, 0.558890720347996\n",
      "best merge is 9, 14, 0.629489436953147\n",
      "best merge is 2, 8, 0.639283369444921\n",
      "best merge is 0, 7, 0.660182896028066\n",
      "best merge is 2, 12, 0.685885844262931\n",
      "best merge is 9, 18, 0.725544494362274\n",
      "best merge is 0, 19, 0.796645597252772\n",
      "best merge is 0, 2, 0.978052827206337\n",
      "order classes\n",
      "7 - 1.43002337481991\n",
      "6 - 1.99650214792083\n",
      "3 - 2.51410087861294\n",
      "2 - 2.25688161966024\n",
      "5 - 2.25029409797791\n",
      "0 5 1 6 11 7 19\n",
      "2 17 8 12 16 13\n",
      "3 20 22\n",
      "4 10\n",
      "9 15 14 21 18\n",
      "Done.Start drawing\n",
      "5.81\t-10.06\n",
      "GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_DN\n",
      "GSE7219_UNSTIM_VS_LPS_AND_ANTI_CD40_STIM_DC_DN\n",
      "GSE46606_UNSTIM_VS_CD40L_IL2_IL5_DAY3_STIMULATED_BCELL_DN\n",
      "GSE11924_TH1_VS_TH2_CD4_TCELL_UP\n",
      "GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "GSE42021_TCONV_PLN_VS_CD24HI_TCONV_THYMUS_UP\n",
      "GSE9988_LOW_LPS_VS_VEHICLE_TREATED_MONOCYTE_DN\n",
      "GSE8678_IL7R_LOW_VS_HIGH_EFF_CD8_TCELL_UP\n",
      "GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "GSE45365_CTRL_VS_MCMV_INFECTION_NK_CELL_UP\n",
      "GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "GSE29617_CTRL_VS_DAY3_TIV_FLU_VACCINE_PBMC_2008_DN\n",
      "GSE5679_PPARG_LIGAND_ROSIGLITAZONE_VS_RARA_AGONIST_AM580_TREATED_DC_DN\n",
      "GSE21774_CD62L_POS_CD56_DIM_VS_CD62L_NEG_CD56_DIM_NK_CELL_UP\n",
      "GSE7218_IGM_VS_IGG_SIGNAL_THGOUGH_ANTIGEN_BCELL_UP\n",
      "GSE17721_PAM3CSK4_VS_CPG_8H_BMDC_DN\n",
      "GSE11924_TFH_VS_TH1_CD4_TCELL_UP\n",
      "GSE13411_IGM_VS_SWITCHED_MEMORY_BCELL_UP\n",
      "GSE14308_TH17_VS_NATURAL_TREG_UP\n",
      "GSE46242_TH1_VS_ANERGIC_TH1_CD4_TCELL_UP\n",
      "GSE360_T_GONDII_VS_B_MALAYI_LOW_DOSE_DC_DN\n",
      "GSE20366_CD103_POS_VS_NEG_TREG_KLRG1NEG_UP\n",
      "GSE7509_UNSTIM_VS_FCGRIIB_STIM_DC_UP\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 5, 0.120969208430764\n",
      "best merge is 0, 1, 0.160861094835309\n",
      "best merge is 6, 11, 0.20265898878909\n",
      "best merge is 9, 15, 0.24066481155955\n",
      "best merge is 0, 6, 0.322810321876976\n",
      "best merge is 3, 20, 0.357886466315949\n",
      "best merge is 12, 16, 0.37714518627478\n",
      "best merge is 4, 10, 0.39250115124526\n",
      "best merge is 2, 17, 0.447681211453734\n",
      "best merge is 14, 21, 0.45412907724918\n",
      "best merge is 12, 13, 0.549256958060896\n",
      "best merge is 3, 22, 0.558890720347996\n",
      "best merge is 9, 14, 0.629489436953147\n",
      "best merge is 2, 8, 0.639283369444921\n",
      "best merge is 0, 7, 0.660182896028066\n",
      "best merge is 2, 12, 0.685885844262931\n",
      "best merge is 9, 18, 0.725544494362274\n",
      "best merge is 0, 19, 0.796645597252772\n",
      "best merge is 0, 2, 0.978052827206337\n",
      "order classes\n",
      "7 - 1.43002337481991\n",
      "6 - 1.99650214792083\n",
      "3 - 2.51410087861294\n",
      "2 - 2.25688161966024\n",
      "5 - 2.25029409797791\n",
      "0 5 1 6 11 7 19\n",
      "2 17 8 12 16 13\n",
      "3 20 22\n",
      "4 10\n",
      "9 15 14 21 18\n",
      "Done.-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_c7/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 17786\n",
      "cat 10083, mi = 0.00546523, GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "cat 13611, mi = 0.00539882, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "cat 7922, mi = 0.00526414, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 12530, mi = 0.00520939, GSE7852_TREG_VS_TCONV_LN_DN\n",
      "cat 12444, mi = 0.00431960, GSE7460_TCONV_VS_TREG_LN_UP\n",
      "cat 12401, mi = 0.00407107, GSE7219_UNSTIM_VS_LPS_AND_ANTI_CD40_STIM_DC_DN\n",
      "cat 238, mi = 0.00406525, AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "cat 1178, mi = 0.00401068, DIRMEIER_LMP1_RESPONSE_EARLY\n",
      "cat 12924, mi = 0.00399804, HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "cat 10565, mi = 0.00399801, GSE29617_CTRL_VS_DAY3_TIV_FLU_VACCINE_PBMC_2008_DN\n",
      "cat 8454, mi = 0.00396885, GSE15330_HSC_VS_PRO_BCELL_UP\n",
      "cat 10353, mi = 0.00385302, GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "cat 12289, mi = 0.00384159, GSE5679_PPARG_LIGAND_ROSIGLITAZONE_VS_RARA_AGONIST_AM580_TREATED_DC_DN\n",
      "cat 5799, mi = 0.00380333, GO_PROTEIN_ACETYLATION\n",
      "cat 8063, mi = 0.00380300, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 10083, mi = 0.005, pvalue = 0.000, GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "Passed: cat 13611, mi = 0.005, pvalue = 0.000, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "Passed: cat 12530, mi = 0.005, pvalue = 0.000, GSE7852_TREG_VS_TCONV_LN_DN\n",
      "Passed: cat 238, mi = 0.004, pvalue = 0.000, AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "Passed: cat 10565, mi = 0.004, pvalue = 0.000, GSE29617_CTRL_VS_DAY3_TIV_FLU_VACCINE_PBMC_2008_DN\n",
      "Passed: cat 10353, mi = 0.004, pvalue = 0.000, GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Passed: cat 5799, mi = 0.004, pvalue = 0.000, GO_PROTEIN_ACETYLATION\n",
      "Passed: cat 1688, mi = 0.004, pvalue = 0.000, GGGCATT_MIR365\n",
      "Passed: cat 10672, mi = 0.004, pvalue = 0.000, GSE3039_ALPHABETA_CD8_TCELL_VS_B2_BCELL_UP\n",
      "Passed: cat 11848, mi = 0.004, pvalue = 0.001, GSE42021_TCONV_PLN_VS_CD24HI_TCONV_THYMUS_UP\n",
      "Passed: cat 12554, mi = 0.004, pvalue = 0.001, GSE8678_IL7R_LOW_VS_HIGH_EFF_CD8_TCELL_UP\n",
      "Passed: cat 7258, mi = 0.004, pvalue = 0.000, GO_SH3_SH2_ADAPTOR_ACTIVITY\n",
      "Passed: cat 10246, mi = 0.004, pvalue = 0.000, GSE2706_UNSTIM_VS_2H_LPS_DC_UP\n",
      "Passed: cat 12969, mi = 0.004, pvalue = 0.001, HOOI_ST7_TARGETS_DN\n",
      "Passed: cat 9419, mi = 0.003, pvalue = 0.001, GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_DN\n",
      "Passed: cat 13713, mi = 0.003, pvalue = 0.000, LIN_NPAS4_TARGETS_DN\n",
      "Passed: cat 12396, mi = 0.003, pvalue = 0.001, GSE7218_IGM_VS_IGG_SIGNAL_THGOUGH_ANTIGEN_BCELL_UP\n",
      "Passed: cat 13705, mi = 0.003, pvalue = 0.001, LINDSTEDT_DENDRITIC_CELL_MATURATION_C\n",
      "Passed: cat 9849, mi = 0.003, pvalue = 0.001, GSE23321_CD8_STEM_CELL_MEMORY_VS_EFFECTOR_MEMORY_CD8_TCELL_DN\n",
      "Passed: cat 15211, mi = 0.003, pvalue = 0.002, PID_FCER1_PATHWAY\n",
      "Passed: cat 12004, mi = 0.003, pvalue = 0.001, GSE45365_CTRL_VS_MCMV_INFECTION_NK_CELL_UP\n",
      "Passed: cat 12462, mi = 0.003, pvalue = 0.003, GSE7509_UNSTIM_VS_FCGRIIB_STIM_DC_UP\n",
      "Passed: cat 8288, mi = 0.003, pvalue = 0.003, GSE14308_TH17_VS_NATURAL_TREG_UP\n",
      "Passed: cat 1107, mi = 0.003, pvalue = 0.003, DARWICHE_SKIN_TUMOR_PROMOTER_UP\n",
      "Didn't pass (cutoff=0.0050): cat 8146, mi = 0.003, pvalue = 0.005, GSE13411_IGM_VS_SWITCHED_MEMORY_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4284, mi = 0.003, pvalue = 0.007, GO_NCRNA_TRANSCRIPTION\n",
      "Didn't pass (cutoff=0.0050): cat 11995, mi = 0.003, pvalue = 0.006, GSE45365_BCELL_VS_CD8_TCELL_DN\n",
      "Passed: cat 12497, mi = 0.003, pvalue = 0.004, GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 11069, mi = 0.003, pvalue = 0.006, GSE360_T_GONDII_VS_B_MALAYI_LOW_DOSE_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4462, mi = 0.003, pvalue = 0.007, GO_NEGATIVE_REGULATION_OF_IMMUNE_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 17083, mi = 0.003, pvalue = 0.008, WATANABE_RECTAL_CANCER_RADIOTHERAPY_RESPONSIVE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 9305, mi = 0.003, pvalue = 0.005, GSE20152_SPHK1_KO_VS_HTNFA_OVEREXPRESS_ANKLE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 17278, mi = 0.003, pvalue = 0.006, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_17\n",
      "Didn't pass (cutoff=0.0050): cat 10726, mi = 0.003, pvalue = 0.009, GSE31082_DN_VS_DP_THYMOCYTE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 9906, mi = 0.003, pvalue = 0.007, GSE2405_0H_VS_24H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 12271, mi = 0.003, pvalue = 0.005, GSE5589_WT_VS_IL10_KO_LPS_STIM_MACROPHAGE_45MIN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 12359, mi = 0.003, pvalue = 0.009, GSE6269_HEALTHY_VS_STAPH_AUREUS_INF_PBMC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4910, mi = 0.003, pvalue = 0.009, GO_ORGANELLE_ENVELOPE_LUMEN\n",
      "Didn't pass (cutoff=0.0050): cat 14556, mi = 0.003, pvalue = 0.010, MORF_CUL1\n",
      "Didn't pass (cutoff=0.0050): cat 11862, mi = 0.003, pvalue = 0.006, GSE42021_TREG_PLN_VS_TREG_PRECURSORS_THYMUS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 9898, mi = 0.003, pvalue = 0.008, GSE23984_CTRL_VS_HYPOCALEMIC_VITAMIND_ANALOG_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 8598, mi = 0.003, pvalue = 0.007, GSE16385_ROSIGLITAZONE_IL4_VS_IL4_ALONE_STIM_MACROPHAGE_12H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14493, mi = 0.003, pvalue = 0.011, MOHANKUMAR_TLX1_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2623, mi = 0.003, pvalue = 0.012, GO_CHAPERONE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 8417, mi = 0.003, pvalue = 0.009, GSE14908_ATOPIC_VS_NONATOPIC_PATIENT_HDM_STIM_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 10771, mi = 0.003, pvalue = 0.008, GSE32423_IL7_VS_IL7_IL4_MEMORY_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 6468, mi = 0.003, pvalue = 0.005, GO_REGULATION_OF_LIPID_STORAGE\n",
      "Didn't pass (cutoff=0.0050): cat 11561, mi = 0.003, pvalue = 0.009, GSE3982_NEUTROPHIL_VS_CENT_MEMORY_CD4_TCELL_DN\n",
      "Total number of categories evaluated = 320.\n",
      "Number of categories that passed the tests = 25\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 8, 23, 0.137107493713052\n",
      "best merge is 1, 6, 0.143614536856623\n",
      "best merge is 2, 18, 0.180379187124694\n",
      "best merge is 9, 19, 0.195693537800015\n",
      "best merge is 4, 7, 0.20839008141674\n",
      "best merge is 3, 16, 0.254974625432282\n",
      "best merge is 1, 24, 0.257710862778228\n",
      "best merge is 2, 8, 0.281177372196072\n",
      "best merge is 9, 17, 0.323519563460626\n",
      "best merge is 0, 2, 0.356944792912741\n",
      "best merge is 5, 20, 0.39250115124526\n",
      "best merge is 11, 13, 0.456925840890028\n",
      "best merge is 5, 12, 0.488792398600501\n",
      "best merge is 15, 21, 0.49712647058821\n",
      "best merge is 1, 10, 0.513397031164451\n",
      "best merge is 0, 4, 0.520305571093907\n",
      "best merge is 3, 15, 0.619231844731146\n",
      "best merge is 14, 22, 0.636224510589328\n",
      "best merge is 9, 11, 0.638032320739022\n",
      "best merge is 0, 1, 0.692284402571462\n",
      "best merge is 0, 9, 0.810718401412509\n",
      "order classes\n",
      "11 - 1.12188624198107\n",
      "4 - 2.52267849605084\n",
      "3 - 2.53719620082641\n",
      "5 - 2.05114097201817\n",
      "2 - 3.9764031911833\n",
      "0 2 18 8 23 4 7 1 6 24 10\n",
      "3 16 15 21\n",
      "5 20 12\n",
      "9 19 17 11 13\n",
      "14 22\n",
      "Done.Start drawing\n",
      "5.81\t-10.06\n",
      "GSE2706_UNSTIM_VS_2H_LPS_DC_UP\n",
      "GO_PROTEIN_ACETYLATION\n",
      "GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_DN\n",
      "GSE8678_IL7R_LOW_VS_HIGH_EFF_CD8_TCELL_UP\n",
      "GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "GSE45365_CTRL_VS_MCMV_INFECTION_NK_CELL_UP\n",
      "GSE42021_TCONV_PLN_VS_CD24HI_TCONV_THYMUS_UP\n",
      "LINDSTEDT_DENDRITIC_CELL_MATURATION_C\n",
      "GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "GSE7852_TREG_VS_TCONV_LN_DN\n",
      "GSE23321_CD8_STEM_CELL_MEMORY_VS_EFFECTOR_MEMORY_CD8_TCELL_DN\n",
      "GSE3039_ALPHABETA_CD8_TCELL_VS_B2_BCELL_UP\n",
      "DARWICHE_SKIN_TUMOR_PROMOTER_UP\n",
      "GSE29617_CTRL_VS_DAY3_TIV_FLU_VACCINE_PBMC_2008_DN\n",
      "GGGCATT_MIR365\n",
      "LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "GSE7218_IGM_VS_IGG_SIGNAL_THGOUGH_ANTIGEN_BCELL_UP\n",
      "GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "PID_FCER1_PATHWAY\n",
      "HOOI_ST7_TARGETS_DN\n",
      "GSE14308_TH17_VS_NATURAL_TREG_UP\n",
      "GO_SH3_SH2_ADAPTOR_ACTIVITY\n",
      "LIN_NPAS4_TARGETS_DN\n",
      "GSE7509_UNSTIM_VS_FCGRIIB_STIM_DC_UP\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 8, 23, 0.137107493713052\n",
      "best merge is 1, 6, 0.143614536856623\n",
      "best merge is 2, 18, 0.180379187124694\n",
      "best merge is 9, 19, 0.195693537800015\n",
      "best merge is 4, 7, 0.20839008141674\n",
      "best merge is 3, 16, 0.254974625432282\n",
      "best merge is 1, 24, 0.257710862778228\n",
      "best merge is 2, 8, 0.281177372196072\n",
      "best merge is 9, 17, 0.323519563460626\n",
      "best merge is 0, 2, 0.356944792912741\n",
      "best merge is 5, 20, 0.39250115124526\n",
      "best merge is 11, 13, 0.456925840890028\n",
      "best merge is 5, 12, 0.488792398600501\n",
      "best merge is 15, 21, 0.49712647058821\n",
      "best merge is 1, 10, 0.513397031164451\n",
      "best merge is 0, 4, 0.520305571093907\n",
      "best merge is 3, 15, 0.619231844731146\n",
      "best merge is 14, 22, 0.636224510589328\n",
      "best merge is 9, 11, 0.638032320739022\n",
      "best merge is 0, 1, 0.692284402571462\n",
      "best merge is 0, 9, 0.810718401412509\n",
      "order classes\n",
      "11 - 1.12188624198107\n",
      "4 - 2.52267849605084\n",
      "3 - 2.53719620082641\n",
      "5 - 2.05114097201817\n",
      "2 - 3.9764031911833\n",
      "0 2 18 8 23 4 7 1 6 24 10\n",
      "3 16 15 21\n",
      "5 20 12\n",
      "9 19 17 11 13\n",
      "14 22\n",
      "Done.-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_full/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 50\n",
      "cat 14, mi = 0.00289097, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 26, mi = 0.00239328, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "cat 31, mi = 0.00230077, HALLMARK_MYC_TARGETS_V1\n",
      "cat 45, mi = 0.00228384, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 15, mi = 0.00218120, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 44, mi = 0.00217278, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "cat 34, mi = 0.00191445, HALLMARK_NOTCH_SIGNALING\n",
      "cat 24, mi = 0.00190782, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "cat 48, mi = 0.00187866, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "cat 2, mi = 0.00185213, HALLMARK_ANDROGEN_RESPONSE\n",
      "cat 10, mi = 0.00180728, HALLMARK_COMPLEMENT\n",
      "cat 21, mi = 0.00180452, HALLMARK_HYPOXIA\n",
      "cat 16, mi = 0.00178090, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 40, mi = 0.00173480, HALLMARK_PROTEIN_SECRETION\n",
      "cat 25, mi = 0.00171044, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 14, mi = 0.003, pvalue = 0.005, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.002, pvalue = 0.016, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.035, HALLMARK_MYC_TARGETS_V1\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.002, pvalue = 0.033, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.002, pvalue = 0.093, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.002, pvalue = 0.071, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.002, pvalue = 0.113, HALLMARK_ANDROGEN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.002, pvalue = 0.090, HALLMARK_COMPLEMENT\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.141, HALLMARK_FATTY_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.002, pvalue = 0.167, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.002, pvalue = 0.125, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 18, mi = 0.002, pvalue = 0.134, HALLMARK_GLYCOLYSIS\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.002, pvalue = 0.202, HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.002, pvalue = 0.155, HALLMARK_KRAS_SIGNALING_UP\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.002, pvalue = 0.171, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.002, pvalue = 0.203, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 49, mi = 0.001, pvalue = 0.258, HALLMARK_XENOBIOTIC_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.001, pvalue = 0.210, HALLMARK_P53_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.001, pvalue = 0.296, HALLMARK_UV_RESPONSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.419, HALLMARK_E2F_TARGETS\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.001, pvalue = 0.635, HALLMARK_APICAL_SURFACE\n",
      "Total number of categories evaluated = 18.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.22306238185255\t42\t0\n",
      "4.25519848771266\t42\t23.5538752362949\n",
      "5.32703213610586\t42\t24.9565217391304\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_msigdb_h/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7491 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7491\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "681 hits go_terms_num = 790\n",
      "cat 449, mi = 0.00183844, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00180162, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 31, mi = 0.00170107, AUH_K562_POSTAR_eCLIP\n",
      "cat 16, mi = 0.00158901, ALKBH1_HeLa_POSTAR_CIMS\n",
      "cat 30, mi = 0.00155393, AUH_HepG2_POSTAR_eCLIP\n",
      "cat 484, mi = 0.00155161, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 237, mi = 0.00154847, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 637, mi = 0.00150643, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 635, mi = 0.00149676, TARDBP_H9_POSTAR_CIMS\n",
      "cat 236, mi = 0.00145359, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 32, mi = 0.00137177, BCCIP_HepG2_POSTAR_eCLIP\n",
      "cat 282, mi = 0.00129417, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "cat 275, mi = 0.00128334, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 672, mi = 0.00126306, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 230, mi = 0.00122052, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.061, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.074, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.109, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.155, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.002, pvalue = 0.168, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.111, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.169, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.166, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.001, pvalue = 0.170, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.001, pvalue = 0.233, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.175, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.167, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.001, pvalue = 0.304, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.300, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.001, pvalue = 0.266, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.001, pvalue = 0.279, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.001, pvalue = 0.346, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 447, mi = 0.001, pvalue = 0.301, NPM1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.334, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.001, pvalue = 0.364, HLTF_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_all_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6816 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6816\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "620 hits go_terms_num = 790\n",
      "cat 449, mi = 0.00319311, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00187035, AUH_K562_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00179222, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 16, mi = 0.00174765, ALKBH1_HeLa_POSTAR_CIMS\n",
      "cat 629, mi = 0.00173817, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "cat 30, mi = 0.00171060, AUH_HepG2_POSTAR_eCLIP\n",
      "cat 237, mi = 0.00170130, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 236, mi = 0.00166858, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 637, mi = 0.00163638, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 635, mi = 0.00162587, TARDBP_H9_POSTAR_CIMS\n",
      "cat 377, mi = 0.00158512, LARP7_K562_POSTAR_eCLIP\n",
      "cat 667, mi = 0.00145139, TNRC6A_HEK293_POSTAR_Piranha_0.01\n",
      "cat 282, mi = 0.00142182, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "cat 275, mi = 0.00141142, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 672, mi = 0.00138598, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 449, mi = 0.003, pvalue = 0.001, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.123, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.120, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.101, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.166, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.186, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.173, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.175, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.002, pvalue = 0.143, LARP7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 667, mi = 0.001, pvalue = 0.198, TNRC6A_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.172, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.001, pvalue = 0.300, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.300, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 749, mi = 0.001, pvalue = 0.278, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 329, mi = 0.001, pvalue = 0.259, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 447, mi = 0.001, pvalue = 0.302, NPM1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.001, pvalue = 0.352, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.339, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.365, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.001, pvalue = 0.366, HNRNPA1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 615, mi = 0.001, pvalue = 0.448, TAF15_Brain_POSTAR_CIMS\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "NSUN2_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.22306238185255\t42\t0\n",
      "4.26843100189036\t42\t23.5538752362949\n",
      "5.34026465028356\t42\t24.9829867674858\n",
      "5.93572778827977\t42\t25.6975425330813\n",
      "6.35916824196597\t42\t26.1739130434783\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.22495274102079\t42\t27.8676748582231\n",
      "15.2911153119093\t42\t28.5822306238185\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6678 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6678\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "607 hits go_terms_num = 790\n",
      "cat 250, mi = 0.00362468, FXR2_K562_POSTAR_eCLIP\n",
      "cat 242, mi = 0.00360265, FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 240, mi = 0.00322535, FXR1_HEK293_POSTAR_PAR-CLIP\n",
      "cat 241, mi = 0.00322535, FXR1_HEK293_POSTAR_PARalyzer\n",
      "cat 221, mi = 0.00319557, FTO_K562_POSTAR_eCLIP\n",
      "cat 774, mi = 0.00312360, YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "cat 784, mi = 0.00282949, ZNF622_K562_POSTAR_eCLIP\n",
      "cat 693, mi = 0.00261849, U2AF2_HeLa_POSTAR_Piranha_0.01\n",
      "cat 692, mi = 0.00259198, U2AF2_HeLa_POSTAR_HITS-CLIP\n",
      "cat 248, mi = 0.00251729, FXR2_HEK293_POSTAR_Piranha_0.01\n",
      "cat 350, mi = 0.00247560, IGF2BP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 641, mi = 0.00246224, TARDBP_K562_POSTAR_eCLIP\n",
      "cat 498, mi = 0.00243755, PUS1_K562_POSTAR_eCLIP\n",
      "cat 479, mi = 0.00237964, PTBP1_HEK293T,plus_SRRM4_POSTAR_Piranha_0.01\n",
      "cat 610, mi = 0.00236195, SUGP2_HepG2_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 250, mi = 0.004, pvalue = 0.000, FXR2_K562_POSTAR_eCLIP\n",
      "Passed: cat 774, mi = 0.003, pvalue = 0.002, YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.002, pvalue = 0.029, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.002, pvalue = 0.026, HNRNPA1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 575, mi = 0.002, pvalue = 0.083, SMNDC1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 691, mi = 0.002, pvalue = 0.098, U2AF2_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 544, mi = 0.002, pvalue = 0.107, RPS5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 513, mi = 0.002, pvalue = 0.136, RBM10_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 492, mi = 0.002, pvalue = 0.110, PUM2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 493, mi = 0.002, pvalue = 0.104, PUM2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 612, mi = 0.002, pvalue = 0.155, SUPV3L1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 468, mi = 0.002, pvalue = 0.099, PRKRA_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 407, mi = 0.002, pvalue = 0.188, MBNL2_hippocampus_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 406, mi = 0.002, pvalue = 0.187, MBNL2_hippocampus_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.002, pvalue = 0.207, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 327, mi = 0.002, pvalue = 0.183, HNRNPU_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 467, mi = 0.002, pvalue = 0.207, PRKRA_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 684, mi = 0.002, pvalue = 0.213, TROVE2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.002, pvalue = 0.275, DDX42_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 627, mi = 0.002, pvalue = 0.291, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 382, mi = 0.001, pvalue = 0.292, LIN28A_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.049, ALKBH5_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 92.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.26\t-10.06\n",
      "YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "FXR2_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.78655352480418\t42\t0\n",
      "5.15339425587468\t42\t24.4268929503916\n",
      "6.26370757180157\t42\t25.8798955613577\n",
      "6.89425587467363\t42\t26.6475195822454\n",
      "7.31919060052219\t42\t27.1409921671018\n",
      "7.64817232375979\t42\t27.4973890339426\n",
      "7.92232375979112\t42\t27.798955613577\n",
      "8.22389033942559\t42\t28.0456919060052\n",
      "8.63511749347259\t42\t28.402088772846\n",
      "9.22454308093995\t42\t28.8681462140992\n",
      "15.7904699738903\t42\t29.5809399477807\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6602 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6602\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "600 hits go_terms_num = 790\n",
      "cat 510, mi = 0.00339917, RBFOX2_K562_POSTAR_eCLIP\n",
      "cat 583, mi = 0.00247017, SRRM4_HEK293T_POSTAR_CIMS\n",
      "cat 0, mi = 0.00240671, AARS_K562_POSTAR_eCLIP\n",
      "cat 459, mi = 0.00238093, PCBP2_HepG2_POSTAR_eCLIP\n",
      "cat 508, mi = 0.00235042, RBFOX2_HepG2_POSTAR_eCLIP\n",
      "cat 221, mi = 0.00226683, FTO_K562_POSTAR_eCLIP\n",
      "cat 48, mi = 0.00224528, CELF2_T_cell_POSTAR_HITS-CLIP\n",
      "cat 313, mi = 0.00223929, HNRNPM_HEK293T_POSTAR_HITS-CLIP\n",
      "cat 314, mi = 0.00223929, HNRNPM_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 408, mi = 0.00216194, METAP2_K562_POSTAR_eCLIP\n",
      "cat 522, mi = 0.00209346, RBM15_HepG2_POSTAR_eCLIP\n",
      "cat 584, mi = 0.00208533, SRRM4_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 788, mi = 0.00206346, ZRANB2_K562_POSTAR_eCLIP\n",
      "cat 476, mi = 0.00204969, PTBP1_HEK293T,no_SRRM4_POSTAR_Piranha_0.01\n",
      "cat 365, mi = 0.00204851, IMP3_HEK293_Plass\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 510, mi = 0.003, pvalue = 0.001, RBFOX2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 221, mi = 0.002, pvalue = 0.035, FTO_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 408, mi = 0.002, pvalue = 0.050, METAP2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 584, mi = 0.002, pvalue = 0.045, SRRM4_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 365, mi = 0.002, pvalue = 0.055, IMP3_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 384, mi = 0.002, pvalue = 0.084, LIN28A_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 190, mi = 0.002, pvalue = 0.086, EWSR1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.002, pvalue = 0.085, FKBP4_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 398, mi = 0.002, pvalue = 0.102, LSM11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 385, mi = 0.002, pvalue = 0.096, LIN28A_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.002, pvalue = 0.095, EIF4G2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 58, mi = 0.002, pvalue = 0.009, CPSF-160_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 585, mi = 0.002, pvalue = 0.077, SRRM4_HEK293T_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.121, AGGF1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 485, mi = 0.002, pvalue = 0.111, PTBP1_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 346, mi = 0.002, pvalue = 0.128, IGF2BP2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 605, mi = 0.002, pvalue = 0.128, STAU1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 189, mi = 0.002, pvalue = 0.098, EWSR1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 338, mi = 0.002, pvalue = 0.070, IGF2BP1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 680, mi = 0.002, pvalue = 0.102, TRA2A_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.002, pvalue = 0.035, CDC40_HepG2_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 32.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "RBFOX2_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.22306238185255\t42\t0\n",
      "4.26843100189036\t42\t23.5538752362949\n",
      "5.35349716446125\t42\t24.9829867674858\n",
      "5.96219281663516\t42\t25.7240075614367\n",
      "6.37240075614367\t42\t26.2003780718336\n",
      "6.68998109640832\t42\t26.5444234404537\n",
      "6.95463137996219\t42\t26.8355387523629\n",
      "7.24574669187146\t42\t27.0737240075614\n",
      "7.64272211720227\t42\t27.4177693761815\n",
      "8.2117202268431\t42\t27.8676748582231\n",
      "15.2778827977316\t42\t28.5557655954631\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6691 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6691\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "608 hits go_terms_num = 790\n",
      "cat 118, mi = 0.00315007, DDX55_K562_POSTAR_eCLIP\n",
      "cat 669, mi = 0.00309290, TNRC6A_K562_POSTAR_eCLIP\n",
      "cat 221, mi = 0.00305514, FTO_K562_POSTAR_eCLIP\n",
      "cat 20, mi = 0.00302753, ALKBH5_HEK293_POSTAR_PARalyzer\n",
      "cat 19, mi = 0.00295378, ALKBH5_HEK293_POSTAR_PAR-CLIP\n",
      "cat 678, mi = 0.00293129, TRA2A_HepG2_POSTAR_eCLIP\n",
      "cat 122, mi = 0.00285427, DDX6_HepG2_POSTAR_eCLIP\n",
      "cat 680, mi = 0.00281228, TRA2A_K562_POSTAR_eCLIP\n",
      "cat 39, mi = 0.00276562, C17orf85_HEK293_Plass\n",
      "cat 248, mi = 0.00273975, FXR2_HEK293_POSTAR_Piranha_0.01\n",
      "cat 185, mi = 0.00273543, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "cat 43, mi = 0.00272157, CAPRIN1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 206, mi = 0.00269692, FBL_HEK293_POSTAR_PARalyzer\n",
      "cat 205, mi = 0.00261761, FBL_HEK293_POSTAR_PAR-CLIP\n",
      "cat 439, mi = 0.00251228, NOP56_HEK293_POSTAR_PAR-CLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 118, mi = 0.003, pvalue = 0.002, DDX55_K562_POSTAR_eCLIP\n",
      "Passed: cat 669, mi = 0.003, pvalue = 0.003, TNRC6A_K562_POSTAR_eCLIP\n",
      "Passed: cat 39, mi = 0.003, pvalue = 0.003, C17orf85_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 439, mi = 0.003, pvalue = 0.019, NOP56_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 440, mi = 0.003, pvalue = 0.018, NOP56_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 22, mi = 0.003, pvalue = 0.015, ALKBH5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 319, mi = 0.002, pvalue = 0.021, HNRNPUL1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 301, mi = 0.002, pvalue = 0.036, HNRNPF_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 302, mi = 0.002, pvalue = 0.034, HNRNPF_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.002, pvalue = 0.044, ATXN2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 382, mi = 0.002, pvalue = 0.049, LIN28A_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.022, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.002, pvalue = 0.051, SFPQ_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 405, mi = 0.002, pvalue = 0.009, MBNL2_hippocampus_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 361, mi = 0.002, pvalue = 0.059, ILF3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 640, mi = 0.002, pvalue = 0.061, TARDBP_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 207, mi = 0.002, pvalue = 0.059, FBL_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 621, mi = 0.002, pvalue = 0.077, TAF15_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 641, mi = 0.002, pvalue = 0.070, TARDBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 593, mi = 0.002, pvalue = 0.099, SRSF7_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 225, mi = 0.002, pvalue = 0.072, FUS_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.037, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.002, pvalue = 0.141, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 70.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.366361275652177\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "5.26\t-10.06\n",
      "TNRC6A_K562_POSTAR_eCLIP\n",
      "C17orf85_HEK293_Plass\n",
      "DDX55_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.366361275652177\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-7.78655352480418\t42\t0\n",
      "5.15339425587468\t42\t24.4268929503916\n",
      "6.25\t42\t25.8798955613577\n",
      "6.8668407310705\t42\t26.6201044386423\n",
      "7.30548302872063\t42\t27.1135770234987\n",
      "7.64817232375979\t42\t27.4973890339426\n",
      "7.92232375979112\t42\t27.798955613577\n",
      "8.22389033942559\t42\t28.0456919060052\n",
      "8.63511749347259\t42\t28.402088772846\n",
      "9.22454308093995\t42\t28.8681462140992\n",
      "15.7904699738903\t42\t29.5809399477807\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 790\n",
      "cat 629, mi = 0.00265758, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "cat 126, mi = 0.00260023, DGCR8_HEK293T_POSTAR_CIMS\n",
      "cat 456, mi = 0.00238196, PAPD5_HEK293_Plass\n",
      "cat 449, mi = 0.00234709, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00220459, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 377, mi = 0.00203127, LARP7_K562_POSTAR_eCLIP\n",
      "cat 236, mi = 0.00197494, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 30, mi = 0.00196888, AUH_HepG2_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00196598, AUH_K562_POSTAR_eCLIP\n",
      "cat 428, mi = 0.00189760, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 237, mi = 0.00187280, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 22, mi = 0.00179580, ALKBH5_HEK293_Plass\n",
      "cat 256, mi = 0.00174680, GNL3_K562_POSTAR_eCLIP\n",
      "cat 671, mi = 0.00169410, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "cat 329, mi = 0.00165302, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.003, pvalue = 0.013, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.003, pvalue = 0.012, DGCR8_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 456, mi = 0.002, pvalue = 0.034, PAPD5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.038, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.053, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.002, pvalue = 0.065, LARP7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.100, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.002, pvalue = 0.068, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.106, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.002, pvalue = 0.084, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.127, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 22, mi = 0.002, pvalue = 0.112, ALKBH5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.002, pvalue = 0.120, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 671, mi = 0.002, pvalue = 0.138, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 329, mi = 0.002, pvalue = 0.158, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.235, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 667, mi = 0.002, pvalue = 0.172, TNRC6A_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 666, mi = 0.002, pvalue = 0.200, TNRC6A_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 675, mi = 0.002, pvalue = 0.198, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.002, pvalue = 0.226, C17orf85_HEK293_Plass\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "read_go_names_data: please enter a valid filename (/flash/hani/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_coding_gene_names/human_ensembl_RBPs_coding_gene_names_names.txt invalid)\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 101\n",
      "cat 9, mi = 0.00160364, ESRP2\n",
      "cat 56, mi = 0.00124640, RBM3\n",
      "cat 13, mi = 0.00120929, FXR2\n",
      "cat 27, mi = 0.00113485, HuR.2\n",
      "cat 62, mi = 0.00104475, RBM46\n",
      "cat 6, mi = 0.00091752, CPEB4\n",
      "cat 88, mi = 0.00091132, SRSF9.1\n",
      "cat 34, mi = 0.00071213, KHDRBS3\n",
      "cat 21, mi = 0.00070426, HNRNPK\n",
      "cat 69, mi = 0.00065348, RBMS3.1\n",
      "cat 68, mi = 0.00048804, RBMS3\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "cat 3, mi = 0.00000000, BRUNOL6\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.284, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.416, RBM3\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.420, FXR2\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.527, HuR.2\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.667, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.675, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.001, pvalue = 0.666, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.919, KHDRBS3\n",
      "Didn't pass: cat 21, mi = 0.001, pvalue = 1.000, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.853, RBMS3.1\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.000, pvalue = 0.942, RBMS3\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, FMR1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.81\t-10.06\n",
      "Outputing EPS file deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/METTL14_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL14_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL14_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL14_peak_m6A_vs_input_human_ensembl_RBPs_DeepBind/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7509 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7509\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "683 hits go_terms_num = 17689\n",
      "cat 2537, mi = 0.00346310, peptidyl-serine phosphorylation\n",
      "cat 916, mi = 0.00307643, meiotic cell cycle\n",
      "cat 2106, mi = 0.00282469, endosome organization\n",
      "cat 10057, mi = 0.00282232, lymphocyte homeostasis\n",
      "cat 2360, mi = 0.00266963, regulation of mitotic nuclear division\n",
      "cat 6756, mi = 0.00264799, execution phase of apoptosis\n",
      "cat 4439, mi = 0.00263452, face morphogenesis\n",
      "cat 2435, mi = 0.00260250, histone acetylation\n",
      "cat 378, mi = 0.00259360, negative regulation of BMP signaling pathway\n",
      "cat 3739, mi = 0.00252712, calcium channel activity\n",
      "cat 101, mi = 0.00249475, negative regulation of translation\n",
      "cat 3210, mi = 0.00249243, homeostasis of number of cells within a tissue\n",
      "cat 5105, mi = 0.00246928, cytoskeletal protein binding\n",
      "cat 1198, mi = 0.00246886, beta-catenin-TCF complex assembly\n",
      "cat 3728, mi = 0.00245070, embryonic camera-type eye morphogenesis\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2537, mi = 0.003, pvalue = 0.000, peptidyl-serine phosphorylation\n",
      "Passed: cat 916, mi = 0.003, pvalue = 0.000, meiotic cell cycle\n",
      "Passed: cat 2106, mi = 0.003, pvalue = 0.001, endosome organization\n",
      "Passed: cat 6756, mi = 0.003, pvalue = 0.003, execution phase of apoptosis\n",
      "Passed: cat 4439, mi = 0.003, pvalue = 0.004, face morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 2435, mi = 0.003, pvalue = 0.005, histone acetylation\n",
      "Didn't pass (cutoff=0.0050): cat 5105, mi = 0.002, pvalue = 0.008, cytoskeletal protein binding\n",
      "Didn't pass (cutoff=0.0050): cat 5947, mi = 0.002, pvalue = 0.009, exonuclease activity\n",
      "Didn't pass (cutoff=0.0050): cat 2898, mi = 0.002, pvalue = 0.009, telomeric DNA binding\n",
      "Passed: cat 4034, mi = 0.002, pvalue = 0.003, cholesterol biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2671, mi = 0.002, pvalue = 0.010, MyD88-dependent toll-like receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 408, mi = 0.002, pvalue = 0.009, protein glycosylation\n",
      "Didn't pass (cutoff=0.0050): cat 1705, mi = 0.002, pvalue = 0.005, toxin transport\n",
      "Passed: cat 2305, mi = 0.002, pvalue = 0.001, histone mRNA catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 2101, mi = 0.002, pvalue = 0.014, centriole\n",
      "Didn't pass (cutoff=0.0050): cat 328, mi = 0.002, pvalue = 0.013, cell redox homeostasis\n",
      "Didn't pass (cutoff=0.0050): cat 4696, mi = 0.002, pvalue = 0.008, osteoblast development\n",
      "Didn't pass (cutoff=0.0050): cat 2443, mi = 0.002, pvalue = 0.017, histone methyltransferase complex\n",
      "Didn't pass (cutoff=0.0050): cat 5066, mi = 0.002, pvalue = 0.007, mRNA 5'-UTR binding\n",
      "Didn't pass (cutoff=0.0050): cat 8602, mi = 0.002, pvalue = 0.017, oligosaccharide metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 4230, mi = 0.002, pvalue = 0.018, negative regulation of ERK1 and ERK2 cascade\n",
      "Didn't pass (cutoff=0.0050): cat 1280, mi = 0.002, pvalue = 0.020, protein K48-linked ubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 3608, mi = 0.002, pvalue = 0.020, polysome\n",
      "Didn't pass (cutoff=0.0050): cat 2611, mi = 0.002, pvalue = 0.012, positive regulation of axon extension\n",
      "Didn't pass (cutoff=0.0050): cat 1329, mi = 0.002, pvalue = 0.026, regulation of inflammatory response\n",
      "Didn't pass (cutoff=0.0050): cat 1914, mi = 0.002, pvalue = 0.029, regulation of protein kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 3834, mi = 0.002, pvalue = 0.019, regulation of signal transduction\n",
      "Passed: cat 14151, mi = 0.002, pvalue = 0.003, nuclear membrane organization\n",
      "Didn't pass (cutoff=0.0050): cat 2145, mi = 0.002, pvalue = 0.037, histone H3-K4 trimethylation\n",
      "Didn't pass (cutoff=0.0050): cat 1382, mi = 0.002, pvalue = 0.023, positive regulation of cell adhesion\n",
      "Didn't pass (cutoff=0.0050): cat 7300, mi = 0.002, pvalue = 0.022, SWI/SNF complex\n",
      "Didn't pass (cutoff=0.0050): cat 7302, mi = 0.002, pvalue = 0.024, nBAF complex\n",
      "Didn't pass (cutoff=0.0050): cat 6456, mi = 0.002, pvalue = 0.009, DNA packaging\n",
      "Didn't pass (cutoff=0.0050): cat 1132, mi = 0.002, pvalue = 0.030, substantia nigra development\n",
      "Didn't pass (cutoff=0.0050): cat 2308, mi = 0.002, pvalue = 0.021, mRNA cleavage\n",
      "Didn't pass (cutoff=0.0050): cat 3903, mi = 0.002, pvalue = 0.036, histone deacetylation\n",
      "Didn't pass (cutoff=0.0050): cat 2239, mi = 0.002, pvalue = 0.013, cellular response to cAMP\n",
      "Didn't pass (cutoff=0.0050): cat 2482, mi = 0.002, pvalue = 0.044, protein N-linked glycosylation via asparagine\n",
      "Didn't pass (cutoff=0.0050): cat 1049, mi = 0.002, pvalue = 0.036, cell adhesion molecule binding\n",
      "Didn't pass (cutoff=0.0050): cat 11275, mi = 0.002, pvalue = 0.030, insulin-like growth factor receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 3242, mi = 0.002, pvalue = 0.042, transcriptional activator activity, RNA polymerase II distal enhancer sequence-specific binding\n",
      "Didn't pass (cutoff=0.0050): cat 1895, mi = 0.002, pvalue = 0.026, epithelial cell differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 1014, mi = 0.002, pvalue = 0.025, positive regulation of reactive oxygen species metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 721, mi = 0.002, pvalue = 0.037, MOZ/MORF histone acetyltransferase complex\n",
      "Didn't pass (cutoff=0.0050): cat 1989, mi = 0.002, pvalue = 0.017, MHC class I protein binding\n",
      "Didn't pass (cutoff=0.0050): cat 2330, mi = 0.002, pvalue = 0.005, positive regulation of vascular smooth muscle cell proliferation\n",
      "Didn't pass (cutoff=0.0050): cat 3198, mi = 0.002, pvalue = 0.006, regulation of store-operated calcium entry\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.002, pvalue = 0.054, negative regulation of cell death\n",
      "Total number of categories evaluated = 80.\n",
      "Number of categories that passed the tests = 8\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 6, 7, 0.1512291614607\n",
      "best merge is 2, 3, 0.271108152913565\n",
      "best merge is 2, 4, 0.438485116851127\n",
      "best merge is 0, 6, 0.652779736074506\n",
      "order classes\n",
      "3 - 0.680342747624189\n",
      "2 - 0.3024583229214\n",
      "6 7\n",
      "1\n",
      "2 3 4\n",
      "5\n",
      "0\n",
      "Done.Start drawing\n",
      "8.47\t-11.35\n",
      "endosome organization, GO:0007032\n",
      "execution phase of apoptosis, GO:0097194\n",
      "face morphogenesis, GO:0060325\n",
      "histone mRNA catabolic process, GO:0071044\n",
      "nuclear membrane organization, GO:0071763\n",
      "peptidyl-serine phosphorylation, GO:0018105\n",
      "meiotic cell cycle, GO:0051321\n",
      "cholesterol biosynthetic process, GO:0006695\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 6, 7, 0.1512291614607\n",
      "best merge is 2, 3, 0.271108152913565\n",
      "best merge is 2, 4, 0.438485116851127\n",
      "best merge is 0, 6, 0.652779736074506\n",
      "order classes\n",
      "3 - 0.680342747624189\n",
      "2 - 0.3024583229214\n",
      "6 7\n",
      "1\n",
      "2 3 4\n",
      "5\n",
      "0\n",
      "Done.-9.73309788092836\t42\t0\n",
      "1.28607467204844\t42\t20.5338042381433\n",
      "2.43037336024218\t42\t22.0383451059536\n",
      "3.11907164480323\t42\t22.8224016145308\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.86326942482341\t42\t25.5136226034309\n",
      "14.106458123108\t42\t26.2129162462159\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6496 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6496\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "591 hits go_terms_num = 181\n",
      "cat 131, mi = 0.00209761, SMARCA4_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00132653, POLR3G_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00131847, BRF2_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00075718, XRCC4_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00071002, PPARGC1A_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00063436, BRF1_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00062640, BDP1_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00055764, POU5F1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.002, pvalue = 0.076, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.477, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.561, BRF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 167, mi = 0.001, pvalue = 1.000, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.962, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.923, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.886, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.001, pvalue = 0.902, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.83\t-11.35\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_encode_tf/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 326\n",
      "cat 279, mi = 0.00233795, chr7q36\n",
      "cat 38, mi = 0.00228546, chr12q15\n",
      "cat 182, mi = 0.00224649, chr3p22\n",
      "cat 4, mi = 0.00215361, chr10p15\n",
      "cat 269, mi = 0.00215281, chr7p22\n",
      "cat 274, mi = 0.00214942, chr7q31\n",
      "cat 190, mi = 0.00210894, chr3q13\n",
      "cat 315, mi = 0.00205473, chrxq13\n",
      "cat 26, mi = 0.00204122, chr11q23\n",
      "cat 139, mi = 0.00196652, chr20q13\n",
      "cat 125, mi = 0.00194804, chr1q23\n",
      "cat 164, mi = 0.00192610, chr2q14\n",
      "cat 223, mi = 0.00192610, chr5p12\n",
      "cat 305, mi = 0.00191233, chr9q31\n",
      "cat 118, mi = 0.00186995, chr1p34\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 279, mi = 0.002, pvalue = 0.026, chr7q36\n",
      "Passed: cat 38, mi = 0.002, pvalue = 0.004, chr12q15\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.002, pvalue = 0.038, chr3p22\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.002, pvalue = 0.044, chr10p15\n",
      "Didn't pass (cutoff=0.0050): cat 269, mi = 0.002, pvalue = 0.043, chr7p22\n",
      "Didn't pass (cutoff=0.0050): cat 274, mi = 0.002, pvalue = 0.038, chr7q31\n",
      "Didn't pass (cutoff=0.0050): cat 190, mi = 0.002, pvalue = 0.054, chr3q13\n",
      "Didn't pass (cutoff=0.0050): cat 315, mi = 0.002, pvalue = 0.066, chrxq13\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.002, pvalue = 0.068, chr11q23\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.002, pvalue = 0.058, chr20q13\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.002, pvalue = 0.095, chr1q23\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.002, pvalue = 0.063, chr2q14\n",
      "Didn't pass (cutoff=0.0050): cat 223, mi = 0.002, pvalue = 0.065, chr5p12\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.002, pvalue = 0.111, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.002, pvalue = 0.091, chr1p34\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.002, pvalue = 0.117, chr14q11\n",
      "Didn't pass (cutoff=0.0050): cat 138, mi = 0.002, pvalue = 0.089, chr20q12\n",
      "Didn't pass (cutoff=0.0050): cat 180, mi = 0.002, pvalue = 0.089, chr3p14\n",
      "Didn't pass (cutoff=0.0050): cat 307, mi = 0.002, pvalue = 0.147, chr9q33\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.002, pvalue = 0.153, chr13q12\n",
      "Didn't pass (cutoff=0.0050): cat 241, mi = 0.002, pvalue = 0.031, chr6p\n",
      "Didn't pass (cutoff=0.0050): cat 86, mi = 0.002, pvalue = 0.038, chr17p12\n",
      "Total number of categories evaluated = 12.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.47\t-11.35\n",
      "chr12q15\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_c1/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4738\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t4738\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 4738\n",
      "cat 4614, mi = 0.00383646, YU_MYC_TARGETS_DN\n",
      "cat 3069, mi = 0.00356559, REACTOME_APOPTOTIC_CLEAVAGE_OF_CELLULAR_PROTEINS\n",
      "cat 4449, mi = 0.00353684, WHITEFORD_PEDIATRIC_CANCER_MARKERS\n",
      "cat 2122, mi = 0.00352168, LY_AGING_OLD_DN\n",
      "cat 1242, mi = 0.00345577, GU_PDEF_TARGETS_UP\n",
      "cat 270, mi = 0.00342650, BIOCARTA_ETS_PATHWAY\n",
      "cat 3753, mi = 0.00321907, ROSTY_CERVICAL_CANCER_PROLIFERATION_CLUSTER\n",
      "cat 2246, mi = 0.00317539, MCBRYAN_PUBERTAL_BREAST_5_6WK_DN\n",
      "cat 2390, mi = 0.00313364, MORI_LARGE_PRE_BII_LYMPHOCYTE_DN\n",
      "cat 3196, mi = 0.00312131, REACTOME_FACTORS_INVOLVED_IN_MEGAKARYOCYTE_DEVELOPMENT_AND_PLATELET_PRODUCTION\n",
      "cat 3826, mi = 0.00312080, SA_B_CELL_RECEPTOR_COMPLEXES\n",
      "cat 2393, mi = 0.00312040, MORI_MATURE_B_LYMPHOCYTE_UP\n",
      "cat 4594, mi = 0.00308799, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_9\n",
      "cat 3228, mi = 0.00308684, REACTOME_GASTRIN_CREB_SIGNALLING_PATHWAY_VIA_PKC_AND_MAPK\n",
      "cat 730, mi = 0.00307020, CONCANNON_APOPTOSIS_BY_EPOXOMICIN_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 4614, mi = 0.004, pvalue = 0.000, YU_MYC_TARGETS_DN\n",
      "Passed: cat 3069, mi = 0.004, pvalue = 0.001, REACTOME_APOPTOTIC_CLEAVAGE_OF_CELLULAR_PROTEINS\n",
      "Passed: cat 4449, mi = 0.004, pvalue = 0.001, WHITEFORD_PEDIATRIC_CANCER_MARKERS\n",
      "Passed: cat 1242, mi = 0.003, pvalue = 0.000, GU_PDEF_TARGETS_UP\n",
      "Passed: cat 270, mi = 0.003, pvalue = 0.000, BIOCARTA_ETS_PATHWAY\n",
      "Passed: cat 2246, mi = 0.003, pvalue = 0.001, MCBRYAN_PUBERTAL_BREAST_5_6WK_DN\n",
      "Passed: cat 3196, mi = 0.003, pvalue = 0.002, REACTOME_FACTORS_INVOLVED_IN_MEGAKARYOCYTE_DEVELOPMENT_AND_PLATELET_PRODUCTION\n",
      "Passed: cat 4594, mi = 0.003, pvalue = 0.001, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_9\n",
      "Passed: cat 3228, mi = 0.003, pvalue = 0.002, REACTOME_GASTRIN_CREB_SIGNALLING_PATHWAY_VIA_PKC_AND_MAPK\n",
      "Passed: cat 74, mi = 0.003, pvalue = 0.003, AMUNDSON_POOR_SURVIVAL_AFTER_GAMMA_RADIATION_2G\n",
      "Passed: cat 515, mi = 0.003, pvalue = 0.001, BRUECKNER_TARGETS_OF_MIRLET7A3_DN\n",
      "Didn't pass (cutoff=0.0050): cat 795, mi = 0.003, pvalue = 0.005, DARWICHE_PAPILLOMA_RISK_LOW_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4234, mi = 0.003, pvalue = 0.006, TURASHVILI_BREAST_DUCTAL_CARCINOMA_VS_LOBULAR_NORMAL_UP\n",
      "Passed: cat 3872, mi = 0.003, pvalue = 0.001, SCHMIDT_POR_TARGETS_IN_LIMB_BUD_UP\n",
      "Passed: cat 44, mi = 0.003, pvalue = 0.004, ALONSO_METASTASIS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 927, mi = 0.003, pvalue = 0.008, ENGELMANN_CANCER_PROGENITORS_UP\n",
      "Passed: cat 2467, mi = 0.003, pvalue = 0.003, NADLER_HYPERGLYCEMIA_AT_OBESITY\n",
      "Didn't pass (cutoff=0.0050): cat 4689, mi = 0.003, pvalue = 0.007, ZHENG_FOXP3_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3286, mi = 0.003, pvalue = 0.007, REACTOME_INFLAMMASOMES\n",
      "Didn't pass (cutoff=0.0050): cat 4660, mi = 0.003, pvalue = 0.008, ZHAN_MULTIPLE_MYELOMA_CD1_VS_CD2_DN\n",
      "Passed: cat 294, mi = 0.003, pvalue = 0.002, BIOCARTA_HSP27_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2064, mi = 0.003, pvalue = 0.010, LIU_VMYB_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 974, mi = 0.003, pvalue = 0.006, FIGUEROA_AML_METHYLATION_CLUSTER_5_DN\n",
      "Passed: cat 1893, mi = 0.003, pvalue = 0.001, KYNG_ENVIRONMENTAL_STRESS_RESPONSE_NOT_BY_UV_IN_OLD\n",
      "Didn't pass (cutoff=0.0050): cat 2153, mi = 0.003, pvalue = 0.012, MANTOVANI_VIRAL_GPCR_SIGNALING_DN\n",
      "Passed: cat 1153, mi = 0.003, pvalue = 0.003, GNATENKO_PLATELET_SIGNATURE\n",
      "Didn't pass (cutoff=0.0050): cat 3643, mi = 0.003, pvalue = 0.009, REACTOME_TRAF6_MEDIATED_INDUCTION_OF_TAK1_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 2732, mi = 0.003, pvalue = 0.016, PID_ARF6_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2749, mi = 0.003, pvalue = 0.016, PID_BMP_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2535, mi = 0.002, pvalue = 0.018, NIKOLSKY_BREAST_CANCER_12Q13_Q21_AMPLICON\n",
      "Didn't pass (cutoff=0.0050): cat 2652, mi = 0.002, pvalue = 0.017, PAPASPYRIDONOS_UNSTABLE_ATEROSCLEROTIC_PLAQUE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 871, mi = 0.002, pvalue = 0.007, DOANE_BREAST_CANCER_CLASSES_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1438, mi = 0.002, pvalue = 0.025, IKEDA_MIR133_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1161, mi = 0.002, pvalue = 0.031, GOLDRATH_IMMUNE_MEMORY\n",
      "Didn't pass (cutoff=0.0050): cat 3276, mi = 0.002, pvalue = 0.014, REACTOME_IKK_COMPLEX_RECRUITMENT_MEDIATED_BY_RIP1\n",
      "Didn't pass (cutoff=0.0050): cat 3557, mi = 0.002, pvalue = 0.011, REACTOME_SIGNALING_BY_BMP\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.002, pvalue = 0.027, CHEOK_RESPONSE_TO_HD_MTX_DN\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.002, pvalue = 0.029, BASSO_B_LYMPHOCYTE_NETWORK\n",
      "Didn't pass (cutoff=0.0050): cat 466, mi = 0.002, pvalue = 0.032, BOYLAN_MULTIPLE_MYELOMA_D_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4363, mi = 0.002, pvalue = 0.032, WANG_ESOPHAGUS_CANCER_VS_NORMAL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4597, mi = 0.002, pvalue = 0.026, YEGNASUBRAMANIAN_PROSTATE_CANCER\n",
      "Didn't pass (cutoff=0.0050): cat 719, mi = 0.002, pvalue = 0.039, COATES_MACROPHAGE_M1_VS_M2_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2534, mi = 0.002, pvalue = 0.028, NIKOLSKY_BREAST_CANCER_11Q12_Q14_AMPLICON\n",
      "Didn't pass (cutoff=0.0050): cat 576, mi = 0.002, pvalue = 0.037, CAFFAREL_RESPONSE_TO_THC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1505, mi = 0.002, pvalue = 0.014, JIANG_HYPOXIA_VIA_VHL\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.002, pvalue = 0.042, ACOSTA_PROLIFERATION_INDEPENDENT_MYC_TARGETS_UP\n",
      "Total number of categories evaluated = 158.\n",
      "Number of categories that passed the tests = 17\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 15, 0.199815488432056\n",
      "best merge is 7, 16, 0.218480137578353\n",
      "best merge is 1, 12, 0.308558553416493\n",
      "best merge is 5, 6, 0.354698306943488\n",
      "best merge is 4, 14, 0.404484345536794\n",
      "best merge is 3, 10, 0.427866919990935\n",
      "best merge is 0, 9, 0.530506079842702\n",
      "best merge is 2, 11, 0.583192474672221\n",
      "best merge is 2, 13, 0.658671560560832\n",
      "best merge is 5, 8, 0.664209184039648\n",
      "best merge is 4, 5, 0.707352618474207\n",
      "best merge is 1, 3, 0.785520110234079\n",
      "best merge is 2, 7, 0.87609161040861\n",
      "order classes\n",
      "2 - 2.25465083933148\n",
      "4 - 2.06045626699511\n",
      "3 - 2.39326704655526\n",
      "6 - 1.4258080106809\n",
      "2 - 0.928540584708002\n",
      "7 16\n",
      "1 12 3 10\n",
      "2 11 13\n",
      "4 15 14 5 6 8\n",
      "0 9\n",
      "Done.Start drawing\n",
      "8.47\t-11.35\n",
      "REACTOME_APOPTOTIC_CLEAVAGE_OF_CELLULAR_PROTEINS\n",
      "ALONSO_METASTASIS_DN\n",
      "MCBRYAN_PUBERTAL_BREAST_5_6WK_DN\n",
      "BIOCARTA_ETS_PATHWAY\n",
      "KYNG_ENVIRONMENTAL_STRESS_RESPONSE_NOT_BY_UV_IN_OLD\n",
      "BIOCARTA_HSP27_PATHWAY\n",
      "GU_PDEF_TARGETS_UP\n",
      "BRUECKNER_TARGETS_OF_MIRLET7A3_DN\n",
      "REACTOME_FACTORS_INVOLVED_IN_MEGAKARYOCYTE_DEVELOPMENT_AND_PLATELET_PRODUCTION\n",
      "REACTOME_GASTRIN_CREB_SIGNALLING_PATHWAY_VIA_PKC_AND_MAPK\n",
      "YU_MYC_TARGETS_DN\n",
      "YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_9\n",
      "GNATENKO_PLATELET_SIGNATURE\n",
      "NADLER_HYPERGLYCEMIA_AT_OBESITY\n",
      "WHITEFORD_PEDIATRIC_CANCER_MARKERS\n",
      "SCHMIDT_POR_TARGETS_IN_LIMB_BUD_UP\n",
      "AMUNDSON_POOR_SURVIVAL_AFTER_GAMMA_RADIATION_2G\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 15, 0.199815488432056\n",
      "best merge is 7, 16, 0.218480137578353\n",
      "best merge is 1, 12, 0.308558553416493\n",
      "best merge is 5, 6, 0.354698306943488\n",
      "best merge is 4, 14, 0.404484345536794\n",
      "best merge is 3, 10, 0.427866919990935\n",
      "best merge is 0, 9, 0.530506079842702\n",
      "best merge is 2, 11, 0.583192474672221\n",
      "best merge is 2, 13, 0.658671560560832\n",
      "best merge is 5, 8, 0.664209184039648\n",
      "best merge is 4, 5, 0.707352618474207\n",
      "best merge is 1, 3, 0.785520110234079\n",
      "best merge is 2, 7, 0.87609161040861\n",
      "order classes\n",
      "2 - 2.25465083933148\n",
      "4 - 2.06045626699511\n",
      "3 - 2.39326704655526\n",
      "6 - 1.4258080106809\n",
      "2 - 0.928540584708002\n",
      "7 16\n",
      "1 12 3 10\n",
      "2 11 13\n",
      "4 15 14 5 6 8\n",
      "0 9\n",
      "Done.-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_c2/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 836\n",
      "cat 177, mi = 0.00307915, CCAWYNNGAAR_UNKNOWN\n",
      "cat 723, mi = 0.00302089, TCTAGAG_MIR517\n",
      "cat 309, mi = 0.00273040, EVI1_03\n",
      "cat 298, mi = 0.00271869, EN1_01\n",
      "cat 399, mi = 0.00257583, GGTGAAG_MIR412\n",
      "cat 39, mi = 0.00255928, ACTACCT_MIR196A_MIR196B\n",
      "cat 174, mi = 0.00253468, CCATCCA_MIR432\n",
      "cat 395, mi = 0.00252176, GGGNRMNNYCAT_UNKNOWN\n",
      "cat 111, mi = 0.00248099, ATACCTC_MIR202\n",
      "cat 57, mi = 0.00247742, AGGCACT_MIR5153P\n",
      "cat 724, mi = 0.00246472, TCTATGA_MIR376A_MIR376B\n",
      "cat 443, mi = 0.00241150, HFH3_01\n",
      "cat 373, mi = 0.00237393, GGAMTNNNNNTCCY_UNKNOWN\n",
      "cat 87, mi = 0.00235190, AP2REP_01\n",
      "cat 415, mi = 0.00234173, GTCAGGA_MIR378\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 177, mi = 0.003, pvalue = 0.003, CCAWYNNGAAR_UNKNOWN\n",
      "Passed: cat 723, mi = 0.003, pvalue = 0.002, TCTAGAG_MIR517\n",
      "Didn't pass (cutoff=0.0050): cat 309, mi = 0.003, pvalue = 0.010, EVI1_03\n",
      "Didn't pass (cutoff=0.0050): cat 298, mi = 0.003, pvalue = 0.007, EN1_01\n",
      "Didn't pass (cutoff=0.0050): cat 399, mi = 0.003, pvalue = 0.014, GGTGAAG_MIR412\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.003, pvalue = 0.013, ACTACCT_MIR196A_MIR196B\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.002, pvalue = 0.012, ATACCTC_MIR202\n",
      "Didn't pass (cutoff=0.0050): cat 724, mi = 0.002, pvalue = 0.020, TCTATGA_MIR376A_MIR376B\n",
      "Didn't pass (cutoff=0.0050): cat 443, mi = 0.002, pvalue = 0.019, HFH3_01\n",
      "Didn't pass (cutoff=0.0050): cat 577, mi = 0.002, pvalue = 0.036, PAX2_01\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.002, pvalue = 0.028, AACWWCAANK_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.002, pvalue = 0.034, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 470, mi = 0.002, pvalue = 0.024, HSF_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 431, mi = 0.002, pvalue = 0.030, GTGTTGA_MIR505\n",
      "Didn't pass (cutoff=0.0050): cat 65, mi = 0.002, pvalue = 0.047, AGTTCTC_MIR146A_MIR146B\n",
      "Didn't pass (cutoff=0.0050): cat 513, mi = 0.002, pvalue = 0.035, MSX1_01\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.002, pvalue = 0.054, ACCAATC_MIR509\n",
      "Didn't pass (cutoff=0.0050): cat 422, mi = 0.002, pvalue = 0.039, GTGCAAA_MIR507\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.002, pvalue = 0.048, CATRRAGC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 436, mi = 0.002, pvalue = 0.057, GTTRYCATRR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 761, mi = 0.002, pvalue = 0.049, TGTGTGA_MIR377\n",
      "Didn't pass (cutoff=0.0050): cat 107, mi = 0.002, pvalue = 0.058, AR_Q2\n",
      "Total number of categories evaluated = 20.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.47\t-11.35\n",
      "TCTAGAG_MIR517\n",
      "CCAWYNNGAAR_UNKNOWN\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_c3/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 858\n",
      "cat 121, mi = 0.00410476, GNF2_CENPF\n",
      "cat 113, mi = 0.00405147, GNF2_CDC2\n",
      "cat 184, mi = 0.00364946, GNF2_PCNA\n",
      "cat 201, mi = 0.00352119, GNF2_RRM1\n",
      "cat 104, mi = 0.00333582, GNF2_CCNA2\n",
      "cat 92, mi = 0.00333204, GNF2_ATM\n",
      "cat 724, mi = 0.00316250, MORF_ETV3\n",
      "cat 114, mi = 0.00315603, GNF2_CDC20\n",
      "cat 304, mi = 0.00314608, MODULE_168\n",
      "cat 190, mi = 0.00313299, GNF2_PTPRC\n",
      "cat 120, mi = 0.00312313, GNF2_CENPE\n",
      "cat 668, mi = 0.00297543, MORF_ANP32B\n",
      "cat 202, mi = 0.00290147, GNF2_RRM2\n",
      "cat 101, mi = 0.00290077, GNF2_CASP8\n",
      "cat 205, mi = 0.00283068, GNF2_SELL\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 121, mi = 0.004, pvalue = 0.000, GNF2_CENPF\n",
      "Passed: cat 92, mi = 0.003, pvalue = 0.001, GNF2_ATM\n",
      "Passed: cat 724, mi = 0.003, pvalue = 0.002, MORF_ETV3\n",
      "Passed: cat 668, mi = 0.003, pvalue = 0.004, MORF_ANP32B\n",
      "Didn't pass (cutoff=0.0050): cat 205, mi = 0.003, pvalue = 0.007, GNF2_SELL\n",
      "Didn't pass (cutoff=0.0050): cat 352, mi = 0.003, pvalue = 0.006, MODULE_213\n",
      "Didn't pass (cutoff=0.0050): cat 65, mi = 0.003, pvalue = 0.009, GCM_PTPRD\n",
      "Didn't pass (cutoff=0.0050): cat 153, mi = 0.003, pvalue = 0.015, GNF2_INPP5D\n",
      "Passed: cat 294, mi = 0.002, pvalue = 0.003, MODULE_157\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.002, pvalue = 0.025, GNF2_RAP1B\n",
      "Didn't pass (cutoff=0.0050): cat 72, mi = 0.002, pvalue = 0.022, GCM_RBM8A\n",
      "Didn't pass (cutoff=0.0050): cat 446, mi = 0.002, pvalue = 0.025, MODULE_325\n",
      "Didn't pass (cutoff=0.0050): cat 318, mi = 0.002, pvalue = 0.036, MODULE_180\n",
      "Didn't pass (cutoff=0.0050): cat 634, mi = 0.002, pvalue = 0.020, MODULE_70\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.047, GNF2_JAK1\n",
      "Didn't pass (cutoff=0.0050): cat 580, mi = 0.002, pvalue = 0.006, MODULE_509\n",
      "Didn't pass (cutoff=0.0050): cat 453, mi = 0.002, pvalue = 0.050, MODULE_331\n",
      "Didn't pass (cutoff=0.0050): cat 846, mi = 0.002, pvalue = 0.036, MORF_TPR\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.002, pvalue = 0.067, GCM_CSNK1A1\n",
      "Didn't pass (cutoff=0.0050): cat 189, mi = 0.002, pvalue = 0.057, GNF2_PTPN6\n",
      "Didn't pass (cutoff=0.0050): cat 61, mi = 0.002, pvalue = 0.025, GCM_PRKAG1\n",
      "Didn't pass (cutoff=0.0050): cat 474, mi = 0.002, pvalue = 0.044, MODULE_36\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.002, pvalue = 0.046, GCM_DEAF1\n",
      "Passed: cat 531, mi = 0.002, pvalue = 0.003, MODULE_432\n",
      "Didn't pass (cutoff=0.0050): cat 242, mi = 0.002, pvalue = 0.018, MODULE_107\n",
      "Didn't pass (cutoff=0.0050): cat 332, mi = 0.002, pvalue = 0.018, MODULE_196\n",
      "Didn't pass (cutoff=0.0050): cat 385, mi = 0.002, pvalue = 0.017, MODULE_253\n",
      "Didn't pass (cutoff=0.0050): cat 475, mi = 0.002, pvalue = 0.053, MODULE_360\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.002, pvalue = 0.067, GCM_RING1\n",
      "Didn't pass (cutoff=0.0050): cat 231, mi = 0.002, pvalue = 0.083, GNF2_VAV1\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.002, pvalue = 0.071, GNF2_PECAM1\n",
      "Didn't pass (cutoff=0.0050): cat 822, mi = 0.002, pvalue = 0.069, MORF_REV3L\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.002, pvalue = 0.051, GCM_GSPT1\n",
      "Didn't pass (cutoff=0.0050): cat 549, mi = 0.002, pvalue = 0.054, MODULE_458\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.002, pvalue = 0.008, GCM_FCGR2B\n",
      "Didn't pass (cutoff=0.0050): cat 546, mi = 0.002, pvalue = 0.029, MODULE_455\n",
      "Didn't pass (cutoff=0.0050): cat 525, mi = 0.002, pvalue = 0.077, MODULE_426\n",
      "Didn't pass (cutoff=0.0050): cat 218, mi = 0.002, pvalue = 0.090, GNF2_STAT6\n",
      "Didn't pass (cutoff=0.0050): cat 109, mi = 0.002, pvalue = 0.120, GNF2_CD48\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.102, MODULE_491\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.002, pvalue = 0.116, GNF2_PAK2\n",
      "Didn't pass (cutoff=0.0050): cat 193, mi = 0.002, pvalue = 0.115, GNF2_RAB7L1\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.002, pvalue = 0.089, MORF_IKBKG\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.002, pvalue = 0.122, GCM_IL6ST\n",
      "Total number of categories evaluated = 87.\n",
      "Number of categories that passed the tests = 6\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.553665492239562\n",
      "best merge is 0, 3, 0.626424360240101\n",
      "order classes\n",
      "2 - 0.830498238359343\n",
      "1 2\n",
      "0\n",
      "3\n",
      "4\n",
      "5\n",
      "Done.Start drawing\n",
      "8.47\t-11.35\n",
      "MODULE_157\n",
      "GNF2_ATM\n",
      "MORF_ETV3\n",
      "MORF_ANP32B\n",
      "GNF2_CENPF\n",
      "MODULE_432\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.553665492239562\n",
      "best merge is 0, 3, 0.626424360240101\n",
      "order classes\n",
      "2 - 0.830498238359343\n",
      "1 2\n",
      "0\n",
      "3\n",
      "4\n",
      "5\n",
      "Done.-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_c4/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 5917\n",
      "cat 1993, mi = 0.00375714, GO_LYMPHOCYTE_HOMEOSTASIS\n",
      "cat 1662, mi = 0.00359515, GO_HOMEOSTASIS_OF_NUMBER_OF_CELLS\n",
      "cat 1913, mi = 0.00359371, GO_LEUKOCYTE_HOMEOSTASIS\n",
      "cat 4835, mi = 0.00341444, GO_REGULATION_OF_RESPONSE_TO_CYTOKINE_STIMULUS\n",
      "cat 2196, mi = 0.00338609, GO_MIDBRAIN_DEVELOPMENT\n",
      "cat 3047, mi = 0.00329339, GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "cat 2051, mi = 0.00327670, GO_MAMMARY_GLAND_DEVELOPMENT\n",
      "cat 1918, mi = 0.00324238, GO_LEUKOCYTE_PROLIFERATION\n",
      "cat 1329, mi = 0.00321912, GO_EXONUCLEASE_ACTIVITY_ACTIVE_WITH_EITHER_RIBO_OR_DEOXYRIBONUCLEIC_ACIDS_AND_PRODUCING_5_PHOSPHOMONOESTERS\n",
      "cat 377, mi = 0.00310844, GO_B_CELL_PROLIFERATION\n",
      "cat 1328, mi = 0.00305854, GO_EXONUCLEASE_ACTIVITY\n",
      "cat 7, mi = 0.00303469, GO_3_5_EXONUCLEASE_ACTIVITY\n",
      "cat 5781, mi = 0.00301610, GO_T_CELL_HOMEOSTASIS\n",
      "cat 1387, mi = 0.00293157, GO_FEMALE_GAMETE_GENERATION\n",
      "cat 1663, mi = 0.00291866, GO_HOMEOSTASIS_OF_NUMBER_OF_CELLS_WITHIN_A_TISSUE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1993, mi = 0.004, pvalue = 0.000, GO_LYMPHOCYTE_HOMEOSTASIS\n",
      "Passed: cat 4835, mi = 0.003, pvalue = 0.001, GO_REGULATION_OF_RESPONSE_TO_CYTOKINE_STIMULUS\n",
      "Passed: cat 3047, mi = 0.003, pvalue = 0.001, GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 1329, mi = 0.003, pvalue = 0.002, GO_EXONUCLEASE_ACTIVITY_ACTIVE_WITH_EITHER_RIBO_OR_DEOXYRIBONUCLEIC_ACIDS_AND_PRODUCING_5_PHOSPHOMONOESTERS\n",
      "Passed: cat 2430, mi = 0.003, pvalue = 0.003, GO_NEGATIVE_REGULATION_OF_BMP_SIGNALING_PATHWAY\n",
      "Passed: cat 5617, mi = 0.003, pvalue = 0.003, GO_TELOMERIC_DNA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2920, mi = 0.003, pvalue = 0.006, GO_NUCLEAR_PERIPHERY\n",
      "Passed: cat 4831, mi = 0.003, pvalue = 0.001, GO_REGULATION_OF_RESPIRATORY_GASEOUS_EXCHANGE\n",
      "Didn't pass (cutoff=0.0050): cat 400, mi = 0.003, pvalue = 0.012, GO_CALCIUM_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3159, mi = 0.003, pvalue = 0.012, GO_PEPTIDYL_ASPARAGINE_MODIFICATION\n",
      "Didn't pass (cutoff=0.0050): cat 5292, mi = 0.003, pvalue = 0.007, GO_RNA_STABILIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 5658, mi = 0.003, pvalue = 0.007, GO_TOXIN_TRANSPORT\n",
      "Passed: cat 1765, mi = 0.003, pvalue = 0.002, GO_INSULIN_RECEPTOR_SUBSTRATE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 320, mi = 0.003, pvalue = 0.013, GO_BETA_CATENIN_TCF_COMPLEX_ASSEMBLY\n",
      "Passed: cat 3658, mi = 0.003, pvalue = 0.004, GO_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 3151, mi = 0.003, pvalue = 0.013, GO_PEPTIDE_N_ACETYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4644, mi = 0.003, pvalue = 0.009, GO_REGULATION_OF_MRNA_3_END_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 3068, mi = 0.003, pvalue = 0.007, GO_OSTEOBLAST_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 5241, mi = 0.003, pvalue = 0.006, GO_RIBOSOMAL_LARGE_SUBUNIT_BIOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 2939, mi = 0.003, pvalue = 0.005, GO_NUCLEOBASE_CONTAINING_COMPOUND_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2156, mi = 0.003, pvalue = 0.013, GO_METHYLTRANSFERASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.003, pvalue = 0.016, GO_AMINOGLYCAN_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1212, mi = 0.002, pvalue = 0.018, GO_ENDOSOME_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.018, GO_ACETYLGLUCOSAMINYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 154, mi = 0.002, pvalue = 0.019, GO_AMINO_SUGAR_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2268, mi = 0.002, pvalue = 0.018, GO_MONOCARBOXYLIC_ACID_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3640, mi = 0.002, pvalue = 0.013, GO_POSITIVE_REGULATION_OF_MRNA_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 2350, mi = 0.002, pvalue = 0.019, GO_MYD88_DEPENDENT_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 4676, mi = 0.002, pvalue = 0.008, GO_REGULATION_OF_NEURONAL_SYNAPTIC_PLASTICITY\n",
      "Didn't pass (cutoff=0.0050): cat 4012, mi = 0.002, pvalue = 0.020, GO_PROTEIN_N_LINKED_GLYCOSYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 4348, mi = 0.002, pvalue = 0.021, GO_REGULATION_OF_CYTOKINESIS\n",
      "Didn't pass (cutoff=0.0050): cat 4010, mi = 0.002, pvalue = 0.027, GO_PROTEIN_MONOUBIQUITINATION\n",
      "Didn't pass (cutoff=0.0050): cat 2250, mi = 0.002, pvalue = 0.029, GO_MLL1_2_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 957, mi = 0.002, pvalue = 0.018, GO_CYTOSOLIC_RIBOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 2572, mi = 0.002, pvalue = 0.022, GO_NEGATIVE_REGULATION_OF_HORMONE_SECRETION\n",
      "Total number of categories evaluated = 78.\n",
      "Number of categories that passed the tests = 9\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 6, 0.345504416479659\n",
      "best merge is 1, 5, 0.406795538708\n",
      "best merge is 2, 3, 0.478491868164445\n",
      "best merge is 0, 8, 0.516484498397191\n",
      "best merge is 1, 4, 0.885513272844404\n",
      "order classes\n",
      "2 - 1.16209012139368\n",
      "2 - 0.915289962093\n",
      "2 - 1.07660670337\n",
      "2 - 0.777384937079232\n",
      "4 6\n",
      "1 5\n",
      "2 3\n",
      "0 8\n",
      "7\n",
      "Done.Start drawing\n",
      "8.47\t-11.35\n",
      "GO_NEGATIVE_REGULATION_OF_BMP_SIGNALING_PATHWAY\n",
      "GO_REGULATION_OF_RESPIRATORY_GASEOUS_EXCHANGE\n",
      "GO_LYMPHOCYTE_HOMEOSTASIS\n",
      "GO_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_ACTIVATION\n",
      "GO_TELOMERIC_DNA_BINDING\n",
      "GO_REGULATION_OF_RESPONSE_TO_CYTOKINE_STIMULUS\n",
      "GO_INSULIN_RECEPTOR_SUBSTRATE_BINDING\n",
      "GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "GO_EXONUCLEASE_ACTIVITY_ACTIVE_WITH_EITHER_RIBO_OR_DEOXYRIBONUCLEIC_ACIDS_AND_PRODUCING_5_PHOSPHOMONOESTERS\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 6, 0.345504416479659\n",
      "best merge is 1, 5, 0.406795538708\n",
      "best merge is 2, 3, 0.478491868164445\n",
      "best merge is 0, 8, 0.516484498397191\n",
      "best merge is 1, 4, 0.885513272844404\n",
      "order classes\n",
      "2 - 1.16209012139368\n",
      "2 - 0.915289962093\n",
      "2 - 1.07660670337\n",
      "2 - 0.777384937079232\n",
      "4 6\n",
      "1 5\n",
      "2 3\n",
      "0 8\n",
      "7\n",
      "Done.-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_c5/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 189\n",
      "cat 183, mi = 0.00344204, VEGF_A_UP.V1_DN\n",
      "cat 115, mi = 0.00292280, MTOR_UP.N4.V1_DN\n",
      "cat 149, mi = 0.00269177, PTEN_DN.V2_UP\n",
      "cat 96, mi = 0.00257795, KRAS.BREAST_UP.V1_UP\n",
      "cat 156, mi = 0.00237424, RB_P107_DN.V1_DN\n",
      "cat 55, mi = 0.00231876, ERB2_UP.V1_DN\n",
      "cat 6, mi = 0.00225967, ATF2_S_UP.V1_DN\n",
      "cat 81, mi = 0.00221860, JAK2_DN.V1_DN\n",
      "cat 45, mi = 0.00215994, DCA_UP.V1_DN\n",
      "cat 118, mi = 0.00214892, MTOR_UP.V1_UP\n",
      "cat 185, mi = 0.00213989, WNT_UP.V1_DN\n",
      "cat 184, mi = 0.00212594, VEGF_A_UP.V1_UP\n",
      "cat 132, mi = 0.00208117, PDGF_UP.V1_DN\n",
      "cat 111, mi = 0.00207177, MEK_UP.V1_DN\n",
      "cat 97, mi = 0.00204692, KRAS.DF.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 183, mi = 0.003, pvalue = 0.001, VEGF_A_UP.V1_DN\n",
      "Passed: cat 115, mi = 0.003, pvalue = 0.003, MTOR_UP.N4.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 149, mi = 0.003, pvalue = 0.009, PTEN_DN.V2_UP\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.026, RB_P107_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.002, pvalue = 0.037, DCA_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.002, pvalue = 0.042, MTOR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.002, pvalue = 0.050, WNT_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 184, mi = 0.002, pvalue = 0.038, VEGF_A_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.002, pvalue = 0.066, PDGF_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.002, pvalue = 0.050, MEK_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.002, pvalue = 0.055, KRAS.DF.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.002, pvalue = 0.059, P53_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 65, mi = 0.002, pvalue = 0.087, GCNP_SHH_UP_EARLY.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 134, mi = 0.002, pvalue = 0.083, PIGF_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.147, TBK1.DN.48HRS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 107, mi = 0.002, pvalue = 0.127, LEF1_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.002, pvalue = 0.159, PTEN_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 110, mi = 0.002, pvalue = 0.120, LTE2_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.002, pvalue = 0.160, BCAT_GDS748_UP\n",
      "Didn't pass (cutoff=0.0050): cat 35, mi = 0.002, pvalue = 0.114, CSR_EARLY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.002, pvalue = 0.142, PDGF_ERK_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 102, mi = 0.002, pvalue = 0.173, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "Total number of categories evaluated = 21.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.47\t-11.35\n",
      "MTOR_UP.N4.V1_DN\n",
      "VEGF_A_UP.V1_DN\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_c6/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 4872\n",
      "cat 1294, mi = 0.00474941, GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_6H_DN\n",
      "cat 59, mi = 0.00465555, GSE10325_CD4_TCELL_VS_MYELOID_UP\n",
      "cat 762, mi = 0.00434000, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "cat 4832, mi = 0.00427504, GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "cat 204, mi = 0.00402862, GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "cat 206, mi = 0.00394839, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "cat 4345, mi = 0.00394361, GSE5542_UNTREATED_VS_IFNA_AND_IFNG_TREATED_EPITHELIAL_CELLS_24H_UP\n",
      "cat 3046, mi = 0.00382533, GSE34156_UNTREATED_VS_6H_NOD2_AND_TLR1_TLR2_LIGAND_TREATED_MONOCYTE_DN\n",
      "cat 4848, mi = 0.00369851, GSE9988_LOW_LPS_VS_CTRL_TREATED_MONOCYTE_DN\n",
      "cat 3085, mi = 0.00361888, GSE3565_DUSP1_VS_WT_SPLENOCYTES_UP\n",
      "cat 4858, mi = 0.00360152, GSE9988_LPS_VS_VEHICLE_TREATED_MONOCYTE_DN\n",
      "cat 330, mi = 0.00356472, GSE13485_DAY3_VS_DAY7_YF17D_VACCINE_PBMC_DN\n",
      "cat 3673, mi = 0.00355786, GSE3982_MAC_VS_TH2_UP\n",
      "cat 4852, mi = 0.00348885, GSE9988_LPS_VS_CTRL_TREATED_MONOCYTE_DN\n",
      "cat 3330, mi = 0.00343695, GSE37301_CD4_TCELL_VS_RAG2_KO_NK_CELL_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1294, mi = 0.005, pvalue = 0.000, GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_6H_DN\n",
      "Passed: cat 762, mi = 0.004, pvalue = 0.000, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "Passed: cat 4832, mi = 0.004, pvalue = 0.000, GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "Passed: cat 4345, mi = 0.004, pvalue = 0.000, GSE5542_UNTREATED_VS_IFNA_AND_IFNG_TREATED_EPITHELIAL_CELLS_24H_UP\n",
      "Passed: cat 3673, mi = 0.004, pvalue = 0.000, GSE3982_MAC_VS_TH2_UP\n",
      "Passed: cat 1695, mi = 0.003, pvalue = 0.001, GSE21927_GMCSF_IL6_VS_GMCSF_GCSF_TREATED_BONE_MARROW_UP\n",
      "Passed: cat 536, mi = 0.003, pvalue = 0.001, GSE14699_DELETIONAL_TOLERANCE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "Passed: cat 3518, mi = 0.003, pvalue = 0.001, GSE3920_IFNA_VS_IFNG_TREATED_ENDOTHELIAL_CELL_DN\n",
      "Passed: cat 2742, mi = 0.003, pvalue = 0.002, GSE29618_MONOCYTE_VS_PDC_DN\n",
      "Passed: cat 684, mi = 0.003, pvalue = 0.002, GSE15930_NAIVE_VS_72H_IN_VITRO_STIM_IFNAB_CD8_TCELL_DN\n",
      "Passed: cat 2121, mi = 0.003, pvalue = 0.003, GSE24574_BCL6_LOW_TFH_VS_TCONV_CD4_TCELL_UP\n",
      "Passed: cat 1952, mi = 0.003, pvalue = 0.003, GSE22935_24H_VS_48H_MBOVIS_BCG_STIM_MYD88_KO_MACROPHAGE_DN\n",
      "Passed: cat 1407, mi = 0.003, pvalue = 0.004, GSE19888_CTRL_VS_A3R_INHIBITOR_TREATED_MAST_CELL_UP\n",
      "Passed: cat 1378, mi = 0.003, pvalue = 0.004, GSE19825_NAIVE_VS_DAY3_EFF_CD8_TCELL_DN\n",
      "Passed: cat 3539, mi = 0.003, pvalue = 0.004, GSE39556_CD8A_DC_VS_NK_CELL_UP\n",
      "Passed: cat 2809, mi = 0.003, pvalue = 0.004, GSE3039_ALPHAALPHA_CD8_TCELL_VS_B2_BCELL_UP\n",
      "Passed: cat 4578, mi = 0.003, pvalue = 0.004, GSE7460_CTRL_VS_TGFB_TREATED_ACT_TREG_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3182, mi = 0.003, pvalue = 0.006, GSE360_L_DONOVANI_VS_L_MAJOR_MAC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 342, mi = 0.003, pvalue = 0.006, GSE13522_CTRL_VS_T_CRUZI_BRAZIL_STRAIN_INF_SKIN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2495, mi = 0.003, pvalue = 0.006, GSE2770_UNTREATED_VS_IL12_TREATED_ACT_CD4_TCELL_6H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3477, mi = 0.003, pvalue = 0.009, GSE37605_C57BL6_VS_NOD_FOXP3_IRES_GFP_TCONV_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4655, mi = 0.003, pvalue = 0.006, GSE7831_UNSTIM_VS_INFLUENZA_STIM_PDC_1H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3362, mi = 0.003, pvalue = 0.009, GSE37301_LYMPHOID_PRIMED_MPP_VS_PRO_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3398, mi = 0.003, pvalue = 0.007, GSE37416_0H_VS_6H_F_TULARENSIS_LVS_NEUTROPHIL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2988, mi = 0.003, pvalue = 0.008, GSE33424_CD161_HIGH_VS_NEG_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3385, mi = 0.003, pvalue = 0.009, GSE37301_RAG2_KO_VS_RAG2_AND_ETS1_KO_NK_CELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 981, mi = 0.003, pvalue = 0.009, GSE17721_CTRL_VS_PAM3CSK4_2H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 239, mi = 0.003, pvalue = 0.012, GSE12845_IGD_NEG_BLOOD_VS_NAIVE_TONSIL_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 477, mi = 0.003, pvalue = 0.013, GSE14413_UNSTIM_VS_IFNB_STIM_RAW264_CELLS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1835, mi = 0.002, pvalue = 0.013, GSE22611_NOD2_TRANSDUCED_VS_CTRL_HEK293T_STIMULATED_WITH_MDP_2H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1392, mi = 0.002, pvalue = 0.012, GSE19888_ADENOSINE_A3R_INH_PRETREAT_AND_ACT_BY_A3R_VS_TCELL_MEMBRANES_ACT_MAST_CELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4610, mi = 0.002, pvalue = 0.015, GSE7509_UNSTIM_VS_TNFA_IL1B_IL6_PGE_STIM_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 473, mi = 0.002, pvalue = 0.024, GSE14413_UNSTIM_VS_IFNB_STIM_L929_CELLS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1992, mi = 0.002, pvalue = 0.013, GSE23321_CD8_STEM_CELL_MEMORY_VS_EFFECTOR_MEMORY_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3059, mi = 0.002, pvalue = 0.026, GSE34205_RSV_VS_FLU_INF_INFANT_PBMC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1431, mi = 0.002, pvalue = 0.020, GSE19941_IL10_KO_VS_IL10_KO_AND_NFKBP50_KO_UNSTIM_MACROPHAGE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2312, mi = 0.002, pvalue = 0.015, GSE26343_WT_VS_NFAT5_KO_MACROPHAGE_DN\n",
      "Total number of categories evaluated = 197.\n",
      "Number of categories that passed the tests = 17\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 8, 9, 0.233107792299503\n",
      "best merge is 14, 15, 0.311140280732158\n",
      "best merge is 0, 5, 0.349259715899531\n",
      "best merge is 12, 13, 0.361149808045866\n",
      "best merge is 6, 7, 0.362179510011005\n",
      "best merge is 8, 14, 0.487032263024491\n",
      "best merge is 0, 2, 0.518322423535101\n",
      "best merge is 1, 3, 0.55876266296007\n",
      "best merge is 4, 16, 0.592548261001531\n",
      "best merge is 6, 11, 0.653865092596104\n",
      "best merge is 1, 10, 0.68651510618377\n",
      "best merge is 0, 8, 0.856242524737897\n",
      "best merge is 6, 12, 0.862430083089898\n",
      "order classes\n",
      "7 - 1.40294031337738\n",
      "3 - 2.43262806522736\n",
      "2 - 2.51833010925651\n",
      "3 - 2.10284924581145\n",
      "2 - 1.53488668419493\n",
      "0 5 2 8 9 14 15\n",
      "1 3 10\n",
      "4 16\n",
      "6 7 11\n",
      "12 13\n",
      "Done.Start drawing\n",
      "8.47\t-11.35\n",
      "GSE3982_MAC_VS_TH2_UP\n",
      "GSE7460_CTRL_VS_TGFB_TREATED_ACT_TREG_DN\n",
      "GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_6H_DN\n",
      "GSE21927_GMCSF_IL6_VS_GMCSF_GCSF_TREATED_BONE_MARROW_UP\n",
      "GSE5542_UNTREATED_VS_IFNA_AND_IFNG_TREATED_EPITHELIAL_CELLS_24H_UP\n",
      "GSE19825_NAIVE_VS_DAY3_EFF_CD8_TCELL_DN\n",
      "GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "GSE39556_CD8A_DC_VS_NK_CELL_UP\n",
      "GSE3039_ALPHAALPHA_CD8_TCELL_VS_B2_BCELL_UP\n",
      "GSE29618_MONOCYTE_VS_PDC_DN\n",
      "GSE19888_CTRL_VS_A3R_INHIBITOR_TREATED_MAST_CELL_UP\n",
      "GSE24574_BCL6_LOW_TFH_VS_TCONV_CD4_TCELL_UP\n",
      "GSE15930_NAIVE_VS_72H_IN_VITRO_STIM_IFNAB_CD8_TCELL_DN\n",
      "GSE14699_DELETIONAL_TOLERANCE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "GSE3920_IFNA_VS_IFNG_TREATED_ENDOTHELIAL_CELL_DN\n",
      "GSE22935_24H_VS_48H_MBOVIS_BCG_STIM_MYD88_KO_MACROPHAGE_DN\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 8, 9, 0.233107792299503\n",
      "best merge is 14, 15, 0.311140280732158\n",
      "best merge is 0, 5, 0.349259715899531\n",
      "best merge is 12, 13, 0.361149808045866\n",
      "best merge is 6, 7, 0.362179510011005\n",
      "best merge is 8, 14, 0.487032263024491\n",
      "best merge is 0, 2, 0.518322423535101\n",
      "best merge is 1, 3, 0.55876266296007\n",
      "best merge is 4, 16, 0.592548261001531\n",
      "best merge is 6, 11, 0.653865092596104\n",
      "best merge is 1, 10, 0.68651510618377\n",
      "best merge is 0, 8, 0.856242524737897\n",
      "best merge is 6, 12, 0.862430083089898\n",
      "order classes\n",
      "7 - 1.40294031337738\n",
      "3 - 2.43262806522736\n",
      "2 - 2.51833010925651\n",
      "3 - 2.10284924581145\n",
      "2 - 1.53488668419493\n",
      "0 5 2 8 9 14 15\n",
      "1 3 10\n",
      "4 16\n",
      "6 7 11\n",
      "12 13\n",
      "Done.-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_c7/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 17786\n",
      "cat 9151, mi = 0.00474941, GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_6H_DN\n",
      "cat 7916, mi = 0.00465555, GSE10325_CD4_TCELL_VS_MYELOID_UP\n",
      "cat 8619, mi = 0.00434000, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "cat 12689, mi = 0.00427504, GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "cat 1749, mi = 0.00410476, GNF2_CENPF\n",
      "cat 1741, mi = 0.00405147, GNF2_CDC2\n",
      "cat 8061, mi = 0.00402862, GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "cat 12791, mi = 0.00397908, HALLMARK_E2F_TARGETS\n",
      "cat 8063, mi = 0.00394839, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "cat 12202, mi = 0.00394361, GSE5542_UNTREATED_VS_IFNA_AND_IFNG_TREATED_EPITHELIAL_CELLS_24H_UP\n",
      "cat 17325, mi = 0.00383646, YU_MYC_TARGETS_DN\n",
      "cat 10903, mi = 0.00382533, GSE34156_UNTREATED_VS_6H_NOD2_AND_TLR1_TLR2_LIGAND_TREATED_MONOCYTE_DN\n",
      "cat 3879, mi = 0.00375714, GO_LYMPHOCYTE_HOMEOSTASIS\n",
      "cat 12705, mi = 0.00369851, GSE9988_LOW_LPS_VS_CTRL_TREATED_MONOCYTE_DN\n",
      "cat 1812, mi = 0.00364946, GNF2_PCNA\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 9151, mi = 0.005, pvalue = 0.000, GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_6H_DN\n",
      "Passed: cat 8619, mi = 0.004, pvalue = 0.000, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "Passed: cat 12689, mi = 0.004, pvalue = 0.000, GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "Passed: cat 1749, mi = 0.004, pvalue = 0.000, GNF2_CENPF\n",
      "Passed: cat 12202, mi = 0.004, pvalue = 0.000, GSE5542_UNTREATED_VS_IFNA_AND_IFNG_TREATED_EPITHELIAL_CELLS_24H_UP\n",
      "Passed: cat 17325, mi = 0.004, pvalue = 0.000, YU_MYC_TARGETS_DN\n",
      "Passed: cat 3879, mi = 0.004, pvalue = 0.000, GO_LYMPHOCYTE_HOMEOSTASIS\n",
      "Passed: cat 15540, mi = 0.004, pvalue = 0.001, REACTOME_APOPTOTIC_CLEAVAGE_OF_CELLULAR_PROTEINS\n",
      "Passed: cat 11530, mi = 0.004, pvalue = 0.000, GSE3982_MAC_VS_TH2_UP\n",
      "Passed: cat 9552, mi = 0.003, pvalue = 0.000, GSE21927_GMCSF_IL6_VS_GMCSF_GCSF_TREATED_BONE_MARROW_UP\n",
      "Passed: cat 427, mi = 0.003, pvalue = 0.000, BIOCARTA_ETS_PATHWAY\n",
      "Passed: cat 8393, mi = 0.003, pvalue = 0.001, GSE14699_DELETIONAL_TOLERANCE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "Passed: cat 11375, mi = 0.003, pvalue = 0.000, GSE3920_IFNA_VS_IFNG_TREATED_ENDOTHELIAL_CELL_DN\n",
      "Passed: cat 3215, mi = 0.003, pvalue = 0.002, GO_EXONUCLEASE_ACTIVITY_ACTIVE_WITH_EITHER_RIBO_OR_DEOXYRIBONUCLEIC_ACIDS_AND_PRODUCING_5_PHOSPHOMONOESTERS\n",
      "Passed: cat 10599, mi = 0.003, pvalue = 0.002, GSE29618_MONOCYTE_VS_PDC_DN\n",
      "Passed: cat 12789, mi = 0.003, pvalue = 0.002, HALLMARK_COMPLEMENT\n",
      "Passed: cat 156, mi = 0.003, pvalue = 0.003, AMUNDSON_POOR_SURVIVAL_AFTER_GAMMA_RADIATION_2G\n",
      "Passed: cat 16743, mi = 0.003, pvalue = 0.002, TCTAGAG_MIR517\n",
      "Passed: cat 9809, mi = 0.003, pvalue = 0.003, GSE22935_24H_VS_48H_MBOVIS_BCG_STIM_MYD88_KO_MACROPHAGE_DN\n",
      "Passed: cat 14521, mi = 0.003, pvalue = 0.004, MORF_ANP32B\n",
      "Passed: cat 679, mi = 0.003, pvalue = 0.002, BRUECKNER_TARGETS_OF_MIRLET7A3_DN\n",
      "Passed: cat 4316, mi = 0.003, pvalue = 0.003, GO_NEGATIVE_REGULATION_OF_BMP_SIGNALING_PATHWAY\n",
      "Passed: cat 9013, mi = 0.003, pvalue = 0.004, GSE17721_POLYIC_VS_PAM3CSK4_24H_BMDC_DN\n",
      "Passed: cat 7503, mi = 0.003, pvalue = 0.003, GO_TELOMERIC_DNA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2812, mi = 0.003, pvalue = 0.007, GO_CYTOKINE_RECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 11039, mi = 0.003, pvalue = 0.005, GSE360_L_DONOVANI_VS_L_MAJOR_MAC_DN\n",
      "Passed: cat 6717, mi = 0.003, pvalue = 0.001, GO_REGULATION_OF_RESPIRATORY_GASEOUS_EXCHANGE\n",
      "Didn't pass (cutoff=0.0050): cat 10352, mi = 0.003, pvalue = 0.005, GSE2770_UNTREATED_VS_IL12_TREATED_ACT_CD4_TCELL_6H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 17407, mi = 0.003, pvalue = 0.007, ZHENG_FOXP3_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1610, mi = 0.003, pvalue = 0.010, GCM_PTPRD\n",
      "Didn't pass (cutoff=0.0050): cat 2286, mi = 0.003, pvalue = 0.011, GO_CALCIUM_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 17378, mi = 0.003, pvalue = 0.006, ZHAN_MULTIPLE_MYELOMA_CD1_VS_CD2_DN\n",
      "Didn't pass (cutoff=0.0050): cat 11334, mi = 0.003, pvalue = 0.010, GSE37605_C57BL6_VS_NOD_FOXP3_IRES_GFP_TCONV_UP\n",
      "Didn't pass (cutoff=0.0050): cat 11219, mi = 0.003, pvalue = 0.011, GSE37301_LYMPHOID_PRIMED_MPP_VS_PRO_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1359, mi = 0.003, pvalue = 0.006, FIGUEROA_AML_METHYLATION_CLUSTER_5_DN\n",
      "Passed: cat 13560, mi = 0.003, pvalue = 0.001, KYNG_ENVIRONMENTAL_STRESS_RESPONSE_NOT_BY_UV_IN_OLD\n",
      "Didn't pass (cutoff=0.0050): cat 11798, mi = 0.003, pvalue = 0.009, GSE41867_NAIVE_VS_DAY15_LCMV_EFFECTOR_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 11242, mi = 0.003, pvalue = 0.009, GSE37301_RAG2_KO_VS_RAG2_AND_ETS1_KO_NK_CELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2301, mi = 0.003, pvalue = 0.013, GO_CARBOHYDRATE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 7178, mi = 0.003, pvalue = 0.008, GO_RNA_STABILIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 9644, mi = 0.003, pvalue = 0.011, GSE22432_PDC_VS_TGFB1_TREATEDCOMMON_DC_PROGENITOR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 7764, mi = 0.003, pvalue = 0.012, GO_VITAMIN_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2206, mi = 0.003, pvalue = 0.014, GO_BETA_CATENIN_TCF_COMPLEX_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 5544, mi = 0.003, pvalue = 0.005, GO_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 8334, mi = 0.003, pvalue = 0.011, GSE14413_UNSTIM_VS_IFNB_STIM_RAW264_CELLS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5037, mi = 0.003, pvalue = 0.013, GO_PEPTIDE_N_ACETYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 6530, mi = 0.003, pvalue = 0.009, GO_REGULATION_OF_MRNA_3_END_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 11487, mi = 0.003, pvalue = 0.013, GSE3982_EFF_MEMORY_CD4_TCELL_VS_TH2_DN\n",
      "Didn't pass (cutoff=0.0050): cat 815, mi = 0.003, pvalue = 0.016, CCATCCA_MIR432\n",
      "Didn't pass (cutoff=0.0050): cat 15359, mi = 0.003, pvalue = 0.013, PLASARI_TGFB1_SIGNALING_VIA_NFIC_1HR_DN\n",
      "Didn't pass (cutoff=0.0050): cat 7127, mi = 0.003, pvalue = 0.006, GO_RIBOSOMAL_LARGE_SUBUNIT_BIOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 2026, mi = 0.003, pvalue = 0.016, GO_AMINOGLYCAN_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3098, mi = 0.002, pvalue = 0.019, GO_ENDOSOME_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.002, pvalue = 0.016, AGGCACT_MIR5153P\n",
      "Didn't pass (cutoff=0.0050): cat 8400, mi = 0.002, pvalue = 0.014, GSE14769_20MIN_VS_360MIN_LPS_BMDM_UP\n",
      "Didn't pass (cutoff=0.0050): cat 9249, mi = 0.002, pvalue = 0.013, GSE19888_ADENOSINE_A3R_INH_PRETREAT_AND_ACT_BY_A3R_VS_TCELL_MEMBRANES_ACT_MAST_CELL_DN\n",
      "Total number of categories evaluated = 351.\n",
      "Number of categories that passed the tests = 26\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 17, 0.16444306662021\n",
      "best merge is 11, 13, 0.173946975341428\n",
      "best merge is 0, 20, 0.194423231958034\n",
      "best merge is 10, 25, 0.199815488432056\n",
      "best merge is 15, 22, 0.215936713498748\n",
      "best merge is 7, 21, 0.252342712448099\n",
      "best merge is 14, 16, 0.311519515959067\n",
      "best merge is 7, 24, 0.32650823496379\n",
      "best merge is 0, 9, 0.331711571836108\n",
      "best merge is 1, 6, 0.342951125602591\n",
      "best merge is 3, 15, 0.387983899520164\n",
      "best merge is 11, 18, 0.408189439042095\n",
      "best merge is 14, 19, 0.457432518437307\n",
      "best merge is 2, 5, 0.514658558428039\n",
      "best merge is 8, 12, 0.529251546759875\n",
      "best merge is 4, 10, 0.577127997699795\n",
      "best merge is 3, 14, 0.597811443110735\n",
      "best merge is 0, 2, 0.625744991104846\n",
      "best merge is 1, 4, 0.670904871674717\n",
      "best merge is 7, 23, 0.800318102046363\n",
      "best merge is 8, 11, 0.820304861130424\n",
      "best merge is 0, 7, 0.909569774751123\n",
      "order classes\n",
      "6 - 1.70180658394534\n",
      "5 - 2.38053996522474\n",
      "6 - 1.82929067951485\n",
      "4 - 2.68637970941825\n",
      "5 - 2.6796251318587\n",
      "0 17 20 9 2 5\n",
      "1 6 4 10 25\n",
      "3 15 22 14 16 19\n",
      "7 21 24 23\n",
      "8 12 11 13 18\n",
      "Done.Start drawing\n",
      "8.47\t-11.35\n",
      "GSE3982_MAC_VS_TH2_UP\n",
      "REACTOME_APOPTOTIC_CLEAVAGE_OF_CELLULAR_PROTEINS\n",
      "GO_NEGATIVE_REGULATION_OF_BMP_SIGNALING_PATHWAY\n",
      "GO_REGULATION_OF_RESPIRATORY_GASEOUS_EXCHANGE\n",
      "GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "GO_LYMPHOCYTE_HOMEOSTASIS\n",
      "BIOCARTA_ETS_PATHWAY\n",
      "KYNG_ENVIRONMENTAL_STRESS_RESPONSE_NOT_BY_UV_IN_OLD\n",
      "MORF_ANP32B\n",
      "GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_6H_DN\n",
      "TCTAGAG_MIR517\n",
      "BRUECKNER_TARGETS_OF_MIRLET7A3_DN\n",
      "GSE21927_GMCSF_IL6_VS_GMCSF_GCSF_TREATED_BONE_MARROW_UP\n",
      "GSE5542_UNTREATED_VS_IFNA_AND_IFNG_TREATED_EPITHELIAL_CELLS_24H_UP\n",
      "GO_TELOMERIC_DNA_BINDING\n",
      "GSE9988_ANTI_TREM1_AND_LPS_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "YU_MYC_TARGETS_DN\n",
      "HALLMARK_COMPLEMENT\n",
      "GSE17721_POLYIC_VS_PAM3CSK4_24H_BMDC_DN\n",
      "GSE29618_MONOCYTE_VS_PDC_DN\n",
      "GNF2_CENPF\n",
      "AMUNDSON_POOR_SURVIVAL_AFTER_GAMMA_RADIATION_2G\n",
      "GSE3920_IFNA_VS_IFNG_TREATED_ENDOTHELIAL_CELL_DN\n",
      "GSE14699_DELETIONAL_TOLERANCE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "GO_EXONUCLEASE_ACTIVITY_ACTIVE_WITH_EITHER_RIBO_OR_DEOXYRIBONUCLEIC_ACIDS_AND_PRODUCING_5_PHOSPHOMONOESTERS\n",
      "GSE22935_24H_VS_48H_MBOVIS_BCG_STIM_MYD88_KO_MACROPHAGE_DN\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 17, 0.16444306662021\n",
      "best merge is 11, 13, 0.173946975341428\n",
      "best merge is 0, 20, 0.194423231958034\n",
      "best merge is 10, 25, 0.199815488432056\n",
      "best merge is 15, 22, 0.215936713498748\n",
      "best merge is 7, 21, 0.252342712448099\n",
      "best merge is 14, 16, 0.311519515959067\n",
      "best merge is 7, 24, 0.32650823496379\n",
      "best merge is 0, 9, 0.331711571836108\n",
      "best merge is 1, 6, 0.342951125602591\n",
      "best merge is 3, 15, 0.387983899520164\n",
      "best merge is 11, 18, 0.408189439042095\n",
      "best merge is 14, 19, 0.457432518437307\n",
      "best merge is 2, 5, 0.514658558428039\n",
      "best merge is 8, 12, 0.529251546759875\n",
      "best merge is 4, 10, 0.577127997699795\n",
      "best merge is 3, 14, 0.597811443110735\n",
      "best merge is 0, 2, 0.625744991104846\n",
      "best merge is 1, 4, 0.670904871674717\n",
      "best merge is 7, 23, 0.800318102046363\n",
      "best merge is 8, 11, 0.820304861130424\n",
      "best merge is 0, 7, 0.909569774751123\n",
      "order classes\n",
      "6 - 1.70180658394534\n",
      "5 - 2.38053996522474\n",
      "6 - 1.82929067951485\n",
      "4 - 2.68637970941825\n",
      "5 - 2.6796251318587\n",
      "0 17 20 9 2 5\n",
      "1 6 4 10 25\n",
      "3 15 22 14 16 19\n",
      "7 21 24 23\n",
      "8 12 11 13 18\n",
      "Done.-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_full/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 50\n",
      "cat 12, mi = 0.00397908, HALLMARK_E2F_TARGETS\n",
      "cat 10, mi = 0.00316089, HALLMARK_COMPLEMENT\n",
      "cat 26, mi = 0.00299759, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "cat 32, mi = 0.00298685, HALLMARK_MYC_TARGETS_V2\n",
      "cat 8, mi = 0.00298207, HALLMARK_CHOLESTEROL_HOMEOSTASIS\n",
      "cat 43, mi = 0.00262343, HALLMARK_TGF_BETA_SIGNALING\n",
      "cat 30, mi = 0.00258936, HALLMARK_MTORC1_SIGNALING\n",
      "cat 4, mi = 0.00257202, HALLMARK_APICAL_JUNCTION\n",
      "cat 23, mi = 0.00214900, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "cat 16, mi = 0.00202355, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 9, mi = 0.00198122, HALLMARK_COAGULATION\n",
      "cat 14, mi = 0.00176699, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 25, mi = 0.00172260, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "cat 38, mi = 0.00166469, HALLMARK_PEROXISOME\n",
      "cat 6, mi = 0.00163111, HALLMARK_APOPTOSIS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 12, mi = 0.004, pvalue = 0.000, HALLMARK_E2F_TARGETS\n",
      "Passed: cat 10, mi = 0.003, pvalue = 0.002, HALLMARK_COMPLEMENT\n",
      "Passed: cat 8, mi = 0.003, pvalue = 0.004, HALLMARK_CHOLESTEROL_HOMEOSTASIS\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.003, pvalue = 0.013, HALLMARK_TGF_BETA_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.003, pvalue = 0.015, HALLMARK_APICAL_JUNCTION\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.002, pvalue = 0.098, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.002, pvalue = 0.123, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.002, pvalue = 0.199, HALLMARK_PEROXISOME\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.144, HALLMARK_APOPTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.002, pvalue = 0.250, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 22, mi = 0.002, pvalue = 0.134, HALLMARK_IL2_STAT5_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 49, mi = 0.002, pvalue = 0.236, HALLMARK_XENOBIOTIC_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.002, pvalue = 0.318, HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.001, pvalue = 0.050, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.322, HALLMARK_KRAS_SIGNALING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.001, pvalue = 0.320, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.001, pvalue = 0.406, HALLMARK_UV_RESPONSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.001, pvalue = 0.385, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 1, mi = 0.001, pvalue = 0.381, HALLMARK_ALLOGRAFT_REJECTION\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.001, pvalue = 0.650, HALLMARK_SPERMATOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.001, pvalue = 0.729, HALLMARK_MYOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.001, pvalue = 0.833, HALLMARK_ANGIOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.001, pvalue = 0.982, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Total number of categories evaluated = 38.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.659072856423168\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "8.47\t-11.35\n",
      "HALLMARK_COMPLEMENT\n",
      "HALLMARK_E2F_TARGETS\n",
      "HALLMARK_CHOLESTEROL_HOMEOSTASIS\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.659072856423168\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-9.72250252270434\t42\t0\n",
      "1.30726538849647\t42\t20.5549949545913\n",
      "2.45156407669021\t42\t22.0595358224016\n",
      "3.12966700302724\t42\t22.8435923309788\n",
      "3.64883955600403\t42\t23.4157416750757\n",
      "4.09384460141271\t42\t23.8819374369324\n",
      "4.49646821392533\t42\t24.305751765893\n",
      "4.87790110998991\t42\t24.6871846619576\n",
      "5.29112008072654\t42\t25.0686175580222\n",
      "5.85267406659939\t42\t25.5136226034309\n",
      "14.095862764884\t42\t26.1917255297679\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_msigdb_h/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7491 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7491\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "681 hits go_terms_num = 790\n",
      "cat 402, mi = 0.00201994, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 230, mi = 0.00181280, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 31, mi = 0.00180213, AUH_K562_POSTAR_eCLIP\n",
      "cat 449, mi = 0.00177027, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 237, mi = 0.00171525, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 672, mi = 0.00170413, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 236, mi = 0.00163398, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 235, mi = 0.00160367, FUS_HeLa_POSTAR_CIMS\n",
      "cat 637, mi = 0.00154549, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 668, mi = 0.00154233, TNRC6A_HEK293_Plass\n",
      "cat 635, mi = 0.00153582, TARDBP_H9_POSTAR_CIMS\n",
      "cat 16, mi = 0.00152328, ALKBH1_HeLa_POSTAR_CIMS\n",
      "cat 676, mi = 0.00149915, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00145331, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 326, mi = 0.00141479, HNRNPU_HeLa_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.038, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.002, pvalue = 0.053, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.089, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.073, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.100, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.105, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.126, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.002, pvalue = 0.044, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.156, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.110, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.160, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.191, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.001, pvalue = 0.131, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.001, pvalue = 0.184, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.169, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.001, pvalue = 0.162, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 300, mi = 0.001, pvalue = 0.229, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 447, mi = 0.001, pvalue = 0.207, NPM1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.001, pvalue = 0.278, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.265, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.47\t-11.35\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_all_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6816 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6816\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "620 hits go_terms_num = 790\n",
      "cat 402, mi = 0.00265854, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 449, mi = 0.00236466, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 256, mi = 0.00231054, GNL3_K562_POSTAR_eCLIP\n",
      "cat 672, mi = 0.00216170, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 230, mi = 0.00213758, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 447, mi = 0.00192856, NPM1_K562_POSTAR_eCLIP\n",
      "cat 173, mi = 0.00191884, EIF4A3_HeLa_POSTAR_CIMS\n",
      "cat 237, mi = 0.00188329, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 676, mi = 0.00185196, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 31, mi = 0.00179534, AUH_K562_POSTAR_eCLIP\n",
      "cat 236, mi = 0.00179330, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 235, mi = 0.00176190, FUS_HeLa_POSTAR_CIMS\n",
      "cat 377, mi = 0.00171895, LARP7_K562_POSTAR_eCLIP\n",
      "cat 637, mi = 0.00170289, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 635, mi = 0.00169238, TARDBP_H9_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.003, pvalue = 0.011, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.024, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.002, pvalue = 0.018, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.047, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.002, pvalue = 0.034, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 447, mi = 0.002, pvalue = 0.060, NPM1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 173, mi = 0.002, pvalue = 0.062, EIF4A3_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.108, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.002, pvalue = 0.080, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.131, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.132, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.002, pvalue = 0.044, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.002, pvalue = 0.100, LARP7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.143, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.151, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.166, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 300, mi = 0.002, pvalue = 0.230, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.205, SLBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.200, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.001, pvalue = 0.268, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.83\t-11.35\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6678 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6678\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "607 hits go_terms_num = 790\n",
      "cat 437, mi = 0.00303651, NONO_K562_POSTAR_eCLIP\n",
      "cat 169, mi = 0.00280427, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 479, mi = 0.00277234, PTBP1_HEK293T,plus_SRRM4_POSTAR_Piranha_0.01\n",
      "cat 260, mi = 0.00260832, GRSF1_HepG2_POSTAR_eCLIP\n",
      "cat 648, mi = 0.00252781, TBRG4_K562_POSTAR_eCLIP\n",
      "cat 109, mi = 0.00250127, DDX3X_K562_POSTAR_eCLIP\n",
      "cat 315, mi = 0.00246183, HNRNPM_HepG2_POSTAR_eCLIP\n",
      "cat 610, mi = 0.00244364, SUGP2_HepG2_POSTAR_eCLIP\n",
      "cat 230, mi = 0.00240091, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 317, mi = 0.00239104, HNRNPM_K562_POSTAR_eCLIP\n",
      "cat 244, mi = 0.00238368, FXR1_K562_POSTAR_eCLIP\n",
      "cat 159, mi = 0.00236694, EIF3B_HEK293T_POSTAR_PAR-CLIP\n",
      "cat 599, mi = 0.00233003, SSB_HEK293_POSTAR_PAR-CLIP\n",
      "cat 190, mi = 0.00232493, EWSR1_K562_POSTAR_eCLIP\n",
      "cat 163, mi = 0.00231094, EIF3D_HEK293T_POSTAR_PARalyzer\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 437, mi = 0.003, pvalue = 0.002, NONO_K562_POSTAR_eCLIP\n",
      "Passed: cat 169, mi = 0.003, pvalue = 0.004, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.002, pvalue = 0.023, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 244, mi = 0.002, pvalue = 0.014, FXR1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.002, pvalue = 0.049, DKC1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 381, mi = 0.002, pvalue = 0.075, LIN28A_H9_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 380, mi = 0.002, pvalue = 0.099, LIN28A_H9_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 691, mi = 0.002, pvalue = 0.138, U2AF2_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 404, mi = 0.002, pvalue = 0.200, MBNL2_frontalcortex_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 627, mi = 0.002, pvalue = 0.165, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 774, mi = 0.002, pvalue = 0.250, YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.002, pvalue = 0.200, CELF2_T_cell_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.002, pvalue = 0.256, CNBP_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 506, mi = 0.002, pvalue = 0.261, QKI_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 626, mi = 0.002, pvalue = 0.295, TARBP2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 601, mi = 0.001, pvalue = 0.322, SSB_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 710, mi = 0.001, pvalue = 0.294, UPF1_HeLa,treated_with_puromycoin_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 142, mi = 0.001, pvalue = 0.360, DICER1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 687, mi = 0.001, pvalue = 0.554, U2AF1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 350, mi = 0.001, pvalue = 0.345, IGF2BP3_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 514, mi = 0.001, pvalue = 0.471, RBM10_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 379, mi = 0.001, pvalue = 0.505, LIN28A_H9_POSTAR_CIMS\n",
      "Total number of categories evaluated = 123.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.83\t-11.35\n",
      "NONO_K562_POSTAR_eCLIP\n",
      "EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.35846724351051\t42\t0\n",
      "6.16563658838071\t42\t25.283065512979\n",
      "7.54140914709518\t42\t27.0482076637824\n",
      "8.37206427688504\t42\t28.0346106304079\n",
      "8.99505562422744\t42\t28.7095179233622\n",
      "9.52719406674907\t42\t29.2805933250927\n",
      "9.99443757725587\t42\t29.7737948084054\n",
      "10.4487021013597\t42\t30.2150803461063\n",
      "10.9548825710754\t42\t30.6823238566131\n",
      "11.6427688504326\t42\t31.2274412855377\n",
      "17.0290482076638\t42\t32.0580964153276\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6602 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6602\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "600 hits go_terms_num = 790\n",
      "cat 140, mi = 0.00266079, DICER1_HEK293_POSTAR_PAR-CLIP\n",
      "cat 724, mi = 0.00262311, WDR33_HEK293_Plass\n",
      "cat 703, mi = 0.00250989, UCHL5_K562_POSTAR_eCLIP\n",
      "cat 373, mi = 0.00250106, LARP4_K562_POSTAR_eCLIP\n",
      "cat 258, mi = 0.00225129, GPKOW_K562_POSTAR_eCLIP\n",
      "cat 346, mi = 0.00222353, IGF2BP2_K562_POSTAR_eCLIP\n",
      "cat 150, mi = 0.00220145, DROSHA_K562_POSTAR_eCLIP\n",
      "cat 228, mi = 0.00219364, FUS_HEK293T_POSTAR_PAR-CLIP\n",
      "cat 229, mi = 0.00219364, FUS_HEK293T_POSTAR_PARalyzer\n",
      "cat 393, mi = 0.00215510, LIN28B_K562_POSTAR_eCLIP\n",
      "cat 743, mi = 0.00214475, YBX3_K562_POSTAR_eCLIP\n",
      "cat 44, mi = 0.00214119, CAPRIN1_HEK293_Plass\n",
      "cat 162, mi = 0.00213992, EIF3D_HEK293T_POSTAR_PAR-CLIP\n",
      "cat 608, mi = 0.00210615, SUB1_HepG2_POSTAR_eCLIP\n",
      "cat 163, mi = 0.00208135, EIF3D_HEK293T_POSTAR_PARalyzer\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.003, pvalue = 0.008, DICER1_HEK293_POSTAR_PAR-CLIP\n",
      "Passed: cat 724, mi = 0.003, pvalue = 0.001, WDR33_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 703, mi = 0.003, pvalue = 0.013, UCHL5_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 373, mi = 0.003, pvalue = 0.015, LARP4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 258, mi = 0.002, pvalue = 0.043, GPKOW_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 346, mi = 0.002, pvalue = 0.049, IGF2BP2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 150, mi = 0.002, pvalue = 0.038, DROSHA_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 228, mi = 0.002, pvalue = 0.037, FUS_HEK293T_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.002, pvalue = 0.037, FUS_HEK293T_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 393, mi = 0.002, pvalue = 0.056, LIN28B_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.002, pvalue = 0.053, YBX3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 162, mi = 0.002, pvalue = 0.042, EIF3D_HEK293T_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 608, mi = 0.002, pvalue = 0.052, SUB1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 163, mi = 0.002, pvalue = 0.052, EIF3D_HEK293T_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 459, mi = 0.002, pvalue = 0.053, PCBP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.002, pvalue = 0.053, FKBP4_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 361, mi = 0.002, pvalue = 0.056, ILF3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 482, mi = 0.002, pvalue = 0.057, PTBP1_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 408, mi = 0.002, pvalue = 0.073, METAP2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 289, mi = 0.002, pvalue = 0.054, HNRNPC_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 781, mi = 0.002, pvalue = 0.086, ZC3H7B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 64, mi = 0.002, pvalue = 0.093, CPSF2_HEK293_POSTAR_PAR-CLIP\n",
      "Total number of categories evaluated = 13.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.83\t-11.35\n",
      "WDR33_HEK293_Plass\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.37144622991347\t42\t0\n",
      "6.15265760197775\t42\t25.2571075401731\n",
      "7.52843016069221\t42\t27.0482076637824\n",
      "8.35908529048208\t42\t28.008652657602\n",
      "8.98207663782448\t42\t28.7095179233622\n",
      "9.5142150803461\t42\t29.2546353522868\n",
      "9.99443757725587\t42\t29.7737948084054\n",
      "10.4487021013597\t42\t30.2150803461063\n",
      "10.9548825710754\t42\t30.6823238566131\n",
      "11.6297898640297\t42\t31.2274412855377\n",
      "17.0160692212608\t42\t32.0321384425216\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6691 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6691\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "608 hits go_terms_num = 790\n",
      "cat 383, mi = 0.00354143, LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "cat 382, mi = 0.00346310, LIN28A_HEK293_POSTAR_CIMS\n",
      "cat 40, mi = 0.00331849, C22orf28_HEK293_Plass\n",
      "cat 386, mi = 0.00320919, LIN28A_HEK293_POSTAR_Piranha_0.01\n",
      "cat 601, mi = 0.00298232, SSB_HEK293_POSTAR_Piranha_0.01\n",
      "cat 398, mi = 0.00295900, LSM11_K562_POSTAR_eCLIP\n",
      "cat 557, mi = 0.00275081, SERBP1_K562_POSTAR_eCLIP\n",
      "cat 179, mi = 0.00274321, ELAVL1_HEK293_POSTAR_HITS-CLIP\n",
      "cat 199, mi = 0.00265556, FAM120A_K562_POSTAR_eCLIP\n",
      "cat 492, mi = 0.00262840, PUM2_HEK293_POSTAR_PAR-CLIP\n",
      "cat 8, mi = 0.00259379, AGO1_HEK293_Plass\n",
      "cat 493, mi = 0.00258554, PUM2_HEK293_POSTAR_PARalyzer\n",
      "cat 441, mi = 0.00253972, NOP56_HEK293_POSTAR_Piranha_0.01\n",
      "cat 757, mi = 0.00252465, YTHDF1_Huh7_POSTAR_PAR-CLIP\n",
      "cat 173, mi = 0.00252183, EIF4A3_HeLa_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 383, mi = 0.004, pvalue = 0.001, LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "Passed: cat 40, mi = 0.003, pvalue = 0.001, C22orf28_HEK293_Plass\n",
      "Passed: cat 398, mi = 0.003, pvalue = 0.004, LSM11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.003, pvalue = 0.006, ELAVL1_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 467, mi = 0.002, pvalue = 0.023, PRKRA_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 468, mi = 0.002, pvalue = 0.023, PRKRA_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 533, mi = 0.002, pvalue = 0.026, RBM47_MDA-MB-231_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.002, pvalue = 0.031, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 359, mi = 0.002, pvalue = 0.035, ILF3_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 223, mi = 0.002, pvalue = 0.056, FUBP3_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 532, mi = 0.002, pvalue = 0.051, RBM47_MDA-MB-231_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 514, mi = 0.002, pvalue = 0.079, RBM10_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.002, pvalue = 0.062, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.002, pvalue = 0.099, DHX30_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 669, mi = 0.002, pvalue = 0.111, TNRC6A_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.173, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 327, mi = 0.002, pvalue = 0.126, HNRNPU_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.002, pvalue = 0.140, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 730, mi = 0.002, pvalue = 0.283, WTAP_HeLa_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 328, mi = 0.001, pvalue = 0.208, HNRNPU_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.307, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 178, mi = 0.001, pvalue = 0.315, ELAVL1_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 299, mi = 0.001, pvalue = 0.293, HNRNPD_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 123.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.89318228236706\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "4.83\t-11.35\n",
      "LSM11_K562_POSTAR_eCLIP\n",
      "LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "C22orf28_HEK293_Plass\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.89318228236706\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-7.35846724351051\t42\t0\n",
      "6.16563658838071\t42\t25.283065512979\n",
      "7.54140914709518\t42\t27.0482076637824\n",
      "8.37206427688504\t42\t28.0346106304079\n",
      "8.99505562422744\t42\t28.7095179233622\n",
      "9.52719406674907\t42\t29.2805933250927\n",
      "9.99443757725587\t42\t29.7737948084054\n",
      "10.4487021013597\t42\t30.2150803461063\n",
      "10.9548825710754\t42\t30.6823238566131\n",
      "11.6427688504326\t42\t31.2274412855377\n",
      "17.0290482076638\t42\t32.0580964153276\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 790\n",
      "cat 256, mi = 0.00282465, GNL3_K562_POSTAR_eCLIP\n",
      "cat 126, mi = 0.00222308, DGCR8_HEK293T_POSTAR_CIMS\n",
      "cat 402, mi = 0.00213142, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 635, mi = 0.00197425, TARDBP_H9_POSTAR_CIMS\n",
      "cat 637, mi = 0.00197425, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 31, mi = 0.00196825, AUH_K562_POSTAR_eCLIP\n",
      "cat 447, mi = 0.00192409, NPM1_K562_POSTAR_eCLIP\n",
      "cat 484, mi = 0.00185923, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 705, mi = 0.00185463, UPF1_HEK293_Plass\n",
      "cat 235, mi = 0.00185419, FUS_HeLa_POSTAR_CIMS\n",
      "cat 321, mi = 0.00181584, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 136, mi = 0.00177498, DHX30_HepG2_POSTAR_eCLIP\n",
      "cat 236, mi = 0.00173842, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 672, mi = 0.00173640, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 237, mi = 0.00169650, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.003, pvalue = 0.007, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.002, pvalue = 0.035, DGCR8_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.065, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.115, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.115, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.106, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 447, mi = 0.002, pvalue = 0.088, NPM1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.102, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 705, mi = 0.002, pvalue = 0.127, UPF1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.002, pvalue = 0.035, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.139, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.002, pvalue = 0.118, DHX30_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.173, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.181, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.191, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 57, mi = 0.002, pvalue = 0.181, CPSF-100_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.002, pvalue = 0.178, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.184, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 329, mi = 0.002, pvalue = 0.218, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 173, mi = 0.001, pvalue = 0.216, EIF4A3_HeLa_POSTAR_CIMS\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.83\t-11.35\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "read_go_names_data: please enter a valid filename (/flash/hani/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_coding_gene_names/human_ensembl_RBPs_coding_gene_names_names.txt invalid)\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 101\n",
      "cat 27, mi = 0.00226776, HuR.2\n",
      "cat 68, mi = 0.00214114, RBMS3\n",
      "cat 9, mi = 0.00210654, ESRP2\n",
      "cat 56, mi = 0.00172726, RBM3\n",
      "cat 21, mi = 0.00132324, HNRNPK\n",
      "cat 13, mi = 0.00120929, FXR2\n",
      "cat 88, mi = 0.00102816, SRSF9.1\n",
      "cat 6, mi = 0.00097716, CPEB4\n",
      "cat 62, mi = 0.00092791, RBM46\n",
      "cat 34, mi = 0.00071213, KHDRBS3\n",
      "cat 69, mi = 0.00027940, RBMS3.1\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "cat 3, mi = 0.00000000, BRUNOL6\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.002, pvalue = 0.046, HuR.2\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.002, pvalue = 0.059, RBMS3\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.053, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.002, pvalue = 0.164, RBM3\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.424, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.422, FXR2\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.001, pvalue = 0.575, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.634, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.792, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.921, KHDRBS3\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.000, pvalue = 0.995, RBMS3.1\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, FMR1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.83\t-11.35\n",
      "Outputing EPS file deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/METTL3_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/METTL3_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/METTL3_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/METTL3_peak_m6A_vs_input_human_ensembl_RBPs_DeepBind/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7509 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7509\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "683 hits go_terms_num = 17689\n",
      "cat 3498, mi = 0.00303104, p53 binding\n",
      "cat 14, mi = 0.00294687, response to organic cyclic compound\n",
      "cat 2491, mi = 0.00277220, retina development in camera-type eye\n",
      "cat 373, mi = 0.00270389, skeletal muscle cell differentiation\n",
      "cat 975, mi = 0.00270049, negative regulation of autophagy\n",
      "cat 1609, mi = 0.00268759, positive regulation of multicellular organism growth\n",
      "cat 84, mi = 0.00268614, animal organ regeneration\n",
      "cat 724, mi = 0.00266267, RNA polymerase II activating transcription factor binding\n",
      "cat 658, mi = 0.00265482, Notch signaling pathway\n",
      "cat 3071, mi = 0.00259754, cytokine production\n",
      "cat 2433, mi = 0.00258699, histone acetyltransferase activity\n",
      "cat 3493, mi = 0.00258301, negative regulation of T cell activation\n",
      "cat 497, mi = 0.00256153, palate development\n",
      "cat 7102, mi = 0.00253501, sprouting angiogenesis\n",
      "cat 111, mi = 0.00252672, cell development\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 3498, mi = 0.003, pvalue = 0.001, p53 binding\n",
      "Passed: cat 14, mi = 0.003, pvalue = 0.002, response to organic cyclic compound\n",
      "Passed: cat 2491, mi = 0.003, pvalue = 0.002, retina development in camera-type eye\n",
      "Passed: cat 975, mi = 0.003, pvalue = 0.004, negative regulation of autophagy\n",
      "Passed: cat 3071, mi = 0.003, pvalue = 0.003, cytokine production\n",
      "Passed: cat 3493, mi = 0.003, pvalue = 0.000, negative regulation of T cell activation\n",
      "Passed: cat 7102, mi = 0.003, pvalue = 0.003, sprouting angiogenesis\n",
      "Passed: cat 2540, mi = 0.002, pvalue = 0.004, rRNA binding\n",
      "Didn't pass (cutoff=0.0050): cat 1872, mi = 0.002, pvalue = 0.010, spliceosomal complex\n",
      "Didn't pass (cutoff=0.0050): cat 2187, mi = 0.002, pvalue = 0.008, positive regulation of protein ubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 826, mi = 0.002, pvalue = 0.005, positive regulation of macroautophagy\n",
      "Didn't pass (cutoff=0.0050): cat 551, mi = 0.002, pvalue = 0.008, chromatin-mediated maintenance of transcription\n",
      "Didn't pass (cutoff=0.0050): cat 1700, mi = 0.002, pvalue = 0.010, Golgi organization\n",
      "Didn't pass (cutoff=0.0050): cat 628, mi = 0.002, pvalue = 0.006, voltage-gated ion channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 2141, mi = 0.002, pvalue = 0.013, chondroitin sulfate biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 1876, mi = 0.002, pvalue = 0.013, catalytic step 2 spliceosome\n",
      "Didn't pass (cutoff=0.0050): cat 516, mi = 0.002, pvalue = 0.012, lipid catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.018, glucose homeostasis\n",
      "Didn't pass (cutoff=0.0050): cat 1774, mi = 0.002, pvalue = 0.010, signal transducer activity, downstream of receptor\n",
      "Didn't pass (cutoff=0.0050): cat 6363, mi = 0.002, pvalue = 0.011, positive regulation of glucose import\n",
      "Didn't pass (cutoff=0.0050): cat 7300, mi = 0.002, pvalue = 0.011, SWI/SNF complex\n",
      "Didn't pass (cutoff=0.0050): cat 7302, mi = 0.002, pvalue = 0.009, nBAF complex\n",
      "Passed: cat 2280, mi = 0.002, pvalue = 0.003, phosphatidylinositol-4,5-bisphosphate 5-phosphatase activity\n",
      "Passed: cat 2480, mi = 0.002, pvalue = 0.005, GPI anchor biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 1684, mi = 0.002, pvalue = 0.020, costamere\n",
      "Passed: cat 6753, mi = 0.002, pvalue = 0.004, regulation of skeletal muscle satellite cell proliferation\n",
      "Didn't pass (cutoff=0.0050): cat 492, mi = 0.002, pvalue = 0.008, embryonic skeletal system morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 2420, mi = 0.002, pvalue = 0.014, keratan sulfate biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2027, mi = 0.002, pvalue = 0.027, thyroid hormone receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.002, pvalue = 0.028, response to stress\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.022, regulation of ion transmembrane transport\n",
      "Passed: cat 7169, mi = 0.002, pvalue = 0.004, dystrophin-associated glycoprotein complex\n",
      "Didn't pass (cutoff=0.0050): cat 1777, mi = 0.002, pvalue = 0.006, mitochondrial respiratory chain complex IV assembly\n",
      "Didn't pass (cutoff=0.0050): cat 2879, mi = 0.002, pvalue = 0.009, cortical cytoskeleton\n",
      "Didn't pass (cutoff=0.0050): cat 599, mi = 0.002, pvalue = 0.035, positive regulation of JNK cascade\n",
      "Didn't pass (cutoff=0.0050): cat 3158, mi = 0.002, pvalue = 0.006, Golgi transport complex\n",
      "Didn't pass (cutoff=0.0050): cat 539, mi = 0.002, pvalue = 0.022, positive regulation of receptor activity\n",
      "Didn't pass (cutoff=0.0050): cat 3179, mi = 0.002, pvalue = 0.029, protein monoubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 792, mi = 0.002, pvalue = 0.011, anion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 7809, mi = 0.002, pvalue = 0.012, ion channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 8241, mi = 0.002, pvalue = 0.010, volume-sensitive anion channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 557, mi = 0.002, pvalue = 0.028, positive regulation of T cell activation\n",
      "Didn't pass (cutoff=0.0050): cat 3523, mi = 0.002, pvalue = 0.037, dendritic spine development\n",
      "Passed: cat 1628, mi = 0.002, pvalue = 0.001, nucleotide metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 10198, mi = 0.002, pvalue = 0.028, nuclear export\n",
      "Didn't pass (cutoff=0.0050): cat 1249, mi = 0.002, pvalue = 0.024, small-subunit processome\n",
      "Passed: cat 2790, mi = 0.002, pvalue = 0.002, GTPase regulator activity\n",
      "Didn't pass (cutoff=0.0050): cat 1432, mi = 0.002, pvalue = 0.044, 14-3-3 protein binding\n",
      "Didn't pass (cutoff=0.0050): cat 2039, mi = 0.002, pvalue = 0.043, Golgi to plasma membrane protein transport\n",
      "Passed: cat 3188, mi = 0.002, pvalue = 0.002, branched-chain amino acid catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 3651, mi = 0.002, pvalue = 0.048, insulin receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 4297, mi = 0.002, pvalue = 0.007, peptidyl-serine dephosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 3886, mi = 0.002, pvalue = 0.052, regulation of cell morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 5283, mi = 0.002, pvalue = 0.029, protein localization to cilium\n",
      "Didn't pass (cutoff=0.0050): cat 661, mi = 0.002, pvalue = 0.050, cartilage development\n",
      "Didn't pass (cutoff=0.0050): cat 1940, mi = 0.002, pvalue = 0.056, regulation of RNA splicing\n",
      "Didn't pass (cutoff=0.0050): cat 2135, mi = 0.002, pvalue = 0.055, negative regulation of type I interferon production\n",
      "Didn't pass (cutoff=0.0050): cat 1443, mi = 0.002, pvalue = 0.029, negative regulation of protein complex assembly\n",
      "Didn't pass (cutoff=0.0050): cat 13890, mi = 0.002, pvalue = 0.020, dendritic cell dendrite assembly\n",
      "Didn't pass (cutoff=0.0050): cat 3013, mi = 0.002, pvalue = 0.018, glucuronosyl-N-acetylgalactosaminyl-proteoglycan 4-beta-N-acetylgalactosaminyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1307, mi = 0.002, pvalue = 0.024, positive regulation of T cell differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 2536, mi = 0.002, pvalue = 0.027, embryo development\n",
      "Didn't pass (cutoff=0.0050): cat 3061, mi = 0.002, pvalue = 0.026, viral nucleocapsid\n",
      "Didn't pass (cutoff=0.0050): cat 13636, mi = 0.002, pvalue = 0.026, beta-galactoside (CMP) alpha-2,3-sialyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 6170, mi = 0.002, pvalue = 0.057, positive regulation of fatty acid beta-oxidation\n",
      "Didn't pass (cutoff=0.0050): cat 2361, mi = 0.002, pvalue = 0.009, wound healing, spreading of cells\n",
      "Didn't pass (cutoff=0.0050): cat 3799, mi = 0.002, pvalue = 0.058, termination of RNA polymerase II transcription\n",
      "Didn't pass (cutoff=0.0050): cat 875, mi = 0.002, pvalue = 0.040, positive regulation of cytosolic calcium ion concentration involved in phospholipase C-activating G-protein coupled signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 5181, mi = 0.002, pvalue = 0.033, synaptic vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 8984, mi = 0.002, pvalue = 0.062, cell-cell contact zone\n",
      "Total number of categories evaluated = 156.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 5, 6, 0.250788947465718\n",
      "best merge is 12, 14, 0.273989382329842\n",
      "best merge is 10, 11, 0.403356273869248\n",
      "best merge is 0, 7, 0.419614648033758\n",
      "best merge is 3, 5, 0.45459735532245\n",
      "best merge is 10, 12, 0.468538007747334\n",
      "best merge is 9, 13, 0.509479303352954\n",
      "best merge is 0, 9, 0.680441285661941\n",
      "best merge is 1, 4, 0.786279862218028\n",
      "best merge is 1, 3, 0.848392386906295\n",
      "best merge is 2, 10, 0.92523962363074\n",
      "order classes\n",
      "4 - 1.71134020032866\n",
      "5 - 1.68878828202394\n",
      "4 - 1.19601454086958\n",
      "10 11 12 14\n",
      "1 4 3 5 6\n",
      "2\n",
      "8\n",
      "0 7 9 13\n",
      "Done.Start drawing\n",
      "7.06\t-12.28\n",
      "regulation of skeletal muscle satellite cell proliferation, GO:0014842\n",
      "dystrophin-associated glycoprotein complex, GO:0016010\n",
      "nucleotide metabolic process, GO:0009117\n",
      "branched-chain amino acid catabolic process, GO:0009083\n",
      "retina development in camera-type eye, GO:0060041\n",
      "sprouting angiogenesis, GO:0002040\n",
      "response to organic cyclic compound, GO:0014070\n",
      "cytokine production, GO:0001816\n",
      "negative regulation of autophagy, GO:0010507\n",
      "negative regulation of T cell activation, GO:0050868\n",
      "p53 binding, GO:0002039\n",
      "rRNA binding, GO:0019843\n",
      "GPI anchor biosynthetic process, GO:0006506\n",
      "GTPase regulator activity, GO:0030695\n",
      "phosphatidylinositol-4,5-bisphosphate 5-phosphatase activity, GO:0004439\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 5, 6, 0.250788947465718\n",
      "best merge is 12, 14, 0.273989382329842\n",
      "best merge is 10, 11, 0.403356273869248\n",
      "best merge is 0, 7, 0.419614648033758\n",
      "best merge is 3, 5, 0.45459735532245\n",
      "best merge is 10, 12, 0.468538007747334\n",
      "best merge is 9, 13, 0.509479303352954\n",
      "best merge is 0, 9, 0.680441285661941\n",
      "best merge is 1, 4, 0.786279862218028\n",
      "best merge is 1, 3, 0.848392386906295\n",
      "best merge is 2, 10, 0.92523962363074\n",
      "order classes\n",
      "4 - 1.71134020032866\n",
      "5 - 1.68878828202394\n",
      "4 - 1.19601454086958\n",
      "10 11 12 14\n",
      "1 4 3 5 6\n",
      "2\n",
      "8\n",
      "0 7 9 13\n",
      "Done.-8.0124095139607\t42\t0\n",
      "4.57238883143743\t42\t23.9751809720786\n",
      "5.49534643226473\t42\t25.1695966907963\n",
      "6.0382626680455\t42\t25.8210961737332\n",
      "6.41830403309204\t42\t26.2554291623578\n",
      "6.71147880041365\t42\t26.5811789038263\n",
      "6.98293691830403\t42\t26.841778697001\n",
      "7.26525336091003\t42\t27.124095139607\n",
      "7.6018614270941\t42\t27.406411582213\n",
      "8.11220268872802\t42\t27.7973112719752\n",
      "15.2135470527404\t42\t28.4270941054809\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6496 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6496\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "591 hits go_terms_num = 181\n",
      "cat 131, mi = 0.00175428, SMARCA4_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00144312, PPARGC1A_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00144123, POLR3G_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00143505, BRF2_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00135905, POU5F1_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00105920, BRF1_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00075718, XRCC4_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00070414, BDP1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.002, pvalue = 0.173, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.305, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.382, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.256, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.001, pvalue = 0.300, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.672, BRF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 167, mi = 0.001, pvalue = 1.000, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.833, BDP1_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.05\t-12.28\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_encode_tf/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 326\n",
      "cat 259, mi = 0.00271236, chr6q25\n",
      "cat 114, mi = 0.00270186, chr1p22\n",
      "cat 307, mi = 0.00251380, chr9q33\n",
      "cat 126, mi = 0.00247843, chr1q24\n",
      "cat 24, mi = 0.00210594, chr11q21\n",
      "cat 41, mi = 0.00209210, chr12q23\n",
      "cat 171, mi = 0.00207739, chr2q33\n",
      "cat 129, mi = 0.00206854, chr1q32\n",
      "cat 34, mi = 0.00204554, chr12q11\n",
      "cat 282, mi = 0.00198547, chr8p21\n",
      "cat 143, mi = 0.00194082, chr21q22\n",
      "cat 28, mi = 0.00192398, chr11q25\n",
      "cat 146, mi = 0.00190348, chr22q12\n",
      "cat 60, mi = 0.00189301, chr14q24\n",
      "cat 6, mi = 0.00183424, chr10q11\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 259, mi = 0.003, pvalue = 0.009, chr6q25\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.003, pvalue = 0.010, chr1p22\n",
      "Didn't pass (cutoff=0.0050): cat 307, mi = 0.003, pvalue = 0.015, chr9q33\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.002, pvalue = 0.019, chr1q24\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.002, pvalue = 0.038, chr11q21\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.002, pvalue = 0.043, chr12q23\n",
      "Didn't pass (cutoff=0.0050): cat 171, mi = 0.002, pvalue = 0.063, chr2q33\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.002, pvalue = 0.059, chr1q32\n",
      "Passed: cat 34, mi = 0.002, pvalue = 0.004, chr12q11\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.002, pvalue = 0.085, chr8p21\n",
      "Didn't pass (cutoff=0.0050): cat 143, mi = 0.002, pvalue = 0.072, chr21q22\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.002, pvalue = 0.077, chr11q25\n",
      "Didn't pass (cutoff=0.0050): cat 146, mi = 0.002, pvalue = 0.088, chr22q12\n",
      "Didn't pass (cutoff=0.0050): cat 60, mi = 0.002, pvalue = 0.095, chr14q24\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.133, chr10q11\n",
      "Didn't pass (cutoff=0.0050): cat 8, mi = 0.002, pvalue = 0.143, chr10q22\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.002, pvalue = 0.106, chr1q42\n",
      "Didn't pass (cutoff=0.0050): cat 82, mi = 0.002, pvalue = 0.084, chr16q22\n",
      "Didn't pass (cutoff=0.0050): cat 223, mi = 0.002, pvalue = 0.084, chr5p12\n",
      "Didn't pass (cutoff=0.0050): cat 243, mi = 0.002, pvalue = 0.102, chr6p12\n",
      "Didn't pass (cutoff=0.0050): cat 35, mi = 0.002, pvalue = 0.102, chr12q12\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.002, pvalue = 0.130, chr1q22\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.002, pvalue = 0.148, chr3p22\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.002, pvalue = 0.100, chr22q13\n",
      "Didn't pass (cutoff=0.0050): cat 218, mi = 0.002, pvalue = 0.145, chr4q31\n",
      "Didn't pass (cutoff=0.0050): cat 70, mi = 0.002, pvalue = 0.168, chr15q23\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.002, pvalue = 0.154, chr17q12\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.124, chr5q31\n",
      "Didn't pass (cutoff=0.0050): cat 89, mi = 0.002, pvalue = 0.133, chr17q11\n",
      "Total number of categories evaluated = 19.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "7.06\t-12.28\n",
      "chr12q11\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_c1/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4738\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t4738\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 4738\n",
      "cat 2183, mi = 0.00438938, MARSON_FOXP3_TARGETS_UP\n",
      "cat 2730, mi = 0.00413907, PID_AP1_PATHWAY\n",
      "cat 1304, mi = 0.00406529, HENDRICKS_SMARCA4_TARGETS_UP\n",
      "cat 3986, mi = 0.00402419, SIMBULAN_UV_RESPONSE_NORMAL_DN\n",
      "cat 1945, mi = 0.00401270, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "cat 2849, mi = 0.00352582, PID_NFAT_TFPATHWAY\n",
      "cat 1326, mi = 0.00352132, HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "cat 4192, mi = 0.00348682, TONKS_TARGETS_OF_RUNX1_RUNX1T1_FUSION_ERYTHROCYTE_UP\n",
      "cat 959, mi = 0.00346959, FERRANDO_T_ALL_WITH_MLL_ENL_FUSION_UP\n",
      "cat 1591, mi = 0.00336920, KEGG_AMYOTROPHIC_LATERAL_SCLEROSIS_ALS\n",
      "cat 1143, mi = 0.00334134, GHANDHI_DIRECT_IRRADIATION_UP\n",
      "cat 3972, mi = 0.00331378, SIG_BCR_SIGNALING_PATHWAY\n",
      "cat 2391, mi = 0.00328265, MORI_LARGE_PRE_BII_LYMPHOCYTE_UP\n",
      "cat 4405, mi = 0.00323881, WATANABE_RECTAL_CANCER_RADIOTHERAPY_RESPONSIVE_UP\n",
      "cat 1673, mi = 0.00317622, KEGG_LONG_TERM_POTENTIATION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2183, mi = 0.004, pvalue = 0.000, MARSON_FOXP3_TARGETS_UP\n",
      "Passed: cat 2730, mi = 0.004, pvalue = 0.000, PID_AP1_PATHWAY\n",
      "Passed: cat 3986, mi = 0.004, pvalue = 0.000, SIMBULAN_UV_RESPONSE_NORMAL_DN\n",
      "Passed: cat 1945, mi = 0.004, pvalue = 0.000, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "Passed: cat 1326, mi = 0.004, pvalue = 0.001, HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "Passed: cat 1591, mi = 0.003, pvalue = 0.000, KEGG_AMYOTROPHIC_LATERAL_SCLEROSIS_ALS\n",
      "Passed: cat 1143, mi = 0.003, pvalue = 0.001, GHANDHI_DIRECT_IRRADIATION_UP\n",
      "Passed: cat 2391, mi = 0.003, pvalue = 0.001, MORI_LARGE_PRE_BII_LYMPHOCYTE_UP\n",
      "Passed: cat 4582, mi = 0.003, pvalue = 0.001, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_13\n",
      "Passed: cat 144, mi = 0.003, pvalue = 0.003, BENPORATH_ES_CORE_NINE_CORRELATED\n",
      "Passed: cat 641, mi = 0.003, pvalue = 0.003, CHENG_IMPRINTED_BY_ESTRADIOL\n",
      "Didn't pass (cutoff=0.0050): cat 2823, mi = 0.003, pvalue = 0.008, PID_INSULIN_PATHWAY\n",
      "Passed: cat 1488, mi = 0.003, pvalue = 0.004, JAZAERI_BREAST_CANCER_BRCA1_VS_BRCA2_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3363, mi = 0.003, pvalue = 0.007, REACTOME_MRNA_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 3115, mi = 0.003, pvalue = 0.007, REACTOME_CHONDROITIN_SULFATE_BIOSYNTHESIS\n",
      "Didn't pass (cutoff=0.0050): cat 591, mi = 0.003, pvalue = 0.010, CAMPS_COLON_CANCER_COPY_NUMBER_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3751, mi = 0.003, pvalue = 0.009, ROSS_AML_WITH_PML_RARA_FUSION\n",
      "Didn't pass (cutoff=0.0050): cat 1340, mi = 0.003, pvalue = 0.010, HOFFMANN_SMALL_PRE_BII_TO_IMMATURE_B_LYMPHOCYTE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1576, mi = 0.003, pvalue = 0.012, KAYO_CALORIE_RESTRICTION_MUSCLE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3359, mi = 0.003, pvalue = 0.008, REACTOME_MRNA_3_END_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 4627, mi = 0.003, pvalue = 0.006, ZEMBUTSU_SENSITIVITY_TO_NIMUSTINE\n",
      "Didn't pass (cutoff=0.0050): cat 3466, mi = 0.003, pvalue = 0.012, REACTOME_PROCESSING_OF_CAPPED_INTRON_CONTAINING_PRE_MRNA\n",
      "Didn't pass (cutoff=0.0050): cat 3057, mi = 0.003, pvalue = 0.009, REACTOME_ANTIGEN_ACTIVATES_B_CELL_RECEPTOR_LEADING_TO_GENERATION_OF_SECOND_MESSENGERS\n",
      "Didn't pass (cutoff=0.0050): cat 3662, mi = 0.003, pvalue = 0.013, REACTOME_TRANSPORT_OF_MATURE_TRANSCRIPT_TO_CYTOPLASM\n",
      "Didn't pass (cutoff=0.0050): cat 418, mi = 0.003, pvalue = 0.008, BOGNI_TREATMENT_RELATED_MYELOID_LEUKEMIA_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1675, mi = 0.002, pvalue = 0.020, KEGG_LYSOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 247, mi = 0.002, pvalue = 0.007, BIOCARTA_CTLA4_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 3809, mi = 0.002, pvalue = 0.012, SARTIPY_BLUNTED_BY_INSULIN_RESISTANCE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3867, mi = 0.002, pvalue = 0.018, SCHLOSSER_SERUM_RESPONSE_AUGMENTED_BY_MYC\n",
      "Didn't pass (cutoff=0.0050): cat 937, mi = 0.002, pvalue = 0.021, FAELT_B_CLL_WITH_VH3_21_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1760, mi = 0.002, pvalue = 0.021, KEGG_VASCULAR_SMOOTH_MUSCLE_CONTRACTION\n",
      "Passed: cat 106, mi = 0.002, pvalue = 0.002, BANDRES_RESPONSE_TO_CARMUSTIN_MGMT_24HR_DN\n",
      "Passed: cat 1333, mi = 0.002, pvalue = 0.000, HOFFMANN_IMMATURE_TO_MATURE_B_LYMPHOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 758, mi = 0.002, pvalue = 0.007, CROONQUIST_STROMAL_STIMULATION_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3319, mi = 0.002, pvalue = 0.012, REACTOME_KERATAN_SULFATE_KERATIN_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 3304, mi = 0.002, pvalue = 0.027, REACTOME_INTERFERON_ALPHA_BETA_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 1647, mi = 0.002, pvalue = 0.032, KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE\n",
      "Didn't pass (cutoff=0.0050): cat 4535, mi = 0.002, pvalue = 0.033, XU_HGF_TARGETS_REPRESSED_BY_AKT1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3126, mi = 0.002, pvalue = 0.027, REACTOME_CLEAVAGE_OF_GROWING_TRANSCRIPT_IN_THE_TERMINATION_REGION_\n",
      "Didn't pass (cutoff=0.0050): cat 1867, mi = 0.002, pvalue = 0.032, KUMAR_AUTOPHAGY_NETWORK\n",
      "Didn't pass (cutoff=0.0050): cat 2034, mi = 0.002, pvalue = 0.034, LINDVALL_IMMORTALIZED_BY_TERT_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2707, mi = 0.002, pvalue = 0.018, PETRETTO_LEFT_VENTRICLE_MASS_QTL_CIS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.021, ABE_VEGFA_TARGETS_2HR\n",
      "Didn't pass (cutoff=0.0050): cat 1790, mi = 0.002, pvalue = 0.033, KIM_MYCN_AMPLIFICATION_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2571, mi = 0.002, pvalue = 0.042, NOUZOVA_METHYLATED_IN_APL\n",
      "Didn't pass (cutoff=0.0050): cat 2394, mi = 0.002, pvalue = 0.036, MORI_PLASMA_CELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1585, mi = 0.002, pvalue = 0.037, KEGG_ALDOSTERONE_REGULATED_SODIUM_REABSORPTION\n",
      "Didn't pass (cutoff=0.0050): cat 2320, mi = 0.002, pvalue = 0.029, MIKKELSEN_MCV6_ICP_WITH_H3K27ME3\n",
      "Didn't pass (cutoff=0.0050): cat 2597, mi = 0.002, pvalue = 0.032, OKAWA_NEUROBLASTOMA_1P36_31_DELETION\n",
      "Didn't pass (cutoff=0.0050): cat 4453, mi = 0.002, pvalue = 0.032, WHITE_NEUROBLASTOMA_WITH_1P36.3_DELETION\n",
      "Didn't pass (cutoff=0.0050): cat 3508, mi = 0.002, pvalue = 0.027, REACTOME_REGULATION_OF_MRNA_STABILITY_BY_PROTEINS_THAT_BIND_AU_RICH_ELEMENTS\n",
      "Didn't pass (cutoff=0.0050): cat 360, mi = 0.002, pvalue = 0.021, BIOCARTA_PS1_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 1408, mi = 0.002, pvalue = 0.026, HUNSBERGER_EXERCISE_REGULATED_GENES\n",
      "Total number of categories evaluated = 176.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.120344249801738\n",
      "best merge is 0, 12, 0.162449734538551\n",
      "best merge is 1, 3, 0.172532091435468\n",
      "best merge is 10, 11, 0.26263727148787\n",
      "best merge is 0, 6, 0.367247468676511\n",
      "best merge is 0, 4, 0.41443424944214\n",
      "best merge is 1, 9, 0.560161546092782\n",
      "best merge is 1, 7, 0.636440148612688\n",
      "best merge is 0, 1, 0.789834382219659\n",
      "best merge is 0, 13, 0.983374688227507\n",
      "order classes\n",
      "9 - 0.852812646231574\n",
      "2 - 0.919230450207545\n",
      "0 2 12 6 4 1 3 9 7\n",
      "5\n",
      "8\n",
      "10 11\n",
      "13\n",
      "Done.Start drawing\n",
      "7.06\t-12.28\n",
      "YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_13\n",
      "PID_AP1_PATHWAY\n",
      "LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "BENPORATH_ES_CORE_NINE_CORRELATED\n",
      "MORI_LARGE_PRE_BII_LYMPHOCYTE_UP\n",
      "MARSON_FOXP3_TARGETS_UP\n",
      "SIMBULAN_UV_RESPONSE_NORMAL_DN\n",
      "BANDRES_RESPONSE_TO_CARMUSTIN_MGMT_24HR_DN\n",
      "GHANDHI_DIRECT_IRRADIATION_UP\n",
      "HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "HOFFMANN_IMMATURE_TO_MATURE_B_LYMPHOCYTE_DN\n",
      "KEGG_AMYOTROPHIC_LATERAL_SCLEROSIS_ALS\n",
      "CHENG_IMPRINTED_BY_ESTRADIOL\n",
      "JAZAERI_BREAST_CANCER_BRCA1_VS_BRCA2_UP\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.120344249801738\n",
      "best merge is 0, 12, 0.162449734538551\n",
      "best merge is 1, 3, 0.172532091435468\n",
      "best merge is 10, 11, 0.26263727148787\n",
      "best merge is 0, 6, 0.367247468676511\n",
      "best merge is 0, 4, 0.41443424944214\n",
      "best merge is 1, 9, 0.560161546092782\n",
      "best merge is 1, 7, 0.636440148612688\n",
      "best merge is 0, 1, 0.789834382219659\n",
      "best merge is 0, 13, 0.983374688227507\n",
      "order classes\n",
      "9 - 0.852812646231574\n",
      "2 - 0.919230450207545\n",
      "0 2 12 6 4 1 3 9 7\n",
      "5\n",
      "8\n",
      "10 11\n",
      "13\n",
      "Done.-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_c2/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 836\n",
      "cat 428, mi = 0.00354764, GTGGTGA_MIR197\n",
      "cat 761, mi = 0.00316605, TGTGTGA_MIR377\n",
      "cat 443, mi = 0.00294858, HFH3_01\n",
      "cat 34, mi = 0.00292260, ACCAATC_MIR509\n",
      "cat 588, mi = 0.00277526, PAX8_01\n",
      "cat 713, mi = 0.00275132, TCCAGAT_MIR5165P\n",
      "cat 276, mi = 0.00257308, E2F_01\n",
      "cat 190, mi = 0.00256266, CDPCR3_01\n",
      "cat 351, mi = 0.00242230, GATGKMRGCG_UNKNOWN\n",
      "cat 242, mi = 0.00237316, CTCCAAG_MIR432\n",
      "cat 222, mi = 0.00235827, CREBP1_01\n",
      "cat 243, mi = 0.00234315, CTCNANGTGNY_UNKNOWN\n",
      "cat 817, mi = 0.00233646, YNTTTNNNANGCARM_UNKNOWN\n",
      "cat 332, mi = 0.00231498, GACAATC_MIR219\n",
      "cat 381, mi = 0.00230569, GGCAGTG_MIR3243P\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 428, mi = 0.004, pvalue = 0.000, GTGGTGA_MIR197\n",
      "Passed: cat 761, mi = 0.003, pvalue = 0.001, TGTGTGA_MIR377\n",
      "Passed: cat 34, mi = 0.003, pvalue = 0.003, ACCAATC_MIR509\n",
      "Passed: cat 588, mi = 0.003, pvalue = 0.003, PAX8_01\n",
      "Didn't pass (cutoff=0.0050): cat 276, mi = 0.003, pvalue = 0.013, E2F_01\n",
      "Didn't pass (cutoff=0.0050): cat 190, mi = 0.003, pvalue = 0.005, CDPCR3_01\n",
      "Didn't pass (cutoff=0.0050): cat 243, mi = 0.002, pvalue = 0.029, CTCNANGTGNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 817, mi = 0.002, pvalue = 0.025, YNTTTNNNANGCARM_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 381, mi = 0.002, pvalue = 0.029, GGCAGTG_MIR3243P\n",
      "Didn't pass (cutoff=0.0050): cat 173, mi = 0.002, pvalue = 0.043, CCANNAGRKGGC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 773, mi = 0.002, pvalue = 0.038, TTCNRGNNNNTTC_HSF_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.002, pvalue = 0.040, ATGTCAC_MIR489\n",
      "Didn't pass (cutoff=0.0050): cat 825, mi = 0.002, pvalue = 0.054, YWATTWNNRGCT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 244, mi = 0.002, pvalue = 0.060, CTCTAGA_MIR526C_MIR518F_MIR526A\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.002, pvalue = 0.009, ACCGAGC_MIR423\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.002, pvalue = 0.070, ATCMNTCCGY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.002, pvalue = 0.088, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.002, pvalue = 0.058, ACTWSNACTNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 602, mi = 0.002, pvalue = 0.093, PPARG_01\n",
      "Didn't pass (cutoff=0.0050): cat 191, mi = 0.002, pvalue = 0.088, CDP_01\n",
      "Didn't pass (cutoff=0.0050): cat 106, mi = 0.002, pvalue = 0.108, AR_03\n",
      "Didn't pass (cutoff=0.0050): cat 723, mi = 0.002, pvalue = 0.116, TCTAGAG_MIR517\n",
      "Didn't pass (cutoff=0.0050): cat 633, mi = 0.002, pvalue = 0.135, RYAAAKNNNNNNTTGW_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 457, mi = 0.002, pvalue = 0.091, HNF3_Q6\n",
      "Total number of categories evaluated = 46.\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.256708867575889\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "7.06\t-12.28\n",
      "GTGGTGA_MIR197\n",
      "TGTGTGA_MIR377\n",
      "ACCAATC_MIR509\n",
      "PAX8_01\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.256708867575889\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_c3/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 858\n",
      "cat 92, mi = 0.00353225, GNF2_ATM\n",
      "cat 563, mi = 0.00317393, MODULE_484\n",
      "cat 190, mi = 0.00314977, GNF2_PTPRC\n",
      "cat 211, mi = 0.00305049, GNF2_SNRK\n",
      "cat 321, mi = 0.00273063, MODULE_183\n",
      "cat 529, mi = 0.00268699, MODULE_430\n",
      "cat 24, mi = 0.00265203, GCM_CSNK1D\n",
      "cat 100, mi = 0.00265053, GNF2_CASP4\n",
      "cat 57, mi = 0.00259402, GCM_NUMA1\n",
      "cat 102, mi = 0.00257721, GNF2_CBFB\n",
      "cat 737, mi = 0.00256164, MORF_GSPT1\n",
      "cat 132, mi = 0.00251210, GNF2_ELAC2\n",
      "cat 751, mi = 0.00246260, MORF_JUND\n",
      "cat 29, mi = 0.00243097, GCM_DENR\n",
      "cat 126, mi = 0.00233739, GNF2_DDX5\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 92, mi = 0.004, pvalue = 0.001, GNF2_ATM\n",
      "Passed: cat 563, mi = 0.003, pvalue = 0.003, MODULE_484\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.003, pvalue = 0.010, MODULE_183\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.003, pvalue = 0.012, GCM_CSNK1D\n",
      "Didn't pass (cutoff=0.0050): cat 100, mi = 0.003, pvalue = 0.006, GNF2_CASP4\n",
      "Didn't pass (cutoff=0.0050): cat 57, mi = 0.003, pvalue = 0.012, GCM_NUMA1\n",
      "Passed: cat 102, mi = 0.003, pvalue = 0.005, GNF2_CBFB\n",
      "Didn't pass (cutoff=0.0050): cat 737, mi = 0.003, pvalue = 0.011, MORF_GSPT1\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.003, pvalue = 0.011, GNF2_ELAC2\n",
      "Didn't pass (cutoff=0.0050): cat 751, mi = 0.002, pvalue = 0.016, MORF_JUND\n",
      "Didn't pass (cutoff=0.0050): cat 29, mi = 0.002, pvalue = 0.027, GCM_DENR\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.002, pvalue = 0.026, GNF2_DDX5\n",
      "Didn't pass (cutoff=0.0050): cat 446, mi = 0.002, pvalue = 0.026, MODULE_325\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.002, pvalue = 0.037, MODULE_175\n",
      "Didn't pass (cutoff=0.0050): cat 128, mi = 0.002, pvalue = 0.035, GNF2_DENR\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.002, pvalue = 0.033, GNF2_CD97\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.040, GNF2_JAK1\n",
      "Didn't pass (cutoff=0.0050): cat 468, mi = 0.002, pvalue = 0.011, MODULE_352\n",
      "Didn't pass (cutoff=0.0050): cat 342, mi = 0.002, pvalue = 0.043, MODULE_204\n",
      "Didn't pass (cutoff=0.0050): cat 715, mi = 0.002, pvalue = 0.039, MORF_EIF3S6\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.002, pvalue = 0.051, GCM_MAP1B\n",
      "Didn't pass (cutoff=0.0050): cat 755, mi = 0.002, pvalue = 0.040, MORF_LTK\n",
      "Didn't pass (cutoff=0.0050): cat 67, mi = 0.002, pvalue = 0.033, GCM_RAB10\n",
      "Didn't pass (cutoff=0.0050): cat 827, mi = 0.002, pvalue = 0.036, MORF_RPA2\n",
      "Didn't pass (cutoff=0.0050): cat 699, mi = 0.002, pvalue = 0.044, MORF_CSNK1D\n",
      "Didn't pass (cutoff=0.0050): cat 201, mi = 0.002, pvalue = 0.057, GNF2_RRM1\n",
      "Didn't pass (cutoff=0.0050): cat 590, mi = 0.002, pvalue = 0.048, MODULE_524\n",
      "Total number of categories evaluated = 22.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.639059226517685\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "7.06\t-12.28\n",
      "GNF2_ATM\n",
      "GNF2_CBFB\n",
      "MODULE_484\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.639059226517685\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_c4/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 5917\n",
      "cat 1133, mi = 0.00361708, GO_EAR_DEVELOPMENT\n",
      "cat 848, mi = 0.00357967, GO_COLUMNAR_CUBOIDAL_EPITHELIAL_CELL_DIFFERENTIATION\n",
      "cat 5406, mi = 0.00324996, GO_SKELETAL_MUSCLE_CELL_DIFFERENTIATION\n",
      "cat 2783, mi = 0.00320332, GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "cat 4971, mi = 0.00312656, GO_REGULATION_OF_T_CELL_DIFFERENTIATION\n",
      "cat 3746, mi = 0.00312313, GO_POSITIVE_REGULATION_OF_RECEPTOR_ACTIVITY\n",
      "cat 4720, mi = 0.00311329, GO_REGULATION_OF_OSSIFICATION\n",
      "cat 873, mi = 0.00310759, GO_CORE_PROMOTER_BINDING\n",
      "cat 574, mi = 0.00306141, GO_CELLULAR_RESPONSE_TO_INSULIN_STIMULUS\n",
      "cat 5299, mi = 0.00304698, GO_RRNA_BINDING\n",
      "cat 92, mi = 0.00303700, GO_ADHERENS_JUNCTION_ASSEMBLY\n",
      "cat 1594, mi = 0.00301448, GO_HEART_TRABECULA_MORPHOGENESIS\n",
      "cat 4207, mi = 0.00300697, GO_REGULATION_OF_BLOOD_VESSEL_ENDOTHELIAL_CELL_MIGRATION\n",
      "cat 2354, mi = 0.00299709, GO_MYELIN_SHEATH\n",
      "cat 4889, mi = 0.00298029, GO_REGULATION_OF_STEM_CELL_PROLIFERATION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1133, mi = 0.004, pvalue = 0.000, GO_EAR_DEVELOPMENT\n",
      "Passed: cat 2783, mi = 0.003, pvalue = 0.001, GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "Passed: cat 3746, mi = 0.003, pvalue = 0.001, GO_POSITIVE_REGULATION_OF_RECEPTOR_ACTIVITY\n",
      "Passed: cat 574, mi = 0.003, pvalue = 0.003, GO_CELLULAR_RESPONSE_TO_INSULIN_STIMULUS\n",
      "Passed: cat 5299, mi = 0.003, pvalue = 0.002, GO_RRNA_BINDING\n",
      "Passed: cat 3871, mi = 0.003, pvalue = 0.005, GO_POTASSIUM_ION_TRANSPORT\n",
      "Passed: cat 544, mi = 0.003, pvalue = 0.003, GO_CELLULAR_RESPONSE_TO_CAMP\n",
      "Didn't pass (cutoff=0.0050): cat 757, mi = 0.003, pvalue = 0.005, GO_CHONDROITIN_SULFATE_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 2043, mi = 0.003, pvalue = 0.005, GO_MAIN_AXON\n",
      "Didn't pass (cutoff=0.0050): cat 1532, mi = 0.003, pvalue = 0.010, GO_GOLGI_ASSOCIATED_VESICLE\n",
      "Didn't pass (cutoff=0.0050): cat 3180, mi = 0.003, pvalue = 0.010, GO_PERIPHERAL_NERVOUS_SYSTEM_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 831, mi = 0.003, pvalue = 0.010, GO_COENZYME_BINDING\n",
      "Passed: cat 1858, mi = 0.003, pvalue = 0.002, GO_KERATAN_SULFATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3640, mi = 0.003, pvalue = 0.006, GO_POSITIVE_REGULATION_OF_MRNA_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 4174, mi = 0.003, pvalue = 0.014, GO_REGULATION_OF_ALPHA_BETA_T_CELL_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 1057, mi = 0.003, pvalue = 0.015, GO_DNA_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5181, mi = 0.003, pvalue = 0.018, GO_RESPONSE_TO_TYPE_I_INTERFERON\n",
      "Didn't pass (cutoff=0.0050): cat 1597, mi = 0.003, pvalue = 0.019, GO_HEAT_SHOCK_PROTEIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 882, mi = 0.002, pvalue = 0.019, GO_COSTAMERE\n",
      "Didn't pass (cutoff=0.0050): cat 1539, mi = 0.002, pvalue = 0.021, GO_GOLGI_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 1352, mi = 0.002, pvalue = 0.013, GO_EXTRINSIC_COMPONENT_OF_ORGANELLE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 3132, mi = 0.002, pvalue = 0.011, GO_PATTERNING_OF_BLOOD_VESSELS\n",
      "Didn't pass (cutoff=0.0050): cat 1548, mi = 0.002, pvalue = 0.006, GO_GPI_ANCHOR_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3737, mi = 0.002, pvalue = 0.024, GO_POSITIVE_REGULATION_OF_PROTEIN_POLYMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 1450, mi = 0.002, pvalue = 0.006, GO_GASTRULATION_WITH_MOUTH_FORMING_SECOND\n",
      "Didn't pass (cutoff=0.0050): cat 760, mi = 0.002, pvalue = 0.022, GO_CHONDROITIN_SULFATE_PROTEOGLYCAN_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1683, mi = 0.002, pvalue = 0.012, GO_HYALURONAN_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1533, mi = 0.002, pvalue = 0.028, GO_GOLGI_ASSOCIATED_VESICLE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 3362, mi = 0.002, pvalue = 0.021, GO_POSITIVE_REGULATION_OF_ALPHA_BETA_T_CELL_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 759, mi = 0.002, pvalue = 0.032, GO_CHONDROITIN_SULFATE_PROTEOGLYCAN_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2393, mi = 0.002, pvalue = 0.018, GO_NBAF_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 5570, mi = 0.002, pvalue = 0.017, GO_SWI_SNF_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 5591, mi = 0.002, pvalue = 0.008, GO_SYNCYTIUM_FORMATION\n",
      "Total number of categories evaluated = 138.\n",
      "Number of categories that passed the tests = 9\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 6, 7, 0.39465934497394\n",
      "best merge is 1, 2, 0.404955252127552\n",
      "best merge is 0, 3, 0.584460214767773\n",
      "best merge is 5, 8, 0.734125904721121\n",
      "best merge is 0, 1, 0.750988968303254\n",
      "order classes\n",
      "2 - 1.31503548322749\n",
      "2 - 0.911149317286993\n",
      "2 - 1.65178328562252\n",
      "2 - 0.887983526191364\n",
      "6 7\n",
      "1 2\n",
      "4\n",
      "5 8\n",
      "0 3\n",
      "Done.Start drawing\n",
      "7.06\t-12.28\n",
      "GO_KERATAN_SULFATE_METABOLIC_PROCESS\n",
      "GO_EAR_DEVELOPMENT\n",
      "GO_CELLULAR_RESPONSE_TO_CAMP\n",
      "GO_MAIN_AXON\n",
      "GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "GO_POSITIVE_REGULATION_OF_RECEPTOR_ACTIVITY\n",
      "GO_RRNA_BINDING\n",
      "GO_CELLULAR_RESPONSE_TO_INSULIN_STIMULUS\n",
      "GO_POTASSIUM_ION_TRANSPORT\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 6, 7, 0.39465934497394\n",
      "best merge is 1, 2, 0.404955252127552\n",
      "best merge is 0, 3, 0.584460214767773\n",
      "best merge is 5, 8, 0.734125904721121\n",
      "best merge is 0, 1, 0.750988968303254\n",
      "order classes\n",
      "2 - 1.31503548322749\n",
      "2 - 0.911149317286993\n",
      "2 - 1.65178328562252\n",
      "2 - 0.887983526191364\n",
      "6 7\n",
      "1 2\n",
      "4\n",
      "5 8\n",
      "0 3\n",
      "Done.-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_c5/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 189\n",
      "cat 104, mi = 0.00339157, KRAS.LUNG_UP.V1_UP\n",
      "cat 68, mi = 0.00308450, GCNP_SHH_UP_LATE.V1_UP\n",
      "cat 180, mi = 0.00293202, TBK1.DN.48HRS_UP\n",
      "cat 102, mi = 0.00268092, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 182, mi = 0.00258081, TGFB_UP.V1_UP\n",
      "cat 132, mi = 0.00257542, PDGF_UP.V1_DN\n",
      "cat 129, mi = 0.00256254, P53_DN.V2_UP\n",
      "cat 96, mi = 0.00253121, KRAS.BREAST_UP.V1_UP\n",
      "cat 16, mi = 0.00245100, BCAT_GDS748_DN\n",
      "cat 64, mi = 0.00241692, ESC_V6.5_UP_LATE.V1_UP\n",
      "cat 9, mi = 0.00236434, ATF2_UP.V1_UP\n",
      "cat 134, mi = 0.00230257, PIGF_UP.V1_DN\n",
      "cat 124, mi = 0.00228500, NRL_DN.V1_DN\n",
      "cat 81, mi = 0.00224356, JAK2_DN.V1_DN\n",
      "cat 34, mi = 0.00217886, CRX_NRL_DN.V1_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 104, mi = 0.003, pvalue = 0.000, KRAS.LUNG_UP.V1_UP\n",
      "Passed: cat 68, mi = 0.003, pvalue = 0.001, GCNP_SHH_UP_LATE.V1_UP\n",
      "Passed: cat 180, mi = 0.003, pvalue = 0.003, TBK1.DN.48HRS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.003, pvalue = 0.008, TGFB_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 96, mi = 0.003, pvalue = 0.018, KRAS.BREAST_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.013, BCAT_GDS748_DN\n",
      "Didn't pass (cutoff=0.0050): cat 64, mi = 0.002, pvalue = 0.022, ESC_V6.5_UP_LATE.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.026, ATF2_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 134, mi = 0.002, pvalue = 0.025, PIGF_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.002, pvalue = 0.032, NRL_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.002, pvalue = 0.033, CRX_NRL_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.002, pvalue = 0.051, ATM_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 187, mi = 0.002, pvalue = 0.038, YAP1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.002, pvalue = 0.071, BCAT.100_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.002, pvalue = 0.063, MTOR_UP.N4.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.002, pvalue = 0.092, RB_P130_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 167, mi = 0.002, pvalue = 0.088, SNF5_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.113, TBK1.DN.48HRS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.002, pvalue = 0.115, CAHOY_ASTROGLIAL\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.002, pvalue = 0.079, MTOR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.002, pvalue = 0.079, CYCLIN_D1_KE_.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.002, pvalue = 0.089, NRL_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 77, mi = 0.002, pvalue = 0.137, IL21_UP.V1_DN\n",
      "Total number of categories evaluated = 25.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.817902468455268\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "7.06\t-12.28\n",
      "GCNP_SHH_UP_LATE.V1_UP\n",
      "TBK1.DN.48HRS_UP\n",
      "KRAS.LUNG_UP.V1_UP\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.817902468455268\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_c6/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 4872\n",
      "cat 2160, mi = 0.00590038, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "cat 2047, mi = 0.00469287, GSE2405_0H_VS_12H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "cat 4640, mi = 0.00439729, GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "cat 762, mi = 0.00431186, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "cat 3944, mi = 0.00429246, GSE41867_NAIVE_VS_DAY30_LCMV_CLONE13_EXHAUSTED_CD8_TCELL_DN\n",
      "cat 2303, mi = 0.00425427, GSE26156_DOUBLE_POSITIVE_VS_CD4_SINGLE_POSITIVE_THYMOCYTE_UP\n",
      "cat 25, mi = 0.00416270, GSE10239_NAIVE_VS_DAY4.5_EFF_CD8_TCELL_UP\n",
      "cat 1338, mi = 0.00408451, GSE1925_CTRL_VS_24H_IFNG_STIM_MACROPHAGE_DN\n",
      "cat 3082, mi = 0.00403803, GSE3565_DUSP1_VS_WT_SPLENOCYTES_DN\n",
      "cat 1672, mi = 0.00401986, GSE21670_UNTREATED_VS_TGFB_IL6_TREATED_STAT3_KO_CD4_TCELL_DN\n",
      "cat 520, mi = 0.00398062, GSE1460_DP_VS_CD4_THYMOCYTE_DN\n",
      "cat 4047, mi = 0.00396779, GSE43863_NAIVE_VS_MEMORY_LY6C_INT_CXCR5POS_CD4_TCELL_D150_LCMV_UP\n",
      "cat 2617, mi = 0.00395655, GSE27896_HDAC6_KO_VS_WT_TREG_UP\n",
      "cat 4497, mi = 0.00393658, GSE6269_FLU_VS_STREP_PNEUMO_INF_PBMC_UP\n",
      "cat 2439, mi = 0.00384027, GSE2770_IL12_AND_TGFB_VS_IL4_TREATED_ACT_CD4_TCELL_48H_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2160, mi = 0.006, pvalue = 0.000, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "Passed: cat 2047, mi = 0.005, pvalue = 0.000, GSE2405_0H_VS_12H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "Passed: cat 4640, mi = 0.004, pvalue = 0.000, GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "Passed: cat 2303, mi = 0.004, pvalue = 0.000, GSE26156_DOUBLE_POSITIVE_VS_CD4_SINGLE_POSITIVE_THYMOCYTE_UP\n",
      "Passed: cat 25, mi = 0.004, pvalue = 0.000, GSE10239_NAIVE_VS_DAY4.5_EFF_CD8_TCELL_UP\n",
      "Passed: cat 1338, mi = 0.004, pvalue = 0.000, GSE1925_CTRL_VS_24H_IFNG_STIM_MACROPHAGE_DN\n",
      "Passed: cat 1930, mi = 0.004, pvalue = 0.000, GSE22886_NAIVE_VS_MEMORY_TCELL_DN\n",
      "Passed: cat 2574, mi = 0.004, pvalue = 0.001, GSE27786_LSK_VS_CD8_TCELL_DN\n",
      "Passed: cat 2854, mi = 0.003, pvalue = 0.001, GSE30971_CTRL_VS_LPS_STIM_MACROPHAGE_WBP7_KO_4H_DN\n",
      "Passed: cat 481, mi = 0.003, pvalue = 0.001, GSE14415_ACT_VS_CTRL_NATURAL_TREG_UP\n",
      "Passed: cat 2465, mi = 0.003, pvalue = 0.001, GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_6H_UP\n",
      "Passed: cat 4579, mi = 0.003, pvalue = 0.001, GSE7460_CTRL_VS_TGFB_TREATED_ACT_TREG_UP\n",
      "Passed: cat 4509, mi = 0.003, pvalue = 0.002, GSE6566_STRONG_VS_WEAK_DC_STIMULATED_CD4_TCELL_UP\n",
      "Passed: cat 780, mi = 0.003, pvalue = 0.002, GSE16522_ANTI_CD3CD28_STIM_VS_UNSTIM_NAIVE_CD8_TCELL_DN\n",
      "Passed: cat 2284, mi = 0.003, pvalue = 0.003, GSE26023_PHD3_KO_VS_WT_NEUTROPHIL_HYPOXIA_DN\n",
      "Passed: cat 2722, mi = 0.003, pvalue = 0.004, GSE29618_BCELL_VS_MONOCYTE_DN\n",
      "Passed: cat 1327, mi = 0.003, pvalue = 0.003, GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "Passed: cat 4505, mi = 0.003, pvalue = 0.004, GSE6269_HEALTHY_VS_STAPH_PNEUMO_INF_PBMC_UP\n",
      "Passed: cat 578, mi = 0.003, pvalue = 0.005, GSE15324_NAIVE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "Passed: cat 629, mi = 0.003, pvalue = 0.004, GSE15659_CD45RA_NEG_CD4_TCELL_VS_ACTIVATED_TREG_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3963, mi = 0.003, pvalue = 0.007, GSE41978_ID2_KO_VS_ID2_KO_AND_BIM_KO_KLRG1_LOW_EFFECTOR_CD8_TCELL_UP\n",
      "Passed: cat 2456, mi = 0.003, pvalue = 0.005, GSE2770_IL4_ACT_VS_ACT_CD4_TCELL_48H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4179, mi = 0.003, pvalue = 0.008, GSE45365_WT_VS_IFNAR_KO_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 687, mi = 0.003, pvalue = 0.006, GSE15930_NAIVE_VS_72H_IN_VITRO_STIM_IL12_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3399, mi = 0.003, pvalue = 0.007, GSE37416_0H_VS_6H_F_TULARENSIS_LVS_NEUTROPHIL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3159, mi = 0.003, pvalue = 0.008, GSE360_DC_VS_MAC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2775, mi = 0.003, pvalue = 0.009, GSE29949_CD8_NEG_DC_SPLEEN_VS_DC_BRAIN_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1458, mi = 0.003, pvalue = 0.012, GSE20198_UNTREATED_VS_IFNA_TREATED_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1050, mi = 0.003, pvalue = 0.010, GSE17721_LPS_VS_PAM3CSK4_2H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.003, pvalue = 0.014, GSE10273_HIGH_IL7_VS_HIGH_IL7_AND_IRF4_IN_IRF4_8_NULL_PRE_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1030, mi = 0.003, pvalue = 0.015, GSE17721_LPS_VS_GARDIQUIMOD_24H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3284, mi = 0.002, pvalue = 0.026, GSE36891_POLYIC_TLR3_VS_PAM_TLR2_STIM_PERITONEAL_MACROPHAGE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.002, pvalue = 0.019, GSE12839_CTRL_VS_IL12_TREATED_PBMC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 690, mi = 0.002, pvalue = 0.016, GSE15930_STIM_VS_STIM_AND_IFNAB_24H_CD8_T_CELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 193, mi = 0.002, pvalue = 0.021, GSE12003_MIR223_KO_VS_WT_BM_PROGENITOR_8D_CULTURE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1556, mi = 0.002, pvalue = 0.022, GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_8H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 70, mi = 0.002, pvalue = 0.025, GSE10463_CD40L_AND_VA347_VS_CD40L_IN_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1654, mi = 0.002, pvalue = 0.022, GSE21670_STAT3_KO_VS_WT_CD4_TCELL_TGFB_IL6_TREATED_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2887, mi = 0.002, pvalue = 0.022, GSE3203_INFLUENZA_INF_VS_IFNB_TREATED_LN_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2413, mi = 0.002, pvalue = 0.019, GSE27291_0H_VS_7D_STIM_GAMMADELTA_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 349, mi = 0.002, pvalue = 0.031, GSE13522_CTRL_VS_T_CRUZI_Y_STRAIN_INF_SKIN_BALBC_MOUSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1814, mi = 0.002, pvalue = 0.027, GSE22601_DOUBLE_NEGATIVE_VS_CD8_SINGLE_POSITIVE_THYMOCYTE_DN\n",
      "Total number of categories evaluated = 288.\n",
      "Number of categories that passed the tests = 21\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 7, 0.229834281971824\n",
      "best merge is 9, 11, 0.282318297454546\n",
      "best merge is 3, 20, 0.284989913362131\n",
      "best merge is 1, 2, 0.292297760059873\n",
      "best merge is 4, 13, 0.326453658898114\n",
      "best merge is 3, 10, 0.382947118500553\n",
      "best merge is 1, 14, 0.435757337757122\n",
      "best merge is 5, 9, 0.489908969467415\n",
      "best merge is 0, 1, 0.50538485209432\n",
      "best merge is 4, 17, 0.528740389106958\n",
      "best merge is 5, 18, 0.540742577627993\n",
      "best merge is 16, 19, 0.58416294269306\n",
      "best merge is 6, 15, 0.640279501228235\n",
      "best merge is 0, 3, 0.669629264001139\n",
      "best merge is 5, 8, 0.760114415607333\n",
      "best merge is 0, 4, 0.776267160854403\n",
      "best merge is 12, 16, 0.865966594815961\n",
      "order classes\n",
      "11 - 1.12133453042871\n",
      "5 - 1.9907400682521\n",
      "2 - 3.36146738144823\n",
      "2 - 3.06685544913856\n",
      "0 7 1 2 14 3 20 10 4 13 17\n",
      "5 9 11 18 8\n",
      "6 15\n",
      "12\n",
      "16 19\n",
      "Done.Start drawing\n",
      "7.06\t-12.28\n",
      "GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "GSE30971_CTRL_VS_LPS_STIM_MACROPHAGE_WBP7_KO_4H_DN\n",
      "GSE2405_0H_VS_12H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "GSE26023_PHD3_KO_VS_WT_NEUTROPHIL_HYPOXIA_DN\n",
      "GSE2770_IL4_ACT_VS_ACT_CD4_TCELL_48H_DN\n",
      "GSE10239_NAIVE_VS_DAY4.5_EFF_CD8_TCELL_UP\n",
      "GSE16522_ANTI_CD3CD28_STIM_VS_UNSTIM_NAIVE_CD8_TCELL_DN\n",
      "GSE26156_DOUBLE_POSITIVE_VS_CD4_SINGLE_POSITIVE_THYMOCYTE_UP\n",
      "GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_6H_UP\n",
      "GSE6269_HEALTHY_VS_STAPH_PNEUMO_INF_PBMC_UP\n",
      "GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "GSE27786_LSK_VS_CD8_TCELL_DN\n",
      "GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "GSE22886_NAIVE_VS_MEMORY_TCELL_DN\n",
      "GSE29618_BCELL_VS_MONOCYTE_DN\n",
      "GSE6566_STRONG_VS_WEAK_DC_STIMULATED_CD4_TCELL_UP\n",
      "GSE1925_CTRL_VS_24H_IFNG_STIM_MACROPHAGE_DN\n",
      "GSE7460_CTRL_VS_TGFB_TREATED_ACT_TREG_UP\n",
      "GSE15324_NAIVE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "GSE14415_ACT_VS_CTRL_NATURAL_TREG_UP\n",
      "GSE15659_CD45RA_NEG_CD4_TCELL_VS_ACTIVATED_TREG_UP\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 7, 0.229834281971824\n",
      "best merge is 9, 11, 0.282318297454546\n",
      "best merge is 3, 20, 0.284989913362131\n",
      "best merge is 1, 2, 0.292297760059873\n",
      "best merge is 4, 13, 0.326453658898114\n",
      "best merge is 3, 10, 0.382947118500553\n",
      "best merge is 1, 14, 0.435757337757122\n",
      "best merge is 5, 9, 0.489908969467415\n",
      "best merge is 0, 1, 0.50538485209432\n",
      "best merge is 4, 17, 0.528740389106958\n",
      "best merge is 5, 18, 0.540742577627993\n",
      "best merge is 16, 19, 0.58416294269306\n",
      "best merge is 6, 15, 0.640279501228235\n",
      "best merge is 0, 3, 0.669629264001139\n",
      "best merge is 5, 8, 0.760114415607333\n",
      "best merge is 0, 4, 0.776267160854403\n",
      "best merge is 12, 16, 0.865966594815961\n",
      "order classes\n",
      "11 - 1.12133453042871\n",
      "5 - 1.9907400682521\n",
      "2 - 3.36146738144823\n",
      "2 - 3.06685544913856\n",
      "0 7 1 2 14 3 20 10 4 13 17\n",
      "5 9 11 18 8\n",
      "6 15\n",
      "12\n",
      "16 19\n",
      "Done.-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_c7/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 17786\n",
      "cat 10017, mi = 0.00590038, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "cat 9904, mi = 0.00469287, GSE2405_0H_VS_12H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "cat 12497, mi = 0.00439729, GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "cat 13865, mi = 0.00438938, MARSON_FOXP3_TARGETS_UP\n",
      "cat 8619, mi = 0.00431186, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "cat 11801, mi = 0.00429246, GSE41867_NAIVE_VS_DAY30_LCMV_CLONE13_EXHAUSTED_CD8_TCELL_DN\n",
      "cat 10160, mi = 0.00425427, GSE26156_DOUBLE_POSITIVE_VS_CD4_SINGLE_POSITIVE_THYMOCYTE_UP\n",
      "cat 7882, mi = 0.00416270, GSE10239_NAIVE_VS_DAY4.5_EFF_CD8_TCELL_UP\n",
      "cat 15155, mi = 0.00413907, PID_AP1_PATHWAY\n",
      "cat 9195, mi = 0.00408451, GSE1925_CTRL_VS_24H_IFNG_STIM_MACROPHAGE_DN\n",
      "cat 12878, mi = 0.00406529, HENDRICKS_SMARCA4_TARGETS_UP\n",
      "cat 10939, mi = 0.00403803, GSE3565_DUSP1_VS_WT_SPLENOCYTES_DN\n",
      "cat 16491, mi = 0.00402419, SIMBULAN_UV_RESPONSE_NORMAL_DN\n",
      "cat 9529, mi = 0.00401986, GSE21670_UNTREATED_VS_TGFB_IL6_TREATED_STAT3_KO_CD4_TCELL_DN\n",
      "cat 13613, mi = 0.00401270, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 10017, mi = 0.006, pvalue = 0.000, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "Passed: cat 9904, mi = 0.005, pvalue = 0.000, GSE2405_0H_VS_12H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "Passed: cat 12497, mi = 0.004, pvalue = 0.000, GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "Passed: cat 13865, mi = 0.004, pvalue = 0.000, MARSON_FOXP3_TARGETS_UP\n",
      "Passed: cat 10160, mi = 0.004, pvalue = 0.000, GSE26156_DOUBLE_POSITIVE_VS_CD4_SINGLE_POSITIVE_THYMOCYTE_UP\n",
      "Passed: cat 7882, mi = 0.004, pvalue = 0.000, GSE10239_NAIVE_VS_DAY4.5_EFF_CD8_TCELL_UP\n",
      "Passed: cat 9195, mi = 0.004, pvalue = 0.000, GSE1925_CTRL_VS_24H_IFNG_STIM_MACROPHAGE_DN\n",
      "Passed: cat 9787, mi = 0.004, pvalue = 0.000, GSE22886_NAIVE_VS_MEMORY_TCELL_DN\n",
      "Passed: cat 12738, mi = 0.004, pvalue = 0.000, GTGGTGA_MIR197\n",
      "Passed: cat 10431, mi = 0.004, pvalue = 0.001, GSE27786_LSK_VS_CD8_TCELL_DN\n",
      "Passed: cat 13513, mi = 0.003, pvalue = 0.000, KRAS.LUNG_UP.V1_UP\n",
      "Passed: cat 10711, mi = 0.003, pvalue = 0.001, GSE30971_CTRL_VS_LPS_STIM_MACROPHAGE_WBP7_KO_4H_DN\n",
      "Passed: cat 8338, mi = 0.003, pvalue = 0.001, GSE14415_ACT_VS_CTRL_NATURAL_TREG_UP\n",
      "Passed: cat 10322, mi = 0.003, pvalue = 0.001, GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_6H_UP\n",
      "Passed: cat 4669, mi = 0.003, pvalue = 0.001, GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "Passed: cat 13315, mi = 0.003, pvalue = 0.001, KEGG_LONG_TERM_POTENTIATION\n",
      "Passed: cat 17274, mi = 0.003, pvalue = 0.002, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_13\n",
      "Passed: cat 12366, mi = 0.003, pvalue = 0.002, GSE6566_STRONG_VS_WEAK_DC_STIMULATED_CD4_TCELL_UP\n",
      "Passed: cat 301, mi = 0.003, pvalue = 0.003, BENPORATH_ES_CORE_NINE_CORRELATED\n",
      "Passed: cat 1637, mi = 0.003, pvalue = 0.003, GCNP_SHH_UP_LATE.V1_UP\n",
      "Passed: cat 7185, mi = 0.003, pvalue = 0.002, GO_RRNA_BINDING\n",
      "Passed: cat 6093, mi = 0.003, pvalue = 0.000, GO_REGULATION_OF_BLOOD_VESSEL_ENDOTHELIAL_CELL_MIGRATION\n",
      "Passed: cat 888, mi = 0.003, pvalue = 0.004, CHENG_IMPRINTED_BY_ESTRADIOL\n",
      "Passed: cat 10141, mi = 0.003, pvalue = 0.003, GSE26023_PHD3_KO_VS_WT_NEUTROPHIL_HYPOXIA_DN\n",
      "Passed: cat 9184, mi = 0.003, pvalue = 0.003, GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "Passed: cat 17154, mi = 0.003, pvalue = 0.005, WILCOX_RESPONSE_TO_PROGESTERONE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5757, mi = 0.003, pvalue = 0.005, GO_POTASSIUM_ION_TRANSPORT\n",
      "Passed: cat 16725, mi = 0.003, pvalue = 0.003, TBK1.DN.48HRS_UP\n",
      "Passed: cat 2643, mi = 0.003, pvalue = 0.004, GO_CHONDROITIN_SULFATE_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 8435, mi = 0.003, pvalue = 0.004, GSE15324_NAIVE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "Passed: cat 1122, mi = 0.003, pvalue = 0.005, DAVICIONI_TARGETS_OF_PAX_FOXO1_FUSIONS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 13115, mi = 0.003, pvalue = 0.006, JAZAERI_BREAST_CANCER_BRCA1_VS_BRCA2_UP\n",
      "Passed: cat 15834, mi = 0.003, pvalue = 0.005, REACTOME_MRNA_PROCESSING\n",
      "Passed: cat 15089, mi = 0.003, pvalue = 0.004, PAX8_01\n",
      "Didn't pass (cutoff=0.0050): cat 10313, mi = 0.003, pvalue = 0.005, GSE2770_IL4_ACT_VS_ACT_CD4_TCELL_48H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3418, mi = 0.003, pvalue = 0.009, GO_GOLGI_ASSOCIATED_VESICLE\n",
      "Didn't pass (cutoff=0.0050): cat 14354, mi = 0.003, pvalue = 0.011, MODULE_430\n",
      "Didn't pass (cutoff=0.0050): cat 11256, mi = 0.003, pvalue = 0.006, GSE37416_0H_VS_6H_F_TULARENSIS_LVS_NEUTROPHIL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 11016, mi = 0.003, pvalue = 0.009, GSE360_DC_VS_MAC_UP\n",
      "Passed: cat 3744, mi = 0.003, pvalue = 0.003, GO_KERATAN_SULFATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 17343, mi = 0.003, pvalue = 0.006, ZEMBUTSU_SENSITIVITY_TO_NIMUSTINE\n",
      "Didn't pass (cutoff=0.0050): cat 1730, mi = 0.003, pvalue = 0.005, GNF2_CBFB\n",
      "Didn't pass (cutoff=0.0050): cat 8907, mi = 0.003, pvalue = 0.013, GSE17721_LPS_VS_PAM3CSK4_2H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 831, mi = 0.003, pvalue = 0.007, CDPCR3_01\n",
      "Didn't pass (cutoff=0.0050): cat 7902, mi = 0.003, pvalue = 0.013, GSE10273_HIGH_IL7_VS_HIGH_IL7_AND_IRF4_IN_IRF4_8_NULL_PRE_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 8887, mi = 0.003, pvalue = 0.013, GSE17721_LPS_VS_GARDIQUIMOD_24H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9509, mi = 0.003, pvalue = 0.011, GSE21670_STAT3_KO_VS_WT_CD4_TCELL_IL6_TREATED_UP\n",
      "Didn't pass (cutoff=0.0050): cat 9314, mi = 0.003, pvalue = 0.012, GSE20198_IL12_VS_IL12_IL18_TREATED_ACT_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 13317, mi = 0.002, pvalue = 0.019, KEGG_LYSOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 8355, mi = 0.002, pvalue = 0.019, GSE14415_TCONV_VS_FOXP3_KO_INDUCED_TREG_DN\n",
      "Didn't pass (cutoff=0.0050): cat 16328, mi = 0.002, pvalue = 0.012, SA_CASPASE_CASCADE\n",
      "Didn't pass (cutoff=0.0050): cat 15157, mi = 0.002, pvalue = 0.018, PID_ARF6_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 11141, mi = 0.002, pvalue = 0.022, GSE36891_POLYIC_TLR3_VS_PAM_TLR2_STIM_PERITONEAL_MACROPHAGE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1322, mi = 0.002, pvalue = 0.021, FAELT_B_CLL_WITH_VH3_21_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3339, mi = 0.002, pvalue = 0.018, GO_GDP_BINDING\n",
      "Passed: cat 257, mi = 0.002, pvalue = 0.001, BANDRES_RESPONSE_TO_CARMUSTIN_MGMT_24HR_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8655, mi = 0.002, pvalue = 0.017, GSE17186_CD21LOW_VS_CD21HIGH_TRANSITIONAL_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8050, mi = 0.002, pvalue = 0.023, GSE12003_MIR223_KO_VS_WT_BM_PROGENITOR_8D_CULTURE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 8877, mi = 0.002, pvalue = 0.020, GSE17721_LPS_VS_CPG_8H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 10353, mi = 0.002, pvalue = 0.021, GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 7927, mi = 0.002, pvalue = 0.026, GSE10463_CD40L_AND_VA347_VS_CD40L_IN_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 11717, mi = 0.002, pvalue = 0.028, GSE41176_UNSTIM_VS_ANTI_IGM_STIM_TAK1_KO_BCELL_3H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9511, mi = 0.002, pvalue = 0.022, GSE21670_STAT3_KO_VS_WT_CD4_TCELL_TGFB_IL6_TREATED_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3419, mi = 0.002, pvalue = 0.025, GO_GOLGI_ASSOCIATED_VESICLE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 1754, mi = 0.002, pvalue = 0.024, GNF2_DDX5\n",
      "Didn't pass (cutoff=0.0050): cat 9671, mi = 0.002, pvalue = 0.027, GSE22601_DOUBLE_NEGATIVE_VS_CD8_SINGLE_POSITIVE_THYMOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 7973, mi = 0.002, pvalue = 0.025, GSE11864_UNTREATED_VS_CSF1_IFNG_IN_MAC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3403, mi = 0.002, pvalue = 0.007, GO_GLYCOPROTEIN_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 12005, mi = 0.002, pvalue = 0.028, GSE45365_HEALTHY_VS_MCMV_INFECTION_BCELL_IFNAR_KO_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9902, mi = 0.002, pvalue = 0.030, GSE2405_0H_VS_1.5H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 17220, mi = 0.002, pvalue = 0.033, XU_HGF_SIGNALING_NOT_VIA_AKT1_48HR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 13432, mi = 0.002, pvalue = 0.028, KIM_MYCN_AMPLIFICATION_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 14948, mi = 0.002, pvalue = 0.035, NOUZOVA_METHYLATED_IN_APL\n",
      "Didn't pass (cutoff=0.0050): cat 7640, mi = 0.002, pvalue = 0.036, GO_TRNA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 8334, mi = 0.002, pvalue = 0.040, GSE14413_UNSTIM_VS_IFNB_STIM_RAW264_CELLS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 15153, mi = 0.002, pvalue = 0.045, PID_ANGIOPOIETIN_RECEPTOR_PATHWAY\n",
      "Total number of categories evaluated = 735.\n",
      "Number of categories that passed the tests = 34\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 9, 0.0436032191949296\n",
      "best merge is 8, 15, 0.054844773409529\n",
      "best merge is 3, 33, 0.136312759977268\n",
      "best merge is 10, 14, 0.189918966923308\n",
      "best merge is 2, 30, 0.191374627378662\n",
      "best merge is 16, 24, 0.239769587286408\n",
      "best merge is 3, 10, 0.251832137704075\n",
      "best merge is 25, 29, 0.260804352702994\n",
      "best merge is 1, 2, 0.279591609153206\n",
      "best merge is 13, 18, 0.299707508735299\n",
      "best merge is 7, 31, 0.335423213786545\n",
      "best merge is 0, 3, 0.348881575179371\n",
      "best merge is 1, 23, 0.384886874682489\n",
      "best merge is 17, 20, 0.386864129003069\n",
      "best merge is 4, 13, 0.420616585544464\n",
      "best merge is 5, 21, 0.44202543348455\n",
      "best merge is 12, 27, 0.513689263825839\n",
      "best merge is 17, 22, 0.523587820117587\n",
      "best merge is 8, 25, 0.537108862970925\n",
      "best merge is 16, 19, 0.564732389315692\n",
      "best merge is 0, 1, 0.579824905830524\n",
      "best merge is 6, 28, 0.632947838209825\n",
      "best merge is 4, 5, 0.634549563020744\n",
      "best merge is 26, 32, 0.647119046076001\n",
      "best merge is 6, 11, 0.696614806971949\n",
      "best merge is 0, 8, 0.759372196569924\n",
      "best merge is 6, 12, 0.773454835043839\n",
      "best merge is 0, 7, 0.78186666509836\n",
      "best merge is 0, 4, 0.869778690622808\n",
      "best merge is 16, 26, 0.95922972676661\n",
      "order classes\n",
      "21 - 1.09861585393356\n",
      "5 - 3.90624407507597\n",
      "3 - 3.44844210675592\n",
      "3 - 3.61165571511854\n",
      "2 - 5.50051189164601\n",
      "0 3 9 33 10 14 1 2 30 23 8 15 25 29 7 31 4 13 18 5 21\n",
      "6 28 11 12 27\n",
      "16 24 19\n",
      "17 20 22\n",
      "26 32\n",
      "Done.Start drawing\n",
      "7.06\t-12.28\n",
      "YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_13\n",
      "GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "GSE30971_CTRL_VS_LPS_STIM_MACROPHAGE_WBP7_KO_4H_DN\n",
      "GCNP_SHH_UP_LATE.V1_UP\n",
      "GSE2405_0H_VS_12H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "GSE26023_PHD3_KO_VS_WT_NEUTROPHIL_HYPOXIA_DN\n",
      "TBK1.DN.48HRS_UP\n",
      "GO_KERATAN_SULFATE_METABOLIC_PROCESS\n",
      "WILCOX_RESPONSE_TO_PROGESTERONE_DN\n",
      "GSE10239_NAIVE_VS_DAY4.5_EFF_CD8_TCELL_UP\n",
      "GO_REGULATION_OF_BLOOD_VESSEL_ENDOTHELIAL_CELL_MIGRATION\n",
      "DAVICIONI_TARGETS_OF_PAX_FOXO1_FUSIONS_DN\n",
      "GSE26156_DOUBLE_POSITIVE_VS_CD4_SINGLE_POSITIVE_THYMOCYTE_UP\n",
      "GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_6H_UP\n",
      "BENPORATH_ES_CORE_NINE_CORRELATED\n",
      "GTGGTGA_MIR197\n",
      "KEGG_LONG_TERM_POTENTIATION\n",
      "GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "MARSON_FOXP3_TARGETS_UP\n",
      "GSE27786_LSK_VS_CD8_TCELL_DN\n",
      "BANDRES_RESPONSE_TO_CARMUSTIN_MGMT_24HR_DN\n",
      "KRAS.LUNG_UP.V1_UP\n",
      "GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "REACTOME_MRNA_PROCESSING\n",
      "GSE22886_NAIVE_VS_MEMORY_TCELL_DN\n",
      "PAX8_01\n",
      "GSE6566_STRONG_VS_WEAK_DC_STIMULATED_CD4_TCELL_UP\n",
      "GO_RRNA_BINDING\n",
      "GSE1925_CTRL_VS_24H_IFNG_STIM_MACROPHAGE_DN\n",
      "GSE15324_NAIVE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "GO_CHONDROITIN_SULFATE_BIOSYNTHETIC_PROCESS\n",
      "GSE14415_ACT_VS_CTRL_NATURAL_TREG_UP\n",
      "CHENG_IMPRINTED_BY_ESTRADIOL\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 9, 0.0436032191949296\n",
      "best merge is 8, 15, 0.054844773409529\n",
      "best merge is 3, 33, 0.136312759977268\n",
      "best merge is 10, 14, 0.189918966923308\n",
      "best merge is 2, 30, 0.191374627378662\n",
      "best merge is 16, 24, 0.239769587286408\n",
      "best merge is 3, 10, 0.251832137704075\n",
      "best merge is 25, 29, 0.260804352702994\n",
      "best merge is 1, 2, 0.279591609153206\n",
      "best merge is 13, 18, 0.299707508735299\n",
      "best merge is 7, 31, 0.335423213786545\n",
      "best merge is 0, 3, 0.348881575179371\n",
      "best merge is 1, 23, 0.384886874682489\n",
      "best merge is 17, 20, 0.386864129003069\n",
      "best merge is 4, 13, 0.420616585544464\n",
      "best merge is 5, 21, 0.44202543348455\n",
      "best merge is 12, 27, 0.513689263825839\n",
      "best merge is 17, 22, 0.523587820117587\n",
      "best merge is 8, 25, 0.537108862970925\n",
      "best merge is 16, 19, 0.564732389315692\n",
      "best merge is 0, 1, 0.579824905830524\n",
      "best merge is 6, 28, 0.632947838209825\n",
      "best merge is 4, 5, 0.634549563020744\n",
      "best merge is 26, 32, 0.647119046076001\n",
      "best merge is 6, 11, 0.696614806971949\n",
      "best merge is 0, 8, 0.759372196569924\n",
      "best merge is 6, 12, 0.773454835043839\n",
      "best merge is 0, 7, 0.78186666509836\n",
      "best merge is 0, 4, 0.869778690622808\n",
      "best merge is 16, 26, 0.95922972676661\n",
      "order classes\n",
      "21 - 1.09861585393356\n",
      "5 - 3.90624407507597\n",
      "3 - 3.44844210675592\n",
      "3 - 3.61165571511854\n",
      "2 - 5.50051189164601\n",
      "0 3 9 33 10 14 1 2 30 23 8 15 25 29 7 31 4 13 18 5 21\n",
      "6 28 11 12 27\n",
      "16 24 19\n",
      "17 20 22\n",
      "26 32\n",
      "Done.-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_full/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 50\n",
      "cat 16, mi = 0.00289025, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 14, mi = 0.00232206, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 12, mi = 0.00229400, HALLMARK_E2F_TARGETS\n",
      "cat 6, mi = 0.00229266, HALLMARK_APOPTOSIS\n",
      "cat 1, mi = 0.00216820, HALLMARK_ALLOGRAFT_REJECTION\n",
      "cat 33, mi = 0.00211521, HALLMARK_MYOGENESIS\n",
      "cat 38, mi = 0.00207368, HALLMARK_PEROXISOME\n",
      "cat 9, mi = 0.00198051, HALLMARK_COAGULATION\n",
      "cat 26, mi = 0.00194109, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "cat 20, mi = 0.00189515, HALLMARK_HEME_METABOLISM\n",
      "cat 28, mi = 0.00188736, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 13, mi = 0.00188284, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 30, mi = 0.00181108, HALLMARK_MTORC1_SIGNALING\n",
      "cat 21, mi = 0.00173293, HALLMARK_HYPOXIA\n",
      "cat 7, mi = 0.00160151, HALLMARK_BILE_ACID_METABOLISM\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 16, mi = 0.003, pvalue = 0.005, HALLMARK_FATTY_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.002, pvalue = 0.023, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.002, pvalue = 0.025, HALLMARK_E2F_TARGETS\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.024, HALLMARK_APOPTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 1, mi = 0.002, pvalue = 0.038, HALLMARK_ALLOGRAFT_REJECTION\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.002, pvalue = 0.062, HALLMARK_MYOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.065, HALLMARK_COAGULATION\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.002, pvalue = 0.059, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.002, pvalue = 0.078, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.002, pvalue = 0.080, HALLMARK_KRAS_SIGNALING_UP\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.002, pvalue = 0.109, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.002, pvalue = 0.114, HALLMARK_HYPOXIA\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.002, pvalue = 0.212, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.002, pvalue = 0.171, HALLMARK_INTERFERON_ALPHA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.001, pvalue = 0.224, HALLMARK_P53_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.001, pvalue = 0.380, HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.001, pvalue = 0.289, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.380, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.001, pvalue = 0.289, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.001, pvalue = 0.298, HALLMARK_G2M_CHECKPOINT\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.001, pvalue = 0.368, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "Total number of categories evaluated = 15.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "7.06\t-12.28\n",
      "HALLMARK_FATTY_ACID_METABOLISM\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.94725956566701\t42\t0\n",
      "4.65925542916236\t42\t24.105480868666\n",
      "5.52792140641158\t42\t25.2130299896587\n",
      "6.04912099276111\t42\t25.8428128231644\n",
      "6.42916235780765\t42\t26.2554291623578\n",
      "6.73319544984488\t42\t26.6028955532575\n",
      "6.99379524301965\t42\t26.8634953464323\n",
      "7.25439503619442\t42\t27.124095139607\n",
      "7.56928645294726\t42\t27.3846949327818\n",
      "8.06876938986557\t42\t27.7538779731127\n",
      "15.1918304033092\t42\t28.3836608066184\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_msigdb_h/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7491 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7491\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "681 hits go_terms_num = 790\n",
      "cat 677, mi = 0.00292019, TNRC6C_HEK293_Plass\n",
      "cat 31, mi = 0.00203463, AUH_K562_POSTAR_eCLIP\n",
      "cat 676, mi = 0.00194727, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 236, mi = 0.00177410, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 275, mi = 0.00176951, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 447, mi = 0.00176761, NPM1_K562_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00176515, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 237, mi = 0.00171525, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 30, mi = 0.00145288, AUH_HepG2_POSTAR_eCLIP\n",
      "cat 668, mi = 0.00134468, TNRC6A_HEK293_Plass\n",
      "cat 230, mi = 0.00129944, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 300, mi = 0.00129277, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "cat 321, mi = 0.00128201, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 235, mi = 0.00123534, FUS_HeLa_POSTAR_CIMS\n",
      "cat 672, mi = 0.00120278, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 677, mi = 0.003, pvalue = 0.001, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.037, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.002, pvalue = 0.036, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.092, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.086, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.087, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.103, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.001, pvalue = 0.224, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.001, pvalue = 0.348, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.343, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.001, pvalue = 0.405, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.001, pvalue = 0.409, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.001, pvalue = 0.535, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 331, mi = 0.001, pvalue = 0.326, HNRNPU_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.001, pvalue = 0.458, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.657, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.001, pvalue = 0.498, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.001, pvalue = 0.577, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.569, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 323, mi = 0.001, pvalue = 0.091, HNRNPU_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.583, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Total number of categories evaluated = 22.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "7.06\t-12.28\n",
      "TNRC6C_HEK293_Plass\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.0124095139607\t42\t0\n",
      "4.56153050672182\t42\t23.9751809720786\n",
      "5.48448810754912\t42\t25.147880041365\n",
      "6.02740434332988\t42\t25.8210961737332\n",
      "6.40744570837642\t42\t26.2337125129266\n",
      "6.71147880041365\t42\t26.5811789038263\n",
      "6.97207859358842\t42\t26.841778697001\n",
      "7.25439503619441\t42\t27.1023784901758\n",
      "7.59100310237849\t42\t27.406411582213\n",
      "8.09048603929679\t42\t27.7755946225439\n",
      "15.2026887280248\t42\t28.4053774560496\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_all_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6816 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6816\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "620 hits go_terms_num = 790\n",
      "cat 677, mi = 0.00288158, TNRC6C_HEK293_Plass\n",
      "cat 173, mi = 0.00223106, EIF4A3_HeLa_POSTAR_CIMS\n",
      "cat 402, mi = 0.00221994, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 275, mi = 0.00207329, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00205234, AUH_K562_POSTAR_eCLIP\n",
      "cat 195, mi = 0.00197159, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00197159, EZH2_HCT116_POSTAR_iCLIP\n",
      "cat 676, mi = 0.00197038, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 236, mi = 0.00190989, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 447, mi = 0.00189210, NPM1_K562_POSTAR_eCLIP\n",
      "cat 671, mi = 0.00183532, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "cat 670, mi = 0.00174756, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "cat 237, mi = 0.00168905, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 635, mi = 0.00167807, TARDBP_H9_POSTAR_CIMS\n",
      "cat 538, mi = 0.00161835, RPS11_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 677, mi = 0.003, pvalue = 0.003, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 173, mi = 0.002, pvalue = 0.024, EIF4A3_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.041, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.058, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.058, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.082, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.002, pvalue = 0.084, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.002, pvalue = 0.059, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.091, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 671, mi = 0.002, pvalue = 0.079, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 670, mi = 0.002, pvalue = 0.096, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.180, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.162, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.186, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.002, pvalue = 0.273, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 667, mi = 0.001, pvalue = 0.242, TNRC6A_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.001, pvalue = 0.214, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.001, pvalue = 0.526, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.463, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.662, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.001, pvalue = 0.428, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 23.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.05\t-12.28\n",
      "TNRC6C_HEK293_Plass\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.53721869590306\t42\t0\n",
      "7.53144835545297\t42\t26.9255626081939\n",
      "8.50086555106752\t42\t28.1373341027121\n",
      "9.09463358338142\t42\t28.864396999423\n",
      "9.5066358915176\t42\t29.3248701673399\n",
      "9.83381419503751\t42\t29.6884016156953\n",
      "10.1246393537219\t42\t29.9792267743797\n",
      "10.4275822273514\t42\t30.2700519330641\n",
      "10.7789959607617\t42\t30.5851125216388\n",
      "11.3242931332949\t42\t30.9728793998846\n",
      "16.8378534333526\t42\t31.6757068667051\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6678 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6678\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "607 hits go_terms_num = 790\n",
      "cat 379, mi = 0.00404595, LIN28A_H9_POSTAR_CIMS\n",
      "cat 677, mi = 0.00316213, TNRC6C_HEK293_Plass\n",
      "cat 380, mi = 0.00306261, LIN28A_H9_POSTAR_HITS-CLIP\n",
      "cat 581, mi = 0.00303914, SND1_K562_POSTAR_eCLIP\n",
      "cat 306, mi = 0.00279466, HNRNPK_HepG2_POSTAR_eCLIP\n",
      "cat 369, mi = 0.00270771, KHSRP_K562_POSTAR_eCLIP\n",
      "cat 171, mi = 0.00265135, EIF3H_HepG2_POSTAR_eCLIP\n",
      "cat 608, mi = 0.00243198, SUB1_HepG2_POSTAR_eCLIP\n",
      "cat 304, mi = 0.00241563, HNRNPH1_HEK293T_POSTAR_HITS-CLIP\n",
      "cat 240, mi = 0.00239382, FXR1_HEK293_POSTAR_PAR-CLIP\n",
      "cat 241, mi = 0.00239382, FXR1_HEK293_POSTAR_PARalyzer\n",
      "cat 502, mi = 0.00239074, QKI_HEK293_POSTAR_Piranha_0.01\n",
      "cat 381, mi = 0.00237660, LIN28A_H9_POSTAR_Piranha_0.01\n",
      "cat 577, mi = 0.00230104, SMNDC1_K562_POSTAR_eCLIP\n",
      "cat 176, mi = 0.00226367, EIF4G2_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 379, mi = 0.004, pvalue = 0.000, LIN28A_H9_POSTAR_CIMS\n",
      "Passed: cat 677, mi = 0.003, pvalue = 0.001, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 437, mi = 0.002, pvalue = 0.037, NONO_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.002, pvalue = 0.055, SFPQ_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 670, mi = 0.002, pvalue = 0.011, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 671, mi = 0.002, pvalue = 0.012, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 563, mi = 0.002, pvalue = 0.060, SF3B4_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 142, mi = 0.002, pvalue = 0.092, DICER1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 76, mi = 0.002, pvalue = 0.045, CPSF6_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 120, mi = 0.002, pvalue = 0.105, DDX59_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 512, mi = 0.002, pvalue = 0.132, RBM10_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 290, mi = 0.002, pvalue = 0.173, HNRNPC_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 288, mi = 0.002, pvalue = 0.157, HNRNPC_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.002, pvalue = 0.262, DDX42_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.238, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 221, mi = 0.002, pvalue = 0.220, FTO_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.206, AGGF1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 534, mi = 0.002, pvalue = 0.242, RBM5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 549, mi = 0.002, pvalue = 0.266, SAFB2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 731, mi = 0.001, pvalue = 0.107, WTAP_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 267, mi = 0.001, pvalue = 0.216, GTF2F1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 130, mi = 0.001, pvalue = 0.345, DGCR8_HepG2_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 100.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.05\t-12.28\n",
      "TNRC6C_HEK293_Plass\n",
      "LIN28A_H9_POSTAR_CIMS\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.52510098095788\t42\t0\n",
      "7.55568378534333\t42\t26.9497980380842\n",
      "8.5129832660127\t42\t28.1615695326024\n",
      "9.09463358338142\t42\t28.864396999423\n",
      "9.5066358915176\t42\t29.3248701673399\n",
      "9.83381419503751\t42\t29.6884016156953\n",
      "10.1246393537219\t42\t29.9792267743797\n",
      "10.4275822273514\t42\t30.2700519330641\n",
      "10.7911136757069\t42\t30.5851125216388\n",
      "11.33641084824\t42\t30.997114829775\n",
      "16.8378534333526\t42\t31.6757068667051\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6602 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6602\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "600 hits go_terms_num = 790\n",
      "cat 765, mi = 0.00299553, YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "cat 567, mi = 0.00243485, SFPQ_HepG2_POSTAR_eCLIP\n",
      "cat 246, mi = 0.00237182, FXR2_HEK293_POSTAR_PAR-CLIP\n",
      "cat 247, mi = 0.00237182, FXR2_HEK293_POSTAR_PARalyzer\n",
      "cat 373, mi = 0.00237085, LARP4_K562_POSTAR_eCLIP\n",
      "cat 701, mi = 0.00236877, UCHL5_HepG2_POSTAR_eCLIP\n",
      "cat 157, mi = 0.00226956, EIF3A_HEK293T_POSTAR_PARalyzer\n",
      "cat 584, mi = 0.00225240, SRRM4_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 422, mi = 0.00224682, NCBP2_HepG2_POSTAR_eCLIP\n",
      "cat 557, mi = 0.00216033, SERBP1_K562_POSTAR_eCLIP\n",
      "cat 640, mi = 0.00214076, TARDBP_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 169, mi = 0.00213930, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 761, mi = 0.00210555, YTHDF2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 156, mi = 0.00207503, EIF3A_HEK293T_POSTAR_PAR-CLIP\n",
      "cat 130, mi = 0.00207489, DGCR8_HepG2_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 765, mi = 0.003, pvalue = 0.004, YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.002, pvalue = 0.027, SFPQ_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 373, mi = 0.002, pvalue = 0.024, LARP4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 701, mi = 0.002, pvalue = 0.011, UCHL5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 577, mi = 0.002, pvalue = 0.085, SMNDC1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.002, pvalue = 0.082, DICER1_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 612, mi = 0.002, pvalue = 0.070, SUPV3L1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 696, mi = 0.002, pvalue = 0.103, U2AF2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 549, mi = 0.002, pvalue = 0.026, SAFB2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 150, mi = 0.002, pvalue = 0.093, DROSHA_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 252, mi = 0.002, pvalue = 0.142, Fip1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.002, pvalue = 0.107, EFTUD2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 398, mi = 0.002, pvalue = 0.152, LSM11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.002, pvalue = 0.167, CSTF2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 723, mi = 0.002, pvalue = 0.180, WDR33_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 513, mi = 0.002, pvalue = 0.129, RBM10_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 171, mi = 0.002, pvalue = 0.159, EIF3H_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 534, mi = 0.002, pvalue = 0.207, RBM5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.223, AGGF1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 96, mi = 0.002, pvalue = 0.272, CstF-64tau_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 339, mi = 0.002, pvalue = 0.135, IGF2BP1_HepG2_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 44.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.05\t-12.28\n",
      "YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.52510098095788\t42\t0\n",
      "7.55568378534333\t42\t26.9497980380842\n",
      "8.5129832660127\t42\t28.1615695326024\n",
      "9.09463358338142\t42\t28.864396999423\n",
      "9.5066358915176\t42\t29.3248701673399\n",
      "9.83381419503751\t42\t29.6884016156953\n",
      "10.1246393537219\t42\t29.9792267743797\n",
      "10.4275822273514\t42\t30.2700519330641\n",
      "10.7789959607617\t42\t30.5851125216388\n",
      "11.3242931332949\t42\t30.9728793998846\n",
      "16.8378534333526\t42\t31.6757068667051\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6691 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6691\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "608 hits go_terms_num = 790\n",
      "cat 225, mi = 0.00323677, FUS_Brain_POSTAR_CIMS\n",
      "cat 595, mi = 0.00282512, SRSF7_K562_POSTAR_eCLIP\n",
      "cat 382, mi = 0.00277042, LIN28A_HEK293_POSTAR_CIMS\n",
      "cat 345, mi = 0.00276969, IGF2BP2_HEK293_POSTAR_Piranha_0.01\n",
      "cat 185, mi = 0.00273100, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "cat 60, mi = 0.00268762, CPSF-73_HEK293_Plass\n",
      "cat 383, mi = 0.00256430, LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "cat 467, mi = 0.00254376, PRKRA_HEK293_POSTAR_PAR-CLIP\n",
      "cat 468, mi = 0.00254376, PRKRA_HEK293_POSTAR_PARalyzer\n",
      "cat 124, mi = 0.00251366, DDX6_K562_POSTAR_eCLIP\n",
      "cat 221, mi = 0.00249823, FTO_K562_POSTAR_eCLIP\n",
      "cat 579, mi = 0.00248889, SND1_HepG2_POSTAR_eCLIP\n",
      "cat 19, mi = 0.00243928, ALKBH5_HEK293_POSTAR_PAR-CLIP\n",
      "cat 492, mi = 0.00241628, PUM2_HEK293_POSTAR_PAR-CLIP\n",
      "cat 411, mi = 0.00240034, METTL14_HeLa_POSTAR_PARalyzer\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 225, mi = 0.003, pvalue = 0.000, FUS_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 595, mi = 0.003, pvalue = 0.006, SRSF7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 382, mi = 0.003, pvalue = 0.006, LIN28A_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 345, mi = 0.003, pvalue = 0.007, IGF2BP2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.003, pvalue = 0.009, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 60, mi = 0.003, pvalue = 0.010, CPSF-73_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 383, mi = 0.003, pvalue = 0.014, LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 467, mi = 0.003, pvalue = 0.010, PRKRA_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 468, mi = 0.003, pvalue = 0.009, PRKRA_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.003, pvalue = 0.010, DDX6_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 221, mi = 0.002, pvalue = 0.018, FTO_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 579, mi = 0.002, pvalue = 0.016, SND1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.002, pvalue = 0.016, ALKBH5_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 492, mi = 0.002, pvalue = 0.024, PUM2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 411, mi = 0.002, pvalue = 0.026, METTL14_HeLa_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 258, mi = 0.002, pvalue = 0.015, GPKOW_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 493, mi = 0.002, pvalue = 0.029, PUM2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 573, mi = 0.002, pvalue = 0.019, SLTM_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 446, mi = 0.002, pvalue = 0.029, NOP58_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 157, mi = 0.002, pvalue = 0.020, EIF3A_HEK293T_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.002, pvalue = 0.025, ALKBH5_HEK293_POSTAR_PARalyzer\n",
      "Total number of categories evaluated = 11.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.05\t-12.28\n",
      "FUS_Brain_POSTAR_CIMS\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.52510098095788\t42\t0\n",
      "7.55568378534333\t42\t26.9497980380842\n",
      "8.5129832660127\t42\t28.1615695326024\n",
      "9.09463358338142\t42\t28.864396999423\n",
      "9.5066358915176\t42\t29.3248701673399\n",
      "9.83381419503751\t42\t29.6884016156953\n",
      "10.1246393537219\t42\t29.9792267743797\n",
      "10.4275822273514\t42\t30.2700519330641\n",
      "10.7789959607617\t42\t30.5851125216388\n",
      "11.3242931332949\t42\t30.9728793998846\n",
      "16.8378534333526\t42\t31.6757068667051\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 790\n",
      "cat 676, mi = 0.00295406, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 705, mi = 0.00278213, UPF1_HEK293_Plass\n",
      "cat 447, mi = 0.00232096, NPM1_K562_POSTAR_eCLIP\n",
      "cat 195, mi = 0.00228913, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00228913, EZH2_HCT116_POSTAR_iCLIP\n",
      "cat 31, mi = 0.00215996, AUH_K562_POSTAR_eCLIP\n",
      "cat 275, mi = 0.00211034, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 449, mi = 0.00209365, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 97, mi = 0.00201818, DCK1_HEK293_Plass\n",
      "cat 402, mi = 0.00201308, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 674, mi = 0.00200729, TNRC6C_HEK293_POSTAR_PAR-CLIP\n",
      "cat 377, mi = 0.00196054, LARP7_K562_POSTAR_eCLIP\n",
      "cat 569, mi = 0.00190796, SLBP_K562_POSTAR_eCLIP\n",
      "cat 236, mi = 0.00184958, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 428, mi = 0.00182647, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 676, mi = 0.003, pvalue = 0.005, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 705, mi = 0.003, pvalue = 0.010, UPF1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 447, mi = 0.002, pvalue = 0.031, NPM1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.031, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.002, pvalue = 0.030, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.058, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.076, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.002, pvalue = 0.086, DCK1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.115, SLBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.147, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 353, mi = 0.002, pvalue = 0.114, IGF2BP3_PL45_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.002, pvalue = 0.160, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.002, pvalue = 0.140, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.160, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 538, mi = 0.002, pvalue = 0.181, RPS11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.321, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 456, mi = 0.002, pvalue = 0.287, PAPD5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 673, mi = 0.001, pvalue = 0.273, TNRC6B_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 446, mi = 0.001, pvalue = 0.318, NOP58_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.001, pvalue = 0.480, AGO4_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.001, pvalue = 0.349, C17orf85_HEK293_Plass\n",
      "Total number of categories evaluated = 25.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.48\t-12.28\n",
      "TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.06682577565632\t42\t0\n",
      "8.48031026252983\t42\t27.8663484486874\n",
      "9.47016706443915\t42\t29.0942720763723\n",
      "10.0841288782816\t42\t29.846062052506\n",
      "10.5101431980907\t42\t30.3221957040573\n",
      "10.8359188544153\t42\t30.6980906921241\n",
      "11.1241050119332\t42\t30.9737470167064\n",
      "11.4373508353222\t42\t31.2744630071599\n",
      "11.813245823389\t42\t31.6002386634845\n",
      "12.3770883054893\t42\t32.0262529832936\n",
      "17.3639618138425\t42\t32.727923627685\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "read_go_names_data: please enter a valid filename (/flash/hani/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_coding_gene_names/human_ensembl_RBPs_coding_gene_names_names.txt invalid)\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 101\n",
      "cat 34, mi = 0.00144682, KHDRBS3\n",
      "cat 69, mi = 0.00136561, RBMS3.1\n",
      "cat 62, mi = 0.00135367, RBM46\n",
      "cat 27, mi = 0.00128402, HuR.2\n",
      "cat 9, mi = 0.00117787, ESRP2\n",
      "cat 6, mi = 0.00097300, CPEB4\n",
      "cat 68, mi = 0.00091532, RBMS3\n",
      "cat 13, mi = 0.00090150, FXR2\n",
      "cat 21, mi = 0.00070426, HNRNPK\n",
      "cat 88, mi = 0.00064493, SRSF9.1\n",
      "cat 56, mi = 0.00032336, RBM3\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "cat 3, mi = 0.00000000, BRUNOL6\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.288, KHDRBS3\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.345, RBMS3.1\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.402, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.404, HuR.2\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.626, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.647, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.001, pvalue = 0.664, RBMS3\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.805, FXR2\n",
      "Didn't pass: cat 21, mi = 0.001, pvalue = 1.000, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.001, pvalue = 0.863, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.000, pvalue = 0.985, RBM3\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, FMR1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.48\t-12.28\n",
      "Outputing EPS file deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/Virma_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/Virma_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/Virma_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/Virma_peak_m6A_vs_input_human_ensembl_RBPs_DeepBind/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7509 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7509\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "683 hits go_terms_num = 17689\n",
      "cat 550, mi = 0.00318730, chromatin remodeling\n",
      "cat 1609, mi = 0.00301277, positive regulation of multicellular organism growth\n",
      "cat 3210, mi = 0.00297899, homeostasis of number of cells within a tissue\n",
      "cat 3512, mi = 0.00287027, positive regulation by host of viral transcription\n",
      "cat 589, mi = 0.00283424, ossification\n",
      "cat 3739, mi = 0.00280109, calcium channel activity\n",
      "cat 4263, mi = 0.00277069, response to iron ion\n",
      "cat 1502, mi = 0.00276710, regulation of mitotic spindle organization\n",
      "cat 1684, mi = 0.00270643, costamere\n",
      "cat 2170, mi = 0.00252881, T cell homeostasis\n",
      "cat 629, mi = 0.00251673, regulation of ion transmembrane transport\n",
      "cat 782, mi = 0.00248305, endoplasmic reticulum-Golgi intermediate compartment membrane\n",
      "cat 4405, mi = 0.00246964, cellular response to organic substance\n",
      "cat 3671, mi = 0.00244319, cation transport\n",
      "cat 1647, mi = 0.00242749, enzyme activator activity\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 550, mi = 0.003, pvalue = 0.001, chromatin remodeling\n",
      "Passed: cat 3210, mi = 0.003, pvalue = 0.001, homeostasis of number of cells within a tissue\n",
      "Passed: cat 3739, mi = 0.003, pvalue = 0.001, calcium channel activity\n",
      "Passed: cat 1684, mi = 0.003, pvalue = 0.002, costamere\n",
      "Didn't pass (cutoff=0.0050): cat 782, mi = 0.002, pvalue = 0.007, endoplasmic reticulum-Golgi intermediate compartment membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1647, mi = 0.002, pvalue = 0.008, enzyme activator activity\n",
      "Passed: cat 42, mi = 0.002, pvalue = 0.001, copper ion binding\n",
      "Didn't pass (cutoff=0.0050): cat 1621, mi = 0.002, pvalue = 0.009, cellular response to transforming growth factor beta stimulus\n",
      "Didn't pass (cutoff=0.0050): cat 359, mi = 0.002, pvalue = 0.006, negative regulation of neuron death\n",
      "Didn't pass (cutoff=0.0050): cat 1693, mi = 0.002, pvalue = 0.006, response to peptide hormone\n",
      "Didn't pass (cutoff=0.0050): cat 264, mi = 0.002, pvalue = 0.014, positive regulation of translation\n",
      "Didn't pass (cutoff=0.0050): cat 5165, mi = 0.002, pvalue = 0.013, mitotic cytokinesis\n",
      "Didn't pass (cutoff=0.0050): cat 5700, mi = 0.002, pvalue = 0.014, myeloid cell differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 6314, mi = 0.002, pvalue = 0.013, tRNA methylation\n",
      "Didn't pass (cutoff=0.0050): cat 3485, mi = 0.002, pvalue = 0.014, negative regulation of MAP kinase activity\n",
      "Passed: cat 82, mi = 0.002, pvalue = 0.001, response to heat\n",
      "Passed: cat 4552, mi = 0.002, pvalue = 0.002, GTP-Rho binding\n",
      "Didn't pass (cutoff=0.0050): cat 4101, mi = 0.002, pvalue = 0.008, positive regulation of hepatocyte proliferation\n",
      "Didn't pass (cutoff=0.0050): cat 1766, mi = 0.002, pvalue = 0.020, receptor tyrosine kinase binding\n",
      "Passed: cat 998, mi = 0.002, pvalue = 0.001, positive regulation of insulin receptor signaling pathway\n",
      "Passed: cat 4097, mi = 0.002, pvalue = 0.004, cellular response to peptide hormone stimulus\n",
      "Didn't pass (cutoff=0.0050): cat 3388, mi = 0.002, pvalue = 0.025, osteoblast differentiation\n",
      "Passed: cat 1883, mi = 0.002, pvalue = 0.004, mitochondrial calcium ion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 696, mi = 0.002, pvalue = 0.023, mitochondrial large ribosomal subunit\n",
      "Didn't pass (cutoff=0.0050): cat 6753, mi = 0.002, pvalue = 0.007, regulation of skeletal muscle satellite cell proliferation\n",
      "Didn't pass (cutoff=0.0050): cat 189, mi = 0.002, pvalue = 0.021, embryonic limb morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 806, mi = 0.002, pvalue = 0.026, response to endoplasmic reticulum stress\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.002, pvalue = 0.026, integrin binding\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.012, cell chemotaxis\n",
      "Didn't pass (cutoff=0.0050): cat 1249, mi = 0.002, pvalue = 0.012, small-subunit processome\n",
      "Didn't pass (cutoff=0.0050): cat 563, mi = 0.002, pvalue = 0.031, calcium-mediated signaling\n",
      "Didn't pass (cutoff=0.0050): cat 1906, mi = 0.002, pvalue = 0.023, lipoprotein metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 4073, mi = 0.002, pvalue = 0.031, mesoderm development\n",
      "Didn't pass (cutoff=0.0050): cat 2500, mi = 0.002, pvalue = 0.035, manganese ion binding\n",
      "Didn't pass (cutoff=0.0050): cat 1527, mi = 0.002, pvalue = 0.035, double-strand break repair via homologous recombination\n",
      "Didn't pass (cutoff=0.0050): cat 5305, mi = 0.002, pvalue = 0.005, positive regulation of viral transcription\n",
      "Didn't pass (cutoff=0.0050): cat 779, mi = 0.002, pvalue = 0.007, antigen processing and presentation of exogenous peptide antigen via MHC class I, TAP-dependent\n",
      "Didn't pass (cutoff=0.0050): cat 3196, mi = 0.002, pvalue = 0.017, translational elongation\n",
      "Didn't pass (cutoff=0.0050): cat 1678, mi = 0.002, pvalue = 0.019, regulation of TOR signaling\n",
      "Didn't pass (cutoff=0.0050): cat 3071, mi = 0.002, pvalue = 0.035, cytokine production\n",
      "Didn't pass (cutoff=0.0050): cat 378, mi = 0.002, pvalue = 0.038, negative regulation of BMP signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 1283, mi = 0.002, pvalue = 0.015, regulation of androgen receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 6940, mi = 0.002, pvalue = 0.015, uterus morphogenesis\n",
      "Total number of categories evaluated = 76.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 5, 7, 0.105077161959632\n",
      "best merge is 4, 8, 0.106462217030548\n",
      "best merge is 2, 6, 0.278466769194812\n",
      "best merge is 4, 9, 0.310741848078472\n",
      "best merge is 1, 4, 0.386068286524756\n",
      "best merge is 0, 3, 0.839496268347489\n",
      "order classes\n",
      "4 - 0.589422116488049\n",
      "2 - 0.69616692298703\n",
      "2 - 0.262692904899079\n",
      "5 7\n",
      "1 4 8 9\n",
      "2 6\n",
      "3\n",
      "0\n",
      "Done.Start drawing\n",
      "8.75\t-11.06\n",
      "response to heat, GO:0009408\n",
      "positive regulation of insulin receptor signaling pathway, GO:0046628\n",
      "chromatin remodeling, GO:0006338\n",
      "costamere, GO:0043034\n",
      "GTP-Rho binding, GO:0017049\n",
      "homeostasis of number of cells within a tissue, GO:0048873\n",
      "copper ion binding, GO:0005507\n",
      "cellular response to peptide hormone stimulus, GO:0071375\n",
      "mitochondrial calcium ion transmembrane transport, GO:0006851\n",
      "calcium channel activity, GO:0005262\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 5, 7, 0.105077161959632\n",
      "best merge is 4, 8, 0.106462217030548\n",
      "best merge is 2, 6, 0.278466769194812\n",
      "best merge is 4, 9, 0.310741848078472\n",
      "best merge is 1, 4, 0.386068286524756\n",
      "best merge is 0, 3, 0.839496268347489\n",
      "order classes\n",
      "4 - 0.589422116488049\n",
      "2 - 0.69616692298703\n",
      "2 - 0.262692904899079\n",
      "5 7\n",
      "1 4 8 9\n",
      "2 6\n",
      "3\n",
      "0\n",
      "Done.-10.0035335689046\t42\t0\n",
      "0.692579505300351\t42\t19.9929328621908\n",
      "1.80565371024735\t42\t21.3922261484099\n",
      "2.51590106007067\t42\t22.2190812720848\n",
      "3.04593639575972\t42\t22.8127208480565\n",
      "3.48056537102473\t42\t23.2791519434629\n",
      "3.87279151943463\t42\t23.6819787985866\n",
      "4.24381625441696\t42\t24.0636042402827\n",
      "4.65724381625441\t42\t24.4240282685512\n",
      "5.19787985865724\t42\t24.8904593639576\n",
      "13.7526501766784\t42\t25.5053003533569\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6496 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6496\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "591 hits go_terms_num = 181\n",
      "cat 115, mi = 0.00175138, PPARGC1A_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00174331, BRF2_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00167465, SMARCA4_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00159692, POU5F1_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00144710, BRF1_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00104004, BDP1_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00101828, POLR3G_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00075718, XRCC4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.002, pvalue = 0.116, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.002, pvalue = 0.113, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.002, pvalue = 0.222, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.002, pvalue = 0.179, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.387, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.602, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.792, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass: cat 167, mi = 0.001, pvalue = 1.000, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.44\t-11.06\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_encode_tf/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 326\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t326\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 326\n",
      "cat 171, mi = 0.00307348, chr2q33\n",
      "cat 146, mi = 0.00297977, chr22q12\n",
      "cat 126, mi = 0.00291140, chr1q24\n",
      "cat 174, mi = 0.00289722, chr2q36\n",
      "cat 139, mi = 0.00278883, chr20q13\n",
      "cat 114, mi = 0.00271355, chr1p22\n",
      "cat 129, mi = 0.00268778, chr1q32\n",
      "cat 229, mi = 0.00244894, chr5q12\n",
      "cat 73, mi = 0.00241449, chr15q26\n",
      "cat 304, mi = 0.00239130, chr9q22\n",
      "cat 12, mi = 0.00237452, chr10q26\n",
      "cat 119, mi = 0.00236283, chr1p35\n",
      "cat 90, mi = 0.00226675, chr17q12\n",
      "cat 79, mi = 0.00226668, chr16q12\n",
      "cat 256, mi = 0.00226663, chr6q22\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 171, mi = 0.003, pvalue = 0.003, chr2q33\n",
      "Passed: cat 146, mi = 0.003, pvalue = 0.004, chr22q12\n",
      "Passed: cat 126, mi = 0.003, pvalue = 0.004, chr1q24\n",
      "Passed: cat 174, mi = 0.003, pvalue = 0.002, chr2q36\n",
      "Didn't pass (cutoff=0.0050): cat 139, mi = 0.003, pvalue = 0.006, chr20q13\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.003, pvalue = 0.009, chr1p22\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.003, pvalue = 0.011, chr1q32\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.002, pvalue = 0.011, chr5q12\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.002, pvalue = 0.027, chr15q26\n",
      "Didn't pass (cutoff=0.0050): cat 304, mi = 0.002, pvalue = 0.024, chr9q22\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.002, pvalue = 0.027, chr10q26\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.002, pvalue = 0.030, chr1p35\n",
      "Didn't pass (cutoff=0.0050): cat 90, mi = 0.002, pvalue = 0.035, chr17q12\n",
      "Didn't pass (cutoff=0.0050): cat 79, mi = 0.002, pvalue = 0.037, chr16q12\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.002, pvalue = 0.036, chr6q22\n",
      "Didn't pass (cutoff=0.0050): cat 205, mi = 0.002, pvalue = 0.032, chr4p16\n",
      "Didn't pass (cutoff=0.0050): cat 296, mi = 0.002, pvalue = 0.039, chr9p13\n",
      "Didn't pass (cutoff=0.0050): cat 190, mi = 0.002, pvalue = 0.046, chr3q13\n",
      "Didn't pass (cutoff=0.0050): cat 102, mi = 0.002, pvalue = 0.045, chr18q23\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.002, pvalue = 0.059, chr15q22\n",
      "Didn't pass (cutoff=0.0050): cat 243, mi = 0.002, pvalue = 0.026, chr6p12\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.002, pvalue = 0.055, chr1p34\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.002, pvalue = 0.080, chr13q12\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.002, pvalue = 0.050, chr2p25\n",
      "Total number of categories evaluated = 15.\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.294711930780006\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "8.75\t-11.06\n",
      "chr2q36\n",
      "chr1q24\n",
      "chr2q33\n",
      "chr22q12\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.294711930780006\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-9.98233215547703\t42\t0\n",
      "0.745583038869256\t42\t20.0353356890459\n",
      "1.85865724381625\t42\t21.4558303886926\n",
      "2.53710247349823\t42\t22.2614840989399\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.45936395759717\t42\t23.2579505300353\n",
      "3.85159010600707\t42\t23.660777385159\n",
      "4.23321554770318\t42\t24.0424028268551\n",
      "4.64664310954063\t42\t24.4240282685512\n",
      "5.1660777385159\t42\t24.86925795053\n",
      "13.7314487632509\t42\t25.4628975265018\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_c1/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4738\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t4738\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 4738\n",
      "cat 4356, mi = 0.00372941, WANG_BARRETTS_ESOPHAGUS_UP\n",
      "cat 3709, mi = 0.00353409, RIGGI_EWING_SARCOMA_PROGENITOR_DN\n",
      "cat 1541, mi = 0.00334403, KAMMINGA_EZH2_TARGETS\n",
      "cat 2719, mi = 0.00329213, PHONG_TNF_TARGETS_UP\n",
      "cat 2968, mi = 0.00329159, QI_HYPOXIA\n",
      "cat 2812, mi = 0.00322122, PID_IL2_1PATHWAY\n",
      "cat 1370, mi = 0.00320599, HONRADO_BREAST_CANCER_BRCA1_VS_BRCA2\n",
      "cat 483, mi = 0.00320243, BROCKE_APOPTOSIS_REVERSED_BY_IL6\n",
      "cat 3753, mi = 0.00313364, ROSTY_CERVICAL_CANCER_PROLIFERATION_CLUSTER\n",
      "cat 828, mi = 0.00308151, DAZARD_UV_RESPONSE_CLUSTER_G6\n",
      "cat 477, mi = 0.00304962, BREDEMEYER_RAG_SIGNALING_NOT_VIA_ATM_DN\n",
      "cat 2527, mi = 0.00295667, NIELSEN_MALIGNAT_FIBROUS_HISTIOCYTOMA_DN\n",
      "cat 454, mi = 0.00294078, BOYAULT_LIVER_CANCER_SUBCLASS_G5_DN\n",
      "cat 4173, mi = 0.00294078, TIAN_TNF_SIGNALING_NOT_VIA_NFKB\n",
      "cat 3719, mi = 0.00291602, RIZ_ERYTHROID_DIFFERENTIATION_CCNE1\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 4356, mi = 0.004, pvalue = 0.001, WANG_BARRETTS_ESOPHAGUS_UP\n",
      "Passed: cat 3709, mi = 0.004, pvalue = 0.001, RIGGI_EWING_SARCOMA_PROGENITOR_DN\n",
      "Passed: cat 1541, mi = 0.003, pvalue = 0.000, KAMMINGA_EZH2_TARGETS\n",
      "Passed: cat 2719, mi = 0.003, pvalue = 0.001, PHONG_TNF_TARGETS_UP\n",
      "Passed: cat 2968, mi = 0.003, pvalue = 0.001, QI_HYPOXIA\n",
      "Passed: cat 2812, mi = 0.003, pvalue = 0.001, PID_IL2_1PATHWAY\n",
      "Passed: cat 477, mi = 0.003, pvalue = 0.002, BREDEMEYER_RAG_SIGNALING_NOT_VIA_ATM_DN\n",
      "Passed: cat 454, mi = 0.003, pvalue = 0.000, BOYAULT_LIVER_CANCER_SUBCLASS_G5_DN\n",
      "Passed: cat 3719, mi = 0.003, pvalue = 0.001, RIZ_ERYTHROID_DIFFERENTIATION_CCNE1\n",
      "Passed: cat 4040, mi = 0.003, pvalue = 0.005, STARK_HYPPOCAMPUS_22Q11_DELETION_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1131, mi = 0.003, pvalue = 0.005, GEORGANTAS_HSC_MARKERS\n",
      "Passed: cat 4569, mi = 0.003, pvalue = 0.004, YANG_BREAST_CANCER_ESR1_UP\n",
      "Passed: cat 1120, mi = 0.003, pvalue = 0.003, GENTILE_UV_LOW_DOSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1945, mi = 0.003, pvalue = 0.006, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2215, mi = 0.003, pvalue = 0.009, MATTIOLI_MGUS_VS_PCL\n",
      "Didn't pass (cutoff=0.0050): cat 1417, mi = 0.003, pvalue = 0.009, HU_GENOTOXIN_ACTION_DIRECT_VS_INDIRECT_24HR\n",
      "Didn't pass (cutoff=0.0050): cat 3686, mi = 0.003, pvalue = 0.008, REN_ALVEOLAR_RHABDOMYOSARCOMA_UP\n",
      "Didn't pass (cutoff=0.0050): cat 973, mi = 0.003, pvalue = 0.013, FIGUEROA_AML_METHYLATION_CLUSTER_4_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1289, mi = 0.003, pvalue = 0.010, HEIDENBLAD_AMPLICON_12P11_12_DN\n",
      "Passed: cat 4619, mi = 0.003, pvalue = 0.001, ZEILSTRA_CD44_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2516, mi = 0.003, pvalue = 0.015, NIELSEN_GIST\n",
      "Didn't pass (cutoff=0.0050): cat 1393, mi = 0.003, pvalue = 0.014, HUANG_DASATINIB_RESISTANCE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2380, mi = 0.002, pvalue = 0.016, MOREAUX_B_LYMPHOCYTE_MATURATION_BY_TACI_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4310, mi = 0.002, pvalue = 0.018, VERHAAK_GLIOBLASTOMA_NEURAL\n",
      "Didn't pass (cutoff=0.0050): cat 4583, mi = 0.002, pvalue = 0.021, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_14\n",
      "Didn't pass (cutoff=0.0050): cat 2884, mi = 0.002, pvalue = 0.014, PID_S1P_META_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 1691, mi = 0.002, pvalue = 0.025, KEGG_NOTCH_SIGNALING_PATHWAY\n",
      "Passed: cat 3435, mi = 0.002, pvalue = 0.001, REACTOME_PHOSPHORYLATION_OF_CD3_AND_TCR_ZETA_CHAINS\n",
      "Didn't pass (cutoff=0.0050): cat 1000, mi = 0.002, pvalue = 0.009, FLOTHO_PEDIATRIC_ALL_THERAPY_RESPONSE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1840, mi = 0.002, pvalue = 0.010, KORKOLA_EMBRYONAL_CARCINOMA_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2693, mi = 0.002, pvalue = 0.019, PENG_LEUCINE_DEPRIVATION_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2028, mi = 0.002, pvalue = 0.019, LINDGREN_BLADDER_CANCER_WITH_LOH_IN_CHR9Q\n",
      "Didn't pass (cutoff=0.0050): cat 1964, mi = 0.002, pvalue = 0.027, LEE_LIVER_CANCER_SURVIVAL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2680, mi = 0.002, pvalue = 0.034, PEDERSEN_METASTASIS_BY_ERBB2_ISOFORM_4\n",
      "Didn't pass (cutoff=0.0050): cat 4612, mi = 0.002, pvalue = 0.036, YUAN_ZNF143_PARTNERS\n",
      "Didn't pass (cutoff=0.0050): cat 262, mi = 0.002, pvalue = 0.005, BIOCARTA_EIF_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 4298, mi = 0.002, pvalue = 0.008, VARELA_ZMPSTE24_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2710, mi = 0.002, pvalue = 0.039, PETROVA_ENDOTHELIUM_LYMPHATIC_VS_BLOOD_UP\n",
      "Didn't pass (cutoff=0.0050): cat 591, mi = 0.002, pvalue = 0.033, CAMPS_COLON_CANCER_COPY_NUMBER_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3254, mi = 0.002, pvalue = 0.042, REACTOME_G_ALPHA_I_SIGNALLING_EVENTS\n",
      "Didn't pass (cutoff=0.0050): cat 1770, mi = 0.002, pvalue = 0.041, KESHELAVA_MULTIPLE_DRUG_RESISTANCE\n",
      "Didn't pass (cutoff=0.0050): cat 4715, mi = 0.002, pvalue = 0.036, ZHU_CMV_24_HR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3160, mi = 0.002, pvalue = 0.039, REACTOME_DIABETES_PATHWAYS\n",
      "Didn't pass (cutoff=0.0050): cat 1187, mi = 0.002, pvalue = 0.041, GRAESSMANN_RESPONSE_TO_MC_AND_SERUM_DEPRIVATION_DN\n",
      "Didn't pass (cutoff=0.0050): cat 673, mi = 0.002, pvalue = 0.034, CHIANG_LIVER_CANCER_SUBCLASS_INTERFERON_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3750, mi = 0.002, pvalue = 0.046, ROSS_AML_WITH_MLL_FUSIONS\n",
      "Didn't pass (cutoff=0.0050): cat 2295, mi = 0.002, pvalue = 0.020, MELLMAN_TUT1_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3337, mi = 0.002, pvalue = 0.038, REACTOME_METABOLISM_OF_AMINO_ACIDS_AND_DERIVATIVES\n",
      "Total number of categories evaluated = 159.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 8, 0.126997281378515\n",
      "best merge is 10, 11, 0.326744624013317\n",
      "best merge is 1, 12, 0.354894683596659\n",
      "best merge is 2, 3, 0.410640633130732\n",
      "best merge is 5, 9, 0.450768343857375\n",
      "best merge is 0, 4, 0.549177843008787\n",
      "best merge is 5, 10, 0.740037236125825\n",
      "best merge is 1, 13, 0.741068580701931\n",
      "best merge is 2, 5, 0.764847977738421\n",
      "best merge is 0, 6, 0.932553884104999\n",
      "order classes\n",
      "2 - 1.92212245053075\n",
      "3 - 1.90507006148202\n",
      "7 - 1.13176458717347\n",
      "2 8 3 5 9 10 11\n",
      "1 12 13\n",
      "0 4\n",
      "6\n",
      "7\n",
      "Done.Start drawing\n",
      "8.75\t-11.06\n",
      "PID_IL2_1PATHWAY\n",
      "STARK_HYPPOCAMPUS_22Q11_DELETION_UP\n",
      "RIZ_ERYTHROID_DIFFERENTIATION_CCNE1\n",
      "BOYAULT_LIVER_CANCER_SUBCLASS_G5_DN\n",
      "KAMMINGA_EZH2_TARGETS\n",
      "PHONG_TNF_TARGETS_UP\n",
      "YANG_BREAST_CANCER_ESR1_UP\n",
      "GENTILE_UV_LOW_DOSE_UP\n",
      "WANG_BARRETTS_ESOPHAGUS_UP\n",
      "RIGGI_EWING_SARCOMA_PROGENITOR_DN\n",
      "ZEILSTRA_CD44_TARGETS_DN\n",
      "QI_HYPOXIA\n",
      "REACTOME_PHOSPHORYLATION_OF_CD3_AND_TCR_ZETA_CHAINS\n",
      "BREDEMEYER_RAG_SIGNALING_NOT_VIA_ATM_DN\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 8, 0.126997281378515\n",
      "best merge is 10, 11, 0.326744624013317\n",
      "best merge is 1, 12, 0.354894683596659\n",
      "best merge is 2, 3, 0.410640633130732\n",
      "best merge is 5, 9, 0.450768343857375\n",
      "best merge is 0, 4, 0.549177843008787\n",
      "best merge is 5, 10, 0.740037236125825\n",
      "best merge is 1, 13, 0.741068580701931\n",
      "best merge is 2, 5, 0.764847977738421\n",
      "best merge is 0, 6, 0.932553884104999\n",
      "order classes\n",
      "2 - 1.92212245053075\n",
      "3 - 1.90507006148202\n",
      "7 - 1.13176458717347\n",
      "2 8 3 5 9 10 11\n",
      "1 12 13\n",
      "0 4\n",
      "6\n",
      "7\n",
      "Done.-9.98233215547703\t42\t0\n",
      "0.745583038869256\t42\t20.0353356890459\n",
      "1.85865724381625\t42\t21.4558303886926\n",
      "2.53710247349823\t42\t22.2614840989399\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.45936395759717\t42\t23.2579505300353\n",
      "3.85159010600707\t42\t23.660777385159\n",
      "4.23321554770318\t42\t24.0424028268551\n",
      "4.64664310954063\t42\t24.4240282685512\n",
      "5.1660777385159\t42\t24.86925795053\n",
      "13.7314487632509\t42\t25.4628975265018\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_c2/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 836\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t836\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 836\n",
      "cat 110, mi = 0.00320736, ATAAGCT_MIR21\n",
      "cat 132, mi = 0.00293245, ATGTCAC_MIR489\n",
      "cat 41, mi = 0.00291752, ACTGAAA_MIR30A3P_MIR30E3P\n",
      "cat 457, mi = 0.00285385, HNF3_Q6\n",
      "cat 125, mi = 0.00263638, ATGCACG_MIR517B\n",
      "cat 431, mi = 0.00263079, GTGTTGA_MIR505\n",
      "cat 174, mi = 0.00250138, CCATCCA_MIR432\n",
      "cat 494, mi = 0.00248110, LXR_DR4_Q3\n",
      "cat 818, mi = 0.00247857, YRCCAKNNGNCGC_UNKNOWN\n",
      "cat 298, mi = 0.00241185, EN1_01\n",
      "cat 246, mi = 0.00233012, CTCTGGA_MIR520A_MIR525\n",
      "cat 400, mi = 0.00231078, GGTGTGT_MIR329\n",
      "cat 126, mi = 0.00229558, ATGCAGT_MIR217\n",
      "cat 408, mi = 0.00228986, GTAAGAT_MIR200A\n",
      "cat 585, mi = 0.00228254, PAX5_01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 110, mi = 0.003, pvalue = 0.002, ATAAGCT_MIR21\n",
      "Passed: cat 132, mi = 0.003, pvalue = 0.004, ATGTCAC_MIR489\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.003, pvalue = 0.006, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 174, mi = 0.003, pvalue = 0.019, CCATCCA_MIR432\n",
      "Didn't pass (cutoff=0.0050): cat 818, mi = 0.002, pvalue = 0.012, YRCCAKNNGNCGC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 298, mi = 0.002, pvalue = 0.023, EN1_01\n",
      "Didn't pass (cutoff=0.0050): cat 400, mi = 0.002, pvalue = 0.025, GGTGTGT_MIR329\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.002, pvalue = 0.024, ATGCAGT_MIR217\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.002, pvalue = 0.038, CCAWWNAAGG_SRF_Q4\n",
      "Didn't pass (cutoff=0.0050): cat 772, mi = 0.002, pvalue = 0.026, TTCCGTT_MIR191\n",
      "Didn't pass (cutoff=0.0050): cat 258, mi = 0.002, pvalue = 0.030, CYTAGCAAY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 692, mi = 0.002, pvalue = 0.024, TAAYNRNNTCC_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 388, mi = 0.002, pvalue = 0.026, GGGACCA_MIR133A_MIR133B\n",
      "Didn't pass (cutoff=0.0050): cat 513, mi = 0.002, pvalue = 0.036, MSX1_01\n",
      "Didn't pass (cutoff=0.0050): cat 436, mi = 0.002, pvalue = 0.041, GTTRYCATRR_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 746, mi = 0.002, pvalue = 0.053, TGCAAAC_MIR452\n",
      "Didn't pass (cutoff=0.0050): cat 708, mi = 0.002, pvalue = 0.058, TAXCREB_02\n",
      "Didn't pass (cutoff=0.0050): cat 245, mi = 0.002, pvalue = 0.067, CTCTATG_MIR368\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.075, CTAGGAA_MIR384\n",
      "Didn't pass (cutoff=0.0050): cat 181, mi = 0.002, pvalue = 0.065, CCCNNNNNNAAGWT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 715, mi = 0.002, pvalue = 0.076, TCCCCAC_MIR491\n",
      "Didn't pass (cutoff=0.0050): cat 401, mi = 0.002, pvalue = 0.095, GKCGCNNNNNNNTGAYG_UNKNOWN\n",
      "Total number of categories evaluated = 28.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.75\t-11.06\n",
      "ATAAGCT_MIR21\n",
      "ATGTCAC_MIR489\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-9.98233215547703\t42\t0\n",
      "0.745583038869256\t42\t20.0353356890459\n",
      "1.85865724381625\t42\t21.4558303886926\n",
      "2.53710247349823\t42\t22.2614840989399\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.45936395759717\t42\t23.2579505300353\n",
      "3.85159010600707\t42\t23.660777385159\n",
      "4.23321554770318\t42\t24.0424028268551\n",
      "4.64664310954063\t42\t24.4240282685512\n",
      "5.1660777385159\t42\t24.86925795053\n",
      "13.7314487632509\t42\t25.4628975265018\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_c3/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 858\n",
      "cat 211, mi = 0.00362094, GNF2_SNRK\n",
      "cat 92, mi = 0.00314960, GNF2_ATM\n",
      "cat 772, mi = 0.00310986, MORF_NME2\n",
      "cat 668, mi = 0.00308183, MORF_ANP32B\n",
      "cat 474, mi = 0.00299914, MODULE_36\n",
      "cat 156, mi = 0.00286080, GNF2_JAK1\n",
      "cat 843, mi = 0.00282546, MORF_THRA\n",
      "cat 598, mi = 0.00282438, MODULE_537\n",
      "cat 193, mi = 0.00260568, GNF2_RAB7L1\n",
      "cat 794, mi = 0.00260417, MORF_PRKAR1A\n",
      "cat 372, mi = 0.00257941, MODULE_239\n",
      "cat 125, mi = 0.00254349, GNF2_DAP3\n",
      "cat 190, mi = 0.00247562, GNF2_PTPRC\n",
      "cat 751, mi = 0.00246401, MORF_JUND\n",
      "cat 73, mi = 0.00245249, GCM_RING1\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 211, mi = 0.004, pvalue = 0.000, GNF2_SNRK\n",
      "Passed: cat 772, mi = 0.003, pvalue = 0.000, MORF_NME2\n",
      "Passed: cat 474, mi = 0.003, pvalue = 0.003, MODULE_36\n",
      "Passed: cat 843, mi = 0.003, pvalue = 0.003, MORF_THRA\n",
      "Didn't pass (cutoff=0.0050): cat 372, mi = 0.003, pvalue = 0.012, MODULE_239\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.002, pvalue = 0.021, GCM_RING1\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.002, pvalue = 0.027, GCM_MAP1B\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.002, pvalue = 0.012, GCM_AQP4\n",
      "Didn't pass (cutoff=0.0050): cat 204, mi = 0.002, pvalue = 0.030, GNF2_S100A4\n",
      "Didn't pass (cutoff=0.0050): cat 65, mi = 0.002, pvalue = 0.029, GCM_PTPRD\n",
      "Didn't pass (cutoff=0.0050): cat 454, mi = 0.002, pvalue = 0.034, MODULE_332\n",
      "Didn't pass (cutoff=0.0050): cat 324, mi = 0.002, pvalue = 0.024, MODULE_188\n",
      "Didn't pass (cutoff=0.0050): cat 98, mi = 0.002, pvalue = 0.030, GNF2_CARD15\n",
      "Didn't pass (cutoff=0.0050): cat 386, mi = 0.002, pvalue = 0.034, MODULE_254\n",
      "Didn't pass (cutoff=0.0050): cat 102, mi = 0.002, pvalue = 0.017, GNF2_CBFB\n",
      "Didn't pass (cutoff=0.0050): cat 143, mi = 0.002, pvalue = 0.037, GNF2_HCK\n",
      "Didn't pass (cutoff=0.0050): cat 434, mi = 0.002, pvalue = 0.038, MODULE_312\n",
      "Didn't pass (cutoff=0.0050): cat 571, mi = 0.002, pvalue = 0.035, MODULE_493\n",
      "Didn't pass (cutoff=0.0050): cat 67, mi = 0.002, pvalue = 0.027, GCM_RAB10\n",
      "Didn't pass (cutoff=0.0050): cat 371, mi = 0.002, pvalue = 0.033, MODULE_238\n",
      "Didn't pass (cutoff=0.0050): cat 108, mi = 0.002, pvalue = 0.017, GNF2_CD33\n",
      "Didn't pass (cutoff=0.0050): cat 540, mi = 0.002, pvalue = 0.014, MODULE_447\n",
      "Didn't pass (cutoff=0.0050): cat 737, mi = 0.002, pvalue = 0.047, MORF_GSPT1\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.002, pvalue = 0.040, GNF2_IL2RB\n",
      "Total number of categories evaluated = 35.\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.769980842369428\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "8.75\t-11.06\n",
      "MORF_THRA\n",
      "MODULE_36\n",
      "GNF2_SNRK\n",
      "MORF_NME2\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.769980842369428\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-9.98233215547703\t42\t0\n",
      "0.745583038869256\t42\t20.0353356890459\n",
      "1.85865724381625\t42\t21.4558303886926\n",
      "2.53710247349823\t42\t22.2614840989399\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.45936395759717\t42\t23.2579505300353\n",
      "3.85159010600707\t42\t23.660777385159\n",
      "4.23321554770318\t42\t24.0424028268551\n",
      "4.64664310954063\t42\t24.4240282685512\n",
      "5.1660777385159\t42\t24.86925795053\n",
      "13.7314487632509\t42\t25.4628975265018\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_c4/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5917\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t5917\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 5917\n",
      "cat 1280, mi = 0.00418145, GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "cat 3849, mi = 0.00368770, GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "cat 5642, mi = 0.00343185, GO_TISSUE_HOMEOSTASIS\n",
      "cat 3847, mi = 0.00339146, GO_POSITIVE_REGULATION_OF_VIRAL_PROCESS\n",
      "cat 1663, mi = 0.00333993, GO_HOMEOSTASIS_OF_NUMBER_OF_CELLS_WITHIN_A_TISSUE\n",
      "cat 4999, mi = 0.00330402, GO_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "cat 3047, mi = 0.00329193, GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "cat 3644, mi = 0.00326000, GO_POSITIVE_REGULATION_OF_MULTICELLULAR_ORGANISM_GROWTH\n",
      "cat 3647, mi = 0.00323485, GO_POSITIVE_REGULATION_OF_MUSCLE_CELL_DIFFERENTIATION\n",
      "cat 1283, mi = 0.00318681, GO_ESTABLISHMENT_OF_EPITHELIAL_CELL_POLARITY\n",
      "cat 89, mi = 0.00317231, GO_ADENYLATE_CYCLASE_MODULATING_G_PROTEIN_COUPLED_RECEPTOR_SIGNALING_PATHWAY\n",
      "cat 3789, mi = 0.00313525, GO_POSITIVE_REGULATION_OF_STRIATED_MUSCLE_CELL_DIFFERENTIATION\n",
      "cat 5182, mi = 0.00310728, GO_RESPONSE_TO_UV\n",
      "cat 4463, mi = 0.00307180, GO_REGULATION_OF_GLUCOSE_IMPORT\n",
      "cat 1356, mi = 0.00305498, GO_EYE_MORPHOGENESIS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1280, mi = 0.004, pvalue = 0.000, GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "Passed: cat 3849, mi = 0.004, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "Passed: cat 5642, mi = 0.003, pvalue = 0.001, GO_TISSUE_HOMEOSTASIS\n",
      "Passed: cat 3047, mi = 0.003, pvalue = 0.002, GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 89, mi = 0.003, pvalue = 0.002, GO_ADENYLATE_CYCLASE_MODULATING_G_PROTEIN_COUPLED_RECEPTOR_SIGNALING_PATHWAY\n",
      "Passed: cat 4463, mi = 0.003, pvalue = 0.003, GO_REGULATION_OF_GLUCOSE_IMPORT\n",
      "Passed: cat 1356, mi = 0.003, pvalue = 0.004, GO_EYE_MORPHOGENESIS\n",
      "Passed: cat 5264, mi = 0.003, pvalue = 0.004, GO_RNA_METHYLATION\n",
      "Passed: cat 882, mi = 0.003, pvalue = 0.004, GO_COSTAMERE\n",
      "Passed: cat 627, mi = 0.003, pvalue = 0.004, GO_CELL_ADHESION_MOLECULE_BINDING\n",
      "Passed: cat 1779, mi = 0.003, pvalue = 0.002, GO_INTERMEDIATE_FILAMENT\n",
      "Didn't pass (cutoff=0.0050): cat 885, mi = 0.003, pvalue = 0.006, GO_CRANIAL_NERVE_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 2058, mi = 0.003, pvalue = 0.007, GO_MANGANESE_ION_BINDING\n",
      "Passed: cat 868, mi = 0.003, pvalue = 0.001, GO_COPPER_ION_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3853, mi = 0.003, pvalue = 0.009, GO_POSTREPLICATION_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 2803, mi = 0.003, pvalue = 0.009, GO_NERVE_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 4670, mi = 0.003, pvalue = 0.005, GO_REGULATION_OF_NECROPTOTIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1534, mi = 0.003, pvalue = 0.008, GO_GOLGI_CISTERNA\n",
      "Passed: cat 4831, mi = 0.003, pvalue = 0.003, GO_REGULATION_OF_RESPIRATORY_GASEOUS_EXCHANGE\n",
      "Didn't pass (cutoff=0.0050): cat 3266, mi = 0.003, pvalue = 0.010, GO_PHOSPHORIC_DIESTER_HYDROLASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4654, mi = 0.003, pvalue = 0.007, GO_REGULATION_OF_MUSCLE_CELL_APOPTOTIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3640, mi = 0.002, pvalue = 0.010, GO_POSITIVE_REGULATION_OF_MRNA_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 1638, mi = 0.002, pvalue = 0.007, GO_HISTONE_EXCHANGE\n",
      "Passed: cat 1564, mi = 0.002, pvalue = 0.002, GO_GTP_RHO_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5728, mi = 0.002, pvalue = 0.018, GO_TRANSLESION_SYNTHESIS\n",
      "Didn't pass (cutoff=0.0050): cat 4868, mi = 0.002, pvalue = 0.007, GO_REGULATION_OF_SKELETAL_MUSCLE_CELL_PROLIFERATION\n",
      "Didn't pass (cutoff=0.0050): cat 1541, mi = 0.002, pvalue = 0.018, GO_GOLGI_STACK\n",
      "Didn't pass (cutoff=0.0050): cat 4671, mi = 0.002, pvalue = 0.028, GO_REGULATION_OF_NECROTIC_CELL_DEATH\n",
      "Didn't pass (cutoff=0.0050): cat 3231, mi = 0.002, pvalue = 0.030, GO_PHOSPHATIDYLINOSITOL_4_5_BISPHOSPHATE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 4142, mi = 0.002, pvalue = 0.015, GO_RECEPTOR_REGULATOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2727, mi = 0.002, pvalue = 0.024, GO_NEGATIVE_REGULATION_OF_RESPONSE_TO_EXTRACELLULAR_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 2821, mi = 0.002, pvalue = 0.013, GO_NEUROMUSCULAR_PROCESS_CONTROLLING_BALANCE\n",
      "Didn't pass (cutoff=0.0050): cat 2783, mi = 0.002, pvalue = 0.031, GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 5725, mi = 0.002, pvalue = 0.023, GO_TRANSLATION_REGULATOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5214, mi = 0.002, pvalue = 0.037, GO_RETROGRADE_TRANSPORT_ENDOSOME_TO_GOLGI\n",
      "Didn't pass (cutoff=0.0050): cat 1746, mi = 0.002, pvalue = 0.040, GO_INORGANIC_ANION_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 5409, mi = 0.002, pvalue = 0.025, GO_SKELETAL_MUSCLE_TISSUE_REGENERATION\n",
      "Didn't pass (cutoff=0.0050): cat 4838, mi = 0.002, pvalue = 0.031, GO_REGULATION_OF_RESPONSE_TO_EXTRACELLULAR_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 416, mi = 0.002, pvalue = 0.043, GO_CARBOHYDRATE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2622, mi = 0.002, pvalue = 0.028, GO_NEGATIVE_REGULATION_OF_MACROAUTOPHAGY\n",
      "Didn't pass (cutoff=0.0050): cat 752, mi = 0.002, pvalue = 0.038, GO_CHLORIDE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 4419, mi = 0.002, pvalue = 0.042, GO_REGULATION_OF_ESTABLISHMENT_OF_PLANAR_POLARITY\n",
      "Didn't pass (cutoff=0.0050): cat 712, mi = 0.002, pvalue = 0.045, GO_CENTROMERE_COMPLEX_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 5186, mi = 0.002, pvalue = 0.048, GO_RESPONSE_TO_VITAMIN\n",
      "Total number of categories evaluated = 157.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.25380411938752\n",
      "best merge is 9, 13, 0.395056722682523\n",
      "best merge is 1, 8, 0.423784356564338\n",
      "best merge is 2, 10, 0.493107389993967\n",
      "best merge is 5, 7, 0.544524242142308\n",
      "best merge is 0, 9, 0.572704072789468\n",
      "best merge is 2, 6, 0.596493185937173\n",
      "best merge is 2, 11, 0.866302198596182\n",
      "best merge is 5, 12, 0.875836444882113\n",
      "best merge is 1, 2, 0.989911374660908\n",
      "order classes\n",
      "3 - 1.59751912264151\n",
      "2 - 1.48324524797518\n",
      "5 - 1.45481442435206\n",
      "3 - 2.38124147012529\n",
      "2 4 10 6 11\n",
      "1 8\n",
      "0 9 13\n",
      "3\n",
      "5 7 12\n",
      "Done.Start drawing\n",
      "8.75\t-11.06\n",
      "GO_REGULATION_OF_GLUCOSE_IMPORT\n",
      "GO_REGULATION_OF_RESPIRATORY_GASEOUS_EXCHANGE\n",
      "GO_TISSUE_HOMEOSTASIS\n",
      "GO_ADENYLATE_CYCLASE_MODULATING_G_PROTEIN_COUPLED_RECEPTOR_SIGNALING_PATHWAY\n",
      "GO_INTERMEDIATE_FILAMENT\n",
      "GO_EYE_MORPHOGENESIS\n",
      "GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "GO_COSTAMERE\n",
      "GO_GTP_RHO_BINDING\n",
      "GO_COPPER_ION_BINDING\n",
      "GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "GO_CELL_ADHESION_MOLECULE_BINDING\n",
      "GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "GO_RNA_METHYLATION\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.25380411938752\n",
      "best merge is 9, 13, 0.395056722682523\n",
      "best merge is 1, 8, 0.423784356564338\n",
      "best merge is 2, 10, 0.493107389993967\n",
      "best merge is 5, 7, 0.544524242142308\n",
      "best merge is 0, 9, 0.572704072789468\n",
      "best merge is 2, 6, 0.596493185937173\n",
      "best merge is 2, 11, 0.866302198596182\n",
      "best merge is 5, 12, 0.875836444882113\n",
      "best merge is 1, 2, 0.989911374660908\n",
      "order classes\n",
      "3 - 1.59751912264151\n",
      "2 - 1.48324524797518\n",
      "5 - 1.45481442435206\n",
      "3 - 2.38124147012529\n",
      "2 4 10 6 11\n",
      "1 8\n",
      "0 9 13\n",
      "3\n",
      "5 7 12\n",
      "Done.-9.98233215547703\t42\t0\n",
      "0.745583038869256\t42\t20.0353356890459\n",
      "1.85865724381625\t42\t21.4558303886926\n",
      "2.53710247349823\t42\t22.2614840989399\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.45936395759717\t42\t23.2579505300353\n",
      "3.85159010600707\t42\t23.660777385159\n",
      "4.23321554770318\t42\t24.0424028268551\n",
      "4.64664310954063\t42\t24.4240282685512\n",
      "5.1660777385159\t42\t24.86925795053\n",
      "13.7314487632509\t42\t25.4628975265018\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_c5/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 189\n",
      "cat 41, mi = 0.00302385, CYCLIN_D1_KE_.V1_DN\n",
      "cat 113, mi = 0.00301304, MEL18_DN.V1_DN\n",
      "cat 116, mi = 0.00278304, MTOR_UP.N4.V1_UP\n",
      "cat 78, mi = 0.00272651, IL21_UP.V1_UP\n",
      "cat 66, mi = 0.00267227, GCNP_SHH_UP_EARLY.V1_UP\n",
      "cat 28, mi = 0.00264763, CAMP_UP.V1_DN\n",
      "cat 52, mi = 0.00263611, EGFR_UP.V1_UP\n",
      "cat 0, mi = 0.00258487, AKT_UP.V1_DN\n",
      "cat 4, mi = 0.00257735, ALK_DN.V1_DN\n",
      "cat 150, mi = 0.00255964, RAF_UP.V1_DN\n",
      "cat 43, mi = 0.00255488, CYCLIN_D1_UP.V1_DN\n",
      "cat 18, mi = 0.00254697, BMI1_DN.V1_DN\n",
      "cat 83, mi = 0.00251458, JNK_DN.V1_DN\n",
      "cat 79, mi = 0.00243864, IL2_UP.V1_DN\n",
      "cat 22, mi = 0.00237145, BRCA1_DN.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 41, mi = 0.003, pvalue = 0.003, CYCLIN_D1_KE_.V1_DN\n",
      "Passed: cat 113, mi = 0.003, pvalue = 0.003, MEL18_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.003, pvalue = 0.006, MTOR_UP.N4.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 78, mi = 0.003, pvalue = 0.009, IL21_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.003, pvalue = 0.010, GCNP_SHH_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 52, mi = 0.003, pvalue = 0.009, EGFR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 0, mi = 0.003, pvalue = 0.011, AKT_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.003, pvalue = 0.016, ALK_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 83, mi = 0.003, pvalue = 0.012, JNK_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 79, mi = 0.002, pvalue = 0.022, IL2_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 102, mi = 0.002, pvalue = 0.028, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 186, mi = 0.002, pvalue = 0.034, WNT_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.002, pvalue = 0.032, PRC2_EED_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.002, pvalue = 0.060, SIRNA_EIF4GI_UP\n",
      "Didn't pass (cutoff=0.0050): cat 143, mi = 0.002, pvalue = 0.054, PRC2_EZH2_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 74, mi = 0.002, pvalue = 0.056, HOXA9_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 77, mi = 0.002, pvalue = 0.100, IL21_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 162, mi = 0.002, pvalue = 0.081, RPS14_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.102, TBK1.DN.48HRS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 103, mi = 0.002, pvalue = 0.099, KRAS.LUNG_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 149, mi = 0.002, pvalue = 0.074, PTEN_DN.V2_UP\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.100, RB_P107_DN.V1_DN\n",
      "Total number of categories evaluated = 20.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.75\t-11.06\n",
      "MEL18_DN.V1_DN\n",
      "CYCLIN_D1_KE_.V1_DN\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-9.98233215547703\t42\t0\n",
      "0.745583038869256\t42\t20.0353356890459\n",
      "1.85865724381625\t42\t21.4558303886926\n",
      "2.53710247349823\t42\t22.2614840989399\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.45936395759717\t42\t23.2579505300353\n",
      "3.85159010600707\t42\t23.660777385159\n",
      "4.23321554770318\t42\t24.0424028268551\n",
      "4.64664310954063\t42\t24.4240282685512\n",
      "5.1660777385159\t42\t24.86925795053\n",
      "13.7314487632509\t42\t25.4628975265018\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_c6/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 4872\n",
      "cat 4297, mi = 0.00433748, GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "cat 65, mi = 0.00424467, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 1563, mi = 0.00423588, GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_UP\n",
      "cat 2134, mi = 0.00384640, GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_DN\n",
      "cat 1347, mi = 0.00373341, GSE1925_CTRL_VS_IFNG_PRIMED_MACROPHAGE_3H_IFNG_STIM_UP\n",
      "cat 2719, mi = 0.00352322, GSE29618_BCELL_VS_MDC_UP\n",
      "cat 508, mi = 0.00346040, GSE1460_CD4_THYMOCYTE_VS_NAIVE_CD4_TCELL_CORD_BLOOD_DN\n",
      "cat 3310, mi = 0.00345946, GSE3720_LPS_VS_PMA_STIM_VD1_GAMMADELTA_TCELL_DN\n",
      "cat 1141, mi = 0.00342454, GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "cat 771, mi = 0.00342093, GSE16450_IMMATURE_VS_MATURE_NEURON_CELL_LINE_UP\n",
      "cat 204, mi = 0.00337489, GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "cat 4240, mi = 0.00337479, GSE46606_IRF4HIGH_VS_WT_CD40L_IL2_IL5_DAY1_STIMULATED_BCELL_DN\n",
      "cat 4675, mi = 0.00334621, GSE7852_TREG_VS_TCONV_THYMUS_DN\n",
      "cat 522, mi = 0.00334289, GSE1460_INTRATHYMIC_T_PROGENITOR_VS_CD4_THYMOCYTE_DN\n",
      "cat 885, mi = 0.00330380, GSE17721_0.5H_VS_8H_PAM3CSK4_BMDC_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 4297, mi = 0.004, pvalue = 0.000, GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "Passed: cat 65, mi = 0.004, pvalue = 0.000, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "Passed: cat 1563, mi = 0.004, pvalue = 0.000, GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_UP\n",
      "Passed: cat 2134, mi = 0.004, pvalue = 0.000, GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_DN\n",
      "Passed: cat 1347, mi = 0.004, pvalue = 0.000, GSE1925_CTRL_VS_IFNG_PRIMED_MACROPHAGE_3H_IFNG_STIM_UP\n",
      "Passed: cat 508, mi = 0.003, pvalue = 0.001, GSE1460_CD4_THYMOCYTE_VS_NAIVE_CD4_TCELL_CORD_BLOOD_DN\n",
      "Passed: cat 1141, mi = 0.003, pvalue = 0.000, GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "Passed: cat 204, mi = 0.003, pvalue = 0.001, GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "Passed: cat 885, mi = 0.003, pvalue = 0.002, GSE17721_0.5H_VS_8H_PAM3CSK4_BMDC_UP\n",
      "Passed: cat 2160, mi = 0.003, pvalue = 0.001, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "Passed: cat 1421, mi = 0.003, pvalue = 0.003, GSE19923_WT_VS_E2A_KO_DP_THYMOCYTE_UP\n",
      "Passed: cat 2461, mi = 0.003, pvalue = 0.003, GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_2H_UP\n",
      "Passed: cat 1112, mi = 0.003, pvalue = 0.002, GSE17721_POLYIC_VS_CPG_0.5H_BMDC_DN\n",
      "Passed: cat 70, mi = 0.003, pvalue = 0.003, GSE10463_CD40L_AND_VA347_VS_CD40L_IN_DC_DN\n",
      "Passed: cat 3289, mi = 0.003, pvalue = 0.004, GSE36891_UNSTIM_VS_POLYIC_TLR3_STIM_PERITONEAL_MACROPHAGE_UP\n",
      "Passed: cat 3404, mi = 0.003, pvalue = 0.003, GSE37416_CTRL_VS_0H_F_TULARENSIS_LVS_NEUTROPHIL_DN\n",
      "Passed: cat 4358, mi = 0.003, pvalue = 0.002, GSE557_WT_VS_CIITA_KO_DC_DN\n",
      "Passed: cat 646, mi = 0.003, pvalue = 0.004, GSE15659_RESTING_VS_ACTIVATED_TREG_DN\n",
      "Passed: cat 584, mi = 0.003, pvalue = 0.004, GSE15330_HSC_VS_GRANULOCYTE_MONOCYTE_PROGENITOR_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1843, mi = 0.003, pvalue = 0.006, GSE22611_NOD2_VS_MUTANT_NOD2_TRANSDUCED_HEK293T_CELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3176, mi = 0.003, pvalue = 0.006, GSE360_L_DONOVANI_VS_B_MALAYI_LOW_DOSE_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2073, mi = 0.003, pvalue = 0.007, GSE24142_ADULT_VS_FETAL_DN2_THYMOCYTE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4172, mi = 0.003, pvalue = 0.009, GSE45365_NK_CELL_VS_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2445, mi = 0.003, pvalue = 0.010, GSE2770_IL12_VS_IL4_TREATED_ACT_CD4_TCELL_48H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3337, mi = 0.003, pvalue = 0.015, GSE37301_COMMON_LYMPHOID_PROGENITOR_VS_PRO_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4852, mi = 0.003, pvalue = 0.010, GSE9988_LPS_VS_CTRL_TREATED_MONOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1438, mi = 0.003, pvalue = 0.014, GSE19941_UNSTIM_VS_LPS_AND_IL10_STIM_IL10_KO_NFKBP50_KO_MACROPHAGE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1202, mi = 0.002, pvalue = 0.019, GSE17974_2.5H_VS_72H_IL4_AND_ANTI_IL12_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3759, mi = 0.002, pvalue = 0.017, GSE40274_CTRL_VS_FOXP3_TRANSDUCED_ACTIVATED_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4490, mi = 0.002, pvalue = 0.021, GSE6269_E_COLI_VS_STREP_PNEUMO_INF_PBMC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2634, mi = 0.002, pvalue = 0.018, GSE28449_WT_VS_LRF_KO_GERMINAL_CENTER_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1458, mi = 0.002, pvalue = 0.017, GSE20198_UNTREATED_VS_IFNA_TREATED_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.002, pvalue = 0.017, GSE16451_CTRL_VS_WEST_EQUINE_ENC_VIRUS_MATURE_NEURON_CELL_LINE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4741, mi = 0.002, pvalue = 0.019, GSE9006_HEALTHY_VS_TYPE_1_DIABETES_PBMC_1MONTH_POST_DX_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4476, mi = 0.002, pvalue = 0.022, GSE6259_FLT3L_INDUCED_33D1_POS_DC_VS_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1646, mi = 0.002, pvalue = 0.023, GSE21670_IL6_VS_TGFB_AND_IL6_TREATED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4777, mi = 0.002, pvalue = 0.026, GSE9509_10MIN_VS_30MIN_LPS_STIM_IL10_KO_MACROPHAGE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2726, mi = 0.002, pvalue = 0.023, GSE29618_BCELL_VS_PDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3424, mi = 0.002, pvalue = 0.033, GSE37532_VISCERAL_ADIPOSE_TISSUE_VS_LN_DERIVED_PPARG_KO_TCONV_CD4_TCELL_DN\n",
      "Total number of categories evaluated = 219.\n",
      "Number of categories that passed the tests = 19\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 17, 0.125836812811185\n",
      "best merge is 14, 16, 0.171411396918862\n",
      "best merge is 0, 1, 0.190445436817526\n",
      "best merge is 4, 8, 0.378751494084889\n",
      "best merge is 3, 10, 0.396885768891637\n",
      "best merge is 2, 14, 0.399629269155976\n",
      "best merge is 0, 4, 0.454130759906787\n",
      "best merge is 12, 15, 0.505749765977844\n",
      "best merge is 6, 18, 0.528650349992331\n",
      "best merge is 2, 7, 0.616516258433975\n",
      "best merge is 12, 13, 0.657229763789387\n",
      "best merge is 0, 9, 0.713787891965359\n",
      "best merge is 0, 11, 0.77347836041092\n",
      "best merge is 3, 6, 0.802153982810874\n",
      "best merge is 2, 3, 0.884752703792216\n",
      "order classes\n",
      "7 - 1.33037439633294\n",
      "4 - 1.67450485937881\n",
      "4 - 2.45465277976318\n",
      "3 - 2.56177604278339\n",
      "0 1 4 17 8 9 11\n",
      "2 14 16 7\n",
      "3 10 6 18\n",
      "5\n",
      "12 15 13\n",
      "Done.Start drawing\n",
      "8.75\t-11.06\n",
      "GSE36891_UNSTIM_VS_POLYIC_TLR3_STIM_PERITONEAL_MACROPHAGE_UP\n",
      "GSE557_WT_VS_CIITA_KO_DC_DN\n",
      "GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_UP\n",
      "GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_DN\n",
      "GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "GSE15330_HSC_VS_GRANULOCYTE_MONOCYTE_PROGENITOR_DN\n",
      "GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_2H_UP\n",
      "GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "GSE1925_CTRL_VS_IFNG_PRIMED_MACROPHAGE_3H_IFNG_STIM_UP\n",
      "GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "GSE19923_WT_VS_E2A_KO_DP_THYMOCYTE_UP\n",
      "GSE17721_POLYIC_VS_CPG_0.5H_BMDC_DN\n",
      "GSE37416_CTRL_VS_0H_F_TULARENSIS_LVS_NEUTROPHIL_DN\n",
      "GSE15659_RESTING_VS_ACTIVATED_TREG_DN\n",
      "GSE17721_0.5H_VS_8H_PAM3CSK4_BMDC_UP\n",
      "GSE1460_CD4_THYMOCYTE_VS_NAIVE_CD4_TCELL_CORD_BLOOD_DN\n",
      "GSE10463_CD40L_AND_VA347_VS_CD40L_IN_DC_DN\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 17, 0.125836812811185\n",
      "best merge is 14, 16, 0.171411396918862\n",
      "best merge is 0, 1, 0.190445436817526\n",
      "best merge is 4, 8, 0.378751494084889\n",
      "best merge is 3, 10, 0.396885768891637\n",
      "best merge is 2, 14, 0.399629269155976\n",
      "best merge is 0, 4, 0.454130759906787\n",
      "best merge is 12, 15, 0.505749765977844\n",
      "best merge is 6, 18, 0.528650349992331\n",
      "best merge is 2, 7, 0.616516258433975\n",
      "best merge is 12, 13, 0.657229763789387\n",
      "best merge is 0, 9, 0.713787891965359\n",
      "best merge is 0, 11, 0.77347836041092\n",
      "best merge is 3, 6, 0.802153982810874\n",
      "best merge is 2, 3, 0.884752703792216\n",
      "order classes\n",
      "7 - 1.33037439633294\n",
      "4 - 1.67450485937881\n",
      "4 - 2.45465277976318\n",
      "3 - 2.56177604278339\n",
      "0 1 4 17 8 9 11\n",
      "2 14 16 7\n",
      "3 10 6 18\n",
      "5\n",
      "12 15 13\n",
      "Done.-9.98233215547703\t42\t0\n",
      "0.745583038869256\t42\t20.0353356890459\n",
      "1.85865724381625\t42\t21.4558303886926\n",
      "2.53710247349823\t42\t22.2614840989399\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.45936395759717\t42\t23.2579505300353\n",
      "3.85159010600707\t42\t23.660777385159\n",
      "4.23321554770318\t42\t24.0424028268551\n",
      "4.64664310954063\t42\t24.4240282685512\n",
      "5.1660777385159\t42\t24.86925795053\n",
      "13.7314487632509\t42\t25.4628975265018\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_c7/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 17786\n",
      "cat 12154, mi = 0.00433748, GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "cat 7922, mi = 0.00424467, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 9420, mi = 0.00423588, GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_UP\n",
      "cat 3166, mi = 0.00418145, GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "cat 9991, mi = 0.00384640, GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_DN\n",
      "cat 9204, mi = 0.00373341, GSE1925_CTRL_VS_IFNG_PRIMED_MACROPHAGE_3H_IFNG_STIM_UP\n",
      "cat 17034, mi = 0.00372941, WANG_BARRETTS_ESOPHAGUS_UP\n",
      "cat 5735, mi = 0.00368770, GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "cat 1839, mi = 0.00362094, GNF2_SNRK\n",
      "cat 16187, mi = 0.00353409, RIGGI_EWING_SARCOMA_PROGENITOR_DN\n",
      "cat 10576, mi = 0.00352322, GSE29618_BCELL_VS_MDC_UP\n",
      "cat 8365, mi = 0.00346040, GSE1460_CD4_THYMOCYTE_VS_NAIVE_CD4_TCELL_CORD_BLOOD_DN\n",
      "cat 11167, mi = 0.00345946, GSE3720_LPS_VS_PMA_STIM_VD1_GAMMADELTA_TCELL_DN\n",
      "cat 7528, mi = 0.00343185, GO_TISSUE_HOMEOSTASIS\n",
      "cat 8998, mi = 0.00342454, GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 12154, mi = 0.004, pvalue = 0.000, GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "Passed: cat 7922, mi = 0.004, pvalue = 0.000, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "Passed: cat 9420, mi = 0.004, pvalue = 0.000, GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_UP\n",
      "Passed: cat 3166, mi = 0.004, pvalue = 0.000, GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "Passed: cat 9991, mi = 0.004, pvalue = 0.000, GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_DN\n",
      "Passed: cat 9204, mi = 0.004, pvalue = 0.000, GSE1925_CTRL_VS_IFNG_PRIMED_MACROPHAGE_3H_IFNG_STIM_UP\n",
      "Passed: cat 17034, mi = 0.004, pvalue = 0.000, WANG_BARRETTS_ESOPHAGUS_UP\n",
      "Passed: cat 5735, mi = 0.004, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "Passed: cat 1839, mi = 0.004, pvalue = 0.001, GNF2_SNRK\n",
      "Passed: cat 16187, mi = 0.004, pvalue = 0.001, RIGGI_EWING_SARCOMA_PROGENITOR_DN\n",
      "Passed: cat 8998, mi = 0.003, pvalue = 0.000, GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "Passed: cat 8061, mi = 0.003, pvalue = 0.001, GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "Passed: cat 4933, mi = 0.003, pvalue = 0.002, GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 10017, mi = 0.003, pvalue = 0.001, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "Passed: cat 9278, mi = 0.003, pvalue = 0.003, GSE19923_WT_VS_E2A_KO_DP_THYMOCYTE_UP\n",
      "Passed: cat 10318, mi = 0.003, pvalue = 0.001, GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_2H_UP\n",
      "Passed: cat 8969, mi = 0.003, pvalue = 0.002, GSE17721_POLYIC_VS_CPG_0.5H_BMDC_DN\n",
      "Passed: cat 1071, mi = 0.003, pvalue = 0.003, CYCLIN_D1_KE_.V1_DN\n",
      "Passed: cat 7927, mi = 0.003, pvalue = 0.003, GSE10463_CD40L_AND_VA347_VS_CD40L_IN_DC_DN\n",
      "Passed: cat 11261, mi = 0.003, pvalue = 0.004, GSE37416_CTRL_VS_0H_F_TULARENSIS_LVS_NEUTROPHIL_DN\n",
      "Passed: cat 17606, mi = 0.003, pvalue = 0.004, chr22q12\n",
      "Passed: cat 12215, mi = 0.003, pvalue = 0.002, GSE557_WT_VS_CIITA_KO_DC_DN\n",
      "Passed: cat 615, mi = 0.003, pvalue = 0.000, BOYAULT_LIVER_CANCER_SUBCLASS_G5_DN\n",
      "Passed: cat 8503, mi = 0.003, pvalue = 0.005, GSE15659_RESTING_VS_ACTIVATED_TREG_DN\n",
      "Passed: cat 10617, mi = 0.003, pvalue = 0.003, GSE29618_PRE_VS_DAY7_POST_LAIV_FLU_VACCINE_MONOCYTE_DN\n",
      "Passed: cat 8441, mi = 0.003, pvalue = 0.004, GSE15330_HSC_VS_GRANULOCYTE_MONOCYTE_PROGENITOR_DN\n",
      "Passed: cat 1648, mi = 0.003, pvalue = 0.004, GENTILE_UV_LOW_DOSE_UP\n",
      "Passed: cat 16068, mi = 0.003, pvalue = 0.001, REACTOME_SOS_MEDIATED_SIGNALLING\n",
      "Didn't pass (cutoff=0.0050): cat 13897, mi = 0.003, pvalue = 0.009, MATTIOLI_MGUS_VS_PCL\n",
      "Didn't pass (cutoff=0.0050): cat 3944, mi = 0.003, pvalue = 0.007, GO_MANGANESE_ION_BINDING\n",
      "Passed: cat 2754, mi = 0.003, pvalue = 0.001, GO_COPPER_ION_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 17589, mi = 0.003, pvalue = 0.011, chr1q32\n",
      "Didn't pass (cutoff=0.0050): cat 12029, mi = 0.003, pvalue = 0.010, GSE45365_NK_CELL_VS_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1635, mi = 0.003, pvalue = 0.007, GCNP_SHH_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 221, mi = 0.003, pvalue = 0.008, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 1358, mi = 0.003, pvalue = 0.014, FIGUEROA_AML_METHYLATION_CLUSTER_4_UP\n",
      "Didn't pass (cutoff=0.0050): cat 10302, mi = 0.003, pvalue = 0.010, GSE2770_IL12_VS_IL4_TREATED_ACT_CD4_TCELL_48H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14647, mi = 0.003, pvalue = 0.015, MORF_PRKAR1A\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.003, pvalue = 0.015, ALK_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5372, mi = 0.003, pvalue = 0.015, GO_POSITIVE_REGULATION_OF_ENDOCYTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 1753, mi = 0.003, pvalue = 0.012, GNF2_DAP3\n",
      "Didn't pass (cutoff=0.0050): cat 2395, mi = 0.003, pvalue = 0.005, GO_CELLULAR_EXTRAVASATION\n",
      "Didn't pass (cutoff=0.0050): cat 13798, mi = 0.002, pvalue = 0.020, LXR_DR4_Q3\n",
      "Didn't pass (cutoff=0.0050): cat 3524, mi = 0.002, pvalue = 0.008, GO_HISTONE_EXCHANGE\n",
      "Didn't pass (cutoff=0.0050): cat 11616, mi = 0.002, pvalue = 0.020, GSE40274_CTRL_VS_FOXP3_TRANSDUCED_ACTIVATED_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1619, mi = 0.002, pvalue = 0.017, GCM_RING1\n",
      "Didn't pass (cutoff=0.0050): cat 17275, mi = 0.002, pvalue = 0.021, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_14\n",
      "Passed: cat 5279, mi = 0.002, pvalue = 0.004, GO_POSITIVE_REGULATION_OF_CALCIUM_ION_TRANSPORT_INTO_CYTOSOL\n",
      "Didn't pass (cutoff=0.0050): cat 9315, mi = 0.002, pvalue = 0.018, GSE20198_UNTREATED_VS_IFNA_TREATED_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8632, mi = 0.002, pvalue = 0.016, GSE16451_CTRL_VS_WEST_EQUINE_ENC_VIRUS_MATURE_NEURON_CELL_LINE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 16962, mi = 0.002, pvalue = 0.006, VANLOO_SP3_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1254, mi = 0.002, pvalue = 0.016, EBAUER_TARGETS_OF_PAX3_FOXO1_FUSION_DN\n",
      "Didn't pass (cutoff=0.0050): cat 13483, mi = 0.002, pvalue = 0.011, KORKOLA_EMBRYONAL_CARCINOMA_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1591, mi = 0.002, pvalue = 0.022, GCM_MAP1B\n",
      "Didn't pass (cutoff=0.0050): cat 5899, mi = 0.002, pvalue = 0.023, GO_PROTEIN_N_TERMINUS_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 7151, mi = 0.002, pvalue = 0.025, GO_RNA_METHYLTRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 9503, mi = 0.002, pvalue = 0.021, GSE21670_IL6_VS_TGFB_AND_IL6_TREATED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 13036, mi = 0.002, pvalue = 0.022, IGLESIAS_E2F_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 12634, mi = 0.002, pvalue = 0.025, GSE9509_10MIN_VS_30MIN_LPS_STIM_IL10_KO_MACROPHAGE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14636, mi = 0.002, pvalue = 0.029, MORF_PPP1CA\n",
      "Didn't pass (cutoff=0.0050): cat 1610, mi = 0.002, pvalue = 0.028, GCM_PTPRD\n",
      "Didn't pass (cutoff=0.0050): cat 13033, mi = 0.002, pvalue = 0.032, IGARASHI_ATF4_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 14554, mi = 0.002, pvalue = 0.030, MORF_CTBP1\n",
      "Didn't pass (cutoff=0.0050): cat 11281, mi = 0.002, pvalue = 0.035, GSE37532_VISCERAL_ADIPOSE_TISSUE_VS_LN_DERIVED_PPARG_KO_TCONV_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 12614, mi = 0.002, pvalue = 0.023, GSE9037_CTRL_VS_LPS_1H_STIM_IRAK4_KO_BMDM_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3886, mi = 0.002, pvalue = 0.026, GO_LYSINE_ACETYLATED_HISTONE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 14268, mi = 0.002, pvalue = 0.036, MODULE_322\n",
      "Didn't pass (cutoff=0.0050): cat 1730, mi = 0.002, pvalue = 0.017, GNF2_CBFB\n",
      "Total number of categories evaluated = 577.\n",
      "Number of categories that passed the tests = 30\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 28, 29, 0.0636652875256613\n",
      "best merge is 5, 23, 0.125836812811185\n",
      "best merge is 0, 1, 0.190445436817526\n",
      "best merge is 3, 14, 0.235434490651208\n",
      "best merge is 7, 25, 0.287221029410587\n",
      "best merge is 2, 21, 0.310653129875547\n",
      "best merge is 9, 13, 0.335127910483962\n",
      "best merge is 16, 20, 0.354130140561958\n",
      "best merge is 0, 5, 0.381871204849991\n",
      "best merge is 24, 26, 0.404519334342374\n",
      "best merge is 6, 8, 0.426936318848102\n",
      "best merge is 15, 17, 0.449120453251836\n",
      "best merge is 3, 9, 0.457677716738753\n",
      "best merge is 16, 19, 0.467225338412148\n",
      "best merge is 7, 10, 0.477214511996996\n",
      "best merge is 11, 24, 0.550972682492288\n",
      "best merge is 12, 18, 0.561402190484092\n",
      "best merge is 0, 28, 0.609430057230998\n",
      "best merge is 2, 27, 0.627798023148228\n",
      "best merge is 6, 7, 0.694401037599554\n",
      "best merge is 2, 11, 0.701959335926029\n",
      "best merge is 4, 12, 0.724944459438881\n",
      "best merge is 0, 15, 0.733773602716411\n",
      "best merge is 3, 22, 0.853060461352253\n",
      "best merge is 4, 16, 0.88254696296272\n",
      "best merge is 2, 6, 0.935479591074421\n",
      "order classes\n",
      "8 - 1.87936833987317\n",
      "6 - 2.60842997189811\n",
      "5 - 2.79048725447961\n",
      "6 - 3.12299849817016\n",
      "5 - 2.800796446968\n",
      "0 1 5 23 28 29 15 17\n",
      "2 21 27 11 24 26\n",
      "3 14 9 13 22\n",
      "4 12 18 16 20 19\n",
      "6 8 7 25 10\n",
      "Done.Start drawing\n",
      "8.75\t-11.06\n",
      "GSE557_WT_VS_CIITA_KO_DC_DN\n",
      "REACTOME_SOS_MEDIATED_SIGNALLING\n",
      "GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_UP\n",
      "GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_DN\n",
      "BOYAULT_LIVER_CANCER_SUBCLASS_G5_DN\n",
      "GENTILE_UV_LOW_DOSE_UP\n",
      "GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "GSE15330_HSC_VS_GRANULOCYTE_MONOCYTE_PROGENITOR_DN\n",
      "GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "WANG_BARRETTS_ESOPHAGUS_UP\n",
      "GNF2_SNRK\n",
      "GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_2H_UP\n",
      "GSE29618_PRE_VS_DAY7_POST_LAIV_FLU_VACCINE_MONOCYTE_DN\n",
      "GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "GSE1925_CTRL_VS_IFNG_PRIMED_MACROPHAGE_3H_IFNG_STIM_UP\n",
      "GO_COPPER_ION_BINDING\n",
      "GO_POSITIVE_REGULATION_OF_CALCIUM_ION_TRANSPORT_INTO_CYTOSOL\n",
      "CYCLIN_D1_KE_.V1_DN\n",
      "GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "GSE19923_WT_VS_E2A_KO_DP_THYMOCYTE_UP\n",
      "RIGGI_EWING_SARCOMA_PROGENITOR_DN\n",
      "GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "GSE17721_POLYIC_VS_CPG_0.5H_BMDC_DN\n",
      "GSE37416_CTRL_VS_0H_F_TULARENSIS_LVS_NEUTROPHIL_DN\n",
      "GSE15659_RESTING_VS_ACTIVATED_TREG_DN\n",
      "GO_ORGANIC_HYDROXY_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "GSE10463_CD40L_AND_VA347_VS_CD40L_IN_DC_DN\n",
      "chr22q12\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 28, 29, 0.0636652875256613\n",
      "best merge is 5, 23, 0.125836812811185\n",
      "best merge is 0, 1, 0.190445436817526\n",
      "best merge is 3, 14, 0.235434490651208\n",
      "best merge is 7, 25, 0.287221029410587\n",
      "best merge is 2, 21, 0.310653129875547\n",
      "best merge is 9, 13, 0.335127910483962\n",
      "best merge is 16, 20, 0.354130140561958\n",
      "best merge is 0, 5, 0.381871204849991\n",
      "best merge is 24, 26, 0.404519334342374\n",
      "best merge is 6, 8, 0.426936318848102\n",
      "best merge is 15, 17, 0.449120453251836\n",
      "best merge is 3, 9, 0.457677716738753\n",
      "best merge is 16, 19, 0.467225338412148\n",
      "best merge is 7, 10, 0.477214511996996\n",
      "best merge is 11, 24, 0.550972682492288\n",
      "best merge is 12, 18, 0.561402190484092\n",
      "best merge is 0, 28, 0.609430057230998\n",
      "best merge is 2, 27, 0.627798023148228\n",
      "best merge is 6, 7, 0.694401037599554\n",
      "best merge is 2, 11, 0.701959335926029\n",
      "best merge is 4, 12, 0.724944459438881\n",
      "best merge is 0, 15, 0.733773602716411\n",
      "best merge is 3, 22, 0.853060461352253\n",
      "best merge is 4, 16, 0.88254696296272\n",
      "best merge is 2, 6, 0.935479591074421\n",
      "order classes\n",
      "8 - 1.87936833987317\n",
      "6 - 2.60842997189811\n",
      "5 - 2.79048725447961\n",
      "6 - 3.12299849817016\n",
      "5 - 2.800796446968\n",
      "0 1 5 23 28 29 15 17\n",
      "2 21 27 11 24 26\n",
      "3 14 9 13 22\n",
      "4 12 18 16 20 19\n",
      "6 8 7 25 10\n",
      "Done.-9.98233215547703\t42\t0\n",
      "0.745583038869256\t42\t20.0353356890459\n",
      "1.85865724381625\t42\t21.4558303886926\n",
      "2.53710247349823\t42\t22.2614840989399\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.45936395759717\t42\t23.2579505300353\n",
      "3.85159010600707\t42\t23.660777385159\n",
      "4.23321554770318\t42\t24.0424028268551\n",
      "4.64664310954063\t42\t24.4240282685512\n",
      "5.1660777385159\t42\t24.86925795053\n",
      "13.7314487632509\t42\t25.4628975265018\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_full/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 50\n",
      "cat 28, mi = 0.00239446, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 45, mi = 0.00221575, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 34, mi = 0.00221327, HALLMARK_NOTCH_SIGNALING\n",
      "cat 32, mi = 0.00215537, HALLMARK_MYC_TARGETS_V2\n",
      "cat 43, mi = 0.00215469, HALLMARK_TGF_BETA_SIGNALING\n",
      "cat 38, mi = 0.00166398, HALLMARK_PEROXISOME\n",
      "cat 48, mi = 0.00165122, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "cat 41, mi = 0.00163682, HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY\n",
      "cat 16, mi = 0.00163388, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 40, mi = 0.00162037, HALLMARK_PROTEIN_SECRETION\n",
      "cat 5, mi = 0.00155107, HALLMARK_APICAL_SURFACE\n",
      "cat 11, mi = 0.00147059, HALLMARK_DNA_REPAIR\n",
      "cat 30, mi = 0.00146654, HALLMARK_MTORC1_SIGNALING\n",
      "cat 21, mi = 0.00142558, HALLMARK_HYPOXIA\n",
      "cat 19, mi = 0.00139845, HALLMARK_HEDGEHOG_SIGNALING\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.002, pvalue = 0.019, HALLMARK_KRAS_SIGNALING_UP\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.002, pvalue = 0.040, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.002, pvalue = 0.038, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.002, pvalue = 0.055, HALLMARK_MYC_TARGETS_V2\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.002, pvalue = 0.051, HALLMARK_TGF_BETA_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.002, pvalue = 0.197, HALLMARK_PEROXISOME\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.002, pvalue = 0.211, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.002, pvalue = 0.204, HALLMARK_REACTIVE_OXIGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.213, HALLMARK_FATTY_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.002, pvalue = 0.240, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.002, pvalue = 0.281, HALLMARK_APICAL_SURFACE\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.001, pvalue = 0.226, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.001, pvalue = 0.226, HALLMARK_MTORC1_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.223, HALLMARK_HYPOXIA\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.001, pvalue = 0.348, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.001, pvalue = 0.235, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.001, pvalue = 0.244, HALLMARK_G2M_CHECKPOINT\n",
      "Didn't pass (cutoff=0.0050): cat 0, mi = 0.001, pvalue = 0.284, HALLMARK_ADIPOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.280, HALLMARK_E2F_TARGETS\n",
      "Didn't pass (cutoff=0.0050): cat 22, mi = 0.001, pvalue = 0.258, HALLMARK_IL2_STAT5_SIGNALING\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.75\t-11.06\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_msigdb_h/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7491 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7491\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "681 hits go_terms_num = 790\n",
      "cat 321, mi = 0.00299641, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 354, mi = 0.00259630, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 402, mi = 0.00206889, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 484, mi = 0.00184703, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 672, mi = 0.00180462, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 31, mi = 0.00171525, AUH_K562_POSTAR_eCLIP\n",
      "cat 465, mi = 0.00164717, PPIL4_K562_POSTAR_eCLIP\n",
      "cat 629, mi = 0.00163006, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "cat 312, mi = 0.00161504, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "cat 270, mi = 0.00146837, HLTF_K562_POSTAR_eCLIP\n",
      "cat 668, mi = 0.00140719, TNRC6A_HEK293_Plass\n",
      "cat 300, mi = 0.00139383, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "cat 676, mi = 0.00137445, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 275, mi = 0.00124801, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 237, mi = 0.00123225, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 321, mi = 0.003, pvalue = 0.001, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.036, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.073, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.091, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.001, pvalue = 0.159, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 300, mi = 0.001, pvalue = 0.229, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.001, pvalue = 0.186, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.389, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.001, pvalue = 0.534, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 749, mi = 0.001, pvalue = 0.343, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.001, pvalue = 0.481, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.001, pvalue = 0.488, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.001, pvalue = 0.477, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.633, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.001, pvalue = 0.811, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.001, pvalue = 0.587, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.578, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 615, mi = 0.001, pvalue = 0.578, TAF15_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.001, pvalue = 0.916, HNRNPA1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass: cat 282, mi = 0.001, pvalue = 1.000, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass: cat 323, mi = 0.001, pvalue = 1.000, HNRNPU_HEK293T_POSTAR_CIMS\n",
      "Total number of categories evaluated = 25.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.75\t-11.06\n",
      "HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-10.0035335689046\t42\t0\n",
      "0.692579505300351\t42\t19.9929328621908\n",
      "1.80565371024735\t42\t21.3922261484099\n",
      "2.51590106007067\t42\t22.2190812720848\n",
      "3.03533568904594\t42\t22.8127208480565\n",
      "3.46996466431095\t42\t23.2579505300353\n",
      "3.87279151943463\t42\t23.6819787985866\n",
      "4.24381625441696\t42\t24.0636042402827\n",
      "4.65724381625441\t42\t24.4240282685512\n",
      "5.19787985865724\t42\t24.8904593639576\n",
      "13.7526501766784\t42\t25.5053003533569\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_all_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6816 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6816\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "620 hits go_terms_num = 790\n",
      "cat 321, mi = 0.00329113, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 354, mi = 0.00272079, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 402, mi = 0.00231024, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 312, mi = 0.00201224, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "cat 672, mi = 0.00198001, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 629, mi = 0.00197100, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "cat 31, mi = 0.00188466, AUH_K562_POSTAR_eCLIP\n",
      "cat 668, mi = 0.00176775, TNRC6A_HEK293_Plass\n",
      "cat 465, mi = 0.00174286, PPIL4_K562_POSTAR_eCLIP\n",
      "cat 484, mi = 0.00164319, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 615, mi = 0.00158900, TAF15_Brain_POSTAR_CIMS\n",
      "cat 270, mi = 0.00156324, HLTF_K562_POSTAR_eCLIP\n",
      "cat 300, mi = 0.00153102, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "cat 676, mi = 0.00150786, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 237, mi = 0.00135508, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 321, mi = 0.003, pvalue = 0.002, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.029, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.077, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.056, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.091, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 615, mi = 0.002, pvalue = 0.197, TAF15_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 300, mi = 0.002, pvalue = 0.222, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.002, pvalue = 0.184, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.375, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.001, pvalue = 0.339, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.342, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.001, pvalue = 0.529, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.001, pvalue = 0.485, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.001, pvalue = 0.554, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 749, mi = 0.001, pvalue = 0.524, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.698, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.001, pvalue = 0.589, HNRNPA1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 667, mi = 0.001, pvalue = 0.603, TNRC6A_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.818, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.001, pvalue = 0.779, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.001, pvalue = 0.777, EZH2_HCT116_POSTAR_iCLIP\n",
      "Total number of categories evaluated = 28.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.44\t-11.06\n",
      "HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.16967741935484\t42\t0\n",
      "6.55483870967742\t42\t25.6606451612903\n",
      "7.96387096774194\t42\t27.4490322580645\n",
      "8.83096774193548\t42\t28.4787096774194\n",
      "9.46774193548387\t42\t29.1832258064516\n",
      "10.0096774193548\t42\t29.7522580645161\n",
      "10.5109677419355\t42\t30.2670967741935\n",
      "10.9851612903226\t42\t30.7548387096774\n",
      "11.5135483870968\t42\t31.2154838709677\n",
      "12.1909677419355\t42\t31.8116129032258\n",
      "17.2851612903226\t42\t32.5703225806452\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6678 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6678\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "607 hits go_terms_num = 790\n",
      "cat 577, mi = 0.00347866, SMNDC1_K562_POSTAR_eCLIP\n",
      "cat 510, mi = 0.00280461, RBFOX2_K562_POSTAR_eCLIP\n",
      "cat 242, mi = 0.00274681, FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 531, mi = 0.00273015, RBM47_MDA-MB-231_POSTAR_CIMS\n",
      "cat 494, mi = 0.00268245, PUM2_HEK293_POSTAR_Piranha_0.01\n",
      "cat 774, mi = 0.00242260, YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "cat 308, mi = 0.00241201, HNRNPK_K562_POSTAR_eCLIP\n",
      "cat 355, mi = 0.00236858, IGF2BP3_PL45_POSTAR_iCLIP\n",
      "cat 533, mi = 0.00236146, RBM47_MDA-MB-231_POSTAR_Piranha_0.01\n",
      "cat 45, mi = 0.00230536, CDC40_HepG2_POSTAR_eCLIP\n",
      "cat 610, mi = 0.00230488, SUGP2_HepG2_POSTAR_eCLIP\n",
      "cat 753, mi = 0.00228751, YTHDC2_HeLa_POSTAR_Piranha_0.01\n",
      "cat 431, mi = 0.00224757, NKRF_HepG2_POSTAR_eCLIP\n",
      "cat 267, mi = 0.00224490, GTF2F1_K562_POSTAR_eCLIP\n",
      "cat 65, mi = 0.00223600, CPSF2_HEK293_POSTAR_PARalyzer\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 577, mi = 0.003, pvalue = 0.001, SMNDC1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 242, mi = 0.003, pvalue = 0.010, FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 531, mi = 0.003, pvalue = 0.007, RBM47_MDA-MB-231_POSTAR_CIMS\n",
      "Passed: cat 494, mi = 0.003, pvalue = 0.004, PUM2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 774, mi = 0.002, pvalue = 0.024, YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.002, pvalue = 0.032, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.053, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 240, mi = 0.002, pvalue = 0.058, FXR1_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 241, mi = 0.002, pvalue = 0.055, FXR1_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.002, pvalue = 0.075, EIF3D_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 504, mi = 0.002, pvalue = 0.063, QKI_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.109, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 626, mi = 0.002, pvalue = 0.189, TARBP2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.002, pvalue = 0.215, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.002, pvalue = 0.254, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 595, mi = 0.002, pvalue = 0.054, SRSF7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 502, mi = 0.002, pvalue = 0.269, QKI_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 627, mi = 0.002, pvalue = 0.322, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 538, mi = 0.001, pvalue = 0.267, RPS11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 514, mi = 0.001, pvalue = 0.413, RBM10_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 506, mi = 0.001, pvalue = 0.470, QKI_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 53, mi = 0.001, pvalue = 0.381, CNBP_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 244, mi = 0.001, pvalue = 0.547, FXR1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 142, mi = 0.001, pvalue = 0.619, DICER1_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 152.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.44\t-11.06\n",
      "PUM2_HEK293_POSTAR_Piranha_0.01\n",
      "SMNDC1_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.15612903225806\t42\t0\n",
      "6.5683870967742\t42\t25.6877419354839\n",
      "7.96387096774194\t42\t27.4490322580645\n",
      "8.83096774193548\t42\t28.4787096774194\n",
      "9.46774193548387\t42\t29.1832258064516\n",
      "10.0096774193548\t42\t29.7522580645161\n",
      "10.4974193548387\t42\t30.2670967741935\n",
      "10.9716129032258\t42\t30.7277419354839\n",
      "11.5\t42\t31.2154838709677\n",
      "12.1638709677419\t42\t31.7845161290323\n",
      "17.2716129032258\t42\t32.5432258064516\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6602 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6602\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "600 hits go_terms_num = 790\n",
      "cat 557, mi = 0.00281351, SERBP1_K562_POSTAR_eCLIP\n",
      "cat 160, mi = 0.00275741, EIF3B_HEK293T_POSTAR_PARalyzer\n",
      "cat 743, mi = 0.00250405, YBX3_K562_POSTAR_eCLIP\n",
      "cat 540, mi = 0.00248665, RPS3_HepG2_POSTAR_eCLIP\n",
      "cat 217, mi = 0.00248347, FMR1_HEK293_Plass\n",
      "cat 159, mi = 0.00248339, EIF3B_HEK293T_POSTAR_PAR-CLIP\n",
      "cat 9, mi = 0.00248201, AGO2_HEK293_Plass\n",
      "cat 75, mi = 0.00233894, CPSF6_HEK293_POSTAR_Piranha_0.01\n",
      "cat 504, mi = 0.00233395, QKI_HepG2_POSTAR_eCLIP\n",
      "cat 8, mi = 0.00227311, AGO1_HEK293_Plass\n",
      "cat 655, mi = 0.00220608, TIA1_K562_POSTAR_eCLIP\n",
      "cat 765, mi = 0.00220546, YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "cat 338, mi = 0.00215771, IGF2BP1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 169, mi = 0.00215406, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 524, mi = 0.00214623, RBM15_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 557, mi = 0.003, pvalue = 0.006, SERBP1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 160, mi = 0.003, pvalue = 0.007, EIF3B_HEK293T_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.003, pvalue = 0.022, YBX3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 540, mi = 0.002, pvalue = 0.020, RPS3_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 217, mi = 0.002, pvalue = 0.006, FMR1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.002, pvalue = 0.019, EIF3B_HEK293T_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.005, AGO2_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 75, mi = 0.002, pvalue = 0.026, CPSF6_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 504, mi = 0.002, pvalue = 0.036, QKI_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 8, mi = 0.002, pvalue = 0.013, AGO1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 655, mi = 0.002, pvalue = 0.051, TIA1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 765, mi = 0.002, pvalue = 0.053, YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 338, mi = 0.002, pvalue = 0.017, IGF2BP1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.020, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 524, mi = 0.002, pvalue = 0.043, RBM15_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 0, mi = 0.002, pvalue = 0.054, AARS_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 306, mi = 0.002, pvalue = 0.071, HNRNPK_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.002, pvalue = 0.078, DICER1_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 424, mi = 0.002, pvalue = 0.074, NCBP2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.002, pvalue = 0.076, DICER1_HEK293_POSTAR_PARalyzer\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.44\t-11.06\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6691 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6691\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "608 hits go_terms_num = 790\n",
      "cat 515, mi = 0.00342198, RBM10_HEK293_Plass\n",
      "cat 526, mi = 0.00334462, RBM22_HepG2_POSTAR_eCLIP\n",
      "cat 51, mi = 0.00324958, CF_Im59_HEK293_Plass\n",
      "cat 9, mi = 0.00299689, AGO2_HEK293_Plass\n",
      "cat 640, mi = 0.00268569, TARDBP_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 40, mi = 0.00261923, C22orf28_HEK293_Plass\n",
      "cat 428, mi = 0.00260595, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 522, mi = 0.00253509, RBM15_HepG2_POSTAR_eCLIP\n",
      "cat 240, mi = 0.00251161, FXR1_HEK293_POSTAR_PAR-CLIP\n",
      "cat 208, mi = 0.00242752, FBL_HEK293_Plass\n",
      "cat 317, mi = 0.00239126, HNRNPM_K562_POSTAR_eCLIP\n",
      "cat 557, mi = 0.00236983, SERBP1_K562_POSTAR_eCLIP\n",
      "cat 548, mi = 0.00235742, RTCB_HEK293_POSTAR_Piranha_0.01\n",
      "cat 403, mi = 0.00234067, MBNL2_frontalcortex_POSTAR_HITS-CLIP\n",
      "cat 145, mi = 0.00231347, DIS3L2_HEK293_POSTAR_HITS-CLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 515, mi = 0.003, pvalue = 0.001, RBM10_HEK293_Plass\n",
      "Passed: cat 526, mi = 0.003, pvalue = 0.000, RBM22_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 640, mi = 0.003, pvalue = 0.011, TARDBP_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 145, mi = 0.002, pvalue = 0.030, DIS3L2_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 146, mi = 0.002, pvalue = 0.032, DIS3L2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 225, mi = 0.002, pvalue = 0.024, FUS_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 356, mi = 0.002, pvalue = 0.046, IGF2BP3_Panc1_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 358, mi = 0.002, pvalue = 0.046, IGF2BP3_Panc1_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.002, pvalue = 0.075, HNRNPH1_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 319, mi = 0.002, pvalue = 0.067, HNRNPUL1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 304, mi = 0.002, pvalue = 0.085, HNRNPH1_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 623, mi = 0.002, pvalue = 0.062, TAF15_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 283, mi = 0.002, pvalue = 0.076, HNRNPA2B1_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 284, mi = 0.002, pvalue = 0.072, HNRNPA2B1_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 536, mi = 0.002, pvalue = 0.082, RBPMS_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 537, mi = 0.002, pvalue = 0.083, RBPMS_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.002, pvalue = 0.108, CNBP_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 538, mi = 0.002, pvalue = 0.047, RPS11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.002, pvalue = 0.022, DCK1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 691, mi = 0.002, pvalue = 0.129, U2AF2_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 595, mi = 0.002, pvalue = 0.120, SRSF7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 710, mi = 0.002, pvalue = 0.138, UPF1_HeLa,treated_with_puromycoin_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 43.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.44\t-11.06\n",
      "RBM22_HepG2_POSTAR_eCLIP\n",
      "RBM10_HEK293_Plass\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.15612903225806\t42\t0\n",
      "6.5683870967742\t42\t25.6877419354839\n",
      "7.96387096774194\t42\t27.4490322580645\n",
      "8.83096774193548\t42\t28.4787096774194\n",
      "9.4541935483871\t42\t29.1832258064516\n",
      "9.99612903225806\t42\t29.7251612903226\n",
      "10.4974193548387\t42\t30.2670967741935\n",
      "10.9716129032258\t42\t30.7277419354839\n",
      "11.5\t42\t31.2154838709677\n",
      "12.1638709677419\t42\t31.7845161290323\n",
      "17.2716129032258\t42\t32.5432258064516\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 790\n",
      "cat 354, mi = 0.00260886, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00246869, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 270, mi = 0.00214397, HLTF_K562_POSTAR_eCLIP\n",
      "cat 195, mi = 0.00211131, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00211131, EZH2_HCT116_POSTAR_iCLIP\n",
      "cat 672, mi = 0.00191497, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 657, mi = 0.00190765, TIA1_NA_POSTAR_CIMS\n",
      "cat 658, mi = 0.00190765, TIA1_NA_POSTAR_iCLIP\n",
      "cat 275, mi = 0.00187893, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 353, mi = 0.00185391, IGF2BP3_PL45_POSTAR_CIMS\n",
      "cat 31, mi = 0.00185103, AUH_K562_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00182023, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 465, mi = 0.00171977, PPIL4_K562_POSTAR_eCLIP\n",
      "cat 666, mi = 0.00170239, TNRC6A_HEK293_POSTAR_PARalyzer\n",
      "cat 208, mi = 0.00168617, FBL_HEK293_Plass\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.003, pvalue = 0.014, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.027, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.002, pvalue = 0.060, HLTF_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.059, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.002, pvalue = 0.061, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.115, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.002, pvalue = 0.121, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.002, pvalue = 0.123, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.132, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 353, mi = 0.002, pvalue = 0.091, IGF2BP3_PL45_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.146, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.162, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.002, pvalue = 0.160, PPIL4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 666, mi = 0.002, pvalue = 0.134, TNRC6A_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 208, mi = 0.002, pvalue = 0.144, FBL_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.177, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.181, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 329, mi = 0.002, pvalue = 0.168, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.233, SLBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 615, mi = 0.002, pvalue = 0.267, TAF15_Brain_POSTAR_CIMS\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.44\t-11.06\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "read_go_names_data: please enter a valid filename (/flash/hani/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_coding_gene_names/human_ensembl_RBPs_coding_gene_names_names.txt invalid)\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_names_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 101\n",
      "cat 6, mi = 0.00230180, CPEB4\n",
      "cat 69, mi = 0.00225742, RBMS3.1\n",
      "cat 9, mi = 0.00210654, ESRP2\n",
      "cat 68, mi = 0.00199236, RBMS3\n",
      "cat 62, mi = 0.00154539, RBM46\n",
      "cat 88, mi = 0.00129663, SRSF9.1\n",
      "cat 13, mi = 0.00120816, FXR2\n",
      "cat 34, mi = 0.00101992, KHDRBS3\n",
      "cat 27, mi = 0.00101726, HuR.2\n",
      "cat 21, mi = 0.00101205, HNRNPK\n",
      "cat 56, mi = 0.00092092, RBM3\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "cat 3, mi = 0.00000000, BRUNOL6\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.044, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.002, pvalue = 0.050, RBMS3.1\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.052, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.002, pvalue = 0.090, RBMS3\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.002, pvalue = 0.283, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.001, pvalue = 0.380, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.455, FXR2\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.835, KHDRBS3\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.613, HuR.2\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.913, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.644, RBM3\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, FMR1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.44\t-11.06\n",
      "Outputing EPS file deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.eps\n",
      "Convert to PDF deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/WTAP_peak_m6A_vs_input.ave.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt --expfile=deseq/WTAP_peak_m6A_vs_input.ave.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/hani/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/WTAP_peak_m6A_vs_input.ave.txt_PAGE/' -> 'ipage/WTAP_peak_m6A_vs_input_human_ensembl_RBPs_DeepBind/'\n"
     ]
    }
   ],
   "source": [
    "%%bash --err error\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "    # ave duplicate: peaks  \n",
    "    awk 'NR>1{arr[$1]   += $3; count[$1] += 1}END{for (a in arr) {print substr(a,1,15)\"\\t\"arr[a]/count[a]}}' deseq/${treat}_peak_m6A_vs_input.txt > deseq/${treat}_peak_m6A_vs_input.ave.txt\n",
    "    for f in `ls -d /flash/hani/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl*`; do\n",
    "        base=`basename \"$f\"`\n",
    "        perl $PAGEDIR/page.pl --expfile=deseq/${treat}_peak_m6A_vs_input.ave.txt \\\n",
    "        --species=$base --exptype=continuous --ebins=11 --nodups=1\n",
    "        mv -v deseq/${treat}_peak_m6A_vs_input.ave.txt_PAGE/ ipage/${treat}_peak_m6A_vs_input_${base}/\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash --err error\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "    for f in `ls -d /flash/hani/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl*`; do\n",
    "        base=`basename \"$f\"`        \n",
    "        magick -density 300 \\\n",
    "        ipage/${treat}_peak_m6A_vs_input_${base}/${treat}_peak_m6A_vs_input.ave.txt.summary.pdf \\\n",
    "        -flatten -quality 90 plots/ipage.${treat}_peak_m6A_vs_input_${base}.png\n",
    "    done\n",
    "done\n",
    "#         echo '<img src=\"plots/ipage.'${treat}_peak_m6A_vs_input_${base}.png'\" style=\"height:300px\">'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TEISER\n",
    "https://github.com/goodarzilab/TEISER\n",
    "\n",
    "(RNA count files (input samples) made in the `internal_control` notebook). \n",
    "\n",
    "Hypo and Hyper-methylation gene set analysis: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "METTL14\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sun Apr 26 07:53:17 UTC 2020\n",
      "Remove duplicates, create teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 58335 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.007387\n",
      "pass = 0.000000\n",
      "z = 170.107849\n",
      "freq 0: 0.000120\t1\t8334\t360\t58335\n",
      "freq 1: 0.000480\t4\t8334\t360\t58335\n",
      "freq 2: 0.017519\t146\t8334\t360\t58335\n",
      "freq 3: 0.004560\t38\t8334\t360\t58335\n",
      "freq 4: 0.000000\t0\t8334\t360\t58335\n",
      "freq 5: 0.019438\t162\t8334\t360\t58335\n",
      "freq 6: 0.001080\t9\t8331\t360\t58335\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.52\t-5.04\n",
      "0\n",
      "Outputing EPS file teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt.summary.eps\n",
      "Convert to PDF teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt.summary.eps teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET' -> 'teiser/METTL14_hyper_methylation.txt_GENESET'\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sun Apr 26 07:53:27 UTC 2020\n",
      "Remove duplicates, create teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 58335 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.107621\n",
      "pass = 0.000000\n",
      "z = 2483.733887\n",
      "freq 0: 0.000120\t1\t8334\t4562\t58335\n",
      "freq 1: 0.001920\t16\t8334\t4562\t58335\n",
      "freq 2: 0.107151\t893\t8334\t4562\t58335\n",
      "freq 3: 0.068755\t573\t8334\t4562\t58335\n",
      "freq 4: 0.000000\t0\t8334\t4562\t58335\n",
      "freq 5: 0.333213\t2777\t8334\t4562\t58335\n",
      "freq 6: 0.036250\t302\t8331\t4562\t58335\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.52\t-5.04\n",
      "0\n",
      "Outputing EPS file teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt.summary.eps\n",
      "Convert to PDF teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt.summary.eps teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET/METTL14_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/METTL14_vs_NT_log2FoldChange.txt_GENESET' -> 'teiser/METTL14_hypo_methylation.txt_GENESET'\n",
      "METTL3\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sun Apr 26 07:53:40 UTC 2020\n",
      "Remove duplicates, create teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 58335 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.011806\n",
      "pass = 0.000000\n",
      "z = 281.204773\n",
      "freq 0: 0.000120\t1\t8334\t651\t58335\n",
      "freq 1: 0.000600\t5\t8334\t651\t58335\n",
      "freq 2: 0.027598\t230\t8334\t651\t58335\n",
      "freq 3: 0.009959\t83\t8334\t651\t58335\n",
      "freq 4: 0.000000\t0\t8334\t651\t58335\n",
      "freq 5: 0.034557\t288\t8334\t651\t58335\n",
      "freq 6: 0.005281\t44\t8331\t651\t58335\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.64\t-4.98\n",
      "0\n",
      "Outputing EPS file teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt.summary.eps\n",
      "Convert to PDF teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt.summary.eps teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET' -> 'teiser/METTL3_hyper_methylation.txt_GENESET'\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sun Apr 26 07:53:51 UTC 2020\n",
      "Remove duplicates, create teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 58335 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.090167\n",
      "pass = 0.000000\n",
      "z = 2127.579834\n",
      "freq 0: 0.000000\t0\t8334\t4439\t58335\n",
      "freq 1: 0.001560\t13\t8334\t4439\t58335\n",
      "freq 2: 0.086513\t721\t8334\t4439\t58335\n",
      "freq 3: 0.069234\t577\t8334\t4439\t58335\n",
      "freq 4: 0.000000\t0\t8334\t4439\t58335\n",
      "freq 5: 0.293377\t2445\t8334\t4439\t58335\n",
      "freq 6: 0.081983\t683\t8331\t4439\t58335\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.64\t-4.98\n",
      "0\n",
      "Outputing EPS file teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt.summary.eps\n",
      "Convert to PDF teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt.summary.eps teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET/METTL3_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/METTL3_vs_NT_log2FoldChange.txt_GENESET' -> 'teiser/METTL3_hypo_methylation.txt_GENESET'\n",
      "Virma\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sun Apr 26 07:54:04 UTC 2020\n",
      "Remove duplicates, create teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 58335 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.007662\n",
      "pass = 0.000000\n",
      "z = 174.683060\n",
      "freq 0: 0.000000\t0\t8334\t390\t58335\n",
      "freq 1: 0.000840\t7\t8334\t390\t58335\n",
      "freq 2: 0.018119\t151\t8334\t390\t58335\n",
      "freq 3: 0.006240\t52\t8334\t390\t58335\n",
      "freq 4: 0.000000\t0\t8334\t390\t58335\n",
      "freq 5: 0.020518\t171\t8334\t390\t58335\n",
      "freq 6: 0.001080\t9\t8331\t390\t58335\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.7\t-4.78\n",
      "0\n",
      "Outputing EPS file teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt.summary.eps\n",
      "Convert to PDF teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt.summary.eps teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/Virma_vs_NT_log2FoldChange.txt_GENESET' -> 'teiser/Virma_hyper_methylation.txt_GENESET'\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sun Apr 26 07:54:14 UTC 2020\n",
      "Remove duplicates, create teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 58335 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.101648\n",
      "pass = 0.000000\n",
      "z = 2356.312988\n",
      "freq 0: 0.000000\t0\t8334\t4556\t58335\n",
      "freq 1: 0.002400\t20\t8334\t4556\t58335\n",
      "freq 2: 0.119030\t992\t8334\t4556\t58335\n",
      "freq 3: 0.079554\t663\t8334\t4556\t58335\n",
      "freq 4: 0.000000\t0\t8334\t4556\t58335\n",
      "freq 5: 0.312815\t2607\t8334\t4556\t58335\n",
      "freq 6: 0.032889\t274\t8331\t4556\t58335\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.7\t-4.78\n",
      "0\n",
      "Outputing EPS file teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt.summary.eps\n",
      "Convert to PDF teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt.summary.eps teiser/Virma_vs_NT_log2FoldChange.txt_GENESET/Virma_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/Virma_vs_NT_log2FoldChange.txt_GENESET' -> 'teiser/Virma_hypo_methylation.txt_GENESET'\n",
      "WTAP\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sun Apr 26 07:54:27 UTC 2020\n",
      "Remove duplicates, create teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 58335 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.010204\n",
      "pass = 0.000000\n",
      "z = 234.331207\n",
      "freq 0: 0.000120\t1\t8334\t584\t58335\n",
      "freq 1: 0.001080\t9\t8334\t584\t58335\n",
      "freq 2: 0.023398\t195\t8334\t584\t58335\n",
      "freq 3: 0.010319\t86\t8334\t584\t58335\n",
      "freq 4: 0.000000\t0\t8334\t584\t58335\n",
      "freq 5: 0.031198\t260\t8334\t584\t58335\n",
      "freq 6: 0.003961\t33\t8331\t584\t58335\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.64\t-4.96\n",
      "0\n",
      "Outputing EPS file teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt.summary.eps\n",
      "Convert to PDF teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt.summary.eps teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET' -> 'teiser/WTAP_hyper_methylation.txt_GENESET'\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sun Apr 26 07:54:37 UTC 2020\n",
      "Remove duplicates, create teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 58335 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.095398\n",
      "pass = 0.000000\n",
      "z = 2232.381104\n",
      "freq 0: 0.000000\t0\t8334\t4487\t58335\n",
      "freq 1: 0.001920\t16\t8334\t4487\t58335\n",
      "freq 2: 0.090713\t756\t8334\t4487\t58335\n",
      "freq 3: 0.067555\t563\t8334\t4487\t58335\n",
      "freq 4: 0.000000\t0\t8334\t4487\t58335\n",
      "freq 5: 0.308975\t2575\t8334\t4487\t58335\n",
      "freq 6: 0.069259\t577\t8331\t4487\t58335\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.64\t-4.96\n",
      "0\n",
      "Outputing EPS file teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt.summary.eps\n",
      "Convert to PDF teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt.summary.eps teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET/WTAP_vs_NT_log2FoldChange.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/WTAP_vs_NT_log2FoldChange.txt_GENESET' -> 'teiser/WTAP_hypo_methylation.txt_GENESET'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir -p teiser\n",
    "\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "    echo $treat\n",
    "    # exp input \n",
    "    awk -F \"\\t\" '{OFS=FS}NR>1{print substr($1,1,15), $3}' deseq/inputs_count_${treat}_vs_NT.txt | \\\n",
    "    awk '!seen[$1]++' > teiser/${treat}_vs_NT_log2FoldChange.txt\n",
    "    # hyper_methylation gene list\n",
    "    awk '$2>=1{print $1}' deseq/${treat}_peak_m6A_vs_input.ave.txt > teiser/${treat}_hyper_methylation.txt    \n",
    "    perl $TEISERDIR/run_mi_gene_list.pl \\\n",
    "    --expfile=teiser/${treat}_vs_NT_log2FoldChange.txt \\\n",
    "    --genefile=teiser/${treat}_hyper_methylation.txt \\\n",
    "    --exptype=continuous \\\n",
    "    --ebins=7 \\\n",
    "    --species=human \\\n",
    "    --doremovedups=0 \\\n",
    "    --doremoveextra=0\n",
    "    mv -v teiser/${treat}_vs_NT_log2FoldChange.txt_GENESET teiser/${treat}_hyper_methylation.txt_GENESET\n",
    "    # hypo_methylation gene list\n",
    "    awk '$2<1 {print $1}' deseq/${treat}_peak_m6A_vs_input.ave.txt > teiser/${treat}_hypo_methylation.txt\n",
    "    perl $TEISERDIR/run_mi_gene_list.pl \\\n",
    "    --expfile=teiser/${treat}_vs_NT_log2FoldChange.txt \\\n",
    "    --genefile=teiser/${treat}_hypo_methylation.txt \\\n",
    "    --exptype=continuous \\\n",
    "    --ebins=7 \\\n",
    "    --species=human \\\n",
    "    --doremovedups=0 \\\n",
    "    --doremoveextra=0\n",
    "    mv -v teiser/${treat}_vs_NT_log2FoldChange.txt_GENESET teiser/${treat}_hypo_methylation.txt_GENESET\n",
    "done "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<img src=\"plots/teiser.METTL14_hypo_methylation_GENESET.png\" style=\"height:300px\">\n",
      "<img src=\"plots/teiser.METTL14_hyper_methylation_GENESET.png\" style=\"height:300px\">\n",
      "<img src=\"plots/teiser.METTL3_hypo_methylation_GENESET.png\" style=\"height:300px\">\n",
      "<img src=\"plots/teiser.METTL3_hyper_methylation_GENESET.png\" style=\"height:300px\">\n",
      "<img src=\"plots/teiser.Virma_hypo_methylation_GENESET.png\" style=\"height:300px\">\n",
      "<img src=\"plots/teiser.Virma_hyper_methylation_GENESET.png\" style=\"height:300px\">\n",
      "<img src=\"plots/teiser.WTAP_hypo_methylation_GENESET.png\" style=\"height:300px\">\n",
      "<img src=\"plots/teiser.WTAP_hyper_methylation_GENESET.png\" style=\"height:300px\">\n"
     ]
    }
   ],
   "source": [
    "%%bash --err error\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "\n",
    "    magick -density 300 \\\n",
    "    teiser/${treat}_hyper_methylation.txt_GENESET/*pdf \\\n",
    "    -flatten -quality 90 plots/teiser.${treat}_hyper_methylation_GENESET.png\n",
    "    \n",
    "    magick -density 300 \\\n",
    "    teiser/${treat}_hypo_methylation.txt_GENESET/*pdf \\\n",
    "    -flatten -quality 90 plots/teiser.${treat}_hypo_methylation_GENESET.png\n",
    "\n",
    "    echo '<img src=\"'plots/teiser.${treat}_hypo_methylation_GENESET.png'\" style=\"height:300px\">'\n",
    "    echo '<img src=\"'plots/teiser.${treat}_hyper_methylation_GENESET.png'\" style=\"height:300px\">'\n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### using bam to bed for cov count \n",
    "\n",
    "1. `intersectBed -split` make it work with bed12 file\n",
    "\n",
    "3. make a intersect gtf to extract names for later. `"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash \n",
    "# bamToBed -i bam/s23.human.METTL3.input.bam | intersectBed -s -split -wo -a - -b exomepeak/control/peak.bed | awk '{print $10}' | sort  tail"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash \n",
    "# bamToBed -i bam/s23.human.METTL3.input.bam | intersectBed -s -split -wo -a - -b exomepeak/control/peak.bed \\\n",
    "# | cut -f10 | sort | uniq -c | awk '{ print $2 \"\\t\" $1}' | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R version 3.5.1 (2018-07-02)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 18.04.2 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /rumi/shams/abe/anaconda3/envs/deseq/lib/R/lib/libRlapack.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      " [1] grid      parallel  stats4    tools     stats     graphics  grDevices\n",
      " [8] utils     datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      " [1] rtracklayer_1.40.6          data.table_1.11.6          \n",
      " [3] Rsamtools_1.32.3            Biostrings_2.48.0          \n",
      " [5] XVector_0.20.0              Gviz_1.24.0                \n",
      " [7] pheatmap_1.0.12             forcats_0.4.0              \n",
      " [9] stringr_1.4.0               dplyr_0.8.3                \n",
      "[11] purrr_0.3.3                 readr_1.3.1                \n",
      "[13] tidyr_1.0.0                 tibble_2.1.3               \n",
      "[15] tidyverse_1.2.1             ggplot2_3.2.1              \n",
      "[17] DESeq2_1.20.0               SummarizedExperiment_1.10.1\n",
      "[19] DelayedArray_0.6.6          BiocParallel_1.14.2        \n",
      "[21] matrixStats_0.55.0          Biobase_2.40.0             \n",
      "[23] GenomicRanges_1.32.7        GenomeInfoDb_1.16.0        \n",
      "[25] IRanges_2.14.12             S4Vectors_0.18.3           \n",
      "[27] BiocGenerics_0.26.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      " [1] colorspace_1.4-1         biovizBase_1.28.2        htmlTable_1.13.2        \n",
      " [4] base64enc_0.1-3          dichromat_2.0-0          rstudioapi_0.10         \n",
      " [7] bit64_0.9-7              AnnotationDbi_1.42.1     lubridate_1.7.4         \n",
      "[10] xml2_1.2.2               splines_3.5.1            geneplotter_1.58.0      \n",
      "[13] knitr_1.25               zeallot_0.1.0            Formula_1.2-3           \n",
      "[16] jsonlite_1.6             broom_0.5.0              annotate_1.58.0         \n",
      "[19] cluster_2.1.0            compiler_3.5.1           httr_1.4.1              \n",
      "[22] backports_1.1.5          assertthat_0.2.1         Matrix_1.2-17           \n",
      "[25] lazyeval_0.2.2           cli_1.1.0                acepack_1.4.1           \n",
      "[28] htmltools_0.4.0          prettyunits_1.0.2        gtable_0.3.0            \n",
      "[31] glue_1.3.1               GenomeInfoDbData_1.1.0   Rcpp_1.0.3              \n",
      "[34] cellranger_1.1.0         vctrs_0.2.0              nlme_3.1-141            \n",
      "[37] xfun_0.10                rvest_0.3.5              lifecycle_0.1.0         \n",
      "[40] ensembldb_2.4.1          XML_3.98-1.20            zlibbioc_1.26.0         \n",
      "[43] scales_1.0.0             BSgenome_1.48.0          VariantAnnotation_1.26.1\n",
      "[46] hms_0.5.2                ProtGenerics_1.12.0      AnnotationFilter_1.4.0  \n",
      "[49] RColorBrewer_1.1-2       curl_4.2                 memoise_1.1.0           \n",
      "[52] gridExtra_2.3            biomaRt_2.36.1           rpart_4.1-15            \n",
      "[55] latticeExtra_0.6-28      stringi_1.4.3            RSQLite_2.1.2           \n",
      "[58] genefilter_1.62.0        checkmate_1.9.4          GenomicFeatures_1.32.3  \n",
      "[61] rlang_0.4.1              pkgconfig_2.0.3          bitops_1.0-6            \n",
      "[64] lattice_0.20-38          GenomicAlignments_1.16.0 htmlwidgets_1.5.1       \n",
      "[67] bit_1.1-14               tidyselect_0.2.5         magrittr_1.5            \n",
      "[70] R6_2.4.0                 Hmisc_4.2-0              DBI_1.0.0               \n",
      "[73] pillar_1.4.2             haven_2.2.0              foreign_0.8-72          \n",
      "[76] withr_2.1.2              survival_2.44-1.1        RCurl_1.95-4.12         \n",
      "[79] nnet_7.3-12              modelr_0.1.5             crayon_1.3.4            \n",
      "[82] progress_1.2.2           locfit_1.5-9.1           readxl_1.3.1            \n",
      "[85] blob_1.2.0               digest_0.6.22            xtable_1.8-4            \n",
      "[88] munsell_0.5.0           \n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
