{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Protocol used by Albertas to make this data: https://benchling.com/anavickas/f/lib_VNejBHt6-protocols/prt_2Bkfh557-meripseq/edit\n",
    "\n",
    "HIV-1 IIIB Virus\n",
    "\n",
    "\n",
    "https://aidsreagent.org/reagentdetail.cfm?t=viruses&id=607\n",
    "\n",
    "We are starting from `fastq` files as raw data: "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Trimming "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cd ~/People/Judd/fastq\n",
    "for f in *_L002_R1_001.fastq.gz; do  \n",
    "    out=${f/\\_S[1-9]*/.trim.fastq.gz}; \n",
    "    cutadapt -j 12 -q 15 -m 20 -a NNNAGATCGGAAGAGCACACGTCTGAACTCCAGTCA -o ../trim/$out $f; \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alignment\n",
    "\n",
    "Unmmaped reads for each `fastq` file saved for later alignment to HIV genome \n",
    "\n",
    "Human genome: `gencode.v28.annotation.gtf`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "STAR --genomeLoad LoadAndExit --genomeDir /rumi/shams/abe/genomes/hg38/\n",
    "for f in trim/*.fastq.gz;do\n",
    "    out=${f/.trim.fastq.gz/_};\n",
    "    out=${out/trim/bam};\n",
    "    STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --runThreadN 16 --genomeDir /rumi/shams/abe/genomes/hg38/ --readFilesIn $f --outFileNamePrefix $out --outReadsUnmapped Fastx;\n",
    "done\n",
    "STAR --genomeLoad Remove --genomeDir /rumi/shams/abe/genomes/hg38/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All `bam`  files (mapped to human genome or hiv genome) renamed in this format: \n",
    "> `<sample>.<species>.<treatment>.<input/m6A>.bam`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Id transfer \n",
    "Read gene names from `gtf` file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import os\n",
    "import feather as ft\n",
    "\n",
    "\n",
    "def gtf_id_transfer(gtf):\n",
    "    # awk code to extract gene ID and Name \n",
    "    gtf_raw = os.popen(\"cat \" + gtf + \" | awk 'BEGIN{FS=\\\"\\t\\\"}\\\n",
    "                       {split($9,a,\\\";\\\"); if($3~\\\"gene\\\") print a[1]\\\"\\t\\\"a[3]}' | sed 's/gene_id \\\"//' | sed 's/gene_name \\\"//' | sed 's/\\\"//g' | sed 's/ //g'\").read().split('\\n')\n",
    "    # data frame \n",
    "    gtf_df = pd.DataFrame(\n",
    "        # col: Names \n",
    "        data=[i.split('\\t') for i in gtf_raw[:len(gtf_raw)-1]],\n",
    "#         # index: ids \n",
    "#         index=[i.split('\\t')[0] for i in gtf_raw[:len(gtf_raw)-1]],\n",
    "        columns=['ID','Name'], dtype=str)\n",
    "    return gtf_df\n",
    "\n",
    "gtf = '/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf'\n",
    "gtf_df = gtf_id_transfer(gtf)\n",
    "\n",
    "# pass data between R and Python using feather-format\n",
    "ft.write_dataframe(gtf_df, 'gtf_df.ft')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(feather)\n",
    "gtf_df = read_feather('gtf_df.ft') %>% remove_rownames %>% column_to_rownames(var=\"ID\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peakcalling \n",
    "\n",
    "## Control\n",
    "\n",
    "We used different conda `environment` for exomepeak (and it takes long time to run!!). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(exomePeak)\n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')\n",
    "setwd(\"/rumi/shams/abe/People/Judd/bam\")\n",
    "print (txdb)\n",
    "res <- exomepeak(\n",
    "    TXDB=txdb,\n",
    "    IP_BAM=list.files(pattern='human.NT.m6A.bam$'),\n",
    "    INPUT_BAM=list.files(pattern='human.NT.input.bam$'),\n",
    "    OUTPUT_DIR='../exomepeak',\n",
    "    EXPERIMENT_NAME='control'\n",
    ")\n",
    "saveRDS(res, paste('../exomepeak','control', 'results.rds', sep='/'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot _Guitar_ \n",
    "Now, let's draw metagene plots for control samples:\n",
    "(using seprate env: `Guitar`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Guitar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"20200325183907\"\n",
      "[1] \"There are 203835 transcripts of 58381 genes in the genome.\"\n",
      "[1] \"total 203835 transcripts extracted ...\"\n",
      "[1] \"total 94682 transcripts left after ambiguity filter ...\"\n",
      "[1] \"total 94682 transcripts left after check chromosome validity ...\"\n",
      "[1] \"total 11625 mRNAs left after component length filter ...\"\n",
      "[1] \"total 56222 ncRNAs left after ncRNA length filter ...\"\n",
      "[1] \"generate components for all tx\"\n",
      "[1] \"generate components for mRNA\"\n",
      "[1] \"generate components for lncRNA\"\n",
      "[1] \"generate chiped transcriptome\"\n",
      "[1] \"generate coverage checking ranges for tx\"\n",
      "[1] \"generate coverage checking ranges for mrna\"\n",
      "[1] \"generate coverage checking ranges for ncrna\"\n",
      "[1] \"20200325184342\"\n",
      "[1] \"import BED file exomepeak/control/peak.bed\"\n",
      "[1] \"sample 10 points for Group1\"\n",
      "[1] \"start figure plotting for tx ...\"\n",
      "[1] \"start figure plotting for mrna ...\"\n",
      "[1] \"start figure plotting for ncrna ...\"\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "GuitarPlot(txTxdb = txdb,stBedFiles = list(\"exomepeak/control/peak.bed\"),miscOutFilePrefix = \"control\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !magick -density 300 plots/control_mrna_test.pdf -flatten -quality 90 plots/control_mrna_test.png"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Motif analysis "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cd ~/People/Judd/exomepeak/control/\n",
    "bedtools getfasta -name -s -fi /rumi/shams/genomes/hg38/hg38.fa -bed peak.bed -split -fo peak.fa\n",
    "# prepare fasta file for FIRE\n",
    "perl /flash/hani/bin/Tools/prep_seqs_for_teiser_run.pl peak.fa peaks\n",
    "# run FIRE --doskipdiscovery=1\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --doskipdiscovery=1 --motiffile_rna=motifs_of_interest.txt --oribiasonly=0\n",
    "mv peaks_teiser.txt_FIRE/ motifs_of_interest_FIRE\n",
    "# run FIRE --doskipdiscovery=0\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --oribiasonly=0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/rumi/shams/abe/People/Judd/exomepeak/control/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.densities.pdf\r\n",
      "/rumi/shams/abe/People/Judd/exomepeak/control/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf\r\n",
      "/rumi/shams/abe/People/Judd/exomepeak/control/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.summary.pdf\r\n",
      "/rumi/shams/abe/People/Judd/exomepeak/control/peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.pdf\r\n",
      "/rumi/shams/abe/People/Judd/exomepeak/control/peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf\r\n",
      "/rumi/shams/abe/People/Judd/exomepeak/control/peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.pdf\r\n"
     ]
    }
   ],
   "source": [
    "ls ~/People/Judd/exomepeak/control/*FIRE/*/*pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "### motifs_of_interest\n",
      "<img src=\"plots/fire.motifs_of_interest.densities.png\" title=\"densities\" style=\"height:600px\">\n",
      "<img src=\"plots/fire.motifs_of_interest.fullmimatrix.png\" title=\"fullmimatrix\" style=\"height:600px\">\n",
      "<img src=\"plots/fire.motifs_of_interest.summary.png\" title=\"summary\" style=\"height:600px\">\n",
      "### discovery_mode\n",
      "<img src=\"plots/fire.discovery_mode.densities.png\" title=\"densities\" style=\"height:600px\">\n",
      "<img src=\"plots/fire.discovery_mode.fullmimatrix.png\" title=\"fullmimatrix\" style=\"height:600px\">\n",
      "<img src=\"plots/fire.discovery_mode.summary.png\" title=\"summary\" style=\"height:600px\">\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "cd ~/People/Judd/\n",
    "echo '### motifs_of_interest'\n",
    "for f in exomepeak/control/motifs_of_interest_FIRE/*/*pdf; do \n",
    "    base=`basename $f`\n",
    "    base=${base/peaks_teiser.txt./}\n",
    "    base=${base/.pdf/}\n",
    "#     magick -density 300 $f -flatten -quality 90 plots/fire.motifs_of_interest.$base.png\n",
    "    echo '<img src=\"plots/fire.motifs_of_interest.'$base'.png\" title=\"'$base'\" style=\"height:600px\">'\n",
    "done \n",
    "\n",
    "echo '### discovery_mode'\n",
    "for f in exomepeak/control/peaks_teiser.txt_FIRE/*/*pdf; do \n",
    "    base=`basename $f`\n",
    "    base=${base/peaks_teiser.txt./}\n",
    "    base=${base/.pdf/}\n",
    "#     magick -density 300 $f -flatten -quality 90 plots/fire.discovery_mode.$base.png\n",
    "    echo '<img src=\"plots/fire.discovery_mode.'$base'.png\" title=\"'$base'\" style=\"height:600px\">'\n",
    "done "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running a custom pipeline \n",
    "\n",
    "Here we're using https://github.com/gold-lab/bed12ToGTF to make gtf file from control exomepeak result.\n",
    "\n",
    "https://metacpan.org/pod/Bio::FeatureIO::gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat exomepeak/control/peak.bed | sort -k4 | \\\n",
    "# rename features peak by peak\n",
    "awk -F \"\\t\" '{OFS=FS}NR>1{$4=$4\"_peak\"}; cnt[$4]++{$4=$4\"_\"cnt[$4]} 1' | \\\n",
    "# bed12 to gtf\n",
    "perl ../../Workflows/bed12ToGTF/bed12ToGTF.1.pl | \\\n",
    "awk '{gsub(/\\\"|\\;/,\"\")}1''{gsub(/transcript_id/,\"gene_id\")}1''{gsub(/ exon_number [[:digit:]]/,\"\")}1' > exomepeak/control/peak.gtf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using `featureCounts` to count reads which are present in control peak cordinates (found by `exomepeak`). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.METTL14.input.bam                ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.METTL14.input.fc             ||\n",
      "||                 Summary : peakcount/s23.human.METTL14.input.fc.summary     ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.METTL14.input.bam...                        ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 38750994                                                  ||\n",
      "||    Successfully assigned reads : 939351 (2.4%)                             ||\n",
      "||    Running time : 0.84 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.MET  ||\n",
      "|| TL14.input.fc.summary\"                                                     ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.METTL14.m6A.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.METTL14.m6A.fc               ||\n",
      "||                 Summary : peakcount/s23.human.METTL14.m6A.fc.summary       ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.METTL14.m6A.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 17154692                                                  ||\n",
      "||    Successfully assigned reads : 9044401 (52.7%)                           ||\n",
      "||    Running time : 0.37 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.MET  ||\n",
      "|| TL14.m6A.fc.summary\"                                                       ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.METTL14.input.bam                ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.METTL14.input.fc             ||\n",
      "||                 Summary : peakcount/s24.human.METTL14.input.fc.summary     ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.METTL14.input.bam...                        ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 64012763                                                  ||\n",
      "||    Successfully assigned reads : 822388 (1.3%)                             ||\n",
      "||    Running time : 1.40 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.MET  ||\n",
      "|| TL14.input.fc.summary\"                                                     ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.METTL14.m6A.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.METTL14.m6A.fc               ||\n",
      "||                 Summary : peakcount/s24.human.METTL14.m6A.fc.summary       ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.METTL14.m6A.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8506966                                                   ||\n",
      "||    Successfully assigned reads : 4543360 (53.4%)                           ||\n",
      "||    Running time : 0.19 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.MET  ||\n",
      "|| TL14.m6A.fc.summary\"                                                       ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.METTL3.input.bam                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.METTL3.input.fc              ||\n",
      "||                 Summary : peakcount/s23.human.METTL3.input.fc.summary      ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.METTL3.input.bam...                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 45742294                                                  ||\n",
      "||    Successfully assigned reads : 1149406 (2.5%)                            ||\n",
      "||    Running time : 1.02 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.MET  ||\n",
      "|| TL3.input.fc.summary\"                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.METTL3.m6A.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.METTL3.m6A.fc                ||\n",
      "||                 Summary : peakcount/s23.human.METTL3.m6A.fc.summary        ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.METTL3.m6A.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 12180232                                                  ||\n",
      "||    Successfully assigned reads : 5997991 (49.2%)                           ||\n",
      "||    Running time : 0.27 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.MET  ||\n",
      "|| TL3.m6A.fc.summary\"                                                        ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.METTL3.input.bam                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.METTL3.input.fc              ||\n",
      "||                 Summary : peakcount/s24.human.METTL3.input.fc.summary      ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.METTL3.input.bam...                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 47661894                                                  ||\n",
      "||    Successfully assigned reads : 714579 (1.5%)                             ||\n",
      "||    Running time : 1.05 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.MET  ||\n",
      "|| TL3.input.fc.summary\"                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.METTL3.m6A.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.METTL3.m6A.fc                ||\n",
      "||                 Summary : peakcount/s24.human.METTL3.m6A.fc.summary        ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.METTL3.m6A.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 11789267                                                  ||\n",
      "||    Successfully assigned reads : 5806586 (49.3%)                           ||\n",
      "||    Running time : 0.25 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.MET  ||\n",
      "|| TL3.m6A.fc.summary\"                                                        ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.Virma.input.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.Virma.input.fc               ||\n",
      "||                 Summary : peakcount/s23.human.Virma.input.fc.summary       ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.Virma.input.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 43594515                                                  ||\n",
      "||    Successfully assigned reads : 860140 (2.0%)                             ||\n",
      "||    Running time : 0.97 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.Vir  ||\n",
      "|| ma.input.fc.summary\"                                                       ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.Virma.m6A.bam                    ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.Virma.m6A.fc                 ||\n",
      "||                 Summary : peakcount/s23.human.Virma.m6A.fc.summary         ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.Virma.m6A.bam...                            ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 15882391                                                  ||\n",
      "||    Successfully assigned reads : 8113156 (51.1%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.Vir  ||\n",
      "|| ma.m6A.fc.summary\"                                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.Virma.input.bam                  ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.Virma.input.fc               ||\n",
      "||                 Summary : peakcount/s24.human.Virma.input.fc.summary       ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.Virma.input.bam...                          ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 56471415                                                  ||\n",
      "||    Successfully assigned reads : 689932 (1.2%)                             ||\n",
      "||    Running time : 1.24 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.Vir  ||\n",
      "|| ma.input.fc.summary\"                                                       ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.Virma.m6A.bam                    ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.Virma.m6A.fc                 ||\n",
      "||                 Summary : peakcount/s24.human.Virma.m6A.fc.summary         ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.Virma.m6A.bam...                            ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 12393273                                                  ||\n",
      "||    Successfully assigned reads : 6371965 (51.4%)                           ||\n",
      "||    Running time : 0.27 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.Vir  ||\n",
      "|| ma.m6A.fc.summary\"                                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.WTAP.input.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.WTAP.input.fc                ||\n",
      "||                 Summary : peakcount/s23.human.WTAP.input.fc.summary        ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.WTAP.input.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 37223063                                                  ||\n",
      "||    Successfully assigned reads : 1025554 (2.8%)                            ||\n",
      "||    Running time : 0.83 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.WTA  ||\n",
      "|| P.input.fc.summary\"                                                        ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.WTAP.m6A.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.WTAP.m6A.fc                  ||\n",
      "||                 Summary : peakcount/s23.human.WTAP.m6A.fc.summary          ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.WTAP.m6A.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 14386845                                                  ||\n",
      "||    Successfully assigned reads : 7796629 (54.2%)                           ||\n",
      "||    Running time : 0.30 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.WTA  ||\n",
      "|| P.m6A.fc.summary\"                                                          ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.WTAP.input.bam                   ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.WTAP.input.fc                ||\n",
      "||                 Summary : peakcount/s24.human.WTAP.input.fc.summary        ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.WTAP.input.bam...                           ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 75448172                                                  ||\n",
      "||    Successfully assigned reads : 867570 (1.1%)                             ||\n",
      "||    Running time : 1.64 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.WTA  ||\n",
      "|| P.input.fc.summary\"                                                        ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.WTAP.m6A.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.WTAP.m6A.fc                  ||\n",
      "||                 Summary : peakcount/s24.human.WTAP.m6A.fc.summary          ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.WTAP.m6A.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 14438949                                                  ||\n",
      "||    Successfully assigned reads : 7201731 (49.9%)                           ||\n",
      "||    Running time : 0.31 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.WTA  ||\n",
      "|| P.m6A.fc.summary\"                                                          ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for enz in \"${StringArray[@]}\"; \n",
    "    do for f in bam/*human.$enz*.bam; \n",
    "        do out=${f/.bam/.fc}; out=${out/bam/peakcount}; \n",
    "            featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/control/peak.gtf -o $out $f; \n",
    "    done;\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.NT.input.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.NT.input.fc                  ||\n",
      "||                 Summary : peakcount/s23.human.NT.input.fc.summary          ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.NT.input.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 41946113                                                  ||\n",
      "||    Successfully assigned reads : 601429 (1.4%)                             ||\n",
      "||    Running time : 1.46 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.NT.  ||\n",
      "|| input.fc.summary\"                                                          ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s23.human.NT.m6A.bam                       ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s23.human.NT.m6A.fc                    ||\n",
      "||                 Summary : peakcount/s23.human.NT.m6A.fc.summary            ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s23.human.NT.m6A.bam...                               ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 15481162                                                  ||\n",
      "||    Successfully assigned reads : 8231587 (53.2%)                           ||\n",
      "||    Running time : 0.52 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s23.human.NT.  ||\n",
      "|| m6A.fc.summary\"                                                            ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.NT.input.bam                     ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.NT.input.fc                  ||\n",
      "||                 Summary : peakcount/s24.human.NT.input.fc.summary          ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.NT.input.bam...                             ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 39882640                                                  ||\n",
      "||    Successfully assigned reads : 823906 (2.1%)                             ||\n",
      "||    Running time : 1.25 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.NT.  ||\n",
      "|| input.fc.summary\"                                                          ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/s24.human.NT.m6A.bam                       ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/s24.human.NT.m6A.fc                    ||\n",
      "||                 Summary : peakcount/s24.human.NT.m6A.fc.summary            ||\n",
      "||              Annotation : exomepeak/control/peak.gtf (GTF)                 ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/control/peak.gtf ...                        ||\n",
      "||    Features : 8499                                                         ||\n",
      "||    Meta-features : 7529                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/s24.human.NT.m6A.bam...                               ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 13525429                                                  ||\n",
      "||    Successfully assigned reads : 7258122 (53.7%)                           ||\n",
      "||    Running time : 0.50 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/s24.human.NT.  ||\n",
      "|| m6A.fc.summary\"                                                            ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in bam/*human.NT*.bam; do \n",
    "    out=${f/.bam/.fc}; out=${out/bam/peakcount}; \n",
    "    featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/control/peak.gtf -o $out $f; \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Differential methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(DESeq2)\n",
    "library(tidyverse)\n",
    "library(ggplot2)\n",
    "require(ggrepel)\n",
    "library(patchwork)\n",
    "library(feather)\n",
    "gtf_df = read_feather('gtf_df.ft') %>% remove_rownames %>% column_to_rownames(var=\"ID\")\n",
    "\n",
    "pca = list(); vol = list(); hist = list()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "using `DESeq2` to find differential methylation peaks: \n",
    "\n",
    "### Contrast design: $\\frac{m6A^{treated}}{input^{treated}}$ vs.$\\frac{m6A^{nontreated}}{input^{nontreated}}$"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n",
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n",
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n",
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "all_dds = list()\n",
    "all_res = list()\n",
    "for (treat in c('METTL14','METTL3','Virma','WTAP'\n",
    "               )){\n",
    "    treat.input <- list.files(path = 'peakcount',pattern=paste('human.',treat,\".input.fc$\",sep = ''))\n",
    "    treat.m6A <- list.files(path = 'peakcount',pattern=paste('human.',treat,\".m6A.fc$\",sep = ''))\n",
    "    \n",
    "    n_treat.input <- list.files(path = 'peakcount',pattern=paste(\"human.NT.input.fc$\",sep = ''))\n",
    "    n_treat.m6A <- list.files(path = 'peakcount',pattern=paste(\"human.NT.m6A.fc$\",sep = ''))\n",
    "    \n",
    "    files = c(treat.input, treat.m6A, n_treat.input, n_treat.m6A)\n",
    "    # meta\n",
    "    reps <- rep(c('s23','s24'),2)\n",
    "    ip <- c(rep('input',2),rep('m6A',2))\n",
    "    conds <- c(paste(treat,ip, sep='.'), paste('NT',ip, sep='.'))\n",
    "    all_conds <- c(paste(reps,treat,ip, sep='.'), paste(reps,'NT',ip, sep='.'))\n",
    "    datalist <- lapply(files, function(x){read.table(file=paste('peakcount/', x, sep = \"\"),skip=1,sep='\\t',header=TRUE,colClasses=c(NA, \"NULL\", \"NULL\", \"NULL\", \"NULL\", \"NULL\", NA),check.names=F)})\n",
    "    m <- Reduce(function(...) merge(..., by=1, all = TRUE), datalist)\n",
    "    rownames(m) <- m[,1]\n",
    "    m <- m[,-1]\n",
    "    m[is.na(m)]<-0\n",
    "    m <- round(m)\n",
    "    colnames(m) <- all_conds\n",
    "    colData <- data.frame(cond = conds, rep = reps, row.names=all_conds)\n",
    "    dds <- DESeqDataSetFromMatrix(countData = m, colData = colData, design = ~ cond + rep)\n",
    "    dds$cond <- relevel(dds$cond, ref=\"NT.input\")\n",
    "    dds <- DESeq(dds)\n",
    "    all_dds[[treat]] = dds\n",
    "    # normalized counts \n",
    "    ncu <- counts(dds, normalized=TRUE)\n",
    "    # contrast design: (treat_m6A - treat_input) - (NT_m6A - NT_input)\n",
    "    res <- results(dds, \n",
    "                   contrast=list(\n",
    "                       c(paste(\"cond_\",treat,\".m6A_vs_NT.input\",sep=\"\")), \n",
    "                       c(paste(\"cond_\",treat,\".input_vs_NT.input\",sep=\"\"),\n",
    "                         \"cond_NT.m6A_vs_NT.input\")), \n",
    "                   listValues=c(1,-1))    \n",
    "    # idtransfer\n",
    "    res <- as.data.frame(res)\n",
    "    peak_ids = strsplit(rownames(m), '_',)\n",
    "    ids = unlist(lapply(peak_ids, '[[', 1))\n",
    "    res$name = gtf_df[ids,]\n",
    "    \n",
    "    all_res[[treat]] = res\n",
    "    # save files \n",
    "    write.table(res, paste('deseq/d_mtyl_',treat,'_vs_NT.txt',sep=''), sep=\"\\t\", quote=FALSE, col.names=NA)\n",
    "    # Raw tabels \n",
    "    write.table(ncu, file=paste('deseq/d_mtyl_',treat,'_peak_count_table_DE2norm.txt',sep=''), sep=\"\\t\", quote=FALSE, col.names=all_conds) \n",
    "    write.table(m, file=paste('deseq/d_mtyl_',treat,'_peak_count_table_raw.txt',sep=''), sep=\"\\t\", quote=FALSE, col.names=all_conds)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plots "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# %%R\n",
    "# ma = list()\n",
    "# for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "#     res = all_res[[treat]]\n",
    "# #     ma[[treat]] = plotMA(res)\n",
    "# }    \n",
    "# p = ma$METTL14 #+ ma$METTL3 + ma$Virma + ma$WTAP\n",
    "# ggsave(\"plots/peak_plotMA.png\", plot = p, device = 'png', dpi = 300)\n",
    "# ggsave(\"plots/peak_plotMA.pdf\", plot = p, device = 'pdf', dpi = 300)\n",
    "# p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////isF19AAAgAElEQVR4nOydB2BT1frAeX/XU3E8EXiKigO3ggqoILtsEFFwoLgHIiiIIjJE9t7Klr2hZRXoSJO0dO91M9o0bdbtyt5p1vmfc2/SjN6UpLRpk5dP6T05+c6535ffvWfdc8/pBCIS1tKpvQ2ISNtKBHCYSwRwmEsEcJhLBHCYSwRwmEsEcJhLBHCYSwRwmEsEcJiLX4AndNoF/6r+3UkMg0h2d/IS8YQZhOaFqPs6aclEEzvFtZXRbSg3dBXGk65uef7OLu/w2tdaP8Q/wI8MgH//eQR5N7UQigz+e3tYYWFBDnksNDu8PrFyvQPwP6NDE7APVwsO5Tg8dQL+51Rm3KBn29daP8Q/wDPu4wMwdIHr8oXy+Tvux8b4VBKw4FFBaAJuxlXHwRUP0jvJgmxfwOIf4DlfLQfCzln+A7YPO2QJTcDNuNoEsOLbF+zBNjBQ8RMw8xmwenop8vpft0ApAjcAvGUcCFXAvl31AhxzS6fnBO1kpv/iJ2D7Y9nPJxBef1AKxQiaB1z2X0nIAvbtqhdgdWli1CBrO9npt/gJGCwc9JC11O8i+hi6+Dv938etbGwQpDlXm9bByn/Rg2xfwOIvYHanecB/wCp46Rd12iNuZWODIM252hRwfafkINsXsPgLGMhMpNdE36EGNAFMxGsUhYc6ZRQaUEyoFtG+XXUCJl396mLWhQG99O1oq1/iN2AopY29/z9BE8BEfNwx4lCIYkIXMPDhqhMw6eqnj97+yPTKdjLTf4kMVYa5RACHuUQAh7lEAIe5RACHuUQAh7lEAIe5RACHufgDeO0OUrZu2UElWyljd2yijN2+jTJ6M3U0ddYttGOB8YaeXneasWl7AKfY4sNMytjtmymjt1GfsGV2rL4UGOAdjqNBQ/m1nvKZqI36UbjVRBmtaqCM1lHGGgOyQ+oM9NVSJnOXY3JHQE79lIjaHi31eCW1sllBGd1gpowOyA6d0459612REcDuEgFMJRHAEcAuiQCOAHaTCOAI4BvaEQF8Q4kAbt6eCGA3iQCOAHaT0AHc8NGoYTs5U9DHjBHDhyTnDhgyVi3tGhXFBmDkzzA2ediwo+jbCGB3CR3AJxYBoCYAy3pXA112jQ6s2iQl5vjUTxxsB/qhjl8/AthdQgdwzAgeAATgQ8sd0eu2Sh+OmmMCu44vTgVxk6a8L0SxEcDuEjqAbet6P0UjAK/6h4ytflFp0YKFm8A4LXs2OPi8MeU9FB0B7C6hAxhK4fPud7BuEDFlOv/j6h5jxjxjO/8FAC+giAhgdwkdwHgDqH+JrINfqQPGPOuk8zoAT7jjj+2nAFjKqBloZ49Fyi7Ah6Kiou6NIdtkojeHDBGi5LRZANA7KZ06wQG8xUCKWmmgErWeKlYvo1TWaSijFVrKaBVlrIbaDhW1HVLnmf0AfAR3KMt0AdijhNHX3hr55qXCrsOHLzAwR4wckXDkvuFDt1ztN2R89UChwZD4tWFz1NBcpKx1/12SBmklL1UapNe1au0/vxRONhiuzDAYJr9c41SQUfrkww6V2hHYGegdbCdFr7FTic5GFWuVUSpbjJTRKhN11pSxhoDskDoDfgA+6jRabgnAHi11NHVsg8IVVvfm2w8uI4Kmhr0n2FPs9sRZ9tjtbzTqBGSHTu8I7I0U0T6k9Qc62PNsnrHudfBnhxrbZJmDn+VyHhw6tPcs+yTTG0EuoiOAm7enOcCaV3/wjHUDHO3WJmswX55K1sExO0EEcCB2tPNQZf2zyzxiXYDxl9BPRLbJTA3m5M9IwMuixtw30akTAXxjO9p7LLryoe3usS7APz4Om9GnQMbI0SOTmUOjRvGdrejIHRxSgEFJlzNusZF+sLuEBWCQcX+8K9YLsHUrOdczAthNQg0wuBilYCkcrWkvwDP6kZ5HALtJyAEGKiEudNSrnoCX9KolAxHAbhJ6gOU4jjuy9gC86+EqRygC2E1CD7C2XMxzGO0O+NR/Cp3BCGA3CT3AQK90fnIDTL//emM4AthNQhCwSxyAbSq54oHLrugIYDcJB8ByoYSLu0VHALtJOABm4bhQ7RYdAewm4QBYLsLLDW7REcBuEg6ALQqZhxcRwG4SUoDNUVGdh0dtnQE7RX9yuo8ZsEfhmChNzswhg+0B2L5r+XLHbxPugF2uts0d3McIaCTgKaChl4KcKA3AlD5KZ7A9AOdvBSWbyWC4A3a52vaAVS8ryInS4MqON5SOIGgPwFcuAM2XZDDcAbtcbWvA3cf03KAgJ0qTM3OIIJLgA2atsKe9D48HFyzYoCFFKddQiZIyVi2jjlZRRsupoxWUsaqA7JA6U/kE7HAVX7Dgd6FDWaYOwB45dXRj7Mv1msRPNJo9y/Pe0Sj6FZc/HBXVS3V8s6afmAwS1lH/AAHZoXD+An/51ciKXnEYzS5Ij4nZbCRFozRSidpAFWuQUSrrtZTRCupoFWVsYHZInWf23cgiXVXFxKyvdijL9AHYo1I3r9xbaZS8ojV+falostE4OHXTUaNxUfyS4aPuG08GkY5WR5lHy+zY5WcrOv0QeQz3Iho0utpmRbR93+CRPwJYRA/9UTGoDoAUtGr8G8rGYDsU0dYlSxyTDcIesMvVtgFst9hk5VwV+SEyZcddwqAfDNlh3HwujnPIKR0RwO4SFoDlEpxXgOPsCOCmEh6AcVxUURYpoqkkHADXneHwWaYGiyM2AthdwgCw6tX5Zr3FFRsB7C6hD7hh1Eeeb59FALtLyAO2Th3h5XAEsLuEOmD7t/21wGp2N4wAbDJ62xokwJbcCwm1TRKGI2BqT1sb8KJetUDPKpO5ldIIsKKMJ/MyNiiA6+c9+PKowV3fuuiVMPwA+/K0lQHH9RAAgOF4lZvBELCVg+MVBk/loAB+fT8xDSzvm789E4YfYF+etjJgYz0gAAs8AdvYOM7z2l8gKICtTQKkhB9gX57eBGByLcNODGB/bAE5E6eg+5iBhbkD3hpWhLnlhopoNYeLLfFsXAerkWXiUSQMP8DAh6c3AZhcy7D/TJD++gJyJk7BFJA6vkYH1q5zZ0k0sqxm7auzPPIIEmDaY0+DjA+9E4YjYGpPbwIwuZbhh+NtczYuIGfiQMCql+A3W7a6Kzu7SfXPLnePDhLgfuKhADznnTAcAVN7ehOAybUMP1xHHxu7gJyJAwFffh+A6peV7sqN/WC+x9IOQQL8GhgKwPPeCcMRMLWnN9fIKnye8yH/tRWxC8jVC2EdPLkS6AZleyi7BjpKupx1RQcJ8KDaoeDqIO+E4QiY2tObAEyuZfghmMOLXUCuXliAVkmyTjrvqew2kpV8b0JjOEiAU1/qOrhLpnfCcARM7elNAKYPHj0wjoNq9dgF5EwcAvCp+6Oi/kJKarmSyNN9qPLCs43BYLWiZZcuSIG3bDOTolWZqUTbQBVrklEqm/SU0UoDZbSGMlZHbYeG2g6pM+DZiqb09GitQ1luCsAeNXW0R6ySI+TXoYBB7m5cY8hA/QMEZIdG6wjsDnAsOiwBU0rbAa6vEgqLUYAEnDJ8+KC/zKXdRw/ILe0+5o1r5sLJ5tJOmea4nyvfHDKY35istQHfQYi32+FYRFN72nYvn+nLcQFROBNFtPQlHNjT0ArxqePhH/6boORdwHlxKqD9YrWBowsak7X6wwaLRX9mtXfCcARM7WkLAWttVNGeyjq5itAiADsWqESvsqBFpnNJwB++zaX9AqP3nWxM1SaPC6d5R4QjYEKaeNoywJICFpVFzbw+uuofkD3mLQT48vuc7kPvvU4Cvv4lBJw35Fluo3ZbAMb7e8eEK+CmnrYIsKmwwjll8sbKbnfwi2gi/ORKSPnYchIwGHYQ3cGXpzZqtzbgLl26PPBQtHfCcARM7WmLABuLK3CupWl8M4ClL9cA+wvkLi3wj72fkgR8+aVf4O+T/FmjdmsDViqVFE3NcARM7WmLANtzsAo5RbzjBXCTp1lkPzht5JjRmxoBg+3rScD2l35hDh85it+o3bqAnS8xeScMP8C+PG1ZHayWGqhOTSirueUyj+K7Pedk3XHH/0o3yZenrd5NsrNwnI92vsNYCtI89MBf1/SCiEy6c5PQmXTXCFgmwSvJ05sVNqyyQmr2MjYC2E1CCLCziIacRTIjytSs0Ffi/FgO5nmGIAFWLJwQFRXlnTAcAVN72gYjWRaikaWqEBYUV2BmBNjIwxlMiccyaEEDPHlNn0vTfvVOGI6AqT1tw6FKdTW8iVVEHaxgX+Lh7QP4ZTAIgOneCcMRMLWnbbkQGqyGhSRgYNNyKjDPnnOQAPcFgxVgrHfCcARM7WlbAjayylC9S3aTzEavoa8gAZ4m3f3M0E+8E4YjYGpP23QpQxvRcm73d5NyE5vkGY6AAaWnwVur0luCBJg6YTgCps457AF3+fI6aCrhCJja0zYC3GTie1MJEmDZjteeWikkw9wlS86RIV+Aia2NcwcMmqDOHTBkgluzf9oUEUjqpATJw4Yd9TKiowB299TlapsAtsg4MleTud3r4NLP/4884/zG9998AJb1rga67Bqdfv2mGh3YsMkZb9OMgX/fe1GpH9rU844CGLg8dXO1TQArRLiYoKrFMD0BuKGpt8ECbLn03oNfEaGi5etXVsIj7fDhzY6Npb12AN+7xBG9bgP8u36Detr4MVXJQ2ae+LXHF4bzm/rWXJow+b0yr12q23AHcB87kfvYAdzlqdNV+eHDq1q2A7hUQXVqpzLO5/NhvlphShE/X6eVGapLSsTeCbTUP0Br7wA+t/uIo47LgPG1sXouPJ5evXqDlhSlQusuS3eSR+6LEq2W96LkwAJt9EzGaK1WMEarGS/tK9n9rDR+ktZTNCotlcjVlNFKyliVgjKaWlkjdQTUHoDdPHW6Wrt69RKRQ1mmCeAUopzibAq3nMr1eVh+nVYrLkpKKi5UqGWKgvK88xm1nspq6h8gIDuUzui/fQJurJVgJ2IdADPI1oGPItoxc0w3MAsQL2gsuwhqxmbOA0A6AcTstPVVnv8CgBc8S4gOU0S7eermasuK6JwKvIriQYZD2YAV1iPvWGJmZrHMblaYuZVpqQJW+7w+Wnf5cj0Z0vxs1zjecPRVBxNbG1snnbKTL2gcXw7ifsr8hQC8LGr0vRNrBtrZ3oNFHQWwm6durrYQMIuZiHvF2ZzKVjaOs1FuyipRQZ2NGIsuprFxTrs8TYp5ZNK4By+QYdqS34vIkM9WNNra+NT9w6L+Il7QMH/y7njcARj9nkqwY9TwEi8jOgpgd09drrYMcO3ZVIzrMTvEKmPLzKSyqRzHy5HTNrUcxaBGlqmIz/Oa5BMkwANh55X7LPCScOwHU3vawik7uQJJpcdXSiEulpMxNkwoxLwWYQE2vfecjiABJgZne3snDBRwLgf4BlxMrnLAmQI43aKi5nK6j30zEeS+PniCmkK77QBTe9rCbpKkvOhivXudKpfguAMwsKhU7g+O2rcfvPiEVrF+hXfCQAEfe4jfDGBylQME2DGnkP8mqKlucPWj3U2mzKM1AFN72kLAOhFbKHC/QA0cAVfnpWySy1B3m7yDG7znUQcJcCeHeCYMuIhe/mSNb8DkKgdugHPfREW050IHTpMp82gNwNSetnSgg4XjEo87s0Fr8lK2YRKcbbSQAx0FRYVeVVWIzcma+7LSN2BilQME+MGhQ1eRL24AldBzoQOnyZR5dLw5WWqekKtSqqzNKJu5OJ5fzFWhWZWcK8nJXs3uoADmOG2p8EwYOGD7Z8P0PgETqxy43cHHYH+6+q1sKu22AuzL0xYPVepVWpZYKGtG2Y6JKq+KcK4BdpOux19I8zpzUACPmhotNmtyVvWJ9UzYgla0eezbTRcKRAIBE6scuAG291Nax52l1G4rwL48vYmxaH0ljnPdU9lqlR7FsEVZA0tyrl4BWEVnzl/y+nmCU0QnvN+1022vrKwHntKSbpK2/1K1kqJLhACjVQ4cregvyRc3Tt03nFzowNtkyhO2QhHtw9ObWQiNi0swdwvkLBHbs/Nnl/HK5bCIlvNLYqu8ThC0Oth7QjaSFvWDVbkCMbup1R2lH0zt6c08TdJhcnff7CweLvQy1Ko2oEaWSVKj8s4jSIBrsrNrmiZsEWBrBo4LmjZrSMAl3kmCDpja01Z6XGg2wORYmecOwej9lXKWyawwsis5TR4wBgUwd2jnZ57uPKzcO2HLRrLSJHhZk7e7SMDHH/Qe1QgyYF+etg5gPYePNQCzGPPyUirAJbCIVqBREO88grPa7CELvPr2veGdsGWAq+UYhT4CTLs/zTs62KvN+vC0dQDLcFysbKqszmKyEWCliJ+Y6l1GBwXwE+ThUe+ErTwWnf2fa02igwzYl6etBJic4k4oG3XOcUorR8CgFcIi2lJy8khSYXssJ9yfvK5f907YuoDLux9qGh1kwL48bR3AJoyLprgjZXV5FeZAZ+HiYpYONbJ08fkFCcgYt8X+gwKYPbDzM890HsQFXtKqgCt7bqaIDjJgX562UiPLToCDyvb8CtdbZnJeWaUC3sFAn5+VfRkC1bFdi/0HqRUtSksTN03YeoANWt1zCyjig9+Kpvb0ZgCrMXmDt7I8Lh2rbDytUSIQswwKYCstyUe/Dlrs3/lliI1FA2rAqjJu/hnqXnPIvx9sKBeJMS9lC7eKmVCNzNLLZEZg40CiSgW8sY3q3Azc7r7YfzAB9/SO2G4lRae2UonOQhVrljWNK0q6TuNR5qE0UkZrKWP11HZoqe2QOgNNATfx9Gi9Q1neEIA9GhStKGDQsy2eyg2YSIgZYNAIA8Vma7FQxFbL4WdLVkpOYo2SVVbsPI/RRJm13ByAHVqdI7DHB+BMUrp6u73FsWRJYDtv18ma7FOdlpzJTKHUDvIO4L48PdLyHcBrY9OvJ5MWGGrwWgOhXF9YWItiFKzKyhKNUV2dL9XJjUadPCUjMyUDOtJ4mqDsAN7pKUJu8Xa7RUV0Q1plmfdwhiIuueTXJk/ZkQS5iPbl6U0U0Toup8wxNUeSyCxV2ghlGxljRbdy4yIsRhYvkZ5D81jsNihFdE+y2dHDO2GLACuLcBzzirPLOFjmPSIK7SAD9uXpTQC2YpU88pm/ObugIOE6R9GobJTLNdW4Y6DarDBW8XABO7ve4xRBATyPHGD60TthiwBXFzQFDOwNDaZvv6bQDjJgX57eTCvapneMXZgKMvMPYjjP2cq0sCSSDNgfJrM0V7OyGVyx98ys0GtFP5DFxZoORQOrqfreJtzDoRXtEru0JJcuxvnEs0irXVF4jS+JE+BVZGJzGf36MSZm1Hs+Vgs9wE8mWqnirSaw+O2m0aEP2OY6u12vl/Mq0Ui8TcbJ54iYdPFFMc4zIMsbzKz0vJxkM8Yv8xiODhLgW5A8MLHKM2GLAA89SKkMAWu6M5pEBx0wtactB6xiey4pbNDZoLJaiJcV4oJMuQpjw0JZUybgaCqSD8Qy5VU4znK3N0iAN29RqTYuPTLKM2GLAH9K2VomniZtf71JiqADpva0xYAbuLj3yyhIWSnGRRerKlCFi4xj4bi4riJj9ZzdmeVeLZQgAR6M/rwCXvVM2CLAi7+nVEaAzb2arOMbdMDUnrYYsImH42WNjGx6rZ1QNrGquCpt41sMcjEukPETV2xbe4ZVwPKouYME+DnYhRE8C/p6JmwR4D0UNS1wPPA/9YyZXB+AFhXVeXjUue7jomYRfWbRgKFDhPo37qEB4o/oDfiRg/Z4JD46VwyQPBgVxfbXDh+AqT1tMWC7rJLfODZrLYiLz7QAnVXOqteQbSm7Sqa2AzPGkpuq9y3+ZvqnuZjXbI/gAD7V7d3J3c7oFnsmbBHgq69SxZKA7f12O9YHAKCPkdi6YON3xLdoRwpr7RwaIP9o4Edi5wrio3PFAMmEAOzwAZja05bXwTad4/RWg0VXkF9AVwCdQkg8F0aiqsL5GrVMqrWYFfkfTx36/a5rno/8g9WKro6OqW6SsEWAS5qMAxJCzslKXep8u9gJ2NjTadxJACBM8o8OfuSQezzCj3GTpryPXuuVPBQ1h2LuXmCtaEpPb7KbZNOodawKtjy/oIAuBzo53jg3B9a+uIQnSMpnqxTlc6d/OuvgVc9dm4LWTdJQQGx2ERYAhv8MAPkOGaf7uDE/qcmiVP0vil6w26zKVf+QRwfghl7EF+SOFI2AU+FHDrnHI/x48HljynvwG20NWEjxJlOA3SQqT28SsEzAYwhwiUxyNT6vAeg0PFGZIwW6gyVCTh4Tr1FgWxZ+8u2BHM+hjiABLut/++1vNNlYt9lFWED9uMF2xztkaJLzlu/IotT2X4pZi26Ave/gx8HuCcgutCOF2x18eSqxcwXx0bliAGxk5X/cNOuAAFN7GiDgBhkxr84J2MLlpl6OE0rksJNEzOjQq5wVLaqDDcWZ8cWSGkXVzmmTP49OqPPMKjiAh+23WfeM8E7Y/BIOuw4uTnW8Q4YA23qSRamtsQC0uxvZCJhcH8BVB89EkY4dKZyATTr4kXgBgvhIrBhgqwNSDdjxR1P/AgJM7WmAgDFy4qgTsI3FLGDE5WMNbsqETQaZTGe2W/OKYmNLtQrxh+OHTPunlOWRVdBWmwX+by/rKGTH1bFmA/IdMoJFLxNRlDY+8LdgXLeVVV0T34n1AUjA3ceNmE3kTe5I8c6jr64l/jCHwI8kYCIOrRhQ0xdc6zvsbYq3bANcbZbK08AA29k4LlDI5bXOH9ZwkVFWqbC7lO21RZgJWNmS4mi2TM/DJWyjWXHpnff6vThtP+7xBmmQAPcpg9VpH++Ezd7B1T1GjnnGRr5DRtzBj5NFqU2mQctCAaAQ425D6x3lzQZqTwO8g+WCyoICMV7Q6J4B48psaD0sI6lcmZCafVGmLscOH0jlqVgSMWYzK86//9ajjw7+7BrLPbcgAb7cZeyYB696J2x2EZbtp/T2pQzyHTKiDp7pKEpruDiOlpKBbUleeuMU6UbAVr27S0EHTO1pgIBtNTmlp1kijKhOia02bAqFHo1qYcRAR11MdOJJmqQkn7l51+FYLDeDbwJmRdqn/Z94bdqHO8XuY5XBakXXxJxv2jZqdhGWQXV6e8oM8h0yWNaOngOVUFFqwytwnGtGb7tXxO4t4zoIOgE3YHyum5nBf9hA6WmgrWhMxExMLM1BXVobVsmTA2WVhCvn41WF0B+dpTQ5+vB2RmGaJPGvjT9uOphelScxmmuu/TzkyV7vfH5ZwHErpIMC2Dn1g8x6/qJfHa3Mlk26q8cEVTJihWQlNvUttqPT5ASsFOG427S8dt03yeVqoIBlFRmZeTnVxOujfBzn2IjNN7CSf6IvGoHOzC26vHbT6V1HYk8f3b3lyLkDjGwuyyi++NPHw4Z+uu1auXs7Ivj7JtnsoGQtGd/CWZVWjWNXTrusfMgIhwNOwNXFXFH7AfbcN8nlaqCAjUVxmIilRj+skYeeD8lFOF9nzk/NzRRBZTz+wnHG1b/Wn1mYcX3n5aNrD8ULeVrpya+nTPloe6nSY6yyXZ4HZ5+Bf8qys7c69h1u8QbRJoXw1W8dQb1ZXlSkVpfSaClal0ZrbBDtww6/NohGruqzs7cHukF0Qz1yhoiuKylWmlXCImmDOSYnJ5UPlWsrCpiX6Bs2HF5xIHbrli1nT+xJzNKJMpePfufj5Ucyit33Ag/eDuCHnAHxoi/QSgcbp09fryRFLlNSiZwyVumpXPH0AuKoUNTlcDhZVRgby3RXVjQGawWSxg/UWSsCsUMhdX7tDbjRU4erwunT51Q2tefGp5ChaKlcrlBW5RXwYUr+mUsna6ByFQtLuxJ3aOeBC2vzdu7dsXXTITpdWp6/b+74WdvW7M/McPNDQX1CKWWsDzvkzugdzQF+yhXkzyGPrTLxXfToNnSwmgyo8aUuYZe6D8a6imgjV1zZuB5Lm87ocPPU6WrLnybJuWwtev0I56GK3ai0wCa2XJ64k55azku5RmNWxJ/cvGHXpuSkamHW/NkzFy75bX96jluxHLwi2uk2PGHdb2SwVQBbch/IAwiwDRNWyQxZFy6mufnkAqyCra/GQZ7gAHa52mLA6Kpl2a0ciaBE5TBDgQmS6fRt285kZBTLpKXxh6N3rl6x9UyhPH/5+oWf/PrTqkOJdmA2OrSDB3iP44gtWrLwplrRnoC1HF4SalXARpZFrbEpS/MK4viaxpQuwDo+Lm58PRo50jT3VgLs9NTN1RYDFp27wmXbQF3s2bNcRx4F11Lj4i8dWJ9wiQcbltqs3fMWLlt2FLadmcuWzZ+5f/Hyy5k2DZfnmGwZepPuvAA738Ox1mFy6Is2Pz89trSqsaMAAcvF1Wq9gqVQsuSNI5s6YJax5N6rwnW8SXf62LMXzsITyXjMDFZZA7BDk3kpBYc3HFi258IVtk6aE7di3W+/r9225YJYduKzb7/7cO7PKy9l2lyvn4UyYOJRIpbS6bKG9gun65uDDiW+OWQC7+wLdx7F07o65maoGmTxqX+dYBTjle6vQ+iIWS7eL0h0PMBSZl7GZRMwXb9w4PjVQrOOxVHYUpnRu6JPXU4+l6fOo8fOm/bZD3P2rbnAVnJnj4t6edjEkUuZOVYImF+rQddzSANGjxJ1V54ebYWA385n5u3NqF+/ycT67LS4/B2HjqqhMufqPwfPXsYl7stX6Ih5at4PFjocYG0tMyMzzQ6wvCMb11zJRK8Ncgsu/LFk7tHreTqrInXb2r2fjvh40Nw9VyR6c+onk/t1HfL2yPmpLJuOXVbOF6A3XkIccO6bAHv/bS4EPPFa3t7TCWeXbwXg+3/k+MNRPxELAKoaqhP3rtnJPCspdzdeh2aasr1nDnQ4wOIqLLHKBOwsZvIZegnLxuLmHtl3+PC8vetWci3YuS0zP/1w8JT3x8XEXq4D5rzpY154vPdr45ecrEcj8hx4ExtCHDB6lNgQO+7Kl6iIHvTW6sqtAUMAACAASURBVA0X9z9Rpgc/ni5Kz1F/81upCQG20qI3HzlXJfewXUfMFffOuqMBVhaSkICi8PqhhAINqLx6YMXyY/tWbz18vlRRsezDsROmvDtz1vyft6xmWM0lox+5/6Fe3d5fe5KYL49Wv2sIccDoUaIseVLV4IO/cN4ppG2dvfGxzeVlhm9WXjpRyt0zil5NNLIK+ZWl3ssfhsabDZo8ByS1uAQTQVKlifGntqyavf8QjcuRF879aNSo72d99MX3X61Yvldjjhv1xH/u7vnWewszYclkVMsxYlZIiAO298tKebtk0/PzOO+JMdrSXt9cFIrlH224evoc8+dvc/kIsKIwLq/UCiwa9+xCA7BWhqHHobqshNNsnA2bTDJ6xsZjV9Prstg8uSVl8TdTo+bOnbPqt+ET3l5d3LDllW6339H55V/+KLTa9VxhOelMiAMG25cnjKFXPTub8155wslf7nqp36qKid16jv/4sT6vXKFVmoBKz8WFedVaMybgu7WpQgJwg7ia6NcJsrJO0QRlUq3NIrp4Jl9QbLTpDQ1a6coVP83ZMm/c96P6R/Wbc069dtTd/77t9m7vHWpQs9LLnfMuQxiw47VKg1CI47IGKZt++O+kS2VFddrSzK1/MumX8i5yyjQQcMW19acvZMcxK9iu/EIBsJXF5WaUyixAkJ2yI4aRzK/C7MCqZOUxMC0w5pQWXNt24uiOJSMHDJ4wePiIj1POLLjvtltve+Dx9RouzqWJReT1HLqArRiffPCpLReyc8vS8oWFsdnndx7alZNTeXBNfGZ6WRFPhKka6tb+NOO7xasvp1wpupEjHQuwoaK8mAFrHKBKPXAxOS0Tx3mwQa2uyy6X5BSl7f/nyM6ti36ZP3v8kLcHDuk/7PutGTMeuPXWWzs/PHNXWiEvu0BM5hm6gHWVOM4mxqL0qlqhiLb4YJaoOOv4vI3nGKf+nrf6GCP+AJNeq1LkLl40//3vFi87e9Bty6pQAGxhlxfQJThLw8k8S0tjwiDHAlRVgnjmhZjcjXNWb1j00Q+/TRr91MA+Lzz+VN9v1sUPvBsCvrVL36VLtp1jY46FLkIXMBqGd66jrBaWHlhzMF5fenTWR38ujVm+ZO7X81PPHbqYKFMVFCz79puBs5auScY4rjmYoQAYVj4VJcIKJYuflrT/EJadD0tmwMIl+3bu/vWfb6bPem/+Bx9PfvOzwb379n6ia88Rv6978t8I8L+7Dv1t+/F9CSwR8ZZh6AK21HIbpw9asJTV2w5slP4z692pE1ce++urFb8sSzx3KY+nV2WnXpg1cdz7y1dgIiy06mA0VGlU6azsspy0RBojoRjNWVEJ8DPnD+xd8/13n8354f2+r/Z/p9/LvZ58+L8vjj/zfjfE99Y7uvQZunDFudjjSUJgkhED9BTS8QHr2GV1rjvSxp73++pDjJgts2bOnPfnnOnz5/yUHHe8SGFXCTf9/uXXkz/8cWtGRVUx5kQRKoBh7kDJSmCUJjDKiLFWo6S6kHbu6N9r582ds+y9F1595cnne3S789/3PDz66PjOiO9t/37kmd5fr4qOO5FYYMckOEZtXMcHjOG426Zg+nO/Lf123e/bV7//9bpvFiz+ZMic/TQeckJFnzW5/5svDfl62aH8q0wJ19ycIx0PsEXGllka6gUSAYvNqoa/VH7hpYxY5oULK7/84MvFY1945dVHn7jv7tv//Z8ek/+aRJTQt9754FOjZ61KSS0WY1YOj3ml6ZtwAdvRHoDtsDIqd22YIzh1cOWUL2b/svjnU6cXzf3yvYnffvzZyuv1AIhXfv/Os88+/eL4OQcYJ+hCnrE5RzoeYIUIFymsmLAoM/pQUlEDqEiLP1PIrtekb925YdQbLz3U6+HH770DUr37qcl/vHMnAfieHo+8Oe03VkZRiQbU0nNpBZRn7PCAgZqdd6Wx+DGU7t265KONa35afoK5a85nY5999enHXp6x5ZoeVP7W94l7u73Q95tlm+ISL1SWarRW3450PMDEi6L6SnFyCi21TKCQndx4eCe9UMTK3PDrb++Nf7XXYz27EfftnQ8v3vMmwffWu7s/0/erWafppSwt0BdyytmU80o7PmCg4EgaW9HCsxt/+vKjz7/86Mel26aNjnrhsf90fezpx99fmm7kfPTE7bfedu9r01cciMsSa+r5VcSqcSEB2C6V67gCjt7EraIzGSncslrhuWN7/z5y4tx5bNtXP7w7aGi3Bx964HZU8d713GfLXiYBd+755Ksf/rD3cnJVncWGCSuKKK0PDuDNOlJUCh2VqLRUsVoZeaxj8Xh5pIacsW77J5OmvjFm5aG5sz/t3fs/93a+u3O3/7w867goc8J90OnbHxu+9vzVOp0sl8crrYcplJQnVFPbQa2slToCGj8AHxY7lGWUPlGfQpxfIJLXKXSq5JhjV2MOnUrhXz2zctfh5QvOHV8595s/p7/1dOfO991FYr137K8vkqH7nuj72uhZC2KupydkCpU1EmrrA7FDp1Q5Ar53AKeWm7qDbRiPS6aza3JOHP706/lffjz74HfLP3396e6d77n/3geff3nECnHSM3fdc88ddz8x+YPTuFFScAUTsrVWu0VNmXXHuoPN3PIKYoNgpQjn0a9cvFZelXB1887vl/yx7Nz2Xetmz36v54Ndb7vzzts6P/jwv+9++tNet916+wNP//fe7k9O+XH7ruT4/cyMOG379oPXSCQS1Oqp50okxAO9GhhBvK4Bj9Xoh62DAfTzGeGR2JRPKhGVEk/BJIKyMlS7yCUCnijz0OnNP3wx5+eVO7Ye+XJ472ceHfT+xyMmDRuyVsiY3P/lJ57v/9Hs7w+LUxIZ184m44pKDh+rQplVw1zR41YdPKL3iYwVMIB8NDhPh86PHkOZ4FFCnl8iQW8MmSWiEomNOL/ID8D/YBIJ8SYZW+jykHiFqZbwUOfysB4GCA8FKanpHOiijJ2bn33pXMJpeiHjRPq5n7f9MHv7xhP/TFn0+8DeLz795rjxUSNGDH1txNQpT7/00sDXh0UN+Gzm8TMxFwvP5RSkFAn4AqeHjb8w8hAXol/Y6SE8kh7qSA/VhIdQSA/Leag5a5FINgUI+MeYmBjkJXY8Jga9mg2uwQg0oCiCx6sIcBIMoHmJNfBIbKudEhN9GE1+lZ89dfqU0aZVp8acOJ0hKU+O/eX7j388kZubvHzKay++ETVrx4avPh7/O7tk/uLRr49674uVS66xN23aviW5WizNPHD6SBy6K2Nhrmj10Cp4jEeAL8IAH7kPj5fR6ZgwwIV21MNjDEqSDo8l8KiKiT4Ug26v3JhoPwBvPRoTk4QCp8/FxKBFUwUwH2ILERoMVMAfthoeL6GIZBhALwbIYo5t2IRKq4wz23aezEo7c3bd2UMLr2Qv27Xgqy1rl55a9c+6YcMGj125aMWqqSOmvvv92t9n/jBs6oRxX3+/csuRdWv2xl1nZsQeP330DLps+DDTBJR7XExMNM9iA6XQDmL7PTr8pgyBRh4iwKnwWAojlCgC3ULZMSdP5qNrMibm57WBAb6ZItpsMpnMWoVKZtbq1DK5ulpSkXUtOzMzPe3K8i+nr9sVe+z4keh0UfG1E/8cOXT5+LHLWcWlpXl5YrEYF4l0Ch1V3h2riIZXkcMeq9EsTU+VWC2aiuT0wnPnE8p4JcUpZ3adPLX74NHjx48cPPT3/sOMv7dsWv/n8gOblq1NzMnl1FRX5+WJTCZ1kxVNdEaj1t7xW9EWvUlvUas1KrVVpdRIa2oreOzCvLSMzPSs9NgkWlJSUuI1+nUsK4kWn5AYH0+7nsvBa+uURm1NbbVKVe89Y5aQjgYYjWSZVSbnapU2rbS6Ul7NLcKKiooKs5nJTCYj8Vp8fFx8QlIiLZceGxt37cq1uLjE6+lZfAmukNYrTSaV94vxdr3JpLd1fMDAarbaTQq1QgssWo1cLWSLsaL01OTrKdevM+nMJFpCIpOenMlMoifRYTiJkVkkhBezHbYJ1XprSHSTEGCrVK2o1pK75FjlOF/IY/OKivKyMtIzryczmPA/Gi2eCS9negbzWhKNQUtMoNNTcvL4fKVUWi9r0Bm8G1l2ncmkC4E7GIrd0qDU6KzwaNLIarmVpUUF6enM6/C6hhd0Io2WQGcmM2h0FGQwGFnF4hrorM1sbLCGRj+YWKtSoa7DNVI1Smaur62oquIU52SlZ2WnMeGVy4SOJSYmwgKLxkxhoGIrPjGJTk9OyeLUaKokNdV2ila0zWy2dvyBDvSVxQhLaTu8JA06ZX2NiMsuzM/JSb3OYCJPk6DjtCT4KyDS8AJnXM8V1cuNtuYc6YCALVJ1TTVeo9XagdWgqZMIpLyi9MyMjPzkFAaDnsREN21CEo2elMSgJ8YzExMTaAwmPSUbE6mEFcIak9UUmk+TrFqdGWhhO8tCcDZpVTUCiagyPzMzK4UO71roOLyqkxKYiTQ6HYJOoDOYKWU1cq2jcRUygO0mtbSutk6rUuv1MrwerxLVl+cWF2QWpjPoDHiz0pIYcfEMWEoxGEmwJoaOoos5OSOXLS0XCHGNXh+agLWooWCGd7CVAGw02zRShYJbXFScCQGjIiuBRotLjE+kwwIrAcFmMNMKK3GZNsQAo90KNHJZtdooV6lkdRJZlaQO4xSWFKfDiohOo8ejmzcJOQyLZ1oiHRZXsNhmpuYVl5VgZZUygzI0AcOGgt5qt5hhCQ3sZp3OBswKTS2ntCAzhUln0Ek/E2lMejy8f+mwkopnMrO5vLoafcgBBrCLIK5T6+T1imqxSFYnNQnLi4pymUmMJEYCqncTYE0E/4O3L6yQExJhJIN5Pa8Uq+SX1ahRSUclHR2wRa939WTtdhi06pQKXn5+OjM5Gd3CiQxUO6HGFvwFEuB1zmBml4sllUpyeY8QAmzTa1RSaU11bS0ulwpq5TK9VCEoZF6HfBFVWDonJjJpCHICqo9psE5iXM/KL+NVVwukMmOI1sHA5r1xg1VTzcvNy01LS4FlFy0hgR6HWtLQ30QmbFkmJDBSs3n88poarcW3Ix0SMNBq1VJVrUIkqlYZ1fJaXCvFq0vTYSc4Cd6tzARafFIc8hL+g+1JOoTMYOYUcRQKiahWYw7hOVleYlaIBQUFuZkZqcmwiIZlNCyXE+JRUxoW11DoKYUSXCIj36kMKcC2BpVaqpDjdbJqrVpVW10vrZOUpcHeIJ0O21UM1EmCzej4xISERKK4SmCmFpfhOqOK6D2HCWCbWSWtryzBsJzCokwmrIvQ2AYsqGE7k05nIt6MDHZVrbhaRi6DH0qAAerx16nxOnWt2axV1tXVqeorsKysvDQmg5YYT6PFw45SImxuJcEaGDajaYxstkipUmss4QPYrjPq62vwKrEELy4uTE1CwzvQYXjfwu4SLLdgOyQ5q7QKdifNodfIIgS2JOtVJmBTqeVKmV1VV8/DCkuyiCEsePtCdxNQSxpKfBKTkSHgGS0GkyHUXwB3iV1n0qv0Kp3RpBLyS3OSmTQmKq/QMA/qKsLamJGSWSxSNzobcoDReVDz0KJS69TAWidX14pLCuE9DCshOi0BdRsYRE8JCiO7utrWgB7GgHABDMwGncJu0ZmtOoVSBJvSDNj2QDUvjRjIgj8CI6OgUmtqJBGKgEmx2+w61GfQWc1SSWkqbEsnobFYJmxL0mhE/5+RUlwp12u0Jn073cGXlv7hmMrZio0si5FcwE+n08ghYToqvBITkOs0NFbLyOZWSrUGt0VYqKTVATe62nqAXcoWFS7iFjJgSyuRiYZlE9DIByypk/O5UpXOYDRb26cOls23l68hg63ZinasVWm3WXV6RWEW7Cwl0hNQP4mOGlxZZTV6rcnlVHAAu1xtC8B2q0av4ZYyriUmoy4wvI4TUGuDnlPKU6p1zv5v8AGnHQFgBjzKJJJtjo2lfewA7mPnbYodwKE0GBoVjGpVRX4a7C0lJSbGw25EIj2drTGb3Pe81lDm0Wo7gHu6apZI9t1gB3BqezQ+NgZ3BS3qylzYI0bPvFHHEPYcGKXlwmpVrUbtcDewHcCp7bjhDuAeEn8OgG/h8ffhw9fKSJFKZVRCHesr2kNFLKhg58F2ZRxqYDJLBbVeiQLK+gbK9T4BO1zlDR8+o6JFJ/bDnnphUW56IjHIAdtYzKwqvEpSU+cjpd8++ZBtgdzBoE2KaIfY7TaLQVrKpMO2ZRzFYv/BKaJdrrZFEe0Ui8mmExLPGxLTcYO3cvCLaPl8e9nNrRdNqUy9ZwMtgcmgig8OYJerbQmYNCI2KT6Rys72aEUvWdL6regOsymHhzS62uaA0VgtZXSY9IM7KOBGiQCmkgjgCGCXRABHALtJBHBQAA/oGxbyJvW15S7H2tvGVpIdLpf8AaxRk3Jgodp/qRwQgLL6y2sBKB/5LQBlYX9nqEnfs6k0OHWjsABOsfKvAJRTpwSgrB5dHIDy2u3OkNsLIf4AdsqJpQEoSwcEoAy+SQ5A+cziG+s0iqJ/IHY4ZaQgAOX1+wJQLng/EDvG8ANQ3ryHIjIQwDWBnK0hJwBlwKKumailtiIAZXN2IHY4JZ9yd2sfIsADUNYUB2JHgR/FTqMIxRSRgQCOSAiK34ArFixaVOf+lLg5sc7/EL1m659yIIoBZRyQyS1M1maetparfgNWGUHqTvenxM2JXX46CfirHIBiYBkHZHILk7WZp63laiBFdNoe92dMzQsyzm9lvxUDzTggk1uYrM08bR1XAwCsm1PT+JTYL+P8VvZbMdCMAzK5hcnazNPWcdU/wFf/PATMS3L8u6CQcse4g/022SUdyNPWcdXvO9i+Hq2/4vaU+MbG+a3st2JgGQdocguTtZmnreOq34DTPliy5Ij7U+JmZd23P+z1WzkAxYAyDszkFiZrM09bydVIPzjMJQI4zCUCOMwlAjjMJQI4zKUtAd/h9UhG/m6PHhMlAHTp3qPH115Kt8Cu3B9P9Hx6gTkUr7kO7GkwAUtjzLYf34NulzZVgm5/Mq4e6NYowwFwB/K0jQFffeXlETwAzjzXb+stRNzFfk63HV9BpZjnBq24BZT/u95h0oY+T16Ev0Lf3pMV8NO6j/tdbMwhfchr/eltaHILpQN72raAax4sA3tfB9XdRWAL4bZt1Fro9sM9e152fEUo8cG6W8CZF50m7QH5j8G7AIA1C+Gni6DyMWcO8pdrgOixG8+tCrZ0YE/bFvD5MQBYb9fETAZARbj9w2Sr47p2fAWVYt6GldYt4PQLTpNgUXanGewePKjPBPipAYC7zI4cLnUdOnTo07w2tLll0oE9bVvAMWMJ36Lfdbg9+200PZZw2/EVUpoCgAYWXHc4Cy74725j7jMKcCnK+cmRw4XBbWjuTUgH9rSti2gO2PMGqO4mApuh27MmEdOfCbcdXyGlx5TgOPz2o3G1QLdK6XA0YSgA0xrdduQg7RIPQGYbmtxC6cCetnEj60qf54fDcubUMwPXPgCKO8FeQ2PTw/EVVIoe9Nmf0O2GRT179FpidjhqnTZ22i+NbjtyABlvPf90QNMSgyMd2NPg9El0AByZ0M45BEc6nKfBAby47ytRgcy5bYscgiMdztMQHFWISCASARzmEgEc5hIBHOYSARzmEgEc5hIBHOYSARzmEgEc5hIBHOYSGOChW9vIjIi0lfgDeNzHxOGLKLA2oW2taXfZ/AT8E93pAPz70oJODhHv/r/t6MsJnTrd1e9q+xoYsPgD+NRdaGEqw71HGmM63qSZVpLcTmIA5jzyBQCKfyUVFha+PQz+MQ/4tS/6csKHnNwvbue2t42BiT+Ajfcdhn9P3a0liui+y8Z1/qvvhnfu7F16/aXOn0LWh/rc9egC6oW8Qk2s95wE4LUtTwFw+Ta0rNjn78A/vC6m/6L96iegVcPv+audTQxQ/KqDvx4J/0z4lKyD+/6Hbtf27XqQNemVQdfT/7sbgH2JQuaTYVI7j/4eaO7U3lMN5hPLQBGA//gWzF4ASMCgy+Z2tS9g8Qtwyv9JQP2tNAfgH2BM3+8ByOyUBMC8aaTKgQ46WypQWfUiiH8TjDoD3kRICcD2J5JA6iM2ArBxQ6es9jYxMPELsP2JDWB7D5sDMCqj+u4CoLJTPQAbRgHA+uC5/97/eBsbGiRJ+Zds8a9g+WzDbURrCgFO6WYF9kfg5T3hX7f8665N7W1hgOJfN+mP3qA/uqIJwPsB+aeqkxKAjVHA3GN2Ef5PjzY1M2hiuuPikEuA3ofxfyr0EQH+utMtt9zyL1hBTfiglG+5UQYdTfwDzOt0rhNqZlAC5qLpvcvDBDAYPPtOGdDf8fOrxCcI2HjfsdLS0ui7dWQdHGri50DHgPuJjgIlYPWdV0Dxf8MF8KLOz8O//TvPIT5BwKe6oCV97Q8fCWvAezoRXX1KwODsE48M3hQugOM7oYWIfu4UQ3yCgMd/ToRmRIU14IiEqkQAh7lEAIe5RACHuUQAh7lEAIe5RACHuUQAh7n4AxiXkCIWSahETBkrEVJHU2uLqKN9xLbQDtsNPV29wCG/LQhAAlJuy6wbxW31Fn8A/xRDyukTMVRyKpoqNvoopfK5M5TRx6mjT1LGBmbHEWfAj53PXj/mUD56LgB7Tp4KQPnsMcroM2cpo1tmx8+Rre18yOv/m1vbRQA3b08EsJtEAEcAu0kEcATwDe1oO8Cc7uPG/KQGnG6j0aFoLAAjOaIBQ4cIyWAEsJuEJuApAGz5DnDe0aMD+CH6xK/AagNHF5DBCGA3CVnAtp4EYHgAir79CMP2nSSDEcBuErKAQS8TAgwPAIxDMz7yhjzLJYMRwG4SsoBtj5N38OMA0KYPJta3ujyVDEYAu0nIAt4yk6yDZwLzGzVZk9FbW8mfkUGHsl3BkrnRiwB2lw4OuPu40XM0qBWNDhs3AfBFHHP4yFF8MuhQ1lbiYrkrUQSwu3RswE5pvh+sEuE4yxUbAewu4QDYxBHxla7YCGB3CQfAwKTSuVnWEQFv1pGiUuioRKWlitXKKJW1aspoOXW0kjJWTW0HtbJW6gho/AEsdijLKH3ycQoFdTR1rIb6d1FrKKMDskOpcgT+jtzBPiQyFk0lEcARwC753wRslsucJ48AdpdwAYyJca5jc7QIYHcJE8B2No4LHGePAHaXMAEMZEIJx+FhBLC7hAtgq0ru3L4yAthdwgWwm5gVGSOGD0l2zAaRdo2KYqNZIBURwC4JbcAVvauBLtsxG0SK1uFCs0B+jQB2SWgD/ns5Ojhmg0gfjpqDfpF9RyOAXRLagBf/gw6O2SAWLVi4Cc0CKY0AdkloA3a7g4nF4/LRWr6X34sAdkloA654pQ4Y8xyzQaBJO/5As0CmRwC7JLQBKzJGjh6Z7JgNkvT6sLcVaBYINwLYJSEOmDI60g92kwjgCGA3OyKAbygRwM3bE5qA7buWL3f8NuEO2OXq/xLg/K2gxLF6fbgDdrn6vwT4ygWg+RIeZRLJNispOrWVSrQWqlizjFK5wUAZrTRSRmsoY/UB2SF1BnwCdrhqlkher3coyxsCsEejDUDZJKeMNpooo+XmALLW6hyBPf4AZq2wp70Pj78PH75WRopUKqMS6lhf0QEpBxR9A+V6n4AdrvKGD+9X0aIT39geYVp5tYRXUes760BOeKM8tvnVyIpecXgWGQr3ItrlalsV0Q1cXFLEkoiwDlREQ0k/RB7DHnCjq20FWF+J44U8HOdYOw5g65IlWx2zEMIdsMvVgAATD+71b9xD0+q+HjdF7fpi2hQRoHdSctFqDmSMhSUSVnNwcUkHAuwm4Q7YJYEAlhEP7q21c2ja6O/BkZWNJ9SMgX+n9FGqidUcSGlQqq0GuWLjrAhgD+nIgA8tJ48Q8Ia/QeEo8/S3x9dkDvvx3O+PfAOu7HhDqSNWc/AQ7D5DBLC7dGTAq/4hjxDwtQ/A1r4n/gBXfsocB4B0ArBPMkHA5GoOHtL3dASwu3RkwG53sH752/MmLLsIasZmziMAx+wExB18eapXBjvGRQC7S0cGLCMe3JOAAdj6z/HlIO6nzF8IwMuixtw3UY5Wc/DKQH5nVQSwm3RkwIB4cA/eefTV5fUjR/1kNX/y7njcARjKG8o4tJqDdw5TVkcAu0mHBtwogYxFX366XQFbci8k1DZJGI6AqT1te8CW/yahgwHDPL8OCuD6eQ++PGpw17cueiUMP8C+PA3C06S5XwPOFFvxoEF3DevRacyA/Y7ooAB+fT8xHJP3zd+eCcMPsC9PgwC4sBcEbC7D8eeUnCmg4WlHdFAAW5sESAk/wL48bV3ADXqKXMx1ELAdEwqf1UPAqt6O6GA1skw8ioThBxj48LRVAddyKzESm9Xoyg72gyFYq1rV28jpPqanY05FsADTHnsaZHzonTAcAVN72qqAc3FcSGwjbmJVsJxvjzoAQ+ljhEfLGyIyOkiA+4mHAvCcd8JwBEztaasCFklwAVHVyyW4pHExQ0/AYFgJGR0kwK+BoQA8750wHAFTe9qqgGUYV0Zs1aQQewHuFhW1GAHuPmboHEd0kAAPqh0Krg7yThiOgKk9bd1Glt1CBsxYGdboX/uORae+1HVwl0zvhOEImNrTtlqEpQHdyRaCdzs/bJBduiAF3hKOgKk9bct+sILLQU2ujvg0KbJWZfOn8GutSkURj1egau+1Ku8gxNvtcLyDqT1twzvYVI7j6B3R9r2DLRaL/sxq74ThCJja0zYEbJfxylHuHaCInuYdEY6ACWniaesANpvsFMp2I+F6+wPG+3vHhCvgpp62CmA1txyz+VRuX8BdunR54KFo74ThCJja01YAbJTniPFKnU/l9gWsVCopmprhCJja05sHbOKIEllugK3eRrUnYKNDvBOGH2Bfnt48YI0QL4stkzmLaAWH5ZWmPQHfccf/SjfJl6c3D9jAw8VFDc5GFvyEszzNav9GVlMJP8C+HLt0oAAAIABJREFUpBXqYDUmNzcqo5fOODYP5QhgdwlBwB7KNqyS5wW0fQErFk6IioryThiOgKk9bfWBDpveu55vX8CT1/S5NO1X74ThCJja07Bfo+NlMAiA6d4JwxEwtadhD7gvGKwAY70ThiNgak/DHvA06e5nhn7inTAcAVN72mb7Jrkm0CLAdo3cO6fgtaJzE5vkGY6AAaWngQNG0zRuCFjH5TfO2UGAlZXiYpmntUECTJ0wHAFT5xwwYAWXq7oxYAzHRWiHWY3KRgBm4VhagSfhYD1s+PI6RcJwBEztaaCAG8pwnGu5IWCZhJghXZmYVG5DgBX81Izyco9fIkiAZTtee2qlY2UY7pIl58hQOAJ299TlaqCAiWkaNwZswrgyKzDFFRTQlQiwtT6WK/FcAjx4dXDp5/9HnnG+wRnVPOCRP9oBp/vYNxPRhG7HdkC5A4aMr6I0oqMABi5P3VwNFDCapqHwo5FFTKBtuFpQQFORrWg1m63yUA4WYMul9x78iggVLV+/spKMbBZw/cS3bAgt/00CMLkdUI0OrF9BaUSHAezy1M3VgOtgYpqGT8ANFo8I8bU4vt3RTbJ5jlAHC/Dc7iOOOrQYXxur54Ibr1W5cc/PV2SZb8uS+qG/8H9Z/aMofuUqf9djnH5ZtusVmez514bcPWhIqmzoTJkss+uI15MDXaoxoLUq3Tx1utraa1XW87KzKl0fcR5fJPaddSAnbPlalY21EuxErANghu3Gq82OVeX9YMW6D7k32Yq9h/63WnvprVbxS3zKFVEpVpvdv9o6c5hC/YrV2ltntdZMGGRB2aSM87mqamusNuvmqdPV1l5ttp4rEmENVoOesEudwaqos7b7arN1ly/XkyHNz3aNH4uRVvcYM/IZGyyZjy13FtFoOyDdoEy/G1nsd8DkNXTGDPQ+FgC7ji9ORdmoXgJtWkS7PHVztXXfTeLBFphVyS2XqmVqU0p8OsaxtvdIVswjk8Y9eIEM05b8XkSGmgO8/RTQL2VAIPZ+SmcdPBNYJ533vxVtf17/ScKaNYdJwOO07Nkom8vv+3SkNQC7e+pytVUBa2XlXKWVg+PFbLyKTY+OTioG7Q14IKTCfdY7YXOAB9UBffIMxHX7erIVjbYDOnV/1Ih1lEY0AQwdG3t4S+37k8oIwLBEGANLhO5jJlf6dKQ1AFN72sovn5kagI2N43mVuCT5enoCDbUk2xcwMTjbG3hJW/aD8/sBsLxPumrCC4AADEsEQJQIpMmUebQGYGpP22AsWs3lCvm4IKuMkYMGs9oZ8OITWkXT/k2ggDdOtfoNeOdUABJvM6q+R6uHQcCwRAApM9oeMLWnbfGwwWK2q+Uqo4xFLjjcvoA7OcQzYaCA9QM/s/sL+DtyGd6g94OpPQ3O40KjTNYkRWjNyZI+t8RfwK/HEocwnZNFIWaFjSXB2Rav6KAA5jhtqPBMGHgdLH5si3+ArXeJiWOQAfvytAWAbaLLmNIvZa1cZUOAzQVMFsd7qlZQAI+aGi02a3JW9Yn1TNiCRhbW5ZBfgFldyGOQAfvytAWAa5kFzEK53OBT2SLHVMgyHVcoUCLAhis5ORe98w9OEZ3wftdOt72ysh54Skta0Wn3xVgbKL7wAnxiJHkMdhHtw9MWABbkFKRuEUualLmNynIxzqqBN6yQduRyFgKsreBySrxtDVodbKb4lVrUTUrYzSbf4PAUL8C/OSY2tkMdTOVpCwDL4/Li4nCc37QeJpSV5dc4vBQWV2fN3bv3cIYeFdElXE6T4i1IgGuys2uaJmxZPzgNx6uaNmu8AI8+Rh6DDpja05Y0sqSV5Tk+7+CGy3lxW68nV+JyS/bR6KO5GgjYUlCQ5V10BAcwd2jnZ57uPKzcO2HLAKffCDCsnBQ9MDIcZMC+PG1hK1qlkDd5j41UVtMyYg4n5OFCZUPenqOJGWZURJfEnjjmPaAXnNVmD8HL0LzvDe+ELQMs4PKwpua5AVaIcEGWIxzs1WZ9eNrybpIBw7yvZqRsuci4cqgygyW3Yhz6lTSikaWPvpZwQuylHBTAT5CHR70TtnCo0mxsvpElx10LwAUZsC9PWwzYhpFLrDRR1pcmVojkdo08m1tSUgcQYHtcdPR1kVceQQHcn7yuX/dO2EZj0RqeiOs0KciAfXnaYsAWLo5XGJoo2+wGVmFavVXDF5+NT8twPGyoiU7I8nYhKIDZAzs/80znQd57/bTZwwa9qrHmCjJgX562GLBdJhB6V0g6oGCzRMSSs3Kcn5yUlYMaYrAVjbHzmjTwgtSKFqWleVcOIDxnVVJ72vI62KpWebejdUYejqdJcJEKqKoqdsXnxaLr2azQCPGqvPYZ6PAh4QiYWlp3LJrYcxTjYRZ4g184c/JiXi6G+sE6Pvd6PLzd7e72BhNwT+8IT8COmbG0GQCc+rPho1HDtrK7jxlYiLbhBNbpo4eUkMqhALind0TrAiZe/7aabMCmkrCZuUkXhTjbChtZ4pPX+AKtmsVyMygogDNJ6eqd0HOtyoJ3dLp1X+liv9bpDi86OF+nE+e/o6ONUVf+EKujv6dLmEYqd+i1Kn152kprVarVpLKmTkrkI2ZjFwtS9p3LLS+Wa2S1JVfoRZikkMfLc3kdlLUqOz1FyC3ebnvdwWheXU/HHRwzggf0bHKK3BwakHwELv1CKnfoO9iXp61zB2vYXIWHMkvETji6q4CRysFsZgVscdGTxWh9ljLXbRuUO7gn2ezo4Z2wKWDQy0QCtq3r/dQVNjlFDgK2ft7vySpSuUMD9uVpqwBG07DQi0du7yaVZJ+gJaQX0ksMsA5WCQTnowvrsEq+2/tnQQE8L404/OidkOIOfhykfQHAYTSxrvBZNjlFDgKO/hoUTnS42ZEB+/K0VQATfWKju7Ilgx6Xk3dpz8nEfLtZYVMmX8vLzm3QuRvcgVrRjpmx0tcs4Ls4vAHUv8AmZ1BBwGcXANFbpLK/gFEzbSdnEgJM7JkuQhujOxprI2dDbKjxBtaPmti460GHb0UryivQvemmLBNx47B8Rm5+NmxkAbswuaAgw/NJW0cCTM6MBQeHjPwR0AePHniRBPzOo6+ubZg6bhCTVPYX8IlFAKgJwI4909HG6BkfgJRPQf3EgXaA9lAHwhEgemFjko4KuPH8RoPdTRmGrYo6kbSamZ+dgYYqZfkXYq/W2zz6zUECfAuSByZWeSZs034waqYBArBzx2Ww7yTZWNt1/LdUQJQMJ1cDlWsHjdYATO3pzQA2y9iez79JZRPGUiixknxuDkZPSTWiGR0VEl6mVs/iyoJfRG/eolJtXHpklGdCvwB/necZ6y9g1EyjEYAde6ajjdHJxto4bT4soxHgnX8B+4uNSVoDMLWnNwNYLnbsg+WpjEkkhVli/slDGxdeyi21EIB5nCL0dKLSLbsgAR6M/rwCXvVM6BfgA908n64G0MgqfN7zDr48lWiska89Nd7BbzXqtwZgak9vCjCOSxTeyjZjYVVx/MU43paDKz7ed6lIih74x/6xlakuEuOCYA90APCcCADBs6CvZ0L/iui1j3o8AvMXMGqmvUTWwcSe6Sa0MTrRWNt+CuiWMgjAgigQ83tjktYATO3pzQDWlAk850pCZStWnnTh8IkDxy4dW/DTnJ8PZubAfnD1sR27j17Iv8JxfzoRJMCnur07udsZ3WLPhH7Wwb+8JHeL9RcwaqbFcboOj1pM7pnORBujE421QXVAlzKDaLyBtaMmuDbBaQ3A1J7eBGCrsrq2wVtZI6jIiIvdfe7g4eTZ876cfyKjsMGswNasmPNjNl6Ju9fYwWpFV0fHVDdJ6Cdg+5dvup24OcBWOcvL8uCPRVN6ehOA5UK8DD0Qtje+1g+VtVUVGUzOkiNH9tD3rd+wJiH3is6s4G/79qOvEnDc/XYIXjdJQ/Hj+duKNo+f6Gr6NwdYKcRFnq/otMPDBipPbwIwC8eFauR2OUeD6l4r8cBfxs0oKji45silrdsWLjpwvbRKYVbIr8778+ASLstjfkCQAJf1v/32N5psrOt3N0k39jqGOc7SHGDYHvG8foMPmNrTm7qDJVyITFeJlgA3Yzy2ASnbzVZVauHhI+eWfrFs3goms0JrVpjpi/cd2yzw9DhIgIftt1n3jPBO6H8/2MyF7pFVS3OAdVxhmeftFHTA1J7eTB1cnZarcgJWinBcRiob2cxUxraN4z746YOl9Gt1drPClPDTt6suCTyLsKCtNgtuantZY0Xj3DME2GZqMvOdbGQZVF7zl9phtVnQutvL2lnEIwYbhopoCFjiWI5SKSo6cXDh/J8nTP1hXnQO6gcLlv3w6cI8ryIsSID7lAHA6eOd0H/ANkxQ5RifgYAbsHKWd6PZn4cNJoPzumg7wNSe3gRg9RVmOV+nUWhNsB1igZ4bSWUtH0vLXD/tk7Gvzl27hoOm7GR98cHbw8/miNqjDr7cZeyYB696JwxgqNKq0TrgQMAeO187FW4MWFleUerIpO0AU3t6E3Oy2KwMWrGiUlJO6KANZTVyTGYFdmVmUfyOD74Y+/bE3zZUoRHNk2Oeef6VGefYZe7PKILViq6JOd/0jY6WjkW3DLCd7RrEa8NWNKWnLQesycE4xXqW21RvUM1mJong3azOvnZ03fgvpn2++I/oDJ25cs+UPr0+mHm21OO3Cf6+SfL5i351tDJbCrgB47G2ec1d9AMw5ppi3FaAPfdNcrnaYsBWjJ6ZWGpTCPHKRuMEKQUFmVotm59fcOXo5x99/uM3Px2+XqQv2fLNC49G/XqgSKC2mRpPE/x9k2x2ULKWjG/x0yTowJLnPVvTfhTROhZX7siyrQB77pvkcrWlgBvqSoRlBQZgVbCUjVZI4nOSY1gZlYLSq/vPH9vxR9Q3C6KZpWrObwN6PNx3dvQFKbwB2E4m7fI8OPsMebyZx4X2r1/3KBn9aWS5npQGbdqsw9UWAlayK+PZEsyry6DFi7OTBHnFzPTYA/t2nNi6dN1vq66WGAXrhg/q3fvjJJ5eLcRxx8t3QQR8yBkQL/oCrXSwcfr09UpS5DIllcgpY5UOZdmkobUOPaSoUDi/r+FX1rqUFcpaUY2fWSsCsUMhdX7tDbjRU4erwunT+1W67PH/FDIBxilKrKzziKyvlcklrNyrlzeu3nfl8KbfVs39489Ff/HqFZJz/R9+7KVfd9KrRaUcTqbTTOoTSiljfdghd0bvaA7wU64gfw78U0ijbTGRolGaqERtpIo1yhwB5aAPDehYnZIsNpkMWke0rqiqssDg1FboFKUVLJl3JmrKE2pVgdghdQQM3oDdPCVc1dBo/WscyjIDVV4+7FHVcKAvrs8audYkLWFl6U3aZObRI8fWH/hrzZ+zp05Z9OcSvl4v3T62R9eeYxcfLaopYBUpHGl0esqsA7JDrXEEdvkDGBYZdb+RwZuc0aF+9Qf4tyErt+CqwVVEozER17RRVYMcx8XeTe62LaIbAbtcbWERrVNiMlcdo+NWsY0YLriUq7BLKw5dZe4/uH3WB1OGvPLJnBPX5Ob6rW890e3x8Uvy8DJTkBtZhOxxHLFFSxbeZCvaKdVPMmDi+IICuqLxaZLCzBKLscaUqgZV2aWLRVay9jWodPZmHGklwE5P3VxtlUl3xGuxGI4xCvJxQ+nZYxn7M7IOjR4xtPfn3626VGIW/fpol86PTZsfI6hwe3zcgSbdNZEbz8lCI1P2tLSM7AYSMOxNCNQNCoXLKVWDLfNIUnwKWvoeGLiiClUzjnTUSXfOtGJcpNSxrybHn71Qa1fk5VWmJSd8NPCV3kMmLv87y8T/9eHO93Qf9v3WDPdWWYgCbvho+Fub2Z0YwP7YAjM+eDa8W7qNG/OTAM/t+9YQtxWbURG9/sSZC0lsPiSrEOM4qxlHOjhgM8aBl6m1PvHC2YRrsItsBsr0zZOefuqRAd8tuVCu5g2//+67Hhjw+YRfGG49ihAFfOKXSrwor/9MkP76AlA/cbAdYO8CsOWLCgFHs2+u0W5RyMg8VUranqWrfo3LhaU40FThInkzjnRwwKChDoO9YTsWl3jpUinxeDT/xDdv9+354jcrssu1Ra93u/vf9z/zctRvVwRuaUITcMzgdBzP+nC8bc7GBeRS3wiwradOqZevXnvqPLsSLycGrVQ1l2PX/fjnocSrGoPZrmDJLc040tEBYxKcDzUVWRcuxDFFdqCMu7hzyeA+PV+YOC1Fbs56++GH7rrvuScm/rnmlKsSDlHAtmXP9TyW9+E6+tjY38ilvhFg0MsEcvo9cfRi8t5TRQJiEVZVbfy+n3/Zsu9aLsbnuH6vEARsUSqK0eQOG9BZKuMKrhfV1ufQ9qz99I3HXxow/Ifd5WbevOcefODZt/oMmbEmHzMo1GRFHKKAgSHrSK/0cfmvLjv5DfNhtNQ3cQc/Dr/Bdg49+9efi48wiatYpefO+v7nk/uv8ETEZA+LzuTbkY4M2I6JxJk8YWl1UXYdMGVj3GuspJ3r//hg7NuPP/Zsv7VXz6slO3t26fLQwDGf/ElLSikRV5EP/kMUMF4oKel1dZJ4Rv7BWcv3o6W+iTp4JjAB3a4hx9fs/GPcFzlIW2WMO7l+3fKdVTlFEokctlMqy9Q+Hek4gG1qJdpWR6uUqx0nR3uhlctUdezUTIYRKLmpW7esXfH9ByP6vPVw12efWXbkiqps5iPdOj/wxuJV+64dTGHyHM3JkAJsV8kUVgIwve/QfruvTsLl8sOzXi9DS30XdR05aqZQQR86YNA1dsLOsV+v+BFdwiol8/y+JYfyk3aupMN2iXOUtqMDlgvEbDPQ4kK8wpE5sVaw0ShlFBTkwF6/PP7chmXbvp373svPPtTt9WHvbyxuyHjn3ttuu/XFadP3ZsaVp7McozshBVjLx/n80koUrawQFJQIyjRGdhWXmOlfzBcmZ3MFsmqBqIyXtuHNL5b8iGavqpR5sxfO3BK/YdnZ3Qo03TQkAKNH1wIt0ObhuNg5xcook8nZFdkJKan5anN+5qHdi3/YPe/Tdwc83q1njyfe+J1hzvjpjltvve2+/j9uY/NxTqZcJ5cZQgywokrIoovz6mRys12vMTdojMBUmEvUyUZuFfscjV6Snpscm3Jq7/mFM36cjQb5VKayVduOLmas3HPhn2pUvnMwo09HOgxg1F4uMwJttQDnkV/ZjBY0lFXFyDqXIbLhB86sn310xvzTi7/t0+uBO+7vOeCX3fr8ibdC6TZo1ZkqGVtmsWPE8uChBNguSrx+gSPOy8XFzodh6JfA0HLY1tKi42uXHz+TKYg7uO/0tpOnD5yuQD2i/2/vSqCbqNb//xwXENwAtycCR88Dl/cURcAFFEUUfQiCC4KIC4gK4hNEQHZRW1YR0ELZoQgUodBKoFknoW2atmmbtkmbpmnSNk2aNM0+k6XZ5v/dLFBgChleW9Oefue0c3Pzzb3fb373fve7kztzbc3mXxLnTz+2bd2m30Ovk/IHWwcSPwQ3m2UONAbbzLgPd6O1WEo5QZq18nx5jVLjyk7duebLFakJB05MHfXgvbfdeueAFw86WY/dBAT3HTFzQ44jGF6IWu2iItguk7nikmC3UlnOV2r4pRhWEPkqINdV8sqK8SChOJG6bFtSprhOceZU6p6UfYf/TC+xIoLta+d+s+jgqaQzFXX2awCJH4LDgqJor6ymyhaJHTyyInYdT1YlxnZ/v2LetJ+2LUv8ZGjfPrf16nX3k7u8R55BBPcc/NYacYGf9DqKNfXlfgqCPRCuoUVdsdvRcQTrdBVWU01GUUFBdCZvquXxpVhZoej8iX17liSlSZiZRSkLNm09UZFVLKv0gouWbHrv3VfmMP+S6y4Q7JaJKWuMR4Iddej2qrFCoysjXD5jdSGjVsIX/rZ41sLZMzatnvr8XTfd1PPm2++dsNW95RFEcI+Hhr7BYVohNpHpLM1UYzB6y5bcGI8EB82KShsZMEoqKqujlgQceWpBniJt9+b5P/28/VBamoj15VeLFkz+fGN6MVNsDtrMaZ+8+vDzkzbsOlkRfeguKNMpK6nWS8cjwbhap5XZK5hnSwvV1aoajVSiEfFY2Udmzp3wzuwpr/7j9p69b+rR67bHE1yr7kJj8I39+ozdeLLKpI0843ElwX5ZndoUly46bG3AZFbXyS4+q0SUn05n71i55oflhzOzOBnHv/t8wsix46fMWnI0tVhltakXPzWg36Pjlh45q48W56/UKdVU7xiLR4KD5gqZu1ynrW6o1pb+KqmvM6sZJ0+eOLNp8UevTHvu2b69evXofUuf+8av0n/eJ0Rw7wdenPJLqri8vr4VgkmfHQ/EKcFIAqYgbm9ptV8oOLt1y4oliceSU5L/SJ762egnnnpy9Gszj+4/J5M2bX764b533v/ozLNY+YUzTLWVFG+cjk+CoXwIJXU6tUmWd3zz0RKlRVp4dt+RH5dMnvjGjJcevv3mm2Fu1GPIczN2fRYm+MaHxnyQkp5RUiALXaTOFEWHJXSjA5fJLt5PL9fwsrcs+ibxyOmt675+98OXRg1+sM/AIc/M/uGAsF46+6l+PXreOfT93/Kje9QE3S5HAyXsuCQYPFbALZObTIX79zAYmTYD93RCyvJP5016Z+774wfcgcbdG2/pO3j5+nk9Q/zeOmjUf/akFdSVhcvonAR7peoLaw19XptcmHFqy6K1KzbvX7h29oevTx475pG7H7h3+Iezxbasj9HI1Osf01IlkTsGod3TO08UHaji8aoCQT+ezzzxU9qf0oD05P7Vs96a8vpbi94aNuOhu0Ok3nTzA5N+Hh/uwD0GTFr6XTKLxzOgydLfS/CWyDJxh9VNJXYXVa7L5HbXnsWYJXjoY312tpQlZh/eNWv+t3O+nDzx42fHDXtxZN8+d90xZMznmYRoZA8E+r7RSZWF+uIGwu22ydTqUhNlhfTsaIoknLEQ3BBRNjkpq7BR59rhhDMiEcPoduezju1bveU8X713x4/z3x014YVxU8Z+MfXxXjdGaB344ehI8p8Lt6dl5gm5knI4zY0TlEXTtiMkV110RyHbIhtLU+4ADtmt7Lzt9ZZwRYLa0LeujHz2Dg739OHEj1bumLlkxqgnnx41fPC/+9zWu/egQaPX6oUDe/a6sWefB4clSnJVmpomv98p12hkNsoK22IH8FYIvsYO4NRbfTtwt7EuTSw+6/D7ywqPpPyZfuZwyi/r35s7e9yrE6a9/s6ENx/pEeIUxuE+H/4nnLr138+uzTwjPXX8fC2qlN4O4NR2xLIDOKVcl4v2aV1EWV1lSfgcB6fg2G6GMPnA9q8Xb529bNPXU9+b8dxbE4a9Pf21txcuWXr6wBNjnxj28kdzJqxmaWrDMwdLZYWts7hoWb2On36m1hcgjWVM7vmkfXvXrdr2zsxVn368bOF78+dNfvb+fn373z1gyKP33Pf5qD533PPYvwa8NmPuz6eEB0//wStB68/+Vhe9gMFgoP0k5CcYjPMogwsZ6JkjLRw56MKeh4QaMhrheA5piBinU6Qeg7zg2BkGw0f6rPsSv1n6ZwZ7194Fy5YtWL1z3vzZ40a+9PJ/Fy7ZOGvBsnlbD3732SeTJ06btexEsX77/qQ9tQEyKD12/Dh6ERrJglLRLxJ1cMTg6D4LCbTcRQ9HJtLIgYQS7DDBkYF4zodjBYke6jxzlIHm0hLGmRgIHpHKYGShxKm/GAwdHOuhHC7KEECiBi6sAY6ZKEMICfTiKDODcfJEOYTP2Km0U+KAvaliT+KPS/dvXb19xqwvvk5IPzR3yQfjXhg5dNIrk8eNGv7ME7uWvvTmxFFj3/x2/dpth9cu3bibWRkgmYz0dPQoXC0Uykel8yCBXlpUDnawUEY2ZKCNFpsQQkRwHhzRgwk2lIFaRzHjZBqK2FwMxnctNueOheCfVKrQxl6GMpUq9CxZHWSgXoTDsRYRXA8J1A6dcEREkzpVpajB4/U7IEMV8Lk8lSzGcabOU804eVpwnpmckZ6y9ccV6zbvTPg9KXnH9lTmhsTvVvzwfeKhYwU14lxBvkIb8KrKSkWh6mqgDESOHY5opZ67AhKhn4vhWIM0tJCwgh0uVB3K0MMxtOpPpSxUoV7QqFLG0oOLIwhLqsJVOiJVhhDa4cK2QKhSoXjQrVJJZVKNtgI9p6DzEB6b+HzG0aPbDu/ZsTP1D7ZMuDdp86IvP5jx8cxv18z8asHCtIPb1y9d/uvv23bvPZByePdRtqgpGKyukiuICEJNyyusqWqJENEXQgi6DXBE/sYTusIhhFIZ8lY+lWoDTYKvy0X7DQQRXRbqdXvcRp1Rj8PImlOmlClL84RZAi4j/VxmJvNc5jmOpCBtX9rZc5lsFiaWWyw6rV6PN1sdDsOVe+OR8eiifWhnLEAJIZGLMFlry5S6EnkWK79QVCzBWGl/HNjyW8rhPb+mHEr9S8gD/8NkcgH3GZZYUVVt8AchDiWoX3EQ19OkFg/+E27CZrKaCbtZoyorKykryc3OFmB8Ho/L4fF5HKwkj8fhYRiHy+MLSxpU8tqaRtzfZLUYKIuOP4LDFXhxm4sgPHYn+mGp0aoSF4pEklzACSAxFpeJcTgcVjafncnictkYxssurjWbvcGgVavDrdQP5cU1wS0/eP0Bl83hxg2KcnmlPF8sEpzH+FzgE0iFuaDwPDDN47C5HH6OuKbJUNdA+MiA20NNSbwR7HB4w+p+lwc3uSCah9rqjMoiSaGwIEcALZjDY2NsaM0sjMPnsTkcLpfL52CYSGkj8GZ3o6GpydQJe/AlH3240xv0GZQl+eJKqThPkJ3Nx1Df5fC4Ag6HD0k2i8OBq5BVqNaajJbAVYDEG8FWi4cI6fudHovF4/KRZNChq5EVKgvzhPyQgG9icpgIIxdjw382D6BieWUm3Nxg1JsMjXinvJN1UYK4x0PYHVpZUWGeuDBflMXHBFyMy2YM9b+VAAAJ80lEQVRz2Vwuh83ngNcCgpEvyylTVBsjeNG7xK4chuON4EZriFQSNWOj0+1E+3y7jPUV0lJxaXEuBvxibD6CChRDg2axgW0WGwjOFqt1lkatvqEJd3dygkkg2Ex4mkrEBaKigtzcULPm8tksNoy9AJrDBMfFYUP75mcptCq92RkIA/E6iSs2bo03gq0WN4on0T5IQbehEV1mn9uulZSKi2QlQiEmYMMIjLHYwCoLWjObH6IavDQ/X6msqdXhNm+gc96LbiF+gmh0uxurCvNKyoqLhNkCAZePMTkscM3QtFHnhVEYYg8uX1hYXllrdYQJDjih519efrwRjDu8iF/CiQcCJrsNvWcoiNsN1ZWSEokoG4WSEGlwQmMQRFmsTC4Myshz8flCubqhXmP1Q8Po7ASDq/URuFFfXVahkJUU5aHQA5wzxoO4AwWVLA6Pw2JyeRhPKCwsk+nDP0J1EoJDF9YHc0Gv1+JwNKFzg36TVC6RSIpz+RgXQUPDDxeYZnLRSISB5+Lws/MqtVazzQ/N326jLLoTEQziszQ26JW1uoaKMvDUfAzmRRhg5oWGXzQaI9D8nGxxgRQG4fC2B53BRUde6e/xoDs7VmuIKr+XUFSJxaXQljGYJwGfPAitEL1cHovNY8PcAQgWVRssxmY0gOEOyho7F8Fej9Wot+hxa1VdORDM4/FCrEIHZvK40MTRFeAC6sISmdET8llE6BU4l0t8Ehxy0aTX4UFmgOcx1aiVpeUKSX42mvCz2RBlQGAFvZfFxjIh7IK5Q16p2ulDYxEBBF+5RzxdO+KAYKfVbnPbTDUKaUEBTI0gikZTBhiXMDT8stmZXE5OsaisqsICvaFVIPFJMBls0RZ9HqJJXaPRqOvU8iIBBJMYB2bCiGKENxPiSiafn1sqi9wN8tn1TV2gB0MbtzuJoM2qqayskmbx2Rw+cl18LrRvFurKKKTOEhUqFNX28MSjUxHcUvy41aDVazRKnaFOKuCjGJLFCnVgFg9QAscQYmVJGiJzwCBhxymt61wEo3VLfo/X1qCqalDUFAg4MHvgIPDgwTgYkwnxNEydRAq1ARyWP9gqkE5AMOnBiSYbYbcDzzIhhBksNAEGHwVDLwvSLIwvEJSUN9rC9QSc+MU79y2lsxGMXoSGW0z1dQ11OlkBBnEG9Fo0PQT3xeMzmRBj5Uqqa3B3hIvOS3CA8Ji8fidh0Wul+VkQT2Ksc4AV2jKTDeEWl8fl5ygVdXZ76B5YkHDglGW3J8EZq1dFNjVoW4JhJmEyWhpMiuI8mCqdg5kSF8Uf0KxhvsgXFMiVGqfTdzUgbU7wBahtuH9wwGsPkj7cardr5ZK8LJgrMc+BcwZuoTdnQqpAVqU2WIjwLDDoclFa344EmxYHqxLDybYlOOi2O4hgs16prCgtzeWyodti7EwYl6Bp80XliqoGT6D93xfdUi5CbdsNokO1Ek5DrVIqkWTxQxPCcyjUAIeFiep0JmNjkyvyg0rH38nKOUSSX4STbUswfIXG1wCBN+nxuiIuD/2OxIUmDf23QIO7m1vMFzqG4ItQ255gMuDz+bzeZndjGR+mg9CSYRbMwTiieo/P2+zFo9P8jieYeYIk58AxNSFhIx4WqwWnEitlrsNEnW2LpmyNjXa7o1HCxyDERLcCOJhQe1lZ1EXbaNnRFEnYWyU4AtWQkDBCE1E2OWhUYaHOvjTX0VgFkwVwU+jXQoFAEbk89khFdjtlGbTssEazW9sB/BKJNmvOwYO/uMJit7qoxO6kynWaKJUJx2UZDkMJxkT3aDlZFfrLS7JRluGgZUdTtOZr9WDzwYMjdBFlE0FZBbU9Vursy3KduLk4dKODzWUBvZcp45dnROygxNSKHTZ7JJEUC8HmxUHF//q+aAqheJ2wH2cyIfIwNnuvmCh0jIu+CLUdXHRLCTbLQ/egKW5a/Q2/JmWsXNkuUTRVrj20WvJK6agoOgq1nQlGIuBcuZMA2VkfXaGQ+NjivTXpAIK9FsrsboJbSDfB3QS3sKOb4GtKN8FXt6fTEzz3YFj27DpIJXsOUOUe2EGpvH8vZXbyPsrs3ZS5e2nZkRStORaCo0bv2E/Dnt17aCjv20mdTX0Brs+OeTQJ3pAQls/eTYhdVo+ioZww4b80lOfQsWPNc9HUVsrHJC6RnVHdMctoVDH9YxrK375GQznhpe9pKH/wUTSVT4/gqBxZTUO56TkayuRnAhrKx1dcW+eCWEbQsSMq42qvrXNBNuymoVz8Hh07xqtoKP+STJFJh2AmVQGtifVTGsrkT0U0lNlJNJTtM+nYEZW5V+4O3rocOk1DuXIpHTvma2koHz5JkUmH4G7phNJNcBeXmAmuXrp8eWPLZQBXE//i99Fj07Ep01GkVTAtk6/ztHZD2lZQYybY5iazk1ouA7iaBM2pXDJWZRqK9AqmZfJ1ntZuSNsKKh0XnZPcchnA1QUZF7NyzIp0C6Zl8nWe1m5I2wYqDYKJb/QXlgHEZFzMyjEr0i2YlsnXeVq7IW0bqLERfHbNAdK7siC2BoWU46MHx2zyRYkjpG0DNeYeHNzAIS9ZBnBt42JWjlmRXsE0Tb7O09oNadtAjZngnKkrVx5quQzgqrJ+zrxdMSvTUKRVMD2Tr/O0dkPaRlC758FdXLoJ7uLSTXAXl26Cu7h0E9zFpT0J7uG+9LN5Sv/+b2pJst+9/fvPvkzpBpjKrXpw0OCl3s7Y5uIYaUcS3JTmDXz9NsCWXqkEsGe8YSSJRGtXIDiOkLYzwWeffHyskiSPPzL81xtCeenDo7AjX4FS2iOjf7yBrOppjJi0cehD6XAVnn5isgU+rf9gePqFEoQvDhvBa0eTr1PiGGn7Eqy/S0HuGkk23Ksht4RgB15dB7DvHzTor8hXISUVuf4G8vi/oiYlk0UDoReQZOIy+JROqgdGSzA/ric1A6kXZv6dEsdI25fgU+NJ0n+zI20ySdpCsOdN9kfadeQrUEqbCIPWDWTqY1GTwJXd4iV3vjB66AT41EySvbyREjLuHjNmzGBlO9p8fRLHSNuX4LTXQ9hOTonAnj8RrX8OwY58hZTeIUkHOK4eUccFf73d4iEWMuOV6KdICadfaEdz/weJY6Tt7aLlZPIzZMM9GvIXgP3VpND69hDsyFdIaaCV/AO+nfaGgSR+tkaAssaQ5PQLsCMlNPVjkqSoHU2+ToljpO0cZDGGPvoy+JljQ55f15cs/T+YNVwIPSJfgdLJ0R+tAdjNywf1/+dKbwSof/rr0xddgB0pgcwd9ehgWutOO0biGGnHzEkIkjw04W8uoWMk7pB2DMErnn7yFTpLuNujhI6RuEPaCe8qdAsd6Sa4i0s3wV1cugnu4tJNcBeXboK7uHQT3MWlm+AuLt0Ed3HpJriLSzfBXVz+H2v99aTVthRVAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "    res = all_res[[treat]]\n",
    "    # significant thr \n",
    "    pvalue_lim = 0.05 #; log2FoldChange_lim = 5\n",
    "    res$sig <- as.factor(\n",
    "        res$pvalue < pvalue_lim\n",
    "#         | (abs(res$log2FoldChange) > log2FoldChange_lim & res$pvalue < pvalue_lim)\n",
    "    )\n",
    "    relevel(res$sig, ref=TRUE)\n",
    "#     ## Volcano plot\n",
    "    vol[[treat]] = res %>% ggplot(aes(x=log2FoldChange, y=-log10(pvalue),colour=sig, fill=sig)) +\n",
    "            geom_point(aes(color = sig),alpha = 1/10) +\n",
    "            ggtitle(treat) +\n",
    "            xlim(c(-20,20)) +\n",
    "            ylim(c(0,11)) +\n",
    "            geom_hline(yintercept=-log10(pvalue_lim), linetype=\"dashed\", alpha = 4/10) +\n",
    "#             geom_vline(xintercept=log2FoldChange_lim, linetype=\"dashed\", alpha = 4/10) +\n",
    "#             geom_vline(xintercept=(-1)*log2FoldChange_lim, linetype=\"dashed\", alpha = 4/10) +\n",
    "            scale_color_manual(values = c(\"grey\", \"black\")) +\n",
    "            theme_bw() + theme(legend.position=\"none\") +\n",
    "            geom_text_repel(\n",
    "                data = subset(res[order(res$pvalue),], sig == TRUE)[1:5,],\n",
    "                aes(label = name),\n",
    "                size = 3,\n",
    "                box.padding = unit(0.35, \"lines\"),\n",
    "                point.padding = unit(0.3, \"lines\")\n",
    "            )\n",
    "}\n",
    "v = vol$METTL14 + vol$METTL3 + vol$Virma + vol$WTAP\n",
    "ggsave(\"plots/peak_Volcanos.png\", plot = v, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/peak_Volcanos.pdf\", plot = v, device = 'pdf', dpi = 300)\n",
    "v"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[Wilcoxon signed rank test](http://www.sthda.com/english/wiki/one-sample-wilcoxon-signed-rank-test-in-r) and t-test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"METTL14\"\n",
      "\n",
      "\tWilcoxon signed rank test with continuity correction\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "V = 13403000, p-value = 2.626e-05\n",
      "alternative hypothesis: true location is less than 0\n",
      "\n",
      "\n",
      "\tOne Sample t-test\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "t = -12.41, df = 7526, p-value < 2.2e-16\n",
      "alternative hypothesis: true mean is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.1661212 -0.1208004\n",
      "sample estimates:\n",
      " mean of x \n",
      "-0.1434608 \n",
      "\n",
      "[1] \"METTL3\"\n",
      "\n",
      "\tWilcoxon signed rank test with continuity correction\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "V = 12628000, p-value < 2.2e-16\n",
      "alternative hypothesis: true location is less than 0\n",
      "\n",
      "\n",
      "\tOne Sample t-test\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "t = -16.202, df = 7525, p-value < 2.2e-16\n",
      "alternative hypothesis: true mean is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.2949141 -0.2312522\n",
      "sample estimates:\n",
      " mean of x \n",
      "-0.2630832 \n",
      "\n",
      "[1] \"Virma\"\n",
      "\n",
      "\tWilcoxon signed rank test with continuity correction\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "V = 13353000, p-value = 8.105e-06\n",
      "alternative hypothesis: true location is less than 0\n",
      "\n",
      "\n",
      "\tOne Sample t-test\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "t = -12.373, df = 7526, p-value < 2.2e-16\n",
      "alternative hypothesis: true mean is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.1777083 -0.1290999\n",
      "sample estimates:\n",
      " mean of x \n",
      "-0.1534041 \n",
      "\n",
      "[1] \"WTAP\"\n",
      "\n",
      "\tWilcoxon signed rank test with continuity correction\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "V = 12742000, p-value = 2.132e-14\n",
      "alternative hypothesis: true location is less than 0\n",
      "\n",
      "\n",
      "\tOne Sample t-test\n",
      "\n",
      "data:  res$log2FoldChange\n",
      "t = -15.143, df = 7526, p-value < 2.2e-16\n",
      "alternative hypothesis: true mean is not equal to 0\n",
      "95 percent confidence interval:\n",
      " -0.2504366 -0.1930277\n",
      "sample estimates:\n",
      " mean of x \n",
      "-0.2217321 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAC+lBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+RkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///+7+53vAAAgAElEQVR4nO2dC3wU1b34V/cvkUf/vf8kRGwgudQSTDEJNQlBDAShucI/5aG1SCwtWmu5+Gy1RVT+XEDAv1at9pZwheuzV8MlJkH5l0e9goASqaU2rby9EB5pk5BNso/ZndnN7/P5n3ntziY7u3NmZ3YyZ8/3ozuv35xzZr7M2eycmXMcQCEah9UFoJgLFUw4VDDhUMGEQwUTDhVMOFQw4VDBhEMFEw4VTDg6Bdc4foM+e652nEezPJscAzhfs0yIbJz9VYdb3Ok7jt8ZU2iKdvQKHnsT+twylhd8x1FEF/p/3syjR//4qTg9ykqCf7vu/0qCt/wTFZx69Ape9tUzAFUreMHL5JVLFyin4fUHRMFnx52lglOPXsGP/GgNnBt1WLvg/pmvclRw6tEt+MMCWL+klRd8hRPxJ0gg+IW5QAVbgG7B/XkthbsFwYtaEQzEF3xizAUq2Ap0C4aVldcGWzVX0W/y17njyruSKiwFH/2Cv3A8CtoF96Cr/E+OuvNJFZaCj37B0OUXBQs/k9phkGBhfV/30VcdHx/18WtoFW0BSQhGtIZvdKyGQYKF9b97U5gc5ddQwRZAb1USDhVMOFQw4VDBhEMFEw4VTDhUMOFQwYSTrOAMhv9PjTUhtRXVx+KmGUVwSkdk4fiU8RUno2ZjTubm5k77Y/yiy2UxuJAaC6gIM5VkBbf0xxPs5FRW7LkjTpqDEnzl0ch85VbYOjNqNubkLMC/FccvulwWgwupsYCKMFPRJ7jufmhxfArLN0tX8P6pxSX7YOfkolmn4NjYs7D6TiHscceMKt+hGTeWfwDMouJvLRBX8FsWbUdZr1pyc6MQt/anAF2ZHvh+afHCbv7cubIAuAwAcV/oHR0ET0AIPT+KBXZUu2I25kQoZEncY5DLYnAhNRZQEWYu+gSfmggbbtoIBadFwR2jD0PQ1Z59AjZPAXijYndBrxiHroXLRe3QluffsQCgJ3Jx5JxHWf8Wusb+nV86N4aDl+4B6ATYsFJx7qR9AYqOQM3Lwo4HC9BHQYtiNuYE4M7s3M/jH4TiCjawkBoLqAgzF51VdN752fu+3ZYvfQc3VvPr3r0VfRMN6wNYOvwPUhg6Vc2jq6qqJpw6m798myd87lgHi7J2A8xvFpZnNcONHwFsml5ZUqM4d9K+6MuwEf58QYgUTsyEFsVszAn66P/1rfGPQSHYwEJqLKAizFx0Cr677gaYtOmeKMENc0TBgbJrd0ph6FQ1Thdn3c0PXOeVz13/VejfgQNVdHN3CMtv3N56HcCRgm5ons2n6c4E8GaE94Xpu+SM2yJVmzQbc8JvZ6/ojHsMCsEGFlJjAduGdhUNb+Yvg5/kvyUJ7sw5DFx3e/YxqKsAeHjZX8ahym3PRYDMTujMQsf9CVxwg3/MOX6FwOQjKOvn4GR2hxDnzV66DmB3FUCtcO5gzEWoz5D3RZVlOxw8Le45bQtsqRJTl2ZjTVx/gf5Nuf1xj4EviwmF1FZAeWI6OgVfdGyDescl+WfSgSmFRfvg/ZLCW05BYzEDWyo5mIT+Nlk3scT38c2FE74Hu4uLJq0TV/D7r32aP3dTJ6IYPg7uvfIcuvxr59Q+Jp67bWWL16O/X8R94bNpIH8Hwxfl48qPi3tJs7Eml6aMyZ11NP4x8GUxoZDaCihPTMeiGx0dJSxG1j9uMrEo6tiikImw6k7We6e1Zx38tZkliYMtCpkAequScKhgwqGCCYcKJhwqmHBUBQd/cefvAZg7V65EPyeb/8+qS/InxVaoCu6//A4veCU/3/WL/pMbpE8BlwKfz6WbJHZlvJZk6/dYkm3AjRWuRTCAIPiuFS+64eDrAMukT4ATLS2ungh+f49uktg1wFiSLeuzJFvOixWuXXC/G5p/A7v+E+A+6RPguSVLfFyEUIjTTTK7WpNtv0XZBrHCtQtG9Dw24Arm6+zOCD5fp26S2JXxWJJtwG1JtmwvVjhGFd0PB38Jl3/Rf2Kj9EkFd9ZKpDBbMwQ/c9/9m6H10See+jv6+/mppy7Jn1SwjQX7wh/xoYJtKlhogY7/rBoVbF/BIW4G+gussyCx4F7lH/9p9zNJFpzCbI35mbTG6XA6nZnrqGB1bC0Y4PHEbgVoFW3LKpqHZRj1FxWoYLsLbswbMXLkSGFWbGyIamaINDZQwTYVXPpleFZsbFA2MygaG6hgmwperFzgBStvUtJblZ22F7yhKRAtWNnMIDc2/Ovy5R42QijE6iaZXYMpzFYyKmVbG1k0N9sI/XhHqypY7Kw9wRV8dO/enr4IgUCfbpLYlfWnMFvJKJrjmPBSrenZRgj6sMJVBUfBC1Y2M6RxY0OkTrZ5Fa1EaGyIamZI38YGYgRnCKhf1DJUsE0FcxznrV9PBQ+EGMECtVTwQGILxvU8RARfLE8s2KeA43y6SWLXIJvCbCWVfLaB8FJtZK1J2UYIBbDCVQVnZWVlXrudXsEDIeYKdrlc7sR6qWDbCgaI+pFMBUsQI/hE+bBhFacUK1ReXUkbwYNU2lzwzFdCwbpZSsGxX12hgm0quIj/KFQKjvHqSteFC5e7IzBMt26S2NXvS0W20SrRCtY7cGWtCdkOgnNjhasKLjkBcEz5VGWsV1cev+UWpj8CCrKC1OQarVLKtnbQWtPBPFpVwTuy5tyavTNqVXq/ukJaFQ3tDe9G9b6W7q+ukCb4QxdA937FinR/dYU0wTcEAdgiSAj5gmtj0Wl7wUIX2oWQECo45sbks1XDKMFTjwAcqkgsOOqPfyJ/JsV02B3zZ1JkY/LZqmHUz6T9OdXVOQcSC3YrCATcukliV9bcbGM6ROs5v8q2WkOyVSPIYIWrCoaOxqb4nSyLEFxFx7ZHShWtFSo4XoyebBOQGsFkNzbE1ZYWggltbNAgFkdwHM9DXTAxtyoxlKaVYLmx4Z31691MBI5jdJPErkGVfXXb0wCfLasni6SPNsRihSd3Be997bU+TwSW9egmiV25gCXZBv3WZMtghesTnI6NDYMgqSO0QUQaG5Q0WDMYxZt7Lcl28yFLsn0+wShu6iTfX/SvrBmNYu3rlmT7c2v+Od/3od49qWA80lAwZUhDBRNOcoKjuuJJKdID26nGmlENkjnY5AQru+JJLeID26nGiiOF5A422So60pFHahEf2E41VhwpJHewRgiW72GmEvGB7VRjxZFCcgebhOCdq1+16AoWcuYf2E41Fl3BkMTBGnEFK+5hpgzxge1UY8WRQnIHm6RgZVc8KUV6YDvVWHCkkNzB0t/BhEMFEw4VTDhUMOFQwYRDruCMAUMTXL4tN/c7FwCyrsnNvXdAkBOAXTU+f8IKlrgTQtrxRBgouLOBDT10OxLcOjgICf7+3A7wbHARd0JIO54IyN3OyUWzTgHUX1/2olNY11QmC5Y2oaCG6yvXOuHk1R3ibo5nS77ehHyXFi/sRkvP3FXWFE7h0Iwbyz+w7ID0QbLg9uwTsHkKXLqmDV4QBIeqNyLBX8vP3yFtEoLOwDNOqJ8k7eaog8/y0PUOsGElWmqCL/PkFC4XtUNbnt/CY9IByYLfvRUgOKyvYSFAjyD4/oVB6QqWNqGghnno69kJ73xT2s2BKu3hLGyaXllSg5YCACNYKYXm0VVVVRNOqWc5FCFZcMMcweL22yTBD87jR6QQBEub+KDvAvShKjpDrqLR/yOZIwXd0DxbXpJSaJxu1bEkAcmC27OPQV0FXMppg+eR4AfmCyOOCIKlTXxQngveQlsXz/0beJ52SUp3VwHUhgVLKXRm7QL4xNKDwodkwfB+SeEtqEZ9u2Daxkz43IF+H4X/yJI2oaDtlT9cjQQHnsjP/cZTrKQ0WDun9rGwYCkF+Pjmwgnfs/aosCFXcAQPwOs1FqdgGekg+MnSybPPWJyCZaSD4LSGCiYcKphwqGDCoYIJhwomHCqYcKhgwqGCCYcKJhwqmHB0Cq5x8G+79VztOI9meTY5BnC+RnxL64XC4VkLbNZIThJ6BY+9CX1uGcsLvuMoogv9P2/m0aN//FScHmUlwVve/uR3lRONKzAFD72Cl331DEDVCl5w+H3KpQuU08h6OOToSqKIlGTQK/iRH62Bc6MOaxLcfd83+wcmQEkRugV/WADrl7Tygq9wIv4EaoIbnI7rzxpQUooudAvuz2sp3C0IXtSK4B8gjy24t3XP7MqgAUWl6EG3YFhZeW2wVdt3sOsKuz0uTg76BX/heBQ0Cu5w7EuiiJRk0C8YuvyiYOFnEj/i4QDBwvq+HzUdbrzpG16DikvBJQnBiNbwjY7VMEiwsP53Pxg3bOySL40oKkUP9FYl4VDBhEMFEw4VTDhUMOFQwYRDBRMOFUw4VDDhJCs4gxnUX5GCNSG1FdXH4qYZRXBKR2Th+JTxFSejZuNN4iGXxeBCGldAQ0hWcEt/PMFOTmXFnjvipDkowVcejcxXboWtM6Nm403iIZfF4EIaV0BD0Ce47n5ocXwKyzdLV/D+qcUl+6S+p46NPQur7xTCHnfMqPKJnUsxi4q/tUBcwW9ZtB1lvWrJzY1C3NqfAnRleuTOqTIYVxYAlxHumKp3dBA8QgcbcH4UC+yodsVsvEm8Y5DLYnAhjSugMegTfGoibLhpIxScFgV3jD4MQZfc99QbFbsLesU4dC1InUvtWADQE7k4cs6jrH8LXWOFfq7PjeHgpXvCnVOFz124Y6qiI1DzsrDjwQL0UdCimI03iXsQiivYwEIaWEBD0FlF552fve/bbfnSd3BjNb9O7nsKlg7/gxSGTpXUudTZ/OXbPOFzxzpYlLUbYH6zsDyrGW78COTOqcLnLtwxVXUj/PmCECmcmAktitl4k7jHoBBsYCENLKAh6BR8d90NMGnTPVGC5b6nAmXX7pTC0KmSO5dyNz9wnVc+d/1XoX8HDlTRzd0hLL9xe+t1AHLnVBmMOxPAmxHpmGr6LjnjtkjVJs3Gm8Q9BoVgAwtpYAENQafgN/OXwU/y35IEd+YcBq5b7nvq4WV/GYcqtz0XATI75c6lLrjBP+Ycv0Jg8hGU9XNwMrtDiPNmL10HIHdOhdIccxHqMyIdU+W0w8HT4p7TtsCWKjF1aTbeJB58WUwopHEFNASdgi86tkG945L8M+nAlMKifVLfU43FDGyp5GAS+ttk3cQSn9i51O7ioknrxBX8/muf5s/d1Ikoho+De688hy5/qXMqlOC2ssXrM8IdU302DeTvYPiifFz5cXEvaTbeJB58WUwopHEFNASLbnR0lOD0zPxja8Z0tUUhE2HVnaz3TmvPOmjNCMU2KWQC6K1KwqGCCYcKJhwqmHBUBYuDe0tjT4sDX1sz/DUlKVQFi4N7i2NPiwNfK4a/9jJej3b8PoxgH1bKjFkpM0MjZS/OqYukrEWwOHKsOPa0OGyuYvDcLs/lTu0E3BjB7m6MYL8HI7jPhRHMeDGCe3swgn0+jOCeXoxgbzhl7YLFsafFga/l4a8fLi212cgj6UwCwSCMPR19BXt7e7u83V3aCbgxgj0ujGC/ByPY3YMRzHgxgvt6MYJ9Pozg3j6clBl5TrtgcexpceBrxfDXtIqOwp5VtDC4tzT2tDjwdWT4ayo4CvsI9oU/4kMFR2EfwUIDZQkVTKjgEDeD47jOAiqYUMFrnA6n05m5LrFgH8v4tBNkMYLNSzngxwjmzEuZwwj2B3SlrCIY4PHEbukVPBi7XME8LMOoP8dOBcfGPoIb80aMHDmSCiZWcGmkOxyxNSmqHSnSmpQegmt5tATbR/DiyKzYmqRsR1K0JlHBUdhH8IamQGSBF6y8Cy3fi9772mt9AZwGMs6PERzAaSDjAhjBWK2WLEpZEKwpZZwWQJbFCGZwTl0kZVXBYmftCsHKdiS5Nemd9evdnJ/RTpDFCGbNSzmgObQ2gpbwgPaUGYbjMILxUg7Kc6qCo1C7gtOgilYI1hJunyp6kGBlO1IatSYRKzhDQJwXWpOi2pHSpzWpdgAJwu0jmOM4b/169YtahmDBA92SJViglgomWvDF8sSC3X6PWzucHyPY7zUrZUZTyjEFJ0rZh1GMQAAj2IeVMivPqQrOysrKvHZ7YsHd3p5u7bBejGCslAM4KXt6tUTFFJxgH3cfRjEYBiO4z42Tsl+eUxXscrncoAFaRUdhpyq6r48KJljwifJhwyqUAw2qvLpCpOCYagkTPPOVULBullJw7FdXqOAo7CO4iP8oVApOo1dX4ghOoNk+gktOABxTPlUZ69WVNfPn+/pDQe1gBYcsSzmhYN0pR9HfjxGsM2VVwTuy5tyavTNq1eBXV9q++KLb1+vSDuvFCPbipBzw4aTcF397QsGqe3rcGMXwMxjBbg9GMOOX51QFQ3vDu1Gdc6XJqysJ3ZJSRX+I3HfvV6xIk1dX0kbwDaj2ZosgIVRwFPYRXMx/FEJCCBKszS0pgqceAThUQQUTK3h/TnV1zoHEggl6dQVDsID2lFWw9NUV6Ghs6oTEEHEF46pVv4ztcwVrxe6C9bmlglUgSnAszaQIJqE1KWm1BAu2cWuSQVrVLBMiWL4XfXTv3h6/u087HIMRzHgwgln/wDXmqEwIRpkDAYxgr09XyvoEy61J/7p8uSfIYhDCiTYxZc6slDmslEPmp5zcFQz2q6IF6PvBCVC2JvV0YWS88zOMYBdOyrtbcFLG+Uf5wSGM4G6cf5T7PzIr5YMfynP6BCtak/D4RaOevbSw6j/MSnnjVrNS/pVpgztsfi7GylR0CE4FKyFQ8Mm/m5Xymb+ZlfJ/XzQr5XNtZqV84WyMlbRLf8IxW3BUXy5GIz21bQKmjV5gUpHVT7PZgpV9uRiO+NS2CZhUXjCtyOqn2fwqOtIThOGIT22bgEnlBROLrHaaUyNYvglmMOJT2yZgUnnBxCKrnWZTBe9c/ap5V7CQOP/UtgmYdwWDWUW29gpW3AQzEvGpbRMwqbxgYpHVTrPpgpV9uRiN9NS2CZhTXjCvyKqnmf4OJhwqmHCoYMKhggmHCiYccgVnDBia4PJtubnfuQCQdU1u7r0DgpwA7Krx+RNWsMSdENKOJ8JAwZ0NbOih25Hg1sFBSPD353aAZ4OLuBNC2vFEQO52Ti6adQqg/vqyF53CuqYyWbC0CQU1XF+51gknr+4Qd3M8W/L1JuS7tHhhN1p65q6ypnAKh2bcWP6BZQekD5IFt2efgM1T4NI1bfCCIDhUvREJ/lp+/g5pkxB0Bp5xQv0kaTdHHXyWh653gA0r0VITfJknp3C5qB3a8mw2tC7Jgt+9FSA4rK9hIUCPIPj+hUHpCpY2oaCGeejr2QnvfFPazYEq7eEsbJpeWVKDlgIAI1gphebRVVVVE06pZzkUIVlwwxzB4vbbJMEPzuNHpBAES5v4oO8C9KEqOkOuotH/I5kjBd3QPFteklJonG7VsSQByYLbs49BXQVcymmD55HgB+YLI44IgqVNfFCeC95CWxfP/Rt4nnZJSndXAdSGBUspdGbtAvjE0oPCh2TB8H5J4S2oRn27YNrGTPjcgX4fhf/IkjahoO2VP1yNBAeeyM/9xlOspDRYO6f2sbBgKQX4+ObCCd+z9qiwIVdwBA/A6zUWp2AZ6SD4ydLJs89YnIJlpIPgtIYKJhwqmHCoYMKhggmHCiYcKphwqGDCoYIJhwomnGQFV71oSDEoZqFP8Ny7hMnds2HjbiNLQzEcfYLfHsF3pub7n6+H19jsQZb0QZ9g5quvoc+3R7qFKrr0X+aOern02QXDi1s/umHUD5DrV0tGjFvBJkyGYj46v4Pv/Tb6qPmB+B1c+r8+6HeXjv73v86fXPnRoTGbAP5tz7kPv06/nYcCOgXvv/ICdPyPvZLg+9Ga0n8G+MTxe4BHpRHEt043qIiUZNApuH/8s/Cr3JAk+GW0pvQ3AF86OgCerQb466Lrx/zDPxpaUIo+9P5MWlUM5StAEvwKiB//7XABPDcb2NwH/3RxS66BxaToRa/gU47/dPwVVAQf5x8uXkMFDwV03+i46R9K+UlMwb3D34fPx1DBQwHdguscv+InMQXDtvFjp/+SCh4K0HvRhEMFEw4VTDhUMOFQwYRDBRMOFUw4VDDhUMGEk6zgDGZQf0UK1oTUVlQfi5tmFMEpHZGF41PGV5yMmo05eewfHYnexZfLYnAhNRZQEWYqyQpu6Y8n2MmprNhzR5w0ByX4yqOR+cqtsHVm1GzMyYG26xIJlsticCE1FlARZir6BNfdDy2OT2H5ZukK3j+1uGSf1PfUsbFnYfWdQtjjjhlVPrFzKWZR8bcWiCv4LYu2o6xXLblZHDJr7U8BujI9cudUGYwrC4DLCHdM1Ts6CB6hgw04P4oFdlS7YjbmBG1OJFgui8GF1FhARZi56BN8aiJsuGkjFJwWBXeMPgxBl9z31BsVuwt6xTh0LUidS+1YANATuThyzqOsfwtdY4XOsc+N4eCle8KdU4XPXbhjqqIjUPOysOPBAvRR0KKYjTmBxIKVV7CBhdRYQEWYueisovPOz9737bZ86Tu4sZpfJ/c9BUuH/0EKQ6dK6lzqbP7ybZ7wuWMdLMraDTC/WVie1Qw3fgRy51ThcxfumKq6Ef58QYgUTsyEFsVszAngCTawkBoLqAgzF52C7667ASZtuidKsNz3VKDs2p1SGDpVcudS7uYHrvPK567/KvTvwIEqurk7hOU3bm+9DkDunCqDcWcCeDMiHVNN3yVn3Bap2qTZmBPAE2xgITUWsG1oV9HwZv4y+En+W5LgzpzDwHXLfU89vOwv41DltuciQGan3LnUBTf4x5zjVwhMPoKyfg5OZncIcd7spesA5M6pUJpjLkJ9RqRjqpx2OHha3HPaFthSJaYuzcacaBDMl8WEQmosYLicJqNT8EXHNqh3XJJ/Jh2YUli0T+p7qrGYgS2VHExCf5usm1jiEzuX2l1cNGmduILff+3T/LmbOhHF8HFw75Xn0OUvdU6FEtxWtnh9Rrhjqs+mgfwdDF+Ujys/Lu4lzcacPJTrHJ0f/xj4sphQSI0FlCamY9GNjo4SnJ6Zf9xkYlHUsUUhE2HVnaz3TmvPOmjaiLsJsEUhE0BvVRIOFUw4VDDhUMGEoypYHFNaGrBaHFnatGGxKeahKlgcU1ocsFocWdq8YbEp5hGnihYECwNWi+PSKkan7eoM4+vEx+3XsZOejPoCKcqol9Wxk1fHPj2clijtgsUBq8WRpeXxpZ9bssTHhQlx+IT69eykY59gmmakXTAIA1ZHX8FH9+7t6QsT6MPHz+rYSU9GDJeijHxBHTv5dezj1ZQRRhUtDFgtjiytGF+aVtEDsGMVLYwpLQ1YLY4sHRlfmgoeAC+4VkT7TlZ/B8eBCh6AXQT7wh9UMBZ2ESy0QJdQwdjYQ3CIm4H+xO4sSCzYF4Hz4cMGdeykJ6NAyjIK+XySYHMz8oe0RKkIXuN0OJ3OzHX0CsbGHlcwwOOJ3VLBsbCLYACWYdRfVKCC1bCL4Ma8ESNHjhRmxdakqHakSGsSFTwAuwgu/TI8K7YmKduRFK1JVPAA7CJ4sXKBF6y8Cy3fi2774otuVxi/Cx9fQMdOejLysinKyMO5XJJg7TsxOjJyB7VEqQre0BSIFqxsR5Jbk9bMn+8LhukP4hPSs9OQzEh0KmZUKy+YkZGMtiNSFewQSHAF0yo6glwr26WKjoIXrGxHoq1Jg7GzYKE1KaodibYmDcJugjME1C9qGSpYwm6COY7z1q+ngjVjN8ECtYkFd0dguvHxBnTspCcjD2tyRqLT7m43J8/XyquMzUimL6glKq7gi+WJBbsjBNz4+DkdO+nJiDE7I9Emyigoz9fKq4zNSMYX0hKlKjgrKyvz2u2JBdMqWiJGFa29oragiuZvlSTWSwWHsZtggKhHLlVfXaGCJewm+ET5sGEVp5SCY7+6QgUPsGkXwTNfCQXrZikF01dXYmNTwUX8R6FiRaxXV95Zv97NhOEYfNigjp10ZRQyK6Mom2JGA1YZlNEAApqOSFVwyQmAYwOeqhz06sre117r84RhPfgEOB076cnIHzQroyibHg8THLTKoIwGwIS0RKkK3pE159bsnYoV9NUVFWxaRUN7w7tRva/RV1cGUTsY+wj+0AXQvR8SQgUnEqxFswWCbwgCsEVUcDxsLbiY/yiEhFDBNhU89QjAoYrEgpN8/8Kmr67EMCgS9epK9AZdGcUhqVdXAPbnVFfnHKBXcGxUBdvnCoaOxqZOSAwVrE1wfM9WP3RHBYdRd0eU4DRsTUpglizB6dWapE2tNsEqmoeaYPletLe3tysC04WPx69jJz0ZuQM40ThOo+jq6mO17q7Iz6fjiHo5LVH6BMutSQ+Xlvq170WxFh1XMKRFFY2HHfvJGkTs1iS3jlKeO61jJz0ZnT+ZoowupCqjS8e1ROkTrGhNSpL6Jw1JJjHv/SxFGf3XfSnKqOUuzB0s6RCcCtaNPQR/vjtFGR17L0UZnWlIUUZtb2PuQLv0JxwqmHBSLziq+x6TkZ7bN51UjWah43hSL1jZfY/ZiM/tm07KRrPQcTxWVNGRzj/MRnxu33RSczCg63isEizf9zQX8bl900nNwYCu40mx4J2rX03ZFSzkxT+3bzopu4IB/3isuoIV9z1NRHxu33RSczCg63gsEKzsvsdkpOf2TSclBwO6jof+DiYcKphwqGDCoYIJh1zBGQN6rr98W27udy4AZF2Tm3vvgCAnALtqfP6EFSxxJ4S044kwUHBnAxt66HYkuHVwEBL8/bkd4NngIu6EkHY8EZC7nZOLZp0CqL++7EWnsK6pTBYsbUJBDddXrnXCyas7xN0cz5Z8vQn5Li1e2I2WnrmrrCmcwqEZN5Z/YNkB6YNkwe3ZJ2DzFLh0TRu8IAgOVW9Egr+Wn79D2iQEnYFnnFA/SdrNUQef5aHrHWDDSrTUBF/mySlcLmqHtjybPVFKsuB3bwUIDutrWAjQIwi+f2FQuoKlTSioYaxXf6EAAAwqSURBVB76enbCO9+UdnOgSns4C5umV5bUoKUAwAhWSqF5dFVV1YRT6lkORUgW3DBHsLj9Nknwg/P4AQsEwdImPui7AH2ois6Qq2j0/0jmSEE3NM+Wl6QUGqdbdSxJQLLg9uxjUFcBl3La4Hkk+IH5woAUgmBpEx+U54K30NbFc/8GnqddktLdVQC1YcFSCp1ZuwA+sfSg8CFZMLxfUngLqlHfLpi2MRM+d6DfR+E/sqRNKGh75Q9XI8GBJ/Jzv/EUKykN1s6pfSwsWEoBPr65cML3rD0qbMgVHMED8HqNxSlYRjoIfrJ08uwzFqdgGekgOK2hggmHCiYcKphwqGDCoYIJhwomHCqYcKhgwqGCCYcKJhwqmHD0CX5+PPrY7tiKPm9Y4ZA4D5uu/BVaU+NwjCjbmSAFSorQJ/gIsgmPjL0boPuK3x89enTeTPTBwk0/L0Uba+48duTuYceNLSdFJ/oEB7/yHwA3vnAdwI6rvGh56QJ+7aks/5i/IsHLANivvGxkKSm60fkd/E//DH3D3V+5BL+4iV8UBa+6Dx5cIQqGrOcNKyIlGXQKfnoS7JoK1fUwdQW/KAjuH/97ODA2xAtmnnUcNrCQFP3oFLz/iq4nfw5rHvRdJfw1JQjenxOE/rF7oeYK5xUjUvDaNUULOgX7M5pmNMMHJf91ZQ+/KAi+1+F0Oq/4AdQsaj3DGVhESjLo/R08/cHhXeDN+Nm3hCVeMPPVN1tbW7eP9NSkqrsKigb0Cn5iFD9+VvmoR4QlXvDbWUE01/+116ngoYRewbscfL9BP3OInXDygv/3UmF22WwqeChBb1USDhVMOFQw4VDBhEMFEw4VTDhUMOFQwYSTrOAMZlB/RQrWhNRWVB+Lm2YUwSkdkYXjU8ZXnIyajTeJh1wWgwtpXAENIVnBLf3xBDsHNjrIK/bcESfNQQm+8mhkvnIrbJ0ZNRtvEg+5LAYX0rgCGoI+wXX3Q4vjU1i+WbqC908tLtkn9T11bOxZWH2nEPa4Y0aVT+xcillU/K0F4gp+y6LtKOtVS25uFOLW/hSgK9Mjd06VwbiyALiMcMdUvaOD4BE62IDzo1hgR7UrZuNN4h2DXBaDC2lcAY1Bn+BTE2HDTRuh4LQouGP0YQi65L6n3qjYXdArxqFrQepcascCgJ7IxZFzHmX9W+gaK/R+fG4MBy/dE+6cKnzuwh1TFR2BGvEZoIMF6KOgRTEbbxL3IBRXsIGFNLCAhqCzis47P3vft9vype/gxmp+ndz3FCwd/gcpDJ0qqXOps/nLt3nC5451sChrN8D8ZmF5VjPc+BHInVOFz124Y6rqRvjzBSFSODETWhSz8SZxj0Eh2MBCGlhAQ9Ap+O66G2DSpnuiBMt9TwXKrpUfmkWnSu5cyt38wHVe+dz1X4X+HThQRTd3h7D8xu2t1wHInVNlMO5MAG9GpGOq6bvkjNsiVZs0G28S9xgUgg0spIEFNASdgt/MXwY/yX9LEtyZcxi4brnvqYeX/WUcqtz2XATI7JQ7l7rgBv+Yc/wKgclHUNbPwcnsDiHOm710HYDcORVKc8xFqM+IdEyV0w4HT4t7TtsCW6rE1KXZeJN48GUxoZDGFdAQdAq+6NgG9Y5L8s+kA1MKi/ZJfU81FjOwpZKDSehvk3UTS3xi51K7i4smrRNX8PuvfZo/d1Mnohg+Du698hy6/KXOqVCC28oWr88Id0z12TSQv4Phi/Jx5cfFvaTZeJN48GUxoZDGFdAQLLrR0VGC0zPzj5tMLIo6tihkIqy6k/Xeae1ZB39tZkniYItCJoDeqiQcKphwqGDCoYIJhwomHFXB4kDd0pDT4gjXqRrnmmIgqoLFgbrFIafFEa4V41x3dWqE7dUayePDCXb7caI9mouM6AKcpC+7caIDfTjRWGekLyDPaREsjgIrDjktjo8rj5Lb/NJLbp9GQgGtkTwcTjAbxIrGCfYBTjCDdYxB885IIHxGtAsWh5wWR7iWx7mmgqOxtWAQhpyOvoJpFR2NvatoYchpcYRrxTjXVLACuwoWBuqWhpwWR7iOjHNNBSuwk2Bf+CM+VLACOwkWWqBLqGBCBYe4GRzHdRYkFuz1aCTIaI3kYXGC/RxOdEBzkRFewEnaG8CJDvpxovHOSFCeUxG8xsl3o5K5LrHgy10aYfu0RvL4cII9AZxor+YiIy4DTtLdHpzogBsnGuuMuFl5TkUwwOOJ3QrQKlqBfapoHpZh1F9UoIJjYSfBjXkjRo4cKcyKjQ1RzQyRxoY0EFwroCXaToJLvwzPio0NymaG9GpsIFPwYuUCL1h5kzJtblXWRtASbifBG5oC0YKVzQxyY8Pjt9zC9GsEtAbqwLS0FYKNTzwlZ0RVsNh1e4IruOvChcvdGuHcWiN5GJxgbwAr2qU9ViFYS7jLi1MQ1oMTjXVGPKw8pyo4Cl6wspkhbRobCK6ilQiNDVHNDOnS2FA7gAThdhKcIaB+UcuQKnigWuIEcxznrV9PBRMrWKCWCiZa8MXyxII1PyY0dJ7JYrRExRacYCcG63kva5/JysrKyrx2O72Cib2CXS6XO7FeKjgKOwkG6OujggkWfKJ82LCKU4oVKq+uECg4tlstmu0keOYroWDdLKXgtHl1JT0EF/EfhUrBMV5dOdTQ0OvWSJDRGsnD4gT7OaxoT4KAhIJV9/T4cQrCmXdGmKA8pyq45ATAMeVTlbFeXfn3FSs8fo2EWK2RPEGcYC6EFR1IEJBQsOqeAQ6nICGsaKwzwobPiKrgHVlzbs0eMPQv8a+uJFRLUBUN7Q3vRvW+lg6vrqST4A9dAN37FSvS4dWVdBJ8QxCALYKEkCJYo1tyBBfzH4WQECpYgZ0ETz2CfgVVJBbcp5WgT3MoIoATzHA40X53rLU4glWTdjM4BeGwonWeEVXB+3Oqq3MOJBbc49II59EaycPgBPtYrOgBRcZRqyBW0r0+nIKwXpxorDPiDZ8RVcHQ0djUCYmxfxWtU3CspO1URWvF1oJ1uk0DwbZvbEhGrbpmYgTburHBKLckC7bfrUpDrapaJkaw3Njw3JIlPk4j/UGtkTyhGOuEkxl77RAjVWdElWC/PJfcFXyipcXVoxHOqzWSx48T7GNxoplejOBewEm6z4cTbeIZ8XLynD7BpDY2DIbcF8DjE2ls0MwjBzF30E7zGtOS9t4SNC3tlXtMS3rvihgrTe4v+r4PTUt6u9YORfDxlJon+JGdiWN08v8eirHSZMHr/pA4RicfvGxa0r4lA0ecNY5fmlenHXo2xkra4zvhUMGEY6LgqH56jEd6ZNsETBy6wKxCq59rEwUr++kxAfGRbRMwrcRgXqHVz7WpVXSklw8TEB/ZNgHTSgwmFlr1XJsuWL7BaTjiI9smYFqJwcRCq55rswTvXP2qmVewkDz/yLYJmHkFg1mFtu4KVtzgNBbxkW0TMK3EYGKhVc+1mYKV/fQYj/TItgmYVWIwsdCq55r+DiYcKphwqGDCoYIJhwomHHIFZwwYmuDybbm537kAkHVNbu69A4KcAOyq8fkTVrDEnRDSjifCQMGdDWzooduR4NbBQUjw9+d2gGeDi7gTQtrxREDudk4umnUKoP76shedwrqmMlmwtAkFNVxfudYJJ6/uEHdzPFvy9Sbku7R4YTdaeuausqZwCodm3Fj+gWUHpA+SBbdnn4DNU+DSNW3wgiA4VL0RCf5afv4OaZMQdAaecUL9JGk3Rx18loeud4ANK9FSE3yZJ6dwuagd2vL8cfIcgpAs+N1bAYLD+hoWAvQIgu9fGJSuYGkTCmqYh76enfDON6XdHKjSHs7CpumVJTVoKQAwgpVSaB5dVVU14ZR6lkMRkgU3zBEsbr9NEvzgPH5ECkGwtIkP+i5AH6qiM+QqGv0/kjlS0A3Ns+UlKYXG6VYdSxKQLLg9+xjUVcClnDZ4Hgl+YL4w4oggWNrEB+W54C20dfHcv4HnaZekdHcVQG1YsJRCZ9YugE8sPSh8SBYM75cU3oJq1LcLpm3MhM8d6PdR+I8saRMK2l75w9VIcOCJ/NxvPMVKSoO1c2ofCwuWUoCPby6c8D1rjwobcgVH8AC8XmNxCpaRDoKfLJ08+4zFKVhGOghOa6hgwqGCCYcKJhwqmHCoYMKhggmHCiYcKphwqGDCoYIJ5/8DQoC8Cws52xgAAAAASUVORK5CYII=\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "for (treat in c('METTL14','METTL3','Virma','WTAP')){\n",
    "    res <- all_res[[treat]]\n",
    "    d <- res$log2FoldChange[!is.na(res$log2FoldChange)]\n",
    "    w <- wilcox.test(res$log2FoldChange,mu = 0, alternative = \"less\")\n",
    "    \n",
    "    t <- t.test(res$log2FoldChange, mu=0)\n",
    "\n",
    "    print (treat)\n",
    "    print (w)\n",
    "    print (t)\n",
    "    \n",
    "    hist[[treat]] = ggplot(res, aes(x=log2FoldChange)) +\n",
    "            ggtitle(\n",
    "                treat, \n",
    "                sprintf(\"wilcox.test (p.value): %.5f \\t t.test (p.value):%.5f\", w$p.value, t$p.value)\n",
    "            ) +\n",
    "            ylim(c(-10,1500)) +\n",
    "            geom_histogram(binwidth=0.3)\n",
    "}\n",
    "h = hist$METTL14 / hist$METTL3 / hist$Virma / hist$WTAP\n",
    "ggsave(\"plots/peak_Histograms.png\", plot = h, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/peak_Histograms.pdf\", plot = h, device = 'pdf', dpi = 300)\n",
    "h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pairwise correlation of each treatments \n",
    "http://www.sthda.com/english/wiki/correlation-matrix-a-quick-start-guide-to-analyze-format-and-visualize-a-correlation-matrix-using-r-software"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Hmisc)\n",
    "library(corrplot)\n",
    "library(pheatmap)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "METTL14  METTL3   Virma    WTAP \n",
      "      2       3       2       2 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "all_fc = data.frame(\n",
    "    METTL14 = all_res$METTL14$log2FoldChange,\n",
    "    METTL3 = all_res$METTL3$log2FoldChange,\n",
    "    Virma = all_res$Virma$log2FoldChange,\n",
    "    WTAP = all_res$WTAP$log2FoldChange,\n",
    "    row.names = rownames(all_res[[1]])\n",
    ")\n",
    "colSums(is.na(all_fc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "png \n",
      "  2 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "res2 <- rcorr(as.matrix(all_fc))\n",
    "# # Extract the correlation coefficients\n",
    "# res2$r\n",
    "# # Extract p-values\n",
    "# res2$P\n",
    "\n",
    "pdf(\"plots/Corrplot.pdf\", width=7, height=7)\n",
    "c = corrplot(res2$r, type=\"upper\", order=\"hclust\", method=\"pie\",\n",
    "         p.mat = res2$P, sig.level = 0.01, insig = \"blank\")\n",
    "dev.off()\n",
    "png(\"plots/Corrplot.png\")\n",
    "c = corrplot(res2$r, type=\"upper\", order=\"hclust\", method=\"pie\",\n",
    "         p.mat = res2$P, sig.level = 0.01, insig = \"blank\")\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nOydZ1wUV9fAz53Zzi4ssPS2gDRBERALIoLdRI3GxBprTGKq6TFFE42J6cYkT55UH0VNURNjNCYajcbexY4N6b2zvcy8H+BFVLayO7PA/f/mwzJz594zzO6ZO+eegmiaBgwGg8G4HgTbAmAwGAymfbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCxmAwGBcFK2gMBoNxUbCCdjHkcqioYFsIDAbjEnDYFqB7M3r0nXvKymDaNODx4K+/2BAIg8G4EFhBs8r+/ZCSAvfee2vPoUOQlgZiMXsyYTAYVwHRNM22DN2Y/Hx4+mlACFatgvBwAAB/f8jJAX9/tiXDYDDsg23QrCKXw7ZtMG8ejB4NS5eCRsO2QBgMxoXACtoFmDABTp8GlQr69IGmJralwWAwrgI2cbgSly7B/v0wezYIhWyLgsFg2AcraAwGg3FRsInD9TAYYNEitoXAMMKuXfDll1BVBRoNvPMOPPww/Pgj2zJhXAg8g3Y9NBoQCgHfly7PJ5/Ahx+CmxuQJIwdCyoVUBSsWwcffQRPPMG2cBiXAPtBs8qCBe3sNBoZlwPDBl98AVu2QGQk+PtDaipMnQoAMGwYLFuGFTSmGWziYJXvvweFAjicOzdMd6CkBGJjwccHvLxg0KCWncOGQV4eq2JhXAisC1glPh6mToWxY2/bqdHAV1+xJBCGQWQyqK8HqRQeeQSk0padDQ0gErEqFsaFwDNoVpk7F7TaO3dyOPD662xIg2GWCROgqAgA4N13QSJp2bl9O/Trx6JQGJcCLxJiMK5EUxMQBLi5sS0HxiXAM2jXQ6+HOXPYFgLDEgIBPPkk20JgXAU8g3Y9sJtddwbffUwb8Awag8FgXBSsoDEYDMZFwW52rLJ+fTs79XrG5XAB1Gq4ehWqq2+93Q8fzqpAzgfffYwlsA2aVRISTB66cIFBOdjm339h8mQQCqG0FIKCID8fEhLg/Hm2xXIy3fzu19TAa6/B2bMQEwMrVkBgINsCuSJYQWNcgKQkeOIJeOQRkMshPx9++w2OHYMVK9gWC+NMpk2Dq1dh8mTYsQO4XNi9m22BXBGsoFll/XqYPh2Ibr8SIJFAfT2QJISEtMRujBkDf/7JtlhOppvffZkMjh2DyEhoagIfH2hqAi6XbZlcju765XARZs4EnY5tIVwADw+oqQEA8PODCxdAre4W+Si6+d2vqYGwMAAAiQSkUqitZVsgVwQvEmJcgDlzYM8emDYNnnkGBg8GoRCmT2dbJozz+eGHlhcIjQY2bwYPj5b9Dz3EolAuBTZxsApCLQa4u+nyPgymuHYNGhshJYVtOZxPN7/73XyN1DqwgmYVhMDbu/1D1dXMioJhHHz3MZbACppVEAK1GgQCtuXAsAG+++1iMOCU6K3gfwTbUBTbErgASiV89x2cOQMKxa2dmzezJxBT4LvflrNnITsbNmyA8nK2RXEVsIJmG50OJ2iHqVOhrg7uv7/b/Svw3QeAigr44QdYuxbOnoXBg+Gzz9gWyIXACppt8BsuAOzdCxUV3TEJcne++1ot/P47rF0LO3dCZCRMnw4FBbBxI/j7sy2ZC4EVNNvs3dt91/FbSUmBmzfNLet3Vbrz3ff3B4RgzhxYurTFaefLL9mWyeXACpptZs5sf3+3Wsdfvx6mToWsrNtmT089xZ5ATNGd735iIhw4AMeOQY8eEBYGMhnbArki2IuDVfA6fjMvvACrV8PgwbcZZH/6iT2BGAHf/fx8yM6G7GwoKICRI+Hff+HSJQgNZVssFwIraFbBP9FmxGK4cAHkcrblYJZufvfbpiI5eBDWroWNG4EkYe5c+PhjtoVzFbCCZpVu/hNtJToajhwxGbXRVenmd//uy1er4bffYM0a2LmTPbFcC6ygWUWj6b6/z7Z8+SX89hu8/PJtNuguv2bYze9+N38+WQdW0Kwydixs2dKyjv/227BwIbi7g04H/fpBTg7bwjFIu55VXT5aoZvf/W6eisQ6sIJmlbaTCJkMcnIgOLjb1XXW66FPH7hwARBiWxRm6eZ3H6cisQLsZodhGy4XeDzQ64HHY1sUDLMUF2MTh3mwgsa4AA8/DJMmwTPPtAQvNNPlbdAYjCWwgmabH39sMcNptbBlC3h6dse6zsuXA9wVtdHlbdCA7z7GAtgGzSo4Z3l3ppvf/TVrYORIXMzbPFhBYzAYNoiKguvXITISMjJgyBDIyIDwcLZlcjmwgmYVM95UffowKAdLnDwJYWHg4wMnT7ZztG9fxgVilm5+9wGgrAwOHID9++HAAbhwAQIDW5T1o4+yLZmrgBU0qyAEPF777mUaDePSMI5cDkuWwLx57Qd55+czLA7TdPO7fwe1tfDtt/DJJ1BZ2S28DK0DLxKySno6lJTAjBkwaxZERbEtDePk54O/P1y8CFu3QmIi29IwTje/+wBgNEJODuzfD/v3w+HDEBgIkydDejrbYrkQeAbNNnl5kJ0N69eDjw/MmgVTpoCXF9syMcjWrbBuHWzfDrGxMHs2TJ8Ofn5sy8Qg3fnujxkDZ85ATAykp8PgwZCWBu7ubMvkcmAF7TIcPAjffw8//QT79kH//mxLwyx1dbBxI2Rnw4kTMHIkzJ4N48cDn8+2WAzSDe9+VBTo9ZCVBYMHw+DB3fQdwhJYQbsGx4/DunWweTOkpMAXX3S7xJut5OXBunXw7begVEJdHdvSMEW3vfttFwmrqmDQoBZlnZTEtmSuArZBs0p+PmzYAOvWgVgMs2bBuXPg48O2TOyhVsPx43DsGFRWQloa29I4H3z3AwJg8mSYPBng/xcJ33kHLxK2Bc+gWYUgICoKHnoI4uLuPPTAA2wIxAY0DQcOQHY2bNrUYoqdNatbzCLx3W+7SHjwIDQ2QkoKZGTAe++xLZmrgBU0q3TzWDIAWLIE1q2D2lqYPBlmz+5eK/jd/O6PGQOHD4NeD/37twSqDBwIQiHbYrkWWEFjWGX0aJg9GyZMwL/MbseKFZCRAampOIuhGbCCxmAwGBeFYFsADAaD6VJMnDhRKpU+4IiFBKygMRgMxpEsXLjwf//7n0O6wgoag8FgHElmZqZEInFIV9gP+hZGo7E7W+QpiiKI7vvA7uaXT9M06m41IdtAkmTby79ZWGKqJU3TIj6Xz+dJJBIOx+n6EyvoW5w4ccJRz71OB0VRtbW1MpmMbUHYQa/XNzY2epuqYdrVUavVGo3G09OTbUHYQavV+vn5BQUFNf+Z/fPWOXNmI0H7/w1aqwjwcR/UP+Wpp57KyMhwtmxYQd+Cz+fHx8ezLQU7GAyGY8eOddvLb2hoyM3N7baXX1JSUlFR0W0vv6qqSqvVtv7Z0KREHmFkQEq7jenG4uievhs3rmdGNqygMRgM5hYEQSBACLVv76IRIgjmbEFYQWMwGMwttDoDIGi/kAIAAFLrDOZ7GDVq1OnTp5VKZXBw8LZt25I6kPsJK2gMBoO5hbtYCIhARPu6ERGkp7ub+R527tzpKGGwgsZgMJhbNCo1gBCYsmMgVNegYkwYrKBZQK83HDx66t/Dx86eu1hcXNzQUKfVaDVqJUGQfIFQIBR6y3xCQ4L7JvcZmjEwqdddqc46ORRNVyl05Y2a8iZtvVqvNVAqneHf33++cPBvgUAocnPzCwgMCQ7qEx+X1jcxMjSYbXkdjJGi6zX6GqWuVq1XaA16I601Gjd//5+rZ47z+QKxu7t/QGBoSHBifM8ByYmBfl0tAamBoquV2iqFrlqpbdIadEZaazD+8Nl7FQXXeVyeh6eXj69veFhYcmLCgKTefjIW6ssIBVyEEEGQ7R6lCULsxlwpCaygGcJopH7asmPdjxvPnDxRVVOHBB6I7wF8d8QVATcU8TjG+vNI5ENw/Wmt9ka++ti1C5v+PEQvXk7rlMFBgemDhzw276EhaZ210DVF03k1qtyKpmtVCgNF+4r5fu4Cfwk/1lcs4pF8Dnl2Q1H46FkegXJtU726ofZ0VfneH35/84NPjU01/kHBWUOHzXpwQs/IMLavw06MFF3coCmsV5U0aGgATyHXS8T1cePJPUUCDsHjENkFl4fMf0Uo8VDW1Sjqa3IrSg5s+GXxO+/rFfUhYeEjRoyYNnFcWJA/29dhJ3ojdbNWdaNGWVSvJhGSufFkYl6Qh1DM5wg4BJ9Dfn3zYvoTK2iaUtXXKutrTpYU7zq2vqrgGmiV4ZFRo0aNnjFxbIAvQ06QBgMFgMDEIiEAMhiZi5bACtrp/LFr37sfrjqfc0oJboR7MJL25Xhx726GEIkILpA8RPKAJ2n7flWuV2/cfebnX39359GZw0Ysff2F3j1jGJO/gxTWqY/k1+bVKHvI3GL9JCNifPmcdr76JIG4Qje+mzvfzd3dP7RtXUJlTcWhc4d/nj6Po1ffO/6+lxbMC/LvNPPKkgbNhYrGKoUuVCoM9RQODPPitPfuTCAkFLuLJB4iiYdPaARAauuhmpKCoycPrn1wOg+oBydPfnzODG+pB4NXYD80DXm1ytPF9bUqXQ+ZOM5XMjLal2zv8kkCCcTuACCUSL1DIqBXamsXNcU3/zx58MtvvpW6CadMnfrE7GkSN5FTxTZQFEImvTgQQgaD0akCtAUraGdB0/TSD7748j//qVEYCK8oFDyMBHu9c7hCwjMSPCOVlHHboWtbBw6XhwSuePvNKRPvcajIjoSm4VRx/d5rVX4S/kC51+SkILtdk9y8/WKyJsZkTdSrlWeO7+k3/N6oqB7LX385vW8fR0rsUCiavljRdLa00V/CTwxw95cI7O7KOyjMOyhswH0zVI31J/f+0TdjeFJy8rJXX0qI6eFAgR2L3kidKKrPKW0I9xJlRMj8JPbaBBDyDonwDolInTBLUVu1a/+Ozz9PG5yRufz1F51n+5IIBYAIRLZv4kAEKXGz/27aSveNbXUqS1Z85hUY+fbK7+rd+5BhGUgSAHZr57YQJOEZzokYUWQMnD5/YVBU4qbfHbZe7ChogKP5te/vuVrWqHkyPWJWamiUj9ghjqNcoVvUkPFj3vjao++YWU+9NGD0xMNnXC6xPUXTZ0oa1p8uVuuNkxMDh0f5dEQ7t0XkLh1w34zHVv0gSxz84JxH7pkyKzcv3yE9OxADRe/Pq/72WAGJ4NEB8jGxfvZr59sRe/mkTpg948N1hoCYoeMmTZzzWEFJuUN6voNGtRYBEAi1uyEEjSqt5V4cBJ5BO5i/9x2eNfeRSiUi/QaQHGc9aZHAkwwdXKlpmDp7wfKYyK2bsuUhgU4ayyaK6tUbz5RE+4qfy+zRrinDIfhEJgx99oOa/Nwp8xakJiV+t3KFl4e7k8ayieIG9b95NdEy8fSk4HZNGQ4hok//iD7988+fvHfStJEjhn+y7HWhgLk5nRmuVil2X6tKDZEuGCgnnJPZAyEiMjUjMjXjxon9g0aMmTp16opXX+ByHanHJEI+IAQmFgkBERIRnkF3QnR6/ahJs0eNn1wtiCGD+oHTtHMrSOBByrMuVZMRcX0WLf3I2cOZx0DRG3NKtp4vm5UaMi7e33nauRVveeyoVz5vcA/rM3DIml//cPZw5tEZqT9zK06XNNwX758aInWedm5F3qvv/I/XVFLCPgMydu4/4uzhzKPUGTecLrpY3jgnNTQ1xNNJ2rktkakZ01esOV1Y3WtgxqHT5xzYs0qrR2ZRafUOHM48WEE7hjPnc4MiEvYcy+VEjkACKZNDI0kQJ3LUh59/l5CaWd/QxOTQrVQptCv3XQ+RCp8aHOEjZs4JCQDCB4wY8vzK5R9/MXHuEzo9c7+ctlQqtD+fLYmSicf39BfzmHsrRYhIvffBSa9/8uyi15969S22cjHm16rWnihIk3tP7BUo4pqYeDoBgsPpN3HOkMfemDHvsdffX+mobhEgAASINLERTKb9wwraAXz2zbqUgRn1bnGET7xjbM22QnDJkLTcaiI8NvHw8TMMD36muGH1scKZfUMGylnwWgUAvtgj8+l3qzhevdOG5hWXMjz6ubLGf65X3xcf0ENmIcDMSbjL/GYt/+paVVPayLE19Q0Mj74/r3p/XvXs1LBwL+c6V5jCMzBs8rKvdx87N3TCFJVa0/EOBXwuQgRBku1uiCAEvHa8sJwEVtAd5fnX3332pVc5ESOQkB311AohlSukfdKHjtq09S/GBt13vfpoQe1zmZH+7iybQWOHTYq579HMkfceP3eJsUEP5deWNGgmJwa689lczkEIZc1YEDf8/kFDR+YVFjMzKE3D7xfL1HrjzL6hbjzmJs53Q5Bk1sMviqP79ssaWV5V08HeVBodIIDmeXR7m0Ktc4TUVoEVdIeYPv/ZVf/9nhM+DEhG3+tNgQRSjnzolBlzvlz9IwPDbbtYXlineixNziNd4ovk0yNhwKNLxz0wdffhk84ei6Zh19UqA0WPifVlwORqDdH9MkYueHXU2PvOXbnu7LGMFP1TTrG3iDcqxs8lLh6g55B7ek+YN3jE6PySso70IxG1uNm1vxGEVMJcBXqX+F11UuY88fLPv/5OhmYAYnP6cCdcESdi+FMLn/9m7c9OHWf7xXKN3jgzNdRF1FMz7v4hQ5/9YMach/cdO+3Ugf6+VikVcoZEuFaO/8Co+PEvrpg0eerl63nOG4Wm4eezJTE+4kHhrnX5ob1SB856fsS944vLKuzupEmtBYQAEaY2Jt3ssIK2k0VvfbDux81k6BDTaQnZg+RxwocuePKZ7bv2OWmE/TeqK5u0D/SxP/zEebh5+Q1d+P6UmXPPX3WWkjqYXyvgkv1CXLEEiU9I+Ljnlk94YGpZRZWThth+qTxUKkwOZnQx3EoCY3oPeOjZ4eMmNjQp7OuBy+UghAgTIIR4zq901QpW0Pbw9dqfP/hkFRk62BW1czMknwwbMv2hWRdyHf+2m1PScKGsaU7/UFe9eHDz9kt/5I3xkyZX1zl+0exsWaNCa8hwscljW3zDIrPmPT964oMarePnev9cr+JziHQXvvyguD7xY6aOmDiFssuthSQI814cBIMGPaygbebq9fzHn3yaEzbEpCu7a4B4YrVn4tBRYx3rfFar0u3MrZw/MMylLBt34xnSI+beOfdMnunYbquU2tzKppHRrp4MRN4rJSbjnhmPPu3YbvNqlMX16hExvo7t1uFEDRgmCo5a8NJiO87V6oyAwIwftMZSwn4HghW0bVAUlTlyLBnYn4E4lI6DRD61lOfoiQ5TUkaKXnOscGZqiIusCponuM8gvcTv+WUfOqpDnZH660rl6BhXWRU0T/KoiUU19Z+v2eCoDhVaw5+5FQ8kuqJd624GPDB/36Gjm3b8beuJ7m58hAhEctrfCFIqxouErsqUec9UagTIzdUnUK0Qsrh/Dx37349bHNLbb+fL+ss9A9n2qLOepEmP/fTzxiM5Fx3S299Xq9LCvDwEzLnBdpBxT73x0Ycf33SQb/gv50rHxwcwGYrSIRAa88xbr7zyiq2+4Y0qLQACZGIDqFM4wNvaSrCCtoFTZy/+8ssvhG8C24LYBhHU/7nnnuu4ObK0UVPcoHa1hXvzECSZNnfRvEcXdDzKrrhBTQNEerMTjWIfXL5wzGMvz13wVMe7OlfW4CPmh0iZmzx2HKG7Z5/xs+cvfMmmswR8s4uEBOHG4BMaK2gbmDRtDhnY13QmbxcFcUUKTsDMR5/rYD8/ny6Z0ifIISIxiTQ4gh8U896X/+tIJxRN77tRkxnZmR5Ozch7pSgpcuP2DmU91Bqog3k1w6M6zYtjK9Fpwy/kXtt31Aa/eIORBoRMJLMjEEJGirmQ+k6ma1jky//9VFRRj0Sd7zsKAIQs5pdffskrsD/G7GRRvdxLxHq4oH0k3jf3v//5XNmBOOAzJQ1xvmIm82w4kFHzn1+yZElH3iH23ageHCHjOT8BljPIevjF515aZH17o5FuqajS7gaE3oAVtOvx1ltLCX/XzRBvAUQQPvHzn3zRvrNpgL3XqkbFuvravSm4AlFw/1FvfPCZfadTNH2psqlPYOeoY3I3Yk9ZUHzfL9faGVyqMRjza5UJAS6R0NUOvILklMjjF6tXCyVCHkIEQXDa3RBBujNYkxAraKv4fv0vNUoK8SRsC2I/hDRs3769JXZFWJ0taeghcxOxmm+hg8QMnfjrzz9qdfZkUThX1tjTV9JurabOwqAH537xuZ3Pp0M3a9Pk3p344gEGTV3wznvvW9lYodEDAiBQuxsCaFDhXBwuxtLl7xI+8WxL0UEQ4R399Mtv2XHmP1erhru866t5SC7Pt/egj7+12eeMpuF8eWNip50+NyOSeHiHx276w2afM72RulqlSPDvrNPnZtx9AzXAPXTqrDWNRQIOQggRZLsbkIREyHO2wK1gBW2Zi1euF5dVMJzl2RkQ0og9O3fYaousaNJ6CLkSVrO1OYSYofdnr1lt61mF9epgDyEDCfidzYD7Znzx5Ze2nnWpoinB370zuH1bIHnstA8/+481LTU6ChACgmh/Q4RGx1zRWKygLbNk+ceEp+sW6LQBglTQbus3/W7TSYdu1rCV6NmxCNw9tQT/zKWrNp11saIx3q8Tm7ZakQXLS8rKK2tqbTrrTElDYmDnnj43ExyfcurkSauiahEy78Vxd4KH7du3R0REyOXy7OzsOw6tXLkyOjq6R48ezz1njxsVVtCW2btnF+ERwrYUjoHwDF/1xdfWt6dpyK1QxPqJnScSk0QOGv3Zd3f+hMygN9J1ar0vszVinEf84JHfbthoffsmrQEA3DtPYI55Qnv3X7/Fcmk0IZc0myyJEN7uzGMwGBYuXLh3796TJ08uXry4sbGx9VB5eflnn3129uzZ3Nzc/fv3nzxpcxZcrKAtkHPxSoPKAEQX+Y4iN79rV2wIqyuqV4d5CTtFZLM1hPRJP7x/n/Xti+pVck92CoU4g/j0ETv+sKF445VKRc8u8fbQTFT/rF+3Wn59VOkMZrL1Q/MqYhuOHz8eHR0dFhYmk8kyMzP37NnTeoimaaPRqNVq9Xo9TdPe3jb70WMFbYFvVm9Aks4XnWEGhYF74MgpKxvnVjTF+nadnyjJ46uNqKLa2tf8gnp1mGdnip0zj7vMr7SsXK+3NtdPXo0ykqU6Xs7Av0fPy+ctrxO6CXjmcnEggqTUeXl5Gk2LW31paWlQUIuKCA4OLikpae0qICDg2WefDQ0NDQwMvO+++8LDw22VGStoC/yzdy+SBLAthSNB4sA1Vr/nXqlSxPh2EftGM36xSb/s2G1l49JGTYCkU8bmmCIoOn7vkeNWNq5R6bxFzHksOB2EeB4+V24WmG+l1BgATNqgKW3T5TPH3n///dOnWypC0DTdtoxs288NDQ1bt27Ny8srKiravn37qVPWToxawQraAuWlxYjfuV2s7gCJZIePHrOysUpnFHd+/422+PZI2HfoiDUt9UaKS6BO7f58N8Gxvf85aNXl13Yx7QwAAIGxiTv+OWC+DYeDEEKIINrdSJFnr4HDv/7667S0tOb2QUFBrbPmkpKSgIBb87n9+/eHhITIZDKxWDx06NBjx6z93bWCFbQ5GhqbGlWOTKbsCiCBR0VpkTUt69V6T2EXMb63Iouw6j0XAGpUem+3rqahgmN6nT1jVTGwKoXOT9JFVkdb8e/R88yZHPNtOARhJpkdQkDenmu3X79+V65cKSwsrKmp2bt377BhwwBgy5YtNE0HBwcfP35coVBotdr9+/dHR0fbKrAzFXRCAhAEFLR5oUhPB4QgJ6flaNuLHjsWNBpz/5U7tvXrISEB1q+/c9BVqyApCbhcWLCgHZHy80EkgtGjrbyC/UdOIH5X8DG6HVTfpLLGG7q8Sevv3tV+olyhm1JpVTGkWpXOi8GQBGbwCggpKrTwjt9MlVIrYzCmmRm8gsOvXsk130ZrMAKAGeWj1t7mB83hcFauXJmZmZmcnLx06VIPDw8AmDZtmtFoTEpKmjp1amJiYnx8fFpa2vDhw20V2Mkz6KioWzo0Lw9KSoDX5hu/ejWo1S3bli0gENz609sbduxo+axUtrNz+vT2RwwOhmXLYMqU9o8+/TSkpFgv/pHjOYjfdZbIWkEcYe41y/X6yhs1fl3LAtuM1gjW+MPWqnReoq72AkGQpEpjVaRylULrI+5qzyc3qXdttYVSjWIB17SFAxEISe96rxo/fnxeXl5BQcGcOXOa92g0Gg6HAwDLli27cePG9evXP/74YzsEdrKCfughWLeu5XN2Njz00G0+3lwuCAQtG5cLALf+bHtUJGpnJ2FC8kmTYNw4kLYX9ffbb4AQ3Huv9eLfLCgAbtfxsroFV3jRilqF9Wq9Z5fTUADAl0gLSy3nJFHojF0gfvJuEEEajZZj4Rq1BndBF7x8jYEy36DFi87kDLor5eKIiwOxGI4fBwBYvx5mOrhAnA0olfDKK/DppzadVFtTgzpDaStbQRxBSVm5xWZaAyXgdOIESaYQeniVlltW0FoD1UkTbJpH5O5RUVVjsZnOQHWKwmYOh88hkYkwb4IAhEDIYNYw59+AWbNg3To4eBB8fOAOG/njj4NM1rK9/bZzxVi2DKZPh4gIm05SKlVAdMFJBBDc2to6i600BqOA2wV/ohy+sL5JabGZ3tg1NRRPKGpQWLbC0wBdJkCpLSSXaz6pIQW0uUhvAlmYgTsU52ufadOgd29QKGDWrDsPvfsuTJzY8tndmWtxly7Bb7/BWavW7tui1WRFqG8AACAASURBVOkAMfW0RMTdMf5OGwspVZaz12v1FJ8pDUXTNAKGLp8gOdYk79dTFNeUJc3REASh1zJU6Y7kcJTqjtY/czg0TSFGahVxuHyFUs3nmTSv640UgLnvYnMDZnC+gvbxgZQU+OknuNtG7ukJwcFOFwAA9u+H4mKQywEAlErQ6UAuh/x8i+cxOYMgfBOYLKbFsULzcghkYKq6T319PU/EWEQMTVqheQmEKJomGfkOTJky5YtfN4ya/wIDYwGAqzl3x8XFVd286hsRy8BYFGXkcc3pPTGf11xQpV0QYjQ5CSMa4YsvYN8+8LorI5peDxpNy2ZXJvXbemhelzcYQKMBoxGMRtBowGAAAJg9G27cgJwcyMmBhQth0CCwzmPczc0NKGvjYjsKk6UOKYOnp+XsqXwuobW0ouIoKisrBe6ezIyl16g9PSw75/BIQsfUXOn+++9HZVebai04GDgEnVrl5eFazqNZWVkF56yNb+wgRr3OTWQufF+pNSAAUyWvEKAGDXOxEYwoBbkc+vdvZ/+8eSAUtmz9+tnTc9seHnkEAOCtt0AohK++gu++A6EQ3ngDAEAoBH//lk0sBh4P/Pys6d7d3R2MzK3YMgZt1Mu8LWcQFXBIjYGh1Lc6nY7gMDQx0WtU3u36+dwOkwoaIfTKK6/s+8GGRIN2o1UpvaSWFTST709ZWVmF508wMxZlNBJm35+EfALMJBslkJjP3AzamSaOCxfa2fn/GUbaP9pKdbXlne32sHw5LF9urudFi2CRtRUkAwMDaf0l13obdAh6VY9IucVWIh6p1DKkoK9WKaOYGQlAVVcVEWbZtibgEmo95cGUF8+oUaM++eSTurJizwDn2v10GrVEbNmaJOSSKp2RGU87mUwWwDPQFIWcb/QXWPLM0RloBObWg7RMzVoAh3qbJ6FnNK23vNzf6aD1qpTePS028xXzKxRMrCZ1pOC0HRi0GqnEcpI2TyG3Ts3o+9OSJUv+WW9z0RNb4ZNWTTm8RbxqJXNricnJyeXXbUiEax9GvY5renmwGRqai3qbnEUz+WXFCtocmYP6gaaBbSkcj7uIK3azHIDj7y6oaGLCtaC+vp7vxpBVlDIa3awLP/ES8WqZzcQyaNCgAI62suCG84ZQNtR5SK2y9fuIedVK5p5PzJiha0sL5BGR5tsIuYT5gipdyw+6M9MzpgetV7EthaMx6kRuVsWv+0n4FU1MzKGqq6sFEoZKPjZWFIXIrXKH9xbxahiMGWtm2bJle7I/d17/FTevxsRZfnkCAB8xv4rBGfSQIUOKLthccMRWqgtvxPW0cPkqnYWFB5WWKccBrKAt4iPzAmOXSmhHq6rj4hOsackhkN5IU863P1RUVAg8GCp7WHX9fGpqqjUtJXxOo4a5n2Izffr06ennXnrtkpP6L75yvn8/qy7fT8wva2TINRsApFJpsAgZDc79rZXmns0alGa+jRufRAhM5eJABBJ3NTe7zkyf5L600nJYcCeCUlbcN3aMlY3DPIWFdWqnygMAVVVVQqZ87CqunB0/MsvKxhI+h0mfqmaWLl26J/sLJ3Wef/7UfSOHWtOSSxIAwJgfCwD079+/7Op5pw5RmZc7bJCF55NKZwRzqTiaa2IxBFbQFnhw4nhKUca2FI6EVlTMmnKflY1j/SSXK5qcKg8AVFdX85kycdQVXR+Ukmhl4zBPYWG9059PdxAdHT0gJjT/vM3VNyxiNOhpTVOgn4+V7eVebvm1zJn4srKyCp1phtY0NXhJ3QV8CzlUSQKQaQAha6KcHAVW0BaYPW0CraxkWwrHYVD7+8qkVscpRPuIc52voCsrK4WMmDhqC6/Fxccjq4MD5Z6imwxqqFbefPNNZ1ii83KO9UvLsL59lMztapVVubMdwuDBg4svWlVMwD7yTh3IGjbCYrOWhP0mFwkRl8NggDFjI3VSuBxObGwcrba2zKiLQzUUjh071vr2Ih7J4xC1Tl4rY2yRMP/4P1MeuN/69lIhV6E1MPma30xoaOg9g/pePWGhOJOtXDyw66HJEy23axXDU1RYp2JgEaIZsVgc7ikw6Jy1Mpl78O9HZ5rIFN+GluhZ03W9m20gzIAVtGWefPwRqtZy9uROAVWX9+aihTadMiDM61iB5dR3HaGiokLo7vwZNE2XnDs878HxNp0U4yO+wuAsspU33njj3x+/caCHuEGnrSu6MWKwhSWytiCAHjLx9WrmQgEGDhxYctlCSSr7UDfWCZAxNkJusaVYwDGXsJ8ADwbrwGEFbZnH504TQxNQzD02nQStqYsIlwcFWBXm3kpikMfZkganTqJqamoYsEGXXjjef2CamTRm7dLTT3LJ+Uaeu/Hx8Zl6z7CLB3Y5qsPz//41ZqxtDycASA6Wniqud5QMFsnKyio87xQz9Lm/f5tmqgzT7Si0BjPTZwBoYtC3BytoyxAEMXLMWKrOieEDzEBVXlz82iu2nsUhULSv+FyJEwN2tFotyXV6daULf/7w5kvP2XqWkEtK+BwmHc5aeemllw5sXE1ZUf3EGo5v/+mVp9sr1GkWmRtPZ6BqmIpYSUtLc8YMmjIarx7Z/ez8uzIetwefQyJApiteEXwuDlRxMb769B2q9howG5HsWGi9ykdCzp46wY5zR8b4/n3ViYnWrlQ6fYpaW3gtUOaZkhBjx7lpYV5HC51r5GkXqVS64KEHc3b/3vGurp061Delb4CvzI5zs6J89ue1lxvHCQiFwmhfd53GwQuzl/79Y/z48UKBVTVwaZpGyIwjB+BQb5fD21M6evQYqq4TW6Kp8jNvvP6afeeK+ZwQqfBSuVPUKE3TDGRqP73pqw/fWWrfuVIhlyRQJSNpSe7gmWeeOfr7DwZ9hyawNE3v2/D1irfsvPuhUmGdWs+YP3h6erpjfTkogyFnx8/vvvaile0NFAWm/aABQM9UDl7ACtp61ny1kqq5wlx6aIdCa+r9PHhPPWJ/TcgxcX7bLpYbnZB/sqGhgSu0nLqoI5Tnngn1lw1O7WN3D+lyr703GJpFtsXNze2lxx8+8cemjnRyYf9faQMH9AgLtbuH4VE+Oy4zFK7lcG/o0zt+enDyFE+rXUvd+BxEIESa3CRC5srgYQVtLb4yz6efespYYXPdLFfAWHp87XdfdaQHdwGnb4h0jxMMHZWVlU514aCMxlM/f/H95x91pBMvEc9fIrjIxmrhY489Vn3yb73WzngZrUp5ZNPqlcuXdESGUE8RlySY8Yl2bDyhsr4m/+ju99+wdvoMAMqWSEJTFg68SOiqrHpviY9A3+l8oqnaa5mD00dkDepgP1lRspySBof7RFdXVwvcnejCcX7bmqlTp0bJ7Z8/NpMW5nWquF7D4OttMzweb/78+Yd/XWff6btWr3zhpZdlXh2NpB8T67f7WhUDKfx5PF58sEyrdMzDYPfX777/3goe1wbHuOZCySZDvRES4Wx2LsuWTT8ai492Ipc7Wtck1hRt+eG7jndFIDQtOXjNsULHRi5UVlYKnDaDrsnPVeZf+MiWCZQpuCTKiPD+M5eFxCyzZs1quHBI3WSzI821U4d4euWTc6xyLzOPG48cJPfadqm8411ZJCMjo+iiAzLbXdq3PS4yfNKY4TadpTfS5kO9dUbmlgmxgraNgamJL77wrLGUofppHYU2GgsP/rA+28PdMfVYQzyFySHSrecdmZykurraSdUIdSrF4dXv/bJhDemg2uRyT5HMjXeSQb/gZkiSfPbZZw9u+p9NZzXWVO5f98VPq79xlBiJgR40TeeUOj1DelZWVsFZq6qGmqGm+Gbu7l/W/ecTW09sDg4ykyyJvsuPY/v27REREXK5PDs7+45DRUVFQ4cO9fX1jYuL02hsdtbECtpmPli6KDEmjKq5wrYgljEWH33mqSfvGWFD+gWLZPaQVSl0Zx3nFl1VVSWQOF5B0zS1/79vvvfu8jgrintZT7rc+0aNsqSBabfoBx98UF9wXlFn7UKl0WDY/N4r//n8Mz+ZI99OxvX0P1ZQ62yHlpSUlPLrHUq4qlUq/ly1+OcN683Xh20XAZdECIBAJjYQ3u4HbTAYFi5cuHfv3pMnTy5evLixsbHt0Xnz5k2fPr2ysnLXrl0cjs2ri1hB28ORf3735TTQTSVsC2IOquJcer/en654w+E9z+kfuudq1TUHLRlVVlY6I9fo0bUfjh01fO4D4xzbLUIwvqf/PzeqGPa6QwgtWrRo3w9WTYdpmtr03ksLHp0/MsOGwG5r4JLE1KTgzedK69RO9LrjcDhJEYHqRjt9zymDYduHL69Y/nZSz2g7TlfpKTCXiqNlFbGV48ePR0dHh4WFyWSyzMzMPXv2tB66efNmfn7+/PnzASAkJAQraIbg83g5x/91V1132UR3VM2VqADRvj82OqNzHkk8mibflFNa5IhUnFVVVQ43cZz55Zu4YJ8vltvp+WseIZecEB+w82olw6mi77nnHmFDcUOlBfsSTdO/r1o2PCP9+UfnOEMMDwH3gd6BP5wuUjgzLXJmZmbheXvM0JTRuO3jVx6ZO3Pm/TYkBWuLG49ABCJIot0NgNY3Vp86dUqpbElRUlpaGhQU1Pw5ODi4pOTWvO369evBwcH3339/UlLSW2+9ZYcwWEHbia/M+8iBPeLGi7SCiWUTm6Cqc0Pd9TlHdlufV9NWxHzOgkHy9SeLCuo6GvTlcAV9evNXvhzdpm8/c2CfdyDhc8bE+P1+qdypE8m7Wbx48Z51/zHTgKaprZ++mRgT8dGb1pautwNfMX9sT/8Np4qanFb8KSsrq+CczWZoymDY9tHLD44f+8oT8+0eWm2gAJApG7RB1Xjl3Olvvvnm0qUWIwxN021/aG0/GwyGo0ePrlix4siRI3v27Pnrr79sFQYraPuJjpSfP31EqrlGNRSyLcstqPKcWH/elTOHbc0KZCteIt4T6eE/nS652LEIQ41G46hEHDRFHVnzfpg7988fv3few6kZmRtvXE//bZfKSxlM05GRkeGHVNXF+e0eNei0G999KT2591cfvO1sScI8RRMSAtefKnJSYdnExMSqm1dtOkWrUvz6zjMzH5y47MWnOjI0iQgzod58iWdy5uivv/66tXBaUFBQ66y5pKQkICCgtavg4OCYmJiYmBiBQDBy5Mjz523278YKukMEB/rnnjsRyK2lKp1bqscqKKOxYH9mv7jzx/dyuUwEO3kIuM9kROzKrTxwo8buThwV/qBTNe3+5IXRg1K3rHZWvag7kAq4k3oF7r1RzWQ+0rfeemvP2nZy+TfVVq157dHZUx/46K1XmZHET8Kf3Cdo49mSGzWOz0dKEES/WLmi1trAqLqyws1vPbHklRdfe/qxDg7N4QBCQBKo3Y0gEPd2p6B+/fpduXKlsLCwpqZm7969w4YNA4AtW7bQNJ2QkEBRVHFxsdFoPHDgQFxcnK3CYAXdUbw9pTcvnRg5IM6YvxeMLKRraIbW1Btu7Hz+qUd2b93g7MljW4RccuGQiDq17qtDN9V6+9zDHSBtdd6lne899eaiFz5e/ELHe7MeNx45JTGoqF69/XIFA0EcAJCSkhLlJSi7kdt2543TRzYuffrLlR9ZmbDNUXiLeA/3CztWWPdnboXD8/pnZWUVnj9hTcurh3f/8/mSzRvWzprkgDVhnYFCZnNxaG7/nnM4nJUrV2ZmZiYnJy9dutTDwwMApk2bZjQaEUKrVq0aOXJkXFxcfHy8TbUyWjrv+PVgSJLc8Uv259+uX/jc86RfInIPYXR4mqaqL3nQNb/v2TGofzKjQwMAAIHQ+ISAc6UNn/574/7egTG+NvhcNzQ08DqWiIMy6HN++15bcvXg339EhTH7nwcAAA6Bhkf5XChv+vlsybAePv4Sq1KmdYSlS5c+8OgzM5d9CQA6jXrX958gVf2hvX/7ejNUGb0tfA4xLSn4cH7N6uMF98UH+IgddvlZWVnfvfBWzyH3mGmjVTbt/ua9YG+Pkwf3ekgc4+zvxuUQCDhE+/MGkkCSu15Px48fP378bbm2W12es7KyWq3VdoBn0A7j6UceKrh6vqcPbczfR+sYeuellZWGG3+Ny+xTevMyK9q5ld6BHk8PjjhWUPf5gbw6lbVLZx1cISw5d2TH8seykmLPH/qHFe3cSoK/ZHxP/1PF9dsvV9j7JmEtcXFx/XoEF146c+Hfv75/cfakMcMP7PiNFe3cDAIYJPeekBD4x+WKbRfLdQ6Khu/Zs2d1ocn8kTRNnft7y9ZlTzz32Nw/f17rKO0MACq9EUzGeQNCd7rZORU8g3YkIYF+Z4/uXr/x92eff7FOzyd9EoBrs5+8ldCaOqr8bHiI/88Hd6ckxjtpFJsQ8zmzUkOuVim+OZIf4yseHu0j5lv4glVVVdlXLrbq+oWcLd9FR4Sd+OevkEDbasQ4CQmfc2+c340a5ebzpVHe4qQgDz7HWROguc+8NHZE5rgHpp0+uM/b08NJo9iEzI03JzU0p7Thu2P5fYKkqSFSbscCOBFC6b1jGqvK3H0CbjtA09eO7T3+65qRI0duOnHITejgnxiXRMh06W4CIefd1rvBCtrxPDR5/EOTx7/z8X9XfrqyTkMS3jFI6MjZDd1USlVfDgvy+2T9NxPutS3PAANE+4hfHhaVU9zw30M3w7xEmT1kvqZfe20NI6QpqvD0v7l7fo0ICdq85qvUXjavujibSG+3cC/R5UrFpnOlYZ7CxEAPd0tPKesxUvS1auXpkvoAd9+C/AJPsbMe/3bTJ9Ajwd/9VHH9t8cK4v0kKSFSMc/+yx86dOjv544nDLuv+U+DTntp3x/nd/82aFDakd07gm0s3mYlzRNl0sRCDsFsxn6soJ3F6y88/voLj3+7btOHH396/fpxwiMUScMQ1357K62upRoK6KbS/gMGrFizKTO9nwOldSwIICnYIynY41J502/nyxo1hr4h0pRgqURw5/etsrLSKhMHTVflXbp5ZFf5lTPDRozatXlDXGSYU0R3BARC8X6Snr6S6zXK3deq9Eaqp58kSiYW2Dvzomi6pEFzsaKpWqnrIXObmBAgZLDqkq1wCNQ/1DM1RHq+rHHz2VICQWKgR6yvxI6JZ1ZW1ucbX4obck/xxVOX/t2hqigcN/GB7IP/yJz50qA30s2LhO2CAPQMJkvCCtq5PDLzwUdmPlhRWf32R19u3bqlJL8MibyRmy8SeiO+pQziNEVr6ml1Da0op7WNsbFxsx9/4tkFs/l8p5fvcxQ9/SU9/SVqvfFkUf3aE4VKnTHUUxjjKw7yEPqIeQRCZmzQRr2uruh6dd6l8twz+rrymITeix6dMWXsGlPvnq4GQhAlc4uSuan1xksVTdsuleuMVKC7IEQqlIl4HgKueV8bjd5Yo9KXNmoK61VqPRXoLugT6MHACqSjIBBKDPRIDPRo1BhySuvXny6iKDrMUyT3EvmI+VIh17zvjlJnqFToSgnP8msXtr71WN/+Az99a1HWwFQGJBfxSQIhjomvGUkgMZ+5pyNW0Ezg5yv74oMlX3ywxGAwbNr615btOy+cP1dReqauSQWIRBweEG10rlFLG7QAtMzTPSQsPDl58IzJEzIHue582SJCLjk4wntwhDdF04V16qtVivOljVUKrZGG3ecLrlw8V3DiH4LDoww6mgatol6rbBKQwOfxI2NiR6SmTnh+dqJdSRVcBCGXTAmWpgRLKZoubdQUN2iuVCoaNHoagCSQgEPwSYJLEgaK1hkpnZHSGymKBiGX9BJy/d0F98T6ufJ82SLuAk5GhCwjQmag6II6VUGd6kxJfb1aDwBckuBzCAGH5JGEnqI0BkpnoHRGCgDceKSPGz9YKszLvSARCpgUWKMzNgeqtH8YIZUWLxJ2UTgczrRJY6dNuuUO2dikKCktLyotr69vIjmEj7eXv59PeGgQ15YU450FAiG5l0juJWrd80z6Z0WlZdV1DfWNCj6X4+Up9fH0CPTzITrJNNkmCISCPYTBHrcMxwaK1hsprYHSGikugXgkweMQPAclR3U1OASK9HaL9L5l5dMbKZ2R0hootZ7icwg+h+CTBI/BJbh24XEQQmDqJpCoJaM/M2AFzTLuErF7TI+4mB5sC8IOQj4vOjwsOpxtOViCQyAOQXbqCXJH4JIElyTcXMxiZ6QAIUSY8INGCBkZLKqDFTQGg8HcgkZ0czrodkGoJaM/M2AFjcFgMLfgkyRCYGotmiSQgME3HqygMRgM5hZagxEBMmNmdnakaFuwgr4FRVFNTR3KnNl5MRqNFEVVVLBQEdUVUKlUBoOh215+U1OTVqvttpff2NjI59/yXxRwSIIAkjSZi0OEZ9CsIJPJKitdtEKKs6Eoiqbp4uJitgVhB4PBoNfru+3l63Q6nU7XbS/fYDDIZLLWP5vjUEzZoAkAHYOrhFhB3yIszHWD05wNRVFqtbp3795sC8IODQ0Nubm5KSkpbAvCDiUlJRUVFcnJbCbbYpGqqiqttk2iYIQIhAgTftDNafsZkgwraAwGg2kLj2hJ2N/uUYKADiaBsgmsoDEYDOYWOopurnjV7lEEyFH5VK0BK2gMBoO5hZBLmE/Y78bDi4QYDAbDBho9hcC0DRq72WG6NtduFu3Ytffk6ZwrV65WVlWqmhp0Oi0CaP5RCIRuYolHaFho74SEtP4p947MFAo6TQo3a2jSGioV2mqFtkqpa9DodUZKb6S3ffvJ1TPHAWg3scTLxy8sNLRnXGy/pN4DknpxOV3qR9qg0VcrdTUqXbVSp9QatEZKb6Q3fLi46PoVkiTdxGIfP//QkNCE+LjUpN6pveJNhVw7Dy6BAIGpZDCIAA62QWO6GGWVVav+u/aPP3YUFeQ1aSgQeSOBFPEkwI1AXgJAhLH4KCGLRQKpgjJU6VQ3L9XtO7191TfraU2dt1SSnDpg1vTJ0+6/p5MmUVLpjNeqFTdqlBVNWncBx1fM9xXzk4I83AXc5vRAuz4qHf3yx0KJVKdWNtVUKmoqdp65uv63HdUF1z093NOHZE6eMC6T1ZJmHaFJa7hRo7xZq6pV6aQCro+Y7y3ihQYJJXwul0Q8ksiuKpz23hoA0Cgam2qrGivLth89v/rnX6uLbvp4e2UOHTbjgYnJ8TEMiUsAgRBpIlSFBEYfGVhBY5yITq9f8enX//vf2sKiEsI9CLkHo8ChFgx4BAfx3YHvjsQthY4aKMM/Z8t2H3jp8SeeSE4d+Oarz2el93e+7A7ASNG5lU2nSxr0RirWVzJI7u1nKaEzT+jmHRzuHRweljigeY9WpSjIObpwyTuKsvyhI8e88Pj82Ei500V3BHojdblScaG8EQFE+4gzIry9RRYSIwnE7gKxu09oZGTf9OY96sb6G6cOzXn6RV191Zhx4xc+OlceFGC+kw5iNAKYTtgPCJgp394MVtAYp1BdW/fYs6/9/ecfSo434dmD06OX/X0RHOQeQrqHaGj6UG7J8HFT/bwkb7y+6Il50x0nr4PRGqjD+bWXK5vifMXjevpLhfYnj+WLxNFpw6PThhv1uhsnD4ydNtvXS7p40ctjMgc5UGDHotIbjxbU5teqevpJxsf7d6TqldBdmpB1b0LWvXqtJvfQ38PGTYqQh7352svpfZMcKHBb+M2LhOYiCZl7jeuUL4wYV6ZJoZowY4FvaOzWfy9ogrJI/yTElzima4SQezApz6oSxDz9ytu+YXHfrf/FMT07Dp2R+vtq5f9OFEiF3AUD5Vk9fDqindtCcnnRA4c98OaX8RPmv7z8w75Zo/85csIhPTsQtd6480rlprMlge7Cuf3CBoR5dUQ7t4XLF/QaOm7me6vDh0567PlXM8ZOOnU+1yE934HOQDUn7DcBqPXMudlhBY1xJO+v+jY4Mm77/nOcHmMIryhATnFIQjwJGTywTtzrsWdeiUlKv3w1zxmj2MGF8sbvjub7ifmPDQxPCvIw5QnQQbxDIu597p2kqU89/sKr42Y8XFFd64xRbIUGOFPS8MOZ4ghv0ey+obG+YieZagOjEx5849PY0dOmzJ0/88nnGxVKx/bPJRFCQJrYCARMVvXGChrjGIpKK3okDnrt7Y/UfumELNZsOjDHgHhiMiwjT+EenzzwmUVvO3s48yi0huyThfm1qof7y3sHejCwjOQdHD7x9VWCqL7Jg4d+99MW5w9ojjq1ft2pogaNfnbf0CiZmIERA6MTZrzzrVLsn5yWsWXXXgf2TFFA/H+0990bAsRgUW+soDGOYN3G3+P79CtQiMmQQUAy6hWHxP6cyNH/Wf1Tj95p1bX1TA7dSl6Ncu3JwiGRsrE9/ZmcXgFAZGrG5KVfr/pu7fiHHtHodEwO3UpupeLX86WjYnwzI2Wm4jucAUKo97DxE1//7NW33pmz8GWjg3IY0UADAEEgUxt1V8L+7du3R0REyOXy7OzsuztUqVRhYWHPPvusHcJgBY3pKLMff2n2/Cc0foOQexA7EhAkGdQ/XyHq0TNp/5FTDA++93rV4fzaef3CwjxFlls7Ab6b+N7n3jH6hCWnD71RxGhGOpqGnVcqr1Q1zUoJ8ROz467u5uk99a0vyjRowPAxlTV1He+QR5IIIdLERiDgkbcZ7gwGw8KFC/fu3Xvy5MnFixc3Njbe0eGKFSv697fT7wgraIz90DQ9ZMyD6zdv50QMBw6jpZfvhvAIVXolZ44Y8/OWHcyMSNPw24UyvZGekRLCel3BXsMnpk5/esSY8SfOXmRmRANF/3K+VCrk3hcfwGT+oHYZOGlu3Oip6cNHXS8o6mBXesqIEBBE+xu6K93o8ePHo6Ojw8LCZDJZZmbmnj172h7Ny8u7ePHiPffcY58wWEFj7MRopHr1H3bobB4ZPJABi7M1IL4HRz5s2qyH/7P6R2ePZaToH88U+Yr5I2N8mY51M0FAVMLIZ9+dOHXGP0dOOnssrYH6Oac41lfcP9TT2WNZSUTSwKGPvjZq7H05l691SwHBYQAAIABJREFUpB8uSRAIkQTR7kZTVGN1+alTpxoaGprbl5aWBgW1vDsGBweXlJS07e3ll19esWKF3cK4xO8K0+mgaTopfWRuiYLwS2RbltvhCjnhw55e+PzqDU70wKNp2Hi2JNZPkib3ct4odiANCLnv1ZWzHn7k4Mkc541ioOhN50r6hXol+Ls7bxQ78I+Mvff5FZMenHzx6nW7O2nOVWfKyU6jbLx6+eKmTZuuX28Zgqbptqnv2n7euXNnaGhoTIz9MZBYQWPsYei4aZfyawnfBLYFaQ+SxwkfOv+xJ3bs3u+kEbZfLg+RCpODpE7qvyNIvH3Hv/LRzDnzcm8UOKP/ZsNOn0CPKJmbM/rvIF5BYaOeWTZx8rSyymr7eiAQNFf1bndz9/QeMGT4e++911reISgoqHXWXFJSEhBwK9Dx6NGjGzdulMvlL7744urVq5ctW2azMPZdA6Y78+izb+w/cprw78O2IKYh+WTYkKnTH7pyPd/hfe+7Uc0nifRwb4f37CjcZf5DH188btKDdY2Or7G580pFqFToanPntvjKowY99MyY+ydr7XJr4SBEICAJ1O5GIODeHmTYr1+/K1euFBYW1tTU7N27d9iwYQCwZcsWmqbffPPN4uLi/Pz8jz76aN68eUuWLLFVGKygMbbx6/a/v1u9mgxJY1sQCyCeWC3tnTlyrMHgyOSQN2tVhXWqETG+DuzTGfjIoxPHz5owY55ju71U0aSn6H4uY3c2RVjv1NDUoTMff86Oc/UU1ZyKo90NAHTG29zsOBzOypUrMzMzk5OTly5d6uHhAQDTpk0zGh3wxcMKGmMDldW1cx+eT4YOdpFVQfMgN98qndvYKXMd1aFSZ9hxuXxS7yAXWRU0T4/+WVq++5KPPndUh3Vq/dGC2tEu/3BqJnnMA1cKir/9cbOtJ/I4JIHAlJsdiZDgLlf38ePH5+XlFRQUzJkzp3mPRqPhtMkTO2fOnE8//dSOq+gEPzOM6zB83BSlKBJxXdH42C6ET/yu3ft+/PUPh/S25XzZ2J7+TBbU6CAZsxauWZt95tLVjndF0/D7xbJxLuBRZz33Pv3msreXl1ZU2XSW3kiBienz/8+gcS4OjOuxesPmi1duEFI524LYBhk84MmnntHrDR3s52qVwl3AYSsaxT4Ikhz+6KKHH3+y412dLqkP93LzcbOQL9Sl4AlFmTOfnv34MzadRSAw4wfdPLl2ksDtCMPYSJhOjcFgfPGlRURgP7YFsRnEEzcir0cWvtqRTvRGavfVyhHRnePtvi2+EbGk1P/L9Rs70olab8wpbXA1n0JriEwZVFGv2LbHBn8eAhACRJjYECAmE/ZjBY2xigXPL24EKeJ2pvljK4RPz+zstZVVNXb3cPBmzYAwL9bDBe0jfcaTH7//vk6vt7uHfTeqGc6z4UCGzXtu0WuvWd/eQNMAgJDJjcmE/VhBYyyj0+s3/bSB8IljWxB7QSThHTv/mVfsO1tnpC5XNPUJ8nCsUIzBd5OEpAz+4L+r7TtdoTNUKrSR3p1m4eEOPHwD3YN7ZG/+3cr2fJJAZtzsCBAw+JzGChpjmedeW67k+gLhmMTzrEB4Rmzf9nttfYMd5x4rqOsX6uWk5M7MkDxuxv+++4aya+p3JL82Te66Tt/WkDZp7ocffWRlYz1Fm549IwRIa8CLhBhX4of16whvpkp2OglEENKIFxe/Z+t5FE2fL29MCu6s0+dmeAKRb2zy9xt/s/VErYEqbtD0cMmgQeuRyPy4Ut+dB45a05iDkLlkSXcFqjgVrKAxFli/6fdGPRfIzrR83y6EV4+tv26y9axrVcoomRuTC/dOIumeKV9//ZWtZ12qaErwl3T6iwdIuXfyqv/815qWFNAIzOXsvysdtBPBChpjgU9WfUl4RbEthSMgefVq+NPGBB2nS+o7r/W5Le4+AbWNyoKSMpvOulDeGO/CUd3WExTT++LFiyq1xnJTGhACAtrfEABW0BhXQavT5+ScRm6dz72sXQjP8A9XWTWNakZroBRag48bO6noHU5s+qj/ZtuQiLVeredxCFHn9F25m8jkQWt/2WaxGYdszsXR/kYg4GETB8ZF2LB5GxL5sC2Fw0CSwJxTx6xvf71aGe3DRIU9ZogakPX3zp3Wt79erYzpQpcfN3jkli2WizcajGDWyw7pKbxIiHENftr4KyFhqZCVM0BkvdJwLc/aPJx5NcrO6152N0J3z+raOrXGitd8AAC4WaeUd6rISfP4hEbevG456p0kwIwNGiHgEriqN8Y1yDlzCon92JbCkRBi/zUbrE2gU9KoDvRguZSXY/GPit9lnTMDDdCkMUiFndi38m5EPkFnL1vQ0ZTpbP0IIQTIyKARGitojEkoiqqure/U7s93g9x89u4/aE1LrYEScMhO7f58N4ExvffsP2RNy1qVzrtTZd6whqDY3rsPHDHfpvl+m0rY32zmYAysoDsPcjmcdHqtubYcPn4G8SRMjsgASOidd/2KNS0rFVofcVfTUH6RPS+cO2tNy2qlrsusjrYSEBl3+swZ820I8wn7CUQymGsXK2jXY9QouKPywtat4O8PS5ZAWBiTguw/cgLxu4KL1W0QHK11RtgqpbbraSgPv8CykmJrWlYru+AMWhYaeeOqJROHJQuGAS8SdmvmzIF1625ztly3DmbMgHnzwOcuhwpDR7NomuHsuYvQ9RQ0QINSo9ZoLTarVuhkXU5DIUQotVZlTapR6rxFXe3yhRKPhoY68224BEJgMls/CcBjMCM2VtCux4QJUFsLBw60/FlXB9u3w6xZt5k4/P3h/fchORl69QJ/f/jyS4iOBrEYXngBiopgyBAQi2HcOFAqb3XbPAGXSCApCf791xpBysrLEVfo2ItzBRBHeC0v32KzJq3BQ9Cl7O/NECSpt+K5rtAZ3AUci806HWq9hUpUBopGyOQ6ISDEYCoOrKBdEKEQJk+G7OyWPzduhNhYSEy8s9n+/XDkCFy6BADw669w+DCcOwerV8OUKfD551BeDrW18N13t9rHxcHx41BfD088AZMng1ptUZCG+nogu9o7PgAAyS8qthxQpzVSvLuKG3UBhBJppRUVr3VGikl/Mtehuao3idrfmpcKmROGuaEw1jN7Nmza1KJDs7Nh9ux22ixcCHx+y4ryokUgk0FEBAweDEOHQu/eIBbDpEnQdj1k2jTw8wOShEceAaEQLFniAECtVgHZBaeQQHJr6yyntdMZKH5XVNAcvqCh7auVKWhG3RUYg8sXKpUqMw1aKqqY2gB7cWDS08HHB7Zuhbw8OHECpk9vp01AwK3P/v4tH0Si2z4rFLfaZGdDaiqEhIBcDqWlUG15DqXX6xEw9WUkSAYL0aJGhWUNpTdSjNXfI0mScuZywm1jcbgKtWUTPGN3HgAIgmDs8hFCWrNjUbTJgoTNG41NHBiYNQuys2HdOhg5EvzaCxWx6Tl+7RosXAirV0NREeTnQ3CwNRlfCAIx5pFPBqYy6THC51l+MyAJZGSqdkavXr0qblxmZiyapnika6XXSE9Pv5ljVfiMQ+CYfe6SBIFQ+z52zW52ON0oBmDWLNi9G779tn37hq00NoJIBFFRAADbtsHNm9acJBK5AcXQvIZRKL3M29NiKz6HYCw1e3p6ekluDjNj6dUqTw8r3NsZzNk2adKkSwdtSBLSEfRajVhkLn7dSFNmps8AuOQVBgDkckhLA6USxo93QG8pKTBtGiQlwZgxcPgwJCRYc5KbmxiMOgeM7moY9f6+MouteCShNVhY8XcU/fv3L71yjpmxtGqlt6flBKocEumZ0kS9e/cWNpZRRib+2zRNEWYXPwmEAAFBoPY3hAgGVwm7oBtN12Hfvtv+zM+/9bm8vP3PP/106/OCBbBgwa0/P/oIWqv+rFhhzfj+AQF0/o2ut1BE61XRPSIsNhPxSJXO6MVIsiChUBguFRgNepLj9FVZvUYtdrOcAUrEJVU65hwNMzIyCs6fCO8zwNkDCTgWzDuIBmR66ooYNc7jGTTGNDFRkaC3Yrm/00EbvaSW7d0yEa9axdwLRN++fcuvX2JgIIF1rileIl6tyv5C4LYyadKkSwd2OXsUg1YjFFpw7W+uqGImW9Ldyzfbt2+PiIiQy+XZrd6xAABw8+bNoUOHBgUFxcXF/fXXX3YIjBU0xiSDBvSlNfVsS+FoaMrLw6ocxz5ifpXCCm8HB5GWllZyyUKaiI6jqK3y9rGq/ILMjVetZO75lJyczK3Jp53sIVFbWhgqDzffhmN+kRAhzu0mDoPBsHDhwr179548eXLx4sWNjY2th2iafvfdd0tKSjZv3jxnzhzK9hhxrKAxJhmWMZDW2FMG25WhtQ2BIValNGFYQaenpxc7X0FXF16PiYuzpqW3G69aydzlI4QGDRpUeOG0U0cpu34psc9dMV+3Y6TNRRIiAMPtU+jjx49HR0eHhYXJZLLMzMw9e/a0HoqIiBgwYAAAxMfHA4DSGvfz28EKGmMSN5HQQyxg1O3T+dCq6v79+lnTUsLn1GuYc2KRyWR+HK2zC96V5p4bnPZ/7d13fBTV9gDwM7O9ZNNDyqYQkhAg9E4ogQjSDAiCgMJDrGBBn1gRFXyKBR8WLE/8oRRBEUUFBRQMhCpNkBpYQnrv2d1sm7m/PxIhwPadnV3wfD/zR7J7585Zysnkzr3nDnSmZYRCUsnjzycAmDx58rl93h3lKNOcHTrAwd++gKLsrCRkLOai/LydO3fW1bXW9CgtLY2Jad3UQq1Wl5SU3Njnb7/9lpSUFBDgcm1ITNDInuSOnUlzja+j4BLRls+cNsnJxmFycQ2Pw9CdOnWqKXZqBqTbSs+fzBqZ4UzLlt/oDXzNYwGA/v37l+SeIt78EVV68czQfr3ttyEE/q66YeXQ67WXL2l27txZVFT0d3tCtVmUQN2wQKGwsPCJJ574vG3dBadhgkb2jBs7mjSV+joKLimFpqED+zjZODFUcamav8ek6enpxWe9+Du+xWQUsqbwEMdzwFvEB8sL6hzXbOEKRVGTRw0rOe9UuWo36OpqIsPDxU6sUQIAWxsSBgWHDL9t1JtvvtmtW7eWljExMVfumktKSqLaLvEFqK+vz8rKWrFiRWpqqhsxY4JG9sy7/172FkrQRF+dkNjR+fbJYYoLVVrH7TiSnp7u1WHo/BMHBw0b7nz7DqGKi9X8fXzw8lyO3EPZI0ePdtiMoigawFa5URqo6zbZ6devX25ubmFhYU1NTXZ2dmZmJgBs3ryZEGIymSZNmvTUU0+1vOgGTNDInvDQIHVMFDHx+r/Ue9j6/Pvvu9f59oEykdHC6k08/ZqflJRUX1bkvf5z9/06Z8ZU59tHq6QVTUY+F86lp6cXnf3TS6Mc5/b9+vDMaU40dHD166ITCoXLly/PyMjo1avX4sWLAwMDAWD69OkMw+zcuTMnJ2fhwoVqtVqtVue3XcrgHEzQyIHZ//oXqdX4OgouEFbB1s2bY63ylG3dolWnyvibyjKgc6K2ttIbPZsMenN95eA+PVw6KzlMqeHxJpqm6azhg8ounuG858bq8kCFNC460mHLlq27adrecZ2srKy8vLyCgoLZs2e3vGIwGIRC4dixYy0WS/HfEhISXA0bEzRy4MV/z2Wbim+BuRxsY3Hv/ulCoWurZ7tFBZ4sa3TcjiPem2x3dvfPWXc6+3T0iu7RqhOl/H188NpcjuPbNs25/wFnWrKE2C1mR/G44xUmaOSIVCK+beRotqHA14F4iq0+t/ytxa6eJRHSkQGSvBqeHhWmp6d7ZbkKIad3/fj8Yw+7el6gVCSggc/5dkOHDi04xfHmyBaTsejEgQenO/3zyXY9aAqs3EF7DyZo5NjKD99kq8/7OgqPEF1l59SUHl1ceEJ4xbDEsD15jstnc6JHjx4Vedz/UV849Pvw4cOdWeB+oyHtw/bn13Ieki1CoXB0eh9u/xD+3P7d3TPuFTn3yxMNFNW6t7fVA/grko4JGjkjXh09OD2drc/3dSDuY8pPfPbhu+6dGygTqSSigjp723BwRSgUdo4JNTVzecNOWPbw91+8/coL7p0eGSAxWhg+54NPnjz5LHejHGaj4a/ff3rpqXlOtmftDnAAAOGxEismaOSU9V98xFadBsLfsgUOkcaiXj26Derfy+0ebksJ35FbyXp5mV+LgQMHlp7nsvToqZ0/jB03Lirihi3hnTYiKXxHrlceXVq/3IgReccPctXbgW//b+7ceXKp1Mn2Ahooykat0b8rjnIVm0OYoJFT1FHt7rtvDltxyteBuI5lmPKTG9d+5kkfgVJRx3DlH4V1XAVlR3p6evE5zoahDU0Nub//sHzJQk86iVBKQuXicxVNXEVln0gkuq1/j6rCS553VV9eXH722HOPOvV4sIXDx+FeXet4HUzQyFkr338jSKC96conMeV/PvbYY4nxag/7GZwYeqKkocno9eocAwcOLOVuNd3OlW++unixwlGNTYcyOoTtz6/lbYuZyZMnn/N4xQoh5JePXlvx/nKBK8/1qJYngbYfEt64mNt7MEEjZ9E0vembr5jiAzfRlDuiq1AHiz5462XPuxJQ1B1dIjeeKPH2QIdCoYhXiTjZRPXCgZ0JkWEzJ433vCuJkM7oEPbT2TLPu3LGyJEjLx3f72EnR376asiggZnp/V09kaLsPCTkdQcLTNDIBRnpfec9/BBTesTXgTjH0syUHtnz2xau+lMHyjpGKH/XeH1GR8+ePSsuezqNoa604NTP69b/70NOQgKApDBFsEx8tJiPEuESiWRI91RPSkeVXjhddCzn03ded/lMAmD3ISGfMEEj16xYtrhz+0i25oKvA3GEsJaCvatXrUyIjeaw1/T2oRVNhvOV3h2NTU9PLz7jUdUkU7Pul/deWr9mtUrpeHcr541ICjtX0VTcwEcFpcmTJ5/bv9O9c7W1Vds+/s8PG9eLHG1wdSOKBgps3j7TFCXAIQ7kz/7Y/XMYXevfVe4IU7j3iUcfmTl1Arf9UgBTu8fsu1yTX+vFWXeDBw8u8eA5IWMx//jWgiWvvtK3W2cOowIAmqImd4v+NbeSh60MRo8erTm6140TTc36TUuf/r/PPu0Q586Dh9aFhDZQAAw+JET+TC6TnvhjT4D+AtFV+ToW65jiQ5OzRr+39CVvdC4S0NN7xm47X+G99XXt2rULp9y8S2UZZuu7zz9837/um8LxD6cWcpFgUtfoH8+UNxi8u2OhTCbrn9q+vsJK/Xs7zEbDxv/Mf/WlF0cNdnP/WYpuWS5ofY5dyyJD3mCCRu6IbBd+MOd3ZdMZoi133JpXhCk6MHJov41ffuS9ayjEgmk91Zv+Ki3x2i/7ycnJdaUuL69nLOYt7zx71x3jnp13vzeiahEkE2V1ifz2ZKm3V6+4WpfD1KzfuOSJfz/x+OwpE92/KrFbiQOAz7FoTNDITR2TEk4fPxhsvORHZTpYhinYMzVr1PbvVnv7UsEy0azesVvPlnupYLQbxfuNOu3m/8yfdfek15553BshtRWhlEzuFr35VFlRvRfHo8eOHXvxcI6TjXX1NRtenrvw2afnzXShpKpVlJ1aHBRQuJIQ3RRiotrlnjoSI2lgK7hc+eYeYtZb8n595olHNqz6gJ8rKiXC2X3j91+uOVjAfamKwYMHu/ScsK60YNPiuYsXPvfCYw9xHoxVwTLRtJ4xuzRVp8u9Ve5OqVT2TIxurHb8W1rZxTNfv/roJx/8d7bHAzutFZFsH1iLA900ggNVl88cHjOkG3N5F1gMvgqDNBbLKw9+vXbVm68+y+d1JUL6X33jdEbLumNFzWYu18GnpKTUluQ72fhczrbsjxd/v2HdPRPHchiDQ0qx8N5esUX1zZtPl5kZr8yOnzRp0rl9v9lpQAg5smXD/jXv/b5ty+hh6Z5fkSVge8crCoDi7/4ZEzTyHE3TW75Z9ekH70jL9/mgoBJjYooOJCh1F04fmzpxDN9XB6Ap6raUiP5xwasOF3A43EFRVN+OCfp6B/fm+vraLe88S0pz/9y/m/M5G84Q0tSY1HbJYYo1x4oKvTDccccdd1z4Y7etdxuryr9+ZV47geHPfdnJ8bGcXJGmKKBsPiSkaD4XEmKCRhx5cNbUS+dP9ktUMpd38bUcnLC1Gml5zsvPPX7x5P52EWG8XNS65HDlnH7xuZXaNUcL65q5md5gfxiaZZjjW9f/8taTrz775E9rVyrkni7m9kRapGpq95jjxfWbT5fpON0hTKVSpanDtHXXLw5izKZ936zctvzFFe+8seq9t0Qi1/ZhsIMAcfSQkD+YoBFnwkODD+z66fsNX4QazjPFB726kyHbUGjRbM/oEVtw4fQrzzzqvQs5TyYS3NElckRS+Hd/lWw5W97o8Sw0W7urEJY9u/vn9c//KzVMdubowSljR3p4IU4ESIQT06J6RAduPFnyu6ZKz92Az5133nl+/9VRDtZiOb7t2y+fmTUoNf704X0jhwzk6kJ/o4Cyg9cczdmPHYRaTBgzYsLlM59+sWHxa69XlhnosI6Uoh1nvbNmti6PrdUMGTJ05W8HUzokcNYzR9RBsgf6J2iqdZv+Kg2SifrHBccEunlv26tXr4pLZ9u+YtA2ntr5Q+6+HWPHjT+xPzssOIiLkLnUPkSeEBKXW6ndeLIkMkDaOyYwXCnxsM8JEya8/tGEPuOn6+prjm/blH90z4RJd3115ECAQs5JzNdpuU22NdmZ52l2mKCRVzxy3/RH7pv+w8+7lix958SJn+mAGCoonpIGu9kdy7BNJaQhP1BCxo6f8PaSzdGREZzGy7GkMEVSmKKovvlwYV15U3mndgFdIgPCFa6lKpFIlBoVbDY0E8LmHd17ft+vtEk3bca9P75/UOnTAQ37KIDUCGVqhPJyrX5ffm1Dszm1XUBquDJIJnKvQ7FCFRUo3/DKPLmQunfWrMc/e1sscrMr5xA7QxkU8PqYEBM08qKJ4zInjsvUNxveWfF/33yzsaTwuJYRUfIIShZKyYKAtvffjJh0xFAL+hpWVxmikvXqO/CZ+UtGZgziLXjPxQbJYoNkFpacr2zaramu0pnC5OL2ofJ2SkmYQiIX2ywTwbCkRm+q1BpDO6Stf/5foaGhQ4dn/mfVx2kpHfiM30PtQ+TtQ+Qmhj1fqd2lqapvNocrJHHBsnCFJFQhktqukmFhSbXOVKM3FdU3lzY0y0SCBa+93Tk6OImjx4D2Ua27eltP0RTN6yAHJmjkdXKZ9JVnHm0ZKd5/6Ng3P/xy9OixgvwTep2uQdsMQFECMVAUABCWAdYMAMEqRWhYRMc+nYYNmf6v6RPDQ0N8/Bk8IKSptEhVWqQKAGr0pvxa/ZmKpmpdjc5kEVAUS4hYQIuFNEuI0cKyBABAQFEhCnG4QvzySy/+7+0lAZwWPOKZWEB3i1J1i1IBQJXWWNTQfKaisVpnMpgZAU2xBMQCWiygLCwxMSwAEAJCmgpRiMMU4q6RqlEpETQFAJ6W83bB39XsrOL5ISEmaMSr9AG90wf0bvsKy7IVlVXNRgtFQWiQSqUK8FVsPAiVi0Pl4uteNDGshSUUgEzkcum1m0u4UnLjkLSJYRmWUBQlFfrHnIW/nxHaepPPWDBBIx+jaToqkruniDchsYC2Pdpx6xMLaPCnj9+Sgm3OduZ1GjROs0MIoWvZX+rt0NatWxMTExMSEtasWeNhJHgHjRBCbVBAUWBrF0OKBvu30BaLZf78+dnZ2QqFonfv3hMnTlSpVG7HgnfQCCF0DU9WEh4+fDglJSU+Pj4sLCwjI2PXrl2eRIJ30KiV2Wyuq6vzdRS+odVqzWZzUVGRrwPxjbq6OoPB8I/9+DqdTiqVXvm2dQTaxn2y0WDQaDQ7d+7s2rVru3ZWnp2UlpbGxMS0fK1Wq0tKXNtw4DqYoBEAAE3TERERTU3e3WrPb5nNZgD4x/58MhgMFovlH/vxTSaTWHzd1BqbS1H0en1JScmxY8ciIyOtJmhCSNvk7uEzRUzQqFVsLB+rAPyTwWAwm82pqam+DsQ3SkpKKioqunXr5utAfKOqqspovGb3MgpsVuUPCQnJGDbs2eees9VbTEzMlbvmkpKS3r1722rpDByDRgihNggBQoCw1g9wUPa6X79+ubm5hYWFNTU12dnZmZmZnsSCd9AIIdQWASBAbCRiQuwMgACAUChcvnx5RkYGwzCLFy8ODAz0JBRM0AghdC1CgLWRoFnWfoIGgKysrKysLE4CwQSNEEJttNwiE9tZmMc9rzBBI8Qfo8n8wy+7cvYfPHXqTGFhQWN9LRBC0YK66nIQKwOVMgCQyuTtIqM6pqb27dVzysQx8eooX0fNGYYlJQ3NFVpjZZOpUmtsMlqENMWw5P8WPVpbVyemWIqmFUqVOja2U6dOPbt1vX3owNAg91d5eIDYTNB2ErcXYIJGyOt+23Pw45WrD+7fW1lVQ8lCKFkIJQ2kxB0gIq1lXRqlzxaoB+mEEgDQMebqOu3p7DPfbdv/7MJXVDJBSmrXuyZPfPzBe2VST4vf+0RJg+FcRdOlGp2ZYdVBssgASWo75bAOoUpJa/75njJkLF7Z8rVR29hYVXqmJD9n7XcvvLJEDJaeffpPmpg1ZfQIgYCvSQ2EAGtjRxjC4h00QreCy4Wlz7+y9LcdvzQYgAqMowN6CoOuL2VnhUBECYJbNjegAfSE/Fladez1D55/4cUePXo+t+DJuyeO9nroXGg0WI4V158qa4xQirtGqQa3D5E6Ua5PolSFK1Xh7VNh8GgAYBlL6dnj76xc9/yzz/QZMOjfjz4yqFdXLwfuMAFjwX6EbmaHjp585Iln/jp1ig5JosMHCexuTeAARVGKCIEiAgBOVVbOuP/xx56g5z8x/6UF8zgLl2tVWuPOi1WNBkvf2KC5gxJEHtz50gKhums/ddd+hLDFp4/OfuIZOcU89+yC6XfczmHA1yOPFrFCAAAgAElEQVQsENt30DzCedAIcen0eU3X/rcNyhx3pkooTBpDhyTb3zjGJZQiQhA3pCGw1yvvfBoSnbT8ky+56pkrtXrz2qNFW86UD0oIeXhgQi91kCfZuS2KomO79hu74J3es55585Mv0gZmbNm1l5Oeb8ACXJkNbePgCyZohLhhMpvvnvN4tz6Dz9eIhImjKGWkt64klAmi+2hD+i5YtLR9l77H/zrnrQu5gmHJzgtV648XZSSFzekfHx/slR1dASAgPGr4Qwv73ffC06+8njlpRlFZBccXIJS9hSqO5kFzCxM0QhzYlXMoKqHzd78eFiaNpgJ4mXchkgli04stkX0GDX/oyZf4uKJtxQ3NK/blycWCR9MTY4P42NBWFRE9dsE7QT1G9B9++wdfbuC0778Xqri1kpBbmKAR8tTTC98YOe7ORlU3OrwLULz+n6Lk4cKk2/9vw08d0vpXVNbweekrjhTV/XCqbGaf2EEJIbxuNwIQ263/na/+b+WG70ffPUvfbOCm0yvzoG2Ob/CXozFBI+Q+k9nca8iY9/63RthhFCX1aFGv+yiBILpPYXNgx259du87zOeVLSxZe6yoosk4L719yA17LfJDJJVnzn0Zojv1HDziYgEXFVMpCggB1sZBgM+0iQkaITc1NDYlpfX/K79BEDsIKB9vq0ep1LqQ3iNGj//q2y38XNFoYb84XNClXcD4zpE0z3fON0gdNr73jPkjx2YdOP6Xp30Ru0MchLW5CtwLMEEj5I6Kyprkrv1KzcF0mL8UKaUkKmH7zJlzHlrx+VfevpbWaPnsYP7QxLBe6iBvX8tJER06j3hi6V0zZu06eNTTvuw/JOTxhxEmaIRc1tDY1K3v4Dqhmg6M93Us1xLKhO0zn3hqwRfrv/PeRYwW9ssjhXd0iewYofTeVdygahcz9pllM+97YN/REx50wxIAANbWQXCaHUJ+y2JhuvXLqBFEUyq/3OJAIBYmjLj/oXk7ft/vje4Zlqw5WpiZHJ4Q4q2JdJ5QhrYb8/TbM2fPyb1c6GYXhAIgtrS04C5eBzBBI+SawaMmlWiFfnfv3JZQIogfOmXa9Ev53G8z+PWfxb3VQZ3aBXDeM1cCwqPSH3hx/KSpjVq9O+dTFBAWWMbGweJDQoT81KLX3zty8hwd3sXXgThAiQOag7oOyRzHcvpE60B+rUoq8p9xZ1vCEjp2GnPP+Gkz3Tm5ZZE3sXFcacALTNAIOevQ0ZOvL31ToB7g60CcQinaVRokd977CFcdljUajhfXj+1kZadUP5TYL0NLy197/zOXz2yZZkcYmwePU1YwQSPkFJZlJ9w1XRDTH+ibpsQYHZG2ZevPW3/d7XlXDEu+PVl6T+9YAe3jGXXOS5/55KeffnpOc9m101qn2dk6/n6IyAtM0Ag5Ze7TL1cbJZQ81NeBuIQSqAfOvv8hzwc69uTV9IgJDJZxVviJBwKReMicZ/71yGMun+lgJSF/MEEj5FhpReXKlSsFEd18HYjLKImqzqKY/8J/POmk0WA5XdY4uH0IV1HxJjI5rVkgX7XxBxfPI8Cy1g+cB42Qv5k6ay4d3gVoHy8XdI8gouvqVZ83aXVu9/Dz2fKsLr5fLuieQfc8/sbrb7DO3/m2jGPYfEpoez9ZL8AEjZAD+UWlB/bvp4MSfB2Iu2ihXhoz999uVryrbzZrTRb/nPXsDJkqODSlx4dfrHf2BIfT7Hish4UJGiEH7n90AR3e2ddReIQOSdnyw3cGo8mNc3deqBqRHM55SHzqOWHWRx9+6Gzr1ll0tu+gicVrkV4PEzRC9mh1zdm/7/LrZSnOoAU6QdhLry939Ty9ianUGjuEKrwRFG+kykBZVMKmbbucak0ou/WgCfA4CI0JGiF7Xn3rfTowjs+pr15Ch3Zct3aNq2cdL67vHevvy1Kc0X3MtA8++sSpppT9YklYsB8hv7Fh/QYquIOvo+CCSFZVpz1x+rxLJ50obegepfJSRHwKie2Ql59f29DouGnLbbK9edDWc/TWrVsTExMTEhLWrLnmB+Hly5dHjBgRExPTqVOn7du3uxQ2JmiEbMorLCmrrKZEN+vzsetQQQmvvf2B8+2rtEaVVCQV3ZRzV27Uod+Ilc4U+aMEQOxMs2OtTuaxWCzz58/Pzs4+evTookWLGhuv/iQghLzxxhslJSWbNm2aPXu2S3PSMUEjZNOKz9bQKrWvo+AMrYrdl5PtfPtzFdq0SP8tiuSq9n2H/fzLL47bsRYH9aBZ840nHT58OCUlJT4+PiwsLCMjY9euq+PdiYmJAwYMAIAuXboAgE7nwnxHTNAI2bRjxw4qINrXUXBHIK6qaah35td8AADQVGuTwvyr4rMngqLiNBqN42rOLbPobAxx6PTNp85d+Pbbb4uLi9ueVFpaGhMT0/K1Wq0uKSm5sePffvstKSkpIMCFn3mYoBGyqaQwn5IG+zoKLlGKcCf3xCIEdCZGJb1pCo84Izgm4fBfZ+23IYQAIYSwVg+zxdzc3FxXV6fX6687i2rzJJm64alyYWHhE0888fnnn7sUMCZohKy7XFjaZOD1kT0PKHn49p27nWlZqTW2C5B4ORy+RSZ33bF7n4NGrQ8JrQ9xBCrl/Xp2e+ihh1JSUgBg2bJlSUlJSUlJMTExV+6aS0pKoqKi2nZZX1+flZW1YsWK1FTXNkjDBI2Qddt37gHZzVd9wj5KFnr+3GlnWlbcigk6PLHTyZOOdpWlBUAIMIz1g73mIeGCBQs0Go1Go+nXr19ubm5hYWFNTU12dnZmZiYAbN68mRBiMpkmTZr01FNPtbzoEkzQiEdpaUDTUFBw9ZXBg4Gi4MSJ1ncp6uoxfjwYDNe8Yv9Ytw7S0mDduusv+thjEBcHUikkJsJ//+t8sEeOnaAkgZ5/aL9CiRW11VXOtKxsMkYob7UEHRyTkKe54KARw9grZUcIMFZWEgqFwuXLl2dkZPTq1Wvx4sWBgYEAMH36dIZhdu7cmZOTs3DhQrVarVar8/PznQ/4lhpgQjeB5GRYtw4WLgQAyMuDkhIQi6++u2oVTJ/e+rVAACIRNDe3fqtWw9q1MHw4AADLAk1f/6JYDG++aeWKd98NCxZAYCDk5sLEiZCWBqNGORNp7oULlOTWmcNwRV2jU7MIKrXGHjG32s8niVypbXT0jJSiCSGEtb5tCiE2a3FkZWVlZWW1fcVgMADA2LFjLRY3V4fjHTTi1733wtq1rV+vWQP33nvNIj2RCKTS1kMkAoCr37Z9Vy638iJt4x/zkCGQkADBwdChAwQEgEbjZKQ1NdUgkrn3Kf0aRRmdKMqhNVlusSeELZotjjasIoy9QhxgM3d7AyZoxK9OnUCphMOHAQDWrYOZbu0a56oXX4SICIiMBJqGKVOcPEnb1EgJbrXf8QGAEkiLSsocNjNZWJHgFswPFC1wtFSE2FuowuN2KoAJGvnArFmwdi3s2wfh4ZCScs1bc+dCWFjr8dprnF3xhRfg2DH46isYPx6cnoVqMhpuot2tXEALa+sbnGl409cfsUYgFBqMRnstKAEQQljG+kFYisdyo7fivz/k56ZPh27dQKuFWbOuf+uNN+DOO1u/VnFXAiIgAAICYNo02LMH3n23dQTcEYZh+Kv8K5LZHKLhHEU1afWOm/GIatmnlZeKVAKhSNdslMtsDl4R5u9dva2/DYSxspLQSzBBI96Fh0Pv3vD11/Duu9e/FRwMam8urSYE8vKcbEvTNH9ZI4bHncIJkUv9a+gmPDxc31gnD+RjUiNjscilYjsNKIEAWNbqVA0AAJahBPylTUzQyBdWrICKCgi54T+k2QwGQ+vXNH3NBA8nte1BIACDAb74Au64AwIDITsb1q6FlSud7EkilQFrBoHrMfg51hwR4dTWt7xVPo6NjS2tLucpQZtNdm6fAVpukInNsqKEEAt/d9A4Bo18ISEB+ve38vqcOSCTtR79+rnTc9seHnwQaBq2bYM+fSAmBhYtgjffhBkznOxJJlfw+cssbwhjjowIc9hMJKDNDE8LKWNjY5uqy/m5FiHsjeuw26KAsrerN/D0S1ULvINGPDptbQ3blRteq+9eUV3t+EWrPWzb5lRsNwiPiMwv0IH45t5MxArCKOSOC6gqxUKt0RIi5+MXiNjYWO3pIzxcCACkQkfVUymKEJbYmunB4o4qCPmBDokJxOz+Tth+KyjAqfLW4UpxpdadPQzdoFar+bmDNhuaZXbHNwCgdZqzze1UbA9PewEmaISs69urBzHU+zoKjhGzLiTU8fgGAEQoJZVau9PRuBMXF8dPgq4tvtQhpaP9NpRACIS1WYuDYUHI32MJTNAIWTdqxBBorvN1FFxrruuQ7CBDtWgXIKls4ilBR0RE6BtqebhQTYEmLS3NfhtiMTkYg7bw9McCmKARsiU1ub1CzOuyMR6w+qpRmcOdadkuQFLWaHDcjgs0TXcI4WNVfVnuidszhjiKRgCEtblQhbVZi8MbMEEjZFNUTCwxOrv/yE2B6CpnTZvgTEuaoqQiWmfiabxVIBDYfC7Hndqiy+m9uzloxBI79aBbF4LzBRM0QjZljsgkTaW+joI7rDk0UBEe6ux046Qwpaaap8ekkZGRujprE3W401RVFhenph2u2GQt9oY4CAGCxZIQ8gPz585hG4t8HQVn2MbiAelDnW+fGqE8U97kvXjaio2N1dZ49zlh3tE9o8eMddxOKCaEJQxj/WBZEIi8GmdbmKARsqljUnx4sBIsPA3FehupL3j+ybnOt49SSat1JhMvy1V4WKuS90f2w/c4UcvQYrI3xEFYYsKHhAj5h8mTp7J1l3wdBRcYY2iAKH1Ab5dO6hqlOl3Gxyi8txN0Q0VxVERoZLgTa9xpIbBgs9woISDEO2iE/MOShU+x9fm+joIDbK3mzslTXT2rtzroSBEfk8FjY2O13kzQp7ZvfOD++51qylrs3kFb3/LKSzBBI2RPWEjQgIGDSGOxrwPxDGHZustvL37O1fNUUmGARFhY1+y4qWe8egdtNjTXak7OmerU9BUAAEIIYa0erTmaL5igEXLgsw/fZqrO+DoKj7B1l8ffcUegSunGuaM6Ruy66NQ+s54IDQ01NDm1jYAb/tr+zb/ue4B2ssgRLbC3kpBlQeComgd3MEEj5EBaanLPHt1v4ptowrI15z7/8G33zg5TiAU0VdLg3SelFEV1CHWqSIirzAZ9ybHdLz7+kJPtHVcTNWO5UYT8yYYvP2EqTtqsEezf2Opz06fPiHDm+ZgNYzu1+/F0mbd/sReLxawXhnf/+ObTx+c/KRI5W7mTEogIYQljsX6wDNbiQMi/pCTGT516N1t11teBuM6sD7BUrlrh5u1zizCFODFUfqTQu5VJoqOjtTWV3PZZW3TJXFnw1AOu7E1sMQGx+5AQa3Eg5G/W/u+/KlJz0638ZooPvv/+conE05u+zOSIA/m1WqMXJzBw/pyQZZjslUtXfvyha6dRNBAAllg/CNaDRsj/iETCb7/+iinafxMNdLDV54cMGjDr7omedyUSUJO6Rq07Vsx6bQ5DbGystqaCww6PfLdy6l2T+3Xv4tppNAXAAmFsHCzQmKAR8j+ZQwc8MGc2U3bM14E4hTTXBpGq7ZvXcNVhXLA8OVzx+0VvVczgdip00V9/mEovLVv0jMtnWhzV4rDxkHDr1q2JiYkJCQlr1lj5M9fr9fHx8U8++aRLsWCCRsgFn72/tGNs6E2wttBiYIoP7Ny+VSrhcgPvEcnhBXX6cxVeKdDB4b4qjRUlxzd+/POm9fa3H7ROJAZC7NXiEFv5I7VYLPPnz8/Ozj569OiiRYsaG68fClu6dGl/q/tw2oUJGiHXHMnZFkoqSVOZrwOxjTVb8rNXr/q8R5pTtfmdRwHM7BObram+XKPntmfgbgy6ubFu+3svrF+7OjLMrW3CTQZ7t89AiMnKjMPDhw+npKTEx8eHhYVlZGTs2rWr7bt5eXlnzpwZO9aJUk3XwgSNkGsUctmff+wJ0OUSndeXb7iDZSz5u5f+Z8nMqVne6F4soGf1if3pTFkF1/utBAUFGXWe3psbtY0/v/30Zys+GNizq5tdCIRAiK1aHFq98fSlwm+//bawsLDtSaWlpTExMS1fq9XqkpKStu8+++yzS5cudSMWTNAIuSw6MuJAzi5l0xmi5WMnPRewZsvlXS8889Rz8x/03kWUEuHMPnEb/izmfAl4UphHe6g3N9ZveXP+u2+9MT7T0bYptpEWLGv1YFm22WCoq6szmUzXndV2OKXt1zt27IiLi+vY0Z3fZjBBI+SO1OT2p48fDDZe8p9SSsSst+TtXPraK6+/9G9vXytELrq/f/wPp8u4HY+WyWSM2c0b8/qywi1vPP7xe+/ePW6kJzEQxgJX0vQNAqTivp2SHnrooaSkJABYtmxZUlJSUlJSTEzMlbvmkpKSqKioKx0eOnRo48aNCQkJCxYsWLVq1ZIlS5wPBhM0Qm6KiWp38fSxWLmOLT8B4OPdC4m+Wlax/+u1X3j13rmtAInw4YEJ+y7X7LlUzdWHV6vVTdXuzLQrPHlwz0eLfvp2Q9ZtwzyMgRZLCEsIw1o/WJYSS680XrBggUaj0Wg0/fr1y83NLSwsrKmpyc7OzszMBIDNmzcTQl555ZXi4uL8/Pxly5bNmTPn5ZdfdiEYDz8MQv9kgSpl3ulDd44aYMnbCWavl3yzha06E2LQ7M/5ferE0XxeVyKkHxiQYGHJF38U6E0cbASlVqtdnWnHMsyhDR8X7/3p0O6dfbp28jwG1mhs3XjQ2kEIYQ1W/qKFQuHy5cszMjJ69eq1ePHiwMBAAJg+fTrDePTH4uz6dISQVRRFfbv647Ubb3/ssSd0sgQ6pAOfVyfGRqb40O2jbtv6zf8JeKyydgUFkJkcfrFa97+D+ZnJ4d2iVZ70Fhsbe+CkCwm6Ov/C7s+Xzpg+7e2Fn3hy3bYooZAQwtpKrCyhRNYL9mdlZWVlXfNg1mC4Zr7H7NmzXQ0G76AR4sDMqRMKNKcHdQq35P1Kmmv5uCRrZsqOBzX9tfW79ds2femT7HxFcphibnr7vBrdykMF1TqT4xNsiI2NbXJuZ0KDtmH352+e+f7TXzZ9/fbCBW5f8UaEsQCxsc6bJUAIwYL9CN10glQBOds37dy6KZItYApyiMFrG5GwFrbqjLRszxNzplQW5o4d6cI+sN4jFdITu0aN69zu+79Kv/6z2L007cxiQqOu6Y9vPvnt7SfnzZj0Z85v3VKT3IrXJkJaRjhsAh6X+uMQB0JcGjFkQNGFP7/9YdszLywqunyMDu1IqWK4Kq9DjE1s7QUlWz9txr3vvblIJpU6Podf0SrpQwMTCur0P54uEwvoAfHBSWFK51fzxcbGam0/JKwtuvTXtm+0pZcenjvv31+tENBeub+kBUJCCLFYT8OEISDkL21igkaIe1MmjpkycczeQ8cXLXkzJ+dnSh5BB8VT8nCg3MkpxKQlDYVsY1FcTORDz8179vEHhEJfDmg4FB8sv79/fHmT8Uhh3c/nKpLCFGmRqrhgmcM9TRQKhclw/RrFupLLl4/syTu6p3183AuPzvVwFp1DrNkEBIjNaTnWVxJ6CSZohLxlyIBeu3/ZyDDMyrWbVq9df/5sTqPeQsnDKHkoSIMokcJmvrYYiLGRNNcSfTUxNiQkxI+ZPnbh0/OiIyP4/QQeiQyQ3NElkhC4WK09UdLw05lymUiQGCpvFyCJUEpC5WKBtbJwSWEKXV1VfWlBVX5uxcUz9WWFHVNT7xg75qEPlwQGeLSMxUm0SAyEJTYeEhKWpUT8/eKCCRoh7xIIBI/MvvuR2XcDQEFx6Ybvfj5w8I+LFy9UVZQzLNPQpAdoqUFMWiZTByllAUHB8YkJvXqOHXf7iNuGDXSn4o/foChICVemhCsBQGdiCmr15VrjmfKmGp3JwhKqpbonRZG/q5hKE9JOf/1eYlLSuD7dRj39QHeuh5gdYk0mQoCw1u+gCUuIib+C/ZigEeJPvDr6+fkPAl9rSfyNQizoHBnQGQLstHls8Are4rGOpoEQmwmaED7rQWOCRgihNii6pRaH9XcJAZq/BwCYoBFCqA2zubVgv1WEsDjEgRBCviGREMKyFusPCVmGpRUy3mLBBI0QQm0Ym1tKcVhHgLlhIqD3YIJGCABAr9eXlpb6OgrfqK+v1+v1Z8+e9XUgvmE0GoODg69+T/9dsN8qQihcqIIQn8RicVRUFLF513SLE4vFQh6Tjh+67q+eEEIYm9Ps+Kws+4/+W0GoBU3TbSus/9M0NDQIBILExERfB+IbVVVVRuPV537EYgEChNhY6k0IsbGrtzdggkYIoasoiZTYXUlIS/EhIUII+QJraLY/zY7R63gLBhM0QghdRYlEhABrewyaEkt4CwYTNEIIXUUsTGvBfutvE2Lhr2A/JmiEEE/OX7y8YdNPR46f0FzIramuZFm2QWsAwhBDPSUNAloYqBBTFK0KDGrfIbln926Ts8ak9+/Jd5SEASB2HhICy8Hui07CBI0Q8iKjyfTeJ2s2fvf9pQvnmgwsJQ+nZKEgVVMRqfB3ArJotgmTxgBAy+Cu1mIoya3fe3LL+5+sIqam5JSU8WPHvfDvh8NCgm1fhzOUqHVXb+tvs6Ttrt7ehlteIYS84udf9/QdNk4Won5x6Yd/lQv0URmC9pl0u26UKoYSK+2dKZRSykg6rJMgfqgweVxec8T7X3wfEd85ufugD1eu83bYrFHfcgdt4yBsM38rCTFBI4Q49smqDVGJXbPuvu9EKREmjxdE9aYU4W7v+0VJA+mINGHSmHxT5JML3wiOSlzw0hsWG7UyPEdJ5IQFwhDrB8vSEv6m2WGCRghx5pvN29oldH7s2cXVsk6ChAxKGclh55Q0SBAzQBc+6L3/2xSmTnr1rQ857PwKYjTY2zKWAMvjlleYoBFCHCgqregxaNSM++bWKroIYgc5GMTwhEBMR/bQR6S/9u6n6uTu+w4d5/oCNAFCWNbqAYTlagtgp0Lh7UoIoVvVso9Wdene90wFEbQfQYntbZjCGYFYENO/QtB+6MhxMx95msM6KpRQAMRmgib8FkvCBI0Qcp/ZbBkx/u7nFv3HEDWUUsXyfHVKFiLscPv6H3+PS+1VVFrBSZ+syQgEgCW2DtbQbPXErVu3JiYmJiQkrFmz5rq3ioqKRowYERER0alTJ4PBhRESTNAIITcVl1XEpnTPOVEgiM8Agdg3QVC0ILJnOROR1rNf9t4/PO+PlsoJS1iGtXoQllByK5uLWyyW+fPnZ2dnHz16dNGiRY2NjW3fnTNnzowZMyorK3/99VeXCgdigkYIuePE6fNpPQfUCOPo8M6+jgWogBh9WL/MMVnrNv7kYVdMs87uQ0LCaq3U4jh8+HBKSkp8fHxYWFhGRsauXbuuvHX58uX8/PwHHngAAGJjYzFBI4S869DRk8NG3K4L6k4p/aVMKyUOELbPnHX/wx+t2uBRPwIhAADDWj20RvOf+YWfffaZRqNpe1ZpaWlMTEzL12q1uqSk5MpbGo1GrVZPmjSpZ8+er776qkvB4EpChJBr/jqbe/vYLH1ob0oa5OtYriWUCttnPj7/3wFy2axpE93shKJanhHaeJtIJZLg4GCx+JohHUIIRV2d3dH2a4vFcujQoRMnTsTHx48cOXLAgAGjR492Mha8g0YIuaCsomrYiNH6kJ5+l51bCMTChBGzH3j4t90H3OuAmE2t5UatHXKBIDUyYsqUKXFxcQCwbNmypKSkpKSkmJiYK3fNJSUlbfd/UKvVHTt27Nixo1QqHTVq1KlTp5wPBhM0QshZDMP0Sb9Nq0imZCG+jsU2oUQQN+SuqdMLS8rdOLv1IaGFtXoQhtDyq1O8FyxYoNFoNBpNv379cnNzCwsLa2pqsrOzMzMzAWDz5s2EkLS0NJZli4uLGYbZu3dvp06dXPgobnwAhNA/09gp91U0S+gIta8DcYCSqPSBndOHjy7M/bPtaIMzGH2z3Xr9wGibbnxdKBQuX748IyODYZjFixcHBgYCwPTp07VarVAofP/990eNGmWxWMaNGzd+/Hjng8EEjRByyso1G3f+niNoP8LXgTiFUkaVlVdMv3/+16s+cO1EsQhYm5vGAktoG9XssrKysrKy2r5yZcrz8OHD3ds0HYc4EEKO1dTVP73gGUHsQF8H4gI6svvGTd+5PDmaYQixPdEOCOGxHjQmaISQYxOn3a+Xtwchf4XcuEAJYvpPn3mfS+cQhgGwuYwQWGDN/O2oggkaIeTA7v1H9h88SAcn+joQl1HSoKpmwYuv/df5U2iJDFhCGNb6wbICGZYbRQj5jdkPzhNE9/V1FG4StOv+0YfvG00mJ9szzXoCYKtePxBgdPzt6o0JGiFkz8+/7Skqq6Zkob4OxF0CsU4Y/vTCN5xsTktlhADL2jxomdyr8V4TDG9XQgjdjOY//Tzdrruvo/AIHd5p7ZovGFvbDF6LNRjs7nhFGFfK0XkIEzRCyKYzFy7l5Rf59bIUZ9AiLaiWrVjlVGMKCLE9xMEScHFitScwQSOEbHpu0et0aLKvo+AAHZb6yaefONOSEomJvYeEhBaKvB3tFZigEUI27du9i1bF+ToKDlDigMKSMmcWf7MGA9gsxQEEgNHjrt4IIV/7aXt2k0UMtMDXgXCDDox/fdkKh80EMrmdaXbAsIIAXvb0AgBM0AghWz5fvZ4KvBVun1tQgXG/7tjhsBmj0/29ZND6HbSl0UotDi/BBI0Qsu7wwf2039Tj9xwlUhQUFlosDtYBUkIhECAMsX6wQItwDBoh5FNGk7myps5nOw16ByUN3rZrr4NGNE0IsAyxehCGAMVf2sQEjRCy4pff9vhpSX4PUPKwn375zX4b1mgEsP2UEHAeNEL/WEuXwtixV79NTr7+2y+/BIqycqxb19omIwOCg6Ely7RIS2ttE3jlMr8AAAzNSURBVB4OM2ZATY0zgezZd/Cmn/58A0oe+tfJk/bbCBRKO7t6sywR4kNChP6hhg6F/fuBYQAAysvBbIbjx69+q9HAqFHQ3Nx6hIbCL7+0fj1jBgBAfj7s3QsUBT9du7n16tVgNsOBA3D+PLz4ojOBnDpzjpKoOP50vkZJVEVFhfbbMFotAACxeZgbGvmIFQAwQSPkX/r2BbMZTpwAAMjJgeHDoWPHq9926ADR0SCVth4AIBK1fk3TAABr1sCAATB7NqxefU23NA1CISQnw913g6NbyBYF+fkg5u9WkSe0qFnvoNQRJRET1voANMsQwhKB1HrBfm/ABI2QPxGLoX9/yMkBAMjJgSFDYPDgq98OHerg9DVr4J574J57YMcOqKi4/l2tFrZvh5QUZwJprK+lbrLqz05p0Dbbb0AsDIDtgv2EEAYL9iP0jzVsWGtG3rsXhgyBIUOufjtsmL0T9+2DggKYOhV694YOHWD9+qtvPfYYREZCWBgAwNKlzkRhMZtvmSUq16IYuxmWtViA2C7YT7BgP0L/ZEOHwr59UFcHVVWQnAyDBsGBA1BXB6dPO7iDXr0aRo1qzcIzZlwzyvH663DyJDQ2QnY2xMQ4FQZ/FYH4RdPNBqOd94Vyhd1dvVmhgr9yo7hpLEJ+ZuBAaGiAzz6D9HQAAJUKoqPhs88gOhrat7d5VnMzbNwIDAORkQAARiPU18PJk9C9OwBAYCC0a8dL9O6g5GH8XQvoJq1OaTvJWnQ6IEBsb+ttxpWECP1zyWTQpw/8978wZEjrK4MHw3//6+D2+YcfQCCAs2fhxAk4cQLOnYMhQ2DNGvfDsJGgvIHP7VoIkAClvVtggUxGCBDG+sGyIFAoeIsWEzRC/mfYMKishMGDW78dMgQqKx2Pb9x3H8TFQWRk6/HYY/DVV+BoZbNNt+oQB2tR2s2wjMFIbBfsBwJMM38LVXCIAyH/s3TpNY/ypk6FqVOtNKuuvvr19u3Xv3vlrNOn3QhBJBYDy9yizwkdszXEQVoKJvEF76ARQlYEh4QTM3+7o/ImUOlg7iAtaSnYb6tYEqGlEn5CBUzQCCGr4uLiwHTLJWjGLFcoHTTRNYO9edBgadJaPXHr1q2JiYkJCQlrbhj6X758eUpKSlJS0lNPPeVSvJigEUJW9OjejRjqfB0Fx4ihLqF9B/tthEpH0+xUVhZYWiyW+fPnZ2dnHz16dNGiRY2NV5eDl5eXf/DBBydPnjx//nxOTs7Ro0edDxgTNELIimGDB9ySCbpbt67221iadPaWEdq4gz58+HBKSkp8fHxYWFhGRsauXbuuXpQQhmGMRqPZbCaEhIaGOh8wJmiEkBW3D08nRv6qAvGD6KqmTBhrvw0lEAIBYK0fOoY5Wlz41ltvnT9/vu1ZpaWlMX+vAFKr1SUlJVfeioqKevLJJ+Pi4qKjoydMmNDezmT2G2CCRghZIRQKotqFg4W/KWU8IMaGEUMHOGgkEgCArWeENECQQpGYmBgYGHhNz4RQ1NWZiW2/bmho+PHHH/Py8oqKirZu3Xrs2DHnA8YEjRCybtDgIWxTqa+j4AwxNnbo4GAAGgCY5mabGxISkFB0+4DgKVOmREVFAcCyZcuSkpKSkpJiYmKu3DWXlJS0vNsiJycnNjY2LCxMqVSOGDHijz/+cD5mTNAIIesef+R+0uCgevJNhDQUTJwwwWEzoSqABWIh1g+GEHHw1XvnBQsWaDQajUbTr1+/3NzcwsLCmpqa7OzszMxMANi8eTMhRK1WHz58WKvVGo3GnJycFOeqCbYG48bnRAj9Ewwd0CtQRmlZM9D8bZPqPWxj8bPzH3TYzNzYRIjtle4ETPVWhuaFQuHy5cszMjIYhlm8eHHLAMj06dO1Wm3Pnj2nTZvWvXt3iqImTJhw2223OR8zJmiEkE0jR4/9btcJOiTZ14F4ihjqUpKSwkMdb+JFSyQEwFZBUhbA1kKVrKysrKystq8Y/t69cMmSJUuWLHEp4NZg3DgHIfQP8cbLz7G1l3wdBQfY6twnHp/nTEvCsATARikOwhJCWNbb0V6BCRohZFOHBHXXtC5Ed8PmLDcXxhgoMs6bM92pthaz7Vl2QAAYk9nb8V6BCRohZM/Hy99kKk75OgqPMFVnH5k718nGYqWSBWBsHCyASMXfXrqYoBFC9qQP6N0xOZFoy3wdiLssBhWpf+1FZ4tgGLU6AGBZYv0AMNTXezPca2CCRgg5sOHLT5nyE2BrkxH/xpQdX/jSSwKBs3VThTIZIcDYOFgWRAEOyi1xCBM0QsiBHmmp48eNY2vOO27qZ4i+Oi4iYMFj9zt/isVgJITYGYO28FiwHxM0Qsixb778SGkuJybrlTb9FMswJX98/42L+34Ru7M4gNjcrtALMEEjhByTSSVrvlzFFO2/iQY6mLKjjz76aM+unVw6SyCXsWB7iIOAQO6g5D+HMEEjhJySNXr4jGlTmTIXav34EFuf3zE29MO3Xnb1RHOT3lED/vYxwASNEHLWus+Wd4hWsXV5vg7EAdJcqzIW7P99qxvnilRyloCdWhwiFT4kRAj5peP7fg0l5f48646YtPLa47t2bA0OdGfCsknbbH+hignvoBFC/kmpkB8/tEepzSW6Kl/HYgUx65mCPZs2bnB16PkKSkgTAJbYOAAoIX87nWOCRgi5Jiaq3cGcXcqmM0Rb7utYrkFMTUz+7xs3rBmZke52J7RASOwU4iBACfmrMYcJGiHkstTk9qePHww25fnPeDTRV8sq/9ix9Ye77rjdk34szUawfQdNCJi0Dp4icggTNELIHTFR7TSnjyWoTEzZcZ/PvWPr8lTaswf27hqZMcjDrsQqJUtsbnnFECINCXTcC0cwQSOE3KQKUFw8uf+eCcMteb8RM3/3lddgLUzR/rRo8eXzJ9JSOahbbWhsAgBi4wCAZmsF+70EEzRCyH0URa353/INq1cqqg6xNRd4vjppKrVc2v7Ckw+fOPCriqMSGQKJxN5CFQChjYL93oAJGiHkqbvvHFNy+Vxm7wSLZjvRVfJwRWJqYgp2JwfpNWeO/+elf3PaNbHzkJAlhGVwqTdC6KaikMt2bF53cPf2eEkNc/l3oq/20oWIScsUHwzWnlr9v/fOHs1JjFdz27/FZLI/D9piNHJ7RTswQSOEONO/dzfNqYPbf/iqg6LOculXti4PiK3t/VxGmkqZ/N3hhnMfvf1yZcH5e6dkOT7HdRJVAAtgIdYPhoA0OMgb17UKEzRCiGMjM9LPH9976siecQOTFeU5TOFetqEQWLd2iiIs0ZYzJYctF3/u1162ffO60kunHrnPqc2r3GNo1AIAITYOAH0tfwX7cVdvhJBXdOnY4ccNKwFgy47dn3y++tjhg1W1jZQsmJKFUtJASqwEobWycIyJmLTE2Eiaa0hzTbBC0rlr93umL3pw5l3OF933hEguIQRYGxMHCSFipZyHMFpggkYIedcdt2fccXsGAOj0zT/8sitn38EzZ88VFl7QVtcDBfXa1vr3QUopEJDKFTHq2NTUHoMG9Ltz3G2REWE8R2sxmgkQFqwnaBbAYjTxFgwmaIQQTxRy2T13jb/nrvG+DsQelmEJAdbGTA1CgGVZ3oLBBI0QQleJWwr223iXBRAr+BviwIeECCF0lVGnBzsPCQk0NzQ57OTOO+8MCgq66667PAwGEzRCCF0lVirsFewHIg0McNjJ/Pnzv/jiC8+DwQSNEEJXtd5B267FYXCiml1GRkZAgOM87hCOQSP0T0dRVENDQ2Fhoa8D8Q2dTtc2mcpUAeVgPE6sV0SqBlNIbflbb701bty4tLQ0b8eGCRqhfzqlUpmQkODrKHwmICBAoVBc+XbkXRNeq6211dhoMsojQmMSE8LDw3mIDRM0Qv90NE0HBwf7Ogp/ERAUOOfZJ904cdmyZZ9++ikAaDQaroLBMWiEEOLAggULNBoNh9kZACji660QEELoFnP77bcfP35cp9OFhIRs2bKlZ8+e7vWDCRohhPwUDnEghJCfwgSNEEJ+ChM0Qgj5KUzQCCHkpzBBI4SQn8IEjRBCfgoTNEII+SlM0Agh5KcwQSOEkJ/CBI0QQn4KEzRCCPkpTNAIIeSnMEEjhJCfwgSNEEJ+ChM0Qgj5KUzQCCHkpzBBI4SQn8IEjRBCfgoTNEII+SlM0Agh5KcwQSOEkJ/CBI0QQn4KEzRCCPkpTNAIIeSnMEEjhJCfwgSNEEJ+ChM0Qgj5KUzQCCHkpzBBI4SQn8IEjRBCfgoTNEII+SlM0Agh5KcwQSOEkJ/CBI0QQn4KEzRCCPkpTNAIIeSnMEEjhJCfwgSNEEJ+ChM0Qgj5KUzQCCHkpzBBI4SQn8IEjRBCfgoTNEII+SlM0Agh5Kf+H8+FWsZMi/kgAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "corrplot(\n",
    "    res2$r, type=\"upper\", \n",
    "    order=\"hclust\", method=\"pie\",\n",
    "    p.mat = res2$P, sig.level = 0.01, insig = \"blank\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nOzde1wTV/o/8DNJSAgy5aaICBIChpvgFra1pVhqsRXaKgpt9aeoW6u2tqv2ol1rq1+tq+uFVr9Vt62u233prrFfXaWiVuutVUSthaqlCCFQEFCLhACBEJJJ5vfHtNlsEmKEkDNhnveLP8iZmeSJymM4c+YzBE3TCAAAAPvwcBcAAADAPgHuAthOLpf//e9/x10FuG9jxoz585//jLsKAPoEGvQ9lJeX/8///E9aWhruQsB9oGk6IyMDdxUA9BU0aNCjzZs3V1RU4K6il6qqql555RXcVfTS5MmTs7KycFcB8IMGDXpUUFDw+eef466il15//XVfX1/cVfTGtWvXvv32W2jQAEGDBg7weDypVIq7Cs65e/cu7hIAW8AqDgAAYClo0AAAwFLQoAEAgKWgQQMAAEtBgwYAAJaCBg0AACwFDRoAAFgKGjQAALAUXKgCgLX29vbY2NiEhAQsr67T6VpbW0tKSrC8+p07d955552ZM2dieXVgBRo0ANYMBkNqauqBAwdwF4LBv/71L61Wi7sK8CuY4gAAAJaCBg0AACwFDRoAAFgKGjQAALAUNGgAAGApaNAAAMBS0KABAICloEEDAABLQYMGAACWggYNAAAsBQ0aAABYCho0AACwFDRoAABgKWjQAADAUtCgAQCApaBBAwAAS0GDBgAAloIGDQAALAUNGgAAWAoaNAAAsBQ0aAAAYClo0AAAwFLQoAEAgKWgQQMAAEtBgwYAAJaCBg0AACwFDRoAAFgKGjQAALAUNGgAAGApaNAAAMBS0KABAICloEEDAABLQYMGAACWggYNAAAsBQ0aAABYCho0AACwFDRoAABgKQHuAu5Nq9XW1dXhevVBgwap1eobN25gefXOzk4fHx+CILC8elRUFK43jhDq7OwcNGgQrpcOCwvD9d6NRqNerxeLxVhe3WAw0DSN670TBCGVSoVCIZZXZyGCpmncNdyDQqH4+d//COpU4S4Eg7LQxLArl/j6btyFYKDKzoqX4WlSeDU1G7Qd3ZJQLv52W3+HTvz9M6GhobgLcUrFg7EOtsb+UNH3l/CAT9AIoeDO5ghVLe4qMKgIiQuuLBd2aXEXgkHni8/GjeRig+bxULvAECfhYoPW6oy4S7gPPC9+v79Ef78AAAD0TktLy+zZs0NDQ6VSaVZWVnl5OUVRBEGE/aayspKiKG9vb9tjDx06RBBERcWvH2MpiuLxeNHR0VKpNDMzs6mpyb1vpZegQQMAWGr69OmhoaE3b96sqalZsWJFTU0NQkgkEjX8JiYmpqdj5XJ5Wlravn37zCNCoVCpVFZXV4eEhOTn5/e9PJ6Q7+Cr78+PPGWKAwDANaWlpUql8tixYzweDyGUmpqKEKIoypljOzo6Lly4cPbs2UmTJq1atcpyE0EQ6enpJ0+e7HuFbpjicGWDvnnz5t27d134hAyVSjXE5U8KAGC3srKy5ORkpjtb6u7ulkgkCCGJRPLNN9/YPbagoCAzM1MmkwUGBpaWliYnJ5s3GY3G48eP5+bm9lfdPRgyZIhIJEIIDRo0qLKy0smjXNmg586d+7vf/c72D7SPSJJ8xrXPCADwBHbXmIlEotraWscHyuXyN954AyE0bdo0uVzONGi9Xh8bG9vY2BgfH5+dnd338u5rHoPP5zc0NNzvS7iyQdM0vW7dOoHAxdMmCoWic7cLJowAAB4kISFh9erVJpPpfj/zqVSqM2fOlJWVEQRhNBoJgti4cSNCSCgUVlRUtLW15ebmbtiwYeXKlX2skO+iiWYH4CQhAICNUlJSoqKili9frtfrEULFxcWFhYXOHHjgwIFZs2bV1dXV1tbW19dHRkYWFRWZt/r5+W3dunXbtm1arVtXrxqNxpEjRyYkJHz22WfOHwUNGgDAUnK5vLGxMTw8PCIiYu3atTKZzO5u3d3d5oV3mzZtksvlU6ZMMW/Nzc3du3ev5f5xcXEZGRm7du3qY3k8L76DL6udr1y5UlVVdeTIkQ8//NDyPwzHYBUHAIClgoKC9uzZYzWo0+ksHwoEAqup6qVLl1o+XLRoke2Bcrm87+Xd1xw0c2IzMjJy0qRJ33//fVpamjNHeUaD7hYIO4U+uKvAgCYIHUmaXD2t7xGMRlrT4dSaqgFG101362lNJ9szGPqDgRqY71qtVhsMhuDg4Kampq+++urjjz928kAP+MnXaDT14THNEVLchWCgp8S/zHyeQAPzX61jbV2ir4q5GJrT3WXwMtHtWhPuQjDQdKLgiC7cVTjL+U/Qt2/fnjJlSkdHh1AofOWVVzIyMpw80AMaNEmSiT6qkSI17kIw+KJFNi70ro+Aiz+r/9ccEffYONxVYHC3robUljyU4AE/my5XcoPCFePXC3ynL1SJj493fu2zJThJCAAALAUNGgDAUvn5+TKZTCaTJSUlFRUVWQYeZWdnq9VquyNW2Un19fXjx48PCwuLioratm0bsghOio6OTkpKOn/+fO/Kc0MWBzRoAAAblZaW7ty5s6SkRKFQnDp1asSIEcgi8Igkyc2bN9sdsbVy5cqGhoaLFy+uX7++vLzcfJRSqVyzZs2yZct6V+F9LbPrnV/nuTZt2nTlypU+PhfzzgEAoO/u3LkzZMgQX19fhFBwcDCySEoiCCIjI8My8Mh2xCw8PDw8PJx5kpiYmFu3blmup9ZqtQEBAf36Rvri1wZ99OjRv//97318rpdeesn5nRcsWFBVVeXMzZwkEsnbTw7vQ10AAM8zbty41atXx8bGjh8//vnnnx837j9njPV6fWFhoWUEku2ILYVCoVAoxowZg37L5dDpdGq1+vTp072r0FXzGA782qCZW4H18bnu605iP//889GjR5l4J8cUCoWpZO89dwMADCRisfjSpUvffffdmTNnZs6cuXz58vnz5zNRdjweLy0tjYlDsh2xq7W19YUXXtixYwdJkhRFMbkcCKGioqIZM2aUl5fz+ffdbd2QxcHFpTwAAI9AEMSYMWPGjBkTHx+/ZcuW+fPnW0XZURRlO2L7PDqdLjs7e/HixVlZWVab0tLSmpubGxsbmTlutoEGDQBgo4qKis7OzpSUFJqmL1++HBER0bvnMRqNU6dOzczMnDNnju3WkpISiqJCQkJ68cweFtgPAACuYjQa33777draWj6fHx8fv2PHDicPZLKTmO8XL16ckpJy+PDhkpKS7du3I4S2bt06ceJEc+q/WCzes2fPfU3PmrlvDhoAAFglISHB9oYptklJ98xOQvaC/+3eCoCFPKBBazSaax0hlzuG4i4EAwPiH6gJvfdKl4GI4rV8f/TfuKvAwGQy8RB142cuBkVRRuQ/zK0xzX0Bn6ARQogkyTEP+UVF3Hu9x8Dz76MtWWFNYk5mcfxdKWsPjMddBQbeXXeTh3cmPRiOuxAMyn+85ePjMbmVMAf9KwGfEAm5eNEjgZCQZxLxudigEUHQPM/49+laJoInEPCFIi6+dx6fm78u9oiLXQ8A4BFaWlpmz54dGhoqlUqzsrLM1yofOnSIIAhmITPjnqkdCKG8vLwhQ4aMGjXKfNTZs2djYmIiIyPfe++93tTnxXf05QoubtBqp9ldrggAAGbTp08PDQ29efNmTU3NihUrampqmHG5XJ6WlrZv3z7moZOpHfPmzTt27Jj5yWmanjt37oEDB6qqqk6ePFlcXHzf9XkJHH25git/jZowYcLrr7/u5M43btzwlBOpAAD3Ky0tVSqVx44dY+7qnZqayox3dHRcuHDh7NmzkyZNWrVqFXI6tSM9PV2pVJqf/+rVq4GBgYmJiQihmTNnHjx40PwS7OHKBr1kyRLnd87MzLQM4tDpdGvWrDEajbZ7kiT5/KQoF9QHAPAcZWVlycnJTHe2VFBQkJmZKZPJAgMDS0tLk5OT7yu1w6yxsXH48F9DfsLDw3v1CZozJwlbWlpKS0v//Oc/225SqVQItbi/JAAAXnZ/yZbL5UzmxrRp0+RyeXJyspOpHQ6e3GTq1Xl47jRohFBgYGBKSortuEKhoLuhQQPALQkJCatXrzaZTJYfolUq1ZkzZ8rKygiCMBqNBEFs3LiRIIh7pnbYCgsLa2hoYL5vaGgwf5pmFVjFAQBgo5SUlKioqOXLl+v1eoRQcXFxYWHhgQMHZs2aVVdXV1tbW19fHxkZWVRUVFFRUVJSghC6r9SO0aNHt7S0XL9+3WAw/POf/5w8efJ9l9j/JwmhQQMAWEoulzc2NoaHh0dERKxdu1Ymk8nl8ilTpph3yM3N3bt3L5PaIZFIoqOjf/zxx7/85S92ny0nJ2fs2LGVlZVhYWG7du3i8Xg7d+7Mzc2VSqVPPvlkWlrafdfX/8vsWDTFAQAAloKCgvbs2WM5YpXOsWjRIrvjyCa1AyF08OBBq5GMjIyqqqo+l9mPPKBB6/X66qquxtt63IVg0NVNX7ztxye4uB6RZ+wm1b25U72n41FdDTeNnZ3duAvBoE2tHT7Uc66Q4NRJQgf8BpmCA7l4DWhdAx0cIvDic7FBV//s5T04GHcVGJg6WkhRU5g/F//S6S7P6c4IES6aaHbAAxq0UCgcFiyQRfT7f1YsdLXCEB1C+Yi4+LNaXO/FH8zFwCBEEIMHqeKivHDXgYFWZxIIPKApuQ3Ok4Rnzpw59Zvz589rNBqMxQAA2OaeCRt2R7y9vS2fpL6+fvz48WFhYVFRUdu2bUMImY+Kjo5OSko6f/58L+vzuCwO5/3xj3+8fv16yW+uX7/e2dmJqxgAANs4mbBhO2Jr5cqVDQ0NFy9eXL9+PZO4xBylVCrXrFmzbNmyXpY4gFdxPPfcc88995z54a1btxyvKgcAcIqTCRs9jZiFh4eHh4czTxITE3Pr1i2ZTGbeqtVqAwIC+vWN9AWrp3uUSmVbW5tKpYoYgrsUAIB73VfChoPMDTOFQqFQKMaMGcPsHxsbq9Pp1Gr16dOne1li/58kZPWFKtOmTdu/f/+VK1dwFwIAcDcmYWP37t1hYWEzZ87861//ihBiEjZiY2N9fX2ZhA3bEbtaW1tfeOGFHTt2kCSJEBIKhRUVFbW1tUePHp0xY4bdmLZ7G8BTHM4gSXL9+vUKhYLuvIi7FgCAu90zYYOiKNsR2+fR6XTZ2dmLFy/Oysqy2pSWltbc3NzY2MjMcbMNqxs0AICzKioqOjs7U1JS7ithw5bRaJw6dWpmZuacOXNst5aUlFAUFRIS0punhgtVAADcxCRs1NbW8vn8+Pj4HTt2OHlgd3d3WFgY8/3ixYtTUlIOHz5cUlKyfft2hNDWrVsnTpzITIwghMRi8Z49e4RCYW9KhAtVAADclJCQcM+EDYFAYDtimyLtzAg7sahB6/V65t6OZuYUbV23SdPJxUu9TTTq0KHencDweCaTSa/FXQQGNKXv1tOaDi7eyt2g94y++SvuTHE88MADnZ2dL774ouVgdXU1Qkij0dTUGBpucTEsqbub+K5KxOPi/02IpvTG+p9wV4EBrdc18I3tHdZhbFyg6aSDR3ThrsJZBHcatK+vr+UNdxnMykeSJJOiTLLhXPxAIT/Df3K0wUeEuw4cPr8gFssexl0FBobmhihx5UOJXPxbL/mpWywW466CRdjSoAEAwMPw+v0TNKsvVAEAcBnbw5J4fEdf/+3s2bMxMTGRkZHvvffefbxCLysDAID+5AFhSU6jaXru3LkHDhyoqqo6efJkcXGxkwdCgwYAsJFVWJLllX5MNJJSqXQwYhYeHv74448ji7Aky619Ckty+hP01atXAwMDExMTBQLBzJkzbW++1eMr9LIyt/Dz83vqqac2bdqEuxAAgLuNGzfOYDDExsa+/vrrZ8+etdzERCPFx8c7GLFlG5YkkUheffXVVatW9bJEpxt0Y2Pj8OHDme/Dw8MbGxudfYVeVuYWBQUFJ0+eXLp0Ke5CAADu5gFhSU6zvC7GfHmHM2AVBwCApdgeluT0Ko6wsLCGhgbm+4aGBvOn6Xu/wn3XBAAA/a+ioqKkpAQhxN6wJKenOEaPHt3S0nL9+nWDwfDPf/5z8uTJTr4CfIIGALCRB4QlOY3H4+3cuTM3N1en082YMSMtLc3JA6FBAwDYyAPCknj3MQORkZFRVVV1v6/gAQ1ao9FcK+NdKufibIyBQvuLhJyM4kBGU1vH1VO4q8CANpnKCNONagPuQjCgjLT/UM9JyOr/Kwk9oEGTJEkFxxoeGIq7EAyImkvZmYPE3lz8z+nr081Pj+HiG1c2oLZadbJfC+5CMLjeHuDj44O7ChbxgAaNEEI8PuJ74S4CAxoRQi+eSMjFPsXjESIu/p0jAR8JCCTicTEdjE94VNwoAVkcAACuamlpmT17dmhoqFQqzcrKKi8vpyiKIIiw31RWVtqGbzAOHTpEEERFRQXz0DK1IzMzs6mpyQX13U8WRy9fwSXPAgAALjd9+vTQ0NCbN2/W1NSsWLGipqYGISQSiRp+ExMT09Oxcrk8LS1t37595hFzakdISEh+fr473kCfecgUBwCAY0pLS5VK5bFjx3g8HkIoNTUV9XAdiq2Ojo4LFy6cPXt20qRJVldyEwSRnp5+8uRJF5TItZOE27ZtM19vY0aS5JDfPYGjHAAANmVlZcnJyTybpWzmJcwSicR2HR6joKAgMzNTJpMFBgaWlpYmJyebNxmNxuPHj+fm5rqgRK416H379v3v//6v1aBKparl4oojALjO7mplq2u77ZLL5Uwux7Rp0+RyOdOgmYCkxsbG+Pj47OzsfqjX9djVoL28vFJSUqwGFQpFbdVdLPUAAHBJSEhYvXq1yWSy/RDtmEqlOnPmTFlZGUEQRqORIIiNGzei3wKS2tracnNzN2zYsHLlyr6WCHdUAQBwU0pKSlRU1PLly/V6PUKouLi4sLDQmQMPHDgwa9asurq62tra+vr6yMjIoqIi81Y/P7+tW7du27ZNq+3zFTE8nqMvV4AGDQBgKblc3tjYGB4eHhERsXbtWplMZnc3JnyDsWnTJrlcPmXKFPPW3NzcvXv3Wu4fFxeXkZGxa9eu/q3eFdg1xQEAAGZBQUF79uyxGrxn+IZVgvyiRYtsD5TL5S6ob+CdJPzyyy9/+eWXnrY2NzfbDur1erqlAXWo+rMuliKM+uLvNXweF9M42rX02VKPuq7MRdo7kL7Lp9PIxWyDFr0w2LmFdKww8Br06tWrX3311fs9qhuJKOTbH/Ww3CD6bnDIIC8vLs5EtbQZQ8K5+JeOftHpvFBIqAh3HRgYevzwxlHubtB+fn7z58/vaavd3zuEQiHl7a8TB/dnXSwl7qiPjhrk48PFmaiaak2cbBDuKjDg8VA7XxcXwcVfm7Q6WiDwoH/tsIoDAMBJdtMzzIPR0dFJSUnnz59nBgmCeOmll5gDu7q6SJKcNm0a8/Du3buTJk0KCQmRSqU//PCDCys00QIHXy55CWjQAACWspuewQwqlco1a9YsW7aMGXzggQdKS0uZBXlHjhyRSqXmJ3n11VcfeeSR27dvX7t2rdf3zcIFGjQAgNWY9AzbEAitVhsQEGDe5+mnnz5x4gRC6Isvvpg6dSozfufOnaKionfeeYcgCJIkAwMDXViYCfEdfLnkJdg13WMwGJjbRFpSqVQI9e8dwwAArGWVnsFcsa3T6dRq9enTp827TZs27aOPPnriiSfq6+vz8vKuX7+OEFIqlSNGjJg9e/YPP/zwyCOPbN26ddAgl53YcNU8hgPsatDTpk3bv3+/1SBJkmLZY1jqAQBgZDc9g7liGyFUVFQ0Y8aM8vJyZjwlJaW8vFwul0+aNMn8DBRFlZaWbt68OTU19eWXX964cePq1avd/0Z6jV0N+o9//KPtoEKhOHLtlvuLAQDg5Tg9Iy0trbm5ubGxMTQ0lBl55pln3n333cuXL5eVlTEjYWFhISEhzF20c3Jy/va3v7mwPBMNqzgAANzWU3pGSUkJRVEhISHmkXnz5q1ZsyY6Oto8Eh0dPWTIEGa64+TJk6NGjXJhYSYkcPDlkpdg1ydoAACwZU7PWLBggTkPWiwW79mzRygUmlP8JRLJa6+9ZnXsp59+mpeXp9PpEhMTP//8czdX3kfQoAEAbCQQCOymZ9iGRAsEgtbWVsuRyZMnT548mfn+kUceYT5Bu5wbpjhc2aA7OzuZdYgOmEy9uVcxz6TnUbp77zfgEIju6DAajbjrwMFopDUaz4llcB2dztRtoDVaLl5JaPCov3APa9AymSw+Pt7xPj///PP9Pq1Go/EzqAKIzt7W5cH0NHXp4E88ujf/q3m6zqDhZy92464CA61Wj7qMbc1cfO8dekGwpAt3FSzi4gZ9z1sxjhs37n6fliRJ37CRPkPDe1uXB7vz3dfpd6+LTff4vWRA+mpE1BMTknBXgUFN1W29ovShkHbchWBQ+gspFotxV+EsmmvroAEAwFO46nJBB2CZHQCApVpaWmbPnh0aGiqVSrOysphrUvLy8oYMGeL8grmNGzfGx8e//vrrdo+1Gqmvrx8/fnxYWFhUVNS2bdtc/YbuGzRoAABLTZ8+PTQ09ObNmzU1NStWrKipqUEIzZs379ixY84cTtO0yWT661//evny5e3bt9s91nZk5cqVDQ0NFy9eXL9+vfkyRbsgzQ4AwFGlpaVKpXLt2rVMQnRqaupzzz2HEEpPTzdnJDF27tyZmJg4atQo5jLuO3fuxMXFvf7661lZWVOnTr1169bEiROZ2xLaHms1Eh4e/vjjjyOEgoODY2Jibt1ydA2zieY7+HLJHwKeOWidTpeamhoUFOTMzhKJ5OkZr/R3SQAAVikrK0tOTubd6/bYlZWVmzZtunLlire39xNPPPHYY4+NGjWqqqqqoKAgJiYGIRQSEvLNN9/c76srFAqFQjFmzJjeFe8qeBq0Xq8PDw//8ssvndlZoVBcu8XFM9oAcJztNSm2ioqKnnvuOT8/P4TQtGnTzp07N2rUqKioKKY7905ra+sLL7ywY8cOkiQd7OaGNDuY4gAAsFFCQkJpaWnvLm3ry1o9nU6XnZ29ePHirKwsx3u6IQ8aGjQAgI1SUlKioqKWL1/OXJ9cXFxcWFhou1taWtrRo0fb2tq6u7v37duXnp7elxc1Go1Tp07NzMycM2dOX57HVaBBAwBYSi6XNzY2hoeHR0RErF27ViaTIYRycnLGjh1bWVkZFha2a9eumJiYt99++9FHH/3d736XmZmZkZHh4AmtjrUd+fbbbw8fPrx9+/awsLCwsLBDhw45eDY3rOKAC1UAACwVFBS0Z88eq8GDBw9ajcyfP3/+/PnmhyEhIVevXjU/vHPnjoNjbUecmfhmeFgWhzOam5ufeuopiqKam5udPESj0bQqr7XV/NivhbETbaQKAlJxV4EHrdIc2nsBdxUYmEwmZBpU0eKDuxAMKBPh/9+hzxzn7gZ9+fLl7u5ujUZj9+YpdpEkOTz+d/7DuJjFUXn+xESyWszzqIwvFzmliXhKosZdBQbVreJ2IZk8kosJWddreD4+HvM/E93bM4FDhgwRiUQIoUGDBlVWVjrY090N2sfHx8fHh8/nE8R9pCny+Hy+FzfvG0sICaOIx8WfVYJHiPhcfON8ghbwCZEX7jpwuNeiZ3YxUM5Ohljh8/m2Nym3y6P+PAAAXGKbxUFRFEEQYb+prKykKMrb29v22EOHDhEEwdxeFiFEURSPx4uOjpZKpZmZmU1NTe59K70EDRoAwFJ2szhEIlHDbxxcjSKXy9PS0vbt22ceEQqFSqWyuro6JCQkPz+/7+UZjCYHXw4ONBqNI0eOTEhI+Oyzzxy/BKziAACwEZPFcezYMeZq79TUVISQ+faDjnV0dFy4cOHs2bOTJk1atWqV5SaCINLT0++ZXO8Mx13YgStXrkgkkp9//nnChAkJCQnMTcftwtOgCYIoLy9ftmyZMzuTJJky/tn+LgkAwCo9ZXGYbxorkUh6CtkoKCjIzMyUyWSBgYGlpaXJycnmTUaj8fjx47m5uf1Vtz2bN29m4vS+/fbb4cOHM/VHRkZOmjTp+++/Z12DJkny3//+t8FgcGZnlUrFxVNFAHCe3SXJIpGotrbW8YFyufyNN95ACE2bNk0ulzMNWq/Xx8bGNjY2xsfHZ2dn97085z9Bv/nmm2+++SbzvVqtNhgMwcHBTU1NX3311ccff+zgQGxTHElJzt7NSKFQ1KhhaSQA3JKQkLB69WqTyXTPQDsrKpXqzJkzZWVlBEEYjUaCIDZu3IgQEgqFFRUVbW1tubm5GzZsWLlyZR8rpIy9WcVx+/btKVOmdHR0CIXCV155xfGlj3CSEADARk5mcdg6cODArFmz6urqamtr6+vrIyMji4qKzFv9/Py2bt26bds2LaYrYuLj4ysrKxsbG3/++ed7TvNCgwYAsJTdLA5b3d3d5oV3mzZtksvlU6ZMMW/Nzc1l0vrN4uLiMjIymCyOvuj1Kg7nwSoOAABL2c3i0Ol0lg8FAoHVVPXSpUstHy5atMj2QLlc3vfyDFS/nx3zgAat1+ub66o1d+/ce9cBx6jvvtgewie4eJZUY+R/U++PuwoM2nRelAB1dHHxt1t1BxEcxcVgg554QINGCDWoTZ1qI+4qMAgl6KFS0ovfyytKPdrdhgfImAjcVWDQ/UsL3XU3ZDgXsw0Mt5xa2cUSrprHcMADGrRQKOwgBrXQAffedcAZSjdFhxp97FzIOvBV3hVGykbgrgIDgkeYWtVx0Vz8W9d2mZhbxHoEQ69WcdwXR38WpaWlLS0tzj9XZ2dnn+sBAID/OHToUE5Ozo0bN2JjYxFCFEV5eXm99dZbH374IUIoPz+/o6Pj/fff9/X1ZaaYv/766wULFpw8eXLEiBHMIEVRQqFQKpUihHx8fLZv3z527FiEUF5e3okTJ4YOHVpWVob1LTriqEEvWLAgJyfH+efq6urqcz0AAPAf5kgN8xXbYrH4wIED77777uDBg612Pnv27B4W9dwAACAASURBVKuvvvrVV19JpVLLi8KZFA6E0Jdffrls2bILFy4ghObNm7d48eKXXnqp17VReKc4fHx8/vSnPzn/XMePH+9zPQAA8Cu7kRpCoXDu3LmbN29eu3at5c4XLlyYM2fOkSNHHCQoabXagIBfJ0vT09OZrt1rnJuDrqqqam9vtxpUqVRYigEA4NVTpMbChQuTkpLeeecd854Gg2HixImnTp1KSEiwfR7mIm+dTqdWq0+fPu2m6l2BXQ36xRdfnDBhgtUgSZJoyINY6gEAYGQ3UgMh5O/vn5eX9/HHH4vFYmZEIBCMHTt2165dlrlIZsxF3gihoqKiGTNmlJeX8/kuuJ0g5z5B+/v7r1+/3mpQoVDs/qZPv4kAADxOT5EajLfeeislJWXOnDnMQ4Ig9u3b9/TTT3/wwQcOQjbS0tKam5sbGxtHjHDBAiE3rOLg4mJ4AAD7OY7UGDx4cG5uruXl2mKx+PDhw/v373eQgl9SUkJRVEhISP+W7jrQoAEAbHTPSI2lS5da3bkqICDg+PHjGzZsOHjwoOU4EyEtkUjy8vL27NkjFAoRQjk5OWPHjq2srAwLC+tdLoeBMjn46sUT2mLXFAcAADCswvjNkRqtra3MN8OGDTPHa5i/GT58OHNnLPOgbVgHw6qJ9wLmZXYu0d7ebjQ6e5W2yWT/DQuQUYj0rivKYxAE6tCh/v9nwEZGo7Gzg4s54Hqd3qinNR1c/Fs3GLiYauBA/zbomzdvpqamxsXFObm/+b8+SxqNZqigbTCfi1fBeOmN35XTPAJ3HTh0dFEXzt3AXQUGVJdWwDO1aT0plcJVNB10cJjH/KR7/CoOvV6flZW1c+dOJ/cfN26c7SBJkl1kmE4c7NLSPEPAL1eeHN3lI8JdBw7yKz5RD6fjrgKDlvqfA3h1KQ95zIksF/qh9Bfzyjn2c0PcKJwkBAAAloIGDQBgqZaWltmzZ4eGhkql0qysrPLycoqiCIIw3z+lsrKSoihvbzvJf4cOHSIIgrk+BSFEURSPx4uOjpZKpZmZmVbLP3qHokwOvvr+/AgaNACAtaZPnx4aGnrz5s2ampoVK1Yw56hEIlHDbxzEbphTlswjTGRSdXV1SEhIfn5+38szGEwOvvr+/AgaNACAnUpLS5VK5dq1a5mE6NTU1Oeee87JY5mUpV27dlk2aAZBEOnp6Q0NDS4ut3+wax20j4/PU089ZTUokUjinp6BpR4AAC5lZWXJyck8nvWHSOaqE4SQRCKxWitt1lPKEkLIaDQeP348Nze37xUaqH6/zRO7GvTRo0dtBxUKxZFrt9xfDAAAL7sXmIhEotraWscH2k1ZYjLtGhsb4+Pjs7Oz+14e5aJ5DAdgigMAwEYJCQmlpaU9XbzmAJOyNHfuXIlEsmnTpi+++IJp9EymXUNDA0mSGzZs6IeSXQ8aNACAjVJSUqKiopYvX67X6xFCxcXFhYWFzhzoOGXJz89v69at27Zt02r7ep2qG7I4oEEDAFhKLpc3NjaGh4dHRESsXbtWJpPZ3a27u9u88G7Tpk33TFmKi4vLyMjoXUCSJcpgdPDVxydnsGsOGgAAzIKCgvbs2WM1aM5FYthmIS1dutTyoTllyfJAuVzuykL7jQc0aI1GM6hV6dNmJ6ZjwOPRxv1FQk5GcSADrS47WYC7CgxMJtMtwlRZwcU7vVEUHejvMQlZbrjU2wMaNEmSjfQwtckPdyEYxBJVj2RkeXl7TDqBC1Wd2/f0Q7iLwEHZiNrooAeTuRg+8+P1Zh8fH9xVOMvgonkMBzygQSOETIgwIhfcQ8zj0DTiC4VeQiHuQjDgEYTIi4vhkwI+EiCeSMTFf/A8Pjd/XewRnCQEALAUZHFAgwYAsBRkcUCDBgCwEWRxINfOQdfV1S1btsxyRK1W19fX9+Kp7ty5s2XLFuZ7kiTRkAddUB8AwHOwP4uD8qwsjsLCQqslivX19QcOHOjFUymVyubm5gULFiCEVCrVuVrKNSUCADwHy7M4XDWP4YArG3RCQoLViJ+fn938I2cMGzYsJSUFIaRQKM7VKvtaHADAoyQkJKxevdpkMtl+iHaMyeIoKysjCMJoNBIEsXHjRvRbFkdbW1tubu6GDRtWrlzZP4W7EsxBAwDYCLI4EDRoAABrDdQsjry8vCFDhowaNco8cvbs2ZiYmMjIyPfee89yT8+4UAUAwEEDNYtj3rx5ixcvfumll5iHNE3PnTu3oKAgLi4uNTX12WefTU1NZTbhadAURe3Zs8dgMPS0g1Kp7OjoYL7X6/VBdIsv6nBXdSwiIIyK0u94PC5eVNapRWd/wF0EDu0dhJ7QdHT2+NMxgKlbuocGe8yKgF7PY6SnpyuV/zmvdvXq1cDAwMTERITQzJkzDx48iLlB37lz57PPPpszZ46zB4h8kIDsz4rYSqsJCRkkEHCxQdepUUgg7iIw0RH8YSF2ro4b8FyV0ukerrpcsLGxcfjw4cz34eHhxcXF5k3YpjhGjhw5f/78nrYWFRWdOHGC+V4oFHZ7+XUKB7urNBZ5QNcYHhnqLRbhLgSDX6rK4iJwF4EDj0e3E+K4OH/chWCg1VLMZSmcYjlFY3UHGc79WQAAPEVLS8ubb7558uRJb2/vmJiYDz/8UCaTeXl5mT9vnj59OioqytfX12piGiF06NChnJycGzduxMbGIoQoihIKhVKp1GQyyWSy3bt3Bwf3NS/w643POrnn5s2bt2/fjhD69ttvzcWbhYWFma9sbGhosNwBVnEAAFiK5VkcznvzzTeVSqVSqbTtzgih0aNHt7S0XL9+3WAw/POf/5w8ebJ5EzRoAAAbDeAsjpycnLFjx1ZWVoaFhe3atYvH4+3cuTM3N1cqlT755JNpaWnmPdk7xXH79u2SkhKEkEqlQpwMgwaAy9ifxdFrBw8etBrJyMioqqqy3ZOlDTo6Onrw4MH79+9HCJEkyZc8grsiAIC7sTyLww1Y2qBDQkLWr1/PfK9QKA5e6U0kHgDAc0EWB4I5aAAAO7E/i8MNoEEDAFiK5VkcbsDSKQ4AABioWRzO6/cG3d3drVarrQbb2trsTv/3hGeiBCbrhehcQCCk7ewyGj3p+ldXMZpojQf8Dup6Oj3qRkaNRo+7EAzcEIHvWfq3QQcGBt69e/fFF1+0Gu/u7naQlGRFo9GIdb8IdC2urs4D8GmqrKTifk+SDAwdBvJU+SDcVWDQpdXyqLZ2VTvuQjDQdNLBwV24q2CRfm/QX331le14Q0PDu+++6+STkCSp4gWrCS4G58ionx5+/EFuZnF8e+L7hzPG4a4Cg/rqGr669KEELk4/ltygxGIx7ipYhIsfzQAAwCNAgwYAsBFFUTweLzo6OiIiIjExccuWLUzSG0VR3t7/yWLNz89ftmwZQohZUdfS0oIQUqvVUqm0vr7eZDI9/PDDEokkIiJiyZIlNE3bjuB6g86ABg0AYCkm3qiurq6wsFAul69du9bBzuHh4QsWLGCa9bJlyxYsWBAeHs7j8Q4fPlxbW1tZWVlcXHzixAnbEXe9m96ABg0AYDuJRPLJJ59s3brV8W5vvvnmpUuXtmzZcvHiReZSb4RQSEgIQshkMpmjlm1HWAv/iYgZM2Y0NTU52EEikQx/BGeyCQAAu9GjR6vVapVK5efnZ85LQgi1t7ebb/3h5eW1adOmzMzMM2fOeHl5mY9NSEioqanJy8ubMGFCTyPshL9B37p16+zZsw52UCgUe8/XuK0eAAAL0TRtni+2zEvKz89vbm427/bVV18NGzasrKxs3Lj/LAH66aefWltbc3JyLl++/Mgjj9gdYSeY4gAAeIBr164FBgYGBQU52Ofq1asnT568dOnShx9+aPV7ub+//4QJE44cOeJghIWgQQMA2K66uvq1115buHChg31oml6wYMGWLVtGjBjx9ttvMxd8NzU11dXVIYRaW1u//PLLuLg42xH3vIXewT/FAQAAdjFzzQaDISAg4OWXX168eLGDnXfu3DlixIinnnoKIfTaa689/PDDRUVFwcHBzz//fHNzs5eX14wZM6ZPn15VVWU14q530xvQoAEAbGSbgmQet4w9WrJkCfPN/PnzzWcL+Xw+cz8mhND169ctD5fJZFYjbIanQXt5eZ05c4b5v66iosLxzhqNJsTYMBQ1uqU0duEj08kvzxG4y8DCRKOT+63vDMQFJpOJoKkbP1O4C8GAMiL/YZyMyOoBngY9dOjQiooKiqIQQpa3sLWLJMlrdx+4qeHiFfpPj2h6clK6yJuLWRwl3347bvxI3FVgUFPd0tJGxI3m4nuvKqvx8fHBXQWLYJviIEmS+caZqDYjTRhMXDyfSdNI4OXlJRTiLgQDHo8nFHFxCk4g4An4PKHQ6967Djg8Phd/zB2APw4AAEvl5+fLZDKZTJaUlFRUVGRO55BKpdnZ2Wq12u6IZVIHQqi+vn78+PFhYWFRUVHbtm1DFikf0dHRSUlJ58+fx/T+7g0aNACAjUpLS3fu3FlSUqJQKE6dOjVixAj0WzpHdXU1SZKbN2+2O2Jr5cqVDQ0NFy9eXL9+fXl5ufkopVK5Zs0aJr6DnaBBAwDY6M6dO0OGDPH19UUIBQcHMw2aQRBERkaGUql0MGIWHh7++OOPM08SExNz69Yty61arTYgIKC/3kOf4Z/joyjK8f9gJEkiZP9mkQCAgWrcuHGrV6+OjY0dP378888/b3nptl6vLywsTE5OdjBiS6FQKBSKMWPGMPvHxsbqdDq1Wn369On+exd9hL9B7969m4lw7YlKpar7rs1t9QAA2EAsFl+6dOm77747c+bMzJkzly9fPn/+fObSFR6Pl5aWxuTV2Y7Y1dra+sILL+zYsYMkSYqihEIhs8C3qKhoxowZ5eXlfD7ffe/NafgbdGRkZGRkpIMdFAoF+u6a2+oBALAEQRBjxowZM2ZMfHz8li1b5s+fbxmThBCiKMp2xPZ5dDpddnb24sWLs7KyrDalpaU1Nzc3NjZaTqGwB8xBAwDYqKKigrkakKbpy5cvR0RE9O55jEbj1KlTMzMz58yZY7u1pKSEoigmIZqF8H+CBgAAW0aj8e23366treXz+fHx8Tt27HDywO7u7rCwMOb7xYsXp6SkHD58uKSkZPv27QihrVu3Tpw40ZwoLRaL9+zZI2TrpQbQoAEAbJSQkPDNN99YDVqmcCCbXA7UQ4KHMyPs5AENWq/XR5KdQ8S6e+864Ij4prIr17l5eZVGoyv6los3amhv0+kpQtvZhbsQDNpbOyKGReGugkU8oEEjhB4dJQgO5mIexeXvu8NDury8uNigtWrT8CFcDAzim0zX6ryUnZ7xs+laviZuJoP1yAP+EQiFwmHDvGVRXIxQuXq9PTpqkI+PB/w1uVxNtSZuJBf/0nkEcaXeu40/GHchGAhog0DAxX/tPfn1z4LP57/44otW2+xelgMAAG5QW1s7fvx4cxdasmRJSEjIG2+8IRQKpVKpyWRKTEz8xz/+QZKkl5fX8OHDmd1Onz4dExOTl5d34sSJoUOHlpWV4XsHLvBrgy4sLLSaa0cIZWdnu70eAABwhInRoGl65syZmzdvXrlypUgkamhosNxn3rx5ixcvfumll3AV6Sq/NmixWCwWWwcus/PSGgAAYMI3Tp48aXdrenr6wJgAYN10j8FgsLohjUqligjDVQ4AgI0swzfMi5olEontyjyPxroGffHixeXLl6elpZlHSJKMCIvBWBIAwP0IwnpFBzNiG75hdbX3QMK6Bk3T9Pjx41etWmUeUSgUNOUxN3kEALhEUFCQZYyaSqVKTExENu3YbvjGgMHFBbYAAPbz9fWVSqUFBQUIodu3b586deqxxx7DXZS7QYMGALDU7t27P/roo+jo6IyMjHXr1kVHRzt5YE5OztixYysrK8PCwnbt2tWvRfYr1k1xAAAAIz4+/ty5c5YjdsM3bJcIHzx4sN+LcwvMDdpoNLa3t1uOaDQa2xwTnc6k0QzkmaaemGjU0WE0GnHXgYPRSGs6uPiXrtOZCJoSmLpxF4IBj+bkv/WeYW7QW7Zs+de//hUUFGQeUavVMTH/tWZDo9HU1GgbbnPx32t3N/3dpV94nJyI6tIazxc1464Cg84u5McTPSDgYqviIV1XFxdTonqCuUF3d3evX7/+6aefNo98++23Z8+etdyHJMmkpACZjHR7dfjJ5bVPPiry8eZihz56pvOZx7h4qVRlnembm0OpwVwMdeM3/2x7xRyXcfEnHwAAPAI0aAAAS7W0tMyePTs0NFQqlWZlZZWXl1MUxePxoqOjpVJpZmZmU1MTQoiiKIIgwn7z7rvvmneLjo5OSko6f/48s5uDQalUmp2drVarMb/n/wYNGgDAUtOnTw8NDb1582ZNTc2KFStqamrQb2FJ1dXVISEh+fn5zJ5MXhLjL3/5i3k3pVK5Zs2aZcuWMbs5GKyuriZJcvPmzTjeaI9gmR0AgI1KS0uVSuWxY8d4PB5CKDU1FVlcN0gQRHp6ek9hSZa0Wm1AQMA9Bx2nL+HCugYdEBBw5MiRCxcumEckEsmSJc9hLAkA4H5lZWXJycm8HtYwGY3G48eP5+bmMg/NeUkIoZUrV86aNUuv18fGxup0OrVaffr0aWaT3UHzJnP6EnuwrkEnJSV9//33liMKhYKmb+CqBwCAi91buzJNtrGxMT4+3pxZbxvQIRQKKyoqEEJFRUUzZswoLy9HCNkdtE1fYg+YgwYAsFFCQkJpaanJZLIaZ5psQ0MDSZIbNmy45/OkpaU1Nzc3Njb2NMg095qamt27d/v6+rrwLfQdNGgAABulpKRERUUtX75cr9cjhIqLiwsLC81b/fz8tm7dum3bNq1W6/h5SkpKKIoKCQm55yALsW6KAwAAGHK5/I033ggPD/f29h41atRHH31kuTUuLi4jI2PXrl0LFizo7u4OC/v1vh7PPvvs9u3bzbPSYrF4z549QqGQoii7g+5+V/cDGjQAgKWCgoL27NljNWgZjSSXy5lvbGerbUcEAoHdQdusJfbwgAat0WiuXWu6dOku7kIwMBjo/V9pbe4swQkmE737qPX8IxcYTYhnqvdSN9571wGHoE1abSTuKljEAxo0SZJjhrZK/bgYoXKwasgzWSFiMRdPFXx9uvnpsew6Y+Meyjr9lYZgUehI3IVgoL9T4+Pjg7sKFvGABo0QEhC0iM/FD1MEQkIhIRJysUHzeIRIyMXfHQR8guDzCYEX7kJw4GZyY8/gjwMAwFIOsjgiIiISExO3bNnCrMMzmUwPP/ywRCKJiIhYsmQJM9f86aefxsXFxcXFTZ48WaPR4H43vQENGgDAUg6yOOrq6goLC+Vy+dq1axFCPB7v8OHDtbW1lZWVxcXFJ06cUKvVK1asKC4uvnHjBkmSHnrjK8+Y4gAAcI3jLA6EkEQi+eSTTzIzM1esWIEQYhY1m0wm5jM1TdM0Tet0OqPRqNPpQkND8byNvullg75w4YLlovFeu3jxovkKeitdXV1r1qwxmUwkST7/ICfn4wDgMMdZHIzRo0er1WqVSsXclSkhIaGmpiYvL2/ChAkEQaxfv37kyJGDBg169NFHX3zxRXcV7kq9bNCnT5+WSCQPPfRQH19eo9H4+fnZ3dTc3Hzt2rUPPvhApVIh6mofXwgA4HHsZnFY7WC5z08//dTa2pqTk3P58uVRo0Z9/vnnFRUVISEh06dP//TTT1999dV+rtf1ej/FIZPJUlJS+vjyJ06c8PLq8dNxUFBQSkqKQqGgldCgAeCWhISE1atXm0wmBx+ir127FhgYaHlTU39//wkTJhw5ckStVvv7+zOXF06ePPn48eOe2KDhJCEAgI0cZ3EghKqrq1977bWFCxcihJqamurq6hBCra2tX375ZVxc3IgRI3744Yfm5maapr/++murW1F7CjhJCABgKbtZHEyehsFgCAgIePnllxcvXowQam1tff7555ubm728vGbMmDF9+nSCIBYvXvzII48QBPG73/1u0aJFuN9Nb0CDBgCwlN0sDrsT0zKZ7Pr161aDf/rTn/70pz/1V3FuwboGff78+Rs3biCEmJOzCCG9Xq9s8WnsEOIuDYMuI1H8XTufz8UL6to7TGcvdeKuAoN2DU11NdHd7E3w6T/GLg1FReOugkVY16DXrVv33HPPWZ059PM2DvVldSpgP6nToOAQHy8vLp4qaFF1hfjf4yT+wEQhtdFb6Bd07z0HHANtxF0Cu7CuQSOEXn75ZW9v7/r6+p9++gkhJBQKQx8wjAzkYljS1TuDoqMG+fiw8a+pv9VUquIiuPirA49AjbSfKHgE7kIwMOl1AgEX/7X3hIsfzQAA7FdbWxsd/Z/pjiVLluTn55uzOKRSaXZ2tlqtpiiKIIiw31RWViKE8vLyhgwZMmrUKHzluwY0aACAJ2GyOKqrq0mS3Lx5M0JIJBI1/IZZTjdv3rxjx47hrtQFoEEDADwPQRAZGRlKpdLu1vT09ICAADeX1B9Y3aBVKlVJSYnl3dQBAAAhpNfrCwsL4+Pj0W8royUSyRNPPIG7Lhdj73z84MGDR48evX//fpIkIx4S4S4HAOBWhM2t3pgRph3zeLy0tLQ33ngDISQSiQbqxzj2NmixWLxu3TqEkEKhoGsP4y4HAOBWQUFBLS0t5ocqlSoxMRHZtGOW35a7j1g9xQEA4CxfX1+pVFpQUIAQun379qlTpx577DHcRbkbNGgAAEvt3r37o48+io6OzsjIWLduneWqO8dycnLGjh1bWVkZFhbmofdSYbB3igMAwHHx8fHnzp2zHBEIBDqdzvEIQujgwYP9Xpxb4G/QHR0darXa/NBotHOtp47iabq5+GHfRKOODrt/JAOf0UhrtFy8klCnRyajwaTT4i4EA5NxIE8o9wLmBp2YmPjxxx9/8skn5pEff/zRKq1Ko9HUNHvXt3ExLKnbyP/uQgOPi20Kdel556+ZcFeBQWcX0uvvGrRcDIqi9V1dXVLcVbAI5gY9ceLEiRMnWo5kZWVZLa8hSTIpRMvNLI59ZYOfHE35cHKR4dHvvZ95lIv/NVXeRMW3hwlCZbgLwYC6rRSLxbirYBEuzhsAAIBHgAYNAGCplpaW2bNnh4aGSqXSrKys8vJy22gkiqK8vb1tjz106BBBEBUVFcxDy5SlzMzMpqYm976VXoIGDQBgqenTp4eGht68ebOmpmbFihU1NTXIXjSSXXK5PC0tbd++feYRc8pSSEhIfn6+O95An0GDBgCwUWlpqVKpXLt2LZMQnZqa+txzzzl5bEdHx4ULF3bt2mXZoBkEQaSnpzc0NLi43P6Bf5ldT1JTUwcNGoQQkkgkS17wyDvyAgB6raysLDk5mcez/hDJZHEghCQSyTfffGP32IKCgszMTJlMFhgYWFpampycbN5kNBqPHz+em5vbX3W7FHsbtEgkOnnyJIIsDgC4yu79YZ2JRmJuB44QmjZtmlwuZxq0Xq+PjY1tbGyMj4/Pzs7uh3pdD6Y4AABslJCQUFpaajLd91p4lUp15syZuXPnSiSSTZs2ffHFF0yjFwqFFRUVDQ0NJElu2LChH0p2PWjQAAA2SklJiYqKWr58uV6vRwgVFxcXFhY6c+CBAwdmzZpVV1dXW1tbX18fGRlZVFRk3urn57d169Zt27ZptR5wrSY0aAAAS8nl8sbGxvDw8IiIiLVr18pk9i/e6e7uNi+827Rpk1wunzJlinlrbm7u3r17LfePi4vLyMjwiBAl9s5BAwA4LigoaM+ePVaDtmFJVlPVS5cutXy4aNEi2wPlcrkrC+03rGvQQqHw2Wef5fF45vMAGo3m2k3yUr0v1rrwMJh4+4u4GcWBTDS9+ys754gGPKMJUXQt1VyPuxAcaKNWOwJ3ESzCugb9f//3f8zc0OTJk5kRkiTHROulwQasdeFx8IrPM0/4iEVcbNFfn+t4+sFu3FVgoLzjdbVVNiici0tLtY1KHx8f3FWwCOsatEgkEolECCHL9Y8CPi3y4uKHKQIhoRchEnKxQfN4hMgLdxE4CHg0wePzBJx88zarnjkO/jgAAGxUW1treQuVJUuW5OfnW0ZqZGdnq9Vq23QOhFBeXt6QIUNGjRqFr3zXgAYNAPAk5kgNkiQ3b96M7KVzzJs379ixY7grdQFo0AAAz0MQREZGhlKptLs1PT09ICDAzSX1B9bNQZvV19cvW7YMIUSS5POPk7jLAQCwiF6vLywsZK7hdiadw0Oxt0EfPny4q6sLIaRSqRCqwF0OAMCtrO6sZB5h2jGPx0tLS2MCN5xJ5/BQ7G3Q8fHxzDcKhYL+BRo0ANwSFBTU0tJifqhSqRITE5FNO6aogXyfWZiDBgCwka+vr1QqLSgoQAjdvn371KlTjz32GO6i3A0aNACApXbv3v3RRx9FR0dnZGSsW7fOctWdYzk5OWPHjq2srAwLC/OIzI2esHeKAwDAcfHx8efOnbMcEQgEtlkcViMIoYMHD/Z7cW7hAQ1ar9crG70aW/i4C8Ggy0AU/6Dnc/Gto3YtcfZHIe4qMGjX8roNvxj1XbgLwYDSaihKirsKFvGABo0Q8vcXBAdycTamTmWs1EloHhc79DBhVchQz/j36WLN9G2VoBMNwl0HBgLExfQVBxz9AJhMJrVabXcTswDOPYRC4bDBfNkILuZRXK00tvkOpQVc/CDp3aaMk3Dxf2UeYbrRSVIPhOIuBAOC0jO3iAUMR38Wjz322CuvvGJ3U3l5eVJSUv+UBAAAiKIoLy+vt95668MPP0QI5efnd3R0vP/++15eXsOHD2f2OX369FtvvZWXl/f//t//QwidO3fuzTff/O677w4fPpyTk3Pjxo3Y2Fic76HPHDXodevW9bTpgw8+GDp0aD/UAwAAvxKLxQcOHHj33XcHDx5sHmSSN8wPt23bNn78+GeeeWbQoEELFy7829/+xufz5XJ5Wlravn37Vq1ahaFuzISBBQAAHdhJREFU1+Hir5AAAI8gFArnzp3LJCL1JDIy8uWXX37//fe3bNmSlpb20EMPdXR0XLhwYdeuXfv27XNbqf2EjdM9SqWyra3N/FClUkUEYiwHAIDNwoULk5KS3nnnHfOIbfLGkiVLUlJStFptSUkJQqigoCAzM1MmkwUGBpaWljJ5HR6KjQ166tSpTz31lPkhSZIRE4ZhrAcAgIu/v39eXt7HH38sFouZEdvkDaFQ+NJLLzU0NPj7+yOE5HI5k9Exbdo0uVwODdrFHnjggfXr15sfKhQKur3Iwf4AgAHsrbfeSklJmTNnjoN9eDwecw8mlUp15syZsrIygiCMRiNBEBs3brTNXfIUMAcNAGC1wYMH5+bmOnnF9oEDB2bNmlVXV1dbW1tfXx8ZGVlU5MEf76BBAwDYbunSpU1NTc7sKZfLp0yZYn6Ym5u7d+/efqur37FxigMAAAQCQWtrK/P9sGHDzIEbtskbCCFm0hkhZBXYv2jRon4ssf+5tUF3d3drtdp77mYymaxGdHpa09k/NbGbiUaI0hG09R8IFxhNtKaTi7dy1+kRMhoIg502NOARpoEc7twLbm3QkydPNhgM95ywr66utnyo0Whq6lHD3f6sjK269YSP+mfksac4+qIL0eevGnFXgUFnF/Ixtgs7a3AXgoHBoHVnjAT7ubVB6/X6kydPMidbHRg3bpzlQ5Ikk2JFMgkX8yjkhe2asFFIIMJdCAa+TWeeeYyLKVGVdabytvDgkaNwF4LBXWW5eTkdQHCSEAAAWAsaNACAjSiKIgji7bffZh7m5+evWrWKoigejxcdHR0REZGYmLhlyxbmlFVtba3l/VaWLFmSn5/PPEPYbyorKxFChw4dIgiioqLC6oUsd2NuxcIQi8UbN25071v/D1jFAQBgKbthSUKhUKlUIoRqa2unTp2q0WhWrFjR0zNYJSshhOzmKNnuxjykaVoikViu23Mz+AQNAGApx2FJEonkk08+2bp1q/NPeL85SkVFRcHBwSNHjnT+JVyLjZ+gSZK0zOKQSCRL/piOsR4AAC62YUmWRo8erVarVSpVT4dbJSv1lKNkG8DE2Lt37/Tp0132Zu4fGxv04cOHLR8qFAq66ztcxQAAMLINS7JE0zRN0wgh28W7zIhVslJPOUq2AUwIIYqiDh48ePXqVVe9l15gY4MGAAAzB2FJ165dCwwMDAoKEolELS0t5nGVSpWYmGi18/3mKJ04cWLUqFHDhuGM0oQ5aAAAq/UUllRdXf3aa68tXLgQIeTr6yuVSgsKChBCt2/fPnXq1GOPPWa1//3mKGGf30DQoAEA7GcZlsTMFw8fPjw7O3vatGnvvfceM7579+6PPvooOjo6IyNj3bp1lqvuGPeVo6TVao8fP56bm+vqt3J/YIoDAMBGPYUlMZPOtuLj48+dO2f1DJbJSj3lKFntxvDx8XFw7tFt+qVBHz582O7al7Kysl48m0ajuXat69IPXLxC30AhQnmRm1kcdwnT7qPcTIlCRrpKXc/FLA6T0aiNCMFdBYv0S4P+8ccfX3vttSeeeMJqvHfrvUmSHPNQQFSkjwsq8zT/Lrh9/GaIwcTFX3SmJzQ8/fgDuKvAQFnb3d4tSE7m4o04r19v9fHh4k96T/rrJ9/X1zcgIMBqkM/vZfaNQECIRFycLicIRNE8A83F987jESIhF391EAgIgZEnEnExKIrH4+LfuANc/MkHALBfT1kc3t7ezFYmlEMqlWZnZ6vVamY3q6gN8/5mJpPp4YcflkgkERERS5Ys6WlGmyWgQQMAWIrJ4mhubra7lQnlqK6uJknSfDm4OWqjp+fk8XiHDx+ura2trKwsLi4+ceJEv5TuItCgAQAs5TiLg0EQREZGBhOf5GTURkhICELIZDLZ3ryJbVhx9un8+fNHjx7taStJks/nxLqzHgAASzjO4kAI6fX6wsJC5qLtnqI2bCUkJNTU1OTl5U2YMKFf6nYRVjToU6dORUdHP/jgg3a3qlQqhDRuLgkAwAYOsjiYK1Z4PF5aWhqTsNFT1Iatn376qbW1NScn5/Lly4888ki/voW+YEWDRgjJZLKUlBS7mxQKBW380c31AABYoqcsDquEI7tRGw6e1t/ff8KECUeOHGFzg4Y5aAAAq/WUxWHFyaiNpqamuro6hFBra+uXX34ZFxfXL0W7CDRoAADbWWZx9KSnqI3u7m7z/as2bdrU2to6ceLE0NDQ0aNHP/HEE9jjkBxjyxQHAABY6imLg/nGNkCjp6gN25XO169f74d6+4W7G/TOnTttA1h/+OEHB9NAer2+ukrTeMs6zYQLunSm+AdURhMXL69q7zCevcjFk8PtGpPBpO/oMOAuBAO1Wh8cTOGugkXc2qBXrFihUCh6cSDfN1wYNMTl9bAfcbNEi3woTs5ECb10ISF2bqIx8BF6TbfPA0OG4q4DA233XdwlsItbG/QTTzxhm6CEEGpsbBw0aFBPRwmFwqCQkNDIqH6sjK1qyn78RU8aaC7ORHl7t8aN5GJuDo8gmjoekMrCcBeCgU7bLRBw8V97T7j40QwAwH62MRqWIyqV6g9/+ENoaGhsbOyjjz566NAh5Gk5G86ABg0A8DzTpk0LDQ2tq6urqKiQy+VVVVXI03I2nAG/TQAAPMz3339fW1t74sQJHo+HEJJIJOZrwT0oZ8MZbGnQCoWip2lolUpFDhvu5noAAKxVXl7+4IMPMt3ZlqfkbDiDFQ16/PjxR48eZfKobJEk+eQkaNAAgP8wzy+//fbbR48epWm6srKSGfGUnA1nsKJBjx07duzYsT1tVSgUqm4urgkFANgVHx+/Zs0ak8nE4/E+/PDD1atXSyQSyx08ImfDGXCSEADgYX7/+9+PGDHivffe0+v1CCHzBYeelbPhDGjQAACWsorRsNz0xRdfNDY2hoeHy2SyqVOnfvrppwghz8rZcEYvpzh4PN7mzZu/+OILu1uvXr0aGRnZh6oAAFwnEAhsFzKb8zcGDx68e/duq60ymcyDcjac0csGvWjRImbhoV2ff/55YKArbxqv1+m0HR0ufEJPQdMmEY/i0wNhwdD9MhppTQcXYxl0OpNeb+jUaHEXgoHBwMW/cQd62aAfeOCBnvL1EULHjx/n811203iNRtOkLO1oKHHVE3oQqhvF+LaYEBfDklo6vA5/y8U3btCZvIlb3a23cReCgaaTHjEsGncVLMKKVRyOkSSZFE3Iwrn4syo/aSrXR1CEF+5CMAj2bhz+YBruKjBov107jC57KFGEuxAMSn7qtr21FZfBSUIAAGApaNAAADaiKIogiLfffpt5mJ+fv2rVKmbQvLSjsrLy2WeflcvlzD7nzp1LSUkxGo2HDh0iCKKiogJf+a4BDRoAwFJisfjAgQPNzc2WgyKRqOE3MTEx27Zte//999va2iiKWrhw4aeffsrn8+VyeVpa2r59+3BV7irQoAEALCUUCufOnbt582YH+0RGRr788svvv//+li1b0tLSHnrooY6OjgsXLuzatWsANGh3nCSkafr3v/99rxfeSSSSJfPHuLYkAIBHWLhwYVJSkjmsDiHU3d3NXNgtkUiY+xAuWbIkJSVFq9WWlJQghAoKCjIzM2UyWWBgYGlpaXJyMp7SXcFNDdrf3//kyZO9O1yhUNBt51xbEgDAI/j7++fl5X388cfm1R0ikai2ttZyH6FQ+NJLLzU0NPj7+yOE5HL5G2+8gRCaNm2aXC6HBg0AAP3lrbfeSklJmTNnjoN9eDwekz6qUqnOnDlTVlZGEITRaCQIYuPGjbY3qvYUMAcNAGC1wYMH5+bm7tq1y5mdDxw4MGvWrLq6utra2vr6+sjIyKKiov6usP9AgwYAsN3SpUubmpqc2VMul0+ZMsX8MDc3d+/evf1WV7+DKQ4AABsJBAJzjuiwYcPMMUnmbywxk84IIea0odmiRYv6scT+5wENWqPRXLtuulSGuw4cDBSKpj1+sX3v6NqJ6nNHcFeBAW0yqpDpRjUXb1JBGWn/oVxMieqJBzRokiTHPDgoKoKL0QT//qpV5TvaxBPiLgSDhM7zT8nacVeBQbXKux2Jk0dyMcLweg3Px8cHdxUs4gENGiEkEBAioaeeh+0LAiETIaB5nvHX5FoEgUQC6zhgLuDzTAKCEHExIAv1cBtY7oI/DwAAG0EWB4IGDQBgLcjigAYNAGApyOJg6eTmp59+ar6akyTJ5yfCHQ4B4CLI4mCjf/3rX1u2bGG+V6lUCDU73h8AMCBBFgcbCQQC8z0PFQoFrYMGDQBHcTmLo18aNEEQX3zxRWlpKfOQpmmraX4AAHCSOYvDcY9mMFkcn332GfMwPT29qKho7Nix/Vxjf+mXk4R/+MMfpk2blmIBbgQJAOg1yOJwpdDQ0NDQUPNDk8m0fv36/nghAMBABVkcCOMc9I8//njx4sWetlpOiej1+mqlrvGO3i11sUtXN+1L1yDkqTNofaEx8b6pJnFXgUFbF48iUEcXF5fAqjuI4CgKdxUsgq1B/+Mf/xCLxSNGjHBmZ78A7+Bg7/4uiYXqGlvE/kGIx8ddCAbC1jsh/kbcVWBAI6JbIAgZxsVrvQ23oTv/F5yrOHJycnpaAWO+dhMhJBQKhw0Ty0b6uqsuFrl6rVXrF0J4cTEoStRxPTaEi782EQTSCPhxUVxs0FqdSSBg6dIyLLj4axQAwFNYpWpQFOXt/esv019//XVUVFRNTU19ff348ePDwsKioqK2bduGr1jXgwYNAGCvnlI1zp49++qrrx47dkwqlSKEVq5c2dDQcPHixfXr15eXl+OotF9AgwYAsFRPqRoXLlyYM2dOYWFhTEwMQig8PPzxxx9HCAUHB8fExNy6dQtPuf2ApdM9FEUxl9UjhFQqVYRTpxIBAAOK3VQNg8EwceLEU6dOJSQkWO2vUCgUCsWYMWNwFNsvWNqgZ8yYsX//fuZ7kiQjRsTjrQcA4H52UzUEAsHYsWN37dpltcSgtbX1hRde2LFjB0kOnNWZLG3Qr776qvl7hUJBm37CWAwAwP3spmoghAiC2Ldv39NPP/3BBx+sXLmS2Vmn02VnZy9evDgrKwtr1S4Gc9AAADZiUjXq6upqa2vr6+sjIyOLioqYTWKx+PDhw/v372cyN4xG49SpU/9/e/ce09TZxwH8OaWFCTQWBeTlMhqGDIqXCV6ggxnAjEsXcYy7LHMhZgnOsA1GMlzMmDG6hG06xRiZ+4NdKJlycxrqNjYNUlyU6Lh3lRVfHFBRiyC0xZ6+f5z3bZr2uBffF85z4Pl9/jo9tuT7cPnZPs9zficlJWUuzToWFyjQAAA++vuuGl5eXi0tLZ988kl9ff2lS5eam5urqqqY+2A1NDTgyLsgeDrFAQAg3JO6ath6cQQEBAwODjLHVuvSvL8wRwV6dnb2wYMH9mfM5qe4SMxotExOkngNKE1b0awRWUm84pmm0aSRxE94plnKRFsnp2jcQTCYNS/NOvs/46JAUxS1YsWK7Oxs+5NarXbHjh1zefnk5OTg9Tv//I3Eq35NaPnMUB+Zv7Meri5tuqWzHD93j2aQBVkeTrH0bFvyJh9ZfZ+dwZ2CRzgq0I2NjQ4nS0pKvLy85vJysVi8Fv21Gt1dgGh8V4c23PZ4jha44g6CQYigI20ziR8dBv4pePhYuGktiQ1YrveYoHe8PRI/QgIAwKIABRoAwFOVlZVhYWFhYWHr1q1j9tjdv3//jTfe8Pf3DwkJSU1N7e3tVSgUtuaXly9fjo6ONplMtoZKDJqmN2/eLJVKg4ODS0tLF9GKIuziAADwUWdnZ3V19fXr18VisV6vZzZv5Ofnb9iw4fbt20KhsL29fXBw8Pjx49u2bUtLS/Pw8Ni7d++XX37p4uLYP10gEDQ3N/v5+RmNxsTERJVKlZKSgmNMTw0KNACAj0ZHR318fDw9PRFCvr6+CKHOzk6tVnvhwgXmBt5yuZx5ZmFh4YcffhgcHBwXF7dp06bHj1l2fPn5+SGEaJqm6cW0PYaPBbqpqenw4cPMDwYhJJVKSzatwBsJAMCxhISEioqK8PDwbdu2ZWZmJiQkdHd3R0VFMdXZXmlpaXR09PT0tK3DGqvIyMjBwcGCgoLk5OSFDD6f+Fig9Xr922+/vXPnTuahRqOhfz2NNxIAgGPLli3r6Oj47bffWltbX3/99fLyck9PT9bpY1dX1zfffHN4eFgikfzNF+zp6TEYDBkZGVevXo2JiVmw4PMJFgkBADxFUdSWLVs++OCDqqqq77//PjIysrOzk3WOQiAQOL+zdiaRSJKTk3/44YcFCLsgoEADAPiov7+fmbKwWq1Xr14NDg6Ojo5+7rnnysvLmeuQ29vbz507N5cvpdfrh4aGEEIGg6GpqSkiImJBk88jPk5xAACAxWIpKSnR6XQuLi4ymezUqVPoPx2ig4KCnnnmmTVr1nz22WesrzWZTIGBgcxxcXFxenp6Zmbm+Pi4SCTauXNnfn4+d8P4/0CBBgDwUWRkpEO/JITQypUrv/76a+cnM339GUKh0Hmq+vfff5/vgFzAVqBdXV2LiopY733w119/2VYIEUKTk5M3kfQqCuYwHV/MIuE/HnbhToHHXReq5kcS30BYaITQbN+tWdxBMHhssUpWTeNOwSPY/gAOHDgwOTnJ+k81NTXe3t62h2KxePOysRDRA9YnL20NU2EzoS9SQhJ7cawcbX15E+4QOGjvoIdTdNTqxbRdd778Pihwd3fHnYJHsBVooVD4pGZJzj8hIUW7UST2zaGQFQmEyEWEOwgGAopyEy2aS3LnkdAFCV0oNxJ/5mgOGzHIAt8PAABP2ffi+Pnnn5kbpkgkErFYzBxrtVqBQBAaGhoSEpKens40nb93796uXbv8/f3Dw8NjY2NtN1gpKCjw8fFZs2YN1jE9HSjQAAA+svXi0Gg0P/300+rVq4eHh4eHhz/66KO33nqLOZZKpa6urlqt9tatW2Kx+PPPP0cI5ebm+vv7Dw0N9ff319bW/vHHH8wX3L1794ULF7CO6amRuAgDAOA/514cf4OiqKSkpB9//PHatWs6nU6lUjHXrUil0rKyMuY5W7du1Wq1Cx17fvGxQAsEgrq6uq6uf+9eEIvFr5G4gwMAojn34vibJ5vN5nPnzkVFRfX29m7YsGEuVxUuCnws0NnZ2S+88ILt4b1795C+DWMeAAD3nHtxFBUVOT/NZDJJpVKBQBAXF/fOO+/U19fbNkGXlJScP3/earUODAxwm33e8LFAi8Xi6Oho20ONRkNDgQaAPEwvji1btshksiNHjrAWaDc3N51OZ3sok8kOHDhA07RAIPj0008rKiqkUilngefdEvkgAABYYpx7cczlVRs3bnz22Wf37dvH9OswGAwLm3KBQYEGAPAR04tDKpWGhoZ2dXUdOnRoji+sq6u7c+dOUFBQWFhYTk7OyZMnmfMZGRnx8fEDAwOBgYGnTy+ODsZ8nOIAAADWXhzIqe0Gcysse97e3jU1Nc4vrK+vn9eAXFgcBXpk1t1C4jVlaNYqQBOjVoHjPdZIYDSjviHcIXDQP6CmjXTfbQp3EAzuPaT8cWfgFYr/N7idnp5merkS6NGjR+7u7hRF4t/qo0ePPDw8cKfAwGKxmM3mZcuW4Q6CAUVRISEhrq4kNp9htQgKNAAAkAkWCQEAgKegQM/J/fv3q6urExMTcQfhwuMnwJ1rwRE7cGcjIyNnz5597733cAch3eJYJMRlamqqubm5tra2o6NDoVDYLupf2uybcdtMTEws+dkwYgeOEKJpuqurq729/cqVK+3t7WazOTY2Vi6X485FOijQ7BobG5VKZWtra2JiYmFh4ZkzZ9zc3HCH4oj93v7Ozs66ujqVSmV/8f1SRezAEUISicTb21uhUCgUioMHD87xqhCw0GCRkB1FUevXr//qq6+ioqJwZ8FgaGiourr6/PnzERERubm5KSkphCysEzvwffv2dXR0GAyGtWvXyuVyuVwuk8mWTMuhxQsKNLu+vr7a2lqlUunh4ZGbm5uTk7Oor+h/WlVVVfv37z98+HBubi7rfSOXKmIHzrBYLF1dXWq1Wq1W9/T0+Pj4tLS04A5FNPgfkl1ERMTHH3+s0Wiqq6v1en18fHxsbOzRo0dx5+LInj17Ojo6xsbGkpKSsrKyzpw5MzMzgzsUF4gdOGNiYmJkZGRkZGRsbEwkEgUEBOBORDp4Bz0nVqv18uXLtbW1tuv6l7DGxkb7h729vUqlUqfTPXz4EFckbhA7cITQrl27urq6JBJJTEyMXC6PjY1dsWIF7lAAFgmfwLkJVlxc3Pr167GE4ZhSqXQ4I5PJZDIZljBcInbgCCG1Wj0xMREcHOzl5eXl5UXmNZw8BO+g2YWHhzuc+e677/Lz8/v7+7HkAWChjY+PM3vsrly50t3dLZPJ5HJ5ZWUl7lxEgwINHD3p0gyhcIl/3iJ24M7MZvP169d/+eWX8vJy3FmIBgWa3ejoqMMZPz8/LEm4R1GUbdO3yWRijk0m05L/VSF24E8SGBg4PDyMOwXRiHtrMEcpKSnMwczMzK1bt1avXt3X14c3EmcCAgJsf5ZSqZS5n1BgYCDOTJwgduCAt6BAs7tx44bteGBg4IsvvsAYBoCFduTIEeeTU1NT3CcB9qBA/3fPP//84OCg1Wolsy8zIAHrVIbFYuE+CbAHBZqdwxz0oUOHyCnQCoXCdrx//37mIDs7G1Mc7hA7cIQQ624N532HgGOwSMjO1iLHaDRqtdrw8PDu7m68kfAidr2I2IEjssfOE/AOmp39HPTNmzdhDhosbQkJCfn5+a+99pr9BYRtbW0YIwEEBXou1q1bp1arcafgDrHrRcQOHCFUVlbW0NBQXFyclJSUl5e3fft2T09PohqE8RMUaHalpaXMAU3TN27ciIyMxJuHS8SuFxE7cIRQampqamqqSqXau3evUql8//334+Pj8/Ly0tPTcUcjGsxBs/vmm2+YA6PRePHixT179mzduhVvJLyInY4kauC23d8ajaawsLCtrQ3qA15QoP87i8Xy4osvdnR04A6Ck4+Pz927d3GnwICogQcFBZWVlSmVyvHx8ZycnPz8fOemNIBLMMXBztaWgabpmzdv6vV6vHmwGx8fxx0BD0IGfuzYsbNnz1IUpdPpjh49unHjRtyJAEJQoJ/Etjzi4uISHBx86tQprHHwI7Z3OyED7+npqaioeOmllwjZ7L9YQIFmR860ozNiNzMQO3CE0PHjxxHbiiiBnfx4Bb777GZmZpqbm728vF5++WXcWbhG7GYGYgeOEPL09GQObG38EEJms5mmaXyhACwSPkFaWppUKv3zzz/lcnlRUVF+fr5KpcIdCieiNjPYI23gto0cCKHQ0FCtVos1DungprHsPD09T5w40dTU1NLSsnLlSkJWigAQCASzs7MIIavVCm+fsYMpDnbu7u56vd7X11coFBqNRpPJhDsRd1hvLPLrr79yHoRrxA7cXlxcXF5e3iuvvNLa2rp27VrccUgHBZqdTqcLDQ2Vy+W9vb1yufzdd9/FnYg7IpHINgtpz2g0ch+GS8QO3N6JEyeOHDly8eLF0NBQon7t+QnmoNnZLktxc3MLCQlZvnw53jxcIm3W1YbYgQPegnfQ7GJiYnBHAACQDhYJAQCAp6BAA0fFxcW4I+BB7MABb8EcNAD/ptFoli9fvmrVKvuTNE0LBPA+BuABv3nAUUZGhsOZS5cuFRQUYAnDpczMzJmZGfszo6OjsBoBMIJFQuBIrVY7XD/m7u5OwoWU4+PjtiZZkZGRPT09fn5+Y2NjWEMBokGBBo7Gx8d37NjhcNLhg/+SJBKJDAaDRCIxGAyDg4PT09MuLi64QwGiQYEGjlatWkXmLczz8vLS0tLS09NVKlVBQUFSUpJAIMjKysKdC5ALFgmBo9OnTxcWFuJOgYHVav3222/7+voUCkVsbGxLSwtFUcnJydAiGeACBRqwO3jwoFwuT0hIwB0EAHJBgQbsAgICkpKSEEKVlZW+vr644wBAIthmB9iJRKKamprdu3dnZWWdPHkSOk8CwD14Bw0cMY2iXn311YaGBoSQxWI5duzY0NCQWq3GHQ0AskCBBo7i4uIQQteuXXO4tXNbWxumRAAQCgo0YLd9+/bm5mbcKQAgGhRoAADgKVgkBAAAnoICDQAAPAUFGgAAeAoKNAAA8BQUaAAA4Cko0AAAwFP/AkckZerwBRZCAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# mostVar Calculate the top n most variable genes in a matrix of gene expression data\n",
    "# https://rdrr.io/github/abc-igmm/transcripTools/man/mostVar.html\n",
    "mostVar <- function(data, n, i_want_most_var = TRUE) {\n",
    "  data.var <- apply(data, 1, stats::var)\n",
    "  data[order(data.var, decreasing = i_want_most_var)[1:n],] \n",
    "}\n",
    "# scale - Z-Score\n",
    "fc <- data.frame(apply(all_fc,2,scale, center=TRUE, scale=TRUE), row.names=rownames(all_fc))\n",
    "\n",
    "# idtransfer\n",
    "peak_ids = strsplit(rownames(all_fc), '_',)\n",
    "ids = unlist(lapply(peak_ids, '[[', 1))\n",
    "\n",
    "# filter most variable genes\n",
    "fc = mostVar(fc,30)\n",
    "# Plot heatmap\n",
    "h1 <- pheatmap(fc, labels_row = gtf_df[ids,])\n",
    "        ggsave('plots/peak_mostVar_Heatmap.png', plot = h1, device = 'png', dpi = 300)\n",
    "ggsave('plots/peak_mostVar_Heatmap.pdf', plot = h1, device = 'pdf', dpi = 300)\n",
    "h1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## iPAGE\n",
    "Enrichment analysis "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7509 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7509\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "683 hits go_terms_num = 17689\n",
      "cat 658, mi = 0.00299231, Notch signaling pathway\n",
      "cat 2542, mi = 0.00298120, negative regulation of proteasomal ubiquitin-dependent protein catabolic process\n",
      "cat 725, mi = 0.00292438, transcriptional repressor activity, RNA polymerase II transcription factor binding\n",
      "cat 1968, mi = 0.00290173, secretory granule membrane\n",
      "cat 869, mi = 0.00286084, regulation of cell adhesion\n",
      "cat 1329, mi = 0.00282004, regulation of inflammatory response\n",
      "cat 3519, mi = 0.00280437, SH3/SH2 adaptor activity\n",
      "cat 84, mi = 0.00275510, animal organ regeneration\n",
      "cat 69, mi = 0.00273984, response to hypoxia\n",
      "cat 5105, mi = 0.00270558, cytoskeletal protein binding\n",
      "cat 6001, mi = 0.00268393, histone deacetylase complex\n",
      "cat 1490, mi = 0.00258134, mitotic spindle pole\n",
      "cat 1532, mi = 0.00257534, RNA polymerase II distal enhancer sequence-specific DNA binding\n",
      "cat 4439, mi = 0.00255886, face morphogenesis\n",
      "cat 1863, mi = 0.00253823, nucleosome assembly\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 658, mi = 0.003, pvalue = 0.001, Notch signaling pathway\n",
      "Passed: cat 2542, mi = 0.003, pvalue = 0.000, negative regulation of proteasomal ubiquitin-dependent protein catabolic process\n",
      "Passed: cat 1968, mi = 0.003, pvalue = 0.001, secretory granule membrane\n",
      "Passed: cat 869, mi = 0.003, pvalue = 0.001, regulation of cell adhesion\n",
      "Passed: cat 1329, mi = 0.003, pvalue = 0.002, regulation of inflammatory response\n",
      "Passed: cat 3519, mi = 0.003, pvalue = 0.003, SH3/SH2 adaptor activity\n",
      "Passed: cat 5105, mi = 0.003, pvalue = 0.002, cytoskeletal protein binding\n",
      "Passed: cat 6001, mi = 0.003, pvalue = 0.002, histone deacetylase complex\n",
      "Passed: cat 1490, mi = 0.003, pvalue = 0.003, mitotic spindle pole\n",
      "Didn't pass (cutoff=0.0050): cat 4439, mi = 0.003, pvalue = 0.005, face morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 1863, mi = 0.003, pvalue = 0.006, nucleosome assembly\n",
      "Didn't pass (cutoff=0.0050): cat 1115, mi = 0.002, pvalue = 0.007, inositol phosphate metabolic process\n",
      "Passed: cat 4968, mi = 0.002, pvalue = 0.003, structural molecule activity conferring elasticity\n",
      "Didn't pass (cutoff=0.0050): cat 3212, mi = 0.002, pvalue = 0.010, negative regulation of extrinsic apoptotic signaling pathway in absence of ligand\n",
      "Passed: cat 1946, mi = 0.002, pvalue = 0.001, phosphatidylinositol-3,4-bisphosphate binding\n",
      "Passed: cat 5060, mi = 0.002, pvalue = 0.001, keratin filament binding\n",
      "Didn't pass (cutoff=0.0050): cat 3347, mi = 0.002, pvalue = 0.006, nucleotide-excision repair, DNA incision, 3'-to lesion\n",
      "Didn't pass (cutoff=0.0050): cat 475, mi = 0.002, pvalue = 0.014, odontogenesis of dentin-containing tooth\n",
      "Didn't pass (cutoff=0.0050): cat 1141, mi = 0.002, pvalue = 0.017, fatty acid metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1670, mi = 0.002, pvalue = 0.012, cellular response to amino acid stimulus\n",
      "Didn't pass (cutoff=0.0050): cat 1243, mi = 0.002, pvalue = 0.018, COPII vesicle coating\n",
      "Didn't pass (cutoff=0.0050): cat 7738, mi = 0.002, pvalue = 0.015, transcriptional repressor activity, RNA polymerase II distal enhancer sequence-specific binding\n",
      "Didn't pass (cutoff=0.0050): cat 3071, mi = 0.002, pvalue = 0.015, cytokine production\n",
      "Passed: cat 3890, mi = 0.002, pvalue = 0.002, pre-miRNA processing\n",
      "Didn't pass (cutoff=0.0050): cat 6968, mi = 0.002, pvalue = 0.017, histone methylation\n",
      "Didn't pass (cutoff=0.0050): cat 1218, mi = 0.002, pvalue = 0.022, regulation of macroautophagy\n",
      "Didn't pass (cutoff=0.0050): cat 3447, mi = 0.002, pvalue = 0.018, positive regulation of protein kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.026, bone development\n",
      "Didn't pass (cutoff=0.0050): cat 541, mi = 0.002, pvalue = 0.025, lysine-acetylated histone binding\n",
      "Didn't pass (cutoff=0.0050): cat 2412, mi = 0.002, pvalue = 0.026, Cajal body\n",
      "Didn't pass (cutoff=0.0050): cat 1498, mi = 0.002, pvalue = 0.026, viral transcription\n",
      "Didn't pass (cutoff=0.0050): cat 1912, mi = 0.002, pvalue = 0.029, retrograde transport, endosome to Golgi\n",
      "Didn't pass (cutoff=0.0050): cat 1690, mi = 0.002, pvalue = 0.030, response to radiation\n",
      "Didn't pass (cutoff=0.0050): cat 135, mi = 0.002, pvalue = 0.018, regulation of cell growth\n",
      "Didn't pass (cutoff=0.0050): cat 1497, mi = 0.002, pvalue = 0.024, mitotic nuclear envelope disassembly\n",
      "Didn't pass (cutoff=0.0050): cat 3022, mi = 0.002, pvalue = 0.023, endodeoxyribonuclease activity\n",
      "Didn't pass (cutoff=0.0050): cat 3250, mi = 0.002, pvalue = 0.025, positive regulation of synapse assembly\n",
      "Passed: cat 1986, mi = 0.002, pvalue = 0.004, ER to Golgi transport vesicle membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1496, mi = 0.002, pvalue = 0.022, tRNA export from nucleus\n",
      "Didn't pass (cutoff=0.0050): cat 2333, mi = 0.002, pvalue = 0.042, retrograde vesicle-mediated transport, Golgi to ER\n",
      "Didn't pass (cutoff=0.0050): cat 823, mi = 0.002, pvalue = 0.043, autophagosome\n",
      "Didn't pass (cutoff=0.0050): cat 4854, mi = 0.002, pvalue = 0.036, intrinsic apoptotic signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 497, mi = 0.002, pvalue = 0.045, palate development\n",
      "Didn't pass (cutoff=0.0050): cat 1514, mi = 0.002, pvalue = 0.027, voltage-gated potassium channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 2163, mi = 0.002, pvalue = 0.024, macrophage activation involved in immune response\n",
      "Didn't pass (cutoff=0.0050): cat 1274, mi = 0.002, pvalue = 0.035, G2/M transition of mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 1956, mi = 0.002, pvalue = 0.032, Rab GTPase binding\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.038, regulation of ion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 3122, mi = 0.002, pvalue = 0.040, peptidyl-threonine phosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 2179, mi = 0.002, pvalue = 0.051, homeostasis of number of cells\n",
      "Didn't pass (cutoff=0.0050): cat 1864, mi = 0.002, pvalue = 0.043, ATP-dependent chromatin remodeling\n",
      "Didn't pass (cutoff=0.0050): cat 4092, mi = 0.002, pvalue = 0.050, cellular response to glucose starvation\n",
      "Didn't pass (cutoff=0.0050): cat 1071, mi = 0.002, pvalue = 0.012, negative regulation of G2/M transition of mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 7450, mi = 0.002, pvalue = 0.016, filamentous actin\n",
      "Didn't pass (cutoff=0.0050): cat 2893, mi = 0.002, pvalue = 0.048, mitochondrion organization\n",
      "Didn't pass (cutoff=0.0050): cat 558, mi = 0.002, pvalue = 0.042, tumor necrosis factor receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 1673, mi = 0.002, pvalue = 0.050, protein complex scaffold\n",
      "Didn't pass (cutoff=0.0050): cat 1404, mi = 0.002, pvalue = 0.027, sphingolipid biosynthetic process\n",
      "Total number of categories evaluated = 103.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 11, 13, 0.0500042751103696\n",
      "best merge is 0, 2, 0.285771895389315\n",
      "best merge is 4, 12, 0.316854236214635\n",
      "best merge is 3, 7, 0.34814385741588\n",
      "best merge is 1, 11, 0.405103880522896\n",
      "best merge is 5, 6, 0.489648363356674\n",
      "best merge is 4, 9, 0.593192180252935\n",
      "best merge is 3, 5, 0.737132733993758\n",
      "best merge is 1, 10, 0.737941617183441\n",
      "best merge is 0, 4, 0.824992655704793\n",
      "order classes\n",
      "2 - 1.0002016338626\n",
      "4 - 1.34489113837159\n",
      "4 - 1.65651638107707\n",
      "3 - 1.55891410030275\n",
      "0 2\n",
      "1 11 13 10\n",
      "3 7 5 6\n",
      "4 12 9\n",
      "8\n",
      "Done.Start drawing\n",
      "5.87\t-9.72\n",
      "phosphatidylinositol-3,4-bisphosphate binding, GO:0043325\n",
      "regulation of cell adhesion, GO:0030155\n",
      "histone deacetylase complex, GO:0000118\n",
      "cytoskeletal protein binding, GO:0008092\n",
      "pre-miRNA processing, GO:0031054\n",
      "structural molecule activity conferring elasticity, GO:0097493\n",
      "Notch signaling pathway, GO:0007219\n",
      "secretory granule membrane, GO:0030667\n",
      "regulation of inflammatory response, GO:0050727\n",
      "SH3/SH2 adaptor activity, GO:0005070\n",
      "negative regulation of proteasomal ubiquitin-dependent protein catabolic process, GO:0032435\n",
      "keratin filament binding, GO:1990254\n",
      "ER to Golgi transport vesicle membrane, GO:0012507\n",
      "mitotic spindle pole, GO:0097431\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 11, 13, 0.0500042751103696\n",
      "best merge is 0, 2, 0.285771895389315\n",
      "best merge is 4, 12, 0.316854236214635\n",
      "best merge is 3, 7, 0.34814385741588\n",
      "best merge is 1, 11, 0.405103880522896\n",
      "best merge is 5, 6, 0.489648363356674\n",
      "best merge is 4, 9, 0.593192180252935\n",
      "best merge is 3, 5, 0.737132733993758\n",
      "best merge is 1, 10, 0.737941617183441\n",
      "best merge is 0, 4, 0.824992655704793\n",
      "order classes\n",
      "2 - 1.0002016338626\n",
      "4 - 1.34489113837159\n",
      "4 - 1.65651638107707\n",
      "3 - 1.55891410030275\n",
      "0 2\n",
      "1 11 13 10\n",
      "3 7 5 6\n",
      "4 12 9\n",
      "8\n",
      "Done.-8.59076330981398\t42\t0\n",
      "3.62668377164849\t42\t22.818473380372\n",
      "4.8120590121873\t42\t24.434894162925\n",
      "5.44515715202053\t42\t25.1892238614497\n",
      "5.88967286722258\t42\t25.7010904425914\n",
      "6.22642719692111\t42\t26.0782552918538\n",
      "6.52277100705581\t42\t26.3745991019885\n",
      "6.84605516356639\t42\t26.6709429121232\n",
      "7.25016035920462\t42\t27.0211674150096\n",
      "7.85631815266196\t42\t27.4791533033996\n",
      "15.1167415009622\t42\t28.2334830019243\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6496 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6496\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "591 hits go_terms_num = 181\n",
      "cat 114, mi = 0.00179924, POU5F1_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00163156, BDP1_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00162673, BRF2_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00156178, BRF1_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00155618, SMARCA4_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00132653, POLR3G_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00132465, PPARGC1A_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00106544, XRCC4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.002, pvalue = 0.111, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.227, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.002, pvalue = 0.156, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.002, pvalue = 0.262, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.002, pvalue = 0.280, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.476, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.567, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 167, mi = 0.001, pvalue = 0.091, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.90\t-9.72\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_encode_tf/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 298\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6547 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6547\t298\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "595 hits go_terms_num = 298\n",
      "cat 192, mi = 0.00333823, chr7q21\n",
      "cat 248, mi = 0.00273236, chr15q26\n",
      "cat 15, mi = 0.00262188, chr3q21\n",
      "cat 52, mi = 0.00257703, chr15q15\n",
      "cat 208, mi = 0.00243469, chr22q12\n",
      "cat 291, mi = 0.00241171, chr16p12\n",
      "cat 147, mi = 0.00237097, chr3p22\n",
      "cat 77, mi = 0.00234047, chr1p22\n",
      "cat 6, mi = 0.00230087, chr11q12\n",
      "cat 13, mi = 0.00222598, chr3q25\n",
      "cat 135, mi = 0.00221214, chr20p11\n",
      "cat 213, mi = 0.00220156, chr6q24\n",
      "cat 102, mi = 0.00216167, chr9q31\n",
      "cat 156, mi = 0.00215848, chr13q22\n",
      "cat 83, mi = 0.00215672, chr22q13\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 192, mi = 0.003, pvalue = 0.001, chr7q21\n",
      "Didn't pass (cutoff=0.0050): cat 248, mi = 0.003, pvalue = 0.011, chr15q26\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.003, pvalue = 0.014, chr3q21\n",
      "Didn't pass (cutoff=0.0050): cat 52, mi = 0.003, pvalue = 0.016, chr15q15\n",
      "Didn't pass (cutoff=0.0050): cat 208, mi = 0.002, pvalue = 0.021, chr22q12\n",
      "Didn't pass (cutoff=0.0050): cat 291, mi = 0.002, pvalue = 0.024, chr16p12\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.002, pvalue = 0.031, chr3p22\n",
      "Didn't pass (cutoff=0.0050): cat 77, mi = 0.002, pvalue = 0.038, chr1p22\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.036, chr11q12\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.002, pvalue = 0.047, chr3q25\n",
      "Didn't pass (cutoff=0.0050): cat 135, mi = 0.002, pvalue = 0.042, chr20p11\n",
      "Didn't pass (cutoff=0.0050): cat 213, mi = 0.002, pvalue = 0.032, chr6q24\n",
      "Didn't pass (cutoff=0.0050): cat 102, mi = 0.002, pvalue = 0.024, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.035, chr13q22\n",
      "Didn't pass (cutoff=0.0050): cat 83, mi = 0.002, pvalue = 0.042, chr22q13\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.068, chr9p24\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.002, pvalue = 0.066, chr16q12\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.002, pvalue = 0.081, chr1q23\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.062, chr17q22\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.002, pvalue = 0.082, chr18p11\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.002, pvalue = 0.074, chr3q27\n",
      "Total number of categories evaluated = 11.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.87\t-9.72\n",
      "chr7q21\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.49647209749839\t42\t0\n",
      "3.74791533033996\t42\t23.0070558050032\n",
      "4.86593970493906\t42\t24.4887748556767\n",
      "5.48556767158435\t42\t25.2431045542014\n",
      "5.90314304041052\t42\t25.7280307889673\n",
      "6.22642719692111\t42\t26.0782552918538\n",
      "6.52277100705581\t42\t26.3745991019885\n",
      "6.83258499037845\t42\t26.6709429121232\n",
      "7.23669018601668\t42\t26.9942270686337\n",
      "7.82937780628608\t42\t27.4791533033996\n",
      "15.0898011545863\t42\t28.1796023091726\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_c1/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5529\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6492 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6492\t5529\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "590 hits go_terms_num = 5529\n",
      "cat 1616, mi = 0.00495250, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "cat 1401, mi = 0.00424559, REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING\n",
      "cat 790, mi = 0.00412275, ACOSTA_PROLIFERATION_INDEPENDENT_MYC_TARGETS_DN\n",
      "cat 3804, mi = 0.00409002, WEIGEL_OXIDATIVE_STRESS_BY_HNE_AND_H2O2\n",
      "cat 257, mi = 0.00406003, BOSCO_ALLERGEN_INDUCED_TH2_ASSOCIATED_MODULE\n",
      "cat 3501, mi = 0.00402276, AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "cat 2060, mi = 0.00391981, DIRMEIER_LMP1_RESPONSE_EARLY\n",
      "cat 785, mi = 0.00366554, ZHENG_FOXP3_TARGETS_IN_THYMUS_UP\n",
      "cat 4156, mi = 0.00358042, PID_HIF1_TFPATHWAY\n",
      "cat 755, mi = 0.00357297, GRABARCZYK_BCL11B_TARGETS_UP\n",
      "cat 4182, mi = 0.00349451, CHAUHAN_RESPONSE_TO_METHOXYESTRADIOL_UP\n",
      "cat 5118, mi = 0.00341044, PID_HDAC_CLASSII_PATHWAY\n",
      "cat 4421, mi = 0.00340776, GALINDO_IMMUNE_RESPONSE_TO_ENTEROTOXIN\n",
      "cat 4743, mi = 0.00339618, KEGG_GLIOMA\n",
      "cat 2423, mi = 0.00332691, PID_REELIN_PATHWAY\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1616, mi = 0.005, pvalue = 0.000, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "Passed: cat 1401, mi = 0.004, pvalue = 0.000, REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING\n",
      "Passed: cat 790, mi = 0.004, pvalue = 0.000, ACOSTA_PROLIFERATION_INDEPENDENT_MYC_TARGETS_DN\n",
      "Passed: cat 3804, mi = 0.004, pvalue = 0.000, WEIGEL_OXIDATIVE_STRESS_BY_HNE_AND_H2O2\n",
      "Passed: cat 257, mi = 0.004, pvalue = 0.000, BOSCO_ALLERGEN_INDUCED_TH2_ASSOCIATED_MODULE\n",
      "Passed: cat 3501, mi = 0.004, pvalue = 0.000, AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "Passed: cat 755, mi = 0.004, pvalue = 0.001, GRABARCZYK_BCL11B_TARGETS_UP\n",
      "Passed: cat 5118, mi = 0.003, pvalue = 0.000, PID_HDAC_CLASSII_PATHWAY\n",
      "Passed: cat 4743, mi = 0.003, pvalue = 0.001, KEGG_GLIOMA\n",
      "Passed: cat 373, mi = 0.003, pvalue = 0.003, FLORIO_NEOCORTEX_BASAL_RADIAL_GLIA_DN\n",
      "Passed: cat 1711, mi = 0.003, pvalue = 0.002, DARWICHE_SKIN_TUMOR_PROMOTER_UP\n",
      "Passed: cat 1699, mi = 0.003, pvalue = 0.005, HINATA_NFKB_TARGETS_FIBROBLAST_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2106, mi = 0.003, pvalue = 0.006, PUIFFE_INVASION_INHIBITED_BY_ASCITES_UP\n",
      "Passed: cat 2207, mi = 0.003, pvalue = 0.003, PID_P38_ALPHA_BETA_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 1074, mi = 0.003, pvalue = 0.006, WATANABE_RECTAL_CANCER_RADIOTHERAPY_RESPONSIVE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4906, mi = 0.003, pvalue = 0.006, REACTOME_SNRNP_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 1216, mi = 0.003, pvalue = 0.009, KOBAYASHI_EGFR_SIGNALING_24HR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 369, mi = 0.003, pvalue = 0.009, DELACROIX_RAR_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4902, mi = 0.003, pvalue = 0.009, GESERICK_TERT_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4501, mi = 0.003, pvalue = 0.011, TURASHVILI_BREAST_DUCTAL_CARCINOMA_VS_DUCTAL_NORMAL_DN\n",
      "Passed: cat 3814, mi = 0.003, pvalue = 0.004, FUJII_YBX1_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5278, mi = 0.003, pvalue = 0.011, BROWNE_HCMV_INFECTION_48HR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 57, mi = 0.003, pvalue = 0.009, HOLLEMAN_PREDNISOLONE_RESISTANCE_ALL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1844, mi = 0.003, pvalue = 0.011, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_0\n",
      "Didn't pass (cutoff=0.0050): cat 564, mi = 0.003, pvalue = 0.006, SASAI_RESISTANCE_TO_NEOPLASTIC_TRANSFROMATION\n",
      "Didn't pass (cutoff=0.0050): cat 568, mi = 0.003, pvalue = 0.012, REACTOME_SUMOYLATION_OF_SUMOYLATION_PROTEINS\n",
      "Didn't pass (cutoff=0.0050): cat 3623, mi = 0.003, pvalue = 0.011, MAYBURD_RESPONSE_TO_L663536_DN\n",
      "Didn't pass (cutoff=0.0050): cat 739, mi = 0.003, pvalue = 0.017, SHEDDEN_LUNG_CANCER_GOOD_SURVIVAL_A4\n",
      "Passed: cat 2301, mi = 0.003, pvalue = 0.003, REACTOME_TYPE_I_HEMIDESMOSOME_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 3130, mi = 0.002, pvalue = 0.023, GARGALOVIC_RESPONSE_TO_OXIDIZED_PHOSPHOLIPIDS_YELLOW_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4659, mi = 0.002, pvalue = 0.027, NELSON_RESPONSE_TO_ANDROGEN_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4858, mi = 0.002, pvalue = 0.008, LEIN_LOCALIZED_TO_DISTAL_AND_PROXIMAL_DENDRITES\n",
      "Didn't pass (cutoff=0.0050): cat 1107, mi = 0.002, pvalue = 0.012, BRIDEAU_IMPRINTED_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 2247, mi = 0.002, pvalue = 0.021, MENSE_HYPOXIA_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5503, mi = 0.002, pvalue = 0.029, GARGALOVIC_RESPONSE_TO_OXIDIZED_PHOSPHOLIPIDS_GREEN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2406, mi = 0.002, pvalue = 0.026, MATTIOLI_MGUS_VS_PCL\n",
      "Didn't pass (cutoff=0.0050): cat 4193, mi = 0.002, pvalue = 0.025, IVANOVA_HEMATOPOIESIS_INTERMEDIATE_PROGENITOR\n",
      "Didn't pass (cutoff=0.0050): cat 1459, mi = 0.002, pvalue = 0.023, KYNG_WERNER_SYNDROM_AND_NORMAL_AGING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 939, mi = 0.002, pvalue = 0.021, LINDGREN_BLADDER_CANCER_WITH_LOH_IN_CHR9Q\n",
      "Didn't pass (cutoff=0.0050): cat 79, mi = 0.002, pvalue = 0.023, REACTOME_TRANSPORT_OF_MATURE_MRNAS_DERIVED_FROM_INTRONLESS_TRANSCRIPTS\n",
      "Didn't pass (cutoff=0.0050): cat 2378, mi = 0.002, pvalue = 0.022, REACTOME_VIRAL_MESSENGER_RNA_SYNTHESIS\n",
      "Didn't pass (cutoff=0.0050): cat 2638, mi = 0.002, pvalue = 0.021, REACTOME_NUCLEAR_PORE_COMPLEX_NPC_DISASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 3730, mi = 0.002, pvalue = 0.015, REACTOME_PHASE_I_FUNCTIONALIZATION_OF_COMPOUNDS\n",
      "Didn't pass (cutoff=0.0050): cat 3970, mi = 0.002, pvalue = 0.006, REACTOME_GAMMA_CARBOXYLATION_HYPUSINE_FORMATION_AND_ARYLSULFATASE_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 1504, mi = 0.002, pvalue = 0.032, REACTOME_HCMV_EARLY_EVENTS\n",
      "Didn't pass (cutoff=0.0050): cat 2565, mi = 0.002, pvalue = 0.018, REACTOME_TRANSPORT_OF_THE_SLBP_DEPENDANT_MATURE_MRNA\n",
      "Didn't pass (cutoff=0.0050): cat 2958, mi = 0.002, pvalue = 0.021, REACTOME_REGULATION_OF_GLUCOKINASE_BY_GLUCOKINASE_REGULATORY_PROTEIN\n",
      "Didn't pass (cutoff=0.0050): cat 5429, mi = 0.002, pvalue = 0.018, REACTOME_NUCLEAR_IMPORT_OF_REV_PROTEIN\n",
      "Didn't pass (cutoff=0.0050): cat 1412, mi = 0.002, pvalue = 0.033, BILANGES_SERUM_SENSITIVE_VIA_TSC2\n",
      "Total number of categories evaluated = 165.\n",
      "Number of categories that passed the tests = 15\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 10, 0.0499489428825097\n",
      "best merge is 5, 12, 0.143738437593462\n",
      "best merge is 6, 11, 0.148905697259962\n",
      "best merge is 1, 5, 0.238772587568389\n",
      "best merge is 7, 14, 0.329617056148821\n",
      "best merge is 2, 3, 0.33694745404229\n",
      "best merge is 2, 4, 0.421421977177477\n",
      "best merge is 0, 13, 0.579571733910848\n",
      "best merge is 0, 7, 0.642287358470333\n",
      "best merge is 1, 2, 0.654593166965003\n",
      "best merge is 0, 6, 0.765404630381867\n",
      "order classes\n",
      "4 - 1.63047104247234\n",
      "7 - 0.915263387651002\n",
      "2 - 0.558396364724856\n",
      "6 11\n",
      "1 5 12 2 3 10 4\n",
      "0 13 7 14\n",
      "8\n",
      "9\n",
      "Done.Start drawing\n",
      "5.87\t-9.72\n",
      "LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "FUJII_YBX1_TARGETS_UP\n",
      "KEGG_GLIOMA\n",
      "REACTOME_TYPE_I_HEMIDESMOSOME_ASSEMBLY\n",
      "ACOSTA_PROLIFERATION_INDEPENDENT_MYC_TARGETS_DN\n",
      "WEIGEL_OXIDATIVE_STRESS_BY_HNE_AND_H2O2\n",
      "DARWICHE_SKIN_TUMOR_PROMOTER_UP\n",
      "BOSCO_ALLERGEN_INDUCED_TH2_ASSOCIATED_MODULE\n",
      "REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING\n",
      "AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "PID_P38_ALPHA_BETA_PATHWAY\n",
      "PID_HDAC_CLASSII_PATHWAY\n",
      "GRABARCZYK_BCL11B_TARGETS_UP\n",
      "HINATA_NFKB_TARGETS_FIBROBLAST_UP\n",
      "FLORIO_NEOCORTEX_BASAL_RADIAL_GLIA_DN\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 10, 0.0499489428825097\n",
      "best merge is 5, 12, 0.143738437593462\n",
      "best merge is 6, 11, 0.148905697259962\n",
      "best merge is 1, 5, 0.238772587568389\n",
      "best merge is 7, 14, 0.329617056148821\n",
      "best merge is 2, 3, 0.33694745404229\n",
      "best merge is 2, 4, 0.421421977177477\n",
      "best merge is 0, 13, 0.579571733910848\n",
      "best merge is 0, 7, 0.642287358470333\n",
      "best merge is 1, 2, 0.654593166965003\n",
      "best merge is 0, 6, 0.765404630381867\n",
      "order classes\n",
      "4 - 1.63047104247234\n",
      "7 - 0.915263387651002\n",
      "2 - 0.558396364724856\n",
      "6 11\n",
      "1 5 12 2 3 10 4\n",
      "0 13 7 14\n",
      "8\n",
      "9\n",
      "Done.-8.48300192431045\t42\t0\n",
      "3.77485567671584\t42\t23.0339961513791\n",
      "4.87940987812701\t42\t24.5157152020526\n",
      "5.49903784477229\t42\t25.2431045542014\n",
      "5.93008338678641\t42\t25.7549711353432\n",
      "6.23989737010905\t42\t26.1051956382296\n",
      "6.52277100705581\t42\t26.3745991019885\n",
      "6.83258499037845\t42\t26.6709429121232\n",
      "7.23669018601668\t42\t26.9942270686337\n",
      "7.82937780628608\t42\t27.4791533033996\n",
      "15.0898011545863\t42\t28.1796023091726\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_c2/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 3733\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6535 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6535\t3733\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "594 hits go_terms_num = 3733\n",
      "cat 453, mi = 0.00420663, MIR12135\n",
      "cat 2218, mi = 0.00419419, ARID3B_TARGET_GENES\n",
      "cat 549, mi = 0.00373257, MIR6837_5P\n",
      "cat 337, mi = 0.00349632, MIR6769A_3P\n",
      "cat 3154, mi = 0.00341463, MIR4273\n",
      "cat 2275, mi = 0.00341143, MIR6793_3P\n",
      "cat 1081, mi = 0.00334397, AGCATTA_MIR155\n",
      "cat 1592, mi = 0.00330786, SRF_01\n",
      "cat 471, mi = 0.00325127, MIR3943\n",
      "cat 1468, mi = 0.00320346, GTAAACC_MIR2995P\n",
      "cat 520, mi = 0.00318948, MIR3925_3P\n",
      "cat 2433, mi = 0.00317781, MIR4685_5P\n",
      "cat 3074, mi = 0.00300976, MIR6772_5P\n",
      "cat 2075, mi = 0.00299771, MIR3173_3P\n",
      "cat 3601, mi = 0.00299268, MIR6891_5P\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 453, mi = 0.004, pvalue = 0.000, MIR12135\n",
      "Passed: cat 2218, mi = 0.004, pvalue = 0.000, ARID3B_TARGET_GENES\n",
      "Passed: cat 549, mi = 0.004, pvalue = 0.001, MIR6837_5P\n",
      "Passed: cat 337, mi = 0.003, pvalue = 0.000, MIR6769A_3P\n",
      "Passed: cat 3154, mi = 0.003, pvalue = 0.001, MIR4273\n",
      "Passed: cat 2275, mi = 0.003, pvalue = 0.001, MIR6793_3P\n",
      "Passed: cat 1592, mi = 0.003, pvalue = 0.001, SRF_01\n",
      "Passed: cat 471, mi = 0.003, pvalue = 0.001, MIR3943\n",
      "Passed: cat 1468, mi = 0.003, pvalue = 0.002, GTAAACC_MIR2995P\n",
      "Passed: cat 3074, mi = 0.003, pvalue = 0.002, MIR6772_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2355, mi = 0.003, pvalue = 0.006, MIR2682_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1452, mi = 0.003, pvalue = 0.006, MIR3159\n",
      "Didn't pass (cutoff=0.0050): cat 3468, mi = 0.003, pvalue = 0.007, MIR4794\n",
      "Didn't pass (cutoff=0.0050): cat 309, mi = 0.003, pvalue = 0.005, MIR4278\n",
      "Didn't pass (cutoff=0.0050): cat 419, mi = 0.003, pvalue = 0.008, KTGGYRSGAA_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 437, mi = 0.003, pvalue = 0.012, MIR3619_3P\n",
      "Didn't pass (cutoff=0.0050): cat 2596, mi = 0.003, pvalue = 0.013, MIR6765_3P\n",
      "Didn't pass (cutoff=0.0050): cat 145, mi = 0.003, pvalue = 0.009, AGGAAGC_MIR5163P\n",
      "Didn't pass (cutoff=0.0050): cat 1243, mi = 0.003, pvalue = 0.009, SATB1_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 2201, mi = 0.003, pvalue = 0.013, PPARG_01\n",
      "Didn't pass (cutoff=0.0050): cat 922, mi = 0.003, pvalue = 0.012, MIR449C_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2932, mi = 0.003, pvalue = 0.013, MIR34B_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1891, mi = 0.003, pvalue = 0.016, HMEF2_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 1451, mi = 0.003, pvalue = 0.017, MIR518D_5P_MIR518F_5P_MIR520C_5P_MIR526A_5P\n",
      "Didn't pass (cutoff=0.0050): cat 225, mi = 0.003, pvalue = 0.011, MIR556_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2681, mi = 0.003, pvalue = 0.019, MIR6886_3P\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.003, pvalue = 0.017, MIR6730_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1289, mi = 0.003, pvalue = 0.018, EVI1_05\n",
      "Didn't pass (cutoff=0.0050): cat 1246, mi = 0.003, pvalue = 0.018, MIR4316\n",
      "Didn't pass (cutoff=0.0050): cat 2544, mi = 0.003, pvalue = 0.020, MIR3162_5P\n",
      "Total number of categories evaluated = 58.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 6, 0.106422334662118\n",
      "best merge is 4, 7, 0.325698715134303\n",
      "best merge is 3, 8, 0.524983735604466\n",
      "best merge is 1, 4, 0.585719059989106\n",
      "best merge is 2, 5, 0.707272834653693\n",
      "best merge is 1, 2, 0.818250382458715\n",
      "order classes\n",
      "3 - 1.10899024823149\n",
      "2 - 1.76818208663423\n",
      "3 - 0.856585041385963\n",
      "3 6 8\n",
      "1 4 7\n",
      "2 5\n",
      "0\n",
      "9\n",
      "Done.Start drawing\n",
      "5.87\t-9.72\n",
      "MIR6769A_3P\n",
      "SRF_01\n",
      "MIR6837_5P\n",
      "ARID3B_TARGET_GENES\n",
      "MIR4273\n",
      "MIR3943\n",
      "MIR6793_3P\n",
      "GTAAACC_MIR2995P\n",
      "MIR6772_5P\n",
      "MIR12135\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 6, 0.106422334662118\n",
      "best merge is 4, 7, 0.325698715134303\n",
      "best merge is 3, 8, 0.524983735604466\n",
      "best merge is 1, 4, 0.585719059989106\n",
      "best merge is 2, 5, 0.707272834653693\n",
      "best merge is 1, 2, 0.818250382458715\n",
      "order classes\n",
      "3 - 1.10899024823149\n",
      "2 - 1.76818208663423\n",
      "3 - 0.856585041385963\n",
      "3 6 8\n",
      "1 4 7\n",
      "2 5\n",
      "0\n",
      "9\n",
      "Done.-8.48300192431045\t42\t0\n",
      "3.7613855035279\t42\t23.0339961513791\n",
      "4.86593970493906\t42\t24.4887748556767\n",
      "5.48556767158435\t42\t25.2431045542014\n",
      "5.90314304041052\t42\t25.7280307889673\n",
      "6.22642719692111\t42\t26.0782552918538\n",
      "6.52277100705581\t42\t26.3745991019885\n",
      "6.83258499037845\t42\t26.6709429121232\n",
      "7.23669018601668\t42\t26.9942270686337\n",
      "7.82937780628608\t42\t27.4791533033996\n",
      "15.0898011545863\t42\t28.1796023091726\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_c3/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3839 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3839\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "349 hits go_terms_num = 858\n",
      "cat 420, mi = 0.00630805, MODULE_480\n",
      "cat 89, mi = 0.00517873, MORF_PPP6C\n",
      "cat 133, mi = 0.00473956, GCM_ZNF198\n",
      "cat 298, mi = 0.00461427, MORF_ANP32B\n",
      "cat 595, mi = 0.00454651, GCM_NCAM1\n",
      "cat 182, mi = 0.00435148, GNF2_ATM\n",
      "cat 25, mi = 0.00430576, MODULE_125\n",
      "cat 444, mi = 0.00429076, MORF_PPP2R4\n",
      "cat 202, mi = 0.00424656, GNF2_DEK\n",
      "cat 450, mi = 0.00416883, GCM_DLG1\n",
      "cat 674, mi = 0.00410312, GCM_RAD21\n",
      "cat 724, mi = 0.00408099, GCM_APEX1\n",
      "cat 703, mi = 0.00397237, GCM_ING1\n",
      "cat 502, mi = 0.00396426, MODULE_236\n",
      "cat 672, mi = 0.00395649, MODULE_484\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 420, mi = 0.006, pvalue = 0.001, MODULE_480\n",
      "Passed: cat 89, mi = 0.005, pvalue = 0.004, MORF_PPP6C\n",
      "Didn't pass (cutoff=0.0050): cat 133, mi = 0.005, pvalue = 0.005, GCM_ZNF198\n",
      "Didn't pass (cutoff=0.0050): cat 298, mi = 0.005, pvalue = 0.012, MORF_ANP32B\n",
      "Didn't pass (cutoff=0.0050): cat 595, mi = 0.005, pvalue = 0.009, GCM_NCAM1\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.004, pvalue = 0.020, GNF2_ATM\n",
      "Didn't pass (cutoff=0.0050): cat 25, mi = 0.004, pvalue = 0.006, MODULE_125\n",
      "Didn't pass (cutoff=0.0050): cat 202, mi = 0.004, pvalue = 0.022, GNF2_DEK\n",
      "Didn't pass (cutoff=0.0050): cat 450, mi = 0.004, pvalue = 0.024, GCM_DLG1\n",
      "Didn't pass (cutoff=0.0050): cat 674, mi = 0.004, pvalue = 0.021, GCM_RAD21\n",
      "Didn't pass (cutoff=0.0050): cat 724, mi = 0.004, pvalue = 0.028, GCM_APEX1\n",
      "Didn't pass (cutoff=0.0050): cat 502, mi = 0.004, pvalue = 0.007, MODULE_236\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.004, pvalue = 0.037, MODULE_484\n",
      "Didn't pass (cutoff=0.0050): cat 709, mi = 0.004, pvalue = 0.030, MODULE_239\n",
      "Didn't pass (cutoff=0.0050): cat 255, mi = 0.004, pvalue = 0.035, MODULE_57\n",
      "Didn't pass (cutoff=0.0050): cat 487, mi = 0.004, pvalue = 0.047, MODULE_537\n",
      "Didn't pass (cutoff=0.0050): cat 571, mi = 0.004, pvalue = 0.027, MODULE_229\n",
      "Didn't pass (cutoff=0.0050): cat 707, mi = 0.004, pvalue = 0.027, GNF2_MCM5\n",
      "Didn't pass (cutoff=0.0050): cat 408, mi = 0.004, pvalue = 0.043, MODULE_87\n",
      "Didn't pass (cutoff=0.0050): cat 543, mi = 0.004, pvalue = 0.032, GNF2_HAT1\n",
      "Didn't pass (cutoff=0.0050): cat 779, mi = 0.004, pvalue = 0.034, GCM_PTK2\n",
      "Didn't pass (cutoff=0.0050): cat 719, mi = 0.004, pvalue = 0.035, GCM_UBE2N\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.04\t-9.45\n",
      "MODULE_480\n",
      "MORF_PPP6C\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.95478131949592\t42\t0\n",
      "6.88435878428466\t42\t26.0904373610082\n",
      "8.06745737583395\t42\t27.6782802075612\n",
      "8.72127501853225\t42\t28.4566345441067\n",
      "9.15715344699777\t42\t28.9859154929577\n",
      "9.49962935507784\t42\t29.3283914010378\n",
      "9.81097108969607\t42\t29.6708673091179\n",
      "10.1378799110452\t42\t29.9510748702743\n",
      "10.5737583395107\t42\t30.3246849518162\n",
      "11.2120088954781\t42\t30.8228317272053\n",
      "16.8005930318755\t42\t31.6011860637509\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_c4/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 10191\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 5908 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 5908\t10191\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "537 hits go_terms_num = 10191\n",
      "cat 8804, mi = 0.00444550, GO_NOTCH_SIGNALING_PATHWAY\n",
      "cat 4444, mi = 0.00422973, GO_CARDIOCYTE_DIFFERENTIATION\n",
      "cat 1475, mi = 0.00404009, GO_POSITIVE_REGULATION_OF_LYMPHOCYTE_DIFFERENTIATION\n",
      "cat 7924, mi = 0.00397482, GO_REGULATION_OF_T_CELL_DIFFERENTIATION\n",
      "cat 2260, mi = 0.00394248, GO_IRON_ION_TRANSPORT\n",
      "cat 6197, mi = 0.00392620, GO_REGULATION_OF_NOTCH_SIGNALING_PATHWAY\n",
      "cat 5183, mi = 0.00381242, GO_REGULATION_OF_RESPONSE_TO_CYTOKINE_STIMULUS\n",
      "cat 5677, mi = 0.00377997, GO_HISTONE_H3_K9_MODIFICATION\n",
      "cat 6855, mi = 0.00371668, GO_ANATOMICAL_STRUCTURE_MATURATION\n",
      "cat 9242, mi = 0.00361648, GO_CD4_POSITIVE_ALPHA_BETA_T_CELL_ACTIVATION\n",
      "cat 4481, mi = 0.00356110, GO_INTERLEUKIN_2_PRODUCTION\n",
      "cat 5008, mi = 0.00355663, GO_POSITIVE_REGULATION_OF_TRANSCRIPTION_OF_NOTCH_RECEPTOR_TARGET\n",
      "cat 5655, mi = 0.00354909, GO_CARDIAC_MUSCLE_CELL_DIFFERENTIATION\n",
      "cat 9068, mi = 0.00349308, GO_REGULATION_OF_CALCIUM_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "cat 6295, mi = 0.00348660, GO_REGULATION_OF_RNA_SPLICING\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 8804, mi = 0.004, pvalue = 0.000, GO_NOTCH_SIGNALING_PATHWAY\n",
      "Passed: cat 1475, mi = 0.004, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_LYMPHOCYTE_DIFFERENTIATION\n",
      "Passed: cat 2260, mi = 0.004, pvalue = 0.000, GO_IRON_ION_TRANSPORT\n",
      "Passed: cat 5677, mi = 0.004, pvalue = 0.002, GO_HISTONE_H3_K9_MODIFICATION\n",
      "Passed: cat 9068, mi = 0.003, pvalue = 0.002, GO_REGULATION_OF_CALCIUM_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Passed: cat 4510, mi = 0.003, pvalue = 0.003, GO_CELLULAR_MODIFIED_AMINO_ACID_METABOLIC_PROCESS\n",
      "Passed: cat 2175, mi = 0.003, pvalue = 0.003, GO_MITOTIC_SPINDLE_POLE\n",
      "Passed: cat 3858, mi = 0.003, pvalue = 0.004, GO_REGULATION_OF_FILOPODIUM_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 2382, mi = 0.003, pvalue = 0.009, GO_ALTERNATIVE_MRNA_SPLICING_VIA_SPLICEOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 9796, mi = 0.003, pvalue = 0.011, GO_ORGANOPHOSPHATE_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 9730, mi = 0.003, pvalue = 0.006, GO_TRANSCRIPTION_PREINITIATION_COMPLEX_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 6873, mi = 0.003, pvalue = 0.009, GO_NEGATIVE_REGULATION_OF_VIRAL_LIFE_CYCLE\n",
      "Didn't pass (cutoff=0.0050): cat 3647, mi = 0.003, pvalue = 0.010, GO_SIGNALING_ADAPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 6208, mi = 0.003, pvalue = 0.014, GO_REGULATION_OF_MYELOID_LEUKOCYTE_MEDIATED_IMMUNITY\n",
      "Didn't pass (cutoff=0.0050): cat 2226, mi = 0.003, pvalue = 0.009, GO_POSITIVE_REGULATION_OF_MYELOID_LEUKOCYTE_MEDIATED_IMMUNITY\n",
      "Didn't pass (cutoff=0.0050): cat 8637, mi = 0.003, pvalue = 0.007, GO_POSITIVE_REGULATION_OF_MYELOID_LEUKOCYTE_CYTOKINE_PRODUCTION_INVOLVED_IN_IMMUNE_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 6477, mi = 0.003, pvalue = 0.013, GO_CRANIAL_NERVE_DEVELOPMENT\n",
      "Passed: cat 4273, mi = 0.003, pvalue = 0.005, GO_PRE_MIRNA_PROCESSING\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.003, pvalue = 0.006, GO_SULFUR_COMPOUND_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 2704, mi = 0.003, pvalue = 0.014, GO_TRANSLATIONAL_INITIATION\n",
      "Didn't pass (cutoff=0.0050): cat 7256, mi = 0.003, pvalue = 0.018, GO_PEPTIDYL_ASPARAGINE_MODIFICATION\n",
      "Didn't pass (cutoff=0.0050): cat 6062, mi = 0.003, pvalue = 0.015, GO_HISTONE_DEACETYLASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 2752, mi = 0.003, pvalue = 0.006, GO_DENDRITIC_CELL_CYTOKINE_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 4329, mi = 0.003, pvalue = 0.007, GO_POSITIVE_REGULATION_OF_DENDRITIC_CELL_CYTOKINE_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 4345, mi = 0.003, pvalue = 0.008, GO_MYELOID_DENDRITIC_CELL_CYTOKINE_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 7950, mi = 0.003, pvalue = 0.015, GO_POSITIVE_REGULATION_OF_TOLL_LIKE_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 546, mi = 0.003, pvalue = 0.022, GO_NEUROTRANSMITTER_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3679, mi = 0.003, pvalue = 0.017, GO_REGULATION_OF_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_DEADENYLATION_DEPENDENT_DECAY\n",
      "Didn't pass (cutoff=0.0050): cat 1619, mi = 0.003, pvalue = 0.019, GO_BASAL_TRANSCRIPTION_MACHINERY_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 7689, mi = 0.003, pvalue = 0.013, GO_NEGATIVE_REGULATION_OF_CHROMOSOME_SEGREGATION\n",
      "Didn't pass (cutoff=0.0050): cat 2495, mi = 0.003, pvalue = 0.021, GO_REGULATION_OF_MRNA_SPLICING_VIA_SPLICEOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 5437, mi = 0.003, pvalue = 0.024, GO_MACROPHAGE_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 1283, mi = 0.003, pvalue = 0.013, GO_REGULATION_OF_ENDOPLASMIC_RETICULUM_STRESS_INDUCED_INTRINSIC_APOPTOTIC_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 8019, mi = 0.003, pvalue = 0.027, GO_REACTIVE_NITROGEN_SPECIES_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 769, mi = 0.003, pvalue = 0.008, GO_HOMOLOGOUS_RECOMBINATION\n",
      "Didn't pass (cutoff=0.0050): cat 9497, mi = 0.003, pvalue = 0.007, GO_AUTOPHAGOSOME_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 1830, mi = 0.003, pvalue = 0.024, GO_NEGATIVE_REGULATION_OF_TOR_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 3868, mi = 0.003, pvalue = 0.023, GO_NEGATIVE_REGULATION_OF_VIRAL_PROCESS\n",
      "Total number of categories evaluated = 120.\n",
      "Number of categories that passed the tests = 9\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 8, 0.181471958782776\n",
      "best merge is 0, 1, 0.188302485176644\n",
      "best merge is 2, 7, 0.273696120059127\n",
      "best merge is 2, 3, 0.394568887835097\n",
      "best merge is 0, 5, 0.450446994105642\n",
      "order classes\n",
      "2 - 0.423680591647449\n",
      "3 - 0.708555930486214\n",
      "2 - 0.408311907261247\n",
      "4 8\n",
      "2 7 3\n",
      "0 1\n",
      "5\n",
      "6\n",
      "Done.Start drawing\n",
      "4.90\t-9.72\n",
      "GO_IRON_ION_TRANSPORT\n",
      "GO_REGULATION_OF_FILOPODIUM_ASSEMBLY\n",
      "GO_HISTONE_H3_K9_MODIFICATION\n",
      "GO_REGULATION_OF_CALCIUM_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "GO_PRE_MIRNA_PROCESSING\n",
      "GO_CELLULAR_MODIFIED_AMINO_ACID_METABOLIC_PROCESS\n",
      "GO_NOTCH_SIGNALING_PATHWAY\n",
      "GO_POSITIVE_REGULATION_OF_LYMPHOCYTE_DIFFERENTIATION\n",
      "GO_MITOTIC_SPINDLE_POLE\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 8, 0.181471958782776\n",
      "best merge is 0, 1, 0.188302485176644\n",
      "best merge is 2, 7, 0.273696120059127\n",
      "best merge is 2, 3, 0.394568887835097\n",
      "best merge is 0, 5, 0.450446994105642\n",
      "order classes\n",
      "2 - 0.423680591647449\n",
      "3 - 0.708555930486214\n",
      "2 - 0.408311907261247\n",
      "4 8\n",
      "2 7 3\n",
      "0 1\n",
      "5\n",
      "6\n",
      "Done.-7.70451436388509\t42\t0\n",
      "5.36662106703147\t42\t24.5909712722298\n",
      "6.54445964432285\t42\t26.1422708618331\n",
      "7.20519835841313\t42\t26.9466484268126\n",
      "7.65047879616963\t42\t27.4637482900137\n",
      "7.99521203830369\t42\t27.8372093023256\n",
      "8.296853625171\t42\t28.1532147742818\n",
      "8.62722298221614\t42\t28.4404924760602\n",
      "9.05813953488372\t42\t28.8139534883721\n",
      "9.66142270861833\t42\t29.3023255813953\n",
      "16.0102599179207\t42\t30.0205198358413\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_c5/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3784 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3784\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "344 hits go_terms_num = 189\n",
      "cat 160, mi = 0.00503441, KRAS.DF.V1_UP\n",
      "cat 28, mi = 0.00433243, SRC_UP.V1_UP\n",
      "cat 51, mi = 0.00432477, MEK_UP.V1_UP\n",
      "cat 172, mi = 0.00419906, KRAS.PROSTATE_UP.V1_DN\n",
      "cat 119, mi = 0.00416301, KRAS.KIDNEY_UP.V1_DN\n",
      "cat 7, mi = 0.00407668, PRC1_BMI_UP.V1_UP\n",
      "cat 155, mi = 0.00402004, PDGF_UP.V1_DN\n",
      "cat 47, mi = 0.00396678, HOXA9_DN.V1_UP\n",
      "cat 69, mi = 0.00390727, E2F3_UP.V1_UP\n",
      "cat 67, mi = 0.00383777, RAPA_EARLY_UP.V1_UP\n",
      "cat 31, mi = 0.00375716, CRX_DN.V1_DN\n",
      "cat 72, mi = 0.00374562, KRAS.LUNG_UP.V1_UP\n",
      "cat 37, mi = 0.00349491, MTOR_UP.V1_UP\n",
      "cat 66, mi = 0.00347543, WNT_UP.V1_DN\n",
      "cat 166, mi = 0.00334009, IL21_UP.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 160, mi = 0.005, pvalue = 0.006, KRAS.DF.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.004, pvalue = 0.021, SRC_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.004, pvalue = 0.022, MEK_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 172, mi = 0.004, pvalue = 0.029, KRAS.PROSTATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.004, pvalue = 0.024, KRAS.KIDNEY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.004, pvalue = 0.025, PRC1_BMI_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 155, mi = 0.004, pvalue = 0.036, PDGF_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.004, pvalue = 0.025, HOXA9_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.004, pvalue = 0.029, E2F3_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 67, mi = 0.004, pvalue = 0.040, RAPA_EARLY_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.004, pvalue = 0.043, CRX_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 72, mi = 0.004, pvalue = 0.068, KRAS.LUNG_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.003, pvalue = 0.066, MTOR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.003, pvalue = 0.076, WNT_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.003, pvalue = 0.110, IL21_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 103, mi = 0.003, pvalue = 0.092, RPS14_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 78, mi = 0.003, pvalue = 0.123, BMI1_DN_MEL18_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 87, mi = 0.003, pvalue = 0.099, LEF1_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 178, mi = 0.003, pvalue = 0.111, E2F3_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.003, pvalue = 0.125, DCA_UP.V1_UP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.32\t-9.45\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_c6/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6489 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6489\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "590 hits go_terms_num = 4872\n",
      "cat 2776, mi = 0.00576571, GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "cat 3030, mi = 0.00490466, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 978, mi = 0.00484356, GSE5679_PPARG_LIGAND_ROSIGLITAZONE_VS_RARA_AGONIST_AM580_TREATED_DC_DN\n",
      "cat 766, mi = 0.00482199, GSE7852_TREG_VS_TCONV_LN_DN\n",
      "cat 4017, mi = 0.00457071, GSE37605_C57BL6_VS_NOD_FOXP3_IRES_GFP_TCONV_DN\n",
      "cat 1810, mi = 0.00404656, KAECH_DAY15_EFF_VS_MEMORY_CD8_TCELL_DN\n",
      "cat 4416, mi = 0.00402854, GSE9601_UNTREATED_VS_PI3K_INHIBITOR_TREATED_HCMV_INF_MONOCYTE_UP\n",
      "cat 4087, mi = 0.00400676, GSE23321_CD8_STEM_CELL_MEMORY_VS_EFFECTOR_MEMORY_CD8_TCELL_DN\n",
      "cat 509, mi = 0.00400274, GSE45365_NK_CELL_VS_CD8_TCELL_MCMV_INFECTION_UP\n",
      "cat 3251, mi = 0.00385619, GSE24142_DN2_VS_DN3_THYMOCYTE_DN\n",
      "cat 1333, mi = 0.00383606, GSE22886_NAIVE_TCELL_VS_NKCELL_UP\n",
      "cat 124, mi = 0.00381195, GSE21670_UNTREATED_VS_TGFB_TREATED_STAT3_KO_CD4_TCELL_DN\n",
      "cat 1139, mi = 0.00378533, GSE29618_PRE_VS_DAY7_POST_LAIV_FLU_VACCINE_PDC_DN\n",
      "cat 2589, mi = 0.00376015, GSE22886_NAIVE_CD4_TCELL_VS_NEUTROPHIL_DN\n",
      "cat 2133, mi = 0.00374657, GSE37605_FOXP3_FUSION_GFP_VS_IRES_GFP_TREG_NOD_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2776, mi = 0.006, pvalue = 0.000, GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "Passed: cat 3030, mi = 0.005, pvalue = 0.000, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "Passed: cat 978, mi = 0.005, pvalue = 0.000, GSE5679_PPARG_LIGAND_ROSIGLITAZONE_VS_RARA_AGONIST_AM580_TREATED_DC_DN\n",
      "Passed: cat 4416, mi = 0.004, pvalue = 0.000, GSE9601_UNTREATED_VS_PI3K_INHIBITOR_TREATED_HCMV_INF_MONOCYTE_UP\n",
      "Passed: cat 509, mi = 0.004, pvalue = 0.000, GSE45365_NK_CELL_VS_CD8_TCELL_MCMV_INFECTION_UP\n",
      "Passed: cat 1139, mi = 0.004, pvalue = 0.000, GSE29618_PRE_VS_DAY7_POST_LAIV_FLU_VACCINE_PDC_DN\n",
      "Passed: cat 4720, mi = 0.004, pvalue = 0.000, GSE13484_3H_UNSTIM_VS_YF17D_VACCINE_STIM_PBMC_DN\n",
      "Passed: cat 4045, mi = 0.004, pvalue = 0.001, GSE7218_IGM_VS_IGG_SIGNAL_THGOUGH_ANTIGEN_BCELL_UP\n",
      "Passed: cat 2425, mi = 0.003, pvalue = 0.001, GSE43955_1H_VS_60H_ACT_CD4_TCELL_WITH_TGFB_IL6_UP\n",
      "Passed: cat 3269, mi = 0.003, pvalue = 0.000, GSE1448_CTRL_VS_ANTI_VBETA5_DP_THYMOCYTE_UP\n",
      "Passed: cat 176, mi = 0.003, pvalue = 0.001, GSE37301_CD4_TCELL_VS_RAG2_KO_NK_CELL_UP\n",
      "Passed: cat 2024, mi = 0.003, pvalue = 0.001, GSE5589_WT_VS_IL6_KO_LPS_STIM_MACROPHAGE_45MIN_UP\n",
      "Passed: cat 371, mi = 0.003, pvalue = 0.002, GSE17721_POLYIC_VS_PAM3CSK4_16H_BMDC_DN\n",
      "Passed: cat 3678, mi = 0.003, pvalue = 0.002, GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Passed: cat 3785, mi = 0.003, pvalue = 0.003, GSE5589_UNSTIM_VS_45MIN_LPS_AND_IL6_STIM_MACROPHAGE_DN\n",
      "Passed: cat 2996, mi = 0.003, pvalue = 0.003, GSE22313_HEALTHY_VS_SLE_MOUSE_CD4_TCELL_UP\n",
      "Passed: cat 577, mi = 0.003, pvalue = 0.002, GSE17974_IL4_AND_ANTI_IL12_VS_UNTREATED_0.5H_ACT_CD4_TCELL_UP\n",
      "Passed: cat 2360, mi = 0.003, pvalue = 0.002, GSE22033_UNTREATED_VS_MRL24_TREATED_MEF_UP\n",
      "Passed: cat 4556, mi = 0.003, pvalue = 0.002, GSE2770_IL12_AND_TGFB_VS_IL4_TREATED_ACT_CD4_TCELL_48H_DN\n",
      "Passed: cat 4093, mi = 0.003, pvalue = 0.003, GSE17974_CTRL_VS_ACT_IL4_AND_ANTI_IL12_12H_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1473, mi = 0.003, pvalue = 0.005, GSE45365_NK_CELL_VS_CD8A_DC_MCMV_INFECTION_DN\n",
      "Passed: cat 2000, mi = 0.003, pvalue = 0.004, GSE15139_GMCSF_TREATED_VS_UNTREATED_NEUTROPHILS_UP\n",
      "Passed: cat 4470, mi = 0.003, pvalue = 0.004, GSE24492_LYVE_NEG_VS_POS_MACROPHAGE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2396, mi = 0.003, pvalue = 0.005, GSE17721_LPS_VS_CPG_16H_BMDC_DN\n",
      "Passed: cat 4401, mi = 0.003, pvalue = 0.003, GSE15324_NAIVE_VS_ACTIVATED_ELF4_KO_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 872, mi = 0.003, pvalue = 0.006, GSE19198_1H_VS_6H_IL21_TREATED_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1955, mi = 0.003, pvalue = 0.007, GSE11924_TH1_VS_TH2_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2053, mi = 0.003, pvalue = 0.007, GSE15324_ELF4_KO_VS_WT_ACTIVATED_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4701, mi = 0.003, pvalue = 0.006, GSE45365_NK_CELL_VS_CD11B_DC_MCMV_INFECTION_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1618, mi = 0.003, pvalue = 0.010, GSE13229_IMM_VS_INTMATURE_NKCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1727, mi = 0.003, pvalue = 0.007, GSE8678_IL7R_LOW_VS_HIGH_EFF_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4030, mi = 0.003, pvalue = 0.009, GSE45365_CTRL_VS_MCMV_INFECTION_NK_CELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4198, mi = 0.003, pvalue = 0.011, GSE17974_IL4_AND_ANTI_IL12_VS_UNTREATED_2H_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1928, mi = 0.003, pvalue = 0.011, GSE22886_NAIVE_CD8_TCELL_VS_MEMORY_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.003, pvalue = 0.015, GSE18281_PERIMEDULLARY_CORTICAL_REGION_VS_WHOLE_MEDULLA_THYMUS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4279, mi = 0.003, pvalue = 0.012, GSE29617_CTRL_VS_DAY3_TIV_FLU_VACCINE_PBMC_2008_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4194, mi = 0.003, pvalue = 0.013, GSE31082_DN_VS_DP_THYMOCYTE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2079, mi = 0.003, pvalue = 0.008, GSE27786_BCELL_VS_NEUTROPHIL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3270, mi = 0.003, pvalue = 0.013, GSE24972_MARGINAL_ZONE_BCELL_VS_FOLLICULAR_BCELL_IRF8_KO_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1264, mi = 0.003, pvalue = 0.017, GSE7459_UNTREATED_VS_IL6_TREATED_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4497, mi = 0.003, pvalue = 0.012, GSE18791_CTRL_VS_NEWCASTLE_VIRUS_DC_18H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 613, mi = 0.003, pvalue = 0.015, GSE17721_CPG_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3960, mi = 0.002, pvalue = 0.023, GSE26928_EFF_MEMORY_VS_CXCR5_POS_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 771, mi = 0.002, pvalue = 0.025, GSE19825_NAIVE_VS_DAY3_EFF_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2105, mi = 0.002, pvalue = 0.023, GSE20366_TREG_VS_TCONV_UP\n",
      "Total number of categories evaluated = 267.\n",
      "Number of categories that passed the tests = 23\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.155600860366982\n",
      "best merge is 2, 18, 0.173395917124669\n",
      "best merge is 2, 3, 0.266716239163591\n",
      "best merge is 14, 20, 0.292164085135661\n",
      "best merge is 4, 11, 0.334027668128235\n",
      "best merge is 0, 2, 0.339686821155561\n",
      "best merge is 19, 21, 0.342651829180522\n",
      "best merge is 10, 13, 0.391468205722091\n",
      "best merge is 15, 16, 0.408013632594255\n",
      "best merge is 0, 8, 0.46809385141733\n",
      "best merge is 5, 22, 0.471058375345692\n",
      "best merge is 17, 19, 0.482649033270765\n",
      "best merge is 4, 12, 0.524568709557782\n",
      "best merge is 0, 6, 0.531535739274452\n",
      "best merge is 9, 14, 0.600154002978591\n",
      "best merge is 0, 15, 0.743589270608738\n",
      "best merge is 0, 9, 0.780523629560755\n",
      "best merge is 10, 17, 0.788062439393786\n",
      "best merge is 0, 4, 0.836573423806969\n",
      "order classes\n",
      "12 - 1.11313364568775\n",
      "3 - 2.35650348197092\n",
      "2 - 2.70858565823773\n",
      "5 - 2.36542772751292\n",
      "0 1 2 18 3 8 6 15 16 9 14 20\n",
      "4 11 12\n",
      "5 22\n",
      "7\n",
      "10 13 17 19 21\n",
      "Done.Start drawing\n",
      "5.87\t-9.72\n",
      "GSE17974_CTRL_VS_ACT_IL4_AND_ANTI_IL12_12H_CD4_TCELL_DN\n",
      "GSE5679_PPARG_LIGAND_ROSIGLITAZONE_VS_RARA_AGONIST_AM580_TREATED_DC_DN\n",
      "GSE45365_NK_CELL_VS_CD8_TCELL_MCMV_INFECTION_UP\n",
      "GSE22033_UNTREATED_VS_MRL24_TREATED_MEF_UP\n",
      "GSE24492_LYVE_NEG_VS_POS_MACROPHAGE_DN\n",
      "GSE13484_3H_UNSTIM_VS_YF17D_VACCINE_STIM_PBMC_DN\n",
      "GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "GSE1448_CTRL_VS_ANTI_VBETA5_DP_THYMOCYTE_UP\n",
      "GSE5589_UNSTIM_VS_45MIN_LPS_AND_IL6_STIM_MACROPHAGE_DN\n",
      "GSE15139_GMCSF_TREATED_VS_UNTREATED_NEUTROPHILS_UP\n",
      "GSE43955_1H_VS_60H_ACT_CD4_TCELL_WITH_TGFB_IL6_UP\n",
      "GSE5589_WT_VS_IL6_KO_LPS_STIM_MACROPHAGE_45MIN_UP\n",
      "GSE17721_POLYIC_VS_PAM3CSK4_16H_BMDC_DN\n",
      "GSE7218_IGM_VS_IGG_SIGNAL_THGOUGH_ANTIGEN_BCELL_UP\n",
      "GSE2770_IL12_AND_TGFB_VS_IL4_TREATED_ACT_CD4_TCELL_48H_DN\n",
      "GSE9601_UNTREATED_VS_PI3K_INHIBITOR_TREATED_HCMV_INF_MONOCYTE_UP\n",
      "GSE22313_HEALTHY_VS_SLE_MOUSE_CD4_TCELL_UP\n",
      "GSE17974_IL4_AND_ANTI_IL12_VS_UNTREATED_0.5H_ACT_CD4_TCELL_UP\n",
      "GSE29618_PRE_VS_DAY7_POST_LAIV_FLU_VACCINE_PDC_DN\n",
      "GSE15324_NAIVE_VS_ACTIVATED_ELF4_KO_CD8_TCELL_DN\n",
      "GSE37301_CD4_TCELL_VS_RAG2_KO_NK_CELL_UP\n",
      "GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.155600860366982\n",
      "best merge is 2, 18, 0.173395917124669\n",
      "best merge is 2, 3, 0.266716239163591\n",
      "best merge is 14, 20, 0.292164085135661\n",
      "best merge is 4, 11, 0.334027668128235\n",
      "best merge is 0, 2, 0.339686821155561\n",
      "best merge is 19, 21, 0.342651829180522\n",
      "best merge is 10, 13, 0.391468205722091\n",
      "best merge is 15, 16, 0.408013632594255\n",
      "best merge is 0, 8, 0.46809385141733\n",
      "best merge is 5, 22, 0.471058375345692\n",
      "best merge is 17, 19, 0.482649033270765\n",
      "best merge is 4, 12, 0.524568709557782\n",
      "best merge is 0, 6, 0.531535739274452\n",
      "best merge is 9, 14, 0.600154002978591\n",
      "best merge is 0, 15, 0.743589270608738\n",
      "best merge is 0, 9, 0.780523629560755\n",
      "best merge is 10, 17, 0.788062439393786\n",
      "best merge is 0, 4, 0.836573423806969\n",
      "order classes\n",
      "12 - 1.11313364568775\n",
      "3 - 2.35650348197092\n",
      "2 - 2.70858565823773\n",
      "5 - 2.36542772751292\n",
      "0 1 2 18 3 8 6 15 16 9 14 20\n",
      "4 11 12\n",
      "5 22\n",
      "7\n",
      "10 13 17 19 21\n",
      "Done.-8.48300192431045\t42\t0\n",
      "3.7613855035279\t42\t23.0339961513791\n",
      "4.86593970493906\t42\t24.4887748556767\n",
      "5.49903784477229\t42\t25.2431045542014\n",
      "5.93008338678641\t42\t25.7549711353432\n",
      "6.25336754329699\t42\t26.1051956382296\n",
      "6.53624118024375\t42\t26.4015394483643\n",
      "6.84605516356639\t42\t26.6709429121232\n",
      "7.25016035920462\t42\t27.0211674150096\n",
      "7.82937780628608\t42\t27.4791533033996\n",
      "15.0898011545863\t42\t28.1796023091726\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_c7/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 17786\n",
      "cat 10083, mi = 0.00596156, GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "cat 12530, mi = 0.00586998, GSE7852_TREG_VS_TCONV_LN_DN\n",
      "cat 13611, mi = 0.00581076, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "cat 7922, mi = 0.00498824, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 12289, mi = 0.00476433, GSE5679_PPARG_LIGAND_ROSIGLITAZONE_VS_RARA_AGONIST_AM580_TREATED_DC_DN\n",
      "cat 12444, mi = 0.00467289, GSE7460_TCONV_VS_TREG_LN_UP\n",
      "cat 1178, mi = 0.00451706, DIRMEIER_LMP1_RESPONSE_EARLY\n",
      "cat 13167, mi = 0.00430157, KAECH_DAY15_EFF_VS_MEMORY_CD8_TCELL_DN\n",
      "cat 238, mi = 0.00424809, AZARE_NEOPLASTIC_TRANSFORMATION_BY_STAT3_UP\n",
      "cat 595, mi = 0.00419949, BOSCO_ALLERGEN_INDUCED_TH2_ASSOCIATED_MODULE\n",
      "cat 9849, mi = 0.00411286, GSE23321_CD8_STEM_CELL_MEMORY_VS_EFFECTOR_MEMORY_CD8_TCELL_DN\n",
      "cat 14386, mi = 0.00410786, MODULE_480\n",
      "cat 17102, mi = 0.00408730, WEIGEL_OXIDATIVE_STRESS_BY_HNE_AND_H2O2\n",
      "cat 12646, mi = 0.00405010, GSE9601_UNTREATED_VS_PI3K_INHIBITOR_TREATED_HCMV_INF_MONOCYTE_UP\n",
      "cat 12969, mi = 0.00403913, HOOI_ST7_TARGETS_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 10083, mi = 0.006, pvalue = 0.000, GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "Passed: cat 12530, mi = 0.006, pvalue = 0.000, GSE7852_TREG_VS_TCONV_LN_DN\n",
      "Passed: cat 13611, mi = 0.006, pvalue = 0.000, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "Passed: cat 1178, mi = 0.005, pvalue = 0.000, DIRMEIER_LMP1_RESPONSE_EARLY\n",
      "Passed: cat 14386, mi = 0.004, pvalue = 0.000, MODULE_480\n",
      "Passed: cat 12969, mi = 0.004, pvalue = 0.000, HOOI_ST7_TARGETS_DN\n",
      "Passed: cat 11188, mi = 0.004, pvalue = 0.000, GSE37301_CD4_TCELL_VS_RAG2_KO_NK_CELL_UP\n",
      "Passed: cat 12497, mi = 0.004, pvalue = 0.000, GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "Passed: cat 9761, mi = 0.004, pvalue = 0.000, GSE22886_NAIVE_CD4_TCELL_VS_NEUTROPHIL_DN\n",
      "Passed: cat 8968, mi = 0.003, pvalue = 0.001, GSE17721_PAM3CSK4_VS_GADIQUIMOD_8H_BMDC_UP\n",
      "Passed: cat 15933, mi = 0.003, pvalue = 0.001, REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING\n",
      "Passed: cat 6791, mi = 0.003, pvalue = 0.001, GO_REGULATION_OF_SYNAPSE_ORGANIZATION\n",
      "Passed: cat 7258, mi = 0.003, pvalue = 0.001, GO_SH3_SH2_ADAPTOR_ACTIVITY\n",
      "Passed: cat 15219, mi = 0.003, pvalue = 0.000, PID_HDAC_CLASSII_PATHWAY\n",
      "Passed: cat 12396, mi = 0.003, pvalue = 0.002, GSE7218_IGM_VS_IGG_SIGNAL_THGOUGH_ANTIGEN_BCELL_UP\n",
      "Passed: cat 10146, mi = 0.003, pvalue = 0.001, GSE26030_TH1_VS_TH17_DAY5_POST_POLARIZATION_UP\n",
      "Passed: cat 15211, mi = 0.003, pvalue = 0.002, PID_FCER1_PATHWAY\n",
      "Passed: cat 17083, mi = 0.003, pvalue = 0.003, WATANABE_RECTAL_CANCER_RADIOTHERAPY_RESPONSIVE_UP\n",
      "Passed: cat 12496, mi = 0.003, pvalue = 0.001, GSE7768_OVA_ALONE_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_UP\n",
      "Passed: cat 8426, mi = 0.003, pvalue = 0.003, GSE15139_GMCSF_TREATED_VS_UNTREATED_NEUTROPHILS_UP\n",
      "Passed: cat 1107, mi = 0.003, pvalue = 0.002, DARWICHE_SKIN_TUMOR_PROMOTER_UP\n",
      "Passed: cat 12259, mi = 0.003, pvalue = 0.003, GSE5589_UNSTIM_VS_45MIN_LPS_AND_IL6_STIM_MACROPHAGE_DN\n",
      "Passed: cat 8437, mi = 0.003, pvalue = 0.003, GSE15324_NAIVE_VS_ACTIVATED_ELF4_KO_CD8_TCELL_DN\n",
      "Passed: cat 12043, mi = 0.003, pvalue = 0.004, GSE45365_WT_VS_IFNAR_KO_CD8A_DC_MCMV_INFECTION_UP\n",
      "Didn't pass (cutoff=0.0050): cat 7992, mi = 0.003, pvalue = 0.005, GSE11924_TH1_VS_TH2_CD4_TCELL_UP\n",
      "Passed: cat 3729, mi = 0.003, pvalue = 0.002, GO_IRON_ION_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 10466, mi = 0.003, pvalue = 0.005, GSE27859_DC_VS_CD11C_INT_F480_INT_DC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.003, pvalue = 0.008, AGGAAGC_MIR5163P\n",
      "Didn't pass (cutoff=0.0050): cat 9065, mi = 0.003, pvalue = 0.007, GSE17974_CTRL_VS_ACT_IL4_AND_ANTI_IL12_12H_CD4_TCELL_DN\n",
      "Passed: cat 5979, mi = 0.003, pvalue = 0.003, GO_PYRIMIDINE_CONTAINING_COMPOUND_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 8865, mi = 0.003, pvalue = 0.010, GSE17721_LPS_VS_CPG_16H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 17553, mi = 0.003, pvalue = 0.010, chr17q23\n",
      "Didn't pass (cutoff=0.0050): cat 2236, mi = 0.003, pvalue = 0.009, GO_BODY_MORPHOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 12534, mi = 0.003, pvalue = 0.007, GSE7852_TREG_VS_TCONV_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5445, mi = 0.003, pvalue = 0.009, GO_POSITIVE_REGULATION_OF_INTERFERON_BETA_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 17506, mi = 0.003, pvalue = 0.013, chr13q14\n",
      "Didn't pass (cutoff=0.0050): cat 3811, mi = 0.003, pvalue = 0.010, GO_LIGAND_GATED_CHANNEL_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 15174, mi = 0.003, pvalue = 0.009, PID_BMP_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 9180, mi = 0.003, pvalue = 0.013, GSE19198_1H_VS_6H_IL21_TREATED_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 8092, mi = 0.003, pvalue = 0.012, GSE12839_CTRL_VS_IL12_TREATED_PBMC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5115, mi = 0.003, pvalue = 0.009, GO_PHOSPHATIDYLINOSITOL_3_PHOSPHATE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 3017, mi = 0.003, pvalue = 0.017, GO_EARLY_ENDOSOME_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 11995, mi = 0.003, pvalue = 0.017, GSE45365_BCELL_VS_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2269, mi = 0.003, pvalue = 0.016, GO_CAJAL_BODY\n",
      "Didn't pass (cutoff=0.0050): cat 10028, mi = 0.003, pvalue = 0.012, GSE24972_MARGINAL_ZONE_BCELL_VS_FOLLICULAR_BCELL_IRF8_KO_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1446, mi = 0.003, pvalue = 0.019, GAANYNYGACNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 224, mi = 0.002, pvalue = 0.014, ATGCTGG_MIR338\n",
      "Didn't pass (cutoff=0.0050): cat 7874, mi = 0.002, pvalue = 0.016, GSE10239_KLRG1INT_VS_KLRG1HIGH_EFF_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1688, mi = 0.002, pvalue = 0.013, GGGCATT_MIR365\n",
      "Didn't pass (cutoff=0.0050): cat 6234, mi = 0.002, pvalue = 0.020, GO_REGULATION_OF_CYTOKINESIS\n",
      "Total number of categories evaluated = 442.\n",
      "Number of categories that passed the tests = 26\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 10, 0.0784231409161187\n",
      "best merge is 19, 20, 0.149190737286732\n",
      "best merge is 1, 16, 0.170029958038255\n",
      "best merge is 5, 15, 0.192088010645299\n",
      "best merge is 2, 17, 0.199008981595118\n",
      "best merge is 6, 24, 0.239836345994215\n",
      "best merge is 3, 9, 0.242433741438595\n",
      "best merge is 0, 4, 0.246565645614294\n",
      "best merge is 19, 21, 0.268447598348515\n",
      "best merge is 2, 12, 0.300698991582631\n",
      "best merge is 6, 25, 0.367650902231895\n",
      "best merge is 0, 1, 0.372104466197937\n",
      "best merge is 7, 23, 0.412085132461855\n",
      "best merge is 3, 19, 0.448685792150997\n",
      "best merge is 7, 13, 0.456771935686523\n",
      "best merge is 2, 5, 0.46819112780683\n",
      "best merge is 14, 18, 0.472449537258201\n",
      "best merge is 6, 8, 0.518399001294253\n",
      "best merge is 0, 3, 0.52140597087005\n",
      "best merge is 2, 7, 0.702209353756026\n",
      "best merge is 0, 2, 0.794196016011667\n",
      "best merge is 6, 14, 0.824323506696838\n",
      "order classes\n",
      "18 - 0.881845796479849\n",
      "4 - 2.05589731290187\n",
      "2 - 3.0709219921783\n",
      "0 4 1 16 3 10 9 19 20 21 2 17 12 5 15 7 23 13\n",
      "6 24 25 8\n",
      "11\n",
      "14 18\n",
      "22\n",
      "Done.Start drawing\n",
      "5.87\t-9.72\n",
      "WATANABE_RECTAL_CANCER_RADIOTHERAPY_RESPONSIVE_UP\n",
      "GO_SH3_SH2_ADAPTOR_ACTIVITY\n",
      "LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "GSE45365_WT_VS_IFNAR_KO_CD8A_DC_MCMV_INFECTION_UP\n",
      "GSE25123_CTRL_VS_IL4_STIM_PPARG_KO_MACROPHAGE_DN\n",
      "MODULE_480\n",
      "GSE7852_TREG_VS_TCONV_LN_DN\n",
      "DIRMEIER_LMP1_RESPONSE_EARLY\n",
      "REACTOME_PRE_NOTCH_EXPRESSION_AND_PROCESSING\n",
      "GSE17721_PAM3CSK4_VS_GADIQUIMOD_8H_BMDC_UP\n",
      "GSE15139_GMCSF_TREATED_VS_UNTREATED_NEUTROPHILS_UP\n",
      "DARWICHE_SKIN_TUMOR_PROMOTER_UP\n",
      "GSE5589_UNSTIM_VS_45MIN_LPS_AND_IL6_STIM_MACROPHAGE_DN\n",
      "GSE7218_IGM_VS_IGG_SIGNAL_THGOUGH_ANTIGEN_BCELL_UP\n",
      "PID_FCER1_PATHWAY\n",
      "HOOI_ST7_TARGETS_DN\n",
      "GSE26030_TH1_VS_TH17_DAY5_POST_POLARIZATION_UP\n",
      "GSE7768_OVA_WITH_LPS_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_DN\n",
      "PID_HDAC_CLASSII_PATHWAY\n",
      "GO_REGULATION_OF_SYNAPSE_ORGANIZATION\n",
      "GSE15324_NAIVE_VS_ACTIVATED_ELF4_KO_CD8_TCELL_DN\n",
      "GSE7768_OVA_ALONE_VS_OVA_WITH_MPL_IMMUNIZED_MOUSE_WHOLE_SPLEEN_6H_UP\n",
      "GSE37301_CD4_TCELL_VS_RAG2_KO_NK_CELL_UP\n",
      "GO_IRON_ION_TRANSPORT\n",
      "GO_PYRIMIDINE_CONTAINING_COMPOUND_CATABOLIC_PROCESS\n",
      "GSE22886_NAIVE_CD4_TCELL_VS_NEUTROPHIL_DN\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 10, 0.0784231409161187\n",
      "best merge is 19, 20, 0.149190737286732\n",
      "best merge is 1, 16, 0.170029958038255\n",
      "best merge is 5, 15, 0.192088010645299\n",
      "best merge is 2, 17, 0.199008981595118\n",
      "best merge is 6, 24, 0.239836345994215\n",
      "best merge is 3, 9, 0.242433741438595\n",
      "best merge is 0, 4, 0.246565645614294\n",
      "best merge is 19, 21, 0.268447598348515\n",
      "best merge is 2, 12, 0.300698991582631\n",
      "best merge is 6, 25, 0.367650902231895\n",
      "best merge is 0, 1, 0.372104466197937\n",
      "best merge is 7, 23, 0.412085132461855\n",
      "best merge is 3, 19, 0.448685792150997\n",
      "best merge is 7, 13, 0.456771935686523\n",
      "best merge is 2, 5, 0.46819112780683\n",
      "best merge is 14, 18, 0.472449537258201\n",
      "best merge is 6, 8, 0.518399001294253\n",
      "best merge is 0, 3, 0.52140597087005\n",
      "best merge is 2, 7, 0.702209353756026\n",
      "best merge is 0, 2, 0.794196016011667\n",
      "best merge is 6, 14, 0.824323506696838\n",
      "order classes\n",
      "18 - 0.881845796479849\n",
      "4 - 2.05589731290187\n",
      "2 - 3.0709219921783\n",
      "0 4 1 16 3 10 9 19 20 21 2 17 12 5 15 7 23 13\n",
      "6 24 25 8\n",
      "11\n",
      "14 18\n",
      "22\n",
      "Done.-8.50994227068634\t42\t0\n",
      "3.73444515715202\t42\t22.9801154586273\n",
      "4.85246953175112\t42\t24.4887748556767\n",
      "5.47209749839641\t42\t25.2161642078255\n",
      "5.90314304041052\t42\t25.7280307889673\n",
      "6.23989737010905\t42\t26.0782552918538\n",
      "6.53624118024375\t42\t26.4015394483643\n",
      "6.84605516356639\t42\t26.6709429121232\n",
      "7.25016035920462\t42\t27.0211674150096\n",
      "7.82937780628608\t42\t27.4791533033996\n",
      "15.0898011545863\t42\t28.1796023091726\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_full/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 1545 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 1545\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "140 hits go_terms_num = 50\n",
      "cat 1, mi = 0.01149582, HALLMARK_KRAS_SIGNALING_DN\n",
      "cat 26, mi = 0.01022778, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "cat 32, mi = 0.00911121, HALLMARK_BILE_ACID_METABOLISM\n",
      "cat 5, mi = 0.00897567, HALLMARK_PROTEIN_SECRETION\n",
      "cat 20, mi = 0.00822935, HALLMARK_MYC_TARGETS_V1\n",
      "cat 48, mi = 0.00821156, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 47, mi = 0.00821046, HALLMARK_COMPLEMENT\n",
      "cat 9, mi = 0.00793002, HALLMARK_SPERMATOGENESIS\n",
      "cat 15, mi = 0.00758129, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 0, mi = 0.00714170, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 36, mi = 0.00693622, HALLMARK_DNA_REPAIR\n",
      "cat 2, mi = 0.00681450, HALLMARK_PEROXISOME\n",
      "cat 29, mi = 0.00674444, HALLMARK_MYC_TARGETS_V2\n",
      "cat 14, mi = 0.00635568, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 27, mi = 0.00604921, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 1, mi = 0.011, pvalue = 0.009, HALLMARK_KRAS_SIGNALING_DN\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.010, pvalue = 0.020, HALLMARK_TNFA_SIGNALING_VIA_NFKB\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.009, pvalue = 0.058, HALLMARK_BILE_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.009, pvalue = 0.056, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.008, pvalue = 0.104, HALLMARK_MYC_TARGETS_V1\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.008, pvalue = 0.070, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.008, pvalue = 0.075, HALLMARK_COMPLEMENT\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.008, pvalue = 0.132, HALLMARK_SPERMATOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.008, pvalue = 0.129, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 0, mi = 0.007, pvalue = 0.206, HALLMARK_FATTY_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.007, pvalue = 0.173, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.007, pvalue = 0.254, HALLMARK_PEROXISOME\n",
      "Didn't pass (cutoff=0.0050): cat 29, mi = 0.007, pvalue = 0.238, HALLMARK_MYC_TARGETS_V2\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.006, pvalue = 0.221, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.006, pvalue = 0.248, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.006, pvalue = 0.240, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.006, pvalue = 0.305, HALLMARK_ANDROGEN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.006, pvalue = 0.272, HALLMARK_GLYCOLYSIS\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.006, pvalue = 0.426, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.006, pvalue = 0.317, HALLMARK_HEME_METABOLISM\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.04\t-9.45\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_msigdb_h/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7491 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7491\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "681 hits go_terms_num = 790\n",
      "cat 275, mi = 0.00181845, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 237, mi = 0.00181631, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 635, mi = 0.00176475, TARDBP_H9_POSTAR_CIMS\n",
      "cat 668, mi = 0.00175740, TNRC6A_HEK293_Plass\n",
      "cat 236, mi = 0.00173504, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 637, mi = 0.00170870, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 569, mi = 0.00164458, SLBP_K562_POSTAR_eCLIP\n",
      "cat 484, mi = 0.00150824, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 321, mi = 0.00148469, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 449, mi = 0.00145670, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00143380, AUH_K562_POSTAR_eCLIP\n",
      "cat 672, mi = 0.00137774, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 32, mi = 0.00136908, BCCIP_HepG2_POSTAR_eCLIP\n",
      "cat 282, mi = 0.00129417, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "cat 277, mi = 0.00123116, HNRNPA1_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.074, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.073, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.082, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.055, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.096, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.092, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.104, SLBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.131, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.001, pvalue = 0.166, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.001, pvalue = 0.177, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.001, pvalue = 0.270, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.234, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.174, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.164, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.001, pvalue = 0.261, HNRNPA1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.001, pvalue = 0.308, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.001, pvalue = 0.319, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.001, pvalue = 0.344, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 553, mi = 0.001, pvalue = 0.588, SBDS_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.636, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.87\t-9.72\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_RBPs_all_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6816 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6816\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "620 hits go_terms_num = 790\n",
      "cat 449, mi = 0.00241435, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 275, mi = 0.00190285, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 237, mi = 0.00188466, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 635, mi = 0.00186788, TARDBP_H9_POSTAR_CIMS\n",
      "cat 637, mi = 0.00180613, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 236, mi = 0.00179467, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 668, mi = 0.00175380, TNRC6A_HEK293_Plass\n",
      "cat 569, mi = 0.00170457, SLBP_K562_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00157521, AUH_K562_POSTAR_eCLIP\n",
      "cat 672, mi = 0.00151207, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 402, mi = 0.00141050, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 256, mi = 0.00139650, GNL3_K562_POSTAR_eCLIP\n",
      "cat 770, mi = 0.00135947, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00135025, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 312, mi = 0.00130440, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.019, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.092, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.097, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.099, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.112, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.125, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.101, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.132, SLBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.259, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.228, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.001, pvalue = 0.291, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.001, pvalue = 0.228, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.001, pvalue = 0.254, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.001, pvalue = 0.306, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.001, pvalue = 0.331, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.001, pvalue = 0.282, LARP7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.001, pvalue = 0.394, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.385, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.330, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 553, mi = 0.001, pvalue = 0.627, SBDS_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.87\t-9.72\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_RBPs_coding_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6678 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6678\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "607 hits go_terms_num = 790\n",
      "cat 250, mi = 0.00406610, FXR2_K562_POSTAR_eCLIP\n",
      "cat 242, mi = 0.00388056, FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 240, mi = 0.00284017, FXR1_HEK293_POSTAR_PAR-CLIP\n",
      "cat 241, mi = 0.00284017, FXR1_HEK293_POSTAR_PARalyzer\n",
      "cat 341, mi = 0.00282470, IGF2BP1_K562_POSTAR_eCLIP\n",
      "cat 221, mi = 0.00270640, FTO_K562_POSTAR_eCLIP\n",
      "cat 575, mi = 0.00264121, SMNDC1_HepG2_POSTAR_eCLIP\n",
      "cat 431, mi = 0.00252706, NKRF_HepG2_POSTAR_eCLIP\n",
      "cat 111, mi = 0.00245146, DDX42_K562_POSTAR_eCLIP\n",
      "cat 774, mi = 0.00242439, YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "cat 693, mi = 0.00242192, U2AF2_HeLa_POSTAR_Piranha_0.01\n",
      "cat 696, mi = 0.00239667, U2AF2_K562_POSTAR_eCLIP\n",
      "cat 315, mi = 0.00236979, HNRNPM_HepG2_POSTAR_eCLIP\n",
      "cat 692, mi = 0.00236250, U2AF2_HeLa_POSTAR_HITS-CLIP\n",
      "cat 648, mi = 0.00235524, TBRG4_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 250, mi = 0.004, pvalue = 0.000, FXR2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 575, mi = 0.003, pvalue = 0.009, SMNDC1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.002, pvalue = 0.018, DDX42_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 774, mi = 0.002, pvalue = 0.020, YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 648, mi = 0.002, pvalue = 0.019, TBRG4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 612, mi = 0.002, pvalue = 0.037, SUPV3L1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 94, mi = 0.002, pvalue = 0.039, CSTF2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.002, pvalue = 0.045, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.002, pvalue = 0.033, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.089, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 413, mi = 0.002, pvalue = 0.157, METTL3_HeLa_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 414, mi = 0.002, pvalue = 0.151, METTL3_HeLa_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 467, mi = 0.002, pvalue = 0.075, PRKRA_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 628, mi = 0.002, pvalue = 0.082, TARBP2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 445, mi = 0.002, pvalue = 0.123, NOP58_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 623, mi = 0.002, pvalue = 0.133, TAF15_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 691, mi = 0.002, pvalue = 0.218, U2AF2_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 626, mi = 0.002, pvalue = 0.226, TARBP2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 356, mi = 0.002, pvalue = 0.198, IGF2BP3_Panc1_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 358, mi = 0.002, pvalue = 0.188, IGF2BP3_Panc1_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.002, pvalue = 0.169, HNRNPA1_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 69.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.90\t-9.72\n",
      "FXR2_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.70451436388509\t42\t0\n",
      "5.36662106703147\t42\t24.5909712722298\n",
      "6.53009575923393\t42\t26.1422708618331\n",
      "7.17647058823529\t42\t26.9179206566347\n",
      "7.63611491108071\t42\t27.4350205198358\n",
      "7.99521203830369\t42\t27.8372093023256\n",
      "8.296853625171\t42\t28.1532147742818\n",
      "8.62722298221614\t42\t28.4404924760602\n",
      "9.05813953488372\t42\t28.8139534883721\n",
      "9.67578659370725\t42\t29.3023255813953\n",
      "16.0246238030096\t42\t30.0492476060192\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6602 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6602\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "600 hits go_terms_num = 790\n",
      "cat 510, mi = 0.00332455, RBFOX2_K562_POSTAR_eCLIP\n",
      "cat 346, mi = 0.00254525, IGF2BP2_K562_POSTAR_eCLIP\n",
      "cat 267, mi = 0.00252488, GTF2F1_K562_POSTAR_eCLIP\n",
      "cat 398, mi = 0.00236601, LSM11_K562_POSTAR_eCLIP\n",
      "cat 605, mi = 0.00233996, STAU1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 489, mi = 0.00227868, PTBP1_K562_POSTAR_eCLIP\n",
      "cat 221, mi = 0.00226683, FTO_K562_POSTAR_eCLIP\n",
      "cat 459, mi = 0.00223757, PCBP2_HepG2_POSTAR_eCLIP\n",
      "cat 583, mi = 0.00218236, SRRM4_HEK293T_POSTAR_CIMS\n",
      "cat 684, mi = 0.00215771, TROVE2_K562_POSTAR_eCLIP\n",
      "cat 508, mi = 0.00215407, RBFOX2_HepG2_POSTAR_eCLIP\n",
      "cat 788, mi = 0.00206346, ZRANB2_K562_POSTAR_eCLIP\n",
      "cat 571, mi = 0.00205969, SLTM_HepG2_POSTAR_eCLIP\n",
      "cat 206, mi = 0.00203062, FBL_HEK293_POSTAR_PARalyzer\n",
      "cat 176, mi = 0.00201026, EIF4G2_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 510, mi = 0.003, pvalue = 0.001, RBFOX2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 346, mi = 0.003, pvalue = 0.017, IGF2BP2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 398, mi = 0.002, pvalue = 0.022, LSM11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 605, mi = 0.002, pvalue = 0.036, STAU1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 221, mi = 0.002, pvalue = 0.035, FTO_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 684, mi = 0.002, pvalue = 0.015, TROVE2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 571, mi = 0.002, pvalue = 0.077, SLTM_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 206, mi = 0.002, pvalue = 0.081, FBL_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.002, pvalue = 0.061, EIF4G2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 751, mi = 0.002, pvalue = 0.030, YTHDC2_HeLa_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 205, mi = 0.002, pvalue = 0.093, FBL_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.002, pvalue = 0.083, FKBP4_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 485, mi = 0.002, pvalue = 0.091, PTBP1_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 58, mi = 0.002, pvalue = 0.010, CPSF-160_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.002, pvalue = 0.130, DICER1_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.002, pvalue = 0.034, CDC40_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.002, pvalue = 0.035, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 439, mi = 0.002, pvalue = 0.044, NOP56_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 440, mi = 0.002, pvalue = 0.043, NOP56_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 431, mi = 0.002, pvalue = 0.122, NKRF_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 190, mi = 0.002, pvalue = 0.143, EWSR1_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 25.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.87\t-9.72\n",
      "RBFOX2_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.48300192431045\t42\t0\n",
      "3.77485567671584\t42\t23.0339961513791\n",
      "4.87940987812701\t42\t24.5157152020526\n",
      "5.49903784477229\t42\t25.2431045542014\n",
      "5.93008338678641\t42\t25.7549711353432\n",
      "6.25336754329699\t42\t26.1051956382296\n",
      "6.53624118024375\t42\t26.4015394483643\n",
      "6.83258499037845\t42\t26.6709429121232\n",
      "7.23669018601668\t42\t26.9942270686337\n",
      "7.82937780628608\t42\t27.4791533033996\n",
      "15.0898011545863\t42\t28.1796023091726\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6691 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6691\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "608 hits go_terms_num = 790\n",
      "cat 28, mi = 0.00358561, ATXN2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 233, mi = 0.00337921, FUS_HEK293_POSTAR_Piranha_0.01\n",
      "cat 185, mi = 0.00331495, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "cat 221, mi = 0.00325133, FTO_K562_POSTAR_eCLIP\n",
      "cat 301, mi = 0.00301246, HNRNPF_HEK293T_POSTAR_HITS-CLIP\n",
      "cat 302, mi = 0.00301246, HNRNPF_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 22, mi = 0.00297992, ALKBH5_HEK293_Plass\n",
      "cat 248, mi = 0.00295784, FXR2_HEK293_POSTAR_Piranha_0.01\n",
      "cat 39, mi = 0.00284586, C17orf85_HEK293_Plass\n",
      "cat 43, mi = 0.00283168, CAPRIN1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 338, mi = 0.00268388, IGF2BP1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 122, mi = 0.00256129, DDX6_HepG2_POSTAR_eCLIP\n",
      "cat 208, mi = 0.00253903, FBL_HEK293_Plass\n",
      "cat 319, mi = 0.00252924, HNRNPUL1_HepG2_POSTAR_eCLIP\n",
      "cat 548, mi = 0.00251746, RTCB_HEK293_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 28, mi = 0.004, pvalue = 0.000, ATXN2_HEK293T_POSTAR_Piranha_0.01\n",
      "Passed: cat 22, mi = 0.003, pvalue = 0.003, ALKBH5_HEK293_Plass\n",
      "Passed: cat 39, mi = 0.003, pvalue = 0.003, C17orf85_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 319, mi = 0.003, pvalue = 0.018, HNRNPUL1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 678, mi = 0.002, pvalue = 0.023, TRA2A_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 313, mi = 0.002, pvalue = 0.018, HNRNPM_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 314, mi = 0.002, pvalue = 0.022, HNRNPM_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.002, pvalue = 0.022, DDX55_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 361, mi = 0.002, pvalue = 0.034, ILF3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.002, pvalue = 0.042, SFPQ_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 621, mi = 0.002, pvalue = 0.038, TAF15_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.023, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 612, mi = 0.002, pvalue = 0.051, SUPV3L1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.002, pvalue = 0.071, HLTF_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 258, mi = 0.002, pvalue = 0.049, GPKOW_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 530, mi = 0.002, pvalue = 0.089, RBM27_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 593, mi = 0.002, pvalue = 0.099, SRSF7_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 514, mi = 0.002, pvalue = 0.103, RBM10_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 225, mi = 0.002, pvalue = 0.074, FUS_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 267, mi = 0.002, pvalue = 0.075, GTF2F1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 669, mi = 0.002, pvalue = 0.103, TNRC6A_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 178, mi = 0.002, pvalue = 0.055, ELAVL1_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.130, ELAVL1_HEK293_POSTAR_HITS-CLIP\n",
      "Total number of categories evaluated = 53.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.999364354596772\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "4.90\t-9.72\n",
      "ALKBH5_HEK293_Plass\n",
      "C17orf85_HEK293_Plass\n",
      "ATXN2_HEK293T_POSTAR_Piranha_0.01\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.999364354596772\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-7.70451436388509\t42\t0\n",
      "5.36662106703147\t42\t24.5909712722298\n",
      "6.53009575923393\t42\t26.1422708618331\n",
      "7.17647058823529\t42\t26.9179206566347\n",
      "7.63611491108071\t42\t27.4350205198358\n",
      "7.98084815321478\t42\t27.8372093023256\n",
      "8.28248974008208\t42\t28.124487004104\n",
      "8.61285909712722\t42\t28.4404924760602\n",
      "9.0437756497948\t42\t28.7852257181943\n",
      "9.67578659370725\t42\t29.3023255813953\n",
      "16.0246238030096\t42\t30.0492476060192\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 790\n",
      "cat 456, mi = 0.00230595, PAPD5_HEK293_Plass\n",
      "cat 686, mi = 0.00228206, TTP_HEK293_Plass\n",
      "cat 22, mi = 0.00201683, ALKBH5_HEK293_Plass\n",
      "cat 275, mi = 0.00200966, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 629, mi = 0.00177309, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "cat 770, mi = 0.00166967, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 312, mi = 0.00164733, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "cat 705, mi = 0.00161049, UPF1_HEK293_Plass\n",
      "cat 256, mi = 0.00160763, GNL3_K562_POSTAR_eCLIP\n",
      "cat 126, mi = 0.00157132, DGCR8_HEK293T_POSTAR_CIMS\n",
      "cat 635, mi = 0.00157111, TARDBP_H9_POSTAR_CIMS\n",
      "cat 637, mi = 0.00157111, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 237, mi = 0.00156274, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 236, mi = 0.00154804, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 31, mi = 0.00154022, AUH_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 456, mi = 0.002, pvalue = 0.044, PAPD5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 686, mi = 0.002, pvalue = 0.036, TTP_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 22, mi = 0.002, pvalue = 0.060, ALKBH5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.095, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.132, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.002, pvalue = 0.154, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.002, pvalue = 0.197, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 705, mi = 0.002, pvalue = 0.211, UPF1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.002, pvalue = 0.170, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.002, pvalue = 0.185, DGCR8_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.255, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.261, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.288, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.292, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.306, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 671, mi = 0.002, pvalue = 0.205, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.001, pvalue = 0.330, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 675, mi = 0.001, pvalue = 0.248, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.001, pvalue = 0.332, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.001, pvalue = 0.260, C17orf85_HEK293_Plass\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.87\t-9.72\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 101\n",
      "cat 56, mi = 0.00188624, RBM3\n",
      "cat 13, mi = 0.00120929, FXR2\n",
      "cat 9, mi = 0.00117787, ESRP2\n",
      "cat 62, mi = 0.00104475, RBM46\n",
      "cat 69, mi = 0.00103557, RBMS3.1\n",
      "cat 27, mi = 0.00102067, HuR.2\n",
      "cat 6, mi = 0.00098212, CPEB4\n",
      "cat 88, mi = 0.00077756, SRSF9.1\n",
      "cat 34, mi = 0.00071213, KHDRBS3\n",
      "cat 21, mi = 0.00070426, HNRNPK\n",
      "cat 68, mi = 0.00059099, RBMS3\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "cat 3, mi = 0.00000000, BRUNOL6\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.002, pvalue = 0.114, RBM3\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.412, FXR2\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.627, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.672, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.563, RBMS3.1\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.597, HuR.2\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.635, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.001, pvalue = 0.774, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.919, KHDRBS3\n",
      "Didn't pass: cat 21, mi = 0.001, pvalue = 1.000, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.001, pvalue = 0.896, RBMS3\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, FMR1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.87\t-9.72\n",
      "Outputing EPS file deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/d_mtyl_METTL14_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL14_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL14_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL14_vs_NT_human_ensembl_RBPs_DeepBind/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7509 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7509\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "683 hits go_terms_num = 17689\n",
      "cat 2545, mi = 0.00289456, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "cat 2530, mi = 0.00288150, proteolysis involved in cellular protein catabolic process\n",
      "cat 2039, mi = 0.00287304, Golgi to plasma membrane protein transport\n",
      "cat 2898, mi = 0.00279146, telomeric DNA binding\n",
      "cat 2611, mi = 0.00273733, positive regulation of axon extension\n",
      "cat 3612, mi = 0.00260745, N-glycan processing\n",
      "cat 378, mi = 0.00255533, negative regulation of BMP signaling pathway\n",
      "cat 11275, mi = 0.00253466, insulin-like growth factor receptor signaling pathway\n",
      "cat 328, mi = 0.00252483, cell redox homeostasis\n",
      "cat 1440, mi = 0.00250974, ventricular septum development\n",
      "cat 4696, mi = 0.00246485, osteoblast development\n",
      "cat 2177, mi = 0.00246372, regulation of B cell differentiation\n",
      "cat 6001, mi = 0.00246329, histone deacetylase complex\n",
      "cat 412, mi = 0.00245409, heparan sulfate proteoglycan biosynthetic process\n",
      "cat 2449, mi = 0.00239659, Set1C/COMPASS complex\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2545, mi = 0.003, pvalue = 0.002, nuclear-transcribed mRNA catabolic process, nonsense-mediated decay\n",
      "Passed: cat 2530, mi = 0.003, pvalue = 0.000, proteolysis involved in cellular protein catabolic process\n",
      "Passed: cat 2039, mi = 0.003, pvalue = 0.000, Golgi to plasma membrane protein transport\n",
      "Passed: cat 3612, mi = 0.003, pvalue = 0.003, N-glycan processing\n",
      "Passed: cat 378, mi = 0.003, pvalue = 0.004, negative regulation of BMP signaling pathway\n",
      "Passed: cat 11275, mi = 0.003, pvalue = 0.002, insulin-like growth factor receptor signaling pathway\n",
      "Passed: cat 328, mi = 0.003, pvalue = 0.004, cell redox homeostasis\n",
      "Passed: cat 4696, mi = 0.002, pvalue = 0.003, osteoblast development\n",
      "Passed: cat 2177, mi = 0.002, pvalue = 0.004, regulation of B cell differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 6001, mi = 0.002, pvalue = 0.008, histone deacetylase complex\n",
      "Didn't pass (cutoff=0.0050): cat 412, mi = 0.002, pvalue = 0.006, heparan sulfate proteoglycan biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 2449, mi = 0.002, pvalue = 0.007, Set1C/COMPASS complex\n",
      "Didn't pass (cutoff=0.0050): cat 415, mi = 0.002, pvalue = 0.009, glycosaminoglycan biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 1280, mi = 0.002, pvalue = 0.012, protein K48-linked ubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 7931, mi = 0.002, pvalue = 0.008, regulation of axonogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 1049, mi = 0.002, pvalue = 0.013, cell adhesion molecule binding\n",
      "Didn't pass (cutoff=0.0050): cat 5066, mi = 0.002, pvalue = 0.006, mRNA 5'-UTR binding\n",
      "Didn't pass (cutoff=0.0050): cat 2106, mi = 0.002, pvalue = 0.016, endosome organization\n",
      "Didn't pass (cutoff=0.0050): cat 1705, mi = 0.002, pvalue = 0.011, toxin transport\n",
      "Passed: cat 4638, mi = 0.002, pvalue = 0.003, maternal placenta development\n",
      "Didn't pass (cutoff=0.0050): cat 2443, mi = 0.002, pvalue = 0.017, histone methyltransferase complex\n",
      "Didn't pass (cutoff=0.0050): cat 2360, mi = 0.002, pvalue = 0.021, regulation of mitotic nuclear division\n",
      "Passed: cat 1989, mi = 0.002, pvalue = 0.004, MHC class I protein binding\n",
      "Passed: cat 7324, mi = 0.002, pvalue = 0.004, G-protein coupled purinergic nucleotide receptor activity\n",
      "Didn't pass (cutoff=0.0050): cat 1141, mi = 0.002, pvalue = 0.027, fatty acid metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 2355, mi = 0.002, pvalue = 0.021, single organismal cell-cell adhesion\n",
      "Didn't pass (cutoff=0.0050): cat 3519, mi = 0.002, pvalue = 0.024, SH3/SH2 adaptor activity\n",
      "Passed: cat 2499, mi = 0.002, pvalue = 0.004, biomineral tissue development\n",
      "Didn't pass (cutoff=0.0050): cat 3212, mi = 0.002, pvalue = 0.025, negative regulation of extrinsic apoptotic signaling pathway in absence of ligand\n",
      "Didn't pass (cutoff=0.0050): cat 103, mi = 0.002, pvalue = 0.014, growth factor activity\n",
      "Didn't pass (cutoff=0.0050): cat 5262, mi = 0.002, pvalue = 0.009, dynein light intermediate chain binding\n",
      "Didn't pass (cutoff=0.0050): cat 1268, mi = 0.002, pvalue = 0.027, mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 2671, mi = 0.002, pvalue = 0.031, MyD88-dependent toll-like receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.002, pvalue = 0.038, neuron apoptotic process\n",
      "Didn't pass (cutoff=0.0050): cat 180, mi = 0.002, pvalue = 0.023, metanephros development\n",
      "Didn't pass (cutoff=0.0050): cat 1940, mi = 0.002, pvalue = 0.036, regulation of RNA splicing\n",
      "Didn't pass (cutoff=0.0050): cat 2383, mi = 0.002, pvalue = 0.038, dendritic spine\n",
      "Didn't pass (cutoff=0.0050): cat 6535, mi = 0.002, pvalue = 0.006, 3',5'-cyclic-nucleotide phosphodiesterase activity\n",
      "Didn't pass (cutoff=0.0050): cat 6536, mi = 0.002, pvalue = 0.006, 3',5'-cyclic-AMP phosphodiesterase activity\n",
      "Didn't pass (cutoff=0.0050): cat 6538, mi = 0.002, pvalue = 0.006, cAMP catabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1609, mi = 0.002, pvalue = 0.039, positive regulation of multicellular organism growth\n",
      "Didn't pass (cutoff=0.0050): cat 3581, mi = 0.002, pvalue = 0.010, negative regulation of epidermal growth factor receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 2037, mi = 0.002, pvalue = 0.031, Golgi stack\n",
      "Didn't pass (cutoff=0.0050): cat 1475, mi = 0.002, pvalue = 0.043, positive regulation of B cell proliferation\n",
      "Didn't pass (cutoff=0.0050): cat 792, mi = 0.002, pvalue = 0.016, anion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 7809, mi = 0.002, pvalue = 0.019, ion channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 8241, mi = 0.002, pvalue = 0.018, volume-sensitive anion channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 1160, mi = 0.002, pvalue = 0.043, Rab guanyl-nucleotide exchange factor activity\n",
      "Total number of categories evaluated = 81.\n",
      "Number of categories that passed the tests = 13\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 11, 0.0971695769661867\n",
      "best merge is 7, 12, 0.194285900474921\n",
      "best merge is 4, 9, 0.199827964663555\n",
      "best merge is 0, 8, 0.324410761393425\n",
      "best merge is 2, 6, 0.398234525906436\n",
      "best merge is 1, 4, 0.468327221028751\n",
      "best merge is 2, 3, 0.766835577286718\n",
      "best merge is 0, 2, 0.862452306799406\n",
      "best merge is 5, 7, 0.870822713817851\n",
      "order classes\n",
      "6 - 1.2540395323478\n",
      "3 - 1.09439046573139\n",
      "2 - 0.631429176543492\n",
      "7 12\n",
      "1 4 9\n",
      "5\n",
      "0 8 2 11 6 3\n",
      "10\n",
      "Done.Start drawing\n",
      "9.21\t-11.27\n",
      "insulin-like growth factor receptor signaling pathway, GO:0048009\n",
      "proteolysis involved in cellular protein catabolic process, GO:0051603\n",
      "negative regulation of BMP signaling pathway, GO:0030514\n",
      "maternal placenta development, GO:0001893\n",
      "regulation of B cell differentiation, GO:0045577\n",
      "N-glycan processing, GO:0006491\n",
      "biomineral tissue development, GO:0031214\n",
      "osteoblast development, GO:0002076\n",
      "nuclear-transcribed mRNA catabolic process, nonsense-mediated decay, GO:0000184\n",
      "Golgi to plasma membrane protein transport, GO:0043001\n",
      "G-protein coupled purinergic nucleotide receptor activity, GO:0045028\n",
      "cell redox homeostasis, GO:0045454\n",
      "MHC class I protein binding, GO:0042288\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 11, 0.0971695769661867\n",
      "best merge is 7, 12, 0.194285900474921\n",
      "best merge is 4, 9, 0.199827964663555\n",
      "best merge is 0, 8, 0.324410761393425\n",
      "best merge is 2, 6, 0.398234525906436\n",
      "best merge is 1, 4, 0.468327221028751\n",
      "best merge is 2, 3, 0.766835577286718\n",
      "best merge is 0, 2, 0.862452306799406\n",
      "best merge is 5, 7, 0.870822713817851\n",
      "order classes\n",
      "6 - 1.2540395323478\n",
      "3 - 1.09439046573139\n",
      "2 - 0.631429176543492\n",
      "7 12\n",
      "1 4 9\n",
      "5\n",
      "0 8 2 11 6 3\n",
      "10\n",
      "Done.-10.31005859375\t42\t0\n",
      "0.1591796875\t42\t19.3798828125\n",
      "1.369140625\t42\t20.9384765625\n",
      "2.107421875\t42\t21.7998046875\n",
      "2.640625\t42\t22.4150390625\n",
      "3.08154296875\t42\t22.8662109375\n",
      "3.4814453125\t42\t23.296875\n",
      "3.86083984375\t42\t23.666015625\n",
      "4.28125\t42\t24.0556640625\n",
      "4.84521484375\t42\t24.5068359375\n",
      "13.591796875\t42\t25.18359375\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6496 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6496\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "591 hits go_terms_num = 181\n",
      "cat 131, mi = 0.00217345, SMARCA4_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00144312, POLR3G_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00131847, BRF2_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00101828, PPARGC1A_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00101126, BDP1_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00075718, XRCC4_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00063436, BRF1_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00045414, POU5F1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.002, pvalue = 0.065, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.306, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.553, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.789, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.624, BDP1_ChIP-seq_regulon\n",
      "Didn't pass: cat 167, mi = 0.001, pvalue = 1.000, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.923, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.000, pvalue = 0.953, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.61\t-11.27\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_encode_tf/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 298\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6547 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6547\t298\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "595 hits go_terms_num = 298\n",
      "cat 165, mi = 0.00325848, chr7p14\n",
      "cat 122, mi = 0.00274993, chr2q36\n",
      "cat 2, mi = 0.00265262, chr1p32\n",
      "cat 158, mi = 0.00249528, chr1p34\n",
      "cat 296, mi = 0.00248989, chr16p11\n",
      "cat 280, mi = 0.00244907, chr7q31\n",
      "cat 18, mi = 0.00242814, chr13q14\n",
      "cat 274, mi = 0.00231352, chr18q22\n",
      "cat 168, mi = 0.00229371, chr7q33\n",
      "cat 74, mi = 0.00229357, chr2p13\n",
      "cat 6, mi = 0.00218575, chr11q12\n",
      "cat 222, mi = 0.00215951, chr5q14\n",
      "cat 8, mi = 0.00214433, chr6q22\n",
      "cat 116, mi = 0.00211186, chr1p13\n",
      "cat 132, mi = 0.00208434, chr12q21\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 165, mi = 0.003, pvalue = 0.001, chr7p14\n",
      "Passed: cat 122, mi = 0.003, pvalue = 0.004, chr2q36\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.003, pvalue = 0.015, chr1p32\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.002, pvalue = 0.018, chr1p34\n",
      "Didn't pass (cutoff=0.0050): cat 296, mi = 0.002, pvalue = 0.021, chr16p11\n",
      "Didn't pass (cutoff=0.0050): cat 280, mi = 0.002, pvalue = 0.008, chr7q31\n",
      "Didn't pass (cutoff=0.0050): cat 18, mi = 0.002, pvalue = 0.026, chr13q14\n",
      "Didn't pass (cutoff=0.0050): cat 274, mi = 0.002, pvalue = 0.021, chr18q22\n",
      "Didn't pass (cutoff=0.0050): cat 168, mi = 0.002, pvalue = 0.012, chr7q33\n",
      "Didn't pass (cutoff=0.0050): cat 74, mi = 0.002, pvalue = 0.040, chr2p13\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.053, chr11q12\n",
      "Didn't pass (cutoff=0.0050): cat 222, mi = 0.002, pvalue = 0.024, chr5q14\n",
      "Didn't pass (cutoff=0.0050): cat 8, mi = 0.002, pvalue = 0.066, chr6q22\n",
      "Didn't pass (cutoff=0.0050): cat 116, mi = 0.002, pvalue = 0.059, chr1p13\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.002, pvalue = 0.080, chr12q21\n",
      "Didn't pass (cutoff=0.0050): cat 249, mi = 0.002, pvalue = 0.068, chr10p15\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.080, chr11q14\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.002, pvalue = 0.073, chr3q13\n",
      "Didn't pass (cutoff=0.0050): cat 183, mi = 0.002, pvalue = 0.083, chr9p13\n",
      "Didn't pass (cutoff=0.0050): cat 35, mi = 0.002, pvalue = 0.094, chr15q22\n",
      "Didn't pass (cutoff=0.0050): cat 135, mi = 0.002, pvalue = 0.092, chr20p11\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.002, pvalue = 0.080, chr6q14\n",
      "Total number of categories evaluated = 12.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-11.27\n",
      "chr2q36\n",
      "chr7p14\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.41587901701323\t42\t0\n",
      "6.16068052930057\t42\t25.1682419659735\n",
      "7.68241965973535\t42\t27.1531190926276\n",
      "8.58223062381852\t42\t28.2117202268431\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.3024574669187\t42\t30.0642722117202\n",
      "10.7920604914934\t42\t30.5406427221172\n",
      "11.3345935727788\t42\t31.0434782608696\n",
      "12.0359168241966\t42\t31.6257088846881\n",
      "17.2230623818526\t42\t32.4461247637051\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_c1/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5529\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6492 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6492\t5529\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "590 hits go_terms_num = 5529\n",
      "cat 4763, mi = 0.00383357, MORI_IMMATURE_B_LYMPHOCYTE_DN\n",
      "cat 800, mi = 0.00370631, BIOCARTA_ETS_PATHWAY\n",
      "cat 104, mi = 0.00369108, YU_MYC_TARGETS_DN\n",
      "cat 5116, mi = 0.00368473, KIM_ALL_DISORDERS_DURATION_CORR_DN\n",
      "cat 4350, mi = 0.00355916, GU_PDEF_TARGETS_UP\n",
      "cat 672, mi = 0.00354411, SA_B_CELL_RECEPTOR_COMPLEXES\n",
      "cat 1616, mi = 0.00352512, LEE_DIFFERENTIATING_T_LYMPHOCYTE\n",
      "cat 2739, mi = 0.00349991, SAMOLS_TARGETS_OF_KHSV_MIRNAS_DN\n",
      "cat 3761, mi = 0.00344879, JOHANSSON_GLIOMAGENESIS_BY_PDGFB_UP\n",
      "cat 2451, mi = 0.00337310, REACTOME_NONSENSE_MEDIATED_DECAY_NMD\n",
      "cat 61, mi = 0.00335274, KYNG_ENVIRONMENTAL_STRESS_RESPONSE_UP\n",
      "cat 58, mi = 0.00334894, SIG_BCR_SIGNALING_PATHWAY\n",
      "cat 4083, mi = 0.00328181, MORI_EMU_MYC_LYMPHOMA_BY_ONSET_TIME_UP\n",
      "cat 3444, mi = 0.00326754, WHITEFORD_PEDIATRIC_CANCER_MARKERS\n",
      "cat 681, mi = 0.00324670, PID_SHP2_PATHWAY\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 4763, mi = 0.004, pvalue = 0.000, MORI_IMMATURE_B_LYMPHOCYTE_DN\n",
      "Passed: cat 800, mi = 0.004, pvalue = 0.000, BIOCARTA_ETS_PATHWAY\n",
      "Passed: cat 104, mi = 0.004, pvalue = 0.000, YU_MYC_TARGETS_DN\n",
      "Passed: cat 5116, mi = 0.004, pvalue = 0.000, KIM_ALL_DISORDERS_DURATION_CORR_DN\n",
      "Passed: cat 4350, mi = 0.004, pvalue = 0.000, GU_PDEF_TARGETS_UP\n",
      "Passed: cat 2739, mi = 0.003, pvalue = 0.000, SAMOLS_TARGETS_OF_KHSV_MIRNAS_DN\n",
      "Passed: cat 61, mi = 0.003, pvalue = 0.000, KYNG_ENVIRONMENTAL_STRESS_RESPONSE_UP\n",
      "Passed: cat 58, mi = 0.003, pvalue = 0.002, SIG_BCR_SIGNALING_PATHWAY\n",
      "Passed: cat 4083, mi = 0.003, pvalue = 0.002, MORI_EMU_MYC_LYMPHOMA_BY_ONSET_TIME_UP\n",
      "Passed: cat 3566, mi = 0.003, pvalue = 0.002, RASHI_RESPONSE_TO_IONIZING_RADIATION_3\n",
      "Passed: cat 5234, mi = 0.003, pvalue = 0.004, BROWNE_HCMV_INFECTION_24HR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5049, mi = 0.003, pvalue = 0.006, JIANG_HYPOXIA_NORMAL\n",
      "Didn't pass (cutoff=0.0050): cat 1611, mi = 0.003, pvalue = 0.005, ZHAN_MULTIPLE_MYELOMA_CD1_VS_CD2_DN\n",
      "Passed: cat 3205, mi = 0.003, pvalue = 0.001, REACTOME_MITOCHONDRIAL_PROTEIN_IMPORT\n",
      "Didn't pass (cutoff=0.0050): cat 4213, mi = 0.003, pvalue = 0.009, QI_HYPOXIA\n",
      "Didn't pass (cutoff=0.0050): cat 1217, mi = 0.003, pvalue = 0.007, GRABARCZYK_BCL11B_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1211, mi = 0.003, pvalue = 0.006, LIU_TARGETS_OF_VMYB_VS_CMYB_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1084, mi = 0.003, pvalue = 0.005, KYNG_RESPONSE_TO_H2O2_VIA_ERCC6_DN\n",
      "Didn't pass (cutoff=0.0050): cat 117, mi = 0.003, pvalue = 0.009, MCBRYAN_PUBERTAL_BREAST_5_6WK_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1192, mi = 0.003, pvalue = 0.016, PURBEY_TARGETS_OF_CTBP1_AND_SATB1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3738, mi = 0.003, pvalue = 0.015, GOLDRATH_IMMUNE_MEMORY\n",
      "Didn't pass (cutoff=0.0050): cat 1771, mi = 0.003, pvalue = 0.018, KEGG_N_GLYCAN_BIOSYNTHESIS\n",
      "Didn't pass (cutoff=0.0050): cat 2294, mi = 0.003, pvalue = 0.018, REACTOME_DISEASES_OF_GLYCOSYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 2681, mi = 0.003, pvalue = 0.008, REACTOME_DISEASES_ASSOCIATED_WITH_N_GLYCOSYLATION_OF_PROTEINS\n",
      "Didn't pass (cutoff=0.0050): cat 1968, mi = 0.003, pvalue = 0.019, KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_HEPARAN_SULFATE\n",
      "Didn't pass (cutoff=0.0050): cat 915, mi = 0.003, pvalue = 0.005, BOYAULT_LIVER_CANCER_SUBCLASS_G5_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1090, mi = 0.003, pvalue = 0.017, REACTOME_CASPASE_MEDIATED_CLEAVAGE_OF_CYTOSKELETAL_PROTEINS\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.003, pvalue = 0.022, LEE_NEURAL_CREST_STEM_CELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1718, mi = 0.003, pvalue = 0.020, PID_BMP_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 4928, mi = 0.003, pvalue = 0.017, ZHENG_FOXP3_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1408, mi = 0.002, pvalue = 0.019, BOYLAN_MULTIPLE_MYELOMA_D_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2498, mi = 0.002, pvalue = 0.017, REACTOME_TRIF_MEDIATED_PROGRAMMED_CELL_DEATH\n",
      "Didn't pass (cutoff=0.0050): cat 4883, mi = 0.002, pvalue = 0.023, MCLACHLAN_DENTAL_CARIES_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1271, mi = 0.002, pvalue = 0.014, VANTVEER_BREAST_CANCER_POOR_PROGNOSIS\n",
      "Total number of categories evaluated = 81.\n",
      "Number of categories that passed the tests = 12\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 11, 0.198952668143169\n",
      "best merge is 0, 3, 0.314165477086407\n",
      "best merge is 4, 7, 0.363605195732212\n",
      "best merge is 2, 6, 0.381803009906751\n",
      "best merge is 2, 5, 0.509562056783423\n",
      "best merge is 4, 8, 0.769295983711358\n",
      "best merge is 4, 9, 0.841985312382822\n",
      "best merge is 1, 4, 0.87171233471243\n",
      "order classes\n",
      "3 - 0.735363219836428\n",
      "3 - 1.24526855419875\n",
      "4 - 1.66055741261377\n",
      "0 11 3\n",
      "1\n",
      "2 6 5\n",
      "4 7 8 9\n",
      "10\n",
      "Done.Start drawing\n",
      "4.60\t-11.27\n",
      "MORI_EMU_MYC_LYMPHOMA_BY_ONSET_TIME_UP\n",
      "BIOCARTA_ETS_PATHWAY\n",
      "RASHI_RESPONSE_TO_IONIZING_RADIATION_3\n",
      "GU_PDEF_TARGETS_UP\n",
      "SIG_BCR_SIGNALING_PATHWAY\n",
      "YU_MYC_TARGETS_DN\n",
      "KYNG_ENVIRONMENTAL_STRESS_RESPONSE_UP\n",
      "SAMOLS_TARGETS_OF_KHSV_MIRNAS_DN\n",
      "BROWNE_HCMV_INFECTION_24HR_UP\n",
      "MORI_IMMATURE_B_LYMPHOCYTE_DN\n",
      "REACTOME_MITOCHONDRIAL_PROTEIN_IMPORT\n",
      "KIM_ALL_DISORDERS_DURATION_CORR_DN\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 11, 0.198952668143169\n",
      "best merge is 0, 3, 0.314165477086407\n",
      "best merge is 4, 7, 0.363605195732212\n",
      "best merge is 2, 6, 0.381803009906751\n",
      "best merge is 2, 5, 0.509562056783423\n",
      "best merge is 4, 8, 0.769295983711358\n",
      "best merge is 4, 9, 0.841985312382822\n",
      "best merge is 1, 4, 0.87171233471243\n",
      "order classes\n",
      "3 - 0.735363219836428\n",
      "3 - 1.24526855419875\n",
      "4 - 1.66055741261377\n",
      "0 11 3\n",
      "1\n",
      "2 6 5\n",
      "4 7 8 9\n",
      "10\n",
      "Done.-7.40264650283554\t42\t0\n",
      "6.18714555765595\t42\t25.1947069943289\n",
      "7.69565217391305\t42\t27.179584120983\n",
      "8.58223062381852\t42\t28.2117202268431\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.3024574669187\t42\t30.0642722117202\n",
      "10.7920604914934\t42\t30.5406427221172\n",
      "11.3213610586011\t42\t31.0434782608696\n",
      "12.0094517958412\t42\t31.5992438563327\n",
      "17.2098298676749\t42\t32.4196597353497\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_c2/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 3733\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6535 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6535\t3733\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "594 hits go_terms_num = 3733\n",
      "cat 1671, mi = 0.00348971, MIR6740_5P\n",
      "cat 1455, mi = 0.00334830, MIR624_3P\n",
      "cat 919, mi = 0.00328663, MIR1288_3P\n",
      "cat 3684, mi = 0.00328226, MIR6880_5P\n",
      "cat 1345, mi = 0.00327885, MIR1226_3P\n",
      "cat 2885, mi = 0.00326642, MIR6819_5P\n",
      "cat 1679, mi = 0.00324717, MIR10524_5P\n",
      "cat 2403, mi = 0.00319945, ZNF33A_TARGET_GENES\n",
      "cat 3179, mi = 0.00311312, MIR640\n",
      "cat 985, mi = 0.00308297, MIR4695_3P\n",
      "cat 2229, mi = 0.00307421, MIR6812_5P\n",
      "cat 1138, mi = 0.00305676, MIR450B_3P\n",
      "cat 1556, mi = 0.00302924, MIR6718_5P\n",
      "cat 3294, mi = 0.00302924, MIR203B_5P\n",
      "cat 3107, mi = 0.00302824, MIR382_3P\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1671, mi = 0.003, pvalue = 0.001, MIR6740_5P\n",
      "Passed: cat 1455, mi = 0.003, pvalue = 0.001, MIR624_3P\n",
      "Passed: cat 919, mi = 0.003, pvalue = 0.001, MIR1288_3P\n",
      "Passed: cat 3684, mi = 0.003, pvalue = 0.002, MIR6880_5P\n",
      "Passed: cat 1345, mi = 0.003, pvalue = 0.002, MIR1226_3P\n",
      "Passed: cat 1679, mi = 0.003, pvalue = 0.002, MIR10524_5P\n",
      "Passed: cat 2403, mi = 0.003, pvalue = 0.002, ZNF33A_TARGET_GENES\n",
      "Passed: cat 1982, mi = 0.003, pvalue = 0.004, EVI1_03\n",
      "Passed: cat 392, mi = 0.003, pvalue = 0.005, MIR4736\n",
      "Passed: cat 471, mi = 0.003, pvalue = 0.005, MIR3943\n",
      "Didn't pass (cutoff=0.0050): cat 1729, mi = 0.003, pvalue = 0.006, E2F_Q2\n",
      "Didn't pass (cutoff=0.0050): cat 1498, mi = 0.003, pvalue = 0.005, MIR6800_5P\n",
      "Didn't pass (cutoff=0.0050): cat 412, mi = 0.003, pvalue = 0.009, MIR4518\n",
      "Didn't pass (cutoff=0.0050): cat 3574, mi = 0.003, pvalue = 0.012, MIR34A_3P\n",
      "Didn't pass (cutoff=0.0050): cat 1757, mi = 0.003, pvalue = 0.011, MIR217_3P\n",
      "Didn't pass (cutoff=0.0050): cat 3616, mi = 0.003, pvalue = 0.010, MSX1_01\n",
      "Didn't pass (cutoff=0.0050): cat 804, mi = 0.003, pvalue = 0.012, MIR4708_3P\n",
      "Didn't pass (cutoff=0.0050): cat 988, mi = 0.003, pvalue = 0.010, CTCTATG_MIR368\n",
      "Didn't pass (cutoff=0.0050): cat 3258, mi = 0.003, pvalue = 0.013, MIR6773_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1480, mi = 0.003, pvalue = 0.014, MIR339_5P\n",
      "Didn't pass (cutoff=0.0050): cat 242, mi = 0.003, pvalue = 0.011, COBLL1_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 1344, mi = 0.003, pvalue = 0.014, MIR1266_5P\n",
      "Didn't pass (cutoff=0.0050): cat 3718, mi = 0.003, pvalue = 0.018, MIR934\n",
      "Didn't pass (cutoff=0.0050): cat 1725, mi = 0.003, pvalue = 0.014, SGCGSSAAA_E2F1DP2_01\n",
      "Didn't pass (cutoff=0.0050): cat 2628, mi = 0.003, pvalue = 0.019, MIR216B_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2155, mi = 0.003, pvalue = 0.012, MIR4640_3P\n",
      "Didn't pass (cutoff=0.0050): cat 3597, mi = 0.003, pvalue = 0.016, MIR4802_5P\n",
      "Didn't pass (cutoff=0.0050): cat 3234, mi = 0.002, pvalue = 0.021, MIR200A_5P\n",
      "Passed: cat 298, mi = 0.002, pvalue = 0.002, ZNF26_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 3052, mi = 0.002, pvalue = 0.021, MIR1843\n",
      "Didn't pass (cutoff=0.0050): cat 1605, mi = 0.002, pvalue = 0.009, RBM17_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.002, pvalue = 0.016, MIR224_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2458, mi = 0.002, pvalue = 0.025, MIR4501\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.027, MIR6502_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1637, mi = 0.002, pvalue = 0.029, MIR1301_5P\n",
      "Didn't pass (cutoff=0.0050): cat 3603, mi = 0.002, pvalue = 0.019, MIR7114_5P\n",
      "Didn't pass (cutoff=0.0050): cat 691, mi = 0.002, pvalue = 0.023, MIR4724_5P\n",
      "Didn't pass (cutoff=0.0050): cat 309, mi = 0.002, pvalue = 0.020, MIR4278\n",
      "Didn't pass (cutoff=0.0050): cat 732, mi = 0.002, pvalue = 0.025, MIR769_5P\n",
      "Didn't pass (cutoff=0.0050): cat 64, mi = 0.002, pvalue = 0.030, MIR3670\n",
      "Didn't pass (cutoff=0.0050): cat 858, mi = 0.002, pvalue = 0.028, MCM2_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 269, mi = 0.002, pvalue = 0.033, MIR7160_3P\n",
      "Didn't pass (cutoff=0.0050): cat 910, mi = 0.002, pvalue = 0.030, PPARA_02\n",
      "Didn't pass (cutoff=0.0050): cat 2364, mi = 0.002, pvalue = 0.027, MIR500A_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1192, mi = 0.002, pvalue = 0.040, MIR4772_5P\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.002, pvalue = 0.033, MIR622\n",
      "Didn't pass (cutoff=0.0050): cat 2099, mi = 0.002, pvalue = 0.045, MIR328_3P\n",
      "Didn't pass (cutoff=0.0050): cat 1038, mi = 0.002, pvalue = 0.042, RYCACNNRNNRNCAG_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 1328, mi = 0.002, pvalue = 0.045, MIR10226\n",
      "Total number of categories evaluated = 96.\n",
      "Number of categories that passed the tests = 11\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 8, 0.352015892797631\n",
      "best merge is 0, 6, 0.363642596072153\n",
      "best merge is 4, 7, 0.558165322363063\n",
      "best merge is 2, 10, 0.671347135412535\n",
      "best merge is 3, 4, 0.726974397524966\n",
      "best merge is 1, 3, 0.838306614598942\n",
      "best merge is 5, 9, 0.920222821599345\n",
      "order classes\n",
      "2 - 1.00001713919842\n",
      "5 - 1.23243932917502\n",
      "2 - 1.84620462238447\n",
      "0 6\n",
      "1 3 4 8 7\n",
      "2 10\n",
      "5\n",
      "9\n",
      "Done.Start drawing\n",
      "4.60\t-11.27\n",
      "MIR624_3P\n",
      "MIR1226_3P\n",
      "MIR4736\n",
      "MIR3943\n",
      "MIR1288_3P\n",
      "ZNF26_TARGET_GENES\n",
      "MIR6740_5P\n",
      "ZNF33A_TARGET_GENES\n",
      "MIR10524_5P\n",
      "MIR6880_5P\n",
      "EVI1_03\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 8, 0.352015892797631\n",
      "best merge is 0, 6, 0.363642596072153\n",
      "best merge is 4, 7, 0.558165322363063\n",
      "best merge is 2, 10, 0.671347135412535\n",
      "best merge is 3, 4, 0.726974397524966\n",
      "best merge is 1, 3, 0.838306614598942\n",
      "best merge is 5, 9, 0.920222821599345\n",
      "order classes\n",
      "2 - 1.00001713919842\n",
      "5 - 1.23243932917502\n",
      "2 - 1.84620462238447\n",
      "0 6\n",
      "1 3 4 8 7\n",
      "2 10\n",
      "5\n",
      "9\n",
      "Done.-7.41587901701323\t42\t0\n",
      "6.16068052930057\t42\t25.1682419659735\n",
      "7.68241965973535\t42\t27.1531190926276\n",
      "8.58223062381852\t42\t28.2117202268431\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.3024574669187\t42\t30.0642722117202\n",
      "10.7920604914934\t42\t30.5406427221172\n",
      "11.3345935727788\t42\t31.0434782608696\n",
      "12.0359168241966\t42\t31.6257088846881\n",
      "17.2230623818526\t42\t32.4461247637051\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_c3/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3839 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3839\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "349 hits go_terms_num = 858\n",
      "cat 298, mi = 0.00724963, MORF_ANP32B\n",
      "cat 165, mi = 0.00596299, MORF_G22P1\n",
      "cat 633, mi = 0.00565283, MORF_UBE2N\n",
      "cat 139, mi = 0.00554966, GCM_PTPRD\n",
      "cat 363, mi = 0.00552990, GNF2_SELL\n",
      "cat 109, mi = 0.00542570, MORF_PPP1CA\n",
      "cat 230, mi = 0.00536388, GNF2_CENPF\n",
      "cat 459, mi = 0.00533348, MODULE_397\n",
      "cat 443, mi = 0.00500721, GNF2_CDC2\n",
      "cat 142, mi = 0.00480989, MORF_TERF1\n",
      "cat 576, mi = 0.00480911, GNF2_PTPN6\n",
      "cat 164, mi = 0.00467442, MORF_BUB1B\n",
      "cat 625, mi = 0.00459596, GNF2_FBL\n",
      "cat 615, mi = 0.00456312, MODULE_285\n",
      "cat 514, mi = 0.00451584, GNF2_ICAM3\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 298, mi = 0.007, pvalue = 0.000, MORF_ANP32B\n",
      "Passed: cat 139, mi = 0.006, pvalue = 0.002, GCM_PTPRD\n",
      "Passed: cat 363, mi = 0.006, pvalue = 0.002, GNF2_SELL\n",
      "Passed: cat 230, mi = 0.005, pvalue = 0.002, GNF2_CENPF\n",
      "Didn't pass (cutoff=0.0050): cat 625, mi = 0.005, pvalue = 0.011, GNF2_FBL\n",
      "Didn't pass (cutoff=0.0050): cat 615, mi = 0.005, pvalue = 0.012, MODULE_285\n",
      "Didn't pass (cutoff=0.0050): cat 417, mi = 0.004, pvalue = 0.016, GNF2_ITGAL\n",
      "Didn't pass (cutoff=0.0050): cat 89, mi = 0.004, pvalue = 0.025, MORF_PPP6C\n",
      "Didn't pass (cutoff=0.0050): cat 528, mi = 0.004, pvalue = 0.016, MODULE_119\n",
      "Didn't pass (cutoff=0.0050): cat 51, mi = 0.004, pvalue = 0.025, GNF2_CASP8\n",
      "Didn't pass (cutoff=0.0050): cat 542, mi = 0.004, pvalue = 0.027, GCM_RING1\n",
      "Passed: cat 87, mi = 0.004, pvalue = 0.002, MODULE_318\n",
      "Passed: cat 238, mi = 0.004, pvalue = 0.003, MODULE_107\n",
      "Didn't pass (cutoff=0.0050): cat 277, mi = 0.004, pvalue = 0.024, MORF_RUNX1\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.004, pvalue = 0.038, MORF_ETV3\n",
      "Didn't pass (cutoff=0.0050): cat 161, mi = 0.004, pvalue = 0.028, MODULE_261\n",
      "Didn't pass (cutoff=0.0050): cat 837, mi = 0.004, pvalue = 0.008, MODULE_509\n",
      "Didn't pass (cutoff=0.0050): cat 490, mi = 0.004, pvalue = 0.036, MODULE_36\n",
      "Didn't pass (cutoff=0.0050): cat 379, mi = 0.004, pvalue = 0.020, MODULE_157\n",
      "Didn't pass (cutoff=0.0050): cat 424, mi = 0.004, pvalue = 0.051, GNF2_SNRK\n",
      "Didn't pass (cutoff=0.0050): cat 319, mi = 0.004, pvalue = 0.059, MODULE_180\n",
      "Didn't pass (cutoff=0.0050): cat 548, mi = 0.004, pvalue = 0.056, MODULE_544\n",
      "Didn't pass (cutoff=0.0050): cat 487, mi = 0.004, pvalue = 0.066, MODULE_537\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.004, pvalue = 0.056, GNF2_ATM\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.004, pvalue = 0.033, MODULE_85\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.004, pvalue = 0.061, GCM_IL6ST\n",
      "Passed: cat 822, mi = 0.004, pvalue = 0.004, MODULE_432\n",
      "Didn't pass (cutoff=0.0050): cat 588, mi = 0.004, pvalue = 0.023, MODULE_196\n",
      "Didn't pass (cutoff=0.0050): cat 760, mi = 0.004, pvalue = 0.023, MODULE_253\n",
      "Didn't pass (cutoff=0.0050): cat 86, mi = 0.004, pvalue = 0.068, MORF_RAD54L\n",
      "Didn't pass (cutoff=0.0050): cat 386, mi = 0.004, pvalue = 0.053, MODULE_70\n",
      "Didn't pass (cutoff=0.0050): cat 127, mi = 0.004, pvalue = 0.085, GCM_SMARCD1\n",
      "Didn't pass (cutoff=0.0050): cat 559, mi = 0.004, pvalue = 0.078, MODULE_248\n",
      "Didn't pass (cutoff=0.0050): cat 286, mi = 0.003, pvalue = 0.066, MODULE_145\n",
      "Didn't pass (cutoff=0.0050): cat 123, mi = 0.003, pvalue = 0.090, GNF2_JAK1\n",
      "Didn't pass (cutoff=0.0050): cat 221, mi = 0.003, pvalue = 0.065, MORF_ESR1\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.003, pvalue = 0.108, GCM_SMARCC1\n",
      "Didn't pass (cutoff=0.0050): cat 375, mi = 0.003, pvalue = 0.083, MORF_RAF1\n",
      "Didn't pass (cutoff=0.0050): cat 746, mi = 0.003, pvalue = 0.029, GCM_SUPT4H1\n",
      "Didn't pass (cutoff=0.0050): cat 630, mi = 0.003, pvalue = 0.120, GNF2_PTPN4\n",
      "Didn't pass (cutoff=0.0050): cat 761, mi = 0.003, pvalue = 0.142, MORF_LMO1\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.003, pvalue = 0.084, MORF_NF1\n",
      "Didn't pass (cutoff=0.0050): cat 812, mi = 0.003, pvalue = 0.145, MODULE_300\n",
      "Didn't pass (cutoff=0.0050): cat 228, mi = 0.003, pvalue = 0.144, GCM_CASP2\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.003, pvalue = 0.145, MODULE_86\n",
      "Didn't pass (cutoff=0.0050): cat 503, mi = 0.003, pvalue = 0.045, GCM_ATM\n",
      "Didn't pass (cutoff=0.0050): cat 390, mi = 0.003, pvalue = 0.154, MODULE_481\n",
      "Total number of categories evaluated = 90.\n",
      "Number of categories that passed the tests = 7\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 5, 0.317175127790436\n",
      "best merge is 1, 3, 0.367792213688021\n",
      "best merge is 0, 2, 0.541721347665655\n",
      "order classes\n",
      "2 - 0.643636373954037\n",
      "2 - 0.555056473633262\n",
      "4 5\n",
      "1 3\n",
      "2\n",
      "0\n",
      "6\n",
      "Done.Start drawing\n",
      "4.20\t-11.27\n",
      "GNF2_SELL\n",
      "GCM_PTPRD\n",
      "MODULE_318\n",
      "MODULE_107\n",
      "MORF_ANP32B\n",
      "GNF2_CENPF\n",
      "MODULE_432\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 5, 0.317175127790436\n",
      "best merge is 1, 3, 0.367792213688021\n",
      "best merge is 0, 2, 0.541721347665655\n",
      "order classes\n",
      "2 - 0.643636373954037\n",
      "2 - 0.555056473633262\n",
      "4 5\n",
      "1 3\n",
      "2\n",
      "0\n",
      "6\n",
      "Done.-6.99547511312217\t42\t0\n",
      "7.02714932126697\t42\t26.0090497737557\n",
      "8.5475113122172\t42\t28.0452488687783\n",
      "9.41628959276018\t42\t29.0497737556561\n",
      "10.0678733031674\t42\t29.7828054298643\n",
      "10.6108597285068\t42\t30.3529411764706\n",
      "11.0859728506787\t42\t30.868778280543\n",
      "11.5610859728507\t42\t31.3031674208145\n",
      "12.0904977375566\t42\t31.8190045248869\n",
      "12.7692307692308\t42\t32.3619909502262\n",
      "17.5882352941177\t42\t33.1764705882353\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_c4/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 10191\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 5908 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 5908\t10191\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "537 hits go_terms_num = 10191\n",
      "cat 6572, mi = 0.00420439, GO_BONE_MORPHOGENESIS\n",
      "cat 296, mi = 0.00409322, GO_REGULATION_OF_BMP_SIGNALING_PATHWAY\n",
      "cat 1318, mi = 0.00396201, GO_ACETYLGLUCOSAMINYLTRANSFERASE_ACTIVITY\n",
      "cat 3386, mi = 0.00379072, GO_REGULATION_OF_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS\n",
      "cat 5176, mi = 0.00370968, GO_SPROUTING_ANGIOGENESIS\n",
      "cat 1667, mi = 0.00365566, GO_DEFENSE_RESPONSE_TO_GRAM_POSITIVE_BACTERIUM\n",
      "cat 4427, mi = 0.00360946, GO_SIGNAL_TRANSDUCTION_IN_RESPONSE_TO_DNA_DAMAGE\n",
      "cat 578, mi = 0.00359935, GO_REGULATION_OF_B_CELL_DIFFERENTIATION\n",
      "cat 4514, mi = 0.00353335, GO_CELLULAR_RESPIRATION\n",
      "cat 5096, mi = 0.00352052, GO_POSITIVE_REGULATION_OF_ACTIN_FILAMENT_BUNDLE_ASSEMBLY\n",
      "cat 5197, mi = 0.00349384, GO_INOSITOL_PHOSPHATE_MEDIATED_SIGNALING\n",
      "cat 676, mi = 0.00348349, GO_CALCINEURIN_MEDIATED_SIGNALING\n",
      "cat 7731, mi = 0.00348349, GO_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_ACTIVATION\n",
      "cat 7963, mi = 0.00340805, GO_ENDOCHONDRAL_BONE_MORPHOGENESIS\n",
      "cat 5221, mi = 0.00337837, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_PLASMA_MEMBRANE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 6572, mi = 0.004, pvalue = 0.000, GO_BONE_MORPHOGENESIS\n",
      "Passed: cat 1318, mi = 0.004, pvalue = 0.001, GO_ACETYLGLUCOSAMINYLTRANSFERASE_ACTIVITY\n",
      "Passed: cat 3386, mi = 0.004, pvalue = 0.001, GO_REGULATION_OF_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS\n",
      "Passed: cat 5176, mi = 0.004, pvalue = 0.002, GO_SPROUTING_ANGIOGENESIS\n",
      "Passed: cat 1667, mi = 0.004, pvalue = 0.001, GO_DEFENSE_RESPONSE_TO_GRAM_POSITIVE_BACTERIUM\n",
      "Passed: cat 4427, mi = 0.004, pvalue = 0.002, GO_SIGNAL_TRANSDUCTION_IN_RESPONSE_TO_DNA_DAMAGE\n",
      "Passed: cat 578, mi = 0.004, pvalue = 0.000, GO_REGULATION_OF_B_CELL_DIFFERENTIATION\n",
      "Passed: cat 4514, mi = 0.004, pvalue = 0.002, GO_CELLULAR_RESPIRATION\n",
      "Passed: cat 5197, mi = 0.003, pvalue = 0.003, GO_INOSITOL_PHOSPHATE_MEDIATED_SIGNALING\n",
      "Passed: cat 7731, mi = 0.003, pvalue = 0.000, GO_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_ACTIVATION\n",
      "Passed: cat 5221, mi = 0.003, pvalue = 0.003, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_PLASMA_MEMBRANE\n",
      "Passed: cat 9880, mi = 0.003, pvalue = 0.004, GO_MAMMARY_GLAND_DEVELOPMENT\n",
      "Passed: cat 6832, mi = 0.003, pvalue = 0.003, GO_TELOMERIC_DNA_BINDING\n",
      "Passed: cat 2279, mi = 0.003, pvalue = 0.005, GO_PYRIMIDINE_CONTAINING_COMPOUND_CATABOLIC_PROCESS\n",
      "Passed: cat 2361, mi = 0.003, pvalue = 0.000, GO_INSULIN_RECEPTOR_COMPLEX\n",
      "Passed: cat 6340, mi = 0.003, pvalue = 0.003, GO_CYTOSOLIC_RIBOSOME\n",
      "Passed: cat 10103, mi = 0.003, pvalue = 0.003, GO_POLYOL_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 9063, mi = 0.003, pvalue = 0.015, GO_CHROMOSOME_SEPARATION\n",
      "Passed: cat 6827, mi = 0.003, pvalue = 0.001, GO_NEGATIVE_REGULATION_OF_MACROPHAGE_ACTIVATION\n",
      "Didn't pass (cutoff=0.0050): cat 8133, mi = 0.003, pvalue = 0.015, GO_HISTONE_UBIQUITINATION\n",
      "Didn't pass (cutoff=0.0050): cat 6062, mi = 0.003, pvalue = 0.013, GO_HISTONE_DEACETYLASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 8563, mi = 0.003, pvalue = 0.017, GO_POLYOL_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3007, mi = 0.003, pvalue = 0.008, GO_INOSITOL_TRISPHOSPHATE_METABOLIC_PROCESS\n",
      "Passed: cat 7707, mi = 0.003, pvalue = 0.003, GO_NEGATIVE_REGULATION_OF_MULTICELLULAR_ORGANISM_GROWTH\n",
      "Didn't pass (cutoff=0.0050): cat 5370, mi = 0.003, pvalue = 0.021, GO_ANION_CHANNEL_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 8640, mi = 0.003, pvalue = 0.020, GO_INOSITOL_PHOSPHATE_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1463, mi = 0.003, pvalue = 0.018, GO_HYDROLASE_ACTIVITY_ACTING_ON_ACID_PHOSPHORUS_NITROGEN_BONDS\n",
      "Didn't pass (cutoff=0.0050): cat 4007, mi = 0.003, pvalue = 0.017, GO_METALLOEXOPEPTIDASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 7658, mi = 0.003, pvalue = 0.016, GO_UNSATURATED_FATTY_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 8451, mi = 0.003, pvalue = 0.019, GO_METHYLTRANSFERASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 7676, mi = 0.003, pvalue = 0.028, GO_IMMUNE_RECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5460, mi = 0.003, pvalue = 0.023, GO_REGULATION_OF_CHROMOSOME_SEPARATION\n",
      "Didn't pass (cutoff=0.0050): cat 1563, mi = 0.003, pvalue = 0.028, GO_MICROTUBULE_ORGANIZING_CENTER_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 7919, mi = 0.003, pvalue = 0.034, GO_CYTOKINE_RECEPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 9708, mi = 0.003, pvalue = 0.006, GO_3_5_CYCLIC_AMP_PHOSPHODIESTERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 7184, mi = 0.003, pvalue = 0.033, GO_HISTONE_MONOUBIQUITINATION\n",
      "Didn't pass (cutoff=0.0050): cat 65, mi = 0.003, pvalue = 0.044, GO_MITOCHONDRIAL_TRANSLATIONAL_TERMINATION\n",
      "Didn't pass (cutoff=0.0050): cat 5906, mi = 0.003, pvalue = 0.026, GO_VOLUME_SENSITIVE_ANION_CHANNEL_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3396, mi = 0.003, pvalue = 0.018, GO_NEGATIVE_REGULATION_OF_TYROSINE_PHOSPHORYLATION_OF_STAT_PROTEIN\n",
      "Didn't pass (cutoff=0.0050): cat 9101, mi = 0.002, pvalue = 0.036, GO_RESPONSE_TO_MISFOLDED_PROTEIN\n",
      "Didn't pass (cutoff=0.0050): cat 5441, mi = 0.002, pvalue = 0.034, GO_HISTONE_METHYLTRANSFERASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 4416, mi = 0.002, pvalue = 0.030, GO_STRUCTURAL_MOLECULE_ACTIVITY_CONFERRING_ELASTICITY\n",
      "Didn't pass (cutoff=0.0050): cat 8900, mi = 0.002, pvalue = 0.048, GO_ACTION_POTENTIAL\n",
      "Didn't pass (cutoff=0.0050): cat 7825, mi = 0.002, pvalue = 0.015, GO_LIGASE_ACTIVITY_FORMING_CARBON_SULFUR_BONDS\n",
      "Total number of categories evaluated = 210.\n",
      "Number of categories that passed the tests = 19\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 15, 17, 0.267216600150242\n",
      "best merge is 4, 14, 0.290322438759258\n",
      "best merge is 2, 5, 0.368771753787509\n",
      "best merge is 6, 13, 0.374504250404728\n",
      "best merge is 8, 10, 0.406610998671779\n",
      "best merge is 11, 12, 0.424491843013813\n",
      "best merge is 3, 7, 0.533831179114456\n",
      "best merge is 1, 9, 0.540979005585518\n",
      "best merge is 3, 8, 0.565390222520117\n",
      "best merge is 1, 2, 0.593683484511002\n",
      "best merge is 11, 18, 0.733370402805165\n",
      "best merge is 0, 6, 0.744458134514339\n",
      "best merge is 3, 11, 0.834283833684214\n",
      "best merge is 4, 16, 0.867688486826604\n",
      "best merge is 0, 3, 1.03444036648268\n",
      "order classes\n",
      "3 - 2.62259184216554\n",
      "4 - 1.95016278909136\n",
      "7 - 1.67340054048587\n",
      "3 - 2.85098435821014\n",
      "2 - 1.26927885071365\n",
      "15 17\n",
      "1 9 2 5\n",
      "3 7 8 10 11 12 18\n",
      "4 14 16\n",
      "0 6 13\n",
      "Done.Start drawing\n",
      "4.60\t-11.27\n",
      "GO_DEFENSE_RESPONSE_TO_GRAM_POSITIVE_BACTERIUM\n",
      "GO_INSULIN_RECEPTOR_COMPLEX\n",
      "GO_NEGATIVE_REGULATION_OF_MULTICELLULAR_ORGANISM_GROWTH\n",
      "GO_POSITIVE_REGULATION_OF_NATURAL_KILLER_CELL_ACTIVATION\n",
      "GO_SPROUTING_ANGIOGENESIS\n",
      "GO_CELLULAR_RESPIRATION\n",
      "GO_REGULATION_OF_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS\n",
      "GO_INOSITOL_PHOSPHATE_MEDIATED_SIGNALING\n",
      "GO_MAMMARY_GLAND_DEVELOPMENT\n",
      "GO_TELOMERIC_DNA_BINDING\n",
      "GO_BONE_MORPHOGENESIS\n",
      "GO_REGULATION_OF_B_CELL_DIFFERENTIATION\n",
      "GO_PYRIMIDINE_CONTAINING_COMPOUND_CATABOLIC_PROCESS\n",
      "GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_PLASMA_MEMBRANE\n",
      "GO_CYTOSOLIC_RIBOSOME\n",
      "GO_NEGATIVE_REGULATION_OF_MACROPHAGE_ACTIVATION\n",
      "GO_SIGNAL_TRANSDUCTION_IN_RESPONSE_TO_DNA_DAMAGE\n",
      "GO_POLYOL_CATABOLIC_PROCESS\n",
      "GO_ACETYLGLUCOSAMINYLTRANSFERASE_ACTIVITY\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 15, 17, 0.267216600150242\n",
      "best merge is 4, 14, 0.290322438759258\n",
      "best merge is 2, 5, 0.368771753787509\n",
      "best merge is 6, 13, 0.374504250404728\n",
      "best merge is 8, 10, 0.406610998671779\n",
      "best merge is 11, 12, 0.424491843013813\n",
      "best merge is 3, 7, 0.533831179114456\n",
      "best merge is 1, 9, 0.540979005585518\n",
      "best merge is 3, 8, 0.565390222520117\n",
      "best merge is 1, 2, 0.593683484511002\n",
      "best merge is 11, 18, 0.733370402805165\n",
      "best merge is 0, 6, 0.744458134514339\n",
      "best merge is 3, 11, 0.834283833684214\n",
      "best merge is 4, 16, 0.867688486826604\n",
      "best merge is 0, 3, 1.03444036648268\n",
      "order classes\n",
      "3 - 2.62259184216554\n",
      "4 - 1.95016278909136\n",
      "7 - 1.67340054048587\n",
      "3 - 2.85098435821014\n",
      "2 - 1.26927885071365\n",
      "15 17\n",
      "1 9 2 5\n",
      "3 7 8 10 11 12 18\n",
      "4 14 16\n",
      "0 6 13\n",
      "Done.-7.38941398865784\t42\t0\n",
      "6.18714555765596\t42\t25.2211720226843\n",
      "7.68241965973535\t42\t27.1531190926276\n",
      "8.58223062381852\t42\t28.2117202268431\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.289224952741\t42\t30.0642722117202\n",
      "10.765595463138\t42\t30.5141776937618\n",
      "11.3081285444234\t42\t31.0170132325142\n",
      "11.9962192816635\t42\t31.5992438563327\n",
      "17.1965973534972\t42\t32.3931947069943\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_c5/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3784 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3784\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "344 hits go_terms_num = 189\n",
      "cat 153, mi = 0.00569095, VEGF_A_UP.V1_DN\n",
      "cat 152, mi = 0.00419147, MYC_UP.V1_UP\n",
      "cat 54, mi = 0.00414247, GCNP_SHH_UP_EARLY.V1_DN\n",
      "cat 71, mi = 0.00411164, RB_P107_DN.V1_DN\n",
      "cat 158, mi = 0.00401592, GCNP_SHH_UP_LATE.V1_DN\n",
      "cat 59, mi = 0.00396277, ATM_DN.V1_DN\n",
      "cat 114, mi = 0.00395478, ESC_J1_UP_EARLY.V1_UP\n",
      "cat 166, mi = 0.00393544, IL21_UP.V1_DN\n",
      "cat 68, mi = 0.00387530, MTOR_UP.N4.V1_DN\n",
      "cat 15, mi = 0.00383360, P53_DN.V1_DN\n",
      "cat 32, mi = 0.00380953, P53_DN.V1_UP\n",
      "cat 36, mi = 0.00367290, RAF_UP.V1_DN\n",
      "cat 112, mi = 0.00364048, JNK_DN.V1_DN\n",
      "cat 42, mi = 0.00363590, VEGF_A_UP.V1_UP\n",
      "cat 177, mi = 0.00351169, MEK_UP.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 153, mi = 0.006, pvalue = 0.002, VEGF_A_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 152, mi = 0.004, pvalue = 0.025, MYC_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.004, pvalue = 0.024, GCNP_SHH_UP_EARLY.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 71, mi = 0.004, pvalue = 0.028, RB_P107_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.004, pvalue = 0.024, GCNP_SHH_UP_LATE.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 59, mi = 0.004, pvalue = 0.041, ATM_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.004, pvalue = 0.038, ESC_J1_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 166, mi = 0.004, pvalue = 0.043, IL21_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.004, pvalue = 0.031, MTOR_UP.N4.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.004, pvalue = 0.043, P53_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.004, pvalue = 0.052, P53_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.004, pvalue = 0.042, RAF_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.004, pvalue = 0.055, JNK_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.004, pvalue = 0.049, VEGF_A_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 177, mi = 0.004, pvalue = 0.062, MEK_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 91, mi = 0.003, pvalue = 0.074, BMI1_DN_MEL18_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 89, mi = 0.003, pvalue = 0.087, BCAT_GDS748_DN\n",
      "Didn't pass (cutoff=0.0050): cat 117, mi = 0.003, pvalue = 0.109, RELA_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 124, mi = 0.003, pvalue = 0.067, IL2_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.003, pvalue = 0.083, MTOR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 172, mi = 0.003, pvalue = 0.118, KRAS.PROSTATE_UP.V1_DN\n",
      "Total number of categories evaluated = 11.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-11.27\n",
      "VEGF_A_UP.V1_DN\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.37618147448015\t42\t0\n",
      "6.21361058601134\t42\t25.2476370510397\n",
      "7.70888468809074\t42\t27.179584120983\n",
      "8.6351606805293\t42\t28.2381852551985\n",
      "9.31001890359168\t42\t29.0321361058601\n",
      "9.85255198487713\t42\t29.5879017013233\n",
      "10.3421550094518\t42\t30.117202268431\n",
      "10.8185255198488\t42\t30.5671077504726\n",
      "11.3478260869565\t42\t31.069943289225\n",
      "12.062381852552\t42\t31.6257088846881\n",
      "17.2495274102079\t42\t32.4990548204159\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_c6/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6489 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6489\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "590 hits go_terms_num = 4872\n",
      "cat 368, mi = 0.00440951, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "cat 1968, mi = 0.00409162, GSE19888_CTRL_VS_A3R_INHIBITOR_TREATED_MAST_CELL_UP\n",
      "cat 3474, mi = 0.00395716, GSE17974_1.5H_VS_72H_IL4_AND_ANTI_IL12_ACT_CD4_TCELL_UP\n",
      "cat 1344, mi = 0.00377523, GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_DN\n",
      "cat 3213, mi = 0.00374639, GSE22886_DC_VS_MONOCYTE_UP\n",
      "cat 2924, mi = 0.00372846, GSE39556_UNTREATED_VS_3H_POLYIC_INJ_MOUSE_NK_CELL_DN\n",
      "cat 3470, mi = 0.00365830, GSE15930_NAIVE_VS_24H_IN_VITRO_STIM_CD8_TCELL_DN\n",
      "cat 1543, mi = 0.00364463, GSE7218_UNSTIM_VS_ANTIGEN_STIM_THROUGH_IGG_BCELL_DN\n",
      "cat 3390, mi = 0.00364004, GSE34006_A2AR_KO_VS_A2AR_AGONIST_TREATED_TREG_DN\n",
      "cat 1866, mi = 0.00356791, GSE28726_ACT_CD4_TCELL_VS_ACT_VA24NEG_NKTCELL_DN\n",
      "cat 3744, mi = 0.00355898, GSE15930_NAIVE_VS_24H_IN_VITRO_STIM_INFAB_CD8_TCELL_DN\n",
      "cat 4555, mi = 0.00355837, GSE13547_2H_VS_12_H_ANTI_IGM_STIM_BCELL_DN\n",
      "cat 4830, mi = 0.00355559, GSE28726_ACT_CD4_TCELL_VS_ACT_NKTCELL_DN\n",
      "cat 1648, mi = 0.00354629, GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP\n",
      "cat 4044, mi = 0.00353978, GSE3039_ALPHAALPHA_CD8_TCELL_VS_B2_BCELL_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 368, mi = 0.004, pvalue = 0.000, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "Passed: cat 1968, mi = 0.004, pvalue = 0.000, GSE19888_CTRL_VS_A3R_INHIBITOR_TREATED_MAST_CELL_UP\n",
      "Passed: cat 3474, mi = 0.004, pvalue = 0.000, GSE17974_1.5H_VS_72H_IL4_AND_ANTI_IL12_ACT_CD4_TCELL_UP\n",
      "Passed: cat 1344, mi = 0.004, pvalue = 0.000, GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_DN\n",
      "Passed: cat 3213, mi = 0.004, pvalue = 0.001, GSE22886_DC_VS_MONOCYTE_UP\n",
      "Passed: cat 3470, mi = 0.004, pvalue = 0.001, GSE15930_NAIVE_VS_24H_IN_VITRO_STIM_CD8_TCELL_DN\n",
      "Passed: cat 1543, mi = 0.004, pvalue = 0.000, GSE7218_UNSTIM_VS_ANTIGEN_STIM_THROUGH_IGG_BCELL_DN\n",
      "Passed: cat 3390, mi = 0.004, pvalue = 0.001, GSE34006_A2AR_KO_VS_A2AR_AGONIST_TREATED_TREG_DN\n",
      "Passed: cat 1648, mi = 0.004, pvalue = 0.000, GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP\n",
      "Passed: cat 357, mi = 0.004, pvalue = 0.001, GSE3982_MAC_VS_TH2_UP\n",
      "Passed: cat 2094, mi = 0.003, pvalue = 0.002, GSE22935_24H_VS_48H_MBOVIS_BCG_STIM_MYD88_KO_MACROPHAGE_DN\n",
      "Passed: cat 463, mi = 0.003, pvalue = 0.002, GSE3720_UNSTIM_VS_PMA_STIM_VD1_GAMMADELTA_TCELL_UP\n",
      "Passed: cat 1034, mi = 0.003, pvalue = 0.004, GSE14350_TREG_VS_TEFF_DN\n",
      "Passed: cat 1635, mi = 0.003, pvalue = 0.003, GSE40274_IRF4_VS_FOXP3_AND_IRF4_TRANSDUCED_ACTIVATED_CD4_TCELL_UP\n",
      "Passed: cat 2173, mi = 0.003, pvalue = 0.002, GSE45365_HEALTHY_VS_MCMV_INFECTION_BCELL_IFNAR_KO_UP\n",
      "Passed: cat 4087, mi = 0.003, pvalue = 0.003, GSE23321_CD8_STEM_CELL_MEMORY_VS_EFFECTOR_MEMORY_CD8_TCELL_DN\n",
      "Passed: cat 874, mi = 0.003, pvalue = 0.004, GSE20715_0H_VS_48H_OZONE_LUNG_DN\n",
      "Passed: cat 1782, mi = 0.003, pvalue = 0.003, GSE2770_UNTREATED_VS_IL12_TREATED_ACT_CD4_TCELL_48H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3711, mi = 0.003, pvalue = 0.006, GSE29164_CD8_TCELL_VS_CD8_TCELL_AND_IL12_TREATED_MELANOMA_DAY7_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3377, mi = 0.003, pvalue = 0.006, GSE34156_UNTREATED_VS_24H_NOD2_AND_TLR1_TLR2_LIGAND_TREATED_MONOCYTE_DN\n",
      "Passed: cat 1797, mi = 0.003, pvalue = 0.005, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "Passed: cat 4688, mi = 0.003, pvalue = 0.005, GSE3982_EOSINOPHIL_VS_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3242, mi = 0.003, pvalue = 0.006, GSE29618_PDC_VS_MDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1597, mi = 0.003, pvalue = 0.006, GSE7460_CTRL_VS_TGFB_TREATED_ACT_TREG_DN\n",
      "Didn't pass (cutoff=0.0050): cat 513, mi = 0.003, pvalue = 0.007, GSE18893_CTRL_VS_TNF_TREATED_TCONV_24H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2502, mi = 0.003, pvalue = 0.009, GSE5589_WT_VS_IL10_KO_LPS_STIM_MACROPHAGE_180MIN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.003, pvalue = 0.009, GSE22025_UNTREATED_VS_TGFB1_TREATED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3606, mi = 0.003, pvalue = 0.006, GSE14000_4H_VS_16H_LPS_DC_TRANSLATED_RNA_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1155, mi = 0.003, pvalue = 0.007, GSE25087_TREG_VS_TCONV_FETUS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4824, mi = 0.003, pvalue = 0.010, GSE17721_CTRL_VS_LPS_6H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4595, mi = 0.003, pvalue = 0.010, GSE8835_HEALTHY_VS_CLL_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 719, mi = 0.003, pvalue = 0.009, GSE26343_WT_VS_NFAT5_KO_MACROPHAGE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1655, mi = 0.003, pvalue = 0.013, GSE13411_NAIVE_VS_MEMORY_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2433, mi = 0.003, pvalue = 0.014, GSE13762_CTRL_VS_125_VITAMIND_DAY5_DC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4507, mi = 0.003, pvalue = 0.014, GSE27786_LIN_NEG_VS_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 713, mi = 0.003, pvalue = 0.019, GSE32423_MEMORY_VS_NAIVE_CD8_TCELL_IL7_IL4_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1581, mi = 0.003, pvalue = 0.014, GSE43863_NAIVE_VS_LY6C_LOW_CXCR5NEG_CD4_EFF_TCELL_D6_LCMV_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4506, mi = 0.003, pvalue = 0.015, GSE37301_RAG2_KO_VS_RAG2_AND_ETS1_KO_NK_CELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2273, mi = 0.003, pvalue = 0.014, GSE8685_IL15_ACT_IL2_STARVED_VS_IL21_ACT_IL2_STARVED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2377, mi = 0.003, pvalue = 0.015, GSE17721_LPS_VS_CPG_24H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 769, mi = 0.003, pvalue = 0.013, GSE32533_WT_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4560, mi = 0.003, pvalue = 0.024, GSE16385_UNTREATED_VS_12H_IFNG_TNF_TREATED_MACROPHAGE_UP\n",
      "Total number of categories evaluated = 150.\n",
      "Number of categories that passed the tests = 20\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 11, 15, 0.205294886419603\n",
      "best merge is 9, 17, 0.312059920964693\n",
      "best merge is 5, 10, 0.325433537772234\n",
      "best merge is 0, 6, 0.356023460472013\n",
      "best merge is 8, 16, 0.414115788538816\n",
      "best merge is 2, 13, 0.46043352238664\n",
      "best merge is 1, 4, 0.48454508054128\n",
      "best merge is 5, 14, 0.571908057769858\n",
      "best merge is 2, 11, 0.634855194381008\n",
      "best merge is 1, 19, 0.646827642308487\n",
      "best merge is 8, 18, 0.659259306919812\n",
      "best merge is 3, 8, 0.68514031403944\n",
      "best merge is 2, 7, 0.788751377916219\n",
      "best merge is 5, 9, 0.79423046348301\n",
      "best merge is 1, 2, 0.887675418716669\n",
      "best merge is 0, 3, 0.951023802189351\n",
      "order classes\n",
      "2 - 1.78011730236006\n",
      "8 - 1.67605362061746\n",
      "4 - 2.36753459031047\n",
      "5 - 2.09494155365591\n",
      "1 4 19 2 13 11 15 7\n",
      "0 6\n",
      "3 8 16 18\n",
      "5 10 14 9 17\n",
      "12\n",
      "Done.Start drawing\n",
      "4.60\t-11.27\n",
      "GSE3982_MAC_VS_TH2_UP\n",
      "GSE14350_TREG_VS_TEFF_DN\n",
      "GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP\n",
      "GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "GSE3982_EOSINOPHIL_VS_BCELL_DN\n",
      "GSE7218_UNSTIM_VS_ANTIGEN_STIM_THROUGH_IGG_BCELL_DN\n",
      "GSE3720_UNSTIM_VS_PMA_STIM_VD1_GAMMADELTA_TCELL_UP\n",
      "GSE34006_A2AR_KO_VS_A2AR_AGONIST_TREATED_TREG_DN\n",
      "GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "GSE23321_CD8_STEM_CELL_MEMORY_VS_EFFECTOR_MEMORY_CD8_TCELL_DN\n",
      "GOLDRATH_EFF_VS_MEMORY_CD8_TCELL_DN\n",
      "GSE17974_1.5H_VS_72H_IL4_AND_ANTI_IL12_ACT_CD4_TCELL_UP\n",
      "GSE40274_IRF4_VS_FOXP3_AND_IRF4_TRANSDUCED_ACTIVATED_CD4_TCELL_UP\n",
      "GSE2770_UNTREATED_VS_IL12_TREATED_ACT_CD4_TCELL_48H_DN\n",
      "GSE22886_DC_VS_MONOCYTE_UP\n",
      "GSE19888_CTRL_VS_A3R_INHIBITOR_TREATED_MAST_CELL_UP\n",
      "GSE15930_NAIVE_VS_24H_IN_VITRO_STIM_CD8_TCELL_DN\n",
      "GSE45365_HEALTHY_VS_MCMV_INFECTION_BCELL_IFNAR_KO_UP\n",
      "GSE20715_0H_VS_48H_OZONE_LUNG_DN\n",
      "GSE22935_24H_VS_48H_MBOVIS_BCG_STIM_MYD88_KO_MACROPHAGE_DN\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 11, 15, 0.205294886419603\n",
      "best merge is 9, 17, 0.312059920964693\n",
      "best merge is 5, 10, 0.325433537772234\n",
      "best merge is 0, 6, 0.356023460472013\n",
      "best merge is 8, 16, 0.414115788538816\n",
      "best merge is 2, 13, 0.46043352238664\n",
      "best merge is 1, 4, 0.48454508054128\n",
      "best merge is 5, 14, 0.571908057769858\n",
      "best merge is 2, 11, 0.634855194381008\n",
      "best merge is 1, 19, 0.646827642308487\n",
      "best merge is 8, 18, 0.659259306919812\n",
      "best merge is 3, 8, 0.68514031403944\n",
      "best merge is 2, 7, 0.788751377916219\n",
      "best merge is 5, 9, 0.79423046348301\n",
      "best merge is 1, 2, 0.887675418716669\n",
      "best merge is 0, 3, 0.951023802189351\n",
      "order classes\n",
      "2 - 1.78011730236006\n",
      "8 - 1.67605362061746\n",
      "4 - 2.36753459031047\n",
      "5 - 2.09494155365591\n",
      "1 4 19 2 13 11 15 7\n",
      "0 6\n",
      "3 8 16 18\n",
      "5 10 14 9 17\n",
      "12\n",
      "Done.-7.41587901701323\t42\t0\n",
      "6.16068052930057\t42\t25.1682419659735\n",
      "7.68241965973535\t42\t27.1531190926276\n",
      "8.58223062381852\t42\t28.2117202268431\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.3024574669187\t42\t30.0642722117202\n",
      "10.7920604914934\t42\t30.5406427221172\n",
      "11.3345935727788\t42\t31.0434782608696\n",
      "12.0359168241966\t42\t31.6257088846881\n",
      "17.2230623818526\t42\t32.4461247637051\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_c7/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 17786\n",
      "cat 8619, mi = 0.00498822, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "cat 663, mi = 0.00411150, BROWNE_HCMV_INFECTION_24HR_UP\n",
      "cat 10877, mi = 0.00390705, GSE34006_A2AR_KO_VS_A2AR_AGONIST_TREATED_TREG_DN\n",
      "cat 12791, mi = 0.00388062, HALLMARK_E2F_TARGETS\n",
      "cat 9264, mi = 0.00376387, GSE19888_CTRL_VS_A3R_INHIBITOR_TREATED_MAST_CELL_UP\n",
      "cat 12638, mi = 0.00375006, GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP\n",
      "cat 7922, mi = 0.00368492, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 7916, mi = 0.00368459, GSE10325_CD4_TCELL_VS_MYELOID_UP\n",
      "cat 17325, mi = 0.00368172, YU_MYC_TARGETS_DN\n",
      "cat 16327, mi = 0.00366892, SA_B_CELL_RECEPTOR_COMPLEXES\n",
      "cat 8219, mi = 0.00365880, GSE13547_2H_VS_12_H_ANTI_IGM_STIM_BCELL_DN\n",
      "cat 8063, mi = 0.00364528, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "cat 427, mi = 0.00359045, BIOCARTA_ETS_PATHWAY\n",
      "cat 8527, mi = 0.00356656, GSE15930_NAIVE_VS_24H_IN_VITRO_STIM_CD8_TCELL_DN\n",
      "cat 8323, mi = 0.00355701, GSE14350_TREG_VS_TEFF_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 8619, mi = 0.005, pvalue = 0.000, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "Passed: cat 663, mi = 0.004, pvalue = 0.000, BROWNE_HCMV_INFECTION_24HR_UP\n",
      "Passed: cat 10877, mi = 0.004, pvalue = 0.000, GSE34006_A2AR_KO_VS_A2AR_AGONIST_TREATED_TREG_DN\n",
      "Passed: cat 12791, mi = 0.004, pvalue = 0.000, HALLMARK_E2F_TARGETS\n",
      "Passed: cat 9264, mi = 0.004, pvalue = 0.000, GSE19888_CTRL_VS_A3R_INHIBITOR_TREATED_MAST_CELL_UP\n",
      "Passed: cat 12638, mi = 0.004, pvalue = 0.000, GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP\n",
      "Passed: cat 427, mi = 0.004, pvalue = 0.000, BIOCARTA_ETS_PATHWAY\n",
      "Passed: cat 8323, mi = 0.004, pvalue = 0.000, GSE14350_TREG_VS_TEFF_DN\n",
      "Passed: cat 14521, mi = 0.004, pvalue = 0.001, MORF_ANP32B\n",
      "Passed: cat 12762, mi = 0.003, pvalue = 0.000, GU_PDEF_TARGETS_UP\n",
      "Passed: cat 4814, mi = 0.003, pvalue = 0.000, GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "Passed: cat 16293, mi = 0.003, pvalue = 0.001, SAMOLS_TARGETS_OF_KHSV_MIRNAS_DN\n",
      "Passed: cat 8021, mi = 0.003, pvalue = 0.001, GSE11961_MARGINAL_ZONE_BCELL_VS_PLASMA_CELL_DAY7_DN\n",
      "Passed: cat 14712, mi = 0.003, pvalue = 0.002, MORI_EMU_MYC_LYMPHOMA_BY_ONSET_TIME_UP\n",
      "Passed: cat 9724, mi = 0.003, pvalue = 0.001, GSE22886_DC_VS_MONOCYTE_UP\n",
      "Passed: cat 13560, mi = 0.003, pvalue = 0.000, KYNG_ENVIRONMENTAL_STRESS_RESPONSE_NOT_BY_UV_IN_OLD\n",
      "Passed: cat 752, mi = 0.003, pvalue = 0.001, CAGCAGG_MIR370\n",
      "Passed: cat 9056, mi = 0.003, pvalue = 0.001, GSE17974_1.5H_VS_72H_IL4_AND_ANTI_IL12_ACT_CD4_TCELL_UP\n",
      "Passed: cat 5898, mi = 0.003, pvalue = 0.002, GO_PROTEIN_N_LINKED_GLYCOSYLATION\n",
      "Passed: cat 1051, mi = 0.003, pvalue = 0.001, CTCTATG_MIR368\n",
      "Passed: cat 10048, mi = 0.003, pvalue = 0.003, GSE25087_TREG_VS_TCONV_FETUS_UP\n",
      "Passed: cat 9809, mi = 0.003, pvalue = 0.002, GSE22935_24H_VS_48H_MBOVIS_BCG_STIM_MYD88_KO_MACROPHAGE_DN\n",
      "Passed: cat 3495, mi = 0.003, pvalue = 0.002, GO_HEPARAN_SULFATE_PROTEOGLYCAN_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 4806, mi = 0.003, pvalue = 0.003, GO_NUCLEAR_PERIPHERY\n",
      "Passed: cat 8569, mi = 0.003, pvalue = 0.002, GSE16266_LPS_VS_HEATSHOCK_AND_LPS_STIM_MEF_DN\n",
      "Passed: cat 12571, mi = 0.003, pvalue = 0.002, GSE8835_HEALTHY_VS_CLL_CD4_TCELL_DN\n",
      "Passed: cat 9150, mi = 0.003, pvalue = 0.004, GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_2H_UP\n",
      "Passed: cat 11652, mi = 0.003, pvalue = 0.002, GSE40274_IRF4_VS_FOXP3_AND_IRF4_TRANSDUCED_ACTIVATED_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5260, mi = 0.003, pvalue = 0.005, GO_POSITIVE_REGULATION_OF_AXON_EXTENSION\n",
      "Passed: cat 12686, mi = 0.003, pvalue = 0.003, GSE9960_HEALTHY_VS_GRAM_POS_SEPSIS_PBMC_UP\n",
      "Passed: cat 1242, mi = 0.003, pvalue = 0.005, E2F_Q2\n",
      "Didn't pass (cutoff=0.0050): cat 6316, mi = 0.003, pvalue = 0.005, GO_REGULATION_OF_EXTENT_OF_CELL_GROWTH\n",
      "Didn't pass (cutoff=0.0050): cat 11530, mi = 0.003, pvalue = 0.005, GSE3982_MAC_VS_TH2_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3215, mi = 0.003, pvalue = 0.005, GO_EXONUCLEASE_ACTIVITY_ACTIVE_WITH_EITHER_RIBO_OR_DEOXYRIBONUCLEIC_ACIDS_AND_PRODUCING_5_PHOSPHOMONOESTERS\n",
      "Passed: cat 4316, mi = 0.003, pvalue = 0.005, GO_NEGATIVE_REGULATION_OF_BMP_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 6048, mi = 0.003, pvalue = 0.008, GO_REGULATION_OF_ACTIN_FILAMENT_LENGTH\n",
      "Didn't pass (cutoff=0.0050): cat 9907, mi = 0.003, pvalue = 0.006, GSE2405_0H_VS_3H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 16291, mi = 0.003, pvalue = 0.007, SAKAI_TUMOR_INFILTRATING_MONOCYTES_DN\n",
      "Passed: cat 17734, mi = 0.003, pvalue = 0.002, chr7q31\n",
      "Passed: cat 4154, mi = 0.003, pvalue = 0.004, GO_MONOCARBOXYLIC_ACID_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 8098, mi = 0.003, pvalue = 0.005, GSE12845_IGD_NEG_BLOOD_VS_PRE_GC_TONSIL_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 15381, mi = 0.003, pvalue = 0.009, PPARA_02\n",
      "Didn't pass (cutoff=0.0050): cat 16070, mi = 0.003, pvalue = 0.013, REACTOME_SPHINGOLIPID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 11242, mi = 0.003, pvalue = 0.011, GSE37301_RAG2_KO_VS_RAG2_AND_ETS1_KO_NK_CELL_UP\n",
      "Passed: cat 5211, mi = 0.003, pvalue = 0.001, GO_POLYOL_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 8902, mi = 0.003, pvalue = 0.011, GSE17721_LPS_VS_PAM3CSK4_16H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2417, mi = 0.003, pvalue = 0.014, GO_CELLULAR_RESPIRATION\n",
      "Didn't pass (cutoff=0.0050): cat 1221, mi = 0.003, pvalue = 0.018, DUTTA_APOPTOSIS_VIA_NFKB\n",
      "Didn't pass (cutoff=0.0050): cat 14132, mi = 0.003, pvalue = 0.011, MODULE_170\n",
      "Passed: cat 17008, mi = 0.003, pvalue = 0.003, VISALA_AGING_LYMPHOCYTE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1893, mi = 0.003, pvalue = 0.017, GO_3_5_EXONUCLEASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 9161, mi = 0.003, pvalue = 0.013, GSE18893_CTRL_VS_TNF_TREATED_TCONV_24H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4032, mi = 0.003, pvalue = 0.019, GO_METANEPHROS_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 14041, mi = 0.002, pvalue = 0.019, MILI_PSEUDOPODIA\n",
      "Didn't pass (cutoff=0.0050): cat 16862, mi = 0.002, pvalue = 0.014, TRAYNOR_RETT_SYNDROM_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1621, mi = 0.002, pvalue = 0.013, GCM_SMARCC1\n",
      "Didn't pass (cutoff=0.0050): cat 7258, mi = 0.002, pvalue = 0.024, GO_SH3_SH2_ADAPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 15297, mi = 0.002, pvalue = 0.020, PID_RAC1_REG_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 9590, mi = 0.002, pvalue = 0.021, GSE22025_UNTREATED_VS_PROGESTERONE_TREATED_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2579, mi = 0.002, pvalue = 0.016, GO_CELL_REDOX_HOMEOSTASIS\n",
      "Didn't pass (cutoff=0.0050): cat 16971, mi = 0.002, pvalue = 0.011, VANTVEER_BREAST_CANCER_POOR_PROGNOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 8869, mi = 0.002, pvalue = 0.020, GSE17721_LPS_VS_CPG_24H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3214, mi = 0.002, pvalue = 0.024, GO_EXONUCLEASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 11635, mi = 0.002, pvalue = 0.015, GSE40274_FOXP3_VS_FOXP3_AND_HELIOS_TRANSDUCED_ACTIVATED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4078, mi = 0.002, pvalue = 0.025, GO_MICROVILLUS\n",
      "Didn't pass (cutoff=0.0050): cat 10771, mi = 0.002, pvalue = 0.018, GSE32423_IL7_VS_IL7_IL4_MEMORY_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8615, mi = 0.002, pvalue = 0.026, GSE16450_CTRL_VS_IFNA_12H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1737, mi = 0.002, pvalue = 0.024, GNF2_CD48\n",
      "Didn't pass (cutoff=0.0050): cat 11603, mi = 0.002, pvalue = 0.022, GSE40274_CTRL_VS_FOXP3_AND_HELIOS_TRANSDUCED_ACTIVATED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8635, mi = 0.002, pvalue = 0.021, GSE16522_ANTI_CD3CD28_STIM_VS_UNSTIM_MEMORY_CD8_TCELL_DN\n",
      "Total number of categories evaluated = 488.\n",
      "Number of categories that passed the tests = 35\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 14, 24, 0.125284635934465\n",
      "best merge is 6, 34, 0.125412297104053\n",
      "best merge is 14, 28, 0.17801256942534\n",
      "best merge is 15, 18, 0.232509920802826\n",
      "best merge is 16, 19, 0.236042404500413\n",
      "best merge is 20, 33, 0.249920257237796\n",
      "best merge is 27, 31, 0.265466038901826\n",
      "best merge is 7, 32, 0.26967369492689\n",
      "best merge is 4, 23, 0.313697948416995\n",
      "best merge is 25, 29, 0.326365051284038\n",
      "best merge is 5, 12, 0.353094843302512\n",
      "best merge is 3, 21, 0.358028309514169\n",
      "best merge is 1, 8, 0.361864274953664\n",
      "best merge is 1, 14, 0.407399758112932\n",
      "best merge is 0, 6, 0.423975733032542\n",
      "best merge is 10, 17, 0.440240493281765\n",
      "best merge is 15, 22, 0.457700539259373\n",
      "best merge is 5, 30, 0.491773673436016\n",
      "best merge is 7, 11, 0.493228012296001\n",
      "best merge is 2, 16, 0.513758408708256\n",
      "best merge is 13, 20, 0.543546139626126\n",
      "best merge is 10, 27, 0.553134555635458\n",
      "best merge is 3, 26, 0.587208028937831\n",
      "best merge is 1, 4, 0.607348563570226\n",
      "best merge is 1, 2, 0.681163007462265\n",
      "best merge is 0, 9, 0.683438682125689\n",
      "best merge is 3, 25, 0.753417519428857\n",
      "best merge is 3, 13, 0.829783363084811\n",
      "best merge is 5, 10, 0.890635739875622\n",
      "best merge is 0, 15, 0.921361537801587\n",
      "best merge is 1, 5, 0.990920014347282\n",
      "order classes\n",
      "7 - 3.10774066581364\n",
      "10 - 1.76699169928121\n",
      "8 - 2.75667075441532\n",
      "7 - 3.04949302518519\n",
      "3 - 3.25663260616009\n",
      "1 8 14 24 28 4 23 2 16 19\n",
      "0 6 34 9 15 18 22\n",
      "3 21 26 25 29 13 20 33\n",
      "5 12 30 10 17 27 31\n",
      "7 32 11\n",
      "Done.Start drawing\n",
      "9.21\t-11.27\n",
      "GSE14350_TREG_VS_TEFF_DN\n",
      "GO_MONOCARBOXYLIC_ACID_CATABOLIC_PROCESS\n",
      "MORI_EMU_MYC_LYMPHOMA_BY_ONSET_TIME_UP\n",
      "GSE9509_LPS_VS_LPS_AND_IL10_STIM_IL10_KO_MACROPHAGE_20MIN_UP\n",
      "GO_NEGATIVE_REGULATION_OF_BMP_SIGNALING_PATHWAY\n",
      "BIOCARTA_ETS_PATHWAY\n",
      "VISALA_AGING_LYMPHOCYTE_UP\n",
      "GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "KYNG_ENVIRONMENTAL_STRESS_RESPONSE_NOT_BY_UV_IN_OLD\n",
      "GO_PROTEIN_N_LINKED_GLYCOSYLATION\n",
      "GO_HEPARAN_SULFATE_PROTEOGLYCAN_BIOSYNTHETIC_PROCESS\n",
      "GSE34006_A2AR_KO_VS_A2AR_AGONIST_TREATED_TREG_DN\n",
      "GU_PDEF_TARGETS_UP\n",
      "chr7q31\n",
      "GO_NUCLEAR_TRANSCRIBED_MRNA_CATABOLIC_PROCESS_NONSENSE_MEDIATED_DECAY\n",
      "GSE17974_1.5H_VS_72H_IL4_AND_ANTI_IL12_ACT_CD4_TCELL_UP\n",
      "GSE11961_MARGINAL_ZONE_BCELL_VS_PLASMA_CELL_DAY7_DN\n",
      "GSE40274_IRF4_VS_FOXP3_AND_IRF4_TRANSDUCED_ACTIVATED_CD4_TCELL_UP\n",
      "SAMOLS_TARGETS_OF_KHSV_MIRNAS_DN\n",
      "BROWNE_HCMV_INFECTION_24HR_UP\n",
      "MORF_ANP32B\n",
      "GSE22886_DC_VS_MONOCYTE_UP\n",
      "GSE16266_LPS_VS_HEATSHOCK_AND_LPS_STIM_MEF_DN\n",
      "GSE9960_HEALTHY_VS_GRAM_POS_SEPSIS_PBMC_UP\n",
      "GO_NUCLEAR_PERIPHERY\n",
      "CAGCAGG_MIR370\n",
      "CTCTATG_MIR368\n",
      "GSE19888_CTRL_VS_A3R_INHIBITOR_TREATED_MAST_CELL_UP\n",
      "GSE25087_TREG_VS_TCONV_FETUS_UP\n",
      "GO_POLYOL_CATABOLIC_PROCESS\n",
      "HALLMARK_E2F_TARGETS\n",
      "GSE22935_24H_VS_48H_MBOVIS_BCG_STIM_MYD88_KO_MACROPHAGE_DN\n",
      "GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_2H_UP\n",
      "GSE8835_HEALTHY_VS_CLL_CD4_TCELL_DN\n",
      "E2F_Q2\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 14, 24, 0.125284635934465\n",
      "best merge is 6, 34, 0.125412297104053\n",
      "best merge is 14, 28, 0.17801256942534\n",
      "best merge is 15, 18, 0.232509920802826\n",
      "best merge is 16, 19, 0.236042404500413\n",
      "best merge is 20, 33, 0.249920257237796\n",
      "best merge is 27, 31, 0.265466038901826\n",
      "best merge is 7, 32, 0.26967369492689\n",
      "best merge is 4, 23, 0.313697948416995\n",
      "best merge is 25, 29, 0.326365051284038\n",
      "best merge is 5, 12, 0.353094843302512\n",
      "best merge is 3, 21, 0.358028309514169\n",
      "best merge is 1, 8, 0.361864274953664\n",
      "best merge is 1, 14, 0.407399758112932\n",
      "best merge is 0, 6, 0.423975733032542\n",
      "best merge is 10, 17, 0.440240493281765\n",
      "best merge is 15, 22, 0.457700539259373\n",
      "best merge is 5, 30, 0.491773673436016\n",
      "best merge is 7, 11, 0.493228012296001\n",
      "best merge is 2, 16, 0.513758408708256\n",
      "best merge is 13, 20, 0.543546139626126\n",
      "best merge is 10, 27, 0.553134555635458\n",
      "best merge is 3, 26, 0.587208028937831\n",
      "best merge is 1, 4, 0.607348563570226\n",
      "best merge is 1, 2, 0.681163007462265\n",
      "best merge is 0, 9, 0.683438682125689\n",
      "best merge is 3, 25, 0.753417519428857\n",
      "best merge is 3, 13, 0.829783363084811\n",
      "best merge is 5, 10, 0.890635739875622\n",
      "best merge is 0, 15, 0.921361537801587\n",
      "best merge is 1, 5, 0.990920014347282\n",
      "order classes\n",
      "7 - 3.10774066581364\n",
      "10 - 1.76699169928121\n",
      "8 - 2.75667075441532\n",
      "7 - 3.04949302518519\n",
      "3 - 3.25663260616009\n",
      "1 8 14 24 28 4 23 2 16 19\n",
      "0 6 34 9 15 18 22\n",
      "3 21 26 25 29 13 20 33\n",
      "5 12 30 10 17 27 31\n",
      "7 32 11\n",
      "Done.-10.26904296875\t42\t0\n",
      "0.2412109375\t42\t19.4619140625\n",
      "1.4306640625\t42\t21.0205078125\n",
      "2.1279296875\t42\t21.8408203125\n",
      "2.65087890625\t42\t22.4150390625\n",
      "3.091796875\t42\t22.88671875\n",
      "3.4814453125\t42\t23.296875\n",
      "3.86083984375\t42\t23.666015625\n",
      "4.28125\t42\t24.0556640625\n",
      "4.82470703125\t42\t24.5068359375\n",
      "13.5712890625\t42\t25.142578125\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_full/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 1545 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 1545\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "140 hits go_terms_num = 50\n",
      "cat 22, mi = 0.01750262, HALLMARK_E2F_TARGETS\n",
      "cat 16, mi = 0.01326330, HALLMARK_MTORC1_SIGNALING\n",
      "cat 12, mi = 0.01160711, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "cat 3, mi = 0.01100837, HALLMARK_XENOBIOTIC_METABOLISM\n",
      "cat 38, mi = 0.01084345, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "cat 18, mi = 0.00965243, HALLMARK_COAGULATION\n",
      "cat 30, mi = 0.00958900, HALLMARK_CHOLESTEROL_HOMEOSTASIS\n",
      "cat 42, mi = 0.00957002, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "cat 21, mi = 0.00915427, HALLMARK_UV_RESPONSE_DN\n",
      "cat 29, mi = 0.00842551, HALLMARK_MYC_TARGETS_V2\n",
      "cat 13, mi = 0.00825053, HALLMARK_APICAL_JUNCTION\n",
      "cat 5, mi = 0.00766082, HALLMARK_PROTEIN_SECRETION\n",
      "cat 41, mi = 0.00738428, HALLMARK_ALLOGRAFT_REJECTION\n",
      "cat 7, mi = 0.00724768, HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY\n",
      "cat 24, mi = 0.00706912, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 22, mi = 0.018, pvalue = 0.000, HALLMARK_E2F_TARGETS\n",
      "Passed: cat 16, mi = 0.013, pvalue = 0.002, HALLMARK_MTORC1_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.012, pvalue = 0.010, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 18, mi = 0.010, pvalue = 0.022, HALLMARK_COAGULATION\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.009, pvalue = 0.041, HALLMARK_UV_RESPONSE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.008, pvalue = 0.152, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.007, pvalue = 0.201, HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.007, pvalue = 0.193, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.007, pvalue = 0.170, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.007, pvalue = 0.228, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.006, pvalue = 0.206, HALLMARK_INFLAMMATORY_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.006, pvalue = 0.051, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.006, pvalue = 0.376, HALLMARK_APICAL_SURFACE\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.006, pvalue = 0.377, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.006, pvalue = 0.390, HALLMARK_SPERMATOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.006, pvalue = 0.284, HALLMARK_APOPTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.005, pvalue = 0.329, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.005, pvalue = 0.417, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.004, pvalue = 0.694, HALLMARK_BILE_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.004, pvalue = 0.842, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.004, pvalue = 0.624, HALLMARK_UV_RESPONSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.004, pvalue = 0.672, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Total number of categories evaluated = 33.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-10.39\n",
      "HALLMARK_E2F_TARGETS\n",
      "HALLMARK_MTORC1_SIGNALING\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.8679119412942\t42\t0\n",
      "5.25883922615077\t42\t24.2641761174116\n",
      "6.8278852568379\t42\t26.2535023348899\n",
      "7.78052034689793\t42\t27.4022681787859\n",
      "8.45296864576384\t42\t28.15877251501\n",
      "9.0133422281521\t42\t28.7471647765177\n",
      "9.51767845230153\t42\t29.2795196797865\n",
      "10.0360240160107\t42\t29.7558372248165\n",
      "10.5963975983989\t42\t30.3162108072048\n",
      "11.2968645763843\t42\t30.8765843895931\n",
      "16.8585723815877\t42\t31.7171447631755\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_msigdb_h/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7491 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7491\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "681 hits go_terms_num = 790\n",
      "cat 354, mi = 0.00215862, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 672, mi = 0.00196724, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 657, mi = 0.00191826, TIA1_NA_POSTAR_CIMS\n",
      "cat 658, mi = 0.00191826, TIA1_NA_POSTAR_iCLIP\n",
      "cat 637, mi = 0.00190795, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 635, mi = 0.00189828, TARDBP_H9_POSTAR_CIMS\n",
      "cat 402, mi = 0.00187051, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 770, mi = 0.00178154, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00177145, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 484, mi = 0.00171330, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 235, mi = 0.00160367, FUS_HeLa_POSTAR_CIMS\n",
      "cat 31, mi = 0.00153486, AUH_K562_POSTAR_eCLIP\n",
      "cat 230, mi = 0.00148816, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 326, mi = 0.00148389, HNRNPU_HeLa_POSTAR_CIMS\n",
      "cat 676, mi = 0.00140682, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.002, pvalue = 0.018, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.050, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.002, pvalue = 0.055, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.002, pvalue = 0.056, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.056, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.055, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.061, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.002, pvalue = 0.052, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.076, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.073, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.002, pvalue = 0.043, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.203, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.001, pvalue = 0.135, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.151, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.001, pvalue = 0.162, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.001, pvalue = 0.196, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.324, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.001, pvalue = 0.322, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.299, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.001, pvalue = 0.548, HNRNPA1_HEK293T_POSTAR_CIMS\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.21\t-11.27\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_RBPs_all_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6816 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6816\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "620 hits go_terms_num = 790\n",
      "cat 256, mi = 0.00287154, GNL3_K562_POSTAR_eCLIP\n",
      "cat 672, mi = 0.00233075, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 637, mi = 0.00227032, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 635, mi = 0.00225981, TARDBP_H9_POSTAR_CIMS\n",
      "cat 354, mi = 0.00224603, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 770, mi = 0.00224040, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00222580, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00218926, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 670, mi = 0.00202200, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "cat 16, mi = 0.00186012, ALKBH1_HeLa_POSTAR_CIMS\n",
      "cat 235, mi = 0.00176190, FUS_HeLa_POSTAR_CIMS\n",
      "cat 484, mi = 0.00170206, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 671, mi = 0.00166403, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "cat 676, mi = 0.00166047, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 657, mi = 0.00157419, TIA1_NA_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 256, mi = 0.003, pvalue = 0.003, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.002, pvalue = 0.027, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 670, mi = 0.002, pvalue = 0.047, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.099, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.002, pvalue = 0.045, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 671, mi = 0.002, pvalue = 0.126, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.002, pvalue = 0.129, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.268, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 300, mi = 0.002, pvalue = 0.235, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 674, mi = 0.001, pvalue = 0.249, TNRC6C_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.487, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 323, mi = 0.001, pvalue = 0.086, HNRNPU_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 675, mi = 0.001, pvalue = 0.539, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.001, pvalue = 0.750, HNRNPA1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 667, mi = 0.001, pvalue = 0.547, TNRC6A_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.809, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 663, mi = 0.001, pvalue = 0.247, TIAL1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 664, mi = 0.001, pvalue = 0.249, TIAL1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.001, pvalue = 0.797, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.001, pvalue = 0.976, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass: cat 357, mi = 0.001, pvalue = 1.000, IGF2BP3_Panc1_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 43.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-11.27\n",
      "GNL3_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.41587901701323\t42\t0\n",
      "6.16068052930057\t42\t25.1682419659735\n",
      "7.68241965973535\t42\t27.1531190926276\n",
      "8.58223062381852\t42\t28.2117202268431\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.3024574669187\t42\t30.0642722117202\n",
      "10.7920604914934\t42\t30.5406427221172\n",
      "11.3345935727788\t42\t31.0434782608696\n",
      "12.0359168241966\t42\t31.6257088846881\n",
      "17.2230623818526\t42\t32.4461247637051\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_RBPs_coding_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6678 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6678\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "607 hits go_terms_num = 790\n",
      "cat 610, mi = 0.00295259, SUGP2_HepG2_POSTAR_eCLIP\n",
      "cat 648, mi = 0.00282766, TBRG4_K562_POSTAR_eCLIP\n",
      "cat 289, mi = 0.00261951, HNRNPC_HEK293_POSTAR_PARalyzer\n",
      "cat 575, mi = 0.00234029, SMNDC1_HepG2_POSTAR_eCLIP\n",
      "cat 769, mi = 0.00228583, YTHDF3_HEK293T_POSTAR_CIMS\n",
      "cat 771, mi = 0.00228583, YTHDF3_HEK293T_POSTAR_iCLIP\n",
      "cat 317, mi = 0.00224980, HNRNPM_K562_POSTAR_eCLIP\n",
      "cat 752, mi = 0.00218803, YTHDC2_HeLa_POSTAR_PARalyzer\n",
      "cat 439, mi = 0.00216808, NOP56_HEK293_POSTAR_PAR-CLIP\n",
      "cat 440, mi = 0.00216808, NOP56_HEK293_POSTAR_PARalyzer\n",
      "cat 43, mi = 0.00215338, CAPRIN1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 768, mi = 0.00214476, YTHDF2_Huh7_POSTAR_Piranha_0.01\n",
      "cat 548, mi = 0.00214039, RTCB_HEK293_POSTAR_Piranha_0.01\n",
      "cat 751, mi = 0.00211324, YTHDC2_HeLa_POSTAR_PAR-CLIP\n",
      "cat 66, mi = 0.00205063, CPSF2_HEK293_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 610, mi = 0.003, pvalue = 0.004, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Passed: cat 648, mi = 0.003, pvalue = 0.002, TBRG4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 769, mi = 0.002, pvalue = 0.018, YTHDF3_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 771, mi = 0.002, pvalue = 0.019, YTHDF3_HEK293T_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 752, mi = 0.002, pvalue = 0.049, YTHDC2_HeLa_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.002, pvalue = 0.052, CAPRIN1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 548, mi = 0.002, pvalue = 0.061, RTCB_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 751, mi = 0.002, pvalue = 0.062, YTHDC2_HeLa_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.002, pvalue = 0.017, CPSF2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.081, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 404, mi = 0.002, pvalue = 0.127, MBNL2_frontalcortex_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 600, mi = 0.002, pvalue = 0.090, SSB_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 599, mi = 0.002, pvalue = 0.090, SSB_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.002, pvalue = 0.135, CNBP_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.002, pvalue = 0.139, DHX30_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 396, mi = 0.002, pvalue = 0.153, LSM11_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 157, mi = 0.002, pvalue = 0.147, EIF3A_HEK293T_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 225, mi = 0.002, pvalue = 0.252, FUS_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.002, pvalue = 0.219, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 579, mi = 0.002, pvalue = 0.234, SND1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 405, mi = 0.002, pvalue = 0.026, MBNL2_hippocampus_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 403, mi = 0.002, pvalue = 0.250, MBNL2_frontalcortex_POSTAR_HITS-CLIP\n",
      "Total number of categories evaluated = 51.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-11.27\n",
      "SUGP2_HepG2_POSTAR_eCLIP\n",
      "TBRG4_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.38941398865784\t42\t0\n",
      "6.20037807183365\t42\t25.2211720226843\n",
      "7.69565217391305\t42\t27.179584120983\n",
      "8.58223062381852\t42\t28.2117202268431\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.3024574669187\t42\t30.0642722117202\n",
      "10.7788279773157\t42\t30.5406427221172\n",
      "11.3081285444234\t42\t31.0170132325142\n",
      "12.0094517958412\t42\t31.5992438563327\n",
      "17.2098298676749\t42\t32.4196597353497\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6602 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6602\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "600 hits go_terms_num = 790\n",
      "cat 228, mi = 0.00311597, FUS_HEK293T_POSTAR_PAR-CLIP\n",
      "cat 229, mi = 0.00311597, FUS_HEK293T_POSTAR_PARalyzer\n",
      "cat 743, mi = 0.00280662, YBX3_K562_POSTAR_eCLIP\n",
      "cat 526, mi = 0.00269038, RBM22_HepG2_POSTAR_eCLIP\n",
      "cat 520, mi = 0.00268258, RBM15_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 703, mi = 0.00263000, UCHL5_K562_POSTAR_eCLIP\n",
      "cat 262, mi = 0.00260697, GRWD1_HepG2_POSTAR_eCLIP\n",
      "cat 784, mi = 0.00252045, ZNF622_K562_POSTAR_eCLIP\n",
      "cat 373, mi = 0.00250106, LARP4_K562_POSTAR_eCLIP\n",
      "cat 696, mi = 0.00246596, U2AF2_K562_POSTAR_eCLIP\n",
      "cat 152, mi = 0.00238307, EFTUD2_HepG2_POSTAR_eCLIP\n",
      "cat 226, mi = 0.00237779, FUS_Brain_POSTAR_HITS-CLIP\n",
      "cat 227, mi = 0.00237779, FUS_Brain_POSTAR_Piranha_0.01\n",
      "cat 201, mi = 0.00234429, FASTKD2_HepG2_POSTAR_eCLIP\n",
      "cat 708, mi = 0.00232446, UPF1_HEK_293_FRT_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 228, mi = 0.003, pvalue = 0.003, FUS_HEK293T_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.002, pvalue = 0.020, FKBP4_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.035, AGGF1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.002, pvalue = 0.039, AGGF1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 508, mi = 0.002, pvalue = 0.052, RBFOX2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 203, mi = 0.002, pvalue = 0.058, FASTKD2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.002, pvalue = 0.035, CAPRIN1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 724, mi = 0.002, pvalue = 0.017, WDR33_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.002, pvalue = 0.062, EIF4G2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 597, mi = 0.002, pvalue = 0.090, SRSF9_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 273, mi = 0.002, pvalue = 0.033, HNRNPA1_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 274, mi = 0.002, pvalue = 0.034, HNRNPA1_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.002, pvalue = 0.168, AKAP8L_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 53, mi = 0.002, pvalue = 0.244, CNBP_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.002, pvalue = 0.241, DICER1_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.002, pvalue = 0.199, CNBP_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 289, mi = 0.002, pvalue = 0.210, HNRNPC_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 571, mi = 0.002, pvalue = 0.241, SLTM_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 575, mi = 0.002, pvalue = 0.310, SMNDC1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 738, mi = 0.001, pvalue = 0.215, XRN2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.001, pvalue = 0.370, DICER1_HEK293_POSTAR_PARalyzer\n",
      "Total number of categories evaluated = 114.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-11.27\n",
      "FUS_HEK293T_POSTAR_PAR-CLIP\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.41587901701323\t42\t0\n",
      "6.16068052930057\t42\t25.1682419659735\n",
      "7.66918714555766\t42\t27.1531190926276\n",
      "8.56899810964083\t42\t28.1852551984877\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.289224952741\t42\t30.0642722117202\n",
      "10.765595463138\t42\t30.5141776937618\n",
      "11.3081285444234\t42\t31.0170132325142\n",
      "12.0094517958412\t42\t31.5992438563327\n",
      "17.2098298676749\t42\t32.4196597353497\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6691 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6691\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "608 hits go_terms_num = 790\n",
      "cat 383, mi = 0.00299843, LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "cat 492, mi = 0.00296060, PUM2_HEK293_POSTAR_PAR-CLIP\n",
      "cat 493, mi = 0.00291774, PUM2_HEK293_POSTAR_PARalyzer\n",
      "cat 428, mi = 0.00281464, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 173, mi = 0.00280724, EIF4A3_HeLa_POSTAR_CIMS\n",
      "cat 382, mi = 0.00268364, LIN28A_HEK293_POSTAR_CIMS\n",
      "cat 373, mi = 0.00268286, LARP4_K562_POSTAR_eCLIP\n",
      "cat 433, mi = 0.00264755, NOL12_HepG2_POSTAR_eCLIP\n",
      "cat 439, mi = 0.00260163, NOP56_HEK293_POSTAR_PAR-CLIP\n",
      "cat 440, mi = 0.00260163, NOP56_HEK293_POSTAR_PARalyzer\n",
      "cat 757, mi = 0.00260063, YTHDF1_Huh7_POSTAR_PAR-CLIP\n",
      "cat 40, mi = 0.00250476, C22orf28_HEK293_Plass\n",
      "cat 201, mi = 0.00246792, FASTKD2_HepG2_POSTAR_eCLIP\n",
      "cat 189, mi = 0.00243902, EWSR1_HEK293_Plass\n",
      "cat 205, mi = 0.00242378, FBL_HEK293_POSTAR_PAR-CLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 383, mi = 0.003, pvalue = 0.003, LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.003, pvalue = 0.018, C22orf28_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 189, mi = 0.002, pvalue = 0.021, EWSR1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.002, pvalue = 0.026, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 223, mi = 0.002, pvalue = 0.027, FUBP3_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.029, ELAVL1_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 467, mi = 0.002, pvalue = 0.044, PRKRA_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 468, mi = 0.002, pvalue = 0.045, PRKRA_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.002, pvalue = 0.048, DDX42_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 398, mi = 0.002, pvalue = 0.046, LSM11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.002, pvalue = 0.053, FKBP4_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 533, mi = 0.002, pvalue = 0.047, RBM47_MDA-MB-231_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.002, pvalue = 0.049, DDX55_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.002, pvalue = 0.066, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 771, mi = 0.002, pvalue = 0.091, YTHDF3_HEK293T_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 669, mi = 0.002, pvalue = 0.087, TNRC6A_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 532, mi = 0.002, pvalue = 0.060, RBM47_MDA-MB-231_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.002, pvalue = 0.098, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 130, mi = 0.002, pvalue = 0.127, DGCR8_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.002, pvalue = 0.132, AGO3_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 731, mi = 0.002, pvalue = 0.131, WTAP_HeLa_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 52.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-11.27\n",
      "LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.38941398865784\t42\t0\n",
      "6.20037807183365\t42\t25.2211720226843\n",
      "7.69565217391305\t42\t27.179584120983\n",
      "8.58223062381852\t42\t28.2117202268431\n",
      "9.24385633270322\t42\t28.952741020794\n",
      "9.79962192816635\t42\t29.5349716446125\n",
      "10.3024574669187\t42\t30.0642722117202\n",
      "10.7788279773157\t42\t30.5406427221172\n",
      "11.3081285444234\t42\t31.0170132325142\n",
      "12.0094517958412\t42\t31.5992438563327\n",
      "17.2098298676749\t42\t32.4196597353497\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 790\n",
      "cat 442, mi = 0.00280732, NOP56_HEK293_Plass\n",
      "cat 635, mi = 0.00253264, TARDBP_H9_POSTAR_CIMS\n",
      "cat 637, mi = 0.00253264, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 256, mi = 0.00250663, GNL3_K562_POSTAR_eCLIP\n",
      "cat 657, mi = 0.00241056, TIA1_NA_POSTAR_CIMS\n",
      "cat 658, mi = 0.00241056, TIA1_NA_POSTAR_iCLIP\n",
      "cat 329, mi = 0.00216088, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "cat 57, mi = 0.00208693, CPSF-100_HEK293_Plass\n",
      "cat 126, mi = 0.00201375, DGCR8_HEK293T_POSTAR_CIMS\n",
      "cat 770, mi = 0.00199295, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 484, mi = 0.00191473, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 456, mi = 0.00187259, PAPD5_HEK293_Plass\n",
      "cat 235, mi = 0.00185419, FUS_HeLa_POSTAR_CIMS\n",
      "cat 354, mi = 0.00183541, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 705, mi = 0.00183201, UPF1_HEK293_Plass\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 442, mi = 0.003, pvalue = 0.008, NOP56_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.003, pvalue = 0.019, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.003, pvalue = 0.021, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.003, pvalue = 0.014, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.002, pvalue = 0.027, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.002, pvalue = 0.025, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 329, mi = 0.002, pvalue = 0.043, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 57, mi = 0.002, pvalue = 0.067, CPSF-100_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 126, mi = 0.002, pvalue = 0.062, DGCR8_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.002, pvalue = 0.072, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.091, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 456, mi = 0.002, pvalue = 0.136, PAPD5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.002, pvalue = 0.036, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.002, pvalue = 0.114, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 705, mi = 0.002, pvalue = 0.138, UPF1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 144, mi = 0.002, pvalue = 0.113, DIS3L2_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.228, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.234, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.002, pvalue = 0.167, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 670, mi = 0.002, pvalue = 0.180, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-11.27\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 101\n",
      "cat 68, mi = 0.00237524, RBMS3\n",
      "cat 27, mi = 0.00227824, HuR.2\n",
      "cat 9, mi = 0.00160477, ESRP2\n",
      "cat 21, mi = 0.00132324, HNRNPK\n",
      "cat 88, mi = 0.00120998, SRSF9.1\n",
      "cat 13, mi = 0.00120929, FXR2\n",
      "cat 6, mi = 0.00110978, CPEB4\n",
      "cat 34, mi = 0.00102106, KHDRBS3\n",
      "cat 62, mi = 0.00092791, RBM46\n",
      "cat 56, mi = 0.00086203, RBM3\n",
      "cat 69, mi = 0.00078954, RBMS3.1\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "cat 3, mi = 0.00000000, BRUNOL6\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.002, pvalue = 0.034, RBMS3\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.002, pvalue = 0.044, HuR.2\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.264, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.421, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.001, pvalue = 0.437, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.420, FXR2\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.539, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.720, KHDRBS3\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.793, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.695, RBM3\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.758, RBMS3.1\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, FMR1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.60\t-11.27\n",
      "Outputing EPS file deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/d_mtyl_METTL3_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_METTL3_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_METTL3_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_METTL3_vs_NT_human_ensembl_RBPs_DeepBind/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7509 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7509\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "683 hits go_terms_num = 17689\n",
      "cat 724, mi = 0.00332940, RNA polymerase II activating transcription factor binding\n",
      "cat 1602, mi = 0.00291278, visual learning\n",
      "cat 35, mi = 0.00277873, aging\n",
      "cat 5210, mi = 0.00274245, negative regulation of cell adhesion\n",
      "cat 516, mi = 0.00273105, lipid catabolic process\n",
      "cat 1891, mi = 0.00268729, muscle organ development\n",
      "cat 84, mi = 0.00268614, animal organ regeneration\n",
      "cat 3498, mi = 0.00267210, p53 binding\n",
      "cat 155, mi = 0.00265888, positive regulation of Notch signaling pathway\n",
      "cat 3493, mi = 0.00258301, negative regulation of T cell activation\n",
      "cat 1684, mi = 0.00254877, costamere\n",
      "cat 1609, mi = 0.00254381, positive regulation of multicellular organism growth\n",
      "cat 1940, mi = 0.00253361, regulation of RNA splicing\n",
      "cat 668, mi = 0.00250673, positive regulation of BMP signaling pathway\n",
      "cat 1124, mi = 0.00250560, sulfotransferase activity\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 724, mi = 0.003, pvalue = 0.000, RNA polymerase II activating transcription factor binding\n",
      "Passed: cat 1602, mi = 0.003, pvalue = 0.001, visual learning\n",
      "Passed: cat 5210, mi = 0.003, pvalue = 0.002, negative regulation of cell adhesion\n",
      "Passed: cat 516, mi = 0.003, pvalue = 0.002, lipid catabolic process\n",
      "Passed: cat 84, mi = 0.003, pvalue = 0.003, animal organ regeneration\n",
      "Didn't pass (cutoff=0.0050): cat 1684, mi = 0.003, pvalue = 0.005, costamere\n",
      "Didn't pass (cutoff=0.0050): cat 1609, mi = 0.003, pvalue = 0.005, positive regulation of multicellular organism growth\n",
      "Didn't pass (cutoff=0.0050): cat 1940, mi = 0.003, pvalue = 0.006, regulation of RNA splicing\n",
      "Didn't pass (cutoff=0.0050): cat 1124, mi = 0.003, pvalue = 0.005, sulfotransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1876, mi = 0.002, pvalue = 0.008, catalytic step 2 spliceosome\n",
      "Passed: cat 5165, mi = 0.002, pvalue = 0.004, mitotic cytokinesis\n",
      "Didn't pass (cutoff=0.0050): cat 2187, mi = 0.002, pvalue = 0.009, positive regulation of protein ubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 762, mi = 0.002, pvalue = 0.012, syntaxin binding\n",
      "Passed: cat 1978, mi = 0.002, pvalue = 0.001, histone H2A monoubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 2128, mi = 0.002, pvalue = 0.014, negative regulation of NF-kappaB transcription factor activity\n",
      "Didn't pass (cutoff=0.0050): cat 2844, mi = 0.002, pvalue = 0.008, regulation of defense response to virus by virus\n",
      "Didn't pass (cutoff=0.0050): cat 878, mi = 0.002, pvalue = 0.013, aggresome\n",
      "Didn't pass (cutoff=0.0050): cat 8428, mi = 0.002, pvalue = 0.014, negative regulation of G1/S transition of mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 1492, mi = 0.002, pvalue = 0.015, protein sumoylation\n",
      "Didn't pass (cutoff=0.0050): cat 982, mi = 0.002, pvalue = 0.018, response to estradiol\n",
      "Didn't pass (cutoff=0.0050): cat 7850, mi = 0.002, pvalue = 0.015, negative regulation of TOR signaling\n",
      "Passed: cat 10999, mi = 0.002, pvalue = 0.004, genitalia development\n",
      "Didn't pass (cutoff=0.0050): cat 3882, mi = 0.002, pvalue = 0.009, focal adhesion assembly\n",
      "Didn't pass (cutoff=0.0050): cat 2931, mi = 0.002, pvalue = 0.013, microvillus\n",
      "Didn't pass (cutoff=0.0050): cat 2420, mi = 0.002, pvalue = 0.007, keratan sulfate biosynthetic process\n",
      "Passed: cat 2280, mi = 0.002, pvalue = 0.002, phosphatidylinositol-4,5-bisphosphate 5-phosphatase activity\n",
      "Didn't pass (cutoff=0.0050): cat 10679, mi = 0.002, pvalue = 0.015, hemidesmosome\n",
      "Didn't pass (cutoff=0.0050): cat 72, mi = 0.002, pvalue = 0.020, response to ethanol\n",
      "Passed: cat 6752, mi = 0.002, pvalue = 0.003, skeletal muscle atrophy\n",
      "Didn't pass (cutoff=0.0050): cat 825, mi = 0.002, pvalue = 0.016, GDP binding\n",
      "Didn't pass (cutoff=0.0050): cat 492, mi = 0.002, pvalue = 0.009, embryonic skeletal system morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 3521, mi = 0.002, pvalue = 0.029, actin filament organization\n",
      "Passed: cat 3930, mi = 0.002, pvalue = 0.003, histone H3-K4 demethylation\n",
      "Didn't pass (cutoff=0.0050): cat 3542, mi = 0.002, pvalue = 0.026, skeletal system morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 907, mi = 0.002, pvalue = 0.030, type I interferon signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 7300, mi = 0.002, pvalue = 0.021, SWI/SNF complex\n",
      "Didn't pass (cutoff=0.0050): cat 7302, mi = 0.002, pvalue = 0.020, nBAF complex\n",
      "Passed: cat 7169, mi = 0.002, pvalue = 0.005, dystrophin-associated glycoprotein complex\n",
      "Didn't pass (cutoff=0.0050): cat 2528, mi = 0.002, pvalue = 0.033, endopeptidase activity\n",
      "Didn't pass (cutoff=0.0050): cat 7324, mi = 0.002, pvalue = 0.009, G-protein coupled purinergic nucleotide receptor activity\n",
      "Didn't pass (cutoff=0.0050): cat 7325, mi = 0.002, pvalue = 0.009, G-protein coupled purinergic nucleotide receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 3158, mi = 0.002, pvalue = 0.005, Golgi transport complex\n",
      "Didn't pass (cutoff=0.0050): cat 792, mi = 0.002, pvalue = 0.010, anion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 7809, mi = 0.002, pvalue = 0.012, ion channel complex\n",
      "Didn't pass (cutoff=0.0050): cat 8241, mi = 0.002, pvalue = 0.014, volume-sensitive anion channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 963, mi = 0.002, pvalue = 0.041, glucose metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 557, mi = 0.002, pvalue = 0.027, positive regulation of T cell activation\n",
      "Didn't pass (cutoff=0.0050): cat 1872, mi = 0.002, pvalue = 0.037, spliceosomal complex\n",
      "Didn't pass (cutoff=0.0050): cat 880, mi = 0.002, pvalue = 0.035, channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 2135, mi = 0.002, pvalue = 0.045, negative regulation of type I interferon production\n",
      "Didn't pass (cutoff=0.0050): cat 1956, mi = 0.002, pvalue = 0.033, Rab GTPase binding\n",
      "Passed: cat 3330, mi = 0.002, pvalue = 0.002, supraspliceosomal complex\n",
      "Didn't pass (cutoff=0.0050): cat 551, mi = 0.002, pvalue = 0.046, chromatin-mediated maintenance of transcription\n",
      "Didn't pass (cutoff=0.0050): cat 1309, mi = 0.002, pvalue = 0.036, cell cortex\n",
      "Didn't pass (cutoff=0.0050): cat 5296, mi = 0.002, pvalue = 0.034, positive regulation of G1/S transition of mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 2335, mi = 0.002, pvalue = 0.014, protein binding involved in protein folding\n",
      "Didn't pass (cutoff=0.0050): cat 5627, mi = 0.002, pvalue = 0.007, synaptic vesicle exocytosis\n",
      "Didn't pass (cutoff=0.0050): cat 1504, mi = 0.002, pvalue = 0.038, N-acetyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 301, mi = 0.002, pvalue = 0.029, response to bacterium\n",
      "Didn't pass (cutoff=0.0050): cat 4031, mi = 0.002, pvalue = 0.044, release of sequestered calcium ion into cytosol\n",
      "Didn't pass (cutoff=0.0050): cat 1332, mi = 0.002, pvalue = 0.042, positive regulation of interleukin-8 production\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.054, glucose homeostasis\n",
      "Didn't pass (cutoff=0.0050): cat 5042, mi = 0.002, pvalue = 0.034, cellular response to peptide\n",
      "Didn't pass (cutoff=0.0050): cat 628, mi = 0.002, pvalue = 0.054, voltage-gated ion channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 2170, mi = 0.002, pvalue = 0.058, T cell homeostasis\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.054, regulation of ion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 3071, mi = 0.002, pvalue = 0.055, cytokine production\n",
      "Didn't pass (cutoff=0.0050): cat 201, mi = 0.002, pvalue = 0.018, positive regulation of cytokine secretion\n",
      "Didn't pass (cutoff=0.0050): cat 2028, mi = 0.002, pvalue = 0.044, positive regulation of insulin secretion\n",
      "Didn't pass (cutoff=0.0050): cat 2463, mi = 0.002, pvalue = 0.020, positive regulation of protein autophosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 7413, mi = 0.002, pvalue = 0.047, ADP binding\n",
      "Didn't pass (cutoff=0.0050): cat 2140, mi = 0.002, pvalue = 0.020, dermatan sulfate biosynthetic process\n",
      "Total number of categories evaluated = 121.\n",
      "Number of categories that passed the tests = 13\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 5, 12, 0.079423623572938\n",
      "best merge is 6, 10, 0.215734223103021\n",
      "best merge is 1, 9, 0.242930521247265\n",
      "best merge is 2, 4, 0.265391559295184\n",
      "best merge is 0, 5, 0.310377620619708\n",
      "best merge is 1, 11, 0.466856004828895\n",
      "best merge is 3, 8, 0.468440642393756\n",
      "best merge is 0, 2, 0.761902335902908\n",
      "best merge is 1, 7, 0.826540104829419\n",
      "order classes\n",
      "5 - 1.1516927634212\n",
      "3 - 1.13306317790857\n",
      "2 - 1.52243208777971\n",
      "2 - 0.70113622508482\n",
      "6 10\n",
      "1 9 11\n",
      "3 8\n",
      "0 5 12 2 4\n",
      "7\n",
      "Done.Start drawing\n",
      "5.76\t-12.09\n",
      "visual learning, GO:0008542\n",
      "skeletal muscle atrophy, GO:0014732\n",
      "dystrophin-associated glycoprotein complex, GO:0016010\n",
      "genitalia development, GO:0048806\n",
      "negative regulation of cell adhesion, GO:0007162\n",
      "animal organ regeneration, GO:0031100\n",
      "RNA polymerase II activating transcription factor binding, GO:0001102\n",
      "mitotic cytokinesis, GO:0000281\n",
      "supraspliceosomal complex, GO:0044530\n",
      "histone H2A monoubiquitination, GO:0035518\n",
      "histone H3-K4 demethylation, GO:0034720\n",
      "lipid catabolic process, GO:0016042\n",
      "phosphatidylinositol-4,5-bisphosphate 5-phosphatase activity, GO:0004439\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 5, 12, 0.079423623572938\n",
      "best merge is 6, 10, 0.215734223103021\n",
      "best merge is 1, 9, 0.242930521247265\n",
      "best merge is 2, 4, 0.265391559295184\n",
      "best merge is 0, 5, 0.310377620619708\n",
      "best merge is 1, 11, 0.466856004828895\n",
      "best merge is 3, 8, 0.468440642393756\n",
      "best merge is 0, 2, 0.761902335902908\n",
      "best merge is 1, 7, 0.826540104829419\n",
      "order classes\n",
      "5 - 1.1516927634212\n",
      "3 - 1.13306317790857\n",
      "2 - 1.52243208777971\n",
      "2 - 0.70113622508482\n",
      "6 10\n",
      "1 9 11\n",
      "3 8\n",
      "0 5 12 2 4\n",
      "7\n",
      "Done.-7.28235294117647\t42\t0\n",
      "6.09411764705882\t42\t25.4352941176471\n",
      "7.12941176470588\t42\t26.7529411764706\n",
      "7.74117647058823\t42\t27.5058823529412\n",
      "8.16470588235294\t42\t27.9764705882353\n",
      "8.49411764705882\t42\t28.3529411764706\n",
      "8.77647058823529\t42\t28.6352941176471\n",
      "9.08235294117647\t42\t28.9176470588235\n",
      "9.44705882352941\t42\t29.2470588235294\n",
      "10\t42\t29.6470588235294\n",
      "16.1764705882353\t42\t30.3529411764706\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6496 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6496\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "591 hits go_terms_num = 181\n",
      "cat 114, mi = 0.00174784, POU5F1_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00163555, PPARGC1A_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00144123, POLR3G_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00143505, BRF2_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00137790, BDP1_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00124974, BRF1_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00100089, SMARCA4_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00075718, XRCC4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.002, pvalue = 0.130, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.002, pvalue = 0.189, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.393, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.263, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.362, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.001, pvalue = 0.537, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.676, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass: cat 167, mi = 0.001, pvalue = 1.000, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.93\t-12.09\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_encode_tf/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 298\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6547 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6547\t298\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "595 hits go_terms_num = 298\n",
      "cat 59, mi = 0.00337304, chr1q32\n",
      "cat 212, mi = 0.00314209, chr3q27\n",
      "cat 77, mi = 0.00291704, chr1p22\n",
      "cat 226, mi = 0.00275680, chr13q33\n",
      "cat 97, mi = 0.00275257, chr1q42\n",
      "cat 230, mi = 0.00254429, chr14q24\n",
      "cat 179, mi = 0.00248025, chr1q43\n",
      "cat 208, mi = 0.00233991, chr22q12\n",
      "cat 278, mi = 0.00225372, chr2q33\n",
      "cat 11, mi = 0.00209235, chr2p24\n",
      "cat 273, mi = 0.00208950, chr4p16\n",
      "cat 147, mi = 0.00206119, chr3p22\n",
      "cat 263, mi = 0.00203670, chr8p11\n",
      "cat 135, mi = 0.00198080, chr20p11\n",
      "cat 64, mi = 0.00195195, chr14q32\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 59, mi = 0.003, pvalue = 0.001, chr1q32\n",
      "Passed: cat 212, mi = 0.003, pvalue = 0.001, chr3q27\n",
      "Passed: cat 77, mi = 0.003, pvalue = 0.004, chr1p22\n",
      "Passed: cat 226, mi = 0.003, pvalue = 0.002, chr13q33\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.003, pvalue = 0.010, chr1q42\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.003, pvalue = 0.018, chr14q24\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.005, chr1q43\n",
      "Didn't pass (cutoff=0.0050): cat 208, mi = 0.002, pvalue = 0.031, chr22q12\n",
      "Didn't pass (cutoff=0.0050): cat 278, mi = 0.002, pvalue = 0.046, chr2q33\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.002, pvalue = 0.017, chr2p24\n",
      "Didn't pass (cutoff=0.0050): cat 273, mi = 0.002, pvalue = 0.056, chr4p16\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.002, pvalue = 0.075, chr3p22\n",
      "Didn't pass (cutoff=0.0050): cat 263, mi = 0.002, pvalue = 0.084, chr8p11\n",
      "Didn't pass (cutoff=0.0050): cat 135, mi = 0.002, pvalue = 0.091, chr20p11\n",
      "Didn't pass (cutoff=0.0050): cat 64, mi = 0.002, pvalue = 0.072, chr14q32\n",
      "Didn't pass (cutoff=0.0050): cat 281, mi = 0.002, pvalue = 0.022, chr2q24\n",
      "Didn't pass (cutoff=0.0050): cat 148, mi = 0.002, pvalue = 0.114, chr5q22\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.002, pvalue = 0.128, chr1q44\n",
      "Didn't pass (cutoff=0.0050): cat 228, mi = 0.002, pvalue = 0.122, chr5q33\n",
      "Didn't pass (cutoff=0.0050): cat 63, mi = 0.002, pvalue = 0.105, chr22q11\n",
      "Didn't pass (cutoff=0.0050): cat 242, mi = 0.002, pvalue = 0.134, chr17q23\n",
      "Didn't pass (cutoff=0.0050): cat 95, mi = 0.002, pvalue = 0.097, chr11q25\n",
      "Didn't pass (cutoff=0.0050): cat 233, mi = 0.002, pvalue = 0.095, chr5q31\n",
      "Didn't pass (cutoff=0.0050): cat 50, mi = 0.002, pvalue = 0.091, chr3p24\n",
      "Total number of categories evaluated = 14.\n",
      "Number of categories that passed the tests = 4\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 3, 0.472176644824334\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.Start drawing\n",
      "4.93\t-12.09\n",
      "chr1q32\n",
      "chr3q27\n",
      "chr1p22\n",
      "chr13q33\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 3, 0.472176644824334\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "Done.-6.58813160987074\t42\t0\n",
      "7.49001175088132\t42\t26.8237367802585\n",
      "8.52643948296122\t42\t28.1562867215041\n",
      "9.1310223266745\t42\t28.8965922444183\n",
      "9.55052878965923\t42\t29.3654524089307\n",
      "9.89600470035253\t42\t29.7356051703878\n",
      "10.2044653349001\t42\t30.0564042303173\n",
      "10.5005875440658\t42\t30.352526439483\n",
      "10.858401880141\t42\t30.6486486486486\n",
      "11.4383078730905\t42\t31.0681551116334\n",
      "16.9042303172738\t42\t31.8084606345476\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_c1/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5529\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6492 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6492\t5529\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "590 hits go_terms_num = 5529\n",
      "cat 4602, mi = 0.00439150, ASTIER_INTEGRIN_SIGNALING\n",
      "cat 3072, mi = 0.00430731, HINATA_NFKB_TARGETS_KERATINOCYTE_UP\n",
      "cat 1665, mi = 0.00405892, HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "cat 1049, mi = 0.00400669, JISON_SICKLE_CELL_DISEASE_UP\n",
      "cat 3990, mi = 0.00382969, BASSO_CD40_SIGNALING_UP\n",
      "cat 2998, mi = 0.00376898, KIM_MYCN_AMPLIFICATION_TARGETS_DN\n",
      "cat 2105, mi = 0.00371848, STARK_HYPPOCAMPUS_22Q11_DELETION_UP\n",
      "cat 3601, mi = 0.00367916, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "cat 3252, mi = 0.00366553, REACTOME_REGULATION_OF_GENE_EXPRESSION_IN_LATE_STAGE_BRANCHING_MORPHOGENESIS_PANCREATIC_BUD_PRECURSOR_CELLS\n",
      "cat 1984, mi = 0.00364572, REACTOME_NOTCH3_INTRACELLULAR_DOMAIN_REGULATES_TRANSCRIPTION\n",
      "cat 2128, mi = 0.00361602, MARSON_FOXP3_TARGETS_UP\n",
      "cat 3978, mi = 0.00358639, REACTOME_TRANSCRIPTIONAL_REGULATION_BY_MECP2\n",
      "cat 2692, mi = 0.00352906, REACTOME_RUNX3_REGULATES_NOTCH_SIGNALING\n",
      "cat 291, mi = 0.00349477, MORI_PRE_BI_LYMPHOCYTE_DN\n",
      "cat 1437, mi = 0.00348585, REACTOME_REGULATION_OF_BETA_CELL_DEVELOPMENT\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 4602, mi = 0.004, pvalue = 0.000, ASTIER_INTEGRIN_SIGNALING\n",
      "Passed: cat 3072, mi = 0.004, pvalue = 0.000, HINATA_NFKB_TARGETS_KERATINOCYTE_UP\n",
      "Passed: cat 1665, mi = 0.004, pvalue = 0.000, HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "Passed: cat 1049, mi = 0.004, pvalue = 0.000, JISON_SICKLE_CELL_DISEASE_UP\n",
      "Passed: cat 2105, mi = 0.004, pvalue = 0.000, STARK_HYPPOCAMPUS_22Q11_DELETION_UP\n",
      "Passed: cat 3601, mi = 0.004, pvalue = 0.001, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "Passed: cat 3252, mi = 0.004, pvalue = 0.000, REACTOME_REGULATION_OF_GENE_EXPRESSION_IN_LATE_STAGE_BRANCHING_MORPHOGENESIS_PANCREATIC_BUD_PRECURSOR_CELLS\n",
      "Passed: cat 205, mi = 0.003, pvalue = 0.001, WILCOX_RESPONSE_TO_PROGESTERONE_DN\n",
      "Passed: cat 58, mi = 0.003, pvalue = 0.001, SIG_BCR_SIGNALING_PATHWAY\n",
      "Passed: cat 4826, mi = 0.003, pvalue = 0.001, MORI_LARGE_PRE_BII_LYMPHOCYTE_UP\n",
      "Passed: cat 4395, mi = 0.003, pvalue = 0.002, AKL_HTLV1_INFECTION_DN\n",
      "Passed: cat 5513, mi = 0.003, pvalue = 0.004, KEGG_LYSOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 1153, mi = 0.003, pvalue = 0.006, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_13\n",
      "Passed: cat 3418, mi = 0.003, pvalue = 0.000, MUELLER_COMMON_TARGETS_OF_AML_FUSIONS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1359, mi = 0.003, pvalue = 0.007, MILI_PSEUDOPODIA\n",
      "Didn't pass (cutoff=0.0050): cat 2348, mi = 0.003, pvalue = 0.006, MULLIGHAN_NPM1_MUTATED_SIGNATURE_1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 765, mi = 0.003, pvalue = 0.006, TAKAO_RESPONSE_TO_UVB_RADIATION_UP\n",
      "Didn't pass (cutoff=0.0050): cat 957, mi = 0.003, pvalue = 0.006, JAZAERI_BREAST_CANCER_BRCA1_VS_BRCA2_UP\n",
      "Passed: cat 753, mi = 0.003, pvalue = 0.004, CHARAFE_BREAST_CANCER_BASAL_VS_MESENCHYMAL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3841, mi = 0.003, pvalue = 0.011, CROMER_METASTASIS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5380, mi = 0.003, pvalue = 0.008, MCCLUNG_CREB1_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1679, mi = 0.003, pvalue = 0.010, SETLUR_PROSTATE_CANCER_TMPRSS2_ERG_FUSION_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1240, mi = 0.003, pvalue = 0.005, OKAWA_NEUROBLASTOMA_1P36_31_DELETION\n",
      "Didn't pass (cutoff=0.0050): cat 3536, mi = 0.003, pvalue = 0.011, MULLIGHAN_NPM1_SIGNATURE_3_DN\n",
      "Didn't pass (cutoff=0.0050): cat 684, mi = 0.003, pvalue = 0.012, ROSS_AML_WITH_PML_RARA_FUSION\n",
      "Didn't pass (cutoff=0.0050): cat 191, mi = 0.003, pvalue = 0.015, VART_KSHV_INFECTION_ANGIOGENIC_MARKERS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1190, mi = 0.003, pvalue = 0.010, MORI_PLASMA_CELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4203, mi = 0.003, pvalue = 0.013, LANDIS_ERBB2_BREAST_TUMORS_324_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1274, mi = 0.003, pvalue = 0.007, KYNG_ENVIRONMENTAL_STRESS_RESPONSE_NOT_BY_GAMMA_IN_OLD\n",
      "Didn't pass (cutoff=0.0050): cat 2394, mi = 0.003, pvalue = 0.009, BOGNI_TREATMENT_RELATED_MYELOID_LEUKEMIA_DN\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.003, pvalue = 0.018, HOFFMANN_SMALL_PRE_BII_TO_IMMATURE_B_LYMPHOCYTE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4206, mi = 0.003, pvalue = 0.012, WHITE_NEUROBLASTOMA_WITH_1P36.3_DELETION\n",
      "Didn't pass (cutoff=0.0050): cat 1281, mi = 0.003, pvalue = 0.011, SARTIPY_BLUNTED_BY_INSULIN_RESISTANCE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4967, mi = 0.003, pvalue = 0.013, REACTOME_PEROXISOMAL_PROTEIN_IMPORT\n",
      "Didn't pass (cutoff=0.0050): cat 4871, mi = 0.003, pvalue = 0.021, MARKEY_RB1_CHRONIC_LOF_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1370, mi = 0.002, pvalue = 0.017, YANG_BREAST_CANCER_ESR1_BULK_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4963, mi = 0.002, pvalue = 0.029, SWEET_KRAS_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3492, mi = 0.002, pvalue = 0.029, XU_HGF_TARGETS_REPRESSED_BY_AKT1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 550, mi = 0.002, pvalue = 0.029, KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_CHONDROITIN_SULFATE\n",
      "Total number of categories evaluated = 138.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 5, 7, 0.0479300283617901\n",
      "best merge is 9, 12, 0.138940732550785\n",
      "best merge is 1, 5, 0.199640410546532\n",
      "best merge is 1, 13, 0.345043565210722\n",
      "best merge is 3, 4, 0.34869907447019\n",
      "best merge is 2, 8, 0.415989007329623\n",
      "best merge is 1, 9, 0.491260023908929\n",
      "best merge is 10, 11, 0.580449410193979\n",
      "best merge is 2, 3, 0.584012445062026\n",
      "best merge is 0, 6, 0.638301454039783\n",
      "order classes\n",
      "6 - 0.719621060784531\n",
      "4 - 1.35657281464596\n",
      "2 - 2.03157293567893\n",
      "1 5 7 13 9 12\n",
      "0\n",
      "2 8 3 4\n",
      "6\n",
      "10 11\n",
      "Done.Start drawing\n",
      "4.93\t-12.09\n",
      "SIG_BCR_SIGNALING_PATHWAY\n",
      "JISON_SICKLE_CELL_DISEASE_UP\n",
      "STARK_HYPPOCAMPUS_22Q11_DELETION_UP\n",
      "ASTIER_INTEGRIN_SIGNALING\n",
      "KEGG_LYSOSOME\n",
      "HINATA_NFKB_TARGETS_KERATINOCYTE_UP\n",
      "LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "WILCOX_RESPONSE_TO_PROGESTERONE_DN\n",
      "MORI_LARGE_PRE_BII_LYMPHOCYTE_UP\n",
      "MUELLER_COMMON_TARGETS_OF_AML_FUSIONS_DN\n",
      "AKL_HTLV1_INFECTION_DN\n",
      "REACTOME_REGULATION_OF_GENE_EXPRESSION_IN_LATE_STAGE_BRANCHING_MORPHOGENESIS_PANCREATIC_BUD_PRECURSOR_CELLS\n",
      "CHARAFE_BREAST_CANCER_BASAL_VS_MESENCHYMAL_UP\n",
      "HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 5, 7, 0.0479300283617901\n",
      "best merge is 9, 12, 0.138940732550785\n",
      "best merge is 1, 5, 0.199640410546532\n",
      "best merge is 1, 13, 0.345043565210722\n",
      "best merge is 3, 4, 0.34869907447019\n",
      "best merge is 2, 8, 0.415989007329623\n",
      "best merge is 1, 9, 0.491260023908929\n",
      "best merge is 10, 11, 0.580449410193979\n",
      "best merge is 2, 3, 0.584012445062026\n",
      "best merge is 0, 6, 0.638301454039783\n",
      "order classes\n",
      "6 - 0.719621060784531\n",
      "4 - 1.35657281464596\n",
      "2 - 2.03157293567893\n",
      "1 5 7 13 9 12\n",
      "0\n",
      "2 8 3 4\n",
      "6\n",
      "10 11\n",
      "Done.-6.58813160987074\t42\t0\n",
      "7.49001175088132\t42\t26.8237367802585\n",
      "8.53877790834313\t42\t28.1562867215041\n",
      "9.15569917743831\t42\t28.9212690951821\n",
      "9.56286721504113\t42\t29.3901292596945\n",
      "9.89600470035253\t42\t29.7356051703878\n",
      "10.2044653349001\t42\t30.0564042303173\n",
      "10.5005875440658\t42\t30.352526439483\n",
      "10.858401880141\t42\t30.6486486486486\n",
      "11.4259694477086\t42\t31.0681551116334\n",
      "16.8918918918919\t42\t31.7837837837838\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_c2/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 3733\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6535 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6535\t3733\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "594 hits go_terms_num = 3733\n",
      "cat 467, mi = 0.00383995, GTGGTGA_MIR197\n",
      "cat 533, mi = 0.00347874, MIR31_5P\n",
      "cat 207, mi = 0.00339311, MIR7976\n",
      "cat 637, mi = 0.00334362, TGTGTGA_MIR377\n",
      "cat 1002, mi = 0.00322345, MIR3678_3P\n",
      "cat 876, mi = 0.00313426, ZNF239_TARGET_GENES\n",
      "cat 3361, mi = 0.00313361, MIR455_5P\n",
      "cat 839, mi = 0.00308489, AGGCACT_MIR5153P\n",
      "cat 3433, mi = 0.00306914, MIR1202\n",
      "cat 1635, mi = 0.00303504, MIR1264\n",
      "cat 28, mi = 0.00301570, MIR3972\n",
      "cat 2870, mi = 0.00299764, MIR342_5P\n",
      "cat 703, mi = 0.00298247, MIR3616_5P\n",
      "cat 326, mi = 0.00298026, MIR376B_5P_MIR376C_5P\n",
      "cat 480, mi = 0.00297894, ACCAATC_MIR509\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 467, mi = 0.004, pvalue = 0.000, GTGGTGA_MIR197\n",
      "Passed: cat 533, mi = 0.003, pvalue = 0.001, MIR31_5P\n",
      "Passed: cat 207, mi = 0.003, pvalue = 0.001, MIR7976\n",
      "Passed: cat 1002, mi = 0.003, pvalue = 0.002, MIR3678_3P\n",
      "Passed: cat 876, mi = 0.003, pvalue = 0.001, ZNF239_TARGET_GENES\n",
      "Passed: cat 3361, mi = 0.003, pvalue = 0.003, MIR455_5P\n",
      "Passed: cat 839, mi = 0.003, pvalue = 0.003, AGGCACT_MIR5153P\n",
      "Passed: cat 3433, mi = 0.003, pvalue = 0.004, MIR1202\n",
      "Passed: cat 480, mi = 0.003, pvalue = 0.005, ACCAATC_MIR509\n",
      "Didn't pass (cutoff=0.0050): cat 496, mi = 0.003, pvalue = 0.005, GCGSCMNTTT_UNKNOWN\n",
      "Passed: cat 946, mi = 0.003, pvalue = 0.004, MIR7847_3P\n",
      "Didn't pass (cutoff=0.0050): cat 258, mi = 0.003, pvalue = 0.009, MIR617\n",
      "Didn't pass (cutoff=0.0050): cat 481, mi = 0.003, pvalue = 0.006, HFH3_01\n",
      "Didn't pass (cutoff=0.0050): cat 2199, mi = 0.003, pvalue = 0.013, MIR875_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2477, mi = 0.003, pvalue = 0.018, MIR887_5P\n",
      "Didn't pass (cutoff=0.0050): cat 3437, mi = 0.003, pvalue = 0.012, MIR1911_5P\n",
      "Didn't pass (cutoff=0.0050): cat 807, mi = 0.003, pvalue = 0.014, MIR623\n",
      "Didn't pass (cutoff=0.0050): cat 1513, mi = 0.003, pvalue = 0.017, MIR625_3P\n",
      "Didn't pass (cutoff=0.0050): cat 2491, mi = 0.003, pvalue = 0.019, PPARA_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 655, mi = 0.003, pvalue = 0.008, MIR4701_3P\n",
      "Didn't pass (cutoff=0.0050): cat 942, mi = 0.003, pvalue = 0.011, MIR1262\n",
      "Didn't pass (cutoff=0.0050): cat 1671, mi = 0.003, pvalue = 0.020, MIR6740_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1126, mi = 0.002, pvalue = 0.007, MIR6738_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1035, mi = 0.002, pvalue = 0.020, MIR345_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2042, mi = 0.002, pvalue = 0.023, TTCNRGNNNNTTC_HSF_Q6\n",
      "Didn't pass (cutoff=0.0050): cat 1695, mi = 0.002, pvalue = 0.024, MIR548AL\n",
      "Didn't pass (cutoff=0.0050): cat 3411, mi = 0.002, pvalue = 0.024, MIR4515\n",
      "Didn't pass (cutoff=0.0050): cat 979, mi = 0.002, pvalue = 0.022, MIR5591_3P\n",
      "Didn't pass (cutoff=0.0050): cat 3250, mi = 0.002, pvalue = 0.029, CTCNANGTGNY_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 938, mi = 0.002, pvalue = 0.036, MIR1587\n",
      "Didn't pass (cutoff=0.0050): cat 3039, mi = 0.002, pvalue = 0.034, MIR3620_5P\n",
      "Total number of categories evaluated = 72.\n",
      "Number of categories that passed the tests = 10\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 8, 0.169325349982117\n",
      "best merge is 4, 5, 0.36626797176466\n",
      "best merge is 6, 7, 0.451085536866046\n",
      "best merge is 1, 2, 0.451794354649641\n",
      "best merge is 0, 1, 0.523413954118637\n",
      "best merge is 3, 4, 0.723794145708774\n",
      "order classes\n",
      "4 - 0.848367350345721\n",
      "2 - 0.915669929411649\n",
      "2 - 1.12771384216511\n",
      "0 8 1 2\n",
      "3\n",
      "4 5\n",
      "6 7\n",
      "9\n",
      "Done.Start drawing\n",
      "4.93\t-12.09\n",
      "ZNF239_TARGET_GENES\n",
      "MIR455_5P\n",
      "AGGCACT_MIR5153P\n",
      "MIR7976\n",
      "GTGGTGA_MIR197\n",
      "ACCAATC_MIR509\n",
      "MIR31_5P\n",
      "MIR1202\n",
      "MIR3678_3P\n",
      "MIR7847_3P\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 8, 0.169325349982117\n",
      "best merge is 4, 5, 0.36626797176466\n",
      "best merge is 6, 7, 0.451085536866046\n",
      "best merge is 1, 2, 0.451794354649641\n",
      "best merge is 0, 1, 0.523413954118637\n",
      "best merge is 3, 4, 0.723794145708774\n",
      "order classes\n",
      "4 - 0.848367350345721\n",
      "2 - 0.915669929411649\n",
      "2 - 1.12771384216511\n",
      "0 8 1 2\n",
      "3\n",
      "4 5\n",
      "6 7\n",
      "9\n",
      "Done.-6.58813160987074\t42\t0\n",
      "7.49001175088132\t42\t26.8237367802585\n",
      "8.52643948296122\t42\t28.1562867215041\n",
      "9.1310223266745\t42\t28.8965922444183\n",
      "9.55052878965923\t42\t29.3654524089307\n",
      "9.89600470035253\t42\t29.7356051703878\n",
      "10.2044653349001\t42\t30.0564042303173\n",
      "10.5005875440658\t42\t30.352526439483\n",
      "10.858401880141\t42\t30.6486486486486\n",
      "11.4259694477086\t42\t31.0681551116334\n",
      "16.8918918918919\t42\t31.7837837837838\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_c3/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3839 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3839\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "349 hits go_terms_num = 858\n",
      "cat 327, mi = 0.00578988, MODULE_183\n",
      "cat 182, mi = 0.00562061, GNF2_ATM\n",
      "cat 698, mi = 0.00543835, MORF_EIF3S6\n",
      "cat 781, mi = 0.00522117, MORF_FLT1\n",
      "cat 424, mi = 0.00511883, GNF2_SNRK\n",
      "cat 564, mi = 0.00495426, GNF2_RRM1\n",
      "cat 230, mi = 0.00484821, GNF2_CENPF\n",
      "cat 120, mi = 0.00484058, MORF_DMPK\n",
      "cat 679, mi = 0.00472571, GNF2_PTPRC\n",
      "cat 741, mi = 0.00469998, MORF_LTK\n",
      "cat 228, mi = 0.00461188, GCM_CASP2\n",
      "cat 839, mi = 0.00460370, MORF_CSNK1D\n",
      "cat 459, mi = 0.00453008, MODULE_397\n",
      "cat 381, mi = 0.00447444, MORF_PHB\n",
      "cat 549, mi = 0.00446156, GNF2_CCNA2\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 327, mi = 0.006, pvalue = 0.002, MODULE_183\n",
      "Passed: cat 182, mi = 0.006, pvalue = 0.002, GNF2_ATM\n",
      "Passed: cat 781, mi = 0.005, pvalue = 0.004, MORF_FLT1\n",
      "Didn't pass (cutoff=0.0050): cat 564, mi = 0.005, pvalue = 0.005, GNF2_RRM1\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.005, pvalue = 0.007, GNF2_CENPF\n",
      "Didn't pass (cutoff=0.0050): cat 228, mi = 0.005, pvalue = 0.012, GCM_CASP2\n",
      "Didn't pass (cutoff=0.0050): cat 839, mi = 0.005, pvalue = 0.014, MORF_CSNK1D\n",
      "Didn't pass (cutoff=0.0050): cat 459, mi = 0.005, pvalue = 0.011, MODULE_397\n",
      "Didn't pass (cutoff=0.0050): cat 381, mi = 0.004, pvalue = 0.015, MORF_PHB\n",
      "Didn't pass (cutoff=0.0050): cat 549, mi = 0.004, pvalue = 0.014, GNF2_CCNA2\n",
      "Didn't pass (cutoff=0.0050): cat 665, mi = 0.004, pvalue = 0.008, MODULE_222\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.004, pvalue = 0.018, GCM_RAB10\n",
      "Didn't pass (cutoff=0.0050): cat 443, mi = 0.004, pvalue = 0.029, GNF2_CDC2\n",
      "Didn't pass (cutoff=0.0050): cat 153, mi = 0.004, pvalue = 0.030, GNF2_CENPE\n",
      "Didn't pass (cutoff=0.0050): cat 332, mi = 0.004, pvalue = 0.028, GNF2_DDX5\n",
      "Didn't pass (cutoff=0.0050): cat 174, mi = 0.004, pvalue = 0.035, GNF2_CD97\n",
      "Didn't pass (cutoff=0.0050): cat 162, mi = 0.004, pvalue = 0.024, MORF_NME2\n",
      "Didn't pass (cutoff=0.0050): cat 714, mi = 0.004, pvalue = 0.040, GNF2_PCNA\n",
      "Didn't pass (cutoff=0.0050): cat 79, mi = 0.004, pvalue = 0.034, MORF_JUND\n",
      "Didn't pass (cutoff=0.0050): cat 241, mi = 0.004, pvalue = 0.031, MODULE_524\n",
      "Didn't pass (cutoff=0.0050): cat 739, mi = 0.004, pvalue = 0.031, MODULE_419\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.004, pvalue = 0.042, GCM_NUMA1\n",
      "Didn't pass (cutoff=0.0050): cat 352, mi = 0.004, pvalue = 0.048, GCM_BECN1\n",
      "Total number of categories evaluated = 29.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.741384186872881\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "4.93\t-9.60\n",
      "GNF2_ATM\n",
      "MODULE_183\n",
      "MORF_FLT1\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.741384186872881\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-7.70061940812113\t42\t0\n",
      "5.29249827942188\t42\t24.5987611837577\n",
      "6.40536820371645\t42\t25.986235375086\n",
      "7.08465244322092\t42\t26.8245010323469\n",
      "7.53269098417068\t42\t27.344803854095\n",
      "7.87955953200275\t42\t27.7205781142464\n",
      "8.21197522367515\t42\t28.0385409497591\n",
      "8.55884377150723\t42\t28.3854094975912\n",
      "8.94907088781831\t42\t28.7322780454233\n",
      "9.55609084652443\t42\t29.1658637302134\n",
      "15.9731589814178\t42\t29.9463179628355\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_c4/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 10191\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 5908 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 5908\t10191\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "537 hits go_terms_num = 10191\n",
      "cat 4640, mi = 0.00388956, GO_EXTRINSIC_COMPONENT_OF_PLASMA_MEMBRANE\n",
      "cat 7432, mi = 0.00365184, GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "cat 1145, mi = 0.00361172, GO_DOSAGE_COMPENSATION\n",
      "cat 2971, mi = 0.00360087, GO_POSITIVE_REGULATION_OF_NOTCH_SIGNALING_PATHWAY\n",
      "cat 4298, mi = 0.00357128, GO_HSP90_PROTEIN_BINDING\n",
      "cat 4655, mi = 0.00351678, GO_NEGATIVE_REGULATION_OF_MUSCLE_CELL_DIFFERENTIATION\n",
      "cat 7791, mi = 0.00351620, GO_POLYSOME\n",
      "cat 1838, mi = 0.00351416, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "cat 6866, mi = 0.00349647, GO_P53_BINDING\n",
      "cat 8794, mi = 0.00344209, GO_REGULATION_OF_TUBE_SIZE\n",
      "cat 2959, mi = 0.00341797, GO_OOCYTE_DIFFERENTIATION\n",
      "cat 5221, mi = 0.00340672, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_PLASMA_MEMBRANE\n",
      "cat 2240, mi = 0.00339232, GO_SPERMATID_DIFFERENTIATION\n",
      "cat 5040, mi = 0.00337879, GO_POSITIVE_REGULATION_OF_MULTICELLULAR_ORGANISM_GROWTH\n",
      "cat 9861, mi = 0.00333415, GO_SPECIFIC_GRANULE_MEMBRANE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 4640, mi = 0.004, pvalue = 0.001, GO_EXTRINSIC_COMPONENT_OF_PLASMA_MEMBRANE\n",
      "Passed: cat 7432, mi = 0.004, pvalue = 0.002, GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "Passed: cat 1145, mi = 0.004, pvalue = 0.000, GO_DOSAGE_COMPENSATION\n",
      "Passed: cat 4298, mi = 0.004, pvalue = 0.001, GO_HSP90_PROTEIN_BINDING\n",
      "Passed: cat 4655, mi = 0.004, pvalue = 0.003, GO_NEGATIVE_REGULATION_OF_MUSCLE_CELL_DIFFERENTIATION\n",
      "Passed: cat 7791, mi = 0.004, pvalue = 0.003, GO_POLYSOME\n",
      "Passed: cat 1838, mi = 0.004, pvalue = 0.002, GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "Passed: cat 8794, mi = 0.003, pvalue = 0.001, GO_REGULATION_OF_TUBE_SIZE\n",
      "Passed: cat 2959, mi = 0.003, pvalue = 0.002, GO_OOCYTE_DIFFERENTIATION\n",
      "Passed: cat 5221, mi = 0.003, pvalue = 0.003, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_PLASMA_MEMBRANE\n",
      "Passed: cat 2240, mi = 0.003, pvalue = 0.003, GO_SPERMATID_DIFFERENTIATION\n",
      "Passed: cat 9861, mi = 0.003, pvalue = 0.004, GO_SPECIFIC_GRANULE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 7054, mi = 0.003, pvalue = 0.007, GO_RNA_POLYMERASE_II_ACTIVATING_TRANSCRIPTION_FACTOR_BINDING\n",
      "Passed: cat 652, mi = 0.003, pvalue = 0.004, GO_SYNAPTIC_VESICLE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 9522, mi = 0.003, pvalue = 0.006, GO_REGULATION_OF_STEM_CELL_POPULATION_MAINTENANCE\n",
      "Didn't pass (cutoff=0.0050): cat 6079, mi = 0.003, pvalue = 0.007, GO_PHOSPHATIDYLINOSITOL_3_4_BISPHOSPHATE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 9491, mi = 0.003, pvalue = 0.008, GO_ATPASE_COMPLEX\n",
      "Passed: cat 7999, mi = 0.003, pvalue = 0.004, GO_KERATAN_SULFATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 9615, mi = 0.003, pvalue = 0.013, GO_EPIDERMAL_GROWTH_FACTOR_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2786, mi = 0.003, pvalue = 0.009, GO_REGULATION_OF_COFACTOR_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1634, mi = 0.003, pvalue = 0.014, GO_RESPONSE_TO_OSMOTIC_STRESS\n",
      "Didn't pass (cutoff=0.0050): cat 151, mi = 0.003, pvalue = 0.011, GO_CYTOSOLIC_TRANSPORT\n",
      "Passed: cat 8254, mi = 0.003, pvalue = 0.002, GO_PHOSPHATIDYLINOSITOL_4_5_BISPHOSPHATE_PHOSPHATASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 9527, mi = 0.003, pvalue = 0.015, GO_PERIPHERAL_NERVOUS_SYSTEM_DEVELOPMENT\n",
      "Passed: cat 4643, mi = 0.003, pvalue = 0.004, GO_ANION_HOMEOSTASIS\n",
      "Didn't pass (cutoff=0.0050): cat 7339, mi = 0.003, pvalue = 0.015, GO_COSTAMERE\n",
      "Didn't pass (cutoff=0.0050): cat 7905, mi = 0.003, pvalue = 0.008, GO_INNERVATION\n",
      "Didn't pass (cutoff=0.0050): cat 4175, mi = 0.003, pvalue = 0.017, GO_DERMATAN_SULFATE_PROTEOGLYCAN_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 9412, mi = 0.003, pvalue = 0.022, GO_BONE_CELL_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 5470, mi = 0.003, pvalue = 0.008, GO_HISTONE_PHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 4879, mi = 0.003, pvalue = 0.022, GO_RETROGRADE_TRANSPORT_ENDOSOME_TO_GOLGI\n",
      "Didn't pass (cutoff=0.0050): cat 7975, mi = 0.003, pvalue = 0.019, GO_DETECTION_OF_LIGHT_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 2642, mi = 0.003, pvalue = 0.028, GO_LYASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 9836, mi = 0.003, pvalue = 0.026, GO_NERVE_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 7119, mi = 0.003, pvalue = 0.013, GO_GLYCOSYL_COMPOUND_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1560, mi = 0.003, pvalue = 0.028, GO_REGULATION_OF_STEROID_BIOSYNTHETIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3096, mi = 0.003, pvalue = 0.014, GO_PEPTIDYL_SERINE_DEPHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 3550, mi = 0.003, pvalue = 0.030, GO_N_TERMINAL_PROTEIN_AMINO_ACID_ACETYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 6758, mi = 0.003, pvalue = 0.017, GO_NBAF_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 9030, mi = 0.003, pvalue = 0.034, GO_9PLUS0_NON_MOTILE_CILIUM\n",
      "Didn't pass (cutoff=0.0050): cat 2872, mi = 0.003, pvalue = 0.007, GO_COLLAGEN_TRIMER\n",
      "Didn't pass (cutoff=0.0050): cat 640, mi = 0.003, pvalue = 0.010, GO_POSITIVE_REGULATION_OF_SIGNAL_TRANSDUCTION_BY_P53_CLASS_MEDIATOR\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.003, pvalue = 0.035, GO_LYTIC_VACUOLE_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 1660, mi = 0.003, pvalue = 0.032, GO_NEGATIVE_REGULATION_OF_RESPONSE_TO_ENDOPLASMIC_RETICULUM_STRESS\n",
      "Didn't pass (cutoff=0.0050): cat 5909, mi = 0.003, pvalue = 0.033, GO_MYELOID_CELL_DEVELOPMENT\n",
      "Total number of categories evaluated = 128.\n",
      "Number of categories that passed the tests = 16\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 8, 0.269704091991648\n",
      "best merge is 2, 9, 0.309614241032028\n",
      "best merge is 3, 10, 0.326025160330259\n",
      "best merge is 1, 6, 0.329630895968833\n",
      "best merge is 3, 14, 0.499406536077557\n",
      "best merge is 13, 15, 0.555251835969107\n",
      "best merge is 4, 11, 0.56797469329863\n",
      "best merge is 0, 13, 0.637865418498849\n",
      "best merge is 1, 2, 0.638451709761027\n",
      "best merge is 7, 12, 0.714848813104364\n",
      "best merge is 1, 5, 0.774559281577864\n",
      "best merge is 4, 7, 0.920905229686678\n",
      "order classes\n",
      "4 - 1.68820880097808\n",
      "5 - 1.61057001020325\n",
      "3 - 1.57017864590859\n",
      "2 - 2.27189877319452\n",
      "2 - 2.85939525241746\n",
      "3 10 14\n",
      "1 6 2 9 5\n",
      "0 8 13 15\n",
      "4 11\n",
      "7 12\n",
      "Done.Start drawing\n",
      "4.93\t-12.09\n",
      "GO_REGULATION_OF_TUBE_SIZE\n",
      "GO_OOCYTE_DIFFERENTIATION\n",
      "GO_ANION_HOMEOSTASIS\n",
      "GO_EXTRINSIC_COMPONENT_OF_PLASMA_MEMBRANE\n",
      "GO_SPECIFIC_GRANULE_MEMBRANE\n",
      "GO_NEGATIVE_REGULATION_OF_MUSCLE_CELL_DIFFERENTIATION\n",
      "GO_POLYSOME\n",
      "GO_LARGE_RIBOSOMAL_SUBUNIT\n",
      "GO_DOSAGE_COMPENSATION\n",
      "GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_PLASMA_MEMBRANE\n",
      "GO_NEGATIVE_REGULATION_OF_TYPE_I_INTERFERON_PRODUCTION\n",
      "GO_KERATAN_SULFATE_METABOLIC_PROCESS\n",
      "GO_SYNAPTIC_VESICLE_MEMBRANE\n",
      "GO_HSP90_PROTEIN_BINDING\n",
      "GO_SPERMATID_DIFFERENTIATION\n",
      "GO_PHOSPHATIDYLINOSITOL_4_5_BISPHOSPHATE_PHOSPHATASE_ACTIVITY\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 8, 0.269704091991648\n",
      "best merge is 2, 9, 0.309614241032028\n",
      "best merge is 3, 10, 0.326025160330259\n",
      "best merge is 1, 6, 0.329630895968833\n",
      "best merge is 3, 14, 0.499406536077557\n",
      "best merge is 13, 15, 0.555251835969107\n",
      "best merge is 4, 11, 0.56797469329863\n",
      "best merge is 0, 13, 0.637865418498849\n",
      "best merge is 1, 2, 0.638451709761027\n",
      "best merge is 7, 12, 0.714848813104364\n",
      "best merge is 1, 5, 0.774559281577864\n",
      "best merge is 4, 7, 0.920905229686678\n",
      "order classes\n",
      "4 - 1.68820880097808\n",
      "5 - 1.61057001020325\n",
      "3 - 1.57017864590859\n",
      "2 - 2.27189877319452\n",
      "2 - 2.85939525241746\n",
      "3 10 14\n",
      "1 6 2 9 5\n",
      "0 8 13 15\n",
      "4 11\n",
      "7 12\n",
      "Done.-6.56345475910693\t42\t0\n",
      "7.52702702702703\t42\t26.8730904817861\n",
      "8.55111633372503\t42\t28.1809635722679\n",
      "9.16803760282021\t42\t28.9212690951821\n",
      "9.58754406580493\t42\t29.4148061104583\n",
      "9.90834312573443\t42\t29.7602820211516\n",
      "10.2044653349001\t42\t30.0564042303173\n",
      "10.5005875440658\t42\t30.352526439483\n",
      "10.8707403055229\t42\t30.6486486486486\n",
      "11.4383078730905\t42\t31.0928319623972\n",
      "16.8918918918919\t42\t31.7837837837838\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_c5/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3784 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3784\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "344 hits go_terms_num = 189\n",
      "cat 20, mi = 0.00734600, TBK1.DN.48HRS_UP\n",
      "cat 72, mi = 0.00615505, KRAS.LUNG_UP.V1_UP\n",
      "cat 60, mi = 0.00455052, JAK2_DN.V1_DN\n",
      "cat 3, mi = 0.00453251, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 52, mi = 0.00448722, EGFR_UP.V1_UP\n",
      "cat 146, mi = 0.00418845, CAHOY_ASTROGLIAL\n",
      "cat 40, mi = 0.00415694, ATM_DN.V1_UP\n",
      "cat 69, mi = 0.00407578, E2F3_UP.V1_UP\n",
      "cat 153, mi = 0.00403212, VEGF_A_UP.V1_DN\n",
      "cat 169, mi = 0.00395270, RB_P130_DN.V1_UP\n",
      "cat 14, mi = 0.00393935, KRAS.300_UP.V1_DN\n",
      "cat 155, mi = 0.00393478, PDGF_UP.V1_DN\n",
      "cat 127, mi = 0.00388368, CRX_DN.V1_UP\n",
      "cat 135, mi = 0.00388132, P53_DN.V2_UP\n",
      "cat 34, mi = 0.00381954, E2F1_UP.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 20, mi = 0.007, pvalue = 0.000, TBK1.DN.48HRS_UP\n",
      "Passed: cat 72, mi = 0.006, pvalue = 0.000, KRAS.LUNG_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 60, mi = 0.005, pvalue = 0.014, JAK2_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 52, mi = 0.004, pvalue = 0.013, EGFR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 146, mi = 0.004, pvalue = 0.030, CAHOY_ASTROGLIAL\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.004, pvalue = 0.029, ATM_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.004, pvalue = 0.022, E2F3_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 153, mi = 0.004, pvalue = 0.031, VEGF_A_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.004, pvalue = 0.042, RB_P130_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.004, pvalue = 0.040, KRAS.300_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 127, mi = 0.004, pvalue = 0.050, CRX_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 136, mi = 0.004, pvalue = 0.039, CYCLIN_D1_KE_.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.004, pvalue = 0.060, CRX_NRL_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.004, pvalue = 0.080, MEL18_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 172, mi = 0.004, pvalue = 0.081, KRAS.PROSTATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.004, pvalue = 0.051, RAF_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1, mi = 0.004, pvalue = 0.073, NOTCH_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 78, mi = 0.003, pvalue = 0.093, BMI1_DN_MEL18_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.003, pvalue = 0.066, GCNP_SHH_UP_LATE.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 29, mi = 0.003, pvalue = 0.119, BRCA1_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 49, mi = 0.003, pvalue = 0.133, BCAT_BILD_ET_AL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.003, pvalue = 0.108, KRAS.BREAST_UP.V1_DN\n",
      "Total number of categories evaluated = 19.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.93\t-12.09\n",
      "TBK1.DN.48HRS_UP\n",
      "KRAS.LUNG_UP.V1_UP\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.52643948296122\t42\t0\n",
      "7.56404230317274\t42\t26.9471210340776\n",
      "8.55111633372503\t42\t28.1809635722679\n",
      "9.15569917743831\t42\t28.9212690951821\n",
      "9.58754406580493\t42\t29.3901292596945\n",
      "9.93301997649824\t42\t29.7849588719154\n",
      "10.2291421856639\t42\t30.0810810810811\n",
      "10.5252643948296\t42\t30.3772032902468\n",
      "10.8954171562867\t42\t30.6733254994125\n",
      "11.4629847238543\t42\t31.117508813161\n",
      "16.9042303172738\t42\t31.8084606345476\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_c6/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6489 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6489\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "590 hits go_terms_num = 4872\n",
      "cat 2361, mi = 0.00532191, GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_UP\n",
      "cat 3178, mi = 0.00462480, GSE27896_HDAC6_KO_VS_WT_TREG_UP\n",
      "cat 4382, mi = 0.00459278, GSE9946_MATURE_STIMULATORY_VS_LISTERIA_INF_MATURE_DC_UP\n",
      "cat 1975, mi = 0.00456230, GSE2405_0H_VS_12H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "cat 3689, mi = 0.00453751, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "cat 3588, mi = 0.00433264, GSE43863_NAIVE_VS_MEMORY_LY6C_INT_CXCR5POS_CD4_TCELL_D150_LCMV_UP\n",
      "cat 2169, mi = 0.00415057, GSE9988_ANTI_TREM1_VS_ANTI_TREM1_AND_LPS_MONOCYTE_UP\n",
      "cat 3, mi = 0.00414944, GSE19401_NAIVE_VS_IMMUNIZED_MOUSE_PLN_FOLLICULAR_DC_UP\n",
      "cat 2698, mi = 0.00409466, GSE21360_PRIMARY_VS_QUATERNARY_MEMORY_CD8_TCELL_UP\n",
      "cat 4622, mi = 0.00395178, GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "cat 2540, mi = 0.00393851, GSE18791_CTRL_VS_NEWCASTLE_VIRUS_DC_6H_DN\n",
      "cat 730, mi = 0.00393589, GSE28726_NAIVE_VS_ACTIVATED_CD4_TCELL_UP\n",
      "cat 4016, mi = 0.00391203, GSE21360_NAIVE_VS_SECONDARY_MEMORY_CD8_TCELL_DN\n",
      "cat 3517, mi = 0.00387202, GSE14000_UNSTIM_VS_4H_LPS_DC_DN\n",
      "cat 2308, mi = 0.00386062, GSE31082_CD4_VS_CD8_SP_THYMOCYTE_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2361, mi = 0.005, pvalue = 0.000, GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_UP\n",
      "Passed: cat 3178, mi = 0.005, pvalue = 0.000, GSE27896_HDAC6_KO_VS_WT_TREG_UP\n",
      "Passed: cat 4382, mi = 0.005, pvalue = 0.000, GSE9946_MATURE_STIMULATORY_VS_LISTERIA_INF_MATURE_DC_UP\n",
      "Passed: cat 3689, mi = 0.005, pvalue = 0.000, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "Passed: cat 2169, mi = 0.004, pvalue = 0.000, GSE9988_ANTI_TREM1_VS_ANTI_TREM1_AND_LPS_MONOCYTE_UP\n",
      "Passed: cat 3, mi = 0.004, pvalue = 0.000, GSE19401_NAIVE_VS_IMMUNIZED_MOUSE_PLN_FOLLICULAR_DC_UP\n",
      "Passed: cat 4622, mi = 0.004, pvalue = 0.000, GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "Passed: cat 2308, mi = 0.004, pvalue = 0.000, GSE31082_CD4_VS_CD8_SP_THYMOCYTE_UP\n",
      "Passed: cat 1680, mi = 0.004, pvalue = 0.000, GSE43863_DAY6_EFF_VS_DAY150_MEM_LY6C_INT_CXCR5POS_CD4_TCELL_DN\n",
      "Passed: cat 4101, mi = 0.004, pvalue = 0.000, GSE27786_LSK_VS_CD8_TCELL_DN\n",
      "Passed: cat 1108, mi = 0.004, pvalue = 0.001, GSE33425_CD161_HIGH_VS_INT_CD8_TCELL_DN\n",
      "Passed: cat 696, mi = 0.003, pvalue = 0.001, GSE32255_UNSTIM_VS_4H_LPS_STIM_DC_DN\n",
      "Passed: cat 3654, mi = 0.003, pvalue = 0.002, GSE42724_MEMORY_VS_B1_BCELL_DN\n",
      "Passed: cat 3439, mi = 0.003, pvalue = 0.003, GSE12003_MIR223_KO_VS_WT_BM_PROGENITOR_8D_CULTURE_UP\n",
      "Passed: cat 2858, mi = 0.003, pvalue = 0.001, GSE16450_CTRL_VS_IFNA_6H_STIM_MATURE_NEURON_CELL_LINE_DN\n",
      "Passed: cat 3426, mi = 0.003, pvalue = 0.003, GSE14415_ACT_VS_CTRL_NATURAL_TREG_UP\n",
      "Passed: cat 1394, mi = 0.003, pvalue = 0.003, GSE2585_CTEC_VS_MTEC_THYMUS_DN\n",
      "Passed: cat 3647, mi = 0.003, pvalue = 0.003, GSE3720_VD1_VS_VD2_GAMMADELTA_TCELL_UP\n",
      "Passed: cat 2782, mi = 0.003, pvalue = 0.003, GSE45365_HEALTHY_VS_MCMV_INFECTION_BCELL_IFNAR_KO_DN\n",
      "Passed: cat 821, mi = 0.003, pvalue = 0.003, GSE2770_TGFB_AND_IL4_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Passed: cat 1735, mi = 0.003, pvalue = 0.003, GSE360_LOW_DOSE_B_MALAYI_VS_M_TUBERCULOSIS_MAC_UP\n",
      "Passed: cat 2665, mi = 0.003, pvalue = 0.002, GSE32255_WT_UNSTIM_VS_JMJD2D_KNOCKDOWN_4H_LPS_STIM_DC_UP\n",
      "Passed: cat 2979, mi = 0.003, pvalue = 0.003, GSE29949_CD8_NEG_DC_SPLEEN_VS_DC_BRAIN_UP\n",
      "Passed: cat 1420, mi = 0.003, pvalue = 0.004, GSE22601_DOUBLE_NEGATIVE_VS_IMMATURE_CD4_SP_THYMOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1310, mi = 0.003, pvalue = 0.005, GSE3982_EOSINOPHIL_VS_BASOPHIL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3565, mi = 0.003, pvalue = 0.010, GSE13411_IGM_VS_SWITCHED_MEMORY_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 713, mi = 0.003, pvalue = 0.012, GSE32423_MEMORY_VS_NAIVE_CD8_TCELL_IL7_IL4_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2131, mi = 0.003, pvalue = 0.011, GSE12839_CTRL_VS_IL12_TREATED_PBMC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3486, mi = 0.003, pvalue = 0.011, GSE20484_MCSG_VS_CXCL4_MONOCYTE_DERIVED_MACROPHAGE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4022, mi = 0.003, pvalue = 0.008, GSE28726_NAIVE_VS_ACTIVATED_VA24NEG_NKTCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2017, mi = 0.003, pvalue = 0.011, GSE37301_LYMPHOID_PRIMED_MPP_VS_COMMON_LYMPHOID_PROGENITOR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3436, mi = 0.003, pvalue = 0.014, GSE20198_UNTREATED_VS_IL12_IL18_TREATED_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 222, mi = 0.003, pvalue = 0.015, GSE37532_WT_VS_PPARG_KO_VISCERAL_ADIPOSE_TISSUE_TREG_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.003, pvalue = 0.013, GSE2405_0H_VS_1.5H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4252, mi = 0.003, pvalue = 0.017, GSE17721_LPS_VS_CPG_2H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1208, mi = 0.003, pvalue = 0.014, GSE8621_LPS_PRIMED_UNSTIM_VS_LPS_PRIMED_AND_LPS_STIM_MACROPHAGE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4751, mi = 0.003, pvalue = 0.021, GSE3982_EOSINOPHIL_VS_TH1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4165, mi = 0.002, pvalue = 0.023, GSE4142_PLASMA_CELL_VS_GC_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2797, mi = 0.002, pvalue = 0.027, GSE17721_LPS_VS_POLYIC_24H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 871, mi = 0.002, pvalue = 0.019, GSE17721_CTRL_VS_CPG_1H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3165, mi = 0.002, pvalue = 0.023, GSE14769_UNSTIM_VS_40MIN_LPS_BMDM_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1852, mi = 0.002, pvalue = 0.025, GSE25087_TREG_VS_TCONV_FETUS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4393, mi = 0.002, pvalue = 0.032, GSE41176_UNSTIM_VS_ANTI_IGM_STIM_TAK1_KO_BCELL_3H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3079, mi = 0.002, pvalue = 0.039, GSE6566_STRONG_VS_WEAK_DC_STIMULATED_CD4_TCELL_UP\n",
      "Total number of categories evaluated = 285.\n",
      "Number of categories that passed the tests = 24\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 5, 0.177271420963679\n",
      "best merge is 6, 19, 0.198842964159118\n",
      "best merge is 8, 14, 0.229729040882629\n",
      "best merge is 3, 9, 0.244280640997698\n",
      "best merge is 1, 4, 0.26975310173209\n",
      "best merge is 2, 7, 0.290213203194628\n",
      "best merge is 8, 13, 0.332026245686594\n",
      "best merge is 6, 18, 0.433374055793285\n",
      "best merge is 1, 12, 0.436476534561607\n",
      "best merge is 0, 21, 0.465393302855107\n",
      "best merge is 20, 23, 0.472132966126977\n",
      "best merge is 0, 1, 0.518374205720839\n",
      "best merge is 3, 8, 0.545318145221393\n",
      "best merge is 2, 10, 0.579203512689541\n",
      "best merge is 0, 3, 0.624001682335286\n",
      "best merge is 20, 22, 0.642190166929202\n",
      "best merge is 15, 17, 0.738898877682885\n",
      "best merge is 0, 2, 0.761211441411895\n",
      "best merge is 11, 16, 0.767275247121574\n",
      "best merge is 6, 11, 0.911341525083091\n",
      "order classes\n",
      "14 - 0.985490614924918\n",
      "3 - 1.894384134659\n",
      "2 - 4.60365148272945\n",
      "2 - 4.43339326609731\n",
      "3 - 3.12269031108512\n",
      "0 21 1 5 4 12 3 9 8 14 13 2 7 10\n",
      "6 19 18\n",
      "11 16\n",
      "15 17\n",
      "20 23 22\n",
      "Done.Start drawing\n",
      "4.93\t-12.09\n",
      "GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "GSE2770_TGFB_AND_IL4_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "GSE22601_DOUBLE_NEGATIVE_VS_IMMATURE_CD4_SP_THYMOCYTE_DN\n",
      "GSE29949_CD8_NEG_DC_SPLEEN_VS_DC_BRAIN_UP\n",
      "GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_UP\n",
      "GSE360_LOW_DOSE_B_MALAYI_VS_M_TUBERCULOSIS_MAC_UP\n",
      "GSE19401_NAIVE_VS_IMMUNIZED_MOUSE_PLN_FOLLICULAR_DC_UP\n",
      "GSE43863_DAY6_EFF_VS_DAY150_MEM_LY6C_INT_CXCR5POS_CD4_TCELL_DN\n",
      "GSE16450_CTRL_VS_IFNA_6H_STIM_MATURE_NEURON_CELL_LINE_DN\n",
      "GSE27896_HDAC6_KO_VS_WT_TREG_UP\n",
      "GSE42724_MEMORY_VS_B1_BCELL_DN\n",
      "GSE9946_MATURE_STIMULATORY_VS_LISTERIA_INF_MATURE_DC_UP\n",
      "GSE31082_CD4_VS_CD8_SP_THYMOCYTE_UP\n",
      "GSE33425_CD161_HIGH_VS_INT_CD8_TCELL_DN\n",
      "GSE9988_ANTI_TREM1_VS_ANTI_TREM1_AND_LPS_MONOCYTE_UP\n",
      "GSE12003_MIR223_KO_VS_WT_BM_PROGENITOR_8D_CULTURE_UP\n",
      "GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "GSE27786_LSK_VS_CD8_TCELL_DN\n",
      "GSE2585_CTEC_VS_MTEC_THYMUS_DN\n",
      "GSE32255_WT_UNSTIM_VS_JMJD2D_KNOCKDOWN_4H_LPS_STIM_DC_UP\n",
      "GSE32255_UNSTIM_VS_4H_LPS_STIM_DC_DN\n",
      "GSE14415_ACT_VS_CTRL_NATURAL_TREG_UP\n",
      "GSE3720_VD1_VS_VD2_GAMMADELTA_TCELL_UP\n",
      "GSE45365_HEALTHY_VS_MCMV_INFECTION_BCELL_IFNAR_KO_DN\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 5, 0.177271420963679\n",
      "best merge is 6, 19, 0.198842964159118\n",
      "best merge is 8, 14, 0.229729040882629\n",
      "best merge is 3, 9, 0.244280640997698\n",
      "best merge is 1, 4, 0.26975310173209\n",
      "best merge is 2, 7, 0.290213203194628\n",
      "best merge is 8, 13, 0.332026245686594\n",
      "best merge is 6, 18, 0.433374055793285\n",
      "best merge is 1, 12, 0.436476534561607\n",
      "best merge is 0, 21, 0.465393302855107\n",
      "best merge is 20, 23, 0.472132966126977\n",
      "best merge is 0, 1, 0.518374205720839\n",
      "best merge is 3, 8, 0.545318145221393\n",
      "best merge is 2, 10, 0.579203512689541\n",
      "best merge is 0, 3, 0.624001682335286\n",
      "best merge is 20, 22, 0.642190166929202\n",
      "best merge is 15, 17, 0.738898877682885\n",
      "best merge is 0, 2, 0.761211441411895\n",
      "best merge is 11, 16, 0.767275247121574\n",
      "best merge is 6, 11, 0.911341525083091\n",
      "order classes\n",
      "14 - 0.985490614924918\n",
      "3 - 1.894384134659\n",
      "2 - 4.60365148272945\n",
      "2 - 4.43339326609731\n",
      "3 - 3.12269031108512\n",
      "0 21 1 5 4 12 3 9 8 14 13 2 7 10\n",
      "6 19 18\n",
      "11 16\n",
      "15 17\n",
      "20 23 22\n",
      "Done.-6.58813160987074\t42\t0\n",
      "7.49001175088132\t42\t26.8237367802585\n",
      "8.53877790834313\t42\t28.1562867215041\n",
      "9.15569917743831\t42\t28.9212690951821\n",
      "9.57520564042304\t42\t29.3901292596945\n",
      "9.90834312573443\t42\t29.7602820211516\n",
      "10.2044653349001\t42\t30.0564042303173\n",
      "10.5005875440658\t42\t30.352526439483\n",
      "10.8707403055229\t42\t30.6486486486486\n",
      "11.4506462984724\t42\t31.0928319623972\n",
      "16.9042303172738\t42\t31.8084606345476\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_c7/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 17786\n",
      "cat 198, mi = 0.00514138, ASTIER_INTEGRIN_SIGNALING\n",
      "cat 9992, mi = 0.00490647, GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_UP\n",
      "cat 10017, mi = 0.00473006, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "cat 12924, mi = 0.00447189, HOEBEKE_LYMPHOID_STEM_CELL_UP\n",
      "cat 10474, mi = 0.00437413, GSE27896_HDAC6_KO_VS_WT_TREG_UP\n",
      "cat 1740, mi = 0.00431121, GNF2_CD97\n",
      "cat 8619, mi = 0.00430530, GSE16450_CTRL_VS_IFNA_6H_STIM_IMMATURE_NEURON_CELL_LINE_DN\n",
      "cat 9445, mi = 0.00429285, GSE21360_NAIVE_VS_SECONDARY_MEMORY_CD8_TCELL_DN\n",
      "cat 16491, mi = 0.00420881, SIMBULAN_UV_RESPONSE_NORMAL_DN\n",
      "cat 10502, mi = 0.00414964, GSE28726_NAIVE_VS_ACTIVATED_CD4_TCELL_UP\n",
      "cat 12694, mi = 0.00411055, GSE9988_ANTI_TREM1_VS_ANTI_TREM1_AND_LPS_MONOCYTE_UP\n",
      "cat 13613, mi = 0.00410045, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "cat 12904, mi = 0.00406334, HINATA_NFKB_TARGETS_KERATINOCYTE_UP\n",
      "cat 11891, mi = 0.00406238, GSE43863_DAY6_EFF_VS_DAY150_MEM_LY6C_INT_CXCR5POS_CD4_TCELL_DN\n",
      "cat 4608, mi = 0.00406001, GO_NEGATIVE_REGULATION_OF_RESPONSE_TO_BIOTIC_STIMULUS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 198, mi = 0.005, pvalue = 0.000, ASTIER_INTEGRIN_SIGNALING\n",
      "Passed: cat 9992, mi = 0.005, pvalue = 0.000, GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_UP\n",
      "Passed: cat 10017, mi = 0.005, pvalue = 0.000, GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "Passed: cat 10474, mi = 0.004, pvalue = 0.000, GSE27896_HDAC6_KO_VS_WT_TREG_UP\n",
      "Passed: cat 1740, mi = 0.004, pvalue = 0.000, GNF2_CD97\n",
      "Passed: cat 13613, mi = 0.004, pvalue = 0.000, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "Passed: cat 12904, mi = 0.004, pvalue = 0.000, HINATA_NFKB_TARGETS_KERATINOCYTE_UP\n",
      "Passed: cat 13137, mi = 0.004, pvalue = 0.000, JISON_SICKLE_CELL_DISEASE_UP\n",
      "Passed: cat 10720, mi = 0.004, pvalue = 0.000, GSE31082_CD4_VS_CD8_SP_THYMOCYTE_UP\n",
      "Passed: cat 7161, mi = 0.004, pvalue = 0.000, GO_RNA_POLYMERASE_II_ACTIVATING_TRANSCRIPTION_FACTOR_BINDING\n",
      "Passed: cat 16725, mi = 0.004, pvalue = 0.000, TBK1.DN.48HRS_UP\n",
      "Passed: cat 12738, mi = 0.003, pvalue = 0.001, GTGGTGA_MIR197\n",
      "Passed: cat 10431, mi = 0.003, pvalue = 0.000, GSE27786_LSK_VS_CD8_TCELL_DN\n",
      "Passed: cat 14146, mi = 0.003, pvalue = 0.001, MODULE_183\n",
      "Passed: cat 11180, mi = 0.003, pvalue = 0.001, GSE3720_VD1_VS_VD2_GAMMADELTA_TCELL_UP\n",
      "Passed: cat 517, mi = 0.003, pvalue = 0.000, BIOCARTA_PS1_PATHWAY\n",
      "Passed: cat 8050, mi = 0.003, pvalue = 0.001, GSE12003_MIR223_KO_VS_WT_BM_PROGENITOR_8D_CULTURE_UP\n",
      "Passed: cat 8616, mi = 0.003, pvalue = 0.001, GSE16450_CTRL_VS_IFNA_12H_STIM_IMMATURE_NEURON_CELL_LINE_UP\n",
      "Passed: cat 9184, mi = 0.003, pvalue = 0.002, GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "Passed: cat 10127, mi = 0.003, pvalue = 0.002, GSE2585_CTEC_VS_MTEC_THYMUS_DN\n",
      "Passed: cat 10632, mi = 0.003, pvalue = 0.002, GSE29949_CD8_NEG_DC_SPLEEN_VS_DC_BRAIN_UP\n",
      "Passed: cat 10188, mi = 0.003, pvalue = 0.002, GSE26488_WT_VS_HDAC7_KO_DOUBLE_POSITIVE_THYMOCYTE_UP\n",
      "Passed: cat 14552, mi = 0.003, pvalue = 0.002, MORF_CSNK1D\n",
      "Passed: cat 10329, mi = 0.003, pvalue = 0.003, GSE2770_TGFB_AND_IL4_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Passed: cat 8489, mi = 0.003, pvalue = 0.004, GSE15659_CD45RA_NEG_CD4_TCELL_VS_RESTING_TREG_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5530, mi = 0.003, pvalue = 0.006, GO_POSITIVE_REGULATION_OF_MULTICELLULAR_ORGANISM_GROWTH\n",
      "Passed: cat 874, mi = 0.003, pvalue = 0.003, CHARAFE_BREAST_CANCER_BASAL_VS_MESENCHYMAL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 16235, mi = 0.003, pvalue = 0.005, ROSS_AML_WITH_PML_RARA_FUSION\n",
      "Didn't pass (cutoff=0.0050): cat 13317, mi = 0.003, pvalue = 0.009, KEGG_LYSOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 12581, mi = 0.003, pvalue = 0.006, GSE8921_UNSTIM_0H_VS_TLR1_2_STIM_MONOCYTE_24H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 7909, mi = 0.003, pvalue = 0.008, GSE10325_BCELL_VS_MYELOID_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2969, mi = 0.003, pvalue = 0.010, GO_DNA_METHYLATION_OR_DEMETHYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 5117, mi = 0.003, pvalue = 0.009, GO_PHOSPHATIDYLINOSITOL_4_5_BISPHOSPHATE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5623, mi = 0.003, pvalue = 0.012, GO_POSITIVE_REGULATION_OF_PROTEIN_POLYMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 1978, mi = 0.003, pvalue = 0.009, GO_ADHERENS_JUNCTION_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 14992, mi = 0.003, pvalue = 0.006, OKAWA_NEUROBLASTOMA_1P36_31_DELETION\n",
      "Didn't pass (cutoff=0.0050): cat 17135, mi = 0.003, pvalue = 0.006, WHITE_NEUROBLASTOMA_WITH_1P36.3_DELETION\n",
      "Didn't pass (cutoff=0.0050): cat 8399, mi = 0.003, pvalue = 0.007, GSE14769_20MIN_VS_360MIN_LPS_BMDM_DN\n",
      "Passed: cat 14346, mi = 0.003, pvalue = 0.005, MODULE_419\n",
      "Didn't pass (cutoff=0.0050): cat 15157, mi = 0.003, pvalue = 0.009, PID_ARF6_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 11055, mi = 0.003, pvalue = 0.008, GSE360_L_MAJOR_VS_B_MALAYI_LOW_DOSE_MAC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9317, mi = 0.003, pvalue = 0.012, GSE20198_UNTREATED_VS_IL12_IL18_TREATED_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4181, mi = 0.003, pvalue = 0.015, GO_MOTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 14621, mi = 0.003, pvalue = 0.017, MORF_MYC\n",
      "Didn't pass (cutoff=0.0050): cat 7601, mi = 0.003, pvalue = 0.014, GO_TRANSITION_METAL_ION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 5460, mi = 0.003, pvalue = 0.014, GO_POSITIVE_REGULATION_OF_INTERLEUKIN_8_PRODUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 17175, mi = 0.003, pvalue = 0.008, WONG_ENDMETRIUM_CANCER_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8508, mi = 0.003, pvalue = 0.016, GSE1566_WT_VS_EZH2_KO_LN_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.014, BIOCARTA_MITOCHONDRIA_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 8776, mi = 0.002, pvalue = 0.015, GSE17721_CPG_VS_GARDIQUIMOD_4H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 11717, mi = 0.002, pvalue = 0.019, GSE41176_UNSTIM_VS_ANTI_IGM_STIM_TAK1_KO_BCELL_3H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 11028, mi = 0.002, pvalue = 0.015, GSE360_LOW_DOSE_B_MALAYI_VS_M_TUBERCULOSIS_MAC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 11492, mi = 0.002, pvalue = 0.015, GSE3982_EOSINOPHIL_VS_BASOPHIL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 10353, mi = 0.002, pvalue = 0.016, GSE2770_UNTREATED_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8871, mi = 0.002, pvalue = 0.019, GSE17721_LPS_VS_CPG_2H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4771, mi = 0.002, pvalue = 0.024, GO_NONMOTILE_PRIMARY_CILIUM\n",
      "Didn't pass (cutoff=0.0050): cat 3339, mi = 0.002, pvalue = 0.019, GO_GDP_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 677, mi = 0.002, pvalue = 0.025, BROWN_MYELOID_CELL_DEVELOPMENT_DN\n",
      "Didn't pass (cutoff=0.0050): cat 16743, mi = 0.002, pvalue = 0.021, TCTAGAG_MIR517\n",
      "Total number of categories evaluated = 495.\n",
      "Number of categories that passed the tests = 27\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 8, 25, 0.13376109120874\n",
      "best merge is 9, 16, 0.156121031685159\n",
      "best merge is 0, 26, 0.186727308824692\n",
      "best merge is 5, 6, 0.191132001497746\n",
      "best merge is 9, 11, 0.223042998994516\n",
      "best merge is 18, 23, 0.228046416226771\n",
      "best merge is 4, 8, 0.23480806534569\n",
      "best merge is 2, 12, 0.30268321105917\n",
      "best merge is 13, 22, 0.304676001945719\n",
      "best merge is 4, 5, 0.35639426890496\n",
      "best merge is 10, 15, 0.39347283048892\n",
      "best merge is 3, 13, 0.419239965288198\n",
      "best merge is 0, 10, 0.468972918945059\n",
      "best merge is 18, 24, 0.535395314254446\n",
      "best merge is 17, 21, 0.542660112549862\n",
      "best merge is 1, 4, 0.55723718834604\n",
      "best merge is 2, 3, 0.567496424602819\n",
      "best merge is 1, 2, 0.595408788584857\n",
      "best merge is 7, 20, 0.611150789662257\n",
      "best merge is 0, 9, 0.676566562627292\n",
      "best merge is 0, 1, 0.82333142201733\n",
      "best merge is 17, 18, 0.834266547540233\n",
      "best merge is 7, 19, 0.929223847118373\n",
      "order classes\n",
      "18 - 0.953757029804923\n",
      "2 - 4.12526783022023\n",
      "5 - 2.95794566317163\n",
      "0 26 10 15 9 16 11 1 4 8 25 5 6 2 12 3 13 22\n",
      "7 20\n",
      "14\n",
      "17 21 18 23 24\n",
      "19\n",
      "Done.Start drawing\n",
      "5.76\t-12.09\n",
      "GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "GSE2770_TGFB_AND_IL4_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "MORF_CSNK1D\n",
      "GSE29949_CD8_NEG_DC_SPLEEN_VS_DC_BRAIN_UP\n",
      "GSE26488_WT_VS_HDAC7_KO_DOUBLE_POSITIVE_THYMOCYTE_UP\n",
      "GSE24634_NAIVE_CD4_TCELL_VS_DAY3_IL4_CONV_TREG_UP\n",
      "JISON_SICKLE_CELL_DISEASE_UP\n",
      "MODULE_419\n",
      "TBK1.DN.48HRS_UP\n",
      "BIOCARTA_PS1_PATHWAY\n",
      "MODULE_183\n",
      "GSE16450_CTRL_VS_IFNA_12H_STIM_IMMATURE_NEURON_CELL_LINE_UP\n",
      "GNF2_CD97\n",
      "CHARAFE_BREAST_CANCER_BASAL_VS_MESENCHYMAL_UP\n",
      "LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "HINATA_NFKB_TARGETS_KERATINOCYTE_UP\n",
      "GSE27896_HDAC6_KO_VS_WT_TREG_UP\n",
      "GO_RNA_POLYMERASE_II_ACTIVATING_TRANSCRIPTION_FACTOR_BINDING\n",
      "GSE12003_MIR223_KO_VS_WT_BM_PROGENITOR_8D_CULTURE_UP\n",
      "GTGGTGA_MIR197\n",
      "ASTIER_INTEGRIN_SIGNALING\n",
      "GSE24671_CTRL_VS_SENDAI_VIRUS_INFECTED_MOUSE_SPLENOCYTES_DN\n",
      "GSE27786_LSK_VS_CD8_TCELL_DN\n",
      "GSE2585_CTEC_VS_MTEC_THYMUS_DN\n",
      "GSE31082_CD4_VS_CD8_SP_THYMOCYTE_UP\n",
      "GSE3720_VD1_VS_VD2_GAMMADELTA_TCELL_UP\n",
      "GSE15659_CD45RA_NEG_CD4_TCELL_VS_RESTING_TREG_DN\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 8, 25, 0.13376109120874\n",
      "best merge is 9, 16, 0.156121031685159\n",
      "best merge is 0, 26, 0.186727308824692\n",
      "best merge is 5, 6, 0.191132001497746\n",
      "best merge is 9, 11, 0.223042998994516\n",
      "best merge is 18, 23, 0.228046416226771\n",
      "best merge is 4, 8, 0.23480806534569\n",
      "best merge is 2, 12, 0.30268321105917\n",
      "best merge is 13, 22, 0.304676001945719\n",
      "best merge is 4, 5, 0.35639426890496\n",
      "best merge is 10, 15, 0.39347283048892\n",
      "best merge is 3, 13, 0.419239965288198\n",
      "best merge is 0, 10, 0.468972918945059\n",
      "best merge is 18, 24, 0.535395314254446\n",
      "best merge is 17, 21, 0.542660112549862\n",
      "best merge is 1, 4, 0.55723718834604\n",
      "best merge is 2, 3, 0.567496424602819\n",
      "best merge is 1, 2, 0.595408788584857\n",
      "best merge is 7, 20, 0.611150789662257\n",
      "best merge is 0, 9, 0.676566562627292\n",
      "best merge is 0, 1, 0.82333142201733\n",
      "best merge is 17, 18, 0.834266547540233\n",
      "best merge is 7, 19, 0.929223847118373\n",
      "order classes\n",
      "18 - 0.953757029804923\n",
      "2 - 4.12526783022023\n",
      "5 - 2.95794566317163\n",
      "0 26 10 15 9 16 11 1 4 8 25 5 6 2 12 3 13 22\n",
      "7 20\n",
      "14\n",
      "17 21 18 23 24\n",
      "19\n",
      "Done.-7.24705882352941\t42\t0\n",
      "6.16470588235294\t42\t25.5058823529412\n",
      "7.17647058823529\t42\t26.8235294117647\n",
      "7.76470588235294\t42\t27.5294117647059\n",
      "8.17647058823529\t42\t28\n",
      "8.50588235294117\t42\t28.3529411764706\n",
      "8.78823529411764\t42\t28.6588235294118\n",
      "9.07058823529411\t42\t28.9176470588235\n",
      "9.4235294117647\t42\t29.2235294117647\n",
      "9.96470588235294\t42\t29.6235294117647\n",
      "16.1529411764706\t42\t30.3058823529412\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_full/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 1545 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 1545\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "140 hits go_terms_num = 50\n",
      "cat 0, mi = 0.01279794, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 5, mi = 0.00978152, HALLMARK_PROTEIN_SECRETION\n",
      "cat 22, mi = 0.00959789, HALLMARK_E2F_TARGETS\n",
      "cat 2, mi = 0.00938698, HALLMARK_PEROXISOME\n",
      "cat 32, mi = 0.00818443, HALLMARK_BILE_ACID_METABOLISM\n",
      "cat 38, mi = 0.00770260, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "cat 7, mi = 0.00663305, HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY\n",
      "cat 39, mi = 0.00653580, HALLMARK_NOTCH_SIGNALING\n",
      "cat 12, mi = 0.00633702, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "cat 15, mi = 0.00606597, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 48, mi = 0.00580062, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 40, mi = 0.00579174, HALLMARK_APICAL_SURFACE\n",
      "cat 23, mi = 0.00564328, HALLMARK_GLYCOLYSIS\n",
      "cat 36, mi = 0.00556837, HALLMARK_DNA_REPAIR\n",
      "cat 14, mi = 0.00547627, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 0, mi = 0.013, pvalue = 0.002, HALLMARK_FATTY_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.010, pvalue = 0.035, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 22, mi = 0.010, pvalue = 0.035, HALLMARK_E2F_TARGETS\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.008, pvalue = 0.094, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.007, pvalue = 0.287, HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.007, pvalue = 0.302, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.006, pvalue = 0.300, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.006, pvalue = 0.275, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.006, pvalue = 0.279, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.006, pvalue = 0.421, HALLMARK_APICAL_SURFACE\n",
      "Didn't pass (cutoff=0.0050): cat 23, mi = 0.006, pvalue = 0.313, HALLMARK_GLYCOLYSIS\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.006, pvalue = 0.335, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.005, pvalue = 0.329, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.005, pvalue = 0.378, HALLMARK_COMPLEMENT\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.005, pvalue = 0.341, HALLMARK_PANCREAS_BETA_CELLS\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.005, pvalue = 0.420, HALLMARK_HYPOXIA\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.005, pvalue = 0.425, HALLMARK_KRAS_SIGNALING_UP\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.005, pvalue = 0.430, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 29, mi = 0.005, pvalue = 0.537, HALLMARK_MYC_TARGETS_V2\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.005, pvalue = 0.458, HALLMARK_ADIPOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.005, pvalue = 0.578, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.93\t-9.10\n",
      "HALLMARK_FATTY_ACID_METABOLISM\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-8.01069137562366\t42\t0\n",
      "4.69707769066287\t42\t23.9786172487527\n",
      "5.87954383464006\t42\t25.4155381325731\n",
      "6.58303635067712\t42\t26.3435495367071\n",
      "7.03207412687099\t42\t26.8225231646472\n",
      "7.37633642195296\t42\t27.2416250890948\n",
      "7.67569493941554\t42\t27.5110477548111\n",
      "8.0199572344975\t42\t27.84034212402\n",
      "8.43905915894512\t42\t28.1995723449751\n",
      "9.08267997148967\t42\t28.6785459729152\n",
      "15.7434069850321\t42\t29.4868139700641\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_msigdb_h/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7491 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7491\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "681 hits go_terms_num = 790\n",
      "cat 677, mi = 0.00387760, TNRC6C_HEK293_Plass\n",
      "cat 31, mi = 0.00235085, AUH_K562_POSTAR_eCLIP\n",
      "cat 230, mi = 0.00230515, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00196769, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 30, mi = 0.00193588, AUH_HepG2_POSTAR_eCLIP\n",
      "cat 275, mi = 0.00189005, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00171620, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 236, mi = 0.00168609, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 569, mi = 0.00155548, SLBP_K562_POSTAR_eCLIP\n",
      "cat 237, mi = 0.00153486, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 447, mi = 0.00152930, NPM1_K562_POSTAR_eCLIP\n",
      "cat 331, mi = 0.00152022, HNRNPU_K562_POSTAR_eCLIP\n",
      "cat 101, mi = 0.00143741, DDX3X_HEK293T_POSTAR_CIMS\n",
      "cat 195, mi = 0.00135954, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00135954, EZH2_HCT116_POSTAR_iCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 677, mi = 0.004, pvalue = 0.000, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.008, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.043, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.002, pvalue = 0.031, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.059, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.096, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.112, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.196, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 331, mi = 0.002, pvalue = 0.125, HNRNPU_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.001, pvalue = 0.249, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.001, pvalue = 0.266, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.001, pvalue = 0.262, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.001, pvalue = 0.244, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.001, pvalue = 0.265, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 553, mi = 0.001, pvalue = 0.501, SBDS_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.387, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.466, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.658, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.001, pvalue = 0.809, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 323, mi = 0.001, pvalue = 0.089, HNRNPU_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.001, pvalue = 0.913, HNRNPA1_HEK293T_POSTAR_CIMS\n",
      "Total number of categories evaluated = 24.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "5.76\t-12.09\n",
      "TNRC6C_HEK293_Plass\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.28235294117647\t42\t0\n",
      "6.08235294117647\t42\t25.4352941176471\n",
      "7.11764705882353\t42\t26.7294117647059\n",
      "7.74117647058823\t42\t27.5058823529412\n",
      "8.16470588235294\t42\t27.9764705882353\n",
      "8.49411764705882\t42\t28.3529411764706\n",
      "8.77647058823529\t42\t28.6352941176471\n",
      "9.07058823529411\t42\t28.9176470588235\n",
      "9.43529411764705\t42\t29.2235294117647\n",
      "9.98823529411764\t42\t29.6470588235294\n",
      "16.1647058823529\t42\t30.3294117647059\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_RBPs_all_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6816 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6816\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "620 hits go_terms_num = 790\n",
      "cat 677, mi = 0.00388727, TNRC6C_HEK293_Plass\n",
      "cat 569, mi = 0.00230248, SLBP_K562_POSTAR_eCLIP\n",
      "cat 275, mi = 0.00222430, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 321, mi = 0.00215998, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 236, mi = 0.00214984, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 447, mi = 0.00213435, NPM1_K562_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00210754, AUH_K562_POSTAR_eCLIP\n",
      "cat 195, mi = 0.00208131, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00208131, EZH2_HCT116_POSTAR_iCLIP\n",
      "cat 230, mi = 0.00201427, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 237, mi = 0.00198282, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 377, mi = 0.00190488, LARP7_K562_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00180721, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 668, mi = 0.00159897, TNRC6A_HEK293_Plass\n",
      "cat 30, mi = 0.00159813, AUH_HepG2_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 677, mi = 0.004, pvalue = 0.000, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.024, SLBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.036, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.047, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.050, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.049, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.060, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.002, pvalue = 0.057, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.073, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.123, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.002, pvalue = 0.192, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.186, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.002, pvalue = 0.254, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.198, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 671, mi = 0.001, pvalue = 0.261, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 670, mi = 0.001, pvalue = 0.305, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 331, mi = 0.001, pvalue = 0.319, HNRNPU_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.604, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.001, pvalue = 0.399, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.417, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 282, mi = 0.001, pvalue = 0.655, HNRNPA2B1_HEK293T_POSTAR_CIMS\n",
      "Total number of categories evaluated = 23.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.93\t-12.09\n",
      "TNRC6C_HEK293_Plass\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.60047003525264\t42\t0\n",
      "7.46533490011751\t42\t26.7990599294947\n",
      "8.51410105757932\t42\t28.1316098707403\n",
      "9.1310223266745\t42\t28.8965922444183\n",
      "9.55052878965923\t42\t29.3654524089307\n",
      "9.89600470035253\t42\t29.7356051703878\n",
      "10.1921269095182\t42\t30.0564042303173\n",
      "10.4882491186839\t42\t30.3278495887192\n",
      "10.8707403055229\t42\t30.6486486486486\n",
      "11.4506462984724\t42\t31.0928319623972\n",
      "16.9042303172738\t42\t31.8084606345476\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_RBPs_coding_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6678 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6678\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "607 hits go_terms_num = 790\n",
      "cat 677, mi = 0.00420571, TNRC6C_HEK293_Plass\n",
      "cat 379, mi = 0.00342807, LIN28A_H9_POSTAR_CIMS\n",
      "cat 350, mi = 0.00302207, IGF2BP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 299, mi = 0.00292580, HNRNPD_HEK293_POSTAR_Piranha_0.01\n",
      "cat 437, mi = 0.00290154, NONO_K562_POSTAR_eCLIP\n",
      "cat 306, mi = 0.00283263, HNRNPK_HepG2_POSTAR_eCLIP\n",
      "cat 158, mi = 0.00275239, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 366, mi = 0.00270219, KHDRBS1_K562_POSTAR_eCLIP\n",
      "cat 369, mi = 0.00262593, KHSRP_K562_POSTAR_eCLIP\n",
      "cat 586, mi = 0.00252425, SRSF1_HepG2_POSTAR_eCLIP\n",
      "cat 304, mi = 0.00241563, HNRNPH1_HEK293T_POSTAR_HITS-CLIP\n",
      "cat 128, mi = 0.00239176, DGCR8_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 127, mi = 0.00238216, DGCR8_HEK293T_POSTAR_HITS-CLIP\n",
      "cat 338, mi = 0.00235375, IGF2BP1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 356, mi = 0.00233109, IGF2BP3_Panc1_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 677, mi = 0.004, pvalue = 0.000, TNRC6C_HEK293_Plass\n",
      "Passed: cat 379, mi = 0.003, pvalue = 0.000, LIN28A_H9_POSTAR_CIMS\n",
      "Passed: cat 437, mi = 0.003, pvalue = 0.004, NONO_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.003, pvalue = 0.006, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 512, mi = 0.002, pvalue = 0.032, RBM10_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.046, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 670, mi = 0.002, pvalue = 0.010, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 671, mi = 0.002, pvalue = 0.010, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.002, pvalue = 0.081, EIF3D_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 142, mi = 0.002, pvalue = 0.093, DICER1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 731, mi = 0.001, pvalue = 0.109, WTAP_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 513, mi = 0.001, pvalue = 0.309, RBM10_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 468, mi = 0.001, pvalue = 0.247, PRKRA_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 514, mi = 0.001, pvalue = 0.400, RBM10_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 579, mi = 0.001, pvalue = 0.403, SND1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 687, mi = 0.001, pvalue = 0.554, U2AF1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 628, mi = 0.001, pvalue = 0.375, TARBP2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 467, mi = 0.001, pvalue = 0.416, PRKRA_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 627, mi = 0.001, pvalue = 0.537, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.344, ALKBH5_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.001, pvalue = 0.430, ELAVL1_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 595, mi = 0.001, pvalue = 0.617, SRSF7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 303, mi = 0.001, pvalue = 0.257, HNRNPH1_HEK293T_POSTAR_CIMS\n",
      "Total number of categories evaluated = 183.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.598851489356983\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "4.93\t-12.09\n",
      "TNRC6C_HEK293_Plass\n",
      "NONO_K562_POSTAR_eCLIP\n",
      "LIN28A_H9_POSTAR_CIMS\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.598851489356983\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-6.58813160987074\t42\t0\n",
      "7.49001175088132\t42\t26.8237367802585\n",
      "8.52643948296122\t42\t28.1562867215041\n",
      "9.1310223266745\t42\t28.8965922444183\n",
      "9.55052878965923\t42\t29.3654524089307\n",
      "9.89600470035253\t42\t29.7356051703878\n",
      "10.2044653349001\t42\t30.0564042303173\n",
      "10.5005875440658\t42\t30.352526439483\n",
      "10.8707403055229\t42\t30.6486486486486\n",
      "11.4383078730905\t42\t31.0928319623972\n",
      "16.8918918918919\t42\t31.7837837837838\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6602 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6602\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "600 hits go_terms_num = 790\n",
      "cat 157, mi = 0.00272232, EIF3A_HEK293T_POSTAR_PARalyzer\n",
      "cat 765, mi = 0.00257605, YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "cat 130, mi = 0.00249291, DGCR8_HepG2_POSTAR_eCLIP\n",
      "cat 246, mi = 0.00248727, FXR2_HEK293_POSTAR_PAR-CLIP\n",
      "cat 247, mi = 0.00248727, FXR2_HEK293_POSTAR_PARalyzer\n",
      "cat 489, mi = 0.00239265, PTBP1_K562_POSTAR_eCLIP\n",
      "cat 567, mi = 0.00239048, SFPQ_HepG2_POSTAR_eCLIP\n",
      "cat 740, mi = 0.00231867, XRN2_K562_POSTAR_eCLIP\n",
      "cat 532, mi = 0.00225414, RBM47_MDA-MB-231_POSTAR_HITS-CLIP\n",
      "cat 152, mi = 0.00215388, EFTUD2_HepG2_POSTAR_eCLIP\n",
      "cat 533, mi = 0.00214218, RBM47_MDA-MB-231_POSTAR_Piranha_0.01\n",
      "cat 169, mi = 0.00213930, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 761, mi = 0.00210628, YTHDF2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 422, mi = 0.00209096, NCBP2_HepG2_POSTAR_eCLIP\n",
      "cat 772, mi = 0.00208494, YTHDF3_Huh7_POSTAR_PAR-CLIP\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 157, mi = 0.003, pvalue = 0.011, EIF3A_HEK293T_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 765, mi = 0.003, pvalue = 0.015, YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "Passed: cat 130, mi = 0.002, pvalue = 0.002, DGCR8_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 489, mi = 0.002, pvalue = 0.034, PTBP1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 532, mi = 0.002, pvalue = 0.026, RBM47_MDA-MB-231_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 533, mi = 0.002, pvalue = 0.025, RBM47_MDA-MB-231_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.025, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 772, mi = 0.002, pvalue = 0.068, YTHDF3_Huh7_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 773, mi = 0.002, pvalue = 0.070, YTHDF3_Huh7_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.063, EIF3A_HEK293T_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 365, mi = 0.002, pvalue = 0.057, IMP3_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 652, mi = 0.002, pvalue = 0.096, TIA1_HeLa_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.002, pvalue = 0.079, DICER1_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 692, mi = 0.002, pvalue = 0.089, U2AF2_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 693, mi = 0.002, pvalue = 0.091, U2AF2_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 492, mi = 0.002, pvalue = 0.084, PUM2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 493, mi = 0.002, pvalue = 0.087, PUM2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 496, mi = 0.002, pvalue = 0.106, PUM2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 356, mi = 0.002, pvalue = 0.029, IGF2BP3_Panc1_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 358, mi = 0.002, pvalue = 0.031, IGF2BP3_Panc1_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 640, mi = 0.002, pvalue = 0.072, TARDBP_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 618, mi = 0.002, pvalue = 0.155, TAF15_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 163, mi = 0.002, pvalue = 0.096, EIF3D_HEK293T_POSTAR_PARalyzer\n",
      "Total number of categories evaluated = 29.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.93\t-12.09\n",
      "DGCR8_HepG2_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-6.58813160987074\t42\t0\n",
      "7.49001175088132\t42\t26.8237367802585\n",
      "8.52643948296122\t42\t28.1562867215041\n",
      "9.1433607520564\t42\t28.8965922444183\n",
      "9.57520564042304\t42\t29.3901292596945\n",
      "9.90834312573443\t42\t29.7602820211516\n",
      "10.2044653349001\t42\t30.0564042303173\n",
      "10.5005875440658\t42\t30.352526439483\n",
      "10.8707403055229\t42\t30.6486486486486\n",
      "11.4506462984724\t42\t31.0928319623972\n",
      "16.9042303172738\t42\t31.8084606345476\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6691 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6691\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "608 hits go_terms_num = 790\n",
      "cat 60, mi = 0.00363894, CPSF-73_HEK293_Plass\n",
      "cat 595, mi = 0.00286158, SRSF7_K562_POSTAR_eCLIP\n",
      "cat 185, mi = 0.00283920, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "cat 383, mi = 0.00282871, LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "cat 199, mi = 0.00277028, FAM120A_K562_POSTAR_eCLIP\n",
      "cat 225, mi = 0.00275071, FUS_Brain_POSTAR_CIMS\n",
      "cat 788, mi = 0.00272016, ZRANB2_K562_POSTAR_eCLIP\n",
      "cat 579, mi = 0.00267603, SND1_HepG2_POSTAR_eCLIP\n",
      "cat 386, mi = 0.00262886, LIN28A_HEK293_POSTAR_Piranha_0.01\n",
      "cat 492, mi = 0.00262498, PUM2_HEK293_POSTAR_PAR-CLIP\n",
      "cat 47, mi = 0.00259438, CELF2_T_cell_POSTAR_CIMS\n",
      "cat 382, mi = 0.00256353, LIN28A_HEK293_POSTAR_CIMS\n",
      "cat 493, mi = 0.00252728, PUM2_HEK293_POSTAR_PARalyzer\n",
      "cat 533, mi = 0.00252693, RBM47_MDA-MB-231_POSTAR_Piranha_0.01\n",
      "cat 356, mi = 0.00252150, IGF2BP3_Panc1_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 60, mi = 0.004, pvalue = 0.000, CPSF-73_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 595, mi = 0.003, pvalue = 0.006, SRSF7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.003, pvalue = 0.007, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 383, mi = 0.003, pvalue = 0.006, LIN28A_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 199, mi = 0.003, pvalue = 0.006, FAM120A_K562_POSTAR_eCLIP\n",
      "Passed: cat 225, mi = 0.003, pvalue = 0.001, FUS_Brain_POSTAR_CIMS\n",
      "Passed: cat 788, mi = 0.003, pvalue = 0.004, ZRANB2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.003, pvalue = 0.012, CELF2_T_cell_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 356, mi = 0.003, pvalue = 0.013, IGF2BP3_Panc1_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 358, mi = 0.003, pvalue = 0.012, IGF2BP3_Panc1_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 753, mi = 0.002, pvalue = 0.029, YTHDC2_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 774, mi = 0.002, pvalue = 0.058, YTHDF3_Huh7_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 626, mi = 0.002, pvalue = 0.111, TARBP2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.002, pvalue = 0.073, DICER1_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.126, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 467, mi = 0.002, pvalue = 0.187, PRKRA_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 468, mi = 0.002, pvalue = 0.192, PRKRA_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.002, pvalue = 0.080, DCK1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 621, mi = 0.002, pvalue = 0.175, TAF15_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 627, mi = 0.002, pvalue = 0.221, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.002, pvalue = 0.139, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 178, mi = 0.002, pvalue = 0.165, ELAVL1_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.279, ELAVL1_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 304, mi = 0.002, pvalue = 0.273, HNRNPH1_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 759, mi = 0.002, pvalue = 0.239, YTHDF1_Huh7_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.002, pvalue = 0.185, DICER1_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 142, mi = 0.002, pvalue = 0.331, DICER1_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 81.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.669844073880855\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "4.93\t-12.09\n",
      "FUS_Brain_POSTAR_CIMS\n",
      "CPSF-73_HEK293_Plass\n",
      "ZRANB2_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.669844073880855\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-6.58813160987074\t42\t0\n",
      "7.49001175088132\t42\t26.8237367802585\n",
      "8.52643948296122\t42\t28.1562867215041\n",
      "9.1310223266745\t42\t28.8965922444183\n",
      "9.55052878965923\t42\t29.3654524089307\n",
      "9.88366627497062\t42\t29.7356051703878\n",
      "10.1797884841363\t42\t30.0317273795535\n",
      "10.4882491186839\t42\t30.3278495887192\n",
      "10.858401880141\t42\t30.6486486486486\n",
      "11.4259694477086\t42\t31.0681551116334\n",
      "16.8918918918919\t42\t31.7837837837838\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 790\n",
      "cat 195, mi = 0.00271603, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00271603, EZH2_HCT116_POSTAR_iCLIP\n",
      "cat 377, mi = 0.00263965, LARP7_K562_POSTAR_eCLIP\n",
      "cat 705, mi = 0.00228036, UPF1_HEK293_Plass\n",
      "cat 275, mi = 0.00226919, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00221544, AUH_K562_POSTAR_eCLIP\n",
      "cat 321, mi = 0.00210409, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 270, mi = 0.00205941, HLTF_K562_POSTAR_eCLIP\n",
      "cat 449, mi = 0.00203704, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 256, mi = 0.00202719, GNL3_K562_POSTAR_eCLIP\n",
      "cat 236, mi = 0.00197040, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 428, mi = 0.00187585, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 331, mi = 0.00179979, HNRNPU_K562_POSTAR_eCLIP\n",
      "cat 237, mi = 0.00179339, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 97, mi = 0.00176871, DCK1_HEK293_Plass\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.003, pvalue = 0.007, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.003, pvalue = 0.007, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.003, pvalue = 0.012, LARP7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 705, mi = 0.002, pvalue = 0.043, UPF1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.002, pvalue = 0.049, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.051, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.065, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.002, pvalue = 0.080, HLTF_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.085, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.002, pvalue = 0.061, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.106, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.002, pvalue = 0.096, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 331, mi = 0.002, pvalue = 0.128, HNRNPU_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.180, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.002, pvalue = 0.149, DCK1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.002, pvalue = 0.133, AGO3_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 456, mi = 0.002, pvalue = 0.202, PAPD5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 670, mi = 0.002, pvalue = 0.154, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.002, pvalue = 0.187, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 469, mi = 0.002, pvalue = 0.175, PRKRA_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.68\t-12.09\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 101\n",
      "cat 62, mi = 0.00216437, RBM46\n",
      "cat 9, mi = 0.00160364, ESRP2\n",
      "cat 34, mi = 0.00144682, KHDRBS3\n",
      "cat 69, mi = 0.00128276, RBMS3.1\n",
      "cat 68, mi = 0.00127287, RBMS3\n",
      "cat 13, mi = 0.00121042, FXR2\n",
      "cat 27, mi = 0.00103570, HuR.2\n",
      "cat 21, mi = 0.00101318, HNRNPK\n",
      "cat 88, mi = 0.00073350, SRSF9.1\n",
      "cat 6, mi = 0.00060233, CPEB4\n",
      "cat 56, mi = 0.00041079, RBM3\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "cat 3, mi = 0.00000000, BRUNOL6\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.002, pvalue = 0.036, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.281, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.288, KHDRBS3\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.395, RBMS3.1\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.001, pvalue = 0.404, RBMS3\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.345, FXR2\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.588, HuR.2\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.887, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.001, pvalue = 0.799, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.890, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.000, pvalue = 0.964, RBM3\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, FMR1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.68\t-12.09\n",
      "Outputing EPS file deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/d_mtyl_Virma_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_Virma_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_Virma_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_Virma_vs_NT_human_ensembl_RBPs_DeepBind/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7509 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7509\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "683 hits go_terms_num = 17689\n",
      "cat 4405, mi = 0.00317042, cellular response to organic substance\n",
      "cat 1445, mi = 0.00294897, thymus development\n",
      "cat 3651, mi = 0.00294733, insulin receptor signaling pathway\n",
      "cat 1684, mi = 0.00287281, costamere\n",
      "cat 1690, mi = 0.00285007, response to radiation\n",
      "cat 4263, mi = 0.00277069, response to iron ion\n",
      "cat 1599, mi = 0.00275574, axonogenesis\n",
      "cat 2170, mi = 0.00270877, T cell homeostasis\n",
      "cat 264, mi = 0.00262451, positive regulation of translation\n",
      "cat 661, mi = 0.00254075, cartilage development\n",
      "cat 1609, mi = 0.00249309, positive regulation of multicellular organism growth\n",
      "cat 5700, mi = 0.00247829, myeloid cell differentiation\n",
      "cat 315, mi = 0.00245731, positive regulation of peptidyl-serine phosphorylation\n",
      "cat 998, mi = 0.00243414, positive regulation of insulin receptor signaling pathway\n",
      "cat 42, mi = 0.00241603, copper ion binding\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 4405, mi = 0.003, pvalue = 0.000, cellular response to organic substance\n",
      "Passed: cat 3651, mi = 0.003, pvalue = 0.002, insulin receptor signaling pathway\n",
      "Passed: cat 1684, mi = 0.003, pvalue = 0.001, costamere\n",
      "Passed: cat 264, mi = 0.003, pvalue = 0.005, positive regulation of translation\n",
      "Passed: cat 661, mi = 0.003, pvalue = 0.005, cartilage development\n",
      "Didn't pass (cutoff=0.0050): cat 5700, mi = 0.002, pvalue = 0.005, myeloid cell differentiation\n",
      "Passed: cat 42, mi = 0.002, pvalue = 0.000, copper ion binding\n",
      "Didn't pass (cutoff=0.0050): cat 3071, mi = 0.002, pvalue = 0.007, cytokine production\n",
      "Didn't pass (cutoff=0.0050): cat 692, mi = 0.002, pvalue = 0.012, structural constituent of ribosome\n",
      "Didn't pass (cutoff=0.0050): cat 3388, mi = 0.002, pvalue = 0.010, osteoblast differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 3306, mi = 0.002, pvalue = 0.010, negative regulation of tumor necrosis factor production\n",
      "Didn't pass (cutoff=0.0050): cat 3512, mi = 0.002, pvalue = 0.009, positive regulation by host of viral transcription\n",
      "Didn't pass (cutoff=0.0050): cat 2878, mi = 0.002, pvalue = 0.011, acrosomal vesicle\n",
      "Passed: cat 749, mi = 0.002, pvalue = 0.001, trophectodermal cell differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 2749, mi = 0.002, pvalue = 0.015, hydrolase activity, acting on ester bonds\n",
      "Didn't pass (cutoff=0.0050): cat 5674, mi = 0.002, pvalue = 0.005, negative regulation of nucleic acid-templated transcription\n",
      "Didn't pass (cutoff=0.0050): cat 550, mi = 0.002, pvalue = 0.013, chromatin remodeling\n",
      "Didn't pass (cutoff=0.0050): cat 696, mi = 0.002, pvalue = 0.011, mitochondrial large ribosomal subunit\n",
      "Passed: cat 2274, mi = 0.002, pvalue = 0.004, innervation\n",
      "Didn't pass (cutoff=0.0050): cat 1867, mi = 0.002, pvalue = 0.007, CENP-A containing nucleosome assembly\n",
      "Didn't pass (cutoff=0.0050): cat 1851, mi = 0.002, pvalue = 0.014, amino acid transmembrane transporter activity\n",
      "Didn't pass (cutoff=0.0050): cat 1853, mi = 0.002, pvalue = 0.021, amino acid transport\n",
      "Didn't pass (cutoff=0.0050): cat 4065, mi = 0.002, pvalue = 0.007, positive regulation of keratinocyte differentiation\n",
      "Didn't pass (cutoff=0.0050): cat 2616, mi = 0.002, pvalue = 0.009, neuromuscular process controlling balance\n",
      "Didn't pass (cutoff=0.0050): cat 2368, mi = 0.002, pvalue = 0.028, ephrin receptor signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 2546, mi = 0.002, pvalue = 0.010, SRP-dependent cotranslational protein targeting to membrane\n",
      "Didn't pass (cutoff=0.0050): cat 2333, mi = 0.002, pvalue = 0.021, retrograde vesicle-mediated transport, Golgi to ER\n",
      "Didn't pass (cutoff=0.0050): cat 2949, mi = 0.002, pvalue = 0.015, negative regulation of cardiac muscle cell apoptotic process\n",
      "Didn't pass (cutoff=0.0050): cat 3671, mi = 0.002, pvalue = 0.025, cation transport\n",
      "Didn't pass (cutoff=0.0050): cat 4230, mi = 0.002, pvalue = 0.028, negative regulation of ERK1 and ERK2 cascade\n",
      "Didn't pass (cutoff=0.0050): cat 2434, mi = 0.002, pvalue = 0.019, acetyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 2874, mi = 0.002, pvalue = 0.022, RNA metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 2136, mi = 0.002, pvalue = 0.020, positive regulation of extrinsic apoptotic signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 2248, mi = 0.002, pvalue = 0.028, bHLH transcription factor binding\n",
      "Didn't pass (cutoff=0.0050): cat 1249, mi = 0.002, pvalue = 0.011, small-subunit processome\n",
      "Didn't pass (cutoff=0.0050): cat 563, mi = 0.002, pvalue = 0.028, calcium-mediated signaling\n",
      "Didn't pass (cutoff=0.0050): cat 1906, mi = 0.002, pvalue = 0.023, lipoprotein metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 640, mi = 0.002, pvalue = 0.021, sodium ion transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 5305, mi = 0.002, pvalue = 0.005, positive regulation of viral transcription\n",
      "Total number of categories evaluated = 72.\n",
      "Number of categories that passed the tests = 8\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 6, 0.249733078377237\n",
      "best merge is 1, 7, 0.607020203213068\n",
      "best merge is 2, 3, 0.86006625944343\n",
      "best merge is 1, 5, 0.961234002368164\n",
      "order classes\n",
      "2 - 0.499466156754474\n",
      "2 - 1.21404040642614\n",
      "2 - 1.72013251888686\n",
      "0 6\n",
      "1 7\n",
      "2 3\n",
      "4\n",
      "5\n",
      "Done.Start drawing\n",
      "9.39\t-11.05\n",
      "insulin receptor signaling pathway, GO:0008286\n",
      "innervation, GO:0060384\n",
      "cellular response to organic substance, GO:0071310\n",
      "trophectodermal cell differentiation, GO:0001829\n",
      "positive regulation of translation, GO:0045727\n",
      "costamere, GO:0043034\n",
      "copper ion binding, GO:0005507\n",
      "cartilage development, GO:0051216\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 6, 0.249733078377237\n",
      "best merge is 1, 7, 0.607020203213068\n",
      "best merge is 2, 3, 0.86006625944343\n",
      "best merge is 1, 5, 0.961234002368164\n",
      "order classes\n",
      "2 - 0.499466156754474\n",
      "2 - 1.21404040642614\n",
      "2 - 1.72013251888686\n",
      "0 6\n",
      "1 7\n",
      "2 3\n",
      "4\n",
      "5\n",
      "Done.-10.4041095890411\t42\t0\n",
      "-0.0890410958904084\t42\t19.1917808219178\n",
      "1.04109589041096\t42\t20.6301369863014\n",
      "1.73972602739726\t42\t21.4520547945205\n",
      "2.25342465753425\t42\t22.027397260274\n",
      "2.68493150684932\t42\t22.4794520547945\n",
      "3.08561643835616\t42\t22.8904109589041\n",
      "3.46575342465754\t42\t23.2808219178082\n",
      "3.87671232876713\t42\t23.6506849315069\n",
      "4.40068493150685\t42\t24.1027397260274\n",
      "13.3493150684932\t42\t24.6986301369863\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6496 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6496\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "591 hits go_terms_num = 181\n",
      "cat 12, mi = 0.00181451, BRF1_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00174331, BRF2_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00159054, POU5F1_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00144312, PPARGC1A_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00132843, POLR3G_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00105738, SMARCA4_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00101619, BDP1_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00075718, XRCC4_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.002, pvalue = 0.145, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.002, pvalue = 0.109, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.002, pvalue = 0.183, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.302, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.001, pvalue = 0.434, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.616, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.623, BDP1_ChIP-seq_regulon\n",
      "Didn't pass: cat 167, mi = 0.001, pvalue = 1.000, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.84\t-11.05\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_encode_tf/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 298\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6547 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6547\t298\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "595 hits go_terms_num = 298\n",
      "cat 183, mi = 0.00370595, chr9p13\n",
      "cat 289, mi = 0.00302467, chr1p35\n",
      "cat 130, mi = 0.00276198, chr9q22\n",
      "cat 122, mi = 0.00250370, chr2q36\n",
      "cat 74, mi = 0.00240309, chr2p13\n",
      "cat 87, mi = 0.00235310, chr1q24\n",
      "cat 14, mi = 0.00229986, chr4q21\n",
      "cat 102, mi = 0.00229297, chr9q31\n",
      "cat 111, mi = 0.00224333, chr17q12\n",
      "cat 260, mi = 0.00215290, chrXq21\n",
      "cat 5, mi = 0.00215176, chr6q25\n",
      "cat 100, mi = 0.00200666, chr6q21\n",
      "cat 2, mi = 0.00199602, chr1p32\n",
      "cat 77, mi = 0.00199047, chr1p22\n",
      "cat 33, mi = 0.00198207, chr12p13\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 183, mi = 0.004, pvalue = 0.000, chr9p13\n",
      "Passed: cat 289, mi = 0.003, pvalue = 0.004, chr1p35\n",
      "Didn't pass (cutoff=0.0050): cat 130, mi = 0.003, pvalue = 0.009, chr9q22\n",
      "Didn't pass (cutoff=0.0050): cat 122, mi = 0.003, pvalue = 0.013, chr2q36\n",
      "Didn't pass (cutoff=0.0050): cat 74, mi = 0.002, pvalue = 0.030, chr2p13\n",
      "Didn't pass (cutoff=0.0050): cat 87, mi = 0.002, pvalue = 0.031, chr1q24\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.002, pvalue = 0.040, chr4q21\n",
      "Didn't pass (cutoff=0.0050): cat 102, mi = 0.002, pvalue = 0.014, chr9q31\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.002, pvalue = 0.044, chr17q12\n",
      "Didn't pass (cutoff=0.0050): cat 260, mi = 0.002, pvalue = 0.009, chrXq21\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.002, pvalue = 0.054, chr6q25\n",
      "Didn't pass (cutoff=0.0050): cat 100, mi = 0.002, pvalue = 0.097, chr6q21\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.002, pvalue = 0.095, chr1p32\n",
      "Didn't pass (cutoff=0.0050): cat 77, mi = 0.002, pvalue = 0.093, chr1p22\n",
      "Didn't pass (cutoff=0.0050): cat 33, mi = 0.002, pvalue = 0.073, chr12p13\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.002, pvalue = 0.100, chr16q12\n",
      "Didn't pass (cutoff=0.0050): cat 165, mi = 0.002, pvalue = 0.102, chr7p14\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.002, pvalue = 0.105, chr5q12\n",
      "Didn't pass (cutoff=0.0050): cat 86, mi = 0.002, pvalue = 0.105, chr3p14\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.055, chr1q43\n",
      "Didn't pass (cutoff=0.0050): cat 211, mi = 0.002, pvalue = 0.087, chrXq26\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.116, chr13q34\n",
      "Total number of categories evaluated = 12.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.84\t-10.88\n",
      "chr1p35\n",
      "chr9p13\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.69656488549618\t42\t0\n",
      "5.52862595419847\t42\t24.6068702290076\n",
      "6.97137404580153\t42\t26.4503816793893\n",
      "7.85305343511451\t42\t27.4923664122137\n",
      "8.49427480916031\t42\t28.2137404580153\n",
      "9.04198473282443\t42\t28.7748091603053\n",
      "9.54961832061069\t42\t29.3091603053435\n",
      "10.030534351145\t42\t29.7900763358779\n",
      "10.5515267175573\t42\t30.2709923664122\n",
      "11.219465648855\t42\t30.8320610687023\n",
      "16.8034351145038\t42\t31.6068702290076\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_c1/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5529\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6492 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6492\t5529\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "590 hits go_terms_num = 5529\n",
      "cat 5006, mi = 0.00383834, VANASSE_BCL2_TARGETS_DN\n",
      "cat 4883, mi = 0.00381157, MCLACHLAN_DENTAL_CARIES_DN\n",
      "cat 5070, mi = 0.00366785, PHONG_TNF_TARGETS_UP\n",
      "cat 2095, mi = 0.00349208, WANG_BARRETTS_ESOPHAGUS_UP\n",
      "cat 567, mi = 0.00348126, RASHI_RESPONSE_TO_IONIZING_RADIATION_2\n",
      "cat 3601, mi = 0.00336704, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "cat 3253, mi = 0.00318994, LUI_THYROID_CANCER_CLUSTER_2\n",
      "cat 3592, mi = 0.00317356, BILBAN_B_CLL_LPL_DN\n",
      "cat 2087, mi = 0.00316971, CHNG_MULTIPLE_MYELOMA_HYPERPLOID_UP\n",
      "cat 2196, mi = 0.00312829, HONRADO_BREAST_CANCER_BRCA1_VS_BRCA2\n",
      "cat 3938, mi = 0.00312364, AMIT_EGF_RESPONSE_60_MCF10A\n",
      "cat 1060, mi = 0.00311874, DOANE_RESPONSE_TO_ANDROGEN_UP\n",
      "cat 1363, mi = 0.00310123, LENAOUR_DENDRITIC_CELL_MATURATION_DN\n",
      "cat 291, mi = 0.00307867, MORI_PRE_BI_LYMPHOCYTE_DN\n",
      "cat 5264, mi = 0.00303664, BIOCARTA_CERAMIDE_PATHWAY\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5006, mi = 0.004, pvalue = 0.000, VANASSE_BCL2_TARGETS_DN\n",
      "Passed: cat 4883, mi = 0.004, pvalue = 0.000, MCLACHLAN_DENTAL_CARIES_DN\n",
      "Passed: cat 5070, mi = 0.004, pvalue = 0.001, PHONG_TNF_TARGETS_UP\n",
      "Passed: cat 2095, mi = 0.003, pvalue = 0.001, WANG_BARRETTS_ESOPHAGUS_UP\n",
      "Passed: cat 3601, mi = 0.003, pvalue = 0.001, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "Passed: cat 3253, mi = 0.003, pvalue = 0.003, LUI_THYROID_CANCER_CLUSTER_2\n",
      "Passed: cat 2087, mi = 0.003, pvalue = 0.001, CHNG_MULTIPLE_MYELOMA_HYPERPLOID_UP\n",
      "Passed: cat 915, mi = 0.003, pvalue = 0.000, BOYAULT_LIVER_CANCER_SUBCLASS_G5_DN\n",
      "Passed: cat 2105, mi = 0.003, pvalue = 0.004, STARK_HYPPOCAMPUS_22Q11_DELETION_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3226, mi = 0.003, pvalue = 0.007, KARLSSON_TGFB1_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5124, mi = 0.003, pvalue = 0.009, KEGG_NOTCH_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 1027, mi = 0.003, pvalue = 0.008, KOKKINAKIS_METHIONINE_DEPRIVATION_96HR_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4525, mi = 0.003, pvalue = 0.011, BARRIER_CANCER_RELAPSE_NORMAL_SAMPLE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4418, mi = 0.003, pvalue = 0.012, GENTILE_UV_RESPONSE_CLUSTER_D8\n",
      "Passed: cat 1379, mi = 0.003, pvalue = 0.003, RIZ_ERYTHROID_DIFFERENTIATION_CCNE1\n",
      "Didn't pass (cutoff=0.0050): cat 1371, mi = 0.003, pvalue = 0.008, KOKKINAKIS_METHIONINE_DEPRIVATION_48HR_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4494, mi = 0.003, pvalue = 0.012, FIGUEROA_AML_METHYLATION_CLUSTER_4_UP\n",
      "Passed: cat 5140, mi = 0.003, pvalue = 0.003, BOYLAN_MULTIPLE_MYELOMA_D_CLUSTER_UP\n",
      "Passed: cat 789, mi = 0.003, pvalue = 0.003, REACTOME_SIGNAL_ATTENUATION\n",
      "Didn't pass (cutoff=0.0050): cat 641, mi = 0.003, pvalue = 0.007, GENTILE_UV_LOW_DOSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5137, mi = 0.003, pvalue = 0.013, REACTOME_RAB_GEFS_EXCHANGE_GTP_FOR_GDP_ON_RABS\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.003, pvalue = 0.009, DURCHDEWALD_SKIN_CARCINOGENESIS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1984, mi = 0.003, pvalue = 0.017, REACTOME_NOTCH3_INTRACELLULAR_DOMAIN_REGULATES_TRANSCRIPTION\n",
      "Didn't pass (cutoff=0.0050): cat 1520, mi = 0.003, pvalue = 0.015, NIKOLSKY_BREAST_CANCER_8Q12_Q22_AMPLICON\n",
      "Didn't pass (cutoff=0.0050): cat 938, mi = 0.003, pvalue = 0.013, BURTON_ADIPOGENESIS_3\n",
      "Passed: cat 3392, mi = 0.003, pvalue = 0.003, THEODOROU_MAMMARY_TUMORIGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 2339, mi = 0.003, pvalue = 0.010, LIANG_HEMATOPOIESIS_STEM_CELL_NUMBER_SMALL_VS_HUGE_DN\n",
      "Passed: cat 5471, mi = 0.003, pvalue = 0.000, REACTOME_APOPTOSIS_INDUCED_DNA_FRAGMENTATION\n",
      "Didn't pass (cutoff=0.0050): cat 1182, mi = 0.003, pvalue = 0.018, PEDERSEN_METASTASIS_BY_ERBB2_ISOFORM_4\n",
      "Didn't pass (cutoff=0.0050): cat 939, mi = 0.003, pvalue = 0.014, LINDGREN_BLADDER_CANCER_WITH_LOH_IN_CHR9Q\n",
      "Didn't pass (cutoff=0.0050): cat 3388, mi = 0.003, pvalue = 0.012, STARK_HYPPOCAMPUS_22Q11_DELETION_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3300, mi = 0.003, pvalue = 0.018, STEARMAN_LUNG_CANCER_EARLY_VS_LATE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2687, mi = 0.003, pvalue = 0.025, HOFMANN_CELL_LYMPHOMA_DN\n",
      "Didn't pass (cutoff=0.0050): cat 331, mi = 0.003, pvalue = 0.021, CAMPS_COLON_CANCER_COPY_NUMBER_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4273, mi = 0.003, pvalue = 0.022, SMID_BREAST_CANCER_ERBB2_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1528, mi = 0.003, pvalue = 0.021, REACTOME_CHROMOSOME_MAINTENANCE\n",
      "Didn't pass (cutoff=0.0050): cat 581, mi = 0.003, pvalue = 0.020, ASGHARZADEH_NEUROBLASTOMA_POOR_SURVIVAL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2237, mi = 0.002, pvalue = 0.022, REACTOME_MITOCHONDRIAL_TRANSLATION\n",
      "Didn't pass (cutoff=0.0050): cat 5278, mi = 0.002, pvalue = 0.021, BROWNE_HCMV_INFECTION_48HR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1614, mi = 0.002, pvalue = 0.019, LANDIS_BREAST_CANCER_PROGRESSION_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3233, mi = 0.002, pvalue = 0.010, FLOTHO_PEDIATRIC_ALL_THERAPY_RESPONSE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1150, mi = 0.002, pvalue = 0.025, PID_HIF2PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 546, mi = 0.002, pvalue = 0.024, REACTOME_CARDIAC_CONDUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 4752, mi = 0.002, pvalue = 0.021, ZHONG_RESPONSE_TO_AZACITIDINE_AND_TSA_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4810, mi = 0.002, pvalue = 0.027, HUANG_DASATINIB_RESISTANCE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4763, mi = 0.002, pvalue = 0.030, MORI_IMMATURE_B_LYMPHOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 395, mi = 0.002, pvalue = 0.021, WU_APOPTOSIS_BY_CDKN1A_VIA_TP53\n",
      "Didn't pass (cutoff=0.0050): cat 2841, mi = 0.002, pvalue = 0.011, LEE_LIVER_CANCER_CIPROFIBRATE_UP\n",
      "Total number of categories evaluated = 107.\n",
      "Number of categories that passed the tests = 14\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 8, 12, 0.256115800624741\n",
      "best merge is 5, 11, 0.323425231050907\n",
      "best merge is 1, 3, 0.355557277665718\n",
      "best merge is 8, 9, 0.387022540032098\n",
      "best merge is 0, 2, 0.419741552658523\n",
      "best merge is 0, 8, 0.605239515147851\n",
      "best merge is 10, 13, 0.627494905022141\n",
      "best merge is 1, 4, 0.637980644575175\n",
      "best merge is 0, 7, 0.785839924483174\n",
      "best merge is 5, 10, 0.97787227049244\n",
      "order classes\n",
      "6 - 1.16803285234357\n",
      "3 - 1.69194518040185\n",
      "2 - 1.13198830867817\n",
      "2 - 2.19623216757749\n",
      "5 11\n",
      "1 3 4\n",
      "0 2 8 12 9 7\n",
      "6\n",
      "10 13\n",
      "Done.Start drawing\n",
      "4.84\t-10.88\n",
      "LUI_THYROID_CANCER_CLUSTER_2\n",
      "REACTOME_SIGNAL_ATTENUATION\n",
      "STARK_HYPPOCAMPUS_22Q11_DELETION_UP\n",
      "THEODOROU_MAMMARY_TUMORIGENESIS\n",
      "RIZ_ERYTHROID_DIFFERENTIATION_CCNE1\n",
      "BOYAULT_LIVER_CANCER_SUBCLASS_G5_DN\n",
      "BOYLAN_MULTIPLE_MYELOMA_D_CLUSTER_UP\n",
      "WANG_BARRETTS_ESOPHAGUS_UP\n",
      "PHONG_TNF_TARGETS_UP\n",
      "LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "VANASSE_BCL2_TARGETS_DN\n",
      "CHNG_MULTIPLE_MYELOMA_HYPERPLOID_UP\n",
      "MCLACHLAN_DENTAL_CARIES_DN\n",
      "REACTOME_APOPTOSIS_INDUCED_DNA_FRAGMENTATION\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 8, 12, 0.256115800624741\n",
      "best merge is 5, 11, 0.323425231050907\n",
      "best merge is 1, 3, 0.355557277665718\n",
      "best merge is 8, 9, 0.387022540032098\n",
      "best merge is 0, 2, 0.419741552658523\n",
      "best merge is 0, 8, 0.605239515147851\n",
      "best merge is 10, 13, 0.627494905022141\n",
      "best merge is 1, 4, 0.637980644575175\n",
      "best merge is 0, 7, 0.785839924483174\n",
      "best merge is 5, 10, 0.97787227049244\n",
      "order classes\n",
      "6 - 1.16803285234357\n",
      "3 - 1.69194518040185\n",
      "2 - 1.13198830867817\n",
      "2 - 2.19623216757749\n",
      "5 11\n",
      "1 3 4\n",
      "0 2 8 12 9 7\n",
      "6\n",
      "10 13\n",
      "Done.-7.68320610687023\t42\t0\n",
      "5.55534351145038\t42\t24.6335877862595\n",
      "6.99809160305344\t42\t26.4770992366412\n",
      "7.86641221374046\t42\t27.5190839694656\n",
      "8.50763358778626\t42\t28.2137404580153\n",
      "9.05534351145038\t42\t28.8015267175573\n",
      "9.54961832061069\t42\t29.3091603053435\n",
      "10.030534351145\t42\t29.7900763358779\n",
      "10.5515267175573\t42\t30.2709923664122\n",
      "11.219465648855\t42\t30.8320610687023\n",
      "16.8034351145038\t42\t31.6068702290076\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_c2/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 3733\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6535 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6535\t3733\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "594 hits go_terms_num = 3733\n",
      "cat 992, mi = 0.00372393, MIR7154_3P\n",
      "cat 2232, mi = 0.00369955, ATGGYGGA_UNKNOWN\n",
      "cat 2756, mi = 0.00365946, MIR499A_5P\n",
      "cat 3627, mi = 0.00363864, MIR134_3P\n",
      "cat 1202, mi = 0.00349746, MIR4709_5P\n",
      "cat 1005, mi = 0.00342658, CCATCCA_MIR432\n",
      "cat 831, mi = 0.00335984, MIR548AT_3P\n",
      "cat 2459, mi = 0.00332111, MIR7157_3P\n",
      "cat 2132, mi = 0.00331001, MIR4529_5P\n",
      "cat 2861, mi = 0.00328516, NKX2_8_TARGET_GENES\n",
      "cat 1150, mi = 0.00327748, MIR8065\n",
      "cat 3152, mi = 0.00323874, MIR6737_3P\n",
      "cat 614, mi = 0.00320836, MIR541_5P\n",
      "cat 315, mi = 0.00314986, MIR6862_5P\n",
      "cat 859, mi = 0.00314240, MIR548AY_3P\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 992, mi = 0.004, pvalue = 0.000, MIR7154_3P\n",
      "Passed: cat 2232, mi = 0.004, pvalue = 0.000, ATGGYGGA_UNKNOWN\n",
      "Passed: cat 2756, mi = 0.004, pvalue = 0.000, MIR499A_5P\n",
      "Passed: cat 3627, mi = 0.004, pvalue = 0.001, MIR134_3P\n",
      "Passed: cat 1202, mi = 0.003, pvalue = 0.001, MIR4709_5P\n",
      "Passed: cat 1005, mi = 0.003, pvalue = 0.001, CCATCCA_MIR432\n",
      "Passed: cat 831, mi = 0.003, pvalue = 0.001, MIR548AT_3P\n",
      "Passed: cat 2132, mi = 0.003, pvalue = 0.001, MIR4529_5P\n",
      "Passed: cat 2861, mi = 0.003, pvalue = 0.002, NKX2_8_TARGET_GENES\n",
      "Passed: cat 614, mi = 0.003, pvalue = 0.002, MIR541_5P\n",
      "Passed: cat 432, mi = 0.003, pvalue = 0.001, MIR6869_5P\n",
      "Passed: cat 2178, mi = 0.003, pvalue = 0.004, MIR4721\n",
      "Passed: cat 3170, mi = 0.003, pvalue = 0.002, ATAAGCT_MIR21\n",
      "Passed: cat 2717, mi = 0.003, pvalue = 0.001, MIR6744_5P\n",
      "Passed: cat 2461, mi = 0.003, pvalue = 0.004, MIR424_3P\n",
      "Didn't pass (cutoff=0.0050): cat 821, mi = 0.003, pvalue = 0.010, MIR4633_3P_MIR6500_5P\n",
      "Didn't pass (cutoff=0.0050): cat 3709, mi = 0.003, pvalue = 0.012, RUVBL1_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 2739, mi = 0.003, pvalue = 0.018, MIR3613_5P\n",
      "Passed: cat 2439, mi = 0.003, pvalue = 0.003, MIR6816_3P\n",
      "Didn't pass (cutoff=0.0050): cat 1877, mi = 0.003, pvalue = 0.021, MIR668_3P\n",
      "Didn't pass (cutoff=0.0050): cat 2497, mi = 0.003, pvalue = 0.019, MIR15B_3P\n",
      "Didn't pass (cutoff=0.0050): cat 2046, mi = 0.003, pvalue = 0.023, HEY2_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 113, mi = 0.002, pvalue = 0.018, MIR6796_3P\n",
      "Didn't pass (cutoff=0.0050): cat 2366, mi = 0.002, pvalue = 0.026, MIR6803_5P\n",
      "Didn't pass (cutoff=0.0050): cat 729, mi = 0.002, pvalue = 0.025, MIR3675_3P\n",
      "Didn't pass (cutoff=0.0050): cat 784, mi = 0.002, pvalue = 0.030, ZSCAN5B_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 125, mi = 0.002, pvalue = 0.033, MIR550A_3P\n",
      "Didn't pass (cutoff=0.0050): cat 1349, mi = 0.002, pvalue = 0.025, MIR558\n",
      "Didn't pass (cutoff=0.0050): cat 2714, mi = 0.002, pvalue = 0.038, PRMT5_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 2228, mi = 0.002, pvalue = 0.021, SQSTM1_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 266, mi = 0.002, pvalue = 0.043, MIR555\n",
      "Didn't pass (cutoff=0.0050): cat 2508, mi = 0.002, pvalue = 0.035, ATGCACG_MIR517B\n",
      "Didn't pass (cutoff=0.0050): cat 296, mi = 0.002, pvalue = 0.049, MIR628_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2532, mi = 0.002, pvalue = 0.044, MIR4453\n",
      "Didn't pass (cutoff=0.0050): cat 3345, mi = 0.002, pvalue = 0.046, MIR4538\n",
      "Didn't pass (cutoff=0.0050): cat 2374, mi = 0.002, pvalue = 0.044, MIR1227_3P\n",
      "Didn't pass (cutoff=0.0050): cat 2385, mi = 0.002, pvalue = 0.047, ZNF165_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 407, mi = 0.002, pvalue = 0.042, MIR192_3P\n",
      "Didn't pass (cutoff=0.0050): cat 3178, mi = 0.002, pvalue = 0.058, MIR10395_5P\n",
      "Total number of categories evaluated = 129.\n",
      "Number of categories that passed the tests = 16\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 10, 11, 0.145304925319119\n",
      "best merge is 6, 7, 0.297283864828014\n",
      "best merge is 3, 4, 0.313805759762048\n",
      "best merge is 0, 12, 0.342503918123866\n",
      "best merge is 10, 14, 0.407470519557301\n",
      "best merge is 9, 13, 0.543717742573295\n",
      "best merge is 6, 8, 0.571840529675836\n",
      "best merge is 2, 5, 0.584112162778379\n",
      "best merge is 0, 1, 0.65672101516188\n",
      "best merge is 2, 3, 0.801343363624779\n",
      "best merge is 2, 6, 0.918588572615601\n",
      "best merge is 0, 2, 0.981065504037469\n",
      "order classes\n",
      "3 - 1.96260260556756\n",
      "7 - 1.54563910648223\n",
      "2 - 2.17487097029318\n",
      "3 - 1.13806929118071\n",
      "10 11 14\n",
      "2 5 3 4 6 7 8\n",
      "9 13\n",
      "0 12 1\n",
      "15\n",
      "Done.Start drawing\n",
      "4.84\t-10.88\n",
      "ATGGYGGA_UNKNOWN\n",
      "MIR7154_3P\n",
      "ATAAGCT_MIR21\n",
      "MIR134_3P\n",
      "MIR4709_5P\n",
      "MIR4529_5P\n",
      "NKX2_8_TARGET_GENES\n",
      "MIR499A_5P\n",
      "MIR6869_5P\n",
      "MIR4721\n",
      "MIR424_3P\n",
      "MIR548AT_3P\n",
      "CCATCCA_MIR432\n",
      "MIR541_5P\n",
      "MIR6744_5P\n",
      "MIR6816_3P\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 10, 11, 0.145304925319119\n",
      "best merge is 6, 7, 0.297283864828014\n",
      "best merge is 3, 4, 0.313805759762048\n",
      "best merge is 0, 12, 0.342503918123866\n",
      "best merge is 10, 14, 0.407470519557301\n",
      "best merge is 9, 13, 0.543717742573295\n",
      "best merge is 6, 8, 0.571840529675836\n",
      "best merge is 2, 5, 0.584112162778379\n",
      "best merge is 0, 1, 0.65672101516188\n",
      "best merge is 2, 3, 0.801343363624779\n",
      "best merge is 2, 6, 0.918588572615601\n",
      "best merge is 0, 2, 0.981065504037469\n",
      "order classes\n",
      "3 - 1.96260260556756\n",
      "7 - 1.54563910648223\n",
      "2 - 2.17487097029318\n",
      "3 - 1.13806929118071\n",
      "10 11 14\n",
      "2 5 3 4 6 7 8\n",
      "9 13\n",
      "0 12 1\n",
      "15\n",
      "Done.-7.69656488549618\t42\t0\n",
      "5.52862595419847\t42\t24.6068702290076\n",
      "6.97137404580153\t42\t26.4503816793893\n",
      "7.85305343511451\t42\t27.4923664122137\n",
      "8.49427480916031\t42\t28.2137404580153\n",
      "9.04198473282443\t42\t28.7748091603053\n",
      "9.54961832061069\t42\t29.3091603053435\n",
      "10.030534351145\t42\t29.7900763358779\n",
      "10.5515267175573\t42\t30.2709923664122\n",
      "11.219465648855\t42\t30.8320610687023\n",
      "16.8034351145038\t42\t31.6068702290076\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_c3/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3839 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3839\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "349 hits go_terms_num = 858\n",
      "cat 490, mi = 0.00561013, MODULE_36\n",
      "cat 109, mi = 0.00548778, MORF_PPP1CA\n",
      "cat 709, mi = 0.00535490, MODULE_239\n",
      "cat 527, mi = 0.00519460, MORF_CTBP1\n",
      "cat 516, mi = 0.00509511, MODULE_93\n",
      "cat 415, mi = 0.00506926, MORF_TPT1\n",
      "cat 95, mi = 0.00472043, GNF2_CARD15\n",
      "cat 482, mi = 0.00470278, GNF2_CD1D\n",
      "cat 834, mi = 0.00467374, GNF2_NPM1\n",
      "cat 267, mi = 0.00463536, MODULE_493\n",
      "cat 165, mi = 0.00449635, MORF_G22P1\n",
      "cat 66, mi = 0.00449375, MORF_DAP\n",
      "cat 542, mi = 0.00448321, GCM_RING1\n",
      "cat 162, mi = 0.00435160, MORF_NME2\n",
      "cat 332, mi = 0.00432740, GNF2_DDX5\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 490, mi = 0.006, pvalue = 0.001, MODULE_36\n",
      "Passed: cat 109, mi = 0.005, pvalue = 0.002, MORF_PPP1CA\n",
      "Passed: cat 709, mi = 0.005, pvalue = 0.002, MODULE_239\n",
      "Passed: cat 415, mi = 0.005, pvalue = 0.002, MORF_TPT1\n",
      "Didn't pass (cutoff=0.0050): cat 95, mi = 0.005, pvalue = 0.008, GNF2_CARD15\n",
      "Passed: cat 482, mi = 0.005, pvalue = 0.003, GNF2_CD1D\n",
      "Didn't pass (cutoff=0.0050): cat 267, mi = 0.005, pvalue = 0.010, MODULE_493\n",
      "Didn't pass (cutoff=0.0050): cat 542, mi = 0.004, pvalue = 0.015, GCM_RING1\n",
      "Didn't pass (cutoff=0.0050): cat 514, mi = 0.004, pvalue = 0.028, GNF2_ICAM3\n",
      "Didn't pass (cutoff=0.0050): cat 341, mi = 0.004, pvalue = 0.028, CAR_TNFRSF25\n",
      "Didn't pass (cutoff=0.0050): cat 219, mi = 0.004, pvalue = 0.031, MODULE_332\n",
      "Didn't pass (cutoff=0.0050): cat 424, mi = 0.004, pvalue = 0.030, GNF2_SNRK\n",
      "Didn't pass (cutoff=0.0050): cat 523, mi = 0.004, pvalue = 0.040, MORF_XPC\n",
      "Didn't pass (cutoff=0.0050): cat 182, mi = 0.004, pvalue = 0.038, GNF2_ATM\n",
      "Didn't pass (cutoff=0.0050): cat 352, mi = 0.004, pvalue = 0.049, GCM_BECN1\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.004, pvalue = 0.048, GNF2_DENR\n",
      "Didn't pass (cutoff=0.0050): cat 679, mi = 0.004, pvalue = 0.051, GNF2_PTPRC\n",
      "Didn't pass (cutoff=0.0050): cat 855, mi = 0.004, pvalue = 0.070, MODULE_451\n",
      "Didn't pass (cutoff=0.0050): cat 681, mi = 0.004, pvalue = 0.022, MODULE_76\n",
      "Didn't pass (cutoff=0.0050): cat 203, mi = 0.004, pvalue = 0.065, MODULE_403\n",
      "Didn't pass (cutoff=0.0050): cat 767, mi = 0.004, pvalue = 0.064, GNF2_CBFB\n",
      "Didn't pass (cutoff=0.0050): cat 595, mi = 0.004, pvalue = 0.054, GCM_NCAM1\n",
      "Didn't pass (cutoff=0.0050): cat 59, mi = 0.003, pvalue = 0.028, GCM_AIP\n",
      "Didn't pass (cutoff=0.0050): cat 91, mi = 0.003, pvalue = 0.089, MORF_PTPN9\n",
      "Didn't pass (cutoff=0.0050): cat 479, mi = 0.003, pvalue = 0.015, GNF2_PA2G4\n",
      "Didn't pass (cutoff=0.0050): cat 370, mi = 0.003, pvalue = 0.092, MODULE_322\n",
      "Total number of categories evaluated = 39.\n",
      "Number of categories that passed the tests = 5\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 4, 0.300702041716451\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.Start drawing\n",
      "3.57\t-10.88\n",
      "MODULE_36\n",
      "MODULE_239\n",
      "MORF_TPT1\n",
      "GNF2_CD1D\n",
      "MORF_PPP1CA\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 4, 0.300702041716451\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "Done.-6.48442906574394\t42\t0\n",
      "7.9757785467128\t42\t27.0311418685121\n",
      "9.47266435986159\t42\t28.9204152249135\n",
      "10.3737024221453\t42\t30.0249134948097\n",
      "11.042214532872\t42\t30.722491349481\n",
      "11.6525951557093\t42\t31.361937716263\n",
      "12.1903114186851\t42\t31.9432525951557\n",
      "12.6698961937716\t42\t32.4373702422145\n",
      "13.2076124567474\t42\t32.9024221453287\n",
      "13.8906574394464\t42\t33.5128027681661\n",
      "18.1342560553633\t42\t34.2685121107266\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_c4/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 10191\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 5908 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 5908\t10191\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "537 hits go_terms_num = 10191\n",
      "cat 7701, mi = 0.00477334, GO_EYE_MORPHOGENESIS\n",
      "cat 4975, mi = 0.00372673, GO_STRUCTURAL_CONSTITUENT_OF_RIBOSOME\n",
      "cat 7149, mi = 0.00370339, GO_CAMERA_TYPE_EYE_MORPHOGENESIS\n",
      "cat 7339, mi = 0.00365825, GO_COSTAMERE\n",
      "cat 4829, mi = 0.00355890, GO_SCHAFFER_COLLATERAL_CA1_SYNAPSE\n",
      "cat 1685, mi = 0.00352722, GO_DNA_CATABOLIC_PROCESS\n",
      "cat 6002, mi = 0.00341833, GO_RESPONSE_TO_IRON_ION\n",
      "cat 8824, mi = 0.00339375, GO_REGULATION_OF_PEPTIDYL_SERINE_PHOSPHORYLATION\n",
      "cat 2079, mi = 0.00337199, GO_RECEPTOR_TYROSINE_KINASE_BINDING\n",
      "cat 9862, mi = 0.00332852, GO_REGULATION_OF_CYTOPLASMIC_TRANSLATION\n",
      "cat 2086, mi = 0.00331553, GO_DNA_CATABOLIC_PROCESS_ENDONUCLEOLYTIC\n",
      "cat 4516, mi = 0.00329787, GO_ENTRY_INTO_HOST\n",
      "cat 776, mi = 0.00326134, GO_REGULATION_OF_STEROID_METABOLIC_PROCESS\n",
      "cat 848, mi = 0.00325417, GO_L_ALPHA_AMINO_ACID_TRANSMEMBRANE_TRANSPORT\n",
      "cat 5008, mi = 0.00323364, GO_POSITIVE_REGULATION_OF_TRANSCRIPTION_OF_NOTCH_RECEPTOR_TARGET\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 7701, mi = 0.005, pvalue = 0.000, GO_EYE_MORPHOGENESIS\n",
      "Passed: cat 4975, mi = 0.004, pvalue = 0.001, GO_STRUCTURAL_CONSTITUENT_OF_RIBOSOME\n",
      "Passed: cat 7339, mi = 0.004, pvalue = 0.002, GO_COSTAMERE\n",
      "Passed: cat 4829, mi = 0.004, pvalue = 0.001, GO_SCHAFFER_COLLATERAL_CA1_SYNAPSE\n",
      "Passed: cat 1685, mi = 0.004, pvalue = 0.003, GO_DNA_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2079, mi = 0.003, pvalue = 0.006, GO_RECEPTOR_TYROSINE_KINASE_BINDING\n",
      "Passed: cat 9862, mi = 0.003, pvalue = 0.001, GO_REGULATION_OF_CYTOPLASMIC_TRANSLATION\n",
      "Passed: cat 4516, mi = 0.003, pvalue = 0.005, GO_ENTRY_INTO_HOST\n",
      "Didn't pass (cutoff=0.0050): cat 776, mi = 0.003, pvalue = 0.006, GO_REGULATION_OF_STEROID_METABOLIC_PROCESS\n",
      "Passed: cat 5008, mi = 0.003, pvalue = 0.004, GO_POSITIVE_REGULATION_OF_TRANSCRIPTION_OF_NOTCH_RECEPTOR_TARGET\n",
      "Didn't pass (cutoff=0.0050): cat 9598, mi = 0.003, pvalue = 0.007, GO_RESPONSE_TO_CAMP\n",
      "Didn't pass (cutoff=0.0050): cat 5420, mi = 0.003, pvalue = 0.006, GO_REGULATION_OF_SPINDLE_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 4764, mi = 0.003, pvalue = 0.008, GO_POSITIVE_REGULATION_OF_SMALL_MOLECULE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4909, mi = 0.003, pvalue = 0.008, GO_PROTEIN_K63_LINKED_UBIQUITINATION\n",
      "Passed: cat 6250, mi = 0.003, pvalue = 0.003, GO_FILAMENTOUS_ACTIN\n",
      "Passed: cat 9390, mi = 0.003, pvalue = 0.001, GO_COPPER_ION_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 4527, mi = 0.003, pvalue = 0.009, GO_CELLULAR_RESPONSE_TO_CARBOHYDRATE_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.003, pvalue = 0.008, GO_MRNA_CLEAVAGE_FACTOR_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 1856, mi = 0.003, pvalue = 0.012, GO_SH2_DOMAIN_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 9721, mi = 0.003, pvalue = 0.014, GO_RESPONSE_TO_AMINO_ACID_STARVATION\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.003, pvalue = 0.012, GO_GOLGI_CISTERNA_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 10168, mi = 0.003, pvalue = 0.014, GO_TRANSLESION_SYNTHESIS\n",
      "Didn't pass (cutoff=0.0050): cat 6742, mi = 0.003, pvalue = 0.017, GO_ORGANELLE_TRANSPORT_ALONG_MICROTUBULE\n",
      "Passed: cat 5358, mi = 0.003, pvalue = 0.002, GO_CATION_CATION_ANTIPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 9100, mi = 0.003, pvalue = 0.009, GO_ESTABLISHMENT_OF_EPITHELIAL_CELL_POLARITY\n",
      "Didn't pass (cutoff=0.0050): cat 1131, mi = 0.003, pvalue = 0.016, GO_NEGATIVE_REGULATION_OF_MRNA_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 1950, mi = 0.003, pvalue = 0.009, GO_ADRENERGIC_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 5977, mi = 0.003, pvalue = 0.012, GO_TRANSMEMBRANE_RECEPTOR_PROTEIN_TYROSINE_KINASE_ADAPTOR_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 3743, mi = 0.003, pvalue = 0.014, GO_NEGATIVE_REGULATION_OF_JNK_CASCADE\n",
      "Didn't pass (cutoff=0.0050): cat 181, mi = 0.003, pvalue = 0.013, GO_RETINOIC_ACID_RECEPTOR_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 9535, mi = 0.003, pvalue = 0.018, GO_POSITIVE_REGULATION_OF_CARBOHYDRATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5689, mi = 0.003, pvalue = 0.012, GO_PYRIDINE_CONTAINING_COMPOUND_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 2645, mi = 0.003, pvalue = 0.018, GO_TERMINATION_OF_RNA_POLYMERASE_II_TRANSCRIPTION\n",
      "Didn't pass (cutoff=0.0050): cat 4778, mi = 0.003, pvalue = 0.020, GO_MOTILE_CILIUM\n",
      "Didn't pass (cutoff=0.0050): cat 8515, mi = 0.003, pvalue = 0.021, GO_GLUCOSE_IMPORT\n",
      "Didn't pass (cutoff=0.0050): cat 777, mi = 0.003, pvalue = 0.013, GO_TOLL_LIKE_RECEPTOR_2_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 6451, mi = 0.003, pvalue = 0.013, GO_REGULATION_OF_TOLL_LIKE_RECEPTOR_2_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 9716, mi = 0.003, pvalue = 0.005, GO_SMALL_SUBUNIT_PROCESSOME\n",
      "Didn't pass (cutoff=0.0050): cat 9961, mi = 0.003, pvalue = 0.017, GO_INSULIN_RECEPTOR_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2539, mi = 0.003, pvalue = 0.023, GO_NEGATIVE_REGULATION_OF_ERK1_AND_ERK2_CASCADE\n",
      "Didn't pass (cutoff=0.0050): cat 3511, mi = 0.003, pvalue = 0.021, GO_POSITIVE_REGULATION_OF_FATTY_ACID_OXIDATION\n",
      "Passed: cat 4492, mi = 0.003, pvalue = 0.003, GO_REGULATION_OF_CHROMATIN_ASSEMBLY_OR_DISASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 2302, mi = 0.003, pvalue = 0.024, GO_CERAMIDE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 3442, mi = 0.003, pvalue = 0.023, GO_PROTEIN_TETRAMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 4732, mi = 0.003, pvalue = 0.026, GO_MITOTIC_CYTOKINESIS\n",
      "Didn't pass (cutoff=0.0050): cat 4599, mi = 0.003, pvalue = 0.025, GO_NEGATIVE_REGULATION_OF_MACROAUTOPHAGY\n",
      "Didn't pass (cutoff=0.0050): cat 2930, mi = 0.003, pvalue = 0.031, GO_REGULATION_OF_UBIQUITIN_PROTEIN_TRANSFERASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 6851, mi = 0.003, pvalue = 0.029, GO_PROTEIN_HETEROOLIGOMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 7067, mi = 0.003, pvalue = 0.026, GO_PROTEIN_K48_LINKED_UBIQUITINATION\n",
      "Didn't pass (cutoff=0.0050): cat 806, mi = 0.003, pvalue = 0.030, GO_REGULATION_OF_FATTY_ACID_OXIDATION\n",
      "Didn't pass (cutoff=0.0050): cat 8614, mi = 0.003, pvalue = 0.028, GO_REGULATION_OF_FATTY_ACID_BETA_OXIDATION\n",
      "Didn't pass (cutoff=0.0050): cat 9409, mi = 0.003, pvalue = 0.028, GO_CENTROMERE_COMPLEX_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 5140, mi = 0.003, pvalue = 0.025, GO_IMMUNOLOGICAL_SYNAPSE\n",
      "Didn't pass (cutoff=0.0050): cat 695, mi = 0.003, pvalue = 0.027, GO_POSITIVE_REGULATION_OF_FATTY_ACID_BETA_OXIDATION\n",
      "Didn't pass (cutoff=0.0050): cat 3161, mi = 0.003, pvalue = 0.027, GO_POSITIVE_REGULATION_OF_FATTY_ACID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 8093, mi = 0.003, pvalue = 0.030, GO_POSITIVE_REGULATION_OF_GLUCOSE_TRANSMEMBRANE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 2889, mi = 0.003, pvalue = 0.019, GO_UBIQUITIN_LIKE_PROTEIN_CONJUGATING_ENZYME_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 4023, mi = 0.003, pvalue = 0.031, GO_NEGATIVE_REGULATION_OF_SMOOTH_MUSCLE_CELL_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 6477, mi = 0.003, pvalue = 0.030, GO_CRANIAL_NERVE_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 9359, mi = 0.003, pvalue = 0.030, GO_POSITIVE_REGULATION_OF_LIPID_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 214, mi = 0.003, pvalue = 0.038, GO_REGULATION_OF_CD4_POSITIVE_ALPHA_BETA_T_CELL_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 7427, mi = 0.003, pvalue = 0.037, GO_RNA_STABILIZATION\n",
      "Total number of categories evaluated = 131.\n",
      "Number of categories that passed the tests = 12\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 11, 0.134389355682468\n",
      "best merge is 9, 10, 0.143184472814624\n",
      "best merge is 4, 6, 0.171848895169134\n",
      "best merge is 0, 5, 0.498774386969403\n",
      "best merge is 1, 4, 0.591517467992419\n",
      "best merge is 3, 9, 0.667347676128862\n",
      "best merge is 0, 7, 0.761630813354046\n",
      "best merge is 1, 8, 0.939452190726943\n",
      "order classes\n",
      "3 - 1.79736534549111\n",
      "4 - 0.844739831249247\n",
      "3 - 1.31367095561986\n",
      "1 4 11 6\n",
      "0 5 7\n",
      "2\n",
      "3 9 10\n",
      "8\n",
      "Done.Start drawing\n",
      "4.84\t-10.88\n",
      "GO_SCHAFFER_COLLATERAL_CA1_SYNAPSE\n",
      "GO_POSITIVE_REGULATION_OF_TRANSCRIPTION_OF_NOTCH_RECEPTOR_TARGET\n",
      "GO_EYE_MORPHOGENESIS\n",
      "GO_COSTAMERE\n",
      "GO_STRUCTURAL_CONSTITUENT_OF_RIBOSOME\n",
      "GO_COPPER_ION_BINDING\n",
      "GO_CATION_CATION_ANTIPORTER_ACTIVITY\n",
      "GO_REGULATION_OF_CYTOPLASMIC_TRANSLATION\n",
      "GO_FILAMENTOUS_ACTIN\n",
      "GO_DNA_CATABOLIC_PROCESS\n",
      "GO_REGULATION_OF_CHROMATIN_ASSEMBLY_OR_DISASSEMBLY\n",
      "GO_ENTRY_INTO_HOST\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 11, 0.134389355682468\n",
      "best merge is 9, 10, 0.143184472814624\n",
      "best merge is 4, 6, 0.171848895169134\n",
      "best merge is 0, 5, 0.498774386969403\n",
      "best merge is 1, 4, 0.591517467992419\n",
      "best merge is 3, 9, 0.667347676128862\n",
      "best merge is 0, 7, 0.761630813354046\n",
      "best merge is 1, 8, 0.939452190726943\n",
      "order classes\n",
      "3 - 1.79736534549111\n",
      "4 - 0.844739831249247\n",
      "3 - 1.31367095561986\n",
      "1 4 11 6\n",
      "0 5 7\n",
      "2\n",
      "3 9 10\n",
      "8\n",
      "Done.-7.66984732824428\t42\t0\n",
      "5.55534351145038\t42\t24.6603053435114\n",
      "6.95801526717558\t42\t26.4503816793893\n",
      "7.8263358778626\t42\t27.4656488549618\n",
      "8.48091603053435\t42\t28.1870229007634\n",
      "9.04198473282443\t42\t28.7748091603053\n",
      "9.54961832061069\t42\t29.3091603053435\n",
      "10.030534351145\t42\t29.7900763358779\n",
      "10.5515267175573\t42\t30.2709923664122\n",
      "11.219465648855\t42\t30.8320610687023\n",
      "16.8034351145038\t42\t31.6068702290076\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_c5/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3784 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3784\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "344 hits go_terms_num = 189\n",
      "cat 5, mi = 0.00559885, CYCLIN_D1_KE_.V1_DN\n",
      "cat 166, mi = 0.00522511, IL21_UP.V1_DN\n",
      "cat 153, mi = 0.00517090, VEGF_A_UP.V1_DN\n",
      "cat 183, mi = 0.00463539, BMI1_DN.V1_DN\n",
      "cat 63, mi = 0.00453714, GCNP_SHH_UP_EARLY.V1_UP\n",
      "cat 121, mi = 0.00437380, PKCA_DN.V1_DN\n",
      "cat 29, mi = 0.00413775, BRCA1_DN.V1_UP\n",
      "cat 113, mi = 0.00409447, CYCLIN_D1_UP.V1_DN\n",
      "cat 74, mi = 0.00399756, CAMP_UP.V1_DN\n",
      "cat 98, mi = 0.00392244, IL21_UP.V1_UP\n",
      "cat 46, mi = 0.00373331, IL2_UP.V1_DN\n",
      "cat 88, mi = 0.00362942, MEL18_DN.V1_DN\n",
      "cat 20, mi = 0.00360360, TBK1.DN.48HRS_UP\n",
      "cat 18, mi = 0.00359893, MTOR_UP.N4.V1_UP\n",
      "cat 159, mi = 0.00358395, ALK_DN.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 5, mi = 0.006, pvalue = 0.003, CYCLIN_D1_KE_.V1_DN\n",
      "Passed: cat 166, mi = 0.005, pvalue = 0.005, IL21_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 183, mi = 0.005, pvalue = 0.011, BMI1_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 121, mi = 0.004, pvalue = 0.020, PKCA_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 74, mi = 0.004, pvalue = 0.023, CAMP_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 98, mi = 0.004, pvalue = 0.038, IL21_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.004, pvalue = 0.078, MEL18_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.004, pvalue = 0.073, TBK1.DN.48HRS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 18, mi = 0.004, pvalue = 0.056, MTOR_UP.N4.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.004, pvalue = 0.073, ALK_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 76, mi = 0.004, pvalue = 0.072, SNF5_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.004, pvalue = 0.054, CYCLIN_D1_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 99, mi = 0.004, pvalue = 0.058, HOXA9_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 143, mi = 0.004, pvalue = 0.070, CSR_LATE_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.003, pvalue = 0.090, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 52, mi = 0.003, pvalue = 0.072, EGFR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.003, pvalue = 0.104, JAK2_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.003, pvalue = 0.084, P53_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 91, mi = 0.003, pvalue = 0.092, BMI1_DN_MEL18_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 83, mi = 0.003, pvalue = 0.096, WNT_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 72, mi = 0.003, pvalue = 0.114, KRAS.LUNG_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.003, pvalue = 0.085, JNK_DN.V1_DN\n",
      "Total number of categories evaluated = 17.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.55\t-10.88\n",
      "CYCLIN_D1_KE_.V1_DN\n",
      "IL21_UP.V1_DN\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.47893713545042\t42\t0\n",
      "6.03564484769929\t42\t25.0421257290992\n",
      "7.54633830200908\t42\t27.0291639662994\n",
      "8.43097861309138\t42\t28.0635126377187\n",
      "9.08425145819832\t42\t28.798444588464\n",
      "9.64225534672716\t42\t29.3700583279326\n",
      "10.1594296824368\t42\t29.9144523655217\n",
      "10.649384316267\t42\t30.4044069993519\n",
      "11.1801685029164\t42\t30.8943616331821\n",
      "11.8470511989631\t42\t31.4659753726507\n",
      "17.1140635126377\t42\t32.2281270252754\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_c6/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6489 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6489\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "590 hits go_terms_num = 4872\n",
      "cat 1028, mi = 0.00423313, GSE11057_EFF_MEM_VS_CENT_MEM_CD4_TCELL_DN\n",
      "cat 3030, mi = 0.00411493, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 635, mi = 0.00404140, GSE8384_CTRL_VS_B_ABORTUS_4H_MAC_CELL_LINE_DN\n",
      "cat 318, mi = 0.00399569, GSE2585_THYMIC_DC_VS_THYMIC_MACROPHAGE_DN\n",
      "cat 1291, mi = 0.00389192, GSE11057_NAIVE_CD4_VS_PBMC_CD4_TCELL_DN\n",
      "cat 4158, mi = 0.00369990, GSE32533_MIR17_KO_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_UP\n",
      "cat 1626, mi = 0.00368240, GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "cat 3898, mi = 0.00365899, GSE17721_0.5H_VS_8H_POLYIC_BMDC_DN\n",
      "cat 1917, mi = 0.00358900, GSE22886_DC_VS_MONOCYTE_DN\n",
      "cat 989, mi = 0.00357606, GSE36476_CTRL_VS_TSST_ACT_40H_MEMORY_CD4_TCELL_OLD_UP\n",
      "cat 4118, mi = 0.00352169, GSE17721_0.5H_VS_4H_PAM3CSK4_BMDC_UP\n",
      "cat 2127, mi = 0.00348258, GSE38696_LIGHT_ZONE_VS_DARK_ZONE_BCELL_UP\n",
      "cat 3283, mi = 0.00339567, GSE43956_WT_VS_SGK1_KO_TH17_DIFFERENTIATED_CD4_TCELL_DN\n",
      "cat 1346, mi = 0.00337979, GSE21063_CTRL_VS_ANTI_IGM_STIM_BCELL_NFATC1_KO_8H_UP\n",
      "cat 734, mi = 0.00335437, GSE17721_POLYIC_VS_CPG_0.5H_BMDC_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1028, mi = 0.004, pvalue = 0.000, GSE11057_EFF_MEM_VS_CENT_MEM_CD4_TCELL_DN\n",
      "Passed: cat 635, mi = 0.004, pvalue = 0.000, GSE8384_CTRL_VS_B_ABORTUS_4H_MAC_CELL_LINE_DN\n",
      "Passed: cat 1291, mi = 0.004, pvalue = 0.000, GSE11057_NAIVE_CD4_VS_PBMC_CD4_TCELL_DN\n",
      "Passed: cat 4158, mi = 0.004, pvalue = 0.001, GSE32533_MIR17_KO_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_UP\n",
      "Passed: cat 1626, mi = 0.004, pvalue = 0.000, GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "Passed: cat 3898, mi = 0.004, pvalue = 0.000, GSE17721_0.5H_VS_8H_POLYIC_BMDC_DN\n",
      "Passed: cat 4118, mi = 0.004, pvalue = 0.001, GSE17721_0.5H_VS_4H_PAM3CSK4_BMDC_UP\n",
      "Passed: cat 734, mi = 0.003, pvalue = 0.001, GSE17721_POLYIC_VS_CPG_0.5H_BMDC_DN\n",
      "Passed: cat 1272, mi = 0.003, pvalue = 0.001, GSE20198_UNTREATED_VS_IFNA_TREATED_ACT_CD4_TCELL_UP\n",
      "Passed: cat 4699, mi = 0.003, pvalue = 0.001, GSE15330_LYMPHOID_MULTIPOTENT_VS_PRO_BCELL_UP\n",
      "Passed: cat 1155, mi = 0.003, pvalue = 0.002, GSE25087_TREG_VS_TCONV_FETUS_UP\n",
      "Passed: cat 4090, mi = 0.003, pvalue = 0.002, GSE17974_CTRL_VS_ACT_IL4_AND_ANTI_IL12_4H_CD4_TCELL_UP\n",
      "Passed: cat 1609, mi = 0.003, pvalue = 0.003, GSE36476_CTRL_VS_TSST_ACT_72H_MEMORY_CD4_TCELL_YOUNG_DN\n",
      "Passed: cat 2474, mi = 0.003, pvalue = 0.003, GSE22886_UNSTIM_VS_IL2_STIM_NKCELL_UP\n",
      "Passed: cat 2554, mi = 0.003, pvalue = 0.003, GSE21360_PRIMARY_VS_SECONDARY_MEMORY_CD8_TCELL_DN\n",
      "Passed: cat 3605, mi = 0.003, pvalue = 0.002, GSE360_HIGH_VS_LOW_DOSE_B_MALAYI_MAC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2110, mi = 0.003, pvalue = 0.005, GSE31082_DN_VS_CD4_SP_THYMOCYTE_UP\n",
      "Passed: cat 1704, mi = 0.003, pvalue = 0.004, GSE46606_DAY1_VS_DAY3_CD40L_IL2_IL5_STIMULATED_IRF4MID_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1483, mi = 0.003, pvalue = 0.007, GSE12507_PDC_CELL_LINE_VS_IMMATUE_T_CELL_LINE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3168, mi = 0.003, pvalue = 0.005, GSE14000_TRANSLATED_RNA_VS_MRNA_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2230, mi = 0.003, pvalue = 0.006, GSE36888_UNTREATED_VS_IL2_TREATED_TCELL_2H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1901, mi = 0.003, pvalue = 0.007, GSE37416_CTRL_VS_0H_F_TULARENSIS_LVS_NEUTROPHIL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1935, mi = 0.003, pvalue = 0.005, GSE360_CTRL_VS_L_MAJOR_MAC_DN\n",
      "Passed: cat 3298, mi = 0.003, pvalue = 0.004, GSE14413_UNSTIM_VS_IFNB_STIM_RAW264_CELLS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 850, mi = 0.003, pvalue = 0.009, GSE43955_1H_VS_10H_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2703, mi = 0.003, pvalue = 0.011, GSE20198_IL12_VS_IL12_IL18_TREATED_ACT_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 491, mi = 0.003, pvalue = 0.010, GSE10240_CTRL_VS_IL17_AND_IL22_STIM_PRIMARY_BRONCHIAL_EPITHELIAL_CELLS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2706, mi = 0.003, pvalue = 0.011, GSE4811_CLASSSICALY_ACTIVATED_VS_TYPE_2_ACTIVATED_MACROPHAGE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3374, mi = 0.003, pvalue = 0.006, GSE37416_0H_VS_6H_F_TULARENSIS_LVS_NEUTROPHIL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4121, mi = 0.003, pvalue = 0.012, GSE22611_UNSTIM_VS_2H_MDP_STIM_NOD2_TRANSDUCED_HEK293T_CELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1741, mi = 0.003, pvalue = 0.011, GSE22886_NAIVE_VS_IGG_IGA_MEMORY_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3958, mi = 0.003, pvalue = 0.013, GSE21670_IL6_VS_TGFB_AND_IL6_TREATED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3515, mi = 0.003, pvalue = 0.011, GSE23398_WT_VS_IL2_KO_CD4_TCELL_SCURFY_MOUSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1966, mi = 0.003, pvalue = 0.009, GSE24574_BCL6_HIGH_VS_LOW_TFH_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3922, mi = 0.003, pvalue = 0.012, GSE27291_0H_VS_7D_STIM_GAMMADELTA_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1251, mi = 0.003, pvalue = 0.013, GSE20152_SPHK1_KO_VS_HTNFA_OVEREXPRESS_ANKLE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4639, mi = 0.003, pvalue = 0.012, GSE2585_THYMIC_DC_VS_MTEC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3720, mi = 0.003, pvalue = 0.010, GSE24814_STAT5_KO_VS_WT_PRE_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3015, mi = 0.003, pvalue = 0.016, GSE3565_DUSP1_VS_WT_SPLENOCYTES_POST_LPS_INJECTION_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1820, mi = 0.003, pvalue = 0.019, GSE6681_DELETED_FOXP3_VS_WT_TREG_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1629, mi = 0.003, pvalue = 0.015, GSE9006_HEALTHY_VS_TYPE_1_DIABETES_PBMC_1MONTH_POST_DX_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1295, mi = 0.003, pvalue = 0.013, GSE11864_CSF1_VS_CSF1_IFNG_IN_MAC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2206, mi = 0.003, pvalue = 0.017, GSE29614_CTRL_VS_DAY7_TIV_FLU_VACCINE_PBMC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4467, mi = 0.003, pvalue = 0.013, GSE21670_UNTREATED_VS_IL6_TREATED_STAT3_KO_CD4_TCELL_UP\n",
      "Total number of categories evaluated = 141.\n",
      "Number of categories that passed the tests = 18\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 5, 9, 0.173642063359159\n",
      "best merge is 0, 3, 0.206475817334316\n",
      "best merge is 7, 14, 0.3078277880973\n",
      "best merge is 1, 5, 0.39429525242458\n",
      "best merge is 0, 13, 0.41455260390381\n",
      "best merge is 10, 17, 0.527585976938351\n",
      "best merge is 6, 8, 0.569517820227246\n",
      "best merge is 4, 11, 0.57197000549948\n",
      "best merge is 2, 15, 0.585730427525252\n",
      "best merge is 0, 6, 0.673488489985891\n",
      "best merge is 7, 10, 0.739474710495302\n",
      "best merge is 2, 16, 0.758472407325584\n",
      "best merge is 4, 12, 0.84413646782839\n",
      "best merge is 0, 1, 0.98715455264698\n",
      "order classes\n",
      "5 - 1.62605657816194\n",
      "3 - 1.28297675761109\n",
      "3 - 2.80356698956856\n",
      "3 - 3.01365725487501\n",
      "4 - 2.13373834144698\n",
      "1 5 9\n",
      "0 3 13 6 8\n",
      "2 15 16\n",
      "4 11 12\n",
      "7 14 10 17\n",
      "Done.Start drawing\n",
      "4.84\t-10.88\n",
      "GSE46606_DAY1_VS_DAY3_CD40L_IL2_IL5_STIMULATED_IRF4MID_BCELL_UP\n",
      "GSE36476_CTRL_VS_TSST_ACT_72H_MEMORY_CD4_TCELL_YOUNG_DN\n",
      "GSE17721_POLYIC_VS_GARDIQUIMOD_2H_BMDC_UP\n",
      "GSE17974_CTRL_VS_ACT_IL4_AND_ANTI_IL12_4H_CD4_TCELL_UP\n",
      "GSE15330_LYMPHOID_MULTIPOTENT_VS_PRO_BCELL_UP\n",
      "GSE22886_UNSTIM_VS_IL2_STIM_NKCELL_UP\n",
      "GSE8384_CTRL_VS_B_ABORTUS_4H_MAC_CELL_LINE_DN\n",
      "GSE17721_0.5H_VS_8H_POLYIC_BMDC_DN\n",
      "GSE11057_EFF_MEM_VS_CENT_MEM_CD4_TCELL_DN\n",
      "GSE20198_UNTREATED_VS_IFNA_TREATED_ACT_CD4_TCELL_UP\n",
      "GSE360_HIGH_VS_LOW_DOSE_B_MALAYI_MAC_UP\n",
      "GSE17721_POLYIC_VS_CPG_0.5H_BMDC_DN\n",
      "GSE32533_MIR17_KO_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_UP\n",
      "GSE11057_NAIVE_CD4_VS_PBMC_CD4_TCELL_DN\n",
      "GSE17721_0.5H_VS_4H_PAM3CSK4_BMDC_UP\n",
      "GSE21360_PRIMARY_VS_SECONDARY_MEMORY_CD8_TCELL_DN\n",
      "GSE25087_TREG_VS_TCONV_FETUS_UP\n",
      "GSE14413_UNSTIM_VS_IFNB_STIM_RAW264_CELLS_DN\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 5, 9, 0.173642063359159\n",
      "best merge is 0, 3, 0.206475817334316\n",
      "best merge is 7, 14, 0.3078277880973\n",
      "best merge is 1, 5, 0.39429525242458\n",
      "best merge is 0, 13, 0.41455260390381\n",
      "best merge is 10, 17, 0.527585976938351\n",
      "best merge is 6, 8, 0.569517820227246\n",
      "best merge is 4, 11, 0.57197000549948\n",
      "best merge is 2, 15, 0.585730427525252\n",
      "best merge is 0, 6, 0.673488489985891\n",
      "best merge is 7, 10, 0.739474710495302\n",
      "best merge is 2, 16, 0.758472407325584\n",
      "best merge is 4, 12, 0.84413646782839\n",
      "best merge is 0, 1, 0.98715455264698\n",
      "order classes\n",
      "5 - 1.62605657816194\n",
      "3 - 1.28297675761109\n",
      "3 - 2.80356698956856\n",
      "3 - 3.01365725487501\n",
      "4 - 2.13373834144698\n",
      "1 5 9\n",
      "0 3 13 6 8\n",
      "2 15 16\n",
      "4 11 12\n",
      "7 14 10 17\n",
      "Done.-7.68320610687023\t42\t0\n",
      "5.55534351145038\t42\t24.6335877862595\n",
      "6.99809160305344\t42\t26.4770992366412\n",
      "7.86641221374046\t42\t27.5190839694656\n",
      "8.50763358778626\t42\t28.2137404580153\n",
      "9.05534351145038\t42\t28.8015267175573\n",
      "9.54961832061069\t42\t29.3091603053435\n",
      "10.043893129771\t42\t29.7900763358779\n",
      "10.5648854961832\t42\t30.2977099236641\n",
      "11.219465648855\t42\t30.8320610687023\n",
      "16.8034351145038\t42\t31.6068702290076\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_c7/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6701 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6701\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "609 hits go_terms_num = 17786\n",
      "cat 3242, mi = 0.00457913, GO_EYE_MORPHOGENESIS\n",
      "cat 12154, mi = 0.00415351, GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "cat 12535, mi = 0.00404208, GSE8384_CTRL_VS_B_ABORTUS_4H_MAC_CELL_LINE_DN\n",
      "cat 3166, mi = 0.00401252, GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "cat 7937, mi = 0.00397459, GSE11057_EFF_MEM_VS_CENT_MEM_CD4_TCELL_DN\n",
      "cat 11977, mi = 0.00391491, GSE43956_WT_VS_SGK1_KO_TH17_DIFFERENTIATED_CD4_TCELL_DN\n",
      "cat 7939, mi = 0.00389960, GSE11057_NAIVE_CD4_VS_PBMC_CD4_TCELL_DN\n",
      "cat 2292, mi = 0.00384239, GO_CAMERA_TYPE_EYE_MORPHOGENESIS\n",
      "cat 7922, mi = 0.00376672, GSE10325_LUPUS_CD4_TCELL_VS_LUPUS_MYELOID_UP\n",
      "cat 17586, mi = 0.00376210, chr1q24\n",
      "cat 10782, mi = 0.00372241, GSE32533_MIR17_KO_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_UP\n",
      "cat 10135, mi = 0.00365398, GSE2585_THYMIC_DC_VS_THYMIC_MACROPHAGE_DN\n",
      "cat 8061, mi = 0.00358300, GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "cat 12968, mi = 0.00355860, HONRADO_BREAST_CANCER_BRCA1_VS_BRCA2\n",
      "cat 15144, mi = 0.00355272, PHONG_TNF_TARGETS_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 3242, mi = 0.005, pvalue = 0.000, GO_EYE_MORPHOGENESIS\n",
      "Passed: cat 12154, mi = 0.004, pvalue = 0.000, GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "Passed: cat 12535, mi = 0.004, pvalue = 0.000, GSE8384_CTRL_VS_B_ABORTUS_4H_MAC_CELL_LINE_DN\n",
      "Passed: cat 3166, mi = 0.004, pvalue = 0.000, GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "Passed: cat 7939, mi = 0.004, pvalue = 0.000, GSE11057_NAIVE_CD4_VS_PBMC_CD4_TCELL_DN\n",
      "Passed: cat 17586, mi = 0.004, pvalue = 0.000, chr1q24\n",
      "Passed: cat 10782, mi = 0.004, pvalue = 0.000, GSE32533_MIR17_KO_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_UP\n",
      "Passed: cat 10135, mi = 0.004, pvalue = 0.000, GSE2585_THYMIC_DC_VS_THYMIC_MACROPHAGE_DN\n",
      "Passed: cat 8061, mi = 0.004, pvalue = 0.000, GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "Passed: cat 12968, mi = 0.004, pvalue = 0.000, HONRADO_BREAST_CANCER_BRCA1_VS_BRCA2\n",
      "Passed: cat 15144, mi = 0.004, pvalue = 0.001, PHONG_TNF_TARGETS_UP\n",
      "Passed: cat 8743, mi = 0.003, pvalue = 0.001, GSE17721_0.5H_VS_8H_POLYIC_BMDC_DN\n",
      "Passed: cat 815, mi = 0.003, pvalue = 0.001, CCATCCA_MIR432\n",
      "Passed: cat 13086, mi = 0.003, pvalue = 0.001, IVANOVA_HEMATOPOIESIS_INTERMEDIATE_PROGENITOR\n",
      "Passed: cat 9066, mi = 0.003, pvalue = 0.001, GSE17974_CTRL_VS_ACT_IL4_AND_ANTI_IL12_12H_CD4_TCELL_UP\n",
      "Passed: cat 6349, mi = 0.003, pvalue = 0.001, GO_REGULATION_OF_GLUCOSE_IMPORT\n",
      "Passed: cat 8732, mi = 0.003, pvalue = 0.002, GSE17721_0.5H_VS_4H_PAM3CSK4_BMDC_UP\n",
      "Passed: cat 10722, mi = 0.003, pvalue = 0.002, GSE31082_DN_VS_CD4_SP_THYMOCYTE_UP\n",
      "Passed: cat 8464, mi = 0.003, pvalue = 0.001, GSE15330_LYMPHOID_MULTIPOTENT_VS_PRO_BCELL_UP\n",
      "Passed: cat 9334, mi = 0.003, pvalue = 0.001, GSE20366_EX_VIVO_VS_HOMEOSTATIC_CONVERSION_TREG_UP\n",
      "Passed: cat 2513, mi = 0.003, pvalue = 0.003, GO_CELL_ADHESION_MOLECULE_BINDING\n",
      "Passed: cat 225, mi = 0.003, pvalue = 0.003, ATGGYGGA_UNKNOWN\n",
      "Passed: cat 1071, mi = 0.003, pvalue = 0.003, CYCLIN_D1_KE_.V1_DN\n",
      "Passed: cat 9927, mi = 0.003, pvalue = 0.004, GSE24102_GRANULOCYSTIC_MDSC_VS_NEUTROPHIL_DN\n",
      "Passed: cat 5735, mi = 0.003, pvalue = 0.004, GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "Passed: cat 12140, mi = 0.003, pvalue = 0.005, GSE4811_CLASSSICALY_ACTIVATED_VS_TYPE_2_ACTIVATED_MACROPHAGE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 13057, mi = 0.003, pvalue = 0.007, IL21_UP.V1_DN\n",
      "Passed: cat 4613, mi = 0.003, pvalue = 0.003, GO_NEGATIVE_REGULATION_OF_RESPONSE_TO_EXTRACELLULAR_STIMULUS\n",
      "Didn't pass (cutoff=0.0050): cat 10299, mi = 0.003, pvalue = 0.006, GSE2770_IL12_VS_IL4_TREATED_ACT_CD4_TCELL_2H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9696, mi = 0.003, pvalue = 0.007, GSE22611_NOD2_TRANSD_VS_CTRL_TRANSD_HEK293_MDP_STIM_6H_UP\n",
      "Passed: cat 7335, mi = 0.003, pvalue = 0.002, GO_SODIUM_ION_TRANSMEMBRANE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 1475, mi = 0.003, pvalue = 0.009, GARGALOVIC_RESPONSE_TO_OXIDIZED_PHOSPHOLIPIDS_GREEN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 11812, mi = 0.003, pvalue = 0.006, GSE41867_NAIVE_VS_DAY8_LCMV_EFFECTOR_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 10362, mi = 0.003, pvalue = 0.006, GSE2770_UNTREATED_VS_TGFB_AND_IL12_TREATED_ACT_CD4_TCELL_48H_UP\n",
      "Passed: cat 2754, mi = 0.003, pvalue = 0.001, GO_COPPER_ION_BINDING\n",
      "Passed: cat 9316, mi = 0.003, pvalue = 0.004, GSE20198_UNTREATED_VS_IFNA_TREATED_ACT_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2944, mi = 0.003, pvalue = 0.009, GO_DNA_CATABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 8265, mi = 0.003, pvalue = 0.009, GSE14000_TRANSLATED_RNA_VS_MRNA_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9860, mi = 0.003, pvalue = 0.008, GSE23398_WT_VS_IL2_KO_CD4_TCELL_SCURFY_MOUSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 12572, mi = 0.003, pvalue = 0.011, GSE8835_HEALTHY_VS_CLL_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 6778, mi = 0.003, pvalue = 0.007, GO_REGULATION_OF_STEROID_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 17756, mi = 0.003, pvalue = 0.007, chr9p13\n",
      "Didn't pass (cutoff=0.0050): cat 12231, mi = 0.003, pvalue = 0.009, GSE5589_LPS_AND_IL10_VS_LPS_AND_IL6_STIM_IL6_KO_MACROPHAGE_45MIN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9314, mi = 0.003, pvalue = 0.010, GSE20198_IL12_VS_IL12_IL18_TREATED_ACT_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14554, mi = 0.003, pvalue = 0.012, MORF_CTBP1\n",
      "Didn't pass (cutoff=0.0050): cat 9437, mi = 0.003, pvalue = 0.008, GSE2128_CTRL_VS_MIMETOPE_NEGATIVE_SELECTION_DP_THYMOCYTE_C57BL6_DN\n",
      "Passed: cat 2945, mi = 0.003, pvalue = 0.003, GO_DNA_CATABOLIC_PROCESS_ENDONUCLEOLYTIC\n",
      "Didn't pass (cutoff=0.0050): cat 1635, mi = 0.003, pvalue = 0.011, GCNP_SHH_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 10133, mi = 0.003, pvalue = 0.011, GSE2585_THYMIC_DC_VS_MTEC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8386, mi = 0.003, pvalue = 0.013, GSE1460_INTRATHYMIC_T_PROGENITOR_VS_NAIVE_CD4_TCELL_CORD_BLOOD_UP\n",
      "Didn't pass (cutoff=0.0050): cat 16072, mi = 0.003, pvalue = 0.006, REACTOME_SRP_DEPENDENT_COTRANSLATIONAL_PROTEIN_TARGETING_TO_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 17698, mi = 0.003, pvalue = 0.014, chr5q33\n",
      "Didn't pass (cutoff=0.0050): cat 15631, mi = 0.003, pvalue = 0.013, REACTOME_DIABETES_PATHWAYS\n",
      "Didn't pass (cutoff=0.0050): cat 14512, mi = 0.003, pvalue = 0.016, MOREAUX_B_LYMPHOCYTE_MATURATION_BY_TACI_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2321, mi = 0.003, pvalue = 0.014, GO_CARBOXYLIC_ESTER_HYDROLASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 8281, mi = 0.003, pvalue = 0.010, GSE14308_NAIVE_CD4_TCELL_VS_NATURAL_TREG_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8151, mi = 0.003, pvalue = 0.014, GSE13411_NAIVE_VS_MEMORY_BCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8333, mi = 0.003, pvalue = 0.008, GSE14413_UNSTIM_VS_IFNB_STIM_RAW264_CELLS_DN\n",
      "Passed: cat 14860, mi = 0.003, pvalue = 0.000, NEWMAN_ERCC6_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 14647, mi = 0.002, pvalue = 0.019, MORF_PRKAR1A\n",
      "Didn't pass (cutoff=0.0050): cat 504, mi = 0.002, pvalue = 0.018, BIOCARTA_P38MAPK_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 3524, mi = 0.002, pvalue = 0.008, GO_HISTONE_EXCHANGE\n",
      "Didn't pass (cutoff=0.0050): cat 11995, mi = 0.002, pvalue = 0.022, GSE45365_BCELL_VS_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 12941, mi = 0.002, pvalue = 0.020, HOFMANN_CELL_LYMPHOMA_DN\n",
      "Didn't pass (cutoff=0.0050): cat 17689, mi = 0.002, pvalue = 0.011, chr5q12\n",
      "Didn't pass (cutoff=0.0050): cat 3944, mi = 0.002, pvalue = 0.021, GO_MANGANESE_ION_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 13412, mi = 0.002, pvalue = 0.021, KESHELAVA_MULTIPLE_DRUG_RESISTANCE\n",
      "Didn't pass (cutoff=0.0050): cat 9451, mi = 0.002, pvalue = 0.015, GSE21360_PRIMARY_VS_SECONDARY_MEMORY_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1753, mi = 0.002, pvalue = 0.020, GNF2_DAP3\n",
      "Didn't pass (cutoff=0.0050): cat 12851, mi = 0.002, pvalue = 0.007, HASLINGER_B_CLL_WITH_MUTATED_VH_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 7614, mi = 0.002, pvalue = 0.018, GO_TRANSLESION_SYNTHESIS\n",
      "Didn't pass (cutoff=0.0050): cat 1385, mi = 0.002, pvalue = 0.013, FLOTHO_PEDIATRIC_ALL_THERAPY_RESPONSE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2993, mi = 0.002, pvalue = 0.025, GO_DNA_TEMPLATED_TRANSCRIPTION_TERMINATION\n",
      "Didn't pass (cutoff=0.0050): cat 17732, mi = 0.002, pvalue = 0.027, chr7q21\n",
      "Didn't pass (cutoff=0.0050): cat 8899, mi = 0.002, pvalue = 0.023, GSE17721_LPS_VS_PAM3CSK4_12H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8547, mi = 0.002, pvalue = 0.022, GSE15930_STIM_VS_STIM_AND_IFNAB_24H_CD8_T_CELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1216, mi = 0.002, pvalue = 0.020, DURCHDEWALD_SKIN_CARCINOGENESIS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14884, mi = 0.002, pvalue = 0.030, NIELSEN_GIST\n",
      "Didn't pass (cutoff=0.0050): cat 626, mi = 0.002, pvalue = 0.007, BOYLAN_MULTIPLE_MYELOMA_D_CLUSTER_UP\n",
      "Total number of categories evaluated = 460.\n",
      "Number of categories that passed the tests = 32\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 27, 28, 0.144646018912583\n",
      "best merge is 11, 18, 0.175582311211\n",
      "best merge is 0, 9, 0.19343552164224\n",
      "best merge is 10, 24, 0.228507324489699\n",
      "best merge is 6, 7, 0.235838502866635\n",
      "best merge is 27, 29, 0.262607812158124\n",
      "best merge is 21, 22, 0.278857379469345\n",
      "best merge is 2, 10, 0.320049894635219\n",
      "best merge is 1, 16, 0.401977993023739\n",
      "best merge is 2, 8, 0.471817080118617\n",
      "best merge is 12, 17, 0.472010087970522\n",
      "best merge is 19, 23, 0.475062260852627\n",
      "best merge is 0, 6, 0.496992842912947\n",
      "best merge is 5, 25, 0.530479169497183\n",
      "best merge is 4, 15, 0.536486015225116\n",
      "best merge is 13, 20, 0.55352403515094\n",
      "best merge is 12, 14, 0.557638393867225\n",
      "best merge is 2, 11, 0.590805763285005\n",
      "best merge is 1, 27, 0.600944909675775\n",
      "best merge is 4, 19, 0.605568024620378\n",
      "best merge is 26, 30, 0.626669562871369\n",
      "best merge is 3, 21, 0.755215190476027\n",
      "best merge is 0, 2, 0.759148449244169\n",
      "best merge is 12, 26, 0.790737550724641\n",
      "best merge is 4, 31, 0.800604361094566\n",
      "best merge is 1, 5, 0.828608041716568\n",
      "best merge is 0, 3, 0.874812685920168\n",
      "best merge is 4, 13, 0.914689576715762\n",
      "order classes\n",
      "13 - 1.62713702795169\n",
      "7 - 2.51784371889821\n",
      "5 - 3.39775376329601\n",
      "5 - 3.56269145237718\n",
      "2 - 4.42819228120752\n",
      "0 9 6 7 2 10 24 8 11 18 3 21 22\n",
      "1 16 27 28 29 5 25\n",
      "4 15 19 23 31\n",
      "12 17 14 26 30\n",
      "13 20\n",
      "Done.Start drawing\n",
      "9.39\t-11.05\n",
      "GO_REGULATION_OF_GLUCOSE_IMPORT\n",
      "GSE24102_GRANULOCYSTIC_MDSC_VS_NEUTROPHIL_DN\n",
      "CYCLIN_D1_KE_.V1_DN\n",
      "GO_EYE_MORPHOGENESIS\n",
      "HONRADO_BREAST_CANCER_BRCA1_VS_BRCA2\n",
      "GSE2585_THYMIC_DC_VS_THYMIC_MACROPHAGE_DN\n",
      "GSE4811_CLASSSICALY_ACTIVATED_VS_TYPE_2_ACTIVATED_MACROPHAGE_UP\n",
      "GO_POSITIVE_REGULATION_OF_VIRAL_TRANSCRIPTION\n",
      "GSE15330_LYMPHOID_MULTIPOTENT_VS_PRO_BCELL_UP\n",
      "GO_NEGATIVE_REGULATION_OF_RESPONSE_TO_EXTRACELLULAR_STIMULUS\n",
      "GSE8384_CTRL_VS_B_ABORTUS_4H_MAC_CELL_LINE_DN\n",
      "PHONG_TNF_TARGETS_UP\n",
      "GSE12366_GC_VS_MEMORY_BCELL_DN\n",
      "GSE17721_0.5H_VS_8H_POLYIC_BMDC_DN\n",
      "GSE17974_CTRL_VS_ACT_IL4_AND_ANTI_IL12_12H_CD4_TCELL_UP\n",
      "ATGGYGGA_UNKNOWN\n",
      "chr1q24\n",
      "GSE20366_EX_VIVO_VS_HOMEOSTATIC_CONVERSION_TREG_UP\n",
      "GSE31082_DN_VS_CD4_SP_THYMOCYTE_UP\n",
      "GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_UP\n",
      "GO_SODIUM_ION_TRANSMEMBRANE_TRANSPORT\n",
      "GO_COPPER_ION_BINDING\n",
      "GSE20198_UNTREATED_VS_IFNA_TREATED_ACT_CD4_TCELL_UP\n",
      "GO_ESTABLISHMENT_OF_CELL_POLARITY\n",
      "GO_CELL_ADHESION_MOLECULE_BINDING\n",
      "GSE32533_MIR17_KO_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_UP\n",
      "NEWMAN_ERCC6_TARGETS_DN\n",
      "GSE17721_0.5H_VS_4H_PAM3CSK4_BMDC_UP\n",
      "CCATCCA_MIR432\n",
      "GO_DNA_CATABOLIC_PROCESS_ENDONUCLEOLYTIC\n",
      "GSE11057_NAIVE_CD4_VS_PBMC_CD4_TCELL_DN\n",
      "IVANOVA_HEMATOPOIESIS_INTERMEDIATE_PROGENITOR\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 27, 28, 0.144646018912583\n",
      "best merge is 11, 18, 0.175582311211\n",
      "best merge is 0, 9, 0.19343552164224\n",
      "best merge is 10, 24, 0.228507324489699\n",
      "best merge is 6, 7, 0.235838502866635\n",
      "best merge is 27, 29, 0.262607812158124\n",
      "best merge is 21, 22, 0.278857379469345\n",
      "best merge is 2, 10, 0.320049894635219\n",
      "best merge is 1, 16, 0.401977993023739\n",
      "best merge is 2, 8, 0.471817080118617\n",
      "best merge is 12, 17, 0.472010087970522\n",
      "best merge is 19, 23, 0.475062260852627\n",
      "best merge is 0, 6, 0.496992842912947\n",
      "best merge is 5, 25, 0.530479169497183\n",
      "best merge is 4, 15, 0.536486015225116\n",
      "best merge is 13, 20, 0.55352403515094\n",
      "best merge is 12, 14, 0.557638393867225\n",
      "best merge is 2, 11, 0.590805763285005\n",
      "best merge is 1, 27, 0.600944909675775\n",
      "best merge is 4, 19, 0.605568024620378\n",
      "best merge is 26, 30, 0.626669562871369\n",
      "best merge is 3, 21, 0.755215190476027\n",
      "best merge is 0, 2, 0.759148449244169\n",
      "best merge is 12, 26, 0.790737550724641\n",
      "best merge is 4, 31, 0.800604361094566\n",
      "best merge is 1, 5, 0.828608041716568\n",
      "best merge is 0, 3, 0.874812685920168\n",
      "best merge is 4, 13, 0.914689576715762\n",
      "order classes\n",
      "13 - 1.62713702795169\n",
      "7 - 2.51784371889821\n",
      "5 - 3.39775376329601\n",
      "5 - 3.56269145237718\n",
      "2 - 4.42819228120752\n",
      "0 9 6 7 2 10 24 8 11 18 3 21 22\n",
      "1 16 27 28 29 5 25\n",
      "4 15 19 23 31\n",
      "12 17 14 26 30\n",
      "13 20\n",
      "Done.-10.3630136986301\t42\t0\n",
      "-0.0273972602739718\t42\t19.2739726027397\n",
      "1.08219178082192\t42\t20.6712328767123\n",
      "1.77054794520548\t42\t21.4931506849315\n",
      "2.26369863013699\t42\t22.0479452054795\n",
      "2.68493150684932\t42\t22.4794520547945\n",
      "3.07534246575342\t42\t22.8904109589041\n",
      "3.44520547945206\t42\t23.2602739726027\n",
      "3.8458904109589\t42\t23.6301369863014\n",
      "4.35958904109589\t42\t24.0616438356164\n",
      "13.3287671232877\t42\t24.6575342465753\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_full/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 1545 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 1545\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "140 hits go_terms_num = 50\n",
      "cat 28, mi = 0.01092962, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 13, mi = 0.01019534, HALLMARK_APICAL_JUNCTION\n",
      "cat 7, mi = 0.00998101, HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY\n",
      "cat 2, mi = 0.00981187, HALLMARK_PEROXISOME\n",
      "cat 15, mi = 0.00795862, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 29, mi = 0.00785509, HALLMARK_MYC_TARGETS_V2\n",
      "cat 22, mi = 0.00779402, HALLMARK_E2F_TARGETS\n",
      "cat 16, mi = 0.00757256, HALLMARK_MTORC1_SIGNALING\n",
      "cat 39, mi = 0.00754602, HALLMARK_NOTCH_SIGNALING\n",
      "cat 5, mi = 0.00709390, HALLMARK_PROTEIN_SECRETION\n",
      "cat 9, mi = 0.00698603, HALLMARK_SPERMATOGENESIS\n",
      "cat 20, mi = 0.00689525, HALLMARK_MYC_TARGETS_V1\n",
      "cat 47, mi = 0.00655809, HALLMARK_COMPLEMENT\n",
      "cat 36, mi = 0.00623145, HALLMARK_DNA_REPAIR\n",
      "cat 31, mi = 0.00608538, HALLMARK_HEDGEHOG_SIGNALING\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.011, pvalue = 0.015, HALLMARK_KRAS_SIGNALING_UP\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.010, pvalue = 0.028, HALLMARK_APICAL_JUNCTION\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.010, pvalue = 0.030, HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.010, pvalue = 0.039, HALLMARK_PEROXISOME\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.008, pvalue = 0.101, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 29, mi = 0.008, pvalue = 0.135, HALLMARK_MYC_TARGETS_V2\n",
      "Didn't pass (cutoff=0.0050): cat 22, mi = 0.008, pvalue = 0.102, HALLMARK_E2F_TARGETS\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.008, pvalue = 0.119, HALLMARK_MTORC1_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.008, pvalue = 0.164, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.007, pvalue = 0.220, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.007, pvalue = 0.225, HALLMARK_SPERMATOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.007, pvalue = 0.215, HALLMARK_MYC_TARGETS_V1\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.007, pvalue = 0.190, HALLMARK_COMPLEMENT\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.006, pvalue = 0.249, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.006, pvalue = 0.347, HALLMARK_HEDGEHOG_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.006, pvalue = 0.388, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.006, pvalue = 0.274, HALLMARK_HEME_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.006, pvalue = 0.442, HALLMARK_BILE_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.005, pvalue = 0.388, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.005, pvalue = 0.378, HALLMARK_TGF_BETA_SIGNALING\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "3.57\t-10.88\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_msigdb_h/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7491 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7491\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "681 hits go_terms_num = 790\n",
      "cat 402, mi = 0.00223567, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 354, mi = 0.00222256, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00202812, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 484, mi = 0.00194302, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 672, mi = 0.00187034, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 465, mi = 0.00174091, PPIL4_K562_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00164953, AUH_K562_POSTAR_eCLIP\n",
      "cat 270, mi = 0.00162253, HLTF_K562_POSTAR_eCLIP\n",
      "cat 16, mi = 0.00158901, ALKBH1_HeLa_POSTAR_CIMS\n",
      "cat 236, mi = 0.00144054, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 629, mi = 0.00142228, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "cat 749, mi = 0.00139018, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 668, mi = 0.00133659, TNRC6A_HEK293_Plass\n",
      "cat 235, mi = 0.00123534, FUS_HeLa_POSTAR_CIMS\n",
      "cat 312, mi = 0.00122421, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.018, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.002, pvalue = 0.016, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.036, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.038, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.063, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.002, pvalue = 0.073, PPIL4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.125, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.002, pvalue = 0.111, HLTF_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.158, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.001, pvalue = 0.230, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.001, pvalue = 0.162, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 749, mi = 0.001, pvalue = 0.195, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.001, pvalue = 0.196, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.001, pvalue = 0.342, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.001, pvalue = 0.304, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.001, pvalue = 0.532, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.001, pvalue = 0.424, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.001, pvalue = 0.481, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.478, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 300, mi = 0.001, pvalue = 0.748, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.39\t-11.05\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_RBPs_all_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6816 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6816\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "620 hits go_terms_num = 790\n",
      "cat 402, mi = 0.00242271, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 321, mi = 0.00222645, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 354, mi = 0.00217894, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 484, mi = 0.00202971, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 672, mi = 0.00198001, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 31, mi = 0.00181239, AUH_K562_POSTAR_eCLIP\n",
      "cat 465, mi = 0.00178691, PPIL4_K562_POSTAR_eCLIP\n",
      "cat 16, mi = 0.00174491, ALKBH1_HeLa_POSTAR_CIMS\n",
      "cat 329, mi = 0.00171274, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "cat 629, mi = 0.00168947, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "cat 236, mi = 0.00158476, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 270, mi = 0.00151845, HLTF_K562_POSTAR_eCLIP\n",
      "cat 749, mi = 0.00137483, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 235, mi = 0.00135841, FUS_HeLa_POSTAR_CIMS\n",
      "cat 312, mi = 0.00134530, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 402, mi = 0.002, pvalue = 0.021, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.038, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.002, pvalue = 0.034, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 484, mi = 0.002, pvalue = 0.051, PTBP1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.078, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.128, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.002, pvalue = 0.112, PPIL4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.159, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 329, mi = 0.002, pvalue = 0.101, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.115, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.233, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.002, pvalue = 0.203, HLTF_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 749, mi = 0.001, pvalue = 0.275, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 235, mi = 0.001, pvalue = 0.214, FUS_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.001, pvalue = 0.309, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.001, pvalue = 0.281, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.001, pvalue = 0.538, DDX3X_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.001, pvalue = 0.476, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.001, pvalue = 0.494, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.001, pvalue = 0.494, TIA1_NA_POSTAR_iCLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.84\t-10.88\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_RBPs_coding_gene_ids/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6678 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6678\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "607 hits go_terms_num = 790\n",
      "cat 242, mi = 0.00338598, FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 577, mi = 0.00301700, SMNDC1_K562_POSTAR_eCLIP\n",
      "cat 230, mi = 0.00301664, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 407, mi = 0.00301556, MBNL2_hippocampus_POSTAR_Piranha_0.01\n",
      "cat 240, mi = 0.00284669, FXR1_HEK293_POSTAR_PAR-CLIP\n",
      "cat 241, mi = 0.00284669, FXR1_HEK293_POSTAR_PARalyzer\n",
      "cat 353, mi = 0.00280579, IGF2BP3_PL45_POSTAR_CIMS\n",
      "cat 418, mi = 0.00276509, MOV10_HEK293_POSTAR_Piranha_0.01\n",
      "cat 355, mi = 0.00274435, IGF2BP3_PL45_POSTAR_iCLIP\n",
      "cat 510, mi = 0.00259263, RBFOX2_K562_POSTAR_eCLIP\n",
      "cat 406, mi = 0.00258567, MBNL2_hippocampus_POSTAR_HITS-CLIP\n",
      "cat 748, mi = 0.00253806, YTHDC2_HEK293T_POSTAR_CIMS\n",
      "cat 750, mi = 0.00253806, YTHDC2_HEK293T_POSTAR_iCLIP\n",
      "cat 158, mi = 0.00253257, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 233, mi = 0.00248093, FUS_HEK293_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 242, mi = 0.003, pvalue = 0.001, FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "Passed: cat 577, mi = 0.003, pvalue = 0.003, SMNDC1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.003, pvalue = 0.011, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 494, mi = 0.002, pvalue = 0.044, PUM2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 504, mi = 0.002, pvalue = 0.049, QKI_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 506, mi = 0.002, pvalue = 0.074, QKI_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.002, pvalue = 0.102, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.002, pvalue = 0.160, EIF3D_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.002, pvalue = 0.115, EIF3A_HEK293T_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 169, mi = 0.002, pvalue = 0.184, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 157, mi = 0.002, pvalue = 0.119, EIF3A_HEK293T_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 294, mi = 0.002, pvalue = 0.177, HNRNPC_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 595, mi = 0.002, pvalue = 0.057, SRSF7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 626, mi = 0.001, pvalue = 0.463, TARBP2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 538, mi = 0.001, pvalue = 0.271, RPS11_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.001, pvalue = 0.397, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.001, pvalue = 0.292, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 616, mi = 0.001, pvalue = 0.420, TAF15_Brain_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 617, mi = 0.001, pvalue = 0.433, TAF15_Brain_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 627, mi = 0.001, pvalue = 0.567, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 142, mi = 0.001, pvalue = 0.624, DICER1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.341, ALKBH5_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 177.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.84\t-10.88\n",
      "FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "SMNDC1_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.66984732824428\t42\t0\n",
      "5.56870229007634\t42\t24.6603053435114\n",
      "6.99809160305344\t42\t26.4770992366412\n",
      "7.87977099236641\t42\t27.5190839694656\n",
      "8.52099236641222\t42\t28.2404580152672\n",
      "9.05534351145038\t42\t28.8015267175573\n",
      "9.54961832061069\t42\t29.3091603053435\n",
      "10.043893129771\t42\t29.7900763358779\n",
      "10.5648854961832\t42\t30.2977099236641\n",
      "11.219465648855\t42\t30.8320610687023\n",
      "16.8034351145038\t42\t31.6068702290076\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6602 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6602\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "600 hits go_terms_num = 790\n",
      "cat 689, mi = 0.00311175, U2AF1_K562_POSTAR_eCLIP\n",
      "cat 75, mi = 0.00297611, CPSF6_HEK293_POSTAR_Piranha_0.01\n",
      "cat 169, mi = 0.00287538, EIF3G_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 557, mi = 0.00257029, SERBP1_K562_POSTAR_eCLIP\n",
      "cat 159, mi = 0.00255455, EIF3B_HEK293T_POSTAR_PAR-CLIP\n",
      "cat 563, mi = 0.00248654, SF3B4_HepG2_POSTAR_eCLIP\n",
      "cat 9, mi = 0.00248201, AGO2_HEK293_Plass\n",
      "cat 290, mi = 0.00243368, HNRNPC_HEK293_POSTAR_Piranha_0.01\n",
      "cat 160, mi = 0.00242564, EIF3B_HEK293T_POSTAR_PARalyzer\n",
      "cat 743, mi = 0.00237344, YBX3_K562_POSTAR_eCLIP\n",
      "cat 433, mi = 0.00233539, NOL12_HepG2_POSTAR_eCLIP\n",
      "cat 526, mi = 0.00233077, RBM22_HepG2_POSTAR_eCLIP\n",
      "cat 164, mi = 0.00232863, EIF3D_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 183, mi = 0.00232275, ELAVL1_HeLa_POSTAR_PAR-CLIP\n",
      "cat 8, mi = 0.00227311, AGO1_HEK293_Plass\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 689, mi = 0.003, pvalue = 0.001, U2AF1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.006, AGO2_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 743, mi = 0.002, pvalue = 0.033, YBX3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 433, mi = 0.002, pvalue = 0.034, NOL12_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 8, mi = 0.002, pvalue = 0.015, AGO1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 313, mi = 0.002, pvalue = 0.020, HNRNPM_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 314, mi = 0.002, pvalue = 0.020, HNRNPM_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 528, mi = 0.002, pvalue = 0.033, RBM22_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 315, mi = 0.002, pvalue = 0.091, HNRNPM_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 443, mi = 0.002, pvalue = 0.101, NOP58_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 444, mi = 0.002, pvalue = 0.100, NOP58_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 217, mi = 0.002, pvalue = 0.061, FMR1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 212, mi = 0.002, pvalue = 0.110, FKBP4_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 243, mi = 0.002, pvalue = 0.043, FXR1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 52, mi = 0.002, pvalue = 0.163, CF_Im68_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 684, mi = 0.002, pvalue = 0.127, TROVE2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 234, mi = 0.002, pvalue = 0.112, FUS_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 419, mi = 0.002, pvalue = 0.237, MOV10_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 53, mi = 0.002, pvalue = 0.218, CNBP_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 338, mi = 0.002, pvalue = 0.180, IGF2BP1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 299, mi = 0.002, pvalue = 0.025, HNRNPD_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 57.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.84\t-10.88\n",
      "U2AF1_K562_POSTAR_eCLIP\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.68320610687023\t42\t0\n",
      "5.55534351145038\t42\t24.6335877862595\n",
      "6.99809160305344\t42\t26.4770992366412\n",
      "7.86641221374046\t42\t27.5190839694656\n",
      "8.50763358778626\t42\t28.2137404580153\n",
      "9.05534351145038\t42\t28.8015267175573\n",
      "9.54961832061069\t42\t29.3091603053435\n",
      "10.043893129771\t42\t29.7900763358779\n",
      "10.5648854961832\t42\t30.2977099236641\n",
      "11.219465648855\t42\t30.8320610687023\n",
      "16.8034351145038\t42\t31.6068702290076\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6691 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6691\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "608 hits go_terms_num = 790\n",
      "cat 17, mi = 0.00341727, ALKBH1_HeLa_POSTAR_HITS-CLIP\n",
      "cat 18, mi = 0.00322658, ALKBH1_HeLa_POSTAR_Piranha_0.01\n",
      "cat 532, mi = 0.00312772, RBM47_MDA-MB-231_POSTAR_HITS-CLIP\n",
      "cat 356, mi = 0.00307044, IGF2BP3_Panc1_POSTAR_CIMS\n",
      "cat 358, mi = 0.00307044, IGF2BP3_Panc1_POSTAR_iCLIP\n",
      "cat 515, mi = 0.00300914, RBM10_HEK293_Plass\n",
      "cat 522, mi = 0.00297118, RBM15_HepG2_POSTAR_eCLIP\n",
      "cat 28, mi = 0.00294823, ATXN2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 548, mi = 0.00288709, RTCB_HEK293_POSTAR_Piranha_0.01\n",
      "cat 533, mi = 0.00282236, RBM47_MDA-MB-231_POSTAR_Piranha_0.01\n",
      "cat 531, mi = 0.00269725, RBM47_MDA-MB-231_POSTAR_CIMS\n",
      "cat 240, mi = 0.00267790, FXR1_HEK293_POSTAR_PAR-CLIP\n",
      "cat 317, mi = 0.00264881, HNRNPM_K562_POSTAR_eCLIP\n",
      "cat 428, mi = 0.00260701, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 530, mi = 0.00258211, RBM27_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 17, mi = 0.003, pvalue = 0.001, ALKBH1_HeLa_POSTAR_HITS-CLIP\n",
      "Passed: cat 532, mi = 0.003, pvalue = 0.002, RBM47_MDA-MB-231_POSTAR_HITS-CLIP\n",
      "Passed: cat 515, mi = 0.003, pvalue = 0.003, RBM10_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 530, mi = 0.003, pvalue = 0.015, RBM27_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 305, mi = 0.003, pvalue = 0.011, HNRNPH1_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 304, mi = 0.003, pvalue = 0.015, HNRNPH1_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 225, mi = 0.002, pvalue = 0.022, FUS_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.002, pvalue = 0.085, CNBP_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 283, mi = 0.002, pvalue = 0.071, HNRNPA2B1_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 284, mi = 0.002, pvalue = 0.071, HNRNPA2B1_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.002, pvalue = 0.105, ELAVL1_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.002, pvalue = 0.027, DCK1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 361, mi = 0.002, pvalue = 0.105, ILF3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 623, mi = 0.002, pvalue = 0.091, TAF15_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 319, mi = 0.002, pvalue = 0.132, HNRNPUL1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.130, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.002, pvalue = 0.175, PPIL4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 86, mi = 0.002, pvalue = 0.141, CSTF2T_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 627, mi = 0.002, pvalue = 0.294, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 130, mi = 0.002, pvalue = 0.268, DGCR8_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 267, mi = 0.001, pvalue = 0.216, GTF2F1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.001, pvalue = 0.290, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.001, pvalue = 0.302, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 106.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.491387220708471\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "4.84\t-10.88\n",
      "ALKBH1_HeLa_POSTAR_HITS-CLIP\n",
      "RBM47_MDA-MB-231_POSTAR_HITS-CLIP\n",
      "RBM10_HEK293_Plass\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 2, 0.491387220708471\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-7.65648854961832\t42\t0\n",
      "5.58206106870229\t42\t24.6870229007634\n",
      "6.99809160305344\t42\t26.4770992366412\n",
      "7.86641221374046\t42\t27.5190839694656\n",
      "8.50763358778626\t42\t28.2137404580153\n",
      "9.05534351145038\t42\t28.8015267175573\n",
      "9.54961832061069\t42\t29.3091603053435\n",
      "10.043893129771\t42\t29.7900763358779\n",
      "10.5648854961832\t42\t30.2977099236641\n",
      "11.219465648855\t42\t30.8320610687023\n",
      "16.8034351145038\t42\t31.6068702290076\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 790\n",
      "cat 353, mi = 0.00300417, IGF2BP3_PL45_POSTAR_CIMS\n",
      "cat 354, mi = 0.00250552, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 449, mi = 0.00240031, NSUN2_K562_POSTAR_eCLIP\n",
      "cat 402, mi = 0.00237862, MBNL2_frontalcortex_POSTAR_CIMS\n",
      "cat 195, mi = 0.00228913, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00228913, EZH2_HCT116_POSTAR_iCLIP\n",
      "cat 749, mi = 0.00211045, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 666, mi = 0.00209169, TNRC6A_HEK293_POSTAR_PARalyzer\n",
      "cat 321, mi = 0.00206644, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 484, mi = 0.00198024, PTBP1_HeLa_POSTAR_CIMS\n",
      "cat 657, mi = 0.00196900, TIA1_NA_POSTAR_CIMS\n",
      "cat 658, mi = 0.00196900, TIA1_NA_POSTAR_iCLIP\n",
      "cat 270, mi = 0.00196614, HLTF_K562_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00190765, AUH_K562_POSTAR_eCLIP\n",
      "cat 208, mi = 0.00188620, FBL_HEK293_Plass\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 353, mi = 0.003, pvalue = 0.004, IGF2BP3_PL45_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.002, pvalue = 0.032, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.029, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.002, pvalue = 0.034, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 749, mi = 0.002, pvalue = 0.068, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.079, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 208, mi = 0.002, pvalue = 0.091, FBL_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.002, pvalue = 0.218, PPIL4_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.002, pvalue = 0.308, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 686, mi = 0.002, pvalue = 0.222, TTP_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.001, pvalue = 0.460, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 684, mi = 0.001, pvalue = 0.292, TROVE2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.001, pvalue = 0.386, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 256, mi = 0.001, pvalue = 0.307, GNL3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.001, pvalue = 0.472, AGO4_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 456, mi = 0.001, pvalue = 0.440, PAPD5_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.001, pvalue = 0.694, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.001, pvalue = 0.416, AGO3_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 673, mi = 0.001, pvalue = 0.765, TNRC6B_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 329, mi = 0.001, pvalue = 0.489, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.534, LIN28A_HEK293_Plass\n",
      "Total number of categories evaluated = 33.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.84\t-10.88\n",
      "IGF2BP3_PL45_POSTAR_CIMS\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.68320610687023\t42\t0\n",
      "5.52862595419847\t42\t24.6335877862595\n",
      "6.95801526717558\t42\t26.4236641221374\n",
      "7.83969465648855\t42\t27.4923664122137\n",
      "8.4675572519084\t42\t28.1870229007634\n",
      "9.01526717557252\t42\t28.7480916030534\n",
      "9.52290076335878\t42\t29.2824427480916\n",
      "10.0038167938931\t42\t29.763358778626\n",
      "10.5381679389313\t42\t30.2442748091603\n",
      "11.206106870229\t42\t30.8320610687023\n",
      "16.7900763358779\t42\t31.5801526717557\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "Reading expression data ... 1 by 7528... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 6482 out of 7528 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 6482\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "589 hits go_terms_num = 101\n",
      "cat 69, mi = 0.00273980, RBMS3.1\n",
      "cat 68, mi = 0.00224072, RBMS3\n",
      "cat 88, mi = 0.00172392, SRSF9.1\n",
      "cat 56, mi = 0.00171849, RBM3\n",
      "cat 9, mi = 0.00168078, ESRP2\n",
      "cat 6, mi = 0.00141947, CPEB4\n",
      "cat 13, mi = 0.00120816, FXR2\n",
      "cat 62, mi = 0.00104362, RBM46\n",
      "cat 34, mi = 0.00101992, KHDRBS3\n",
      "cat 21, mi = 0.00101205, HNRNPK\n",
      "cat 27, mi = 0.00058997, HuR.2\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "cat 3, mi = 0.00000000, BRUNOL6\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.003, pvalue = 0.009, RBMS3.1\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.002, pvalue = 0.048, RBMS3\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.002, pvalue = 0.163, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.002, pvalue = 0.166, RBM3\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.190, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.328, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.466, FXR2\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.723, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.835, KHDRBS3\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.916, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.001, pvalue = 0.902, HuR.2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, FMR1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "4.84\t-10.88\n",
      "Outputing EPS file deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/d_mtyl_WTAP_vs_NT.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_WTAP_vs_NT.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_WTAP_vs_NT.c.txt_PAGE/' -> 'ipage/d_mtyl_WTAP_vs_NT_human_ensembl_RBPs_DeepBind/'\n"
     ]
    }
   ],
   "source": [
    "%%bash --err error\n",
    "mkdir -p ipage\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "    # remove peak number \n",
    "    awk 'NR>1{split($1,id,\".\"); print id[1]\"\\t\"$3}' deseq/d_mtyl_${treat}_vs_NT.txt > deseq/d_mtyl_${treat}_vs_NT.c.txt\n",
    "    for f in `ls -d /flash/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl*`; do\n",
    "        base=`basename \"$f\"`\n",
    "        perl $PAGEDIR/page.pl --expfile=deseq/d_mtyl_${treat}_vs_NT.c.txt \\\n",
    "        --species=$base --exptype=continuous --ebins=11 --nodups=1\n",
    "        mv -v deseq/d_mtyl_${treat}_vs_NT.c.txt_PAGE/ ipage/d_mtyl_${treat}_vs_NT_${base}/\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "%%bash \n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "    for f in `ls -d /flash/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl*`; do\n",
    "        base=`basename \"$f\"`        \n",
    "        magick -density 300 \\\n",
    "        ipage/d_mtyl_${treat}_vs_NT_${base}/d_mtyl_${treat}_vs_NT.c.txt.summary.pdf \\\n",
    "        -flatten -quality 90 plots/ipage.d_mtyl_${treat}_vs_NT_${base}.png\n",
    "        echo '<table>'\n",
    "        echo '  <tr>'\n",
    "        echo '    <td><img src=plots/ipage.d_mtyl_'${treat}_vs_NT_${base}'.png style=\"width:600px\">'\n",
    "        echo -e \"\\\\t\"$base\n",
    "        echo '  <tr>'\n",
    "        echo '<table>'\n",
    "    done\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Hypo and Hyper-methylation gene set analysis: \n",
    "\n",
    "Let's try Salmon for the alignment and experssion quntification; We are doing this because `STAR` + `featurecount` didn't quntify most of the transcripts in the input samples. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "salmon index -t ~/genomes/hg38/gencode.v28.transcripts.fa \\\n",
    "-i ~/genomes/hg38/gencode.v28.transcripts.salmon_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/03_IN_23_METTL14_S5_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s23.METTL14.input > _sh/_log/salmon/s23.METTL14.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/08_IN_24_METTL14_S10_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s24.METTL14.input > _sh/_log/salmon/s24.METTL14.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/02_IN_23_METTL3_S4_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s23.METTL3.input > _sh/_log/salmon/s23.METTL3.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/07_IN_24_METTL3_S9_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s24.METTL3.input > _sh/_log/salmon/s24.METTL3.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/04_IN_23_Virma_S6_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s23.Virma.input > _sh/_log/salmon/s23.Virma.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/09_IN_24_Virma_S11_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s24.Virma.input > _sh/_log/salmon/s24.Virma.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/05_IN_23_WTAP_S7_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s23.WTAP.input > _sh/_log/salmon/s23.WTAP.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/10_IN_24_WTAP_S12_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s24.WTAP.input > _sh/_log/salmon/s24.WTAP.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/01_IN_23_NT_S3_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s23.NT.input > _sh/_log/salmon/s23.NT.input.salmon.out&\n",
      "nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r fastq/06_IN_24_NT_S8_L002_trim.fastq.gz -p 8 --validateMappings -o quants/s24.NT.input > _sh/_log/salmon/s24.NT.input.salmon.out&\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# I needed seprate conda env for this purpose \n",
    "mkdir -p quants\n",
    "mkdir -p _sh/_log/salmon\n",
    "# -l A or -l SR ?!\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP' 'NT')\n",
    "for treat in \"${StringArray[@]}\"; do \n",
    "    for f in fastq/*_IN_2[3-4]_$treat*trim*; do \n",
    "        base=`basename \"$f\"`;\n",
    "        out=${base/*_IN_/s};\n",
    "        out=${out/_S*/.input};\n",
    "        out=${out/_/.};        \n",
    "        echo 'nohup salmon quant -i ~/genomes/hg38/gencode.v28.transcripts.salmon_index/ -l A -r '$f' -p 12 --validateMappings -o quants/'$out' > _sh/_log/salmon/'$out'.salmon.out&'\n",
    "    done;\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(DESeq2)\n",
    "library(tidyverse)\n",
    "library(ggplot2)\n",
    "library(tximport)\n",
    "library(readr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://rdrr.io/github/jmw86069/splicejam/man/makeTx2geneFromGtf.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Parsed with column specification:\n",
      "cols(\n",
      "  TXNAME = col_character(),\n",
      "  GENEID = col_character()\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "files <- list.files(pattern=\"quant.sf\",full.names = TRUE, recursive=T)\n",
    "tx2gene <- read_tsv(\"/rumi/shams/genomes/hg38/gencode.v28.basic.tx2gene.cvs\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 6 x 2\n",
      "  TXNAME            GENEID           \n",
      "  <chr>             <chr>            \n",
      "1 ENST00000456328.2 ENSG00000223972.5\n",
      "2 ENST00000450305.2 ENSG00000223972.5\n",
      "3 ENST00000473358.1 ENSG00000243485.5\n",
      "4 ENST00000469289.1 ENSG00000243485.5\n",
      "5 ENST00000607096.1 ENSG00000284332.1\n",
      "6 ENST00000606857.1 ENSG00000268020.3\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "head (tx2gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: reading in files with read_tsv\n",
      "\n",
      "R[write to console]: 1 \n",
      "R[write to console]: 2 \n",
      "R[write to console]: 3 \n",
      "R[write to console]: 4 \n",
      "R[write to console]: 5 \n",
      "R[write to console]: 6 \n",
      "R[write to console]: 7 \n",
      "R[write to console]: 8 \n",
      "R[write to console]: 9 \n",
      "R[write to console]: 10 \n",
      "R[write to console]: \n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "names(files) <- gsub(\"./(\\\\S+)_quant/quant.sf\",\"\\\\1\",files)\n",
    "txi <- tximport(files, type = \"salmon\", tx2gene = tx2gene, txOut=T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: transcripts missing from tx2gene: 102647\n",
      "\n",
      "R[write to console]: summarizing abundance\n",
      "\n",
      "R[write to console]: summarizing counts\n",
      "\n",
      "R[write to console]: summarizing length\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "txi.gene <- summarizeToGene(txi, tx2gene, ignoreAfterBar= TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: using counts and average transcript lengths from tximport\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "lfc.cutoff = 1\n",
    "# meta \n",
    "reps <- c(rep('s23',5),rep('s24',5))\n",
    "treats <- rep(c('METTL14','METTL3','NT','Virma','WTAP'),2)\n",
    "\n",
    "colData <- data.frame(cond=treats, rep=reps, row.names=colnames(txi$abundance))\n",
    "\n",
    "dds <- DESeqDataSetFromTximport(txi.gene, colData = colData, design = ~cond + rep)\n",
    "dds$cond<- relevel(dds$cond, ref=\"NT\")\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "all_res <- list()\n",
    "for (t in c('METTL14','METTL3','Virma','WTAP')){\n",
    "    all_res[[t]]<- results(dds, contrast=c('cond',t,'NT'))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "for (t in c('METTL14','METTL3','Virma','WTAP')){\n",
    "    res = all_res[[t]]\n",
    "    res = data.frame(res [!is.na(res$padj) ,c(2,6)])\n",
    "    write.table(res, paste(\"teiser/\",t,\"_vs_NT_lfc.txt\", sep = ''), sep=\"\\t\", quote=FALSE, col.names=NA )\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TEISER\n",
    "https://github.com/goodarzilab/TEISER\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "prepare inputs "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir -p teiser\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "    ### exp input \n",
    "    # remove dups \n",
    "    awk -F \"\\t\" '{OFS=FS}NR>1{$1=substr($1,1,15); print $1, $2}' teiser/${treat}_vs_NT_lfc.txt | awk '!seen[$1]++' > teiser/${treat}_vs_NT_lfc.c.txt\n",
    "    ### hyper_methylation gene list\n",
    "    # subset by threshold\n",
    "    awk '$2>=2{print $1}' deseq/d_mtyl_${treat}_vs_NT.c.txt | awk '!seen[$1]++' > teiser/${treat}_hyper_methylation.txt\n",
    "    # intersect with exp\n",
    "    awk 'NR==FNR{A[$1];next}$1 in A' teiser/${treat}_vs_NT_lfc.c.txt teiser/${treat}_hyper_methylation.txt > teiser/${treat}_hyper_methylation.c.txt\n",
    "    ### hypo_methylation gene list\n",
    "    # subset by threshold\n",
    "    awk '$2<-2 {print $1}' deseq/d_mtyl_${treat}_vs_NT.c.txt | awk '!seen[$1]++' > teiser/${treat}_hypo_methylation.txt\n",
    "    # intersect with exp\n",
    "    awk 'NR==FNR{A[$1];next}$1 in A' teiser/${treat}_vs_NT_lfc.c.txt teiser/${treat}_hypo_methylation.txt > teiser/${treat}_hypo_methylation.c.txt\n",
    "done "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sat May  2 01:49:52 UTC 2020\n",
      "Remove duplicates, create teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 17979 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000255\n",
      "pass = 0.404800\n",
      "z = 0.047909\n",
      "freq 0: 0.001947\t5\t2568\t52\t17979\n",
      "freq 1: 0.004673\t12\t2568\t52\t17979\n",
      "freq 2: 0.003115\t8\t2568\t52\t17979\n",
      "freq 3: 0.003894\t10\t2568\t52\t17979\n",
      "freq 4: 0.002336\t6\t2568\t52\t17979\n",
      "freq 5: 0.001558\t4\t2568\t52\t17979\n",
      "freq 6: 0.002723\t7\t2571\t52\t17979\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.84\t-1.36\n",
      "0\n",
      "Outputing EPS file teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt.summary.eps\n",
      "Convert to PDF teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt.summary.eps teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/METTL14_vs_NT_lfc.c.txt_GENESET' -> 'teiser/METTL14_hyper_methylation_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sat May  2 01:50:04 UTC 2020\n",
      "Remove duplicates, create teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 17979 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001782\n",
      "pass = 0.000000\n",
      "z = 10.792479\n",
      "freq 0: 0.004283\t11\t2568\t294\t17979\n",
      "freq 1: 0.015187\t39\t2568\t294\t17979\n",
      "freq 2: 0.021417\t55\t2568\t294\t17979\n",
      "freq 3: 0.018692\t48\t2568\t294\t17979\n",
      "freq 4: 0.017134\t44\t2568\t294\t17979\n",
      "freq 5: 0.022975\t59\t2568\t294\t17979\n",
      "freq 6: 0.014780\t38\t2571\t294\t17979\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.84\t-1.36\n",
      "0\n",
      "Outputing EPS file teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt.summary.eps\n",
      "Convert to PDF teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt.summary.eps teiser/METTL14_vs_NT_lfc.c.txt_GENESET/METTL14_vs_NT_lfc.c.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/METTL14_vs_NT_lfc.c.txt_GENESET' -> 'teiser/METTL14_hypo_methylation_GENESET'\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sat May  2 01:50:11 UTC 2020\n",
      "Remove duplicates, create teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 17225 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000648\n",
      "pass = 0.019300\n",
      "z = 2.682979\n",
      "freq 0: 0.002438\t6\t2461\t97\t17225\n",
      "freq 1: 0.007314\t18\t2461\t97\t17225\n",
      "freq 2: 0.008127\t20\t2461\t97\t17225\n",
      "freq 3: 0.007720\t19\t2461\t97\t17225\n",
      "freq 4: 0.006095\t15\t2461\t97\t17225\n",
      "freq 5: 0.002844\t7\t2461\t97\t17225\n",
      "freq 6: 0.004880\t12\t2459\t97\t17225\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.98\t-1.64\n",
      "0\n",
      "Outputing EPS file teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt.summary.eps\n",
      "Convert to PDF teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt.summary.eps teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/METTL3_vs_NT_lfc.c.txt_GENESET' -> 'teiser/METTL3_hyper_methylation_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sat May  2 01:50:17 UTC 2020\n",
      "Remove duplicates, create teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 17225 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.003783\n",
      "pass = 0.000000\n",
      "z = 24.007784\n",
      "freq 0: 0.007314\t18\t2461\t532\t17225\n",
      "freq 1: 0.022755\t56\t2461\t532\t17225\n",
      "freq 2: 0.035351\t87\t2461\t532\t17225\n",
      "freq 3: 0.033320\t82\t2461\t532\t17225\n",
      "freq 4: 0.037383\t92\t2461\t532\t17225\n",
      "freq 5: 0.042666\t105\t2461\t532\t17225\n",
      "freq 6: 0.037414\t92\t2459\t532\t17225\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.98\t-1.64\n",
      "0\n",
      "Outputing EPS file teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt.summary.eps\n",
      "Convert to PDF teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt.summary.eps teiser/METTL3_vs_NT_lfc.c.txt_GENESET/METTL3_vs_NT_lfc.c.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/METTL3_vs_NT_lfc.c.txt_GENESET' -> 'teiser/METTL3_hypo_methylation_GENESET'\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sat May  2 01:50:24 UTC 2020\n",
      "Remove duplicates, create teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 17225 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000391\n",
      "pass = 0.165800\n",
      "z = 0.925938\n",
      "freq 0: 0.002844\t7\t2461\t69\t17225\n",
      "freq 1: 0.005689\t14\t2461\t69\t17225\n",
      "freq 2: 0.005689\t14\t2461\t69\t17225\n",
      "freq 3: 0.003251\t8\t2461\t69\t17225\n",
      "freq 4: 0.004876\t12\t2461\t69\t17225\n",
      "freq 5: 0.004063\t10\t2461\t69\t17225\n",
      "freq 6: 0.001627\t4\t2459\t69\t17225\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.94\t-1.12\n",
      "0\n",
      "Outputing EPS file teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt.summary.eps\n",
      "Convert to PDF teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt.summary.eps teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/Virma_vs_NT_lfc.c.txt_GENESET' -> 'teiser/Virma_hyper_methylation_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sat May  2 01:50:31 UTC 2020\n",
      "Remove duplicates, create teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 17225 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001538\n",
      "pass = 0.000000\n",
      "z = 8.886864\n",
      "freq 0: 0.008533\t21\t2461\t309\t17225\n",
      "freq 1: 0.012597\t31\t2461\t309\t17225\n",
      "freq 2: 0.017066\t42\t2461\t309\t17225\n",
      "freq 3: 0.018692\t46\t2461\t309\t17225\n",
      "freq 4: 0.019098\t47\t2461\t309\t17225\n",
      "freq 5: 0.029663\t73\t2461\t309\t17225\n",
      "freq 6: 0.019927\t49\t2459\t309\t17225\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.94\t-1.12\n",
      "0\n",
      "Outputing EPS file teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt.summary.eps\n",
      "Convert to PDF teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt.summary.eps teiser/Virma_vs_NT_lfc.c.txt_GENESET/Virma_vs_NT_lfc.c.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/Virma_vs_NT_lfc.c.txt_GENESET' -> 'teiser/Virma_hypo_methylation_GENESET'\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sat May  2 01:50:38 UTC 2020\n",
      "Remove duplicates, create teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 14962 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000056\n",
      "pass = 0.977300\n",
      "z = -1.394586\n",
      "freq 0: 0.004679\t10\t2137\t73\t14962\n",
      "freq 1: 0.005615\t12\t2137\t73\t14962\n",
      "freq 2: 0.005147\t11\t2137\t73\t14962\n",
      "freq 3: 0.003744\t8\t2137\t73\t14962\n",
      "freq 4: 0.005615\t12\t2137\t73\t14962\n",
      "freq 5: 0.004679\t10\t2137\t73\t14962\n",
      "freq 6: 0.004673\t10\t2140\t73\t14962\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.86\t-1.28\n",
      "0\n",
      "Outputing EPS file teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt.summary.eps\n",
      "Convert to PDF teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt.summary.eps teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/WTAP_vs_NT_lfc.c.txt_GENESET' -> 'teiser/WTAP_hyper_methylation_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /flash/bin/TEISERv1.1\n",
      "Sat May  2 01:50:44 UTC 2020\n",
      "Remove duplicates, create teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 14962 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.002482\n",
      "pass = 0.000000\n",
      "z = 12.938136\n",
      "freq 0: 0.011231\t24\t2137\t462\t14962\n",
      "freq 1: 0.028077\t60\t2137\t462\t14962\n",
      "freq 2: 0.029481\t63\t2137\t462\t14962\n",
      "freq 3: 0.030884\t66\t2137\t462\t14962\n",
      "freq 4: 0.035564\t76\t2137\t462\t14962\n",
      "freq 5: 0.043519\t93\t2137\t462\t14962\n",
      "freq 6: 0.037383\t80\t2140\t462\t14962\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.86\t-1.28\n",
      "0\n",
      "Outputing EPS file teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt.summary.eps\n",
      "Convert to PDF teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt.summary.eps teiser/WTAP_vs_NT_lfc.c.txt_GENESET/WTAP_vs_NT_lfc.c.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'teiser/WTAP_vs_NT_lfc.c.txt_GENESET' -> 'teiser/WTAP_hypo_methylation_GENESET'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "for treat in \"${StringArray[@]}\"; do\n",
    "    echo hyper_methylation\n",
    "    perl $TEISERDIR/run_mi_gene_list.pl \\\n",
    "    --expfile=teiser/${treat}_vs_NT_lfc.c.txt \\\n",
    "    --genefile=teiser/${treat}_hyper_methylation.c.txt \\\n",
    "    --exptype=continuous \\\n",
    "    --ebins=7 \\\n",
    "    --species=human \\\n",
    "    --doremovedups=0 \\\n",
    "    --doremoveextra=0\n",
    "    mv -v teiser/${treat}_vs_NT_lfc.c.txt_GENESET teiser/${treat}_hyper_methylation_GENESET\n",
    "    \n",
    "    magick -density 300 \\\n",
    "    teiser/${treat}_hyper_methylation_GENESET/*pdf \\\n",
    "    -flatten -quality 90 plots/teiser.${treat}_hyper_methylation_GENESET.png\n",
    "    \n",
    "    echo hypo_methylation\n",
    "    perl $TEISERDIR/run_mi_gene_list.pl \\\n",
    "    --expfile=teiser/${treat}_vs_NT_lfc.c.txt \\\n",
    "    --genefile=teiser/${treat}_hypo_methylation.c.txt \\\n",
    "    --exptype=continuous \\\n",
    "    --ebins=7 \\\n",
    "    --species=human \\\n",
    "    --doremovedups=0 \\\n",
    "    --doremoveextra=0\n",
    "    mv -v teiser/${treat}_vs_NT_lfc.c.txt_GENESET teiser/${treat}_hypo_methylation_GENESET\n",
    "    \n",
    "    magick -density 300 \\\n",
    "    teiser/${treat}_hypo_methylation_GENESET/*pdf \\\n",
    "    -flatten -quality 90 plots/teiser.${treat}_hypo_methylation_GENESET.png\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<!-- ### using bam to bed for cov count \n",
    "\n",
    "1. `intersectBed -split` make it work with bed12 file\n",
    "\n",
    "3. make a intersect gtf to extract names for later. ` --> "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash \n",
    "# bamToBed -i bam/s23.human.METTL3.input.bam | intersectBed -s -split -wo -a - -b exomepeak/control/peak.bed | awk '{print $10}' | sort  tail"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash \n",
    "# bamToBed -i bam/s23.human.METTL3.input.bam | intersectBed -s -split -wo -a - -b exomepeak/control/peak.bed \\\n",
    "# | cut -f10 | sort | uniq -c | awk '{ print $2 \"\\t\" $1}' | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R version 3.5.1 (2018-07-02)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 18.04.2 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /rumi/shams/abe/anaconda3/envs/deseq/lib/R/lib/libRlapack.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      " [1] parallel  stats4    tools     stats     graphics  grDevices utils    \n",
      " [8] datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      " [1] tximport_1.10.1             forcats_0.4.0              \n",
      " [3] stringr_1.4.0               dplyr_0.8.3                \n",
      " [5] purrr_0.3.3                 readr_1.3.1                \n",
      " [7] tidyr_1.0.0                 tibble_2.1.3               \n",
      " [9] ggplot2_3.2.1               tidyverse_1.2.1            \n",
      "[11] DESeq2_1.20.0               SummarizedExperiment_1.10.1\n",
      "[13] DelayedArray_0.6.6          BiocParallel_1.14.2        \n",
      "[15] matrixStats_0.55.0          Biobase_2.40.0             \n",
      "[17] GenomicRanges_1.32.7        GenomeInfoDb_1.16.0        \n",
      "[19] IRanges_2.14.12             S4Vectors_0.18.3           \n",
      "[21] BiocGenerics_0.26.0        \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      " [1] nlme_3.1-141           bitops_1.0-6           lubridate_1.7.4       \n",
      " [4] bit64_0.9-7            RColorBrewer_1.1-2     httr_1.4.1            \n",
      " [7] backports_1.1.5        utf8_1.1.4             R6_2.4.0              \n",
      "[10] rpart_4.1-15           Hmisc_4.2-0            DBI_1.0.0             \n",
      "[13] lazyeval_0.2.2         colorspace_1.4-1       nnet_7.3-12           \n",
      "[16] withr_2.1.2            tidyselect_0.2.5       gridExtra_2.3         \n",
      "[19] bit_1.1-14             compiler_3.5.1         cli_1.1.0             \n",
      "[22] rvest_0.3.5            htmlTable_1.13.2       xml2_1.2.2            \n",
      "[25] scales_1.0.0           checkmate_1.9.4        genefilter_1.62.0     \n",
      "[28] digest_0.6.22          foreign_0.8-72         XVector_0.20.0        \n",
      "[31] base64enc_0.1-3        pkgconfig_2.0.3        htmltools_0.4.0       \n",
      "[34] htmlwidgets_1.5.1      rlang_0.4.1            readxl_1.3.1          \n",
      "[37] rstudioapi_0.10        RSQLite_2.1.2          jsonlite_1.6          \n",
      "[40] acepack_1.4.1          RCurl_1.95-4.12        magrittr_1.5          \n",
      "[43] GenomeInfoDbData_1.1.0 Formula_1.2-3          Matrix_1.2-17         \n",
      "[46] fansi_0.4.0            Rcpp_1.0.3             munsell_0.5.0         \n",
      "[49] lifecycle_0.1.0        stringi_1.4.3          zlibbioc_1.26.0       \n",
      "[52] grid_3.5.1             blob_1.2.0             crayon_1.3.4          \n",
      "[55] lattice_0.20-38        haven_2.2.0            splines_3.5.1         \n",
      "[58] annotate_1.58.0        hms_0.5.2              locfit_1.5-9.1        \n",
      "[61] zeallot_0.1.0          knitr_1.25             pillar_1.4.2          \n",
      "[64] geneplotter_1.58.0     XML_3.98-1.20          glue_1.3.1            \n",
      "[67] latticeExtra_0.6-28    data.table_1.11.6      modelr_0.1.5          \n",
      "[70] vctrs_0.2.0            cellranger_1.1.0       gtable_0.3.0          \n",
      "[73] assertthat_0.2.1       xfun_0.10              xtable_1.8-4          \n",
      "[76] broom_0.5.0            survival_2.44-1.1      AnnotationDbi_1.42.1  \n",
      "[79] memoise_1.1.0          cluster_2.1.0         \n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
