{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metagene plots\n",
    "Now, let's draw **metagene plots**: (using seprate conda env and ipython kernel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Guitar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/abe/genomes/hg38/gencode.v34/gencode.v34.annotation.gtf',organism='Homo sapiens')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"20200729162348\"\n",
      "[1] \"There are 228048 transcripts of 60669 genes in the genome.\"\n",
      "[1] \"total 228048 transcripts extracted ...\"\n",
      "[1] \"total 97941 transcripts left after ambiguity filter ...\"\n",
      "[1] \"total 97941 transcripts left after check chromosome validity ...\"\n",
      "[1] \"total 10939 mRNAs left after component length filter ...\"\n",
      "[1] \"total 60598 ncRNAs left after ncRNA length filter ...\"\n",
      "[1] \"generate components for all tx\"\n",
      "[1] \"generate components for mRNA\"\n",
      "[1] \"generate components for lncRNA\"\n",
      "[1] \"generate chiped transcriptome\"\n",
      "[1] \"generate coverage checking ranges for tx\"\n",
      "[1] \"generate coverage checking ranges for mrna\"\n",
      "[1] \"generate coverage checking ranges for ncrna\"\n",
      "[1] \"20200729162947\"\n",
      "[1] \"import BED file radar/result.sig.bed\"\n",
      "[1] \"sample 10 points for Group1\"\n",
      "[1] \"start figure plotting for tx ...\"\n",
      "[1] \"start figure plotting for mrna ...\"\n",
      "[1] \"start figure plotting for ncrna ...\"\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "GuitarPlot(txTxdb = txdb,stBedFiles = list(\"radar/result.sig.bed\"),miscOutFilePrefix = \"Guitar\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !for f in *.pdf; do bash ~/Workflows/my_scripts/pdf2png.sh $f; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'Guitar_mrna_test.pdf' -> 'plots/Guitar_mrna_test.pdf'\r\n",
      "renamed 'Guitar_mrna_test.png' -> 'plots/Guitar_mrna_test.png'\r\n",
      "renamed 'Guitar_ncrna_test.pdf' -> 'plots/Guitar_ncrna_test.pdf'\r\n",
      "renamed 'Guitar_ncrna_test.png' -> 'plots/Guitar_ncrna_test.png'\r\n",
      "renamed 'Guitar_tx_test.pdf' -> 'plots/Guitar_tx_test.pdf'\r\n",
      "renamed 'Guitar_tx_test.png' -> 'plots/Guitar_tx_test.png'\r\n"
     ]
    }
   ],
   "source": [
    "mv -v Guitar_* plots/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# FIRE \n",
    "https://github.com/goodarzilab/FIRE\n",
    "### Motif analysis "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[AG]GAC\r\n",
      "[AGT][AG]AC[ACT]\r\n"
     ]
    }
   ],
   "source": [
    "cat fire/motifs_of_interest.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "\n",
    "bedtools getfasta -name -s -fi /rumi/shams/genomes/hg38/hg38.fa -bed radar/result.sig.bed -split -fo radar/result.sig.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p fire "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# prepare fasta file for FIRE\n",
    "perl /flash/bin/TEISERv1.1/prep_seqs_for_teiser_run.pl radar/result.sig.fa fire/peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Aug  6 13:55:32 UTC 2020\n",
      "RNA, Processing peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt\n",
      "RNA, Step 2: skip seed optimization, use -motiffile_rna option instead.\n",
      "RNA, Step 3: evaluation of motif significances.\n",
      "Processing motif [AG]GAC.\n",
      "MI=0.00492797\n",
      "Shuffle rank=0\n",
      "Z-score=44.607\n",
      "Sigificance category= OK-NO-SEED\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [AGT][AG]AC[ACT].\n",
      "MI=0.00674459\n",
      "Shuffle rank=0\n",
      "Z-score=62.008\n",
      "Sigificance category= OK-NO-SEED\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "RNA, Step 3.5: creating binary expression profiles.\n",
      "RNA, Step 4: discovery of distance constraints.\n",
      "Read 2 motifs.\n",
      "Read 9374 (genes) x 2 (conditions) matrix.\n",
      "Read 9374 (genes) x 1 (conditions) matrix.\n",
      "Reading sequences and building motif profiles ...Done.\n",
      "Average sequence length is 50.\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AG]GAC, present in 3271 genes.\n",
      "       Average distance: MI=0.0008, rank=1730/10000, Z=0.7684, mbins=3\n",
      "  Orientation bias {5'): MI=0.0049, rank=0/10000, Z=44.8914\n",
      "  Orientation bias (3'): MI=0.0007, rank=29/10000, Z=5.4831\n",
      "       Copy number (1+): 3271 genes, MI=0.0049, rank=0/10000, Z=44.1359\n",
      "       Copy number (2+): 797 genes, MI=0.0001, rank=3039/10000, Z=0.1173\n",
      "       Copy number (3+): 144 genes, MI=0.0012, rank=1/10000, Z=9.9201\n",
      "       Copy number (4+): 19 genes, MI=0.0002, rank=1669/10000, Z=1.0995\n",
      "       Copy number (5+): 2 genes, MI=0.0000, rank=10000/10000, Z=-1.0104\n",
      "       Copy number (6+): 2 genes, MI=0.0000, rank=10000/10000, Z=-0.9938\n",
      "       Copy number (7+): 2 genes, MI=0.0000, rank=10000/10000, Z=-1.0036\n",
      "       Copy number (8+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT][AG]AC[ACT], present in 5874 genes.\n",
      "       Average distance: MI=0.0011, rank=101/10000, Z=3.6015, mbins=3\n",
      "  Orientation bias {5'): MI=0.0067, rank=0/10000, Z=62.2616\n",
      "  Orientation bias (3'): MI=0.0031, rank=0/10000, Z=27.6077\n",
      "       Copy number (1+): 5874 genes, MI=0.0067, rank=0/10000, Z=60.3224\n",
      "       Copy number (2+): 2679 genes, MI=0.0001, rank=3354/10000, Z=-0.0267\n",
      "       Copy number (3+): 923 genes, MI=0.0014, rank=0/10000, Z=11.7770\n",
      "       Copy number (4+): 256 genes, MI=0.0040, rank=0/10000, Z=36.0522\n",
      "       Copy number (5+): 72 genes, MI=0.0013, rank=0/10000, Z=11.1348\n",
      "       Copy number (6+): 17 genes, MI=0.0002, rank=1348/10000, Z=1.3615\n",
      "       Copy number (7+): 7 genes, MI=0.0001, rank=4510/10000, Z=0.1496\n",
      "       Copy number (8+): 3 genes, MI=0.0003, rank=2525/10000, Z=1.7205\n",
      "\n",
      "RNA, Step 5.5: conservation.\n",
      "Setting -rna to 1.\n",
      "[AG]GAC\t98.12\n",
      "[AGT][AG]AC[ACT]\t21.42\n",
      "RNA, Step 5: summarize information.\n",
      "Reading RNA motifs mi_dist output ...Done.\n",
      "Parsing RNA motif file ... Done.\n",
      "Reading mi_signif output file for RNA motifs ...!!!! 641688!!!! -3323984Done.\n",
      "Got 2 motifs\n",
      "Writing good columns to peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.signifcolumns ...Done.\n",
      "Outputing summary ... Done.\n",
      "HI!HELLO!peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.binary created.\n",
      "RNA, Step 5.3: generate profiles for motifs in summary file.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [AGT][AG]AC[ACT] ... Done.\n",
      "Generating profile for [AG]GAC ... Done.\n",
      "RNA, Step 5.3 bis: Generate profiles for /flash/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [AGT][AG]AC[ACT] ... Done.\n",
      "Generating profile for [AG]GAC ... Done.\n",
      "RNA, Determining target gene sets.\n",
      "RNA, Drawing position histograms.\n",
      "RNA, Step 6: cluster motifs.\n",
      "Using positive correlation.\n",
      "Doing all stats.\n",
      "Summary file has 0 DNA motifs and 2 RNA motifs.\n",
      "Read in RNA sequences and build motif profiles ...Done.\n",
      "Outputing MI matrix ... Done.\n",
      "[AGT][AG]AC[ACT]\t0\n",
      "[AGT][AG]AC[ACT]\t[AG]GAC\t0.2946\t0\t2683.4893\t0.6562\t576\t1.8661\t576\t1.8661\n",
      "[AG]GAC\t0\n",
      "Finish calculated rank, z-scores for remaining entries ... [AGT][AG]AC[ACT]\t[AGT][AG]AC[ACT]\t0.9216\t0\t8594.7727\t1.0000\t0\t10.9940\t0\t10.9940\n",
      "[AG]GAC\t[AG]GAC\t0.7101\t0\t6494.4804\t1.0000\t3478\t0.0748\t3478\t0.0748\n",
      "Done.\n",
      "RNA, Step 5.6: generate genome-wide GO enrichments for motifs in summary file.\n",
      "RNA, draw motif maps for co-localizing pairs\n",
      "Draw motif map for [AGT][AG]AC[ACT] .. Done.\n",
      "Draw motif map for [AG]GAC .. Done.\n",
      "Draw motif map for [AGT][AG]AC[ACT],[AGT][AG]AC[ACT] ..  (peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.mimatrix_OUT/[AGT][AG]AC[ACT]_[AGT][AG]AC[ACT].eps) Done.\n",
      "RNA, Step 6.8: identifying motifs that match micrornas.\n",
      "RNA, Step 7: draw matrix figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 400, xbase = 35, ybase = 250\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 0.eps ... Done.\n",
      "Processing [AG]GAC ... Outputing motif 1.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 0.png ... \n",
      "Processing [AG]GAC ... Outputing motif 1.png ... \n",
      "RNA, Step 7b: draw density figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 400, xbase = 35, ybase = 250\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 0.eps ... Done.\n",
      "Processing [AG]GAC ... Outputing motif 1.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 0.png ... \n",
      "Processing [AG]GAC ... Outputing motif 1.png ... \n",
      "Parsing distance report file ...  Done.\n",
      "[AGT][AG]AC[ACT] has no position or orientation bias.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 230, ysize = 310, xbase = 35, ybase = 115\n",
      "[AGT][AG]AC[ACT]\t0\n",
      "[AG]GAC\t1\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf ... Done.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 250, ysize = 305, xbase = 75, ybase = 130\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.html[AGT][AG]AC[ACT]\t0\n",
      "[AG]GAC\t1\n",
      "RNA, Step 10: motif report.\n",
      "Thu Aug  6 13:56:30 UTC 2020\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Segmentation fault (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /flash/bin/FIRE-1.1/SCRIPTS/draw_position_histogram.pl line 134.\n",
      "Can't locate Hypergeom.pm in @INC (you may need to install the Hypergeom module) (@INC contains: /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.12.4/darwin-thread-multi-2level/nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.10.0/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.18.2/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.22.0/x86_64-linux/Hypergeom.pm /flash/bin/FIRE-1.1/SCRIPTS /rumi/shams/abe/perl5/lib/perl5/x86_64-linux-gnu-thread-multi /rumi/shams/abe/perl5/lib/perl5 /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.26.1 /usr/local/share/perl/5.26.1 /usr/lib/x86_64-linux-gnu/perl5/5.26 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.26 /usr/share/perl/5.26 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /flash/bin/FIRE-1.1/SCRIPTS/GroupEnrichment.pm line 24.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "cd fire/\n",
    "# run FIRE --doskipdiscovery=1\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --doskipdiscovery=1 --motiffile_rna=motifs_of_interest.txt --oribiasonly=0\n",
    "mv peaks_teiser.txt_FIRE/ motifs_of_interest_FIRE\n",
    "cd ../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Aug  6 14:08:07 UTC 2020\n",
      "RNA, Processing peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt\n",
      "RNA, Step 1: seed discovery.\n",
      "Setting -shuffle to 16384.\n",
      "Expression file is a 9374 by 1 matrix.\n",
      "Min expression value read is 0, max is 1\n",
      "Reading sequences ... Done\n",
      "Number of ORFs = 9374\n",
      "Calculating Mutual Information for all seeds ... Done\n",
      "top 10 seeds...\n",
      "CCCCCCC\n",
      "Determining threshold ...\n",
      "Evaluating seed #0, CCCCCCC ... passed.\n",
      "Evaluating seed #100, AGGTGGA ... not passed.\n",
      "Decreasing intervals phase.\n",
      "Evaluating seed #50, GCTAGGC ... not passed.\n",
      "Evaluating seed #25, TACAAAT ... passed.\n",
      "Evaluating seed #37, GAGGAAG ... passed.\n",
      "Evaluating seed #43, GAAACCC ... not passed.\n",
      "Searches for 5 consecutive 'not passed'.\n",
      "Threshold set to seed #37 (included).\n",
      "CCCCCCC\t0.0073\n",
      "GGGGGGG\t0.0061\n",
      "AATGTGG\t0.0048\n",
      "CCCCCCT\t0.0043\n",
      "TCCCCCC\t0.0040\n",
      "GAGAAAC\t0.0039\n",
      "AGGGGGG\t0.0039\n",
      "GCCCCCC\t0.0037\n",
      "AGAAACC\t0.0037\n",
      "TTTTTTT\t0.0036\n",
      "GTGGCAA\t0.0034\n",
      "TTCATAC\t0.0033\n",
      "GAATGTG\t0.0031\n",
      "GGGGGGA\t0.0031\n",
      "CCCCCCA\t0.0031\n",
      "GAGAGAG\t0.0029\n",
      "TCTCTCA\t0.0028\n",
      "TGGGGGG\t0.0028\n",
      "ATGTGGA\t0.0026\n",
      "ATCCCCC\t0.0024\n",
      "CATACTG\t0.0024\n",
      "TCATACT\t0.0023\n",
      "AAGCCTT\t0.0022\n",
      "GGGGGGT\t0.0022\n",
      "GGGGGGC\t0.0022\n",
      "TACAAAT\t0.0021\n",
      "ATCACTC\t0.0021\n",
      "ACTGGAG\t0.0021\n",
      "GGAAGAA\t0.0020\n",
      "ATTCATA\t0.0020\n",
      "CCCCCTG\t0.0020\n",
      "AGAATTC\t0.0019\n",
      "TGAATGT\t0.0019\n",
      "GAATTCA\t0.0019\n",
      "GTGAGAG\t0.0019\n",
      "CAGGGGG\t0.0018\n",
      "GGCCCAG\t0.0018\n",
      "GAGGAAG\t0.0018\n",
      "RNA, Step 2: seed optimization.\n",
      "Using minr = 2.0000.\n",
      "Motif length will be 9.\n",
      "Creating prefix tree for storing patterns ... Done.\n",
      "Reading expression data ... Done.\n",
      "Creating hash table for gene names ... Done.\n",
      "Reading sequences and building the index.\n",
      "160952 k-mers in index\n",
      "Matrix occupancy = 396291 / 1508764048.000000\n",
      "Done.                                \n",
      "\n",
      "Evaluating seed CCCCCCC ... \n",
      "optimizing.\n",
      "-243387712\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [CGT].[CT]CCCCC[ACT]\n",
      "Mutual information: 0.009685\n",
      "Optimization stability: 1\n",
      "Z-score: 87.208\n",
      "Starting seed: CCCCCCC\n",
      "Seed MI: 0.007106\n",
      "\n",
      "Evaluating seed GGGGGGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=65.4329\n",
      "optimizing.\n",
      "-242668880\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [AGT]G[GT]G[AGT]G.[GT].\n",
      "Mutual information: 0.009994\n",
      "Optimization stability: 1\n",
      "Z-score: 88.192\n",
      "Starting seed: GGGGGGG\n",
      "Seed MI: 0.005734\n",
      "\n",
      "Evaluating seed AATGTGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=40.3512\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=7.2926\n",
      "optimizing.\n",
      "-241950048\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [CG][AG][ACT].GTG[CG][ACG]\n",
      "Mutual information: 0.009527\n",
      "Optimization stability: 1\n",
      "Z-score: 87.439\n",
      "Starting seed: AATGTGG\n",
      "Seed MI: 0.004672\n",
      "\n",
      "Evaluating seed CCCCCCT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0195\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.019 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed TCCCCCC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0151\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.015 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GAGAAAC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=22.7304\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=44.1552\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=1.9581\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.958 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed AGGGGGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=424.4717\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.1220\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.122 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed GCCCCCC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0169\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.017 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed AGAAACC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=20.1452\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=8.7020\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=1.3096\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.310 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed TTTTTTT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=66.3992\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=5.4692\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=8.9992\n",
      "optimizing.\n",
      "-240781168\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [AGT][GT]TT[GT]TT[CGT][CGT]\n",
      "Mutual information: 0.005533\n",
      "Optimization stability: 4\n",
      "Z-score: 51.238\n",
      "Starting seed: TTTTTTT\n",
      "Seed MI: 0.003746\n",
      "\n",
      "Evaluating seed GTGGCAA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=90.7343\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=5.6903\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.2950\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.295 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed TTCATAC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=14.2513\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=2.9823\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=72.2644\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=907.3990\n",
      "optimizing.\n",
      "-239987328\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: .[GT][ACT]CA[GT][AC]..\n",
      "Mutual information: 0.007099\n",
      "Optimization stability: 1\n",
      "Z-score: 64.026\n",
      "Starting seed: TTCATAC\n",
      "Seed MI: 0.003117\n",
      "\n",
      "Evaluating seed GAATGTG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=24.5738\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=28.8872\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.1006\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.101 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GGGGGGA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=35.0455\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.1413\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.141 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed CCCCCCA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0025\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.002 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GAGAGAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=26.6763\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=1.3343\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.334 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed TCTCTCA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=10.2738\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=1506.4942\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=4.4152\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=404.8468\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=55.0115\n",
      "optimizing.\n",
      "-238968464\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [ACT]TCTCTCA.\n",
      "Mutual information: 0.003120\n",
      "Optimization stability: 10\n",
      "Z-score: 27.632\n",
      "Starting seed: TCTCTCA\n",
      "Seed MI: 0.002705\n",
      "\n",
      "Evaluating seed TGGGGGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=31.9973\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.0837\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.084 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed ATGTGGA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=74.1334\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=23.1235\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.2198\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.220 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed ATCCCCC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0381\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.038 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed CATACTG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=61.4274\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=2.5441\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=3.2105\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=7.6543\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=0.4049\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.405 for .[GT][ACT]CA[GT][AC].. (< minr=2.000).\n",
      "\n",
      "Evaluating seed TCATACT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=9.8089\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=4.2091\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=9.8884\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=10.8367\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=0.3048\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.305 for .[GT][ACT]CA[GT][AC].. (< minr=2.000).\n",
      "\n",
      "Evaluating seed AAGCCTT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=6.5666\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=17.2742\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.7838\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.784 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GGGGGGT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=24.3867\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.0638\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.064 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed GGGGGGC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=173.2965\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.1338\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.134 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed TACAAAT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=8.3689\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=6.2097\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.8060\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.806 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed ATCACTC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=23.5214\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=179.0895\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=10.6221\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=25.4679\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=34.5897\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=3.2066\n",
      "optimizing.\n",
      "-237574560\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: .[ACG].C[AG]C[AGT][AC][AGT]\n",
      "Mutual information: 0.008361\n",
      "Optimization stability: 1\n",
      "Z-score: 75.298\n",
      "Starting seed: ATCACTC\n",
      "Seed MI: 0.002137\n",
      "\n",
      "Evaluating seed ACTGGAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=34.1494\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=36.2412\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=7.9176\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=2496.2476\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=3.6553\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=19.6046\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT] : I(B;E|A)/I(A;B)=8.2936\n",
      "optimizing.\n",
      "-236855728\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [ACT].[ACT][ACT]GG[AT][ACG][ACG]\n",
      "Mutual information: 0.012394\n",
      "Optimization stability: 1\n",
      "Z-score: 112.285\n",
      "Starting seed: ACTGGAG\n",
      "Seed MI: 0.002035\n",
      "\n",
      "Evaluating seed GGAAGAA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=18.5903\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=49.2675\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=2427.5032\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=147.9026\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=6.4242\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=23.8996\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT] : I(B;E|A)/I(A;B)=15.8394\n",
      "Motif [ACT].[ACT][ACT]GG[AT][ACG][ACG] : I(B;E|A)/I(A;B)=0.7094\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.709 for [ACT].[ACT][ACT]GG[AT][ACG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed ATTCATA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=9.0893\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=1.8732\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=36.1699\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.873 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed CCCCCTG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0155\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.016 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed AGAATTC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=5.9938\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=8.4715\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=92.4097\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=6.6479\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=1.7733\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.773 for .[GT][ACT]CA[GT][AC].. (< minr=2.000).\n",
      "\n",
      "Evaluating seed TGAATGT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=46.8529\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=37.8191\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.2125\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.212 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GAATTCA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=5.5340\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=67.6018\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=6.6976\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=6.1424\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=1.0244\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.024 for .[GT][ACT]CA[GT][AC].. (< minr=2.000).\n",
      "\n",
      "Evaluating seed GTGAGAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=68.3707\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.0567\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.057 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed CAGGGGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=1099.4616\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.2377\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.238 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed GGCCCAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=35.3417\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=255.4496\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=2687.7706\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=46.6215\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=101.6700\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=37.1945\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT] : I(B;E|A)/I(A;B)=35.5731\n",
      "Motif [ACT].[ACT][ACT]GG[AT][ACG][ACG] : I(B;E|A)/I(A;B)=4.4844\n",
      "optimizing.\n",
      "-235495584\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: .[CG]GCC[AC]A.[ACG]\n",
      "Mutual information: 0.003056\n",
      "Optimization stability: 1\n",
      "Z-score: 27.215\n",
      "Starting seed: GGCCCAG\n",
      "Seed MI: 0.001813\n",
      "\n",
      "Evaluating seed GAGGAAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=9.1527\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=8.7599\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=9.1585\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=11.5250\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=383.4880\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=16.8900\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT] : I(B;E|A)/I(A;B)=5.4151\n",
      "Motif [ACT].[ACT][ACT]GG[AT][ACG][ACG] : I(B;E|A)/I(A;B)=2.6727\n",
      "Motif .[CG]GCC[AC]A.[ACG] : I(B;E|A)/I(A;B)=16.2077\n",
      "optimizing.\n",
      "-234776752\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: GGAGG[AT]A[GT][AGT]\n",
      "Mutual information: 0.003509\n",
      "Optimization stability: 4\n",
      "Z-score: 31.012\n",
      "Starting seed: GAGGAAG\n",
      "Seed MI: 0.001711\n",
      "RNA, Step 3: evaluation of motif significances.\n",
      "Processing motif [CGT].[CT]CCCCC[ACT].\n",
      "MI=0.00968482\n",
      "Shuffle rank=0\n",
      "Z-score=89.057\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [AGT]G[GT]G[AGT]G.[GT]..\n",
      "MI=0.00999403\n",
      "Shuffle rank=0\n",
      "Z-score=93.256\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [CG][AG][ACT].GTG[CG][ACG].\n",
      "MI=0.00952664\n",
      "Shuffle rank=0\n",
      "Z-score=86.088\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [AGT][GT]TT[GT]TT[CGT][CGT].\n",
      "MI=0.00553276\n",
      "Shuffle rank=0\n",
      "Z-score=50.901\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif .[GT][ACT]CA[GT][AC]...\n",
      "MI=0.00709922\n",
      "Shuffle rank=0\n",
      "Z-score=64.596\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [ACT]TCTCTCA..\n",
      "MI=0.00312046\n",
      "Shuffle rank=0\n",
      "Z-score=27.633\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif .[ACG].C[AG]C[AGT][AC][AGT].\n",
      "MI=0.00836072\n",
      "Shuffle rank=0\n",
      "Z-score=76.441\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [ACT].[ACT][ACT]GG[AT][ACG][ACG].\n",
      "MI=0.01239392\n",
      "Shuffle rank=0\n",
      "Z-score=112.063\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif .[CG]GCC[AC]A.[ACG].\n",
      "MI=0.00305585\n",
      "Shuffle rank=0\n",
      "Z-score=27.115\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif GGAGG[AT]A[GT][AGT].\n",
      "MI=0.00350923\n",
      "Shuffle rank=0\n",
      "Z-score=31.792\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "RNA, Step 3.5: creating binary expression profiles.\n",
      "RNA, Step 4: discovery of distance constraints.\n",
      "Read 10 motifs.\n",
      "Read 9374 (genes) x 10 (conditions) matrix.\n",
      "Read 9374 (genes) x 1 (conditions) matrix.\n",
      "Reading sequences and building motif profiles ...Done.\n",
      "Average sequence length is 50.\n",
      "Quantized E vector into 2 bins\n",
      "Motif [CGT].[CT]CCCCC[ACT], present in 280 genes.\n",
      "       Average distance: MI=0.0041, rank=4856/10000, Z=-0.2231, mbins=3\n",
      "  Orientation bias {5'): MI=0.0097, rank=0/10000, Z=88.0621\n",
      "  Orientation bias (3'): MI=0.0059, rank=0/10000, Z=53.3486\n",
      "       Copy number (1+): 280 genes, MI=0.0097, rank=0/10000, Z=85.5798\n",
      "       Copy number (2+): 100 genes, MI=0.0057, rank=0/10000, Z=53.0047\n",
      "       Copy number (3+): 56 genes, MI=0.0047, rank=0/10000, Z=41.0431\n",
      "       Copy number (4+): 32 genes, MI=0.0027, rank=0/10000, Z=23.3255\n",
      "       Copy number (5+): 16 genes, MI=0.0017, rank=0/10000, Z=14.6532\n",
      "       Copy number (6+): 8 genes, MI=0.0009, rank=80/10000, Z=6.4446\n",
      "       Copy number (7+): 5 genes, MI=0.0005, rank=613/10000, Z=3.4452\n",
      "       Copy number (8+): 3 genes, MI=0.0003, rank=2513/10000, Z=1.7260\n",
      "       Copy number (9+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]., present in 1207 genes.\n",
      "       Average distance: MI=0.0003, rank=4746/10000, Z=-0.3127, mbins=2\n",
      "  Orientation bias {5'): MI=0.0100, rank=0/10000, Z=91.9656\n",
      "  Orientation bias (3'): MI=0.0082, rank=0/10000, Z=73.8283\n",
      "       Copy number (1+): 1207 genes, MI=0.0100, rank=0/10000, Z=92.0017\n",
      "       Copy number (2+): 390 genes, MI=0.0057, rank=0/10000, Z=51.4639\n",
      "       Copy number (3+): 150 genes, MI=0.0047, rank=0/10000, Z=42.3123\n",
      "       Copy number (4+): 61 genes, MI=0.0026, rank=0/10000, Z=23.0176\n",
      "       Copy number (5+): 36 genes, MI=0.0013, rank=0/10000, Z=11.3198\n",
      "       Copy number (6+): 21 genes, MI=0.0007, rank=73/10000, Z=5.4270\n",
      "       Copy number (7+): 14 genes, MI=0.0006, rank=114/10000, Z=4.8928\n",
      "       Copy number (8+): 7 genes, MI=0.0001, rank=4530/10000, Z=0.1403\n",
      "       Copy number (9+): 5 genes, MI=0.0000, rank=10000/10000, Z=-0.5672\n",
      "       Copy number (10+): 3 genes, MI=0.0000, rank=10000/10000, Z=-0.5784\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG], present in 569 genes.\n",
      "       Average distance: MI=0.0006, rank=7816/10000, Z=-0.7646, mbins=3\n",
      "  Orientation bias {5'): MI=0.0095, rank=0/10000, Z=83.2066\n",
      "  Orientation bias (3'): MI=0.0009, rank=7/10000, Z=7.2755\n",
      "       Copy number (1+): 569 genes, MI=0.0095, rank=0/10000, Z=85.8462\n",
      "       Copy number (2+): 33 genes, MI=0.0004, rank=316/10000, Z=2.9522\n",
      "       Copy number (3+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT], present in 257 genes.\n",
      "       Average distance: MI=0.0075, rank=1147/10000, Z=1.1719, mbins=2\n",
      "  Orientation bias {5'): MI=0.0055, rank=0/10000, Z=49.6126\n",
      "  Orientation bias (3'): MI=0.0008, rank=19/10000, Z=6.4529\n",
      "       Copy number (1+): 257 genes, MI=0.0055, rank=0/10000, Z=49.2060\n",
      "       Copy number (2+): 89 genes, MI=0.0028, rank=0/10000, Z=24.8781\n",
      "       Copy number (3+): 42 genes, MI=0.0013, rank=0/10000, Z=11.3944\n",
      "       Copy number (4+): 19 genes, MI=0.0003, rank=628/10000, Z=2.3384\n",
      "       Copy number (5+): 11 genes, MI=0.0004, rank=650/10000, Z=2.4829\n",
      "       Copy number (6+): 6 genes, MI=0.0006, rank=287/10000, Z=4.5011\n",
      "       Copy number (7+): 5 genes, MI=0.0005, rank=650/10000, Z=3.3694\n",
      "       Copy number (8+): 3 genes, MI=0.0003, rank=2456/10000, Z=1.7525\n",
      "       Copy number (9+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "       Copy number (10+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif .[GT][ACT]CA[GT][AC].., present in 2416 genes.\n",
      "       Average distance: MI=0.0013, rank=3454/10000, Z=0.1670, mbins=5\n",
      "  Orientation bias {5'): MI=0.0071, rank=0/10000, Z=64.8691\n",
      "  Orientation bias (3'): MI=0.0017, rank=0/10000, Z=14.6125\n",
      "       Copy number (1+): 2416 genes, MI=0.0071, rank=0/10000, Z=64.1633\n",
      "       Copy number (2+): 360 genes, MI=0.0023, rank=0/10000, Z=20.9995\n",
      "       Copy number (3+): 44 genes, MI=0.0006, rank=88/10000, Z=4.7106\n",
      "       Copy number (4+): 5 genes, MI=0.0000, rank=10000/10000, Z=-0.5646\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [ACT]TCTCTCA., present in 53 genes.\n",
      "       Average distance: MI=0.0624, rank=2595/10000, Z=0.9381, mbins=3\n",
      "  Orientation bias {5'): MI=0.0031, rank=0/10000, Z=26.8242\n",
      "  Orientation bias (3'): MI=0.0014, rank=0/10000, Z=12.0169\n",
      "       Copy number (1+): 53 genes, MI=0.0031, rank=0/10000, Z=27.9427\n",
      "       Copy number (2+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT], present in 1400 genes.\n",
      "       Average distance: MI=0.0009, rank=4402/10000, Z=-0.1684, mbins=3\n",
      "  Orientation bias {5'): MI=0.0084, rank=0/10000, Z=78.3081\n",
      "  Orientation bias (3'): MI=0.0018, rank=0/10000, Z=15.8822\n",
      "       Copy number (1+): 1400 genes, MI=0.0084, rank=0/10000, Z=77.6687\n",
      "       Copy number (2+): 282 genes, MI=0.0035, rank=0/10000, Z=31.0258\n",
      "       Copy number (3+): 66 genes, MI=0.0016, rank=0/10000, Z=13.8825\n",
      "       Copy number (4+): 21 genes, MI=0.0009, rank=15/10000, Z=7.5218\n",
      "       Copy number (5+): 4 genes, MI=0.0000, rank=10000/10000, Z=-0.7124\n",
      "       Copy number (6+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "       Copy number (7+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "       Copy number (8+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [ACT].[ACT][ACT]GG[AT][ACG][ACG], present in 2784 genes.\n",
      "       Average distance: MI=0.0004, rank=4430/10000, Z=-0.1826, mbins=3\n",
      "  Orientation bias {5'): MI=0.0124, rank=0/10000, Z=110.0754\n",
      "  Orientation bias (3'): MI=0.0062, rank=0/10000, Z=57.6482\n",
      "       Copy number (1+): 2784 genes, MI=0.0124, rank=0/10000, Z=113.5838\n",
      "       Copy number (2+): 413 genes, MI=0.0041, rank=0/10000, Z=37.4894\n",
      "       Copy number (3+): 49 genes, MI=0.0006, rank=84/10000, Z=4.6189\n",
      "       Copy number (4+): 7 genes, MI=0.0000, rank=10000/10000, Z=-0.6025\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif .[CG]GCC[AC]A.[ACG], present in 346 genes.\n",
      "       Average distance: MI=0.0084, rank=1312/10000, Z=1.0083, mbins=3\n",
      "  Orientation bias {5'): MI=0.0031, rank=0/10000, Z=27.5556\n",
      "  Orientation bias (3'): MI=0.0001, rank=1935/10000, Z=0.6425\n",
      "       Copy number (1+): 346 genes, MI=0.0031, rank=0/10000, Z=27.2677\n",
      "       Copy number (2+): 14 genes, MI=0.0000, rank=10000/10000, Z=-0.7121\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif GGAGG[AT]A[GT][AGT], present in 61 genes.\n",
      "       Average distance: MI=0.0426, rank=5782/10000, Z=-0.0247, mbins=4\n",
      "  Orientation bias {5'): MI=0.0035, rank=0/10000, Z=31.5304\n",
      "  Orientation bias (3'): MI=0.0003, rank=966/10000, Z=1.6355\n",
      "       Copy number (1+): 61 genes, MI=0.0035, rank=0/10000, Z=31.9871\n",
      "       Copy number (2+): 2 genes, MI=0.0002, rank=5048/10000, Z=0.9904\n",
      "\n",
      "RNA, Step 5.5: conservation.\n",
      "Setting -rna to 1.\n",
      "[CGT].[CT]CCCCC[ACT]\t96.91\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t99.55\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t96.47\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t98.58\n",
      ".[GT][ACT]CA[GT][AC]..\t90.26\n",
      "[ACT]TCTCTCA.\t51.96\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t93.84\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t93.20\n",
      ".[CG]GCC[AC]A.[ACG]\t96.44\n",
      "GGAGG[AT]A[GT][AGT]\t34.93\n",
      "RNA, Step 5: summarize information.\n",
      "Reading RNA motifs mi_dist output ...Done.\n",
      "Parsing RNA motif file ... Done.\n",
      "Reading mi_signif output file for RNA motifs ...!!!! 254088!!!! -170544!!!! -4138504!!!! -147808!!!! -186944!!!! -186536!!!! -186632!!!! -186656!!!! -186656!!!! -186728Done.\n",
      "Got 10 motifs\n",
      "Writing good columns to peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.signifcolumns ...Done.\n",
      "Outputing summary ... Done.\n",
      "HI!HELLO!peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.binary created.\n",
      "RNA, Step 5.3: generate profiles for motifs in summary file.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Done.\n",
      "Generating profile for [AGT]G[GT]G[AGT]G.[GT]. ... Done.\n",
      "Generating profile for [CGT].[CT]CCCCC[ACT] ... Done.\n",
      "Generating profile for [CG][AG][ACT].GTG[CG][ACG] ... Done.\n",
      "Generating profile for .[ACG].C[AG]C[AGT][AC][AGT] ... Done.\n",
      "Generating profile for .[GT][ACT]CA[GT][AC].. ... Done.\n",
      "Generating profile for [AGT][GT]TT[GT]TT[CGT][CGT] ... Done.\n",
      "Generating profile for GGAGG[AT]A[GT][AGT] ... Done.\n",
      "Generating profile for [ACT]TCTCTCA. ... Done.\n",
      "Generating profile for .[CG]GCC[AC]A.[ACG] ... Done.\n",
      "RNA, Step 5.3 bis: Generate profiles for /flash/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Done.\n",
      "Generating profile for [AGT]G[GT]G[AGT]G.[GT]. ... Done.\n",
      "Generating profile for [CGT].[CT]CCCCC[ACT] ... Done.\n",
      "Generating profile for [CG][AG][ACT].GTG[CG][ACG] ... Done.\n",
      "Generating profile for .[ACG].C[AG]C[AGT][AC][AGT] ... Done.\n",
      "Generating profile for .[GT][ACT]CA[GT][AC].. ... Done.\n",
      "Generating profile for [AGT][GT]TT[GT]TT[CGT][CGT] ... Done.\n",
      "Generating profile for GGAGG[AT]A[GT][AGT] ... Done.\n",
      "Generating profile for [ACT]TCTCTCA. ... Done.\n",
      "Generating profile for .[CG]GCC[AC]A.[ACG] ... Done.\n",
      "RNA, Determining target gene sets.\n",
      "RNA, Drawing position histograms.\n",
      "RNA, Step 6: cluster motifs.\n",
      "Using positive correlation.\n",
      "Doing all stats.\n",
      "Summary file has 0 DNA motifs and 10 RNA motifs.\n",
      "Read in RNA sequences and build motif profiles ...Done.\n",
      "Outputing MI matrix ... Done.\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t0\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[AGT]G[GT]G[AGT]G.[GT].\t0.0250\t0\t229.7973\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[CGT].[CT]CCCCC[ACT]\t0.0070\t0\t62.6175\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[CG][AG][ACT].GTG[CG][ACG]\t0.0104\t0\t98.1567\t0.1386\t2941\t0.2245\t2941\t0.2245\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t0\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.0281\t0\t254.9889\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t.[GT][ACT]CA[GT][AC]..\t0.0200\t0\t182.5732\t0.1517\t1897\t0.7163\t1897\t0.7163\n",
      ".[GT][ACT]CA[GT][AC]..\t[CG][AG][ACT].GTG[CG][ACG]\t0.0025\t0\t21.1851\t0.0659\t2526\t0.4954\t2526\t0.4954\n",
      ".[GT][ACT]CA[GT][AC]..\t0\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0059\t0\t52.7105\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\tGGAGG[AT]A[GT][AGT]\t0.0026\t0\t23.0743\t0.0589\t1990\t1.1417\t1990\t1.1417\n",
      "GGAGG[AT]A[GT][AGT]\t[CG][AG][ACT].GTG[CG][ACG]\t0.0009\t5\t6.8129\t0.0438\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[ACT]TCTCTCA.\t0.0015\t0\t12.4063\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t.[CG]GCC[AC]A.[ACG]\t0.0079\t0\t72.4243\t0.1187\t6963\t-0.5912\t6963\t-0.5912\n",
      ".[CG]GCC[AC]A.[ACG]\t[CG][AG][ACT].GTG[CG][ACG]\t0.0009\t9\t7.3260\t0.0444\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t1\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[CGT].[CT]CCCCC[ACT]\t0.0001\t2843\t0.1400\t0.0123\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.0007\t19\t6.1670\t0.0040\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0009\t4\t7.9129\t0.0074\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\tGGAGG[AT]A[GT][AGT]\t0.0007\t69\t5.8415\t0.0000\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[ACT]TCTCTCA.\t0.0002\t1886\t0.5795\t0.0079\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t.[CG]GCC[AC]A.[ACG]\t0.0031\t0\t28.4740\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t2\n",
      "[CGT].[CT]CCCCC[ACT]\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.0006\t67\t5.1690\t0.0149\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0001\t4923\t-0.1848\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\tGGAGG[AT]A[GT][AGT]\t0.0002\t2924\t1.1914\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t[ACT]TCTCTCA.\t0.0002\t3339\t0.9824\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t.[CG]GCC[AC]A.[ACG]\t0.0009\t32\t7.0777\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t3\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0000\t8112\t-0.6569\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\tGGAGG[AT]A[GT][AGT]\t0.0008\t50\t6.4601\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t[ACT]TCTCTCA.\t0.0019\t0\t15.3332\t0.0440\t10000\t0.0000\t10000\t0.0000\n",
      "[ACT]TCTCTCA.\t3\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t.[CG]GCC[AC]A.[ACG]\t0.0035\t0\t30.6598\t0.0000\tnan\tnan\tnan\tnan\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t4\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\tGGAGG[AT]A[GT][AGT]\t0.0002\t3386\t0.9019\t0.0000\tnan\tnan\tnan\tnan\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t.[CG]GCC[AC]A.[ACG]\t0.0007\t72\t5.9122\t0.0000\tnan\tnan\tnan\tnan\n",
      "GGAGG[AT]A[GT][AGT]\t5\n",
      "GGAGG[AT]A[GT][AGT]\t.[CG]GCC[AC]A.[ACG]\t0.0000\t6400\t-0.6496\t0.0046\tnan\tnan\tnan\tnan\n",
      ".[CG]GCC[AC]A.[ACG]\t6\n",
      "Finish calculated rank, z-scores for remaining entries ... [ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t0.6765\t0\t6213.1363\t1.0000\t240\t2.5391\t240\t2.5391\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[AGT]G[GT]G[AGT]G.[GT].\t0.4160\t0\t3836.7599\t1.0000\t330\t2.3212\t330\t2.3212\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[CG][AG][ACT].GTG[CG][ACG]\t0.0057\t0\t50.2295\t0.0000\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t.[GT][ACT]CA[GT][AC]..\t0.0207\t0\t190.8069\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t[CGT].[CT]CCCCC[ACT]\t0.1656\t0\t1475.5923\t1.0000\t3164\t0.5083\t3164\t0.5083\n",
      "[CGT].[CT]CCCCC[ACT]\t[CG][AG][ACT].GTG[CG][ACG]\t0.0016\t0\t13.6778\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t.[GT][ACT]CA[GT][AC]..\t0.0058\t0\t52.0300\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t[CG][AG][ACT].GTG[CG][ACG]\t0.2597\t0\t2452.8707\t1.0000\t3176\t0.4756\t3176\t0.4756\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.0064\t0\t57.5484\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0014\t2\t11.4588\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t[ACT]TCTCTCA.\t0.0003\t1271\t2.1718\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.4489\t0\t3976.8975\t1.0000\t658\t1.7217\t658\t1.7217\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t.[GT][ACT]CA[GT][AC]..\t0.0233\t0\t223.5651\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[GT][ACT]CA[GT][AC]..\t.[GT][ACT]CA[GT][AC]..\t0.6111\t0\t5645.9196\t1.0000\t3514\t0.1667\t3514\t0.1667\n",
      ".[GT][ACT]CA[GT][AC]..\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0049\t0\t43.7631\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[GT][ACT]CA[GT][AC]..\tGGAGG[AT]A[GT][AGT]\t0.0003\t540\t2.1355\t0.0168\tnan\tnan\tnan\tnan\n",
      ".[GT][ACT]CA[GT][AC]..\t[ACT]TCTCTCA.\t0.0012\t8\t9.7887\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[GT][ACT]CA[GT][AC]..\t.[CG]GCC[AC]A.[ACG]\t0.0006\t56\t5.1236\t0.0224\tnan\tnan\tnan\tnan\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.1453\t0\t1254.5912\t1.0000\t8451\t-0.7616\t8451\t-0.7616\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t[ACT]TCTCTCA.\t0.0001\t6305\t-0.2851\t0.0010\tnan\tnan\tnan\tnan\n",
      "GGAGG[AT]A[GT][AGT]\tGGAGG[AT]A[GT][AGT]\t0.0519\t0\t813.7128\t1.0000\t10100\t-100.0000\t10100\t-100.0000\n",
      "GGAGG[AT]A[GT][AGT]\t[ACT]TCTCTCA.\t0.0000\t10000\t-0.3202\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT]TCTCTCA.\t[ACT]TCTCTCA.\t0.0463\t0\t749.4929\t1.0000\t10100\t-100.0000\t10100\t-100.0000\n",
      "[ACT]TCTCTCA.\t.[CG]GCC[AC]A.[ACG]\t0.0002\t3302\t0.9838\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[CG]GCC[AC]A.[ACG]\t.[CG]GCC[AC]A.[ACG]\t0.1662\t0\t1455.9965\t1.0000\t946\t2.0945\t946\t2.0945\n",
      "Done.\n",
      "RNA, Step 5.6: generate genome-wide GO enrichments for motifs in summary file.\n",
      "RNA, draw motif maps for co-localizing pairs\n",
      "Draw motif map for [ACT].[ACT][ACT]GG[AT][ACG][ACG] .. Done.\n",
      "Draw motif map for [AGT]G[GT]G[AGT]G.[GT]. .. Done.\n",
      "Draw motif map for [CGT].[CT]CCCCC[ACT] .. Done.\n",
      "Draw motif map for [CG][AG][ACT].GTG[CG][ACG] .. Done.\n",
      "Draw motif map for .[ACG].C[AG]C[AGT][AC][AGT] .. Done.\n",
      "Draw motif map for .[GT][ACT]CA[GT][AC].. .. Done.\n",
      "Draw motif map for [AGT][GT]TT[GT]TT[CGT][CGT] .. Done.\n",
      "Draw motif map for GGAGG[AT]A[GT][AGT] .. Done.\n",
      "Draw motif map for [ACT]TCTCTCA. .. Done.\n",
      "Draw motif map for .[CG]GCC[AC]A.[ACG] .. Done.\n",
      "RNA, Step 6.8: identifying motifs that match micrornas.\n",
      "RNA, Step 7: draw matrix figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 700, xbase = 35, ybase = 250\n",
      "Processing [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Outputing motif 0.eps ... Done.\n",
      "Processing [CG][AG][ACT].GTG[CG][ACG] ... Outputing motif 3.eps ... Done.\n",
      "Processing .[GT][ACT]CA[GT][AC].. ... Outputing motif 5.eps ... Done.\n",
      "Processing [AGT]G[GT]G[AGT]G.[GT]. ... Outputing motif 1.eps ... Done.\n",
      "Processing [CGT].[CT]CCCCC[ACT] ... Outputing motif 2.eps ... Done.\n",
      "Processing .[ACG].C[AG]C[AGT][AC][AGT] ... Outputing motif 4.eps ... Done.\n",
      "Processing [ACT]TCTCTCA. ... Outputing motif 8.eps ... Done.\n",
      "Processing [AGT][GT]TT[GT]TT[CGT][CGT] ... Outputing motif 6.eps ... Done.\n",
      "Processing GGAGG[AT]A[GT][AGT] ... Outputing motif 7.eps ... Done.\n",
      "Processing .[CG]GCC[AC]A.[ACG] ... Outputing motif 9.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Outputing motif 0.png ... \n",
      "Processing [CG][AG][ACT].GTG[CG][ACG] ... Outputing motif 3.png ... \n",
      "Processing .[GT][ACT]CA[GT][AC].. ... Outputing motif 5.png ... \n",
      "Processing [AGT]G[GT]G[AGT]G.[GT]. ... Outputing motif 1.png ... \n",
      "Processing [CGT].[CT]CCCCC[ACT] ... Outputing motif 2.png ... \n",
      "Processing .[ACG].C[AG]C[AGT][AC][AGT] ... Outputing motif 4.png ... \n",
      "Processing [ACT]TCTCTCA. ... Outputing motif 8.png ... \n",
      "Processing [AGT][GT]TT[GT]TT[CGT][CGT] ... Outputing motif 6.png ... \n",
      "Processing GGAGG[AT]A[GT][AGT] ... Outputing motif 7.png ... \n",
      "Processing .[CG]GCC[AC]A.[ACG] ... Outputing motif 9.png ... \n",
      "RNA, Step 7b: draw density figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 700, xbase = 35, ybase = 250\n",
      "Processing [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Outputing motif 0.eps ... Done.\n",
      "Processing [CG][AG][ACT].GTG[CG][ACG] ... Outputing motif 3.eps ... Done.\n",
      "Processing .[GT][ACT]CA[GT][AC].. ... Outputing motif 5.eps ... Done.\n",
      "Processing [AGT]G[GT]G[AGT]G.[GT]. ... Outputing motif 1.eps ... Done.\n",
      "Processing [CGT].[CT]CCCCC[ACT] ... Outputing motif 2.eps ... Done.\n",
      "Processing .[ACG].C[AG]C[AGT][AC][AGT] ... Outputing motif 4.eps ... Done.\n",
      "Processing [ACT]TCTCTCA. ... Outputing motif 8.eps ... Done.\n",
      "Processing [AGT][GT]TT[GT]TT[CGT][CGT] ... Outputing motif 6.eps ... Done.\n",
      "Processing GGAGG[AT]A[GT][AGT] ... Outputing motif 7.eps ... Done.\n",
      "Processing .[CG]GCC[AC]A.[ACG] ... Outputing motif 9.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Outputing motif 0.png ... \n",
      "Processing [CG][AG][ACT].GTG[CG][ACG] ... Outputing motif 3.png ... \n",
      "Processing .[GT][ACT]CA[GT][AC].. ... Outputing motif 5.png ... \n",
      "Processing [AGT]G[GT]G[AGT]G.[GT]. ... Outputing motif 1.png ... \n",
      "Processing [CGT].[CT]CCCCC[ACT] ... Outputing motif 2.png ... \n",
      "Processing .[ACG].C[AG]C[AGT][AC][AGT] ... Outputing motif 4.png ... \n",
      "Processing [ACT]TCTCTCA. ... Outputing motif 8.png ... \n",
      "Processing [AGT][GT]TT[GT]TT[CGT][CGT] ... Outputing motif 6.png ... \n",
      "Processing GGAGG[AT]A[GT][AGT] ... Outputing motif 7.png ... \n",
      "Processing .[CG]GCC[AC]A.[ACG] ... Outputing motif 9.png ... \n",
      "Parsing distance report file ...  Done.\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG] has no position or orientation bias.\n",
      "[AGT]G[GT]G[AGT]G.[GT]. has no position or orientation bias.\n",
      "[CGT].[CT]CCCCC[ACT] has no position or orientation bias.\n",
      ".[ACG].C[AG]C[AGT][AC][AGT] has no position or orientation bias.\n",
      ".[GT][ACT]CA[GT][AC].. has no position or orientation bias.\n",
      "[ACT]TCTCTCA. has no position or orientation bias.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 470, ysize = 550, xbase = 35, ybase = 115\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t0\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t3\n",
      ".[GT][ACT]CA[GT][AC]..\t5\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t1\n",
      "[CGT].[CT]CCCCC[ACT]\t2\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t4\n",
      "[ACT]TCTCTCA.\t8\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t6\n",
      "GGAGG[AT]A[GT][AGT]\t7\n",
      ".[CG]GCC[AC]A.[ACG]\t9\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf ... Done.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 410, ysize = 465, xbase = 75, ybase = 130\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.html[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t0\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t3\n",
      ".[GT][ACT]CA[GT][AC]..\t5\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t1\n",
      "[CGT].[CT]CCCCC[ACT]\t2\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t4\n",
      "[ACT]TCTCTCA.\t8\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t6\n",
      "GGAGG[AT]A[GT][AGT]\t7\n",
      ".[CG]GCC[AC]A.[ACG]\t9\n",
      "RNA, Step 10: motif report.\n",
      "Thu Aug  6 14:16:33 UTC 2020\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genregexp: malloc.c:2401: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "genregexp: malloc.c:2401: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Segmentation fault (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Segmentation fault (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /flash/bin/FIRE-1.1/SCRIPTS/draw_position_histogram.pl line 134.\n",
      "Can't locate Hypergeom.pm in @INC (you may need to install the Hypergeom module) (@INC contains: /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.12.4/darwin-thread-multi-2level/nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.10.0/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.18.2/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.22.0/x86_64-linux/Hypergeom.pm /flash/bin/FIRE-1.1/SCRIPTS /rumi/shams/abe/perl5/lib/perl5/x86_64-linux-gnu-thread-multi /rumi/shams/abe/perl5/lib/perl5 /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.26.1 /usr/local/share/perl/5.26.1 /usr/lib/x86_64-linux-gnu/perl5/5.26 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.26 /usr/share/perl/5.26 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /flash/bin/FIRE-1.1/SCRIPTS/GroupEnrichment.pm line 24.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "cd fire/\n",
    "# run FIRE --doskipdiscovery=0\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --oribiasonly=0\n",
    "mv peaks_teiser.txt_FIRE/ discovery_FIRE\n",
    "cd ../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'fire/discovery_FIRE/RNA/peaks_teiser.txt.densities.pdf' -> 'fire/discovery_FIRE_densities.pdf'\n",
      "renamed 'fire/discovery_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf' -> 'fire/discovery_FIRE_fullmimatrix.pdf'\n",
      "renamed 'fire/discovery_FIRE/RNA/peaks_teiser.txt.summary.pdf' -> 'fire/discovery_FIRE_summary.pdf'\n",
      "renamed 'fire/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.densities.pdf' -> 'fire/motifs_of_interest_FIRE_densities.pdf'\n",
      "renamed 'fire/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf' -> 'fire/motifs_of_interest_FIRE_fullmimatrix.pdf'\n",
      "renamed 'fire/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.summary.pdf' -> 'fire/motifs_of_interest_FIRE_summary.pdf'\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in  fire/*/RNA/*.pdf; do \n",
    "    o=${f/\\/RNA\\/peaks_teiser.txt./_}\n",
    "    mv -v $f $o; \n",
    "done "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fire/discovery_FIRE_densities.pdf > fire/discovery_FIRE_densities.png\n",
      "done!\n",
      "fire/discovery_FIRE_fullmimatrix.pdf > fire/discovery_FIRE_fullmimatrix.png\n",
      "done!\n",
      "fire/discovery_FIRE_summary.pdf > fire/discovery_FIRE_summary.png\n",
      "done!\n",
      "fire/motifs_of_interest_FIRE_densities.pdf > fire/motifs_of_interest_FIRE_densities.png\n",
      "done!\n",
      "fire/motifs_of_interest_FIRE_fullmimatrix.pdf > fire/motifs_of_interest_FIRE_fullmimatrix.png\n",
      "done!\n",
      "fire/motifs_of_interest_FIRE_summary.pdf > fire/motifs_of_interest_FIRE_summary.png\n",
      "done!\n"
     ]
    }
   ],
   "source": [
    "!for f in fire/*pdf; do bash /rumi/shams/abe/Workflows/my_scripts/pdf2png.sh $f; done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DRACH & RGAC motifs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=fire/motifs_of_interest.png style=\"width:1100px\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discover over/under represented motifs "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=fire/discovery_FIRE.png style=\"width:1100px\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# iPAGE\n",
    "https://github.com/goodarzilab/PAGE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4656 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4656\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "423 hits go_terms_num = 17689\n",
      "cat 4182, mi = 0.00634442, intracellular transport\n",
      "cat 2448, mi = 0.00497632, histone methyltransferase activity (H3-K4 specific)\n",
      "cat 1268, mi = 0.00441760, mitotic cell cycle\n",
      "cat 4144, mi = 0.00435438, positive regulation of substrate adhesion-dependent cell spreading\n",
      "cat 2446, mi = 0.00434639, histone H3-K4 methylation\n",
      "cat 1684, mi = 0.00421286, costamere\n",
      "cat 1905, mi = 0.00415751, Cul3-RING ubiquitin ligase complex\n",
      "cat 476, mi = 0.00406981, camera-type eye development\n",
      "cat 5204, mi = 0.00406172, protein targeting to Golgi\n",
      "cat 1286, mi = 0.00400341, midbody\n",
      "cat 1776, mi = 0.00386987, Ras protein signal transduction\n",
      "cat 7253, mi = 0.00382194, regulation of type I interferon production\n",
      "cat 1324, mi = 0.00372986, nucleotide-binding oligomerization domain containing signaling pathway\n",
      "cat 1949, mi = 0.00371670, fibroblast growth factor receptor signaling pathway\n",
      "cat 2239, mi = 0.00369737, cellular response to cAMP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 4182, mi = 0.006, pvalue = 0.000, intracellular transport\n",
      "Passed: cat 2448, mi = 0.005, pvalue = 0.000, histone methyltransferase activity (H3-K4 specific)\n",
      "Passed: cat 1268, mi = 0.004, pvalue = 0.002, mitotic cell cycle\n",
      "Passed: cat 4144, mi = 0.004, pvalue = 0.004, positive regulation of substrate adhesion-dependent cell spreading\n",
      "Didn't pass (cutoff=0.0050): cat 1684, mi = 0.004, pvalue = 0.005, costamere\n",
      "Didn't pass (cutoff=0.0050): cat 1905, mi = 0.004, pvalue = 0.006, Cul3-RING ubiquitin ligase complex\n",
      "Passed: cat 476, mi = 0.004, pvalue = 0.003, camera-type eye development\n",
      "Passed: cat 7253, mi = 0.004, pvalue = 0.002, regulation of type I interferon production\n",
      "Didn't pass (cutoff=0.0050): cat 2239, mi = 0.004, pvalue = 0.009, cellular response to cAMP\n",
      "Didn't pass (cutoff=0.0050): cat 898, mi = 0.004, pvalue = 0.007, antigen processing and presentation of exogenous peptide antigen via MHC class II\n",
      "Didn't pass (cutoff=0.0050): cat 4036, mi = 0.004, pvalue = 0.010, PDZ domain binding\n",
      "Didn't pass (cutoff=0.0050): cat 308, mi = 0.004, pvalue = 0.019, neuron projection development\n",
      "Didn't pass (cutoff=0.0050): cat 1872, mi = 0.004, pvalue = 0.022, spliceosomal complex\n",
      "Didn't pass (cutoff=0.0050): cat 1714, mi = 0.003, pvalue = 0.020, nucleotidyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 7688, mi = 0.003, pvalue = 0.005, histone monoubiquitination\n",
      "Didn't pass (cutoff=0.0050): cat 1405, mi = 0.003, pvalue = 0.016, secretory granule lumen\n",
      "Didn't pass (cutoff=0.0050): cat 84, mi = 0.003, pvalue = 0.024, animal organ regeneration\n",
      "Passed: cat 875, mi = 0.003, pvalue = 0.002, positive regulation of cytosolic calcium ion concentration involved in phospholipase C-activating G-protein coupled signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 4166, mi = 0.003, pvalue = 0.015, S100 protein binding\n",
      "Didn't pass (cutoff=0.0050): cat 931, mi = 0.003, pvalue = 0.007, cell fate commitment\n",
      "Passed: cat 5849, mi = 0.003, pvalue = 0.001, positive regulation of ATPase activity\n",
      "Didn't pass (cutoff=0.0050): cat 5326, mi = 0.003, pvalue = 0.022, ankyrin binding\n",
      "Didn't pass (cutoff=0.0050): cat 417, mi = 0.003, pvalue = 0.035, positive regulation of catalytic activity\n",
      "Didn't pass (cutoff=0.0050): cat 2659, mi = 0.003, pvalue = 0.022, ribonucleoprotein complex binding\n",
      "Didn't pass (cutoff=0.0050): cat 1192, mi = 0.003, pvalue = 0.042, positive regulation of cell growth\n",
      "Didn't pass (cutoff=0.0050): cat 3508, mi = 0.003, pvalue = 0.044, liver development\n",
      "Didn't pass (cutoff=0.0050): cat 599, mi = 0.003, pvalue = 0.041, positive regulation of JNK cascade\n",
      "Didn't pass (cutoff=0.0050): cat 2596, mi = 0.003, pvalue = 0.043, microtubule cytoskeleton organization\n",
      "Didn't pass (cutoff=0.0050): cat 439, mi = 0.003, pvalue = 0.040, histone acetyltransferase binding\n",
      "Didn't pass (cutoff=0.0050): cat 83, mi = 0.003, pvalue = 0.047, response to toxic substance\n",
      "Didn't pass (cutoff=0.0050): cat 9059, mi = 0.003, pvalue = 0.029, DNA-directed RNA polymerase II, holoenzyme\n",
      "Didn't pass (cutoff=0.0050): cat 1935, mi = 0.003, pvalue = 0.028, endoplasmic reticulum unfolded protein response\n",
      "Didn't pass (cutoff=0.0050): cat 2874, mi = 0.003, pvalue = 0.046, RNA metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1186, mi = 0.003, pvalue = 0.048, regulation of cyclin-dependent protein serine/threonine kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1979, mi = 0.003, pvalue = 0.054, apoptotic signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 5110, mi = 0.003, pvalue = 0.024, mitotic spindle organization\n",
      "Didn't pass (cutoff=0.0050): cat 4928, mi = 0.003, pvalue = 0.009, clathrin coat assembly\n",
      "Didn't pass (cutoff=0.0050): cat 1145, mi = 0.003, pvalue = 0.065, transcription regulatory region sequence-specific DNA binding\n",
      "Didn't pass (cutoff=0.0050): cat 4968, mi = 0.003, pvalue = 0.040, structural molecule activity conferring elasticity\n",
      "Didn't pass (cutoff=0.0050): cat 9495, mi = 0.003, pvalue = 0.038, regulation of voltage-gated calcium channel activity\n",
      "Didn't pass (cutoff=0.0050): cat 1956, mi = 0.003, pvalue = 0.055, Rab GTPase binding\n",
      "Total number of categories evaluated = 68.\n",
      "Number of categories that passed the tests = 8\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 6, 0.283236490709247\n",
      "best merge is 2, 3, 0.53264786142253\n",
      "best merge is 1, 5, 0.566384552259294\n",
      "best merge is 0, 7, 0.719338708268853\n",
      "order classes\n",
      "2 - 1.13276910451859\n",
      "3 - 0.799130200624774\n",
      "2 3 6\n",
      "1 5\n",
      "0\n",
      "4\n",
      "7\n",
      "Done.Start drawing\n",
      "6.80\t-6.64\n",
      "positive regulation of ATPase activity, GO:0032781\n",
      "mitotic cell cycle, GO:0000278\n",
      "positive regulation of cytosolic calcium ion concentration involved in phospholipase C-activating G-protein coupled signaling pathway, GO:0051482\n",
      "regulation of type I interferon production, GO:0032479\n",
      "positive regulation of substrate adhesion-dependent cell spreading, GO:1900026\n",
      "camera-type eye development, GO:0043010\n",
      "histone methyltransferase activity (H3-K4 specific), GO:0042800\n",
      "intracellular transport, GO:0046907\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 6, 0.283236490709247\n",
      "best merge is 2, 3, 0.53264786142253\n",
      "best merge is 1, 5, 0.566384552259294\n",
      "best merge is 0, 7, 0.719338708268853\n",
      "order classes\n",
      "2 - 1.13276910451859\n",
      "3 - 0.799130200624774\n",
      "2 3 6\n",
      "1 5\n",
      "0\n",
      "4\n",
      "7\n",
      "Done.-15.65625\t42\t0\n",
      "-7.796875\t42\t8.6875\n",
      "-3.46875\t42\t15.71875\n",
      "-2.0625\t42\t17.34375\n",
      "0.703125\t42\t18.53125\n",
      "3.390625\t42\t22.875\n",
      "4.5625\t42\t23.90625\n",
      "6.328125\t42\t25.21875\n",
      "10.46875\t42\t27.4375\n",
      "14.84375\t42\t33.5\n",
      "19.09375\t42\t36.1875\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4033 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4033\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "367 hits go_terms_num = 181\n",
      "cat 114, mi = 0.00317517, POU5F1_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00281330, BRF1_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00240357, POLR3G_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00189123, PPARGC1A_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00168932, SMARCA4_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00131497, BDP1_ChIP-seq_regulon\n",
      "cat 13, mi = 0.00121974, BRF2_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "cat 7, mi = 0.00000000, BCL3_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.003, pvalue = 0.082, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.003, pvalue = 0.060, BRF1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.002, pvalue = 0.162, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.002, pvalue = 0.253, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.002, pvalue = 0.710, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.873, BDP1_ChIP-seq_regulon\n",
      "Didn't pass: cat 13, mi = 0.001, pvalue = 1.000, BRF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Didn't pass: cat 15, mi = 0.000, pvalue = 1.000, CBX3_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.80\t-6.64\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_encode_tf/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_encode_tf/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_encode_tf.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 298\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4106 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4106\t298\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "373 hits go_terms_num = 298\n",
      "cat 242, mi = 0.00524666, chr17q23\n",
      "cat 89, mi = 0.00469890, chr11q23\n",
      "cat 35, mi = 0.00449557, chr15q22\n",
      "cat 2, mi = 0.00428949, chr1p32\n",
      "cat 77, mi = 0.00427058, chr1p22\n",
      "cat 248, mi = 0.00397392, chr15q26\n",
      "cat 161, mi = 0.00382856, chr2q35\n",
      "cat 83, mi = 0.00365408, chr22q13\n",
      "cat 16, mi = 0.00357750, chr12p11\n",
      "cat 104, mi = 0.00349923, chr20q11\n",
      "cat 6, mi = 0.00337482, chr11q12\n",
      "cat 84, mi = 0.00332889, chr7q11\n",
      "cat 140, mi = 0.00332607, chrXp22\n",
      "cat 200, mi = 0.00331846, chr17q25\n",
      "cat 198, mi = 0.00312063, chr1q25\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 242, mi = 0.005, pvalue = 0.001, chr17q23\n",
      "Passed: cat 89, mi = 0.005, pvalue = 0.004, chr11q23\n",
      "Passed: cat 35, mi = 0.004, pvalue = 0.004, chr15q22\n",
      "Didn't pass (cutoff=0.0050): cat 2, mi = 0.004, pvalue = 0.007, chr1p32\n",
      "Didn't pass (cutoff=0.0050): cat 77, mi = 0.004, pvalue = 0.011, chr1p22\n",
      "Didn't pass (cutoff=0.0050): cat 248, mi = 0.004, pvalue = 0.019, chr15q26\n",
      "Didn't pass (cutoff=0.0050): cat 161, mi = 0.004, pvalue = 0.019, chr2q35\n",
      "Didn't pass (cutoff=0.0050): cat 83, mi = 0.004, pvalue = 0.034, chr22q13\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.004, pvalue = 0.035, chr12p11\n",
      "Didn't pass (cutoff=0.0050): cat 104, mi = 0.003, pvalue = 0.054, chr20q11\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.003, pvalue = 0.060, chr11q12\n",
      "Didn't pass (cutoff=0.0050): cat 84, mi = 0.003, pvalue = 0.076, chr7q11\n",
      "Didn't pass (cutoff=0.0050): cat 140, mi = 0.003, pvalue = 0.069, chrXp22\n",
      "Didn't pass (cutoff=0.0050): cat 200, mi = 0.003, pvalue = 0.057, chr17q25\n",
      "Didn't pass (cutoff=0.0050): cat 198, mi = 0.003, pvalue = 0.107, chr1q25\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.003, pvalue = 0.100, chr1q42\n",
      "Didn't pass (cutoff=0.0050): cat 43, mi = 0.003, pvalue = 0.136, chr6p22\n",
      "Didn't pass (cutoff=0.0050): cat 213, mi = 0.003, pvalue = 0.094, chr6q24\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.003, pvalue = 0.153, chr3q25\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.003, pvalue = 0.151, chr8p21\n",
      "Didn't pass (cutoff=0.0050): cat 206, mi = 0.003, pvalue = 0.169, chr1q22\n",
      "Didn't pass (cutoff=0.0050): cat 100, mi = 0.003, pvalue = 0.159, chr6q21\n",
      "Didn't pass (cutoff=0.0050): cat 249, mi = 0.003, pvalue = 0.114, chr10p15\n",
      "Total number of categories evaluated = 13.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.497152509489558\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "6.80\t-6.64\n",
      "chr17q23\n",
      "chr15q22\n",
      "chr11q23\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 1, 0.497152509489558\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-15.40625\t42\t0\n",
      "-7.5\t42\t9.1875\n",
      "-3.390625\t42\t15.8125\n",
      "-2.046875\t42\t17.40625\n",
      "0.640625\t42\t18.5\n",
      "3.28125\t42\t22.78125\n",
      "4.4375\t42\t23.78125\n",
      "6.140625\t42\t25.09375\n",
      "10.234375\t42\t27.1875\n",
      "14.6875\t42\t33.28125\n",
      "19.046875\t42\t36.09375\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c1/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c1/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c1.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5529\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4050 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4050\t5529\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "368 hits go_terms_num = 5529\n",
      "cat 3353, mi = 0.00560006, REACTOME_PTEN_REGULATION\n",
      "cat 1771, mi = 0.00529152, KEGG_N_GLYCAN_BIOSYNTHESIS\n",
      "cat 3662, mi = 0.00523873, HOSHIDA_LIVER_CANCER_SUBCLASS_S2\n",
      "cat 1648, mi = 0.00519871, REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION\n",
      "cat 3338, mi = 0.00519711, NADLER_HYPERGLYCEMIA_AT_OBESITY\n",
      "cat 5348, mi = 0.00495390, REACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING\n",
      "cat 791, mi = 0.00494332, REACTOME_DNA_DOUBLE_STRAND_BREAK_RESPONSE\n",
      "cat 2522, mi = 0.00490727, KEGG_AXON_GUIDANCE\n",
      "cat 4691, mi = 0.00483204, RHEIN_ALL_GLUCOCORTICOID_THERAPY_UP\n",
      "cat 4184, mi = 0.00482051, PICCALUGA_ANGIOIMMUNOBLASTIC_LYMPHOMA_DN\n",
      "cat 1487, mi = 0.00478670, REACTOME_REGULATION_OF_SIGNALING_BY_CBL\n",
      "cat 3964, mi = 0.00475417, RIZ_ERYTHROID_DIFFERENTIATION_HEMGN\n",
      "cat 1012, mi = 0.00474852, REACTOME_TRANSCRIPTIONAL_ACTIVITY_OF_SMAD2_SMAD3_SMAD4_HETEROTRIMER\n",
      "cat 3981, mi = 0.00470352, REACTOME_GOLGI_TO_ER_RETROGRADE_TRANSPORT\n",
      "cat 42, mi = 0.00470231, REACTOME_FORMATION_OF_TC_NER_PRE_INCISION_COMPLEX\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 3353, mi = 0.006, pvalue = 0.000, REACTOME_PTEN_REGULATION\n",
      "Passed: cat 1771, mi = 0.005, pvalue = 0.001, KEGG_N_GLYCAN_BIOSYNTHESIS\n",
      "Passed: cat 3662, mi = 0.005, pvalue = 0.002, HOSHIDA_LIVER_CANCER_SUBCLASS_S2\n",
      "Passed: cat 1648, mi = 0.005, pvalue = 0.001, REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION\n",
      "Passed: cat 3338, mi = 0.005, pvalue = 0.001, NADLER_HYPERGLYCEMIA_AT_OBESITY\n",
      "Passed: cat 2522, mi = 0.005, pvalue = 0.003, KEGG_AXON_GUIDANCE\n",
      "Passed: cat 4691, mi = 0.005, pvalue = 0.004, RHEIN_ALL_GLUCOCORTICOID_THERAPY_UP\n",
      "Passed: cat 4184, mi = 0.005, pvalue = 0.005, PICCALUGA_ANGIOIMMUNOBLASTIC_LYMPHOMA_DN\n",
      "Passed: cat 3964, mi = 0.005, pvalue = 0.000, RIZ_ERYTHROID_DIFFERENTIATION_HEMGN\n",
      "Passed: cat 5351, mi = 0.005, pvalue = 0.004, THUM_SYSTOLIC_HEART_FAILURE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5455, mi = 0.005, pvalue = 0.007, ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION\n",
      "Passed: cat 1961, mi = 0.005, pvalue = 0.001, CERVERA_SDHB_TARGETS_2\n",
      "Didn't pass (cutoff=0.0050): cat 4534, mi = 0.004, pvalue = 0.006, BURTON_ADIPOGENESIS_6\n",
      "Didn't pass (cutoff=0.0050): cat 3063, mi = 0.004, pvalue = 0.010, MIKKELSEN_IPS_LCP_WITH_H3K4ME3\n",
      "Didn't pass (cutoff=0.0050): cat 2968, mi = 0.004, pvalue = 0.010, REACTOME_PROCESSING_OF_INTRONLESS_PRE_MRNAS\n",
      "Didn't pass (cutoff=0.0050): cat 2009, mi = 0.004, pvalue = 0.006, SCIBETTA_KDM5B_TARGETS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2898, mi = 0.004, pvalue = 0.021, REACTOME_TRANSCRIPTIONAL_ACTIVATION_OF_MITOCHONDRIAL_BIOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 3137, mi = 0.004, pvalue = 0.024, ODONNELL_METASTASIS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 5235, mi = 0.004, pvalue = 0.017, ROSS_LEUKEMIA_WITH_MLL_FUSIONS\n",
      "Passed: cat 1376, mi = 0.004, pvalue = 0.003, CHESLER_BRAIN_HIGHEST_GENETIC_VARIANCE\n",
      "Passed: cat 4049, mi = 0.004, pvalue = 0.004, BURTON_ADIPOGENESIS_4\n",
      "Didn't pass (cutoff=0.0050): cat 2306, mi = 0.004, pvalue = 0.006, LEIN_MEDULLA_MARKERS\n",
      "Didn't pass (cutoff=0.0050): cat 3601, mi = 0.004, pvalue = 0.013, LEE_EARLY_T_LYMPHOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4075, mi = 0.004, pvalue = 0.014, SCHEIDEREIT_IKK_TARGETS\n",
      "Didn't pass (cutoff=0.0050): cat 1167, mi = 0.004, pvalue = 0.024, SMIRNOV_RESPONSE_TO_IR_6HR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3660, mi = 0.004, pvalue = 0.028, CHEBOTAEV_GR_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4975, mi = 0.004, pvalue = 0.009, SAKAI_CHRONIC_HEPATITIS_VS_LIVER_CANCER_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5075, mi = 0.004, pvalue = 0.013, REACTOME_MEIOTIC_RECOMBINATION\n",
      "Didn't pass (cutoff=0.0050): cat 4879, mi = 0.004, pvalue = 0.031, RODWELL_AGING_KIDNEY_DN\n",
      "Didn't pass (cutoff=0.0050): cat 57, mi = 0.004, pvalue = 0.028, HOLLEMAN_PREDNISOLONE_RESISTANCE_ALL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1196, mi = 0.004, pvalue = 0.027, POMEROY_MEDULLOBLASTOMA_DESMOPLASIC_VS_CLASSIC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2449, mi = 0.004, pvalue = 0.017, MARCINIAK_ER_STRESS_RESPONSE_VIA_CHOP\n",
      "Didn't pass (cutoff=0.0050): cat 3996, mi = 0.004, pvalue = 0.020, GAVIN_FOXP3_TARGETS_CLUSTER_T7\n",
      "Didn't pass (cutoff=0.0050): cat 2106, mi = 0.004, pvalue = 0.038, PUIFFE_INVASION_INHIBITED_BY_ASCITES_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3030, mi = 0.004, pvalue = 0.037, DASU_IL6_SIGNALING_SCAR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 616, mi = 0.004, pvalue = 0.039, REACTOME_RNA_POLYMERASE_II_TRANSCRIBES_SNRNA_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 3990, mi = 0.004, pvalue = 0.043, BASSO_CD40_SIGNALING_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1394, mi = 0.004, pvalue = 0.019, BERTUCCI_INVASIVE_CARCINOMA_DUCTAL_VS_LOBULAR_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4887, mi = 0.004, pvalue = 0.020, ABRAMSON_INTERACT_WITH_AIRE\n",
      "Didn't pass (cutoff=0.0050): cat 1090, mi = 0.004, pvalue = 0.019, REACTOME_CASPASE_MEDIATED_CLEAVAGE_OF_CYTOSKELETAL_PROTEINS\n",
      "Didn't pass (cutoff=0.0050): cat 138, mi = 0.004, pvalue = 0.020, VANTVEER_BREAST_CANCER_BRCA1_DN\n",
      "Total number of categories evaluated = 121.\n",
      "Number of categories that passed the tests = 13\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 11, 12, 0.106351257923284\n",
      "best merge is 6, 8, 0.265653691094948\n",
      "best merge is 2, 9, 0.297902938336796\n",
      "best merge is 1, 10, 0.305638568808374\n",
      "best merge is 2, 3, 0.556725990398905\n",
      "best merge is 0, 6, 0.594107518115331\n",
      "best merge is 4, 7, 0.821899212817638\n",
      "best merge is 2, 11, 0.948480083593828\n",
      "best merge is 0, 4, 1.03529839669389\n",
      "order classes\n",
      "3 - 1.40002173742466\n",
      "2 - 0.993325348627216\n",
      "5 - 1.49938602915314\n",
      "2 - 2.67117244165732\n",
      "1 10\n",
      "0 6 8\n",
      "2 9 3 11 12\n",
      "4 7\n",
      "5\n",
      "Done.Start drawing\n",
      "6.80\t-6.64\n",
      "HOSHIDA_LIVER_CANCER_SUBCLASS_S2\n",
      "THUM_SYSTOLIC_HEART_FAILURE_DN\n",
      "REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION\n",
      "RHEIN_ALL_GLUCOCORTICOID_THERAPY_UP\n",
      "RIZ_ERYTHROID_DIFFERENTIATION_HEMGN\n",
      "NADLER_HYPERGLYCEMIA_AT_OBESITY\n",
      "REACTOME_PTEN_REGULATION\n",
      "CHESLER_BRAIN_HIGHEST_GENETIC_VARIANCE\n",
      "BURTON_ADIPOGENESIS_4\n",
      "PICCALUGA_ANGIOIMMUNOBLASTIC_LYMPHOMA_DN\n",
      "KEGG_AXON_GUIDANCE\n",
      "KEGG_N_GLYCAN_BIOSYNTHESIS\n",
      "CERVERA_SDHB_TARGETS_2\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 11, 12, 0.106351257923284\n",
      "best merge is 6, 8, 0.265653691094948\n",
      "best merge is 2, 9, 0.297902938336796\n",
      "best merge is 1, 10, 0.305638568808374\n",
      "best merge is 2, 3, 0.556725990398905\n",
      "best merge is 0, 6, 0.594107518115331\n",
      "best merge is 4, 7, 0.821899212817638\n",
      "best merge is 2, 11, 0.948480083593828\n",
      "best merge is 0, 4, 1.03529839669389\n",
      "order classes\n",
      "3 - 1.40002173742466\n",
      "2 - 0.993325348627216\n",
      "5 - 1.49938602915314\n",
      "2 - 2.67117244165732\n",
      "1 10\n",
      "0 6 8\n",
      "2 9 3 11 12\n",
      "4 7\n",
      "5\n",
      "Done.-15.390625\t42\t0\n",
      "-7.46875\t42\t9.21875\n",
      "-3.359375\t42\t15.84375\n",
      "-2.015625\t42\t17.4375\n",
      "0.65625\t42\t18.53125\n",
      "3.28125\t42\t22.78125\n",
      "4.421875\t42\t23.78125\n",
      "6.109375\t42\t25.0625\n",
      "10.1875\t42\t27.15625\n",
      "14.65625\t42\t33.21875\n",
      "19.046875\t42\t36.09375\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c2/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c2/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c2.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 3733\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4090 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4090\t3733\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "372 hits go_terms_num = 3733\n",
      "cat 3691, mi = 0.00607516, MIR6730_3P\n",
      "cat 2331, mi = 0.00575984, IRX3_TARGET_GENES\n",
      "cat 709, mi = 0.00559804, MIR1_5P\n",
      "cat 1182, mi = 0.00506968, MIR3064_3P\n",
      "cat 3376, mi = 0.00502933, MIR31_3P\n",
      "cat 189, mi = 0.00488184, MIR4782_3P\n",
      "cat 1329, mi = 0.00485660, MIR8062\n",
      "cat 3026, mi = 0.00479113, MIR4647_MIR4662B\n",
      "cat 1762, mi = 0.00477258, MIR4514\n",
      "cat 1424, mi = 0.00473141, PAX5_01\n",
      "cat 3399, mi = 0.00463836, MIR4436B_5P\n",
      "cat 337, mi = 0.00463340, MIR6769A_3P\n",
      "cat 2912, mi = 0.00462250, MIR6732_5P\n",
      "cat 731, mi = 0.00460148, MIR4666B\n",
      "cat 3170, mi = 0.00450660, ATAAGCT_MIR21\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 3691, mi = 0.006, pvalue = 0.000, MIR6730_3P\n",
      "Passed: cat 2331, mi = 0.006, pvalue = 0.000, IRX3_TARGET_GENES\n",
      "Passed: cat 709, mi = 0.006, pvalue = 0.001, MIR1_5P\n",
      "Passed: cat 3376, mi = 0.005, pvalue = 0.001, MIR31_3P\n",
      "Passed: cat 189, mi = 0.005, pvalue = 0.003, MIR4782_3P\n",
      "Passed: cat 1329, mi = 0.005, pvalue = 0.003, MIR8062\n",
      "Passed: cat 3026, mi = 0.005, pvalue = 0.003, MIR4647_MIR4662B\n",
      "Didn't pass (cutoff=0.0050): cat 1424, mi = 0.005, pvalue = 0.006, PAX5_01\n",
      "Didn't pass (cutoff=0.0050): cat 3399, mi = 0.005, pvalue = 0.006, MIR4436B_5P\n",
      "Passed: cat 337, mi = 0.005, pvalue = 0.003, MIR6769A_3P\n",
      "Didn't pass (cutoff=0.0050): cat 3335, mi = 0.004, pvalue = 0.010, MIR211_3P\n",
      "Passed: cat 1880, mi = 0.004, pvalue = 0.001, ZNF433_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 470, mi = 0.004, pvalue = 0.011, MIR4777_5P\n",
      "Didn't pass (cutoff=0.0050): cat 566, mi = 0.004, pvalue = 0.011, MIR138_1_3P\n",
      "Didn't pass (cutoff=0.0050): cat 1025, mi = 0.004, pvalue = 0.012, MIR3122\n",
      "Didn't pass (cutoff=0.0050): cat 2692, mi = 0.004, pvalue = 0.012, MIR12113\n",
      "Didn't pass (cutoff=0.0050): cat 1027, mi = 0.004, pvalue = 0.013, TAAWWATAG_RSRFC4_Q2\n",
      "Didn't pass (cutoff=0.0050): cat 808, mi = 0.004, pvalue = 0.011, MIR4539\n",
      "Didn't pass (cutoff=0.0050): cat 175, mi = 0.004, pvalue = 0.015, MIR2116_5P\n",
      "Didn't pass (cutoff=0.0050): cat 135, mi = 0.004, pvalue = 0.018, MIR671_5P\n",
      "Didn't pass (cutoff=0.0050): cat 613, mi = 0.004, pvalue = 0.016, MIR6868_5P\n",
      "Didn't pass (cutoff=0.0050): cat 2320, mi = 0.004, pvalue = 0.021, MIR935\n",
      "Didn't pass (cutoff=0.0050): cat 2310, mi = 0.004, pvalue = 0.019, MIR331_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1005, mi = 0.004, pvalue = 0.022, CCATCCA_MIR432\n",
      "Didn't pass (cutoff=0.0050): cat 730, mi = 0.004, pvalue = 0.017, MIR603\n",
      "Didn't pass (cutoff=0.0050): cat 1588, mi = 0.004, pvalue = 0.027, MIR496\n",
      "Didn't pass (cutoff=0.0050): cat 988, mi = 0.004, pvalue = 0.015, CTCTATG_MIR368\n",
      "Didn't pass (cutoff=0.0050): cat 3517, mi = 0.004, pvalue = 0.024, MIR129_1_3P_MIR129_2_3P\n",
      "Didn't pass (cutoff=0.0050): cat 1376, mi = 0.004, pvalue = 0.024, MIR342_3P\n",
      "Didn't pass (cutoff=0.0050): cat 2881, mi = 0.004, pvalue = 0.026, MIR6795_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1267, mi = 0.004, pvalue = 0.026, MIR936\n",
      "Didn't pass (cutoff=0.0050): cat 1918, mi = 0.004, pvalue = 0.016, MIR4740_5P\n",
      "Total number of categories evaluated = 42.\n",
      "Number of categories that passed the tests = 9\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.25835387560782\n",
      "best merge is 5, 7, 0.366340585212371\n",
      "best merge is 0, 2, 0.496935029268382\n",
      "best merge is 0, 5, 0.721798540461281\n",
      "best merge is 1, 3, 0.742903033144162\n",
      "order classes\n",
      "5 - 0.856707229745948\n",
      "0 2 4 5 7\n",
      "1\n",
      "3\n",
      "6\n",
      "8\n",
      "Done.Start drawing\n",
      "6.80\t-6.64\n",
      "ZNF433_TARGET_GENES\n",
      "MIR6730_3P\n",
      "MIR1_5P\n",
      "MIR4782_3P\n",
      "IRX3_TARGET_GENES\n",
      "MIR8062\n",
      "MIR6769A_3P\n",
      "MIR31_3P\n",
      "MIR4647_MIR4662B\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 2, 4, 0.25835387560782\n",
      "best merge is 5, 7, 0.366340585212371\n",
      "best merge is 0, 2, 0.496935029268382\n",
      "best merge is 0, 5, 0.721798540461281\n",
      "best merge is 1, 3, 0.742903033144162\n",
      "order classes\n",
      "5 - 0.856707229745948\n",
      "0 2 4 5 7\n",
      "1\n",
      "3\n",
      "6\n",
      "8\n",
      "Done.-15.421875\t42\t0\n",
      "-7.515625\t42\t9.15625\n",
      "-3.390625\t42\t15.8125\n",
      "-2.046875\t42\t17.40625\n",
      "0.640625\t42\t18.5\n",
      "3.28125\t42\t22.78125\n",
      "4.4375\t42\t23.78125\n",
      "6.140625\t42\t25.09375\n",
      "10.234375\t42\t27.1875\n",
      "14.71875\t42\t33.28125\n",
      "19.078125\t42\t36.15625\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c3/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c3/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c3.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 2383 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 2383\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "217 hits go_terms_num = 858\n",
      "cat 154, mi = 0.00753718, MORF_RAD23B\n",
      "cat 619, mi = 0.00739995, MORF_HAT1\n",
      "cat 563, mi = 0.00736032, MORF_EI24\n",
      "cat 415, mi = 0.00728964, MORF_TPT1\n",
      "cat 654, mi = 0.00714131, MODULE_113\n",
      "cat 817, mi = 0.00708677, MORF_PTEN\n",
      "cat 147, mi = 0.00707624, MORF_ESPL1\n",
      "cat 55, mi = 0.00697197, MODULE_358\n",
      "cat 449, mi = 0.00691502, MORF_LCAT\n",
      "cat 463, mi = 0.00691502, MORF_WNT1\n",
      "cat 814, mi = 0.00690459, MORF_RFC4\n",
      "cat 530, mi = 0.00664117, MODULE_486\n",
      "cat 526, mi = 0.00650279, MORF_CDC16\n",
      "cat 439, mi = 0.00646536, MODULE_197\n",
      "cat 721, mi = 0.00641454, MORF_NPM1\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 154, mi = 0.008, pvalue = 0.011, MORF_RAD23B\n",
      "Didn't pass (cutoff=0.0050): cat 619, mi = 0.007, pvalue = 0.011, MORF_HAT1\n",
      "Didn't pass (cutoff=0.0050): cat 563, mi = 0.007, pvalue = 0.011, MORF_EI24\n",
      "Passed: cat 415, mi = 0.007, pvalue = 0.001, MORF_TPT1\n",
      "Didn't pass (cutoff=0.0050): cat 654, mi = 0.007, pvalue = 0.008, MODULE_113\n",
      "Didn't pass (cutoff=0.0050): cat 817, mi = 0.007, pvalue = 0.019, MORF_PTEN\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.007, pvalue = 0.017, MORF_ESPL1\n",
      "Didn't pass (cutoff=0.0050): cat 55, mi = 0.007, pvalue = 0.018, MODULE_358\n",
      "Passed: cat 449, mi = 0.007, pvalue = 0.002, MORF_LCAT\n",
      "Didn't pass (cutoff=0.0050): cat 814, mi = 0.007, pvalue = 0.021, MORF_RFC4\n",
      "Didn't pass (cutoff=0.0050): cat 530, mi = 0.007, pvalue = 0.029, MODULE_486\n",
      "Didn't pass (cutoff=0.0050): cat 526, mi = 0.007, pvalue = 0.031, MORF_CDC16\n",
      "Didn't pass (cutoff=0.0050): cat 439, mi = 0.006, pvalue = 0.025, MODULE_197\n",
      "Didn't pass (cutoff=0.0050): cat 44, mi = 0.006, pvalue = 0.042, GCM_DENR\n",
      "Didn't pass (cutoff=0.0050): cat 509, mi = 0.006, pvalue = 0.043, MORF_EIF4E\n",
      "Didn't pass (cutoff=0.0050): cat 396, mi = 0.006, pvalue = 0.014, CAR_MYST2\n",
      "Didn't pass (cutoff=0.0050): cat 599, mi = 0.006, pvalue = 0.048, MORF_SNRP70\n",
      "Didn't pass (cutoff=0.0050): cat 812, mi = 0.006, pvalue = 0.010, MODULE_300\n",
      "Didn't pass (cutoff=0.0050): cat 57, mi = 0.006, pvalue = 0.063, MODULE_166\n",
      "Didn't pass (cutoff=0.0050): cat 245, mi = 0.006, pvalue = 0.058, GCM_PPM1D\n",
      "Didn't pass (cutoff=0.0050): cat 498, mi = 0.006, pvalue = 0.061, MORF_PDCD1\n",
      "Didn't pass (cutoff=0.0050): cat 618, mi = 0.006, pvalue = 0.062, GNF2_ITGB2\n",
      "Didn't pass (cutoff=0.0050): cat 141, mi = 0.006, pvalue = 0.072, MODULE_240\n",
      "Didn't pass (cutoff=0.0050): cat 640, mi = 0.006, pvalue = 0.070, MODULE_156\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.006, pvalue = 0.052, MODULE_533\n",
      "Didn't pass (cutoff=0.0050): cat 566, mi = 0.006, pvalue = 0.073, MODULE_175\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.006, pvalue = 0.048, MODULE_257\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.006, pvalue = 0.067, MORF_RRM1\n",
      "Didn't pass (cutoff=0.0050): cat 621, mi = 0.006, pvalue = 0.068, MORF_IFNA1\n",
      "Total number of categories evaluated = 23.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.79\t-6.44\n",
      "MORF_TPT1\n",
      "MORF_LCAT\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-15.015873015873\t42\t0\n",
      "-7.12698412698412\t42\t9.96825396825397\n",
      "-3.52380952380952\t42\t15.7777777777778\n",
      "-2.28571428571428\t42\t17.1746031746032\n",
      "0.38095238095238\t42\t18.2539682539683\n",
      "3\t42\t22.5079365079365\n",
      "4.12698412698413\t42\t23.4920634920635\n",
      "5.77777777777778\t42\t24.7619047619048\n",
      "9.96825396825397\t42\t26.7936507936508\n",
      "14.6349206349206\t42\t33.1428571428571\n",
      "19.0634920634921\t42\t36.1269841269841\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c4/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c4/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c4.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 10191\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 3617 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 3617\t10191\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "329 hits go_terms_num = 10191\n",
      "cat 9728, mi = 0.00768433, GO_NUCLEAR_PORE\n",
      "cat 10180, mi = 0.00742000, GO_NLS_BEARING_PROTEIN_IMPORT_INTO_NUCLEUS\n",
      "cat 10036, mi = 0.00669425, GO_ALCOHOL_BIOSYNTHETIC_PROCESS\n",
      "cat 2193, mi = 0.00663410, GO_POSITIVE_REGULATION_OF_SMALL_GTPASE_MEDIATED_SIGNAL_TRANSDUCTION\n",
      "cat 7339, mi = 0.00651423, GO_COSTAMERE\n",
      "cat 151, mi = 0.00631442, GO_CYTOSOLIC_TRANSPORT\n",
      "cat 4467, mi = 0.00616274, GO_HISTONE_METHYLTRANSFERASE_ACTIVITY_H3_K4_SPECIFIC\n",
      "cat 8564, mi = 0.00615483, GO_RAN_GTPASE_BINDING\n",
      "cat 9499, mi = 0.00604328, GO_REGULATION_OF_SUBSTRATE_ADHESION_DEPENDENT_CELL_SPREADING\n",
      "cat 6636, mi = 0.00600354, GO_PCG_PROTEIN_COMPLEX\n",
      "cat 5210, mi = 0.00584439, GO_SUBSTRATE_ADHESION_DEPENDENT_CELL_SPREADING\n",
      "cat 7008, mi = 0.00579784, GO_TRANSLATION_FACTOR_ACTIVITY_RNA_BINDING\n",
      "cat 2401, mi = 0.00548933, GO_CYSTEINE_TYPE_PEPTIDASE_ACTIVITY\n",
      "cat 8895, mi = 0.00546192, GO_CARDIAC_SEPTUM_DEVELOPMENT\n",
      "cat 3371, mi = 0.00541698, GO_MAMMARY_GLAND_EPITHELIAL_CELL_PROLIFERATION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 9728, mi = 0.008, pvalue = 0.000, GO_NUCLEAR_PORE\n",
      "Passed: cat 10036, mi = 0.007, pvalue = 0.000, GO_ALCOHOL_BIOSYNTHETIC_PROCESS\n",
      "Passed: cat 2193, mi = 0.007, pvalue = 0.001, GO_POSITIVE_REGULATION_OF_SMALL_GTPASE_MEDIATED_SIGNAL_TRANSDUCTION\n",
      "Passed: cat 7339, mi = 0.007, pvalue = 0.000, GO_COSTAMERE\n",
      "Passed: cat 151, mi = 0.006, pvalue = 0.001, GO_CYTOSOLIC_TRANSPORT\n",
      "Passed: cat 4467, mi = 0.006, pvalue = 0.001, GO_HISTONE_METHYLTRANSFERASE_ACTIVITY_H3_K4_SPECIFIC\n",
      "Passed: cat 6636, mi = 0.006, pvalue = 0.001, GO_PCG_PROTEIN_COMPLEX\n",
      "Passed: cat 7008, mi = 0.006, pvalue = 0.001, GO_TRANSLATION_FACTOR_ACTIVITY_RNA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 2401, mi = 0.005, pvalue = 0.005, GO_CYSTEINE_TYPE_PEPTIDASE_ACTIVITY\n",
      "Passed: cat 3371, mi = 0.005, pvalue = 0.002, GO_MAMMARY_GLAND_EPITHELIAL_CELL_PROLIFERATION\n",
      "Didn't pass (cutoff=0.0050): cat 4301, mi = 0.005, pvalue = 0.008, GO_CELLULAR_RESPONSE_TO_TOPOLOGICALLY_INCORRECT_PROTEIN\n",
      "Didn't pass (cutoff=0.0050): cat 9395, mi = 0.005, pvalue = 0.008, GO_ENDOPLASMIC_RETICULUM_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 6666, mi = 0.005, pvalue = 0.008, GO_RNA_POLYMERASE_II_TRANSCRIPTION_FACTOR_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 1117, mi = 0.005, pvalue = 0.008, GO_POSITIVE_REGULATION_OF_ORGANELLE_ASSEMBLY\n",
      "Didn't pass (cutoff=0.0050): cat 7054, mi = 0.005, pvalue = 0.008, GO_RNA_POLYMERASE_II_ACTIVATING_TRANSCRIPTION_FACTOR_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5678, mi = 0.005, pvalue = 0.010, GO_TERTIARY_GRANULE\n",
      "Didn't pass (cutoff=0.0050): cat 8661, mi = 0.005, pvalue = 0.012, GO_UBIQUITIN_LIKE_PROTEIN_SPECIFIC_PROTEASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1698, mi = 0.005, pvalue = 0.013, GO_RETINA_DEVELOPMENT_IN_CAMERA_TYPE_EYE\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.005, pvalue = 0.009, GO_CALCINEURIN_MEDIATED_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 3402, mi = 0.005, pvalue = 0.011, GO_TRANSCRIPTION_COUPLED_NUCLEOTIDE_EXCISION_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 9600, mi = 0.005, pvalue = 0.007, GO_MRNA_CLEAVAGE\n",
      "Didn't pass (cutoff=0.0050): cat 8933, mi = 0.005, pvalue = 0.006, GO_CELL_RECOGNITION\n",
      "Didn't pass (cutoff=0.0050): cat 4082, mi = 0.005, pvalue = 0.016, GO_INTERLEUKIN_1_MEDIATED_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 6191, mi = 0.005, pvalue = 0.014, GO_RUFFLE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 1619, mi = 0.005, pvalue = 0.021, GO_BASAL_TRANSCRIPTION_MACHINERY_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5217, mi = 0.005, pvalue = 0.011, GO_TERTIARY_GRANULE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 5581, mi = 0.005, pvalue = 0.021, GO_RNA_POLYMERASE_CORE_ENZYME_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 5863, mi = 0.005, pvalue = 0.010, GO_TRANSCRIPTION_BY_RNA_POLYMERASE_I\n",
      "Didn't pass (cutoff=0.0050): cat 9743, mi = 0.004, pvalue = 0.024, GO_LEADING_EDGE_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 4104, mi = 0.004, pvalue = 0.021, GO_REGULATION_OF_CALCINEURIN_MEDIATED_SIGNALING\n",
      "Total number of categories evaluated = 54.\n",
      "Number of categories that passed the tests = 9\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 3, 7, 0.201670679722278\n",
      "best merge is 6, 8, 0.311674923394497\n",
      "best merge is 1, 5, 0.45026235422721\n",
      "best merge is 2, 6, 0.623109923079836\n",
      "best merge is 0, 4, 0.760946344991259\n",
      "order classes\n",
      "2 - 1.01309029701122\n",
      "3 - 1.03859651303611\n",
      "2 - 0.453759029375126\n",
      "3 7\n",
      "1 5\n",
      "2 6 8\n",
      "0\n",
      "4\n",
      "Done.Start drawing\n",
      "6.79\t-6.64\n",
      "GO_CYTOSOLIC_TRANSPORT\n",
      "GO_COSTAMERE\n",
      "GO_TRANSLATION_FACTOR_ACTIVITY_RNA_BINDING\n",
      "GO_POSITIVE_REGULATION_OF_SMALL_GTPASE_MEDIATED_SIGNAL_TRANSDUCTION\n",
      "GO_PCG_PROTEIN_COMPLEX\n",
      "GO_MAMMARY_GLAND_EPITHELIAL_CELL_PROLIFERATION\n",
      "GO_ALCOHOL_BIOSYNTHETIC_PROCESS\n",
      "GO_HISTONE_METHYLTRANSFERASE_ACTIVITY_H3_K4_SPECIFIC\n",
      "GO_NUCLEAR_PORE\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 3, 7, 0.201670679722278\n",
      "best merge is 6, 8, 0.311674923394497\n",
      "best merge is 1, 5, 0.45026235422721\n",
      "best merge is 2, 6, 0.623109923079836\n",
      "best merge is 0, 4, 0.760946344991259\n",
      "order classes\n",
      "2 - 1.01309029701122\n",
      "3 - 1.03859651303611\n",
      "2 - 0.453759029375126\n",
      "3 7\n",
      "1 5\n",
      "2 6 8\n",
      "0\n",
      "4\n",
      "Done.-15.1995532390171\t42\t0\n",
      "-7.25614296351452\t42\t9.60089352196575\n",
      "-3.31571109456441\t42\t15.8868205510052\n",
      "-1.98659717051378\t42\t17.481757259866\n",
      "0.671630677587491\t42\t18.5450483991065\n",
      "3.29858525688757\t42\t22.7982129560685\n",
      "4.42442293373045\t42\t23.7989575577066\n",
      "6.0819061801936\t42\t25.0498883097543\n",
      "10.2099776619509\t42\t27.1139240506329\n",
      "14.7758749069248\t42\t33.3060312732688\n",
      "19.1228592702904\t42\t36.2457185405808\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c5/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c5/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c5.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 2321 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 2321\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "211 hits go_terms_num = 189\n",
      "cat 151, mi = 0.00903460, SNF5_DN.V1_UP\n",
      "cat 41, mi = 0.00890633, PDGF_UP.V1_UP\n",
      "cat 70, mi = 0.00832739, EGFR_UP.V1_DN\n",
      "cat 54, mi = 0.00795446, GCNP_SHH_UP_EARLY.V1_DN\n",
      "cat 94, mi = 0.00737989, AKT_UP.V1_DN\n",
      "cat 31, mi = 0.00730906, CRX_DN.V1_DN\n",
      "cat 121, mi = 0.00721793, PKCA_DN.V1_DN\n",
      "cat 147, mi = 0.00721510, AKT_UP.V1_UP\n",
      "cat 118, mi = 0.00720951, DCA_UP.V1_DN\n",
      "cat 159, mi = 0.00714324, ALK_DN.V1_DN\n",
      "cat 32, mi = 0.00683593, P53_DN.V1_UP\n",
      "cat 132, mi = 0.00648004, GCNP_SHH_UP_LATE.V1_UP\n",
      "cat 173, mi = 0.00637967, YAP1_DN\n",
      "cat 176, mi = 0.00636879, LTE2_UP.V1_DN\n",
      "cat 6, mi = 0.00636262, RB_DN.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 151, mi = 0.009, pvalue = 0.002, SNF5_DN.V1_UP\n",
      "Passed: cat 41, mi = 0.009, pvalue = 0.003, PDGF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 70, mi = 0.008, pvalue = 0.006, EGFR_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 54, mi = 0.008, pvalue = 0.009, GCNP_SHH_UP_EARLY.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 94, mi = 0.007, pvalue = 0.015, AKT_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.007, pvalue = 0.016, CRX_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 121, mi = 0.007, pvalue = 0.018, PKCA_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 147, mi = 0.007, pvalue = 0.015, AKT_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 118, mi = 0.007, pvalue = 0.015, DCA_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 159, mi = 0.007, pvalue = 0.009, ALK_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.007, pvalue = 0.029, P53_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.006, pvalue = 0.036, GCNP_SHH_UP_LATE.V1_UP\n",
      "Passed: cat 173, mi = 0.006, pvalue = 0.003, YAP1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.006, pvalue = 0.037, LTE2_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.006, pvalue = 0.048, RB_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 45, mi = 0.006, pvalue = 0.045, RELA_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 37, mi = 0.006, pvalue = 0.064, MTOR_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.006, pvalue = 0.053, MEL18_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.005, pvalue = 0.101, HOXA9_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.005, pvalue = 0.102, KRAS.AMP.LUNG_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1, mi = 0.005, pvalue = 0.140, NOTCH_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 60, mi = 0.005, pvalue = 0.120, JAK2_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 26, mi = 0.005, pvalue = 0.137, NRL_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.005, pvalue = 0.116, ESC_V6.5_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 129, mi = 0.005, pvalue = 0.163, PKCA_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.005, pvalue = 0.136, PRC2_EED_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.005, pvalue = 0.139, ATF2_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 117, mi = 0.005, pvalue = 0.139, RELA_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 119, mi = 0.005, pvalue = 0.163, KRAS.KIDNEY_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 111, mi = 0.005, pvalue = 0.141, ERBB2_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.005, pvalue = 0.213, CRX_NRL_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.005, pvalue = 0.208, IL2_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 155, mi = 0.005, pvalue = 0.248, PDGF_UP.V1_DN\n",
      "Total number of categories evaluated = 30.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.314710008688675\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "6.80\t-6.50\n",
      "SNF5_DN.V1_UP\n",
      "PDGF_UP.V1_UP\n",
      "YAP1_DN\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.314710008688675\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-14.9315789473684\t42\t0\n",
      "-7.08421052631579\t42\t10.1368421052632\n",
      "-3.53157894736842\t42\t15.6947368421053\n",
      "-2.20526315789474\t42\t17.2421052631579\n",
      "0.494736842105262\t42\t18.3473684210526\n",
      "3.14736842105263\t42\t22.6421052631579\n",
      "4.33157894736842\t42\t23.6526315789474\n",
      "6.03684210526315\t42\t25.0105263157895\n",
      "10.1105263157895\t42\t27.0631578947368\n",
      "14.6736842105263\t42\t33.1578947368421\n",
      "19.0947368421053\t42\t36.1894736842105\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c6/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c6/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c6.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4003 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4003\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "364 hits go_terms_num = 4872\n",
      "cat 3723, mi = 0.00681993, GSE29617_CTRL_VS_DAY7_TIV_FLU_VACCINE_PBMC_2008_UP\n",
      "cat 968, mi = 0.00643995, GSE20151_CTRL_VS_FUSOBACT_NUCLEATUM_NEUTROPHIL_DN\n",
      "cat 3558, mi = 0.00611566, GSE7460_CTRL_VS_TGFB_TREATED_ACT_CD8_TCELL_UP\n",
      "cat 1876, mi = 0.00606990, GSE29618_PRE_VS_DAY7_POST_TIV_FLU_VACCINE_MDC_UP\n",
      "cat 1797, mi = 0.00595055, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "cat 419, mi = 0.00578439, GSE21063_WT_VS_NFATC1_KO_BCELL_DN\n",
      "cat 3605, mi = 0.00561348, GSE360_HIGH_VS_LOW_DOSE_B_MALAYI_MAC_UP\n",
      "cat 2150, mi = 0.00555041, GSE1740_UNSTIM_VS_IFNA_STIMULATED_MCSF_DERIVED_MACROPHAGE_UP\n",
      "cat 3080, mi = 0.00551121, GSE45365_CD8A_DC_VS_CD11B_DC_DN\n",
      "cat 2415, mi = 0.00543825, GSE11961_FOLLICULAR_BCELL_VS_MARGINAL_ZONE_BCELL_UP\n",
      "cat 3491, mi = 0.00542066, GSE40274_CTRL_VS_FOXP3_AND_LEF1_TRANSDUCED_ACTIVATED_CD4_TCELL_DN\n",
      "cat 4137, mi = 0.00537769, GSE7852_THYMUS_VS_FAT_TREG_UP\n",
      "cat 2123, mi = 0.00537759, GSE36476_YOUNG_VS_OLD_DONOR_MEMORY_CD4_TCELL_16H_TSST_ACT_UP\n",
      "cat 3161, mi = 0.00537695, GSE21546_WT_VS_ELK1_KO_ANTI_CD3_STIM_DP_THYMOCYTES_UP\n",
      "cat 911, mi = 0.00536605, GSE11057_NAIVE_VS_MEMORY_CD4_TCELL_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 3723, mi = 0.007, pvalue = 0.000, GSE29617_CTRL_VS_DAY7_TIV_FLU_VACCINE_PBMC_2008_UP\n",
      "Passed: cat 968, mi = 0.006, pvalue = 0.000, GSE20151_CTRL_VS_FUSOBACT_NUCLEATUM_NEUTROPHIL_DN\n",
      "Passed: cat 3558, mi = 0.006, pvalue = 0.000, GSE7460_CTRL_VS_TGFB_TREATED_ACT_CD8_TCELL_UP\n",
      "Passed: cat 1797, mi = 0.006, pvalue = 0.000, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "Passed: cat 419, mi = 0.006, pvalue = 0.001, GSE21063_WT_VS_NFATC1_KO_BCELL_DN\n",
      "Passed: cat 3605, mi = 0.006, pvalue = 0.001, GSE360_HIGH_VS_LOW_DOSE_B_MALAYI_MAC_UP\n",
      "Passed: cat 2150, mi = 0.006, pvalue = 0.001, GSE1740_UNSTIM_VS_IFNA_STIMULATED_MCSF_DERIVED_MACROPHAGE_UP\n",
      "Passed: cat 2415, mi = 0.005, pvalue = 0.001, GSE11961_FOLLICULAR_BCELL_VS_MARGINAL_ZONE_BCELL_UP\n",
      "Passed: cat 3491, mi = 0.005, pvalue = 0.002, GSE40274_CTRL_VS_FOXP3_AND_LEF1_TRANSDUCED_ACTIVATED_CD4_TCELL_DN\n",
      "Passed: cat 4137, mi = 0.005, pvalue = 0.002, GSE7852_THYMUS_VS_FAT_TREG_UP\n",
      "Passed: cat 2123, mi = 0.005, pvalue = 0.002, GSE36476_YOUNG_VS_OLD_DONOR_MEMORY_CD4_TCELL_16H_TSST_ACT_UP\n",
      "Passed: cat 1233, mi = 0.005, pvalue = 0.002, GSE17721_POLYIC_VS_CPG_16H_BMDC_DN\n",
      "Passed: cat 4548, mi = 0.005, pvalue = 0.003, GSE17721_POLYIC_VS_CPG_8H_BMDC_UP\n",
      "Passed: cat 3422, mi = 0.005, pvalue = 0.003, GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_48H_UP\n",
      "Passed: cat 1657, mi = 0.005, pvalue = 0.003, GSE40277_GATA1_AND_SATB1_TRANSDUCED_VS_CTRL_CD4_TCELL_DN\n",
      "Passed: cat 2249, mi = 0.005, pvalue = 0.005, GSE15330_HSC_VS_LYMPHOID_PRIMED_MULTIPOTENT_PROGENITOR_IKAROS_KO_UP\n",
      "Passed: cat 2862, mi = 0.005, pvalue = 0.003, GSE7831_CPG_VS_INFLUENZA_STIM_PDC_4H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2617, mi = 0.005, pvalue = 0.006, GSE360_T_GONDII_VS_B_MALAYI_LOW_DOSE_DC_UP\n",
      "Passed: cat 1889, mi = 0.005, pvalue = 0.004, GSE11961_FOLLICULAR_BCELL_VS_MEMORY_BCELL_DAY40_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3344, mi = 0.005, pvalue = 0.006, GSE22432_CONVENTIONAL_CDC_VS_PLASMACYTOID_PDC_UP\n",
      "Passed: cat 3876, mi = 0.005, pvalue = 0.004, GSE17301_CTRL_VS_48H_IFNA2_STIM_CD8_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4086, mi = 0.005, pvalue = 0.005, GSE45739_UNSTIM_VS_ACD3_ACD28_STIM_NRAS_KO_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4363, mi = 0.005, pvalue = 0.007, GSE360_L_DONOVANI_VS_T_GONDII_MAC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 960, mi = 0.005, pvalue = 0.006, GSE5099_MONOCYTE_VS_ALTERNATIVE_M2_MACROPHAGE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4631, mi = 0.005, pvalue = 0.007, GSE5589_WT_VS_IL10_KO_LPS_STIM_MACROPHAGE_45MIN_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1305, mi = 0.005, pvalue = 0.006, GSE20727_CTRL_VS_DNFB_ALLERGEN_TREATED_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1070, mi = 0.005, pvalue = 0.006, GSE37301_HEMATOPOIETIC_STEM_CELL_VS_MULTIPOTENT_PROGENITOR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2127, mi = 0.005, pvalue = 0.007, GSE38696_LIGHT_ZONE_VS_DARK_ZONE_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1110, mi = 0.005, pvalue = 0.008, GSE17721_CTRL_VS_PAM3CSK4_1H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 660, mi = 0.005, pvalue = 0.006, GSE22589_HEALTHY_VS_HIV_AND_SIV_INFECTED_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 678, mi = 0.005, pvalue = 0.007, GSE17721_LPS_VS_PAM3CSK4_6H_BMDC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.005, pvalue = 0.007, GSE13547_CTRL_VS_ANTI_IGM_STIM_ZFX_KO_BCELL_12H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4519, mi = 0.005, pvalue = 0.008, GSE32034_LY6C_HIGH_VS_LOW_MONOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3286, mi = 0.005, pvalue = 0.008, GSE22886_NEUTROPHIL_VS_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4640, mi = 0.005, pvalue = 0.009, GSE15659_RESTING_TREG_VS_NONSUPPRESSIVE_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4095, mi = 0.004, pvalue = 0.009, GSE37533_UNTREATED_VS_PIOGLIZATONE_TREATED_CD4_TCELL_FOXP3_TRASDUCED_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2416, mi = 0.004, pvalue = 0.008, GSE19374_UNINF_VS_LISTERIA_INFECTED_MACROPHAGE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2879, mi = 0.004, pvalue = 0.009, GSE360_CTRL_VS_T_GONDII_MAC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4254, mi = 0.004, pvalue = 0.011, GSE3982_MAST_CELL_VS_EFF_MEMORY_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1245, mi = 0.004, pvalue = 0.012, GSE38697_LIGHT_ZONE_VS_DARK_ZONE_BCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4705, mi = 0.004, pvalue = 0.012, GSE26928_NAIVE_VS_CXCR5_POS_CD4_TCELL_UP\n",
      "Total number of categories evaluated = 72.\n",
      "Number of categories that passed the tests = 19\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 4, 13, 0.230775083461361\n",
      "best merge is 7, 14, 0.300037312896552\n",
      "best merge is 10, 15, 0.366459688719728\n",
      "best merge is 1, 5, 0.446446370623353\n",
      "best merge is 7, 16, 0.47082464836114\n",
      "best merge is 0, 18, 0.479985181892931\n",
      "best merge is 11, 12, 0.522374129321771\n",
      "best merge is 6, 17, 0.551880099656326\n",
      "best merge is 2, 3, 0.55281246598284\n",
      "best merge is 0, 8, 0.647552764371715\n",
      "best merge is 1, 9, 0.656192925999365\n",
      "best merge is 10, 11, 0.687605191943289\n",
      "best merge is 4, 7, 0.721366478234526\n",
      "best merge is 1, 6, 0.835131635264793\n",
      "best merge is 4, 10, 0.87705247605146\n",
      "order classes\n",
      "3 - 2.49827581496969\n",
      "5 - 2.22573664869562\n",
      "2 - 2.62585921341849\n",
      "5 - 1.76340081099615\n",
      "4 - 2.16080741032745\n",
      "4 13 7 14 16\n",
      "1 5 9 6 17\n",
      "2 3\n",
      "0 18 8\n",
      "10 15 11 12\n",
      "Done.Start drawing\n",
      "6.80\t-6.64\n",
      "GSE360_HIGH_VS_LOW_DOSE_B_MALAYI_MAC_UP\n",
      "GSE40274_CTRL_VS_FOXP3_AND_LEF1_TRANSDUCED_ACTIVATED_CD4_TCELL_DN\n",
      "GSE36476_YOUNG_VS_OLD_DONOR_MEMORY_CD4_TCELL_16H_TSST_ACT_UP\n",
      "GSE15330_HSC_VS_LYMPHOID_PRIMED_MULTIPOTENT_PROGENITOR_IKAROS_KO_UP\n",
      "GSE17721_POLYIC_VS_CPG_16H_BMDC_DN\n",
      "GSE17721_POLYIC_VS_CPG_8H_BMDC_UP\n",
      "GSE21063_WT_VS_NFATC1_KO_BCELL_DN\n",
      "GSE20151_CTRL_VS_FUSOBACT_NUCLEATUM_NEUTROPHIL_DN\n",
      "GSE17301_CTRL_VS_48H_IFNA2_STIM_CD8_TCELL_UP\n",
      "GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_48H_UP\n",
      "GSE40277_GATA1_AND_SATB1_TRANSDUCED_VS_CTRL_CD4_TCELL_DN\n",
      "GSE7831_CPG_VS_INFLUENZA_STIM_PDC_4H_DN\n",
      "GSE1740_UNSTIM_VS_IFNA_STIMULATED_MCSF_DERIVED_MACROPHAGE_UP\n",
      "GSE11961_FOLLICULAR_BCELL_VS_MEMORY_BCELL_DAY40_DN\n",
      "GSE7852_THYMUS_VS_FAT_TREG_UP\n",
      "GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "GSE29617_CTRL_VS_DAY7_TIV_FLU_VACCINE_PBMC_2008_UP\n",
      "GSE11961_FOLLICULAR_BCELL_VS_MARGINAL_ZONE_BCELL_UP\n",
      "GSE7460_CTRL_VS_TGFB_TREATED_ACT_CD8_TCELL_UP\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 4, 13, 0.230775083461361\n",
      "best merge is 7, 14, 0.300037312896552\n",
      "best merge is 10, 15, 0.366459688719728\n",
      "best merge is 1, 5, 0.446446370623353\n",
      "best merge is 7, 16, 0.47082464836114\n",
      "best merge is 0, 18, 0.479985181892931\n",
      "best merge is 11, 12, 0.522374129321771\n",
      "best merge is 6, 17, 0.551880099656326\n",
      "best merge is 2, 3, 0.55281246598284\n",
      "best merge is 0, 8, 0.647552764371715\n",
      "best merge is 1, 9, 0.656192925999365\n",
      "best merge is 10, 11, 0.687605191943289\n",
      "best merge is 4, 7, 0.721366478234526\n",
      "best merge is 1, 6, 0.835131635264793\n",
      "best merge is 4, 10, 0.87705247605146\n",
      "order classes\n",
      "3 - 2.49827581496969\n",
      "5 - 2.22573664869562\n",
      "2 - 2.62585921341849\n",
      "5 - 1.76340081099615\n",
      "4 - 2.16080741032745\n",
      "4 13 7 14 16\n",
      "1 5 9 6 17\n",
      "2 3\n",
      "0 18 8\n",
      "10 15 11 12\n",
      "Done.-15.296875\t42\t0\n",
      "-7.359375\t42\t9.40625\n",
      "-3.328125\t42\t15.875\n",
      "-1.984375\t42\t17.46875\n",
      "0.6875\t42\t18.5625\n",
      "3.296875\t42\t22.8125\n",
      "4.421875\t42\t23.78125\n",
      "6.109375\t42\t25.0625\n",
      "10.21875\t42\t27.15625\n",
      "14.71875\t42\t33.28125\n",
      "19.078125\t42\t36.15625\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c7/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c7/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c7.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4159 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4159\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "378 hits go_terms_num = 17786\n",
      "cat 9427, mi = 0.00650968, GSE21063_WT_VS_NFATC1_KO_BCELL_DN\n",
      "cat 12416, mi = 0.00614788, GSE7400_CTRL_VS_CSF3_IN_VIVO_TREATED_PBMC_UP\n",
      "cat 10568, mi = 0.00589816, GSE29617_CTRL_VS_DAY7_TIV_FLU_VACCINE_PBMC_2008_UP\n",
      "cat 10624, mi = 0.00584002, GSE29618_PRE_VS_DAY7_POST_TIV_FLU_VACCINE_MDC_UP\n",
      "cat 6784, mi = 0.00574226, GO_REGULATION_OF_SUBSTRATE_ADHESION_DEPENDENT_CELL_SPREADING\n",
      "cat 11024, mi = 0.00565483, GSE360_HIGH_VS_LOW_DOSE_B_MALAYI_MAC_UP\n",
      "cat 14591, mi = 0.00564333, MORF_HAT1\n",
      "cat 11070, mi = 0.00561302, GSE360_T_GONDII_VS_B_MALAYI_LOW_DOSE_DC_UP\n",
      "cat 12051, mi = 0.00550729, GSE45739_UNSTIM_VS_ACD3_ACD28_STIM_NRAS_KO_CD4_TCELL_DN\n",
      "cat 3539, mi = 0.00546707, GO_HISTONE_METHYLTRANSFERASE_ACTIVITY_H3_K4_SPECIFIC_\n",
      "cat 325, mi = 0.00544842, BERTUCCI_MEDULLARY_VS_DUCTAL_BREAST_CANCER_DN\n",
      "cat 13242, mi = 0.00544056, KEGG_AXON_GUIDANCE\n",
      "cat 2768, mi = 0.00538308, GO_COSTAMERE\n",
      "cat 12014, mi = 0.00537926, GSE45365_HEALTHY_VS_MCMV_INFECTION_CD8A_DC_UP\n",
      "cat 11102, mi = 0.00534778, GSE36476_YOUNG_VS_OLD_DONOR_MEMORY_CD4_TCELL_16H_TSST_ACT_UP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 9427, mi = 0.007, pvalue = 0.000, GSE21063_WT_VS_NFATC1_KO_BCELL_DN\n",
      "Passed: cat 12416, mi = 0.006, pvalue = 0.000, GSE7400_CTRL_VS_CSF3_IN_VIVO_TREATED_PBMC_UP\n",
      "Passed: cat 10568, mi = 0.006, pvalue = 0.000, GSE29617_CTRL_VS_DAY7_TIV_FLU_VACCINE_PBMC_2008_UP\n",
      "Passed: cat 10624, mi = 0.006, pvalue = 0.000, GSE29618_PRE_VS_DAY7_POST_TIV_FLU_VACCINE_MDC_UP\n",
      "Passed: cat 6784, mi = 0.006, pvalue = 0.000, GO_REGULATION_OF_SUBSTRATE_ADHESION_DEPENDENT_CELL_SPREADING\n",
      "Passed: cat 11024, mi = 0.006, pvalue = 0.001, GSE360_HIGH_VS_LOW_DOSE_B_MALAYI_MAC_UP\n",
      "Passed: cat 14591, mi = 0.006, pvalue = 0.001, MORF_HAT1\n",
      "Passed: cat 11070, mi = 0.006, pvalue = 0.001, GSE360_T_GONDII_VS_B_MALAYI_LOW_DOSE_DC_UP\n",
      "Passed: cat 12051, mi = 0.006, pvalue = 0.001, GSE45739_UNSTIM_VS_ACD3_ACD28_STIM_NRAS_KO_CD4_TCELL_DN\n",
      "Passed: cat 325, mi = 0.005, pvalue = 0.001, BERTUCCI_MEDULLARY_VS_DUCTAL_BREAST_CANCER_DN\n",
      "Passed: cat 13242, mi = 0.005, pvalue = 0.001, KEGG_AXON_GUIDANCE\n",
      "Passed: cat 12014, mi = 0.005, pvalue = 0.001, GSE45365_HEALTHY_VS_MCMV_INFECTION_CD8A_DC_UP\n",
      "Passed: cat 11102, mi = 0.005, pvalue = 0.001, GSE36476_YOUNG_VS_OLD_DONOR_MEMORY_CD4_TCELL_16H_TSST_ACT_UP\n",
      "Passed: cat 7607, mi = 0.005, pvalue = 0.000, GO_TRANSLATION_FACTOR_ACTIVITY_RNA_BINDING\n",
      "Passed: cat 11663, mi = 0.005, pvalue = 0.002, GSE40277_GATA1_AND_SATB1_TRANSDUCED_VS_CTRL_CD4_TCELL_DN\n",
      "Passed: cat 12271, mi = 0.005, pvalue = 0.002, GSE5589_WT_VS_IL10_KO_LPS_STIM_MACROPHAGE_45MIN_DN\n",
      "Passed: cat 8694, mi = 0.005, pvalue = 0.003, GSE1740_UNSTIM_VS_IFNA_STIMULATED_MCSF_DERIVED_MACROPHAGE_UP\n",
      "Passed: cat 16211, mi = 0.005, pvalue = 0.003, RODRIGUES_THYROID_CARCINOMA_DN\n",
      "Passed: cat 8973, mi = 0.005, pvalue = 0.003, GSE17721_POLYIC_VS_CPG_16H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8447, mi = 0.005, pvalue = 0.005, GSE15330_HSC_VS_LYMPHOID_PRIMED_MULTIPOTENT_PROGENITOR_IKAROS_KO_UP\n",
      "Passed: cat 15545, mi = 0.005, pvalue = 0.004, REACTOME_ASPARAGINE_N_LINKED_GLYCOSYLATION\n",
      "Passed: cat 12157, mi = 0.005, pvalue = 0.004, GSE5099_MONOCYTE_VS_ALTERNATIVE_M2_MACROPHAGE_DN\n",
      "Passed: cat 11309, mi = 0.005, pvalue = 0.004, GSE37533_UNTREATED_VS_PIOGLIZATONE_TREATED_CD4_TCELL_FOXP3_TRASDUCED_CD4_TCELL_DN\n",
      "Passed: cat 9596, mi = 0.005, pvalue = 0.005, GSE22033_UNTREATED_VS_MRL24_TREATED_MEF_UP\n",
      "Passed: cat 10039, mi = 0.005, pvalue = 0.005, GSE25085_FETAL_LIVER_VS_FETAL_BM_SP4_THYMIC_IMPLANT_DN\n",
      "Passed: cat 8000, mi = 0.005, pvalue = 0.005, GSE11961_FOLLICULAR_BCELL_VS_MARGINAL_ZONE_BCELL_UP\n",
      "Passed: cat 9911, mi = 0.005, pvalue = 0.005, GSE2405_0H_VS_9H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9145, mi = 0.005, pvalue = 0.006, GSE18791_UNSTIM_VS_NEWCATSLE_VIRUS_DC_18H_DN\n",
      "Passed: cat 8063, mi = 0.005, pvalue = 0.004, GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "Passed: cat 702, mi = 0.005, pvalue = 0.001, BURTON_ADIPOGENESIS_4\n",
      "Didn't pass (cutoff=0.0050): cat 8446, mi = 0.005, pvalue = 0.005, GSE15330_HSC_VS_LYMPHOID_PRIMED_MULTIPOTENT_PROGENITOR_IKAROS_KO_DN\n",
      "Didn't pass (cutoff=0.0050): cat 11857, mi = 0.005, pvalue = 0.006, GSE42021_TREG_PLN_VS_CD24INT_TREG_THYMUS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 11208, mi = 0.005, pvalue = 0.007, GSE37301_HEMATOPOIETIC_STEM_CELL_VS_MULTIPOTENT_PROGENITOR_UP\n",
      "Didn't pass (cutoff=0.0050): cat 9557, mi = 0.004, pvalue = 0.007, GSE21927_SPLEEN_C57BL6_VS_EL4_TUMOR_BALBC_MONOCYTES_DN\n",
      "Didn't pass (cutoff=0.0050): cat 10320, mi = 0.004, pvalue = 0.007, GSE2770_TGFB_AND_IL4_ACT_VS_ACT_CD4_TCELL_48H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 8986, mi = 0.004, pvalue = 0.008, GSE17721_POLYIC_VS_CPG_8H_BMDC_UP\n",
      "Passed: cat 851, mi = 0.004, pvalue = 0.001, CERVERA_SDHB_TARGETS_2\n",
      "Didn't pass (cutoff=0.0050): cat 12505, mi = 0.004, pvalue = 0.007, GSE7831_CPG_VS_INFLUENZA_STIM_PDC_4H_DN\n",
      "Passed: cat 15821, mi = 0.004, pvalue = 0.002, REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION\n",
      "Didn't pass (cutoff=0.0050): cat 1937, mi = 0.004, pvalue = 0.008, GO_ACTIVATING_TRANSCRIPTION_FACTOR_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 10733, mi = 0.004, pvalue = 0.010, GSE32034_LY6C_HIGH_VS_LOW_MONOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4138, mi = 0.004, pvalue = 0.012, GO_MODIFICATION_OF_MORPHOLOGY_OR_PHYSIOLOGY_OF_OTHER_ORGANISM\n",
      "Didn't pass (cutoff=0.0050): cat 14750, mi = 0.004, pvalue = 0.011, MULLIGHAN_NPM1_SIGNATURE_3_DN\n",
      "Didn't pass (cutoff=0.0050): cat 5212, mi = 0.004, pvalue = 0.009, GO_POLYOL_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 10222, mi = 0.004, pvalue = 0.014, GSE26928_NAIVE_VS_CXCR5_POS_CD4_TCELL_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2682, mi = 0.004, pvalue = 0.012, GO_CILIUM_ORGANIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 16215, mi = 0.004, pvalue = 0.014, RODWELL_AGING_KIDNEY_DN\n",
      "Didn't pass (cutoff=0.0050): cat 7100, mi = 0.004, pvalue = 0.009, GO_RETROGRADE_TRANSPORT_ENDOSOME_TO_GOLGI\n",
      "Didn't pass (cutoff=0.0050): cat 8228, mi = 0.004, pvalue = 0.013, GSE13547_CTRL_VS_ANTI_IGM_STIM_ZFX_KO_BCELL_12H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 14390, mi = 0.004, pvalue = 0.015, MODULE_486\n",
      "Didn't pass (cutoff=0.0050): cat 17363, mi = 0.004, pvalue = 0.013, ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION\n",
      "Didn't pass (cutoff=0.0050): cat 11737, mi = 0.004, pvalue = 0.012, GSE411_UNSTIM_VS_100MIN_IL6_STIM_SOCS3_KO_MACROPHAGE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8751, mi = 0.004, pvalue = 0.014, GSE17721_12H_VS_24H_PAM3CSK4_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4436, mi = 0.004, pvalue = 0.013, GO_NEGATIVE_REGULATION_OF_GENE_EXPRESSION_EPIGENETIC\n",
      "Didn't pass (cutoff=0.0050): cat 3179, mi = 0.004, pvalue = 0.014, GO_ESTABLISHMENT_OF_PROTEIN_LOCALIZATION_TO_GOLGI\n",
      "Didn't pass (cutoff=0.0050): cat 1634, mi = 0.004, pvalue = 0.018, GCNP_SHH_UP_EARLY.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8831, mi = 0.004, pvalue = 0.017, GSE17721_CTRL_VS_PAM3CSK4_12H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 10885, mi = 0.004, pvalue = 0.016, GSE34156_NOD2_LIGAND_VS_NOD2_AND_TLR1_TLR2_LIGAND_24H_TREATED_MONOCYTE_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8834, mi = 0.004, pvalue = 0.015, GSE17721_CTRL_VS_PAM3CSK4_1H_BMDC_UP\n",
      "Total number of categories evaluated = 166.\n",
      "Number of categories that passed the tests = 30\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 9, 17, 0.131469520780578\n",
      "best merge is 16, 27, 0.261904946523984\n",
      "best merge is 0, 14, 0.305321814328228\n",
      "best merge is 11, 12, 0.309140995432078\n",
      "best merge is 4, 22, 0.318238929960432\n",
      "best merge is 8, 18, 0.349909926490274\n",
      "best merge is 19, 28, 0.359555918822623\n",
      "best merge is 0, 24, 0.381802514947563\n",
      "best merge is 1, 8, 0.411291066396296\n",
      "best merge is 20, 26, 0.436365102734992\n",
      "best merge is 21, 23, 0.458235478116273\n",
      "best merge is 11, 25, 0.475246181706444\n",
      "best merge is 10, 15, 0.487678442116481\n",
      "best merge is 7, 29, 0.513975609426575\n",
      "best merge is 3, 13, 0.547358849264646\n",
      "best merge is 20, 21, 0.557290897215874\n",
      "best merge is 5, 7, 0.623504117376965\n",
      "best merge is 0, 10, 0.635142583157657\n",
      "best merge is 1, 4, 0.66700299110447\n",
      "best merge is 2, 20, 0.701129480884576\n",
      "best merge is 1, 9, 0.717512185166043\n",
      "best merge is 6, 16, 0.76329874554849\n",
      "best merge is 1, 11, 0.790946154339452\n",
      "best merge is 2, 3, 0.84406458719048\n",
      "best merge is 0, 1, 0.911351049337057\n",
      "best merge is 2, 19, 0.937719727925928\n",
      "order classes\n",
      "15 - 1.45073522062975\n",
      "7 - 2.60926299960744\n",
      "3 - 3.91329743151223\n",
      "3 - 3.97444986137992\n",
      "2 - 2.69666939116967\n",
      "0 14 24 10 15 1 8 18 4 22 9 17 11 12 25\n",
      "2 20 26 21 23 3 13\n",
      "5 7 29\n",
      "6 16 27\n",
      "19 28\n",
      "Done.Start drawing\n",
      "6.80\t-6.64\n",
      "GSE36476_YOUNG_VS_OLD_DONOR_MEMORY_CD4_TCELL_16H_TSST_ACT_UP\n",
      "GSE360_T_GONDII_VS_B_MALAYI_LOW_DOSE_DC_UP\n",
      "REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION\n",
      "GSE45739_UNSTIM_VS_ACD3_ACD28_STIM_NRAS_KO_CD4_TCELL_DN\n",
      "GSE17721_POLYIC_VS_CPG_16H_BMDC_DN\n",
      "GSE21063_WT_VS_NFATC1_KO_BCELL_DN\n",
      "GSE29618_PRE_VS_DAY7_POST_TIV_FLU_VACCINE_MDC_UP\n",
      "GO_TRANSLATION_FACTOR_ACTIVITY_RNA_BINDING\n",
      "BERTUCCI_MEDULLARY_VS_DUCTAL_BREAST_CANCER_DN\n",
      "RODRIGUES_THYROID_CARCINOMA_DN\n",
      "GSE45365_HEALTHY_VS_MCMV_INFECTION_CD8A_DC_UP\n",
      "GSE2405_0H_VS_9H_A_PHAGOCYTOPHILUM_STIM_NEUTROPHIL_DN\n",
      "GSE40277_GATA1_AND_SATB1_TRANSDUCED_VS_CTRL_CD4_TCELL_DN\n",
      "GSE11961_FOLLICULAR_BCELL_VS_MARGINAL_ZONE_BCELL_UP\n",
      "GSE7400_CTRL_VS_CSF3_IN_VIVO_TREATED_PBMC_UP\n",
      "GO_REGULATION_OF_SUBSTRATE_ADHESION_DEPENDENT_CELL_SPREADING\n",
      "GSE22033_UNTREATED_VS_MRL24_TREATED_MEF_UP\n",
      "GSE5589_WT_VS_IL10_KO_LPS_STIM_MACROPHAGE_45MIN_DN\n",
      "MORF_HAT1\n",
      "GSE1740_UNSTIM_VS_IFNA_STIMULATED_MCSF_DERIVED_MACROPHAGE_UP\n",
      "BURTON_ADIPOGENESIS_4\n",
      "GSE29617_CTRL_VS_DAY7_TIV_FLU_VACCINE_PBMC_2008_UP\n",
      "GSE5099_MONOCYTE_VS_ALTERNATIVE_M2_MACROPHAGE_DN\n",
      "GSE12366_GC_VS_NAIVE_BCELL_DN\n",
      "GSE37533_UNTREATED_VS_PIOGLIZATONE_TREATED_CD4_TCELL_FOXP3_TRASDUCED_CD4_TCELL_DN\n",
      "GSE25085_FETAL_LIVER_VS_FETAL_BM_SP4_THYMIC_IMPLANT_DN\n",
      "GSE360_HIGH_VS_LOW_DOSE_B_MALAYI_MAC_UP\n",
      "KEGG_AXON_GUIDANCE\n",
      "REACTOME_ASPARAGINE_N_LINKED_GLYCOSYLATION\n",
      "CERVERA_SDHB_TARGETS_2\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 9, 17, 0.131469520780578\n",
      "best merge is 16, 27, 0.261904946523984\n",
      "best merge is 0, 14, 0.305321814328228\n",
      "best merge is 11, 12, 0.309140995432078\n",
      "best merge is 4, 22, 0.318238929960432\n",
      "best merge is 8, 18, 0.349909926490274\n",
      "best merge is 19, 28, 0.359555918822623\n",
      "best merge is 0, 24, 0.381802514947563\n",
      "best merge is 1, 8, 0.411291066396296\n",
      "best merge is 20, 26, 0.436365102734992\n",
      "best merge is 21, 23, 0.458235478116273\n",
      "best merge is 11, 25, 0.475246181706444\n",
      "best merge is 10, 15, 0.487678442116481\n",
      "best merge is 7, 29, 0.513975609426575\n",
      "best merge is 3, 13, 0.547358849264646\n",
      "best merge is 20, 21, 0.557290897215874\n",
      "best merge is 5, 7, 0.623504117376965\n",
      "best merge is 0, 10, 0.635142583157657\n",
      "best merge is 1, 4, 0.66700299110447\n",
      "best merge is 2, 20, 0.701129480884576\n",
      "best merge is 1, 9, 0.717512185166043\n",
      "best merge is 6, 16, 0.76329874554849\n",
      "best merge is 1, 11, 0.790946154339452\n",
      "best merge is 2, 3, 0.84406458719048\n",
      "best merge is 0, 1, 0.911351049337057\n",
      "best merge is 2, 19, 0.937719727925928\n",
      "order classes\n",
      "15 - 1.45073522062975\n",
      "7 - 2.60926299960744\n",
      "3 - 3.91329743151223\n",
      "3 - 3.97444986137992\n",
      "2 - 2.69666939116967\n",
      "0 14 24 10 15 1 8 18 4 22 9 17 11 12 25\n",
      "2 20 26 21 23 3 13\n",
      "5 7 29\n",
      "6 16 27\n",
      "19 28\n",
      "Done.-15.4375\t42\t0\n",
      "-7.546875\t42\t9.125\n",
      "-3.421875\t42\t15.78125\n",
      "-2.0625\t42\t17.375\n",
      "0.640625\t42\t18.5\n",
      "3.296875\t42\t22.78125\n",
      "4.46875\t42\t23.8125\n",
      "6.15625\t42\t25.125\n",
      "10.234375\t42\t27.1875\n",
      "14.71875\t42\t33.28125\n",
      "19.078125\t42\t36.15625\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_full/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_full/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_full.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 967 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 967\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "88 hits go_terms_num = 50\n",
      "cat 6, mi = 0.01412483, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "cat 28, mi = 0.01355679, HALLMARK_KRAS_SIGNALING_UP\n",
      "cat 24, mi = 0.01305564, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "cat 9, mi = 0.01263923, HALLMARK_SPERMATOGENESIS\n",
      "cat 11, mi = 0.01254958, HALLMARK_ADIPOGENESIS\n",
      "cat 5, mi = 0.01254160, HALLMARK_PROTEIN_SECRETION\n",
      "cat 12, mi = 0.01237537, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "cat 20, mi = 0.01204376, HALLMARK_MYC_TARGETS_V1\n",
      "cat 48, mi = 0.01141286, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "cat 39, mi = 0.01057097, HALLMARK_NOTCH_SIGNALING\n",
      "cat 36, mi = 0.01054688, HALLMARK_DNA_REPAIR\n",
      "cat 17, mi = 0.00992149, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 19, mi = 0.00991993, HALLMARK_APOPTOSIS\n",
      "cat 13, mi = 0.00990413, HALLMARK_APICAL_JUNCTION\n",
      "cat 3, mi = 0.00979681, HALLMARK_XENOBIOTIC_METABOLISM\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.014, pvalue = 0.068, HALLMARK_OXIDATIVE_PHOSPHORYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 28, mi = 0.014, pvalue = 0.086, HALLMARK_KRAS_SIGNALING_UP\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.013, pvalue = 0.116, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.013, pvalue = 0.100, HALLMARK_SPERMATOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.013, pvalue = 0.108, HALLMARK_ADIPOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 5, mi = 0.013, pvalue = 0.147, HALLMARK_PROTEIN_SECRETION\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.012, pvalue = 0.140, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.012, pvalue = 0.147, HALLMARK_MYC_TARGETS_V1\n",
      "Didn't pass (cutoff=0.0050): cat 48, mi = 0.011, pvalue = 0.155, HALLMARK_ESTROGEN_RESPONSE_EARLY\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.011, pvalue = 0.253, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.011, pvalue = 0.217, HALLMARK_DNA_REPAIR\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.010, pvalue = 0.232, HALLMARK_IL2_STAT5_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 19, mi = 0.010, pvalue = 0.277, HALLMARK_APOPTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.010, pvalue = 0.306, HALLMARK_APICAL_JUNCTION\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.010, pvalue = 0.357, HALLMARK_XENOBIOTIC_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.010, pvalue = 0.344, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.010, pvalue = 0.329, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Didn't pass (cutoff=0.0050): cat 46, mi = 0.010, pvalue = 0.380, HALLMARK_UV_RESPONSE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.009, pvalue = 0.310, HALLMARK_INTERFERON_GAMMA_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.009, pvalue = 0.310, HALLMARK_MTORC1_SIGNALING\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.43\t-6.64\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_h/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_h/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_h.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4632 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4632\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "421 hits go_terms_num = 790\n",
      "cat 236, mi = 0.00350782, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 237, mi = 0.00338301, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 32, mi = 0.00297331, BCCIP_HepG2_POSTAR_eCLIP\n",
      "cat 31, mi = 0.00268790, AUH_K562_POSTAR_eCLIP\n",
      "cat 275, mi = 0.00251644, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 195, mi = 0.00249171, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00249171, EZH2_HCT116_POSTAR_iCLIP\n",
      "cat 635, mi = 0.00245648, TARDBP_H9_POSTAR_CIMS\n",
      "cat 16, mi = 0.00232558, ALKBH1_HeLa_POSTAR_CIMS\n",
      "cat 553, mi = 0.00208244, SBDS_K562_POSTAR_eCLIP\n",
      "cat 272, mi = 0.00202426, HNRNPA1_HEK293T_POSTAR_CIMS\n",
      "cat 668, mi = 0.00202343, TNRC6A_HEK293_Plass\n",
      "cat 321, mi = 0.00194986, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 637, mi = 0.00193493, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 354, mi = 0.00193412, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.004, pvalue = 0.018, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.003, pvalue = 0.025, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.003, pvalue = 0.048, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.003, pvalue = 0.055, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.003, pvalue = 0.130, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.185, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.002, pvalue = 0.176, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.002, pvalue = 0.177, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.240, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 553, mi = 0.002, pvalue = 0.350, SBDS_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.002, pvalue = 0.351, HNRNPA1_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.251, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.355, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.002, pvalue = 0.367, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.002, pvalue = 0.333, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.002, pvalue = 0.346, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.002, pvalue = 0.406, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.002, pvalue = 0.320, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 300, mi = 0.002, pvalue = 0.419, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 270, mi = 0.002, pvalue = 0.373, HLTF_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.80\t-6.64\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_ids/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_ids/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_all_gene_ids.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_names/d_mtyl_T_vs_U.txt_PAGE'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4255 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4255\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "387 hits go_terms_num = 790\n",
      "cat 32, mi = 0.00367994, BCCIP_HepG2_POSTAR_eCLIP\n",
      "cat 377, mi = 0.00315250, LARP7_K562_POSTAR_eCLIP\n",
      "cat 637, mi = 0.00300830, TARDBP_H9_POSTAR_iCLIP\n",
      "cat 236, mi = 0.00284474, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "cat 775, mi = 0.00277958, YWHAG_K562_POSTAR_eCLIP\n",
      "cat 321, mi = 0.00277639, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 173, mi = 0.00277620, EIF4A3_HeLa_POSTAR_CIMS\n",
      "cat 237, mi = 0.00276431, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "cat 275, mi = 0.00273701, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 195, mi = 0.00271372, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "cat 196, mi = 0.00271372, EZH2_HCT116_POSTAR_iCLIP\n",
      "cat 635, mi = 0.00267444, TARDBP_H9_POSTAR_CIMS\n",
      "cat 469, mi = 0.00254436, PRKRA_HEK293_POSTAR_Piranha_0.01\n",
      "cat 30, mi = 0.00253111, AUH_HepG2_POSTAR_eCLIP\n",
      "cat 629, mi = 0.00247604, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.004, pvalue = 0.027, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.003, pvalue = 0.058, LARP7_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.003, pvalue = 0.106, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.003, pvalue = 0.145, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.003, pvalue = 0.101, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.003, pvalue = 0.152, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 173, mi = 0.003, pvalue = 0.103, EIF4A3_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.003, pvalue = 0.148, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.003, pvalue = 0.166, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.003, pvalue = 0.174, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.003, pvalue = 0.179, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.003, pvalue = 0.179, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 469, mi = 0.003, pvalue = 0.171, PRKRA_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.003, pvalue = 0.247, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.205, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.002, pvalue = 0.254, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.256, ALKBH1_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.282, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.002, pvalue = 0.182, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.002, pvalue = 0.183, TIA1_NA_POSTAR_iCLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.80\t-6.64\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4157 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4157\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "378 hits go_terms_num = 790\n",
      "cat 653, mi = 0.00423335, TIA1_HepG2_POSTAR_eCLIP\n",
      "cat 306, mi = 0.00406477, HNRNPK_HepG2_POSTAR_eCLIP\n",
      "cat 577, mi = 0.00405461, SMNDC1_K562_POSTAR_eCLIP\n",
      "cat 267, mi = 0.00396739, GTF2F1_K562_POSTAR_eCLIP\n",
      "cat 260, mi = 0.00378598, GRSF1_HepG2_POSTAR_eCLIP\n",
      "cat 327, mi = 0.00375225, HNRNPU_HeLa_POSTAR_HITS-CLIP\n",
      "cat 69, mi = 0.00373933, CPSF3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 242, mi = 0.00361543, FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 548, mi = 0.00353055, RTCB_HEK293_POSTAR_Piranha_0.01\n",
      "cat 273, mi = 0.00352008, HNRNPA1_HEK293T_POSTAR_HITS-CLIP\n",
      "cat 274, mi = 0.00352008, HNRNPA1_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 308, mi = 0.00342449, HNRNPK_K562_POSTAR_eCLIP\n",
      "cat 526, mi = 0.00335301, RBM22_HepG2_POSTAR_eCLIP\n",
      "cat 534, mi = 0.00334803, RBM5_HepG2_POSTAR_eCLIP\n",
      "cat 581, mi = 0.00332863, SND1_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 653, mi = 0.004, pvalue = 0.011, TIA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 306, mi = 0.004, pvalue = 0.017, HNRNPK_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 577, mi = 0.004, pvalue = 0.016, SMNDC1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 267, mi = 0.004, pvalue = 0.019, GTF2F1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 260, mi = 0.004, pvalue = 0.029, GRSF1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 327, mi = 0.004, pvalue = 0.030, HNRNPU_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.004, pvalue = 0.031, CPSF3_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 242, mi = 0.004, pvalue = 0.028, FXR1_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 548, mi = 0.004, pvalue = 0.042, RTCB_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 273, mi = 0.004, pvalue = 0.045, HNRNPA1_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 274, mi = 0.004, pvalue = 0.045, HNRNPA1_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 308, mi = 0.003, pvalue = 0.060, HNRNPK_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 526, mi = 0.003, pvalue = 0.064, RBM22_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 534, mi = 0.003, pvalue = 0.055, RBM5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 581, mi = 0.003, pvalue = 0.075, SND1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 424, mi = 0.003, pvalue = 0.059, NCBP2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 655, mi = 0.003, pvalue = 0.052, TIA1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 107, mi = 0.003, pvalue = 0.047, DDX3X_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 502, mi = 0.003, pvalue = 0.009, QKI_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 328, mi = 0.003, pvalue = 0.097, HNRNPU_HeLa_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.80\t-6.64\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_3UTR/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_3UTR/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_3UTR.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4089 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4089\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "372 hits go_terms_num = 790\n",
      "cat 14, mi = 0.00643637, AKAP8L_K562_POSTAR_eCLIP\n",
      "cat 150, mi = 0.00602713, DROSHA_K562_POSTAR_eCLIP\n",
      "cat 203, mi = 0.00471105, FASTKD2_K562_POSTAR_eCLIP\n",
      "cat 530, mi = 0.00465400, RBM27_K562_POSTAR_eCLIP\n",
      "cat 122, mi = 0.00425381, DDX6_HepG2_POSTAR_eCLIP\n",
      "cat 632, mi = 0.00408420, TARDBP_Brain_POSTAR_CIMS\n",
      "cat 634, mi = 0.00408420, TARDBP_Brain_POSTAR_iCLIP\n",
      "cat 86, mi = 0.00403340, CSTF2T_K562_POSTAR_eCLIP\n",
      "cat 120, mi = 0.00383516, DDX59_HepG2_POSTAR_eCLIP\n",
      "cat 653, mi = 0.00354907, TIA1_HepG2_POSTAR_eCLIP\n",
      "cat 606, mi = 0.00354514, STAU1_HEK293_POSTAR_iCLIP\n",
      "cat 508, mi = 0.00349824, RBFOX2_HepG2_POSTAR_eCLIP\n",
      "cat 655, mi = 0.00349083, TIA1_K562_POSTAR_eCLIP\n",
      "cat 373, mi = 0.00348729, LARP4_K562_POSTAR_eCLIP\n",
      "cat 158, mi = 0.00348443, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 14, mi = 0.006, pvalue = 0.000, AKAP8L_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 632, mi = 0.004, pvalue = 0.019, TARDBP_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 634, mi = 0.004, pvalue = 0.020, TARDBP_Brain_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 606, mi = 0.004, pvalue = 0.049, STAU1_HEK293_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 158, mi = 0.003, pvalue = 0.018, EIF3A_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 226, mi = 0.003, pvalue = 0.067, FUS_Brain_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 227, mi = 0.003, pvalue = 0.073, FUS_Brain_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 610, mi = 0.003, pvalue = 0.045, SUGP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 772, mi = 0.003, pvalue = 0.079, YTHDF3_Huh7_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 773, mi = 0.003, pvalue = 0.085, YTHDF3_Huh7_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 545, mi = 0.003, pvalue = 0.046, RPS5_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 246, mi = 0.003, pvalue = 0.100, FXR2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 247, mi = 0.003, pvalue = 0.104, FXR2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 186, mi = 0.003, pvalue = 0.080, EWSR1_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 187, mi = 0.003, pvalue = 0.080, EWSR1_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 757, mi = 0.003, pvalue = 0.062, YTHDF1_Huh7_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 758, mi = 0.003, pvalue = 0.058, YTHDF1_Huh7_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 510, mi = 0.003, pvalue = 0.139, RBFOX2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 233, mi = 0.003, pvalue = 0.131, FUS_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 604, mi = 0.003, pvalue = 0.116, STAU1_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 53, mi = 0.003, pvalue = 0.099, CNBP_HEK293_POSTAR_PAR-CLIP\n",
      "Total number of categories evaluated = 28.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.80\t-6.64\n",
      "AKAP8L_K562_POSTAR_eCLIP\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-15.421875\t42\t0\n",
      "-7.5\t42\t9.15625\n",
      "-3.359375\t42\t15.84375\n",
      "-2.015625\t42\t17.4375\n",
      "0.65625\t42\t18.53125\n",
      "3.296875\t42\t22.78125\n",
      "4.46875\t42\t23.8125\n",
      "6.1875\t42\t25.125\n",
      "10.296875\t42\t27.25\n",
      "14.765625\t42\t33.34375\n",
      "19.09375\t42\t36.1875\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_5UTR/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_5UTR/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_5UTR.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4166 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4166\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "379 hits go_terms_num = 790\n",
      "cat 583, mi = 0.00514112, SRRM4_HEK293T_POSTAR_CIMS\n",
      "cat 65, mi = 0.00511424, CPSF2_HEK293_POSTAR_PARalyzer\n",
      "cat 306, mi = 0.00491612, HNRNPK_HepG2_POSTAR_eCLIP\n",
      "cat 64, mi = 0.00483193, CPSF2_HEK293_POSTAR_PAR-CLIP\n",
      "cat 748, mi = 0.00424447, YTHDC2_HEK293T_POSTAR_CIMS\n",
      "cat 575, mi = 0.00384669, SMNDC1_HepG2_POSTAR_eCLIP\n",
      "cat 788, mi = 0.00377550, ZRANB2_K562_POSTAR_eCLIP\n",
      "cat 544, mi = 0.00362648, RPS5_HepG2_POSTAR_eCLIP\n",
      "cat 433, mi = 0.00355830, NOL12_HepG2_POSTAR_eCLIP\n",
      "cat 120, mi = 0.00355586, DDX59_HepG2_POSTAR_eCLIP\n",
      "cat 443, mi = 0.00352451, NOP58_HEK293_POSTAR_PAR-CLIP\n",
      "cat 444, mi = 0.00352451, NOP58_HEK293_POSTAR_PARalyzer\n",
      "cat 386, mi = 0.00348915, LIN28A_HEK293_POSTAR_Piranha_0.01\n",
      "cat 651, mi = 0.00341765, TIA1_HeLa_POSTAR_Piranha_0.01\n",
      "cat 625, mi = 0.00334058, TAF_HEK293_Plass\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 583, mi = 0.005, pvalue = 0.001, SRRM4_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 544, mi = 0.004, pvalue = 0.030, RPS5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 625, mi = 0.003, pvalue = 0.069, TAF_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 365, mi = 0.003, pvalue = 0.067, IMP3_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 221, mi = 0.003, pvalue = 0.082, FTO_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 260, mi = 0.003, pvalue = 0.092, GRSF1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 489, mi = 0.003, pvalue = 0.066, PTBP1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.003, pvalue = 0.093, EIF4G2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 294, mi = 0.003, pvalue = 0.081, HNRNPC_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 363, mi = 0.003, pvalue = 0.086, IMP1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 691, mi = 0.003, pvalue = 0.064, U2AF2_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.003, pvalue = 0.115, AKAP8L_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 641, mi = 0.003, pvalue = 0.100, TARDBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 179, mi = 0.003, pvalue = 0.078, ELAVL1_HEK293_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 203, mi = 0.003, pvalue = 0.137, FASTKD2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 234, mi = 0.003, pvalue = 0.151, FUS_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 364, mi = 0.003, pvalue = 0.107, IMP2_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 223, mi = 0.003, pvalue = 0.168, FUBP3_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 515, mi = 0.003, pvalue = 0.115, RBM10_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 130, mi = 0.003, pvalue = 0.202, DGCR8_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 395, mi = 0.003, pvalue = 0.240, LIN28_HEK293_Plass\n",
      "Total number of categories evaluated = 50.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.80\t-6.64\n",
      "SRRM4_HEK293T_POSTAR_CIMS\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-15.40625\t42\t0\n",
      "-7.484375\t42\t9.1875\n",
      "-3.359375\t42\t15.84375\n",
      "-2.015625\t42\t17.4375\n",
      "0.65625\t42\t18.53125\n",
      "3.28125\t42\t22.78125\n",
      "4.421875\t42\t23.78125\n",
      "6.09375\t42\t25.0625\n",
      "10.203125\t42\t27.125\n",
      "14.71875\t42\t33.28125\n",
      "19.078125\t42\t36.15625\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_coding_exons.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4087 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4087\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "372 hits go_terms_num = 790\n",
      "cat 39, mi = 0.00529508, C17orf85_HEK293_Plass\n",
      "cat 456, mi = 0.00447087, PAPD5_HEK293_Plass\n",
      "cat 32, mi = 0.00381821, BCCIP_HepG2_POSTAR_eCLIP\n",
      "cat 684, mi = 0.00331764, TROVE2_K562_POSTAR_eCLIP\n",
      "cat 329, mi = 0.00301970, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "cat 312, mi = 0.00301061, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "cat 775, mi = 0.00299094, YWHAG_K562_POSTAR_eCLIP\n",
      "cat 446, mi = 0.00287365, NOP58_HEK293_Plass\n",
      "cat 173, mi = 0.00286778, EIF4A3_HeLa_POSTAR_CIMS\n",
      "cat 275, mi = 0.00284638, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 749, mi = 0.00267144, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 686, mi = 0.00265182, TTP_HEK293_Plass\n",
      "cat 503, mi = 0.00264269, QKI_HEK293_Plass\n",
      "cat 553, mi = 0.00260380, SBDS_K562_POSTAR_eCLIP\n",
      "cat 469, mi = 0.00255572, PRKRA_HEK293_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 39, mi = 0.005, pvalue = 0.002, C17orf85_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 32, mi = 0.004, pvalue = 0.027, BCCIP_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 684, mi = 0.003, pvalue = 0.058, TROVE2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 329, mi = 0.003, pvalue = 0.087, HNRNPU_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.003, pvalue = 0.129, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.003, pvalue = 0.090, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 173, mi = 0.003, pvalue = 0.113, EIF4A3_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.003, pvalue = 0.163, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 553, mi = 0.003, pvalue = 0.274, SBDS_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 469, mi = 0.003, pvalue = 0.216, PRKRA_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 236, mi = 0.003, pvalue = 0.283, FUS_HeLa_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.290, EZH2_HCT116_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 196, mi = 0.002, pvalue = 0.294, EZH2_HCT116_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 237, mi = 0.002, pvalue = 0.271, FUS_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.002, pvalue = 0.282, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 30, mi = 0.002, pvalue = 0.334, AUH_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.002, pvalue = 0.287, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.333, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.002, pvalue = 0.220, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 230, mi = 0.002, pvalue = 0.318, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 377, mi = 0.002, pvalue = 0.304, LARP7_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 18.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.80\t-6.64\n",
      "C17orf85_HEK293_Plass\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-15.390625\t42\t0\n",
      "-7.46875\t42\t9.21875\n",
      "-3.359375\t42\t15.84375\n",
      "-1.984375\t42\t17.4375\n",
      "0.71875\t42\t18.59375\n",
      "3.34375\t42\t22.84375\n",
      "4.484375\t42\t23.84375\n",
      "6.1875\t42\t25.125\n",
      "10.28125\t42\t27.25\n",
      "14.765625\t42\t33.3125\n",
      "19.109375\t42\t36.21875\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_introns/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_introns/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_introns.pdf'\n",
      "Reading expression data ... 1 by 4696... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4087 out of 4696 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4087\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "372 hits go_terms_num = 101\n",
      "cat 69, mi = 0.00420135, RBMS3.1\n",
      "cat 88, mi = 0.00297098, SRSF9.1\n",
      "cat 56, mi = 0.00224242, RBM3\n",
      "cat 6, mi = 0.00216671, CPEB4\n",
      "cat 62, mi = 0.00214214, RBM46\n",
      "cat 9, mi = 0.00188106, ESRP2\n",
      "cat 73, mi = 0.00188106, SNRNP70\n",
      "cat 34, mi = 0.00186597, KHDRBS3\n",
      "cat 68, mi = 0.00177172, RBMS3\n",
      "cat 27, mi = 0.00160547, HuR.2\n",
      "cat 13, mi = 0.00120345, FXR2\n",
      "cat 21, mi = 0.00084645, HNRNPK\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "cat 1, mi = 0.00000000, BRUNOL4\n",
      "cat 2, mi = 0.00000000, BRUNOL5\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.004, pvalue = 0.011, RBMS3.1\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.003, pvalue = 0.105, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.002, pvalue = 0.366, RBM3\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.002, pvalue = 0.444, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.002, pvalue = 0.538, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.002, pvalue = 0.652, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.002, pvalue = 0.659, SNRNP70\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.002, pvalue = 0.260, KHDRBS3\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.002, pvalue = 0.498, RBMS3\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.002, pvalue = 0.686, HuR.2\n",
      "Didn't pass: cat 13, mi = 0.001, pvalue = 1.000, FXR2\n",
      "Didn't pass: cat 21, mi = 0.001, pvalue = 1.000, HNRNPK\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, DAZAP1\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, ENOX1\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "6.80\t-6.64\n",
      "Outputing EPS file d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.eps\n",
      "Convert to PDF d_mtyl_T_vs_U.txt_PAGE/d_mtyl_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt --expfile=d_mtyl_T_vs_U.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'radar/d_mtyl_T_vs_U.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_DeepBind/d_mtyl_T_vs_U.txt_PAGE'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_DeepBind/d_mtyl_T_vs_U.txt.summary.pdf' -> 'plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_DeepBind.pdf'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "command failed: /flash/bin/iPAGEv1.0/PROGRAMS/page -expfile d_mtyl_T_vs_U.txt -goindexfile /flash/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_all_gene_names/human_ensembl_RBPs_all_gene_names_index.txt -gonamesfile /flash/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_all_gene_names/human_ensembl_RBPs_all_gene_names_names.txt -catmaxcount 200 -catmincount 0 -logfile d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt.log  -quantized 0 -pvaluematrixfile d_mtyl_T_vs_U.txt_PAGE/pvmatrix.txt -max_p 0.005  -F 1  -P 1  -C 1  -independence 1  -ebins 11  at /flash/bin/iPAGEv1.0/page.pl line 300.\n",
      "cp: cannot stat 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_names/d_mtyl_T_vs_U.txt.summary.pdf': No such file or directory\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir -p ipage\n",
    "\n",
    "for f in `ls -d /flash/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl*`; do \n",
    "    base=`basename \"$f\"`; \n",
    "    cd radar\n",
    "    perl $PAGEDIR/page.pl \\\n",
    "        --expfile=d_mtyl_T_vs_U.txt \\\n",
    "        --species=$base \\\n",
    "        --exptype=continuous --ebins=11 --nodups=1; \n",
    "    cd ..\n",
    "    mv -v radar/d_mtyl_T_vs_U.txt_PAGE/ ipage/d_mtyl_T_vs_U_${base}/\n",
    "    cp -v ipage/d_mtyl_T_vs_U_${base}/d_mtyl_T_vs_U.txt.summary.pdf plots/d_mtyl_T_vs_U.ipage.${base}.pdf\n",
    "\n",
    "#     magick -density 300 plots/ipage.d_mtyl_T_vs_U_${base}.pdf -flatten -quality 90 plots/ipage.d_mtyl_T_vs_U_${base}.png\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "# for f in plots/*.pdf; do bash ~/Workflows/my_scripts/pdf2png.sh $f; done \n",
    "# plots/boxPlot.pdf > plots/boxPlot.png"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# iPAGE\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c1.png\" title=\"msigdb_c1\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c2.png\" title=\"msigdb_c2\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c3.png\" title=\"msigdb_c3\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c4.png\" title=\"msigdb_c4\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c5.png\" title=\"msigdb_c5\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c6.png\" title=\"msigdb_c6\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c7.png\" title=\"msigdb_c7\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_full.png\" title=\"msigdb_full\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl.png\" title=\"ensembl\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_5UTR.png\" title=\"RBPs_coding_gene_ids_by_5UTR\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_coding_exons.png\" title=\"RBPs_coding_gene_ids_by_coding_exons\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_introns.png\" title=\"RBPs_coding_gene_ids_by_introns\" style=\"width:600px\">\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BURTON_ADIPOGENESIS_4, BURTON_ADIPOGENESIS_4\r\n",
      "\tELVIDGE_HIF1A_AND_HIF2A_TARGETS_UP, ELVIDGE_HIF1A_AND_HIF2A_TARGETS_UP\r\n",
      "CERVERA_SDHB_TARGETS_2, CERVERA_SDHB_TARGETS_2\r\n",
      "\tAMBROSINI_FLAVOPIRIDOL_TREATMENT_TP53, AMBROSINI_FLAVOPIRIDOL_TREATMENT_TP53\r\n",
      "HOSHIDA_LIVER_CANCER_SUBCLASS_S2, HOSHIDA_LIVER_CANCER_SUBCLASS_S2\r\n",
      "\tKEGG_MELANOGENESIS, KEGG_MELANOGENESIS\r\n",
      "\tPID_SHP2_PATHWAY, PID_SHP2_PATHWAY\r\n",
      "\tREACTOME_NUCLEAR_PORE_COMPLEX_NPC_DISASSEMBLY, REACTOME_NUCLEAR_PORE_COMPLEX_NPC_DISASSEMBLY\r\n",
      "\tHEDENFALK_BREAST_CANCER_BRACX_DN, HEDENFALK_BREAST_CANCER_BRACX_DN\r\n",
      "\tREACTOME_INTERACTIONS_OF_VPR_WITH_HOST_CELLULAR_PROTEINS, REACTOME_INTERACTIONS_OF_VPR_WITH_HOST_CELLULAR_PROTEINS\r\n",
      "\tREACTOME_NS1_MEDIATED_EFFECTS_ON_HOST_PATHWAYS, REACTOME_NS1_MEDIATED_EFFECTS_ON_HOST_PATHWAYS\r\n",
      "\tKAAB_FAILED_HEART_ATRIUM_UP, KAAB_FAILED_HEART_ATRIUM_UP\r\n",
      "\tREACTOME_HCMV_LATE_EVENTS, REACTOME_HCMV_LATE_EVENTS\r\n",
      "\tPID_FAK_PATHWAY, PID_FAK_PATHWAY\r\n",
      "\tREACTOME_SEMAPHORIN_INTERACTIONS, REACTOME_SEMAPHORIN_INTERACTIONS\r\n",
      "\tREACTOME_INTERACTIONS_OF_REV_WITH_HOST_CELLULAR_PROTEINS, REACTOME_INTERACTIONS_OF_REV_WITH_HOST_CELLULAR_PROTEINS\r\n",
      "\tREACTOME_VIRAL_MESSENGER_RNA_SYNTHESIS, REACTOME_VIRAL_MESSENGER_RNA_SYNTHESIS\r\n",
      "\tREACTOME_REGULATION_OF_GLUCOKINASE_BY_GLUCOKINASE_REGULATORY_PROTEIN, REACTOME_REGULATION_OF_GLUCOKINASE_BY_GLUCOKINASE_REGULATORY_PROTEIN\r\n",
      "\tREACTOME_EXPORT_OF_VIRAL_RIBONUCLEOPROTEINS_FROM_NUCLEUS, REACTOME_EXPORT_OF_VIRAL_RIBONUCLEOPROTEINS_FROM_NUCLEUS\r\n",
      "\tREACTOME_NUCLEAR_IMPORT_OF_REV_PROTEIN, REACTOME_NUCLEAR_IMPORT_OF_REV_PROTEIN\r\n",
      "KEGG_AXON_GUIDANCE, KEGG_AXON_GUIDANCE\r\n",
      "\tREACTOME_APOPTOTIC_CLEAVAGE_OF_CELLULAR_PROTEINS, REACTOME_APOPTOTIC_CLEAVAGE_OF_CELLULAR_PROTEINS\r\n",
      "\tHAHTOLA_MYCOSIS_FUNGOIDES_CD4_DN, HAHTOLA_MYCOSIS_FUNGOIDES_CD4_DN\r\n",
      "\tREACTOME_APOPTOTIC_EXECUTION_PHASE, REACTOME_APOPTOTIC_EXECUTION_PHASE\r\n",
      "KEGG_N_GLYCAN_BIOSYNTHESIS, KEGG_N_GLYCAN_BIOSYNTHESIS\r\n",
      "\tGROSS_HYPOXIA_VIA_HIF1A_UP, GROSS_HYPOXIA_VIA_HIF1A_UP\r\n",
      "\tKOBAYASHI_EGFR_SIGNALING_24HR_UP, KOBAYASHI_EGFR_SIGNALING_24HR_UP\r\n",
      "NADLER_HYPERGLYCEMIA_AT_OBESITY, NADLER_HYPERGLYCEMIA_AT_OBESITY\r\n",
      "\tHOFFMANN_PRE_BI_TO_LARGE_PRE_BII_LYMPHOCYTE_DN, HOFFMANN_PRE_BI_TO_LARGE_PRE_BII_LYMPHOCYTE_DN\r\n",
      "PICCALUGA_ANGIOIMMUNOBLASTIC_LYMPHOMA_DN, PICCALUGA_ANGIOIMMUNOBLASTIC_LYMPHOMA_DN\r\n",
      "\tMATTIOLI_MGUS_VS_PCL, MATTIOLI_MGUS_VS_PCL\r\n",
      "\tLEE_DIFFERENTIATING_T_LYMPHOCYTE, LEE_DIFFERENTIATING_T_LYMPHOCYTE\r\n",
      "\tYAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_1, YAO_TEMPORAL_RESPONSE_TO_PROGESTERONE_CLUSTER_1\r\n",
      "\tADDYA_ERYTHROID_DIFFERENTIATION_BY_HEMIN, ADDYA_ERYTHROID_DIFFERENTIATION_BY_HEMIN\r\n",
      "\tKIM_WT1_TARGETS_8HR_UP, KIM_WT1_TARGETS_8HR_UP\r\n",
      "REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION, REACTOME_MHC_CLASS_II_ANTIGEN_PRESENTATION\r\n",
      "\tREACTOME_GOLGI_TO_ER_RETROGRADE_TRANSPORT, REACTOME_GOLGI_TO_ER_RETROGRADE_TRANSPORT\r\n",
      "\tWANG_RESPONSE_TO_FORSKOLIN_UP, WANG_RESPONSE_TO_FORSKOLIN_UP\r\n",
      "\tREACTOME_FACTORS_INVOLVED_IN_MEGAKARYOCYTE_DEVELOPMENT_AND_PLATELET_PRODUCTION, REACTOME_FACTORS_INVOLVED_IN_MEGAKARYOCYTE_DEVELOPMENT_AND_PLATELET_PRODUCTION\r\n",
      "\tREACTOME_COPI_INDEPENDENT_GOLGI_TO_ER_RETROGRADE_TRAFFIC, REACTOME_COPI_INDEPENDENT_GOLGI_TO_ER_RETROGRADE_TRAFFIC\r\n",
      "\tWANG_RESPONSE_TO_ANDROGEN_UP, WANG_RESPONSE_TO_ANDROGEN_UP\r\n",
      "REACTOME_PTEN_REGULATION, REACTOME_PTEN_REGULATION\r\n",
      "\tREACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING, REACTOME_FC_EPSILON_RECEPTOR_FCERI_SIGNALING\r\n",
      "\tREACTOME_DNA_DOUBLE_STRAND_BREAK_RESPONSE, REACTOME_DNA_DOUBLE_STRAND_BREAK_RESPONSE\r\n",
      "\tREACTOME_REGULATION_OF_SIGNALING_BY_CBL, REACTOME_REGULATION_OF_SIGNALING_BY_CBL\r\n",
      "\tREACTOME_TRANSCRIPTIONAL_ACTIVITY_OF_SMAD2_SMAD3_SMAD4_HETEROTRIMER, REACTOME_TRANSCRIPTIONAL_ACTIVITY_OF_SMAD2_SMAD3_SMAD4_HETEROTRIMER\r\n",
      "\tREACTOME_FORMATION_OF_TC_NER_PRE_INCISION_COMPLEX, REACTOME_FORMATION_OF_TC_NER_PRE_INCISION_COMPLEX\r\n",
      "\tREACTOME_ABC_FAMILY_PROTEINS_MEDIATED_TRANSPORT, REACTOME_ABC_FAMILY_PROTEINS_MEDIATED_TRANSPORT\r\n",
      "\tREACTOME_REGULATION_OF_INNATE_IMMUNE_RESPONSES_TO_CYTOSOLIC_DNA, REACTOME_REGULATION_OF_INNATE_IMMUNE_RESPONSES_TO_CYTOSOLIC_DNA\r\n",
      "\tREACTOME_SIGNALING_BY_MET, REACTOME_SIGNALING_BY_MET\r\n",
      "\tREACTOME_SMAD2_SMAD3_SMAD4_HETEROTRIMER_REGULATES_TRANSCRIPTION, REACTOME_SMAD2_SMAD3_SMAD4_HETEROTRIMER_REGULATES_TRANSCRIPTION\r\n",
      "\tREACTOME_OXIDATIVE_STRESS_INDUCED_SENESCENCE, REACTOME_OXIDATIVE_STRESS_INDUCED_SENESCENCE\r\n",
      "\tREACTOME_CLEC7A_DECTIN_1_SIGNALING, REACTOME_CLEC7A_DECTIN_1_SIGNALING\r\n",
      "\tREACTOME_FCERI_MEDIATED_NF_KB_ACTIVATION, REACTOME_FCERI_MEDIATED_NF_KB_ACTIVATION\r\n",
      "\tREACTOME_AGGREPHAGY, REACTOME_AGGREPHAGY\r\n",
      "\tREACTOME_TRANSCRIPTION_COUPLED_NUCLEOTIDE_EXCISION_REPAIR_TC_NER, REACTOME_TRANSCRIPTION_COUPLED_NUCLEOTIDE_EXCISION_REPAIR_TC_NER\r\n",
      "\tREACTOME_JNK_C_JUN_KINASES_PHOSPHORYLATION_AND_ACTIVATION_MEDIATED_BY_ACTIVATED_HUMAN_TAK1, REACTOME_JNK_C_JUN_KINASES_PHOSPHORYLATION_AND_ACTIVATION_MEDIATED_BY_ACTIVATED_HUMAN_TAK1\r\n",
      "\tREACTOME_INTERLEUKIN_1_SIGNALING, REACTOME_INTERLEUKIN_1_SIGNALING\r\n",
      "\tREACTOME_METALLOPROTEASE_DUBS, REACTOME_METALLOPROTEASE_DUBS\r\n",
      "\tREACTOME_SELECTIVE_AUTOPHAGY, REACTOME_SELECTIVE_AUTOPHAGY\r\n",
      "\tREACTOME_HEDGEHOG_LIGAND_BIOGENESIS, REACTOME_HEDGEHOG_LIGAND_BIOGENESIS\r\n",
      "\tLANG_MYB_FAMILY_TARGETS, LANG_MYB_FAMILY_TARGETS\r\n",
      "\tREACTOME_IRAK2_MEDIATED_ACTIVATION_OF_TAK1_COMPLEX, REACTOME_IRAK2_MEDIATED_ACTIVATION_OF_TAK1_COMPLEX\r\n",
      "\tREACTOME_TICAM1_TRAF6_DEPENDENT_INDUCTION_OF_TAK1_COMPLEX, REACTOME_TICAM1_TRAF6_DEPENDENT_INDUCTION_OF_TAK1_COMPLEX\r\n",
      "\tREACTOME_REGULATION_OF_PTEN_GENE_TRANSCRIPTION, REACTOME_REGULATION_OF_PTEN_GENE_TRANSCRIPTION\r\n",
      "\tREACTOME_ABC_TRANSPORTER_DISORDERS, REACTOME_ABC_TRANSPORTER_DISORDERS\r\n",
      "\tREACTOME_TAK1_ACTIVATES_NFKB_BY_PHOSPHORYLATION_AND_ACTIVATION_OF_IKKS_COMPLEX, REACTOME_TAK1_ACTIVATES_NFKB_BY_PHOSPHORYLATION_AND_ACTIVATION_OF_IKKS_COMPLEX\r\n",
      "\tREACTOME_RUNX1_REGULATES_GENES_INVOLVED_IN_MEGAKARYOCYTE_DIFFERENTIATION_AND_PLATELET_FUNCTION, REACTOME_RUNX1_REGULATES_GENES_INVOLVED_IN_MEGAKARYOCYTE_DIFFERENTIATION_AND_PLATELET_FUNCTION\r\n",
      "\tGNATENKO_PLATELET_SIGNATURE, GNATENKO_PLATELET_SIGNATURE\r\n",
      "\tCHAUHAN_RESPONSE_TO_METHOXYESTRADIOL_DN, CHAUHAN_RESPONSE_TO_METHOXYESTRADIOL_DN\r\n",
      "\tREACTOME_DEFECTIVE_CFTR_CAUSES_CYSTIC_FIBROSIS, REACTOME_DEFECTIVE_CFTR_CAUSES_CYSTIC_FIBROSIS\r\n",
      "\tREACTOME_NEGATIVE_REGULATION_OF_MET_ACTIVITY, REACTOME_NEGATIVE_REGULATION_OF_MET_ACTIVITY\r\n",
      "\tREACTOME_NOTCH2_ACTIVATION_AND_TRANSMISSION_OF_SIGNAL_TO_THE_NUCLEUS, REACTOME_NOTCH2_ACTIVATION_AND_TRANSMISSION_OF_SIGNAL_TO_THE_NUCLEUS\r\n",
      "\tFLECHNER_BIOPSY_KIDNEY_TRANSPLANT_REJECTED_VS_OK_UP, FLECHNER_BIOPSY_KIDNEY_TRANSPLANT_REJECTED_VS_OK_UP\r\n",
      "\tREACTOME_DNA_REPLICATION_PRE_INITIATION, REACTOME_DNA_REPLICATION_PRE_INITIATION\r\n",
      "\tREACTOME_CIRCADIAN_CLOCK, REACTOME_CIRCADIAN_CLOCK\r\n",
      "\tREACTOME_TRAF6_MEDIATED_INDUCTION_OF_TAK1_COMPLEX_WITHIN_TLR4_COMPLEX, REACTOME_TRAF6_MEDIATED_INDUCTION_OF_TAK1_COMPLEX_WITHIN_TLR4_COMPLEX\r\n",
      "\tREACTOME_ACTIVATED_TAK1_MEDIATES_P38_MAPK_ACTIVATION, REACTOME_ACTIVATED_TAK1_MEDIATES_P38_MAPK_ACTIVATION\r\n",
      "\tREACTOME_CHAPERONE_MEDIATED_AUTOPHAGY, REACTOME_CHAPERONE_MEDIATED_AUTOPHAGY\r\n",
      "\tREACTOME_IRON_UPTAKE_AND_TRANSPORT, REACTOME_IRON_UPTAKE_AND_TRANSPORT\r\n",
      "\tREACTOME_REGULATION_OF_TP53_EXPRESSION_AND_DEGRADATION, REACTOME_REGULATION_OF_TP53_EXPRESSION_AND_DEGRADATION\r\n",
      "\tREACTOME_ONCOGENE_INDUCED_SENESCENCE, REACTOME_ONCOGENE_INDUCED_SENESCENCE\r\n",
      "\tREACTOME_DNA_REPLICATION, REACTOME_DNA_REPLICATION\r\n",
      "\tGUO_HEX_TARGETS_UP, GUO_HEX_TARGETS_UP\r\n",
      "\tREACTOME_ASSEMBLY_OF_THE_PRE_REPLICATIVE_COMPLEX, REACTOME_ASSEMBLY_OF_THE_PRE_REPLICATIVE_COMPLEX\r\n",
      "\tREACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE, REACTOME_SWITCHING_OF_ORIGINS_TO_A_POST_REPLICATIVE_STATE\r\n",
      "\tREACTOME_APC_C_CDH1_MEDIATED_DEGRADATION_OF_CDC20_AND_OTHER_APC_C_CDH1_TARGETED_PROTEINS_IN_LATE_MITOSIS_EARLY_G1, REACTOME_APC_C_CDH1_MEDIATED_DEGRADATION_OF_CDC20_AND_OTHER_APC_C_CDH1_TARGETED_PROTEINS_IN_LATE_MITOSIS_EARLY_G1\r\n",
      "\tREACTOME_REGULATION_OF_RAS_BY_GAPS, REACTOME_REGULATION_OF_RAS_BY_GAPS\r\n",
      "RHEIN_ALL_GLUCOCORTICOID_THERAPY_UP, RHEIN_ALL_GLUCOCORTICOID_THERAPY_UP\r\n",
      "\tCHIARETTI_T_ALL_RELAPSE_PROGNOSIS, CHIARETTI_T_ALL_RELAPSE_PROGNOSIS\r\n",
      "\tBASSO_B_LYMPHOCYTE_NETWORK, BASSO_B_LYMPHOCYTE_NETWORK\r\n",
      "\tZHAN_MULTIPLE_MYELOMA_CD1_DN, ZHAN_MULTIPLE_MYELOMA_CD1_DN\r\n",
      "\tMATTHEWS_SKIN_CARCINOGENESIS_VIA_JUN, MATTHEWS_SKIN_CARCINOGENESIS_VIA_JUN\r\n",
      "\tLIU_SMARCA4_TARGETS, LIU_SMARCA4_TARGETS\r\n",
      "\tBERTUCCI_MEDULLARY_VS_DUCTAL_BREAST_CANCER_DN, BERTUCCI_MEDULLARY_VS_DUCTAL_BREAST_CANCER_DN\r\n",
      "\tFIGUEROA_AML_METHYLATION_CLUSTER_3_DN, FIGUEROA_AML_METHYLATION_CLUSTER_3_DN\r\n",
      "\tOSADA_ASCL1_TARGETS_DN, OSADA_ASCL1_TARGETS_DN\r\n",
      "RIZ_ERYTHROID_DIFFERENTIATION_HEMGN, RIZ_ERYTHROID_DIFFERENTIATION_HEMGN\r\n",
      "\tTONKS_TARGETS_OF_RUNX1_RUNX1T1_FUSION_GRANULOCYTE_UP, TONKS_TARGETS_OF_RUNX1_RUNX1T1_FUSION_GRANULOCYTE_UP\r\n",
      "THUM_SYSTOLIC_HEART_FAILURE_DN, THUM_SYSTOLIC_HEART_FAILURE_DN\r\n",
      "\tCHANDRAN_METASTASIS_TOP50_DN, CHANDRAN_METASTASIS_TOP50_DN\r\n",
      "\tLEIN_CEREBELLUM_MARKERS, LEIN_CEREBELLUM_MARKERS\r\n"
     ]
    }
   ],
   "source": [
    "cat ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c2/pvmatrix.txt.killed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R version 3.5.1 (2018-07-02)\n",
      "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 18.04.4 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS: /rumi/shams/abe/anaconda3/envs/imRIP/radar/lib/R/lib/libRblas.so\n",
      "LAPACK: /rumi/shams/abe/anaconda3/envs/imRIP/radar/lib/R/lib/libRlapack.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      " [1] stats4    parallel  tools     stats     graphics  grDevices utils    \n",
      " [8] datasets  methods   base     \n",
      "\n",
      "other attached packages:\n",
      " [1] bindrcpp_0.2.2            patchwork_1.0.0          \n",
      " [3] forcats_0.3.0             stringr_1.3.1            \n",
      " [5] dplyr_0.7.6               purrr_0.2.5              \n",
      " [7] readr_1.1.1               tidyr_0.8.1              \n",
      " [9] tibble_1.4.2              tidyverse_1.2.1          \n",
      "[11] ggrepel_0.8.2             RADAR_0.2.3              \n",
      "[13] qvalue_2.16.0             RcppArmadillo_0.8.600.0.0\n",
      "[15] Rcpp_1.0.4.6              RColorBrewer_1.1-2       \n",
      "[17] gplots_3.0.1              doParallel_1.0.11        \n",
      "[19] iterators_1.0.10          foreach_1.4.4            \n",
      "[21] ggplot2_3.0.0             Rsamtools_1.34.0         \n",
      "[23] Biostrings_2.50.2         XVector_0.22.0           \n",
      "[25] GenomicFeatures_1.34.1    AnnotationDbi_1.44.0     \n",
      "[27] Biobase_2.42.0            GenomicRanges_1.34.0     \n",
      "[29] GenomeInfoDb_1.18.1       IRanges_2.16.0           \n",
      "[31] S4Vectors_0.20.1          BiocGenerics_0.28.0      \n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      " [1] colorspace_1.3-2            htmlTable_1.12             \n",
      " [3] base64enc_0.1-3             rstudioapi_0.7             \n",
      " [5] bit64_0.9-7                 lubridate_1.7.4            \n",
      " [7] xml2_1.2.0                  codetools_0.2-15           \n",
      " [9] splines_3.5.1               geneplotter_1.60.0         \n",
      "[11] knitr_1.20                  Formula_1.2-3              \n",
      "[13] jsonlite_1.5                broom_0.5.0                \n",
      "[15] annotate_1.60.1             cluster_2.0.7-1            \n",
      "[17] compiler_3.5.1              httr_1.3.1                 \n",
      "[19] backports_1.1.2             assertthat_0.2.0           \n",
      "[21] Matrix_1.2-14               lazyeval_0.2.1             \n",
      "[23] cli_1.0.0                   acepack_1.4.1              \n",
      "[25] htmltools_0.3.6             prettyunits_1.0.2          \n",
      "[27] gtable_0.2.0                glue_1.3.0                 \n",
      "[29] GenomeInfoDbData_1.2.1      reshape2_1.4.3             \n",
      "[31] cellranger_1.1.0            gdata_2.18.0               \n",
      "[33] nlme_3.1-137                rtracklayer_1.42.1         \n",
      "[35] rvest_0.3.2                 gtools_3.8.1               \n",
      "[37] XML_3.98-1.12               zlibbioc_1.28.0            \n",
      "[39] scales_0.5.0                hms_0.4.2                  \n",
      "[41] SummarizedExperiment_1.12.0 memoise_1.1.0              \n",
      "[43] gridExtra_2.3               biomaRt_2.38.0             \n",
      "[45] rpart_4.1-13                latticeExtra_0.6-28        \n",
      "[47] stringi_1.2.4               RSQLite_2.1.1              \n",
      "[49] genefilter_1.64.0           checkmate_1.8.5            \n",
      "[51] caTools_1.17.1.1            BiocParallel_1.16.6        \n",
      "[53] rlang_0.4.7                 pkgconfig_2.0.1            \n",
      "[55] matrixStats_0.56.0          bitops_1.0-6               \n",
      "[57] lattice_0.20-35             bindr_0.1.1                \n",
      "[59] GenomicAlignments_1.18.1    htmlwidgets_1.2            \n",
      "[61] labeling_0.3                bit_1.1-14                 \n",
      "[63] tidyselect_0.2.4            plyr_1.8.4                 \n",
      "[65] magrittr_1.5                DESeq2_1.22.1              \n",
      "[67] R6_2.2.2                    Hmisc_4.1-1                \n",
      "[69] DelayedArray_0.8.0          DBI_1.0.0                  \n",
      "[71] pillar_1.3.0                haven_1.1.2                \n",
      "[73] foreign_0.8-71              withr_2.1.2                \n",
      "[75] survival_2.42-6             RCurl_1.95-4.11            \n",
      "[77] nnet_7.3-12                 modelr_0.1.2               \n",
      "[79] crayon_1.3.4                KernSmooth_2.23-15         \n",
      "[81] progress_1.2.2              locfit_1.5-9.4             \n",
      "[83] grid_3.5.1                  readxl_1.1.0               \n",
      "[85] data.table_1.11.6           blob_1.1.1                 \n",
      "[87] digest_0.6.15               xtable_1.8-2               \n",
      "[89] munsell_0.5.0              \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:root] *",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
