{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Metagene plots\n",
    "Now, let's draw **metagene plots**: (using seprate conda env and ipython kernel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Guitar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/abe/genomes/hg38/gencode.v34/gencode.v34.annotation.gtf',organism='Homo sapiens')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"20201206021304\"\n",
      "[1] \"There are 228048 transcripts of 60669 genes in the genome.\"\n",
      "[1] \"total 228048 transcripts extracted ...\"\n",
      "[1] \"total 97941 transcripts left after ambiguity filter ...\"\n",
      "[1] \"total 97941 transcripts left after check chromosome validity ...\"\n",
      "[1] \"total 10939 mRNAs left after component length filter ...\"\n",
      "[1] \"total 60598 ncRNAs left after ncRNA length filter ...\"\n",
      "[1] \"generate components for all tx\"\n",
      "[1] \"generate components for mRNA\"\n",
      "[1] \"generate components for lncRNA\"\n",
      "[1] \"generate chiped transcriptome\"\n",
      "[1] \"generate coverage checking ranges for tx\"\n",
      "[1] \"generate coverage checking ranges for mrna\"\n",
      "[1] \"generate coverage checking ranges for ncrna\"\n",
      "[1] \"20201206021840\"\n",
      "[1] \"import BED file radar/result.sig.bed\"\n",
      "[1] \"sample 10 points for Group1\"\n",
      "[1] \"start figure plotting for tx ...\"\n",
      "[1] \"start figure plotting for mrna ...\"\n",
      "[1] \"start figure plotting for ncrna ...\"\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "GuitarPlot(txTxdb = txdb,stBedFiles = list(\"radar/result.sig.bed\"),miscOutFilePrefix = \"Guitar\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# !for f in *.pdf; do bash ~/Workflows/my_scripts/pdf2png.sh $f; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'Guitar_mrna_test.pdf' -> 'plots/Guitar_mrna_test.pdf'\r\n",
      "renamed 'Guitar_ncrna_test.pdf' -> 'plots/Guitar_ncrna_test.pdf'\r\n",
      "renamed 'Guitar_tx_test.pdf' -> 'plots/Guitar_tx_test.pdf'\r\n"
     ]
    }
   ],
   "source": [
    "mv -v Guitar_* plots/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# FIRE \n",
    "https://github.com/goodarzilab/FIRE\n",
    "### Motif analysis "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[AG]GAC\r\n",
      "[AGT][AG]AC[ACT]\r\n"
     ]
    }
   ],
   "source": [
    "cat fire/motifs_of_interest.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "# get exon sequences for significant hits from RADAR \n",
    "bedtools getfasta -name -s -fi /rumi/shams/genomes/hg38/hg38.fa \\\n",
    "-bed radar/result.sig.bed \\\n",
    "-split -fo radar/result.sig.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p fire "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Can't locate Markov.pm in @INC (you may need to install the Markov module) (@INC contains: /flash/bin/FIRE-1.1/SCRIPTS /rumi/shams/abe/perl5/lib/perl5/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/perl5/lib/perl5/5.26.2 /rumi/shams/abe/perl5/lib/perl5/x86_64-linux-thread-multi /rumi/shams/abe/perl5/lib/perl5 /rumi/shams/abe/anaconda3/envs/perl/lib/site_perl/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/anaconda3/envs/perl/lib/site_perl/5.26.2 /rumi/shams/abe/anaconda3/envs/perl/lib/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/anaconda3/envs/perl/lib/5.26.2 .) at /rumi/shams/abe/prep_seqs_for_teiser_run.pl line 5.\n",
      "BEGIN failed--compilation aborted at /rumi/shams/abe/prep_seqs_for_teiser_run.pl line 5.\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'\\n# prepare fasta file for FIRE\\n~/anaconda3/envs/perl/bin/perl ~/prep_seqs_for_teiser_run.pl radar/result.sig.fa fire/peaks\\n'' returned non-zero exit status 2.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-9254eabce3a0>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'\\n# prepare fasta file for FIRE\\n~/anaconda3/envs/perl/bin/perl ~/prep_seqs_for_teiser_run.pl radar/result.sig.fa fire/peaks\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/bedenv/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2369\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2370\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2371\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2372\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2373\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bedenv/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bedenv/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/bedenv/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'\\n# prepare fasta file for FIRE\\n~/anaconda3/envs/perl/bin/perl ~/prep_seqs_for_teiser_run.pl radar/result.sig.fa fire/peaks\\n'' returned non-zero exit status 2."
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "\n",
    "# prepare fasta file for FIRE\n",
    "~/anaconda3/envs/perl/bin/perl ~/prep_seqs_for_teiser_run.pl radar/result.sig.fa fire/peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mon Dec  7 01:04:26 UTC 2020\n",
      "RNA, Processing peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt\n",
      "RNA, Step 2: skip seed optimization, use -motiffile_rna option instead.\n",
      "RNA, Step 3: evaluation of motif significances.\n",
      "Processing motif [AG]GAC.\n",
      "MI=0.00492797\n",
      "Shuffle rank=0\n",
      "Z-score=44.607\n",
      "Sigificance category= OK-NO-SEED\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [AGT][AG]AC[ACT].\n",
      "MI=0.00674459\n",
      "Shuffle rank=0\n",
      "Z-score=62.008\n",
      "Sigificance category= OK-NO-SEED\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "RNA, Step 3.5: creating binary expression profiles.\n",
      "RNA, Step 4: discovery of distance constraints.\n",
      "Read 2 motifs.\n",
      "Read 9374 (genes) x 2 (conditions) matrix.\n",
      "Read 9374 (genes) x 1 (conditions) matrix.\n",
      "Reading sequences and building motif profiles ...Done.\n",
      "Average sequence length is 50.\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AG]GAC, present in 3271 genes.\n",
      "       Average distance: MI=0.0008, rank=1730/10000, Z=0.7684, mbins=3\n",
      "  Orientation bias {5'): MI=0.0049, rank=0/10000, Z=44.8914\n",
      "  Orientation bias (3'): MI=0.0007, rank=29/10000, Z=5.4831\n",
      "       Copy number (1+): 3271 genes, MI=0.0049, rank=0/10000, Z=44.1359\n",
      "       Copy number (2+): 797 genes, MI=0.0001, rank=3039/10000, Z=0.1173\n",
      "       Copy number (3+): 144 genes, MI=0.0012, rank=1/10000, Z=9.9201\n",
      "       Copy number (4+): 19 genes, MI=0.0002, rank=1669/10000, Z=1.0995\n",
      "       Copy number (5+): 2 genes, MI=0.0000, rank=10000/10000, Z=-1.0104\n",
      "       Copy number (6+): 2 genes, MI=0.0000, rank=10000/10000, Z=-0.9938\n",
      "       Copy number (7+): 2 genes, MI=0.0000, rank=10000/10000, Z=-1.0036\n",
      "       Copy number (8+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT][AG]AC[ACT], present in 5874 genes.\n",
      "       Average distance: MI=0.0011, rank=101/10000, Z=3.6015, mbins=3\n",
      "  Orientation bias {5'): MI=0.0067, rank=0/10000, Z=62.2616\n",
      "  Orientation bias (3'): MI=0.0031, rank=0/10000, Z=27.6077\n",
      "       Copy number (1+): 5874 genes, MI=0.0067, rank=0/10000, Z=60.3224\n",
      "       Copy number (2+): 2679 genes, MI=0.0001, rank=3354/10000, Z=-0.0267\n",
      "       Copy number (3+): 923 genes, MI=0.0014, rank=0/10000, Z=11.7770\n",
      "       Copy number (4+): 256 genes, MI=0.0040, rank=0/10000, Z=36.0522\n",
      "       Copy number (5+): 72 genes, MI=0.0013, rank=0/10000, Z=11.1348\n",
      "       Copy number (6+): 17 genes, MI=0.0002, rank=1348/10000, Z=1.3615\n",
      "       Copy number (7+): 7 genes, MI=0.0001, rank=4510/10000, Z=0.1496\n",
      "       Copy number (8+): 3 genes, MI=0.0003, rank=2525/10000, Z=1.7205\n",
      "\n",
      "RNA, Step 5.5: conservation.\n",
      "Setting -rna to 1.\n",
      "[AG]GAC\t98.12\n",
      "[AGT][AG]AC[ACT]\t21.42\n",
      "RNA, Step 5: summarize information.\n",
      "Reading RNA motifs mi_dist output ...Done.\n",
      "Parsing RNA motif file ... Done.\n",
      "Reading mi_signif output file for RNA motifs ...!!!! 133544!!!! 72400Done.\n",
      "Got 2 motifs\n",
      "Writing good columns to peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.signifcolumns ...Done.\n",
      "Outputing summary ... Done.\n",
      "HI!HELLO!peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.binary created.\n",
      "RNA, Step 5.3: generate profiles for motifs in summary file.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [AGT][AG]AC[ACT] ... /flash/bin/FIRE-1.1/PROGRAMS/genregexp -re [AGT][AG]AC[ACT] -fastafile peaks_teiser.fa -flank 25  -rna 1  -reldist 0  > genregexp8731116303710Done.\n",
      "Generating profile for [AG]GAC ... /flash/bin/FIRE-1.1/PROGRAMS/genregexp -re [AG]GAC -fastafile peaks_teiser.fa -flank 25  -rna 1  -reldist 0  > genregexp3960211613073Done.\n",
      "RNA, Step 5.3 bis: Generate profiles for /flash/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [AGT][AG]AC[ACT] ... /flash/bin/FIRE-1.1/PROGRAMS/genregexp -re [AGT][AG]AC[ACT] -fastafile /flash/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa -flank 25  -rna 1  -reldist 0  > genregexp2866103012731Done.\n",
      "Generating profile for [AG]GAC ... /flash/bin/FIRE-1.1/PROGRAMS/genregexp -re [AG]GAC -fastafile /flash/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa -flank 25  -rna 1  -reldist 0  > genregexp4750013631371Done.\n",
      "RNA, Determining target gene sets.\n",
      "RNA, Drawing position histograms.\n",
      "RNA, Step 6: cluster motifs.\n",
      "Using positive correlation.\n",
      "Doing all stats.\n",
      "Summary file has 0 DNA motifs and 2 RNA motifs.\n",
      "Read in RNA sequences and build motif profiles ...Done.\n",
      "Outputing MI matrix ... Done.\n",
      "[AGT][AG]AC[ACT]\t0\n",
      "[AGT][AG]AC[ACT]\t[AG]GAC\t0.2946\t0\t2683.4893\t0.6562\t576\t1.8661\t576\t1.8661\n",
      "[AG]GAC\t0\n",
      "Finish calculated rank, z-scores for remaining entries ... [AGT][AG]AC[ACT]\t[AGT][AG]AC[ACT]\t0.9216\t0\t8594.7727\t1.0000\t0\t10.9940\t0\t10.9940\n",
      "[AG]GAC\t[AG]GAC\t0.7101\t0\t6494.4804\t1.0000\t3478\t0.0748\t3478\t0.0748\n",
      "Done.\n",
      "RNA, Step 5.6: generate genome-wide GO enrichments for motifs in summary file.\n",
      "RNA, draw motif maps for co-localizing pairs\n",
      "Draw motif map for [AGT][AG]AC[ACT] .. Done.\n",
      "Draw motif map for [AG]GAC .. Done.\n",
      "Draw motif map for [AGT][AG]AC[ACT],[AGT][AG]AC[ACT] ..  (peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.mimatrix_OUT/[AGT][AG]AC[ACT]_[AGT][AG]AC[ACT].eps) Done.\n",
      "RNA, Step 6.8: identifying motifs that match micrornas.\n",
      "RNA, Step 7: draw matrix figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 400, xbase = 35, ybase = 250\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 0.eps ... Done.\n",
      "Processing [AG]GAC ... Outputing motif 1.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 0.png ... \n",
      "Processing [AG]GAC ... Outputing motif 1.png ... \n",
      "RNA, Step 7b: draw density figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 400, xbase = 35, ybase = 250\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 0.eps ... Done.\n",
      "Processing [AG]GAC ... Outputing motif 1.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 0.png ... \n",
      "Processing [AG]GAC ... Outputing motif 1.png ... \n",
      "Parsing distance report file ...  Done.\n",
      "[AGT][AG]AC[ACT] has no position or orientation bias.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 230, ysize = 310, xbase = 35, ybase = 115\n",
      "[AGT][AG]AC[ACT]\t0\n",
      "[AG]GAC\t1\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf ... Done.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 250, ysize = 305, xbase = 75, ybase = 130\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.html[AGT][AG]AC[ACT]\t0\n",
      "[AG]GAC\t1\n",
      "RNA, Step 10: motif report.\n",
      "Mon Dec  7 01:05:31 UTC 2020\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Segmentation fault (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /flash/bin/FIRE-1.1/SCRIPTS/draw_position_histogram.pl line 134.\n",
      "Can't locate Hypergeom.pm in @INC (you may need to install the Hypergeom module) (@INC contains: /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.12.4/darwin-thread-multi-2level/nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.10.0/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.18.2/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.22.0/x86_64-linux/Hypergeom.pm /flash/bin/FIRE-1.1/SCRIPTS /rumi/shams/abe/perl5/lib/perl5/x86_64-linux-gnu-thread-multi /rumi/shams/abe/perl5/lib/perl5 /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.26.1 /usr/local/share/perl/5.26.1 /usr/lib/x86_64-linux-gnu/perl5/5.26 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.26 /usr/share/perl/5.26 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /flash/bin/FIRE-1.1/SCRIPTS/GroupEnrichment.pm line 24.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "cd fire/\n",
    "# run FIRE --doskipdiscovery=1\n",
    "perl $FIREDIR/fire.pl \n",
    "    --expfile=peaks_teiser.txt \\\n",
    "    --exptype=discrete \n",
    "    --fastafile_rna=peaks_teiser.fa \n",
    "    --nodups=1 --dodna=0 --dodnarna=0 \n",
    "    --species=human \n",
    "    --doskipdiscovery=1 \n",
    "    --motiffile_rna=motifs_of_interest.txt \n",
    "    --oribiasonly=0\n",
    "\n",
    "mv peaks_teiser.txt_FIRE/ motifs_of_interest_FIRE\n",
    "cd ../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Aug  6 14:08:07 UTC 2020\n",
      "RNA, Processing peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt\n",
      "RNA, Step 1: seed discovery.\n",
      "Setting -shuffle to 16384.\n",
      "Expression file is a 9374 by 1 matrix.\n",
      "Min expression value read is 0, max is 1\n",
      "Reading sequences ... Done\n",
      "Number of ORFs = 9374\n",
      "Calculating Mutual Information for all seeds ... Done\n",
      "top 10 seeds...\n",
      "CCCCCCC\n",
      "Determining threshold ...\n",
      "Evaluating seed #0, CCCCCCC ... passed.\n",
      "Evaluating seed #100, AGGTGGA ... not passed.\n",
      "Decreasing intervals phase.\n",
      "Evaluating seed #50, GCTAGGC ... not passed.\n",
      "Evaluating seed #25, TACAAAT ... passed.\n",
      "Evaluating seed #37, GAGGAAG ... passed.\n",
      "Evaluating seed #43, GAAACCC ... not passed.\n",
      "Searches for 5 consecutive 'not passed'.\n",
      "Threshold set to seed #37 (included).\n",
      "CCCCCCC\t0.0073\n",
      "GGGGGGG\t0.0061\n",
      "AATGTGG\t0.0048\n",
      "CCCCCCT\t0.0043\n",
      "TCCCCCC\t0.0040\n",
      "GAGAAAC\t0.0039\n",
      "AGGGGGG\t0.0039\n",
      "GCCCCCC\t0.0037\n",
      "AGAAACC\t0.0037\n",
      "TTTTTTT\t0.0036\n",
      "GTGGCAA\t0.0034\n",
      "TTCATAC\t0.0033\n",
      "GAATGTG\t0.0031\n",
      "GGGGGGA\t0.0031\n",
      "CCCCCCA\t0.0031\n",
      "GAGAGAG\t0.0029\n",
      "TCTCTCA\t0.0028\n",
      "TGGGGGG\t0.0028\n",
      "ATGTGGA\t0.0026\n",
      "ATCCCCC\t0.0024\n",
      "CATACTG\t0.0024\n",
      "TCATACT\t0.0023\n",
      "AAGCCTT\t0.0022\n",
      "GGGGGGT\t0.0022\n",
      "GGGGGGC\t0.0022\n",
      "TACAAAT\t0.0021\n",
      "ATCACTC\t0.0021\n",
      "ACTGGAG\t0.0021\n",
      "GGAAGAA\t0.0020\n",
      "ATTCATA\t0.0020\n",
      "CCCCCTG\t0.0020\n",
      "AGAATTC\t0.0019\n",
      "TGAATGT\t0.0019\n",
      "GAATTCA\t0.0019\n",
      "GTGAGAG\t0.0019\n",
      "CAGGGGG\t0.0018\n",
      "GGCCCAG\t0.0018\n",
      "GAGGAAG\t0.0018\n",
      "RNA, Step 2: seed optimization.\n",
      "Using minr = 2.0000.\n",
      "Motif length will be 9.\n",
      "Creating prefix tree for storing patterns ... Done.\n",
      "Reading expression data ... Done.\n",
      "Creating hash table for gene names ... Done.\n",
      "Reading sequences and building the index.\n",
      "160952 k-mers in index\n",
      "Matrix occupancy = 396291 / 1508764048.000000\n",
      "Done.                                \n",
      "\n",
      "Evaluating seed CCCCCCC ... \n",
      "optimizing.\n",
      "-243387712\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [CGT].[CT]CCCCC[ACT]\n",
      "Mutual information: 0.009685\n",
      "Optimization stability: 1\n",
      "Z-score: 87.208\n",
      "Starting seed: CCCCCCC\n",
      "Seed MI: 0.007106\n",
      "\n",
      "Evaluating seed GGGGGGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=65.4329\n",
      "optimizing.\n",
      "-242668880\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [AGT]G[GT]G[AGT]G.[GT].\n",
      "Mutual information: 0.009994\n",
      "Optimization stability: 1\n",
      "Z-score: 88.192\n",
      "Starting seed: GGGGGGG\n",
      "Seed MI: 0.005734\n",
      "\n",
      "Evaluating seed AATGTGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=40.3512\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=7.2926\n",
      "optimizing.\n",
      "-241950048\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [CG][AG][ACT].GTG[CG][ACG]\n",
      "Mutual information: 0.009527\n",
      "Optimization stability: 1\n",
      "Z-score: 87.439\n",
      "Starting seed: AATGTGG\n",
      "Seed MI: 0.004672\n",
      "\n",
      "Evaluating seed CCCCCCT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0195\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.019 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed TCCCCCC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0151\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.015 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GAGAAAC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=22.7304\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=44.1552\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=1.9581\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.958 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed AGGGGGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=424.4717\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.1220\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.122 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed GCCCCCC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0169\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.017 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed AGAAACC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=20.1452\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=8.7020\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=1.3096\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.310 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed TTTTTTT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=66.3992\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=5.4692\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=8.9992\n",
      "optimizing.\n",
      "-240781168\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [AGT][GT]TT[GT]TT[CGT][CGT]\n",
      "Mutual information: 0.005533\n",
      "Optimization stability: 4\n",
      "Z-score: 51.238\n",
      "Starting seed: TTTTTTT\n",
      "Seed MI: 0.003746\n",
      "\n",
      "Evaluating seed GTGGCAA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=90.7343\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=5.6903\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.2950\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.295 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed TTCATAC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=14.2513\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=2.9823\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=72.2644\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=907.3990\n",
      "optimizing.\n",
      "-239987328\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: .[GT][ACT]CA[GT][AC]..\n",
      "Mutual information: 0.007099\n",
      "Optimization stability: 1\n",
      "Z-score: 64.026\n",
      "Starting seed: TTCATAC\n",
      "Seed MI: 0.003117\n",
      "\n",
      "Evaluating seed GAATGTG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=24.5738\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=28.8872\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.1006\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.101 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GGGGGGA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=35.0455\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.1413\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.141 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed CCCCCCA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0025\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.002 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GAGAGAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=26.6763\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=1.3343\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.334 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed TCTCTCA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=10.2738\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=1506.4942\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=4.4152\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=404.8468\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=55.0115\n",
      "optimizing.\n",
      "-238968464\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [ACT]TCTCTCA.\n",
      "Mutual information: 0.003120\n",
      "Optimization stability: 10\n",
      "Z-score: 27.632\n",
      "Starting seed: TCTCTCA\n",
      "Seed MI: 0.002705\n",
      "\n",
      "Evaluating seed TGGGGGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=31.9973\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.0837\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.084 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed ATGTGGA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=74.1334\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=23.1235\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.2198\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.220 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed ATCCCCC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0381\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.038 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed CATACTG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=61.4274\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=2.5441\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=3.2105\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=7.6543\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=0.4049\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.405 for .[GT][ACT]CA[GT][AC].. (< minr=2.000).\n",
      "\n",
      "Evaluating seed TCATACT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=9.8089\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=4.2091\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=9.8884\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=10.8367\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=0.3048\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.305 for .[GT][ACT]CA[GT][AC].. (< minr=2.000).\n",
      "\n",
      "Evaluating seed AAGCCTT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=6.5666\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=17.2742\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.7838\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.784 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GGGGGGT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=24.3867\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.0638\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.064 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed GGGGGGC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=173.2965\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.1338\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.134 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed TACAAAT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=8.3689\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=6.2097\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.8060\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.806 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed ATCACTC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=23.5214\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=179.0895\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=10.6221\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=25.4679\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=34.5897\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=3.2066\n",
      "optimizing.\n",
      "-237574560\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: .[ACG].C[AG]C[AGT][AC][AGT]\n",
      "Mutual information: 0.008361\n",
      "Optimization stability: 1\n",
      "Z-score: 75.298\n",
      "Starting seed: ATCACTC\n",
      "Seed MI: 0.002137\n",
      "\n",
      "Evaluating seed ACTGGAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=34.1494\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=36.2412\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=7.9176\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=2496.2476\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=3.6553\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=19.6046\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT] : I(B;E|A)/I(A;B)=8.2936\n",
      "optimizing.\n",
      "-236855728\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: [ACT].[ACT][ACT]GG[AT][ACG][ACG]\n",
      "Mutual information: 0.012394\n",
      "Optimization stability: 1\n",
      "Z-score: 112.285\n",
      "Starting seed: ACTGGAG\n",
      "Seed MI: 0.002035\n",
      "\n",
      "Evaluating seed GGAAGAA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=18.5903\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=49.2675\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=2427.5032\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=147.9026\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=6.4242\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=23.8996\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT] : I(B;E|A)/I(A;B)=15.8394\n",
      "Motif [ACT].[ACT][ACT]GG[AT][ACG][ACG] : I(B;E|A)/I(A;B)=0.7094\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.709 for [ACT].[ACT][ACT]GG[AT][ACG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed ATTCATA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=9.0893\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=1.8732\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=36.1699\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.873 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed CCCCCTG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=0.0155\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.016 for [CGT].[CT]CCCCC[ACT] (< minr=2.000).\n",
      "\n",
      "Evaluating seed AGAATTC ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=5.9938\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=8.4715\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=92.4097\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=6.6479\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=1.7733\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.773 for .[GT][ACT]CA[GT][AC].. (< minr=2.000).\n",
      "\n",
      "Evaluating seed TGAATGT ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=46.8529\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=37.8191\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=0.2125\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.212 for [CG][AG][ACT].GTG[CG][ACG] (< minr=2.000).\n",
      "\n",
      "Evaluating seed GAATTCA ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=5.5340\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=67.6018\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=6.6976\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=6.1424\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=1.0244\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 1.024 for .[GT][ACT]CA[GT][AC].. (< minr=2.000).\n",
      "\n",
      "Evaluating seed GTGAGAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=68.3707\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.0567\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.057 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed CAGGGGG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=1099.4616\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=0.2377\n",
      "Break early (motif is too close to existing optimized motif).\n",
      "not optimized, minratio = 0.238 for [AGT]G[GT]G[AGT]G.[GT]. (< minr=2.000).\n",
      "\n",
      "Evaluating seed GGCCCAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=35.3417\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=255.4496\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=2687.7706\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=46.6215\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=101.6700\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=37.1945\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT] : I(B;E|A)/I(A;B)=35.5731\n",
      "Motif [ACT].[ACT][ACT]GG[AT][ACG][ACG] : I(B;E|A)/I(A;B)=4.4844\n",
      "optimizing.\n",
      "-235495584\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: .[CG]GCC[AC]A.[ACG]\n",
      "Mutual information: 0.003056\n",
      "Optimization stability: 1\n",
      "Z-score: 27.215\n",
      "Starting seed: GGCCCAG\n",
      "Seed MI: 0.001813\n",
      "\n",
      "Evaluating seed GAGGAAG ... \n",
      "Motif [CGT].[CT]CCCCC[ACT] : I(B;E|A)/I(A;B)=9.1527\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]. : I(B;E|A)/I(A;B)=8.7599\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG] : I(B;E|A)/I(A;B)=9.1585\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT] : I(B;E|A)/I(A;B)=11.5250\n",
      "Motif .[GT][ACT]CA[GT][AC].. : I(B;E|A)/I(A;B)=383.4880\n",
      "Motif [ACT]TCTCTCA. : I(B;E|A)/I(A;B)=16.8900\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT] : I(B;E|A)/I(A;B)=5.4151\n",
      "Motif [ACT].[ACT][ACT]GG[AT][ACG][ACG] : I(B;E|A)/I(A;B)=2.6727\n",
      "Motif .[CG]GCC[AC]A.[ACG] : I(B;E|A)/I(A;B)=16.2077\n",
      "optimizing.\n",
      "-234776752\t-279398400\t9374\t2\t2\n",
      "Best optimized motif: GGAGG[AT]A[GT][AGT]\n",
      "Mutual information: 0.003509\n",
      "Optimization stability: 4\n",
      "Z-score: 31.012\n",
      "Starting seed: GAGGAAG\n",
      "Seed MI: 0.001711\n",
      "RNA, Step 3: evaluation of motif significances.\n",
      "Processing motif [CGT].[CT]CCCCC[ACT].\n",
      "MI=0.00968482\n",
      "Shuffle rank=0\n",
      "Z-score=89.057\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [AGT]G[GT]G[AGT]G.[GT]..\n",
      "MI=0.00999403\n",
      "Shuffle rank=0\n",
      "Z-score=93.256\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [CG][AG][ACT].GTG[CG][ACG].\n",
      "MI=0.00952664\n",
      "Shuffle rank=0\n",
      "Z-score=86.088\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [AGT][GT]TT[GT]TT[CGT][CGT].\n",
      "MI=0.00553276\n",
      "Shuffle rank=0\n",
      "Z-score=50.901\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif .[GT][ACT]CA[GT][AC]...\n",
      "MI=0.00709922\n",
      "Shuffle rank=0\n",
      "Z-score=64.596\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [ACT]TCTCTCA..\n",
      "MI=0.00312046\n",
      "Shuffle rank=0\n",
      "Z-score=27.633\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif .[ACG].C[AG]C[AGT][AC][AGT].\n",
      "MI=0.00836072\n",
      "Shuffle rank=0\n",
      "Z-score=76.441\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [ACT].[ACT][ACT]GG[AT][ACG][ACG].\n",
      "MI=0.01239392\n",
      "Shuffle rank=0\n",
      "Z-score=112.063\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif .[CG]GCC[AC]A.[ACG].\n",
      "MI=0.00305585\n",
      "Shuffle rank=0\n",
      "Z-score=27.115\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif GGAGG[AT]A[GT][AGT].\n",
      "MI=0.00350923\n",
      "Shuffle rank=0\n",
      "Z-score=31.792\n",
      "Sigificance category= OK\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "RNA, Step 3.5: creating binary expression profiles.\n",
      "RNA, Step 4: discovery of distance constraints.\n",
      "Read 10 motifs.\n",
      "Read 9374 (genes) x 10 (conditions) matrix.\n",
      "Read 9374 (genes) x 1 (conditions) matrix.\n",
      "Reading sequences and building motif profiles ...Done.\n",
      "Average sequence length is 50.\n",
      "Quantized E vector into 2 bins\n",
      "Motif [CGT].[CT]CCCCC[ACT], present in 280 genes.\n",
      "       Average distance: MI=0.0041, rank=4856/10000, Z=-0.2231, mbins=3\n",
      "  Orientation bias {5'): MI=0.0097, rank=0/10000, Z=88.0621\n",
      "  Orientation bias (3'): MI=0.0059, rank=0/10000, Z=53.3486\n",
      "       Copy number (1+): 280 genes, MI=0.0097, rank=0/10000, Z=85.5798\n",
      "       Copy number (2+): 100 genes, MI=0.0057, rank=0/10000, Z=53.0047\n",
      "       Copy number (3+): 56 genes, MI=0.0047, rank=0/10000, Z=41.0431\n",
      "       Copy number (4+): 32 genes, MI=0.0027, rank=0/10000, Z=23.3255\n",
      "       Copy number (5+): 16 genes, MI=0.0017, rank=0/10000, Z=14.6532\n",
      "       Copy number (6+): 8 genes, MI=0.0009, rank=80/10000, Z=6.4446\n",
      "       Copy number (7+): 5 genes, MI=0.0005, rank=613/10000, Z=3.4452\n",
      "       Copy number (8+): 3 genes, MI=0.0003, rank=2513/10000, Z=1.7260\n",
      "       Copy number (9+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT]G[GT]G[AGT]G.[GT]., present in 1207 genes.\n",
      "       Average distance: MI=0.0003, rank=4746/10000, Z=-0.3127, mbins=2\n",
      "  Orientation bias {5'): MI=0.0100, rank=0/10000, Z=91.9656\n",
      "  Orientation bias (3'): MI=0.0082, rank=0/10000, Z=73.8283\n",
      "       Copy number (1+): 1207 genes, MI=0.0100, rank=0/10000, Z=92.0017\n",
      "       Copy number (2+): 390 genes, MI=0.0057, rank=0/10000, Z=51.4639\n",
      "       Copy number (3+): 150 genes, MI=0.0047, rank=0/10000, Z=42.3123\n",
      "       Copy number (4+): 61 genes, MI=0.0026, rank=0/10000, Z=23.0176\n",
      "       Copy number (5+): 36 genes, MI=0.0013, rank=0/10000, Z=11.3198\n",
      "       Copy number (6+): 21 genes, MI=0.0007, rank=73/10000, Z=5.4270\n",
      "       Copy number (7+): 14 genes, MI=0.0006, rank=114/10000, Z=4.8928\n",
      "       Copy number (8+): 7 genes, MI=0.0001, rank=4530/10000, Z=0.1403\n",
      "       Copy number (9+): 5 genes, MI=0.0000, rank=10000/10000, Z=-0.5672\n",
      "       Copy number (10+): 3 genes, MI=0.0000, rank=10000/10000, Z=-0.5784\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [CG][AG][ACT].GTG[CG][ACG], present in 569 genes.\n",
      "       Average distance: MI=0.0006, rank=7816/10000, Z=-0.7646, mbins=3\n",
      "  Orientation bias {5'): MI=0.0095, rank=0/10000, Z=83.2066\n",
      "  Orientation bias (3'): MI=0.0009, rank=7/10000, Z=7.2755\n",
      "       Copy number (1+): 569 genes, MI=0.0095, rank=0/10000, Z=85.8462\n",
      "       Copy number (2+): 33 genes, MI=0.0004, rank=316/10000, Z=2.9522\n",
      "       Copy number (3+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT][GT]TT[GT]TT[CGT][CGT], present in 257 genes.\n",
      "       Average distance: MI=0.0075, rank=1147/10000, Z=1.1719, mbins=2\n",
      "  Orientation bias {5'): MI=0.0055, rank=0/10000, Z=49.6126\n",
      "  Orientation bias (3'): MI=0.0008, rank=19/10000, Z=6.4529\n",
      "       Copy number (1+): 257 genes, MI=0.0055, rank=0/10000, Z=49.2060\n",
      "       Copy number (2+): 89 genes, MI=0.0028, rank=0/10000, Z=24.8781\n",
      "       Copy number (3+): 42 genes, MI=0.0013, rank=0/10000, Z=11.3944\n",
      "       Copy number (4+): 19 genes, MI=0.0003, rank=628/10000, Z=2.3384\n",
      "       Copy number (5+): 11 genes, MI=0.0004, rank=650/10000, Z=2.4829\n",
      "       Copy number (6+): 6 genes, MI=0.0006, rank=287/10000, Z=4.5011\n",
      "       Copy number (7+): 5 genes, MI=0.0005, rank=650/10000, Z=3.3694\n",
      "       Copy number (8+): 3 genes, MI=0.0003, rank=2456/10000, Z=1.7525\n",
      "       Copy number (9+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "       Copy number (10+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif .[GT][ACT]CA[GT][AC].., present in 2416 genes.\n",
      "       Average distance: MI=0.0013, rank=3454/10000, Z=0.1670, mbins=5\n",
      "  Orientation bias {5'): MI=0.0071, rank=0/10000, Z=64.8691\n",
      "  Orientation bias (3'): MI=0.0017, rank=0/10000, Z=14.6125\n",
      "       Copy number (1+): 2416 genes, MI=0.0071, rank=0/10000, Z=64.1633\n",
      "       Copy number (2+): 360 genes, MI=0.0023, rank=0/10000, Z=20.9995\n",
      "       Copy number (3+): 44 genes, MI=0.0006, rank=88/10000, Z=4.7106\n",
      "       Copy number (4+): 5 genes, MI=0.0000, rank=10000/10000, Z=-0.5646\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [ACT]TCTCTCA., present in 53 genes.\n",
      "       Average distance: MI=0.0624, rank=2595/10000, Z=0.9381, mbins=3\n",
      "  Orientation bias {5'): MI=0.0031, rank=0/10000, Z=26.8242\n",
      "  Orientation bias (3'): MI=0.0014, rank=0/10000, Z=12.0169\n",
      "       Copy number (1+): 53 genes, MI=0.0031, rank=0/10000, Z=27.9427\n",
      "       Copy number (2+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif .[ACG].C[AG]C[AGT][AC][AGT], present in 1400 genes.\n",
      "       Average distance: MI=0.0009, rank=4402/10000, Z=-0.1684, mbins=3\n",
      "  Orientation bias {5'): MI=0.0084, rank=0/10000, Z=78.3081\n",
      "  Orientation bias (3'): MI=0.0018, rank=0/10000, Z=15.8822\n",
      "       Copy number (1+): 1400 genes, MI=0.0084, rank=0/10000, Z=77.6687\n",
      "       Copy number (2+): 282 genes, MI=0.0035, rank=0/10000, Z=31.0258\n",
      "       Copy number (3+): 66 genes, MI=0.0016, rank=0/10000, Z=13.8825\n",
      "       Copy number (4+): 21 genes, MI=0.0009, rank=15/10000, Z=7.5218\n",
      "       Copy number (5+): 4 genes, MI=0.0000, rank=10000/10000, Z=-0.7124\n",
      "       Copy number (6+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "       Copy number (7+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "       Copy number (8+): 1 genes, MI=0.0001, rank=10000/10000, Z=0.0000\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [ACT].[ACT][ACT]GG[AT][ACG][ACG], present in 2784 genes.\n",
      "       Average distance: MI=0.0004, rank=4430/10000, Z=-0.1826, mbins=3\n",
      "  Orientation bias {5'): MI=0.0124, rank=0/10000, Z=110.0754\n",
      "  Orientation bias (3'): MI=0.0062, rank=0/10000, Z=57.6482\n",
      "       Copy number (1+): 2784 genes, MI=0.0124, rank=0/10000, Z=113.5838\n",
      "       Copy number (2+): 413 genes, MI=0.0041, rank=0/10000, Z=37.4894\n",
      "       Copy number (3+): 49 genes, MI=0.0006, rank=84/10000, Z=4.6189\n",
      "       Copy number (4+): 7 genes, MI=0.0000, rank=10000/10000, Z=-0.6025\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif .[CG]GCC[AC]A.[ACG], present in 346 genes.\n",
      "       Average distance: MI=0.0084, rank=1312/10000, Z=1.0083, mbins=3\n",
      "  Orientation bias {5'): MI=0.0031, rank=0/10000, Z=27.5556\n",
      "  Orientation bias (3'): MI=0.0001, rank=1935/10000, Z=0.6425\n",
      "       Copy number (1+): 346 genes, MI=0.0031, rank=0/10000, Z=27.2677\n",
      "       Copy number (2+): 14 genes, MI=0.0000, rank=10000/10000, Z=-0.7121\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif GGAGG[AT]A[GT][AGT], present in 61 genes.\n",
      "       Average distance: MI=0.0426, rank=5782/10000, Z=-0.0247, mbins=4\n",
      "  Orientation bias {5'): MI=0.0035, rank=0/10000, Z=31.5304\n",
      "  Orientation bias (3'): MI=0.0003, rank=966/10000, Z=1.6355\n",
      "       Copy number (1+): 61 genes, MI=0.0035, rank=0/10000, Z=31.9871\n",
      "       Copy number (2+): 2 genes, MI=0.0002, rank=5048/10000, Z=0.9904\n",
      "\n",
      "RNA, Step 5.5: conservation.\n",
      "Setting -rna to 1.\n",
      "[CGT].[CT]CCCCC[ACT]\t96.91\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t99.55\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t96.47\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t98.58\n",
      ".[GT][ACT]CA[GT][AC]..\t90.26\n",
      "[ACT]TCTCTCA.\t51.96\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t93.84\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t93.20\n",
      ".[CG]GCC[AC]A.[ACG]\t96.44\n",
      "GGAGG[AT]A[GT][AGT]\t34.93\n",
      "RNA, Step 5: summarize information.\n",
      "Reading RNA motifs mi_dist output ...Done.\n",
      "Parsing RNA motif file ... Done.\n",
      "Reading mi_signif output file for RNA motifs ...!!!! 254088!!!! -170544!!!! -4138504!!!! -147808!!!! -186944!!!! -186536!!!! -186632!!!! -186656!!!! -186656!!!! -186728Done.\n",
      "Got 10 motifs\n",
      "Writing good columns to peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.signifcolumns ...Done.\n",
      "Outputing summary ... Done.\n",
      "HI!HELLO!peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.binary created.\n",
      "RNA, Step 5.3: generate profiles for motifs in summary file.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Done.\n",
      "Generating profile for [AGT]G[GT]G[AGT]G.[GT]. ... Done.\n",
      "Generating profile for [CGT].[CT]CCCCC[ACT] ... Done.\n",
      "Generating profile for [CG][AG][ACT].GTG[CG][ACG] ... Done.\n",
      "Generating profile for .[ACG].C[AG]C[AGT][AC][AGT] ... Done.\n",
      "Generating profile for .[GT][ACT]CA[GT][AC].. ... Done.\n",
      "Generating profile for [AGT][GT]TT[GT]TT[CGT][CGT] ... Done.\n",
      "Generating profile for GGAGG[AT]A[GT][AGT] ... Done.\n",
      "Generating profile for [ACT]TCTCTCA. ... Done.\n",
      "Generating profile for .[CG]GCC[AC]A.[ACG] ... Done.\n",
      "RNA, Step 5.3 bis: Generate profiles for /flash/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Done.\n",
      "Generating profile for [AGT]G[GT]G[AGT]G.[GT]. ... Done.\n",
      "Generating profile for [CGT].[CT]CCCCC[ACT] ... Done.\n",
      "Generating profile for [CG][AG][ACT].GTG[CG][ACG] ... Done.\n",
      "Generating profile for .[ACG].C[AG]C[AGT][AC][AGT] ... Done.\n",
      "Generating profile for .[GT][ACT]CA[GT][AC].. ... Done.\n",
      "Generating profile for [AGT][GT]TT[GT]TT[CGT][CGT] ... Done.\n",
      "Generating profile for GGAGG[AT]A[GT][AGT] ... Done.\n",
      "Generating profile for [ACT]TCTCTCA. ... Done.\n",
      "Generating profile for .[CG]GCC[AC]A.[ACG] ... Done.\n",
      "RNA, Determining target gene sets.\n",
      "RNA, Drawing position histograms.\n",
      "RNA, Step 6: cluster motifs.\n",
      "Using positive correlation.\n",
      "Doing all stats.\n",
      "Summary file has 0 DNA motifs and 10 RNA motifs.\n",
      "Read in RNA sequences and build motif profiles ...Done.\n",
      "Outputing MI matrix ... Done.\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t0\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[AGT]G[GT]G[AGT]G.[GT].\t0.0250\t0\t229.7973\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[CGT].[CT]CCCCC[ACT]\t0.0070\t0\t62.6175\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[CG][AG][ACT].GTG[CG][ACG]\t0.0104\t0\t98.1567\t0.1386\t2941\t0.2245\t2941\t0.2245\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t0\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.0281\t0\t254.9889\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t.[GT][ACT]CA[GT][AC]..\t0.0200\t0\t182.5732\t0.1517\t1897\t0.7163\t1897\t0.7163\n",
      ".[GT][ACT]CA[GT][AC]..\t[CG][AG][ACT].GTG[CG][ACG]\t0.0025\t0\t21.1851\t0.0659\t2526\t0.4954\t2526\t0.4954\n",
      ".[GT][ACT]CA[GT][AC]..\t0\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0059\t0\t52.7105\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\tGGAGG[AT]A[GT][AGT]\t0.0026\t0\t23.0743\t0.0589\t1990\t1.1417\t1990\t1.1417\n",
      "GGAGG[AT]A[GT][AGT]\t[CG][AG][ACT].GTG[CG][ACG]\t0.0009\t5\t6.8129\t0.0438\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[ACT]TCTCTCA.\t0.0015\t0\t12.4063\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t.[CG]GCC[AC]A.[ACG]\t0.0079\t0\t72.4243\t0.1187\t6963\t-0.5912\t6963\t-0.5912\n",
      ".[CG]GCC[AC]A.[ACG]\t[CG][AG][ACT].GTG[CG][ACG]\t0.0009\t9\t7.3260\t0.0444\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t1\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[CGT].[CT]CCCCC[ACT]\t0.0001\t2843\t0.1400\t0.0123\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.0007\t19\t6.1670\t0.0040\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0009\t4\t7.9129\t0.0074\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\tGGAGG[AT]A[GT][AGT]\t0.0007\t69\t5.8415\t0.0000\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[ACT]TCTCTCA.\t0.0002\t1886\t0.5795\t0.0079\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t.[CG]GCC[AC]A.[ACG]\t0.0031\t0\t28.4740\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t2\n",
      "[CGT].[CT]CCCCC[ACT]\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.0006\t67\t5.1690\t0.0149\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0001\t4923\t-0.1848\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\tGGAGG[AT]A[GT][AGT]\t0.0002\t2924\t1.1914\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t[ACT]TCTCTCA.\t0.0002\t3339\t0.9824\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t.[CG]GCC[AC]A.[ACG]\t0.0009\t32\t7.0777\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t3\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0000\t8112\t-0.6569\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\tGGAGG[AT]A[GT][AGT]\t0.0008\t50\t6.4601\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t[ACT]TCTCTCA.\t0.0019\t0\t15.3332\t0.0440\t10000\t0.0000\t10000\t0.0000\n",
      "[ACT]TCTCTCA.\t3\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t.[CG]GCC[AC]A.[ACG]\t0.0035\t0\t30.6598\t0.0000\tnan\tnan\tnan\tnan\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t4\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\tGGAGG[AT]A[GT][AGT]\t0.0002\t3386\t0.9019\t0.0000\tnan\tnan\tnan\tnan\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t.[CG]GCC[AC]A.[ACG]\t0.0007\t72\t5.9122\t0.0000\tnan\tnan\tnan\tnan\n",
      "GGAGG[AT]A[GT][AGT]\t5\n",
      "GGAGG[AT]A[GT][AGT]\t.[CG]GCC[AC]A.[ACG]\t0.0000\t6400\t-0.6496\t0.0046\tnan\tnan\tnan\tnan\n",
      ".[CG]GCC[AC]A.[ACG]\t6\n",
      "Finish calculated rank, z-scores for remaining entries ... [ACT].[ACT][ACT]GG[AT][ACG][ACG]\t[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t0.6765\t0\t6213.1363\t1.0000\t240\t2.5391\t240\t2.5391\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[AGT]G[GT]G[AGT]G.[GT].\t0.4160\t0\t3836.7599\t1.0000\t330\t2.3212\t330\t2.3212\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t[CG][AG][ACT].GTG[CG][ACG]\t0.0057\t0\t50.2295\t0.0000\tnan\tnan\tnan\tnan\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t.[GT][ACT]CA[GT][AC]..\t0.0207\t0\t190.8069\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t[CGT].[CT]CCCCC[ACT]\t0.1656\t0\t1475.5923\t1.0000\t3164\t0.5083\t3164\t0.5083\n",
      "[CGT].[CT]CCCCC[ACT]\t[CG][AG][ACT].GTG[CG][ACG]\t0.0016\t0\t13.6778\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CGT].[CT]CCCCC[ACT]\t.[GT][ACT]CA[GT][AC]..\t0.0058\t0\t52.0300\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t[CG][AG][ACT].GTG[CG][ACG]\t0.2597\t0\t2452.8707\t1.0000\t3176\t0.4756\t3176\t0.4756\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.0064\t0\t57.5484\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0014\t2\t11.4588\t0.0000\tnan\tnan\tnan\tnan\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t[ACT]TCTCTCA.\t0.0003\t1271\t2.1718\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t.[ACG].C[AG]C[AGT][AC][AGT]\t0.4489\t0\t3976.8975\t1.0000\t658\t1.7217\t658\t1.7217\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t.[GT][ACT]CA[GT][AC]..\t0.0233\t0\t223.5651\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[GT][ACT]CA[GT][AC]..\t.[GT][ACT]CA[GT][AC]..\t0.6111\t0\t5645.9196\t1.0000\t3514\t0.1667\t3514\t0.1667\n",
      ".[GT][ACT]CA[GT][AC]..\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.0049\t0\t43.7631\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[GT][ACT]CA[GT][AC]..\tGGAGG[AT]A[GT][AGT]\t0.0003\t540\t2.1355\t0.0168\tnan\tnan\tnan\tnan\n",
      ".[GT][ACT]CA[GT][AC]..\t[ACT]TCTCTCA.\t0.0012\t8\t9.7887\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[GT][ACT]CA[GT][AC]..\t.[CG]GCC[AC]A.[ACG]\t0.0006\t56\t5.1236\t0.0224\tnan\tnan\tnan\tnan\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t[AGT][GT]TT[GT]TT[CGT][CGT]\t0.1453\t0\t1254.5912\t1.0000\t8451\t-0.7616\t8451\t-0.7616\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t[ACT]TCTCTCA.\t0.0001\t6305\t-0.2851\t0.0010\tnan\tnan\tnan\tnan\n",
      "GGAGG[AT]A[GT][AGT]\tGGAGG[AT]A[GT][AGT]\t0.0519\t0\t813.7128\t1.0000\t10100\t-100.0000\t10100\t-100.0000\n",
      "GGAGG[AT]A[GT][AGT]\t[ACT]TCTCTCA.\t0.0000\t10000\t-0.3202\t0.0000\tnan\tnan\tnan\tnan\n",
      "[ACT]TCTCTCA.\t[ACT]TCTCTCA.\t0.0463\t0\t749.4929\t1.0000\t10100\t-100.0000\t10100\t-100.0000\n",
      "[ACT]TCTCTCA.\t.[CG]GCC[AC]A.[ACG]\t0.0002\t3302\t0.9838\t0.0000\tnan\tnan\tnan\tnan\n",
      ".[CG]GCC[AC]A.[ACG]\t.[CG]GCC[AC]A.[ACG]\t0.1662\t0\t1455.9965\t1.0000\t946\t2.0945\t946\t2.0945\n",
      "Done.\n",
      "RNA, Step 5.6: generate genome-wide GO enrichments for motifs in summary file.\n",
      "RNA, draw motif maps for co-localizing pairs\n",
      "Draw motif map for [ACT].[ACT][ACT]GG[AT][ACG][ACG] .. Done.\n",
      "Draw motif map for [AGT]G[GT]G[AGT]G.[GT]. .. Done.\n",
      "Draw motif map for [CGT].[CT]CCCCC[ACT] .. Done.\n",
      "Draw motif map for [CG][AG][ACT].GTG[CG][ACG] .. Done.\n",
      "Draw motif map for .[ACG].C[AG]C[AGT][AC][AGT] .. Done.\n",
      "Draw motif map for .[GT][ACT]CA[GT][AC].. .. Done.\n",
      "Draw motif map for [AGT][GT]TT[GT]TT[CGT][CGT] .. Done.\n",
      "Draw motif map for GGAGG[AT]A[GT][AGT] .. Done.\n",
      "Draw motif map for [ACT]TCTCTCA. .. Done.\n",
      "Draw motif map for .[CG]GCC[AC]A.[ACG] .. Done.\n",
      "RNA, Step 6.8: identifying motifs that match micrornas.\n",
      "RNA, Step 7: draw matrix figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 700, xbase = 35, ybase = 250\n",
      "Processing [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Outputing motif 0.eps ... Done.\n",
      "Processing [CG][AG][ACT].GTG[CG][ACG] ... Outputing motif 3.eps ... Done.\n",
      "Processing .[GT][ACT]CA[GT][AC].. ... Outputing motif 5.eps ... Done.\n",
      "Processing [AGT]G[GT]G[AGT]G.[GT]. ... Outputing motif 1.eps ... Done.\n",
      "Processing [CGT].[CT]CCCCC[ACT] ... Outputing motif 2.eps ... Done.\n",
      "Processing .[ACG].C[AG]C[AGT][AC][AGT] ... Outputing motif 4.eps ... Done.\n",
      "Processing [ACT]TCTCTCA. ... Outputing motif 8.eps ... Done.\n",
      "Processing [AGT][GT]TT[GT]TT[CGT][CGT] ... Outputing motif 6.eps ... Done.\n",
      "Processing GGAGG[AT]A[GT][AGT] ... Outputing motif 7.eps ... Done.\n",
      "Processing .[CG]GCC[AC]A.[ACG] ... Outputing motif 9.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Outputing motif 0.png ... \n",
      "Processing [CG][AG][ACT].GTG[CG][ACG] ... Outputing motif 3.png ... \n",
      "Processing .[GT][ACT]CA[GT][AC].. ... Outputing motif 5.png ... \n",
      "Processing [AGT]G[GT]G[AGT]G.[GT]. ... Outputing motif 1.png ... \n",
      "Processing [CGT].[CT]CCCCC[ACT] ... Outputing motif 2.png ... \n",
      "Processing .[ACG].C[AG]C[AGT][AC][AGT] ... Outputing motif 4.png ... \n",
      "Processing [ACT]TCTCTCA. ... Outputing motif 8.png ... \n",
      "Processing [AGT][GT]TT[GT]TT[CGT][CGT] ... Outputing motif 6.png ... \n",
      "Processing GGAGG[AT]A[GT][AGT] ... Outputing motif 7.png ... \n",
      "Processing .[CG]GCC[AC]A.[ACG] ... Outputing motif 9.png ... \n",
      "RNA, Step 7b: draw density figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 700, xbase = 35, ybase = 250\n",
      "Processing [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Outputing motif 0.eps ... Done.\n",
      "Processing [CG][AG][ACT].GTG[CG][ACG] ... Outputing motif 3.eps ... Done.\n",
      "Processing .[GT][ACT]CA[GT][AC].. ... Outputing motif 5.eps ... Done.\n",
      "Processing [AGT]G[GT]G[AGT]G.[GT]. ... Outputing motif 1.eps ... Done.\n",
      "Processing [CGT].[CT]CCCCC[ACT] ... Outputing motif 2.eps ... Done.\n",
      "Processing .[ACG].C[AG]C[AGT][AC][AGT] ... Outputing motif 4.eps ... Done.\n",
      "Processing [ACT]TCTCTCA. ... Outputing motif 8.eps ... Done.\n",
      "Processing [AGT][GT]TT[GT]TT[CGT][CGT] ... Outputing motif 6.eps ... Done.\n",
      "Processing GGAGG[AT]A[GT][AGT] ... Outputing motif 7.eps ... Done.\n",
      "Processing .[CG]GCC[AC]A.[ACG] ... Outputing motif 9.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [ACT].[ACT][ACT]GG[AT][ACG][ACG] ... Outputing motif 0.png ... \n",
      "Processing [CG][AG][ACT].GTG[CG][ACG] ... Outputing motif 3.png ... \n",
      "Processing .[GT][ACT]CA[GT][AC].. ... Outputing motif 5.png ... \n",
      "Processing [AGT]G[GT]G[AGT]G.[GT]. ... Outputing motif 1.png ... \n",
      "Processing [CGT].[CT]CCCCC[ACT] ... Outputing motif 2.png ... \n",
      "Processing .[ACG].C[AG]C[AGT][AC][AGT] ... Outputing motif 4.png ... \n",
      "Processing [ACT]TCTCTCA. ... Outputing motif 8.png ... \n",
      "Processing [AGT][GT]TT[GT]TT[CGT][CGT] ... Outputing motif 6.png ... \n",
      "Processing GGAGG[AT]A[GT][AGT] ... Outputing motif 7.png ... \n",
      "Processing .[CG]GCC[AC]A.[ACG] ... Outputing motif 9.png ... \n",
      "Parsing distance report file ...  Done.\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG] has no position or orientation bias.\n",
      "[AGT]G[GT]G[AGT]G.[GT]. has no position or orientation bias.\n",
      "[CGT].[CT]CCCCC[ACT] has no position or orientation bias.\n",
      ".[ACG].C[AG]C[AGT][AC][AGT] has no position or orientation bias.\n",
      ".[GT][ACT]CA[GT][AC].. has no position or orientation bias.\n",
      "[ACT]TCTCTCA. has no position or orientation bias.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 470, ysize = 550, xbase = 35, ybase = 115\n",
      "[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t0\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t3\n",
      ".[GT][ACT]CA[GT][AC]..\t5\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t1\n",
      "[CGT].[CT]CCCCC[ACT]\t2\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t4\n",
      "[ACT]TCTCTCA.\t8\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t6\n",
      "GGAGG[AT]A[GT][AGT]\t7\n",
      ".[CG]GCC[AC]A.[ACG]\t9\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf ... Done.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 410, ysize = 465, xbase = 75, ybase = 130\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.html[ACT].[ACT][ACT]GG[AT][ACG][ACG]\t0\n",
      "[CG][AG][ACT].GTG[CG][ACG]\t3\n",
      ".[GT][ACT]CA[GT][AC]..\t5\n",
      "[AGT]G[GT]G[AGT]G.[GT].\t1\n",
      "[CGT].[CT]CCCCC[ACT]\t2\n",
      ".[ACG].C[AG]C[AGT][AC][AGT]\t4\n",
      "[ACT]TCTCTCA.\t8\n",
      "[AGT][GT]TT[GT]TT[CGT][CGT]\t6\n",
      "GGAGG[AT]A[GT][AGT]\t7\n",
      ".[CG]GCC[AC]A.[ACG]\t9\n",
      "RNA, Step 10: motif report.\n",
      "Thu Aug  6 14:16:33 UTC 2020\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genregexp: malloc.c:2401: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "genregexp: malloc.c:2401: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Segmentation fault (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Segmentation fault (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /flash/bin/FIRE-1.1/SCRIPTS/draw_position_histogram.pl line 134.\n",
      "Can't locate Hypergeom.pm in @INC (you may need to install the Hypergeom module) (@INC contains: /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.12.4/darwin-thread-multi-2level/nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.10.0/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.18.2/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.22.0/x86_64-linux/Hypergeom.pm /flash/bin/FIRE-1.1/SCRIPTS /rumi/shams/abe/perl5/lib/perl5/x86_64-linux-gnu-thread-multi /rumi/shams/abe/perl5/lib/perl5 /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.26.1 /usr/local/share/perl/5.26.1 /usr/lib/x86_64-linux-gnu/perl5/5.26 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.26 /usr/share/perl/5.26 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /flash/bin/FIRE-1.1/SCRIPTS/GroupEnrichment.pm line 24.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "cd fire/\n",
    "# run FIRE --doskipdiscovery=0\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --oribiasonly=0\n",
    "mv peaks_teiser.txt_FIRE/ discovery_FIRE\n",
    "cd ../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'fire/discovery_FIRE/RNA/peaks_teiser.txt.densities.pdf' -> 'fire/discovery_FIRE_densities.pdf'\n",
      "renamed 'fire/discovery_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf' -> 'fire/discovery_FIRE_fullmimatrix.pdf'\n",
      "renamed 'fire/discovery_FIRE/RNA/peaks_teiser.txt.summary.pdf' -> 'fire/discovery_FIRE_summary.pdf'\n",
      "renamed 'fire/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.densities.pdf' -> 'fire/motifs_of_interest_FIRE_densities.pdf'\n",
      "renamed 'fire/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf' -> 'fire/motifs_of_interest_FIRE_fullmimatrix.pdf'\n",
      "renamed 'fire/motifs_of_interest_FIRE/RNA/peaks_teiser.txt.summary.pdf' -> 'fire/motifs_of_interest_FIRE_summary.pdf'\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in  fire/*/RNA/*.pdf; do \n",
    "    o=${f/\\/RNA\\/peaks_teiser.txt./_}\n",
    "    mv -v $f $o; \n",
    "done "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fire/discovery_FIRE_densities.pdf > fire/discovery_FIRE_densities.png\n",
      "done!\n",
      "fire/discovery_FIRE_fullmimatrix.pdf > fire/discovery_FIRE_fullmimatrix.png\n",
      "done!\n",
      "fire/discovery_FIRE_summary.pdf > fire/discovery_FIRE_summary.png\n",
      "done!\n",
      "fire/motifs_of_interest_FIRE_densities.pdf > fire/motifs_of_interest_FIRE_densities.png\n",
      "done!\n",
      "fire/motifs_of_interest_FIRE_fullmimatrix.pdf > fire/motifs_of_interest_FIRE_fullmimatrix.png\n",
      "done!\n",
      "fire/motifs_of_interest_FIRE_summary.pdf > fire/motifs_of_interest_FIRE_summary.png\n",
      "done!\n"
     ]
    }
   ],
   "source": [
    "!for f in fire/*pdf; do bash /rumi/shams/abe/Workflows/my_scripts/pdf2png.sh $f; done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DRACH & RGAC motifs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=fire/motifs_of_interest.png style=\"width:1100px\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Discover over/under represented motifs "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=fire/discovery_FIRE.png style=\"width:1100px\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# iPAGE\n",
    "https://github.com/goodarzilab/PAGE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nohup: ignoring input and redirecting stderr to stdout\r\n"
     ]
    }
   ],
   "source": [
    "!nohup bash ipage_human_ensembl.sh d_mtyl_T_vs_U.txt > ipage_human_ensembl.out "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_h.png\" title=\"msigdb_c1\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c1.png\" title=\"msigdb_c1\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c2.png\" title=\"msigdb_c2\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c3.png\" title=\"msigdb_c3\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c4.png\" title=\"msigdb_c4\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c5.png\" title=\"msigdb_c5\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c6.png\" title=\"msigdb_c6\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_c7.png\" title=\"msigdb_c7\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_msigdb_full.png\" title=\"msigdb_full\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl.png\" title=\"ensembl\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_5UTR.png\" title=\"RBPs_coding_gene_ids_by_5UTR\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_coding_exons.png\" title=\"RBPs_coding_gene_ids_by_coding_exons\" style=\"width:600px\">\n",
    "<img src=\"plots/d_mtyl_T_vs_U.ipage.human_ensembl_RBPs_coding_gene_ids_by_introns.png\" title=\"RBPs_coding_gene_ids_by_introns\" style=\"width:600px\">\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "UsageError: Cell magic `%%R` not found.\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:bedenv]",
   "language": "python",
   "name": "conda-env-bedenv-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
