{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rename bams "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'hl60-bam/120h_DMSO_rep1.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/120h_DMSO_rep1.bam'\n",
      "renamed 'hl60-bam/120h_DMSO_rep2.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/120h_DMSO_rep2.bam'\n",
      "renamed 'hl60-bam/120h_treated_rep1.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/120h_treated_rep1.bam'\n",
      "renamed 'hl60-bam/120h_treated_rep2.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/120h_treated_rep2.bam'\n",
      "renamed 'hl60-bam/6h_DMSO_rep1.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/6h_DMSO_rep1.bam'\n",
      "renamed 'hl60-bam/6h_DMSO_rep2.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/6h_DMSO_rep2.bam'\n",
      "renamed 'hl60-bam/6h_treated_rep1.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/6h_treated_rep1.bam'\n",
      "renamed 'hl60-bam/6h_treated_rep2.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/6h_treated_rep2.bam'\n",
      "renamed 'hl60-bam/72h_DMSO_rep1.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/72h_DMSO_rep1.bam'\n",
      "renamed 'hl60-bam/72h_DMSO_rep2.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/72h_DMSO_rep2.bam'\n",
      "renamed 'hl60-bam/72h_treated_rep1.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/72h_treated_rep1.bam'\n",
      "renamed 'hl60-bam/72h_treated_rep2.bamAligned.sortedByCoord.out.bam' -> 'hl60-bam/72h_treated_rep2.bam'\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "\n",
    "for f in hl60-bam/*.bam; do \n",
    "    o=${f/.bamAligned.sortedByCoord.out.bam/.bam};\n",
    "    mv -v $f $o; \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "fastq QC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for kg1_t_1.fastq.gz\n",
      "Analysis complete for kg1_t_2.fastq.gz\n",
      "Analysis complete for kg1_t_3.fastq.gz\n",
      "Analysis complete for kg1_u_1.fastq.gz\n",
      "Analysis complete for kg1_u_2.fastq.gz\n",
      "Analysis complete for kg1_u_3.fastq.gz\n",
      "Analysis complete for molm14_t_1.fastq.gz\n",
      "Analysis complete for molm14_t_2.fastq.gz\n",
      "Analysis complete for molm14_t_3.fastq.gz\n",
      "Analysis complete for molm14_u_1.fastq.gz\n",
      "Analysis complete for molm14_u_2.fastq.gz\n",
      "Analysis complete for molm14_u_3.fastq.gz\n",
      "Analysis complete for ociaml2_t_1.fastq.gz\n",
      "Analysis complete for ociaml2_t_2.fastq.gz\n",
      "Analysis complete for ociaml2_t_3.fastq.gz\n",
      "Analysis complete for ociaml2_u_1.fastq.gz\n",
      "Analysis complete for ociaml2_u_2.fastq.gz\n",
      "Analysis complete for ociaml2_u_3.fastq.gz\n",
      "Analysis complete for ociaml3_t_1.fastq.gz\n",
      "Analysis complete for ociaml3_t_2.fastq.gz\n",
      "Analysis complete for ociaml3_t_3.fastq.gz\n",
      "Analysis complete for ociaml3_u_1.fastq.gz\n",
      "Analysis complete for ociaml3_u_2.fastq.gz\n",
      "Analysis complete for ociaml3_u_3.fastq.gz\n",
      "Analysis complete for thp1_t_1.fastq.gz\n",
      "Analysis complete for thp1_t_2.fastq.gz\n",
      "Analysis complete for thp1_t_3.fastq.gz\n",
      "Analysis complete for thp1_u_1.fastq.gz\n",
      "Analysis complete for thp1_u_2.fastq.gz\n",
      "Analysis complete for thp1_u_3.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of kg1_t_1.fastq.gz\n",
      "Approx 5% complete for kg1_t_1.fastq.gz\n",
      "Approx 10% complete for kg1_t_1.fastq.gz\n",
      "Approx 15% complete for kg1_t_1.fastq.gz\n",
      "Approx 20% complete for kg1_t_1.fastq.gz\n",
      "Approx 25% complete for kg1_t_1.fastq.gz\n",
      "Approx 30% complete for kg1_t_1.fastq.gz\n",
      "Approx 35% complete for kg1_t_1.fastq.gz\n",
      "Approx 40% complete for kg1_t_1.fastq.gz\n",
      "Approx 45% complete for kg1_t_1.fastq.gz\n",
      "Approx 50% complete for kg1_t_1.fastq.gz\n",
      "Approx 55% complete for kg1_t_1.fastq.gz\n",
      "Approx 60% complete for kg1_t_1.fastq.gz\n",
      "Approx 65% complete for kg1_t_1.fastq.gz\n",
      "Approx 70% complete for kg1_t_1.fastq.gz\n",
      "Approx 75% complete for kg1_t_1.fastq.gz\n",
      "Approx 80% complete for kg1_t_1.fastq.gz\n",
      "Approx 85% complete for kg1_t_1.fastq.gz\n",
      "Approx 90% complete for kg1_t_1.fastq.gz\n",
      "Approx 95% complete for kg1_t_1.fastq.gz\n",
      "Started analysis of kg1_t_2.fastq.gz\n",
      "Approx 5% complete for kg1_t_2.fastq.gz\n",
      "Approx 10% complete for kg1_t_2.fastq.gz\n",
      "Approx 15% complete for kg1_t_2.fastq.gz\n",
      "Approx 20% complete for kg1_t_2.fastq.gz\n",
      "Approx 25% complete for kg1_t_2.fastq.gz\n",
      "Approx 30% complete for kg1_t_2.fastq.gz\n",
      "Approx 35% complete for kg1_t_2.fastq.gz\n",
      "Approx 40% complete for kg1_t_2.fastq.gz\n",
      "Approx 45% complete for kg1_t_2.fastq.gz\n",
      "Approx 50% complete for kg1_t_2.fastq.gz\n",
      "Approx 55% complete for kg1_t_2.fastq.gz\n",
      "Approx 60% complete for kg1_t_2.fastq.gz\n",
      "Approx 65% complete for kg1_t_2.fastq.gz\n",
      "Approx 70% complete for kg1_t_2.fastq.gz\n",
      "Approx 75% complete for kg1_t_2.fastq.gz\n",
      "Approx 80% complete for kg1_t_2.fastq.gz\n",
      "Approx 85% complete for kg1_t_2.fastq.gz\n",
      "Approx 90% complete for kg1_t_2.fastq.gz\n",
      "Approx 95% complete for kg1_t_2.fastq.gz\n",
      "Started analysis of kg1_t_3.fastq.gz\n",
      "Approx 5% complete for kg1_t_3.fastq.gz\n",
      "Approx 10% complete for kg1_t_3.fastq.gz\n",
      "Approx 15% complete for kg1_t_3.fastq.gz\n",
      "Approx 20% complete for kg1_t_3.fastq.gz\n",
      "Approx 25% complete for kg1_t_3.fastq.gz\n",
      "Approx 30% complete for kg1_t_3.fastq.gz\n",
      "Approx 35% complete for kg1_t_3.fastq.gz\n",
      "Approx 40% complete for kg1_t_3.fastq.gz\n",
      "Approx 45% complete for kg1_t_3.fastq.gz\n",
      "Approx 50% complete for kg1_t_3.fastq.gz\n",
      "Approx 55% complete for kg1_t_3.fastq.gz\n",
      "Approx 60% complete for kg1_t_3.fastq.gz\n",
      "Approx 65% complete for kg1_t_3.fastq.gz\n",
      "Approx 70% complete for kg1_t_3.fastq.gz\n",
      "Approx 75% complete for kg1_t_3.fastq.gz\n",
      "Approx 80% complete for kg1_t_3.fastq.gz\n",
      "Approx 85% complete for kg1_t_3.fastq.gz\n",
      "Approx 90% complete for kg1_t_3.fastq.gz\n",
      "Approx 95% complete for kg1_t_3.fastq.gz\n",
      "Started analysis of kg1_u_1.fastq.gz\n",
      "Approx 5% complete for kg1_u_1.fastq.gz\n",
      "Approx 10% complete for kg1_u_1.fastq.gz\n",
      "Approx 15% complete for kg1_u_1.fastq.gz\n",
      "Approx 20% complete for kg1_u_1.fastq.gz\n",
      "Approx 25% complete for kg1_u_1.fastq.gz\n",
      "Approx 30% complete for kg1_u_1.fastq.gz\n",
      "Approx 35% complete for kg1_u_1.fastq.gz\n",
      "Approx 40% complete for kg1_u_1.fastq.gz\n",
      "Approx 45% complete for kg1_u_1.fastq.gz\n",
      "Approx 50% complete for kg1_u_1.fastq.gz\n",
      "Approx 55% complete for kg1_u_1.fastq.gz\n",
      "Approx 60% complete for kg1_u_1.fastq.gz\n",
      "Approx 65% complete for kg1_u_1.fastq.gz\n",
      "Approx 70% complete for kg1_u_1.fastq.gz\n",
      "Approx 75% complete for kg1_u_1.fastq.gz\n",
      "Approx 80% complete for kg1_u_1.fastq.gz\n",
      "Approx 85% complete for kg1_u_1.fastq.gz\n",
      "Approx 90% complete for kg1_u_1.fastq.gz\n",
      "Approx 95% complete for kg1_u_1.fastq.gz\n",
      "Started analysis of kg1_u_2.fastq.gz\n",
      "Approx 5% complete for kg1_u_2.fastq.gz\n",
      "Approx 10% complete for kg1_u_2.fastq.gz\n",
      "Approx 15% complete for kg1_u_2.fastq.gz\n",
      "Approx 20% complete for kg1_u_2.fastq.gz\n",
      "Approx 25% complete for kg1_u_2.fastq.gz\n",
      "Approx 30% complete for kg1_u_2.fastq.gz\n",
      "Approx 35% complete for kg1_u_2.fastq.gz\n",
      "Approx 40% complete for kg1_u_2.fastq.gz\n",
      "Approx 45% complete for kg1_u_2.fastq.gz\n",
      "Approx 50% complete for kg1_u_2.fastq.gz\n",
      "Approx 55% complete for kg1_u_2.fastq.gz\n",
      "Approx 60% complete for kg1_u_2.fastq.gz\n",
      "Approx 65% complete for kg1_u_2.fastq.gz\n",
      "Approx 70% complete for kg1_u_2.fastq.gz\n",
      "Approx 75% complete for kg1_u_2.fastq.gz\n",
      "Approx 80% complete for kg1_u_2.fastq.gz\n",
      "Approx 85% complete for kg1_u_2.fastq.gz\n",
      "Approx 90% complete for kg1_u_2.fastq.gz\n",
      "Approx 95% complete for kg1_u_2.fastq.gz\n",
      "Started analysis of kg1_u_3.fastq.gz\n",
      "Approx 5% complete for kg1_u_3.fastq.gz\n",
      "Approx 10% complete for kg1_u_3.fastq.gz\n",
      "Approx 15% complete for kg1_u_3.fastq.gz\n",
      "Approx 20% complete for kg1_u_3.fastq.gz\n",
      "Approx 25% complete for kg1_u_3.fastq.gz\n",
      "Approx 30% complete for kg1_u_3.fastq.gz\n",
      "Approx 35% complete for kg1_u_3.fastq.gz\n",
      "Approx 40% complete for kg1_u_3.fastq.gz\n",
      "Approx 45% complete for kg1_u_3.fastq.gz\n",
      "Approx 50% complete for kg1_u_3.fastq.gz\n",
      "Approx 55% complete for kg1_u_3.fastq.gz\n",
      "Approx 60% complete for kg1_u_3.fastq.gz\n",
      "Approx 65% complete for kg1_u_3.fastq.gz\n",
      "Approx 70% complete for kg1_u_3.fastq.gz\n",
      "Approx 75% complete for kg1_u_3.fastq.gz\n",
      "Approx 80% complete for kg1_u_3.fastq.gz\n",
      "Approx 85% complete for kg1_u_3.fastq.gz\n",
      "Approx 90% complete for kg1_u_3.fastq.gz\n",
      "Approx 95% complete for kg1_u_3.fastq.gz\n",
      "Started analysis of molm14_t_1.fastq.gz\n",
      "Approx 5% complete for molm14_t_1.fastq.gz\n",
      "Approx 10% complete for molm14_t_1.fastq.gz\n",
      "Approx 15% complete for molm14_t_1.fastq.gz\n",
      "Approx 20% complete for molm14_t_1.fastq.gz\n",
      "Approx 25% complete for molm14_t_1.fastq.gz\n",
      "Approx 30% complete for molm14_t_1.fastq.gz\n",
      "Approx 35% complete for molm14_t_1.fastq.gz\n",
      "Approx 40% complete for molm14_t_1.fastq.gz\n",
      "Approx 45% complete for molm14_t_1.fastq.gz\n",
      "Approx 50% complete for molm14_t_1.fastq.gz\n",
      "Approx 55% complete for molm14_t_1.fastq.gz\n",
      "Approx 60% complete for molm14_t_1.fastq.gz\n",
      "Approx 65% complete for molm14_t_1.fastq.gz\n",
      "Approx 70% complete for molm14_t_1.fastq.gz\n",
      "Approx 75% complete for molm14_t_1.fastq.gz\n",
      "Approx 80% complete for molm14_t_1.fastq.gz\n",
      "Approx 85% complete for molm14_t_1.fastq.gz\n",
      "Approx 90% complete for molm14_t_1.fastq.gz\n",
      "Approx 95% complete for molm14_t_1.fastq.gz\n",
      "Started analysis of molm14_t_2.fastq.gz\n",
      "Approx 5% complete for molm14_t_2.fastq.gz\n",
      "Approx 10% complete for molm14_t_2.fastq.gz\n",
      "Approx 15% complete for molm14_t_2.fastq.gz\n",
      "Approx 20% complete for molm14_t_2.fastq.gz\n",
      "Approx 25% complete for molm14_t_2.fastq.gz\n",
      "Approx 30% complete for molm14_t_2.fastq.gz\n",
      "Approx 35% complete for molm14_t_2.fastq.gz\n",
      "Approx 40% complete for molm14_t_2.fastq.gz\n",
      "Approx 45% complete for molm14_t_2.fastq.gz\n",
      "Approx 50% complete for molm14_t_2.fastq.gz\n",
      "Approx 55% complete for molm14_t_2.fastq.gz\n",
      "Approx 60% complete for molm14_t_2.fastq.gz\n",
      "Approx 65% complete for molm14_t_2.fastq.gz\n",
      "Approx 70% complete for molm14_t_2.fastq.gz\n",
      "Approx 75% complete for molm14_t_2.fastq.gz\n",
      "Approx 80% complete for molm14_t_2.fastq.gz\n",
      "Approx 85% complete for molm14_t_2.fastq.gz\n",
      "Approx 90% complete for molm14_t_2.fastq.gz\n",
      "Approx 95% complete for molm14_t_2.fastq.gz\n",
      "Started analysis of molm14_t_3.fastq.gz\n",
      "Approx 5% complete for molm14_t_3.fastq.gz\n",
      "Approx 10% complete for molm14_t_3.fastq.gz\n",
      "Approx 15% complete for molm14_t_3.fastq.gz\n",
      "Approx 20% complete for molm14_t_3.fastq.gz\n",
      "Approx 25% complete for molm14_t_3.fastq.gz\n",
      "Approx 30% complete for molm14_t_3.fastq.gz\n",
      "Approx 35% complete for molm14_t_3.fastq.gz\n",
      "Approx 40% complete for molm14_t_3.fastq.gz\n",
      "Approx 45% complete for molm14_t_3.fastq.gz\n",
      "Approx 50% complete for molm14_t_3.fastq.gz\n",
      "Approx 55% complete for molm14_t_3.fastq.gz\n",
      "Approx 60% complete for molm14_t_3.fastq.gz\n",
      "Approx 65% complete for molm14_t_3.fastq.gz\n",
      "Approx 70% complete for molm14_t_3.fastq.gz\n",
      "Approx 75% complete for molm14_t_3.fastq.gz\n",
      "Approx 80% complete for molm14_t_3.fastq.gz\n",
      "Approx 85% complete for molm14_t_3.fastq.gz\n",
      "Approx 90% complete for molm14_t_3.fastq.gz\n",
      "Approx 95% complete for molm14_t_3.fastq.gz\n",
      "Started analysis of molm14_u_1.fastq.gz\n",
      "Approx 5% complete for molm14_u_1.fastq.gz\n",
      "Approx 10% complete for molm14_u_1.fastq.gz\n",
      "Approx 15% complete for molm14_u_1.fastq.gz\n",
      "Approx 20% complete for molm14_u_1.fastq.gz\n",
      "Approx 25% complete for molm14_u_1.fastq.gz\n",
      "Approx 30% complete for molm14_u_1.fastq.gz\n",
      "Approx 35% complete for molm14_u_1.fastq.gz\n",
      "Approx 40% complete for molm14_u_1.fastq.gz\n",
      "Approx 45% complete for molm14_u_1.fastq.gz\n",
      "Approx 50% complete for molm14_u_1.fastq.gz\n",
      "Approx 55% complete for molm14_u_1.fastq.gz\n",
      "Approx 60% complete for molm14_u_1.fastq.gz\n",
      "Approx 65% complete for molm14_u_1.fastq.gz\n",
      "Approx 70% complete for molm14_u_1.fastq.gz\n",
      "Approx 75% complete for molm14_u_1.fastq.gz\n",
      "Approx 80% complete for molm14_u_1.fastq.gz\n",
      "Approx 85% complete for molm14_u_1.fastq.gz\n",
      "Approx 90% complete for molm14_u_1.fastq.gz\n",
      "Approx 95% complete for molm14_u_1.fastq.gz\n",
      "Started analysis of molm14_u_2.fastq.gz\n",
      "Approx 5% complete for molm14_u_2.fastq.gz\n",
      "Approx 10% complete for molm14_u_2.fastq.gz\n",
      "Approx 15% complete for molm14_u_2.fastq.gz\n",
      "Approx 20% complete for molm14_u_2.fastq.gz\n",
      "Approx 25% complete for molm14_u_2.fastq.gz\n",
      "Approx 30% complete for molm14_u_2.fastq.gz\n",
      "Approx 35% complete for molm14_u_2.fastq.gz\n",
      "Approx 40% complete for molm14_u_2.fastq.gz\n",
      "Approx 45% complete for molm14_u_2.fastq.gz\n",
      "Approx 50% complete for molm14_u_2.fastq.gz\n",
      "Approx 55% complete for molm14_u_2.fastq.gz\n",
      "Approx 60% complete for molm14_u_2.fastq.gz\n",
      "Approx 65% complete for molm14_u_2.fastq.gz\n",
      "Approx 70% complete for molm14_u_2.fastq.gz\n",
      "Approx 75% complete for molm14_u_2.fastq.gz\n",
      "Approx 80% complete for molm14_u_2.fastq.gz\n",
      "Approx 85% complete for molm14_u_2.fastq.gz\n",
      "Approx 90% complete for molm14_u_2.fastq.gz\n",
      "Approx 95% complete for molm14_u_2.fastq.gz\n",
      "Started analysis of molm14_u_3.fastq.gz\n",
      "Approx 5% complete for molm14_u_3.fastq.gz\n",
      "Approx 10% complete for molm14_u_3.fastq.gz\n",
      "Approx 15% complete for molm14_u_3.fastq.gz\n",
      "Approx 20% complete for molm14_u_3.fastq.gz\n",
      "Approx 25% complete for molm14_u_3.fastq.gz\n",
      "Approx 30% complete for molm14_u_3.fastq.gz\n",
      "Approx 35% complete for molm14_u_3.fastq.gz\n",
      "Approx 40% complete for molm14_u_3.fastq.gz\n",
      "Approx 45% complete for molm14_u_3.fastq.gz\n",
      "Approx 50% complete for molm14_u_3.fastq.gz\n",
      "Approx 55% complete for molm14_u_3.fastq.gz\n",
      "Approx 60% complete for molm14_u_3.fastq.gz\n",
      "Approx 65% complete for molm14_u_3.fastq.gz\n",
      "Approx 70% complete for molm14_u_3.fastq.gz\n",
      "Approx 75% complete for molm14_u_3.fastq.gz\n",
      "Approx 80% complete for molm14_u_3.fastq.gz\n",
      "Approx 85% complete for molm14_u_3.fastq.gz\n",
      "Approx 90% complete for molm14_u_3.fastq.gz\n",
      "Approx 95% complete for molm14_u_3.fastq.gz\n",
      "Started analysis of ociaml2_t_1.fastq.gz\n",
      "Approx 5% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 10% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 15% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 20% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 25% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 30% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 35% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 40% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 45% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 50% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 55% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 60% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 65% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 70% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 75% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 80% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 85% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 90% complete for ociaml2_t_1.fastq.gz\n",
      "Approx 95% complete for ociaml2_t_1.fastq.gz\n",
      "Started analysis of ociaml2_t_2.fastq.gz\n",
      "Approx 5% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 10% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 15% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 20% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 25% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 30% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 35% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 40% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 45% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 50% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 55% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 60% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 65% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 70% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 75% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 80% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 85% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 90% complete for ociaml2_t_2.fastq.gz\n",
      "Approx 95% complete for ociaml2_t_2.fastq.gz\n",
      "Started analysis of ociaml2_t_3.fastq.gz\n",
      "Approx 5% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 10% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 15% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 20% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 25% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 30% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 35% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 40% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 45% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 50% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 55% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 60% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 65% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 70% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 75% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 80% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 85% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 90% complete for ociaml2_t_3.fastq.gz\n",
      "Approx 95% complete for ociaml2_t_3.fastq.gz\n",
      "Started analysis of ociaml2_u_1.fastq.gz\n",
      "Approx 5% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 10% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 15% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 20% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 25% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 30% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 35% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 40% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 45% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 50% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 55% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 60% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 65% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 70% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 75% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 80% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 85% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 90% complete for ociaml2_u_1.fastq.gz\n",
      "Approx 95% complete for ociaml2_u_1.fastq.gz\n",
      "Started analysis of ociaml2_u_2.fastq.gz\n",
      "Approx 5% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 10% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 15% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 20% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 25% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 30% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 35% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 40% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 45% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 50% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 55% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 60% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 65% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 70% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 75% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 80% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 85% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 90% complete for ociaml2_u_2.fastq.gz\n",
      "Approx 95% complete for ociaml2_u_2.fastq.gz\n",
      "Started analysis of ociaml2_u_3.fastq.gz\n",
      "Approx 5% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 10% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 15% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 20% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 25% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 30% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 35% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 40% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 45% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 50% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 55% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 60% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 65% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 70% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 75% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 80% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 85% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 90% complete for ociaml2_u_3.fastq.gz\n",
      "Approx 95% complete for ociaml2_u_3.fastq.gz\n",
      "Started analysis of ociaml3_t_1.fastq.gz\n",
      "Approx 5% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 10% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 15% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 20% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 25% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 30% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 35% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 40% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 45% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 50% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 55% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 60% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 65% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 70% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 75% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 80% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 85% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 90% complete for ociaml3_t_1.fastq.gz\n",
      "Approx 95% complete for ociaml3_t_1.fastq.gz\n",
      "Started analysis of ociaml3_t_2.fastq.gz\n",
      "Approx 5% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 10% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 15% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 20% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 25% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 30% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 35% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 40% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 45% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 50% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 55% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 60% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 65% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 70% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 75% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 80% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 85% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 90% complete for ociaml3_t_2.fastq.gz\n",
      "Approx 95% complete for ociaml3_t_2.fastq.gz\n",
      "Started analysis of ociaml3_t_3.fastq.gz\n",
      "Approx 5% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 10% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 15% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 20% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 25% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 30% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 35% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 40% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 45% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 50% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 55% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 60% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 65% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 70% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 75% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 80% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 85% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 90% complete for ociaml3_t_3.fastq.gz\n",
      "Approx 95% complete for ociaml3_t_3.fastq.gz\n",
      "Started analysis of ociaml3_u_1.fastq.gz\n",
      "Approx 5% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 10% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 15% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 20% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 25% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 30% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 35% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 40% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 45% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 50% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 55% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 60% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 65% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 70% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 75% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 80% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 85% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 90% complete for ociaml3_u_1.fastq.gz\n",
      "Approx 95% complete for ociaml3_u_1.fastq.gz\n",
      "Started analysis of ociaml3_u_2.fastq.gz\n",
      "Approx 5% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 10% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 15% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 20% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 25% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 30% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 35% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 40% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 45% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 50% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 55% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 60% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 65% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 70% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 75% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 80% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 85% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 90% complete for ociaml3_u_2.fastq.gz\n",
      "Approx 95% complete for ociaml3_u_2.fastq.gz\n",
      "Started analysis of ociaml3_u_3.fastq.gz\n",
      "Approx 5% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 10% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 15% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 20% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 25% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 30% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 35% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 40% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 45% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 50% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 55% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 60% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 65% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 70% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 75% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 80% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 85% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 90% complete for ociaml3_u_3.fastq.gz\n",
      "Approx 95% complete for ociaml3_u_3.fastq.gz\n",
      "Started analysis of thp1_t_1.fastq.gz\n",
      "Approx 5% complete for thp1_t_1.fastq.gz\n",
      "Approx 10% complete for thp1_t_1.fastq.gz\n",
      "Approx 15% complete for thp1_t_1.fastq.gz\n",
      "Approx 20% complete for thp1_t_1.fastq.gz\n",
      "Approx 25% complete for thp1_t_1.fastq.gz\n",
      "Approx 30% complete for thp1_t_1.fastq.gz\n",
      "Approx 35% complete for thp1_t_1.fastq.gz\n",
      "Approx 40% complete for thp1_t_1.fastq.gz\n",
      "Approx 45% complete for thp1_t_1.fastq.gz\n",
      "Approx 50% complete for thp1_t_1.fastq.gz\n",
      "Approx 55% complete for thp1_t_1.fastq.gz\n",
      "Approx 60% complete for thp1_t_1.fastq.gz\n",
      "Approx 65% complete for thp1_t_1.fastq.gz\n",
      "Approx 70% complete for thp1_t_1.fastq.gz\n",
      "Approx 75% complete for thp1_t_1.fastq.gz\n",
      "Approx 80% complete for thp1_t_1.fastq.gz\n",
      "Approx 85% complete for thp1_t_1.fastq.gz\n",
      "Approx 90% complete for thp1_t_1.fastq.gz\n",
      "Approx 95% complete for thp1_t_1.fastq.gz\n",
      "Started analysis of thp1_t_2.fastq.gz\n",
      "Approx 5% complete for thp1_t_2.fastq.gz\n",
      "Approx 10% complete for thp1_t_2.fastq.gz\n",
      "Approx 15% complete for thp1_t_2.fastq.gz\n",
      "Approx 20% complete for thp1_t_2.fastq.gz\n",
      "Approx 25% complete for thp1_t_2.fastq.gz\n",
      "Approx 30% complete for thp1_t_2.fastq.gz\n",
      "Approx 35% complete for thp1_t_2.fastq.gz\n",
      "Approx 40% complete for thp1_t_2.fastq.gz\n",
      "Approx 45% complete for thp1_t_2.fastq.gz\n",
      "Approx 50% complete for thp1_t_2.fastq.gz\n",
      "Approx 55% complete for thp1_t_2.fastq.gz\n",
      "Approx 60% complete for thp1_t_2.fastq.gz\n",
      "Approx 65% complete for thp1_t_2.fastq.gz\n",
      "Approx 70% complete for thp1_t_2.fastq.gz\n",
      "Approx 75% complete for thp1_t_2.fastq.gz\n",
      "Approx 80% complete for thp1_t_2.fastq.gz\n",
      "Approx 85% complete for thp1_t_2.fastq.gz\n",
      "Approx 90% complete for thp1_t_2.fastq.gz\n",
      "Approx 95% complete for thp1_t_2.fastq.gz\n",
      "Started analysis of thp1_t_3.fastq.gz\n",
      "Approx 5% complete for thp1_t_3.fastq.gz\n",
      "Approx 10% complete for thp1_t_3.fastq.gz\n",
      "Approx 15% complete for thp1_t_3.fastq.gz\n",
      "Approx 20% complete for thp1_t_3.fastq.gz\n",
      "Approx 25% complete for thp1_t_3.fastq.gz\n",
      "Approx 30% complete for thp1_t_3.fastq.gz\n",
      "Approx 35% complete for thp1_t_3.fastq.gz\n",
      "Approx 40% complete for thp1_t_3.fastq.gz\n",
      "Approx 45% complete for thp1_t_3.fastq.gz\n",
      "Approx 50% complete for thp1_t_3.fastq.gz\n",
      "Approx 55% complete for thp1_t_3.fastq.gz\n",
      "Approx 60% complete for thp1_t_3.fastq.gz\n",
      "Approx 65% complete for thp1_t_3.fastq.gz\n",
      "Approx 70% complete for thp1_t_3.fastq.gz\n",
      "Approx 75% complete for thp1_t_3.fastq.gz\n",
      "Approx 80% complete for thp1_t_3.fastq.gz\n",
      "Approx 85% complete for thp1_t_3.fastq.gz\n",
      "Approx 90% complete for thp1_t_3.fastq.gz\n",
      "Approx 95% complete for thp1_t_3.fastq.gz\n",
      "Started analysis of thp1_u_1.fastq.gz\n",
      "Approx 5% complete for thp1_u_1.fastq.gz\n",
      "Approx 10% complete for thp1_u_1.fastq.gz\n",
      "Approx 15% complete for thp1_u_1.fastq.gz\n",
      "Approx 20% complete for thp1_u_1.fastq.gz\n",
      "Approx 25% complete for thp1_u_1.fastq.gz\n",
      "Approx 30% complete for thp1_u_1.fastq.gz\n",
      "Approx 35% complete for thp1_u_1.fastq.gz\n",
      "Approx 40% complete for thp1_u_1.fastq.gz\n",
      "Approx 45% complete for thp1_u_1.fastq.gz\n",
      "Approx 50% complete for thp1_u_1.fastq.gz\n",
      "Approx 55% complete for thp1_u_1.fastq.gz\n",
      "Approx 60% complete for thp1_u_1.fastq.gz\n",
      "Approx 65% complete for thp1_u_1.fastq.gz\n",
      "Approx 70% complete for thp1_u_1.fastq.gz\n",
      "Approx 75% complete for thp1_u_1.fastq.gz\n",
      "Approx 80% complete for thp1_u_1.fastq.gz\n",
      "Approx 85% complete for thp1_u_1.fastq.gz\n",
      "Approx 90% complete for thp1_u_1.fastq.gz\n",
      "Approx 95% complete for thp1_u_1.fastq.gz\n",
      "Started analysis of thp1_u_2.fastq.gz\n",
      "Approx 5% complete for thp1_u_2.fastq.gz\n",
      "Approx 10% complete for thp1_u_2.fastq.gz\n",
      "Approx 15% complete for thp1_u_2.fastq.gz\n",
      "Approx 20% complete for thp1_u_2.fastq.gz\n",
      "Approx 25% complete for thp1_u_2.fastq.gz\n",
      "Approx 30% complete for thp1_u_2.fastq.gz\n",
      "Approx 35% complete for thp1_u_2.fastq.gz\n",
      "Approx 40% complete for thp1_u_2.fastq.gz\n",
      "Approx 45% complete for thp1_u_2.fastq.gz\n",
      "Approx 50% complete for thp1_u_2.fastq.gz\n",
      "Approx 55% complete for thp1_u_2.fastq.gz\n",
      "Approx 60% complete for thp1_u_2.fastq.gz\n",
      "Approx 65% complete for thp1_u_2.fastq.gz\n",
      "Approx 70% complete for thp1_u_2.fastq.gz\n",
      "Approx 75% complete for thp1_u_2.fastq.gz\n",
      "Approx 80% complete for thp1_u_2.fastq.gz\n",
      "Approx 85% complete for thp1_u_2.fastq.gz\n",
      "Approx 90% complete for thp1_u_2.fastq.gz\n",
      "Approx 95% complete for thp1_u_2.fastq.gz\n",
      "Started analysis of thp1_u_3.fastq.gz\n",
      "Approx 5% complete for thp1_u_3.fastq.gz\n",
      "Approx 10% complete for thp1_u_3.fastq.gz\n",
      "Approx 15% complete for thp1_u_3.fastq.gz\n",
      "Approx 20% complete for thp1_u_3.fastq.gz\n",
      "Approx 25% complete for thp1_u_3.fastq.gz\n",
      "Approx 30% complete for thp1_u_3.fastq.gz\n",
      "Approx 35% complete for thp1_u_3.fastq.gz\n",
      "Approx 40% complete for thp1_u_3.fastq.gz\n",
      "Approx 45% complete for thp1_u_3.fastq.gz\n",
      "Approx 50% complete for thp1_u_3.fastq.gz\n",
      "Approx 55% complete for thp1_u_3.fastq.gz\n",
      "Approx 60% complete for thp1_u_3.fastq.gz\n",
      "Approx 65% complete for thp1_u_3.fastq.gz\n",
      "Approx 70% complete for thp1_u_3.fastq.gz\n",
      "Approx 75% complete for thp1_u_3.fastq.gz\n",
      "Approx 80% complete for thp1_u_3.fastq.gz\n",
      "Approx 85% complete for thp1_u_3.fastq.gz\n",
      "Approx 90% complete for thp1_u_3.fastq.gz\n",
      "Approx 95% complete for thp1_u_3.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in other-fastq/*; do\n",
    "    fastqc $f\n",
    "done "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.9\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/other-fastq\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/other-bam\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/other-exp\n",
      "\u001b[?25lSearching 577 files..  [####################################]  100%          \u001b[?25h\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          salmon : Found 30 meta reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          salmon : Found 30 fragment length distributions\n",
      "\u001b[1;30m[INFO   ]\u001b[0m            star : Found 30 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 30 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : other-multiqc.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : other-multiqc_data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc other-fastq other-bam other-exp -n other-multiqc.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# hl-60"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis complete for 120h_DMSO_rep1.fastq.gz\n",
      "Analysis complete for 120h_DMSO_rep2.fastq.gz\n",
      "Analysis complete for 120h_treated_rep1.fastq.gz\n",
      "Analysis complete for 120h_treated_rep2.fastq.gz\n",
      "Analysis complete for 6h_DMSO_rep1.fastq.gz\n",
      "Analysis complete for 6h_DMSO_rep2.fastq.gz\n",
      "Analysis complete for 6h_treated_rep1.fastq.gz\n",
      "Analysis complete for 6h_treated_rep2.fastq.gz\n",
      "Analysis complete for 72h_DMSO_rep1.fastq.gz\n",
      "Analysis complete for 72h_DMSO_rep2.fastq.gz\n",
      "Analysis complete for 72h_treated_rep1.fastq.gz\n",
      "Analysis complete for 72h_treated_rep2.fastq.gz\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Started analysis of 120h_DMSO_rep1.fastq.gz\n",
      "Approx 5% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 10% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 15% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 20% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 25% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 30% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 35% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 40% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 45% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 50% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 55% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 60% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 65% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 70% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 75% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 80% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 85% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 90% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Approx 95% complete for 120h_DMSO_rep1.fastq.gz\n",
      "Started analysis of 120h_DMSO_rep2.fastq.gz\n",
      "Approx 5% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 10% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 15% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 20% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 25% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 30% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 35% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 40% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 45% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 50% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 55% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 60% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 65% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 70% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 75% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 80% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 85% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 90% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Approx 95% complete for 120h_DMSO_rep2.fastq.gz\n",
      "Started analysis of 120h_treated_rep1.fastq.gz\n",
      "Approx 5% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 10% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 15% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 20% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 25% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 30% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 35% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 40% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 45% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 50% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 55% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 60% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 65% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 70% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 75% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 80% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 85% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 90% complete for 120h_treated_rep1.fastq.gz\n",
      "Approx 95% complete for 120h_treated_rep1.fastq.gz\n",
      "Started analysis of 120h_treated_rep2.fastq.gz\n",
      "Approx 5% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 10% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 15% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 20% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 25% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 30% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 35% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 40% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 45% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 50% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 55% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 60% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 65% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 70% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 75% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 80% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 85% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 90% complete for 120h_treated_rep2.fastq.gz\n",
      "Approx 95% complete for 120h_treated_rep2.fastq.gz\n",
      "Started analysis of 6h_DMSO_rep1.fastq.gz\n",
      "Approx 5% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 10% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 15% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 20% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 25% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 30% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 35% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 40% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 45% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 50% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 55% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 60% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 65% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 70% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 75% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 80% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 85% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 90% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Approx 95% complete for 6h_DMSO_rep1.fastq.gz\n",
      "Started analysis of 6h_DMSO_rep2.fastq.gz\n",
      "Approx 5% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 10% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 15% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 20% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 25% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 30% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 35% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 40% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 45% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 50% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 55% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 60% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 65% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 70% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 75% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 80% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 85% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 90% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Approx 95% complete for 6h_DMSO_rep2.fastq.gz\n",
      "Started analysis of 6h_treated_rep1.fastq.gz\n",
      "Approx 5% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 10% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 15% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 20% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 25% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 30% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 35% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 40% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 45% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 50% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 55% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 60% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 65% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 70% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 75% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 80% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 85% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 90% complete for 6h_treated_rep1.fastq.gz\n",
      "Approx 95% complete for 6h_treated_rep1.fastq.gz\n",
      "Started analysis of 6h_treated_rep2.fastq.gz\n",
      "Approx 5% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 10% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 15% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 20% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 25% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 30% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 35% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 40% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 45% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 50% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 55% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 60% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 65% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 70% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 75% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 80% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 85% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 90% complete for 6h_treated_rep2.fastq.gz\n",
      "Approx 95% complete for 6h_treated_rep2.fastq.gz\n",
      "Started analysis of 72h_DMSO_rep1.fastq.gz\n",
      "Approx 5% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 10% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 15% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 20% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 25% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 30% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 35% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 40% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 45% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 50% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 55% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 60% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 65% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 70% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 75% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 80% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 85% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 90% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Approx 95% complete for 72h_DMSO_rep1.fastq.gz\n",
      "Started analysis of 72h_DMSO_rep2.fastq.gz\n",
      "Approx 5% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 10% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 15% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 20% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 25% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 30% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 35% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 40% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 45% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 50% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 55% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 60% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 65% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 70% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 75% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 80% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 85% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 90% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Approx 95% complete for 72h_DMSO_rep2.fastq.gz\n",
      "Started analysis of 72h_treated_rep1.fastq.gz\n",
      "Approx 5% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 10% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 15% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 20% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 25% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 30% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 35% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 40% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 45% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 50% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 55% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 60% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 65% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 70% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 75% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 80% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 85% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 90% complete for 72h_treated_rep1.fastq.gz\n",
      "Approx 95% complete for 72h_treated_rep1.fastq.gz\n",
      "Started analysis of 72h_treated_rep2.fastq.gz\n",
      "Approx 5% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 10% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 15% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 20% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 25% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 30% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 35% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 40% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 45% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 50% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 55% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 60% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 65% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 70% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 75% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 80% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 85% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 90% complete for 72h_treated_rep2.fastq.gz\n",
      "Approx 95% complete for 72h_treated_rep2.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in hl60-fastq/*; do\n",
    "    fastqc $f\n",
    "done "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.9\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-fastq\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-bam\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-exp\n",
      "\u001b[?25lSearching 4981 files..  [####################################]  100%          \u001b[?25h\n",
      "\u001b[1;30m[INFO   ]\u001b[0m  feature_counts : Found 1 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          salmon : Found 12 meta reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          salmon : Found 12 fragment length distributions\n",
      "\u001b[1;30m[INFO   ]\u001b[0m            star : Found 12 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          fastqc : Found 12 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : hl60-multiqc.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : hl60-multiqc_data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc hl60-fastq hl60-bam hl60-exp -n hl60-multiqc.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.9\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep1_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep1_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep1_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep2_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep2_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep2_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep1_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep1_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep1_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep2_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep2_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep2_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep1_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep1_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep1_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep2_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep2_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep2_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep1_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep1_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep1_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep2_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep2_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep2_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep1_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep1_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep1_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep2_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep2_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep2_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep1_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep1_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep1_exons.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep2_exons.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep2_exons.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep2_exons.txt.summary\n",
      "\u001b[?25lSearching 36 files..  [####################################]  100%          \u001b[?25h\n",
      "\u001b[1;30m[INFO   ]\u001b[0m  feature_counts : Found 12 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : hl60-stbl/counts-exons-multiqc.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : hl60-stbl/counts-exons-multiqc_data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.9\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep1_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep1_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep1_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep2_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep2_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_DMSO_rep2_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep1_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep1_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep1_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep2_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep2_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/120h_treated_rep2_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep1_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep1_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep1_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep2_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep2_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_DMSO_rep2_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep1_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep1_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep1_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep2_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep2_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/6h_treated_rep2_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep1_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep1_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep1_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep2_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep2_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_DMSO_rep2_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep1_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep1_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep1_Introns.txt.summary\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep2_Introns.ol.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep2_Introns.txt\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-stbl/counts/72h_treated_rep2_Introns.txt.summary\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?25lSearching 36 files..  [####################################]  100%          \u001b[?25h\n",
      "\u001b[1;30m[INFO   ]\u001b[0m  feature_counts : Found 12 reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : hl60-stbl/counts-introns-multiqc.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : hl60-stbl/counts-introns-multiqc_data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc hl60-stbl/counts/*_exons* -n hl60-stbl/counts-exons-multiqc.html\n",
    "!multiqc hl60-stbl/counts/*_Introns* -n hl60-stbl/counts-introns-multiqc.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : This is MultiQC v1.9\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Template    : default\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Searching   : /rumi/shams/abe/Gilbertlab/Decitabine_treatment/RNA-seq/hl60-exp/scallop\n",
      "\u001b[?25lSearching 340 files..  [####################################]  100%          \u001b[?25h\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          salmon : Found 12 meta reports\n",
      "\u001b[1;30m[INFO   ]\u001b[0m          salmon : Found 12 fragment length distributions\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Compressing plot data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Report      : hl60-exp/scallop-multiqc.html\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : Data        : hl60-exp/scallop-multiqc_data\n",
      "\u001b[1;30m[INFO   ]\u001b[0m         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc hl60-exp/scallop/ -n hl60-exp/scallop-multiqc.html"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:alignment]",
   "language": "python",
   "name": "conda-env-alignment-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
