{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "> We have already established that using annotated HeLa m6A sites, we can observe changes in genes with m6A sites (HL-60) cells. In order to confirm this m6A sites, we performed MeRIP-seq in treated and untreated cells, and did observe a general increase in m6A levels upon treatments for a large number of annotated sites. Here, our goal is to indpendently analyze the MeRIP data without relying on HeLa annotations and use it to define a **treatment-induced hyper-methylation sites**. We will then assess the location and behaviour of these targets across the other datasets generated in this study."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# meRIP-seq data "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Meta-gene plot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As a quality control, we confirm mRNA Methylation sites enriched in 3â€² UTRs and near stop codons same as this Cell paper \n",
    "https://doi.org/10.1016/j.cell.2012.05.003\n",
    "\n",
    "<img src=\"meRIP-seq/plots/Guitar_mrna_test.png\" title=\"DRACH and RGAC\" style=\"width:600px\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Motif analysis "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"meRIP-seq/fire/motifs_of_interest_FIRE_summary.png\" title=\"DRACH and RGAC\" style=\"width:1000px\">\n",
    "\n",
    "\n",
    "__Results from FIRE__ shown above indicate that the known m6A site **DRACH** and **RGAC** (or `[AG]GAC` and `[AGT][AG]AC[ACT]` in regular expression form) is significantly enriched among the **~4700** methylations sites."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test enrichment of treatment-induced hyper/hypo-methylation sites\n",
    "[TEISER](https://github.com/goodarzilab/TEISER)\n",
    "Here, I aim to identify the genes that are hyper or hypo methylated as genesets, and see if they have enriched accross all genes; the input table is list of genes with control vs. treated fold change of RNA expression, RNA stability and translational efficiency. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir -p combined_analysis/\n",
    "\n",
    "### hyper_methylation gene list\n",
    "# subset by threshold - logFC >= 2\n",
    "awk '$2>=2{print $1}' meRIP-seq/d_mtyl_T_vs_U.txt | \\\n",
    "awk '!seen[$1]++' > combined_analysis/hyper_mtyl.txt\n",
    "\n",
    "### hypo_methylation gene list\n",
    "# subset by threshold - logFC < 2\n",
    "awk '$2<-2 {print $1}' meRIP-seq/d_mtyl_T_vs_U.txt | \\\n",
    "awk '!seen[$1]++' > combined_analysis/hypo_mtyl.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hyper_mtyl.txt  hypo_mtyl.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls combined_analysis/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNA-seq/hl60-stbl/120h_delta_stbl.txt  RNA-seq/hl60-stbl/6h_delta_stbl.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls RNA-seq/hl60-stbl/*delta_stbl.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNA-seq/other-stbl/kg1_delta_stbl.txt\r\n",
      "RNA-seq/other-stbl/molm14_delta_stbl.txt\r\n",
      "RNA-seq/other-stbl/ociaml2_delta_stbl.txt\r\n",
      "RNA-seq/other-stbl/ociaml3_delta_stbl.txt\r\n",
      "RNA-seq/other-stbl/thp1_delta_stbl.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls RNA-seq/other-stbl/*delta_stbl.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Decitabine.Rmd              \u001b[0m\u001b[01;34mlnTE_T_vs_U\u001b[0m/    README.md\r\n",
      "fit1_deci_fdr_qval_g_0.csv  make_readme.sh  report.ipynb\r\n",
      "fit1_deci_fdr_qval_g.csv    \u001b[01;34mplots\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls Ribo-seq/Results/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "!source ~/.bashrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ribo-seq/lnTE_T_vs_U.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:17:01 UTC 2020\n",
      "Remove duplicates, create Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 6309 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001502\n",
      "pass = 0.041900\n",
      "z = 2.126055\n",
      "freq 0: 0.091010\t82\t901\t596\t6309\n",
      "freq 1: 0.081021\t73\t901\t596\t6309\n",
      "freq 2: 0.085461\t77\t901\t596\t6309\n",
      "freq 3: 0.086570\t78\t901\t596\t6309\n",
      "freq 4: 0.086570\t78\t901\t596\t6309\n",
      "freq 5: 0.112098\t101\t901\t596\t6309\n",
      "freq 6: 0.118494\t107\t903\t596\t6309\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.76\t-0.4\n",
      "0\n",
      "Outputing EPS file Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.eps\n",
      "Convert to PDF Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.eps Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.eps'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.profile'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.script'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.matrix'\n",
      "removed directory 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/Motifs'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.q'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.pre'\n",
      "removed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt'\n",
      "removed directory 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET'\n",
      "renamed 'Ribo-seq/lnTE_T_vs_U.txt_GENESET' -> 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:17:03 UTC 2020\n",
      "Remove duplicates, create Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 6309 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000604\n",
      "pass = 0.503000\n",
      "z = -0.228189\n",
      "freq 0: 0.047725\t43\t901\t329\t6309\n",
      "freq 1: 0.046615\t42\t901\t329\t6309\n",
      "freq 2: 0.053274\t48\t901\t329\t6309\n",
      "freq 3: 0.053274\t48\t901\t329\t6309\n",
      "freq 4: 0.052164\t47\t901\t329\t6309\n",
      "freq 5: 0.045505\t41\t901\t329\t6309\n",
      "freq 6: 0.066445\t60\t903\t329\t6309\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.76\t-0.4\n",
      "0\n",
      "Outputing EPS file Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.eps\n",
      "Convert to PDF Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.eps Ribo-seq/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt.script'\n",
      "removed directory 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET/lnTE_T_vs_U.txt'\n",
      "removed directory 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt_GENESET'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.script'\n",
      "removed directory 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary'\n",
      "removed directory 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET'\n",
      "renamed 'Ribo-seq/lnTE_T_vs_U.txt_GENESET' -> 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET'\n",
      "RNA-seq/hl60-exp/6h_delta_exp.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.874371072979637).\n",
      "Thu Dec 10 16:17:05 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.011873\n",
      "pass = 0.000000\n",
      "z = 286.411835\n",
      "freq 0: 0.012317\t106\t8606\t1208\t60239\n",
      "freq 1: 0.055543\t478\t8606\t1208\t60239\n",
      "freq 2: 0.004067\t35\t8606\t1208\t60239\n",
      "freq 3: 0.000581\t5\t8606\t1208\t60239\n",
      "freq 4: 0.033581\t289\t8606\t1208\t60239\n",
      "freq 5: 0.023588\t203\t8606\t1208\t60239\n",
      "freq 6: 0.010694\t92\t8603\t1208\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.06\t-0.76\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt.summary.eps RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET' -> 'combined_analysis/hyper_mtyl_6h_delta_exp_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.874371072979637).\n",
      "Thu Dec 10 16:17:19 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.005906\n",
      "pass = 0.000000\n",
      "z = 140.285660\n",
      "freq 0: 0.006275\t54\t8606\t570\t60239\n",
      "freq 1: 0.024983\t215\t8606\t570\t60239\n",
      "freq 2: 0.002092\t18\t8606\t570\t60239\n",
      "freq 3: 0.000000\t0\t8606\t570\t60239\n",
      "freq 4: 0.017546\t151\t8606\t570\t60239\n",
      "freq 5: 0.011736\t101\t8606\t570\t60239\n",
      "freq 6: 0.003603\t31\t8603\t570\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.06\t-0.76\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt.summary.eps RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET/6h_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.eps'\n",
      "removed directory 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary'\n",
      "removed directory 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/hl60-exp/6h_delta_exp.txt_GENESET' -> 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET'\n",
      "RNA-seq/hl60-exp/72h_delta_exp.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.874371072979637).\n",
      "Thu Dec 10 16:17:32 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.009121\n",
      "pass = 0.000000\n",
      "z = 217.247604\n",
      "freq 0: 0.021380\t184\t8606\t1208\t60239\n",
      "freq 1: 0.047525\t409\t8606\t1208\t60239\n",
      "freq 2: 0.003718\t32\t8606\t1208\t60239\n",
      "freq 3: 0.000581\t5\t8606\t1208\t60239\n",
      "freq 4: 0.030328\t261\t8606\t1208\t60239\n",
      "freq 5: 0.018824\t162\t8606\t1208\t60239\n",
      "freq 6: 0.018017\t155\t8603\t1208\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.28\t-1\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt.summary.eps RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET' -> 'combined_analysis/hyper_mtyl_72h_delta_exp_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.874371072979637).\n",
      "Thu Dec 10 16:17:46 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.004467\n",
      "pass = 0.000000\n",
      "z = 105.926483\n",
      "freq 0: 0.008018\t69\t8606\t570\t60239\n",
      "freq 1: 0.023821\t205\t8606\t570\t60239\n",
      "freq 2: 0.002208\t19\t8606\t570\t60239\n",
      "freq 3: 0.000116\t1\t8606\t570\t60239\n",
      "freq 4: 0.013595\t117\t8606\t570\t60239\n",
      "freq 5: 0.009993\t86\t8606\t570\t60239\n",
      "freq 6: 0.008485\t73\t8603\t570\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.28\t-1\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt.summary.eps RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET/72h_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed directory 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.script'\n",
      "removed directory 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/hl60-exp/72h_delta_exp.txt_GENESET' -> 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET'\n",
      "RNA-seq/hl60-exp/120h_delta_exp.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.874371072979637).\n",
      "Thu Dec 10 16:17:59 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.008813\n",
      "pass = 0.000000\n",
      "z = 214.356140\n",
      "freq 0: 0.023356\t201\t8606\t1208\t60239\n",
      "freq 1: 0.045782\t394\t8606\t1208\t60239\n",
      "freq 2: 0.003602\t31\t8606\t1208\t60239\n",
      "freq 3: 0.000581\t5\t8606\t1208\t60239\n",
      "freq 4: 0.029514\t254\t8606\t1208\t60239\n",
      "freq 5: 0.018708\t161\t8606\t1208\t60239\n",
      "freq 6: 0.018831\t162\t8603\t1208\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.36\t-1\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt.summary.eps RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET' -> 'combined_analysis/hyper_mtyl_120h_delta_exp_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.874371072979637).\n",
      "Thu Dec 10 16:18:13 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.004362\n",
      "pass = 0.000000\n",
      "z = 104.171501\n",
      "freq 0: 0.007785\t67\t8606\t570\t60239\n",
      "freq 1: 0.022891\t197\t8606\t570\t60239\n",
      "freq 2: 0.002092\t18\t8606\t570\t60239\n",
      "freq 3: 0.000116\t1\t8606\t570\t60239\n",
      "freq 4: 0.013944\t120\t8606\t570\t60239\n",
      "freq 5: 0.010458\t90\t8606\t570\t60239\n",
      "freq 6: 0.008950\t77\t8603\t570\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.36\t-1\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt.summary.eps RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET/120h_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.profile'\n",
      "removed directory 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/Motifs'\n",
      "removed directory 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/hl60-exp/120h_delta_exp.txt_GENESET' -> 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET'\n",
      "RNA-seq/hl60-stbl/120h_delta_stbl.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.0128942238963777).\n",
      "Thu Dec 10 16:18:26 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 4271 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.002417\n",
      "pass = 0.028200\n",
      "z = 2.357615\n",
      "freq 0: 0.104918\t64\t610\t423\t4271\n",
      "freq 1: 0.083607\t51\t610\t423\t4271\n",
      "freq 2: 0.101639\t62\t610\t423\t4271\n",
      "freq 3: 0.113115\t69\t610\t423\t4271\n",
      "freq 4: 0.121311\t74\t610\t423\t4271\n",
      "freq 5: 0.101639\t62\t610\t423\t4271\n",
      "freq 6: 0.067103\t41\t611\t423\t4271\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.3\t-0.48\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt.summary.eps RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET' -> 'combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.0128942238963777).\n",
      "Thu Dec 10 16:18:27 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 4271 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001208\n",
      "pass = 0.307000\n",
      "z = 0.332500\n",
      "freq 0: 0.049180\t30\t610\t239\t4271\n",
      "freq 1: 0.062295\t38\t610\t239\t4271\n",
      "freq 2: 0.063934\t39\t610\t239\t4271\n",
      "freq 3: 0.054098\t33\t610\t239\t4271\n",
      "freq 4: 0.068852\t42\t610\t239\t4271\n",
      "freq 5: 0.054098\t33\t610\t239\t4271\n",
      "freq 6: 0.039280\t24\t611\t239\t4271\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.3\t-0.48\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt.summary.eps RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET/120h_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt'\n",
      "removed directory 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/cmdline.txt'\n",
      "removed directory 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET'\n",
      "renamed 'RNA-seq/hl60-stbl/120h_delta_stbl.txt_GENESET' -> 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET'\n",
      "RNA-seq/hl60-stbl/6h_delta_stbl.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.0892799734926366).\n",
      "Thu Dec 10 16:18:29 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 4271 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001711\n",
      "pass = 0.116400\n",
      "z = 1.188418\n",
      "freq 0: 0.104918\t64\t610\t423\t4271\n",
      "freq 1: 0.093443\t57\t610\t423\t4271\n",
      "freq 2: 0.126230\t77\t610\t423\t4271\n",
      "freq 3: 0.093443\t57\t610\t423\t4271\n",
      "freq 4: 0.106557\t65\t610\t423\t4271\n",
      "freq 5: 0.093443\t57\t610\t423\t4271\n",
      "freq 6: 0.075286\t46\t611\t423\t4271\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.26\t-0.32\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt.summary.eps RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET' -> 'combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.0892799734926366).\n",
      "Thu Dec 10 16:18:31 UTC 2020\n",
      "Remove duplicates, create RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 4271 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001710\n",
      "pass = 0.116100\n",
      "z = 1.173816\n",
      "freq 0: 0.065574\t40\t610\t239\t4271\n",
      "freq 1: 0.073770\t45\t610\t239\t4271\n",
      "freq 2: 0.063934\t39\t610\t239\t4271\n",
      "freq 3: 0.047541\t29\t610\t239\t4271\n",
      "freq 4: 0.050820\t31\t610\t239\t4271\n",
      "freq 5: 0.050820\t31\t610\t239\t4271\n",
      "freq 6: 0.039280\t24\t611\t239\t4271\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.26\t-0.32\n",
      "0\n",
      "Outputing EPS file RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt.summary.eps RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET/6h_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf'\n",
      "removed directory 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt'\n",
      "removed directory 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET'\n",
      "renamed 'RNA-seq/hl60-stbl/6h_delta_stbl.txt_GENESET' -> 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET'\n",
      "RNA-seq/other-exp/kg1_delta_exp.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.472544886114928).\n",
      "Thu Dec 10 16:18:32 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.016129\n",
      "pass = 0.000000\n",
      "z = 390.032257\n",
      "freq 0: 0.012317\t106\t8606\t1208\t60239\n",
      "freq 1: 0.054729\t471\t8606\t1208\t60239\n",
      "freq 2: 0.003835\t33\t8606\t1208\t60239\n",
      "freq 3: 0.000697\t6\t8606\t1208\t60239\n",
      "freq 4: 0.003835\t33\t8606\t1208\t60239\n",
      "freq 5: 0.051127\t440\t8606\t1208\t60239\n",
      "freq 6: 0.013832\t119\t8603\t1208\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.1\t-0.9\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt.summary.eps RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/cmdline.txt'\n",
      "removed directory 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/Motifs'\n",
      "removed 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.script'\n",
      "removed 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.pre'\n",
      "removed 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt'\n",
      "removed directory 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/other-exp/kg1_delta_exp.txt_GENESET' -> 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.472544886114928).\n",
      "Thu Dec 10 16:18:46 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.009312\n",
      "pass = 0.000000\n",
      "z = 217.226715\n",
      "freq 0: 0.004067\t35\t8606\t570\t60239\n",
      "freq 1: 0.028004\t241\t8606\t570\t60239\n",
      "freq 2: 0.001511\t13\t8606\t570\t60239\n",
      "freq 3: 0.000349\t3\t8606\t570\t60239\n",
      "freq 4: 0.000930\t8\t8606\t570\t60239\n",
      "freq 5: 0.026377\t227\t8606\t570\t60239\n",
      "freq 6: 0.004998\t43\t8603\t570\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.1\t-0.9\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt.summary.eps RNA-seq/other-exp/kg1_delta_exp.txt_GENESET/kg1_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.profile'\n",
      "removed directory 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt'\n",
      "removed directory 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/other-exp/kg1_delta_exp.txt_GENESET' -> 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET'\n",
      "RNA-seq/other-exp/molm14_delta_exp.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.0151001620633903).\n",
      "Thu Dec 10 16:18:59 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.017403\n",
      "pass = 0.000000\n",
      "z = 415.330780\n",
      "freq 0: 0.018011\t155\t8606\t1208\t60239\n",
      "freq 1: 0.051360\t442\t8606\t1208\t60239\n",
      "freq 2: 0.000930\t8\t8606\t1208\t60239\n",
      "freq 3: 0.001046\t9\t8606\t1208\t60239\n",
      "freq 4: 0.001743\t15\t8606\t1208\t60239\n",
      "freq 5: 0.052986\t456\t8606\t1208\t60239\n",
      "freq 6: 0.014297\t123\t8603\t1208\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.2\t-0.92\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt.summary.eps RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-exp/molm14_delta_exp.txt_GENESET' -> 'combined_analysis/hyper_mtyl_molm14_delta_exp_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.0151001620633903).\n",
      "Thu Dec 10 16:19:13 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.008988\n",
      "pass = 0.000000\n",
      "z = 214.750580\n",
      "freq 0: 0.007320\t63\t8606\t570\t60239\n",
      "freq 1: 0.025215\t217\t8606\t570\t60239\n",
      "freq 2: 0.000349\t3\t8606\t570\t60239\n",
      "freq 3: 0.000116\t1\t8606\t570\t60239\n",
      "freq 4: 0.000581\t5\t8606\t570\t60239\n",
      "freq 5: 0.025796\t222\t8606\t570\t60239\n",
      "freq 6: 0.006858\t59\t8603\t570\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.2\t-0.92\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt.summary.eps RNA-seq/other-exp/molm14_delta_exp.txt_GENESET/molm14_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.pdf'\n",
      "removed directory 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.matrix'\n",
      "removed directory 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/other-exp/molm14_delta_exp.txt_GENESET' -> 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET'\n",
      "RNA-seq/other-exp/ociaml2_delta_exp.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:19:25 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.020551\n",
      "pass = 0.000000\n",
      "z = 493.394806\n",
      "freq 0: 0.013247\t114\t8606\t1208\t60239\n",
      "freq 1: 0.061701\t531\t8606\t1208\t60239\n",
      "freq 2: 0.000813\t7\t8606\t1208\t60239\n",
      "freq 3: 0.000465\t4\t8606\t1208\t60239\n",
      "freq 4: 0.000697\t6\t8606\t1208\t60239\n",
      "freq 5: 0.050198\t432\t8606\t1208\t60239\n",
      "freq 6: 0.013251\t114\t8603\t1208\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.8\t-0.72\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt.summary.eps RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_exp_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:19:39 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.010285\n",
      "pass = 0.000000\n",
      "z = 245.077881\n",
      "freq 0: 0.004997\t43\t8606\t570\t60239\n",
      "freq 1: 0.032303\t278\t8606\t570\t60239\n",
      "freq 2: 0.000232\t2\t8606\t570\t60239\n",
      "freq 3: 0.000116\t1\t8606\t570\t60239\n",
      "freq 4: 0.000581\t5\t8606\t570\t60239\n",
      "freq 5: 0.022078\t190\t8606\t570\t60239\n",
      "freq 6: 0.005928\t51\t8603\t570\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.8\t-0.72\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt.summary.eps RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/cmdline.txt'\n",
      "removed directory 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt'\n",
      "removed directory 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/other-exp/ociaml2_delta_exp.txt_GENESET' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET'\n",
      "RNA-seq/other-exp/ociaml3_delta_exp.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:19:52 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.016333\n",
      "pass = 0.000000\n",
      "z = 385.536682\n",
      "freq 0: 0.013828\t119\t8606\t1208\t60239\n",
      "freq 1: 0.058912\t507\t8606\t1208\t60239\n",
      "freq 2: 0.000116\t1\t8606\t1208\t60239\n",
      "freq 3: 0.001046\t9\t8606\t1208\t60239\n",
      "freq 4: 0.009180\t79\t8606\t1208\t60239\n",
      "freq 5: 0.044852\t386\t8606\t1208\t60239\n",
      "freq 6: 0.012438\t107\t8603\t1208\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.36\t-0.86\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt.summary.eps RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_exp_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:05 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.008632\n",
      "pass = 0.000000\n",
      "z = 206.069855\n",
      "freq 0: 0.006042\t52\t8606\t570\t60239\n",
      "freq 1: 0.030909\t266\t8606\t570\t60239\n",
      "freq 2: 0.000116\t1\t8606\t570\t60239\n",
      "freq 3: 0.000116\t1\t8606\t570\t60239\n",
      "freq 4: 0.003602\t31\t8606\t570\t60239\n",
      "freq 5: 0.020102\t173\t8606\t570\t60239\n",
      "freq 6: 0.005347\t46\t8603\t570\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.36\t-0.86\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt.summary.eps RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET/ociaml3_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.pre'\n",
      "removed directory 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/Motifs'\n",
      "removed directory 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/other-exp/ociaml3_delta_exp.txt_GENESET' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET'\n",
      "RNA-seq/other-exp/thp1_delta_exp.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.280321722159322).\n",
      "Thu Dec 10 16:20:19 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.014817\n",
      "pass = 0.000000\n",
      "z = 354.066742\n",
      "freq 0: 0.009993\t86\t8606\t1208\t60239\n",
      "freq 1: 0.051592\t444\t8606\t1208\t60239\n",
      "freq 2: 0.008599\t74\t8606\t1208\t60239\n",
      "freq 3: 0.000465\t4\t8606\t1208\t60239\n",
      "freq 4: 0.003370\t29\t8606\t1208\t60239\n",
      "freq 5: 0.051127\t440\t8606\t1208\t60239\n",
      "freq 6: 0.015227\t131\t8603\t1208\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.72\t-1.02\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt.summary.eps RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-exp/thp1_delta_exp.txt_GENESET' -> 'combined_analysis/hyper_mtyl_thp1_delta_exp_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.280321722159322).\n",
      "Thu Dec 10 16:20:32 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 60239 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.007519\n",
      "pass = 0.000000\n",
      "z = 180.205811\n",
      "freq 0: 0.003718\t32\t8606\t570\t60239\n",
      "freq 1: 0.024634\t212\t8606\t570\t60239\n",
      "freq 2: 0.003835\t33\t8606\t570\t60239\n",
      "freq 3: 0.000232\t2\t8606\t570\t60239\n",
      "freq 4: 0.001743\t15\t8606\t570\t60239\n",
      "freq 5: 0.025796\t222\t8606\t570\t60239\n",
      "freq 6: 0.006277\t54\t8603\t570\t60239\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.72\t-1.02\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt.summary.eps RNA-seq/other-exp/thp1_delta_exp.txt_GENESET/thp1_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary'\n",
      "removed directory 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.pre'\n",
      "removed directory 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET'\n",
      "renamed 'RNA-seq/other-exp/thp1_delta_exp.txt_GENESET' -> 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET'\n",
      "RNA-seq/other-stbl/kg1_delta_stbl.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.0379691762535842).\n",
      "Thu Dec 10 16:20:45 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001902\n",
      "pass = 0.408100\n",
      "z = 0.018183\n",
      "freq 0: 0.108434\t36\t332\t253\t2324\n",
      "freq 1: 0.102410\t34\t332\t253\t2324\n",
      "freq 2: 0.090361\t30\t332\t253\t2324\n",
      "freq 3: 0.090361\t30\t332\t253\t2324\n",
      "freq 4: 0.123494\t41\t332\t253\t2324\n",
      "freq 5: 0.138554\t46\t332\t253\t2324\n",
      "freq 6: 0.108434\t36\t332\t253\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.14\t-1.42\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt.summary.eps RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET' -> 'combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "WARNING: your file might not contain a header line (0.0379691762535842).\n",
      "Thu Dec 10 16:20:47 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001473\n",
      "pass = 0.585500\n",
      "z = -0.382990\n",
      "freq 0: 0.078313\t26\t332\t154\t2324\n",
      "freq 1: 0.066265\t22\t332\t154\t2324\n",
      "freq 2: 0.045181\t15\t332\t154\t2324\n",
      "freq 3: 0.060241\t20\t332\t154\t2324\n",
      "freq 4: 0.081325\t27\t332\t154\t2324\n",
      "freq 5: 0.066265\t22\t332\t154\t2324\n",
      "freq 6: 0.066265\t22\t332\t154\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.14\t-1.42\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt.summary.eps RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET/kg1_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "removed directory 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary'\n",
      "removed directory 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET'\n",
      "renamed 'RNA-seq/other-stbl/kg1_delta_stbl.txt_GENESET' -> 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET'\n",
      "RNA-seq/other-stbl/molm14_delta_stbl.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:48 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001000\n",
      "pass = 0.783800\n",
      "z = -0.811569\n",
      "freq 0: 0.117470\t39\t332\t253\t2324\n",
      "freq 1: 0.090361\t30\t332\t253\t2324\n",
      "freq 2: 0.126506\t42\t332\t253\t2324\n",
      "freq 3: 0.111446\t37\t332\t253\t2324\n",
      "freq 4: 0.114458\t38\t332\t253\t2324\n",
      "freq 5: 0.105422\t35\t332\t253\t2324\n",
      "freq 6: 0.096386\t32\t332\t253\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.64\t-1.66\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt.summary.eps RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET' -> 'combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:50 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001073\n",
      "pass = 0.759400\n",
      "z = -0.729699\n",
      "freq 0: 0.066265\t22\t332\t154\t2324\n",
      "freq 1: 0.060241\t20\t332\t154\t2324\n",
      "freq 2: 0.066265\t22\t332\t154\t2324\n",
      "freq 3: 0.069277\t23\t332\t154\t2324\n",
      "freq 4: 0.054217\t18\t332\t154\t2324\n",
      "freq 5: 0.060241\t20\t332\t154\t2324\n",
      "freq 6: 0.087349\t29\t332\t154\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.64\t-1.66\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt.summary.eps RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET/molm14_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary'\n",
      "removed directory 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf'\n",
      "removed directory 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET'\n",
      "renamed 'RNA-seq/other-stbl/molm14_delta_stbl.txt_GENESET' -> 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET'\n",
      "RNA-seq/other-stbl/ociaml2_delta_stbl.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:51 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001190\n",
      "pass = 0.700500\n",
      "z = -0.650355\n",
      "freq 0: 0.102410\t34\t332\t253\t2324\n",
      "freq 1: 0.090361\t30\t332\t253\t2324\n",
      "freq 2: 0.105422\t35\t332\t253\t2324\n",
      "freq 3: 0.120482\t40\t332\t253\t2324\n",
      "freq 4: 0.123494\t41\t332\t253\t2324\n",
      "freq 5: 0.123494\t41\t332\t253\t2324\n",
      "freq 6: 0.096386\t32\t332\t253\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.32\t-1.52\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt.summary.eps RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:53 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.001113\n",
      "pass = 0.740400\n",
      "z = -0.711295\n",
      "freq 0: 0.060241\t20\t332\t154\t2324\n",
      "freq 1: 0.054217\t18\t332\t154\t2324\n",
      "freq 2: 0.078313\t26\t332\t154\t2324\n",
      "freq 3: 0.078313\t26\t332\t154\t2324\n",
      "freq 4: 0.075301\t25\t332\t154\t2324\n",
      "freq 5: 0.057229\t19\t332\t154\t2324\n",
      "freq 6: 0.060241\t20\t332\t154\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.32\t-1.52\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt.summary.eps RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET/ociaml2_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf'\n",
      "removed directory 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/Motifs'\n",
      "removed directory 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET'\n",
      "renamed 'RNA-seq/other-stbl/ociaml2_delta_stbl.txt_GENESET' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET'\n",
      "RNA-seq/other-stbl/ociaml3_delta_stbl.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:54 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.003553\n",
      "pass = 0.073600\n",
      "z = 1.567256\n",
      "freq 0: 0.087349\t29\t332\t253\t2324\n",
      "freq 1: 0.150602\t50\t332\t253\t2324\n",
      "freq 2: 0.090361\t30\t332\t253\t2324\n",
      "freq 3: 0.093373\t31\t332\t253\t2324\n",
      "freq 4: 0.123494\t41\t332\t253\t2324\n",
      "freq 5: 0.093373\t31\t332\t253\t2324\n",
      "freq 6: 0.123494\t41\t332\t253\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.48\t-0.96\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt.summary.eps RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:55 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000974\n",
      "pass = 0.802100\n",
      "z = -0.839975\n",
      "freq 0: 0.054217\t18\t332\t154\t2324\n",
      "freq 1: 0.084337\t28\t332\t154\t2324\n",
      "freq 2: 0.060241\t20\t332\t154\t2324\n",
      "freq 3: 0.072289\t24\t332\t154\t2324\n",
      "freq 4: 0.063253\t21\t332\t154\t2324\n",
      "freq 5: 0.060241\t20\t332\t154\t2324\n",
      "freq 6: 0.069277\t23\t332\t154\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.48\t-0.96\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt.summary.eps RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET/ociaml3_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "removed directory 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.matrix'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.pre'\n",
      "removed directory 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET'\n",
      "renamed 'RNA-seq/other-stbl/ociaml3_delta_stbl.txt_GENESET' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET'\n",
      "RNA-seq/other-stbl/thp1_delta_stbl.txt\n",
      "hyper_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:57 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.003260\n",
      "pass = 0.111500\n",
      "z = 1.285587\n",
      "freq 0: 0.093373\t31\t332\t253\t2324\n",
      "freq 1: 0.087349\t29\t332\t253\t2324\n",
      "freq 2: 0.129518\t43\t332\t253\t2324\n",
      "freq 3: 0.093373\t31\t332\t253\t2324\n",
      "freq 4: 0.150602\t50\t332\t253\t2324\n",
      "freq 5: 0.108434\t36\t332\t253\t2324\n",
      "freq 6: 0.099398\t33\t332\t253\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.06\t-1.26\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt.summary.eps RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "renamed 'RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET' -> 'combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET'\n",
      "hypo_methylation\n",
      "The TEISERDIR environment variable is /rumi/shams/abe/TEISERv1.1\n",
      "Thu Dec 10 16:20:58 UTC 2020\n",
      "Remove duplicates, create RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt\n",
      "Quantizing the input file.\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 2324 values...\n",
      "Adding small values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 7\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.002475\n",
      "pass = 0.248800\n",
      "z = 0.570503\n",
      "freq 0: 0.084337\t28\t332\t154\t2324\n",
      "freq 1: 0.054217\t18\t332\t154\t2324\n",
      "freq 2: 0.057229\t19\t332\t154\t2324\n",
      "freq 3: 0.081325\t27\t332\t154\t2324\n",
      "freq 4: 0.072289\t24\t332\t154\t2324\n",
      "freq 5: 0.042169\t14\t332\t154\t2324\n",
      "freq 6: 0.072289\t24\t332\t154\t2324\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.06\t-1.26\n",
      "0\n",
      "Outputing EPS file RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt.summary.eps\n",
      "Convert to PDF RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt.summary.eps RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET/thp1_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.script'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.eps'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.pre'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.matrix'\n",
      "removed directory 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/Motifs'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.profile'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.q'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/cmdline.txt'\n",
      "removed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary'\n",
      "removed directory 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET'\n",
      "renamed 'RNA-seq/other-stbl/thp1_delta_stbl.txt_GENESET' -> 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_6h_delta_exp_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_72h_delta_exp_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_120h_delta_exp_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_molm14_delta_exp_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_ociaml2_delta_exp_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_ociaml3_delta_exp_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_thp1_delta_exp_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "rm: cannot remove 'combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET': No such file or directory\n",
      "Option suffix requires an argument\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "declare -a Experiments=(\n",
    "# Ribo-seq\n",
    "'Ribo-seq/lnTE_T_vs_U.txt' \n",
    "\n",
    "## HL-60 RNA-seq \n",
    "# RNA experssion \n",
    "'RNA-seq/hl60-exp/6h_delta_exp.txt' 'RNA-seq/hl60-exp/72h_delta_exp.txt' 'RNA-seq/hl60-exp/120h_delta_exp.txt' \n",
    "# RNA stability  \n",
    "'RNA-seq/hl60-stbl/120h_delta_stbl.txt'  'RNA-seq/hl60-stbl/6h_delta_stbl.txt'\n",
    "\n",
    "## 5 other AML cell lines RNA-seq\n",
    "# RNA experssion \n",
    "'RNA-seq/other-exp/kg1_delta_exp.txt' 'RNA-seq/other-exp/molm14_delta_exp.txt'\n",
    "'RNA-seq/other-exp/ociaml2_delta_exp.txt' 'RNA-seq/other-exp/ociaml3_delta_exp.txt'\n",
    "'RNA-seq/other-exp/thp1_delta_exp.txt'\n",
    "# RNA stability  \n",
    "'RNA-seq/other-stbl/kg1_delta_stbl.txt' 'RNA-seq/other-stbl/molm14_delta_stbl.txt' \n",
    "'RNA-seq/other-stbl/ociaml2_delta_stbl.txt' 'RNA-seq/other-stbl/ociaml3_delta_stbl.txt'\n",
    "'RNA-seq/other-stbl/thp1_delta_stbl.txt'\n",
    ")\n",
    "\n",
    "for exp in \"${Experiments[@]}\"; do\n",
    "    echo $exp\n",
    "    base=`basename $exp`\n",
    "    base=${base/.txt/}\n",
    "    # get intersect \n",
    "    awk 'NR==FNR{A[$1];next}$1 in A' $exp combined_analysis/hyper_mtyl.txt > combined_analysis/hyper_mtyl_${base}.txt\n",
    "\n",
    "    echo hyper_methylation\n",
    "    perl $TEISERDIR/run_mi_gene_list.pl \\\n",
    "        --expfile=$exp \\\n",
    "        --genefile=combined_analysis/hyper_mtyl_${base}.txt \\\n",
    "        --exptype=continuous \\\n",
    "        --ebins=7 \\\n",
    "        --species=human \\\n",
    "        --doremovedups=0 \\\n",
    "        --doremoveextra=0\n",
    "    # remove results from previous run \n",
    "    rm -vr combined_analysis/hyper_mtyl_${base}_GENESET\n",
    "    mv -v ${exp}_GENESET combined_analysis/hyper_mtyl_${base}_GENESET\n",
    "        \n",
    "    awk 'NR==FNR{A[$1];next}$1 in A' $exp combined_analysis/hypo_mtyl.txt > combined_analysis/hypo_mtyl_${base}.txt\n",
    "    \n",
    "    echo hypo_methylation\n",
    "    perl $TEISERDIR/run_mi_gene_list.pl \\\n",
    "        --expfile=$exp \\\n",
    "        --genefile=combined_analysis/hypo_mtyl_${base}.txt \\\n",
    "        --exptype=continuous \\\n",
    "        --ebins=7 \\\n",
    "        --species=human \\\n",
    "        --doremovedups=0 \\\n",
    "        --doremoveextra=0\n",
    "    # remove results from previous run \n",
    "    rm -vr combined_analysis/hypo_mtyl_${base}_GENESET\n",
    "    mv -v ${exp}_GENESET combined_analysis/hypo_mtyl_${base}_GENESET\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "combined_analysis/hyper_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_120h_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_120h_delta_exp.png'\n",
      "combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_120h_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_120h_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_6h_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_6h_delta_exp.png'\n",
      "combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_6h_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_6h_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_72h_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_72h_delta_exp.png'\n",
      "combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_kg1_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_kg1_delta_exp.png'\n",
      "combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_kg1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_kg1_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf > combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_lnTE_T_vs_U.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.png' -> 'combined_analysis/hyper_mtyl_lnTE_T_vs_U.png'\n",
      "combined_analysis/hyper_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_molm14_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_molm14_delta_exp.png'\n",
      "combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_molm14_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_molm14_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_exp.png'\n",
      "combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_exp.png'\n",
      "combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_thp1_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_thp1_delta_exp.png'\n",
      "combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_thp1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_thp1_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_120h_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_120h_delta_exp_GENESET/120h_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_120h_delta_exp.png'\n",
      "combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_120h_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_120h_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_6h_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_6h_delta_exp_GENESET/6h_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_6h_delta_exp.png'\n",
      "combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_6h_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_6h_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_72h_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_72h_delta_exp_GENESET/72h_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_72h_delta_exp.png'\n",
      "combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_kg1_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_kg1_delta_exp_GENESET/kg1_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_kg1_delta_exp.png'\n",
      "combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_kg1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_kg1_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf > combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_lnTE_T_vs_U.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.png' -> 'combined_analysis/hypo_mtyl_lnTE_T_vs_U.png'\n",
      "combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_molm14_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_molm14_delta_exp_GENESET/molm14_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_molm14_delta_exp.png'\n",
      "combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_molm14_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_molm14_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_ociaml2_delta_exp_GENESET/ociaml2_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_exp.png'\n",
      "combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_ociaml3_delta_exp_GENESET/ociaml3_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_exp.png'\n",
      "combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_thp1_delta_exp.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_thp1_delta_exp_GENESET/thp1_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_thp1_delta_exp.png'\n",
      "combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_thp1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_thp1_delta_stbl.png'\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for pdf in combined_analysis/*_GENESET/*.txt.summary.pdf; do \n",
    "    png=${pdf/.pdf/.png}\n",
    "    di=`dirname $pdf`\n",
    "    out=`basename $di`\n",
    "    outpng=${out/_GENESET/.png}\n",
    "    outpdf=${out/_GENESET/.pdf}\n",
    "    \n",
    "    bash /rumi/shams/abe/Workflows/my_scripts/pdf2png.sh $pdf\n",
    "    cp -v $pdf combined_analysis/$outpdf\n",
    "    mv -v $png combined_analysis/$outpng\n",
    "done "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "touch combined_analysis/README.md\n",
    "for f in combined_analysis/*.png; do \n",
    "    b=`basename $f`\n",
    "    t=${b/.png/}\n",
    "    echo '### '$t >> combined_analysis/README.md\n",
    "    echo -e \"<img src=\\\"\"$f\"\\\" title=\\\"\"$t\"\\\" style=\\\"width:1000px\\\">\\n\" >> combined_analysis/README.md\n",
    "done "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RNA Expression\n",
    "\n",
    "I've used `salmon` for alignment and `DESeq2` for ANNOVA like analysis (see [DESeq2 for time-series-experiments](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#time-series-experiments.)). Mainly, the linear model designed as `~condition + time + condition:time` to include both time and treatment together in the model. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HL-60 cell line - Time series experiment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5 other AML cell lines "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RNA Stability \n",
    "\n",
    "Separately, I've used `STAR` for alignment and then, I've used `featureCount`\n",
    "to evalute intronic and exonic counts (similar to [CRIES](https://github.com/csglab/CRIES)). \n",
    "Then, [REMBRANDTS](https://github.com/csglab/REMBRANDTS) is the tool to estimate unbias transcript stability. Then, I use `limma` for differential analysis. Here we only check the log2FC of treated vs. control "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HL-60 cell line - Time series experiment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's redraw heatmaps using `--min=-3 --max=3` thresholds. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6h hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.26\t-0.32\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.eps combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "6h hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.26\t-0.32\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.eps combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "120h hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.3\t-0.48\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.eps combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf\n",
      "Finished.\n",
      "120h hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.3\t-0.48\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.eps combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf\n",
      "Finished.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "declare -a Genesets=('hyper_mtyl' 'hypo_mtyl')\n",
    "declare -a Times=('6h' '120h')\n",
    "for time in \"${Times[@]}\"; do\n",
    "    for geneset in \"${Genesets[@]}\"; do\n",
    "        echo $time $geneset    \n",
    "        perl $TEISERDIR/Scripts/teiser_draw_matrix.pl \\\n",
    "        --pvmatrixfile=combined_analysis/${geneset}_${time}_delta_stbl_GENESET/${time}_delta_stbl.txt.matrix \\\n",
    "        --summaryfile=combined_analysis/${geneset}_${time}_delta_stbl_GENESET/${time}_delta_stbl.txt.summary \\\n",
    "        --expfile=combined_analysis/${geneset}_${time}_delta_stbl_GENESET/${time}_delta_stbl.txt \\\n",
    "        --quantized=0 \\\n",
    "        --colmap=$TEISERDIR/Scripts/HEATMAPS/cmap_1.txt \\\n",
    "        --order=0 \\\n",
    "        --min=-3 --max=3 \\\n",
    "        --cluster=5 \n",
    "    done \n",
    "done "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5 other AML cell lines "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.14\t-1.42\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.eps combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.32\t-1.52\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.eps combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.64\t-1.66\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.eps combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.48\t-0.96\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.eps combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.06\t-1.26\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.eps combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.32\t-1.52\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.eps combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.14\t-1.42\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.eps combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.32\t-1.52\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.eps combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.64\t-1.66\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.eps combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.48\t-0.96\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.eps combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.06\t-1.26\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.eps combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.pdf\n",
      "Finished.\n",
      "hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "1.32\t-1.52\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.eps combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf\n",
      "Finished.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "declare -a Genesets=('hyper_mtyl' 'hypo_mtyl')\n",
    "declare -a Lines=('kg1' 'ociaml2' 'molm14' 'ociaml3' 'thp1' 'ociaml2')\n",
    "\n",
    "for geneset in \"${Genesets[@]}\"; do\n",
    "    for line in \"${Lines[@]}\"; do\n",
    "        echo $geneset\n",
    "        perl $TEISERDIR/Scripts/teiser_draw_matrix.pl \\\n",
    "        --pvmatrixfile=combined_analysis/${geneset}_${line}_delta_stbl_GENESET/${line}_delta_stbl.txt.matrix \\\n",
    "        --summaryfile=combined_analysis/${geneset}_${line}_delta_stbl_GENESET/${line}_delta_stbl.txt.summary \\\n",
    "        --expfile=combined_analysis/${geneset}_${line}_delta_stbl_GENESET/${line}_delta_exp.txt \\\n",
    "        --quantized=0 \\\n",
    "        --colmap=$TEISERDIR/Scripts/HEATMAPS/cmap_1.txt \\\n",
    "        --order=0 \\\n",
    "        --min=-3 --max=3 \\\n",
    "        --cluster=5 \n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_120h_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_120h_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_6h_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_6h_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_kg1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_kg1_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_kg1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_kg1_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_molm14_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_molm14_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_molm14_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_molm14_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_ociaml2_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_ociaml3_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.pdf > combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_thp1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.png' -> 'combined_analysis/hyper_mtyl_thp1_delta_stbl.png'\n",
      "combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf > combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_thp1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.png' -> 'combined_analysis/hyper_mtyl_thp1_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_120h_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_120h_delta_stbl_GENESET/120h_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_120h_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_6h_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_6h_delta_stbl_GENESET/6h_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_6h_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_kg1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_kg1_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_kg1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_kg1_delta_stbl_GENESET/kg1_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_kg1_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_molm14_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_molm14_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_molm14_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_molm14_delta_stbl_GENESET/molm14_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_molm14_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl_GENESET/ociaml2_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_ociaml2_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl_GENESET/ociaml3_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_ociaml3_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.pdf > combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_thp1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_exp.txt.summary.png' -> 'combined_analysis/hypo_mtyl_thp1_delta_stbl.png'\n",
      "combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf > combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_thp1_delta_stbl.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_thp1_delta_stbl_GENESET/thp1_delta_stbl.txt.summary.png' -> 'combined_analysis/hypo_mtyl_thp1_delta_stbl.png'\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for pdf in combined_analysis/*stbl_GENESET/*.txt.summary.pdf; do \n",
    "    png=${pdf/.pdf/.png}\n",
    "    di=`dirname $pdf`\n",
    "    out=`basename $di`\n",
    "    outpng=${out/_GENESET/.png}\n",
    "    outpdf=${out/_GENESET/.pdf}\n",
    "    \n",
    "    bash /rumi/shams/abe/Workflows/my_scripts/pdf2png.sh $pdf\n",
    "    cp -v $pdf combined_analysis/$outpdf\n",
    "    mv -v $png combined_analysis/$outpng\n",
    "done "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ribo-seq "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We have used [Ribolog](https://github.com/goodarzilab/Ribolog) to estimate differential translational efficiency (lnTE). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hyper_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.76\t-0.4\n",
      "0\n",
      "Outputing EPS file combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.eps combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf\n",
      "Finished.\n",
      "hypo_mtyl\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0.76\t-0.4\n",
      "0\n",
      "Outputing EPS file combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.eps\n",
      "Convert to PDF combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.eps combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf\n",
      "Finished.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "declare -a Genesets=('hyper_mtyl' 'hypo_mtyl')\n",
    "for geneset in \"${Genesets[@]}\"; do\n",
    "    echo $time $geneset    \n",
    "    perl $TEISERDIR/Scripts/teiser_draw_matrix.pl \\\n",
    "    --pvmatrixfile=combined_analysis/${geneset}_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.matrix \\\n",
    "    --summaryfile=combined_analysis/${geneset}_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary \\\n",
    "    --expfile=combined_analysis/${geneset}_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt \\\n",
    "    --quantized=0 \\\n",
    "    --colmap=$TEISERDIR/Scripts/HEATMAPS/cmap_1.txt \\\n",
    "    --order=0 \\\n",
    "    --min=-3 --max=3 \\\n",
    "    --cluster=5 \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf > combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf' -> 'combined_analysis/hyper_mtyl_lnTE_T_vs_U.pdf'\n",
      "renamed 'combined_analysis/hyper_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.png' -> 'combined_analysis/hyper_mtyl_lnTE_T_vs_U.png'\n",
      "combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf > combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.png\n",
      "done!\n",
      "'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.pdf' -> 'combined_analysis/hypo_mtyl_lnTE_T_vs_U.pdf'\n",
      "renamed 'combined_analysis/hypo_mtyl_lnTE_T_vs_U_GENESET/lnTE_T_vs_U.txt.summary.png' -> 'combined_analysis/hypo_mtyl_lnTE_T_vs_U.png'\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for pdf in combined_analysis/*lnTE_T_vs_U_GENESET/*.txt.summary.pdf; do \n",
    "    png=${pdf/.pdf/.png}\n",
    "    di=`dirname $pdf`\n",
    "    out=`basename $di`\n",
    "    outpng=${out/_GENESET/.png}\n",
    "    outpdf=${out/_GENESET/.pdf}\n",
    "    \n",
    "    bash /rumi/shams/abe/Workflows/my_scripts/pdf2png.sh $pdf\n",
    "    cp -v $pdf combined_analysis/$outpdf\n",
    "    mv -v $png combined_analysis/$outpng\n",
    "done "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
