{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Main goal: \n",
    "- Hani is looking for the final list of genes which are drug induced significantly methylated, expressed and stabilized. \n",
    "\n",
    "    ○ Venn diagram \n",
    "\n",
    "    ○ UpsetR \n",
    "\n",
    "=> table of these genes with all logFCs, pvalues and screening scores. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cp -v radar/result.all.txt hl60_delta_mtyl_table.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import numpy as np\n",
    "# Teated vs. non-treated \n",
    "comparisons = [\n",
    "    # HL-60 meRIP-seq - logFC\n",
    "    'meRIP-seq/hl60_delta_mtyl_table.txt' ,\n",
    "    # HL-60 Ribo-seq - lnTE\n",
    "    'Ribo-seq/hl60_delta_te_table.txt' ,\n",
    "    # HL-60 RNA-seq \n",
    "    # RNA experssion - log2FC\n",
    "    'RNA-seq/hl60-exp/hl60_delta_exp_table.txt',\n",
    "    # RNA stability  - logFC\n",
    "    'RNA-seq/hl60-stbl/hl60_delta_stbl_table.txt',\n",
    "    ## 5 other AML cell lines RNA-seq\n",
    "    # RNA experssion - log2FC\n",
    "    'RNA-seq/other-exp/kg1_delta_exp_table.txt', 'RNA-seq/other-exp/molm14_delta_exp_table.txt',\n",
    "    'RNA-seq/other-exp/ociaml2_delta_exp_table.txt', 'RNA-seq/other-exp/ociaml3_delta_exp_table.txt',\n",
    "    'RNA-seq/other-exp/thp1_delta_exp_table.txt',\n",
    "    # RNA stability - logFC\n",
    "    'RNA-seq/other-stbl/kg1_delta_stbl_table.txt', 'RNA-seq/other-stbl/molm14_delta_stbl_table.txt',\n",
    "    'RNA-seq/other-stbl/ociaml2_delta_stbl_table.txt', 'RNA-seq/other-stbl/ociaml3_delta_stbl_table.txt',\n",
    "    'RNA-seq/other-stbl/thp1_delta_stbl_table.txt'\n",
    "]\n",
    "\n",
    "\n",
    "# extract cell line name experiment name \n",
    "names = [c.split('/')[-1].replace('.txt','').replace('_table','') for c in comparisons]\n",
    "tables = [pd.read_csv(x, sep = '\\t') for x in comparisons]\n",
    "cells = [names[i].split('_')[0] for i,x in enumerate(comparisons)]\n",
    "experiments = [names[i].replace(cells[i]+'_', '') for i,x in enumerate(comparisons)]\n",
    "\n",
    "# read data into Pandas Data.Frames \n",
    "Data = dict(((c,{}) for c in cells))\n",
    "for i,x in enumerate(comparisons): \n",
    "    cel = cells[i]\n",
    "    exp = experiments[i]\n",
    "    # read data and write into dictionary \n",
    "    Data [cel][exp] = tables[i]\n",
    "    \n",
    "# include CRISPR screening scores \n",
    "Data['hl60']['rho'] = pd.read_csv('screen/CRISPRi_HL60_rho.txt', header=None, sep = '\\t').dropna().set_index(0)\n",
    "Data['hl60']['gamma'] = pd.read_csv('screen/CRISPRi_HL60_gamma.txt', header=None, sep = '\\t').dropna().set_index(0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Intersectional genes \n",
    "Find top genes across all cell lines and high throughput experiments:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "Top = {}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Translational efficiency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(55, 4)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TE = Data['hl60']['delta_te'][['gene_id','gene_name','Estimate_treatmentDRUG','fdr_Pr...z.._treatmentDRUG']]\n",
    "\n",
    "te_thr=1 # Estimate_treatmentDRUG\n",
    "fdr_thr=0.50 # fdr_Pr...z.._treatmentDRUG\n",
    "\n",
    "Top['TE'] = TE.iloc[\n",
    "    [i for i,l in enumerate(\n",
    "        np.array([\n",
    "            np.array(TE.iloc[:,2].abs() >= te_thr),\n",
    "            np.array(TE.iloc[:,3] < fdr_thr)]).all(axis=0)) if l == 1]\n",
    "        ,:] \n",
    "\n",
    "Top['TE'].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## $\\Delta$RNA methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(449, 4) (49, 4)\n"
     ]
    }
   ],
   "source": [
    "Mtyl = Data['hl60']['delta_mtyl'][['ensembl','name','logFC','p_value']]\n",
    "Mtyl.columns = ['gene_id','gene_name','logFC','pval']\n",
    "\n",
    "fc_thr = 4\n",
    "pv_thr = 0.0001\n",
    "\n",
    "Top['Hyper_m6A'] = Mtyl.iloc[\n",
    "    [i for i,l in enumerate(\n",
    "        np.array([\n",
    "            np.array(Mtyl.iloc[:,2] >= fc_thr),\n",
    "            np.array(Mtyl.iloc[:,3] < pv_thr)]).all(axis=0)) if l == 1]\n",
    "        ,:].sort_values('pval').drop_duplicates(subset='gene_id', keep=\"last\")\n",
    "\n",
    "\n",
    "Top['Hypo_m6A'] = Mtyl.iloc[\n",
    "    [i for i,l in enumerate(\n",
    "        np.array([\n",
    "            np.array(Mtyl.iloc[:,2] <= -fc_thr),\n",
    "            np.array(Mtyl.iloc[:,3] < pv_thr)]).all(axis=0)) if l == 1]\n",
    "        ,:].sort_values('pval').drop_duplicates(subset='gene_id', keep=\"last\")\n",
    "\n",
    "print (Top['Hyper_m6A'].shape, Top['Hypo_m6A'].shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "___\n",
    "\n",
    "Function!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# it works for both below stablity and experssion data frames \n",
    "def find_top(\n",
    "    df, n_line, \n",
    "    pv_thr=0.05, fc_thr=2,\n",
    "    fc = range(1,13,2),\n",
    "    pv  = range(2,13,2)):\n",
    "    \n",
    "    # Select rows (genes) which has abs(log2FC) >= fc_thr & pval < pv_thr \n",
    "    # in more than n_line cell lines \n",
    "\n",
    "    top = df.iloc[\n",
    "        [i for i,l in enumerate(\n",
    "            np.array([\n",
    "                np.array(df.iloc[:,fc].abs() >= fc_thr).sum(axis=1) >= n_line,\n",
    "                np.array(df.iloc[:,pv] < pv_thr).sum(axis=1) >= n_line\n",
    "            ]).all(axis=0).tolist()) if l == 1\n",
    "        ],:]\n",
    "    \n",
    "    return top"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## $\\Delta$RNA expression "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "E1 = Data['hl60']['delta_exp']\n",
    "E2 = Data['kg1']['delta_exp']\n",
    "E3 = Data['molm14']['delta_exp']\n",
    "E4 = Data['ociaml2']['delta_exp']\n",
    "E5 = Data['ociaml3']['delta_exp']\n",
    "E6 = Data['thp1']['delta_exp']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "load R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library (tidyverse)\n",
    "library (UpSetR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -i E1,E2,E3,E4,E5,E6 -o exp_df\n",
    "\n",
    "exp_ids = E1[,'gene_id']\n",
    "exp_names = E1[,'gene_name']\n",
    "\n",
    "Exp = list(\n",
    "    # hl60\n",
    "    E1 %>% dplyr::select('gene_id','log2FC_120h','pval_120h') %>% \n",
    "            rename(hl60.log2FC=log2FC_120h,\n",
    "                   hl60.pval=pval_120h), \n",
    "    # kg1\n",
    "    E2 %>% dplyr::select('gene_id','log2FoldChange','pvalue') %>%\n",
    "            rename(kg1.log2FC=log2FoldChange,\n",
    "                   kg1.pval=pvalue),  \n",
    "    # molm14\n",
    "    E3 %>% dplyr::select('gene_id','log2FoldChange','pvalue') %>%\n",
    "            rename(molm14.log2FC=log2FoldChange,\n",
    "                   molm14.pval=pvalue), \n",
    "    # ociaml2\n",
    "    E4 %>% dplyr::select('gene_id','log2FoldChange','pvalue') %>%\n",
    "            rename(ociaml2.log2FC=log2FoldChange,\n",
    "                   ociaml2.pval=pvalue), \n",
    "    # ociaml3    \n",
    "    E5 %>% dplyr::select('gene_id','log2FoldChange','pvalue') %>% \n",
    "            rename(ociaml3.log2FC=log2FoldChange,\n",
    "                   ociaml3.pval=pvalue), \n",
    "    # thp1\n",
    "    E6 %>% dplyr::select('gene_id','log2FoldChange','pvalue') %>%\n",
    "            rename(thp1.log2FC=log2FoldChange,\n",
    "                   thp1.pval=pvalue) \n",
    ")\n",
    "\n",
    "for (i in 1:6){\n",
    "    Exp[[i]] = Exp[[i]] %>% data.frame %>% remove_rownames %>% column_to_rownames('gene_id')\n",
    "    Exp[[i]] = Exp[[i]][exp_ids,]\n",
    "    \n",
    "}\n",
    "\n",
    "exp_df <- bind_cols(Exp) %>% add_column(gene_name=exp_names, .before=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(309, 13)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Top['Exp'] = find_top(exp_df, n_line=3, fc_thr=1)\n",
    "\n",
    "Top['Exp'].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## $\\Delta$RNA stability"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "S1 = Data['hl60']['delta_stbl']\n",
    "S2 = Data['kg1']['delta_stbl']\n",
    "S3 = Data['molm14']['delta_stbl']\n",
    "S4 = Data['ociaml2']['delta_stbl']\n",
    "S5 = Data['ociaml3']['delta_stbl']\n",
    "S6 = Data['thp1']['delta_stbl']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -i S1,S2,S3,S4,S5,S6 -o stbl_df\n",
    "\n",
    "stbl_ids = S1[,'ensembl_id']\n",
    "stbl_names = S1[,'gene_name']\n",
    "\n",
    "Stbl = list(\n",
    "    # hl60\n",
    "    S1 %>% dplyr::select('ensembl_id','logFC_120h','P.Value_120h') %>% \n",
    "            rename(hl60.log2FC=logFC_120h,\n",
    "                   hl60.pval=P.Value_120h), \n",
    "    # kg1\n",
    "    S2 %>% dplyr::select('ensembl_id','logFC','P.Value') %>%\n",
    "            rename(kg1.log2FC=logFC,\n",
    "                   kg1.pval=P.Value),  \n",
    "    # molm14\n",
    "    S3 %>% dplyr::select('ensembl_id','logFC','P.Value') %>%\n",
    "            rename(molm14.log2FC=logFC,\n",
    "                   molm14.pval=P.Value), \n",
    "    # ociaml2\n",
    "    S4 %>% dplyr::select('ensembl_id','logFC','P.Value') %>%\n",
    "            rename(ociaml2.log2FC=logFC,\n",
    "                   ociaml2.pval=P.Value), \n",
    "    # ociaml3    \n",
    "    S5 %>% dplyr::select('ensembl_id','logFC','P.Value') %>% \n",
    "            rename(ociaml3.log2FC=logFC,\n",
    "                   ociaml3.pval=P.Value), \n",
    "    # thp1\n",
    "    S6 %>% dplyr::select('ensembl_id','logFC','P.Value') %>%\n",
    "            rename(thp1.log2FC=logFC,\n",
    "                   thp1.pval=P.Value) \n",
    ")\n",
    "\n",
    "for (i in 1:6){\n",
    "    Stbl[[i]] = Stbl[[i]] %>% data.frame %>% remove_rownames %>% column_to_rownames('ensembl_id')\n",
    "    Stbl[[i]] = Stbl[[i]][stbl_ids,]\n",
    "    \n",
    "}\n",
    "\n",
    "stbl_df <- bind_cols(Stbl) %>% add_column(gene_name=stbl_names, .before=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(71, 13)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Top['Stbl'] = find_top(stbl_df, n_line=3, fc_thr=0.2)\n",
    "\n",
    "Top['Stbl'].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CRISPR Screen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "Rho = Data['hl60']['rho']\n",
    "\n",
    "Top['Rho'] = Rho.iloc[\n",
    "    np.array([i for i,l in enumerate(np.array(np.array(Rho[1].abs()) > 0.5).tolist()) if l])\n",
    "    ,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "Gamma = Data['hl60']['gamma']\n",
    "\n",
    "Top['Gamma'] = Gamma.iloc[\n",
    "    np.array([i for i,l in enumerate(np.array(np.array(Gamma[1].abs()) > 0.5).tolist()) if l])\n",
    "    ,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Venn diagram \n",
    "\n",
    "https://python-graph-gallery.com/172-custom-venn-diagram/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 586,
   "metadata": {},
   "outputs": [],
   "source": [
    "Exp = Top['Exp'].gene_name.tolist()\n",
    "Stbl = Top['Stbl'].gene_name.tolist()\n",
    "TE = Top['TE'].gene_name.tolist()\n",
    "\n",
    "Rho = Top['Rho'].index.tolist()\n",
    "Gamma = Top['Gamma'].index.tolist()\n",
    "\n",
    "Hyper_m6A = Top['Hyper_m6A'].gene_name.tolist()\n",
    "Hypo_m6A = Top['Hypo_m6A'].gene_name.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 587,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 588,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(UpSetR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 589,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAACW1BMVEUKCgoLCwsbGxscHBwoKCgpKSkrKyssLCwuLi4xMTEzMzM5OTk6Ojo7Ozs9PT0+Pj4/Pz9AQEBBQUFCQkJERERFRUVGRkZHR0dJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d5eXl6enp7e3t9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqcnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+xsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////bsPf6AAARpklEQVR4nO2dj19VZZ7Hm9mZ2dndrK3dmJkaM20rI6LUbZtyrLAhJSoUcdJsdESkMhYUJDPFERo2qQVvpUX+QKcmnUypyM1NDJCf3z9r7nPvFZB7uOfc55zznOd8+Lx9Pej58n2e85zz9l4uly/Pc4MQaG6IegIkXCgYHAoGh4LBoWBwKBic+Aju3RP1DGJJfASf3Rb1DGIJBYNDweBQMDj2Ch65eHFQhvok1YSCNbFX8PG1TacvrW9+SzV1TMFaWCy4+qOJA8elUjV1TMFa2Cv4u2Pt23d+IZWqqWMK1sJewUkq2j6WStVEls0vomAd7BV85PWX2vuralNN5PNT71GwDvYKloExkYnhdBM+RWtiseAZULAWFAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDorg0e3pdqD6S1MTigcogpufTrXDb8u4qQnFAxDBZ9rWpVrNxt8fNzajWAAieNP2hw6pVt/dt8XYjGIBiGCRdan21foNR81MJy7ACM4wPhz2PGIGmmAyAwoGh4LBoWBw4i94nhNmp2YzFAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8Hg2Cx4YNjL9rIUnBOLBQ8+2eVle1kKzonFghv2dXnZXpaCc2Kv4OMdHV1T28vOvjklBefEXsENNWXl+z1sL0vBObFXsEhH19T2snyK1sRmweJpe1kKzonlgqdBwVpQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4c0Xwp9W1wyP1DWNyoPrLkCdqF3NF8Ji0de47drTt8NsyHvJE7WKuCJaWxy6/PHC5umbj74+HO0/LmDOCpWffK3299fXdfVtCnaZtzBXBiYa1n56rWtf71foNR0OeqF3MFcFyJfmld3RMZHw43GnaxpwRPFehYHAoGBwKBsdewRNffy3BbC87p/8H2Cv47N4ddcFsL0vBdjJxYWsw28tSsJ0crpi2vWzd5k0UrIPFgkUq2ia3l6VgTcwK/rZ81fgbHnM/rNv6ZjDby1KwMea/WygLvSYPjgS0vSwFG2ORFHsXPAMK1sKs4KVVBRsf1exLwVqYFTy2q6xhTLMvBWthVvCPakQKNftSsBZmBS8ue3iQgo1iVnChHLz1Js2+FKyFWcG7RL5YotmXgrUwKbh0vFSh2ZuCtTApOCEJhWZvCtbCpOA3++VIQcGHmr0pWAuTgm8W+cnBE3yRZRSTggvk839UH/WgYC1MCl645vY1MsJHsFFMCh58ffeYXPizZm8K1sLqH/hfBwVrQcHgUDA4ZgWfXbuypES3LwXrYFbwr97p7u7W7EvBWpgV/JSPvhSshVnBBUXl5eWafSlYC7OCOxSafSlYC7OC+zf+18Z+zb4UrIVZwXe9evSVRZp9KVgLw4IzTQcK1sKs4HvXt1Ut1uxLwVqYFXx1x+odVzX7UrAWrMkCx2xN1ruJRILfJhnF7FN0sahfQNODgrUwKfid1T9fvbr0Vs3eFKyFScHnOh/v7PzgB83eFKyF2afooUzTgYK1MPzjwmS7U7MvBWthVvCCTNOBgrUwK/jhF1pfWKbZl4K1MCt4tKFsF3/D3yj2LqM0EwrWwuJllGZAwVpwGSVwuIwSOFxGCRyzgtfI7n/aqNmXgrUwK3ih/NvwLzT7UrAWZgX/++5ivpNlFqOCx19c983ods3OFKyF2Ufwb330pWAtDD9F3/csf3XFLPzVFXD4XjQ4fC8aHL4XDQ7fiwbHrOCBXWUNrKo0CgvfwbG48L3vIncf9Y+9he8fNFW3cPdR39i8TtbABu4+6huL18ma2HpmavdRCtbE4nWyXv1EpnYfXTa/iIJ1sHedrLYVNfundh/9/NR7FKyD5T9s4O6jfjEpuDiNZm8K1sKk4O40mr0pWAuuFw0OBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg0PB4FAwOBQMDgWDQ8HgUDA4FAwOBYNDweBQMDgUDA4Fg2Ox4JF+bi/rH3sF925ZI9xe1jf2ChZZI9xe1jd2C57aXpabU2pit2BuL+sbewX31yx9Y2p7WT5Fa2Kv4BTcXtYvlgueBgVrQcHgUDA4FAwOBYNDweBQ8BTdtduO9b16PPCZRwoFT+eV81fefyvgeUcMBU9j/HmRkxQcEQYEv3+QgqPDgODKIelbW3Iq6JlHCgWDQ8HgUDA4FAwOBXvLiC0U7C0jzV9ran9QLZALMgMFe8tIc+hYus2Ksv9pde1wPhcWLhTsLSPNiZoX+lSbdZLK/pi0deZ3aWFCwd4yrnFyf7rNQsp+y2OX87mwcKFgbxlpDjeU96qWY5rKfs++fC4sXCjYW0aG/vF0mw1lP9Gw9tN8LixcKNhbhleU/Ss5/gcYh4K9ZcQWCvaWEVso2FtGbKFgTxnuJ7EVCvaUEcRJooGCPWVQcPhQsBYU7CmDgsOHgrWgYE8ZFBw+FKwFBXvKoODwoWAtKNhTBgWHDwVrQcGeMig4fChYCwr2lEHB4UPBWlCwpwwKDh8K1oKCPWVQcPhQsBYU7CmDgsOHgrWgYE8ZFBw+MIJP/rEu9++2uC+nmMeCixTsKSNIwZcm9h/JeaXuyynmseAiBXvKCFKwyIv/n/tS3Vfb874eHwV7yghU8PYel0ul4OnETnDzE7V/zXml7ssp5rHgIgV7yghSsFko2FMGBYeIge1lKThCTGwvS8ERYmJ7WQqOkPT2snWb1xWUOvIPTrhmuI8Rz5NMsjlz++wXnN5ednbBxJH4CHbbXpbkxH7BbtvLkpzEQHAGCtYiRoJPj5I8yNy2GAkeInlwbcnq+AiWqG9ZvKBgcCgYHAoGx5zgvxyexrc+Bor6lsULc4JXTX/7vdXHQFHfsnhBweAACu7t7fWb4T5E8hrCP0kQGXiCe3vdrto1w32I1EWEfZIgMqIVfK64/NGeZ4qOtpQs3/xs0WjmLxVRn21bVnL54qNPv3d+ySNtLSsqLi5/spmCsxNsFvzM38ZuG96zaWn9brlFXurO/KUi6rPrXvlOyntEVp+eWFC/U8p3dNxBwflmRCt45YWJX1YePLe4vl0KZVsi85eKqM+OtS84ufJLkafOy8Lkp1Y2dyaCE+w+Rs4Eb4J9niSIK4lW8OnFq25vWLbqgesFq4j67GtVC8+evr/yf48tfmJ78lOfFVVt8iKYXEekgq+cef9xuZKVeS3y3ajIxPfJh/Jg+vjSROqvqG9ZvDAneMPd03gvFRqo3THgmDtUXl4+6wvtqG9ZvOB70eBQMDgUDA4Fg0PB4FAwODEUTHSgYHAoGBwKBoeCwaFgcMIXfHDzNE6HfjpyPTGqySI6UDA4Ngm+0DfzH3nR1+fSrS+YjNwJ6ir9nsR1CPeTTBKd4L3vzMxsbM3Ee85onKfP9ca5ZgQwRHo5pZCH8JAxSXSClz+oimJT1bNPrahWtbKNreooGW/tSpXMJsN5nMebviAyciXM8ybYfQx/J5kiOsHJx2v9TklVzzZJkaqVbWxVR8l4Y2uqZDYZzuM8QI/gGyEewU1Jwe2SqZ4tVrWyja3qqEkJzpTMFudzIpyvwTdCfA0+cd++pMJM9WyxqpVtbFVHyXhja6ZkNi/BOByZ57Ylw/q7vI4V+avo6bWyU0eTJbNzkngJ3ls5jROhnw6BeAkmeUPB4FAwOBQMDgWDQ8HgzFHBUVcam+PwvKMuGZUL3MaIYV10QHcvBlAwOBQMDgWDQ8HgxEfwV8UzOOxjsIDuXgyIj+C/zZsBFyP1AqLg89+qYkr1Z9bBArp7MQBKcKa4sr5dpOdMY2vPmexqywwuVxTE+nBqiqGfxH2ItnmHXDLK59ux0p2LYFVKufzBNlVLWb+kZFtrV2Nra1cy8Fz/RElIgnNmzAtIsN+1Kj0JtmKtShfB6VLKdGVlndy7s7Ux9Uc6tv3P9sgE5zRsh+DeuAhWpZRNrenKyj3ywH+n9CYDcv/S7IUO+QielhETwaqU8sR9+1KVlfc9X5p+/CYD8offOAyW+4ICWSfdza+hZeWTgl0yYvI1OFNKma6lnLZ65dXSTxwGc7kiIKBeRTtx5KBTNKC7FwPiI/iHt2fwlY/BArp7MSA+ggMloLsXAygYHAoGh4LBoWBwKBicOSp47jBH66LnDhQMDgWDQ8HgRCS4/7J/Rryfbg4TkeBbZv5oSAPHHxORGVAwOPCCQ1qMVNxXD3Mdwv0krgnu07jRVfBz891Ocg0rBYe0GKmagPsYPk/iaS1Zl4UKXZcy7HtuvtfHQHSCLy5/snn/vk9ebvltybaWFRXqUJXRqhbOYqRBLCMZQIbrLDwK9mg4OsHlOzrukCWLhlTBbN3O1KEqo02X0oayGGlsBLtmxELwyubOhCy5d1QVzL7WnjpUZbTpVUk1FiMFeor2JDj3SSaJTvBnRVWbtryVWK0KZpMu1aEqo02vSqqzGCnOiyzX/4gxeZF1aUJ9rG8fmDxUZbTpUlrkxUh/9bLvIeL0bdKpc96ngIGtgr/7P/9c1bkaNGwVTAKCgsGhYHAoGBwKBoeCnQmoJjV6frnR9xCQZbO+74otULAzvu+KLVCwM77vii1QsDO+74otULAzvu+KLUQm+Pyh0PmGgiMU3BjAj4tc4GKkQxRMwe7EQbBaX3ZWxv7SLzKjNMLtkub5X8UsgGXOzKy25j5G9ILV+rKzcfWeTfXS9eP+fASnxne7b75XGbRjvUT3MWwQ/MiKarWsrCqbVTWzLSsqRFpKlm9+tmi0/uU3v5dHG2spePYh7BfcJEVqWVlVNlu2o+OO+p0quFtukZe6Kx56d/7AQ+P/QcG6Y9ggWBVN3r/0iiqbXdHcmUg9ZSc/FMq2xGu75dd1dz72L6cCF5x7jIAE5z5JQIJzn8QWwX/4jaiyWVUzW9+eqL4m+PvCZ35317D0PJ2HYHId0QtWqGVl02Wz6RLaKX6QLKK+ZfHCSfCf7g+dzuudqWVlPZfNRn3L4gXfiwaHgsGhYHAoGBwKBmem4FNHjfEZBRtgpuDXq67jzpKqbNbdvMYh+rvbHYJV//mAU/Tu5ckPu7UfwyR/ZnmKfqjLITj+zw7bVclH9zgN8Npap+gTB7zOiwTELIKfPuoQHJ8/4BA9udxpgD1bnaIVhzxOiwRFjF5kER0oGBwnwQPDMtQnqTbJxNdfS3a076JDUEb6naKK7AgJGQfBg092XVrf/JZqU8Gze3fUZUU/aKpuyU7t3bJGsqOK7AgJGwfBDfu6DhyXStWmRScubHWIDmxwCMoacYqKZEdI2GQLPt7R0bXzC6lUbVr4cIVDdGLrGYfUpGCnqEh2hIRNtuCGmrLy/R9LZdvHM2xUtGVFX/1EsoNKsFNUJDtCwsbpRVZHV39VbbtqU7EP67a+mRVtW1GzPzu1v2bpG9nR1GeyIiRsZvk2aWI43aYYHHGKOqbmjhKj8PtgcCgYHAoGh4LByU/w4MLb/nVv6l/n3pk8TpQGPy0SFPkJfr1C5HLqX52rp47HA58VCYz8BDfdcUZkbOmiBRcf/3nx1cxxonRX4eIbUtGQJkn0yU/w2As/++lHu9bJu+XpR3DqWD1Fl9WkomHMkPgi7xdZfyooKyguXJUWnDpOCj74sKSiwc6NBEB+gi+MyMlfNGwUGf+gdPI4UfrFrYOSioYyReKH/AT/+aaCmz8ce2Th3Xuu3LJoKHOcKF1yU2FhKhrSJIk+/D4YHAoGh4LBoWBwKBgcCgaHgsGhYHAoGJy/A5vnbP1nHtNtAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R -i Exp,Stbl,TE,Hyper_m6A,Rho,Gamma\n",
    "\n",
    "Vecs <- list(\n",
    "    gamma_score=unlist(Gamma), rho_score=unlist(Rho), \n",
    "    translation=unlist(TE), expression=unlist(Exp), \n",
    "    stability=unlist(Stbl), hyper.m6A=unlist(Hyper_m6A)\n",
    ")\n",
    "intersection <- Reduce(intersect,Vecs)\n",
    "upset(fromList(Vecs), order.by = \"freq\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Final merged table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "uniq_genes = list(\n",
    "    set(TE.gene_name.tolist()) &\n",
    "    set(Mtyl.gene_name.tolist()) &\n",
    "    set(stbl_df.gene_name.tolist()) &\n",
    "    set(exp_df.gene_name.tolist()) &\n",
    "    set(Gamma.index.tolist()) &\n",
    "    set(Rho.index.tolist())\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "662"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(uniq_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I'm not sure which subset of genes are related to our purpose."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%R -i uniq_genes,TE\n",
    "# # ,Mtyl,stbl_df,exp_df,Gamma,Rho\n",
    "# t = TE %>% select(-'gene_id') %>% remove_rownames %>% column_to_rownames('gene_name')\n",
    "\n",
    "# # unlist(uniq_genes)%>% head\n",
    "# t %>% head\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "R version 4.0.3 (2020-10-10)\n",
      "Platform: x86_64-conda-linux-gnu (64-bit)\n",
      "Running under: Ubuntu 18.04.4 LTS\n",
      "\n",
      "Matrix products: default\n",
      "BLAS/LAPACK: /rumi/shams/abe/anaconda3/envs/deseq2/lib/libopenblasp-r0.3.10.so\n",
      "\n",
      "locale:\n",
      " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
      " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
      " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
      " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
      " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
      "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
      "\n",
      "attached base packages:\n",
      "[1] tools     stats     graphics  grDevices utils     datasets  methods  \n",
      "[8] base     \n",
      "\n",
      "other attached packages:\n",
      " [1] UpSetR_1.4.0    forcats_0.5.0   stringr_1.4.0   dplyr_1.0.2    \n",
      " [5] purrr_0.3.4     readr_1.4.0     tidyr_1.1.2     tibble_3.0.4   \n",
      " [9] ggplot2_3.3.2   tidyverse_1.3.0\n",
      "\n",
      "loaded via a namespace (and not attached):\n",
      " [1] Rcpp_1.0.4.6     plyr_1.8.6       pillar_1.4.6     compiler_4.0.3  \n",
      " [5] cellranger_1.1.0 dbplyr_1.4.4     jsonlite_1.7.1   lubridate_1.7.9 \n",
      " [9] lifecycle_0.2.0  gtable_0.3.0     pkgconfig_2.0.3  rlang_0.4.7     \n",
      "[13] reprex_0.3.0     cli_2.1.0        rstudioapi_0.11  DBI_1.1.0       \n",
      "[17] haven_2.3.1      gridExtra_2.3    withr_2.3.0      xml2_1.3.2      \n",
      "[21] httr_1.4.2       fs_1.5.0         generics_0.0.2   vctrs_0.3.4     \n",
      "[25] hms_0.5.3        grid_4.0.3       tidyselect_1.1.0 glue_1.4.2      \n",
      "[29] R6_2.4.1         fansi_0.4.1      readxl_1.3.1     modelr_0.1.8    \n",
      "[33] blob_1.2.1       magrittr_1.5     ps_1.4.0         backports_1.1.10\n",
      "[37] scales_1.1.1     ellipsis_0.3.1   rvest_0.3.6      assertthat_0.2.1\n",
      "[41] colorspace_1.4-1 stringi_1.5.3    munsell_0.5.0    broom_0.7.2     \n",
      "[45] crayon_1.3.4    \n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://stackoverflow.com/questions/2617600/importing-data-from-a-json-file-into-r"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:deseq2]",
   "language": "python",
   "name": "conda-env-deseq2-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
