{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "According to Matt's epxeriance ([link](https://github.com/goodarzilab/khorms/blob/master/side_projects/COVID_DMSseq_library.ipynb)) I'm using this genome:\n",
    "\n",
    "- https://www.ncbi.nlm.nih.gov/nuccore/MN908947?%3Fdb=nucleotide\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peaks\n",
    "<img src=\"plots/igv_snapshot.png\" style=\"height:400px\">\n",
    "\n",
    "# _N_\n",
    "<img src=\"plots/N1.png\" style=\"height:400px\">\n",
    "<img src=\"plots/N2.png\" style=\"height:400px\">\n",
    "<img src=\"plots/N3.png\" style=\"height:400px\">\n",
    "\n",
    "# _ORF10_\n",
    "<img src=\"plots/orf10.png\" style=\"height:400px\">\n",
    "\n",
    "# _S_\n",
    "<img src=\"plots/S1.png\" style=\"height:400px\">\n",
    "<img src=\"plots/S2.png\" style=\"height:400px\">\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loci</th>\n",
       "      <th>DRACH</th>\n",
       "      <th>RGAC</th>\n",
       "      <th>AAGAA</th>\n",
       "      <th>Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>S</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>GAACC,AGACT,TGACC,AAACA,AAACT,TAACT,AGAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>S</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>AGACT,AGACT,AGAC,AGAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>GGACC,GGAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>N</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TAACC,AAACA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>N</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>AGACC,GGACA,TAACA,AGAC,GGAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>N</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>AGAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GAACT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TAACA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>GAACT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>N</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>AAACC,GGACC,GGAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>AGACA,AAACA,AAACT,TGACT,AGAC,AAGAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>ORF10</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TAACT,TAACT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     loci  DRACH  RGAC  AAGAA                                  Sequence\n",
       "1       S      6     1      0  GAACC,AGACT,TGACC,AAACA,AAACT,TAACT,AGAC\n",
       "7       S      2     2      0                     AGACT,AGACT,AGAC,AGAC\n",
       "9       N      1     1      0                                GGACC,GGAC\n",
       "10      N      2     0      0                               TAACC,AAACA\n",
       "12      N      3     2      0               AGACC,GGACA,TAACA,AGAC,GGAC\n",
       "13      N      0     1      0                                      AGAC\n",
       "14      N      1     0      0                                     GAACT\n",
       "16      N      1     0      0                                     TAACA\n",
       "18      N      1     0      0                                     GAACT\n",
       "19      N      2     1      0                          AAACC,GGACC,GGAC\n",
       "23      N      4     1      1        AGACA,AAACA,AAACT,TGACT,AGAC,AAGAA\n",
       "24  ORF10      2     0      0                               TAACT,TAACT"
      ]
     },
     "execution_count": 287,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "import pandas as pd \n",
    "\n",
    "DRACH = re.compile('[AGT][AG]AC[ACT]') \n",
    "RGAC = re.compile('[AG]GAC') \n",
    "AAGAA = re.compile('AAGAA') \n",
    "\n",
    "def read_fasta(path):\n",
    "    file = open(path)\n",
    "    lines = file.read().splitlines()\n",
    "    ids = [s[1:] for s in lines if '>' in s]\n",
    "    n = [i for i,s in enumerate(lines) if '>' in s]\n",
    "    n.append(len(lines))\n",
    "    sequences = [''.join(lines[i+1:j]) for i,j in zip(n[:-1],n[1:])]\n",
    "    file.close()\n",
    "    fa = dict(zip(ids, sequences))\n",
    "    return fa\n",
    "\n",
    "def find_motifs(sam):\n",
    "    motifs = [[f.split('::')[0], \n",
    "         len(DRACH.findall(sam[f])), \n",
    "         len(RGAC.findall(sam[f])), \n",
    "         len(AAGAA.findall(sam[f])),\n",
    "         ','.join(DRACH.findall(sam[f]) + RGAC.findall(sam[f]) + AAGAA.findall(sam[f]))\n",
    "             ]\n",
    "        for n, f in enumerate(sam)\n",
    "              \n",
    "    ]\n",
    "    \n",
    "    df = pd.DataFrame(motifs, columns=[\"loci\",\"DRACH\", \"RGAC\", \"AAGAA\",\"Sequence\"])\n",
    "    \n",
    "    return df\n",
    "\n",
    "\n",
    "peaks = read_fasta('lowCt_peaks.fa')\n",
    "\n",
    "df = find_motifs(peaks)\n",
    "df.replace(\"\", nan_value, inplace=True)\n",
    "\n",
    "df.dropna(subset=[\"Sequence\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX4AAAEGCAYAAABiq/5QAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAYIklEQVR4nO3de5QcZZ3G8e/DMAujogMyYBKCcbOYlQVNcMy66oqgEggKgbMI6kEWOUY94gHXjRIviJc9oOMNz7q4QdC4i1yUEBHRwGa5rOdoYHIhCYYs6EZlEpNRGQEdMSS//aNq4mTS3VNzqa7urudzTp/ufru76mHo+nXlrbfeUkRgZmblsV/RAczMrL5c+M3MSsaF38ysZFz4zcxKxoXfzKxk9i86QBaHHnpozJgxo+gYZmZNZfXq1b+OiK6R7U1R+GfMmEFvb2/RMczMmoqkn1dqd1ePmVnJuPCbmZWMC7+ZWcm48JuZlYwLv5lZyTTFqJ5Ws3xtHz0rNrN1YJCpnR0smjeLBXOmFR3LzErChb/Olq/tY/GyDQzu3AVA38Agi5dtAHDxN7O6cFdPnfWs2Lyn6A8Z3LmLnhWbC0pkZmXjwl9nWwcGx9RuZjbZXPjrbGpnx5jazcwmmwt/nS2aN4uO9ra92jra21g0b1ZBicysbHxwt86GDuB6VI+ZFcWFvwAL5kxzoTezwrirx8ysZFz4zcxKxoXfzKxkciv8kg6UdJ+kByQ9KOnjaftlkvokrUtv8/PKYGZm+8rz4O5TwIkR8aSkduCHkr6fvvaFiPhsjus2M7Mqciv8ERHAk+nT9vQWea3PzMyyybWPX1KbpHXADuDOiFiVvnShpPWSrpV0cJXPLpTUK6m3v78/z5hmZqWSa+GPiF0RMRs4Apgr6RjgKmAmMBvYBnyuymeXRER3RHR3de1zkXgzMxunuozqiYgB4G7g5IjYnv4g7AauBubWI4OZmSXyHNXTJakzfdwBvA54SNKUYW87A9iYVwYzM9tXnqN6pgBLJbWR/MDcFBG3SfoPSbNJDvRuAd6ZYwYzMxshz1E964E5FdrPzWudZmY2Op+5a2ZWMi78ZmYl07LTMi9f2+c5783MKmjJwr98bR+Ll23Yc1HzvoFBFi/bAODib2al15JdPT0rNu8p+kMGd+6iZ8XmghKZmTWOliz8WwcGx9RuZlYmLVn4p3Z2jKndzKxMWrLwL5o3i472tr3aOtrbWDRvVkGJzMwaR0se3B06gOtRPWZm+2rJwg9J8XehNzPbV0t29ZiZWXUu/GZmJePCb2ZWMi78ZmYl48JvZlYyLvxmZiXjwm9mVjIu/GZmJZPnxdYPlHSfpAckPSjp42n7IZLulPRwen9wXhnMzGxfee7xPwWcGBEvAWYDJ0t6OXAJsDIijgJWps/NzKxOciv8kXgyfdqe3gI4HViati8FFuSVwczM9pVrH7+kNknrgB3AnRGxCjg8IrYBpPeHVfnsQkm9knr7+/vzjGlmViq5Fv6I2BURs4EjgLmSjhnDZ5dERHdEdHd1deUX0sysZOoyqiciBoC7gZOB7ZKmAKT3O+qRwczMEnmO6umS1Jk+7gBeBzwE3Aqcl77tPOA7eWUwM7N95Tkf/xRgqaQ2kh+YmyLiNkk/Am6SdAHwC+CsHDOYmdkIuRX+iFgPzKnQ/hvgtXmt18zMavOZu2ZmJePCb2ZWMi78ZmYl48JvZlYyLvxmZiXjwm9mVjIu/GZmJePCb2ZWMi78ZmYl48JvZlYyLvxmZiXjwm9mVjIu/GZmJePCb2ZWMi78ZmYlM2rhl3SRpGcrcY2kNZJOqkc4MzObfFn2+N8eEY8DJwFdwPnAFbmmMjOz3GQp/Erv5wNfi4gHhrVV/5A0XdJdkjZJelDSRWn7ZZL6JK1Lb/PHH9/MzMYqy6UXV0u6A3gBsFjSQcDuDJ97Gnh/RKxJP7Na0p3pa1+IiM+OL7KZmU1ElsJ/ATAb+FlE/EHSc0m6e2qKiG3AtvTxE5I2AdMmEtbMzCYuS1fPZRGxJiIG0ucDwAfHshJJM0guvL4qbbpQ0npJ10o6uMpnFkrqldTb398/ltWZmVkNWQr/kZIWA0g6ALgFeDjrCiQ9C7gZuDg9SHwVMJPkXxHbgM9V+lxELImI7ojo7urqyro6MzMbRZbCfz5wbFr8vwvcFRGXZVm4pHaSon9dRCwDiIjtEbErInYDVwNzx5XczMzGpWrhl3ScpONIumiuBM4m2dO/J22vSZKAa4BNEfH5Ye1Thr3tDGDjOLObmdk41Dq4O7IL5jHg6LQ9gBNHWfYrgXOBDZLWpW0fAt4saXa6jC3AO8eY2czMJqBq4Y+IEyay4Ij4IZXH+98+keWamdnEZBnOiaRTgb8BDhxqi4hP5BXKzMzyk2Wunq+Q9O+/l2QP/izg+TnnMjOznGQZ1fOKiHgb8FhEfBz4O2B6vrHMzCwvWQr/YHr/B0lTgZ0k0zeYmVkTytLHf5ukTqAHWEMyGueruaYyM7PcjFr4I+KT6cObJd0GHBgRv8s3lpmZ5SXLwd1nSPqopKsj4ingMElvqEM2MzPLQZY+/q8BT5Ec1AV4FPhUbonMzCxXWQr/zIj4DMlBXSJikAwXYjEzs8aUpfD/SVIHyUFdJM0k+ReAmZk1oSyjej4G/ACYLuk6kjl4/jHPUGZmlp8so3rulLQGeDlJF89FEfHr3JOZmVkuMs3VAxwPvIqku6ed5GIsZmbWhLIM5/w34F3ABpK5898p6ct5BzMzs3xk2eM/HjgmIoYO7i4l+REwM7MmlGVUz2bgyGHPpwPr84ljZmZ5y7LH/1xgk6T70ucvA34k6VaAiDgtr3BmZjb5shT+S3NPYWZmdZNlOOc941mwpOnAN4DnAbuBJRFxpaRDgBuBGSTX3H1TRDw2nnWYmdnYZenjH6+ngfdHxItIzgF4j6SjgUuAlRFxFLAyfW5mZnWSW+GPiG0RsSZ9/ASwCZgGnA4sTd+2FFiQVwYzM9tX1cIvaWV6/+mJrkTSDGAOsAo4PCK2QfLjABxW5TMLJfVK6u3v759oBDMzS9Xq458i6XjgNEk3MGJGzqG9+dFIehZwM3BxRDwuZZvYMyKWAEsAuru7I9OHzMxsVLUK/6Uk/e9HAJ8f8VoAJ462cEntJEX/uohYljZvlzQlIrZJmgLsGHtsMzMbr6qFPyK+DXxb0keHXX4xMyW79tcAmyJi+A/HrcB5wBXp/XfGumwzMxu/TNfclXQa8Oq06e6IuC3Dsl8JnAtskLQubfsQScG/SdIFwC+As8Ye28zMxmvUwi/pcmAucF3adJGkV0bE4lqfi4gfUv1KXa8dU0ozM5s0Wc7cPRWYHRG7Yc8kbWuBmoXfzMwaU9Zx/J3DHj8njyBmZlYfWfb4LwfWSrqLpOvm1Xhv38ysaWU5uHu9pLtJZuUU8MGI+FXewczMLB+ZLr2YnmF7a85ZzMysDvKcpM3MzBqQC7+ZWcnU7OqRtB+wPiKOqVOeprB8bR89KzazdWCQqZ0dLJo3iwVzppVm/VZe/u61hpqFPyJ2S3pA0pER8Yt6hWpky9f2sXjZBgZ37gKgb2CQxcuSa8/XYwMoev1WXv7utY4sXT1TgAclrZR069At72CNqmfF5j1f/CGDO3fRs2JzKdZv5eXvXuvIMqrn47mnaCJbBwbH1N5q67fy8nevdYy6x59ec3cL0J4+vh/INBd/K5ra2TGm9lZbv5WXv3utY9TCL+kdwLeBf0+bpgHL8wzVyBbNm0VHe9tebR3tbSyaN6sU67fy8nevdWTp6nkPyeycqwAi4mFJFS+XWAZDB7GKGtlQ9PqtvPzdax2KqH1VQ0mrIuJvJa2NiDmS9gfWRMSL6xMxufRib29vvVZnZtYSJK2OiO6R7VlG9dwj6UNAh6TXA98CvjvZAc3MrD6yFP5LgH5gA/BO4HbgI3mGMjOz/GSZnXN3evGVVSQXWd8co/UPmZlZw8oyqudU4KfAl4B/BR6RdEqGz10raYekjcPaLpPUJ2ldeps/kfBmZjZ2WUb1fA44ISIeAZA0E/ge8P1RPvd1kh+Kb4xo/0JEfHaMOc3MbJJk6ePfMVT0Uz8Ddoz2oYi4F/jteIOZmVk+qu7xSzozffigpNuBm0j6+M8iOXt3vC6U9DagF3h/RDxWZf0LgYUARx555ARWZ2Zmw9Xa439jejsQ2A4cD7yGZITPweNc31XATGA2sI2kG6miiFgSEd0R0d3V1TXO1ZmZ2UhV9/gj4vzJXllEbB96LOlq4LbJXoeZmdU26sFdSS8A3gvMGP7+iDhtrCuTNCW9fi/AGcDGWu83M7PJl2VUz3LgGpKzdXdnXbCk60m6hg6V9CjwMeA1kmaTHCvYQnJCmJmZ1VGWwv/HiPjSWBccEW+u0HzNWJdjZmaTK0vhv1LSx4A7gKeGGiOitHPym5k1syyF/1jgXOBE/tzVE+lzMzNrMlkK/xnAX0bEn/IOY2Zm+cty5u4DQGfeQczMrD6y7PEfDjwk6X727uMf83BOMzMrXpbC/7HcU5iZWd1kmY//nnoEMTOz+shy5u4TJKN4AP4CaAd+HxHPzjOYNb7la/t84W2zJpRlj/+g4c8lLQDm5pbImsLytX0sXraBwZ27AOgbGGTxsg0ALv5mDS7LqJ69RMRyPIa/9HpWbN5T9IcM7txFz4rNBSUys6yydPWcOezpfkA3f+76sZLaOjA4pnYzaxxZRvW8cdjjp0kmVzs9lzTWNKZ2dtBXochP7ewoII2ZjUWWPv5Jn5ffmt+iebP26uMH6GhvY9G8WQWmMrMsal168dIan4uI+GQOeaxJDB3A9ages+ZTa4//9xXanglcADwXcOEvuQVzprnQmzWhWpde3HM9XEkHARcB5wM3UONauWZm1thq9vFLOgT4J+CtwFLguIh4rB7BzMwsH7X6+HuAM4ElwLER8eRYFizpWuANwI6IOCZtOwS4keT6vVuAN/mHxGrx2cFmk6/WCVzvB6YCHwG2Sno8vT0h6fEMy/46cPKItkuAlRFxFLAyfW5W0dDZwX0DgwR/Pjt4+dq+oqOZNbWqhT8i9ouIjog4KCKePex2UJZ5eiLiXuC3I5pPJ+kyIr1fMO7k1vJ8drBZPsY8ZcMEHR4R2wDS+8OqvVHSQkm9knr7+/vrFtAah88ONstHvQt/ZhGxJCK6I6K7q6ur6DhWgGpnAfvsYLOJqXfh3y5pCkB6v6PO67cmsmjeLDra2/Zq89nBZhNX78J/K3Be+vg84Dt1Xr81kQVzpnH5mccyrbMDAdM6O7j8zGM9qsdsgrJM0jYukq4HXgMcKulRkks4XgHcJOkC4BfAWXmt31qDzw42m3y5Ff6IeHOVl16b1zrNzGx0DXtw18zM8uHCb2ZWMi78ZmYl48JvZlYyLvxmZiXjwm9mVjIu/GZmJePCb2ZWMi78ZmYl48JvZlYyLvxmZiXjwm9mVjIu/GZmJePCb2ZWMi78ZmYl48JvZlYyLvxmZiXjwm9mVjK5XXqxFklbgCeAXcDTEdFdRA4zszIqpPCnToiIXxe4fjOzUnJXj5lZyRRV+AO4Q9JqSQsrvUHSQkm9knr7+/vrHM/MrHUVVfhfGRHHAacA75H06pFviIglEdEdEd1dXV31T2hm1qIKKfwRsTW93wHcAswtIoeZWRnVvfBLeqakg4YeAycBG+udw8ysrIoY1XM4cIukofV/MyJ+UEAOM7NSqnvhj4ifAS+p93rNzCzh4ZxmZiXjwm9mVjJFnrlrZtbwlq/to2fFZrYODDK1s4NF82axYM60omNNiAu/mVkVy9f2sXjZBgZ37gKgb2CQxcs2ADR18XdXj5lZFT0rNu8p+kMGd+6iZ8XmghJNDhd+M7Mqtg4Mjqm9Wbjwm5lVMbWzY0ztzcKF38ysikXzZtHR3rZXW0d7G4vmzSoo0eTwwV0zsyqGDuB6VI+ZWYksmDOt6Qv9SO7qMTMrGRd+M7OSceE3MysZF34zs5Jx4TczKxkXfjOzknHhNzMrGRd+M7OSKeQELkknA1cCbcBXI+KKInKYFa0V53q3icv7e1H3wi+pDfgy8HrgUeB+SbdGxE/qncWsSK0617tNTD2+F0V09cwFHomIn0XEn4AbgNMLyGFWqFad690mph7fiyIK/zTgl8OeP5q27UXSQkm9knr7+/vrFs6sXlp1rnebmHp8L4oo/KrQFvs0RCyJiO6I6O7q6qpDLLP6atW53m1i6vG9KKLwPwpMH/b8CGBrATnMCtWqc73bxNTje1HEqJ77gaMkvQDoA84B3lJADrNCtepc7zYx9fheKGKfXpbcSZoPfJFkOOe1EfEvtd7f3d0dvb29dclmZtYqJK2OiO6R7YWM44+I24Hbi1i3mVnZ+cxdM7OSceE3MysZF34zs5Jx4TczK5lCRvWMlaR+4OfAocCvC44zVs5cP82YuxkzQ3PmLmPm50fEPmfANkXhHyKpt9LQpEbmzPXTjLmbMTM0Z25n/jN39ZiZlYwLv5lZyTRb4V9SdIBxcOb6acbczZgZmjO3M6eaqo/fzMwmrtn2+M3MbIJc+M3MSqZhC7+kayXtkLSxwmv/LCkkHVpEtmqqZZb0XkmbJT0o6TNF5aukUmZJsyX9WNK69Cpoc4vMOJKk6ZLukrQp/ZtelLYfIulOSQ+n9wcXnXW4Grl7JD0kab2kWyR1Fp11SLXMw15vuG2xVuYG3xarfT8mf3uMiIa8Aa8GjgM2jmifDqwgPaGr6JyjZQZOAP4LOCB9fljROTNkvgM4JX08H7i76JwjMk8BjksfHwT8L3A08BngkrT9EuDTRWfNmPskYP+0/dONlLta5vR5Q26LNf7Ojb4tVss96dtjw+7xR8S9wG8rvPQF4ANUuFxj0apkfjdwRUQ8lb5nR92D1VAlcwDPTh8/hwa7QlpEbIuINenjJ4BNJNdtPh1Ymr5tKbCgmISVVcsdEXdExNPp235MclW6hlDjbw0Nui3WyNzo22K13JO+PTZs4a9E0mlAX0Q8UHSWMXgh8PeSVkm6R9LLig6UwcVAj6RfAp8FFhecpypJM4A5wCrg8IjYBslGBBxWXLLaRuQe7u3A9+udJ4vhmZtlWxzxd26abXFE7knfHpum8Et6BvBh4NKis4zR/sDBwMuBRcBNkipdcL6RvBt4X0RMB94HXFNwnookPQu4Gbg4Ih4vOk9W1XJL+jDwNHBdUdmqGZ6ZJGPDb4sV/s5NsS1WyD3p22PTFH5gJvAC4AFJW0j+ObxG0vMKTTW6R4FlkbgP2E0y8VIjOw9Ylj7+FtBQB3cBJLWTbBzXRcRQ1u2SpqSvTwEa6p/yUDU3ks4D3gC8NdLO3EZRIXPDb4tV/s4Nvy1WyT3p22PTFP6I2BARh0XEjIiYQfI/8biI+FXB0UazHDgRQNILgb+g8WcI3Aocnz4+EXi4wCz7SPfSrgE2RcTnh710K8lGQnr/nXpnq6VabkknAx8ETouIPxSVr5JKmRt9W6zx/WjobbFG7snfHos+kl3jCPf1wDZgJ8kX64IRr2+hgUYSVMtM8uX6T2AjsAY4seicGTK/ClgNPEDSx/jSonOOyPwqkgNe64F16W0+8FxgZbphrAQOKTprxtyPAL8c1vaVorOOlnnEexpqW6zxd270bbFa7knfHj1lg5lZyTRNV4+ZmU0OF34zs5Jx4TczKxkXfjOzknHhNzMrGRd+KxVJz5N0g6SfSvqJpNvTMd3V3v9/kmaNaPuipA9UeG+mWTYlvTBd7yPpTIw3STo8nYVx/sT/K81qc+G30khPkLmFZHbDmRFxNPAh4PAaH7sBOGfYMvYD/gG4scJ77wSOiYgXk8ysuM+cKpIOBL4HXBURfxURLwKuArqA2STjts1y5cJvZXICsDMivjLUEBHrIuJ/lOiRtFHSBklnp2+5nmGFn2Qa6y0R8fORC49ss2y+BfhRRHx32OfuIvmh+ARwdjrv+tkVPms2KfYvOoBZHR1DcgZkJWeS7HG/hGT+lvsl3RsR6yXtlvSSSGaiPIfkx2A0b6fyvwoqZoiIP0m6FOiOiAszLN9s3LzHb5Z4FXB9ROyKiO3APcDQtL3XA+dI2p9kzv9v1VpQI8+yaQYu/FYuDwIvrfJarel5rwfeBLwOWB/pBTwkfS3tlrl9z0JGn2WzVgazunDhtzL5b+AASe8YapD0MknHA/eS9K+3Seoi6cu/DyAifgr8BriCYd08EXF+RMyOiPnpsrLMsvlN4BWSTh2W4WRJxwJPkFxyzyxXnqTNSkXSVOCLJHvdfySZWfJikhkyPwOcQjJD4qci4sZhn3sfcDnJVb5+V2XZjwAHkPxIAPw4It5V4X1/nWaYSTIr6nrgovTxCqAduHz4+s0mkwu/mVnJuKvHzKxkXPjNzErGhd/MrGRc+M3MSsaF38ysZFz4zcxKxoXfzKxk/h8zTc5TM51d+QAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd \n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "qPCR = pd.read_csv('RTqPCR.txt', sep='\\t',index_col=0)\n",
    "# create some x data and some integers for the y axis\n",
    "x = np.array(qPCR.iloc[:,0])\n",
    "y = np.array(qPCR.iloc[:,1], dtype=int)\n",
    "\n",
    "# plot the data\n",
    "# fig = plt.figure()\n",
    "# plt.scatter(x, y)\n",
    "# plt.xlabel('CoV-2 Ct', fontsize=10)\n",
    "# plt.ylabel('Number of peaks', fontsize=10)\n",
    "# fig.savefig('plots/Ct_vs_peak.pdf')\n",
    "plt.scatter(x, y)\n",
    "plt.xlabel('CoV-2 Ct', fontsize=10)\n",
    "plt.ylabel('Number of peaks', fontsize=10)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## internal control \n",
    "m6A+ m6A- [N6-Methyladenosine Enrichment Kit](https://international.neb.com/-/media/nebus/files/manuals/manuale1610.pdf?rev=c064d5e232414f709ef8bccee56f7687&hash=AAA94B8FDA043DD5A610E14686552B5E0209EA94)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CoV-2 Ct</th>\n",
       "      <th>Number of peaks</th>\n",
       "      <th>m6A_pos_ctrl</th>\n",
       "      <th>m6A_neg_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>S0008</td>\n",
       "      <td>19.6</td>\n",
       "      <td>10</td>\n",
       "      <td>3.683781</td>\n",
       "      <td>-2.090802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0009</td>\n",
       "      <td>18.5</td>\n",
       "      <td>12</td>\n",
       "      <td>6.804162</td>\n",
       "      <td>-0.219748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0014</td>\n",
       "      <td>16.3</td>\n",
       "      <td>37</td>\n",
       "      <td>2.141356</td>\n",
       "      <td>-3.560361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0017</td>\n",
       "      <td>24.0</td>\n",
       "      <td>17</td>\n",
       "      <td>1.651047</td>\n",
       "      <td>-7.253435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0025</td>\n",
       "      <td>22.1</td>\n",
       "      <td>0</td>\n",
       "      <td>6.203816</td>\n",
       "      <td>-1.010379</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0026</td>\n",
       "      <td>15.6</td>\n",
       "      <td>17</td>\n",
       "      <td>8.585372</td>\n",
       "      <td>2.581687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0030</td>\n",
       "      <td>17.0</td>\n",
       "      <td>17</td>\n",
       "      <td>8.069236</td>\n",
       "      <td>1.410737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0042</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.671282</td>\n",
       "      <td>-0.919416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0057</td>\n",
       "      <td>26.7</td>\n",
       "      <td>3</td>\n",
       "      <td>8.344047</td>\n",
       "      <td>-1.969256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S0085</td>\n",
       "      <td>14.6</td>\n",
       "      <td>33</td>\n",
       "      <td>6.607715</td>\n",
       "      <td>0.843608</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        CoV-2 Ct  Number of peaks  m6A_pos_ctrl  m6A_neg_ctrl\n",
       "Sample                                                       \n",
       "S0008       19.6               10      3.683781     -2.090802\n",
       "S0009       18.5               12      6.804162     -0.219748\n",
       "S0014       16.3               37      2.141356     -3.560361\n",
       "S0017       24.0               17      1.651047     -7.253435\n",
       "S0025       22.1                0      6.203816     -1.010379\n",
       "S0026       15.6               17      8.585372      2.581687\n",
       "S0030       17.0               17      8.069236      1.410737\n",
       "S0042       28.0                0      4.671282     -0.919416\n",
       "S0057       26.7                3      8.344047     -1.969256\n",
       "S0085       14.6               33      6.607715      0.843608"
      ]
     },
     "execution_count": 364,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qPCR = pd.read_csv('RTqPCR.txt', sep='\\t',index_col=0)\n",
    "\n",
    "res = pd.read_csv('internal_control/result.txt', sep='\\t')\n",
    "\n",
    "pos  = [np.log2(res.loc[i+1,'pos']/res.loc[i,'pos']) for i in list(range(0,res.shape[0],2))]\n",
    "neg  = [np.log2(res.loc[i+1,'neg']/res.loc[i,'neg']) for i in list(range(0,res.shape[0],2))]\n",
    "\n",
    "Samples = [res.iloc[i,0].split('_')[0] for i in list(range(0,res.shape[0],2))]\n",
    "\n",
    "log2res = pd.DataFrame(index=Samples)\n",
    "\n",
    "qPCR['m6A_pos_ctrl'] = pos\n",
    "qPCR['m6A_neg_ctrl'] = neg\n",
    "\n",
    "qPCR\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 421,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34m200429_NS500257_0111_AH2V3CBGXF\u001b[0m/  \u001b[01;34mfastqI\u001b[0m/            \u001b[01;34mplots\u001b[0m/      \u001b[01;34mvirus\u001b[0m/\r\n",
      "\u001b[01;34mbam\u001b[0m/                              \u001b[01;34minternal_control\u001b[0m/  \u001b[01;34mqc_star\u001b[0m/    \u001b[01;34mvirus_bam\u001b[0m/\r\n",
      "\u001b[01;34mcell_paper\u001b[0m/                       lowCt_peaks.bed    RTqPCR.txt  \u001b[01;34mvirus_count\u001b[0m/\r\n",
      "demultiplex_index.txt             lowCt_peaks.c.bed  \u001b[01;34mrumi\u001b[0m/       \u001b[01;34mvirus_fastq\u001b[0m/\r\n",
      "\u001b[01;34mexomepeak\u001b[0m/                        lowCt_peaks.fa     \u001b[01;34m_sh\u001b[0m/        \u001b[01;31mvirus.zip\u001b[0m\r\n",
      "\u001b[01;34mfastq\u001b[0m/                            notebook.ipynb     \u001b[01;34mtrim\u001b[0m/\r\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 422,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S0008</th>\n",
       "      <th>S0009</th>\n",
       "      <th>S0014</th>\n",
       "      <th>S0017</th>\n",
       "      <th>S0025</th>\n",
       "      <th>S0026</th>\n",
       "      <th>S0030</th>\n",
       "      <th>S0042</th>\n",
       "      <th>S0057</th>\n",
       "      <th>S0085</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>N_peak</td>\n",
       "      <td>-1.419539</td>\n",
       "      <td>3.094327</td>\n",
       "      <td>-5.804827</td>\n",
       "      <td>-1.772947</td>\n",
       "      <td>2.837167</td>\n",
       "      <td>3.443329</td>\n",
       "      <td>3.418020</td>\n",
       "      <td>2.783654</td>\n",
       "      <td>0.171242</td>\n",
       "      <td>2.433774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_10</td>\n",
       "      <td>-0.850783</td>\n",
       "      <td>4.094327</td>\n",
       "      <td>-5.197406</td>\n",
       "      <td>-1.772098</td>\n",
       "      <td>2.935536</td>\n",
       "      <td>2.589521</td>\n",
       "      <td>1.674362</td>\n",
       "      <td>2.786430</td>\n",
       "      <td>0.165586</td>\n",
       "      <td>1.135859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_11</td>\n",
       "      <td>-2.074677</td>\n",
       "      <td>4.657112</td>\n",
       "      <td>-3.714559</td>\n",
       "      <td>-1.777989</td>\n",
       "      <td>2.869887</td>\n",
       "      <td>3.204982</td>\n",
       "      <td>1.099996</td>\n",
       "      <td>2.809540</td>\n",
       "      <td>0.161692</td>\n",
       "      <td>1.556074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_12</td>\n",
       "      <td>-1.448000</td>\n",
       "      <td>4.036723</td>\n",
       "      <td>-1.126578</td>\n",
       "      <td>-1.793937</td>\n",
       "      <td>2.918386</td>\n",
       "      <td>3.147463</td>\n",
       "      <td>2.667740</td>\n",
       "      <td>2.836804</td>\n",
       "      <td>0.153936</td>\n",
       "      <td>2.528701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_13</td>\n",
       "      <td>-2.492014</td>\n",
       "      <td>3.499571</td>\n",
       "      <td>-2.246635</td>\n",
       "      <td>-1.794465</td>\n",
       "      <td>2.754888</td>\n",
       "      <td>3.156091</td>\n",
       "      <td>2.930574</td>\n",
       "      <td>2.838393</td>\n",
       "      <td>0.150073</td>\n",
       "      <td>2.597607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_14</td>\n",
       "      <td>-2.176355</td>\n",
       "      <td>4.270089</td>\n",
       "      <td>0.599944</td>\n",
       "      <td>-1.803421</td>\n",
       "      <td>2.861803</td>\n",
       "      <td>3.304903</td>\n",
       "      <td>1.235407</td>\n",
       "      <td>2.838393</td>\n",
       "      <td>0.150073</td>\n",
       "      <td>1.546360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_15</td>\n",
       "      <td>-0.905478</td>\n",
       "      <td>3.881819</td>\n",
       "      <td>-6.441363</td>\n",
       "      <td>-1.797105</td>\n",
       "      <td>2.872943</td>\n",
       "      <td>2.004188</td>\n",
       "      <td>1.919031</td>\n",
       "      <td>2.838393</td>\n",
       "      <td>0.153936</td>\n",
       "      <td>1.407446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_16</td>\n",
       "      <td>-2.490233</td>\n",
       "      <td>3.783606</td>\n",
       "      <td>-5.275955</td>\n",
       "      <td>-1.819611</td>\n",
       "      <td>2.757938</td>\n",
       "      <td>1.941162</td>\n",
       "      <td>2.566483</td>\n",
       "      <td>2.827078</td>\n",
       "      <td>0.138547</td>\n",
       "      <td>1.812151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_2</td>\n",
       "      <td>-1.673872</td>\n",
       "      <td>2.795986</td>\n",
       "      <td>-2.128884</td>\n",
       "      <td>-1.772947</td>\n",
       "      <td>2.793446</td>\n",
       "      <td>3.933286</td>\n",
       "      <td>3.410968</td>\n",
       "      <td>2.775465</td>\n",
       "      <td>0.167295</td>\n",
       "      <td>2.376512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_3</td>\n",
       "      <td>-2.568117</td>\n",
       "      <td>2.921217</td>\n",
       "      <td>-0.797945</td>\n",
       "      <td>-1.765427</td>\n",
       "      <td>2.870850</td>\n",
       "      <td>4.309560</td>\n",
       "      <td>2.948174</td>\n",
       "      <td>2.779276</td>\n",
       "      <td>0.177331</td>\n",
       "      <td>2.717156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_4</td>\n",
       "      <td>-2.121263</td>\n",
       "      <td>3.764622</td>\n",
       "      <td>-1.829798</td>\n",
       "      <td>-1.758617</td>\n",
       "      <td>3.001554</td>\n",
       "      <td>4.087463</td>\n",
       "      <td>2.636141</td>\n",
       "      <td>2.783098</td>\n",
       "      <td>0.177331</td>\n",
       "      <td>1.638275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_5</td>\n",
       "      <td>-2.527613</td>\n",
       "      <td>2.920076</td>\n",
       "      <td>-4.007915</td>\n",
       "      <td>-1.783325</td>\n",
       "      <td>2.934002</td>\n",
       "      <td>3.014523</td>\n",
       "      <td>3.360256</td>\n",
       "      <td>2.779276</td>\n",
       "      <td>0.173406</td>\n",
       "      <td>1.741232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_6</td>\n",
       "      <td>-2.773115</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>-3.846962</td>\n",
       "      <td>-1.773171</td>\n",
       "      <td>2.874981</td>\n",
       "      <td>3.056565</td>\n",
       "      <td>3.622800</td>\n",
       "      <td>2.783098</td>\n",
       "      <td>0.173406</td>\n",
       "      <td>1.985089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_7</td>\n",
       "      <td>-2.215455</td>\n",
       "      <td>3.963931</td>\n",
       "      <td>-1.237134</td>\n",
       "      <td>-1.766182</td>\n",
       "      <td>2.995408</td>\n",
       "      <td>3.862312</td>\n",
       "      <td>2.631278</td>\n",
       "      <td>2.783098</td>\n",
       "      <td>0.165586</td>\n",
       "      <td>2.262494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_8</td>\n",
       "      <td>-2.165132</td>\n",
       "      <td>3.690212</td>\n",
       "      <td>0.515587</td>\n",
       "      <td>-1.764024</td>\n",
       "      <td>2.995408</td>\n",
       "      <td>4.608671</td>\n",
       "      <td>3.041087</td>\n",
       "      <td>2.783098</td>\n",
       "      <td>0.161692</td>\n",
       "      <td>1.923623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>N_peak_9</td>\n",
       "      <td>-2.272276</td>\n",
       "      <td>3.645569</td>\n",
       "      <td>-1.083211</td>\n",
       "      <td>-1.774243</td>\n",
       "      <td>2.995408</td>\n",
       "      <td>3.631117</td>\n",
       "      <td>1.764446</td>\n",
       "      <td>2.785875</td>\n",
       "      <td>0.165586</td>\n",
       "      <td>1.985561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ORF10_peak</td>\n",
       "      <td>-2.473275</td>\n",
       "      <td>3.876932</td>\n",
       "      <td>-3.018750</td>\n",
       "      <td>-1.831506</td>\n",
       "      <td>2.799758</td>\n",
       "      <td>5.025034</td>\n",
       "      <td>4.748832</td>\n",
       "      <td>2.845687</td>\n",
       "      <td>0.134725</td>\n",
       "      <td>3.288537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S_peak</td>\n",
       "      <td>-1.641502</td>\n",
       "      <td>4.135481</td>\n",
       "      <td>-1.767023</td>\n",
       "      <td>-2.117893</td>\n",
       "      <td>3.277483</td>\n",
       "      <td>3.479752</td>\n",
       "      <td>2.364363</td>\n",
       "      <td>2.788731</td>\n",
       "      <td>3.187179</td>\n",
       "      <td>2.701556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S_peak_2</td>\n",
       "      <td>-1.866901</td>\n",
       "      <td>3.950561</td>\n",
       "      <td>0.621642</td>\n",
       "      <td>-2.142688</td>\n",
       "      <td>3.153304</td>\n",
       "      <td>3.337010</td>\n",
       "      <td>2.162595</td>\n",
       "      <td>2.743000</td>\n",
       "      <td>-1.659925</td>\n",
       "      <td>3.914647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S_peak_3</td>\n",
       "      <td>-1.644825</td>\n",
       "      <td>5.029134</td>\n",
       "      <td>-3.613739</td>\n",
       "      <td>-2.121836</td>\n",
       "      <td>3.277483</td>\n",
       "      <td>2.611435</td>\n",
       "      <td>2.531343</td>\n",
       "      <td>2.783698</td>\n",
       "      <td>-1.650765</td>\n",
       "      <td>1.892170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S_peak_4</td>\n",
       "      <td>-1.691631</td>\n",
       "      <td>4.017020</td>\n",
       "      <td>-3.780658</td>\n",
       "      <td>-2.121836</td>\n",
       "      <td>3.214058</td>\n",
       "      <td>2.443879</td>\n",
       "      <td>1.316422</td>\n",
       "      <td>2.766051</td>\n",
       "      <td>-1.650765</td>\n",
       "      <td>3.024820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S_peak_5</td>\n",
       "      <td>-1.667873</td>\n",
       "      <td>4.212276</td>\n",
       "      <td>-3.485427</td>\n",
       "      <td>-2.119208</td>\n",
       "      <td>3.290219</td>\n",
       "      <td>2.804131</td>\n",
       "      <td>1.481127</td>\n",
       "      <td>2.766051</td>\n",
       "      <td>-1.650765</td>\n",
       "      <td>1.886521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S_peak_6</td>\n",
       "      <td>-1.652548</td>\n",
       "      <td>4.299560</td>\n",
       "      <td>-3.480067</td>\n",
       "      <td>-2.119866</td>\n",
       "      <td>3.229655</td>\n",
       "      <td>1.822859</td>\n",
       "      <td>1.410358</td>\n",
       "      <td>2.770442</td>\n",
       "      <td>-1.650765</td>\n",
       "      <td>1.292429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S_peak_7</td>\n",
       "      <td>-1.696461</td>\n",
       "      <td>4.028614</td>\n",
       "      <td>-1.237665</td>\n",
       "      <td>-2.151717</td>\n",
       "      <td>3.405992</td>\n",
       "      <td>2.719731</td>\n",
       "      <td>2.023514</td>\n",
       "      <td>2.781360</td>\n",
       "      <td>-1.650765</td>\n",
       "      <td>2.420010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>S_peak_8</td>\n",
       "      <td>-1.732137</td>\n",
       "      <td>3.905759</td>\n",
       "      <td>-4.039405</td>\n",
       "      <td>-2.152360</td>\n",
       "      <td>3.199672</td>\n",
       "      <td>1.780219</td>\n",
       "      <td>1.219527</td>\n",
       "      <td>2.780802</td>\n",
       "      <td>-1.659925</td>\n",
       "      <td>1.897726</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               S0008     S0009     S0014     S0017     S0025     S0026  \\\n",
       "N_peak     -1.419539  3.094327 -5.804827 -1.772947  2.837167  3.443329   \n",
       "N_peak_10  -0.850783  4.094327 -5.197406 -1.772098  2.935536  2.589521   \n",
       "N_peak_11  -2.074677  4.657112 -3.714559 -1.777989  2.869887  3.204982   \n",
       "N_peak_12  -1.448000  4.036723 -1.126578 -1.793937  2.918386  3.147463   \n",
       "N_peak_13  -2.492014  3.499571 -2.246635 -1.794465  2.754888  3.156091   \n",
       "N_peak_14  -2.176355  4.270089  0.599944 -1.803421  2.861803  3.304903   \n",
       "N_peak_15  -0.905478  3.881819 -6.441363 -1.797105  2.872943  2.004188   \n",
       "N_peak_16  -2.490233  3.783606 -5.275955 -1.819611  2.757938  1.941162   \n",
       "N_peak_2   -1.673872  2.795986 -2.128884 -1.772947  2.793446  3.933286   \n",
       "N_peak_3   -2.568117  2.921217 -0.797945 -1.765427  2.870850  4.309560   \n",
       "N_peak_4   -2.121263  3.764622 -1.829798 -1.758617  3.001554  4.087463   \n",
       "N_peak_5   -2.527613  2.920076 -4.007915 -1.783325  2.934002  3.014523   \n",
       "N_peak_6   -2.773115  3.000000 -3.846962 -1.773171  2.874981  3.056565   \n",
       "N_peak_7   -2.215455  3.963931 -1.237134 -1.766182  2.995408  3.862312   \n",
       "N_peak_8   -2.165132  3.690212  0.515587 -1.764024  2.995408  4.608671   \n",
       "N_peak_9   -2.272276  3.645569 -1.083211 -1.774243  2.995408  3.631117   \n",
       "ORF10_peak -2.473275  3.876932 -3.018750 -1.831506  2.799758  5.025034   \n",
       "S_peak     -1.641502  4.135481 -1.767023 -2.117893  3.277483  3.479752   \n",
       "S_peak_2   -1.866901  3.950561  0.621642 -2.142688  3.153304  3.337010   \n",
       "S_peak_3   -1.644825  5.029134 -3.613739 -2.121836  3.277483  2.611435   \n",
       "S_peak_4   -1.691631  4.017020 -3.780658 -2.121836  3.214058  2.443879   \n",
       "S_peak_5   -1.667873  4.212276 -3.485427 -2.119208  3.290219  2.804131   \n",
       "S_peak_6   -1.652548  4.299560 -3.480067 -2.119866  3.229655  1.822859   \n",
       "S_peak_7   -1.696461  4.028614 -1.237665 -2.151717  3.405992  2.719731   \n",
       "S_peak_8   -1.732137  3.905759 -4.039405 -2.152360  3.199672  1.780219   \n",
       "\n",
       "               S0030     S0042     S0057     S0085  \n",
       "N_peak      3.418020  2.783654  0.171242  2.433774  \n",
       "N_peak_10   1.674362  2.786430  0.165586  1.135859  \n",
       "N_peak_11   1.099996  2.809540  0.161692  1.556074  \n",
       "N_peak_12   2.667740  2.836804  0.153936  2.528701  \n",
       "N_peak_13   2.930574  2.838393  0.150073  2.597607  \n",
       "N_peak_14   1.235407  2.838393  0.150073  1.546360  \n",
       "N_peak_15   1.919031  2.838393  0.153936  1.407446  \n",
       "N_peak_16   2.566483  2.827078  0.138547  1.812151  \n",
       "N_peak_2    3.410968  2.775465  0.167295  2.376512  \n",
       "N_peak_3    2.948174  2.779276  0.177331  2.717156  \n",
       "N_peak_4    2.636141  2.783098  0.177331  1.638275  \n",
       "N_peak_5    3.360256  2.779276  0.173406  1.741232  \n",
       "N_peak_6    3.622800  2.783098  0.173406  1.985089  \n",
       "N_peak_7    2.631278  2.783098  0.165586  2.262494  \n",
       "N_peak_8    3.041087  2.783098  0.161692  1.923623  \n",
       "N_peak_9    1.764446  2.785875  0.165586  1.985561  \n",
       "ORF10_peak  4.748832  2.845687  0.134725  3.288537  \n",
       "S_peak      2.364363  2.788731  3.187179  2.701556  \n",
       "S_peak_2    2.162595  2.743000 -1.659925  3.914647  \n",
       "S_peak_3    2.531343  2.783698 -1.650765  1.892170  \n",
       "S_peak_4    1.316422  2.766051 -1.650765  3.024820  \n",
       "S_peak_5    1.481127  2.766051 -1.650765  1.886521  \n",
       "S_peak_6    1.410358  2.770442 -1.650765  1.292429  \n",
       "S_peak_7    2.023514  2.781360 -1.650765  2.420010  \n",
       "S_peak_8    1.219527  2.780802 -1.659925  1.897726  "
      ]
     },
     "execution_count": 422,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "#list the files\n",
    "filelist = os.listdir('virus_count')\n",
    "\n",
    "#read them into pandas\n",
    "df_list = [pd.read_csv('virus_count/' + file, header=None, sep='\\t', index_col=0) for file in filelist]\n",
    "#concatenate them together\n",
    "big_df = pd.concat(df_list, axis=1)\n",
    "big_df.columns = [f.split('.')[0] for f in filelist]\n",
    "\n",
    "Peak_names = big_df.index.tolist()\n",
    "Samples = np.unique([f.split('_')[0] for f in filelist]).tolist()\n",
    "\n",
    "res = list()\n",
    "for s in Samples:\n",
    "    data = big_df.filter(like=s)\n",
    "    res.append([s, np.log2 (np.array(data.filter(like='RIP'), dtype=int) / np.array(data.filter(like='IN'), dtype=int) )])\n",
    "    \n",
    "\n",
    "res_df = pd.DataFrame(data=np.concatenate([r[1] for r in res], axis=1), \n",
    "             columns=[r[0] for r in res],\n",
    "             index=Peak_names)\n",
    "res_df.to_csv('log2RIPvsIN.txt')\n",
    "res_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "###################### pre processing #############################\r\n",
      "conda activate deseq \r\n",
      "\r\n",
      "mkdir -p trim\r\n",
      "for fq in fastq/S00*R2*; do\r\n",
      "    fq=`basename $fq`;\r\n",
      "    out=${fq/_001.fastq.gz/.trim.fastq.gz};\r\n",
      "    echo -------------------------$fq------------------------\r\n",
      "    cutadapt -j 12 -u 3 -o trim/$out fastq/$fq\r\n",
      "done\r\n",
      "\r\n",
      "# map to human genome \r\n",
      "mkdir -p bam\r\n",
      "\r\n",
      "STAR --genomeLoad LoadAndExit --genomeDir /rumi/shams/genomes/hg38/\r\n",
      "\r\n",
      "for fq in fastq/*_R1*; do\r\n",
      "    fq1=`basename $fq`;\r\n",
      "    fq2=${fq1/R1_001.fastq.gz/R2.trim.fastq.gz};\r\n",
      "    out=${fq1/R1_001.fastq.gz/};\r\n",
      "    STAR \\\r\n",
      "    --outSAMtype BAM SortedByCoordinate \\\r\n",
      "    --readFilesCommand zcat \\\r\n",
      "    --runThreadN 16 \\\r\n",
      "    --genomeDir /rumi/shams/genomes/hg38/ \\\r\n",
      "    --readFilesIn fastq/$fq1 trim/$fq2 \\\r\n",
      "    --outFileNamePrefix bam/$out \\\r\n",
      "    --outReadsUnmapped Fastx;\r\n",
      "done\r\n",
      "\r\n",
      "STAR --genomeLoad Remove --genomeDir /rumi/shams/genomes/hg38/\r\n",
      "rm Aligned.out.sam Log*out\r\n",
      "rm -r bam/Undetermined_S0*\r\n",
      "\r\n",
      "mkdir -p virus_fastq\r\n",
      "mv bam/*_Unmapped.out.mate[1-2] virus_fastq/\r\n",
      "\r\n",
      "mkdir -p qc_star/\r\n",
      "mkdir -p qc_star/hg38\r\n",
      "\r\n",
      "mv -v bam/*out qc_star/hg38/\r\n",
      "mv -v bam/*tab qc_star/hg38/\r\n",
      "\r\n",
      "for f in bam/*; do r=${f/_S[0-9]*_Aligned.sortedByCoord.out.bam/.bam}; mv -v $f $r; done \r\n",
      "\r\n",
      "# map to virus genome \r\n",
      "# https://www.ncbi.nlm.nih.gov/nuccore/MN908947?%3Fdb=nucleotide\r\n",
      "STAR --runThreadN 16 \\\r\n",
      "--runMode genomeGenerate \\\r\n",
      "--genomeDir virus/ \\\r\n",
      "--genomeFastaFiles virus/coronavirus_2_isolate_Wuhan-Hu-1.fasta \\\r\n",
      "--sjdbGTFfile virus/coronavirus_2_isolate_Wuhan-Hu-1.gff3 \\\r\n",
      "--sjdbOverhang 99 --genomeSAindexNbases 8 --sjdbGTFfeatureExon CDS\r\n",
      "\r\n",
      "mkdir -p virus_bam\r\n",
      "STAR --genomeLoad LoadAndExit --genomeDir virus/\r\n",
      "\r\n",
      "for fq in virus_fastq/*mate1; do\r\n",
      "    fq1=`basename $fq`;\r\n",
      "    fq2=${fq1/mate1/mate2};\r\n",
      "    out=${fq1/_Unmapped.out.mate1/};\r\n",
      "    STAR \\\r\n",
      "    --outSAMtype BAM SortedByCoordinate \\\r\n",
      "    --readFilesCommand cat \\\r\n",
      "    --runThreadN 16 \\\r\n",
      "    --genomeDir virus/ \\\r\n",
      "    --readFilesIn virus_fastq/$fq1 virus_fastq/$fq2 \\\r\n",
      "    --outFileNamePrefix virus_bam/$out \\\r\n",
      "    --limitBAMsortRAM 1000000000;\r\n",
      "done\r\n",
      "STAR --genomeLoad Remove --genomeDir virus/\r\n",
      "\r\n",
      "###################### peak callingg #############################\r\n",
      "conda activate exomepeak\r\n",
      "for i in `awk 'NR>1{print $1}' RTqPCR.txt`; do \r\n",
      "Rscript _sh/exompeak.R \\\r\n",
      "~/Projects/COVID19_m6A \\\r\n",
      "~/Projects/COVID19_m6A/exomepeak/human/ \\\r\n",
      "$i _IN _RIP;\r\n",
      "done\r\n",
      "\r\n",
      "conda activate guitar\r\n",
      "for i in `awk 'NR>1{print $1}' RTqPCR.txt`; do \r\n",
      "Rscript _sh/guitar.R ./exomepeak/human/$i peak.bed $i\r\n",
      "done \r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "###################### internal ctrl #############################\r\n",
      "conda activate bowtie2\r\n",
      "\r\n",
      "bowtie2-build internal_control/m6A_pos.fa internal_control/index/m6A_pos\r\n",
      "\r\n",
      "for f in virus_fastq/*mate1; \r\n",
      "\tdo f=`basename $f`; \r\n",
      "\tf2=${f/mate1/mate2}; \r\n",
      "\to=${f/_Unmapped.out.mate1/.bam}; \r\n",
      "\tbowtie2 -p 12 --sensitive -N 1 -x internal_control/index/m6A_pos -1 virus_fastq/$f -2 virus_fastq/$f2 | samtools sort -o internal_control/virus/m6A_pos.$o; \r\n",
      "done\r\n",
      "\r\n",
      "bowtie2-build internal_control/m6A_neg.fa internal_control/index/m6A_neg\r\n",
      "\r\n",
      "for f in virus_fastq/*mate1; \r\n",
      "        do f=`basename $f`; \r\n",
      "        f2=${f/mate1/mate2}; \r\n",
      "        o=${f/_Unmapped.out.mate1/.bam}; \r\n",
      "        bowtie2 -p 12 --sensitive -N 1 -x internal_control/index/m6A_neg -1 virus_fastq/$f -2 virus_fastq/$f2 | samtools sort -o internal_control/virus/m6A_neg.$o; \r\n",
      "done\r\n",
      "\r\n",
      "###################### motif analysis #############################\r\n",
      "conda activate cgat \r\n",
      "# to select low Ct samples \r\n",
      "awk 'NR>1 && $2<20 {print $1, $2}' RTqPCR.txt\r\n",
      "\r\n",
      "intersectBed -u \\\r\n",
      "-a exomepeak/virus/S0085/S0085/peak.bed \\\r\n",
      "-b exomepeak/virus/S0008/S0008/peak.bed \\\r\n",
      "exomepeak/virus/S0009/S0009/peak.bed \\\r\n",
      "exomepeak/virus/S0014/S0014/peak.bed \\\r\n",
      "exomepeak/virus/S0026/S0026/peak.bed \\\r\n",
      "exomepeak/virus/S0030/S0030/peak.bed > lowCt_peaks.bed\r\n",
      "\r\n",
      "cat lowCt_peaks.bed | sort -k1,1 -k2,2n | cgat bed2bed --method=merge --merge-by-name |  awk '! /#/' | bedtools getfasta -name -s -fi virus/coronavirus_2_isolate_Wuhan-Hu-1.fasta -bed - -split -fo lowCt_peaks.fa\r\n",
      "awk -F \"\\t\" '{OFS=FS}{$4=$4\"_peak\"}; cnt[$4]++{$4=$4\"_\"cnt[$4]} 1' lowCt_peaks.bed > lowCt_peaks.c.bed\r\n",
      "# count peaks \r\n",
      "for f in virus_bam/*bam; \r\n",
      "\tdo base=`basename $f`; \r\n",
      "\tout=${base/.bam/.fc}; \r\n",
      "\tbamToBed -i $f | intersectBed -split -s -wo -a - -b lowCt_peaks.c.bed | cut -f10 | sort | uniq -c | awk '{ print $2 \"\\t\" $1}' > virus_count/$out; \r\n",
      "done \r\n"
     ]
    }
   ],
   "source": [
    "cat _sh/pipeline.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
