{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mk_dir(p):\n",
    "    if not os.path.exists(p):\n",
    "        os.mkdir(p)\n",
    "\n",
    "mk_dir('./qc_fastq')\n",
    "mk_dir('./deseq')\n",
    "mk_dir('./plots')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Input_T1_S43_L006_R1_001.fastq.gz\" \"Input_T2_S44_L006_R1_001.fastq.gz\"\n",
      "[3] \"Input_U1_S41_L006_R1_001.fastq.gz\" \"Input_U2_S42_L006_R1_001.fastq.gz\"\n",
      "[5] \"IP_T1_S47_L006_R1_001.fastq.gz\"    \"IP_T2_S48_L006_R1_001.fastq.gz\"   \n",
      "[7] \"IP_U1_S45_L006_R1_001.fastq.gz\"    \"IP_U2_S46_L006_R1_001.fastq.gz\"   \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "list.files('fastq')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I've started from bam files previously mapped by Hani or Alex. All `bam`  files renamed in this format: \n",
    "> `<treatment>.<input/m6A>.bam`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd bam/\n",
    "for f in Input_*; do \n",
    "    o=${f/Input_/};o=${o/_S4[1-4].bam/.input.bam}; mv $f $o; \n",
    "done\n",
    "for f in IP_*; do \n",
    "    o=${f/IP_/};o=${o/_S4[5-8].bam/.m6A.bam}; mv $f $o; \n",
    "done\n",
    "cd ../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"T1.input.bam\" \"T1.m6A.bam\"   \"T2.input.bam\" \"T2.m6A.bam\"   \"U1.input.bam\"\n",
      "[6] \"U1.m6A.bam\"   \"U2.input.bam\" \"U2.m6A.bam\"  \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "list.files('bam')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peakcalling \n",
    "\n",
    "## Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TxDb object:\n",
      "# Db type: TxDb\n",
      "# Supporting package: GenomicFeatures\n",
      "# Data source: /rumi/shams/genomes/hg38/gencode.v28.annotation.gtf\n",
      "# Organism: Homo sapiens\n",
      "# Taxonomy ID: 9606\n",
      "# miRBase build ID: NA\n",
      "# Genome: NA\n",
      "# transcript_nrow: 203835\n",
      "# exon_nrow: 693612\n",
      "# cds_nrow: 273085\n",
      "# Db created by: GenomicFeatures package from Bioconductor\n",
      "# Creation time: 2020-04-08 01:53:01 +0000 (Wed, 08 Apr 2020)\n",
      "# GenomicFeatures version at creation time: 1.32.3\n",
      "# RSQLite version at creation time: 2.1.5\n",
      "# DBSCHEMAVERSION: 1.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Divide transcriptome into chr-gene-batch sections ...\"\n",
      "[1] \"Get Reads Count ...\"\n",
      "[1] \"This step may take a few hours ...\"\n",
      "[1] \"0.286 %\"\n",
      "[1] \"0.571 %\"\n",
      "[1] \"0.857 %\"\n",
      "[1] \"1.14 %\"\n",
      "[1] \"1.43 %\"\n",
      "[1] \"1.71 %\"\n",
      "[1] \"2 %\"\n",
      "[1] \"2.29 %\"\n",
      "[1] \"2.57 %\"\n",
      "[1] \"2.86 %\"\n",
      "[1] \"3.14 %\"\n",
      "[1] \"3.43 %\"\n",
      "[1] \"3.71 %\"\n",
      "[1] \"4 %\"\n",
      "[1] \"4.29 %\"\n",
      "[1] \"4.57 %\"\n",
      "[1] \"4.86 %\"\n",
      "[1] \"5.14 %\"\n",
      "[1] \"5.43 %\"\n",
      "[1] \"5.71 %\"\n",
      "[1] \"6 %\"\n",
      "[1] \"6.29 %\"\n",
      "[1] \"6.57 %\"\n",
      "[1] \"6.86 %\"\n",
      "[1] \"7.14 %\"\n",
      "[1] \"7.43 %\"\n",
      "[1] \"7.71 %\"\n",
      "[1] \"8 %\"\n",
      "[1] \"8.29 %\"\n",
      "[1] \"8.57 %\"\n",
      "[1] \"8.86 %\"\n",
      "[1] \"9.14 %\"\n",
      "[1] \"9.43 %\"\n",
      "[1] \"9.71 %\"\n",
      "[1] \"10 %\"\n",
      "[1] \"10.3 %\"\n",
      "[1] \"10.6 %\"\n",
      "[1] \"10.9 %\"\n",
      "[1] \"11.1 %\"\n",
      "[1] \"11.4 %\"\n",
      "[1] \"11.7 %\"\n",
      "[1] \"12 %\"\n",
      "[1] \"12.3 %\"\n",
      "[1] \"12.6 %\"\n",
      "[1] \"12.9 %\"\n",
      "[1] \"13.1 %\"\n",
      "[1] \"13.4 %\"\n",
      "[1] \"13.7 %\"\n",
      "[1] \"14 %\"\n",
      "[1] \"14.3 %\"\n",
      "[1] \"14.6 %\"\n",
      "[1] \"14.9 %\"\n",
      "[1] \"15.1 %\"\n",
      "[1] \"15.4 %\"\n",
      "[1] \"15.7 %\"\n",
      "[1] \"16 %\"\n",
      "[1] \"16.3 %\"\n",
      "[1] \"16.6 %\"\n",
      "[1] \"16.9 %\"\n",
      "[1] \"17.1 %\"\n",
      "[1] \"17.4 %\"\n",
      "[1] \"17.7 %\"\n",
      "[1] \"18 %\"\n",
      "[1] \"18.3 %\"\n",
      "[1] \"18.6 %\"\n",
      "[1] \"18.9 %\"\n",
      "[1] \"19.1 %\"\n",
      "[1] \"19.4 %\"\n",
      "[1] \"19.7 %\"\n",
      "[1] \"20 %\"\n",
      "[1] \"20.3 %\"\n",
      "[1] \"20.6 %\"\n",
      "[1] \"20.9 %\"\n",
      "[1] \"21.1 %\"\n",
      "[1] \"21.4 %\"\n",
      "[1] \"21.7 %\"\n",
      "[1] \"22 %\"\n",
      "[1] \"22.3 %\"\n",
      "[1] \"22.6 %\"\n",
      "[1] \"22.9 %\"\n",
      "[1] \"23.1 %\"\n",
      "[1] \"23.4 %\"\n",
      "[1] \"23.7 %\"\n",
      "[1] \"24 %\"\n",
      "[1] \"24.3 %\"\n",
      "[1] \"24.6 %\"\n",
      "[1] \"24.9 %\"\n",
      "[1] \"25.1 %\"\n",
      "[1] \"25.4 %\"\n",
      "[1] \"25.7 %\"\n",
      "[1] \"26 %\"\n",
      "[1] \"26.3 %\"\n",
      "[1] \"26.6 %\"\n",
      "[1] \"26.9 %\"\n",
      "[1] \"27.1 %\"\n",
      "[1] \"27.4 %\"\n",
      "[1] \"27.7 %\"\n",
      "[1] \"28 %\"\n",
      "[1] \"28.3 %\"\n",
      "[1] \"28.6 %\"\n",
      "[1] \"28.9 %\"\n",
      "[1] \"29.1 %\"\n",
      "[1] \"29.4 %\"\n",
      "[1] \"29.7 %\"\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "library(exomePeak)\n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')\n",
    "setwd(\"bam/\")\n",
    "print (txdb)\n",
    "res <- exomepeak(\n",
    "    TXDB=txdb,\n",
    "    IP_BAM=c(\"U1.m6A.bam\",\"U2.m6A.bam\"),\n",
    "    INPUT_BAM=c(\"U1.input.bam\",\"U2.input.bam\"),\n",
    "    OUTPUT_DIR='../exomepeak',\n",
    "    EXPERIMENT_NAME='ctrl'\n",
    ")\n",
    "saveRDS(res, paste('../exomepeak/ctrl.rds', sep='/'))\n",
    "setwd(\"../\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# chr\tchromStart\tchromEnd\tname\tscore\tstrand\tthickStart\tthickEnd\titemRgb\tblockCount\tblockSizes\tblockStarts\r\n",
      "chr1\t853842\t854022\tENSG00000228794.8\t7.4e-09\t+\t853842\t854022\t0\t1\t180,\t0\r\n",
      "chr1\t1055278\t1055369\tENSG00000188157.14\t0.00032\t+\t1055278\t1055369\t0\t1\t91,\t0\r\n",
      "chr1\t1055518\t1055669\tENSG00000188157.14\t8.9e-06\t+\t1055518\t1055669\t0\t1\t151,\t0\r\n",
      "chr1\t1232951\t1233310\tENSG00000176022.4\t3.2e-33\t+\t1232951\t1233310\t0\t1\t359,\t0\r\n",
      "chr1\t1233458\t1234235\tENSG00000176022.4\t9.99999999999999e-288\t+\t1233458\t1234235\t0\t1\t777,\t0\r\n",
      "chr1\t1234622\t1235011\tENSG00000176022.4\t0\t+\t1234622\t1235011\t0\t1\t389,\t0\r\n",
      "chr1\t1327767\t1328065\tENSG00000224051.6\t5e-11\t+\t1327767\t1328065\t0\t1\t298,\t0\r\n",
      "chr1\t1328361\t1328511\tENSG00000224051.6\t4.8e-08\t+\t1328361\t1328511\t0\t1\t150,\t0\r\n",
      "chr1\t1328569\t1328838\tENSG00000224051.6\t1.6e-17\t+\t1328569\t1328838\t0\t1\t269,\t0\r\n",
      "chr1\t1400709\t1401571\tENSG00000224870.6\t0\t+\t1400709\t1401571\t0\t1\t862,\t0\r\n",
      "chr1\t1672764\t1672944\tENSG00000269737.2\t7.8e-08\t+\t1672764\t1672944\t0\t1\t180,\t0\r\n",
      "chr1\t2228844\t2228935\tENSG00000157933.9\t1.3e-05\t+\t2228844\t2228935\t0\t1\t91,\t0\r\n",
      "chr1\t2229443\t2232546\tENSG00000157933.9\t1e-220\t+\t2229443\t2232546\t0\t2\t292,68,\t0,3035\r\n",
      "chr1\t2303047\t2303378\tENSG00000157933.9\t2.2e-08\t+\t2303047\t2303378\t0\t2\t56,94,\t0,237\r\n",
      "chr1\t2306944\t2307364\tENSG00000157933.9\t0\t+\t2306944\t2307364\t0\t1\t420,\t0\r\n",
      "chr1\t2307483\t2308382\tENSG00000157933.9\t0\t+\t2307483\t2308382\t0\t1\t899,\t0\r\n",
      "chr1\t2403107\t2403497\tENSG00000157916.19\t9.999987484956e-319\t+\t2403107\t2403497\t0\t1\t390,\t0\r\n",
      "chr1\t2403586\t2403767\tENSG00000157916.19\t4e-19\t+\t2403586\t2403767\t0\t1\t181,\t0\r\n",
      "chr1\t2563530\t2565053\tENSG00000157873.17\t4e-85\t+\t2563530\t2565053\t0\t2\t299,1,\t0,1522\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 20 exomepeak/ctrl/peak.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot _Guitar_ \n",
    "Now, let's draw metagene plots for control samples:\n",
    "(using seprate env: `Guitar`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/rumi/shams/abe/anaconda3/envs/guitar/lib/python3.8/site-packages/rpy2/robjects/pandas2ri.py:14: FutureWarning: pandas.core.index is deprecated and will be removed in a future version.  The public classes are available in the top-level namespace.\n",
      "  from pandas.core.index import Index as PandasIndex\n"
     ]
    }
   ],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Guitar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"20200409182002\"\n",
      "[1] \"There are 203835 transcripts of 58381 genes in the genome.\"\n",
      "[1] \"total 203835 transcripts extracted ...\"\n",
      "[1] \"total 94682 transcripts left after ambiguity filter ...\"\n",
      "[1] \"total 94682 transcripts left after check chromosome validity ...\"\n",
      "[1] \"total 11625 mRNAs left after component length filter ...\"\n",
      "[1] \"total 56222 ncRNAs left after ncRNA length filter ...\"\n",
      "[1] \"generate components for all tx\"\n",
      "[1] \"generate components for mRNA\"\n",
      "[1] \"generate components for lncRNA\"\n",
      "[1] \"generate chiped transcriptome\"\n",
      "[1] \"generate coverage checking ranges for tx\"\n",
      "[1] \"generate coverage checking ranges for mrna\"\n",
      "[1] \"generate coverage checking ranges for ncrna\"\n",
      "[1] \"20200409182409\"\n",
      "[1] \"import BED file exomepeak/ctrl/peak.bed\"\n",
      "[1] \"sample 10 points for Group1\"\n",
      "[1] \"start figure plotting for tx ...\"\n",
      "[1] \"start figure plotting for mrna ...\"\n",
      "[1] \"start figure plotting for ncrna ...\"\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "GuitarPlot(txTxdb=txdb,stBedFiles=\"exomepeak/ctrl/peak.bed\",miscOutFilePrefix='plot')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "!magick -density 300 plot_mrna_test.pdf -flatten -quality 90 plot_mrna_test.png\n",
    "!magick -density 300 plot_ncrna_test.pdf -flatten -quality 90 plot_ncrna_test.png\n",
    "!magick -density 300 plot_tx_test.pdf -flatten -quality 90 plot_tx_test.png"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'plot_mrna_test.pdf' -> 'plots/plot_mrna_test.pdf'\r\n",
      "renamed 'plot_mrna_test.png' -> 'plots/plot_mrna_test.png'\r\n",
      "renamed 'plot_ncrna_test.pdf' -> 'plots/plot_ncrna_test.pdf'\r\n",
      "renamed 'plot_ncrna_test.png' -> 'plots/plot_ncrna_test.png'\r\n",
      "renamed 'plot_tx_test.pdf' -> 'plots/plot_tx_test.pdf'\r\n",
      "renamed 'plot_tx_test.png' -> 'plots/plot_tx_test.png'\r\n"
     ]
    }
   ],
   "source": [
    "mv -v *p[nd][gf] plots/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/plot_mrna_test.png\" style=\"width:600px\">\n",
    "  <tr>\n",
    "<table>\n",
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/plot_ncrna_test.png\" style=\"width:300px\">\n",
    "    <td><img src=\"plots/plot_tx_test.png\" style=\"width:300px\">\n",
    "  <tr>\n",
    "<table>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Motif analysis "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "using `cgat` env to use [CGAT - Computational Genomics Analysis Tools](https://www.cgat.org/downloads/public/cgat/documentation/) and use "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cd ~/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated/exomepeak/ctrl/\n",
    "# sort | merge by name \n",
    "cat peak.bed | sort -k1,1 -k2,2n peak.bed | cgat bed2bed --method=merge --merge-by-name |  awk '! /#/' > peak.edit.bed\n",
    "# get fasta for the cordinate sequnces (-split removes introns)\n",
    "bedtools getfasta -name -s -fi /rumi/shams/genomes/hg38/hg38.fa -bed peak.edit.bed -split -fo peak.fa\n",
    "# prepare fasta file for FIRE\n",
    "perl /flash/hani/bin/Tools/prep_seqs_for_teiser_run.pl peak.fa peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cd ~/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated/exomepeak/ctrl/\n",
    "# run FIRE --doskipdiscovery=0\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --oribiasonly=0\n",
    "mv peaks_teiser.txt_FIRE/ skipdiscovery.0_FIRE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To run `FIRE` non-discovery mode we need to list in a file like `motifs_of_interest.txt` "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tue Apr  7 22:16:09 UTC 2020\n",
      "RNA, Processing peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt\n",
      "RNA, Step 2: skip seed optimization, use -motiffile_rna option instead.\n",
      "RNA, Step 3: evaluation of motif significances.\n",
      "Processing motif [AG]GAC.\n",
      "MI=0.00287461\n",
      "Shuffle rank=0\n",
      "Z-score=57.222\n",
      "Sigificance category= OK-NO-SEED\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [AGT][AG]AC[ACT].\n",
      "MI=0.00103316\n",
      "Shuffle rank=0\n",
      "Z-score=19.393\n",
      "Sigificance category= OK-NO-SEED\n",
      "Robustness (jn_f=3)=9/10\n",
      "\n",
      "RNA, Step 3.5: creating binary expression profiles.\n",
      "RNA, Step 4: discovery of distance constraints.\n",
      "Read 2 motifs.\n",
      "Read 20328 (genes) x 2 (conditions) matrix.\n",
      "Read 20328 (genes) x 1 (conditions) matrix.\n",
      "Reading sequences and building motif profiles ...Done.\n",
      "Average sequence length is 426.\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AG]GAC, present in 17970 genes.\n",
      "       Average distance: MI=0.0037, rank=0/10000, Z=64.9084, mbins=2\n",
      "  Orientation bias {5'): MI=0.0029, rank=0/10000, Z=57.2122\n",
      "  Orientation bias (3'): MI=0.0000, rank=8526/10000, Z=-0.6713\n",
      "       Copy number (1+): 17970 genes, MI=0.0029, rank=0/10000, Z=57.8862\n",
      "       Copy number (2+): 14811 genes, MI=0.0028, rank=0/10000, Z=54.8047\n",
      "       Copy number (3+): 11616 genes, MI=0.0026, rank=0/10000, Z=50.7668\n",
      "       Copy number (4+): 9000 genes, MI=0.0018, rank=0/10000, Z=34.1610\n",
      "       Copy number (5+): 6873 genes, MI=0.0012, rank=0/10000, Z=22.7328\n",
      "       Copy number (6+): 5293 genes, MI=0.0008, rank=0/10000, Z=16.0874\n",
      "       Copy number (7+): 4027 genes, MI=0.0005, rank=5/10000, Z=8.8324\n",
      "       Copy number (8+): 3046 genes, MI=0.0002, rank=259/10000, Z=2.9094\n",
      "       Copy number (9+): 2324 genes, MI=0.0001, rank=664/10000, Z=1.7934\n",
      "       Copy number (10+): 1783 genes, MI=0.0000, rank=8073/10000, Z=-0.6626\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT][AG]AC[ACT], present in 19816 genes.\n",
      "       Average distance: MI=0.0011, rank=0/10000, Z=19.3071, mbins=2\n",
      "  Orientation bias {5'): MI=0.0010, rank=0/10000, Z=19.9326\n",
      "  Orientation bias (3'): MI=0.0006, rank=1/10000, Z=10.8323\n",
      "       Copy number (1+): 19816 genes, MI=0.0010, rank=0/10000, Z=19.8805\n",
      "       Copy number (2+): 18855 genes, MI=0.0011, rank=0/10000, Z=21.2587\n",
      "       Copy number (3+): 17474 genes, MI=0.0013, rank=0/10000, Z=24.6523\n",
      "       Copy number (4+): 15943 genes, MI=0.0008, rank=0/10000, Z=15.9567\n",
      "       Copy number (5+): 14339 genes, MI=0.0003, rank=31/10000, Z=5.6152\n",
      "       Copy number (6+): 12709 genes, MI=0.0003, rank=14/10000, Z=6.3201\n",
      "       Copy number (7+): 11271 genes, MI=0.0002, rank=244/10000, Z=2.8455\n",
      "       Copy number (8+): 9995 genes, MI=0.0001, rank=938/10000, Z=1.3412\n",
      "       Copy number (9+): 8847 genes, MI=0.0001, rank=1254/10000, Z=0.9624\n",
      "       Copy number (10+): 7783 genes, MI=0.0000, rank=2699/10000, Z=0.1805\n",
      "\n",
      "RNA, Step 5.5: conservation.\n",
      "Setting -rna to 1.\n",
      "[AG]GAC\t98.12\n",
      "[AGT][AG]AC[ACT]\t21.42\n",
      "RNA, Step 5: summarize information.\n",
      "Reading RNA motifs mi_dist output ...Done.\n",
      "Parsing RNA motif file ... Done.\n",
      "Reading mi_signif output file for RNA motifs ...!!!! 851816!!!! 619296Done.\n",
      "Got 2 motifs\n",
      "Writing good columns to peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.signifcolumns ...Done.\n",
      "Outputing summary ... Done.\n",
      "HI!HELLO!peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.binary created.\n",
      "RNA, Step 5.3: generate profiles for motifs in summary file.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [AG]GAC ... Done.\n",
      "Generating profile for [AGT][AG]AC[ACT] ... Done.\n",
      "RNA, Step 5.3 bis: Generate profiles for /flash/hani/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [AG]GAC ... Done.\n",
      "Generating profile for [AGT][AG]AC[ACT] ... Done.\n",
      "RNA, Determining target gene sets.\n",
      "RNA, Drawing position histograms.\n",
      "RNA, Step 6: cluster motifs.\n",
      "Using positive correlation.\n",
      "Doing all stats.\n",
      "Summary file has 0 DNA motifs and 2 RNA motifs.\n",
      "Read in RNA sequences and build motif profiles ...Done.\n",
      "Outputing MI matrix ... Done.\n",
      "[AG]GAC\t0\n",
      "[AG]GAC\t[AGT][AG]AC[ACT]\t0.7815\t0\t15470.3103\t0.8752\t1854\t0.7416\t1854\t0.7416\n",
      "[AGT][AG]AC[ACT]\t0\n",
      "Finish calculated rank, z-scores for remaining entries ... [AG]GAC\t[AG]GAC\t0.9934\t0\t19590.5947\t1.0000\t0\t16.2939\t0\t16.2939\n",
      "[AGT][AG]AC[ACT]\t[AGT][AG]AC[ACT]\t0.9997\t0\t19904.0798\t1.0000\t3\t6.7154\t3\t6.7154\n",
      "Done.\n",
      "RNA, Step 5.6: generate genome-wide GO enrichments for motifs in summary file.\n",
      "RNA, draw motif maps for co-localizing pairs\n",
      "Draw motif map for [AG]GAC .. Done.\n",
      "Draw motif map for [AGT][AG]AC[ACT] .. Done.\n",
      "Draw motif map for [AGT][AG]AC[ACT],[AGT][AG]AC[ACT] ..  (peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.mimatrix_OUT/[AGT][AG]AC[ACT]_[AGT][AG]AC[ACT].eps) Done.\n",
      "Draw motif map for [AG]GAC,[AG]GAC ..  (peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.mimatrix_OUT/[AG]GAC_[AG]GAC.eps) Done.\n",
      "RNA, Step 6.8: identifying motifs that match micrornas.\n",
      "RNA, Step 7: draw matrix figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 400, xbase = 35, ybase = 250\n",
      "Processing [AG]GAC ... Outputing motif 0.eps ... Done.\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 1.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [AG]GAC ... Outputing motif 0.png ... \n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 1.png ... \n",
      "RNA, Step 7b: draw density figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 400, xbase = 35, ybase = 250\n",
      "Processing [AG]GAC ... Outputing motif 0.eps ... Done.\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 1.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [AG]GAC ... Outputing motif 0.png ... \n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 1.png ... \n",
      "Parsing distance report file ...  Done.\n",
      "Processing matrix 2 x 2.\n",
      "Processing matrix 2 x 2.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 230, ysize = 310, xbase = 35, ybase = 115\n",
      "[AG]GAC\t0\n",
      "[AGT][AG]AC[ACT]\t1\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf ... Done.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 250, ysize = 305, xbase = 75, ybase = 130\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.html[AG]GAC\t0\n",
      "[AGT][AG]AC[ACT]\t1\n",
      "RNA, Step 10: motif report.\n",
      "Tue Apr  7 22:21:16 UTC 2020\n",
      "renamed 'peaks_teiser.txt_FIRE/' -> 'skipdiscovery.1_FIRE'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Segmentation fault (core dumped)\n",
      "Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /flash/hani/bin/FIRE-1.1/SCRIPTS/draw_position_histogram.pl line 134.\n",
      "Can't locate Hypergeom.pm in @INC (you may need to install the Hypergeom module) (@INC contains: /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.12.4/darwin-thread-multi-2level/nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.10.0/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.18.2/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.22.0/x86_64-linux/Hypergeom.pm /flash/hani/bin/FIRE-1.1/SCRIPTS /rumi/shams/abe/perl5/lib/perl5/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/perl5/lib/perl5/5.26.2 /rumi/shams/abe/perl5/lib/perl5/x86_64-linux-thread-multi /rumi/shams/abe/perl5/lib/perl5 /rumi/shams/abe/anaconda3/envs/deseq/lib/site_perl/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/anaconda3/envs/deseq/lib/site_perl/5.26.2 /rumi/shams/abe/anaconda3/envs/deseq/lib/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/anaconda3/envs/deseq/lib/5.26.2 .) at /flash/hani/bin/FIRE-1.1/SCRIPTS/GroupEnrichment.pm line 24.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/over.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/under.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/C0.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/C1.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/motif.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/location.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/mi.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/z.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/robustness.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/pos.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/ori.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/cons.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/seed.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/name.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/parray.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/over.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/under.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/C0.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/C1.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/motif.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/location.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/mi.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/z.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/robustness.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/pos.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/ori.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/cons.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/seed.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/name.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/parray.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/3UTR_pos.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/3UTR_pos.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "cd ~/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated/exomepeak/T_vs_U/\n",
    "# run FIRE --doskipdiscovery=1\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --doskipdiscovery=1 --motiffile_rna=motifs_of_interest.txt --oribiasonly=0\n",
    "rm -r skipdiscovery.1_FIRE\n",
    "mv -v peaks_teiser.txt_FIRE/ skipdiscovery.1_FIRE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/rumi/shams/abe/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated\n"
     ]
    }
   ],
   "source": [
    "!cd /rumi/shams/abe/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated\n",
    "\n",
    "!magick -density 300 exomepeak/T_vs_U/skipdiscovery.0_FIRE/RNA/peaks_teiser.txt.summary.pdf -flatten -quality 90 exomepeak/T_vs_U/skipdiscovery.0_FIRE/RNA/peaks_teiser.txt.summary.png\n",
    "!magick -density 300 exomepeak/T_vs_U/skipdiscovery.1_FIRE/RNA/peaks_teiser.txt.summary.pdf -flatten -quality 90 exomepeak/T_vs_U/skipdiscovery.1_FIRE/RNA/peaks_teiser.txt.summary.png"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"exomepeak/T_vs_U/skipdiscovery.1_FIRE/RNA/peaks_teiser.txt.summary.png\" style=\"width:900px\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential peaks \n",
    "\n",
    "### Running a custom pipeline\n",
    "Here we're using https://github.com/gold-lab/bed12ToGTF to make gtf file from control exomepeak result.\n",
    "\n",
    "https://metacpan.org/pod/Bio::FeatureIO::gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat exomepeak/ctrl/peak.bed | sort -k4 | \\\n",
    "# rename features peak by peak\n",
    "awk -F \"\\t\" '{OFS=FS}NR>1{$4=$4\"_peak\"}; cnt[$4]++{$4=$4\"_\"cnt[$4]} 1' | \\\n",
    "# bed12 to gtf\n",
    "perl ~/Workflows/bed12ToGTF/bed12ToGTF.1.pl | \\\n",
    "awk '{gsub(/\\\"|\\;/,\"\")}1''{gsub(/transcript_id/,\"gene_id\")}1''{gsub(/ exon_number [[:digit:]]/,\"\")}1' > exomepeak/ctrl/peak.gtf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# chr\tblank\ttranscript\t1\tchromEnd\t.\tstrand\t.\tgene_id name\r\n",
      "chr1\tblank\ttranscript\t169853401\t169853641\t.\t-\t.\tgene_id ENSG00000000457.13_peak\r\n",
      "chr1\tblank\texon\t169853401\t169853641\t.\t-\t.\tgene_id ENSG00000000457.13_peak\r\n",
      "chr1\tblank\ttranscript\t169853390\t169853630\t.\t+\t.\tgene_id ENSG00000000460.16_peak\r\n",
      "chr1\tblank\texon\t169853390\t169853630\t.\t+\t.\tgene_id ENSG00000000460.16_peak\r\n",
      "chr6\tblank\ttranscript\t53498597\t53498777\t.\t-\t.\tgene_id ENSG00000001084.11_peak\r\n",
      "chr6\tblank\texon\t53498597\t53498777\t.\t-\t.\tgene_id ENSG00000001084.11_peak\r\n",
      "chr6\tblank\ttranscript\t41097505\t41098607\t.\t+\t.\tgene_id ENSG00000001167.14_peak\r\n",
      "chr6\tblank\texon\t41097505\t41098607\t.\t+\t.\tgene_id ENSG00000001167.14_peak\r\n",
      "chr6\tblank\ttranscript\t41098756\t41099143\t.\t+\t.\tgene_id ENSG00000001167.14_peak_2\r\n",
      "cat: write error: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "cat exomepeak/ctrl/peak.gtf | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "mk_dir('./peakcount')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o peakcount/T1.input.fc bam/T1.input.bam\n",
      "featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o peakcount/T1.m6A.fc bam/T1.m6A.bam\n",
      "featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o peakcount/T2.input.fc bam/T2.input.bam\n",
      "featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o peakcount/T2.m6A.fc bam/T2.m6A.bam\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/T1.input.bam                               ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/T1.input.fc                            ||\n",
      "||                 Summary : peakcount/T1.input.fc.summary                    ||\n",
      "||              Annotation : exomepeak/ctrl/peak.gtf (GTF)                    ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/ctrl/peak.gtf ...                           ||\n",
      "||    Features : 11111                                                        ||\n",
      "||    Meta-features : 9367                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/T1.input.bam...                                       ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 125909291                                                 ||\n",
      "||    Successfully assigned reads : 2393935 (1.9%)                            ||\n",
      "||    Running time : 4.01 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/T1.input.fc.s  ||\n",
      "|| ummary\"                                                                    ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/T1.m6A.bam                                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/T1.m6A.fc                              ||\n",
      "||                 Summary : peakcount/T1.m6A.fc.summary                      ||\n",
      "||              Annotation : exomepeak/ctrl/peak.gtf (GTF)                    ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/ctrl/peak.gtf ...                           ||\n",
      "||    Features : 11111                                                        ||\n",
      "||    Meta-features : 9367                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/T1.m6A.bam...                                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 46222685                                                  ||\n",
      "||    Successfully assigned reads : 29690222 (64.2%)                          ||\n",
      "||    Running time : 1.24 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/T1.m6A.fc.sum  ||\n",
      "|| mary\"                                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/T2.input.bam                               ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/T2.input.fc                            ||\n",
      "||                 Summary : peakcount/T2.input.fc.summary                    ||\n",
      "||              Annotation : exomepeak/ctrl/peak.gtf (GTF)                    ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/ctrl/peak.gtf ...                           ||\n",
      "||    Features : 11111                                                        ||\n",
      "||    Meta-features : 9367                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/T2.input.bam...                                       ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 117775402                                                 ||\n",
      "||    Successfully assigned reads : 2228238 (1.9%)                            ||\n",
      "||    Running time : 3.76 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/T2.input.fc.s  ||\n",
      "|| ummary\"                                                                    ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/T2.m6A.bam                                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/T2.m6A.fc                              ||\n",
      "||                 Summary : peakcount/T2.m6A.fc.summary                      ||\n",
      "||              Annotation : exomepeak/ctrl/peak.gtf (GTF)                    ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/ctrl/peak.gtf ...                           ||\n",
      "||    Features : 11111                                                        ||\n",
      "||    Meta-features : 9367                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/T2.m6A.bam...                                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 48438862                                                  ||\n",
      "||    Successfully assigned reads : 30578524 (63.1%)                          ||\n",
      "||    Running time : 1.69 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/T2.m6A.fc.sum  ||\n",
      "|| mary\"                                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# %%bash --err error\n",
    "for f in bam/T*.bam; do \n",
    "    out=${f/.bam/.fc}; \n",
    "    out=${out/bam/peakcount}; \n",
    "    echo featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o $out $f;\n",
    "    featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o $out $f;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Id transfer\n",
    "Read gene names from gtf file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import os\n",
    "\n",
    "def gtf_id_transfer(gtf):\n",
    "    # awk code to extract gene ID and Name \n",
    "    gtf_raw = os.popen(\"cat \" + gtf + \" | awk 'BEGIN{FS=\\\"\\t\\\"}{split($9,a,\\\";\\\"); if($3~\\\"gene\\\") print a[1]\\\"\\t\\\"a[3]}' | sed 's/gene_id \\\"//' | sed 's/gene_name \\\"//' | sed 's/\\\"//g' | sed 's/ //g'\").read().split('\\n')\n",
    "    # data frame \n",
    "    gtf_df = pd.DataFrame(\n",
    "        # col: Names \n",
    "        data=[i.split('\\t')[1] for i in gtf_raw[:len(gtf_raw)-1]],\n",
    "        # index: ids \n",
    "        index=[i.split('\\t')[0] for i in gtf_raw[:len(gtf_raw)-1]],\n",
    "        columns=['Name'], dtype=str)\n",
    "    return gtf_df\n",
    "\n",
    "gtf = '/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf'\n",
    "gtf_df = gtf_id_transfer(gtf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "load R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Differential methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(DESeq2)\n",
    "library(tidyverse)\n",
    "library(ggplot2)\n",
    "require(ggrepel)\n",
    "library(patchwork)\n",
    "\n",
    "pca = list(); vol = list(); hist = list()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "using `DESeq2` to find differential methylation peaks:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R -i gtf_df\n",
    "\n",
    "files.input <- list.files(path = 'peakcount',pattern=\".input.fc$\")\n",
    "files.m6A <- list.files(path = 'peakcount',pattern=\".m6A.fc$\")\n",
    "files = c(files.input,files.m6A)\n",
    "\n",
    "datalist <- lapply(files, function(x){read.table(file=paste('peakcount/', x, sep = \"\"),skip=1,sep='\\t',header=TRUE,colClasses=c(NA, \"NULL\", \"NULL\", \"NULL\", \"NULL\", \"NULL\", NA),check.names=F)})\n",
    "m <- Reduce(function(...) merge(..., by=1, all = TRUE), datalist)\n",
    "rownames(m) <- m[,1]\n",
    "m <- m[,-1]\n",
    "m[is.na(m)]<-0\n",
    "m <- round(m)\n",
    "colnames(m) <- substr(colnames(m), 5 , nchar(colnames(m)) - 4)\n",
    "\n",
    "# Design: m6A vs input \n",
    "ip = c(rep('input',2),rep('m6A',2))\n",
    "colData <- data.frame(cond = ip, row.names=colnames(m))\n",
    "# diff_peak \n",
    "dds <- DESeqDataSetFromMatrix(countData = m, colData = colData, design = ~ cond)\n",
    "dds$cond <- relevel(dds$cond, ref=\"input\")\n",
    "dds <- DESeq(dds)\n",
    "    \n",
    "# normalized counts \n",
    "ncu <- counts(dds, normalized=TRUE)\n",
    "res <- results(dds)\n",
    "res <- as.data.frame(res)\n",
    "# idtransfer\n",
    "peak_ids = strsplit(rownames(m), '_',)\n",
    "ids = unlist(lapply(peak_ids, '[[', 1))\n",
    "res$name = gtf_df[ids,]\n",
    "# save files \n",
    "write.table(res, 'deseq/peak_m6A_vs_input.txt', sep=\"\\t\", quote=FALSE, col.names=NA)\n",
    "write.table(ncu, file='deseq/peak_count_table_DE2norm.txt',sep=\"\\t\", quote=FALSE, col.names=colnames(m)) \n",
    "write.table(m, file='deseq/peak_count_table_raw.txt',sep=\"\\t\", quote=FALSE, col.names=colnames(m))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAC8VBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVHR0dISEhJSUlLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///8SjB8yAAAVu0lEQVR4nO3df3wU9Z3H8b1yB/ijP/glNhr8UVBOIZ7hh1UwLRSEVm2vKjnxqrUerdqzVxtELkgBJWiv5c56gr/Ow7NX+SWE44eogEKRUltBLQLiD5Sk5jfZTTbZJPv56+bzmZndmc3szneS2SQ7vF+POpvd+c7ks/t0JwEkDREKdKHeHgBlNwAHPAAHPD+Ai5anPFD4VJodqMfzADwzFBp43uw/dN5Rto1o4q8tD5jAvCNNtvWKe9RXoERegG86dHDT9X+9znGnI3CGEutbMp7J/XOhzHkBnsPbW4e3Umzu8M9/4z2itvvzBl76Cl+Jbw6FQhfSswWnn3tfTANeNnPgyO36JbpwycwvXvwqJY7R09cX/nzGmb82dxgHG3seue60sW+/dumZt7QkDzVOpa9AankGPhjaTXOvfOPDeedHaW7+lo+27BRHeVc9+dLHOy5g1C89+f7cz9fowMP20vIvxxLHGMn6wi+9Eg+bO8yD9T1D/+vday+b9Nqe4Ssocah5KryDPeQZuC30fGTAUe2DC7aG+2+RHQlg7pnJGs8sovhFy3XgnxG1fu7P5jHmyXTGO4lsO/hgfc8PifaGXib6aXFyhXEqAHupC8C/+VNIWvHHUFh2JIDfvfHi4V88T+P5lXbnlh/qwCu1j7+wyzzGPJnOqG0SO8yD9T2PE30QqiJ6ZFpyhXEqAHupC5foPX8INcgDb6YAx/LufuvE03kG8GwDmL/h+sIO8xgznVHbZ+5IHJzY82GonugXUxMrzFMB2Evev8k6uzXc/7fygPUSPfnfid4LaV9iF+WlXKJ1FfMYM14v+8wdiYMTe0zgxKEmsKxAann7ZdLb2i+TNhDNG/rCsT13f0z35m86tnmHOM7+7om6k6dtpgPD84xvss6stQInjjHi9fo+Y0fi4MQeEzhxqHkqWYHU8vYbHQPOm/2m9lHbwvz++d8/Sa0lZw0cowMfKOh/Ia05/5zJ/5anoZbNGDDyJbIBm8cYyXrZZ+4wD07sSQCbK8xTyQqkFn4vOuABOOD1MPCH+i95pvbsZz2Vwzs44AE44AE44AE44AE44AE44HUFeECU/5euRR3pHph2KOM5bbVPqEreeW/C+ROP2D6038zIy7vyj1SVl5c3JMNg1ll8HtJtQPtNj9YV4H3xTMD92tI88NJ3M5yz0wmf+mny40nP0DNfs31ov/mI6MmxsnvJrMyjm7P4PKTbgPabHk0deOWdtC/0e/rRE8Y7eNcVYwt20pbLxkw5SofO+YgW3iTL5oWuLmrec/Xl41+h6I1j/+46/QHec+M67dMtmH3VBlm3+CdENYMidHPh2Ovr+LWrH0zUNoBIP5ZODm2nSKss/eTMGMXOrLR8aL+R6Qp4G79ge8bnYM7i85BuA3Yet+dSBz56ES39ahmNel8Hrhr6BrXXVw45TE9MIHpu4rZRxh8kaO+F2jGVdDy/ZdN1RA3JN8ewT7RP9xuqOeczvvfx8DZ69DaiaqKl91teO+NYojH7aab+B7+7R2mbUfssH9pviG4akneAd796Xjzzk7C8g30c0m3A1HF7Mg+X6PxPpu78xvERxtfgDdP4sRena1+J+jcSfe8087+n1V6q8qFFRUUjj3404kdrIonXLhaKaZ8uTHRtudyfUk6Xv0a0YvKkgpmW1844VvtiuIEOfior5YUZuc/yof1G28Qfm867b17k8hwswD4O6TZgp3F7MA/At668lC5ZcZsNeP01OnDruLO3GMu0l2rDZP3DcPldFzaZr138b7R/D0LahW7GJrn/3HfevpBo/6g6Kp/K5wwPImoakDiWJif++63jyUub8aH9hnfE/kp7lzWckfjz5jRZgH0c0m3AzuP2XB6A/2fEHPqnEc8bwNXD3qC2usohh2jlRKJ/nvPOudrF7aUTRIOqqXqw9rz30qdhahn+MT8gXbZf+3S/oCNDqmRd05DvLSHaVkRULK8dDT9BqweYx2oXy0ra/b5+5JVP09NF+tmND2039e9QfEWednF+fLrbc+BZsjCky4ApNz2aB+AToTW0OlRh/jLp9Qmjx+ykzQWjv36UNoyN0tOT2ugS7XuTJRcVNP/uqtEjb6BtY8dcskR/gI9f/CC/dldcpK3hdXT757T3W3vxNcX36q/dmnGzHtK+f9GPpTevJPNrMP15/Lnj39OPMj603VRMGJ435U/aunFr3J4Dz5KFIV0GTLnp0XrwNzqqCmIePt0PNmZxlPTlxJBe6snfyfq/99U/Xftj2ZwkQzkxpIfwW5UBD8ABD8ABD8ABD8ABTx24OdIUi6gX9bDUy1oPI2Rr3Bb1tc1N6mtbPaxVGtcjcF11bbxauZqI+trmJvW1HpZWU42HGdSXRqLqa8O16ms76tTXKo0LYOsM6ksBzAFYArBMC2AOwByAJQBzAAZwcgb1pQDmACwBWKYFMAdgDsASgDkAAzg5g/pSAHMAlgAs0wKYAzAHYAnAHIABnJxBfSmAOQBLAJZpAcwBmAOwBGAOwABOzqC+FMAcgCUAy7QA5gDMAVgCMAdgACdnUF8KYA7AEoBl2twDLtZzXQtgmRbAHIA5AEsA5gAM4OQM6ksBzAFYyiXgSDhCYeUireprYx7Welga1kbOxnlbYuGwAey+1sMI8Sb1tSrj1nsEbqhriNcpV9+svjYa9bBWfWkd1WflvM0tdXUGsOvapgb183acVF+rMm4NLtHWGdSXBvUSDWAjAHMAlgAs0wKYAzAHYAnAHIABnJxBfSmAOQBLAJZpAcwBmAOwBGAOwABOzqC+FMAcgCUAy7QA5gDMAVgCMAdgACdnUF8KYA7AEoBl2lwCLrbmuhrAMi2AOQBzAJYAzAEYwMkZVBcCWA/AEoBlWgBzAOYALAGYAzCAkzOoLgSwHoAlAMu0AOYAzAFYAjAHYAAnZ1BdCGA9AEsAlmkBzAGYA7AEYM5f4GKHXA8CsEwLYA7AHIAlAHMABnByBtcVALYFYAnAMm3OArsyA1imBTAHYA7AEoA5ADsDxx9ftEjbWf7AggpzC+AgAb+5nA7+kmpK4keWGlsAVwcJePMGaryNdq8immNsAVwdJOB3F8d330Bb1xLdYWyJygoLjyl/tQ5m6YF7e7I0NcnW6ZusdYv/+66Ud3D05Mnamrp4jXK1Teprm5s9rFVfWkO16mujrivSA2c4KFKnPkJHvfpa93G10gIT7XmWakvih8uMrYRLdFAu0e2lpcuj2vfPpaUV5hbAQQJOE4ABbAnAEoBlWgBzAOYALAGYAzCAkzO4rgCwLQBLAJZpAcwBmAOwBGAOwABOzuC6AsC2ACwBWKYFMAdgDsASgDkAAzg5g+sKANsCsARgmRbAHIA5AEsA5gAM4OQMrisAbAvAEoBlWgBzAOYALAGYAzCAkzO4rgCwrSABp6cFsNq0AOYAzPVF4AzOAJZpAcwBmAOw67iJAGydIcM+ADsEYAnAMi2AOQBzAHYdNxGArTNk2AdghwAsAVimBTAHYA7AruMmArB1hgz7AOwQgCUAy7QA5gDMAdh13EQAts6QYR+AHQKwBGCZFsBcbwM31DXE65Srb1ZfG416WKu+tI7q/TmvIrDToU0N6iN0nPRnXLNMPxDcqUg4QmHlIq3qa2Me1npYGtZG9uW8isBOh7Z4GCHe5M+4ZvW4RFtnyLDvFLlEAxjAlgAsAVimBTAHYA7AruMmArB1hgz7AOwQgCUAy7QA5gDMAdh13EQAts6QYR+AHQKwBGCZFsAcgDkAu46bCMDWGTLsA7BDAJYALNMCmAMwB2DXcRMB2DpDhn0AdgjAEoBlWgBzAOYA7DpuIgBbZ3B6UFEWwG7TApgDMAfg9OOmBmDrDE4PAjh9AJYALNMCmAMwB+D046YGYOsMTg8COH0AlgAs0wKYAzAH4PTjpgZg6wxODwI4fQCWACzTApgDMAfg9OOmBmDrDE4PAjh9AJYALNMCmAMwB+D046YGYOsMTg8COH0AlgAs0wKYAzAH4PTjpgZg6wxODwI4fQCW+ibwe6Wla4nKH1hQYW4BHCTg1pJmbVtTEj+y1NgCOFDAby16eMkHtHsV0RxjC+BAAb96e7TiJ7RVu0rfYWyJnp09+3hbG7Wp1+FhqZe1Hkbo9rjegLs9bns3x00pbANuTmxo/zLtXdthfwcf2r69qjFMjeq1qC9tbc3KaRsp7GEGpwe9ATudIephhHikm+OmVGcDLuJNgXzY+C/xxruotiR+uMzY4hKtktMZ+soluqPtau09XT1Kp9xeOu8t7fvn0tIKcwvgHAde1C/Ur1+/QUs6f0FOBuAcBiaal8kWwLkPTBSLRqMAthUk4A35p59xxhkAthUk4MIPcInuVJCAZ7n5Aji3gZdubAVwakECDkkAthUkYPcADGBLpyCwk3PfAR4gAdhWkIDb2tqaVj8EYFtBApaKAWwraMAnxgPYVpCABw8ePOjsdQC2FSTg+vr6cCZeAOc4MFFjI4BTChLw4fH9+088CmBbQQL+2lMd7SunANhWkIDH8GY0gG0FCbjgMNGhAgDbChLwpsHXTB+yBcC2ggRMletfrMzkC+DcBt5RrxHuArCtIAFf2k4UGwNgW0ECHssbfBdt1iXaPg18xX6iPRMBbBQ84F3Dpk0b9jqAjYIHTFUbNlZn8gVwjgO7BmAAWwKwBGCZFsAcgDkAO43rHICtM9jvAtgtAEsAlmkBzAGYA7DTuM4B2DqD/S6A3QKwBGCZFsAcgDkAO43rHICtM9jvAtgtAEsAlmkBzAGYA7DTuM4B2DqD/S6A3QKwlEvAzc1R7R/VojH1tW1tHtaqL22maJfP2w1g23liHkaIt3R5XMdO4h1sncF+91R8BwMYwJYALAFYpgUwB2AOwE7jOgdg6wz2uwB2C8ASgGVaAHMA5gDsNK5zALbOYL8LYLcALAFYpgUwB2AOwE7jOgdg6wz2uwB2C8ASgGVaAHMA5gDsNK5zALbOYL8LYLcALAFYpgUwB2AOwE7jOgdg6wz2u90AtjEDWKYFMAdgDsBO4zoHYOsMxm23ZQHsMC2AOQBzALaNmzEAW2cwbgGsGIAlAMu0AOYAzAHYNm7GAGydwbgFsGIAlgAs0wKYAzAHYNu4GQOwdQbjFsCKAVgCsEwLYA7AHIBt42YMwNYZjFsAKwZgCcAyLYA5AHMAto2bMQBbZzBuAawYgCUAy7QA5gDMAdg2bsbSAdeWzP/ZUaLyBxZUmFsABwm4I04Hy6imJH5kqbEFcKCAtfatpt2riOYYW6Idjz5a4eknvjdn6Se+ezhtl37iu1/Axula++RPfP9k/q2HaOtaojuMLYCDBUx07J6UdzAu0R4zTtcnL9Exos/mat9qxQ+XGVsABwr4nfml9/N30aWlFeYWwEECThOAAWwJwBKAZdq+AOyXLIAdpgUwB2AOwPq4KosAbJ0BwAAGsCUASwCWaQHMAZgDsD6uyiIAW2cAMIABbAnAEoBlWgBzAOYArI+rsgjA1hkADGAAWwKwBGCZFsAcgDkA6+OqLAKwdQYAAxjAlgAsAVimBTAHYA7A+rgqiwBsnQHAAAawJQBLAJZpAwhsBGCZFsAcgDkA6+OqLAKwdQYAAxjAlgAM4OS0AOYAzGUBODu0ALZMC2AOwByAJQBzAAYwB2AAA7hTAAZwcloAcwDmACwBmAMwgDkA6zV7+onv0Sz9xHcPS1V/4ns2gWN99Ce+4x3sU7n0DgYwgC0BGMDJaQHMAZgDsARgDsAA5gAMYAB3CsAATk4LYA7AnJ/A2ZQFsHVaAHMA5gAsAZgDMIABDGAAOwZgACenDS6wntq4AOYALAGYAzCAAQxgADsGYAAnpwUwB2AOwBKAOQADGMDBBO4xWgDLtADmAMwBWAIw12eA1ZwBzAFYAjAH4FMPuHdkAay+FsASgDkAAxjAAAawYwBOV9pxAcwBWAIwB2AA90JpxwUwB2AJwByATwng3gZNLe24AOZyH9io87gA5gAs+Qdc/sCCij4NnPb16uPARpZxewW4piR+ZGm3gTs/Gfu0jsDOx2QG7g0iH1N/eX0D3r2KaI52++G+fdUNjdSgnDGy7Y71oYbOT8897bCWTp8i7SfKvdRf3hb3JQ11KsBb1xLdod0+O3v28bY2alOvw8NSL2s9jJBr47b7e96wl3dw9y7RbtebvvD/H6y+NBJVX9vnf4RDbUn8cBmAbQUKmMpLS/v2d9EZArACcDIAGwGYA7AEYJkWwByAOQBLAOYADODkDOpLAcwBWAKwTAtgDsAcgCUAcwAGcHIG9aUA5gAsAVimBTAHYA7AUnaAq4+rT1ATVl9bWam+1sO/N9UfV2XlvH+pUF/b6AH4+Gfqa5XG9QhMVPVVD4s9tPw/s3LaeOHJrJz3uUVZOS1NP5aNswLYc8EFbnw4GxMQvbwzK6eNPxTNynnf2JyV09J/VGXjrF6AUQ4G4ICnDNxectPLRO/fN3/+Zz6PEH9s4f0HfD6n3ofXZuHLWvIv8/hcVqZVB47XvqABN0Tpdb+/JTr2c6oq8fmceg/P8/8ls/xlHp/LxrSeLtEMrLV7pc8jnFwUP/agz+eU3nm+zP+XzPJXAfwtK9N2AThyT6XPI3Qsv+f7h30+p7SoKQsvmfmXeXwvK9OqAm9Z+KwBHCv9va+fXzvz3keo9se+ntQ471rKoXdwdqb1/A6OP7zd9xH2rqDoD3w/K9H/zl94S4nvv9dh+cs8vpadab0AL7vjzido942lpat8HqH9kYVzd/l8TqNsvCeSf5nH73r7HYxyMQAHPAAHPAAHPAAHvIACD0j5k8Lab+flffNTosFn5eXdnrKon/ar+wXnjxh5XyyIr0YAnxKXCly9Ptbx4+9owG93XqQB3zyjiiJL64P4agTwKXGa3ZbLxkw5SrT64nHL+8ljG8eZwMYubdH6iyct7kdHBhp/1h56pOCCjZp34djr67R7y/5h3MbEGfZcffn4V3rtCXW5wAJXDjlMT0ygirOO068EuGNamQb85REjNhm7ZNExWtaPVl9iHBZaSW/ma+93oqX3a/c20gf55hlqx1TS8fyWXnxOXSuwwC9OJ2rv37j+eqIGAb7z+nbjHWzs0hat/5b25bkfvfC3xmEh7aJ9WoxWTJ5UMFO710p0esw4Q/nQoqKikUd77yl1scACr79GFNd92wC++1salw5s7OJFf0/UqF2iB5iXaO2fM6L7R9VR+VTznnGGDZN77cl0q8ACVw45RCsnUsWw4/RLDfiua9lXBzZ28aL8enpe2ztrxl8o8mC9QbqtiKg4AWycoXrwVqK9vfqkulRggWlzweiva1fU3466smwQHQhpvz5KfJNl7NIWrZv0jws14Nb5I/K+UhozSNuLrym+NwFsnIF+d9XokTf07rPqSgEFThYhWjWzl8/QmwUe+F8LL5vavT+G6/4ZerPAA5/qATjgATjgATjgATjg/T/Yaa2R6zve6QAAAABJRU5ErkJggg==\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R \n",
    "d <- res$log2FoldChange[!is.na(res$log2FoldChange)]\n",
    "w <- wilcox.test(res$log2FoldChange,mu = 0, alternative = \"less\")\n",
    "\n",
    "t <- t.test(res$log2FoldChange, mu=0)\n",
    "\n",
    "h = ggplot(res, aes(x=log2FoldChange)) +\n",
    "        ggtitle(\n",
    "            'Decitabine_treatment',\n",
    "            sprintf(\"wilcox.test (p.value): %.5f \\t t.test (p.value):%.5f\", w$p.value, t$p.value)\n",
    "        ) +\n",
    "        ylim(c(0,1100)) +\n",
    "        geom_histogram(binwidth=0.3)\n",
    "\n",
    "ggsave(\"plots/peak_Histograms.png\", plot = h, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/peak_Histograms.pdf\", plot = h, device = 'pdf', dpi = 300)\n",
    "h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAACwVBMVEUAAAABAQECAgIDAwMEBAQGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSksLCwtLS0uLi4vLy8wMDAyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVHR0dISEhJSUlLS0tNTU1OTk5PT09QUFBSUlJUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1fX19gYGBhYWFiYmJjY2NkZGRlZWVnZ2doaGhpaWlqampsbGxvb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyOjo6Pj4+QkJCRkZGSkpKTk5OVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///8kM4q0AAAQo0lEQVR4nO3d/Z9cVX3A8SlpEgQqEKCx0RTEoBJSthSLIa2UCgWsQDRqH0KEWGEJSRcqpSSoGClINLaCLQpijYsmGDAFNlAfaiGI4UEaSxLC7mZ3M/s8u/P9K3rnzJzZObN3z/eezUxm7+zn88Pdyb3nnj2z78zNJjfz2oxQU5dp9AKovgHc5AHc5NUOeMXmqh0tW6c4QMewYODLM5n5Z6762eQDmx4XufDeih0WuHBgipzxCY8kH0EyHeBrXnrhsat++7uxB2OBPZXHD3tn0j8XTVU48HWF7acWjsjozQtP+rNfieTWL5p/7pOFK/HHM5nM2fKNZW9717rRCPiuy+cv2Vm8RLfccfnJ7/uxlM8pVhzf8rnLTrzXHiidXDryhSuPP2/P0+ee+InhiVNLUxVHkNY0gV/IdMjNf/xfr99y1pDcvHj7/27/T+NoXlVf+9G+Xe8uoJ7ytVdvPqmrCHzGc7L590bL55Qy41tOeTKftQfsycUjp//ri1f8wQef3r3wfimfaqfiFZyoaQLnMv/eP++V6MG7d2TnbjcHysCF/mV5xHOtSP69m4vAN4mMHPdLe46drMh4vYhzoHBy8cgakecyT4h8duXEiNJUACdr2sAP/SJjuv+/M1lzoAz84tXvW3jymRHPl6JffGJNEXhL9PjtT9lz7GRFxmhTPmBPLh75isivM2+JfOHSiRGlqQBO1rQv0bt/luk1O35eBTy66Ib/2f/1RSXgVSXgrdHjt++y59iKjNExe6B8cvnI65kekS9eUh5hpwI4WdP9JusdI9m53zY7Ki/Ry78s8qtM9Efs7YuqLtFbpaBiz7EVxptj9kD55PIRC1w+1QKbEaQ1nb8m7Yn+mrRN5JbTH35t9w375MbFj732w13GcdVf7j/cd/wP5fmFi0rfZJ3YXQlcPqdUYXzxWOlA+eTyEQtcPtVOZUaQ1nT+oWPemat+Hj3K/cPiuYv/uk9GWn93/tIi8PPL5p4t3znrncvvLgBvumzekh+JA2zPKWXGm2P2gD25fKQMbEfYqcwI0uLfops8gJu8hgC/XvwrzyWN+NyzLV7BTV7NgLkpODMDuMkDuMkLBm5p+9PTluyULy2NHmeP31W+OVgA/v1thX+l3lN1U5AaWTjw7zwrD5zU9X+/9ZLItxaOl+/vVQJX3RSkBhYO/LfR5vzNctHtIlfdUNxXuL9XAVx9U5AaWDjwPdHmU2vknnMlO/+ZiZuDFcDVNwWpgYUD3x1tPrZGDhz3y4cW5Sfu7xWAz4qAhzN7qm8KUgMLB/6oSP49kebFn7vyMxX39wrAf/R1kT2ZPdU3BamBhQOfvOXVm07sFrnvPfOfFSnf3ysA33TxQO9fRN9kVd0UpAYWDly6CSiH5rwrH32w9/cKwIevPOWc70XAVTcFqYGFA2+twyqobgHc5AHc5HG7sMkDuMkDuMkDuMkDuMkLBe7riWksbqenwcDxo0fqO/9wf9j4bOxXYeoGAxfUF7ie/uHY3dMCPtw5uS7pitnraTBseOdYT9j4gcD5R46EjT8SuJ7BwCfcE7qekdjdANsABtgJYCWA/QGsBLAJYBvAADsBrASwP4CVADYBbAMYYCeAlQD2B7ASwCaAbQAD7ASwEsD+AFYC2ASwDWA/8MqVK8MWBLA/gJUANgFsAxhgJ4CVAPYHsBLAJoBtAAPsBLASwP4AVgLYBLANYICdAFYC2B/ASgCbALYBDLATwEoA+wNYCWATwDaAAXYCWAlgfwArAWwC2AYwwE4AKwHsD2AlgE0A2wAG2AlgJYD9AawEsAlgG8AAOwGsBLA/gJUANgFsAxhgJ4CVAPYHsBLAJoBtAAPs1MzAgzFFwHG7py4XNnwwP1zf+cdGwsaPhK4ncEHDoesZi9vbxyvYxisYYCeAlQD2B7ASwCaAbQAD7ASwEsD+AFYC2ASwDWCAnQBWAtgfwEoAmwC2AQywE8BKAPsDWAlgE8A2gAF2AlgJYH8AKwFsAtgGMMBOACsB7A9gJYBNANsABtgJYCWA/QGsBLAJYBvAADsBrASwP4CVADYBbAMYYCeAlQD2B7ASwCaAbQAD7ASwEsD+AFYC2ASwDWCAnQBWAtgfwEoAmwC2AQywE8BKAPsDWAlgE8A2gAF2AlgJYH8AKwFsAtgGMMBOACsB7A9gJYBNANsABtgJYCWA/QGsBLAJYBvAADulHPjVdRs2HBJpv+3WA+4W4FIpB+4dkmfuk67W/MsbnS3AtpQDR3VskY4HRa5ztgDbUg/cv/ag7HhUZLWzFdm2bt0bw5OLgGP2ehoLGz6cH6nv/OO5sPG50bDxY4ELGg1dz3jc3iNTAo+2/URiX8HPPfDAm/2Ti4Bj9noaDRvenx+s7/xjw2Hjh0PXE7igwdD1jMXt7Z0KOP/5ndG2uzW/d5Oz5RJtS/kluuPqtrboJdve1nbA3QJcKuXA3gDuBFgNYH8AKwFsAtgGMMBOACsB7A9gJYBNANsABtgJYCWA/QGsBLAJYBvAADsBrASwP4CVADYBbAMYYCeAlQD2B7ASwCaAbQAD7ASwEsD+AFYC2ASwDWCAnQBWAtgfwEoAmwC2AQywE8BKAPsDWAlgE8A2gAF2AlgJYH8AKwFsAtgGMMBOACsB7A9gJYBNANsABtgJYCWA/QGsBLAJYBvAADsBrASwP4CVADYBbAMYYCeAlQD2B7ASwCaAbQAD7ASwEsD+AFYC2ASwDWCAnQBWAtgfwEoAmwC2zQbgwfIGYLU0Aq8obJYBnKj0AY/nLs7lcp3n6MA9XZOLgGP2ehoMG9411lvf+UeyYeOzcV8FT4OBC+oJXc9I7O4K4NvnZObMmXPqHTrwQMyPmuYngCvNhJ8AfotuyyXalr5LdKHRoaEhgBOVRuBti992wgknAJyoNAK3/JpLdOLSCHxtMl+AC6UReOP3RwBOWhqBMyaAE5VG4KQB3AmwGsD+6gE8zwRwotIInMvlBh65E+BEpRHYtBLgRKUVeP8FACcqjcALFiw49R3fBThRaQTu6enJ6rwAm9IILHLkCMAJSyPw3gvmzr3wFYATlUbgP9k6PrblQwAnKo3ASwub9wOcqDQCL9sr8hL/qzJZaQR+bMGH//y07QAnKo3AcvA/vndQ9wW4UBqBd/VEeE8BnKg0Ap87JjK6FOBEpRH4vMKG76KTlUbgD/xUZPeFACcqjcBPnXHppWc8A3Ci0ggsb237fqfuC3ChVAInDOBOgNUA9gewEsAmgG0AA+wEsBLA/gBWAtgEsA1ggJ0AVgLYH8BKAJsAtgEMsBPASgD7A1gJYBPANoABdgJYCWB/ACsBbALYBjDATgArAewPYCWATQDbAAbYCWAlgP0BrASwCWAbwAA7AawEsD+AlQA2AWwDGGAngJUA9ldn4LHWa56IPrTfdusBdwtwqZQD57sfjoC7WvMvb3S2ANtSDixSAO54UOQ6ZwuwrSmAdzwqstrZimxqaXktZngEHDcLNbYBsw16BQ/19XXH/Kz4CDjsZ84PhQ3vGusNGz8YOP9INmx8NnQ9gQvqDV3PSOxuP3B3a37vJmdr4hLdmf5L9F2rr/9q9J1zW9sBdwtwqbQD+wK4E2A1gP0BrASwCWAbwAA7AawEsD+AlQA2AWwDGGAngJUA9gewEsAmgG0AA+wEsBLA/gBWAtgEsA1ggJ0AVgLYH8BKAJsAtgEMsBPASgD7A1gJYBPANoABdgJYCWB/ACsBbALYBjDATgArAewPYCWATQDbAAbYCWAlgP0BrASwCWAbwAA7AawEsD+AlQA2AWwDGGAngJUA9gewEsAmgG0AA+wEsBLA/gBWAtgEsA1ggJ0AVgLYH8BKAJsAtgEMsBPASgD7A1gJYBPANoABdpqFwCuDkAH2B7ASwCaAbQAD7ASwEsD+Ggncn51cCTjmyBSNJB9qGh8IGx86f24obPxQ6HoCFzQQup5c3N6eaQH3Hp5cCTjmyBQNJR9qGu8LGx86/2h/2Pj+0PUELqgvdD2jcXs9PwHcE5fozua+RAPcCbAawP4AVgLYBLANYICdAFYC2B/ASgCbALYBDLATwEoA+wNYCWATwDaAAXYCWAlgfwArAWwC2AYwwE4AKwHsD2AlgE0A2wAG2AlgJYD9AawEsAlgG8AAOwGsBLC/GQqcHBlgfwArAWwC2AYwwE4AKwHsD2AlgE0A2wAG2AlgJYD9zWDgZMgA+wNYCWATwDaAAXYCWAlgfwArAWyqJ7AfGmB/MxxYfyUD7A9gJYBNANsABtgJYICdmhs4xhpgfykBLskC3MzA8VdsgP0BrASwqUHAk73VAPY3c4DjX9BqAPubGcCeF7T2BFxg/QyATY0F9mArvwcArm7GA1dhV9tXiFbbl/ZVLQhg08wETu5e8anjVtOZ7PJf6NgC66uajcD+PCup3F3x2HkCLvCkw5O/oA0Grn7aswN4+mdZ4CmGVByyX1DfZJMJp56oeoidDOCaNO3fFnHA0/v0HuCKFVb/5qmeBODmrUbA7bfdegDgmdzRAXe15l/eCPBM7uiAOx4UuS76+NLOnW8dmVyjnxxFf2pP7nBy4B2PiqyOPn77059+YzQmidvpaTxwfD4XNn4scP7xwBPGAtczHviEc6HrycftzYa/gqe4RHdJV+wf8lPG7UJ/x/wfOrpb83s3AVyu6YClva3N8100wEozH3gigDsBVgPYH8BKAJsAtgEMsBPASgD7A1gJYBPANoABdgJYCWB/ACsBbALYBjDATgArAexvpgF37gtbT2d/4Pg3DtV3/v1vho3vi/0qTN3Bg2HjD/eFjX9zf+zuaQHHNdwyXINZPF3xYn3n/7sf1Hf+e++p7/zb1059DGABWCt3Z64Gs3i670B953/k+frO/+Mn6zv/C49MfawWwDSDA7jJOxrg8htaig8m3t9So+yEr67bsOGQDF2zfv0v6jJ/aeaar788497169e21v4JjLVe80TlJ4p9AkcBXH5DS/FBxftbalN5wt4heeY+GVpfy8kr5y/OXPP1OzM+uq32TyDf/fATFZ8o/gkcBXD5v8MXH1T87/jaVDlhxxYZ+ti6zdm6zF+cuebrd2Zc2137JyBSAvYBHAWwfUNL6UH5l7WqYsL+tQcln5X2r9Rl/uLMNV9/5Yy/+Xup/RMoA/sAUvEKHm37ifnYe2N95jcz1/UV/G+P209T0+r7Ci6+oSXfU3pQ8f6W2lSeP//5ndEvh/LScXdd5i/OXPP1T3wCkeuzdXgCRWAF4Ki+iy68oaV/jX1ny8T7W2qUnb/j6ra2B2XPZze0HarL/KWZa77+iS/Q3n+KflX7J3DX6uu/qgHw9+AmD+AmD+AmD+AmD+AmD+BS84YavYL6BHApgJunzK2rLtom//gZka5T++XjLedddbgA3LNAJDdPZPfF51/wZKPXWLNmJfBD0vXOQ/sW5uSf/0qkU2Tj+grg7qUH5TeL6/yfkI5dsxI4K3JFu3yoXc5/WuT+5R9cdnkFcPvpK1asWPJKoxdZq2YlcHRFvuwx+eZH9pwt8tNzDkv7JQXg7KkiA/Nk2/JGr6+mzUrgL8rLp70lA6d98g6Rx1eIrDTAsnC/PDJPOhfsEHmu0WusWbMT+APv3RZ9/Jvj9omMrfzwyhuLwN/5w2vvjL7Jevai9y/5aKPXWLNmJXCjF3Asm1VPttSses6z6snOxgBu8gBu8gBu8gBu8v4foxRFQSp3PoIAAAAASUVORK5CYII=\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R \n",
    "ggplot(res, aes(x=pvalue)) +\n",
    "        ggtitle('Decitabine_treatment\\npvalue') +\n",
    "#         ylim(c(0,1100)) +\n",
    "        geom_histogram(binwidth=0.01)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### volcanos "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAADAFBMVEUAAAABAQECAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OUlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////isF19AAAgAElEQVR4nO29B5wkZZ3/37KySxIEBFQQMKAgIuCaEJDzOP+nx3mmUwRRUc6/6TxPFJGMIGAgHHggCCcCkjYAS9qd2JPzTE/oNNM5x6rqyt1V3fX8nqe6uqdT9fTM7DBbM89nX9tdXU/Vt57u9zzP832yCWBtaJnWOwJYaysMeIMLA97g2h+AL7qv5sT2R3UCsN5wLQPwJSbTIadeMVEfcFcbAJ/4U8WJEmAUoKOq61sMaf0KrLKWA/hSx+zLX3rzroaBDQE3Ufn6bFNLSz8Lq7mWA/gH6PXKt+eA9Ku3v+WfnADI1514yIe6UE78TZPJ9F7w+NmHvetaCQL+3SWHnNZRzKK3337JW0/vBuV7iipev/3Wzx/xp1KAdrMW8ocvHvrhub4PHfGt7OKtmqniFVitadmAZ00D4FefGvH9+t0i+NXJr/lf61E5qqnqL+0B83sQ1KP/4v7VW9JFwMcPg/veKZXv0aRev/3oLoUtBZRuLoYc91fbv51zQd/g2/8MyreWTOEUvAwtG7Bs+ju3zQUP3rOP3fqaGlAGjPR/F0I83wBA+cB9RcC/BCB3kL10T8lYEeOPAagKQDcXQ34IwLCpE4CrL1u8QjOFAS9HKwD8tMWk6s9TJlYNKAO2ff30t7/1VIjnXvjhWz8sAn4YHh/VW7qnZKyIEb6UA0o3F0MeAsBrSgLwh88uXqGZwoCXoxVk0YMTpox6YrIGsHTif05HHjtRA3yFBhg5XEeZS/eUVMQIw0oB5ZvLIT4TBcAfLy5fUTKFAS9Hy3ey3pFjtz6rnqjMoi/8HwCcJljE/ubEmiy6SKV0T0noejWsFFC+uRxSAly+tQRYvQKrNS2vmjQHq0kvAvDr457zDP5nAPzi5Jc9r5pVjlf8e4SkD30VzLz9RM3JOoKoBFy+RxO6vhimBZRvLoeUAJdvLZlSr8BqTctr6Nh26hWT8Ei+5eStJ3+PBrlrTjjkrCLgmbO3vhfsePdJF96NAN/1+W2ntYMqwKV7NKnXq2GlgNLN5ZAy4NIVJVPqFVitCbdFb3BhwBtcbzBgX7HKc/Eb+9TNLJyCN7gw4A0uDHiDCwPe4MKAN7haB8zRdcrUn1qm1soCBQOo9Y3CG2qhoQFleYB/+bc6PfZ4/bnl6cHVWnj80UZnr/7h7Xf+5o6//e0vjz788KN/af6MRx5bZRT+9pfVGnjskdVaeLT+O/51O7s8wA/UnxILLd+to5SySgMFodHZq37zyr7uWwHIswzN5JtboOsGDTVUalw3iGvJQBOJ9NLXNJdQT6KwkQFf/oDEObfCAyWfX+oJLQL279ENwoD1tUaAv/IoAISpYVCdMGBNRgL8L08BwG8Nt2QBA9ZkJMCf2QlA7viZliy0CJiY1g3CgPW1RoDPexW+nNPdkoUWATcRBqyvNQJ8bid8+cedLVnAgDUZCfDpg/Dlaw+3ZAED1vTGAr5qO3oNmV6Bh6arAepJ7keHpoNO/HYELJ7sFdDJLaf8hATg/G+ie/pNc+DUSVhDuuqOlqLQImCZ1w3CgPXVEuBDtnpLgM902F457R/A4kkE+EyH9amjv1oJ+O026EVf8ouWooC9aE3rB/ii875RAozO3nOQuHgSAUYnb9kqVwD+FkzP/KVXthQFDFjTOgIefNPYImDf+VulxZMlwL+H1BcBI/Hf+0JLUcCANb3BgIuDt4qAwVcvKpXBWw4ymW4AiyeLgBXLqZ8CtYD/61MtRQED1vQGA/6gBapNA+w6+GWtDJ6buP7TObB4UnWyIPULXLWAlc4PtBQF7GRpWscsGoCfnuEul8Ff/wlYPKk6WXMOBl39L5eg15dNwaKJ+be1FIXNV03KX3MpaifYc/NN0dIr0roCTh11XRmwbcvc4slSGYx09TtR2r7hcM1mektL0dt8gBXiOQg4fY2ycKf2qmpdAYO7Di0DBp//0uLJSsDeQ7/ca7n/sFu1j4WDWprBtPkAA4AADzwBwA+0VwAcHR1/zNaJFevPLU8pPQtXfgS9ekwvwMML4VHmJFN36ew+00D5ZBejnUQa+dwJh591L1/8wHQfY0PvS8WR4lqKaXRMN4huyUATsdRqLTD131JYAvC+nQB8X3sF4Nkf/ej2TJ1Iqv7c8pRarQGK1AuJ/em9XTCKsXi6eSyXCC5p7lndIKIlA01Epldtof5LkMtKwcBYbdFI/GOffB0ACf4py00tbMZqEgJMXKPM36W9qjIe4EueUgGLUlMLmxDw777/40eg/3zjjdHSK5LRAGe7vv0/AHzP9EUWxvJh0ylgsVcCJP/rlIOP/4ZDvQ4PutNkNMAA/PdNEOq7tsbg4UdPPgUs9krETvngTlv7P71lFF21Gb3ohjIe4Nt/jKpTn74bgOnDrjkFLPZKXHoMAY/y289AT8eANRkPcP9TCPCTZwDwk+/cfgoo90rwW69Xw3eb0MKbGLAmowFWVHBXXSS8dVB8a78GWO2VsJmeUa+YNz0NWgZc0HfVMGB9rakXjV6vugj8+LuPfwCogLVeCasG2Km+b0IvurEMC3ji8O2/zauAtV4J/uBiFr1rOVk0BrwivRGApXMODknlMhjp6ytwsjDgFemNACynY2I14Mi7ztxl7/jsEcupJmHAK9IaAi6k0CsErEispFQBBomfnHzwcctr6MhRukEYsL7WtJrUmnA1SRMGrCcMWF8YMBIGrC99wNJYaxZaBEzZdIMwYH2tuRe9tLAXrQkD1hMGrC8MGAkD1leTDv/XW7PQIuDEoG4QBqyvA9mLViSu+WiukjBgfR3IgDkxyy+xxpZ24WqjgAHra+0AK+ySAzI1YcD6OoABA0kQ2Za+IAasrwPZi1ZkqeL7YS96RTqQAVcLA16RMGAkDFhfGLCqzQg4H2nNQouAxaRuEAasrwPZi25ZGLC+MGAkDFhfGLCqzQhYam014VYB0wu6QRiwvrAXjYQB6wsDVoUB6wsD1mQ0wOJLrVloEXCsRzcIA9YX9qKRMGB9YcCqMGB9YcCaMGA9YcD6WkPAwlM1J/KNx9+0CDioP0gTA9bXG1hNkgSBa/Q0XE3SZHDABS6b5RvFFwPWZHDACgTMYcBNZDTAeW/1Z5ljGxbCLQLm9ccPYMD6eiO9aJ1HYS9a071cnWi2/tzylFqtATazWgskvVoL1GoNZMhVW6gnwWyKFCzLS04+wSlYk9EAyw74X2jsWFWqRcCsXzcIA9bXGnvRbDYrLjW9CHvRmowIWGphgiAGrMmIgBWZlZd6AgasyWiAhedbs9Ai4EiHbhAGrC/cm4SEAesLA1aFAesLA9aEAesJA9YXdrKQMGB94XHRqjBgfWHAmjBgPWHA+lrjzoZWhDsbNBkNcKvCXrQmDFhPGLC+3mDAilQ/LgsD1mQ0wLWD7pBksX75STzoTpPRADfyoqVsNlu7yST2ojVtBMB5PsuvMAVjwCvSG10PLtQPwsOANRkNcN3kMx3hyWeajAa4VWEvWhMGrCcMWF9rDnjpYXcYsKYDAzD9i3dvO/4zO8sHBeH8b6Lz/aY5cKYJ6aMA/PnTRx91Qbt6fV7IirUVoxqLGHBRBwbgL5707Fzf3feWDyoBOy0Wy8DW3wNw/m37zF85+B50XoY1X7apRbyMkqYDArD05oeqDyoBIz12cKJ4gfzWL6rh3FJbp+BqkqYDAjB459e4qoNawB//unZh9i1XqO/5paafYcCaDgzAnSdt++Qv+hcPagBbTJ3ahT8/4f7WLGLAmg4MwCA/fN8lb/p5+aAG8A/fpxm886hBvJwwejMcYKQ/mEKlg4DwL5ego5dNQfjKHfn7YsANx4y3agx70ZoOIMAvmKZKBxPC1e/MwaMbDkeWHt2quljKfx3fYvIFGHBZBwRg+Zz7e+eef/d75NJBTvAe+uVey/2H3YqCP1Z0sf7jkGdgjSnQmkkMWNMBAVi59bxjtp36g2D5oCCAqX95++EffgD5yiUX63C1xeNrrZnE2+poOiAA16mqqTJXGYKHzW48wKfvqPiAAW88wH1HDS1+wIA3HmDw7HGu8jEGvAEBg5tPJ5drAXvRmgwBWPnWRTmdK/WEAWsyBGCQvfAHypIL61QJA9ZkDMAg/ZjVaW3exV+tZoALHNuCKWMCnr/uup9dA8RLr7vOUjxhEMCADEVCqCDO6g+DrFIzwFI2K2i9jYlB3auMCRhq54tAvK78ySiAiXAkTID940Wz2WxWGy6wEb3onxFAvPza+7QRL0YBLNi9dnRyfwCWebG0z8MGBBy8AXqlLNiDxsbctX37TSmDKBqMorfg/+wHW4lEQjuaemo/mHsjlVgS8FNt6lvmF/CFCofvy9eJl+rPLU8peZUGJE4vRBxpzUJGaOmy9KxuENPag/TFZ1ZrgasnIS0J+MfwAlEBA3cXPxoli16uNm01af638GXu6utv1IYpYsB6MijgWmHAesKA9bWGgAup1iy0CDhH6QZhwPrC62QhYcD6woBVYcD6woA1GQywHPvTUvutFIUBazIW4ILV77IuGm+wNkdJLQIu6HcrYcD6WjPA2YVIZKGMrsCJunMMcTVJk7EA5+2RiL2camVRf5YwBqzJWIABb7X2l/s680LFAmh5uSraGLAmgwEG2a70aXdox4rESqUnybzIVsa7RcAp/dlsGLC+1raa5Dn+7/Xn0X6GlQ4T9qI1GRAw6DuqfqCNJFZvWIkBazIiYPBcxUh4TZXZNRIGrMlogHNooQdwy5Ij4VsETEzrBmHA+lr7hdCuXGqpjlrAilydwpcWBqyvA7A/WF5y6bRaYcD62n+Ar1InfR8OD7aXPqjLzT5hOr8qFL4ddUF38dSWU35CArrzCyea7lu02GjR8ObCgPW1HwF/0AI1qwGGH6YGUOnrfMeF51eFnulwDP77oVH1yPrU0V8F9EvX7Ty0AnCea7hruMzrxgED1td+BLy94gD9V73o7NlPfef82lCwYOrQjm7ZKqMs+vAKwDpLp2EvekVaa8A/vgLUA+ZvOCKhnfr9QWId4MbCgFek/V0GX1xRBh8JwO73MBrgcuibtm0zvXUfUK9TLKd+CmDAmg50wGop614sg4dvA8HjRoAGuBwKy+CxX71lEjHfcpDpAhcGXNKBDrg2i1YY8KJpy5YtbzJt6akpg8GHrkCo5xwMAK0Cxk7WirSGgKGYOagvnjvH1QI+8yuLN7QIuIkwYH3t7yzakq8ErKoii0ahahZterICMGGxHPpLlH2vVBiwvva3k2VK1QBWKp0sFApfjzj7UVABuLvkgK1UGLC+1rCpUhoDqNWCW2I98FY7GyibbhAGrK817i5UuGyWXeIRuLtQkxEBF/hslq95RD5ffQID1mREwArHczV9f5wgVHclYMCajAZYXWVHpumqEXZqmuZWMqpyI66yUy2jAVZV1/dXXyrjYbOajAm4ZoQdTNO1fjUGrMmQgAsszy0RSQxYkyEBo5aOJYQBazIaYLU/uHaMbANhL1qTIQG3MIIOA9ZkTMBNphVqwoA1GRJwgRWWaozGgDUZDXA+oobLS8UR75ukyWiAVSmLMSxIUsPoYi9akxEBFziutBxwgRbEhh1LGLAmAwJWRKHUkFWgSYrjG8UXA9Z0L1cnmq0/tzylVmuAzeiFUCNMMhFLEbT6iUlHE+Fko+tIuqUHxWZ0g6iWDDRRhly1hXoSzEZPwfxjsiBQhNZSmYcZtJaAax6JvWhNxgOcF7ICU3qAxBZRKxK/ou5CDHhFWlPAisRVsNSeJIs1bVsYsCajARZfatjTgNq2qoi0CDjWoxuEAevrjZ8AjkbxVPUQYy9a0wYBDJR8dawxYE0bBXCtNglgefzFtnhTGxiwngwAOHn128767IXHnf9SExtGAyw81ZqFFgEH9bdCNADgjz+qPmLiP/5X34bRAOMV3ysAl7tMm/SdYsB6MgBgqGzdmn81woD1ZAjAHSefBoYubWbDaIDz3tYstAiYj+gGGQLwR0MXAXB6MxtGA9yqNoUXDcBHwEUAnNHMhkEBFwpV4zrqtUkAXxC/CLx2QTMbxgQs8byMpjeozynIDXbm2CSA+z903IXHDjezYTTAsgNokwlz2eK4DoUThfpB0i0CZv26QYYADNJ7Xmy+m6PRABeHzULAbE4sTjFEW3PUD5LeJF700jIkYDSZEGbRxaF3Ba6mp1DVJgG8TVUzG8YEDBQVbfFUnmswcnaTAJZlmX/+jrqrKmQ0wMLzrVloEXCkQzfIEIBVXdbMhtEAt6pN4kUjRT7WzAYGrCdDAD722GOPeceuZjYwYD0ZAjBFUUtMs8SA9WQAwKKmZjaMBngVTpZcntG0KGM7Wdu2bdhqUiFfKMj5Jo+pB5znG9SXjV5NWlrGBJznBJbJ8lobdL7BbOEGgBstC4ABr0hrDliCiZEr8crzYv18/3rABVbk69qsjQ6YvO6Siy9uumay0QCrnQ1oJTSWLa1213DPqwZlcCFfP3TJ6J0NX7rz7D2X/bKZDaMBLkrhOClXWkpJFrP13UmbwosG4CxwAQBXNLNhNMBKQbWtVDxBkbgVdxc2kSEAbwcXkuBzzWwYDHCBFSrWmZXY+oqPpk0C+LLUn99/0Teb2TAYYIl3Lpa3qCdYbz20TTLoDmq8vemKUkYDTD68OA+4YcVH0ybpLlw6kgYDXEg+uJgrw/yabzAcS9UmAXzsd/uWsGEwwIB/tOKDIhdWWQYbHXD6gY+89/ZAMxtGA1w9+UxmGzjQqjbF5DNVc985qJkNowGuFieQaabhwzaJFw3kPV952/ea2TA04AJHk6nGaXiTAP7vE/7xSf39NZEMDRhk40mCbOhnbRLAzctfJGMDlpkUlWhYDdwkgEHi5Zf1F8xFMhpgsWq5AoUV2FzD62oBK7LU6A/B6Mso7T7p3z7/theb2TAa4Jr5wQVJbvysWsCSmOUbEDZ6NelTaQCcH6h86KXXXWcBYM/NN0WLJwwOWFe1gNlsVtyAgNVm6A9XPvQ69Jq+Rlm4s3hiswDeoCn4hqdZ8ve3VT708mvvY8HAEwD8oHjCaIClmdYs6JbBharGL3pB14IhAJs0lcMVFux5COzbCcD34ae9d9xxB1unDFN/bnlKrdYAQ63WApFpbDgdT2i2Gbr591x1FDLEai1Q9TGklx6TlflFOQUP/+1vvxPqxPD155an1Got8PRyruZYru4cyTS8NJNIxdNq7Pg0ka6/rfLS5UShkWhy1Rbqf0euErBDQyq5y3RFBQzcDYhrlPm7iieMlkU3upgTubrGDp16MEMymYwaOym7xOZaBsiiP/vvu0ISM/bbs18ph89dff2NCehF33ijQb3ofH0PfcNtshoBLuQVjiNTxYpzcaiewbfVafvacaaDz7m9WVOH0QA38KLrFgNHagA4zwlcXsppib04+MfoXjT8Q13it94AgBtuVdkAMKwKZyuycnX4ntEBx0ZHY81tbADADZeAbwAYZspCbVltbMDOi454/2lH/IN+VQ9sEMANVAE4rzVnFjiuLqUbG/DHH4d/sNJfPtHMhtEA60mRqxulFwEX+PqE24oMAPjdxXPvamZjgwBWWFGoSp6LgCXNyYYJeFmcDQD4Y8UU/PFmNgwNuFCeQFoQslmuMeA8X3SyOTHbcBM8PRkAsP1TR7z//Udc4Gxmw7CAoQ+MJpFqcVW4rKizrY4iS2ra5hr3J+nKAIABCA4MhJrbMBrgrDYMkuO4PFdR76ldLKvei5YEoX6f0sSgbhwMAXhpGQ2w5kWjWSuclC3tQlqvRks4FN0wpbI7ydhetKZTmtkwMuACV27hUGpn+euOyarYfBgYHfBwUcc1s2FMwAra6Sy/OIm0INXO8tcFLIuVM9aMDdj0XlVbmtkwGmBpDL7IPC9VZrQcn+Ra3ZyyuheJsunGwQCATyk6WCc2s2E0wKrYbFaoSK+ykGXTotDa3oUFlmdb+oIGAHz1gPr202Y2jAq4skoLC2Qxw1Z3KOmPi1Z0J6xVywCAW5EhActc1XwVReLZ2hruZhn4vqSMBrigLmBfsx9Hg0TZIuAcpRtkCMBbkI75V5++DaMBLvUm5aXmPd2NOvzVXqVC1ZItxvaiAbjn3kzmjzc/8Vl9GwYFXGqaLMjZxqAbLmWI/DBJrOxcMjrgC9HLOeBcfRsGBaw2dMDrmDTJN1xpp0FTZbFXia1q/jI64NOD8Dt8AGzXt2FQwAqr1mZlIcMQXKP4NkjBQvEWXqy4weiAnz3+y186/nnuBn0bRgOsaOCKs85knqTZ+i4EoFMGo8K3ulGzoD9y1hCAQXTX7mhTG0YDXCOJzmQbRnfTVJMYprkNIwPOwSxXgUmytSxaU8tRMwTg+Y9t3fqJplsIGxiwSNJpSJjlGy2W1QCwUoCZs8zWtYnoyBCA/+HRQv7hf2xmw2iAs13lwwzD0EJxIff65Ugb9QfzIiujVaYrYpca142DIQCfhV421PayFcNmBYqIcuo+hhmhbuCkzsD3HPpzqPh+Rveiz54HwHF2MxsGBlwgkhkhDySOFeqXBG808B01cXBC1drSRgf88rGf++e3vdbMhiEBwwoPiiNb7NpVFJnP1pbDdYALhYLEydAlqyqCjQ4YxHa/0HzuitEA5/rVdkp1/xRZKLVZQOI1rlMFYHVAtMQLjRamJaZ142AAwBtx36RiqDYIWi6PpJR4Trc/mEPDA2rGCLQgAwDeiPsmIaF2yuoGaHVLcJ2pKwpiK6MxAhsOcCsyImDUTllzgtMHrLY+K3mW01lRS09GAfx4cxtGA6zQZdsVi9cp0JWuZl7pZKkzXJSGcZL1V/I0CuD3NrdhMMAK/WCZpCyKDF0eGF3zyE2yIDjYaIALqYfL63GwIktxegupbB7ADze3YTDAeQhY++ULGSpJ8XrbrmwewEvIYIBB8iFtA8J8hmUYUiy6xoW6KGPAmowGuEAVY1fgWJoUs8XOe46vm9vdIuAN4GQtIaMBLikvCGRKm6KABmjVNlNtmg7/pWRUwAon8ipDpTwCr0oYsCajAgYFWc2VZY7LA4lnm2fRBZEX83prh+sIA9bXGnc2FKXkOYZH/bssrAQvMbNBzDBkpsEO7wbvbGhFRgNc7g+WuSRL0gxdPUKjrGrACYZJk7WtmUgb34u+X64Tn6s/tzylpFUayHF6IfRfiu8SzaXi6TQdY2ih0XUZvvITmyITBMvWX+l+QTcOTGtR1RefWa0Frp5EbrmA78vWiRXrzy1PqdVaEBm9EPLh4jsPM91oPM4wPN/wOoqr+shSjMA2+GILu3TjQLcQz6ZiqdVaYOojLGz0LLo86A560WJWfwZai1600QfdLS2jAS5LkfPNnoKrSZoMC3gJYcCajA64IDcep6EDWGZbrgxjwPp6AyaflT6iRUgbXdcYcJ7jiKohaoaffLak1g8wW9rz52JwlelqeMJn6gfgKjTV1XLCl7P0L9697fjP7Ky9q2a9aLUZeul6cEHMqO3WMklT0cpGr41fD14/wAWLxXKHachicYOrDtnqrQDcfeSPOPDFk56d67v73tq7KgAruVy+wNWOv9NU05JFMQT6YWRYH05XTk7CgFeklrPop0xqdK+66LxvLALese03BUF680MN71gErGQokinIXMP53zWAGZrJoJ9bYeIUWbm+JQa8Ii0f8OCbxkqA//fgR1AZ/M6vNfyFFwHLaYYhdEfCVgNmSSpd7HsS2aoV0zDgFWn5gMFXL9IAbzX9WXWyOk/a9slf9NfdUVxGCUlmGCalRlXJS3lZYqs6/WvKYC6WKX4rRaq6zOjLKC2tAwWw6+CXi4BPO/e9AdWLzg/fd8mbfq5zI9qiIUdTRYqSICS4NFs1sr0asNRoa6UlhAHrawWAwU/PcBezaOoTp3hK1aQ/mBqvZ65woshpQ2ULUkbkKJ4kRVEXsJytm364pDBgfa0EcOqo6zQni7nwRIcG+AXTVMP7CqyY5YsPgX40ywkkz6XFyvpSJWBFpmmeW97EFQy4mVYCGNx1aKmaxP/T2yflc+7vnXv+3e+pHamhLicM8kmK1HCije0oVmA5qaptuhIwWgJN50sZeznhVnTgABbfVW7oEP/l2Klbzztm26k/CNbegbzoPCx0yVKLVKG4qUrtcpWVgBvuXlkU9qJXpNU2VVJysxEdBTbLsovL1cmSBMvYTE19uDYF6+ztgAGvSKsFfOX3mgHOw/I3yVWu6i0L0MvSXfFdyedrp4eXhQGvSKsFnD7tNr2g7OvqXNFcscDVFmSQshzNCrnK6yrmB3OLDpbMVm++g7fVWZlW3Zu0cNzTTULl0kAOWFtS+5JkUUhRiao13xcBoxIYwULLPPCNRlbqCAPW1+q7CzuOGtILq7Bd6ktSJE5ixGzjerCsLUzLiqIgZAW61eoSBqyv/dAf/OxxjRfoq9ojFlWHSxy46l2yKrJoCY2NB2Ka5FiOJblWK8QYsL72R4f/zWeIUoOIcJWLqSiZJFV2pjmB02voQP8ZiiFyisC33KSFAetrfwBWnog7bfVz/5IPVyRUSRRJuoS1UNBr6FBDOZ6FuTNqlMYd/gcEYMD6IhFrXQj9oLBY4ZWyLM2JkjrrKE8z+r1J6s4svJQvFCS2aqQtBrwi7RfAjD8SqW9I5P9SsT50gWNI5EJlYW6epDJMRcTrxmQphUIOMq6ZxYQBr0j7BXDe6nbWL3bNPiiUewUVpZBjBY5LQfdJSjNMuqGTtWiUy9ZtEY0Br0j7Z1RlobrpoijRW25XltAShYosxuMklcszmUxCx8nSpK6AWKgeLy8mdeOAAetrDYfNykKp6x6tbscqBTaZJtNUHi3uX1n/qQdckLOwioU6kvHWdiuPUlFrOS667CUVV6zkRDKVTNT/lPVlMMsxOQX9VdRm0zrCgPX1xsxskNDSK2yWjSSp+vy4DrBE0HRCzaYbjqOuFwasr7XcP3imfKigxZNknk3yDZjVAc4RDEMraG5D5dejF3TjgAHra+0AS6HbqGrjBYkr9SCgSm7pdCFQGX0AACAASURBVH0WTRMEswln+BsMcHrhbjdVM3+sPPYdrdlfcrTqnSyZFutboTHgFWntANvcdzvjApfn2AYtymzF7u01gGF2zol8/T0Y8Iq0doAp+18mfs/yTMPlvWV+cUeVGsASR3ONBmbFenTjgAHraw2XE+ao1Pnf5RiUVtUWC0Vmyy0XldsSVgNGo3xSQuud/UgYsL7W0IvOS2LgvX+QeJFm1e5BWO42XKq/ZuC7kGXj6cyyvhgGrK81XC8aDZJ1HPtCvrTLBtrtqtGIyWrACsen+GXuyoEB62sN14tGQ5wLe4/szaOeAy6nwLTJF/Ick6uJdA1gMUOKahOW3pIP9cKA9bWGKTj9uChJ4sPv8EFHOkaSvCJLhQLPU+makTjVgIVMJkMjJ1pmSaKyVST4um4cMGB9rWEZzDwiK3I2+7OPSEAi0PxRRZJyAssQou6+SWhoD8NQKP0qHJGheDzDf+VRKmqN16pUOI6+NwfyKYZJFKCXxWSSUVrU33UFfiIzKRSuUPFUjMOAVx6lotZ6MVKlkEd1o2wymUOj8HiKzSSz+nOT0DzvmNrDnGdS4UTllRjwirQmgMlrP7DthH/c4Spcpa7Nc3hehgdbTvlRjM/8+dwjDzvrz/Cau9Qg9BM8/oGt739i0RwjCAUFjbsTBTSwp2hrVwHws/91ysHHf8MBCp/+DFpa/FP/H3zt/cKJpvtUwM+dXmEFANeW96vv5RWAilfqaF0Au6+9/voEEC+97jpL8YRhAMfe8/Y/DY498N4p/ttnDPcPDA1mmCvOmBh58ugvZ/jH73n11Rvf9DcI+P0WKBj/roNunLvrTa9o97IEQwr5HCNmaXUD8JItB4id8sGdtvZ/esso8B95LwB3HBOBl7923c5DVcBdB91SYQWAa688SW35Kq8AVLxSR+sCOCOC/geBeF35hGEAX/pWdRtVJp298lyBohPBUPTSD9OUcMtWJssmWY756JUQcGmv5M9fALPoSz6tHivZVIai2CxLkiJXaUvgwKXHEPAgv/0MBTx+iHXq4Oe1+w9XAX/+H0DZCpR8Qs+134TvVSsAHX5gAYYaeBiIl197H7rONzp6t1QnLld/bnlKrdZCjq09k9n6a/Wd57hvn5uhotFQMPC1s6Lh1J0HxblMKhXfcdhfJem3hxx5zGf64WVH3sxL0n3bRPUWJplOEdkMm0qSYqWtxaMdphFJ+tLZH7y89LjD75YYaOV2qWwFXXVKbvaQBDx4x1cpqeJKHXGUXkirYut/x+ySgLmfxYDCgj3oj/DxK664laoTUX9qmUqt2kJdHEZMj6rvyf7YZWpBe4HTeekHvYG97/pYNJka3mJ68+0wdOf/tj9/8ZvbqajpfpKi/mpyqbFJxvz+GBlLJqPpKlvo6Pfq+zg6s3Dw8YHS4w67E0YhavoTVbYCdfE1FHXu7+DBS+/c9tH/3Fu+Uu87pFfzC6gWGpxaCrB0o7oGAsj8ovjZKFm01fSM+s4/Vrjqg7CcHZ+NRS83bTnI9CmbkAdc36u/PnR38Ur59EsBb3oQZtHPmdQ1l/IckUqTGbm8f2XJFjoq/gxOdOb3h20pz3hTs2je9FegWtmzZcuW+4H/IBf8vT6kmlxcAegAy6KV33egRytg4O7iCaMA5g++vvj+GFrlQRZ5kUhd8QHzsEtEs/ez0Th55Ye1S795HgBH3gIBP7C1WOMt0Kl4Mi3W2YJHb/6a+r7LNAFmtj79vfeV5soUy+Aj7wSqFdbhcJDgJhPkvMVU+iPQVgA6wAAPfP3GG58Ac1dff2OieGIdAaPqzlEXdGtHJtNT5XoI/HzQid+OVF399aPVGKcfhIDzdBpmuPS3P0xxNMMk2LxI0fRlpxcvlM/4erWTBRMckYhTZKHWFnSy/vWIkpOV+/DXAHPqj7UrGjhZ+ZNunYP63He1S7QVgA4wwHVaT8BnOhyD/35oFB6hTNdClushV51psb1y2j9UXR099R0PDo0/8J574dUTg53mzlD6irO5LC8kKfaqp19/+acHweT2/WeGXvvcmwdqq0kgmw7FKDQ3qcoWrCZNHXfmLnvHZ48YBde8A6Lu3dIOgzmL5dBfWuZqqkmvHBRFb08fRi+uAFS80t34+2HAaEGdBVOHdgQW6yFXbYdZ9D0HVa3sDNK/PG3bCRe/oGjp3eRhvn0uL+QYiqF+dNohx3wcVoPB/3/qthP+eQRd/fj7Dz6t1EShSHQyDPPoZHn/rLItkPjJyQcf9w0H6NuyFwX84iQKgH7VPvr7eu4Di1b+7UL1jT30IaW8AlC/th5yQ2HAaEmsG45ILALefapiP4QsAvadv7XRRN6Cun8ZTSSSYkHOy1KWisciDXaYrGiqzLNsmiSTyYSPIpKtr3eHmyr11SLgN23bZnrrPqCVwScA8M83AfDR+9Fn6B6bbmhwj8JmUWMFlxXYLMtzEkvEgwmaLUi1k8UrAEtZLhNLxOa9fg9JovWUlHzzb3g7cp4x4CZqvQwe+9VbJrUy2LpYD7nqzImJ6z/dYOqZOiGJK8BEyWRYArU8ZjLRVBaN0WJ1OhsKLMEIGW+ICASC1qC6GjiaKtwkXo+9I4DeMGB9LaMMBh+6opxFl+shahn89Z80uEdJ/l3doC6f4VJRJsuwFBNNsQJa2KE61mXADJH0RQmCSXtCYW8ykyxA5CRJazGMdNQ9oucotRDHgJtoOYDP/ErpaLEeogK2bYF+7I4f1phiH0FzzxQ64CfTcTafzyRJiuNoUTcFU0QwlCZ4Ip0iUgmG5/OgkKLLU4nruwvtx75QPMCA9bWcLNr0ZAnwYj1EBQw+/yXI6axbqm8qrvieI4lgsDihIZ8h+CwryXplsBAJBRIxkc9CX4vj1bmjNEWW9m2oA5x63z3aEQasr9YbOo44+1FQArxYDykC7jKNAeB7+9+qbioCzpJMPCFwyF+iuQzZoEhdLINjHl+QzmZFFmKl1bF5Ei+UFkKsBSx84kelQwxYX/uzqXLirVXFpOxQw5NkWh3wLnEkSYUS9f7YohctEKlANESG48kYyfHqwJ1crhRB1l9xj2i1zv6oPPIHA9bXfm2LfuXo2aqwPFrKrCBQ6tAqLsul01zT6aOSQCai6WgyFAqTaU6UdTfTUqyRiHPxTwUD1tf+7Wy459S4SJURwpow9KOLE7kVhWa5DMGw8FCurgkvAoZOcyydDEY8gVAQTSUGMssJjSaBy85IxLP4YAxYX/u5N+knH3MGPRn1kCcoNPFMUddvkDg6wxCsPxpmFLFmFnjlcsLZZMoXjsX9iSSDph9mSZpouD542h+wLo64xID1tZ8B5/8vqC2ZJTgDfVFRbcvihRzMnymBi0ZD8RCZojhBdxEWBVaU2UCGiPJSTlEydJooZdF8ZZdVns5UZN0YsL72d38w642EULceyATdd9tkOS8xAk3LbJYnoAsVc4d9RBqmyqq1GRYBF/KFvJAVE0SaZGhW4ggyUV7UAw+bXZH2N2CFshGqd8u73H/w55FrRQgiJ3Msy0aTQacnFeXZqlWUKpcThpWiPJsh0zChJ+HfBMvHfKmMXj14URiwvtZu4Ds98cCPvgX9LJ6CkEGO5WkyFPb6fdEkm69aX2URMJr3L/MpIhkOhZ10ls9w6QRNad4yBrwireHMBuHJ2Bm/QVudMYxQgOwy4WjE5Y2TnKTAYnnRz6qoJsFEm00yTNwfDPiSLJfPxqjypod48tmKtJYr3WU453F/z8syLWQ5NsuFgm5/KAy96YJMUxTdYIZ/gU6xLJmhAx53IJoRFWQjmVs6hhiwvtZw+ihHZbiOowYUtP0CK2TisWQsEEqFA1mlkKJhzttgjQ60LHzKH00RqXCMbv27YcD6WsN1skSOShJPvM3CUlkxI4rJNBGKJJI0zwGZ5rgUV9pbdBFwXhS4ZIyIJzg2kc6ombgisTr7KFUIA9bX2gGWBTFBktwvz0gwWSnDJuNpJkFSCZJIUCxNM2x5JZ0yYOhEE76wMxpLs1yCyVBouenKBbV0hQHraw3LYPYZhqcI4moHX1Ay8VSUkXkKOloZOpwmaUkob5tTBsxmRXiZ3+FaSJIxMh2BXlbRsVZD8TJKK9Jazg9+FNaPkhTLSyDPRIIhJkcnKJoU0gvBJF2QyhuilQAXeJoIJcnQgt+TiISinjhqrMyxpf5CXE1akfY7YNak6WL+QpPpoHd+Kwjj+L0Ph8OXmr6QZWnpIdOJvsCt5x191AXtxRueMJ2nviucmIwwsYTPHVgIxn3RUAZmzrK6P4cqDHhF2u+ACxaL5Q7TkMXi5i880zqx630XwcL12x8OJq44ces8EU585KQTg9GPXv9S51cOnkTXO9/xqSLgAs8mUjzLJ7yB+QWXN+KjYIUZrcCkO2SnLAxYX2uSRRe3opW+vF2kMrcdFBf5b53tDV92/vk3+YOdh/70XTFfKJwS5KN/Ay/Knv3U5VoKpjLJhUCYjhCeGOkLJuIkqNp7CS8nvCKtIWBw1fYMbf34wSki9u1zwsnLz3vwtFjgu5defzJPROMRUjwSTYv98RVAAwxoNpaIB0PpRDxKh4PRFCqaC1xVLUlusAotwICbaZWAlYIuYMH6VXVQ/M9CEd83zk6kLv+k7chdtiNfvfZdXDYRTwV+/nYCgN3vYUqA80kiGI/4/f5wIszQiSQnSXmJq6oEC3a3rdEQbAxYX6sDLFgdqQaTURBgxRq57LTxvl+e548GQt88O5O4/JOhKy9/4H3xm09W5Hg8dvNRowAEjxsBJcCcwCYSIb8/GM5wqUCAygjZpJCtWvSdiERCZINoYMD6Wh1gayTiStefRoDzzuCXPsSDwlf+IxIhvnMulb7sk76Owz9yG3XTKbJEMr86GvIFL6Lx828ybelRV9QiSY6JJ8K+aDwaSSUJLktULlopJgERxoCXq1UBVmyRiLfBdkZqFp22feyDkM7slt5M6jvnQMCfCMXOOng+fePJQpb5wXHq0kEMGj5/yTlziFCWyJCclI76vbF4Op6OEBRD8dnFoQHQi85a562NxuBhwPpaXQqmXN7ZBmOmVMBK4rxzJAXIn/3XbP675xLUpR8LxqPzCfLWk7Pit7b9ddJiCRSvLnnRNC9yLBGPRFPJKBFLE/FkPFcxSB5VkwqNu5YwYH2t0skSOUnXi6bP/4goK9w+U3/+qo/Eg1/52LwjE0+QN57MZg5Tm0K+U7y65EVLPBSbicWikUQ0IyYYJlPpUeF68Iq0hk2VmYfRBsDFxaKZkN/qcLhTwQyTyrCRNJOJ17dFFwpZkojGYhF/OJJkQwSVxoAPZMB5PsvLQOZ5DvUNJxbsfkciFE0GmGyaJOOL00gr+oMz8bDf5/P6I/5kOhUOtrizHQbcRKsDrHBMTshzOozzEtqtQWRF6DZH/HZPiI4zqXCMYGOxBMeJpVWFNcAFSWBiibh1fsbm8LoCaZJvcWM7DLiZVgeY8PhnKavX1aDuIqTTIprTQKSCfph+AwuRWDpHc4k0nQgFYslkNlfa/rkIWGF5iox6HNNzTrs7lkgGIvFYMYdWlvySGLC+VldNskciHr8vErHXWcnbI2EbDctZf9A5H0sRiXA4TorRSCSUDscSiVhqcdHoIuACL1BEwuOdGJ+zeUMxXzzsDMfQNTLHLTWmAwPW16obOuYj7kb7u2ddEe8jCVbkIkHfQjAWDnhs/nCQjEYC4UQ0EQ5B36t0aSkFC1w6Fow6LNb5YDIeiPntrjAH1L2i1bI4MagbDQxYX60DLqRSdU8TrTaCJ+3W+ubKgjVku0lkYQHtc8zPhwO+kNPjXUjEggGCCYS9JJ9kazr8C3JOpp1evzPudydCft/03IIH/u4KrwHGXvSK1DLgwoLZvFD/OAV60YVGNiTS9SAakqFkfP6wL5b2OTzuuajXD2tKNJGJcwKqQ6kqj+jgktHAvH3BG/SFfM6Add7rYBSgqFvTAgx4hWoZMNsxNdXWoNWqvpokZWiOFgFgHhJRV0EhThNev985b7XOOhIZhs5mMgyZEbM1XjTgsuFZ57zDC4v1kCtA+N3BCI/QaX8/GPCK1DJgfu/U1F6x/nwd4Lw1YB/1O3kgjRR3tKMyMa/bGvXbLU4v9J/jJBFJCwzHc9VZtMJmJifGJqZd7oDH4ZjxRwIJtnKrNMqmG7cNAvherk40W39ueUq1fOXCjh0LDU6zmZoT6VnXRLvDFSqfoBMBz/S0dX56fG5u3uOxe6KxWJLhylEnae0658T40OC002a1221uRzweJ8hUS1+QavlL6ChDrtpCfUQZY6VgWF9pWFupS8GSI2LvCgWo8gV5QE/bJmcdDptldsbnc7kiSabR9FHaPjE6PTA2M2W1TVudPoKROa61vc82SApeb8CNVV8Gs8Mv2edKU8ZgNZbnEnNWy3TYMTUx47LZrI6ZYCpV0U1U8qIDwemxSat9ampiYswy5wpn0CpbLUUPA9bXGgBmvJEAiWaS+fPqOh1CymuZnp10O6cnp13OGadrPhjyRqhyQ3OpDA7EnFa31TE9Mjw4PDu9EGLyCp9cnEqco2ofVBYGrK81AEyE0SoOStr2O2sB+cZk0uWYtUWDVvusfT4UWPAtOK2W+UC5faqURWecLo/TNzc1MjE2MjU97Q3msyRRXskQe9Er0xoA5hbCngyQ5t13u0Q5gzoAF+advljQbp+32zxoNTuHfWYuwNQCzpKZqMs+Nz40PmKxuZ3OIJP2h4KpklkMeEVai+5CjmAUkLe773bymXQsFg447QHoWzkc/vn5hYAvzCzYZ6xxWMRWdBcq+UI+RTosEyMWi2XMMjO34ETzlCKBTMkqBrwirV1/MDd2C5lJh4PRSNDvC0G0jjmb3eZw2P1J74Ldm84XoPuFno4AFziBk+XkwvTg4NDw1OzshN3q8KdjyVSiDA8DXpHWcAI4S7EZwueOx3y+oD/knhgfHpy0TlusAZ/PGQ7n0MxQtUsJAZZFNMCDtNpGewdHx2yuWYdvPhAmAgsuspSTF/QXjsaAK2+p7kbfj4AVlqz8pSEzIcXHvclkKhYJROfHRkdh6ToxNjE6YfPPuwkEWG2OVgELWZ6kY/bJ/oGR6em5WYvV4wv5SX8olmrht8eAF0UteKyVrQf7ETDtCS/Ar8SnCbWjPi9kBZ4lE6kYkSJj4aBtarh/aGRsYnJkbMERSIalAsurHUoIsCJx0ZQ3GLEMDg+NW512t8fqDqajDo8/1GgpwxphwGWhLnpv5e+xHwETkUiEQA1ZYatqVOIkRZF4mk4nUim/b2JoaGB4cGRkArrIFm8oxslK8S+t6EVLZDrgss9MDw+P2PyOgN8ZiKRIn28h0miuSo0w4LIUa9i+b6Ji7cD9CJgKRHw0EDyRiFMtN7NdAKVjkuZi4ZjPPjU+NDQ8OmYZnbA6vC53iExqHcJFwPlkLDA1PjUxNDo+Yp2wBkOJRJpMpCPRUhmcGteNAwa8KG6iwxaqGICxHwEXSBulANkWChZTsLrFu8xE4nSEJDwO28jowPDo9JzH73YFvIF4MBFj1OuKgGXK53fNzE6PjE8NjkxbHP44QctMIimUOp2wF92aMt5IxJYr752+/73oLEkVEx3/aL4gcamg3xsJuq226cGhqdGJiRH7dMQ5bQ/M+0PpYl9CsR5Mc7RzbnJgYHBwZHh4fGrOFg5ykkxz5dlJGHBrkmx+t985G9LuXLNqUi49cY/AMnzE73GHff7AxOjw1OjwyMjotMVln52bD/rjWh6tAs6nqODk8GDf8EDP4EDv0OTEpCcWIskUQZScLAy4RUkMY7MOdUWKv9uaAbaG3U9GhQzrtdtnPXMOz9z05MDI0MjQtMVmm3UuuELRWDqbLzV0gDyTso+MDfQPDph7ByDk0UlbIJSIRxYngBPTunHAgKuVnx2YGpwreqdrBtjmHBwNEkJmwW21Wixz1tnxgb6+weGR3sGh8bHJ6Yk5dzgtoM06ioALTHBsHNLtMZvNff0wn7bYfd5QVF1KaUlhwDVK7+23zaTVvG/NALv/uvP/ehM5IZoMuZwLTuv4xMDA0MDodH//cP9w3+iUxZ+IFvfIKtaDIwtjfX3mnl6zubvH3DcyMO2Ow6LbHadaiCAGXCtixjIW9KL5CGsFWJnsbO8YYPIcF5x3ztndcz7nYH9PT++QuafPPNA7MDw+74ujFf+LgCUx4pkb6B1EKbhroN/cPzY5G4mlAmSmhZUMMeB6FWClVV16f81S8ExP374+upAXotHAgsPvclnN3WZzDwTYYx7q7esfccVJMklrKZjNJhamhgcG1fQLr+sdmZgNxonE4sZnuC16eeLckRBafGHNAAvj+24nCFpMBoKBcCDinZ3s7+3pRzmweXgAFrOj9hDFiWgKqZqC2cTM+NBgHwztMiPKw1NzYYJmSYJXCsVxs6oX3XhcGAZcL4WypVGSWCvA4tzE7INEkkoScbc3Eo1YZ4YhXQSwp29ocLCvb3jaFWSyJcCFVHhuarQXsTX39vT09Q7OzDh8tCQKkqT1KiLApNOZafAwDFhPrLVfd3pvi2oMWJ4Z/vuDv03QRJSM+2Nx0jMx0T/U39fX3Qed5L6+/oGBwakpf0LdQhYCzlNR78xwXzGBw1LY3D84OWYPRdMMJ8RITu1VhIBz86Vm0GphwDqSnJ7+XmZVJqoA57PF+CiE47kn2174YTyTdPs9/ngqYJ8Y7INOdC8i2NXbZ+7rHhqZm08wWnchFZ6fHu3vUTPo7q4uc0//wOTEtMcfnvcHohmalWCFGS3CsgABNyiKMWAdCc6el596iViVjQrAgn3Bqm44mJhwDv7x1R1PkhnvfMwdJIPT0IEa6jP3QoBdxVK2Z2jcMh9mit2FhWAiMD3U091t7u7s6lJT8cDo+KTT5QpEAy6CIhPpZC5HyXzCtdAothiwjvKT5r++/uxT860NL2+sCsDpSCSQKVC28OzI6Ozjr+z1i7D89S14kl67bWxgeBA1YnR3dXcjL7lvZGTM5lO7GeiskojOT8HMG2bO3V1m9M88AlO40xePJ0J+ns0Eg8FIXrR5ZlPZRnHAgPUkDjz3yP0P/N9AC3XNClW5sjWAqbgrYrGZh15Lk7TCpJMBhy/g8c/Zxwb6B1DKhIm0GybVHvPwlM0TVrSmSjHhmh6Eibur21y8pH/CMuLwh0mfbz5JJiKBBa+PToWdPaON1kHDgPWVG7vviV0vjxHLIQxd2Yox6FVZtGsuPt4/Nzs09/yrCQkIRDgSXnD5570TU2OD/YgcKmURwL6evpFxb1JiJRWwIiYGezthBg3/9SDIPb2jU1ar2+Oec8YSZMTudMSZtL93wrbQKAljwPpK2V987YmpUSfRcnUp59Q8HUVG91Q7WYLL2fdEx96Orp4XxyMvx+Mhp3VhenZ8xKw2MiP/uKO7qx1l070Dg2Pz4SQnFatJ3qEuc2d3dwdCbEaF8eDY+HTAPTs7MzOfCHsDoXRwemZvt9/VYIIjBtxEKcnXNbUnEHHEWxgaowrVVeYh4JzVOTZtZSsBZ2lqITy1x2F/afSJ/xno/UM66JyYsw2PDPcjuIgczIY7YRnb1tnR3T087ghGkrkiYMcITNxtnZ3ICUOJuLt/ZGp8wWqZmBqccEZimTTn3qMk5l1pBY27zlR/bwxYXykF5AVbxHX39V/vaM3XUogF1ZUlwoFue8RBE+U4iI6AfXZsZC7i2fnqn/72+NO/Ti2EZqxzoyOQL0yz3T1d3R0dnR2dZnN7ByqP+ydh0aoBtvZC+F3d7WoO3qW2h4xMTM9MzI5a7Pa4wCVTM7vgXxDaLEuYj/iri2IMWF9qSxa50P39G6/83PUN1hVtpOL6KUTE32+PTMzarKVIUKGQudtnSTonvK/uu/+pJ24Lz0cds7OTM8Pmou+ECtlulErNbSgt947ZHOGCCpif7EMuVldXB/Sgu9R8umdg0O51OYdH7O4UHwlGh57UnpMp7WBbFgasr2JTpWj57/++7JvXP6o3gU/JpKnaRwkOz5559ytBT3krbtrvaZ8ML2RZe2R219+ffuoeH5P0uP2e2emJ3l6tjtTe1dXZ3gUTcEcXTMkT87DsRy1ZkWn0F9DV3t3WDQtps5qOe3pmHAvuKevEXIzyB6ITe9BSL4rAc65IoLo2jAHrS2uLZn592b9e8qcHp1ONWy5pX8RfB1/icww9F/KU/Z5C6sWdvVZbng5E/DvN9z/UNRUJzcdZym4Z7h3oQ3w728ydSKi9A2XVPVN0sSXLMdILnayOTvgX0NkFS+keNaPun7PPO1w2VzQWDi8MWl4dZQHh9qRpgqwuTcqAC9TivPPlNLJvfMCA2/Oz39zT1eFxNljAvTjkOZ1rVEazVstieSh4fLZRNhOdCAf6unfsME9Ne8PeZNTrnh4cGuzrRdXgtg7UkNEG6bbDapF5JCBDwHlidsDc1dG5r629qx3y7+nuVJuzBoft0xNjloX5BYIMeHqmzA4BuvCuuppSCXBhoavDrR4phN3ayN/W0SYADEvQyZExWyTsd7vrWwMZd8Q1Mm9vBF+uWHkHtRXbSH/E6ifZsYH+/omFYMjtddssE/09aj24p70T/mtv72yDoDvbe7p6bGQeAvaN9XR17oUeWBdMxR0dqK2rs6vdbB6esY0NjVvs3lAg6uyftLl4B6yj5UB2zlrZq1QCTHWVForhvI1WZ9PVpgAM+cSCEVeneeef06AmsSosEfE1/smK1SSFIdC3oew2rjjBAZD28Zl4yu1C66n0D/X398IkCn2oduhBtbd3dHS0QX4woVqCXCYr+YY64bn2jn0wCNaVYE7dbYb5uHlwcmx8aGRqyuWJsr497S5rIT0yRuTiz3XaFioKkxJgum1q6jW1uscEah2xptokgEGesIZfealtR697PlnbtgUL1oaAc2k0Mp32hF3o6xQUhfL5p1FtNZt4OB2Zt9umZ8YHR/p6+vphyoTuVWdXR0dXe3vb3o7ONuhPDQcpKivOswy6PQAAC0hJREFUmmEQJNuNAuF1MIc2w7y8uw/SHZmYHLLM2YLDz9FMPm8L+CJz7c+Yh22LOYdSHtfj3ftaWD2Qrb6FRj3HOjowAF+9e/duD3yPwHd1jLB593O7HPA9BU/sRg8Ygu8z8J1GJ1BWNQ7f0YwPEZ1AX2IGvqM1HwvoBJo8P7Rr924zMrYHnohAXs9cc8P19zjdNs8LO3ft9sGAEAx4Ff1kTz7wzG60OVUCnngB3dK/e9fOaSYjRnbvePQhTzgFRnfv3jGYibic//f0jl2Z9EP+zkfv/59nZ8eGe++87bbbXoGe1WM333Lz3e1tHR233nrbLTugN/XUn/7+1F5qpLv99ptuvmVH9759f7/11pvvhJQ7/wgveaZ/cPT1u/541/1jI46Ja3fs2mVxen1t999663/3Dojlr0s88dyzO9DXndj93PPjoIC+7o6nE42+rmN39dcFHvjehgDvhQelr/sKuqILHsxXf93du9GfOIWMoXwCft3daPc2Hr4/i35dCzwYge8yuoJcNuC7wuEwukWE7yhuIBn2BFFvbg6eCKOUmIbv6O9WRidQLkvCd+QD5dEJ1LyYge9opI6CTqBI+kLhsFr9jcAT6EeKP7O7vdPjnfN5Zpx+TlEADwOi6EcK+hefr6aTVHDBE6EzWTK4MNZrdyQkIuCwzYcdrvDIjNeZTv6ve3akr398ZmKgeyf8xqhu9MLOp595aR9Mu0/veOaZV2Byfv01t8Pnh/nxDgjpZQj2tWeef+b5LnNHx64dO3a+3m3uH9z7+kt7Jgan+u+bdDgWRudt/d1dzz7SF6cL6Osir5myeT3+0tclOI6VG35d5Jsx4YqvizJ5Dr7HEWDoqKO1T4GgfV2Q0H7ubPnrwne69HOj9ERoz0ePc6Mfk0LPRz+V+usegFm0qjzLRR1uS5AlyTCVicO8R/chUjYrkpQg5VkyPBESeZkjAvNut8PrDfDJdPh38zYnrAbPjY6hDoQe5EB1dsL6b4faoLWvDblTXf0cFe7rhi5VW1c38qLb2jtgedzZY+5QWzXNQyPTI0O9/UPjr/zZHaSioVF/LGlxeAgmVS5EZKJi0rO8uEZi6zowsug3boa/QqUJjkkmUpFoIpqh/P5s4xWr0I8psnmg0KlUiszyuXQ4YHe65tz+GEml3K9bZqanrdO28cEe1NfQ3d3ZAbFBP3ov8rIg7XbUbhWNz5tV3xm5WPANVqHa2rvb9rUjwN3m3qGR8dGJ0VHL1IhnLhxLJkieoaUETVcspkbLi05haVr5srTZACPJWSIZj82nEz6vNxjOUwFPCv2KhVyF96pILCcL6h9ELJlh2GTIF3VbF7y+ZCwQ8HrGR0bHRsdHBof6zD1D5jboRne372vfuw/WkWBd6fVO6ElDp3p0YaKnA7VudcJE3N6+F/nYMB1D+qje3D08NDk2NzU+7Zudm5lyxVIJUeIFDv7XmMqsVKhsyVI4nl32AIbNCBiWNYlMwptIhvzBgJcMW50uWN5IlCcYKPc7cdG4WOBkWSQZMsZkSCIVC/vDocg86V2Yt89ODsHq0djY0PCIuae3D1Z+oHPcta8dke5sM7d374MwYXruNe+DCbgNNXHs64Cn2traUXNIF+o17Ok3D4xOjc44Z11hp8vjC9EpjoYFQxmhul54dVPlstqwilpPwHtuvilaPHrDF2GRRVHIMDGX3x2LeJzzC4G8wqVikfKqGaLN5XZwaVFgCSoVjaZiGTpJ8uFQ2JOcX/BZ58aHh3oGhwYGzWpvYRdMnhBtuxmmTlhXaoe1og4IuKutbV8b/IgSdWfbvo727s59MPHC3Lu9s7tnoGegv39o2D5vDcXD8y53OkOKyWzFhneykM2yDVY5Xp7WEXD6GmXhzuLhuqyyUyjIVDDNsoE5m9sP0yx0JeOlXyPq8Hq8LJXNMkwqnSSjsTQH3W5/IuVN+dxue2C3ZXzaMjc0NNzX293XAytJsJrbvrcblrivowx5H8Tc1g1BIrZtnfu6OtQcGvpbMF23oS4Hs7nX3Ns7MDlmc8wnI+aYP0DTZJbmOKkca4UVRcnInQ0DTwDwg+Lhui2jpEBHmk/GKJhsZNIVCpaSjzjnci9kCZhHSlmGpDIZHo3xYNNkNJOMhb2R+xyOOW/QPjs3DGtLZrWjsHsvrCvBbLkbVoX3tnW07evaB19gBo0+Ile6DZa/XV2oI6LLPICK74H+4ZlZpzPG2Z9jeUbMoPXfK+NckGG+stpfYR0B79sJwPfh+32f+czN6Tql6k8tU8u1kEonFj+EFtzxNKwvQiOpeCBUDElGI/FELJaIu/7gcrrRugwTY9CP7kVtU2aYilGmrDY4w2y5o031tFDVyIxK4b2ooO5GOXfv8NjwyOjwxOi4ZXLa5vRFQhN/jUbjqVSyQYxX/TOkVm+h/lRyWSkY1qHvy9eJl+rPLU8peZUGJE4vhPlLXpJlOceSFJWMRUIL0Bl2zDps0+P9XWj0VQfymvd1FDNoyHwfLJu7h8dtc+NTM/5EjKID4Ug44g0Gk7wEY+l5IZfTiSyzyu+Q5zOrtcDVk5BaAkxco8zfVTw02Ep3QHyp7tq8okhyIS/lsiJHJvxTnd1mVC2Cybhz0hePxFklDxnmsgW11g1fCovV71iPbhyMnEWDPTfeuF5edEvSB9yq6IaD2ZcjQwNeFAasJwxYXxgwEgasLwxY1WYELDzVmoUWAQdf1w3CgPW1hoDRUoatqEXAeCG0FQkDRsKA9YUBq9qMgPPe1iy0CJiP6AZhwPrCXjQSBqwvDFgVBqwvDFiT0QDLjtYstAiY9esGYcD6wl40EgasLwxYFQasLwxY0w+urdM1v6o/tzz9aLUWfvVLvZBfXt6ahZ9e3dJl/3WlbpBuFFrV1T9drYUGUfjVdm55gNt373/t2P7sGlhdnr5x63rHYPct31gLq21geYDXQtL2hjP631D9/JX1jgF4+edrZ3t9AefvaHV5rLXT8zPrHQMws2PtbK8vYKw1Fwa8wbWOgPPXXNoJKmc/rYvW+fFr/iOsI2CFeK6zavbTemidH7/2P8K6ZtHou1XMfloPrfPjwZr/COsOuDT7aZ20zo8Ha/4jrBfg1255HKdgVRs9BVfMfloPrfPjwZr/COsJ+Hff//EjlbOf1kXr/Pg1/xFwPXiDCwPe4MKAN7gw4A0uDHiDa4MC3laz3jrx5RNP/NcwAMeecOKJV9VctAUA6aZ3n3LatdJG/DU24FdCqgWc2i0VfvoVCHiu/iII+JufTwLuTmoj/hob8CshQXavnXPWP7oAeP70j963RT330kdLgLUgeNHu0y+4bQtYOETb0cf0h7Pf8xLkvf3DXyLhp99d/tGXyhYGP/2Rj3Wt2xdasTYs4Njb5sEjHwfRE4LgXhVw4bN3QcDvPOWUl7Ug9SIP+N0W8PyZ2m2mh8HkyTC9A3DndfDTS8B7cskCcVYMBE9e9SyIN1wbFvAL/wxAfiuz+0sAZFTAP/5SXkvBWhC8aPcXYPG8BTz3Qe02E8y0D5XAny+84OxL4KccAIdJmoU9x1100UWnudbvK61QGxbw7s+pFHd9WQP8n19Aw79UwFoQuuirADAwi95WyqLh/8PF8feTYM/FpU+ahRcvXLcvsyptWMCxtznAw58A0eOD4B4I+Cf/pg7vUwFrQeiikynwdxj6jc/HAfdbSkPadhEAl5UBaxZSx+4DYHgdv9IKtWEBg1fPPuMzMEd99v2fuusYMGOC9aOyk6UFwYt2XfDtWyDg3PWnnPi+GyUNaf6yz132izJgzQIYOv+M0762vt9qJdqggBfFAfDEJetsYT214QHfsP2ciz3rbGE9teEBb3ZhwBtcGPAGFwa8wYUBb3D9P+hitMQnnke6AAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# significant thr \n",
    "pvalue_lim = 0.005; log2FoldChange_lim = 3\n",
    "res$sig <- as.factor(\n",
    "    res$pvalue < pvalue_lim &\n",
    "    (res$log2FoldChange > log2FoldChange_lim & res$pvalue < pvalue_lim) |\n",
    "    (res$log2FoldChange < (-1)*log2FoldChange_lim & res$pvalue < pvalue_lim)\n",
    ")\n",
    "\n",
    "relevel(res$sig, ref=TRUE)\n",
    "# Volcano plot\n",
    "\n",
    "v = res %>% ggplot(aes(x=log2FoldChange, y=-log10(pvalue),colour=sig, fill=sig)) +\n",
    "        geom_point(aes(color = sig),alpha = 1/10) +\n",
    "        ggtitle('Decitabine_treatment') +\n",
    "        xlim(c(-15,15)) +\n",
    "        ylim(c(0,100)) +\n",
    "        geom_hline(yintercept=-log10(pvalue_lim), linetype=\"dashed\", alpha = 4/10) +\n",
    "        geom_vline(xintercept=log2FoldChange_lim, linetype=\"dashed\", alpha = 4/10) +\n",
    "        geom_vline(xintercept=(-1)*log2FoldChange_lim, linetype=\"dashed\", alpha = 4/10) +\n",
    "        scale_color_manual(values = c(\"grey\", \"black\")) +\n",
    "        theme_bw() + theme(legend.position=\"none\") +\n",
    "        geom_text_repel(\n",
    "            data = subset(res[order(res$pvalue),], sig == TRUE)[1:10,],\n",
    "            aes(label = name),\n",
    "            size = 5,\n",
    "            box.padding = unit(0.35, \"lines\"),\n",
    "            point.padding = unit(0.3, \"lines\")\n",
    "        )\n",
    "ggsave(\"plots/peak_Volcanos.png\", plot = v, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/peak_Volcanos.pdf\", plot = v, device = 'pdf', dpi = 300)\n",
    "v\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>name</th>\n",
       "      <th>sig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ENSG00000196230.12_peak</td>\n",
       "      <td>32934.670168</td>\n",
       "      <td>-2.633326</td>\n",
       "      <td>0.067697</td>\n",
       "      <td>-38.898739</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>TUBB</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000153187.18_peak</td>\n",
       "      <td>18565.943231</td>\n",
       "      <td>-3.177571</td>\n",
       "      <td>0.072183</td>\n",
       "      <td>-44.021117</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>HNRNPU</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000185624.14_peak</td>\n",
       "      <td>53230.012498</td>\n",
       "      <td>-2.516796</td>\n",
       "      <td>0.060242</td>\n",
       "      <td>-41.778367</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>P4HB</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000005381.7_peak</td>\n",
       "      <td>72644.854943</td>\n",
       "      <td>-3.132771</td>\n",
       "      <td>0.059094</td>\n",
       "      <td>-53.013172</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>MPO</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000262831.1_peak</td>\n",
       "      <td>25918.529538</td>\n",
       "      <td>-2.425949</td>\n",
       "      <td>0.068558</td>\n",
       "      <td>-35.385574</td>\n",
       "      <td>2.846204e-274</td>\n",
       "      <td>5.325817e-271</td>\n",
       "      <td>RP11-498C9.2</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000145990.10_peak</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GFOD1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000149115.13_peak_3</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TNKS1BP1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000162946.22_peak_2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DISC1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000179583.19_peak</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CIITA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000182324.6_peak</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>KCNJ14</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9367 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               baseMean  log2FoldChange     lfcSE       stat  \\\n",
       "ENSG00000196230.12_peak    32934.670168       -2.633326  0.067697 -38.898739   \n",
       "ENSG00000153187.18_peak    18565.943231       -3.177571  0.072183 -44.021117   \n",
       "ENSG00000185624.14_peak    53230.012498       -2.516796  0.060242 -41.778367   \n",
       "ENSG00000005381.7_peak     72644.854943       -3.132771  0.059094 -53.013172   \n",
       "ENSG00000262831.1_peak     25918.529538       -2.425949  0.068558 -35.385574   \n",
       "...                                 ...             ...       ...        ...   \n",
       "ENSG00000145990.10_peak        0.000000             NaN       NaN        NaN   \n",
       "ENSG00000149115.13_peak_3      0.000000             NaN       NaN        NaN   \n",
       "ENSG00000162946.22_peak_2      0.000000             NaN       NaN        NaN   \n",
       "ENSG00000179583.19_peak        0.000000             NaN       NaN        NaN   \n",
       "ENSG00000182324.6_peak         0.000000             NaN       NaN        NaN   \n",
       "\n",
       "                                  pvalue           padj          name    sig  \n",
       "ENSG00000196230.12_peak     0.000000e+00   0.000000e+00          TUBB  FALSE  \n",
       "ENSG00000153187.18_peak     0.000000e+00   0.000000e+00        HNRNPU   TRUE  \n",
       "ENSG00000185624.14_peak     0.000000e+00   0.000000e+00          P4HB  FALSE  \n",
       "ENSG00000005381.7_peak      0.000000e+00   0.000000e+00           MPO   TRUE  \n",
       "ENSG00000262831.1_peak     2.846204e-274  5.325817e-271  RP11-498C9.2  FALSE  \n",
       "...                                  ...            ...           ...    ...  \n",
       "ENSG00000145990.10_peak              NaN            NaN         GFOD1    NaN  \n",
       "ENSG00000149115.13_peak_3            NaN            NaN      TNKS1BP1    NaN  \n",
       "ENSG00000162946.22_peak_2            NaN            NaN         DISC1    NaN  \n",
       "ENSG00000179583.19_peak              NaN            NaN         CIITA    NaN  \n",
       "ENSG00000182324.6_peak               NaN            NaN        KCNJ14    NaN  \n",
       "\n",
       "[9367 rows x 8 columns]"
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res = %Rget res\n",
    "res.sort_values('pvalue')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### iPAGE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "mk_dir('ipage')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "awk 'NR>1{arr[$1]   += $3; count[$1] += 1}END{for (a in arr) {print substr(a,1,15)\"\\t\"arr[a]/count[a]}}' deseq/peak_m6A_vs_input.txt > deseq/peak_m6A_vs_input.ave.txt\n",
    "for f in `ls -d /flash/hani/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl*`; do\n",
    "    base=`basename \"$f\"`\n",
    "    perl $PAGEDIR/page.pl --expfile=deseq/peak_m6A_vs_input.ave.txt \\\n",
    "    --species=$base --exptype=continuous --ebins=11 --nodups=1\n",
    "    mv -v deseq/peak_m6A_vs_input.ave.txt_PAGE/ ipage/peak_m6A_vs_input_${base}/\n",
    "    cp -v ipage/peak_m6A_vs_input_${base}/peak_m6A_vs_input.ave.txt.summary.pdf plots/ipage.peak_m6A_vs_input_${base}.pdf\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_encode_tf.png\" style=\"weight:100px\">\n",
      "encode_tf\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c1.png\" style=\"weight:100px\">\n",
      "msigdb_c1\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c2.png\" style=\"weight:100px\">\n",
      "msigdb_c2\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c3.png\" style=\"weight:100px\">\n",
      "msigdb_c3\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c4.png\" style=\"weight:100px\">\n",
      "msigdb_c4\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c5.png\" style=\"weight:100px\">\n",
      "msigdb_c5\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c6.png\" style=\"weight:100px\">\n",
      "msigdb_c6\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c7.png\" style=\"weight:100px\">\n",
      "msigdb_c7\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_full.png\" style=\"weight:100px\">\n",
      "msigdb_full\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_h.png\" style=\"weight:100px\">\n",
      "msigdb_h\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl.png\" style=\"weight:100px\">\n",
      "plots/ipage.peak_m6A_vs_input_human_ensembl\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_all_gene_ids.png\" style=\"weight:100px\">\n",
      "RBPs_all_gene_ids\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_3UTR.png\" style=\"weight:100px\">\n",
      "RBPs_coding_gene_ids_by_3UTR\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_5UTR.png\" style=\"weight:100px\">\n",
      "RBPs_coding_gene_ids_by_5UTR\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_coding_exons.png\" style=\"weight:100px\">\n",
      "RBPs_coding_gene_ids_by_coding_exons\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_introns.png\" style=\"weight:100px\">\n",
      "RBPs_coding_gene_ids_by_introns\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids.png\" style=\"weight:100px\">\n",
      "RBPs_coding_gene_ids\n",
      "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_DeepBind.png\" style=\"weight:100px\">\n",
      "RBPs_DeepBind\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for i in `ls -d plots/ipage*png`; do\n",
    "    o=${i/pdf/png};\n",
    "    n=${i/plots\\/ipage.peak_m6A_vs_input_human_ensembl_/};\n",
    "    n=${n/.png/};\n",
    "#     magick -density 300 $i -flatten -quality 90 $o\n",
    "    echo '    <td><img src=\"'$o'\" style=\"weight:100px\">'\n",
    "    echo $n\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c1.png\" style=\"weight:100px\">\n",
    "msigdb_c1\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c2.png\" style=\"weight:100px\">\n",
    "msigdb_c2\n",
    "  <tr>\n",
    "<table>\n",
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c4.png\" style=\"weight:100px\">\n",
    "msigdb_c4\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c5.png\" style=\"weight:100px\">\n",
    "msigdb_c5\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c6.png\" style=\"weight:100px\">\n",
    "msigdb_c6\n",
    "  <tr>\n",
    "<table>\n",
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_c7.png\" style=\"weight:100px\">\n",
    "msigdb_c7\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_full.png\" style=\"weight:100px\">\n",
    "msigdb_full\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_msigdb_h.png\" style=\"weight:100px\">\n",
    "msigdb_h\n",
    "  <tr>\n",
    "<table>\n",
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_encode_tf.png\" style=\"weight:100px\">\n",
    "encode_tf\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl.png\" style=\"weight:100px\">\n",
    "human_ensembl\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_all_gene_ids.png\" style=\"weight:100px\">\n",
    "RBPs_all_gene_ids\n",
    "  <tr>\n",
    "<table>\n",
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_3UTR.png\" style=\"weight:100px\">\n",
    "RBPs_coding_gene_ids_by_3UTR\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_5UTR.png\" style=\"weight:100px\">\n",
    "RBPs_coding_gene_ids_by_5UTR\n",
    "    <td><img src=\"plots/ipage.peak_m6A_vs_input_human_ensembl_RBPs_coding_gene_ids_by_coding_exons.png\" style=\"weight:100px\">\n",
    "RBPs_coding_gene_ids_by_coding_exons\n",
    "  <tr>\n",
    "<table>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### TEISER\n",
    "https://github.com/goodarzilab/TEISER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%bash\n",
    "# declare -a StringArray=('METTL14' 'METTL3' 'Virma' 'WTAP')\n",
    "# for treat in \"${StringArray[@]}\"; do\n",
    "#     # ave duplicate: peaks  \n",
    "# #     awk 'NR>1{arr[$1]   += $3; count[$1] += 1}END{for (a in arr) {print substr(a,1,15)\"\\t\"arr[a]/count[a]}}' deseq/${treat}_peak_m6A_vs_input.txt > deseq/${treat}_peak_m6A_vs_input.ave.txt\n",
    "#     # ave duplicate: exons \n",
    "# #     awk 'NR>1{arr[$1]   += $2; count[$1] += 1}END{for (a in arr) {print a\"\\t\"arr[a]/count[a]}}' deseq/${treat}_peak_m6A_vs_input.ave.txt > deseq/${treat}_peak_m6A_vs_input.ave.c.txt\n",
    "#     perl $TEISERDIR/run_mi_gene_list.pl \\\n",
    "#     --expfile=deseq/${treat}_peak_m6A_vs_input.ave.c.txt \\\n",
    "# #     --genefile=miRDB_miR-296-5p.predicted_targets_ens.list \\\n",
    "#     --exptype=continuous \\\n",
    "#     --ebins=7 \\\n",
    "#     --species=human \\\n",
    "#     --doremovedups=0 \\\n",
    "#     --doremoveextra=0\n",
    "# done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
