{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "mkdir -p qc_fastq\n",
    "mkdir -p deseq\n",
    "mkdir -p plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Input_T1_S43_L006_R1_001.fastq.gz\" \"Input_T2_S44_L006_R1_001.fastq.gz\"\n",
      "[3] \"Input_U1_S41_L006_R1_001.fastq.gz\" \"Input_U2_S42_L006_R1_001.fastq.gz\"\n",
      "[5] \"IP_T1_S47_L006_R1_001.fastq.gz\"    \"IP_T2_S48_L006_R1_001.fastq.gz\"   \n",
      "[7] \"IP_U1_S45_L006_R1_001.fastq.gz\"    \"IP_U2_S46_L006_R1_001.fastq.gz\"   \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "list.files('fastq')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@CO\tuser command line: STAR --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat --runThreadN 16 --sjdbGTFfile /avicenna/genomes/hg38/hg38_genes.gtf --genomeDir /avicenna/genomes/hg38 --readFilesIn Input_T1_S43_L006_R1_001.fastq.gz --outFileNamePrefix Input_T1_S43_\r\n"
     ]
    }
   ],
   "source": [
    "!samtools view -H bam/T1.input.bam | tail -n 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I've started from bam files previously mapped by Hani or Alex. All `bam`  files renamed in this format: \n",
    "> `<treatment>.<input/m6A>.bam`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd bam/\n",
    "for f in Input_*; do \n",
    "    o=${f/Input_/};o=${o/_S4[1-4].bam/.input.bam}; mv $f $o; \n",
    "done\n",
    "for f in IP_*; do \n",
    "    o=${f/IP_/};o=${o/_S4[5-8].bam/.m6A.bam}; mv $f $o; \n",
    "done\n",
    "cd ../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"T1.input.bam\" \"T1.m6A.bam\"   \"T2.input.bam\" \"T2.m6A.bam\"   \"U1.input.bam\"\n",
      "[6] \"U1.m6A.bam\"   \"U2.input.bam\" \"U2.m6A.bam\"  \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "list.files('bam')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## QC \n",
    "PCA plot "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/T1.input.bam                               ||\n",
      "||                                                                            ||\n",
      "||             Output file : feature_count/T1.input.fc                        ||\n",
      "||                 Summary : feature_count/T1.input.fc.summary                ||\n",
      "||              Annotation : /rumi/shams/genomes/hg38/hg38_genes.gtf (GTF)    ||\n",
      "||      Dir for temp files : feature_count                                    ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /rumi/shams/genomes/hg38/hg38_genes.gtf ...           ||\n",
      "||    Features : 535393                                                       ||\n",
      "||    Meta-features : 26485                                                   ||\n",
      "||    Chromosomes/contigs : 255                                               ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/T1.input.bam...                                       ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 125909291                                                 ||\n",
      "||    Successfully assigned reads : 19752656 (15.7%)                          ||\n",
      "||    Running time : 3.56 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"feature_count/T1.input.  ||\n",
      "|| fc.summary\"                                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/T1.m6A.bam                                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : feature_count/T1.m6A.fc                          ||\n",
      "||                 Summary : feature_count/T1.m6A.fc.summary                  ||\n",
      "||              Annotation : /rumi/shams/genomes/hg38/hg38_genes.gtf (GTF)    ||\n",
      "||      Dir for temp files : feature_count                                    ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /rumi/shams/genomes/hg38/hg38_genes.gtf ...           ||\n",
      "||    Features : 535393                                                       ||\n",
      "||    Meta-features : 26485                                                   ||\n",
      "||    Chromosomes/contigs : 255                                               ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/T1.m6A.bam...                                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 46222685                                                  ||\n",
      "||    Successfully assigned reads : 36665869 (79.3%)                          ||\n",
      "||    Running time : 1.25 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"feature_count/T1.m6A.fc  ||\n",
      "|| .summary\"                                                                  ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/T2.input.bam                               ||\n",
      "||                                                                            ||\n",
      "||             Output file : feature_count/T2.input.fc                        ||\n",
      "||                 Summary : feature_count/T2.input.fc.summary                ||\n",
      "||              Annotation : /rumi/shams/genomes/hg38/hg38_genes.gtf (GTF)    ||\n",
      "||      Dir for temp files : feature_count                                    ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /rumi/shams/genomes/hg38/hg38_genes.gtf ...           ||\n",
      "||    Features : 535393                                                       ||\n",
      "||    Meta-features : 26485                                                   ||\n",
      "||    Chromosomes/contigs : 255                                               ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/T2.input.bam...                                       ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 117775402                                                 ||\n",
      "||    Successfully assigned reads : 18015207 (15.3%)                          ||\n",
      "||    Running time : 3.48 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"feature_count/T2.input.  ||\n",
      "|| fc.summary\"                                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/T2.m6A.bam                                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : feature_count/T2.m6A.fc                          ||\n",
      "||                 Summary : feature_count/T2.m6A.fc.summary                  ||\n",
      "||              Annotation : /rumi/shams/genomes/hg38/hg38_genes.gtf (GTF)    ||\n",
      "||      Dir for temp files : feature_count                                    ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /rumi/shams/genomes/hg38/hg38_genes.gtf ...           ||\n",
      "||    Features : 535393                                                       ||\n",
      "||    Meta-features : 26485                                                   ||\n",
      "||    Chromosomes/contigs : 255                                               ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/T2.m6A.bam...                                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 48438862                                                  ||\n",
      "||    Successfully assigned reads : 37822021 (78.1%)                          ||\n",
      "||    Running time : 1.29 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"feature_count/T2.m6A.fc  ||\n",
      "|| .summary\"                                                                  ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/U1.input.bam                               ||\n",
      "||                                                                            ||\n",
      "||             Output file : feature_count/U1.input.fc                        ||\n",
      "||                 Summary : feature_count/U1.input.fc.summary                ||\n",
      "||              Annotation : /rumi/shams/genomes/hg38/hg38_genes.gtf (GTF)    ||\n",
      "||      Dir for temp files : feature_count                                    ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /rumi/shams/genomes/hg38/hg38_genes.gtf ...           ||\n",
      "||    Features : 535393                                                       ||\n",
      "||    Meta-features : 26485                                                   ||\n",
      "||    Chromosomes/contigs : 255                                               ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/U1.input.bam...                                       ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 111917626                                                 ||\n",
      "||    Successfully assigned reads : 18257410 (16.3%)                          ||\n",
      "||    Running time : 3.21 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"feature_count/U1.input.  ||\n",
      "|| fc.summary\"                                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/U1.m6A.bam                                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : feature_count/U1.m6A.fc                          ||\n",
      "||                 Summary : feature_count/U1.m6A.fc.summary                  ||\n",
      "||              Annotation : /rumi/shams/genomes/hg38/hg38_genes.gtf (GTF)    ||\n",
      "||      Dir for temp files : feature_count                                    ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /rumi/shams/genomes/hg38/hg38_genes.gtf ...           ||\n",
      "||    Features : 535393                                                       ||\n",
      "||    Meta-features : 26485                                                   ||\n",
      "||    Chromosomes/contigs : 255                                               ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/U1.m6A.bam...                                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 40981072                                                  ||\n",
      "||    Successfully assigned reads : 32565060 (79.5%)                          ||\n",
      "||    Running time : 1.33 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"feature_count/U1.m6A.fc  ||\n",
      "|| .summary\"                                                                  ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/U2.input.bam                               ||\n",
      "||                                                                            ||\n",
      "||             Output file : feature_count/U2.input.fc                        ||\n",
      "||                 Summary : feature_count/U2.input.fc.summary                ||\n",
      "||              Annotation : /rumi/shams/genomes/hg38/hg38_genes.gtf (GTF)    ||\n",
      "||      Dir for temp files : feature_count                                    ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /rumi/shams/genomes/hg38/hg38_genes.gtf ...           ||\n",
      "||    Features : 535393                                                       ||\n",
      "||    Meta-features : 26485                                                   ||\n",
      "||    Chromosomes/contigs : 255                                               ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/U2.input.bam...                                       ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 140156753                                                 ||\n",
      "||    Successfully assigned reads : 22365756 (16.0%)                          ||\n",
      "||    Running time : 4.32 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"feature_count/U2.input.  ||\n",
      "|| fc.summary\"                                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/U2.m6A.bam                                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : feature_count/U2.m6A.fc                          ||\n",
      "||                 Summary : feature_count/U2.m6A.fc.summary                  ||\n",
      "||              Annotation : /rumi/shams/genomes/hg38/hg38_genes.gtf (GTF)    ||\n",
      "||      Dir for temp files : feature_count                                    ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /rumi/shams/genomes/hg38/hg38_genes.gtf ...           ||\n",
      "||    Features : 535393                                                       ||\n",
      "||    Meta-features : 26485                                                   ||\n",
      "||    Chromosomes/contigs : 255                                               ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/U2.m6A.bam...                                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 38607844                                                  ||\n",
      "||    Successfully assigned reads : 30956734 (80.2%)                          ||\n",
      "||    Running time : 1.32 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"feature_count/U2.m6A.fc  ||\n",
      "|| .summary\"                                                                  ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir -p feature_count\n",
    "for f in bam/*.bam; do \n",
    "    out=${f/.bam/.fc}; out=${out/bam/feature_count};\n",
    "    featureCounts -T 12 -O -t exon -g gene_id \\\n",
    "    -a /rumi/shams/genomes/hg38/hg38_genes.gtf -o $out $f;\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "library(DESeq2)\n",
    "library(tidyverse)\n",
    "library(ggplot2)\n",
    "require(ggrepel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "files = list.files(path = 'feature_count',pattern=\".fc$\")\n",
    "# meta \n",
    "reps <- rep(c(rep('1',2), rep('2',2)),2)\n",
    "treat <- c(rep('T',4), rep('U',4))\n",
    "ips <- rep(c('input','m6A'),4)\n",
    "\n",
    "# read files\n",
    "datalist <- lapply(files, function(x){read.table(\n",
    "    file=paste('feature_count/', x, sep = \"\"),skip=1,sep='\\t',header=TRUE,\n",
    "    colClasses=c(NA, \"NULL\", \"NULL\", \"NULL\", \"NULL\", \"NULL\", NA),check.names=F)})\n",
    "m <- Reduce(function(...) merge(..., by=1, all = TRUE), datalist)\n",
    "rownames(m) <- m[,1]\n",
    "m <- m[,-1]\n",
    "m[is.na(m)]<-0\n",
    "m <- round(m)\n",
    "colnames(m) <- paste(treat,reps,ips, sep='.') \n",
    "\n",
    "# design \n",
    "colData <- data.frame(cond = treat, rep = reps, ip = ips, row.names=colnames(m))\n",
    "dds <- DESeqDataSetFromMatrix(countData = m, colData = colData, design = ~ cond + rep + ip)\n",
    "dds$cond <- relevel(dds$cond, ref='U')\n",
    "dds <- DESeq(dds)\n",
    "# normalized counts\n",
    "ncu <- counts(dds, normalized=TRUE)\n",
    "# results \n",
    "res <- results(dds)\n",
    "vsd <- varianceStabilizingTransformation(dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nO3de1xU1d4/8LX2zDDMBR0uGigqIVikgh21UFF51DK1QE20EhAtSlBTy0rLjKNp8tRzvFS/jo8nJSsztUelMnyyIsPLUcg0IgOxEElRbgbIcJnZvz/2aZ45w3CTmb2WzOf9Oq/z2iw2e39n791nlmvfqCiKBAAA+COwLgAAAOxDQAMAcAoBDQDAKUVKSgrrGjps2LBhgiD85S9/aef8kZGRVVVV4eHhHV3RTf9he7T0Kdqz0s4UxufWAIDm5O5BG41G2oJLly45aaUPPPDAXXfd5aSFO5yzq721tgaAK1PKvD61Wn369GlpetWqVdXV1Rs2bJB+vO2225y00uXLlztpyc7g7Gpvra0B4Mrk7kFTSof8ycvLq3v37tL0wIEDV65c6efn161bt/vuu++XX36R5m9qanrxxRf9/f01Gs3gwYO//vprqf369esxMTFarTYkJOSbb76RGocNG/bqq68++OCDnp6e1u2RkZEbN25sZYFpaWlDhgzR6XR9+/Zdvnx5Y2NjKx9hw4YNoaGhlh9ramq0Wm1mZiYh5OOPPw4JCXF3d/f19U1KSmpza5SXlz/44IMajWbAgAGHDx9uXm11dfXcuXMNBoOHh8fMmTPLysqaL6SleSoqKqZOnSotPCMjg1L6ww8/OHxrAIDz8HKScOXKlVlZWfv37z979uywYcMmTZpkNBoJIS+99NKHH3743//93+fOnUtNTVUoFNL8qamp9913X05Ozrhx42JjYy0h8uabb65cubKysvKpp56ybrewu8DGxsbXX3/9559/3rFjx549e95+++1WSo2JicnNzT137pz046efftq9e/cxY8ZcvXo1Li7u5ZdfLiwszMjIuOeee9r81KmpqdHR0bm5udOmTZs+fXp5ebnNDIsXLz558uShQ4e+++67X3/9dd68ec0X0tI8ixYtKikpOX78+M6dO9esWWO3gM5vDQBwIpGdOXPmREdHi6JYU1OjVqsLCgosvwoMDMzIyKiurnZzc/v8889t/nDo0KFPPfWUNF1dXU0IycvLk9qXLVsmtdfX1wuCILWPHTt2w4YN0sx2F2jt3XffHT16tDRt+UMbo0aN+utf/ypNT506deHChaIo/vDDD25ubuXl5e38+EOHDn3kkUekabPZfMcdd0jrsqy0srJSoVB888030jxS//f8+fPtmaeiokKhUJw8eVJql7rGp0+fdsbWAAAn4aIHXVBQUF9fHxwcbDlheOHChV9//bWgoKChoWHMmDHN/yQsLEya0Ov13bp1u3btmvRjUFCQNOHm5ubh4WFpt6zI7gLz8vJmzZoVEhLi5+f37LPPFhcXt17wzJkz9+zZQwipqanJyMiYNWsWIWTQoEEjRowICgqKjY396KOP6uvr2/zgll42pfSee+6xDOxICgsLTSaTZZ6wsDCNRtPOeQoLC0VRvPvuu6X2oUOHNl+7o7YGADgJFwFtMpkIIVVVVdZfHfPnzxdbvg1dpVJZpimlZrNZmraMgUgs7RK7C2xsbLz//vt79uy5a9eunJycN954o81R15iYmLy8vJ9//jk9Pd3b23vUqFHSqr/++uu9e/f27t17+fLlo0aN6tDobfPaWvn4bc5j004pbeff3sTWAAAn4SKg77jjDjc3ty+++MKmfcCAAW5ubkeOHHHUiuwu8MKFCyUlJa+//npYWFivXr1KSkraXI6fn19ERMTu3bt3794dExNjiT9BEMaNG5eamvr999/n5OTk5eW1vpyTJ09KE6Ionjp16o477rD+bVBQkEKhOHHihPTj6dOn6+rq2jlPUFAQpdRywUx2drbztgYAOAkXAa3X65955pmnn376448/vnDhwrFjxxYtWnTx4kW9Xr9o0aKkpKRPP/30woULn3/+uXSxRGdW1HyBfn5+Go3mq6++IoScPXv2nXfeac+iZs2a9cEHHxw6dGjmzJlSy6lTp1JTU8+ePVtSUrJz506NRtO3b19CyBtvvJGcnGx3IYcOHdq6dWthYeHy5ctLSkri4+Otf2swGObMmbNw4cJjx45lZ2c/8cQTDz30UP/+/dszj6en56xZs5KTk3/44Yfs7OyVK1eSZv1oB24NAHAGLgKaELJmzZrk5OTnn38+JCTk0UcfvXHjhsFgIISsW7du1qxZiYmJAwcOXLFiRedX1HyB3bp1e++99xYtWtSnT5+FCxcuW7asPcuZMWPGr7/+2qNHD8uddd26dTt8+PC4ceOCg4PT0tL27dvn6elJCDl//vzZs2ftLuT555/ft2/fwIED9+3b9z//8z9eXl42M2zcuPHee++dPHlyZGRk//79t23b1nwhLc2zefPmXr16hYeHP/roo8899xwhRK1WO2lrAIAz0PYMdMKt7sSJE6NHj75+/bpWq2VdCwC0l9x3EoJssrKyampqwsLCioqKkpOTp0+fjnQGuLUgoLsso9G4ZMmS3377zdvbe8qUKf/1X//FuiIA6BgMcQAAcIqXk4QAAGADAQ0AwCkENAAApxDQAACcQkADAHCKwWV2lZWV0tORJIIgeHp6Nn8Uspwope7u7nV1dQxr0Ov1ZrP5xo0bDGvQ6XS1tbUMC6CUent7230vgZw1aDQa5jtCFEXmNfBwMEjP7219Th8fH3lKkh960AAAnEJAAwBwCgENAMApBDQAAKcQ0AAAnEJAAwBwCgENAMApBDQAAKcQ0AAAnEJAAwBwCgENAOztqKhiXQKPENAAAJxCQAMAY1L3GZ3o5jr7NLuKior169erVKqGhoannnoqKCiIEJKenp6TkyOKYlJSkp+fnyPqBABwOZ3tQRsMhtTU1LVr18bHx+/du5cQUl5enpWVlZKSEhcXl5aW5oAaAaDrsu44oxNto7MBLQgCpZQQUldXFxgYSAg5d+7coEGDKKXBwcFFRUUOqBEAuqjmiYyMtuaAB/ZfunTpnXfe+f3331944QVCSE1NjVarlX5lNpsts23cuDE9PZ0Qsm3btn79+lkvQXoyd+cr6SRL2UxQSkVR1Gg0DGsghLi7u7MtgBDCw8HAdkdInR5XOBh09Y3NG60PAC8vr9aXYP32j66Htvm2gna6cOHC5s2bN27cePTo0cLCwvj4eELI/Pnz//73v0szVFVVSS9ocHd3F4T/67kLgtCtW7eqKpZfm5RSNze3+vp6hjVotVqz2Ww0GhnWoNFo2L5WhlJqMBgqKyvZ1qBWq5nvCEII230hw8HwXnmLO3qOt6d0MFRVVbWeUaIothnit67O9qAbGxtVKhUhRK/Xq9VqQkhISMiBAwdEUSwoKAgICLDMaTAYDAYDafbKK2nrs/0alHqvbGsQRZGHGpjvCIKDwWUOBut/YdswmUyWg8FRnchbUWcDOj8/f+fOnYIgmEymxMREQoiXl1dERMSqVasIIcnJyQ6oEQC6nNbHmndUVM3x9pStGG45bIij/fDSWLvw0liCl8b+yRVeGtvmyUBK6dLg/nhpLACArHCpRjs54CoOAIB2SkpK+vrrr/38/DIzM1ufUxqDdnHoQQOAfOLi4j744APWVdwyENAAIJ+RI0dKV3NBeyCgAQA4hYAGAOAUAhoAgFMIaAAATiGgAUA+CQkJ0dHR+fn5oaGhe/bsYV0O73AdNADIB8+I7xD0oAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTDN5J6O7ubv0jpZRSqtPp5K/EugaFQiEILL+ulEqlKIpst4NKpWK+IwghbGsghCiVSuY7gvy5NRjWwHxHEEK0Wm3rM5jNZnkqYYJBQBuNRpPJZPlREAS1Wl1bWyt/JRaUUnd397q6OoY16PV6s9l848YNhjXodDoedgTzGjQaDfMdIYoi8xo42RGiKLY+Jw9fJE6CIQ4AAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE4pO/n3hYWFW7duVSgUhJDFixf37NmTEJKenp6TkyOKYlJSkp+fnwPKBAAXsKOiihAS72VgXQgvOtuD9vHxSUlJWbt27aRJk/bu3UsIKS8vz8rKSklJiYuLS0tLc0CNAAAuqbM96O7du0sTlFKpH33u3LlBgwZRSoODg4uKiixz/vbbb1evXiWEBAYGqlQqS7sgCJRS6xb5ScWzrUEQBEII8xqY7wjCeiPgYLDUIHMBaWUVUoZ8eL06wcdLOhiUyjYyShRFOYpjpLMBLamtrd2zZ8/y5csJITU1NVqtVmo3m82WeTIzM7/66itCyLp16wwG23/C6HQ6h1Ry0wRBaPNQcHYBpB2Ho7NrkP4LYYuHg4HtdpAOBuY1yFyAurrWMv1Rde0Tvj1JOw4Gk8nk3LKYckAcNDY2rl+/fvbs2b6+voQQvV5fWloq/Uo6ziQJCQkJCQmEkMrKyqqqKku7IAienp7WLfKjlLq7u9fV1TGsQa/Xm83mGzduMKxBp9PV1ta2PZ/TUEq9vb2ZHwwajYb5jhBFkXkNch4M0uiztevXr3t7e1+/fr3NPrJarXZaXYx1dgxaFMUNGzaMHTt2+PDhUktISEhubq4oivn5+QEBAZ0tEAC6uubp3FKjq+lsD/rYsWM5OTnV1dXffvttcHBwfHy8l5dXRETEqlWrCCHJycmOKBIAwBVR+YfYKysrrYeNpCGO8vJymcuwhiEOCSdDHGVlZWxrwBAHkfFgaKmnTCnV6XQPq1VtZpSPj48T6uICblQBAOAUAhoAmGlzoNnFR6IR0ADANVfOaJZX3QKACyopKVm6dOnPP//s5ua2ePHi+Ph4u7NJJyTKy8u79q0orUNAA4CsRFFcsmTJiBEjSktLJ0yYMGbMGFyP2xIMcQCArPz9/UeOHEkp9fX1DQwMvHLlCuuK+IWABgA2CgoKiouLw8LCWBfCLwQ0ADBQVVWVmJi4adMmjUbDuhZ+IaABQG5GozE2NnbBggVjxoxhXQvXENAAIKumpqZ58+ZNmjQpJiaGdS28Q0ADgKyOHz9++PDhLVu2hIaGhoaGZmRksK6IX7jMDgBkNXr0aOndHdAm9KABADiFgAYA4BQCGgCAUwhoAABOIaABADiFgAYA4BQCGgCAUwhoAABOIaABADiFgAYA4BQCGgCAUwhoAABOIaABADiFgAYA4BQCGgCAUwhoAABOMXhgvyAIzX9UKBTyV2JBKRUEgXkNlFLmNTAvgHBwMPCwHXiogXkBhBCFQiGKYiuztf7bWx2DgFYqldbbVDoWVSqV/JVY16BQKNjWIAgC8+0gCALzHUEIYV4D84NBSiUXPxgkSmUbGYWAdrCGhgaTyWT5URAErVZrNBrlr8RCygW2NSiVSrPZzLYGhULBfEfodDrmNVBKme8IURSZ18B8R+j1+vr6+jYj2MPDQ56S5IcxaAAATiGgAQA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFPKTv69yWRasWLFxYsXExMTx48fLzWmp6fn5OSIopiUlOTn59fpIgEAXFFne9CCICxfvnzatGmWlvLy8qysrJSUlLi4uLS0tE4uHwDAZXU2oCmlXl5e1i3nzp0bNGgQpTQ4OLioqKiTywcAcFmdHeJorqamRqvVStNms9nSnpGR8f333xNC5s6d6+npaWmnlFJK9Xq9wyvpEKVSqVAoGBagUqlEURQElmcFlEol2x1BKSWE8HAwMN8RhBDmNTDfEYQQnU7X+gzWIdN+q1evXrlyZSe3sEMW0jrHB7Rery8tLZWmrUs3GAy9e/eWGk0mk6Vd+m/SukV+lFKz2cy2BoVCIYoi8xqY7wjCwcHAw44grLcD84NB0mYNoijexGJXr1794osv2mRrU1OT9NXYmYU4luMDOiQk5MCBA6IoFhQUBAQEWNrDw8PDw8MJIZWVlXV1dZZ2QRA0Go11i/wope7u7mxrUCgUZrOZbQ2CIDDfEVqtlnkNhBDmO0IUReY1MN8ROp3OaDS2GcEd7emvWLHCZDKNHz+eUvrFF19otdq1a9eePn06MjJywYIFx44dW7FiRU1NjUKhWL9+/bhx4wghsbGx586da2xsDAwM3LZtm6enp81CNBrNzX/UltGb+/6xlpqaWlhYqFKpwsLCnnzySUJIenr6qVOnCCHJycnNr+KorKy0/lYUBMHT07O8vLyTZXQGDwGt1+vNZvONGzcY1qDT6WpraxkWQCn19vYuKytjW4NGo2G+I0RRZF4DDwdDeXl5mxnl4+PT0YUrlUqj0Sj1lymlH3744WOPPUYIqaioiIyM/N///V9fX9/i4uKIiIj8/Hy1Wl1WViat5bXXXquurl63bp3NQpzEAYt+4YUXbFqioqKioqI6v2QAABlMnTpVmsjKyrpy5cojjzwi/ahWq4uLi4OCgvbu3btz505RFKurq/39/WUrzInZDwBwS7C+ruHOO+/MzMy0/m12dvaGDRtOnDjh6emZnp6+efNm2QrDnYQA4HK6d+9eVVXVvD0iIiIvL+/QoUPSjydOnCCEVFRU+Pn5Sdee7dq1q82FOBACGgBcztKlSyMiIoYMGWJz5snHx+fTTz9ds2bNXXfdNWDAgL/97W+EkPHjx/fq1WvSpEmPPfZYr1692lyIAzngJGFH4SShXThJSHCS8E84SUicfJLwVoEeNAAAp1oM6Pr6+vPnz8tZCgAAWLMf0IcPHx4wYMDkyZMJIcePH7dcdAIAALKxH9ArVqw4evSoNBw+YsSIM2fOyFsVAAC0ENBms9n6Ymzp5lcAAJCT/RtVtFqt5YFHBw8e9Pb2lrEkAADn6ujVa6w6qfYD+rXXXpswYUJpaemYMWPy8vI+++wzmcsCAHASs9nc0SsIPTw8nFRM6+wHdERERGZm5tGjR81mc0RERBe+zBAAgFv2x6AzMzMVCkVUVNTUqVMVCsWRI0dkLgsAAOwH9KJFiyxder1ev3DhQhlLAgAAQloKaEEQLO9/UqlUTU1NMpYEAACEtBTQWq02Oztbmj527Fi3bt1kLAkAQHaiSGprSPseyNPQ0BAZGRkZGWkwGMLDwyMjI9PT051RlP2ThKmpqVOmTAkLCyOEnDlz5pNPPnHGugEAuCCKwq/nSdGv1GwSB4SY+wS0Prubm5v0zOjIyMi0tDTrd/s5lv2AHjNmTG5u7tGjRymlo0aNwlUcANCV/XGdXPyNdDeIhNBf8shtfsRNzbomQlp5o0qPHj2mTJkiPRfUaDS6u7vLWBUAgHwopZQK4p/TRO5nMLfI/hj0/v37+/XrZzAYfP4kc1kAALIR9R6ifx/6x3V6vdLU/w6i5qL7TFrqQb/66quZmZm33367zNUAADAgCObAYNrTlwiCqNOzrub/2A/o4OBgpDMAuBBKRQ/uLlezP8QRGhp64MCBhoYGmasBALiFZGZmOu8SDtJSD/rFF1+0aZH/1YUAAC7OfkAjjgEAmMNLYwEAOGW/B11ZWfn666+fPXvWaDRKLYcPH5axKgAAZxEEgdXznTvKfkDPmzfvnnvuuXTp0urVq3ft2tW7d2+ZywIAcBKz2WzperaTVqt1UjGtsz/EUVhYuGLFCg8Pj6ioqJ07d165ckXmsgAAnMfUQazqtN+DdnNzI4RQSisrKz09PcvKyhy4SrVabX0SklJKKdVoNA5cRUdRSpXKFu96l4dCoRAEge12UCqVzHcEIYRtDYSD7SAdjcxrYL4jCCFtPmSia1/RYD+VBgwYUFZW9thjj4WHh/v5+Vm/4bvzGhoarL+RpFQqLCxcsmTJzz//rFarFy9eHB8f78A1tolS6u7u3tF/9TiWQqG4iX95OZYgCGwLoJRqtVrmNVBKmR8MoijiYNDpdPX19W1GsF7P0b1/jmU/oHfu3EkImT9//rBhwyorK8eNG+fAVYqiaL3FpWmz2bxkyZIRI0aUlpZOmDBh9OjRTr38u82qWGFeA/MCUAP582hkvh2YF0C4+Q+TFdsxaKPRKH11SwYNGjR69OjGxkZn1+Hv7z9y5EhKqa+vb2BgIEa9AYBnWVlZsbGx0nR2dvaMGTOcsRbbHrTBYDhy5MiYMWNs2mX7x05BQUFxcbH0rgAAAHmcqb3xfnkFITT5th6BajfW5fyLbUBLQcxq7KmqqioxMXHTpk08nJ0AABdR1tT0SdX1gRoNIeTvpdfW+PdSC5R1UYTYvcyuqanpnXfekb8Uo9EYGxu7YMGC5v13AADnqTGZdcK/wlArCDfMzK6rs2EnoJVK5ZdffilzHU1NTfPmzZs0aVJMTIzMqwYAF3ebSnmtsbHWbP7DZBrhoeuuaPuiW41GYxlpMBqNTvpHv/067r777p07d06dOlW2+2eOHz9++PDh3NzcLVu2EEL+8z//84EHHthRUWUzW7yXQZ56AMB1aARhVe9e+UajktI73Ns1vBEUFPTTTz/V19er1eqsrCwnnTazH9CrVq2yaXH2lS6jR4++evWqdYt1Op+ovWGZ+H99ejm1EgBwQd0UwjBdB/qj3bt3f+mllyZMmKDVag0Gw7Zt25xRFaePG7WksyWaLZKLfw/XadGVBgC2YmNjLVfaOcmt+rjR5qMfAABdDNePG23efQYAcB32e9Dz5s3z8PC4dOnS008/3bNnz7vvvlvmsgAAwH4PurCwcN++fQcPHoyKioqKioqLi5O5LAAAJ5Eejsa6inZh8LjRTgrvyJlWAAAblFKVSsW6inaxP8Rh/bjRyMhIb29vmcvCRRoAAAweN9pO8V6GeC9DcvHvlhbrvjMSHAC6PPsBXVtbq9PpCCHDhg2Ttx5b/69Pr5buJ0xKSvr666/9/PwyMzMZVAYA4GT2A7pfv35RUVFz584dPXq0zAU111JnOS4ubt68ec8995zM9QAAyMP+GPQvv/xy9913L1myJCgo6NVXX7148aLMZbXHyJEjDQYMdABAl2U/oL29vRctWpSTk7N///7z58/ffvvtMpcFAAAtPlWvqanp4MGD77333pEjRxISEmQsCQAACGkpoJcuXfrRRx8NHDgwISHh/fffl+2howAAYGE/oL29vU+ePNm3b1+ZqwEAAAv7Y9ArV67kP50TEhKio6Pz8/NDQ0P37NnDuhwAAAdr+80u3EpLS2NdAgCAE92qz4MGAOjyENAAAJxqLaB//PHH+++/f+TIkYcOHZKtIAAAkNiOQTc2NloexLd69erNmzd369Zt9uzZEydOlL02AACXZtuDnjBhwtGjR6VpURRVKpVS2eETienp6a+88sqqVasuX77sgBoBAFySbUB/8sknW7duTUpKqqqqevnll5944okHH3zw+eefb/8Sy8vLs7KyUlJS4uLicKEFAMBNs+0d+/j4pKWlffXVVw899NCiRYu++eabji7x3LlzgwYNopQGBwcXFRU5qE4AAJdjG9Bms/no0aOenp5ffvnlmjVrdu7cuWnTpn79+rV/iTU1NZZbw81ms6V948aN6enphJBt27bZLJBSKv9LW5pje0c7pVQURY1Gw7AGQggP72rj4WBguyMopcxrIHwcDF5eXq3PYDKZ5KmECduAnjNnjiAIVVVVw4cPX7t27U8//ZSYmDhx4sRnn322nUvU6/WlpaXStCD83xBKQkJCTEwMIcTd3b2q6v+ewS8IQrdu3axb5EcpdXNzq6+vZ1iDVqs1m81Go5FhDRqNpq6ujmEBlFKDwcD8YFCr1cx3BCGE7b7g5GC4fv26KIqtzCaKYpshfuuyDWij0SjdNi2F6cCBAw8dOrRly5b2LzEkJOTAgQOiKBYUFAQEBFjaDQaD9PjmyspK6y89aeuz/RqUeq9saxBFkYcamO8IgoMBBwMhxOpgaD2guzbbgDaZTMuWLbt+/bplFIJSOn/+/PYv0cvLKyIiYtWqVYSQ5ORkRxUKAOBqbAP6o48++vzzz7VabWcufI6KioqKiupcYQAAziINtPIwyN4624BWq9XTp0+3/FhWVubt7S39WwMAoGs4cuSIWq1mXUXbbK+DzsnJCQsLGzJkyNmzZydOnNizZ88+ffr88MMPTIoDAHCGMWPGSBcFUEpXrVoVFxcXERGxf/9+1nXZsg3ohQsXPv300/Pnz588efK0adPq6+tXrFixbNkyJsW1qaSkZObMmYMHDx46dOiOHTtYlwMAt54777zz/fffP3DgwKJFi65evcq6nH9Dbc6Q+vv7X7p0iRDSo0ePq1evSme0+/TpIzU6hM1VHIIgeHp6lpeX38SiLl26dPHixREjRpSWlk6YMOGzzz6zvm6k/Sil7u7ubC8q0uv1ZrP5xo0bDGvQ6XS1tbUMC5CuiC8rK2Nbg0ajYb4jRFFkXgMPB0N5eXmbV3H4+Ph0dOGWMWhKaXV1tV6vJ4RER0c//vjjXJ0/s+1BW4abx48fL01LGS13Xe3j7+8/cuRISqmvr29gYOCVK1dYVwQAt5jGxkbLBG/n22wDevDgwcXFxYSQXbt2SS25ubkdupPQGVRnclRnctw//x9pQvqf9QwFBQXFxcVhYWGsKgSAW9S7775LCCkoKDh16lR4eDjrcv6N7VUcBw8etGnx9/f/5JNP5KrHliWIFZeKLP9v8u8n/aoxbCghpKqqKjExcdOmTcxvjQWAW9GIESMqKyu3bt3aozb2JLQAABjWSURBVEcP1rX8GwbDF+0fg7ZJZ2tSRhNCqu8YOGPGjDlz5ki3Pt4cjEFLOBl2xBg0xqCJk8egrdfC7RAuuSVeedU8nS2aTKZ58+ZNmjSpM+kMAMCnWyCg7ZJS+9jZHw8fPrxly5bQ0NDQ0NCMjAzWdQHArYTn7jNpPgbNVlJS0tdff+3n55eZmWlzGtCuMXcPqfzyC2kkGgCgi+GrBx0XF/fBBx9I0+2MXaQzAHRV9gO6oaHhrbfeKikpIYQUFha+9dZb8lQzcuRI6ZGk1iznA9tsBADoSuwHdGNjY0ZGRkVFBSHkypUrGNsFgK7E3EGs6rQ/Bq3T6T777DNpetSoUZZpmTWGDZVGok3+/ayv5ZC6zxjcAICbYDabO3oFoYeHh5OKaR1fJwmbk1JYdSYHoQwArsZ2iKOkpOThhx8ePnx4amqq5XaS4cOHy17Yv2kMGyr9j20ZAABysg3o+fPn9+7d+6WXXjp8+PDMmTOlx4hcvnxZnmoSEhKio6Pz8/NDQ0OlVyMCALgs24A+ffr0xo0bp06deujQodtuuy0mJsbyqCcZpKWl5eXl/f7772fPnsXNgQDArdzc3ISEhJv72ytXrqSnp7dnTtuAtpyyFATh7bff7tWr14wZM5qamm6uDgCALmngwIFbt269ub+9+YAePnz4559/Lk1TSt9++21/f//S0tKbqwMA4Faxffv27du3t3Pmn376KTExMTs7e8yYMcuWLXvooYekN2ZlZ2dHREQsXbr04Ycf3rdvHyEkMzNz+fLlhJCmpibpcaYffPDBkSNHEhISCgoKWl+L7VUc27dvt77oT8rop556qt2fEQDg1mOJ5u3bt8+dO7f9f6hWq994443q6uqHHnpo6tSphBCz2bxhwwaTyTR8+HCpxUZsbGxVVdU//vGPNhdu24P28PBQKv8ttauqqkJCQtpfLgCA6+jfvz8hxMPDw3Jt9R133EEIUSgU3bt3r6io6MxbWmwDeuXKlVu2bLFuefPNN9esWXPTKwAA4J+l19yh7rM1y4PxTp8+bTab6+rqrl275uXlZXnevWVAw83NrZ0n9myHOPbt23fmzBnrlsWLFw8dOnT16tU3VzQAwC3hpqPZho+PT2xs7KVLl9asWUMpHTx48I0bNxYsWGB5sUBQUNClS5diYmLWrFlz5513trIo27cJBAQE/PbbbzYzWV717RAOfKu3o+CNKhJOXqKBN6rgjSrEmW9Uceqt3tnZ2f/4xz/+/ve/d2j5LbHzVm+b/zyuXLmiUqkcsjIAAGg/24B+7LHH5s6da8noq1evJiQkxMXFyV4YAMCtZ9iwYY7qPpPmAf3KK6/o9fo+ffoMGjRo4MCBffv29fHxefnllx21PgAAaCf7b7QtKCj4/vvvKaV33313cHCwY1dpM9TLw/gvpVShULC9YVKlUomiyLwGOe/sb46Hg4EQolQqme8IQgjbfcH8YCCEaDSaNg8Gs9ms0+k6tNhb+3GjZWVlVVVV999/v6enpzNWaTQabU4SqtVq5qcjmOcCThKSP3cE8xpwkpDwcTBIO6LNk4QdDWhBEFgFbkfZBvSuXbsSEhK0Wm19ff3u3bunTJnCpCwAACeRLlLu0J909DvAUWzHoFNSUjIzMysqKnbv3v3Xv/6VSU0AAE51q7zyyjagLY/zmDJlytWrV1mUBAAAhDQf4mhoaMjNzZWmGxsbLdODBg2StS4AAJdnG9A3btyIjIy0/GiZZntzFwCAC7INaAQxAEAnnThxYuXKlYSQCRMmSA+DJoTMmjXLbDZ36GV+vL/VGwBAHtIjoTv/yCSj0fjss89mZGRYX8xXU1Nz7do1s9n8xx9/dOvWrZ2Lsj1JCAAAbcrOzr7vvvsSEhKGDRu2f//+2NjYESNGHDhwgBCSlZVlMBieeOKJ6Ohoy8NB9+3b9/DDD8fExEgvXmkn9KABAG7mjSo3btxIS0v74YcfHn744V9++aW6unratGnR0dGXL1/Oy8v78ccfr1y5MmvWrJycHELI3r173333XUEQ5s6dGx8f386qENAA4Ora/ypCa4MHDyaE+Pr6hoSEKJVKT0/PmpoaQoiXl9ewYcP0en1QUFB1dbXJZKqsrMzOzo6NjSWE5ObmXrt2rUePHu1ZBQIaAFydox7VLxkxYsQrr7xiNpurqqrc3NwUCsXu3btXr179+OOPE0Lee++9vXv3JiUltWdRGIMGAHAkLy+vBQsWTJw4cerUqZs2bSKEfPTRR/fdd5/024kTJ+7ataudi7L/NDunwhtV7MLDkgjeqPInPCyJ3LJvVHEs9KABADiFgAYA4BQCGgCAU7iKAwBci3TOiXUV7YKABgDXQimVXirGPwxxAABwCgENAMApBDQAAKcQ0AAAnEJAAwBwCgENAMApBDQAAKcQ0AAAnEJAAwBwCgENAMApBDQAAKc6+ywOk8m0YsWKixcvJiYmjh8/XmpMT0/PyckRRTEpKcnPz6/TRQIAuKLO9qAFQVi+fPm0adMsLeXl5VlZWSkpKXFxcWlpaZ1cPgCAy+psD5pS6uXlZd1y7ty5QYMGUUqDg4OLioos7b/99tvVq1cJIYGBgdaPkhIEgfnDpSilCoWCbQ2CIBBCmNfAfEcQ1hsBB4OlBuY7ghCiVLaRUfK/tE9Ojn/caE1NjVarlabNZrOlPTMz86uvviKErFu3zmAw2PyVTqdzeCUdIghCm4eCswsg7TgcnV2DQqFgWICEh4OB7XaQDgbmNdwSB4P1C067npuMg4MHD548eTIgICAhIcHmV3q9vrS0VJqWjjNJQkKCNHNlZWVVVZWlXXpprHWL/PDSWAkn7wllfjDgpbGEm4Ph+vXrbfaR1Wq1PCXJ7yYDevLkyZMnT7b7q5CQkAMHDoiiWFBQEBAQcPOlAQC4Ngf8gzo1NbWwsFClUhUWFj755JNeXl4RERGrVq0ihCQnJ3d++QAArskBAf3CCy/YtERFRUVFRXV+yQAArgw3qgAAcAoBDQDAKQQ0AACnENAAAJxCQAMAcAoBDQDAKQQ0AACnENAAAJxCQAMAcAoBDQDAKQQ0AACnENAAAJxCQAMAcAoBDQDAKQQ0AACnENAAAJxCQAMAcAoBDQDAKQQ0AACnENAAAJxCQAMAcAoBDQDAKQQ0AACnENAAAJxCQAMAcAoBDQDAKQQ0AACnENAAAJxSyr9KQRCa/6hQKOSvxIJSKggC8xoopcxrYF4A4eBg4GE78FAD8wIIIQqFQhTFVmZr/be3OgYBrVQqrbepdCyqVCr5K7GuQaFQsK1BEATm20EQBOY7ghDCvAbmB4OUSi5+MEiUyjYyCgHtYA0NDSaTyfKjIAhardZoNMpfiYWUC2xrUCqVZrOZbQ0KhYL5jtDpdMxroJQy3xGiKDKvgfmO0Ov19fX1bUawh4eHPCXJD2PQAACcQkADAHAKAQ0AwCkENAAApxDQAACcQkADAHAKAQ0AwCkENAAApxDQAACcQkADAHAKAQ0AwCkENAAApxDQAACcQkADAHAKAQ0AwCkENAAApxDQAACcQkADAHAKAQ0AwCkENAAApxDQAACcQkADAHAKAQ0AwCkENAAApxDQAACcQkADAHAKAQ0AwCkENAAAp5Sd/PvCwsKtW7cqFApCyOLFi3v27EkISU9Pz8nJEUUxKSnJz8/PAWUCALiezvagfXx8UlJS1q5dO2nSpL179xJCysvLs7KyUlJS4uLi0tLSHFAjAIBL6mwPunv37tIEpVTqR587d27QoEGU0uDg4KKios4WCADgqjob0JLa2to9e/YsX76cEFJTU6PVaqV2s9lsmScjI+P7778nhMydO9fT09PSTimllOr1eodUctOUSqX0BcOKSqUSRVEQWJ4VUCqVbHcEpZQQwsPBwHxHEEKY18B8RxBCdDpd6zNYh0zXc5MBffDgwZMnTwYEBCQkJDQ2Nq5fv3727Nm+vr6EEL1eX1paKs1mfYQZDIbevXtLjSaTydIu/Tdp3SI/SqnZbGZbg0KhEEWReQ3MdwTh4GDgYUcQ1tuB+cEgabMGURTlqYSJmwzoyZMnT548mRAiiuKGDRvGjh07fPhw6VchISEHDhwQRbGgoCAgIMDyJ+Hh4eHh4YSQysrKuro6S7sgCBqNxrpFfpRSd3d3tjUoFAqz2cy2BkEQmO8IrVbLvAZCCPMdIYoi8xqY7widTmc0GtuMYB56+k7S2SGOY8eO5eTkVFdXf/vtt8HBwfHx8V5eXhEREatWrSKEJCcnO6JIAABX1NmAHjVq1KhRo2wao6KioqKiOrlkAAAXhxtVAAA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFMIaAAATiGgAQA4hYAGAOAUAhoAgFMIaADoMNWZHNWZHNZVdH0IaAAATiGgAaBjLH1ndKKdDQENADcPGe1UCGgA6AAkspwQ0ADQXnbTGZHtPAhoAABOIaABoF1a6SmjE+0kCGgAAE4hoAGgbW32kdGJdgYENAA4BjLa4ZSsCwAATpWUlCxduvTnn392c3NbvHhxfHw864pcDgIaAOwTRXHJkiUjRowoLS2dMGHCmDFjAgICWBflWjDEAQD2+fv7jxw5klLq6+sbGBh45coV1hW5HAY9aLVaLYqi5UdKKaVUo9HIX4l1DUol439MKBQKQRDYbgelUsl8RxBC2NZAONgO0tHIvAZLAfn5+ZcuXbr33nvlL8nd3b31GazDpOthkEoNDQ0mk8nyo5RKRqNR/kosKKXu7u5sa1AoFGazmW0NgiAw3xFarZZ5DZRS5geDKIqcHAxVVVVxcXGbNm2S+fCglOp0uvr6+jYjWK/Xy1OS/BgEtCiK1ltcmmb+NWhTFcMyXLwA1ED+PBqZbwfpS2L27NkLFiwYPXo0k3p42A4MYQwaAOxramqaN2/epEmTYmJiWNfiohDQAGDf8ePHDx8+vGXLltDQ0NDQ0IyMDDnXPn/+fB8fn7Fjx8q5Ut7gMjsAsG/06NFXr15ltfb4+PhnnnkmMTGRVQE8QA8aAHg0cuRIT09P1lUwhoAGAOAUAhoAuIA3hTeHgAYA4BQCGgDYw5vC7UJAAwCP5syZExkZmZ+fHxoaumfPHtblsIHL7ACAMZtes+pMTmPY0Pfee8/b27u8vBx3EgIAcAQDHRIENACwhCxuBQIaAJjBm8Jbh4AGAOAUAhoA2Gizj2z651F5KuEWAhoAgFMIaABgoJ1DzC4+Eo3roAGAgcawoa3PQClVeHs3lpcTXAcNAAC8QUADAHAKAQ0AwCkENAAApxDQAACcQkADAHAKAQ0AwCkENAAApxDQAACcQkADAHCqs7d6V1RUrF+/XqVSNTQ0PPXUU0FBQYSQ9PT0nJwcURSTkpL8/PwcUScAgMvpbA/aYDCkpqauXbs2Pj5+7969hJDy8vKsrKyUlJS4uLi0tDQH1AgA4JI624MWhH9FfF1dXWBgICHk3LlzgwYNopQGBwcXFRV1tkAAAFflgKfZXbp06Z133vn9999feOEFQkhNTY1Wq5V+ZTabLbNt3LgxPT2dELJt27Z+/fpZ2s1m8+XLl3kYCbGUzcT169cFQfD29mZYAyHE3d2d4drNZnNJSUmvXr0Y1iDRaDQM1379+nVKqYsfDKIolpSU+Pn5UUpbmc1kMslWkvxuMqAPHjx48uTJgICAhIQEf3//tWvXXrhwYfPmzRs3btTr9aWlpdJslv41ISQhISEmJoYQctttt1lv8YqKihkzZhw7dqwTn6Ir2LJli4+Pz+OPP866EJZqa2ujo6Ozs7NZF8LY9u3b3d3dk5KSWBfCUn19fXR09LFjx9zc3FqZTansys9MvsnPNnny5MmTJxNCGhsbVSoVIUSv16vVakJISEjIgQMHRFEsKCgICAiw/InBYDAYDA4oGQDANXT2yyc/P3/nzp2CIJhMpsTEREKIl5dXRETEqlWrCCHJycltLsHd3X3q1KmdLKMLCAsL8/DwYF0FYyqVavr06ayrYG/w4MFSv8eVKRSK6dOnW/8r3AVR0YXfVgAAwDOX/nYCAOCZ3OPrJpNpxYoVFy9eTExMHD9+vNTY/MYWF7nVxWg0JiQkSJcnzpw5c8iQIcRlPntzLvvBiWsfCciE1ojyMpvN5eXlu3btOnz4sNRSVlb23HPPmc3m/Pz8devW2W3pqurq6lasWGHd4jqf3YbLfnCJKx8JyIRWyD3EQSn18vKybml+Y4tL3epSVFS0fPnyjRs31tTUEBf77NZc9oNbuOyRgExoBfsx6OY3trR0q0vXo1art2zZsn79+sDAwPfff5+40me34bIfXIIjwZorZ4INmQL64MGDKSkpdh/Nodfrb9y48a9qBMFuSxdj2RqUUr1eTwgZO3ZsYWEhcYHP3hKX/eASHAnWXDATWiLTSULLjS3NNb+xpaVbXboMy9YwGo1qtZpSmpubK5336PKfvSUu+8ElOBKsuWAmtITBddCpqamFhYUqlSosLOzJJ58khKSnp586dYoQkpycbDlja9PSJeXm5ko39QqCsGjRop49exKX+ezNuewHJy5/JCATWoIbVQAAOOVCozkAALcWBDQAAKcQ0AAAnEJAAwBwCgHt0nx8fPr27RsUFBQSErJr1y6psbGxcdWqVYGBgQEBAQMGDFi+fHljYyMhZNmyZbfffjul9MSJE20u+ZNPPgkNDQ0JCRk3blxxcbHU+Msvv9x7772BgYHh4eEFBQWEkPr6+mnTpsXExMTHx0s3IJSVlY0dO1Zao6O4u7sbjUYHLhBAJsxuMgcOeHt7//jjj6IoHj16VK1WX7t2TRTF2bNnT5o06erVq6Io1tTUrFu3rrKyUhTF77777uLFi/379z9+/Hjriy0uLjYYDAUFBaIobt++PTo6WmqPiIh49913RVF89913IyMjRVH85JNPnnnmGVEU4+Pjjxw5IorinDlzsrKyHPUBGxsbRVH85z//aTabHbVMANkgoF2aJaBFUfT09Dx58mR+fr67u7uUznbZBHRNTU19fb3NPF9++WVoaKg0XVpaKgjCH3/8UVxcrNfrGxoaRFFsaGjQ6/WXL1/+4osvYmNjRVGMjo7Oyck5fPjw/Pnz7a539erVS5YskabLysq8vLxqampEUZw9e/bQoUNDQ0OnTp1aUVEhzUAIWbt27YwZM9566y1RFNVqdV1dXSszr1+//rHHHhs2bNj+/fulxm+//TY8PDw0NDQsLCwzM1MUxaNHj44ZM+Yvf/nL8OHDv/rqq3ZuYYDOQEC7NEtAf/HFF1qttqKi4uOPPx44cGArf2IT0FOmTHnzzTdt5vn999+7d+8uPRzyb3/7GyHkp59+ysrKGjBggGWeAQMGSB3bZ599dsaMGatXr66rq4uIiJB6680VFRX5+vpKPeLNmzfPnTtXapd6/aIorlu3zvJAOELIhx9+aPlbS0C3NLOUyxcuXOjbt68oilevXu3Ro8eJEydEUWxqaqqsrCwvLx88ePDly5dFUbx48WLfvn2NRmMrWwnAIbry+xahPSZOnKhSqXr06LF7925PT0/psGj/n7/22ms2jyIjhPj5+X344YcLFy5saGh44IEHunXr1vwFTtJaKKVvvPGG1PLSSy8tWrSoqKgoMTFRFMVXXnll8ODBlvn79u171113HTx4MCoqKi0tbePGjVL73r17d+7cKYpidXW1v7+/ZX6771FraeZJkyYRQm6//faysrLGxsajR48OGTLk3nvvJYQoFAqDwZCenn7lypVHHnlEml+tVhcXFwcFBbV/QwHcDHbfDcCe9RCHJD8/X61Wt3+Io03nz5/XaDQNDQ0XL15sPsRhmS03N1caqo6MjDx//nxeXt59991ns6gdO3ZMnz79xx9/7N+/v9Ry6tSpAQMGSIMVBw4cGD9+vNRuc2BLPej2zKzT6erq6vbt22ez9n379o0ePbr9nxrAIXAVB/yb4ODgadOmzZkzp7S0lBBSW1u7du3aqqqqluY/evSo9PQ1G+fPnyeE1NTULF269Omnn1apVH369AkNDd2xYwchZMeOHUOHDvX19ZVmFkVx6dKlUqe4tra2W7duBoNBeiyytYcffvjIkSNvvPFGQkKC1FJRUeHn5+fp6UkIsVyF0pL2zxwREXHmzJl//vOfhBBpiCMiIiIvL+/QoUPSDO25jgXAAVh/QwBLzXvQoijW19e/+OKL/fr16927d1BQ0MqVK6Vu76JFi3r37q1QKHr06NGvXz9pZrtj0KIoxsXF+fn59ezZc+HChZaziHl5ecOHD+/Tp8/w4cPPnTtnmfmdd955/fXXpenPP//8P/7jPyIjIzMyMpov9vHHHxcEoaioSPqxqanp0UcffeCBBx599NFnn3229R50e2aWetCiKH733Xf33HNPSEjI4MGDpZOEx44dGzVqVEhISHBwcExMTJvbFqDz8LAkAABOYYgDAIBTCGgAAE4hoAEAOIWABgDgFAIaAIBTCGgAAE79f1jTg3OxqxkVAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# PCA Plot \n",
    "z <- plotPCA(vsd,intgroup=c(\"cond\"), returnData=TRUE)\n",
    "z$rep <- reps\n",
    "z$treat <- treat\n",
    "z$ip <- ips\n",
    "\n",
    "percentVar <- round(100 * attr(z, \"percentVar\"))\n",
    "pca <- ggplot(z, aes(PC1, PC2)) +\n",
    "        ggtitle('Technical vs. biological') + \n",
    "        geom_point(aes(size = 2,  colour=treat, shape=ip), alpha = 4/10) +\n",
    "        geom_text_repel(aes(label = rep),size = 3.5) +\n",
    "        xlab(paste0(\"PC1: \",percentVar[1],\"% variance\")) +\n",
    "        ylab(paste0(\"PC2: \",percentVar[2],\"% variance\")) +\n",
    "        guides(size = FALSE) +\n",
    "        theme()\n",
    "\n",
    "ggsave(\"plots/PCA.png\", plot = pca, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/PCA.pdf\", plot = pca, device = 'pdf', dpi = 300)\n",
    "\n",
    "pca"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Peakcalling \n",
    "\n",
    "## Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TxDb object:\n",
      "# Db type: TxDb\n",
      "# Supporting package: GenomicFeatures\n",
      "# Data source: /rumi/shams/genomes/hg38/gencode.v28.annotation.gtf\n",
      "# Organism: Homo sapiens\n",
      "# Taxonomy ID: 9606\n",
      "# miRBase build ID: NA\n",
      "# Genome: NA\n",
      "# transcript_nrow: 203835\n",
      "# exon_nrow: 693612\n",
      "# cds_nrow: 273085\n",
      "# Db created by: GenomicFeatures package from Bioconductor\n",
      "# Creation time: 2020-04-08 01:53:01 +0000 (Wed, 08 Apr 2020)\n",
      "# GenomicFeatures version at creation time: 1.32.3\n",
      "# RSQLite version at creation time: 2.1.5\n",
      "# DBSCHEMAVERSION: 1.2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 'select()' returned 1:many mapping between keys and columns\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Divide transcriptome into chr-gene-batch sections ...\"\n",
      "[1] \"Get Reads Count ...\"\n",
      "[1] \"This step may take a few hours ...\"\n",
      "[1] \"0.286 %\"\n",
      "[1] \"0.571 %\"\n",
      "[1] \"0.857 %\"\n",
      "[1] \"1.14 %\"\n",
      "[1] \"1.43 %\"\n",
      "[1] \"1.71 %\"\n",
      "[1] \"2 %\"\n",
      "[1] \"2.29 %\"\n",
      "[1] \"2.57 %\"\n",
      "[1] \"2.86 %\"\n",
      "[1] \"3.14 %\"\n",
      "[1] \"3.43 %\"\n",
      "[1] \"3.71 %\"\n",
      "[1] \"4 %\"\n",
      "[1] \"4.29 %\"\n",
      "[1] \"4.57 %\"\n",
      "[1] \"4.86 %\"\n",
      "[1] \"5.14 %\"\n",
      "[1] \"5.43 %\"\n",
      "[1] \"5.71 %\"\n",
      "[1] \"6 %\"\n",
      "[1] \"6.29 %\"\n",
      "[1] \"6.57 %\"\n",
      "[1] \"6.86 %\"\n",
      "[1] \"7.14 %\"\n",
      "[1] \"7.43 %\"\n",
      "[1] \"7.71 %\"\n",
      "[1] \"8 %\"\n",
      "[1] \"8.29 %\"\n",
      "[1] \"8.57 %\"\n",
      "[1] \"8.86 %\"\n",
      "[1] \"9.14 %\"\n",
      "[1] \"9.43 %\"\n",
      "[1] \"9.71 %\"\n",
      "[1] \"10 %\"\n",
      "[1] \"10.3 %\"\n",
      "[1] \"10.6 %\"\n",
      "[1] \"10.9 %\"\n",
      "[1] \"11.1 %\"\n",
      "[1] \"11.4 %\"\n",
      "[1] \"11.7 %\"\n",
      "[1] \"12 %\"\n",
      "[1] \"12.3 %\"\n",
      "[1] \"12.6 %\"\n",
      "[1] \"12.9 %\"\n",
      "[1] \"13.1 %\"\n",
      "[1] \"13.4 %\"\n",
      "[1] \"13.7 %\"\n",
      "[1] \"14 %\"\n",
      "[1] \"14.3 %\"\n",
      "[1] \"14.6 %\"\n",
      "[1] \"14.9 %\"\n",
      "[1] \"15.1 %\"\n",
      "[1] \"15.4 %\"\n",
      "[1] \"15.7 %\"\n",
      "[1] \"16 %\"\n",
      "[1] \"16.3 %\"\n",
      "[1] \"16.6 %\"\n",
      "[1] \"16.9 %\"\n",
      "[1] \"17.1 %\"\n",
      "[1] \"17.4 %\"\n",
      "[1] \"17.7 %\"\n",
      "[1] \"18 %\"\n",
      "[1] \"18.3 %\"\n",
      "[1] \"18.6 %\"\n",
      "[1] \"18.9 %\"\n",
      "[1] \"19.1 %\"\n",
      "[1] \"19.4 %\"\n",
      "[1] \"19.7 %\"\n",
      "[1] \"20 %\"\n",
      "[1] \"20.3 %\"\n",
      "[1] \"20.6 %\"\n",
      "[1] \"20.9 %\"\n",
      "[1] \"21.1 %\"\n",
      "[1] \"21.4 %\"\n",
      "[1] \"21.7 %\"\n",
      "[1] \"22 %\"\n",
      "[1] \"22.3 %\"\n",
      "[1] \"22.6 %\"\n",
      "[1] \"22.9 %\"\n",
      "[1] \"23.1 %\"\n",
      "[1] \"23.4 %\"\n",
      "[1] \"23.7 %\"\n",
      "[1] \"24 %\"\n",
      "[1] \"24.3 %\"\n",
      "[1] \"24.6 %\"\n",
      "[1] \"24.9 %\"\n",
      "[1] \"25.1 %\"\n",
      "[1] \"25.4 %\"\n",
      "[1] \"25.7 %\"\n",
      "[1] \"26 %\"\n",
      "[1] \"26.3 %\"\n",
      "[1] \"26.6 %\"\n",
      "[1] \"26.9 %\"\n",
      "[1] \"27.1 %\"\n",
      "[1] \"27.4 %\"\n",
      "[1] \"27.7 %\"\n",
      "[1] \"28 %\"\n",
      "[1] \"28.3 %\"\n",
      "[1] \"28.6 %\"\n",
      "[1] \"28.9 %\"\n",
      "[1] \"29.1 %\"\n",
      "[1] \"29.4 %\"\n",
      "[1] \"29.7 %\"\n"
     ]
    }
   ],
   "source": [
    "%%R \n",
    "library(exomePeak)\n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')\n",
    "setwd(\"bam/\")\n",
    "print (txdb)\n",
    "res <- exomepeak(\n",
    "    TXDB=txdb,\n",
    "    IP_BAM=c(\"U1.m6A.bam\",\"U2.m6A.bam\"),\n",
    "    INPUT_BAM=c(\"U1.input.bam\",\"U2.input.bam\"),\n",
    "    OUTPUT_DIR='../exomepeak',\n",
    "    EXPERIMENT_NAME='ctrl'\n",
    ")\n",
    "saveRDS(res, paste('../exomepeak/ctrl.rds', sep='/'))\n",
    "setwd(\"../\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# chr\tchromStart\tchromEnd\tname\tscore\tstrand\tthickStart\tthickEnd\titemRgb\tblockCount\tblockSizes\tblockStarts\r\n",
      "chr1\t853842\t854022\tENSG00000228794.8\t7.4e-09\t+\t853842\t854022\t0\t1\t180,\t0\r\n",
      "chr1\t1055278\t1055369\tENSG00000188157.14\t0.00032\t+\t1055278\t1055369\t0\t1\t91,\t0\r\n",
      "chr1\t1055518\t1055669\tENSG00000188157.14\t8.9e-06\t+\t1055518\t1055669\t0\t1\t151,\t0\r\n",
      "chr1\t1232951\t1233310\tENSG00000176022.4\t3.2e-33\t+\t1232951\t1233310\t0\t1\t359,\t0\r\n",
      "chr1\t1233458\t1234235\tENSG00000176022.4\t9.99999999999999e-288\t+\t1233458\t1234235\t0\t1\t777,\t0\r\n",
      "chr1\t1234622\t1235011\tENSG00000176022.4\t0\t+\t1234622\t1235011\t0\t1\t389,\t0\r\n",
      "chr1\t1327767\t1328065\tENSG00000224051.6\t5e-11\t+\t1327767\t1328065\t0\t1\t298,\t0\r\n",
      "chr1\t1328361\t1328511\tENSG00000224051.6\t4.8e-08\t+\t1328361\t1328511\t0\t1\t150,\t0\r\n",
      "chr1\t1328569\t1328838\tENSG00000224051.6\t1.6e-17\t+\t1328569\t1328838\t0\t1\t269,\t0\r\n",
      "chr1\t1400709\t1401571\tENSG00000224870.6\t0\t+\t1400709\t1401571\t0\t1\t862,\t0\r\n",
      "chr1\t1672764\t1672944\tENSG00000269737.2\t7.8e-08\t+\t1672764\t1672944\t0\t1\t180,\t0\r\n",
      "chr1\t2228844\t2228935\tENSG00000157933.9\t1.3e-05\t+\t2228844\t2228935\t0\t1\t91,\t0\r\n",
      "chr1\t2229443\t2232546\tENSG00000157933.9\t1e-220\t+\t2229443\t2232546\t0\t2\t292,68,\t0,3035\r\n",
      "chr1\t2303047\t2303378\tENSG00000157933.9\t2.2e-08\t+\t2303047\t2303378\t0\t2\t56,94,\t0,237\r\n",
      "chr1\t2306944\t2307364\tENSG00000157933.9\t0\t+\t2306944\t2307364\t0\t1\t420,\t0\r\n",
      "chr1\t2307483\t2308382\tENSG00000157933.9\t0\t+\t2307483\t2308382\t0\t1\t899,\t0\r\n",
      "chr1\t2403107\t2403497\tENSG00000157916.19\t9.999987484956e-319\t+\t2403107\t2403497\t0\t1\t390,\t0\r\n",
      "chr1\t2403586\t2403767\tENSG00000157916.19\t4e-19\t+\t2403586\t2403767\t0\t1\t181,\t0\r\n",
      "chr1\t2563530\t2565053\tENSG00000157873.17\t4e-85\t+\t2563530\t2565053\t0\t2\t299,1,\t0,1522\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 20 exomepeak/ctrl/peak.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot _Guitar_ \n",
    "Now, let's draw metagene plots for control samples:\n",
    "(using seprate env: `Guitar`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/rumi/shams/abe/anaconda3/envs/guitar/lib/python3.8/site-packages/rpy2/robjects/pandas2ri.py:14: FutureWarning: pandas.core.index is deprecated and will be removed in a future version.  The public classes are available in the top-level namespace.\n",
      "  from pandas.core.index import Index as PandasIndex\n"
     ]
    }
   ],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(Guitar)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Import genomic features from the file as a GRanges object ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Prepare the 'metadata' data frame ... \n",
      "R[write to console]: OK\n",
      "\n",
      "R[write to console]: Make the TxDb object ... \n",
      "R[write to console]: OK\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "txdb <- makeTxDbFromGFF('/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf',organism='Homo sapiens')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"20200409182002\"\n",
      "[1] \"There are 203835 transcripts of 58381 genes in the genome.\"\n",
      "[1] \"total 203835 transcripts extracted ...\"\n",
      "[1] \"total 94682 transcripts left after ambiguity filter ...\"\n",
      "[1] \"total 94682 transcripts left after check chromosome validity ...\"\n",
      "[1] \"total 11625 mRNAs left after component length filter ...\"\n",
      "[1] \"total 56222 ncRNAs left after ncRNA length filter ...\"\n",
      "[1] \"generate components for all tx\"\n",
      "[1] \"generate components for mRNA\"\n",
      "[1] \"generate components for lncRNA\"\n",
      "[1] \"generate chiped transcriptome\"\n",
      "[1] \"generate coverage checking ranges for tx\"\n",
      "[1] \"generate coverage checking ranges for mrna\"\n",
      "[1] \"generate coverage checking ranges for ncrna\"\n",
      "[1] \"20200409182409\"\n",
      "[1] \"import BED file exomepeak/ctrl/peak.bed\"\n",
      "[1] \"sample 10 points for Group1\"\n",
      "[1] \"start figure plotting for tx ...\"\n",
      "[1] \"start figure plotting for mrna ...\"\n",
      "[1] \"start figure plotting for ncrna ...\"\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "GuitarPlot(txTxdb=txdb,stBedFiles=\"exomepeak/ctrl/peak.bed\",miscOutFilePrefix='plot')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "!magick -density 300 plot_mrna_test.pdf -flatten -quality 90 plot_mrna_test.png\n",
    "!magick -density 300 plot_ncrna_test.pdf -flatten -quality 90 plot_ncrna_test.png\n",
    "!magick -density 300 plot_tx_test.pdf -flatten -quality 90 plot_tx_test.png"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "renamed 'plot_mrna_test.pdf' -> 'plots/plot_mrna_test.pdf'\r\n",
      "renamed 'plot_mrna_test.png' -> 'plots/plot_mrna_test.png'\r\n",
      "renamed 'plot_ncrna_test.pdf' -> 'plots/plot_ncrna_test.pdf'\r\n",
      "renamed 'plot_ncrna_test.png' -> 'plots/plot_ncrna_test.png'\r\n",
      "renamed 'plot_tx_test.pdf' -> 'plots/plot_tx_test.pdf'\r\n",
      "renamed 'plot_tx_test.png' -> 'plots/plot_tx_test.png'\r\n"
     ]
    }
   ],
   "source": [
    "mv -v *p[nd][gf] plots/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/plot_mrna_test.png\" style=\"width:600px\">\n",
    "  <tr>\n",
    "<table>\n",
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"plots/plot_ncrna_test.png\" style=\"width:300px\">\n",
    "    <td><img src=\"plots/plot_tx_test.png\" style=\"width:300px\">\n",
    "  <tr>\n",
    "<table>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Motif analysis "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "using `cgat` env to use [CGAT - Computational Genomics Analysis Tools](https://www.cgat.org/downloads/public/cgat/documentation/) and use "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cd ~/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated/exomepeak/ctrl/\n",
    "# sort | merge by name \n",
    "cat peak.bed | sort -k1,1 -k2,2n peak.bed | cgat bed2bed --method=merge --merge-by-name |  awk '! /#/' > peak.edit.bed\n",
    "# get fasta for the cordinate sequnces (-split removes introns)\n",
    "bedtools getfasta -name -s -fi /rumi/shams/genomes/hg38/hg38.fa -bed peak.edit.bed -split -fo peak.fa\n",
    "# prepare fasta file for FIRE\n",
    "perl /flash/hani/bin/Tools/prep_seqs_for_teiser_run.pl peak.fa peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash --err error\n",
    "cd ~/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated/exomepeak/ctrl/\n",
    "# run FIRE --doskipdiscovery=0\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --oribiasonly=0\n",
    "mv peaks_teiser.txt_FIRE/ skipdiscovery.0_FIRE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To run `FIRE` non-discovery mode we need to list in a file like `motifs_of_interest.txt` "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tue Apr  7 22:16:09 UTC 2020\n",
      "RNA, Processing peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt\n",
      "RNA, Step 2: skip seed optimization, use -motiffile_rna option instead.\n",
      "RNA, Step 3: evaluation of motif significances.\n",
      "Processing motif [AG]GAC.\n",
      "MI=0.00287461\n",
      "Shuffle rank=0\n",
      "Z-score=57.222\n",
      "Sigificance category= OK-NO-SEED\n",
      "Robustness (jn_f=3)=10/10\n",
      "\n",
      "Processing motif [AGT][AG]AC[ACT].\n",
      "MI=0.00103316\n",
      "Shuffle rank=0\n",
      "Z-score=19.393\n",
      "Sigificance category= OK-NO-SEED\n",
      "Robustness (jn_f=3)=9/10\n",
      "\n",
      "RNA, Step 3.5: creating binary expression profiles.\n",
      "RNA, Step 4: discovery of distance constraints.\n",
      "Read 2 motifs.\n",
      "Read 20328 (genes) x 2 (conditions) matrix.\n",
      "Read 20328 (genes) x 1 (conditions) matrix.\n",
      "Reading sequences and building motif profiles ...Done.\n",
      "Average sequence length is 426.\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AG]GAC, present in 17970 genes.\n",
      "       Average distance: MI=0.0037, rank=0/10000, Z=64.9084, mbins=2\n",
      "  Orientation bias {5'): MI=0.0029, rank=0/10000, Z=57.2122\n",
      "  Orientation bias (3'): MI=0.0000, rank=8526/10000, Z=-0.6713\n",
      "       Copy number (1+): 17970 genes, MI=0.0029, rank=0/10000, Z=57.8862\n",
      "       Copy number (2+): 14811 genes, MI=0.0028, rank=0/10000, Z=54.8047\n",
      "       Copy number (3+): 11616 genes, MI=0.0026, rank=0/10000, Z=50.7668\n",
      "       Copy number (4+): 9000 genes, MI=0.0018, rank=0/10000, Z=34.1610\n",
      "       Copy number (5+): 6873 genes, MI=0.0012, rank=0/10000, Z=22.7328\n",
      "       Copy number (6+): 5293 genes, MI=0.0008, rank=0/10000, Z=16.0874\n",
      "       Copy number (7+): 4027 genes, MI=0.0005, rank=5/10000, Z=8.8324\n",
      "       Copy number (8+): 3046 genes, MI=0.0002, rank=259/10000, Z=2.9094\n",
      "       Copy number (9+): 2324 genes, MI=0.0001, rank=664/10000, Z=1.7934\n",
      "       Copy number (10+): 1783 genes, MI=0.0000, rank=8073/10000, Z=-0.6626\n",
      "\n",
      "Quantized E vector into 2 bins\n",
      "Motif [AGT][AG]AC[ACT], present in 19816 genes.\n",
      "       Average distance: MI=0.0011, rank=0/10000, Z=19.3071, mbins=2\n",
      "  Orientation bias {5'): MI=0.0010, rank=0/10000, Z=19.9326\n",
      "  Orientation bias (3'): MI=0.0006, rank=1/10000, Z=10.8323\n",
      "       Copy number (1+): 19816 genes, MI=0.0010, rank=0/10000, Z=19.8805\n",
      "       Copy number (2+): 18855 genes, MI=0.0011, rank=0/10000, Z=21.2587\n",
      "       Copy number (3+): 17474 genes, MI=0.0013, rank=0/10000, Z=24.6523\n",
      "       Copy number (4+): 15943 genes, MI=0.0008, rank=0/10000, Z=15.9567\n",
      "       Copy number (5+): 14339 genes, MI=0.0003, rank=31/10000, Z=5.6152\n",
      "       Copy number (6+): 12709 genes, MI=0.0003, rank=14/10000, Z=6.3201\n",
      "       Copy number (7+): 11271 genes, MI=0.0002, rank=244/10000, Z=2.8455\n",
      "       Copy number (8+): 9995 genes, MI=0.0001, rank=938/10000, Z=1.3412\n",
      "       Copy number (9+): 8847 genes, MI=0.0001, rank=1254/10000, Z=0.9624\n",
      "       Copy number (10+): 7783 genes, MI=0.0000, rank=2699/10000, Z=0.1805\n",
      "\n",
      "RNA, Step 5.5: conservation.\n",
      "Setting -rna to 1.\n",
      "[AG]GAC\t98.12\n",
      "[AGT][AG]AC[ACT]\t21.42\n",
      "RNA, Step 5: summarize information.\n",
      "Reading RNA motifs mi_dist output ...Done.\n",
      "Parsing RNA motif file ... Done.\n",
      "Reading mi_signif output file for RNA motifs ...!!!! 851816!!!! 619296Done.\n",
      "Got 2 motifs\n",
      "Writing good columns to peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.signifcolumns ...Done.\n",
      "Outputing summary ... Done.\n",
      "HI!HELLO!peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.binary created.\n",
      "RNA, Step 5.3: generate profiles for motifs in summary file.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [AG]GAC ... Done.\n",
      "Generating profile for [AGT][AG]AC[ACT] ... Done.\n",
      "RNA, Step 5.3 bis: Generate profiles for /flash/hani/bin/FIRE-1.1/FIRE_DATA/HUMAN/CONSERVATION/human_chicken_3_0_300_chicken_orthologs.fa.\n",
      "Setting -rna to 1.\n",
      "Generating profile for [AG]GAC ... Done.\n",
      "Generating profile for [AGT][AG]AC[ACT] ... Done.\n",
      "RNA, Determining target gene sets.\n",
      "RNA, Drawing position histograms.\n",
      "RNA, Step 6: cluster motifs.\n",
      "Using positive correlation.\n",
      "Doing all stats.\n",
      "Summary file has 0 DNA motifs and 2 RNA motifs.\n",
      "Read in RNA sequences and build motif profiles ...Done.\n",
      "Outputing MI matrix ... Done.\n",
      "[AG]GAC\t0\n",
      "[AG]GAC\t[AGT][AG]AC[ACT]\t0.7815\t0\t15470.3103\t0.8752\t1854\t0.7416\t1854\t0.7416\n",
      "[AGT][AG]AC[ACT]\t0\n",
      "Finish calculated rank, z-scores for remaining entries ... [AG]GAC\t[AG]GAC\t0.9934\t0\t19590.5947\t1.0000\t0\t16.2939\t0\t16.2939\n",
      "[AGT][AG]AC[ACT]\t[AGT][AG]AC[ACT]\t0.9997\t0\t19904.0798\t1.0000\t3\t6.7154\t3\t6.7154\n",
      "Done.\n",
      "RNA, Step 5.6: generate genome-wide GO enrichments for motifs in summary file.\n",
      "RNA, draw motif maps for co-localizing pairs\n",
      "Draw motif map for [AG]GAC .. Done.\n",
      "Draw motif map for [AGT][AG]AC[ACT] .. Done.\n",
      "Draw motif map for [AGT][AG]AC[ACT],[AGT][AG]AC[ACT] ..  (peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.mimatrix_OUT/[AGT][AG]AC[ACT]_[AGT][AG]AC[ACT].eps) Done.\n",
      "Draw motif map for [AG]GAC,[AG]GAC ..  (peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.mimatrix_OUT/[AG]GAC_[AG]GAC.eps) Done.\n",
      "RNA, Step 6.8: identifying motifs that match micrornas.\n",
      "RNA, Step 7: draw matrix figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 400, xbase = 35, ybase = 250\n",
      "Processing [AG]GAC ... Outputing motif 0.eps ... Done.\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 1.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [AG]GAC ... Outputing motif 0.png ... \n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 1.png ... \n",
      "RNA, Step 7b: draw density figure.\n",
      "Now doing the graphical display.\n",
      "xsize = 1200, ysize = 400, xbase = 35, ybase = 250\n",
      "Processing [AG]GAC ... Outputing motif 0.eps ... Done.\n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 1.eps ... Done.\n",
      "Plotting significance boxes.\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.densities.pdf ... Done.\n",
      "Now doing the graphical display.\n",
      "Processing [AG]GAC ... Outputing motif 0.png ... \n",
      "Processing [AGT][AG]AC[ACT] ... Outputing motif 1.png ... \n",
      "Parsing distance report file ...  Done.\n",
      "Processing matrix 2 x 2.\n",
      "Processing matrix 2 x 2.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 230, ysize = 310, xbase = 35, ybase = 115\n",
      "[AG]GAC\t0\n",
      "[AGT][AG]AC[ACT]\t1\n",
      "Creating peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.eps ...Done.\n",
      "Creating PDF peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.pdf ... Done.\n",
      "Read in summary file ... Done.\n",
      "Now doing the graphical display.\n",
      "xsize = 250, ysize = 305, xbase = 75, ybase = 130\n",
      "peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.fullmimatrix.html[AG]GAC\t0\n",
      "[AGT][AG]AC[ACT]\t1\n",
      "RNA, Step 10: motif report.\n",
      "Tue Apr  7 22:21:16 UTC 2020\n",
      "renamed 'peaks_teiser.txt_FIRE/' -> 'skipdiscovery.1_FIRE'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "corrupted size vs. prev_size\n",
      "Aborted (core dumped)\n",
      "Segmentation fault (core dumped)\n",
      "Can't use 'defined(@array)' (Maybe you should just omit the defined()?) at /flash/hani/bin/FIRE-1.1/SCRIPTS/draw_position_histogram.pl line 134.\n",
      "Can't locate Hypergeom.pm in @INC (you may need to install the Hypergeom module) (@INC contains: /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.12.4/darwin-thread-multi-2level/nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.10.0/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.18.2/darwin-thread-multi-2level/Hypergeom.pm /nvme/bins/FIRE-1.1/SCRIPTS/../modules/lib/perl5/site_perl/5.22.0/x86_64-linux/Hypergeom.pm /flash/hani/bin/FIRE-1.1/SCRIPTS /rumi/shams/abe/perl5/lib/perl5/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/perl5/lib/perl5/5.26.2 /rumi/shams/abe/perl5/lib/perl5/x86_64-linux-thread-multi /rumi/shams/abe/perl5/lib/perl5 /rumi/shams/abe/anaconda3/envs/deseq/lib/site_perl/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/anaconda3/envs/deseq/lib/site_perl/5.26.2 /rumi/shams/abe/anaconda3/envs/deseq/lib/5.26.2/x86_64-linux-thread-multi /rumi/shams/abe/anaconda3/envs/deseq/lib/5.26.2 .) at /flash/hani/bin/FIRE-1.1/SCRIPTS/GroupEnrichment.pm line 24.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/over.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/under.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/C0.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/C1.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/motif.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/location.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/mi.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/z.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/robustness.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/pos.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/ori.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/cons.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/seed.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/name.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/parray.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/over.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/under.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/C0.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/C1.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/motif.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/location.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/mi.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/z.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/robustness.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/pos.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/ori.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/cons.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/seed.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/name.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/parray.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/3UTR_pos.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `peaks_teiser.txt_FIRE/RNA/peaks_teiser.txt.summary_OUT/3UTR_pos.png' @ warning/png.c/MagickPNGWarningHandler/1744.\n"
     ]
    }
   ],
   "source": [
    "%%bash --err error\n",
    "cd ~/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated/exomepeak/T_vs_U/\n",
    "# run FIRE --doskipdiscovery=1\n",
    "perl $FIREDIR/fire.pl --expfile=peaks_teiser.txt --exptype=discrete --fastafile_rna=peaks_teiser.fa --nodups=1 --dodna=0 --dodnarna=0 --species=human --doskipdiscovery=1 --motiffile_rna=motifs_of_interest.txt --oribiasonly=0\n",
    "rm -r skipdiscovery.1_FIRE\n",
    "mv -v peaks_teiser.txt_FIRE/ skipdiscovery.1_FIRE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/rumi/shams/abe/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated\n"
     ]
    }
   ],
   "source": [
    "!cd /rumi/shams/abe/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated\n",
    "\n",
    "!magick -density 300 exomepeak/T_vs_U/skipdiscovery.0_FIRE/RNA/peaks_teiser.txt.summary.pdf -flatten -quality 90 exomepeak/T_vs_U/skipdiscovery.0_FIRE/RNA/peaks_teiser.txt.summary.png\n",
    "!magick -density 300 exomepeak/T_vs_U/skipdiscovery.1_FIRE/RNA/peaks_teiser.txt.summary.pdf -flatten -quality 90 exomepeak/T_vs_U/skipdiscovery.1_FIRE/RNA/peaks_teiser.txt.summary.png"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"exomepeak/T_vs_U/skipdiscovery.1_FIRE/RNA/peaks_teiser.txt.summary.png\" style=\"width:900px\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'exomepeak/T_vs_U/skipdiscovery.0_FIRE/RNA/peaks_teiser.txt.summary.pdf' -> 'plots/peaks_teiser.txt.summary.pdf'\r\n",
      "'exomepeak/T_vs_U/skipdiscovery.0_FIRE/RNA/peaks_teiser.txt.summary.png' -> 'plots/peaks_teiser.txt.summary.png'\r\n"
     ]
    }
   ],
   "source": [
    "cp -v exomepeak/T_vs_U/skipdiscovery.0_FIRE/RNA/peaks_teiser.txt.summary.p[dn][fg] plots/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential peaks \n",
    "\n",
    "### Running a custom pipeline\n",
    "Here we're using https://github.com/gold-lab/bed12ToGTF to make gtf file from control exomepeak result.\n",
    "\n",
    "https://metacpan.org/pod/Bio::FeatureIO::gtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat exomepeak/ctrl/peak.bed | sort -k4 | \\\n",
    "# rename features peak by peak\n",
    "awk -F \"\\t\" '{OFS=FS}NR>1{$4=$4\"_peak\"}; cnt[$4]++{$4=$4\"_\"cnt[$4]} 1' | \\\n",
    "# bed12 to gtf\n",
    "perl ~/Workflows/bed12ToGTF/bed12ToGTF.1.pl | \\\n",
    "awk '{gsub(/\\\"|\\;/,\"\")}1''{gsub(/transcript_id/,\"gene_id\")}1''{gsub(/ exon_number [[:digit:]]/,\"\")}1' > exomepeak/ctrl/peak.gtf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# chr\tblank\ttranscript\t1\tchromEnd\t.\tstrand\t.\tgene_id name\r\n",
      "chr1\tblank\ttranscript\t169853401\t169853641\t.\t-\t.\tgene_id ENSG00000000457.13_peak\r\n",
      "chr1\tblank\texon\t169853401\t169853641\t.\t-\t.\tgene_id ENSG00000000457.13_peak\r\n",
      "chr1\tblank\ttranscript\t169853390\t169853630\t.\t+\t.\tgene_id ENSG00000000460.16_peak\r\n",
      "chr1\tblank\texon\t169853390\t169853630\t.\t+\t.\tgene_id ENSG00000000460.16_peak\r\n",
      "chr6\tblank\ttranscript\t53498597\t53498777\t.\t-\t.\tgene_id ENSG00000001084.11_peak\r\n",
      "chr6\tblank\texon\t53498597\t53498777\t.\t-\t.\tgene_id ENSG00000001084.11_peak\r\n",
      "chr6\tblank\ttranscript\t41097505\t41098607\t.\t+\t.\tgene_id ENSG00000001167.14_peak\r\n",
      "chr6\tblank\texon\t41097505\t41098607\t.\t+\t.\tgene_id ENSG00000001167.14_peak\r\n",
      "chr6\tblank\ttranscript\t41098756\t41099143\t.\t+\t.\tgene_id ENSG00000001167.14_peak_2\r\n",
      "cat: write error: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!cat exomepeak/ctrl/peak.gtf | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir -p peakcount\n",
    "# %%bash --err error\n",
    "for f in bam/T*.bam; do \n",
    "    out=${f/.bam/.fc}; \n",
    "    out=${out/bam/peakcount}; \n",
    "    echo featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o $out $f;\n",
    "    featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o $out $f;\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/U1.input.bam                               ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/U1.input.fc                            ||\n",
      "||                 Summary : peakcount/U1.input.fc.summary                    ||\n",
      "||              Annotation : exomepeak/ctrl/peak.gtf (GTF)                    ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/ctrl/peak.gtf ...                           ||\n",
      "||    Features : 11111                                                        ||\n",
      "||    Meta-features : 9367                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/U1.input.bam...                                       ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 111917626                                                 ||\n",
      "||    Successfully assigned reads : 2290115 (2.0%)                            ||\n",
      "||    Running time : 3.58 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/U1.input.fc.s  ||\n",
      "|| ummary\"                                                                    ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/U1.m6A.bam                                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/U1.m6A.fc                              ||\n",
      "||                 Summary : peakcount/U1.m6A.fc.summary                      ||\n",
      "||              Annotation : exomepeak/ctrl/peak.gtf (GTF)                    ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/ctrl/peak.gtf ...                           ||\n",
      "||    Features : 11111                                                        ||\n",
      "||    Meta-features : 9367                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/U1.m6A.bam...                                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 40981072                                                  ||\n",
      "||    Successfully assigned reads : 27100118 (66.1%)                          ||\n",
      "||    Running time : 1.20 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/U1.m6A.fc.sum  ||\n",
      "|| mary\"                                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/U2.input.bam                               ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/U2.input.fc                            ||\n",
      "||                 Summary : peakcount/U2.input.fc.summary                    ||\n",
      "||              Annotation : exomepeak/ctrl/peak.gtf (GTF)                    ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/ctrl/peak.gtf ...                           ||\n",
      "||    Features : 11111                                                        ||\n",
      "||    Meta-features : 9367                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/U2.input.bam...                                       ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 140156753                                                 ||\n",
      "||    Successfully assigned reads : 2722568 (1.9%)                            ||\n",
      "||    Running time : 4.16 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/U2.input.fc.s  ||\n",
      "|| ummary\"                                                                    ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S bam/U2.m6A.bam                                 ||\n",
      "||                                                                            ||\n",
      "||             Output file : peakcount/U2.m6A.fc                              ||\n",
      "||                 Summary : peakcount/U2.m6A.fc.summary                      ||\n",
      "||              Annotation : exomepeak/ctrl/peak.gtf (GTF)                    ||\n",
      "||      Dir for temp files : peakcount                                        ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 12                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : counted                                          ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file exomepeak/ctrl/peak.gtf ...                           ||\n",
      "||    Features : 11111                                                        ||\n",
      "||    Meta-features : 9367                                                    ||\n",
      "||    Chromosomes/contigs : 23                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file bam/U2.m6A.bam...                                         ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 38607844                                                  ||\n",
      "||    Successfully assigned reads : 25471463 (66.0%)                          ||\n",
      "||    Running time : 1.19 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"peakcount/U2.m6A.fc.sum  ||\n",
      "|| mary\"                                                                      ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for f in bam/U*.bam; do \n",
    "    out=${f/.bam/.fc}; out=${out/bam/peakcount}\n",
    "    featureCounts -T 12 -O -t exon -g gene_id -a exomepeak/ctrl/peak.gtf -o $out $f\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Id transfer\n",
    "Read gene names from gtf file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import os\n",
    "\n",
    "def gtf_id_transfer(gtf):\n",
    "    # awk code to extract gene ID and Name \n",
    "    gtf_raw = os.popen(\"cat \" + gtf + \" | awk 'BEGIN{FS=\\\"\\t\\\"}{split($9,a,\\\";\\\"); if($3~\\\"gene\\\") print a[1]\\\"\\t\\\"a[3]}' | sed 's/gene_id \\\"//' | sed 's/gene_name \\\"//' | sed 's/\\\"//g' | sed 's/ //g'\").read().split('\\n')\n",
    "    # data frame \n",
    "    gtf_df = pd.DataFrame(\n",
    "        # col: Names \n",
    "        data=[i.split('\\t')[1] for i in gtf_raw[:len(gtf_raw)-1]],\n",
    "        # index: ids \n",
    "        index=[i.split('\\t')[0] for i in gtf_raw[:len(gtf_raw)-1]],\n",
    "        columns=['Name'], dtype=str)\n",
    "    return gtf_df\n",
    "\n",
    "gtf = '/rumi/shams/genomes/hg38/gencode.v28.annotation.gtf'\n",
    "gtf_df = gtf_id_transfer(gtf)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "load R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%reload_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Differential methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "library(DESeq2)\n",
    "library(tidyverse)\n",
    "library(ggplot2)\n",
    "require(ggrepel)\n",
    "library(patchwork)\n",
    "\n",
    "pca = list(); vol = list(); hist = list()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "using `DESeq2` to find differential methylation peaks:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: converting counts to integer mode\n",
      "\n",
      "R[write to console]: estimating size factors\n",
      "\n",
      "R[write to console]: estimating dispersions\n",
      "\n",
      "R[write to console]: gene-wise dispersion estimates\n",
      "\n",
      "R[write to console]: mean-dispersion relationship\n",
      "\n",
      "R[write to console]: -- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "R[write to console]: final dispersion estimates\n",
      "\n",
      "R[write to console]: fitting model and testing\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Intercept\"               \"cond_T.input_vs_U.input\"\n",
      "[3] \"cond_T.m6A_vs_U.input\"   \"cond_U.m6A_vs_U.input\"  \n"
     ]
    }
   ],
   "source": [
    "%%R -i gtf_df\n",
    "\n",
    "files.input <- list.files(path = 'peakcount',pattern=\".input.fc$\")\n",
    "files.m6A <- list.files(path = 'peakcount',pattern=\".m6A.fc$\")\n",
    "files = c(files.input,files.m6A)\n",
    "files\n",
    "treats <- rep(c(rep('T',2),rep('U',2)),2)\n",
    "ips <- c(rep('input',4),rep('m6A',4))\n",
    "reps <- rep(c('1','2'),4)\n",
    "conds <- paste(treats,ips,sep='.')\n",
    "all_conds <- paste(treats, ips,reps,sep='.')\n",
    "\n",
    "datalist <- lapply(files, function(x){read.table(file=paste('peakcount/', x, sep = \"\"),skip=1,sep='\\t',header=TRUE,colClasses=c(NA, \"NULL\", \"NULL\", \"NULL\", \"NULL\", \"NULL\", NA),check.names=F)})\n",
    "m <- Reduce(function(...) merge(..., by=1, all = TRUE), datalist)\n",
    "rownames(m) <- m[,1]\n",
    "m <- m[,-1]\n",
    "m[is.na(m)]<-0\n",
    "m <- round(m)\n",
    "colnames(m) <- all_conds\n",
    "\n",
    "colData <- data.frame(cond = conds,rep=reps, row.names=colnames(m))\n",
    "colData\n",
    "# diff_peak \n",
    "dds <- DESeqDataSetFromMatrix(countData = m, colData = colData, design = ~ cond)\n",
    "dds$cond <- relevel(dds$cond, ref=\"U.input\")   \n",
    "dds <- DESeq(dds)\n",
    "resultsNames(dds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p deseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R \n",
    "# normalized counts \n",
    "ncu <- counts(dds, normalized=TRUE)\n",
    "# contrast design: (T_m6A - T_input) - (U_m6A - U_input)\n",
    "res <- results(dds, contrast=list(\n",
    "    c(\"cond_T.m6A_vs_U.input\"), \n",
    "    c(\"cond_T.input_vs_U.input\",\"cond_U.m6A_vs_U.input\")),listValues=c(1,-1))\n",
    "\n",
    "res <- as.data.frame(res)            \n",
    "# idtransfer\n",
    "peak_ids = strsplit(rownames(m), '_',)\n",
    "ids = unlist(lapply(peak_ids, '[[', 1))\n",
    "res$name = gtf_df[ids,]\n",
    "# save files \n",
    "write.table(res, 'deseq/d_mtyl_T_vs_U.txt',sep=\"\\t\", quote=FALSE, col.names=NA)\n",
    "# Raw tabels \n",
    "write.table(ncu, file='deseq/d_mtyl_decitabine_DE2norm.txt', sep=\"\\t\", quote=FALSE, col.names=all_conds) \n",
    "write.table(m, file='deseq/d_mtyl_decitabine_raw.txt', sep=\"\\t\", quote=FALSE, col.names=all_conds)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAAC5VBMVEUAAAACAgIDAwMEBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVHR0dISEhJSUlLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCSkpKTk5OUlJSWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWnp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///8t37xeAAAUtUlEQVR4nO3ce3xU9ZnH8Wl2CSq9cBMbDV4WlNUQ1nBxFUiVItAq7dZLKt3quiyturtdbRDYoItUwmotaytYdVm6tqJgCMtFa1ELVZayVajlJiogieQ2yVxzmczz955zfudMZjLPM/lNciCZk+/n1U4m5/LkN+ftnAQUfIQ8na+/F4DObgD2eAD2eG4Al67ptqHkOWEHOudlATzP5xt66YLfp++o3Ek07adJGxxgc4dQyvGae/SPQImyAb7j0IGt8/9sE7uTBc5Q4vjWjJN6/looc9kALzIf7x7TRu2Lx3z+q4eJOpYWDL3mDfNOfJfP57uC1heff8nD7Qbw6nlDx72ubtElK+d98arfUOIclTq+5N/mDvups8M+2d7zxK3nTTz49jXDvtPadao9Sh2B9Moa+IBvNy2+/t2Pl1wWpcWF2z/Z/qblaL2rfv7aiV2Xm6hf+vmHiz/foIBHv0NrvtyeOMfOOr7kS2/Eg84O52S1Z9R/fnDLpOlv7xmzlhKnOqPwDs6irIE7fC+G8o8ZTy7fERyy3dqRADZ7YYbBcydR/Mo1CviHRG2f+5NzjjNMMd5HlLLDPFnt+R7RO75fEz1Y1nWEPQrA2dQL4F/+wWe19v98QWtHAviD268a88VLDZ6fGJ9853sKeJ3x/AtvOec4wxSj8ZDY4Zys9jxD9JGvjuiJ2V1H2KMAnE29uEXv+b2v2dqwvxtwe8ED751+vsAGXmADmz9wfWGXc46TYjT2OTsSJyf2fOzzEz05K3GEMwrA2ZT9D1kXtQWH/MrakHyLnvEfRId9xrfYFQXdbtFKxTnHyTze2ufsSJyc2OMAJ051gK0jkF7Z/TLpoPHLpCqiJaNeOr7ngRP0UOHW49t2WY4LvnW6qeW8bfT+mAL7h6xhjcnAiXPszOPVPntH4uTEHgc4caozyjoC6ZXdb3TkX7pgv/Gs49HCIYV/10Jt5RcOLVLA7xcPuYJevuziGT82gSvn5o97jVKAnXPsrOOtfc4O5+TEngSwc4QzyjoC6YXfi/Z4APZ45xj4Y/VLnlnn9qsO5vAO9ngA9ngA9ngA9ngA9ngA9ni9Ac6Pmv+TWtEpbZh9KOPMlGJT67o+OTz1smlHU55m+pApZy0uL9K9Bbpeb4D3xjMB53UIG177VoaZaQOfe7Dr+fQX6IWvpDzN9CFTzlpcXqR7C3Q9feB199Fe3//S95+138FvXTex+E3aPqnopmN06OJP6NE7rMOW+GaWRvbMvHbKGxS9feJf3ao2mHtu32R8ueULbqiyjnvsB0QNw0N0V8nE+U3mtfOPIOrIJ1LnUsuoGIXarENPDWun9mG1SU8zfcj0Gpy1uLxI9xbofvrAx66kVX9dSeM/VMB1o96lmL925BF6dirRL6btHG//iwTjvdBYVEsnC1u33krU3PXmGH3K+HK/pIaLz5ifnRjTQU/fQ1RPtGpp0rWzzyUq2kfz1L/43T3eeBi/N+lppg8ZX0TSO9jFRbq4QNfL4hZdeGrWm189Odb+Hlw129z26s3Gd6IhAaLvnuf897TGpaoeVVpaOu7YJ2O//3Ioce3afe3GlwsS3VJtfX5TNV37NtHaGdOL5yVdO/tc45thFR341DrSujDj9iY9zfQh42tIAnZxkS4u0PWyAL573TV09dp7UoA3z1HAbZMv2m4fZlyqqhnqabD6/ivCzrWL/7nxz4HPuNHN3Wp9/otvHryCaN/4JqqeZc4MDicK5yfOpRmJ/37rZNetzX6a6UPG15AE7OIiXVyg62UB/N9jF9E/jH3RBq4f/S51NNWOPETrphH906I/XmLc3F47TTS8nupHGK/7Hfo0SK1jTpgbrCbtM77ck3R0ZJ11XHjkd1cS7SwlKrOuHY05TRvznXONm2Ut7f5QnXn98/R8qZpuP830IVPmWs7CIt1boOtlAXza9zJt9NU4v0z67dQJRW/StuIJNx6jqolRen56B11t/Gyy8sriyO9umDDuNto5sejqlWqDef5jPzKv3XVXGseYx9G9nzthvP3L5pQ9pK7dy5PvfNz4+UWdS/uvJ+d7MP1pyiVTDquz7KeZPmTKXMtZWKR7C3S9c/gbHXXF7Vl8ub/fchaXIpcTi8ymc/k7Wf/zof6Xi/3sbK4kQzmxyCzCb1V6PAB7PAB7PAB7PAB7PH3gSIgp0sZtZYpqHhfWHRgJaw5sd3lgSHsge8X6MDCsO7BNvZQsgZvqmZo7uK1MoUa94/ydmgOD7HrSa4prDgz49Y5rIM2Bzc2aB1KD3nH+gObAuLo2AE4JwNwoAMsBmAvAcgDmAjCAVQDmRgFYDsBcAJYDMBeAAawCMDcKwHIA5gKwHIC5AAxgFYC5UQCWAzAXgOUAzAVgAKsAzI0CsByAuQAsB2AuAANYBWBuFIDlAMwFYDkAcwEYwCoAc6MALAdgLgDLAZgLwABWAZgbBWA5AHMBWA7AXAAGsGrQA4eCTJFObitTK3t6euG47sCwywOjmgNDpDkwEtE8kL+06YWjmgPj1kvxZwnc3MQUiHFbmSJ+veNaOjUHhtn1pNcc1xwYCugd5yfNgYGg5oGke21CmgPj1rVpwC06pUF/iwawCsBsAJYDMBeA5QDMBWBuFIDlAMwFYDkAcwEYwCoAc6MALAdgLgDLAZgLwABWAZgbBWA5AHMBWA7AXAAGsArA3CgAywGYC8ByAOYCMIBVAOZGAVgOwFwAlgMwF4ABrAIwNwrAcgDmArAcgLkADGAVgLlRAJYDMBeA5QDMBWAAqwDMjQKwHIC5ACwHYC4AA1gFYG4UgOUAzAVgOQBzARjAKgBzowAsB2AuAMsBmAvAAFYBmBsFYLkBCxx/ZsUKY2f1I8trnEcAd5X7wPvX0IGnqKE8fnSV/QjgpHIfeFsVBe6h3RuIFtmPAE4qAVxmluHAAQv8wWPx3bfRjleIFtqPRJUlJce1v1sPkizg/l5EhsLWI/dD1qbH/uv+bu/gaEtLYwNTSwe3lSncpHdcc6fmwJBf77imuObAYLPecY1kP7GAMxzY0qL5lYm9tOk1BzUHxtW1EYGJ9qynxvL4kUr70Qq3aFXu36JjFRVrosbPzxUVNc4jgLvKfWAhAKsAzAZgOQBzAVgOwFwA5kYBWA7AXACWAzAXgAGsAjA3CsByAOYCsByAuQAMYBWAuVEAlgMwF4DlAMwFYACrAMyNArAcgLkALAdgLgADWAVgbhSA5QDMBWA5AHMBGMAqAHOjACwHYC4AywGYC8AAVgGYGwVgOQBzAVgOwFwABrAKwNwoAMsBmAvAcgDmAjCAVQDmRgFYDsBcAJYDMBeAAawCMDcKwHIA5hqgwEmYAGYDsByAuQAsB2AuAHOjACyXa8DNTUyBGLeVKeLXO66lU3NgmF1Pes1xzYGhQIadFqZ66qf0bWyBoOZXJt1rE9IcGLeuTaa/EJwrFGSKdHJbmVrZ09MLx3UHhl0eGM000MJUT0OUvo0tEtH8yvylTS8c1RwYt16KH7folAb9LRrA6imA2QAsB2AuAMsBmAvA3CgAywGYC8ByAOYCMIBVAOZGAVgOwFwAlgMwF4ABrAIwNwrAcgDmArAcgLkADGAVgLlRAJYDMBeA5QDMBWAAqwDMjQKwHIC5ACwHYC4AA1gFYG4UgOUAzAVgOQBzARjAKgBzowAsB2AuAMsBmAvAAFYBmBsFYDkAcwFYDsBcAAawCsDcKADLAZgLwHIA5gIwgFUA5kYBWA7AXACWAzAXgAGsAjA3CsByAxf4cEXFK0TVjyyvcR4B7CXgtvKI8dhQHj+6yn4EsKeA31vx7ys/ot0biBbZjwD2FPBv7o3W/IB2GHfphfYj0foFC052MMXi3FamTs3jYqQ7MKZ5oO7AWKaBFma3gcnbsh6YnCsrTBloHRhMAY4kHmjfauNd25n6Dj70+ut1AaZwJ7eVqTWod1xId2A0pDkwrjkwkmmghameBil9G1s4rPmVSffaRDQHxq2X0pQCXGo+FFtPA/8SD9xPjeXxI5X2I27RuX6L7uyYabyn68crytcrlrxn/PxcUVHjPAI4x4FX5Pny8vKGr0z/htwVgNXTnAQmWpLJFsC5D0zUHo1GASyU+8BVhedfcMEFABbKfeCSj3CLzrAz94Hv7MkXwLkNvGpLG4Dlch/YZwVgodwH7jkAq6cAZgOw3DkBzrcCsFDuA3d0dIQ3Pg5godwHtioDsJA3gE9PAbBQ7gOPGDFi+EWbACyU+8B+vz+YiRfAOQ5MFAgAWCz3gY9MGTJk2jEAC+U+8Fee64ytuwnAQrkPXGQ+TACwUO4DFx8hOlQMYKHcB946Ys7NI7cDWCj3gal286u1mXwBnNvAu/wG4VsAFsp94GtiRO1FABbKfeCJ5gN+ipbKfeDr9hHtmQZgodwHfmv07NmjfwtgodwHprqqLfWZfAGc48A9BmD1FMBsAJYDMBeA5QDMBWBuFIDlAMwFYDkAcwEYwCoAc6MALAdgLgDLAZgLwABWAZgbBWA5AHMBWA7AXAAGsArA3KhBAFyWSH0OYDYAyw0Q4AhXaye7Ob32qN5xrXHNgW2aA6P8wpmBrfz2LuBuA5O2sbUKA9Mi3WvTpjkwbg1swTs4pUH/Dgaw+hzAbACWAzAXgOUAzAVgbhSA5QDMBWA5AHMBGMAqAHOjACwHYC4AywGYC8AAVgGYGwVgOQBzAVgOwFwABrAKwNwoAMsBmAvAcgDmAjCAVQDmRgFYDsBcAJYDMNdZBs6ADGAuAMsBmAvAAFYBmBsFYACrACwHYC4AywGYC8DcKAADWAVgOQBzAVgOwFwA5kYBGMAqAMsBmAvAcgDmAjA3CsAAVgFYDsBcAJYDMBeAuVEABrAKwHIA5gKwHIC5Bg9wY/myHx4jqn5keY3zCGAvAXfG6UAlNZTHj66yHwHsKWCjvRtp9waiRfYj0a6nn65h/xLywf43vndtT2vg/o3vp5bdfYh2vEK00H4EsLeAiY7/c7d3MG7RHrpFtxOdWWz8qBU/Umk/AthTwH9cVrHU/Cm6oqLGeQSwl4CFAKw+BzBbbgKXpaS2AZgNwHIA5gKwHIC5AMyNAjCAVQCWAzAXgOUAzAVgbhSAAawCsByAuQAsB2AuAHOjAAxgFYDlAMwFYDkAcwGYGwVgAKsALAdgLgDLAZgLwNwoAANYBWA5AHMBWA7AXADmRgEYwCoAywGYC8ByAOYCMDcKwABWAVgOwFwAlgMwF4C5UQAGsArAcgDmArAcgLkAzI0CsKwMYC4AywGYq7fAZempHQBmA7AcgLkALAdgLgBzowAMYBWA5QDMBWC53gHzfwm5Z//Gdwa420BuZ3ID+W98xzt4EL6DBw8wYwvgngKwHIC5ACwHYC4Ac6MADGAVgOUAzAVgOQBzAZgbBWAAqwAsB2AuAMsBmAvA3CgAA1gFYDkAcwFYDsBcAOZGARjAqhwBlmwB3FMAlgMwF4DlAMyV0RbAPQVgOQBzDTTgdGUAcwFYDsBcAAawCsDcKAADWAVgOQBzAVgOwFwA5kZ5A7hnWwD3FIDlAMwFYDkAd6VpC+CeArAcgLkGJHCKMoC5ACwHYC4AD17gbHAB3EMAlgMwF4DlBjdwFwKAAQxgAIvHApjrHAH3KnU+gNkGBHDvbb0LXP3I8pocAU6V4HAAnFZDefzoqrMLnHTt6pOBU7en5QD3ES2b1BfMArj7qUL9Cbx7A9Ei4+PHe/caC04vFOO2qrq9ZL2dwcRn3XamPmtujgS7zzjrqRW2EP8SNE5NvwLWM2qRL2JywbDecc3xgPnYpAO84xWihcbH9QsWnOxgisW5rUydmsfFSHdgTPNA3YExDw20Dgxm8w4+i7fo1AbCn/DPXOIW3VM58T24sTx+pBLAyXkLmKorKnLlp2g+AOsGYBWA2QAsB2AuAMsBmAvA3CgAywGYC8ByAOYCMIBVAOZGAVgOwFwAlgMwF4ABrAIwNwrAcgDmArBcfwCf+VTzawY1X8RnpzQHBjSB605oDmzRBNYe6NcFPlGnd1xTi+7AM9aHLIHZ9pX17fy0Dn/N5YGnZ7o8MFDS6fLEyc0uD7zxVNdzAGfb4AL+5Lm+nZ/WZz9zeWDzky4PjD4ed3ni4xGXBz7V1PW8j8BooAdgj9cX4Fj5Hb82bll3LF36B3cWowYm/XEZF3JxeVauLs7M3QWmXcK+AMcbXzKBl7qxsKSBSX9cxoVcXJ6Zu4szc3eBaZewb7doC/jbD68JurC0xMCk/9jehVxdntuLM3N5gd0vYd+B40GqfsaFhSUGOn9cxp1cXZ7bizNzeYHdL2Gvgbc/ul4NM2p+yIWFJQa6+CaxZrqzPDv338Hk7gLdfwdH47T7xy4sLDEw6Y/LuJCry3N7cWYuL7D7JewT8OqF9z1LBx9cVnHGjZU5A5P+uIwLubk8M1cXZ+byArtfQvw62OMB2OMB2OMB2OMB2ON5FDg/mvp54zcKCr72KdGICwsK7u12UB5R+/LLxo57uN2LV8ODL8msO3D95vbOf/ymAXww/SAD+K65dRRa5ffi1fDgSzIz7LZPKrrpGNHGqyavybO2bZnsANu7jIM2XzX9sTw6OrROneZ7ovjyLYZ3ycT5TcZnq789eUtiwp6Z1055o99eUK/zLHDtyCP07FSqufAk/cQC7pxdaQB/eezYrfYu66DjtDqPNl5tn+ZbR/sLjfc70aqlxmdb6KNCZ0JjUS2dLGztx9fUuzwL/OrNRLEhgc3ziZot4Pvmx+x3sL3LOGjz141vz3n00l/ap/mMm/Z57bR2xvTiecZnbUTnt9sTqkeVlpaOO9Z/L6mXeRZ48xxLcdM3bOAHvm5wKWB7l3nQ3xAFjFt0vnOLNv5/QXTf+CaqnuV8Zk+omtFvL6ZPeRa4duQhWjeNakafpKcM4PtvMX0VsL3LPKjQTy8ae++c+xmFfuS3SXeWEpUlgO0J9SN2EL3Try+qV3kWmLYVT7jRuKP+avz1lcPpfZ/x66PED1n2LuOgTdP/9lEDuG3Z2IK/qGi3SWNlc8oeSgDbE+h3N0wYd1v/vqre5FHgrkJEG+b184T+zPPA/1oyadbxfp7Qn3keeLAHYI8HYI8HYI8HYI/3/3g3fDiN+N7TAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R \n",
    "d <- res$log2FoldChange[!is.na(res$log2FoldChange)]\n",
    "w <- wilcox.test(res$log2FoldChange,mu = 0, alternative = \"less\")\n",
    "\n",
    "t <- t.test(res$log2FoldChange, mu=0)\n",
    "\n",
    "h = ggplot(res, aes(x=log2FoldChange)) +\n",
    "        ggtitle(\n",
    "            'Decitabine_treatment',\n",
    "            sprintf(\"wilcox.test (p.value): %.5f \\t t.test (p.value):%.5f\", w$p.value, t$p.value)\n",
    "        ) +\n",
    "        ylim(c(0,1100)) +\n",
    "        geom_histogram(binwidth=0.3)\n",
    "\n",
    "ggsave(\"plots/peak_Histograms.png\", plot = h, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/peak_Histograms.pdf\", plot = h, device = 'pdf', dpi = 300)\n",
    "h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAMAAABKCk6nAAACx1BMVEUAAAABAQECAgIDAwMEBAQGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUWFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgpKSksLCwtLS0uLi4vLy8wMDAyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg6Ojo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVHR0dISEhJSUlLS0tNTU1OTk5PT09QUFBRUVFSUlJUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1fX19gYGBhYWFiYmJjY2NkZGRlZWVnZ2doaGhpaWlqampsbGxvb29wcHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGCgoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyOjo6Pj4+QkJCRkZGSkpKTk5OVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWmpqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///9rtMqIAAAS9klEQVR4nO3c/4MU9X3H8a0UMGqjopaUhGoMJhGp1JoapMbYaNG2ETGk6RdANBVPPLrQWBswCaFJFEzSJLS10WBCzkYULFUPbb5YqxfMoQlqBHre99u927vd23n/EZ397HzmZmY/85n5LDvszntfzx+Wu5l5z8znHux6uBwZQqzLNPsGULIBmHkAZl4jgJfvCGxY+o2QHeiUZwB8QyYz98LVP6vdse1Joiu/5tkggSs7QvIdH3NP/COQmwnwyldfefym33xMuVMJrMk9flJ7puhrIX0mwGsrj5+ZP0XFu+ef9fFfEJU6F8y99OnKK/GnMpnMxfTtJe9638aiDXz/DXMX7a++RC+974azP/Sf5M5Uqx6/9HPXn/k1ucMZdvZ88cbTL+t59tIzPz05M+qcqnoEipcx8CuZbrr7D//76D0XFejuhU+88cR/CUfxrPr6U28efH8F9Zyvv373WQNV4AteoB2/U3RnnMTxS8952srJHXK4uuf8b/18xe999NlD83eSOypPhWewQcbApcy/5ee8Zn/w/n252U+IHS5wpX9eZvPcQmR9cEcV+C6iqdMOyxl5sirjeiLfjspwdc86ohcyB4juXDVzhHMqAJtUB/DDL2VEO/8nkxM7XOCf3/yh+WdfaPN82f7k0+uqwLvsj9/9jJyRJ6sy2g/uDjlc3fMg0a8y7xB98bqZI5xTAdikOl6iD/0sMyI2vBgALi64/X+PfXOBA7zaAa58w/Xug3JGVmW098kd7rC752hmmOhL17pHyFMB2CTzb7LeM5Wb/V2xwfsSveyfiH6Rsf8Te++CwEt0VUXOyCrHi31yhzvs7pHA7qgEFkegeJn9ManH/mPSXqJ7zn/kl4duf5M2LHz8lz86KBxX//mxodHTf0Qvz1/gfJN15qAX2J1xqhxf3efscIfdPRLYHZWnEkegeJn9j445F65+0f6o9PcLZy/8q1Ga6vjtuYurwC8vmX0xfe+i9y7bXgHedv2cRU+RD1jOOInjxT65Qw67e1xgeYQ8lTgCxQv/L5p5AGbeKQY+Wv0jz7Wn9qrtHJ7BzGsAMN4UbOUAzDwAM88AeGn2mvMW7acvL7Y/zp1+0H1zsAL8u3sr/5e6J/CmIGp+JsC/9Tx956yBX//Gq0T/Pr/svr/nBQ68KYiangnw39gPl++gq+4luun26rbK+3se4OCbgqjpmQB/xX74zDr6yqWUm/vczJuDHuDgm4Ko6ZkAb7cfbl1Hx087/PACa+b9vQrwRTbwZKYn+KYganomwJ8ksj5ga179uRs/63l/rwL8B98k6sn0BN8URE3PBPjsXa/fdeYg0QMfmPs8kfv+XgX4rqvHR/7E/iYr8KYganomwM6bgNQ3632W/Yt8f68CPHTjOZf8wAYOvCmImp4JcPTfdkYtF4CZB2Dm4e1C5gGYeQBmHoCZB2DmxQceHVY0mVNt1TRhePxY0XBg1PSOJvOGA6ZLGDVegukdFdQDhsBD/YqKY6qtmiYMjx+eNh0wvaOpxJdQNhwYyhkOTKoHACwCMID9AVgXgKMDsHYAwDIAiwAMYH8A1gXg6ACsHQCwDMAiAAPYH4B1ATg6AGsHACwDsAjAAPYHYF0Ajg7A2gEAywAsArAGeJWd2b0AODoAawcALAOwCMAA9gdgXQCODsDaAQDLACwCsFNurLYKsGKzpimzw8fGy4YD+QnDgVLBcMB0CXnLdMD0jtRLGDIEVv3oSgXY8KcsDH8qI1c2HBgbNxwomg6YLmHMdAmjxktQ/jTNIF6iRXiJBrA/AOsCcHQA1g4AWAZgEYAB7A/AugAcHYC1AwCWAVgEYAD7A7AuAEcHYO0AgGUAFgEYwP4ArAvA0QFYOwBgGYBFAI4ANkMGcHQA1g4AWAZgEYAB7A/AugAcHYC1AwCWAVgEYAD7A7AuAEcHYO0AgGUAFgEYwP4ArAvA0QFYOwBgGYBFAAawPwDrAnB0ANYOAFgGYFH7Ab++cdOmPqKuLZuPy0cAe0o98EiBnnuABjqsI1udRwB7Sz2wXfcu6t5NtNZ5JBp+++2BodoksGJXSIX4h4rGpg0HRvOGA0XTAdMljJYNB0bGDQemlAMD4cD5O07Qvj1Ea5xHoh3XXHPUqk0CK3Y1Kkrw3KfmAs1awngocDH7Ewo8g/ESPVPqX6KtL+y3Hwc7rN5tziOAvaUeuPvmbNZ+4nZls8flI4A9pR44JAA7AVgXgKMDsHYAwDIAiwAMYH8A1gXg6ACsHQCwDMAiAAPYH4B1ATg6AGsHACwDsAjAAPYHYF0Ajg7A2gEAywAsAjCA/QFYF4CjA7B2AMAyAIsADGB/ANYF4OgArB0AsAzAIgAD2B+AdQE4OgBrBwAsA7AIwAD2B2BdAI4OwNoBAMsALAIwgP0BWBeAowOwdgDAMgCLAOyUz9UmgRW7QpqKf6hoomw4MF4wHCiZDpguYdwyHZg0HCgpB4YNgUcU/5iaBI7/j7LhX7qLMWB6R8b/0h1eonVxfYkGsBOAdQE4OgBrBwAsA7AIwAD2B2BdAI4OwNoBAMsALAIwgP0BWBeAowOwdgDAsgYCx0cGcHQA1g4AWAZgEYAB7A/AugAcHYC1AwCWAVgEYAD7A7AuAEcHYO0AgGUAFgEYwP4ArAvA0QFYOwBgGYBFAAawPwDrAnB0ANYOAFgGYBGAAewPwLoAHB2AtQMAlgFYBGAA+0s98HTHygNEhZWdnS8RdW3ZfBzAvlIPbA0+UgHurHw80GEd2QpgX6kHJhLAt27ckaPu3URr7S0nDh/uH67NC6zYrWgy3mFuubLhwNi44UDRdMB0CWOmSxg1XsKEauugHtjKUdeDtG8P0Rp7y84VK96Yrs0LrNityIp3mFuZTAfKhgOW8YDh8c1aQl4PbDeywX0G4yV6Ji4v0RZ1b6fBDqt3G4B9pR/4/jXrH6KeOzdl++zvorNZfBftL/3A6gDsBGBdAI4OwNoBAMsALAIwgP0BWBeAowOwdgDAMgCLAAxgfwDWBeDoAKwdALAMwCIAA9gfgHUBODoAawcALAOwCMAxgR3jCGsARwdg7QCAZQAWARjA/gAMYF8A1gXgGAFYF4AB7A/AAPYFYF0AjhGAdQEYwP4ADGBfANYF4BgBWBeAAewPwAD2BWBdAI4RgHUBGMD+uALnc7UFgOUmxZGyKc0+VRNlw4HxguFAyXTAdAnjlunApOFASTkw7AOecB/CGhmqLQAsNymOlBU0+1SNTRsOjOYNB4qmA6ZLGC2bDpje0dS4auuAD3h55WEJXqJjlMaX6HLp6lKp1H8JgGOURuB7Z2VmzZp17n0AjlEagYnu0dkC2Fs6gYmKhUIBwDFKJ/Dehe8644wzAByjdAIv/RVeomOWTuBbonwBLEsn8NYfTgE4XukEzogAHKN0AkcHYCcAA9hXawDPEQE4RukELpVK449+HsAxSiewaBWAY5Re4GNXADhG6QSeN2/eue95DMAxSifw8PBwTscLYLd0AhONjSUG7NkG4OgSAe69YvbsK18DcIzSCfxH3yhP7/oYgGOUTuDFlYcPAzhG6QRe0kv06hIAxyidwI/P+8Qfn/fESQJ7oYOHOR9OqDeHBuAYxfou+sT3f3BC5wtgWTqBDw7bhM+0BrB3O4BjFAf40mmi4uKWA47zJPcHYDcf8GWVh5P9LhrAqloD+CM/JTp0JYBjlE7gZy647roLngNwjNIJTO/s/WG/zrceYPkxgI1q3b+TFQUc4Gpv4PClAVgXgN0ALAKwObAKNoivWSKADa8AYF3tBzzdsfKA/UvXls3H5WNTgBUTMQKwWxiwNfiIDTzQYR3Z6jwC2FvqgYkqwN27idY6j0QnDh/uH65NAyx3Bw8LTCtOKrcrJmI0Nm52/HDRdGDS8PixsnJz+NJGjZcwodo6qAfet4dojfNItHPFijema9MAy93BwwLTipPK7YqJ0JGZyuXwfcppSzOgzDI8vkzxb6Y6YHxHyoG8yTO4VV6iw0Zm0r1EK6fb9SV6sMPq3eY8Athb+oHvX7P+Ifv752z2uHwEsKf0A6urE1i1PTAdsvSw3yr6xTIHVk61FLDnBtsTOHopANYFYDcAi2zg6LN6A/ApBA6crT8esO8siQJXTqwADlt5NQD7Dg1MAditJYE92wCsPiGAI786AGYMXHPRkwKOmowJ7D+LAA45s/K3RlOAIwq72fqAvWduDLBGDsAA9t0fgEMW0BTgsAt5qhM4uGrv6Co9cOD+ai4KYMXtuhsBfEqAFdvCdge/gP1NAlYo1YysArBC0Pk4bHfwC9ivB1Z8dXxncYA9F1VdyFOywJXNuX716hUXFQFYcbvufgA3BVhxaPCLqDhTpWSAPZ8DOE7Bm415aOCromrMszvkq+M7U0OAg/fmlUsE2HcxAIdMiI0AThmwd8Tza9hhyQB7ttcBHHqvAA6MABjAJw8cdm9mwKrdirOrpjkCxzx7U4HlJgADOObKABw8JM6hAE4hsMGhKQZ2zwBg3SEAThw4udFWANbdh+HXEcB1HArg9gYOnq2xwDHvMXB/NRcEsO6QcGDV2QAc/14BHPceA/dXc8FWBG6Z6gMOPZt6X3LAYjOANQEYwJ7d/Z7DdIfE22wagOsIwLLxfG0NuOlml5+edJfhrChsZfmZ/aFnU+9r0Ncq7L5Kkwqb/Igh8PBAbQ246VbKWVHYygZm9mvOoNqX7NdqYCqvsBkYwEu0YTFeoiMmkwn/DW5QAGYegJkHYOYBGDUlADMPwMwDMPMAzDwAMw/AzAMw8wDMPAAzD8DMAzDzAMw8ADMPwMwDMPMAzDwAMw/AzAMw8wDMPAAzD8DMAzDzAMw8ADMPwMwDMPMAzDwAMw/AzKsPuLCys/Mloq4tm48DuLWrE7iz8jjQYR3ZCuDWrk7gWzfuyFH3bqK19mdHf/zj/pHamr00VGmkOKGwGRnSA1s56nqQ9u0hWmN/9u3Vq98q1dbspaFKJaussCnl9MB2IxvcZzBeolu4Ol+iLereToMdVu82ALd29QH33Lkp22d/F53N4rvoFg9/DmYegJkHYOYBmHkAZh6AmQdg5gGYeQBugwDMPAAzD8DMAzDzAMw8ADMPwMwDMPMAzDwAMw/AzAMw8wDMPAAzD8DMAzDzAMw8ADMPwMwDMPMAzDwAMw/AzAMw8wDMPAAzD8DMawDwhKJmLwvJFDijeAYzCi/RzAMw8wDMPAAzD8DMAzDzAMw8ADMPwMwDMPMAzDwAMw/AzAMw8wDMPAAzD8DMAzDzAMw8ADMPwMwDMPMAzDwAMw/AzAMw8wDMPAAzD8DMAzDzAMw8ADMPwMwDMPMAzDwAM69u4K4tm48DuPWrF3igwzqyFcCtX73A3buJ1tq/vrp//ztjtTV7WUimwBmKA7xvD9Ea+9fv3nbb20VF1rRqq6ay4fEly3TA+I5abgnFxiwhZ/IMDnmJ7i+OqbZqmjA8fnjadMD0jqYSX0LZcGAoZzgwqR6IAzzYYfVuA7AvVsDUlc1qvosGcIxaG3gmADsBWBeAowOwdgDAMgCLAAxgfwDWBeDoAKwdALAMwCIAA9gfgHUBODoAawcALAOwCMAA9gdgXQCOrpWAj/UZ3kve8Pi+tw0HhkYNB479n+GA8RJ+bTgwZPpb7rh6CYbAym576uTmI3vlTxO+AP3tfyR8gSPXJ3wBuuex8H0ABrCuhw+f3Hxkx3YmfAF69OWEL9D31YQvQN9/MXzfSQKjVg/AzKsP2P2BluoHMz/f0rDkKV/fuGlTHxVWdna+lMwFnFMnuITezs47OhJYwnTHygPeK6mXUBew+wMt1Q88P9/SqNxTjhTouQeo0NnY03suUD11kkuw27M3gSVYg48c8FwpZAl1Abt/Hb76gedvxzcq7ym7d1Hh1o07cslcoHrqhJdwx2ACSyBygLUKdQHLH2hxPnA/bVyeU+bvOEFWjroeTOYC1VMnu4S3/o4SWIILrFVo+WdwMfsT8evIhoQuIE6d7DP4X5+U12lsiT2Dqz/QYg07H3h+vqVRuVewvrDf/rRgUff2ZC5QPXWSSyBan0tiCVXgKIU6v4uu/EBLfp38yZaZn29pWPIK3Tdns7up585N2b5kLuCcOsElUO8/2p8lsIT716x/KFIBfw5mHoCZB2DmAZh5AGYegCvNKTT7DhILwJUAzKfM5tVX7aV/+CzRwLl5+tTSy24aqgAPzyMqzSE6dPXlVzzd7HtsZO0H/DANvLfvzfkl+upfEvUTbe30AA8uPkFvLZxs9k02sPYDzhGt6KKPddHlzxLtXPbRJTd4gLvOX758+aLXmn2TDaz9gO1X5Osfp3/5s56LiX56yRB1XVsBzp1LND6H9i5r9v01uvYD/hIdOe8dGj/vL+4jenI50SoBTPOP0aNzqH/ePqIXmn2PjawNgT/ywb32r3992ptE06s+sWpDFfh7v3/L5+1vsp6/6sOLPtnse2xk7Qfc7Bs4xbXbettuwe223rYLwMwDMPMAzDwAM+//AffNvK8iLQmnAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R \n",
    "ggplot(res, aes(x=pvalue)) +\n",
    "        ggtitle('Decitabine_treatment\\npvalue') +\n",
    "#         ylim(c(0,1100)) +\n",
    "        geom_histogram(binwidth=0.01)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## volcanos "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n",
      "R[write to console]: Saving 6.67 x 6.67 in image\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAeAAAAHgCAIAAADytinCAAAgAElEQVR4nOzdeXgUVfY38G9V790JWSFhh7AlJCAIAkqAGWHUAWVAZ5Sfsg3DoKAoo6CA+6szCrgw4BgFFxAXVMQFVEAUZB8QySBLQEDCGiFk772q7vtHxU6n09k63V3VnfN58vh0qrtunY7pw82te8/lGGMghBCiPrzSARBCCPGPEjQhhKgUJWhCCFGpKEzQv/vd7xYvXtyoU/r37//GG28EpSlCCAkWtSTom2++meM4juNMJlPnzp0nTJiwf//+wJq66aabevbsKT8eNGjQK6+8EnBU3k0FRRPjCWlryl6FEFKTWhI0gDvuuOPo0aN79+5dsmRJRUXFoEGDPvnkkwDamTt37g033BCUkILYVL2cTmd4LkQIiRhMHUaNGnX33Xd7H5k8eXJqaqrT6WSMuVyuhx9+ODU1NTY2dsSIEXl5efJr3G73vHnz2rZtazQas7Kyvv32W8bYsGHDXn75ZcbYXXfd5XmbXbp0YYy9/fbbV111ldlsbt++/SOPPOJyueR2+vXr9/zzz48aNcpoNHbr1u2bb76Rj3uakl/zzDPPjBo1Kj4+Pj09/bvvvpOP1xZbTTXj6dev31NPPfXHP/4xJiZm6dKltTXlN2y/rS1cuPBPf/qTyWTq3bv3Tz/9tG3btqysrJiYmAkTJjgcjroD9vsGa16FEBI26k3QBw8eBLBjxw7G2MMPP3zdddft2bPnl19+mTt3bufOne12u3y8Q4cOX3755enTp7/88sutW7ey6ll14MCBS5cu9bS5bNmyTZs25efnb9myJS0tzTv5JiQkLFu27MSJEw8//HBsbGxhYSGrkaBbtWq1e/duxtjLL7/cpk0bOVHWFptfPvHI1/32228lSSovL6+tqdrCrtlay5Yt33rrrcOHD48ePbpPnz7Z2dnbtm3buXNnampqTk6O/LLarlLbG/S5CiEkbNSboN1uN4B33323oqLCYDD8/PPPnqfS0tI2bNhQXl6u1+u//PJLn6bqSNDe3nzzzSFDhsiP+/XrN27cOPmxJEk9evSQW/BJ0LNnz5YfO51OnuePHDlSW2y1vc2aKXXGjBny4wY25R12zdbuuece+fHu3bsBbN68Wf72wQcf/L//+7+6r+L3Dda8CiEkbLRhGEVpCo7jfv75Z6fT2a1bN+/jv/zyS6tWrVwu19ChQxve2pEjR55++umDBw+WlJQ4HI74+HjPUwMGDPBcccCAAceOHat5eteuXeUHer0+Njb28uXLTqfTb2wNDykjI0N+UNvbrDtsH71795YfpKSkeH+bmpr6008/1X0Vv2/QEx4hJPzUm6CPHj0KoFOnTqIoAigpKYmLi/N+wY8//tioBt1u9w033DB27NjVq1e3bNny66+/fvLJJ/2+ktWy/F2j0Xh/K0lSbbE1nNFolB/U1lTDwwag0+nkBxzH+XwrSVIdV6ntDQb2pgghQaGiWRw+XnrppdatW/fv379Hjx56vf7rr7/2eUH37t31ev22bdvqaESv18spCcCpU6fOnz+/aNGiq666qk2bNufPn/d+5d69e+UHjLF9+/b16NGjIUHWFltD4mlgU3WEXUdrwQo4sKsQQoJCRQm6pKQkLy/v0KFD69atGzt27Lvvvvvqq6/q9fqYmJgHH3zw/vvv//DDD0+dOrVr166ZM2eeOXMmJiZm5syZ06dPX7du3alTp+SbhD5tduzYcceOHRcuXCguLm7durXJZPr2228BHDx4MCcnx/uVGzduXL58+cmTJ+fOnXv+/PmJEyc2JObaYqvt9d7xNLCpOsKuo7VgBRzYVQghwaH0IHilUaNGyfEYDIZOnTqNHz9+//79nmfdbveTTz7ZoUMHvV7foUOHKVOmlJaWMsacTuecOXNSUlKMRmOvXr22bNnCqt/Z+9///nfVVVfp9Xp5ithHH33UuXPndu3aDRky5IUXXmjbtq38sn79+j333HN//OMfDQZDt27dNm3aJB/3uUm4fPlyT0hxcXHy5WqLzS+feHzarK2p2sKuozV5WLm4uFj+dtGiRcOHD6/7KrW9QZ+rEELChmNUbpQQQlRJRUMchBBCvFGCDr7Tp09zNYwYMULpuAghEYaGOAghRKWoB00IISpFCZoQQlSKEjQhhKgUJWhCCFEpVdTisFqtgS0mZozJRScil3yTNgreBb0FxdFbiAKxsbHePwFVzOKYM2dOVlZWACe6XC69Xh/0eMLJ5XJxHOepahSJJEmSJEmrVcU/9oGRJMnpdJpMJqUDaZIo+DjY7Xa9Xu9TtCuyiKLIcRzPN3pwQpKk//znP1u3bo2JifEcVMWHqkOHDpMmTQrgRKvVarFYgh5POFVUVGg0mohODaIout1uT1m+Jrp06dLZs2f79esXlNYaSBCE8vLyhISEcF7U7XZ///33QZwgHwUfh5KSErPZHNH/zLhcLp7nA+ivyAna5yCNQRN1sVqtFy9eVDqKcBBF8dSpU0pHQVSNEjQhhKgUJWhCCFEpStBEXWJjYzt06KB0FOGg1WppRzFSN1XcJCTEIzk5OTk5WekowkGr1Q4ZMkTpKIiqUQ+aEEJUihI0IYSoFCVooi6CINhsNqWjCAfGWFlZmdJREFWjBE3U5ezZs999953SUYSD0+lcvXq10lEQVaMETQghKkUJmhBCVIoSNCGEqBTNgybq0r59+5SUFKWjCAeDwTBu3DiloyCqRgmaqItWq43oyqUNx3FcixYtlI6CqFqAnwRRFOfNm3fmzJm///3vw4cPlw9+8cUX+/fvZ4xNnz69devW3q+v4ylCCCF+BTgGzfP83Llzx44d6zly5cqVHTt2PPXUUxMmTFixYoX3i+t4ihBCSG0C7EFzHJeYmOh9JC8vLysri+O4bt265efnN+SpvLy8c+fOAXA6nU6nM4AwRFEM7ET1EEWRMRbR70KSpCD+j7hy5cqFCxd69eoVlNYaSJH/C4Ig/Pe//x08eHCwGoyCj4MkSW63Ww3bPAVMEASO4wLYw0+SpJoHgzbYV1FRYTab/V6ptqdyc3O3bdsGoE2bNg6HI4CLyp+rACNWB1EU5S2jlA4kcIwxxlhgu0rWVFhYePLkyW7dugWltQaS30Jgv4QBczgcP/30UxD3jomCjwNjzOVyCYKgdCCBkySJ47gAdlYMbYKOiYn59ddf5cc++3HV9tS4cePku9hLly6Ni4sL4KJRsMcPbXnlIzY21mKxBPb7EDB5y6swX9RgMJhMpiBeNAo+Ds18y6uaB4M2DzojI+PQoUOMsePHj3fq1AkAY6ykpMTvU4QQQuoVeA96wYIFJ0+e1Ol0J0+enDZtWmJiYnZ29hNPPAFgxowZAGw227x583Jycmo+RUhtLBZLM5nno9Fo0tLSlI6CqBqnhkGrpUuXzpw5M4ATo+BvOhriUANFdvUOuij4ODTzIY4BAwZs3bo1JibGc5CWehNCiEpRgiaEEJWiBE3URZ4Jq3QUYRLp05ZJqFGCJuqSn5+/ceNGpaMIB4fDsXLlSqWjIKpGCZoQQlSKEjQhhKgUJWhCCFGpZlF4l0SQNm3a+NThilZ6vf7WW29VOgqiapSgiboYDAaDwaB0FOHA83xycrLSURBVoyEOQghRKUrQhBCiUpSgiboUFRXl5eUpHUU4CIKwd+9epaMgqkYJmqhLaWnpiRMnlI4iHARByM3NVToKomqUoAkhRKUoQRNCiEpRgibqYjKZmsnkM57n27Vrp3QURNVoHjRRl9TU1NTUVKWjCAe9Xj9y5EiloyCqRj1oQghRKUrQhBCiUpSgCSFEpShBE3X55Zdf1q9fr3QU4eBwOJYtW6Z0FETVKEETQohKUYImhBCVogRNCCEqRfOgibqkpqbGxsYqHUU46PX6UaNGKR0FUTVK0ERdTCaTyWRSOopw4Hm+bdu2SkdBVI2GOAghRKUoQRNCiEpRgibqUlJSQvWgCZFRgibqUlxcTDuqECKjBE0IISpFCZoQQlSKEjRRF4PBEBcXp3QU4cDzfMuWLZWOgqgaJeiQKS7G3LlIT4fRiNRUDB+OTz6BJAHA1Kno39/PKd7Hp04Fx4HjoNWiUyfcdx+Kiyufeu01DBuGxETEx2PIEHzzTVjeT5i0adNmyJAhSkcRDnq9fuzYsUpHQVSNFqqERkEBBg+GzYZHH8XVV0Onw549eOQRZGYiPb2hjWRmYs0aiCIOHMD996OgAGvWAMC77+LGGzF/PgwGLF2KUaOwZw+uvjp074YQoghK0KExaxaKinD0KDy7N11zDaZOrexBN5DRWJnNMzNx4gSeew6CAK0WO3ZUvSY7G61aYf16StCERB8a4ggBmw2ffooZM+Czt57JBIslwDbNZggCBMH3uChCFEFDmYREI0rQIXD6NFwuZGYGpzXGkJuLnBwMGgSj0ffZefNgNuOOO4JzLRXIz8/fuHGj0lGEg9PpXLlypdJREFWjIY4QYAwAOK6p7ezfD60WjEGSkJ2Nt9/2fcFzz+Gtt7BpExITm3ot1ZAkye12Kx1FODDGnE6n0lEQVaMEHQKdO0Onw6FDTW0nMxOrV0OrRdu2qFmB87HHkJODzZv9TwghhEQ+GuIIAbMZY8ciJweXLlU7brfDam1EO0YjsrKQnu6bnRnDAw9g+XJs2ULZmZAoRj3o0Fi8GNddhz598Nhj6NsXOh1278a//4316ysnZtjtyM0FwNtsvEaDuLhGTL+bNg3vvou33oIkyY0gMREdOoTorYRZq1atjDWH2qORXq8fMWKE0lEQVaMEHRqtW+OHH/D881i8GGfOID4eWVlYtAg9elS+4MgR9O0LwCx/26MHGl4h6IMP4HDgzjurjkyahBUrgha8oiwWiyXguS4Rhef5tLQ0paMgqsYx+Y6Wol5++eVp06YFcKLb7dbpdEGPJ5zcbjfHcVptBP9LyRgTRTGi34J8Z9JgMCgdSJNEwcfB6XRqtVqNRqN0IIETRZHjOJ5v9OixJEm///3vt27dGhMT4zmoig+VVqsNrNNktVojsrd1/DjOncP11wOoqKjQaDQRvcmTKIputzuixyUEQRAEISJ/l7xE6sfBi/yLpNfrlQ4kcC6Xi+f5APorkr9VbHSTMOwuXcKNN+LYMaXjUKny8vIzZ84oHUU4iKJ49OhRpaMgqkYJOrwcDowZg+HDMX260qGoVGFh4cGDB5WOIhzcbvf27duVjoKoGiXoMGIMU6ciNhavvaZ0KISQCKCKMejm4okn8OOP2LULkXw/jRASNpQpwmX1arz+OnbvRny80qGomlarNZvNSkcRDjzPt2jRQukoiKpRgg4lxmC3Q5JQUoJ77sGGDejSRemY1K59+/bt27dXOopw0Ov148aNUzoKomqUoEOppARFRdBoYLXi4MGoWexHCAkPStAhI0koKIC8vR5jSEpSOiBCSIShWRwhw3FV+6eIIhq/sogQ0sxR1miCulfJcxzS0lBcjNJSJCUhktcKhtPZs2e/++47paMIB5fLtXr1aqWjIKpGQxyBKinBhQtgDGlptSZfkwk9ewLBKN7fbAiCYLPZlI4iHCRJKisrUzoKomrUgw6Iw4HCQsTHIyEBp07V1ZXmOMrOhJDAUIIOiCjCU3CL5+sZ6yCEkIDQEEdAjEbYbOA4iCJSUugGYBAlJyf37t1b6SjCQafTDRkyROkoiKpRgg6IRoP0dNjt4Hm6+xdcsbGxsTU3YIxGGo0mIyND6SiIqlGCDpRGA6+62oQQEnT0tzkhhKgUJWiiLlar9eLFi0pHEQ6SJJ06dUrpKIiqUYIm6nLp0qX9+/crHUU4uFyuzZs3Kx0FUTVK0IQQolKUoAkhRKWiK0GXlWH2bKSlwWhESgquvx5r1lQ+lZ2N8eOrvXjHDnAcDh2q58SpUytXA2q1SE5GdjYWLoTVWq2pX3/F1KlISYHRiJ498fnnoX6jUYzneZ1Op3QU4cBxnMFgUDoKomrRNc1u4kTs349Fi5CVheJi7N2Ls2eDcGJmJtasgSShqAg7d+LFF/Hmm9i+Ha1aAUBZGbKzkZKCnBy0bYvz55GQEJJ31zx07NixY8eOSkcRDgaDYdKkSUpHQVQtihK0240vv8SSJfDsUtHAZVr1nmg0Ij298nF2Nu64A336YN48vPkmACxYAFHEt9+CekOEkKCKoiEOnQ6tWmHLFt/xh6Cf2KkT7roLa9dWluD47DNcfz2mTEGrVsjKwlNPweVqdPCEEFJDFCVoAO+8g927kZSEa6/F7NnYsSNUJ/bqhZISFBcDwMmTWLUKSUnYsAHz52PJEsyZ06R3QQghAKItQQ8fjtOnsXUr7rgDeXkYOhQPPhiSE+W+s1xHlDF07Yp//xtXX40778TcuXjtNQhCk99MM3XhwoXt27crHUU4uFyuTz/9VOkoiKpFV4IGoNFg0CDMmoX167FgAV5+GefOAUBcHEpKqr1S7v/KewbWcaJfhw4hIQHx8QDQujUyMqqKPmdkwOVCYGvhvGeMdOqE++6rDNL7Kc+XpxJIHVNQtm3D6NFo1w4ch8WLAwkp7JxOZ2lpqdJRhIMkSZcvX1Y6CqJqUXSTsKauXQHg8mW0a4f0dKxeDZcLen3ls//9LywWtGtXz4k1nTmD997Dn/9cmZQHD8bBg1XPHjsGvR6tWwcYszxjRBRx4ADuvx8FBVXZtmdPvPde1Ss9BanrmIJSUYGsLEyciIkTA4yHEKKcKErQgoBrrsFf/4o+fZCYiCNHMHcu0tLQqxcA3HcfcnIwbhxmzUKLFti2DS+/jIcfBsfVcyIAhwN5eWAMxcXYuRMvvYSUFPzrX5XPPvQQBg3CnDmYMgVHj+L553HffdAG+oP1zBjJzMSJE3juOQhCZWsmE/r08X193VNQRo7EyJEAMHlygPEQQpQTRQlao8GYMVi9Gk8/DasVrVvjxhvx6KOV2a1zZ+zciccewx13oLwcXbrg+ecxY0b9JwI4fBgZGdBoEBeH9HTMmoV7760aYbj6aqxbh3nzsHQp2rbFffdh/vzgvCOzGYJQlaD98kxBmTgRFktwrquohISEdM+kxqim1WoHDBigdBRE1aIoQXMcnnwSTz5Z6wv69sWXXzb6xDfewBtv1HPpG2/EjTc2ONAGYAz/+x9ycjBoEIzGyoP791fb3nD4cMildt55B5MnIykJffti8GCMGYPs7GAGE17x8fHx8uB+tNNqtX1q/klEiJcoStDRYf9+aLVgDJKE7Gy8/XbVUz5j0J5tR+QpKPv2Yc8ebN6MoUMxaxZeeimsYRNCQoAStMpkZmL1ami1aNsWPjs/+R2DlslTUORZKIsW4eGH8eCD/u9wEkIiR9RNs4t0RiOyspCe7pudG84zBSUy2e32wsJCpaMIB0mSzp8/r3QURNWoBx057Hbk5lY70qsXGKtrCorVip9/BgBJwvnzyM1FbCy6dFEg+AYrKCg4fPjwzTffrHQgIedyub788stp06YpHQhRL0rQkePIEfTtW+3I5ctISqprCsqBA1Wz7l54AS+8UHVrkRCiepSg1aSO6SJ1TCapYwpKdnblqnRCSASiMWhCCFEp6kETdencuXPnzp2VjiIcjEYjDUCTulEPmhBCVEqtCXrqVPTvX/VYrt+m0aBdO0yahAsXql4ZXbXcCCHEI0KGODy7Ap46hQcfxF13YcuWyqeolhuphSRJPK/WLgghDRC0BH38+PEVK1YAsNlser1+4cKFnqccDsfkyZPT0tIA3H777YHUH/DUeOvZE8ePY84cOBwwGqmWW/QpKCg4ffr0oEGDmtIIY8xms8kPzGazOtO0y+XavHnzSPlXlBB/gpagu3fv/q9//QvAmjVrtDWqr6Wlpf3LU5+zKU6fxtq10GorqyFHWS03xlBSgosXkZqKuLiqis/NSVBWEoqiqNFoOI6TH6szQUuSdK6OTSEICcUY9I4dO4YOHepzMD8/f+7cuYsXL66oqAikUbmEkEZTWTV0zhzodJVPBbwPoQqVl6O0FPHxqKhAWZnS0RBCFBbkMeizZ8/GxMQkJiZ6HzQYDK+//npMTMwXX3yxatWq6dOny8eff/75NWvWABg5cqRPpynG4dAKQklhofxYl55etmwZ53Lp16/X/fe/pTNm4LfX26+6Cj/8oD1wQLd/v+777/UvvWS/+27rM894t5bEmM1qtau4woPVagXAFRdzDodcU5Q5nUwUlY6rEQL8p7eG4uLisrKyJnaiGWNOp5PjOMaYXq9vYA86zDVAnE5nRUVFcC9qt9uD2Joiyppr10SSpJoHg5ygv//++5rdZ47jYmJiAAwbNuwZr9R5zz33TJgwAcDHH3+ckJDgfQpvMECjkQ/yBgPM5hbXXQcAv/sd/3//l/TPf0pLlgCw2WxmsxkA/vAH/OEPmDtXeuEF09y5+rlzvWu5cRxnMpmM1S+hEjabjed5o1z02WDAuXMwm+F0IjkZLVooHV2DiKIoCILBYAhKa4wxg8GQEIz/WZIkcRzHeRfRroUoilartUV4f+CCIFx33XVBeaeyqo9DxCorKzOZTDrP38cRyO128zyvafz4ZDgS9O7duxcsWCA/ZoyVlpbGx8c7HA6DwcBx3KFDh1p7bdbnKc2u0Wh83w/HgeMqD3o/BvDUU+jdWzN9OrKy/PwguncHoCkqQseO3od5nlfnkC7HcVXvIjYWHTvC5YJOB7MZDcgsKiFJUgC/kX61bNmyZcuWQWmq4SExxhr1+qDQaDRNvBfqI7C8oCrVPg6RSb7nEcBb8NuTCGaCPn78eJs2bWJ+2wvKZrPNmzcvJyfnxIkTb7/9ttFo5Hl+5syZTb1Mz5644QY8/jg+/RSCgL59o6mWG8xmRHgniBASLMFM0N27d3/00Uc931oslpycHABZWVkvvvhiEC+E2bMxfDj27UNGBtVyI4REK7UuVPGu3Fazitv111cWabNaqZZblHE6nQ6HIy4uTulAQk6SpKKiouTkZKUDIeqlxvmhRBk//FC5pN7ny7OFq/f6ewC5uUhNxV/+AqezrgX3AIqLMXcu0tNhNCI1FcOH45NP4H1L5MQJaLXo0QPAhQsXtm/fXnVFjsNDD1W98vRpcFzVZMrXXsOwYUhMRHw8hgzBN98E/8cSMi6Xa+3atUpHQVRNrT1oEn6ZmThwoNqR48cxfjwmTPDz4i1bMGYMxo/H0qVgTD9uHA4c8L/gvqAAgwfDZsOjj+Lqq6HTYc8ePPIIMjMrV4cCeOMNTJiAzZvx/ffo0KHahYxGvPIK7rsPfkvcvfsubrwR8+fDYMDSpRg1Cnv24Oqrm/qjIEQdKEGT3/hsSltUhNtvx/XX+6kz9fHHmDAB8+fjiScAwOHgv/661gX3s2ahqAhHjyI1tfLINddg6tSqHrQgYMUKfPghUlKwfDmqz2HHwIFwuTB/Pj74wE/M3uuSsrPRqhXWr6cETaIGDXEQf9xu3HYb9Hp89JHv9MT//Ad33YUlSyqzMwCdjrVsiS1bYLX6tmOz4dNPMWNGVXaWmUxVS/O/+AImE4YOxaRJ+OQTvrTUt5EXXsCHH2LfvnpiFkWIIoI0RY8QNYj2BO13bLSiwv9gK8dhxIjKE71GRQHUc0okj4T6d889OHIE69f7Lpb56Sfcdx+WLEH1SvPuN9/0v+D+9Gm4XMjMrOtay5ZhwgRwHDIykJXV4fvvb7zxxmovuO463Hor5sypJ+Z582A24447GvYOlWc0GidNmqR0FETVon2Iw28xUrO5arD1q6/w6KPYtQsmEwDExlYe9x4VHTasnlMmTYqqkdAFC/Dee9iyBZ06+T7VsSNiYvDCCxg50nuwWPr973H6NPbtw5492LwZQ4di1iy89FLlLJo6Vtzk5+Obb/DKK5XfTpzILVume+AB35c9/zx69sS6dZUz3Gt67jm89RY2bUL1MgMqF6zllyRqMRVYsmRJYCdWFBczp5NJkv+nXS6m1bJXX62riVWrGMDKy6sddLtZSgrbupU98gi7664GneJ9bkICe/rpBr6F8vJym83WwBeHw9q1jOfZe+/5eepvf2P9+rHiYjZwIOvYkZ08KR8WBMFut1d75cKFDGBnzzKrlel0bP78Wi/3+OMMYBpN1RfAdu2quuKwYZWPZ85kGRnsxAkGsO3bqzXy6KMsMZHt2xfQG2aMMbfbXVRUFPDpKlFRUaF0CE1VXFzsdDqVjqJJnE6n2+0O4ERRFPv161dePbFE8hBHWRn/yy/Iz0dRkf/5zp5ipDXHRutWfVQUxcWNODeiR0J//BHjx+Pxx3HnnbW+Jj4e33yDDh0wdCiOH/f/mq5dAeDyZZjNGDsWOTm4dKnaC+x2WK0QRbz9Np56Crm5VV833YTly/20+cQTuHABb75Z7SBjeOABLF+OLVuqTQEkJCpEbIKWJJw/z2JjYbGgtBQOh/+XBVaMtPqoKFatakRgkTYSWuXCBdxyC3r3xpgx1TJmbi7c7mqvjI3Fhg3IyMCwYTh8GIKgHzgQS5Zg2zYcOoSPPsJDD1UtuF+8GHFx6NMHr76K3bvxww9YuhS9euHsWXz9NS5cwLRpyMryfJX96U/SBx/4qbaanIy5c32nlEybhmXLsHgxJKky1DNnQvkzCia3272Z1rWSOkVsgvZR24rB4cNx+jS2bsUddyAvD0OH4sEH62lKHhX17JI1caL/Dp1f8kjo559H1khope++w4UL2LMHffv6fv36q++LzWasW4d+/fC733GHDkmjR2P1aowdi/798cgjuOEGbN1aueC+dWv88APuuguLF+P3v8fNN+Pzz7FoEXr0wPLlGDwYXvWzABQPGcIkCe+95yfCWbPgs+7ugw/gcODOO6tC9cwtUT1RFE+dOqV0FETdgjTw0iQBjkGXltp+/JEdP84KC2sdhvbhGRv1qDmgXPeoqN9TZAGNhKpuDLrx/IxBN8GpU6fWrVsXrNYaSJExaLvd/vrrrwexQRqDVgMag/5NixZS587o0AGJiQ2tzOkZG61No0ZFPWgktKa9e6kQCsO45+AAACAASURBVCFNFOHT7HQ61DFRSRBwzTW1FiP1yzMq6v1394QJ+PvfsXhxrRX0p03Du+/irbcqR0IBJCb6LlluVt56C489hlOnIG9E0BixsbEdmsePTqvVZmRkKB0FUbVI7kHXS6OpLEbqd2zUL3+johg9Goz5HxWVRfJIaPBt24YHH8TatQFkZwDJycm9e/cOelAqpNVqh3iviSekhgjvQdeN4+oqRiobPx7jx1d9+/nnfl4TEwObrdZTAARpO75ocPIk/vxnvPYagrpXCCHNU1T3oEmYXbmCP/4RDz5YVTWJENIElKBJkLjd+MtfMHgw5s5tSjOCINi8/16JXoyxZruDNWkgStCkaUQRNltlRVAAr7/exPbOnj373XffBSEw1XM6natXr1Y6CqJqlKBJE7jdyMvDr79y+fkYPx6ffQa9XumYCIkeUX2TkISazYbYWHAcNBrYbLVOQySEBIR60E3GGGw2WK3VNtlrJni+cjVKiNekyOusGK18Ic0M9aCbrKgIpaXgONhsyMgA35z+zbNY4HLh1185hwM9ewalyfbt26ekpHgfkSTJbrfzPC+Kotls5qPlJ2wwGMbRdBdSJ0rQTSMIKCysLPMvSXC5AludEal4HklJiImRJClYb1yr1WqrryQSRVE+IufoqEnQHMe1oEEhUqco+V1XDM9XjWyIYvPqPntotQ2thdI0NMRBmptmmVCCiOfRvj2Ki1FSgpQUmsMQChqNRhAEURQ9XWlCmglK0E0WE4OePZGRgbg4pUOJBkVFRXl5ed5HeJ63WCwmk8lsNnNh6aqHhyAIe/fuVToKomqUoINB3t67NozBbkdFBQQhjDFFqtLS0hMnTtQ8znFcNGVnAIIg5Mq1DwmpBf3BGHolJbhyBVotrFb06FFXLT1CCPFCPegQYwwXL8Jshl6PmJha904khJAaKEGHGMdVLeIQhGY6zaMxTCZTss/Gg1GK5/l27dopHQVRNcoXIfb99zh1CsXFKC1FYiJMpgDb+eGHypFun6/4+MoXTJ1a7eCQIdiypep072flr5gY36e0WiQnIzsbCxfCaq129V9/xdSpSEmB0YiePf1XzQ6S1NTUQc2jlrRerx85cqTSURBVo/HQUDp0CGPH4v33kZkJxpo0WTgzEwcOVDty/DjGj8eECdVes2YNABQV4eWXMWoUTp6s2h2mZ89qm8JoNL4nShKKirBzJ158EW++ie3b0aoVAJSVITsbKSnIyUHbtjh/HgkJgb8RQkiDUYIOmYIC3HwznnoKN90EoKlLOUwm9OlT9W1REW6/Hddfj8WLqw4ajUhPr3zcsiXWrMHhw1UJ2qcFb94nZmfjjjvQpw/mzcObbwLAggUQRXz7bV3bPxJCQoCGOELDbsfYsbjhBtx/f/Abd7tx223Q6/HRR9U6wh42G1auREwMAtvcr1Mn3HUX1q6tHD3/7DNcfz2mTEGrVsjKwlNPweVqUvyEkIahBB0CjOFvf0NcHF59NSTt33MPjhzB+vW+5T1//BFGI4xGWCz4z3+wZk3lGIVs//5qY9AjRtR1iV69UFKC4mIAOHkSq1YhKQkbNmD+fCxZgjlzQvCuKv3yyy/r168PXfvq4XA4li1bpnQURNVoiCME5s/H//6HXbtCMuV5wQK89x62bEGnTr5P9exZOQZdXo41a/CXv2DrVlx9ddWz3mPQcoGn2sh9Z3lYhjF07Yp//xsch6uvxrlzePxxvPgiTegmJNToMxZsRUX47DNs3BiSld+ffor587FqFa691s+z3kPJ11yDr77Cyy9j1arKI3WMQdd06BASEiqniLRujYyMqjH0jAy4XLh4Ee3bB/5GCCENQAk62BITcfRoSFr+8UeMH4/HH8eddzbo9fJOAgE4cwbvvYc//7kyKQ8ejIMHq549dgx6fdW9R0JIyNAY9G/KyjB7NtLSYDQiJQXXX185XOBx4gS0WvTo4efc4mLMnYv0dBiNSE3F8OH45JPKMqT1NttAFy7gllvQuzfGjEFubrUvt7vyNQ4H8vKQl4d9+/DIIzh8GGPGVLVgt/ueKIrVTjx6FLt2YdEiDByIlBT861+Vzz70EI4dw5w5OHoUa9fi+edx332hG98I0TxoxpjaqpXq9fpRo0YpHQVRNepB/2biROzfj0WLkJWF4mLs3YuzZ6u94I03MGECNm/G999j2LCq4wUFGDwYNhsefRRXXw2dDnv24JFHkJmJ9PT6m22g777DhQu4cAF9+/o+dfYs5AVphw8jIwMAYmLQpQuWL682S/rIEd9zL1+GvGZPPlGjQVwc0tMxaxbuvbdqJcvVV2PdOsybh6VL0bYt7ruvcgPv0DCZTKaAl/PUQhAEh8PBcZxer9fpdMFtPGA8z7dt21bpKIiqcWroVixdunTmzJkBnGi1Wi0WSxAicLthNmPJEkyf7v8FgoB27fDhh/j6a5w7h3ffrXpq3Dhs3IijR5GaWnXQbockQa+vp1mgoqJCo9EEPSWFkyiKbrfbqOKtZCoqKuS8LIqi0WisuSeLIAjl5eUJEb4AJ2gfB+WUlJSYzWZ9JNdVd7lcPM8HULhckqQBAwZs3bo1xtM3oiGOSjodWrXCli2+S5w9vvgCJhOGDsWkSfjkk8r5ZwBsNnz6KWbMqJadAZhMsFjqb1Ylzp3DffdFazVUxpinTinHqaJHQkgDqWKIQxAEa0ApzO12B3ZiTZplywx3380lJUlXXSUOGiTecovoNVPCmJMjjRvnstnQoYOpZ0/hzTfd06cD4PPyTC6Xs2tXoZYw6m4WgLxXiKTsjuAmk2nvXnH6dJf3usQGk7fcFj0j2k1TWlp65cqVtLS0oLQmE0XR4XAAkCTJZDLVrCstSRJjLFi/Sw0kCMKRI0d6B7aYyJ8gfhyUIkmSw+Fwe26rRCBRFDmOC2DnTL9JQBUJWqvVBvanWTD/phs1Cvn52LeP37OH37xZd8MNmDULL70EAPn5+O47TU6OTr7W5Mn6Zcv0s2cDkHdKNRiNhtrCqKNZACoZ4rBY8NVX/KBBuuXLMWtWY88O7hDHpUuX8vPze/XqFZTWPOQUzPO836r/giAIghDm8QGHw/HTTz9d63fGZECiYIhD/kVqtkMcNQ/SEIcXjQaDBmHWLKxfjwUL8PLLOHcOAN58E5KE9HRotdBq8Y9/4NAh7N4NAJ07Q6fDoUOBNKsqyclYtw7PPBPSSnUK4nleo9FE2Z4sJOpRgq5F164AcPkyRBFvv42nnqo2Qe2mm7B8OQCYzRg7Fjk5uHSp2ul2u/9xZ0+zKpSRgQ8+wMSJ2LdP6VAIIYBKhjiUJwi45hr89a/o0weJiThyBHPnIi0NvXrh669x4QKmTau2NGPCBPz971i8GC1aYPFiXHcd+vTBY4+hb1/odNi9G//+N9avR9eutTarTjfcgJdewpgx2LNHqYWCBoMhrnlsv8vzfMuWLZWOgqgaJWgAgEaDMWOwejWefhpWK1q3xo034tFHodVi+XIMHuy7cG70aEydivfew/TpaN0aP/yA55/H4sU4cwbx8cjKwqJFlUtaamtWtf72N5w+ja+/xvDhABAXh/Dub9KmTZs2bdqE84pK0ev1Y8eOVToKomqqmHWk/Dxo5ajiJmFNn3+ONm2g1eLiRQwdCq+JmTWpfx50vWgetErQPGiaB03q43JBp6vs5rdoEWBBD0JIk/lP84Ig5Obmnjt3zmw2X3XVVSkpKWEOiyhJr4fbDbcbGg3KyuruPhNCQse3B3358uWHHnqodevWU6ZMefXVV5999tlevXplZ2d/HqWzr4h/N96Ili0RH4927TB8OK5cCduV8/PzN27cGLbLKcjpdK5cuVLpKIiq+fagb7755r///e8nT55s4bVbx/79+1977bVz587de++94Q2PKMRoROfOlY9798bYsfjmm/DsSShJUkQvJGs4xpjT6VQ6CqJqvgl6165dmhrb3PXr12/58uXBWstLIswrr+CPf8TkyXj//aZufUsIaQzfIQ5PdnY6nSdOnPD7FGledDp89BEOHKiqEE0ICQv/szg2b97cvXv3kSNHAti9e/e4cePCGxVRmcREfPUVliyptqthaLRq1apfv36hvooa6PX6EXVv3UuaPf8Jet68eTt37pTXC1x77bX/+9//whsVUZ+0NHz8Me69F3v3hvQ6FouldfPYT4vn+eAW7SPRx3+CliSpnbxJBwCASswQABg6FCtWoLxc6TgIaS78z4M2m82//vqr/Pirr75KSkoKY0hExbw3OSSEhJj/HvRzzz03YsSII0eODB06dOLEiYsWLQpzWCSYfvgBHOfnKz6+6jV17Hu7ciX690d8PCwW9O6N116rOmvqVPTvX+1a586B47B+feW327Zh9Gi0aweOg89uALU8VV5efubMGaDOXXqjgiiKR0O0ATyJFv570NnZ2Vu3bt25c6ckSdnZ2cnhLZdDgiwzEwcOVDty/DjGj6/aUrbufW8NBtx7L7p2Bc9j40bMmAGTCZMmNejSFRXIysLEiZg4sYFPFRYWHj58uEOHDrXu0hst3G739u3bM+R9fgnxp9aKHklJSaNHjw5nKCRUTCb06VP1bVERbr8d119f1W+dNQtFRdX2vb3mGkydWtmD9p7DM3gwvv4aW7c2NEGPHImRIwFg8uRGPAVAELBiBT78ECkpWL48KhM0IfXyn6BrViaTt3QjEc/txm23Qa/HRx9Bntgu73s7e7affW9rnvvttzhyBAGVHmwczy69rVrh6quxdCkivNQcIQHwn6ArKirkBy6Xa/369T4rVkgEu+ceHDmC//4XnqX8p0/D5UJmZl1n5eejSxeIIrRaLFxYbVBi/36NVhvEJUxardZsNuOVVzBhAjgOGRnIysKqVbj//uBdRBV4nvcuqEBITf4TtKeYqVarvf322++8884whkRCZsECvPcetmxBp05VB+WC4HXPpGzTBrm5qKjA5s149FF07Ihbb618qmdPadUqQRAqa/heuoQbb2xKjO3bt28vSfjmG7zySuWhiROxbFn0JWi9Xk9LwEjd6q8qfeHCBepBR4NPP8X8+Vi1Cj7bSDdk31udDllZADBoEM6exdNPVyVok4lddZXkdssbnAdnP1zPLr0eoojdu30jJyTa+U/QnmkbjDGDwbB06dIwhkRC4McfMX48Hn8cNf8Y8ux7+8ADaNWq6rjdDklCzR06nE64XCEM1bNL7223VR2cMwfLl1OCJs2N/wTt6TJrtdoYqtce6S5cwC23oHdvjBmD3NxqT2VmQqera9/b9HRMn45hw9CxIxwObNiAVavw7LMNvbTVip9/BgBJwvnzyM1FbCy6dKnrqXp36SWk+WDV2WvBQmnJkiWBnVhRURHcSMKvvLzcZrOF9hqrVjHA/9fZs5WvKSxks2ezbt2YwcBSUtjw4WztWiZJjDH2wAOsWzdmNLLERDZwIFuxoqrlv/2N9esnCELVb8jZswxg69ZVfrt9u+8Vhw+v+ynbH/5Q3KuX71soL2cmE3v11ZD8fBhzu91FRUUharw2Tqfzgw8+CGKDUfBxKC4udjqdSkfRJE6n0+12B3CiKIr9+vUrLy/3Pujbg473Xl3mhabZRbDx4zF+fD2vSUrCokXwu2R08WLfRYAeb7wBAN6Fwtu1g/c2xNnZqG1X4lqeKnj99cOHD9/sczQmJvq2RpQkqaysTOkoiKr5JmhKxIQQohK0qzchhKiU/5uExcXFixYtOnjwoKdDvXnz5jBGRZqv5OTk3r17Kx1FOOh0uiFDhigdBVE1/z3oKVOmxMbGnjt37v7772/VqlXfvn3DHBZptmJjYzt06KB0FOGg0WioUhKpm/8EffLkyXnz5sXGxo4ePfr9998vKCgIc1iEEEL8J2h52S7HccXFxQAKCwvDGhQhhJDaEnT37t0LCwvvvPPOQYMG/e53v6MdVUjYWK3WixcvKh1FOEiSdOrUKaWjIKrm/ybh+++/D+Cee+7p379/cXHx9ddfH96omhOXC5IEjQZysaFm79KlS4cPH775Zt+Z0NHH5XJt3rx52rRpSgdC1Mt/grZarRaLBUB/nw2NSHDZbPwvv3AWCwoL0bGjnxLMhJBmzP8QR8eOHadMmbJ9+/YwR9PsOBysRQvo9YiNhd2udDSEEHXxn6CPHTvWt2/fWbNmde3a9dlnn63cxJMEHcdVbislSfVUZG42eJ7X6XRBbFAURUEQJPnnHAKMMUEQKioq5AoMDT+R4ziDwRCiqEh08J+gk5KSZs6cuX///s8+++zEiROdO3cOc1jNQkEBVq/mysq4sjKYzYiNVTogVejYseONTSv5700QBJfLJZdzCnqOllOz0+l0uVw6nU7+l6DhpxsMhkkN3NqRNFe1FuwXBOGrr75auXLltm3bJvvd1pM0hc2G0aORnS1NmMABoJquoeFwOOT+uEajkSSJ5xtX28DtdjudTqPR6NljyJvNZuN5XhRFl8tlsVg4+huIBJv/BP2Pf/zjgw8+yMzMnDx58qpVq8xmc5jDinKM4W9/Q1ISFi6Ew1G5eSsJAYPBIIoix3GMscYmUEEQRFHU6XRutxte+8DJ5AbljC9JkiiKNV9DSBP5/31KSkrau3dvM1lxq4B583DwIHbtAn2eQ0zOmHIvuLHdZ09O53ne7+CyJEkajYbneb1er9PpeJ7X0L+1JKj8/8o+9thjRqNx3bp169atu3z5cphjinIrVuCdd/D114iLUzoUNbpw4UIQpw9xHKfT6WJiYrRabWN70BzHycPWoijWTO4cx5nNZkEQBEGwWCw6na6x2dnlcn366aeNOoU0N/4T9Nq1a/v16/fGG2/k5OT07Nnzs88+C3NYUWvbNsyahbVrQX+d1MLpdJaWliodBQBotVq9Xs/zvMFg8Jt8eZ63WCwWiyWwjrMkSdT7IXXz/yf2iy++mJubK6/wPnbs2J/+9KcxY8bU3ZDD4Zg8eXJaWhqA22+/vU+fPt7PfvHFF/v372eMTZ8+vbX3XnPNzV134c03MWiQ0nGQBtFoNDRqQRTkP0F37tzZU3+jR48eDZytmZaW9q9//avm8StXruzYsWPBggUnTpxYsWLFvHnzAg434h06VDWyIUmYMcPy8cdcURG+/ho33aRoZIQQ1fE/xNGpU6f333+/oqKiuLh44cKFf/7znxvSVn5+/ty5cxcvXlxRUeF9PC8vLysri+O4bt265efnByHqYCkuxty5SE+H0YjUVAwfjk8+qVw5sm0bRo9Gu3bgON8d+axWPPIIOneG0YirrsL69Y24ove481df4d137evXOy5exIgRyMoCx1X7uuaayleuXIn+/REfD4sFvXvjtdea+L7VLCEhIT09XekowkGr1Q4YMEDpKIiq+e9B//Of//Q58sQTTwCoY6GUwWB4/fXXY2Jivvjii1WrVk2fPt3zVEVFhWeinvdigQ0bNvz444+e1wQQvbyCK4ATAXC//moaMYKz211z5kh9+jCtVrNvn27OHEfnzlL37prLlzXdu0u3326cNs3pdLq9rmKcNk2zZYtzyRKpa1fNhg2GW2+1b9woepJpQzidMBh0hw/r2rd3pqfzPC84HPyqVZ7V3pzNZho1yvWnP7kqKgBoGeOmTpXS0sDzms2b9TNmODhOuOuuwN540DHGJElq1BqNOmi12tTU1ID/twZGkiRJksJ8UQBdu3YN4kWb8nFQCVEU7Xa7y+VSOpDAySmusVOGUD03evhP0I1asSrjOC4mJgbAsGHDnnnmGe+nYmJifv31V/mxd9zx8fFt27YFUFBQENhInyiKAQ8R6ufO5YqLnbm5LCUFAAdIAwY4p0zhJEmj0WDkSHHkSAC4555q06dcLu2aNe7nn2c338wBUnq6uHWrfvFi1+rVckC6f/5T8847XGGh1K2b8MQT4i23VF7u1ltZaioMBs0nnyApScrI0Hz2GYCExETWooWjoABe3UbNihVgTJo4Ub4uu+MOz/8PcfBg6ZtvdDt3sokTA3vjQSf/YkX0WC3HcYIgRPRbQNM+DirBcVwUjPt7Jsg39qyaB30TdF5ent8/MN1u95kzZ7p06VJb6w6Hw2AwcBx36NAh+TYgY6y0tDQ+Pj4jI+Pzzz9njP3888+dOnXynDJo0KBBgwYBWLp0qSmgQm6SJAV2Imw2fPEFZs82esUDwG89OZ1Op/McZwyCoIuPrzrSogW2bq0M46mnsGwZXnsNvXvz77+vHzcO//1v5UiFRoP338fzzyM/Hy6XJj4eL7yAFSsq9uzRaDS+72LFCowda6w508PtxrffIi8PDzwQ4BsPAVEU3W630WhUOpDAySvCg/UjlfvjPM8H8Clt4nXV81sRGKfTqdfr9ZFcetflcvE8H8CSpQb1oO+///64uLhx48YNHDgwJSXF4XDk5eVt2rTp448/fvbZZ+tI0CdOnHj77bfl5QAzZ84EYLPZ5s2bl5OTk5iYmJ2dLQ+SzJgxo7Fxh8Tp03C5kJnZ6BPNZgwbhpdfxrBh6NwZmzZh/Xo4HHC7AWDRIixYgL/8BQCefho7d+K557B2beW5V12Ff/yjspE65OZi717cey9+/hlJSUhMBID8fHTpAlGEVouFC6Ga7nPQ2e12q9WanJysdCABkqtzyEvAzWZzHTlakqSLFy/Kf0QS4pdvgt60adOmTZveeOON6dOnX758WafTZWZm3nbbbd98803Lli3raCgrK+vFF1/0PmKxWHJycuTHo0ePHj16dHBDbxJ5DCew4gnvvIOpU9GtG3geXbpg3DisWAGex8mTsNngvU9zdjY++KDq2wZuVp2Tg9atkZWFigocO4YRI2A0ok0b5OaiogKbN+PRR9GxI269NZDgVa+goCCiC/Z7qn9wHOd3hYuHy+X68ssvqWA/qYOffvgNN9xwww03AHC73QGsv4oMnTtDp8OhQ4Gc2749Nm6E3Y6iIrRpg2nT0LFjg+ppNGQQwGrFBx/g9tsr//Ewm+FwwGiEToesLAAYNAhnz+Lpp6M1QUc6T/UPBHSniBBv/n+BCgoK9u7de+DAAc/NvWhjNmPsWOTk4NKlasftdlitDWrBZELbtigqwocfYuxYAOjcGWYzvJcp79jR6FGUDz6A04mhQ+F2w+1GebmfQndOJyL5Nnd002q1ct3RKLjZRRTn24M+duzY3XffvX///jZt2jDGLl682L9//2XLlnXr1k2R+EJo8WJcdx369MFjj6FvX+h02L0b//431q9HejqsVvz8MwBIEs6fR24uYmMhD8Fv2IBz59CzJ86cwf/7f0hMxGOPAYBOh4cewpNPIjUVvXvjvffw3XfYvbtxUS1bhjFjcPvtkP9pzMqCVovp0zFsGDp2hMOBDRuwahWefTa4PwwSLBzHydvFERIErLoBAwa8/fbb8t4QjDGXy7Vs2bKBAweyUFqyZElgJ1ZUVDTpwoWFbPZs1q0bMxhYSgobPpytXcskiTHGtm9nQLWv4cMrz9qwgaWnM72eJSWxCRPYuXNVDbrdbP581qYN0+lYVhZbs6bqqVGj2L33Vrv6okUsM7O8pMRmtVYeOXCAAWzzZt84H3iAdevGjEaWmMgGDmQrVjTpXQebXA4/FC3LZTxD0bIPt9tdVFQUhguFVFM/DipQXFzsdDqVjqJJnE6nJ382iiiK/fr1Ky8v9z7IsepTntPS0mpuBd+hQ4eQ7nq1dOlSeeJHY3k2t41IgoDSUtvp0xqOM/TogYh9IyGaZidJks1mkwdzzWZzSO+FCIJQXl6ekJAQukuEQWR/HAAAJSUlZrO52U6zGzBgwNatW2O8RjV9x6CTk5NXrFjhWRXmdruXL1/erMsbhQJjKCrCoUPYt4/p9Sw+HqdPo/GLg6Kb3W7X6XRarVaj0cjl8AlpbnwT9MqVK5cvX56QkNCjR48ePXokJia+884777zzjiLBRS2bDSUliIlBQgJ/9iwA8DwlaEKID99+eEZGxs6dO8+ePSuPaXTs2LFdu3ZKBBbVGINGA50OdjvcblitaN8eNCULAFBQUHD69OlBgwYZjUaHw8HzvCRJUbn7tcvl2rx580i5ogAh/vgfKGnfvn379u3DHEozYjSiogJmM1q3ltq35xMTER+vdExqYbfbCwsLAWg0GrPZzBjjOC78k/HlezMhva4kSefOnQtd+yQKNGgku1OnTqdPnw5xJM2JVov0dDid4HkmCLRpbG0USc0A3G63vGO3wWCI6BtWJNL5Jug9e/bUfJHNZgtLMM2JRlNZkSPC60NGH0mSSktL5bxcVlaWkJBA602IUnwT9LXXXluzIlJRUVG44iHNgNMJSYLR6LcWSlxcXNeuXcMflA+5567RaFjIbt5qtVqfneEI8eGboDt27Lh161afG4N0n1BdGAuwzJMalJTg0iVoNHA4kJFR89ZoYmJiolzATyE8z+v1ek/Z9dB1n2lHFVIv3wR922235efn+2TkW6kuj0pIEkpKUFAAxtCtGyJueFSSUFCAFi0AgOPgdPotwK24mJgYt9sNQKfTRWexMBIhfBO0T8lQ2ZIlS8ISDKmP1YqyMsTHgzGUl+O3jX0jiacqub/y5E3HGJMkiTGm0WgCzq3y7cHgBkZIAGjubUSRpMopHxyHSCw0yPNo3x6lpSgvR2Ki3/qrTqeztLQ04CvIe6MIgmCz2UI3fBwUkiTJEwoJqY3/BK2tLikp6ZZbbonymXYlJTh8GIcPq7qSpzyBWl7bEqE3BmJj0b07unRBYqLfkfQLFy5s9y7ZWie5mpJnryDGmNvt5nlenp+n8gTtcrnWenbbIcQf//OgFy5cyHHclClTACxfvry8vLxLly7Tpk3btGlTeMMLF5sNV64gIQGM4eefA9kKKzwMBnTtCocDWq06R28bJEi33URRdDgccqUOo9FYucHub0lZXuESlAvJ3G630+mUL0QD0yQ8/Cfozz77bNu2bfLj2bNn9+3b98CBA4sXLw5jYOEl7/UHgOPAcaqeCIhyZQAAIABJREFUJqHTQadTOghVkCRJrhmm1WolSZLzprxAHIC8hXGwriUIgiAIOp1OLtvkU6tMkiS54KrRaAygjBkhtfE/xHH58uWzchEfID8/3263I7r37zEaYbVCEOB0onVr9WZnUp3cX/buLGs0GovFYrFYgpsovS8h/wPgTRRFrVar0+ncbjcV3iNB5P+X+Mknn+zfv//gwYMZY7t27Vq6dKnVar3pppvCHFz46HTo2hV2OzSayK3LHB0aXp9Lo9G4XC4Aer0+1Iv9eJ53u90ajUaSpJqVr51Op2ej2IYPfBuNxkmTJgU5UBJd/CfocePGDRs2bNeuXRzHvfrqq3I96Geje5slvT7yphVHI57nG/i3Gs/zFovFu28rSZIkSQ1voeE0Go3BYJAHVWr+Y2A0GuUy7aIoNqp2B03mI3Wr9c/A1q1by3t7x8bGhjEeQhrHk51FUXQ6nXKW9NwzbArGmCAInhuDdWwCq9Vq5TD0en00jwSSsPP/y3T8+PEBAwYkJycnJycPGjToxIkTYQ6LEJmcJQVBkOpb2OK5TyjfM2z6pQVBEEVRHlmut0E5fVN2JsHl//fp7rvvnjZtmt1ut9lsf/3rX+++++4wh0WarUuXLu3fv9/zrfCbeheeeMZ/gzXBzul0yu2EaEq12+3evHlz0Jsl0cR/gr5y5crUqVPlSjF33333xYsXwxwWabasVqvn940x5nK55FUn9RaWk/uwgiDUMRbRKHq93uVyWa1WeXy56Q36EEWx5gbNhHjz/2vH8/zx48flx3l5eVSznCiu3kEGjuN0Op3FYglWhSN52obBYKCBC6IU/zcJn3nmmeuuu+6aa65hjO3fv3/lypVhDosQABzHmUwmeWax0WgMZ6JkjDHG5MnUcvWlsF2aEA//CfqWW245dOiQPM3u2muvTU1NDXNYpNmKjY3t0KGD59tQzJlrCI7jDAaDIAjyrrWhiEGr1WZkZAS9WRJNfBO0Z5VUfHy8Z79hh8NRc3J+88IYnE4AfguwkSCS5w4pHQXgVQw6RFWhtVrtkCFDgt4siSa+CTq+lu2lay5vbUYYQ1ERiosBIDERiu73QcKJCmsQZdXag25G6i2N5HKhqKhyCfjly4iNpXJFISWKImNMqcENQtSjng/AihUrwhKGQlwuHD6MvDxcuVLXBh8cV+1ZKqUUSk6ns6ioSBAEu90e3YWHGGNlZWVKR0FUrZ4EHeX1NyoqEB+PuDhUVKCiotaX6fVISUF5OcrKkJoK+rM3lM6cObNjx44gLghULafTuXr1aqWjIKrWvHNNQQHkMXeNBnV31uLjERMDjgtWsXlSG8+yPXnpdhNbk5ui+vokQtXTg54zZ0544lBG+/aVO0iVl8NsrufFWi1l5zCQK1oIguC3blzDydtfWa1Wm80m74xltVobUlWDEPWopwcd5VU45Nt9goA2bQK/7+d0oqICBQXo1IlqSQeFVqu1NPknKYqiXOoIv+0kK++H4nK5zGYz9alJRGjeQxxo8rxmxnDiBOLjkZCAc+fQuTMVlW6i9u3bp6SkBLdNh8MhT5iT83LQtysMjMFgGDdunNJREFWjaUxNI29mKH/a5c44aRqtVmuud7ipAeRxEkmS3G63wWCQp+5JkqTX69WQnQFwHNeiRQuloyCqRgm6abRaJCdDECCKcDjU2H1mrJ77n1FK3m9Fr9fLeZnnecaYTqfzFNevlyiKbrfb7XZTIQ6ilGY/xNF08fHQasEYUlJUNwPPZsMvv0CjQatWSEhobjO45USs1WrlNS9ut9tkMjXwXHl/Fk+NU7kihzz5L4QRE1Id9aCbTKNBXBzi49XYff7lFyQkoEULlJTAZlM6mgYpLCw8ePBgsFrz7iw3amRDFEVJkgRBYIzZ7Xan0ymPlgjBG8USBGH79u3Bao1EJUrQ0ct7CTvPI0L+Ti8vLz9z5kywWuN5XqvVynuyNLz7zBhzOp3yELbVatXpdPKic57ng1gLQRCEo0ePBqs1EpVU8feavO9cACfKfZygxxNO8h/OoXoXrVrh8mVotbDZ0Lp1iO5hyntpB+styNPjGtKa9yKUOhakcBwn3xiU46ztot6/hN7DGlqtVn5KHsWW032gb64a+R+AIP6vj4KPA2Osgf/3VUv+XQrgRL+/nKpI0PI6ggBOlP93Bj2ecJJvYYXqXcgTvSUJBgOAEN0tlBNfsN6CPLZQd2uSJDkcDnkLFY1GI9/NAyDfAwzsop7/yuSts+SpIHL3WZ6cx3FcON9po9DHQQ0CXgml3gQtbywUwImCIAR2onp49lUK1QVC//OR82Ow3kJCQkKHDh3qbs3tdstT8RhjcoKWhy8EQfAMRzSKIAhOp9P7onq9Xu4KNXzWR2PxPN+9e/cg/q+Pgo+D3W7X6XQRvccex3HywFpjT1RvgibEo1WrVq1atWr46+UxoqCH4XfChvwRClYRVJ1ON2LEiKA0RaIVJWgSeTQajTwHTu478zzvcrkA6HQ6OV+HImXL68Xlq+ioIDgJC0rQJPLIi1A8g8L47TaGIAh6vd7hcFgsluDmaHleh9ynlkelaDMBEgb0S0bUQRRRUQGbTfrtjl/d5JE+OQs7HA55xSB+6+eWl5cH8UaTIAgVFRWexeJB3OTbKW90SUgtKEETFXC5cOQICgtx8WL+Tz9t3LixUWd70qVcYlSj0Wi1WrvdHpTKonIBPL1ez/O83W63Wq0A7HZ703O0w+FYuXJl0yMkUYwSNAk9QYDVitp6i8XFOHQIv/4KlwsmEy5f9rumpo5+q9FoFEVRr9c7nU553gVjzOFw2Gy2JlbSkDO+fGl53DkmJkaezBfRc3VJpKAxaBJiLhdOnIDJBJcLbdogNrbas04nCgsRHw+HAxcvokuXmnv4MsZsNpu804rBYKg5uUKutiGKYlJSksPh4HneZrOZTCatVit3qAMej7bZbACcTqc8Cc/n0qIoyiMtgTVOSL3od4uEmN0OjkN+Pi5eRM2VzZIEnofRiIQEWK0oLkb79jVeIvE8Lw9c1DZoK8/FdrlcBoNBnmXhXQA6MJ7qSCaTSe4vazSa8vJyuSKH0+l0uVx+d7aVh6qpBh5pOupBkxCTJFy4gMREMAa3G4JQreafwQCHAxwHnQ4DByIpqY3TmdimTSOvIMk9ZQDyhikmk0m+09iU2vzy6nC5Ay4vPeA4Ti5hit+WHcrLwb235hJFUe7FS5IkTwGsrX29Xn/rrbcGFhtpJihBk4DI3cOG5D6TCbGxKC+HKCIlpXKLAw+eR8+elTnaaARgMBh8lsNpNBq5QypJkrFhO+DIS8DR5EUlJpNJro5kMBg88/nk+ddyv77mKXL5DgDyovA6AuB5Pjk5uSnhkahHCZo0EmMoKUFBASQJaWmot0ScwYAuXVBcDJ5HXJyfoqwcV28j8qxn1DJkwfO8fIcQXpm0KalZvjcod4SNRqNWq5X7xQBMJhPHcXKdaHndSm1F8lSybwuJaJSgUfmnt0ZDm3Y3iMOBK1cQFwcAp04hM7Oe13McEhIq9340mQLeNKDefMcYkyfDBda+dztWq1VelsJxnNPplEfAvfexles0+S3TodFo5MQNICh7d5HmrNnfJBRFHDmC/HwcOwa7XeloIoEkVY1RNLDMNM/DYoHFggZkz6Kiory8PJ+DjLE67rnJU5V1Op1cXL/+eOokD2LIm7C4XK46Mn5t3Xl5ELze1YyCIOzdu7eJ0ZLo1uwTtNWK2FhYLIiPj5Q9RxRmNMJqhcsFux2pqUHfRqu0tPTEiRPeR6xWq7xCxHvhiacupZy7PakwWHMn9Hq99BvvHF1HUWmPBs69EwQhNze3qYGSqNbshzi8P88FBUhKUi6UCKHRID0ddjs0GjTsll1TyHWZvZd0y3f/bDabZx9Y+a6d/G3T623KEzbkkY0WLVp4z30WBMFms8kVouPj45t4IULq1ewTtMWCc+dgMMDtRufOSkcTITQaxMSE84LyWhW51wzAbDbLQxAAnE6nxWIxm83yU5om30iQx5flARNUH8f4/+2daYwkd3nG36q+qu85d2cPe9drr2GNAWNsAz4EkiGHiRFGSoQECY5EkOyERPkQyf6CiBCXogAJihwIUWwSJQSwFW8ICiFBJBgHbBzMYbzYu7Z3vd71HjPT093Vd3flw2/rdW1f29PTM907838+jHr6qP5Xdfdbbz3v8z4vfd6RSKTTP9rAYD2w5QN0OCwHDkitJuHwxM3k3pKIx+NB8RnOovShoOXoZB7wtOtkFRqNBoNXHMcZPHDDnLRaLRxNo9FosBjIu9DWuMY9tW179+7da9yIweaGCUlyrpPNYDKwsLCwsLCg/1qWlUgkqtWqNp7UarVsNssNz/N6qdx4AgYa5XJ5cAPScrkcDodd10W9B6FBfI/FYsVikc7y4WZrBbPyaDR62223DbERg60DE6ANJh1wDvAJnudls1np7ZqvETBYyhtcexfMi9lI8LVk4szZGm6wlrH8N1gVTIDuiw7jnrWiUpFGQ6LRLv0aBr0RCoVgmfsLJKCGPc/TSKpGdAOmzwTlUqnUarVqtRotKkGuuWtgHWQUlrH8NxgCJkD3Rj4vx4+L58mePaOpiRWL8tJL4jhSKslllxleZVXQqmAvBCNgtVpNJBLDVQ4ty0JjRwzVzL1XiK/X60q/JBIJ00BoMEKYc3gP1Oty8qRMTcn0tLz4ooxkPEetJum0RCKSyZimmF54/vnnv/GNb/R5QqPRqNfrXT3k9DZRkp7sVUVnNkKsJwGv1+ulUglr6a4voZZINO8jkSbuM5eLjL5SqXzxi18cfG0GWxAmQPdAs/lK53coJKOYzSEi57bTbA7SU2fQCdJVnDHaoqFGwGazOTin0QYoFMqPxOjl5WWGADQajf5jtC6o64hEIo7jxOPxoZdnsNVgKI4eiEalWpVwWDxPZmdHo8BLp+XkSbFt2bZtg3XEG4RmUwoFabUkGl2nHaxWq7DAWNyRtMJNh0KhkZjYtVqtaDQKAV0oFCqVir5j1xCsjneDOIEY3tlgVTABugdsWw4ckFLpnNea5jtrKRtGIvKa15yzqN+UWFmRYlHCYVlclF27ZB2sgmKxGHNMCMo0sDDjKhwORyIRjYBDO0HzFrZtMyg2Ho+7rkupkI23CezwUVqL8bSBQS+YAN0btn1eGui68sILYlmye7dkMmva7NiB3drIlSQvvyw0QDuODDCZuysWFhbSbWOxAtAaIDqNZrNJFwm2c7T/1Wq1SqUSCoWYrrLaBfAWtKgEFXVk0I1GQ+cTgjYzkMERjUbf+c53rvZVBlsKJkAPhlZLjh6V6WkRkZMnxXEuYp3c8rKcOSMiMjcnMzOj3PKOHVIoSDgs5bL0sqJvtc6ZUiUSXc9V8Xi8V++JBPqwA9s7N9CEEFmv113XjUQitVoN3mO1rAJvAZdNhTAej6fTabUEUZoFULTsNS+xD2zb3rVr16rWZrDVYAL0YGi1XqkZhsMjqxluPOp1OX363OTWs2cllRrlmSabFcuSVksyme4e/J4nTz8tGCun0zIz04sv0rRU/Tc6Z7/Sk21ZFqagiUSC9DmY8A69KwwhpJoXDocdx6HDUERoLyS5Fl/ULL6cwxAdBiOECdCDIRyWuTnJ58WypFK5iNPndYVtS3+Pt1pNYrFzR29xUdLprkeSVpFQKEQcjMVi9JvQNiJ+8CViortQATLxVHV4Q8doQj/vHgqFKpUKKg5mbjUaDUiVPsm+gcHaMQF86MWCmRnZtk3m5uTVr54IHnk4RCKybZsUClIoyPz8Rp9pQqFXFOVBIWMAuVzumWeeodwXbE4pl8tYfeL5qRo74rIGYozuiNrEUPTLqx2zbVlWsVis1+tQKJguiUilUqnX69wmccabNJ/Pr5ZLMX7QBhfERRtoNh6WJcmkpFIX/WSs6WnZt0/27TtHqW8kwmHZuVOWl2V5WS65pOuRXF5ePnTokFpqUJer1Wq1Wo2sFglHuVwul8uk2HK+fI3ZsujwRKRarZZKpWKxSHwfcKWe56VSKYTV6PkAkw95U209D4fD6XSaKbGDHwwzUcXggjAUx5bEGJ160ukLjjGE3CBLjcVihULBcRyc5LQHL51O0zwSDoeDo//aRqtYlkXNkGy3UqkMaGvH0FgYcF7F9JZYLJZIJDgxOI5j23az2dRmRZ7PWWSoo2NgcB5MgDbYQDSb4nkX7PqxLCuVSpG3EgepzjmOg/kno7uhmzsntzqOQ72O4dyYadi2DTVRLBYty4rH4/1bwB3HKRQKOIuSI/MuiUSi2WxWq1W1bWLL2jKjJqhrHx1gYGACtMFgaDSkWBTPk2RySOY6n5cTJ8TzJJWS6eleMrtYLIahqFrj09fHQ+jb1LdTpRRBhMPhYErLGNl6vY56OhqNyoUcohGQUBIk0NOlzXrK5TLLQDGC4FpnJHIuIenufzxs256fnx/06BlsSZgAbTAAPE9++UvJZMSy5ORJufLKVfe+N5vy0kuSyUguJ0eOyI4d0mjIVVd1yux27ty5c+dOCYxnJU2WAH1BYuv56Iyzeo9lWbFYjIR6ZmZGDY/6UxCIoGdnZ13XJQpTKmyzmVZwPmg0GhecJxtENBq94447Bn++wRaECdAGA6DZlHD4XE2PLsEhzElsWxoNWV4+p7SzbSkUJBaTjsl+sL2VSoW43Na5p0BZ0Wq1+vMJxHeKjcVikSdHo9H+MZqgTKuL4zi5XA72mdcS6IOd5SKC2ENpmVUdGwODrjABelLRaonrSrMp8XhnCNtohELSaJzzIWF+4xBb2LZNTp2SUkl27ZJoVE6fPqe0m50NNjSqX534eS6+ReJL5chkKc0RQLWZsA9IbzOZTL1ex7Wjz5Nt2y6VSnQS0lCOTSgUc1DJF3yVUuemSNgOzxPXlVpNIhFJpUY8BGNTwwToSUUuJ/m8hEJy8qRcfvmYW2MsS664QlxXRGRqakgRCLzz3Jy8/LKcOSPR6DlLk0BDI2U9Ii8uSARBqOTgGO/VRkD6s2G0B/GcU8ukQqGwsrISnLnFWaHPa9kLxn531jC3IopFOX1a4nHJ5cTz1mRls8VgdNATCc+Tl1+WaFRCIUkkpFodflP1urju0NZFryAWk5kZmZlZk0ddLCZzc7J3r1iWnDhxLuKfbxB49OjRb3/727ZtY4hh2zbdemSspK6kzJrYklBTo+v9zufaEdGBXHClmp5Xq1UGrBSLRW5f8LX0HEYikf4W0tVq9YEHHrjg1jYD6vVzrf+OIwNL0Q1kq2fQnifF4qTQCEFYlnjemigFUC7L0aMSi0mlMilDthoNOX5c5uclFpPDh+XSS2XXLk3JMSqqVCoM5FYVHcVAnqM9gcifEb3RGu44Tq95rPDFrVarjTjuBWjrcrkci8Xi8TguH6FQiJYZVf51hbopDTKj64Ir2QwIh6VWk2hU6nW5kLjFIIitnUHncnL2rBQK8vzza8pS1wP794vjyPKybN/e3XhoEJTLkslILCaZzDkPubFDbacyGbnsMrnssjb7Dq3gEU9LpRKsQiwWo887EolUq1XMMUQEs3yV1iFD7vrOQUf/QRAOh1OpVDqdrtfrjUYjHo/TF8MCms2m67rFYrGzO9FxHOqcA2brmx/ptExPy/KyZLPS20vWoBNbO4Nm6qDIORphopLoaFRmZ2V2dq3bYT5AqzUplZlI5FyxSETm53ude7qqm8lq6QwUERpPsPqs1+vkzms0zierRUDSbDZJlhFvMJ+QubRQ5JDaZMGaUJPsw8YMwndvCViWZLOSzY57HRcftnaA9rxz8Ws43diA2x9jRxlDtsJhaTRkx46xLSMIyzo3qkZEEonO08a2bduuu+46bDdEpFMPp//C8JLAhkIh5BkXbA/pj2q1Wi6XLcvK5/PhcBgNNbkzTSjo/xzHYRmlUklEWq1WMpkkFlPbxAS1v1YkGo2+/e1vX8tqDTY9RhaVjhw58jd/8zckPn/0R3+0bds2fahSqdx555379u0Tkd/6rd+65pprRvWma8WVV0qhIC+/vC7zmWo1efZZCYVkfl6mpsYTpiMRueqqcyehCcmgpWNUzflIJpNMkKL6p7bO6iZK0CRRVZKaUBiPxxFRyIVaUXqBBsVqtUoebdt2LpdzXRfiRUQIzclkElM9+HGtZwbZFVbeZxm2bfOjGBT1ujQa50rHBlsDIwvQc3NzH/3oRx3HeeSRR77+9a/ffffdwUf37dv3iU98YlTvNTJEIueUCeuBYlGmpsSyxHUlHB7b9Z1lXaS/ZxJSjOuw3dCRgAyj4v5oNEq8xqDDdV2CJsF6tW8aiURKpRI6aNu2mR1O2C2VSul0mv5v5NixWIy0XV8e9D7Vi4DRwHXlxRclEpFKRa68cpx2VwYbiJEF6KwfgLqqRI8ePXrPPfcsLCx88IMfTPnZ0wsvvHD69GkRqdVq9aF0YNCCa1j1eqJet5BYtVpetdpL6EZz8OTuxQBA3zbCXSD4QvvWajVEFCsrK/F43LKs5eVlZHY4yRGvW60W3kbaP6I8cn/ZMqCmB6mNjGRmZgbLOigUulTQ89HDov6iuHA4jsMR0KkCmO2N6phYhw+fs4e1bW9lpev5fqJ/DoNBr5zGvZDhgeJziEkRXX0CRky8uq77ta997Z577gneGYvFvvCFL6RSqYMHD/793//9XXfdxf3f/e53/+u//ktErrjiChdJ7CrRX/o6XliNhn3qlBeLWZVKc88e6bGDfCoD/ZI9z6pWPcuarGKmXxYbVWgoFotLS0sLCwsigukzTSIUA7GjQ9QMqxCNRhFy0L+H8xz/ql9oPB7vr19mF/gSVqtVxgzK+Z+O4zg6KADLJGQh1AOxZFoVms3mkSNHrrzyykGebKXTluuKZUmj4YVCXreSyST/HAZEq9WqVqsQTRcphmbY1j1A1+v1T33qU+973/v4dSlogRWRt771rR/72Mf0/jvvvPPOO+8Ukc9//vNT/Ucl9YDrumssCq0jpqZkbu4cadj7grRYLIZCoQtPTmq15OmnxXGk1ZK5uTF47fcG6fOo3CeWl5ePHj16xRVXKMVMIA6FQslkkjIdY6gajQZcB6k0OWyj0VCr0lAopFPAM5lMH8aj0WgUCgW+hKlUynVdtoADdaVSaTab8M68V7lczmazU1NTcB1DzA4XkUql8uSTT95www0DPdtx5LnnJBQSy5Jdu7rSVhP9cxgMuVwukUgM0g00scBda1Xjg8H6BmjP8z772c++9a1vvf766/WelZWVqampSqUCP/jzn/98x4RoCTYG3ZyAhkS1KvH4uU6TU6ckk5ksZtnzeo2w6vLMalX6XgfALJ89exbxXKvVikajqVSq2WxGo9Hp6WnSVcdx4DqQcKDoYKIr/vq6QYiIgXfFowkQK7t8Ps8CXNflL6SK67o499u2zZljwO0PCceRV7/6nGvVxcwAGKwKI/tWPfroo0888UShUPjv//7v/fv3/87v/E6pVLr33nvvu+++w4cP/93f/R3jJz784Q+P6h03D1otWVk519DYK5VmWrY+f6J+ouWy/cwzkkyeS+37lOY8T5aWZHlZWi2Zn+9zHcBVRTgcVtNRdBShUAhTDnrBMbSLRCIMxHIch0SeTutQKFQsFguFQiQSOXPmzMzMzCCpGRpnfSPoFIhREmo8+5UWx8RuIxz6bfsiHoZpMBRGFqBvuummm266KXhPMpm87777ROTqq6/+8z//81G90eaDtbIijYaEQnLmjOzd2z21xArjzBlpteSSS8bzQy2XpVKRUEhSqfMW8Nxz3vS0xGKSz0ss1kdCJ9WqLC+fUzSePt31OoARVsg2GAkY8REOhwmd1WpV7UAZbjIzMyMBKj/YQk2gr1Qq+Xw+m832pyNo6S6VSrVaLeZ/EFDhMzMzjuOUy2XCcTwe52KWZpnglIABYdt2xtgGGfTF1m5UmQxYL78ssPbxeM+GRsuSmZlzlvljITcqFTl2TFIpaTSk2Twv+dV0PhSSCzIJ+oQe3vaXXHLJJZdcIiJEw1arhcAOuoOiHxJjSklQw+TaegMFSCQSwRtPfJ88dBp96AhEGrFYjFmIquJASULXDKIRqohqEjKEpC8ajb73ve9d7asMthTMFdP44W3ffi5aVasX0Leqa/7Go1aTeFwsSyIROXnyvId27xbXlWpVCoUL2IY4jszOSqEg+XyvSpciFAqlUimoDBQakBiQHvhBqy8zpUK4YBhknbdSKpUw9MBcqVwu9+GjGRG7srKSz+cbjcbU1FQ6nd6xY8fCwgKZciwWYw0MNuR9oe9WdTgNDAbBRGTQL7/88ve+9z1uX3755Uw8EpHFxcVf/OIX3A6FQjfeeKO+5MknnywUCjTd7tmz59JLL+X+fD7/k5/8RJ9200036S/nqaeeWlpa4vauXbu0iatUKj3xxBP6kje96U3KVP7yl79EqS0i27Zte9WrXsXtWq32wx/+UF/yxje+UQdLP/fccy+99BK3Z2ZmXuNPsG61Wt///vf1Ja973etUOX4snz/13HMiIuFw2nWDnZaPPvqoFruuuuqqWd+a48SJE0eOHOF2PB6/7rrr9CWPPfaYXuDv379fFTVnzpw5dOgQtyORyJvf/GZ9yf/93/+pzPGyyy7bvXs3t5eXl3/+85+fe1Klcstll0k4LK2WbN/+s5/9LJfL8cjOnTt3XXKJRCISixXL5R//+Me65be85S2arj799NNnz54VEWk2FxYW9vtX95VK5fHHH9eXXHvttbFYDJb5mWeeOXLkCNFw27Zt11xzDblzo9F44oknYD9s237ta1+LDKbVaj3zzDMnTpzgCCSTyauuuopG7Xg8/vjjj+uw1+uvv14P5vHjxw8fPlwqlehgZE4gwuf/+Z//IQePx+N79uxRKcji4uILL7yQSCTq9Xo0Gr355pt1N3/0ox+p5K7Xl9m27SAf+JOf/CSfz3N7iC/zzp07L7/8cm73+TI/88wzp05f2FVTAAAgAElEQVSd4nafL/O1116rUpDgl3l6evrqq6/mdr8v87FjR48e5XY6nR7ky3zy5MnDhw9zu9FoBI/MIF/mcDj8lre8RV/S68ucy+V+9rOf6dNuueUWvR38Ml9yySV79+7ldrFYvPCXWWRhYWH//v3crlQqjz76KAoiEbnuuutUoHX48OGTfmYzNzd34MABbjebzUcffVR8fV4bJiJARyKRtO9xFSzjhMNhvb8tQ+E7FA6H2ySuoVAoHbDLCqoR4/G4PhQL0Ai2bQ/ykqASzrKs4EuCa4vFYoO8JMhXxhKJ9K5dwf1S0LrG7eCFeTQa1a216dtSqZQekOCR4SArRUAGqi/RXeh1/K1MRrZtk+PHZWFBMplELqcLiyFWcRwZ+GA6gd0MvgTqgOXRdYIzXDQa5WCWSiVKhewLubP69+OJkclktL4XDodJeNFQp1IpdHvBHwMH07btRCLBtCqCAmVGtHrhcHhqaioej7OAeDw+OzvLOtPptPLazWYzkUjoJzXgl5n37XxJ25c5iPMOZuAL0Of4O47T9ZvZ9pLgNzP4kkTAC6Hflznw/R/wyxz8+bep6ft/mTs3Jb2/zH0OJlVfXb/eP+iXueP4a4AOftC9jr+IcH/XAG0N0fEycnz+858fTt2xCYSfg+qgRwTkaJZlkVTqF0h7HPS7NThGqIMuFounTp06fPjw2972NiaS0HJNpoxsQz3tEE40m01k0bp3nucVi0ViN4GeYL2yskIoh6oO/vbQQadSqUqlcvLkyXK5TD8YAZpSnuM42WwWNztko/V6HQmpFjApXXJiYMt9SOparfbQQw+NkIbeBD+HLa6DvuGGG7773e+mAmV2Q5xtLXCNL76Vj95POQ4GdlWjqUcLOgbppdZ7IJS1b7DlQ0NecNo3amiiM/7RuMrBUOMizYVX1wVUKpXp6Wm2Q6M5pqOVSqVarRaLRa48OIyNRoN+mXw+X6vVIMpRjOgGG41GpVIpl8udzaKtVkuZDQODrpgIimMTYy3OausBRu1RcOuap5ADdi15rdFqeRCQsbI20mcyVpyJaNvDIwntM/fj60/VjhYVnka4J0ajhMtms2yka4LDpUyhUEgmk5yl0JDU6/VUKkWnDPy4iBC49RhiFU2qrkep2WxWKhWKnJVKZYhRigZbHCZAryNwYqM1ebhu4JFDA1Ob8ADFsY77a3sVaSCM8Lq2Y3BtqAOlHMchAtbrdWZ70ytImGs0GrC3rutyv+u66XSa+MuVZqPRIGLi8IChktrRsZ3gcVhaWiIXpjuR9ehfLi+42tA+GrVn4nzgui7lr1ar5bouGX0wahsYDA4ToNcREKAiQh8at5VY6PqLXe+MW62T26Cj/9SKXoH1BIK2crm83mng3Nzc6173Om7rcD/aRlghgmi1h4Z5IESS6nKok8lkrVaDW4deKBQK2WyWrNa2bVhszjrqvs9xIATDqJDO03HuOM7y8jKnikKhsLCwQKKtliCk22qUSt85C6D7JrinkUgkqCUwMOiECdDrCI0v+stElsvtTsNiaFNs24azMx4avdo3yE85bZClrvdK0uk05Tty+UKhEI/HKf0xCZBqHg2H3MnhLRQKoVBoZWWFNJ+qfaVSOX36dKPRcBwH5RzvohPBQ6EQAjvuV4dS9Bhw34g64vE4xcN0Oq3Wo6pt4Cjxwemhk/OlIMHdZCWvfvWr1/t4GlzUMAF6ZOikaB3HIVkjmRIRKlQ82kn1asjg9iT0PpAYqviJBHNj3rdcLpMI1+v1TCZD/K3ValNTU81ms1gsciYjM3VdF5lHtVqt1+vcCQEdjUZVYYJ5KUILzknky8RZ27ZrtRqlxZmZmXK5DANOq4sErjNYYZDtUamcfmrQLNyOnd8dysJEpNVqdWbWBgYKE6BHgF6ZLz/mYGVJAnG8u+xxwn6rMBtIiXvVFdd7AYjYlDR3XZf2EJpZSGZRU9DtDTWhmgp0INVqNZlMoqsl+RX/g4PooHfcsiwqilQdEXIgAhF/QCLjYqempoJhl7cLrtyyLOJ+2/1k/VxaqSx9/Q+kwUWJ8edomwBcL6sCQe+vVCrRaFSJSBGhbEUc6fxZ4jWMrHgS0mdANSwWi0Hvrvfbua5LwxXELiGMdmqkb0RJFW5z2aEWoBzeQqFQKBQIxKjcKPpFIhEIZWZ1E8qV36AWSnpLdRe5Hlk5G5+dnd27d++OHTsGURx3Ru3gna1W6/nnn5+0U7LBRMFk0KOEFZjlrB0KErjsxXWeiNP5y6SZrdej4wKJ6nq/C81+oVDo9OnTTz311G/8xm8gnEADIyKwDaiPM5lMsVikkwXdsUZzKGM0J8RxjmosFiPp1nfEIU/fXUQYDkvcxw+aDWqzIkpqbvcSIw4CuK9yufy9731PzQAMDDphAvQIoJ0LGk1EBGZAjTGD9/e5pO3/6GaF+sy19ciQw4qIqtm4BEEUQf0N6TFWc+Lb0RGd6XBBpAGNzmmvK5NuWVY2m4XLrtfr9OzATfH5Tk9Pi++KB7lMMi6r195wnmbxaztyBpscJkCPALCN2toAQUHhXo3WJicjnjQEOVnxk1ntPVHzz+DliPotVCoVtHRQHNFodGlpyfMnfNNGr9Y51BtrtVomk+mqNcQsgjHhIhKJRCjVMmOF2M0FEIRJqVTKZrN6atkIz36DLQYToLsDulMG1i2Q3xWLRS0AQnGaFOmCUEJW81AiYyQSITqrd5IOby2Xy+SwCDxoW6eUNzU1pUoMGkls2y6VSmqTBAfSNUCHQqGZmRnbtulqsfxpADAkpL3oDnE0DYfDrutSvWSuCuEbN7ugI0evHY9N2Pxfg0mDCdBdwG9Y3Sr6NAFSwtJCPPanSMS4ytar4K0JMtZoNNr/PBcOh2k2wdXz0ksvpVLH8UeYTJ1Q/PYffJHo1gsK6fggdIAsZUA0NrwwSDd1AmIKp3+VYRBn+aBx50ACqMR3cAs8JCLakdjrvWKx2Ac+8IHVHE6DLQcToLtAqU9yqD4RFkMGWMtgDFKWA/PM9cujN8Af44IL4GIfSie4GNVXUHbrdRjRY2AfSlSFuGd+CpkyCpnFxcVkMlksFrk6gbugtRo7PT4OZqyQhi8tLdEfyKeAiV2fxBZbO/R5RHNGWxGdz549C/cNiwK1ooXENj8/c/FksHaYAN0FyN347fV30dRuBYwdqCmRo3GDLG/kv1UEvASv9fPHGKT2BWkA/9DWjqgnj/7nOZWdWZZVqVSIgJoykywzworbjUZjeXlZRCCFw+FwtVpFAkhbEOfLRqORSqXgQ0iZbduenp7uf0GDIJKcHZUIe0EIRqLDB8oJiYMPwcJekLBTouwjGx/7mdXgooAJ0F3AtTa/0gvSiPzSSLpTqRQ8aSwWw6MSQdWqykfaDt6HGaAUydrK5XKqz5zWocAJQFuc+yyemCgiFNOCD3GeI7b2Oc9xYlPelhMbUY8wrS5OmIXiOUderM9hU47jkO1alhWPx+k2hIjAr4rF9Crbwk1RZoSggGvGmoNeGF6r29Rub7UApjKMXruXKSDHFpLahGmDPjBXYd2hDWz9n8aFMBfy/OZR10I9J5NJ/l2VwzIxiM0G2156YT1+4QQRFh90N+6EziuB8A0+xHmOJpdgiO9soaT+lkqlksnk8ePHH3/8cWRz+MNBVkAyQPrzbyQSUZc4yze55ujx2kQiwXwTin6UFlzXzefzi4uLbcM7xK892LaN5wZbg8KCfdbQr2o/TsOcoaFoisVirVar1Wq9hh/C/EQikXK5/OCDD67mYzHYcjAZ9FpBjO68PxiSVhVDgxZLXX/h4qerBKz1br8m9PTaBaJhrVZrm88C+dOmY1Hb0k6Fg0ZY4ikz7gj6MMswzmytWq1mMhkaMqnmcYMcOZfL8RLqeNT60IRgpUS2m8vl8ObXNcBjsEEa9z3P49RLmNZzMK0xUC6cOdTj37ZtfZRA3Hk8Vepzsm38roHB+TAB+jx4/uSOtbPGXKRzJbsqjljbW/p4XxB0vB6zlDpBNBffW+OCzydnJEL19zJGqEAUk4AtiZzfqifn653ZwV4rV+EdjdpcSSQSCdVOWJaVTqchHOCRxLc6KhQK5MscHMqDlPtwL8IxldDZdv4LeuZxeJHuBUuCWhW0fXNRTKsxIGXv+l94YfmEdnArK3wMBoEJ0K9AnYKp3a8xRpNqDUEyKokZtFjquv0BtwypTaAhlAzyKlpvggzvgO9FTU9E2pQtbYu/4HYIx+FwmCofmkXkHNpSz7+0fUNBlMvlSqVC718ymYSqisVixWIR3sN1XV7bdgA5tYjPwJCAi6+I5yQBJR2NRqempvL5PNYckDxw6PgoVatV3E27Mu/a0N//8zUwEBOggyBFEr+6NZIfDyGAjNIeeB5rr+xytS3FgOtubb0bsKI4XNN5kJZpy0+JsBAI6g/XiWw2u2fPHnwwSMZVhoxssVarZbNZsldGFEIZwzYgcCT75rX1ej0ej2PQ0Wg0kskkJb5O9QvOoloMVIU7/LX2rbiuGw6HCc2o/SgqOo5DXM5ms1yp9LlEYFjXDTfcsNojbLClYAL0K9DA14v5HQ46SJvL4aHjvuu6lj/0ZJDQSUMdWgIvMMaF6XnrNIULfhn5Ryc3Qs7rBZyUVS+okpVYLHbllVeKyMrKCpcgyuRQdcQOtK3qyGbhIpjQyp5iJUqLdq1Wg7iIx+O0j7ctjysGbTgMh8PaMQitQfhmWCLFT7QlBGjP8+hX7KN8J+iLz7Zfc801ozz6BpsO5grrFfC7UhHFqDarIz/I7IbbCBk9C+svqwCVSgXf5Fwuh+AB1QE30EV0zpkeCZh1gnyi89G2y4igjLpNsoICRFNyQnO1Wq1Wq0o4uK5bLpfJW20fxGuSa9d1saCD0OeN1MED0pm3o4dQB2Uh26AjlBWSdxOsOUmsrKzkcjlOnCKyuLi4srLSJ3GGa2LxpVJptHnA5EA7d8a9kM0AE6BfAZfeaONGqF1To5+NHJKSz+fhsjUiJ5NJfDK1SbLrT2hVvysoBeySgy/s5Mc1c+zcArQ1ZxTCKM9khSiXdaS353kQuEyWIsm1LCubzc7OztLzTf0APVwqlYKj4JgwQAteG9o6GCip3XHEOJGwNtt38dbGGUgw+mXgoGGWmYbV6xgGjWQ3q/yZcw+qm6HTEQOFCdDrDr3mXUvhMcir9m9uBKlUip+Htq3rdshuOs8WBCPy7l4hptN3olqtlstl13V11mIn6PUgBAe3gAajWCyePXsWJgG//Fwuh4IiHo9Ho9FMJkOIpBAnfh9QLpdTvw5aP9LpdCwWw7GIAiPVwkwmMzU1hXhDW2DE17R4/ggryJZUKkXQp3IAGc1nh7s/k2SR/amnEqO7dVJ4VwS5JgiZl156qdeTL1Jw5JG4DKLiN+gPE6DXHdq9spb02bKsZDLJNfggBDSdMuVyuU22AYWqf4MvIUmE/ej60yJ8q4INxpaQR2Dq+iqEa8RK63zHZzWcQzlHnnvq1Kn//d//rVQqy8vLuAMqV44jcy6XK5fLBOt8Po8rqe07n/BG8/PziUQimUxCtsBXkF8XCgVmDHKuUlcmAitkiLaGIuFgvxzHyWQynAPgOjgCXJ2k02n0G8qGd/0QKUJwPeG67je+8Y0LfpQXHfRqabNeJWwkTJHwYsLgIZ78UWdOB0E87fParuwHgcyyrEKhkEgkyOK5nCfsEq16bQ2lGuEP9Vuj0YCQFT+IU7hjIOzZs2ej0ShRWDUV6XSabhSuAOA34A1arRYpfLlcTqfT7GMul8ODaWVlReddYRaaSqVyuRw3lJTAibBYLPKOXGqwgyi4C4UCbyr+HHTbtlHLcRBIpbkEwRWv8+Czv+zdJiMBuCRCTjNgJmHQHyaDNjgHgqbXrWkbtPy5saTY2smN7riPQCUSiRSLRdJh+F9iGYoLfbkyvDrxGgkdVwCVSgWXDE4DxWJR/KsT13VPnDghIsy1UsV0JpMhx6c7MZfL6bAVBO8qEYHNUH0IXSQtf3QAGTQCwaWlJS4aIDfi8TjECKGZ5cHAFgqFzm5yOd8rcZ3qtOMC/EYikUilUobfGAlMgDY4B0yI1H+jrcjD9b74pU7iWigUSqfT8/PzyMs6t8mF/MrKSiwWI6PUbZJ1Tk1NxePxRCIB/0A0JDPVZr98Po9mA2KBjBXH0VAotLS0RFa7vLxMNCen5kxTqVSq1WqhUFCOW0TwyiArJ9sl1YXAUY0H69QIrjZ1bBnnUthqiHvawXVWgGVZXaUawTLAEGIhFj+Zqbfle4dt7kLoRsJQHK+A34z43qHjXs46wuvhrQFXoCbOi4uLsBkklfF4XMezDnh8gkSEmtIxJFDpXVq0laPft2/f3r17c7kcvX9wF2S+8XicqMoWSIF5DrNO0FdAUzDnUPwrA2UtcDSlV56PW6Ok9nASiLlT/aBh0rmRzWbJo9W8kBCvZwgR6aUF4sqj1WplMpm77rprVR9cqVTCzKTRaMzPz6/qtRsAVEOc/AYpZRtcECZAvwJ+xlz/okhb1csHD1vjBcI4y3dl0xGo/Ov57qmVSoWCm0Y0bTMZvP8b6hZTZjX54/wHvUu0VW9+PoJoNJpOp2mxgQlRaz1oYhYA3YkrNAw4KrdKpULoJ1jQTsL5gH+JyIVCAYkIujoJTE5RXxGAfgPKhXBfKBQYYIj4Tx2rPc+jCxwzPOmRRQ5n9sK5Cm9rpW5Wu5H1hnpAXhS/hcnHxH3AY4TaL9i+JfGAL1QrhgvOdho71OtSfJcJvdBWlygt98Hn6nWr+FVKz/MQzF2wHRGDZippbXZ35L+o2TjUdEuTluoKNREmULquy+xtng87ob4csOc6rpCQWigUtJOTwbKFQiGbzUZ8BJekIdjzu+r1oopyIpE67M/cSiaTpNinTp2KRCJTU1OtVov8erhe+T5gYXwQE8Lw6kVn8JQzyd//iw4mQJ8HLzAEZPBXlctlskJtcFi3Ba4VwTQfTx/9lzCHwMDzR4RoChnciOZuvcqJCj1dtbnX65uGQiFqhhqUbd/Sc2VlBQZZzwq0bhO1WSS2ojAwdAlSiiwUCuqGAaWAX4cW5VB3tA0nhMDhXBukOPSgsddQ4clkks8dAmRmZgYmB7eTdbKpi8fjKysrnm+ztx5vsSpo/yf+fyY0jxzj/4wnB/z46ThY7bdfw/qEd7iiTyC0obsg08R6jaoafX2O48AGdGr1sNqQwfaX6Nl2PPHcQHk9MzMTj8eJmCLy8ssvP/nkkwRWpB2pVAo5M2c+rQHqx8S/GGVoBg1nglqDawXqgXDH3F+r1VDgsST4ZX05VDKd3NAjdB5yhkCN12w2C4UCbkr8VWXIBVGr1b75zW8O8kzALkxPT3PExq4A8fy5M5wIJ7NuebHDZNCvAEudIV7IXCvb9wgd+cJGCC21KY+stqKtVmtpaQklMiGsq5JXfOKCy38y2SFWooMOdAYg/LJ2EtIQSCqdTqdFRJ0xCLjcKSLVapU9oqdRyWI2i3cSiZ4KNtBXkFmXy+WpqSlWQvqsRDxBh0yZHdc8naop3YxKPjC7oF6vMyOx/xFotVrHjx8f/IgFGxGHU4CMFsEd7KWCN1gjTAY9AvBbpQ94tV9TcvY+3dUjBySMprQqB2YNdG9T++JG5xbC4TCxm5aTtSxGa3pk05QlRYSaW6FQIDiWSiWiHgGUabDcH5y0QhQmrHMpwJ1a29RuSSp7xWIRJgcChAirz9Trg5Y/OqBYLDLIijZ0LgKg13ka1wFUO9dyWHphdnaWcyft7+vxFquCTv9ai02jQR+YDHo04NupDXUD0tDar8EF8nD5u/jX4/Zg01K6AvEWGSWkM0Gt1/PbdAje+UNm2x6S3py+51uP0q2nWSEJNTZMElDCYR9KlyCN3VwBEHkJ35xayNC5DtA4y1AVtBY0yKjsWnyWI9iordoMvRPihTVDFmFbCmUULGkO9TlcAAjAuzaIjgV8DcxcmPWDOemNDAQIbWge5CXB8BSMC6vCSPzDHMchhaRthBSSaNjnVSpfY691AAqPkpWjYtaFqTBDA3fLN3Ui0iWTyb1796KjIL2lkNhsNqvVKobO1WoV73/XdSuVCnwFqWXTHwyYyWTEP3XR6aekRzQaXV5e5q35lxom0hTLhwS6/igMUppDmEhSz4gWdXCF+dHzU//P1PhBG1wQE5FBNxoNLlRXC36iI1+PrD4XFt9sSG8PQhF6vtE7l+F9DOEGfF/MIobYCKEEvkUvVzXUAkKk5Y/yIv23/AkjlmURoPXqnq0R6dQXCdUd8VQ5aLqobduG0Ni1a1c+nxffAKRarZLGxuNxnEAo5Wl3dct3UFNRCrbRZLgUD7XDMOjB7wV87IIsE3E5KDHkJMQ5g2sLFs95glZ1TjyULmlh56ygRlGdh/01r3nNCL/A6/dz2DBAeXVtkb9YQBIwnM69886JCNCIloZ4IXbsI18PkQg6EoPgQV6lk1NW9SpOA1SfhiMo2t5XXdZGzgkWi0XSUpJr9W4mM+VMFovF0uk0wUsXJn6PGZGRhdXrdVJ1IixboBWQWF+v1zOZDHOqxLcZQuyM3i4SiWC+yikKGTXN5WF/3jYHRNNhzhnqewc1wfFn8mFbJNXaF6uCeadvhaXCn2A0SI0RyoW6MYcL9fQGiC/X6eewkdAhvONeyPAgBRkiT5rcAD1p8AJ2NjqI+oLgI4FTHvzjge1dC4unhvQiglsbzOmAk7EGB4kkE54Qz2l2SQqpk6t0YdhvEqEIo0GRnPZcrKyssBeEObbMYUylUgRWTpmongnoCLfZIJwyxHQmkymXy2TliJSR1hBtCcq2bdNKzrKJy9QkVdOiO04op7IKi8LibdteWloi3MPGUDUlZQ6SGxMuvjSYWJgA3QXaSUiYG/BVlu/7vK5r6wTqQM8f2qLtHqtqhuwDrUDGYjEsOlu+268aJaPEIH3jkp/yEVI5yBC6YEQExyKdXYv/J0ePWOm6bnBilpIn4ncP6j2VSiUej6u4O0hf8KYiop8IBDFnF84N09PTQa5D79dQ3vIN7fhXlc7EX/Jiwr1ac+D2xzK0BbyXk9TS0tLc3NzaPyODzQpTJOwCrmG5Kh+72nRABOtaI9wsHC41MfHnAKAisCwLKyXHcXDHpypIsVQzUJJu5ZFLpRItMGTicv6oxmazeeLEiUceeaTtPMc4lVwuVygUaN3mhEFkJ/mFmIZQplSonfdtKheSdKIq/f2EaXJhzfGVOOIezdypoDabTdd1c7lctVo9e/bs0tISbAzOeexdNBplwkvXY1ur1R566KERflgGmw8XR/TZeFwscbkNZHYwHiMRY2lDhG3biCi0+IZqQrvINE3WZYgItTURqdfry8vLqt4NtiAmEglGrzqOk06nz5w5Q8xVMQwbyefzxH0ckZgXUy6XSdUpV2qOrIMHyY5bgfmw1PdYM2UGJYg5i0igzqN8tObjtu8n1fJtTHT7y8vLyWSSNnSeCeez9k/BYMviogxDBr2gEosRtnVpRklCqhEqaKVPRCNgqXxNX05Rkfkp2nmos3RbrRbaXuUuiO/lcjmVSjE8m45qaoPFYhEKGGs6JECajJPbkqGzVMhoTiFqpIfcIhwO45oU3FOVSBKsgwk+7lHK4SC88zwPyTbnFSbFTE9Ptya+rdRg8mEC9CbEaqMzOWAbYa29Jyh/0QtLYOyWhlc1k1PTuFgspnqMlj9QCkkDVsjisxYiwjOVjiDCshHxu67RThC1iYOEYNhe3pcF4FUEl60t4MFmGd6dZPns2bOhUIi3Vk6D5zcDkwnZEaxQtahLSq46kEQiwRmLwYmqO1zbJ2mw1WG+QFsdBDWCkRqSESIpu/G3s/hJ2ugFnNWIkqFQ6PTp0+SPjuNg/aEpJ/EOUpuqoFp/UFjbuXPn7t27JWBXhAYOEYhqP5aXl8lVRUSV13Rgi4jy0UFGXp2V9G+tVisUCoyDISirQlGjs8Z37SrUJ9DyHo1G6VcMhUJnz57dtm0bk1YueOQdx/nABz6w1s/PYFPDBOitDjoAJZBH03ABvUvI7pqSq6JZS4hwwVTJcMOAE4A1lvOzb/WAhpGAESbi425BjKPzBcWxxlb6ULiH3hbPb0/HbhQWgjYZK2CsoRcK2lKIXI8BWqxWI7VWBaFlNNYrkcJ5RQ8jao1KpaIuTkFwRaIKRe40HIhBf5gAve7QvKxXO9mEQGMQwgaGWJP59soHiZjkrSj8lPEQEWhiks02/oTNwjPQbwJlrGVGaA3CK+GY3pCmPzcrFovxck3em81msVjkIXTTxFaepoFey4YS6J8M3qkctLqbamDlfiJ7IpHgTFYsFhkGlk6n1ZKl81jRj84uZDIZUzw0GARGZre+gEDgRz52A9+uoEcZeoFUlFDVarXUi6P/eUVTSzRqOvEoFAox909FDidOnCBOhcNhioSpVIoJ0CTRhDA8kmg2IR8nseVVIgItrnNPGPOKdoUea/HtloL9+rp34qt0tH6I9IUmQBVv0HDMa4m8qkjhQkGtgtCBwNdrC3sQumuc+dqmvsKkdw3rA6KNzDHYNDABen3h+aYNZILjXk4XhEIh5jZR17IsKx6Pk+jFYjGu+oPPD8YC+qqJpHAFJJLxeNxxnGw2CxtQKpXOnDmD5mxlZQU3zlwuJyIw3YRyKm+Li4u/+MUvEGaoCz6nCp0VyQKIjPF4PJvN8i9zPYjd7Iv2oXDWsc6fMKDhmzaTYrFo+xARldDpFqyANYfqRjAAwW8Wdz3YjOBBU8clCQy1qdVq3/72t6nEUjIdLkajyC6Xy12HiBtc1DABen1h+e123gTPOW7T/IZCoVQqhTEF1LA+hKYNwZz4KmnP8+gTcRwnmUwSVZW1IAKSWXObDFT8WI/PqoatUqmEjT1JMQRFLpcj9Nfr9ZmZGfJQGlVIwPERVW0yMbpcLut4FPG9MdULW0Ud+XyeOSlc5agpHX3qQaMl8ae10odCmC4UCgToTCYTDuNh4iYAABwkSURBVIeZtMIRCB5kzoJQ23Df+Xz+0KFDOi8t7I8DXy0qlQoqGtuMNdl0MBz0+oJfo8oAxr2cQUF5EMI3eMFOLBDfEYZARkmQiJZIJBjJallWPp9Ha4w+RBV7kUhkeXlZRXWJRAJ2GFoWWw9tlY7FYtQMKf2VSiUmasNXVCqVM2fOEPU4DSC2g9xQ3qAtr6SESIAul8vYcfB87uSTIiUPZtNyfnmTvYaADmrAxfchCXL3PJOTNCIZoqpampjk16ATJoNed5Du8WMe91pWgc5Ody8wcBYQSckldUqeavWgSsij0+k0RkKzs7NYFIkIr8UEDn4WVkT8IbNBcQUCD6YBoAsk/CUSCSa9Qiwo3Q+NLr6Vh2pRPN9mGtkfayAjVrdScm014uA5eq7SLWOMmc/nkXNAMmhQ7vpx6zKsQCOPuusNdwrnU1Aj7CG2YDCxMBn0hAI5RD6fZ7zTJPzwaA8ha1btcyqVyufzRDS8h0gPxXdu0m7DZDKZTCZhHrQ3BJsOcnC46XQ6vX379mg0WiqVCOuMR0FSgjAjkUi4rouuQ9UXLAxtn3LEEpgJgDBO02RVNLuuOzMzowroZrMJK8UKEX4Qr23fJBo5NkSKbdtnz56dmpqiLEm4jMVifVrtLX8A/P79+3GaXaOdIcR35xnU4GKHCdATilqtVi6XSULFN9ZQWzV9Gqyr+MK19V6VtnprrmfbdiqVUlUyS6XeRU4dNN9otVqFQgFZSzabDfnTZMhA2buZmRncOaCYcTJSKpxOQgqbOq5Q16N0f9v7Kgdt+a6hnt8GqR7QkUiEyS9QJeJLLwjrWmBUobRWHXWFmCKps1LXWNlqtUj2ydZvvfXWkXxqvbTqBhc7xp+XGXSF6tVUAKDjnVSo63ke/7ZareGmsQwBok/wHo16/AvRoZU6XaplWewCcc11XXQdJKfNZhMDUs+fMhMOh5eXlynZkZMSpmEYoIBJ5KF3Lb/rWuVxGoitwGAtslcJnDNgMPL5PGEd3pnXEm01cQ4WS1khh0L1f9o43jaMRg9CqVRSpZ21DkMVDDYZzPdjQqEzQWAnNS+jV43neH4vspzv6TMIUMi5rtsMjE9EybDGapXnjzsIhUKYGWHwxokE5w2isLr7t1otEnDCXCKRoLqou6a9ghpVCfFsk60RQJW4UMcPnaKiO6inBMuyisUiObIGcY6AbkdjKCcG3TXxO3E8f8g3SkE9CF0PjoZ4k/AaDAIToCcU0Wg0k8nEYrFUKtWr64zeB3jY1fKPpHJwC0QT7RDplNMqkdJrGV3vgXlIJpMYG+VyOdJbNM6wJZyHyJ0LhYKOhtGoyr7DZqhnk8ZHJNKIuCORCEeMCI40W6V1nj+6UMkNTfBRIlN4VOGgfb7XqGbQrYBRHy59yWQylUrBAjf9EbpUQduODItv+aBzcvCPzGALwnDQkwsdIyJ+gqx6CfEdivUHr9qJPlA1QtCunrBF53TQeFN5DPW063wXpVaUrBA/u8QLlGBHeksSnc1mlWegHIdXkTbXHT169Kc//entt98uvh8TRUKiIVkzc2BhP8Q/URFb9TJCqSElN5i3ooYeOuFFp2dxv5LObRI9bePmAHr+UC5oZTZI2zot7/yFbtaDqcVVtChf/vKXP/ShDw37BTHY/DAB+uKAZVmMj1ImVEUFtm8H2n8LUA1B9RiBj4eCKsC2vE+1z3K+0q7RaOB9QSZOig33Go1GO8eIsHJ4DMI61AeS51gshl0nWS1+IHTEkAKzDJ6sg191RjijILV/RNlnPVCk6toHiNqEf1HpcRYhOjf9WbfqH60qDj3mnN5YGJoT9oUNcs4g/a9UKpzYtJw71FfAYCvCBOiLCcGaErkeGZyaoqnAoFPUodQwwQvfCe3vEJF4PI7AIBaL6WuDwVrlENzGBo8EmaCjFv4aVcUX52GRgWRCfAUe/Ewmk/H8aQDikw9E8JA/+g9LEwZuscjFxUUyWZ1WhQJPGRsCsSqaW/7gcDvgqsE5o1KpZLNZzJX05AcJwzZ1DpaGfvVd4lTBpQxPwF6V7ROpuXBBna2lRcNBGwwCE6AvVti2jYdc0J2HKEDxip4Ofb4GLPJH27YhECjQEXzbXiK+5RDNJnp5DluNCbJGn1BgjrV1fl9cOBxOpVLYaIgImjy0Fp4/xpBAXCgUlMkJvmm9XnccJxKJrKysIIbTWKkyDLJaCQzc4oBYAbcN3k75ZTJoyrAancl8qf5JQD2tR0YT7UQigcUSFA0tkdpywrI5LUHrw0etrKwgmlxjMdZgK8AE6IsYQY0B6JwKqFDTZ3Wto+NZfG6alDxImOrz4Ss0u9QuZ5Ta4rt50DPNmaCtsBnUqLF9JN5QGYVCgTJgLBabmpp685vfrMQCvtIiwuwr27ZxENXto8BzHMd1XfHN+ZS+YE91PAoOUPQZkt7CGlFO1MQ5qIFrBfzwJMAsEXxDoRAKPyTb9KdQsSRAq2KvWCwSoLlq4SR6xx13jOi7YLA5YQL0poIVsFvrvIhu69vW+J7P59PptOV7v+lzyJRJojVCKQ3NPBGyb9U+t7WxdIKXk7zTisIbEfRrtdrU1BSOd+K3ddj+gFcyWa4buFZgC6FQCJpF6YugAIPdhK0mU1Z2Qk8bmGkQwcU/hQQLjHrlEeR52GWIJnghPEAk0BRDkZO4rycGy5+RiPdeeKQzJA02E0yA3lQgeAX7+vpAGQ+yOTI7fVQNfcRv1OZ+lazBVmt5EPR/Uzo1iI+IN8hDRQQegKQe+QrhslQq4eRJJVCLikTqcDiso7jJapWPbhPJoanQ8xahk5hu+SNXyHnJzYnj1vnjrzT0B6UgODehCeFprutyjxUYhxgOh7na0NYbHeTo+Z60pm/FoA0mQG8qkC0OYuyAmgIrDM/zqLxNT08HY3RQ1xHUb+ApKqsPKKVSaWVlJZlMUs1LpVLKjdBYCFGAIwf5ezqdphBaKpWgoeHE+Yt2gjDHmUMHIyh9YfkdhrY/ToXTElVK8e3/xVezIE8mYqrYw/NHagVVdzAY7EgkEiFZlvN1Gtiiip/XWwGTJlUE6tSxrrMfDbYyzBl76wKOguKVlu+CAUhbKtTmTaFE6uBvh7sInZBYaogfm1ShjBz7xRdfhFbetWvX9PQ0GS70LvcHmwwJu2TcGoWVu7B98yNlFcSP3ZzJCO5qBdfwp1tp/VD8ExXnJCWRyJ05gMHkWk8MNMJQAkUNooeLdVYqlccee0wVLwgHTeXQIAgToLc0oIM1/cQBmYfIMa2OcYJDgwhI2MIPWnxFBA0sqVSq0Wjkcrl8Pr9t27b5+flQKOQ4zvT09I4dO9LpNB5yKysr4s/9g1xmkgvvQgmUmKgEDu2Ftm+pQfaqbeKEV1XRqWBDw2Uw71brakIqVwOEbOa5UCRUaQq8vA4x0KNBk+RTTz2lc8XEZ2NGcrQNNgdMgN7S0JjVarUgHDRbhNt1XZf+b2TCrYAL/mqhBvyhUGh6ehrXJHJzNG3xeHxqampmZga+O+xPDSdKlkol13WJejAJBEFIbcqVIoKej/2iO9H2nTdUy6EtjuLr/IKJtuc3hevR8ALDrrhuILzGYjHqq3hS277HSDgcdl1X+RMKg9AajcAoLPJoNoUkHJLawEBhOOgtDcuyksmk4ziEOa0TBhXTyAxWVlbi8Tg91iLCzBTPb34ZBOFweHp6emVlhXdBH6KaNnWOJovXRJ5RLGfOnOFVS0tLKNsI1gRBdHVIO5aXl7Xih6JOuY6WP9ubc4wGXy3Q0RGj6bBWAlu+pz4yEmjiqampVCpl+4NgyNNpd4TIJlJzVuM5vY4V+25yZ4NOmABtIKFQKJ1OB+/Rpg/yTegOlGooH1D1avjuv30tMEajUTUPQbpH/mj7bYdI1mZnZyGpCZR0rFBIJLyqZJuMnk53ojydhIRX7ffzAqb+yDk0arf596uKro3TgP8hxJPaM3EcaxFSde1wgQBhp1B3aB95sAZo2/bu3bu5vfborPtosJlgArRBF5AY0gVHcqpOFzyhWq1yPW73tTklvHqeR2VPRBh1KP4AFPG1d5wGQqHQjh075ufneRRtCb3sym+gehYRSost30qfXFVLeeKXECEuLN87lBAc8j34gww7ZyBtYNGmEtJnFdJpJh7yLadd11V2m+NGQq3SbI5SsIdeRKLR6G233TaSz6vZbGIn4rruILZZBhcLTIA26ALLn1rCvypfQ8CLFao+pG4ebfB8vw7xYx8KM+JpPB6v1WrwEthxsKlqtcqUP9ufS0K4aTQaTL2yLAs1tDaGaOOJBPqnSaKDcmaEz54P8YUZRF42CB0f1BSqSk8rijR5E8EbjUalUsEnhP3CLgqio1Qq4aTKvq/xQ9Ec2epw+odaUTL9gtc0BhcLzAdpcGHAHiAs0+BFdkyVr08STSBTHZsVGBNOZIQ6EN/pSe3zQ6FQoVAgPtZqtUwmgw8RbYeUBDXKawSk3kh2zDs2/TlYGq/1JCG+N5MSzdrnogyJ+nuwfTgKDfGVSqVYLIoIhT7OQ3TDI01hdzoPizawDP4pKJnTbDbV2MRgc8MEaINVQPNZ8S2QxM9DuyISiRSLRTJcGjqCj6ouAlWGapNFBPGcGpnS0k0ox7IDklpFxG11vJA/gUXF0ZZviuSdP2bQDjQKsoMk2jwBQllPMLTMoGmBPed8QwcN22F3mv74mE7bVbJ1WJpOd6o+UKMVzj3BuE/bDm86KlmkwSTAyOwM1gsEr0gkwhzuIFuNgI/ebqV90TwcPXr0P/7jPyKRCFPAQ/5QFUR1juNkMhmiIUI9aowIKjQF1sirwjiKgaqe1usAz29b5199Iew2pwS1DNRCH/uFABHZiW3bjFah2Yf9CmqcFbA61Fe/8IUvrOp4agHT6ugboovHZNabDCZAG6wjbN8zSM7vwqClUEVv6B9C/sRVBn1lMplkMomvKU9jqBW8MwIJiAXitZInJJJBSQbwzveJVpUIIVjVdUqwKPuMbE7jOKoMSpSu65bL5XK5zPIcx0mn081mc3Z2FnF05zHRqxDLH187IDBa6WWu1ElMG2wCGIrDYH2hso2gusDrPQcg7IOB5dPT0yKCTALD6HQ6DY9cKBTopikWi5iX8hxybZ2crYU18YOj5bv98b5quEE4VkM7z7fl43yg5wDOKFQs9fSgXYjT09NTU1NewCywDUqFq6R6QJCkD/MZGFy0MAHaYH3RNU5hN2oHpgfAKasQTYez5PN5pH7wvxpSuajXPhc6aCx/3CIUtgQ6wtWVP6h31soh6SfkA5SFUiJaPxSfW1AfJfHH6aLvFt/kWnwNCYqXtoTXsixEIKtq8zHYmjAB2mAMUOdo5UBSqRThb2FhAa0Iz8RGg0odXHO5XI5Go6lUCguOcrlM2NUtE5QJx+L37xF2g4ITjY9a/SMBt/yubgl0Y8NBa5imdAkbE4lEpqam4vE4z0GaUiqV1Opa23AU7LLjOO985zs35HgbXKwwAdpgPOhMHomJ8XicPFRNP3SeC2QCbsvEOHXFo1SI8ZP28tHnotsP+ZOrVPKhbIMugPKdJtfarqIibv6Kn0HPzs7Ozc1xBYA2o3PSWJ8jsGvXrhEdToPNCROgDSYRpKuYKWMBqoFPeWTP87Zv3w4HDcOA9oNuER1viCRZRMig0c8hxWv601Uk4NesHAjWTmTuZNaoMlqtFl2UjuMwXVBJahX8xWIxOHGjezNYC0YZoA8ePPjEE094nnfXXXft2LFjwIcMDDpBrFR2IpiHMptVfBHxzp07sdwLh8O4HYmvHmm1WvT1BUd5EXDFL0jqsG0No8RxSBUSarZM+lytVmdmZlKpFOaitm3TEN+2fpymZRQmGwZbGSOrUSwuLj7yyCMf/ehHf/u3f/v+++8f8CEDgzbkcrnDhw8H72mLcZbf1kicDYfDmUxmfn5+eno6lUpls9lsNpvJZKLR6OzsbDqdxlZfh5UERdCI6giv0Bp6YqD2SGimIDk3N5dMJvfs2bNjx45kMhnM6OG70cBpoIcV6ROgG43Gk08+OeLDZ7C5MLIM+tChQ1dffbVlWfv37z969OiADxkYtGF5efnQoUNXXHHF4C/pDIKRSGR+fr7RaEBBnDp1ihm1jUajXC4Xi8WghEN8elpbV3Q+C9F5x44djuNQmVQ7DhHBMRXeI5lMam4+IBqNxmOPPXbNNdcM/hKDrYaRBehisciPQXxt0wUf+tznPnfw4EERueWWWxYXF4d4UxqIh1zxJKFUKo17CWsCjm4j2dTy8nKhUBju+9AJ1M14YsApE2qLxaJlWepTqryz6uoikUg8Hk8mk7FYDMaZ+Y3amC6+6m7ob2C1WnVdd1R7Kpvi5+B5Xj6f35q8UFdDm5EF6FQqderUKW635RG9Hrrzzjt/8zd/U0S+9rWvTU1NDfGmpVJJQ/9FCh0IMu6FDA9Uw5087HBYXl5OJpPDfR8uCPVjsm3bdd0zZ84cO3ZMFXjKaezcuRPlnIiEVjl6cXBUKhWGyIxqg5vg55DP5+Px+EU9ObersHIQrG+APnDgwMMPP+x53rPPPrt3714R8TxvZWVlamqq8yEwNTXFt1MHeawWwx2IiYIqbce9kDVhhFqFRCLRNlx8tFAvPcdxZmZmXvWqV33nO99BCt1sNg8cODA/P6/d2OuKSCSyffv2Ee7pJvgibYKfA6f/IXah61duZAF6Zmbm5ptv/shHPiIid999t4iUSqV77733vvvu63zIwKAXdu7cuXPnzo15L34St956K2bTtJVvGKLR6B133LGR72hw0WGUMrt3vetd73rXu/TfZDJ53333dX3IwMDAwOCCMFYABgYGBhMKE6ANDAwMJhQmQBtMFo4ePfqtb31r3KvYCFSr1QceeGDcqzCYaJgAbTBZYFLUuFexEfA8TyXVBgZdYQK0gYGBwYTCBGgDAwODCYWxGzWYLGzbtu2i7qscHNFo9O1vf/u4V2Ew0TAB2mCywDDvca9iI2Db9r59+8a9CoOJhqE4DAwMDCYUJkAbGBgYTChMgDaYLBQKhWPHjo17FRuBZrP59NNPj3sVBhMNE6ANJgtnz5796U9/Ou5VbATq9fr3vve9ca/CYKJhArSBgYHBhMIEaAMDA4MJxUTI7J566ql77rlniBcy9HPk69lIlMtl27ZHNY5kLND5fiPZWqlUKhaLjzzyyEi2NiCazWa5XE6lUhv5pq1W66WXXnruuedGtcFN8HMoFouxWOyinqjSarWGG/jA4Pn2Eck6jn6M+Pa3v10oFMa9ivHg3/7t32ZnZ9/85jePeyFbGidPnnzooYd+//d/f9wL2eq4//77b7nllssvv3zcCxkPUqnUr/zKrwTvmYgM+h3veMe4lzA2PPXUU3v27HnPe94z7oVsaTz99NPf+ta3zKcwdhw8ePCmm2668cYbx72QScFEBOitjGuvvXZ2dnbcq9jqmJ6e/rVf+7Vxr8JA3va2t23fvn3cq5ggTATFYWBgYGDQCaPiMDAwMJhQGIpjDGg2m/fee++xY8d+7/d+79Zbb+XOgwcPPvHEE57n3XXXXTt27BjvCrcUzJEfI8xvoT9CH/3oR8e9hi0Hy7Kuu+66aDSaSCTwM1tcXPynf/qnj3/845dccsk//uM/3nLLLeNe41aBOfLjhfkt9IehOMYAy7JmZmaC9xw6dOjqq6+2LGv//v1Hjx4d18K2IMyRHy/Mb6E/TICeCBSLxUQiwe1WqzXexWwpmCM/aTCfSBCGg944fPOb33zsscf27t175513tj2USqVOnTrF7VG15BkMAnPkJw3mEwnCBOiNw2233Xbbbbd1fejAgQMPP/yw53nPPvvs3r17N3ZdWxrmyE8azCcShNFBjwef/vSnjxw5EolEXv/613/oQx8SkYMHDz7++OMicvfdd2/xyvUGwxz58cL8FvrABGgDAwODCcVWp3gMDAwMJhYmQBsYGBhMKEyANjAwMJhQmABtYGBgMKEwAdpg/HAcp1KpDP78paWl97znPbt37969e/ftt9/+0ksvcf/c3NzCwgL3f/CDH1zV24XD5ySn9Xr9Ix/5yL59+/bu3XvllVfec8899XpdOkZdGBhsAEyANrj40Gq13v/+9z///PPHjh277LLL/vAP/1Af+s///M/jx48fP378S1/60nAb/93f/d0f/ehHP/zhD1944YUf//jH2WzWdd0RLdzAYHUwAdpggvDNb37zDW94w+te97pbb7318OHD3PnVr371wIED119//ec+9zny3Lm5ufe85z2RSMS27VtvvfXYsWOr2qDioYceOnDgwC233PKxj32Me5599tkHH3zwgQcemJ+fF5FkMnnvvfdOTU3x6J/92Z9dc801l19++cMPP8w973//+6+77rrXv/71d9xxx/LyMndalvXpT3/6fe973/XXX6/P7NwLEXn00Uff+ta3vvGNb7zhhhu+853vrPn4GWw6eAYG40YsFiuXyydPnpybm/vlL3/ped4XvvCFG264wfO8EydObN++/dixY57nfeYznwmFQsEXNpvNd7zjHZ/85Cf5d3Z2dufOnXv27NmzZ8/Bgwe7brDt7Y4cOeJ53qc+9Sm2/M///M+vec1rui5SRP76r//a87wnnnji0ksv5c4zZ85w4xOf+MS9996rz/yXf/kXz/Oee+45ntl1LxYXF1/72teePHnS87xjx45deumllUplJMfTYNPABGiD8YOI+dBDD/3qr/4q9zQajWg0ms/nH3zwwXe/+93cmcvl2gL03Xff/e53v7vRaPDv7Ozsz372M3206wb17R588MHbb7+dRxcXF9nyV77ylauuuqrrIkWkXC5zOx6P12o1z/Puu+++W2655eabb37961//zne+U59ZrVa5nUgkarVa1714+OGH5+fn3+pj//79zz777HAH0GCzwnhxGEwKPM/rLMR1vRN8+MMffvHFF7/+9a+HQqHBNxhENBrlRiQS4ca111575MiRM2fOQHG0wXEcbti23Ww2f/KTn3z2s5/9wQ9+MD09ffDgwb/8y7/s3LJlWc1ms+tKWq3Wq1/96u9+97t9VmiwxWE4aINJwY033vijH/3o0KFDIvKlL33pDW94QzqdvvHGG7///e+/+OKLIvK3f/u3+uQ/+IM/OHbs2Ne//nUNhQNuMPjo448/nsvlROTgwYPcuX///jvuuOMDH/gAhmqu63784x/nOZ1YWlrasWPH9PS0iHzlK1/pv2ude3HzzTf/4he/+Na3vsW/P/jBDy5wgAy2HkwGbTApWFhYuP/++9/73vfWarWFhYV/+Id/EJEdO3b8xV/8xdvf/va5ubnbb789m82KyE9/+tO/+qu/2r59OzM4duzYgbfOIBsMPvqZz3zm9ttv37dv32WXXab3P/DAA3/6p3/6pje9qdFoxOPx9773vclksuuCb7311vvvv//Xf/3Xp6end+7cefr06V671nUv5ubm/vVf//VP/uRP/viP/7jRaFxzzTVf/epXhzlwBpsXxizJYNLhui4h8stf/vJXv/rVb3zjG+Ne0TDYHHthsMEwGbTBpOOTn/zkv//7vzebzdnZ2S9+8YvjXs6Q2Bx7YbDBMBm0gYGBwYTCFAkNDAwMJhQmQBsYGBhMKEyANjAwMJhQmABtYGBgMKH4f7AJxOEm2cEyAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "# significant thr \n",
    "pvalue_lim = 0.005; log2FoldChange_lim = 3\n",
    "res$sig <- as.factor(\n",
    "    res$pvalue < pvalue_lim &\n",
    "    (res$log2FoldChange > log2FoldChange_lim & res$pvalue < pvalue_lim) |\n",
    "    (res$log2FoldChange < (-1)*log2FoldChange_lim & res$pvalue < pvalue_lim)\n",
    ")\n",
    "\n",
    "relevel(res$sig, ref=TRUE)\n",
    "# Volcano plot\n",
    "\n",
    "v = res %>% ggplot(aes(x=log2FoldChange, y=-log10(pvalue),colour=sig, fill=sig)) +\n",
    "        geom_point(aes(color = sig),alpha = 1/10) +\n",
    "        ggtitle('Decitabine_treatment') +\n",
    "        xlim(c(-15,15)) +\n",
    "        ylim(c(0,10)) +\n",
    "        geom_hline(yintercept=-log10(pvalue_lim), linetype=\"dashed\", alpha = 4/10) +\n",
    "        geom_vline(xintercept=log2FoldChange_lim, linetype=\"dashed\", alpha = 4/10) +\n",
    "        geom_vline(xintercept=(-1)*log2FoldChange_lim, linetype=\"dashed\", alpha = 4/10) +\n",
    "        scale_color_manual(values = c(\"grey\", \"red\")) +\n",
    "        theme_bw() + theme(legend.position=\"none\") +\n",
    "        geom_text_repel(\n",
    "            data = subset(res[order(res$pvalue),], sig == TRUE)[1:10,],\n",
    "            aes(label = name),\n",
    "            size = 5,\n",
    "            box.padding = unit(0.35, \"lines\"),\n",
    "            point.padding = unit(0.3, \"lines\")\n",
    "        )\n",
    "ggsave(\"plots/peak_Volcanos.png\", plot = v, device = 'png', dpi = 300)\n",
    "ggsave(\"plots/peak_Volcanos.pdf\", plot = v, device = 'pdf', dpi = 300)\n",
    "v\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>name</th>\n",
       "      <th>sig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ENSG00000106665.15_peak</td>\n",
       "      <td>65.956153</td>\n",
       "      <td>-10.823620</td>\n",
       "      <td>1.826728</td>\n",
       "      <td>-5.925140</td>\n",
       "      <td>3.120320e-09</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>CLIP2</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000100647.7_peak</td>\n",
       "      <td>46.647159</td>\n",
       "      <td>-11.770360</td>\n",
       "      <td>2.001166</td>\n",
       "      <td>-5.881751</td>\n",
       "      <td>4.059484e-09</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>SUSD6</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000160075.11_peak</td>\n",
       "      <td>2428.149124</td>\n",
       "      <td>-1.127583</td>\n",
       "      <td>0.203241</td>\n",
       "      <td>-5.548009</td>\n",
       "      <td>2.889417e-08</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>SSU72</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000100029.17_peak</td>\n",
       "      <td>85.335107</td>\n",
       "      <td>-9.566796</td>\n",
       "      <td>1.728609</td>\n",
       "      <td>-5.534391</td>\n",
       "      <td>3.123110e-08</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>PES1</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000169047.5_peak_3</td>\n",
       "      <td>28.471412</td>\n",
       "      <td>-14.965875</td>\n",
       "      <td>2.782342</td>\n",
       "      <td>-5.378877</td>\n",
       "      <td>7.495187e-08</td>\n",
       "      <td>0.000140</td>\n",
       "      <td>IRS1</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000223959.8_peak_2</td>\n",
       "      <td>40.992590</td>\n",
       "      <td>0.001346</td>\n",
       "      <td>1.616918</td>\n",
       "      <td>0.000832</td>\n",
       "      <td>9.993358e-01</td>\n",
       "      <td>0.999656</td>\n",
       "      <td>AFG3L1P</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000141524.15_peak_2</td>\n",
       "      <td>407.786635</td>\n",
       "      <td>0.000311</td>\n",
       "      <td>0.465432</td>\n",
       "      <td>0.000667</td>\n",
       "      <td>9.994677e-01</td>\n",
       "      <td>0.999663</td>\n",
       "      <td>TMC6</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000100324.13_peak_2</td>\n",
       "      <td>594.708856</td>\n",
       "      <td>-0.000262</td>\n",
       "      <td>0.471343</td>\n",
       "      <td>-0.000556</td>\n",
       "      <td>9.995562e-01</td>\n",
       "      <td>0.999663</td>\n",
       "      <td>TAB1</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000105245.9_peak</td>\n",
       "      <td>429.217922</td>\n",
       "      <td>-0.000004</td>\n",
       "      <td>0.416239</td>\n",
       "      <td>-0.000011</td>\n",
       "      <td>9.999915e-01</td>\n",
       "      <td>0.999991</td>\n",
       "      <td>NUMBL</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000145990.10_peak</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GFOD1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9367 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              baseMean  log2FoldChange     lfcSE      stat  \\\n",
       "ENSG00000106665.15_peak      65.956153      -10.823620  1.826728 -5.925140   \n",
       "ENSG00000100647.7_peak       46.647159      -11.770360  2.001166 -5.881751   \n",
       "ENSG00000160075.11_peak    2428.149124       -1.127583  0.203241 -5.548009   \n",
       "ENSG00000100029.17_peak      85.335107       -9.566796  1.728609 -5.534391   \n",
       "ENSG00000169047.5_peak_3     28.471412      -14.965875  2.782342 -5.378877   \n",
       "...                                ...             ...       ...       ...   \n",
       "ENSG00000223959.8_peak_2     40.992590        0.001346  1.616918  0.000832   \n",
       "ENSG00000141524.15_peak_2   407.786635        0.000311  0.465432  0.000667   \n",
       "ENSG00000100324.13_peak_2   594.708856       -0.000262  0.471343 -0.000556   \n",
       "ENSG00000105245.9_peak      429.217922       -0.000004  0.416239 -0.000011   \n",
       "ENSG00000145990.10_peak       0.000000             NaN       NaN       NaN   \n",
       "\n",
       "                                 pvalue      padj     name    sig  \n",
       "ENSG00000106665.15_peak    3.120320e-09  0.000019    CLIP2   TRUE  \n",
       "ENSG00000100647.7_peak     4.059484e-09  0.000019    SUSD6   TRUE  \n",
       "ENSG00000160075.11_peak    2.889417e-08  0.000073    SSU72  FALSE  \n",
       "ENSG00000100029.17_peak    3.123110e-08  0.000073     PES1   TRUE  \n",
       "ENSG00000169047.5_peak_3   7.495187e-08  0.000140     IRS1   TRUE  \n",
       "...                                 ...       ...      ...    ...  \n",
       "ENSG00000223959.8_peak_2   9.993358e-01  0.999656  AFG3L1P  FALSE  \n",
       "ENSG00000141524.15_peak_2  9.994677e-01  0.999663     TMC6  FALSE  \n",
       "ENSG00000100324.13_peak_2  9.995562e-01  0.999663     TAB1  FALSE  \n",
       "ENSG00000105245.9_peak     9.999915e-01  0.999991    NUMBL  FALSE  \n",
       "ENSG00000145990.10_peak             NaN       NaN    GFOD1    NaN  \n",
       "\n",
       "[9367 rows x 8 columns]"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res = %Rget res\n",
    "res.sort_values('pvalue')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Enrichment analysis\n",
    "### iPAGE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17689\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 9343 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 9343\t17689\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "849 hits go_terms_num = 17689\n",
      "cat 210, mi = 0.00307092, regulation of G2/M transition of mitotic cell cycle\n",
      "cat 1308, mi = 0.00256618, spindle pole\n",
      "cat 4456, mi = 0.00249268, histone acetyltransferase complex\n",
      "cat 2836, mi = 0.00235159, DNA biosynthetic process\n",
      "cat 1598, mi = 0.00232790, transforming growth factor beta receptor signaling pathway\n",
      "cat 1876, mi = 0.00222928, catalytic step 2 spliceosome\n",
      "cat 567, mi = 0.00220096, positive regulation of osteoclast differentiation\n",
      "cat 8718, mi = 0.00218311, histone H4-K5 acetylation\n",
      "cat 8719, mi = 0.00218311, histone H4-K8 acetylation\n",
      "cat 1274, mi = 0.00213743, G2/M transition of mitotic cell cycle\n",
      "cat 2092, mi = 0.00213609, filopodium\n",
      "cat 4488, mi = 0.00207964, RNA phosphodiester bond hydrolysis, exonucleolytic\n",
      "cat 3295, mi = 0.00207051, damaged DNA binding\n",
      "cat 183, mi = 0.00205180, single fertilization\n",
      "cat 3305, mi = 0.00203632, inactivation of MAPK activity\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 210, mi = 0.003, pvalue = 0.000, regulation of G2/M transition of mitotic cell cycle\n",
      "Passed: cat 4456, mi = 0.002, pvalue = 0.001, histone acetyltransferase complex\n",
      "Passed: cat 1598, mi = 0.002, pvalue = 0.001, transforming growth factor beta receptor signaling pathway\n",
      "Passed: cat 1876, mi = 0.002, pvalue = 0.003, catalytic step 2 spliceosome\n",
      "Passed: cat 2092, mi = 0.002, pvalue = 0.003, filopodium\n",
      "Passed: cat 4488, mi = 0.002, pvalue = 0.004, RNA phosphodiester bond hydrolysis, exonucleolytic\n",
      "Passed: cat 183, mi = 0.002, pvalue = 0.001, single fertilization\n",
      "Passed: cat 3305, mi = 0.002, pvalue = 0.003, inactivation of MAPK activity\n",
      "Didn't pass (cutoff=0.0050): cat 1406, mi = 0.002, pvalue = 0.007, ficolin-1-rich granule lumen\n",
      "Didn't pass (cutoff=0.0050): cat 4771, mi = 0.002, pvalue = 0.008, DNA-dependent ATPase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1968, mi = 0.002, pvalue = 0.009, secretory granule membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1124, mi = 0.002, pvalue = 0.008, sulfotransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 1714, mi = 0.002, pvalue = 0.009, nucleotidyltransferase activity\n",
      "Didn't pass (cutoff=0.0050): cat 945, mi = 0.002, pvalue = 0.010, snRNA transcription from RNA polymerase II promoter\n",
      "Didn't pass (cutoff=0.0050): cat 6848, mi = 0.002, pvalue = 0.006, response to molecule of bacterial origin\n",
      "Didn't pass (cutoff=0.0050): cat 3800, mi = 0.002, pvalue = 0.013, mRNA 3'-end processing\n",
      "Didn't pass (cutoff=0.0050): cat 2704, mi = 0.002, pvalue = 0.005, receptor signaling complex scaffold activity\n",
      "Didn't pass (cutoff=0.0050): cat 1161, mi = 0.002, pvalue = 0.016, cellular response to insulin stimulus\n",
      "Didn't pass (cutoff=0.0050): cat 1976, mi = 0.002, pvalue = 0.013, PcG protein complex\n",
      "Didn't pass (cutoff=0.0050): cat 2797, mi = 0.002, pvalue = 0.017, chloride transmembrane transport\n",
      "Didn't pass (cutoff=0.0050): cat 6379, mi = 0.002, pvalue = 0.014, vitamin D receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 4537, mi = 0.002, pvalue = 0.011, negative regulation of Notch signaling pathway\n",
      "Didn't pass (cutoff=0.0050): cat 3699, mi = 0.002, pvalue = 0.012, NAD+ ADP-ribosyltransferase activity\n",
      "Passed: cat 2768, mi = 0.002, pvalue = 0.002, interleukin-1 receptor binding\n",
      "Didn't pass (cutoff=0.0050): cat 6958, mi = 0.002, pvalue = 0.012, embryonic hindlimb morphogenesis\n",
      "Didn't pass (cutoff=0.0050): cat 3608, mi = 0.002, pvalue = 0.023, polysome\n",
      "Didn't pass (cutoff=0.0050): cat 2135, mi = 0.002, pvalue = 0.022, negative regulation of type I interferon production\n",
      "Passed: cat 4493, mi = 0.002, pvalue = 0.003, regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 432, mi = 0.002, pvalue = 0.027, embryonic organ development\n",
      "Didn't pass (cutoff=0.0050): cat 6186, mi = 0.002, pvalue = 0.025, isotype switching\n",
      "Didn't pass (cutoff=0.0050): cat 1495, mi = 0.002, pvalue = 0.028, regulation of glycolytic process\n",
      "Didn't pass (cutoff=0.0050): cat 11749, mi = 0.002, pvalue = 0.017, microtubule minus-end\n",
      "Didn't pass (cutoff=0.0050): cat 2138, mi = 0.002, pvalue = 0.030, necroptotic process\n",
      "Didn't pass (cutoff=0.0050): cat 935, mi = 0.002, pvalue = 0.029, cellular response to glucose stimulus\n",
      "Passed: cat 8012, mi = 0.002, pvalue = 0.003, ribonuclease III activity\n",
      "Didn't pass (cutoff=0.0050): cat 5296, mi = 0.002, pvalue = 0.019, positive regulation of G1/S transition of mitotic cell cycle\n",
      "Didn't pass (cutoff=0.0050): cat 4044, mi = 0.002, pvalue = 0.008, activation of NF-kappaB-inducing kinase activity\n",
      "Didn't pass (cutoff=0.0050): cat 2854, mi = 0.002, pvalue = 0.023, DNA-dependent DNA replication\n",
      "Didn't pass (cutoff=0.0050): cat 175, mi = 0.002, pvalue = 0.029, protein-DNA complex\n",
      "Didn't pass (cutoff=0.0050): cat 1155, mi = 0.002, pvalue = 0.006, white fat cell differentiation\n",
      "Passed: cat 15689, mi = 0.002, pvalue = 0.002, meiotic mismatch repair\n",
      "Didn't pass (cutoff=0.0050): cat 4827, mi = 0.002, pvalue = 0.018, carbohydrate biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 4552, mi = 0.002, pvalue = 0.022, GTP-Rho binding\n",
      "Didn't pass (cutoff=0.0050): cat 7790, mi = 0.002, pvalue = 0.020, integral component of peroxisomal membrane\n",
      "Didn't pass (cutoff=0.0050): cat 1778, mi = 0.002, pvalue = 0.020, fatty acid biosynthetic process\n",
      "Didn't pass (cutoff=0.0050): cat 1977, mi = 0.002, pvalue = 0.007, PRC1 complex\n",
      "Didn't pass (cutoff=0.0050): cat 4664, mi = 0.002, pvalue = 0.006, presynaptic active zone\n",
      "Didn't pass (cutoff=0.0050): cat 6968, mi = 0.002, pvalue = 0.037, histone methylation\n",
      "Didn't pass (cutoff=0.0050): cat 1059, mi = 0.002, pvalue = 0.025, fibrillar center\n",
      "Didn't pass (cutoff=0.0050): cat 1431, mi = 0.002, pvalue = 0.042, Rho protein signal transduction\n",
      "Didn't pass (cutoff=0.0050): cat 841, mi = 0.002, pvalue = 0.032, thiol-dependent ubiquitin-specific protease activity\n",
      "Didn't pass (cutoff=0.0050): cat 2504, mi = 0.002, pvalue = 0.015, ATP metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 1115, mi = 0.002, pvalue = 0.041, inositol phosphate metabolic process\n",
      "Didn't pass (cutoff=0.0050): cat 3745, mi = 0.002, pvalue = 0.015, positive regulation of peptidyl-threonine phosphorylation\n",
      "Didn't pass (cutoff=0.0050): cat 8710, mi = 0.002, pvalue = 0.023, positive regulation of DNA repair\n",
      "Didn't pass (cutoff=0.0050): cat 803, mi = 0.002, pvalue = 0.033, chaperone binding\n",
      "Didn't pass (cutoff=0.0050): cat 2925, mi = 0.002, pvalue = 0.019, cardiovascular system development\n",
      "Didn't pass (cutoff=0.0050): cat 3024, mi = 0.002, pvalue = 0.022, telomeric D-loop disassembly\n",
      "Didn't pass (cutoff=0.0050): cat 408, mi = 0.002, pvalue = 0.031, protein glycosylation\n",
      "Didn't pass (cutoff=0.0050): cat 5706, mi = 0.002, pvalue = 0.041, peroxisome\n",
      "Didn't pass (cutoff=0.0050): cat 2783, mi = 0.002, pvalue = 0.036, ubiquitin ligase complex\n",
      "Total number of categories evaluated = 109.\n",
      "Number of categories that passed the tests = 12\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 7, 11, 0.120846810322703\n",
      "best merge is 1, 6, 0.246157297430771\n",
      "best merge is 3, 7, 0.501712496767229\n",
      "best merge is 5, 9, 0.516477106284329\n",
      "best merge is 4, 8, 0.685708777110108\n",
      "best merge is 1, 5, 0.706359330028422\n",
      "best merge is 0, 10, 0.709071872759605\n",
      "best merge is 1, 2, 0.950214524046987\n",
      "order classes\n",
      "2 - 2.12721561827881\n",
      "4 - 1.3455268964358\n",
      "3 - 0.999352714539698\n",
      "2 - 2.05712633133032\n",
      "3 7 11\n",
      "1 6 5 9\n",
      "2\n",
      "0 10\n",
      "4 8\n",
      "Done.Start drawing\n",
      "9.06\t-14.97\n",
      "filopodium, GO:0030175\n",
      "transforming growth factor beta receptor signaling pathway, GO:0007179\n",
      "histone acetyltransferase complex, GO:0000123\n",
      "single fertilization, GO:0007338\n",
      "RNA phosphodiester bond hydrolysis, exonucleolytic, GO:0090503\n",
      "catalytic step 2 spliceosome, GO:0071013\n",
      "inactivation of MAPK activity, GO:0000188\n",
      "meiotic mismatch repair, GO:0000710\n",
      "regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle, GO:0051439\n",
      "regulation of G2/M transition of mitotic cell cycle, GO:0010389\n",
      "ribonuclease III activity, GO:0004525\n",
      "interleukin-1 receptor binding, GO:0005149\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 7, 11, 0.120846810322703\n",
      "best merge is 1, 6, 0.246157297430771\n",
      "best merge is 3, 7, 0.501712496767229\n",
      "best merge is 5, 9, 0.516477106284329\n",
      "best merge is 4, 8, 0.685708777110108\n",
      "best merge is 1, 5, 0.706359330028422\n",
      "best merge is 0, 10, 0.709071872759605\n",
      "best merge is 1, 2, 0.950214524046987\n",
      "order classes\n",
      "2 - 2.12721561827881\n",
      "4 - 1.3455268964358\n",
      "3 - 0.999352714539698\n",
      "2 - 2.05712633133032\n",
      "3 7 11\n",
      "1 6 5 9\n",
      "2\n",
      "0 10\n",
      "4 8\n",
      "Done.-8.07116104868914\t42\t0\n",
      "4.40823970037453\t42\t23.8576779026217\n",
      "5.22097378277154\t42\t24.9588014981273\n",
      "5.65792759051186\t42\t25.4831460674157\n",
      "5.94631710362047\t42\t25.832709113608\n",
      "6.15605493133583\t42\t26.059925093633\n",
      "6.34831460674157\t42\t26.2521847690387\n",
      "6.54931335830212\t42\t26.4444444444444\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.14357053682896\t42\t26.916354556804\n",
      "14.685393258427\t42\t27.3707865168539\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 181\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8163 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8163\t181\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "742 hits go_terms_num = 181\n",
      "cat 13, mi = 0.00152439, BRF2_ChIP-seq_regulon\n",
      "cat 131, mi = 0.00148790, SMARCA4_ChIP-seq_regulon\n",
      "cat 115, mi = 0.00098122, PPARGC1A_ChIP-seq_regulon\n",
      "cat 9, mi = 0.00074801, BDP1_ChIP-seq_regulon\n",
      "cat 114, mi = 0.00064546, POU5F1_ChIP-seq_regulon\n",
      "cat 167, mi = 0.00060282, XRCC4_ChIP-seq_regulon\n",
      "cat 112, mi = 0.00047613, POLR3G_ChIP-seq_regulon\n",
      "cat 12, mi = 0.00039383, BRF1_ChIP-seq_regulon\n",
      "cat 0, mi = 0.00000000, ARID3A_ChIP-seq_regulon\n",
      "cat 1, mi = 0.00000000, ATF1_ChIP-seq_regulon\n",
      "cat 2, mi = 0.00000000, ATF2_ChIP-seq_regulon\n",
      "cat 3, mi = 0.00000000, ATF3_ChIP-seq_regulon\n",
      "cat 4, mi = 0.00000000, BACH1_ChIP-seq_regulon\n",
      "cat 5, mi = 0.00000000, BATF_ChIP-seq_regulon\n",
      "cat 6, mi = 0.00000000, BCL11A_ChIP-seq_regulon\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.002, pvalue = 0.051, BRF2_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 131, mi = 0.001, pvalue = 0.132, SMARCA4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.001, pvalue = 0.569, PPARGC1A_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.663, BDP1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 114, mi = 0.001, pvalue = 0.703, POU5F1_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 167, mi = 0.001, pvalue = 0.826, XRCC4_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.000, pvalue = 0.958, POLR3G_ChIP-seq_regulon\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.000, pvalue = 0.961, BRF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ARID3A_ChIP-seq_regulon\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, ATF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, ATF2_ChIP-seq_regulon\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, ATF3_ChIP-seq_regulon\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, BACH1_ChIP-seq_regulon\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, BATF_ChIP-seq_regulon\n",
      "Didn't pass: cat 6, mi = 0.000, pvalue = 1.000, BCL11A_ChIP-seq_regulon\n",
      "Didn't pass: cat 7, mi = 0.000, pvalue = 1.000, BCL3_ChIP-seq_regulon\n",
      "Didn't pass: cat 8, mi = 0.000, pvalue = 1.000, BCLAF1_ChIP-seq_regulon\n",
      "Didn't pass: cat 10, mi = 0.000, pvalue = 1.000, BHLHE40_ChIP-seq_regulon\n",
      "Didn't pass: cat 11, mi = 0.000, pvalue = 1.000, BRCA1_ChIP-seq_regulon\n",
      "Didn't pass: cat 14, mi = 0.000, pvalue = 1.000, CBX2_ChIP-seq_regulon\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.06\t-14.97\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_encode_tf/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_encode_tf/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_encode_tf.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 298\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8271 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8271\t298\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "752 hits go_terms_num = 298\n",
      "cat 77, mi = 0.00313190, chr1p22\n",
      "cat 52, mi = 0.00266521, chr15q15\n",
      "cat 229, mi = 0.00233614, chr4q31\n",
      "cat 153, mi = 0.00222546, chr6p24\n",
      "cat 134, mi = 0.00205590, chr2p16\n",
      "cat 53, mi = 0.00198826, chr4p15\n",
      "cat 16, mi = 0.00197861, chr12p11\n",
      "cat 164, mi = 0.00192661, chr2p23\n",
      "cat 195, mi = 0.00187819, chr13q34\n",
      "cat 217, mi = 0.00178365, chr10q24\n",
      "cat 192, mi = 0.00175566, chr7q21\n",
      "cat 296, mi = 0.00168310, chr16p11\n",
      "cat 204, mi = 0.00165513, chr4q12\n",
      "cat 210, mi = 0.00164894, chr18q12\n",
      "cat 239, mi = 0.00163418, chr6p12\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 77, mi = 0.003, pvalue = 0.000, chr1p22\n",
      "Passed: cat 52, mi = 0.003, pvalue = 0.001, chr15q15\n",
      "Didn't pass (cutoff=0.0050): cat 229, mi = 0.002, pvalue = 0.005, chr4q31\n",
      "Passed: cat 153, mi = 0.002, pvalue = 0.004, chr6p24\n",
      "Didn't pass (cutoff=0.0050): cat 134, mi = 0.002, pvalue = 0.019, chr2p16\n",
      "Didn't pass (cutoff=0.0050): cat 53, mi = 0.002, pvalue = 0.020, chr4p15\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.002, pvalue = 0.019, chr12p11\n",
      "Didn't pass (cutoff=0.0050): cat 164, mi = 0.002, pvalue = 0.025, chr2p23\n",
      "Didn't pass (cutoff=0.0050): cat 195, mi = 0.002, pvalue = 0.035, chr13q34\n",
      "Didn't pass (cutoff=0.0050): cat 217, mi = 0.002, pvalue = 0.033, chr10q24\n",
      "Didn't pass (cutoff=0.0050): cat 192, mi = 0.002, pvalue = 0.047, chr7q21\n",
      "Didn't pass (cutoff=0.0050): cat 296, mi = 0.002, pvalue = 0.048, chr16p11\n",
      "Didn't pass (cutoff=0.0050): cat 204, mi = 0.002, pvalue = 0.023, chr4q12\n",
      "Didn't pass (cutoff=0.0050): cat 210, mi = 0.002, pvalue = 0.062, chr18q12\n",
      "Didn't pass (cutoff=0.0050): cat 239, mi = 0.002, pvalue = 0.063, chr6p12\n",
      "Didn't pass (cutoff=0.0050): cat 112, mi = 0.002, pvalue = 0.067, chrXq13\n",
      "Didn't pass (cutoff=0.0050): cat 178, mi = 0.002, pvalue = 0.033, chr4q22\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.002, pvalue = 0.052, chr17p12\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.002, pvalue = 0.096, chr3q21\n",
      "Didn't pass (cutoff=0.0050): cat 115, mi = 0.002, pvalue = 0.056, chr7q32\n",
      "Didn't pass (cutoff=0.0050): cat 142, mi = 0.002, pvalue = 0.118, chr10p11\n",
      "Didn't pass (cutoff=0.0050): cat 107, mi = 0.002, pvalue = 0.064, chr4q24\n",
      "Didn't pass (cutoff=0.0050): cat 232, mi = 0.001, pvalue = 0.115, chr7p13\n",
      "Didn't pass (cutoff=0.0050): cat 132, mi = 0.001, pvalue = 0.128, chr12q21\n",
      "Total number of categories evaluated = 14.\n",
      "Number of categories that passed the tests = 3\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.804228581496284\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.Start drawing\n",
      "9.06\t-14.97\n",
      "chr1p22\n",
      "chr6p24\n",
      "chr15q15\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 2, 0.804228581496284\n",
      "order classes\n",
      "0\n",
      "1\n",
      "2\n",
      "Done.-8.00124843945069\t42\t0\n",
      "4.52184769038701\t42\t23.9975031210986\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.69288389513109\t42\t25.5355805243446\n",
      "5.96379525593009\t42\t25.8501872659176\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.55805243445693\t42\t26.4619225967541\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.13483146067416\t42\t26.916354556804\n",
      "14.6766541822722\t42\t27.3533083645443\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c1/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c1/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c1.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 5529\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8198 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8198\t5529\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "745 hits go_terms_num = 5529\n",
      "cat 1608, mi = 0.00300813, BYSTROEM_CORRELATED_WITH_IL5_DN\n",
      "cat 1821, mi = 0.00294094, GERHOLD_ADIPOGENESIS_DN\n",
      "cat 1734, mi = 0.00288718, TONKS_TARGETS_OF_RUNX1_RUNX1T1_FUSION_HSC_DN\n",
      "cat 4572, mi = 0.00274752, PID_PDGFRA_PATHWAY\n",
      "cat 278, mi = 0.00272557, BIOCARTA_GLEEVEC_PATHWAY\n",
      "cat 2524, mi = 0.00271120, STAMBOLSKY_TARGETS_OF_MUTATED_TP53_DN\n",
      "cat 4001, mi = 0.00267640, BIOCARTA_MET_PATHWAY\n",
      "cat 1893, mi = 0.00265004, TAKEDA_TARGETS_OF_NUP98_HOXA9_FUSION_8D_UP\n",
      "cat 3928, mi = 0.00259722, SCHLOSSER_SERUM_RESPONSE_AUGMENTED_BY_MYC\n",
      "cat 924, mi = 0.00259146, BIOCARTA_INTEGRIN_PATHWAY\n",
      "cat 2768, mi = 0.00251348, SCHLOSSER_MYC_TARGETS_AND_SERUM_RESPONSE_DN\n",
      "cat 58, mi = 0.00250780, SIG_BCR_SIGNALING_PATHWAY\n",
      "cat 2241, mi = 0.00249030, LIAO_HAVE_SOX4_BINDING_SITES\n",
      "cat 3983, mi = 0.00246240, DAZARD_RESPONSE_TO_UV_SCC_DN\n",
      "cat 178, mi = 0.00243260, KEGG_ACUTE_MYELOID_LEUKEMIA\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 1608, mi = 0.003, pvalue = 0.000, BYSTROEM_CORRELATED_WITH_IL5_DN\n",
      "Passed: cat 1821, mi = 0.003, pvalue = 0.000, GERHOLD_ADIPOGENESIS_DN\n",
      "Passed: cat 1734, mi = 0.003, pvalue = 0.001, TONKS_TARGETS_OF_RUNX1_RUNX1T1_FUSION_HSC_DN\n",
      "Passed: cat 2524, mi = 0.003, pvalue = 0.001, STAMBOLSKY_TARGETS_OF_MUTATED_TP53_DN\n",
      "Passed: cat 4001, mi = 0.003, pvalue = 0.001, BIOCARTA_MET_PATHWAY\n",
      "Passed: cat 2768, mi = 0.003, pvalue = 0.002, SCHLOSSER_MYC_TARGETS_AND_SERUM_RESPONSE_DN\n",
      "Passed: cat 2241, mi = 0.002, pvalue = 0.001, LIAO_HAVE_SOX4_BINDING_SITES\n",
      "Passed: cat 4792, mi = 0.002, pvalue = 0.004, KORKOLA_SEMINOMA_UP\n",
      "Passed: cat 5170, mi = 0.002, pvalue = 0.003, POMEROY_MEDULLOBLASTOMA_PROGNOSIS_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2782, mi = 0.002, pvalue = 0.006, REACTOME_AURKA_ACTIVATION_BY_TPX2\n",
      "Didn't pass (cutoff=0.0050): cat 5409, mi = 0.002, pvalue = 0.006, HEIDENBLAD_AMPLICON_8Q24_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4046, mi = 0.002, pvalue = 0.008, MARKEY_RB1_CHRONIC_LOF_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4010, mi = 0.002, pvalue = 0.007, REACTOME_SLC_TRANSPORTER_DISORDERS\n",
      "Didn't pass (cutoff=0.0050): cat 176, mi = 0.002, pvalue = 0.005, REACTOME_RECRUITMENT_OF_NUMA_TO_MITOTIC_CENTROSOMES\n",
      "Didn't pass (cutoff=0.0050): cat 2224, mi = 0.002, pvalue = 0.006, REACTOME_REGULATION_OF_PLK1_ACTIVITY_AT_G2_M_TRANSITION\n",
      "Didn't pass (cutoff=0.0050): cat 450, mi = 0.002, pvalue = 0.009, MAHAJAN_RESPONSE_TO_IL1A_DN\n",
      "Didn't pass (cutoff=0.0050): cat 294, mi = 0.002, pvalue = 0.005, CAFFAREL_RESPONSE_TO_THC_24HR_5_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2190, mi = 0.002, pvalue = 0.009, HUANG_DASATINIB_RESISTANCE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3783, mi = 0.002, pvalue = 0.009, IKEDA_MIR30_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4016, mi = 0.002, pvalue = 0.011, LIU_BREAST_CANCER\n",
      "Didn't pass (cutoff=0.0050): cat 3716, mi = 0.002, pvalue = 0.015, REACTOME_RUNX1_INTERACTS_WITH_CO_FACTORS_WHOSE_PRECISE_EFFECT_ON_RUNX1_TARGETS_IS_NOT_KNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 138, mi = 0.002, pvalue = 0.011, VANTVEER_BREAST_CANCER_BRCA1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4460, mi = 0.002, pvalue = 0.016, REACTOME_INOSITOL_PHOSPHATE_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 1961, mi = 0.002, pvalue = 0.016, CERVERA_SDHB_TARGETS_2\n",
      "Didn't pass (cutoff=0.0050): cat 2613, mi = 0.002, pvalue = 0.011, WENG_POR_TARGETS_LIVER_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4567, mi = 0.002, pvalue = 0.015, REACTOME_NOTCH2_INTRACELLULAR_DOMAIN_REGULATES_TRANSCRIPTION\n",
      "Didn't pass (cutoff=0.0050): cat 2008, mi = 0.002, pvalue = 0.014, YAMAZAKI_TCEB3_TARGETS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3951, mi = 0.002, pvalue = 0.014, BERTUCCI_MEDULLARY_VS_DUCTAL_BREAST_CANCER_DN\n",
      "Didn't pass (cutoff=0.0050): cat 986, mi = 0.002, pvalue = 0.021, REACTOME_SUMOYLATION_OF_RNA_BINDING_PROTEINS\n",
      "Total number of categories evaluated = 61.\n",
      "Number of categories that passed the tests = 9\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.142703392193928\n",
      "best merge is 6, 8, 0.212938661512565\n",
      "best merge is 2, 5, 0.394671240238136\n",
      "best merge is 0, 1, 0.474625060278271\n",
      "best merge is 0, 7, 0.761206549749715\n",
      "order classes\n",
      "3 - 0.727969008500314\n",
      "2 - 0.888010290535806\n",
      "2 - 0.479111988403272\n",
      "6 8\n",
      "2 5\n",
      "3\n",
      "0 4 1\n",
      "7\n",
      "Done.Start drawing\n",
      "9.06\t-14.97\n",
      "BIOCARTA_MET_PATHWAY\n",
      "GERHOLD_ADIPOGENESIS_DN\n",
      "TONKS_TARGETS_OF_RUNX1_RUNX1T1_FUSION_HSC_DN\n",
      "SCHLOSSER_MYC_TARGETS_AND_SERUM_RESPONSE_DN\n",
      "BYSTROEM_CORRELATED_WITH_IL5_DN\n",
      "LIAO_HAVE_SOX4_BINDING_SITES\n",
      "POMEROY_MEDULLOBLASTOMA_PROGNOSIS_DN\n",
      "KORKOLA_SEMINOMA_UP\n",
      "STAMBOLSKY_TARGETS_OF_MUTATED_TP53_DN\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 0, 4, 0.142703392193928\n",
      "best merge is 6, 8, 0.212938661512565\n",
      "best merge is 2, 5, 0.394671240238136\n",
      "best merge is 0, 1, 0.474625060278271\n",
      "best merge is 0, 7, 0.761206549749715\n",
      "order classes\n",
      "3 - 0.727969008500314\n",
      "2 - 0.888010290535806\n",
      "2 - 0.479111988403272\n",
      "6 8\n",
      "2 5\n",
      "3\n",
      "0 4 1\n",
      "7\n",
      "Done.-8.00124843945069\t42\t0\n",
      "4.52184769038701\t42\t23.9975031210986\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.70162297128589\t42\t25.5355805243446\n",
      "5.97253433208489\t42\t25.8676654182272\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.55805243445693\t42\t26.4619225967541\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.13483146067416\t42\t26.916354556804\n",
      "14.6766541822722\t42\t27.3533083645443\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c2/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c2/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c2.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 3733\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8258 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8258\t3733\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "751 hits go_terms_num = 3733\n",
      "cat 206, mi = 0.00276434, MIR6895_3P\n",
      "cat 1569, mi = 0.00271830, MIR1270\n",
      "cat 640, mi = 0.00260111, MIR4741\n",
      "cat 1459, mi = 0.00253436, ZBTB1_TARGET_GENES\n",
      "cat 1513, mi = 0.00252065, MIR625_3P\n",
      "cat 1200, mi = 0.00250871, MIR620\n",
      "cat 2994, mi = 0.00248059, MIR3165\n",
      "cat 0, mi = 0.00247575, MIR34C_3P\n",
      "cat 1924, mi = 0.00246808, MIR4675\n",
      "cat 728, mi = 0.00244898, MIR4288\n",
      "cat 1507, mi = 0.00237947, MIR1224_3P\n",
      "cat 2106, mi = 0.00236877, MIR4742_5P\n",
      "cat 1947, mi = 0.00231868, MIR8081\n",
      "cat 1360, mi = 0.00229411, ATF6_01\n",
      "cat 2603, mi = 0.00226503, MIR27B_5P\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 206, mi = 0.003, pvalue = 0.001, MIR6895_3P\n",
      "Passed: cat 1569, mi = 0.003, pvalue = 0.001, MIR1270\n",
      "Passed: cat 640, mi = 0.003, pvalue = 0.002, MIR4741\n",
      "Passed: cat 1459, mi = 0.003, pvalue = 0.002, ZBTB1_TARGET_GENES\n",
      "Passed: cat 1513, mi = 0.003, pvalue = 0.002, MIR625_3P\n",
      "Passed: cat 2994, mi = 0.002, pvalue = 0.003, MIR3165\n",
      "Passed: cat 0, mi = 0.002, pvalue = 0.003, MIR34C_3P\n",
      "Passed: cat 2106, mi = 0.002, pvalue = 0.003, MIR4742_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1947, mi = 0.002, pvalue = 0.006, MIR8081\n",
      "Didn't pass (cutoff=0.0050): cat 2603, mi = 0.002, pvalue = 0.008, MIR27B_5P\n",
      "Didn't pass (cutoff=0.0050): cat 3000, mi = 0.002, pvalue = 0.009, MIR219B_3P\n",
      "Didn't pass (cutoff=0.0050): cat 3346, mi = 0.002, pvalue = 0.009, MIR12114\n",
      "Didn't pass (cutoff=0.0050): cat 1437, mi = 0.002, pvalue = 0.008, MIR1251_3P\n",
      "Didn't pass (cutoff=0.0050): cat 3334, mi = 0.002, pvalue = 0.011, MIR4438\n",
      "Didn't pass (cutoff=0.0050): cat 741, mi = 0.002, pvalue = 0.011, MIR5196_3P\n",
      "Didn't pass (cutoff=0.0050): cat 1720, mi = 0.002, pvalue = 0.013, MIR3161\n",
      "Didn't pass (cutoff=0.0050): cat 1161, mi = 0.002, pvalue = 0.013, MIR10A_3P\n",
      "Didn't pass (cutoff=0.0050): cat 926, mi = 0.002, pvalue = 0.014, MIR3935\n",
      "Didn't pass (cutoff=0.0050): cat 2513, mi = 0.002, pvalue = 0.014, MIR182_3P\n",
      "Didn't pass (cutoff=0.0050): cat 1272, mi = 0.002, pvalue = 0.018, MIR1272\n",
      "Didn't pass (cutoff=0.0050): cat 3704, mi = 0.002, pvalue = 0.019, GCTGAGT_MIR5125P\n",
      "Didn't pass (cutoff=0.0050): cat 2333, mi = 0.002, pvalue = 0.011, MIR4694_5P\n",
      "Didn't pass (cutoff=0.0050): cat 3645, mi = 0.002, pvalue = 0.020, TCTAGAG_MIR517\n",
      "Didn't pass (cutoff=0.0050): cat 2533, mi = 0.002, pvalue = 0.017, KCCGNSWTTT_UNKNOWN\n",
      "Didn't pass (cutoff=0.0050): cat 3139, mi = 0.002, pvalue = 0.017, MIR4696\n",
      "Didn't pass (cutoff=0.0050): cat 2977, mi = 0.002, pvalue = 0.019, MIR6778_5P\n",
      "Didn't pass (cutoff=0.0050): cat 1331, mi = 0.002, pvalue = 0.025, ZNF785_TARGET_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 316, mi = 0.002, pvalue = 0.031, SKP2_TARGET_GENES\n",
      "Total number of categories evaluated = 51.\n",
      "Number of categories that passed the tests = 8\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 6, 7, 0.421715609067771\n",
      "best merge is 3, 6, 0.668510797526769\n",
      "best merge is 4, 5, 0.788834488403681\n",
      "best merge is 0, 3, 0.827267993916787\n",
      "order classes\n",
      "3 - 1.04221463947929\n",
      "2 - 1.57766897680736\n",
      "3 6 7\n",
      "1\n",
      "2\n",
      "0\n",
      "4 5\n",
      "Done.Start drawing\n",
      "9.06\t-14.97\n",
      "MIR4741\n",
      "ZBTB1_TARGET_GENES\n",
      "MIR34C_3P\n",
      "MIR4742_5P\n",
      "MIR625_3P\n",
      "MIR3165\n",
      "MIR1270\n",
      "MIR6895_3P\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 6, 7, 0.421715609067771\n",
      "best merge is 3, 6, 0.668510797526769\n",
      "best merge is 4, 5, 0.788834488403681\n",
      "best merge is 0, 3, 0.827267993916787\n",
      "order classes\n",
      "3 - 1.04221463947929\n",
      "2 - 1.57766897680736\n",
      "3 6 7\n",
      "1\n",
      "2\n",
      "0\n",
      "4 5\n",
      "Done.-8.00124843945069\t42\t0\n",
      "4.52184769038701\t42\t23.9975031210986\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.69288389513109\t42\t25.5355805243446\n",
      "5.96379525593009\t42\t25.8501872659176\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.55805243445693\t42\t26.4619225967541\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.13483146067416\t42\t26.916354556804\n",
      "14.6766541822722\t42\t27.3533083645443\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c3/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c3/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c3.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 858\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4744 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4744\t858\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "431 hits go_terms_num = 858\n",
      "cat 355, mi = 0.00494177, GNF2_DAP3\n",
      "cat 381, mi = 0.00469044, MORF_PHB\n",
      "cat 403, mi = 0.00431181, GNF2_RPA1\n",
      "cat 189, mi = 0.00410320, MODULE_436\n",
      "cat 5, mi = 0.00377876, MORF_RAB1A\n",
      "cat 12, mi = 0.00375730, GNF2_KPNB1\n",
      "cat 448, mi = 0.00369668, MORF_CDK2\n",
      "cat 169, mi = 0.00364541, GNF2_HDAC1\n",
      "cat 340, mi = 0.00361681, MODULE_177\n",
      "cat 669, mi = 0.00351961, MORF_CUL1\n",
      "cat 407, mi = 0.00349131, GNF2_MKI67\n",
      "cat 839, mi = 0.00345150, MORF_CSNK1D\n",
      "cat 668, mi = 0.00344537, MODULE_533\n",
      "cat 523, mi = 0.00344324, MORF_XPC\n",
      "cat 767, mi = 0.00340542, GNF2_CBFB\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 355, mi = 0.005, pvalue = 0.000, GNF2_DAP3\n",
      "Passed: cat 381, mi = 0.005, pvalue = 0.002, MORF_PHB\n",
      "Didn't pass (cutoff=0.0050): cat 189, mi = 0.004, pvalue = 0.005, MODULE_436\n",
      "Didn't pass (cutoff=0.0050): cat 340, mi = 0.004, pvalue = 0.015, MODULE_177\n",
      "Didn't pass (cutoff=0.0050): cat 407, mi = 0.003, pvalue = 0.023, GNF2_MKI67\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.003, pvalue = 0.020, MODULE_533\n",
      "Didn't pass (cutoff=0.0050): cat 523, mi = 0.003, pvalue = 0.025, MORF_XPC\n",
      "Didn't pass (cutoff=0.0050): cat 565, mi = 0.003, pvalue = 0.025, GCM_BAG5\n",
      "Didn't pass (cutoff=0.0050): cat 607, mi = 0.003, pvalue = 0.024, MODULE_171\n",
      "Didn't pass (cutoff=0.0050): cat 519, mi = 0.003, pvalue = 0.034, MODULE_128\n",
      "Didn't pass (cutoff=0.0050): cat 838, mi = 0.003, pvalue = 0.032, MODULE_79\n",
      "Didn't pass (cutoff=0.0050): cat 362, mi = 0.003, pvalue = 0.036, MODULE_146\n",
      "Didn't pass (cutoff=0.0050): cat 533, mi = 0.003, pvalue = 0.035, MORF_CDH4\n",
      "Didn't pass (cutoff=0.0050): cat 160, mi = 0.003, pvalue = 0.036, MODULE_438\n",
      "Didn't pass (cutoff=0.0050): cat 665, mi = 0.003, pvalue = 0.037, MODULE_222\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.003, pvalue = 0.038, MORF_PRKAR1A\n",
      "Didn't pass (cutoff=0.0050): cat 220, mi = 0.003, pvalue = 0.044, GCM_CSNK1A1\n",
      "Didn't pass (cutoff=0.0050): cat 105, mi = 0.003, pvalue = 0.028, MORF_SART1\n",
      "Didn't pass (cutoff=0.0050): cat 467, mi = 0.003, pvalue = 0.012, GNF2_ZAP70\n",
      "Didn't pass (cutoff=0.0050): cat 630, mi = 0.003, pvalue = 0.041, GNF2_PTPN4\n",
      "Didn't pass (cutoff=0.0050): cat 465, mi = 0.003, pvalue = 0.046, MORF_RRM1\n",
      "Didn't pass (cutoff=0.0050): cat 803, mi = 0.003, pvalue = 0.049, MORF_PSMC1\n",
      "Total number of categories evaluated = 27.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.04\t-14.97\n",
      "GNF2_DAP3\n",
      "MORF_PHB\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.40547588005215\t42\t0\n",
      "5.70925684485007\t42\t25.1890482398957\n",
      "6.47588005215124\t42\t26.2294654498044\n",
      "6.88657105606259\t42\t26.722294654498\n",
      "7.16036505867015\t42\t27.0508474576271\n",
      "7.36114732724903\t42\t27.2698826597132\n",
      "7.54367666232073\t42\t27.4524119947849\n",
      "7.72620599739244\t42\t27.6349413298566\n",
      "7.93611473272491\t42\t27.8174706649283\n",
      "8.2555410691004\t42\t28.0547588005215\n",
      "15.2281616688396\t42\t28.4563233376793\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c4/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c4/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c4.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 10191\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 7375 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 7375\t10191\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "670 hits go_terms_num = 10191\n",
      "cat 7689, mi = 0.00337565, GO_NEGATIVE_REGULATION_OF_CHROMOSOME_SEGREGATION\n",
      "cat 6424, mi = 0.00309632, GO_FORMATION_OF_PRIMARY_GERM_LAYER\n",
      "cat 194, mi = 0.00307845, GO_NEGATIVE_REGULATION_OF_SUPRAMOLECULAR_FIBER_ORGANIZATION\n",
      "cat 6466, mi = 0.00304090, GO_MESODERM_MORPHOGENESIS\n",
      "cat 2622, mi = 0.00303396, GO_REGULATION_OF_DNA_TEMPLATED_TRANSCRIPTION_IN_RESPONSE_TO_STRESS\n",
      "cat 6628, mi = 0.00287966, GO_SULFOTRANSFERASE_ACTIVITY\n",
      "cat 8408, mi = 0.00283551, GO_NEGATIVE_REGULATION_OF_NUCLEAR_DIVISION\n",
      "cat 9731, mi = 0.00282723, GO_POSITIVE_REGULATION_OF_INTERFERON_ALPHA_PRODUCTION\n",
      "cat 5447, mi = 0.00279763, GO_NEGATIVE_REGULATION_OF_CYTOSKELETON_ORGANIZATION\n",
      "cat 994, mi = 0.00273581, GO_TRANSFERASE_ACTIVITY_TRANSFERRING_SULFUR_CONTAINING_GROUPS\n",
      "cat 3587, mi = 0.00272119, GO_POSITIVE_REGULATION_OF_HEMOPOIESIS\n",
      "cat 6019, mi = 0.00266105, GO_CONDENSED_CHROMOSOME_OUTER_KINETOCHORE\n",
      "cat 3570, mi = 0.00264447, GO_ENDONUCLEASE_COMPLEX\n",
      "cat 7417, mi = 0.00262641, GO_FILOPODIUM\n",
      "cat 2827, mi = 0.00262204, GO_DYNEIN_COMPLEX_BINDING\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 7689, mi = 0.003, pvalue = 0.000, GO_NEGATIVE_REGULATION_OF_CHROMOSOME_SEGREGATION\n",
      "Passed: cat 6424, mi = 0.003, pvalue = 0.001, GO_FORMATION_OF_PRIMARY_GERM_LAYER\n",
      "Passed: cat 2622, mi = 0.003, pvalue = 0.001, GO_REGULATION_OF_DNA_TEMPLATED_TRANSCRIPTION_IN_RESPONSE_TO_STRESS\n",
      "Passed: cat 6628, mi = 0.003, pvalue = 0.001, GO_SULFOTRANSFERASE_ACTIVITY\n",
      "Passed: cat 3587, mi = 0.003, pvalue = 0.004, GO_POSITIVE_REGULATION_OF_HEMOPOIESIS\n",
      "Passed: cat 3570, mi = 0.003, pvalue = 0.005, GO_ENDONUCLEASE_COMPLEX\n",
      "Passed: cat 7417, mi = 0.003, pvalue = 0.005, GO_FILOPODIUM\n",
      "Passed: cat 2110, mi = 0.003, pvalue = 0.000, GO_HYALURONAN_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5807, mi = 0.003, pvalue = 0.005, GO_PYRUVATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 7774, mi = 0.003, pvalue = 0.008, GO_CARDIAC_MUSCLE_CONTRACTION\n",
      "Passed: cat 8451, mi = 0.002, pvalue = 0.004, GO_METHYLTRANSFERASE_COMPLEX\n",
      "Didn't pass (cutoff=0.0050): cat 7092, mi = 0.002, pvalue = 0.009, GO_INORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 1267, mi = 0.002, pvalue = 0.005, GO_TYROSINE_PHOSPHORYLATION_OF_STAT_PROTEIN\n",
      "Didn't pass (cutoff=0.0050): cat 4059, mi = 0.002, pvalue = 0.007, GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_THE_CH_CH_GROUP_OF_DONORS\n",
      "Didn't pass (cutoff=0.0050): cat 2892, mi = 0.002, pvalue = 0.006, GO_ACTIVATION_OF_ADENYLATE_CYCLASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 9117, mi = 0.002, pvalue = 0.010, GO_NEGATIVE_REGULATION_OF_PROTEIN_POLYMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 9548, mi = 0.002, pvalue = 0.012, GO_MITOTIC_G2_M_TRANSITION_CHECKPOINT\n",
      "Didn't pass (cutoff=0.0050): cat 8825, mi = 0.002, pvalue = 0.010, GO_INTRACELLULAR_LIPID_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 1860, mi = 0.002, pvalue = 0.014, GO_STRIATED_MUSCLE_CONTRACTION\n",
      "Didn't pass (cutoff=0.0050): cat 8155, mi = 0.002, pvalue = 0.010, GO_ADP_METABOLIC_PROCESS\n",
      "Passed: cat 7396, mi = 0.002, pvalue = 0.001, GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_THE_CH_CH_GROUP_OF_DONORS_NAD_OR_NADP_AS_ACCEPTOR\n",
      "Didn't pass (cutoff=0.0050): cat 7348, mi = 0.002, pvalue = 0.014, GO_REGULATION_OF_NUCLEOTIDE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 4321, mi = 0.002, pvalue = 0.012, GO_NEGATIVE_REGULATION_OF_ACTIN_FILAMENT_DEPOLYMERIZATION\n",
      "Didn't pass (cutoff=0.0050): cat 2184, mi = 0.002, pvalue = 0.019, GO_CHLORIDE_TRANSPORT\n",
      "Passed: cat 2723, mi = 0.002, pvalue = 0.000, GO_ACTIVATED_T_CELL_PROLIFERATION\n",
      "Didn't pass (cutoff=0.0050): cat 4896, mi = 0.002, pvalue = 0.007, GO_TOLERANCE_INDUCTION\n",
      "Didn't pass (cutoff=0.0050): cat 2591, mi = 0.002, pvalue = 0.011, GO_NEUTRAL_AMINO_ACID_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 5479, mi = 0.002, pvalue = 0.017, GO_THIOESTER_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 6529, mi = 0.002, pvalue = 0.018, GO_VESICLE_COAT\n",
      "Didn't pass (cutoff=0.0050): cat 9922, mi = 0.002, pvalue = 0.015, GO_MEMBRANE_COAT\n",
      "Didn't pass (cutoff=0.0050): cat 3894, mi = 0.002, pvalue = 0.016, GO_MULTIVESICULAR_BODY_SORTING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 2335, mi = 0.002, pvalue = 0.011, GO_POTASSIUM_ION_IMPORT_ACROSS_PLASMA_MEMBRANE\n",
      "Didn't pass (cutoff=0.0050): cat 1416, mi = 0.002, pvalue = 0.017, GO_NUCLEOSIDE_DIPHOSPHATE_METABOLIC_PROCESS\n",
      "Didn't pass (cutoff=0.0050): cat 5110, mi = 0.002, pvalue = 0.018, GO_FUCOSYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 343, mi = 0.002, pvalue = 0.011, GO_POSITIVE_REGULATION_OF_TYROSINE_PHOSPHORYLATION_OF_STAT_PROTEIN\n",
      "Didn't pass (cutoff=0.0050): cat 9630, mi = 0.002, pvalue = 0.022, GO_REGULATION_OF_SYNAPTIC_VESICLE_CYCLE\n",
      "Didn't pass (cutoff=0.0050): cat 4592, mi = 0.002, pvalue = 0.021, GO_G2_DNA_DAMAGE_CHECKPOINT\n",
      "Didn't pass (cutoff=0.0050): cat 1023, mi = 0.002, pvalue = 0.023, GO_CATALYTIC_STEP_2_SPLICEOSOME\n",
      "Didn't pass (cutoff=0.0050): cat 5412, mi = 0.002, pvalue = 0.022, GO_NEGATIVE_REGULATION_OF_NOTCH_SIGNALING_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.002, pvalue = 0.019, GO_PROTEIN_PHOSPHATASE_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 6504, mi = 0.002, pvalue = 0.028, GO_STRUCTURAL_CONSTITUENT_OF_CYTOSKELETON\n",
      "Didn't pass (cutoff=0.0050): cat 9320, mi = 0.002, pvalue = 0.027, GO_PIGMENT_GRANULE\n",
      "Didn't pass (cutoff=0.0050): cat 3963, mi = 0.002, pvalue = 0.017, GO_REGULATION_OF_SYNAPTIC_VESICLE_EXOCYTOSIS\n",
      "Didn't pass (cutoff=0.0050): cat 1383, mi = 0.002, pvalue = 0.024, GO_SUBSTANTIA_NIGRA_DEVELOPMENT\n",
      "Didn't pass (cutoff=0.0050): cat 4654, mi = 0.002, pvalue = 0.017, GO_POSITIVE_REGULATION_OF_RELEASE_OF_CYTOCHROME_C_FROM_MITOCHONDRIA\n",
      "Total number of categories evaluated = 76.\n",
      "Number of categories that passed the tests = 11\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 9, 10, 0.0632949215960714\n",
      "best merge is 3, 7, 0.204428036845605\n",
      "best merge is 2, 4, 0.220273063475967\n",
      "best merge is 1, 3, 0.450153794754193\n",
      "best merge is 2, 8, 0.500471724464947\n",
      "best merge is 0, 1, 0.590210814392345\n",
      "best merge is 2, 5, 0.659767180277925\n",
      "order classes\n",
      "4 - 0.98840777390129\n",
      "3 - 0.995065306404776\n",
      "2 - 0.174061034389196\n",
      "9 10\n",
      "2 4 8\n",
      "5\n",
      "6\n",
      "0 1 3 7\n",
      "Done.Start drawing\n",
      "8.52\t-14.97\n",
      "GO_FILOPODIUM\n",
      "GO_POSITIVE_REGULATION_OF_HEMOPOIESIS\n",
      "GO_ENDONUCLEASE_COMPLEX\n",
      "GO_OXIDOREDUCTASE_ACTIVITY_ACTING_ON_THE_CH_CH_GROUP_OF_DONORS_NAD_OR_NADP_AS_ACCEPTOR\n",
      "GO_ACTIVATED_T_CELL_PROLIFERATION\n",
      "GO_REGULATION_OF_DNA_TEMPLATED_TRANSCRIPTION_IN_RESPONSE_TO_STRESS\n",
      "GO_METHYLTRANSFERASE_COMPLEX\n",
      "GO_NEGATIVE_REGULATION_OF_CHROMOSOME_SEGREGATION\n",
      "GO_FORMATION_OF_PRIMARY_GERM_LAYER\n",
      "GO_SULFOTRANSFERASE_ACTIVITY\n",
      "GO_HYALURONAN_METABOLIC_PROCESS\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 9, 10, 0.0632949215960714\n",
      "best merge is 3, 7, 0.204428036845605\n",
      "best merge is 2, 4, 0.220273063475967\n",
      "best merge is 1, 3, 0.450153794754193\n",
      "best merge is 2, 8, 0.500471724464947\n",
      "best merge is 0, 1, 0.590210814392345\n",
      "best merge is 2, 5, 0.659767180277925\n",
      "order classes\n",
      "4 - 0.98840777390129\n",
      "3 - 0.995065306404776\n",
      "2 - 0.174061034389196\n",
      "9 10\n",
      "2 4 8\n",
      "5\n",
      "6\n",
      "0 1 3 7\n",
      "Done.-7.70753512132823\t42\t0\n",
      "5.1213282247765\t42\t24.5849297573435\n",
      "5.89910600255428\t42\t25.6577266922095\n",
      "6.30140485312899\t42\t26.1404853128991\n",
      "6.56960408684547\t42\t26.4623243933589\n",
      "6.77522349936143\t42\t26.6768837803321\n",
      "6.96296296296296\t42\t26.8735632183908\n",
      "7.15070242656449\t42\t27.0523627075351\n",
      "7.38314176245211\t42\t27.2490421455939\n",
      "7.73180076628352\t42\t27.5172413793103\n",
      "14.9731800766284\t42\t27.9463601532567\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c5/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c5/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c5.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 189\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 4679 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 4679\t189\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "425 hits go_terms_num = 189\n",
      "cat 180, mi = 0.00446842, NOTCH_DN.V1_UP\n",
      "cat 66, mi = 0.00386301, WNT_UP.V1_DN\n",
      "cat 6, mi = 0.00360188, RB_DN.V1_DN\n",
      "cat 24, mi = 0.00345911, AKT_UP_MTOR_DN.V1_DN\n",
      "cat 148, mi = 0.00338759, LEF1_UP.V1_DN\n",
      "cat 154, mi = 0.00333256, KRAS.DF.V1_DN\n",
      "cat 41, mi = 0.00332044, PDGF_UP.V1_UP\n",
      "cat 64, mi = 0.00331375, CAHOY_ASTROCYTIC\n",
      "cat 156, mi = 0.00325546, KRAS.BREAST_UP.V1_DN\n",
      "cat 9, mi = 0.00314097, PRC2_EED_UP.V1_DN\n",
      "cat 81, mi = 0.00309214, BRCA1_DN.V1_DN\n",
      "cat 38, mi = 0.00297159, KRAS.LUNG.BREAST_UP.V1_DN\n",
      "cat 3, mi = 0.00296868, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "cat 91, mi = 0.00293040, BMI1_DN_MEL18_DN.V1_UP\n",
      "cat 53, mi = 0.00290581, PRC2_SUZ12_UP.V1_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 180, mi = 0.004, pvalue = 0.002, NOTCH_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.004, pvalue = 0.010, WNT_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.004, pvalue = 0.015, RB_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.003, pvalue = 0.016, AKT_UP_MTOR_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 148, mi = 0.003, pvalue = 0.020, LEF1_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 154, mi = 0.003, pvalue = 0.022, KRAS.DF.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 41, mi = 0.003, pvalue = 0.020, PDGF_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 64, mi = 0.003, pvalue = 0.034, CAHOY_ASTROCYTIC\n",
      "Didn't pass (cutoff=0.0050): cat 156, mi = 0.003, pvalue = 0.034, KRAS.BREAST_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.003, pvalue = 0.029, PRC2_EED_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 81, mi = 0.003, pvalue = 0.040, BRCA1_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 38, mi = 0.003, pvalue = 0.062, KRAS.LUNG.BREAST_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3, mi = 0.003, pvalue = 0.068, KRAS.LUNG.BREAST_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 91, mi = 0.003, pvalue = 0.056, BMI1_DN_MEL18_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 53, mi = 0.003, pvalue = 0.061, PRC2_SUZ12_UP.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 134, mi = 0.003, pvalue = 0.051, PRC2_EZH2_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 101, mi = 0.003, pvalue = 0.050, TGFB_UP.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 59, mi = 0.003, pvalue = 0.086, ATM_DN.V1_DN\n",
      "Didn't pass (cutoff=0.0050): cat 40, mi = 0.003, pvalue = 0.088, ATM_DN.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 63, mi = 0.003, pvalue = 0.073, GCNP_SHH_UP_EARLY.V1_UP\n",
      "Didn't pass (cutoff=0.0050): cat 155, mi = 0.003, pvalue = 0.106, PDGF_UP.V1_DN\n",
      "Total number of categories evaluated = 11.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "8.04\t-14.97\n",
      "NOTCH_DN.V1_UP\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.46936114732725\t42\t0\n",
      "5.61799217731421\t42\t25.0612777053455\n",
      "6.43024771838331\t42\t26.1747066492829\n",
      "6.85006518904824\t42\t26.6857887874837\n",
      "7.1238591916558\t42\t27.0143415906128\n",
      "7.33376792698827\t42\t27.2333767926988\n",
      "7.52542372881356\t42\t27.4341590612777\n",
      "7.71707953063886\t42\t27.6166883963494\n",
      "7.96349413298567\t42\t27.8174706649283\n",
      "8.32855280312908\t42\t28.109517601043\n",
      "15.2737940026076\t42\t28.5475880052151\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c6/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c6/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c6.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 4872\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8185 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8185\t4872\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "744 hits go_terms_num = 4872\n",
      "cat 2646, mi = 0.00323489, GSE2706_R848_VS_LPS_2H_STIM_DC_DN\n",
      "cat 1889, mi = 0.00308297, GSE11961_FOLLICULAR_BCELL_VS_MEMORY_BCELL_DAY40_DN\n",
      "cat 204, mi = 0.00300061, GSE24972_WT_VS_IRF8_KO_SPLEEN_FOLLICULAR_BCELL_UP\n",
      "cat 4696, mi = 0.00291079, GSE23321_CD8_STEM_CELL_MEMORY_VS_NAIVE_CD8_TCELL_DN\n",
      "cat 3383, mi = 0.00290567, GSE37301_MULTIPOTENT_PROGENITOR_VS_LYMPHOID_PRIMED_MPP_UP\n",
      "cat 1113, mi = 0.00282204, GSE22045_TREG_VS_TCONV_DN\n",
      "cat 3253, mi = 0.00279177, GSE21063_3H_VS_16H_ANTI_IGM_STIM_NFATC1_KOBCELL_UP\n",
      "cat 1494, mi = 0.00274604, GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_MACROPHAGE_DN\n",
      "cat 1188, mi = 0.00274315, GSE15659_NAIVE_CD4_TCELL_VS_RESTING_TREG_UP\n",
      "cat 808, mi = 0.00268868, GSE15330_HSC_VS_GRANULOCYTE_MONOCYTE_PROGENITOR_UP\n",
      "cat 521, mi = 0.00265016, GSE2706_UNSTIM_VS_2H_R848_DC_DN\n",
      "cat 3628, mi = 0.00262225, GSE43863_DAY6_EFF_VS_DAY150_MEM_LY6C_INT_CXCR5POS_CD4_TCELL_UP\n",
      "cat 3623, mi = 0.00261412, GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_DN\n",
      "cat 280, mi = 0.00260953, GSE9960_GRAM_NEG_VS_GRAM_NEG_AND_POS_SEPSIS_PBMC_DN\n",
      "cat 4761, mi = 0.00260039, GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 2646, mi = 0.003, pvalue = 0.000, GSE2706_R848_VS_LPS_2H_STIM_DC_DN\n",
      "Passed: cat 1889, mi = 0.003, pvalue = 0.000, GSE11961_FOLLICULAR_BCELL_VS_MEMORY_BCELL_DAY40_DN\n",
      "Passed: cat 204, mi = 0.003, pvalue = 0.000, GSE24972_WT_VS_IRF8_KO_SPLEEN_FOLLICULAR_BCELL_UP\n",
      "Passed: cat 4696, mi = 0.003, pvalue = 0.001, GSE23321_CD8_STEM_CELL_MEMORY_VS_NAIVE_CD8_TCELL_DN\n",
      "Passed: cat 3383, mi = 0.003, pvalue = 0.001, GSE37301_MULTIPOTENT_PROGENITOR_VS_LYMPHOID_PRIMED_MPP_UP\n",
      "Passed: cat 1113, mi = 0.003, pvalue = 0.000, GSE22045_TREG_VS_TCONV_DN\n",
      "Passed: cat 3253, mi = 0.003, pvalue = 0.001, GSE21063_3H_VS_16H_ANTI_IGM_STIM_NFATC1_KOBCELL_UP\n",
      "Passed: cat 1494, mi = 0.003, pvalue = 0.001, GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_MACROPHAGE_DN\n",
      "Passed: cat 1188, mi = 0.003, pvalue = 0.001, GSE15659_NAIVE_CD4_TCELL_VS_RESTING_TREG_UP\n",
      "Passed: cat 3623, mi = 0.003, pvalue = 0.002, GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_DN\n",
      "Passed: cat 280, mi = 0.003, pvalue = 0.001, GSE9960_GRAM_NEG_VS_GRAM_NEG_AND_POS_SEPSIS_PBMC_DN\n",
      "Passed: cat 445, mi = 0.003, pvalue = 0.001, GSE21546_ELK1_KO_VS_SAP1A_KO_AND_ELK1_KO_DP_THYMOCYTES_DN\n",
      "Passed: cat 3524, mi = 0.003, pvalue = 0.002, GSE40277_GATA1_AND_SATB1_TRANSDUCED_VS_CTRL_CD4_TCELL_UP\n",
      "Passed: cat 548, mi = 0.003, pvalue = 0.002, GSE4142_GC_BCELL_VS_MEMORY_BCELL_DN\n",
      "Passed: cat 1014, mi = 0.003, pvalue = 0.002, GSE18791_CTRL_VS_NEWCASTLE_VIRUS_DC_1H_UP\n",
      "Passed: cat 3444, mi = 0.003, pvalue = 0.003, GSE14769_UNSTIM_VS_20MIN_LPS_BMDM_UP\n",
      "Passed: cat 4825, mi = 0.002, pvalue = 0.002, GSE2770_IL4_ACT_VS_ACT_CD4_TCELL_48H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 207, mi = 0.002, pvalue = 0.006, GSE26495_NAIVE_VS_PD1HIGH_CD8_TCELL_UP\n",
      "Passed: cat 858, mi = 0.002, pvalue = 0.004, GSE10325_CD4_TCELL_VS_LUPUS_CD4_TCELL_DN\n",
      "Passed: cat 4277, mi = 0.002, pvalue = 0.004, GSE17721_POLYIC_VS_CPG_12H_BMDC_UP\n",
      "Passed: cat 837, mi = 0.002, pvalue = 0.004, GSE5542_IFNG_VS_IFNA_AND_IFNG_TREATED_EPITHELIAL_CELLS_24H_UP\n",
      "Passed: cat 2593, mi = 0.002, pvalue = 0.004, GSE10240_IL22_VS_IL17_STIM_PRIMARY_BRONCHIAL_EPITHELIAL_CELLS_UP\n",
      "Passed: cat 4162, mi = 0.002, pvalue = 0.005, GSE25085_FETAL_LIVER_VS_ADULT_BM_SP4_THYMIC_IMPLANT_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2456, mi = 0.002, pvalue = 0.005, GSE27786_CD8_TCELL_VS_ERYTHROBLAST_UP\n",
      "Passed: cat 397, mi = 0.002, pvalue = 0.005, GSE17721_POLYIC_VS_GARDIQUIMOD_24H_BMDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2859, mi = 0.002, pvalue = 0.007, GSE24210_TCONV_VS_TREG_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2558, mi = 0.002, pvalue = 0.006, GSE22611_NOD2_TRANSDUCED_VS_CTRL_HEK293T_STIMULATED_WITH_MDP_2H_DN\n",
      "Didn't pass (cutoff=0.0050): cat 532, mi = 0.002, pvalue = 0.007, GSE9006_HEALTHY_VS_TYPE_1_DIABETES_PBMC_AT_DX_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3093, mi = 0.002, pvalue = 0.006, GSE9006_TYPE_1_DIABETES_AT_DX_VS_1MONTH_POST_DX_PBMC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3500, mi = 0.002, pvalue = 0.010, GSE2770_UNTREATED_VS_TGFB_AND_IL4_TREATED_ACT_CD4_TCELL_48H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 2674, mi = 0.002, pvalue = 0.008, GSE360_L_DONOVANI_VS_M_TUBERCULOSIS_DC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1990, mi = 0.002, pvalue = 0.009, GSE14699_DELETIONAL_TOLERANCE_VS_ACTIVATED_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2231, mi = 0.002, pvalue = 0.010, GSE8835_CD4_VS_CD8_TCELL_CLL_PATIENT_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4245, mi = 0.002, pvalue = 0.015, GSE13411_IGM_MEMORY_BCELL_VS_PLASMA_CELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 272, mi = 0.002, pvalue = 0.018, GSE9946_MATURE_STIMULATORY_VS_PROSTAGLANDINE2_TREATED_MATURE_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2810, mi = 0.002, pvalue = 0.014, GSE29618_MONOCYTE_VS_MDC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4759, mi = 0.002, pvalue = 0.013, GSE22886_TCELL_VS_BCELL_NAIVE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4712, mi = 0.002, pvalue = 0.015, GSE22611_NOD2_TRANSD_VS_CTRL_TRANSD_HEK293_MDP_STIM_6H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 4681, mi = 0.002, pvalue = 0.017, GSE42021_CD24INT_VS_CD24LOW_TREG_THYMUS_UP\n",
      "Didn't pass (cutoff=0.0050): cat 710, mi = 0.002, pvalue = 0.013, GSE21927_UNTREATED_VS_GMCSF_GCSF_TREATED_BONE_MARROW_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4734, mi = 0.002, pvalue = 0.020, GSE14000_TRANSLATED_RNA_VS_MRNA_4H_LPS_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 4203, mi = 0.002, pvalue = 0.016, GSE11864_CSF1_PAM3CYS_VS_CSF1_IFNG_PAM3CYS_IN_MAC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 1142, mi = 0.002, pvalue = 0.018, GSE2585_CTEC_VS_THYMIC_MACROPHAGE_UP\n",
      "Didn't pass (cutoff=0.0050): cat 3315, mi = 0.002, pvalue = 0.023, GSE34179_THPOK_KO_VS_WT_VA14I_NKTCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 1779, mi = 0.002, pvalue = 0.019, GSE42021_CD24LO_TREG_VS_CD24LO_TCONV_THYMUS_UP\n",
      "Total number of categories evaluated = 141.\n",
      "Number of categories that passed the tests = 23\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 8, 10, 0.191466729017407\n",
      "best merge is 0, 7, 0.229678788122079\n",
      "best merge is 13, 19, 0.26835639083093\n",
      "best merge is 12, 20, 0.378186173359373\n",
      "best merge is 9, 21, 0.380675449334158\n",
      "best merge is 5, 15, 0.427781333532485\n",
      "best merge is 0, 18, 0.441146098360556\n",
      "best merge is 13, 22, 0.506189816066245\n",
      "best merge is 4, 12, 0.524337218568507\n",
      "best merge is 6, 14, 0.539354363128763\n",
      "best merge is 3, 16, 0.541143637580662\n",
      "best merge is 2, 4, 0.66127881813107\n",
      "best merge is 9, 11, 0.679458169834911\n",
      "best merge is 0, 3, 0.684709096825867\n",
      "best merge is 5, 6, 0.684791070184769\n",
      "best merge is 1, 13, 0.699356901279881\n",
      "best merge is 0, 2, 0.899447539227503\n",
      "best merge is 8, 17, 0.925338678821741\n",
      "best merge is 0, 1, 0.928554736140347\n",
      "order classes\n",
      "9 - 1.71386390182395\n",
      "4 - 2.4285173348897\n",
      "4 - 2.66390310875648\n",
      "3 - 3.47920844394078\n",
      "3 - 2.96374897385863\n",
      "0 7 18 3 16 2 4 12 20\n",
      "1 13 19 22\n",
      "5 15 6 14\n",
      "8 10 17\n",
      "9 21 11\n",
      "Done.Start drawing\n",
      "9.06\t-14.97\n",
      "GSE37301_MULTIPOTENT_PROGENITOR_VS_LYMPHOID_PRIMED_MPP_UP\n",
      "GSE4142_GC_BCELL_VS_MEMORY_BCELL_DN\n",
      "GSE10325_CD4_TCELL_VS_LUPUS_CD4_TCELL_DN\n",
      "GSE11961_FOLLICULAR_BCELL_VS_MEMORY_BCELL_DAY40_DN\n",
      "GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_DN\n",
      "GSE40277_GATA1_AND_SATB1_TRANSDUCED_VS_CTRL_CD4_TCELL_UP\n",
      "GSE10240_IL22_VS_IL17_STIM_PRIMARY_BRONCHIAL_EPITHELIAL_CELLS_UP\n",
      "GSE24972_WT_VS_IRF8_KO_SPLEEN_FOLLICULAR_BCELL_UP\n",
      "GSE17721_POLYIC_VS_GARDIQUIMOD_24H_BMDC_DN\n",
      "GSE25085_FETAL_LIVER_VS_ADULT_BM_SP4_THYMIC_IMPLANT_UP\n",
      "GSE2706_R848_VS_LPS_2H_STIM_DC_DN\n",
      "GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_MACROPHAGE_DN\n",
      "GSE23321_CD8_STEM_CELL_MEMORY_VS_NAIVE_CD8_TCELL_DN\n",
      "GSE5542_IFNG_VS_IFNA_AND_IFNG_TREATED_EPITHELIAL_CELLS_24H_UP\n",
      "GSE15659_NAIVE_CD4_TCELL_VS_RESTING_TREG_UP\n",
      "GSE9960_GRAM_NEG_VS_GRAM_NEG_AND_POS_SEPSIS_PBMC_DN\n",
      "GSE21546_ELK1_KO_VS_SAP1A_KO_AND_ELK1_KO_DP_THYMOCYTES_DN\n",
      "GSE17721_POLYIC_VS_CPG_12H_BMDC_UP\n",
      "GSE2770_IL4_ACT_VS_ACT_CD4_TCELL_48H_UP\n",
      "GSE21063_3H_VS_16H_ANTI_IGM_STIM_NFATC1_KOBCELL_UP\n",
      "GSE18791_CTRL_VS_NEWCASTLE_VIRUS_DC_1H_UP\n",
      "GSE22045_TREG_VS_TCONV_DN\n",
      "GSE14769_UNSTIM_VS_20MIN_LPS_BMDM_UP\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 8, 10, 0.191466729017407\n",
      "best merge is 0, 7, 0.229678788122079\n",
      "best merge is 13, 19, 0.26835639083093\n",
      "best merge is 12, 20, 0.378186173359373\n",
      "best merge is 9, 21, 0.380675449334158\n",
      "best merge is 5, 15, 0.427781333532485\n",
      "best merge is 0, 18, 0.441146098360556\n",
      "best merge is 13, 22, 0.506189816066245\n",
      "best merge is 4, 12, 0.524337218568507\n",
      "best merge is 6, 14, 0.539354363128763\n",
      "best merge is 3, 16, 0.541143637580662\n",
      "best merge is 2, 4, 0.66127881813107\n",
      "best merge is 9, 11, 0.679458169834911\n",
      "best merge is 0, 3, 0.684709096825867\n",
      "best merge is 5, 6, 0.684791070184769\n",
      "best merge is 1, 13, 0.699356901279881\n",
      "best merge is 0, 2, 0.899447539227503\n",
      "best merge is 8, 17, 0.925338678821741\n",
      "best merge is 0, 1, 0.928554736140347\n",
      "order classes\n",
      "9 - 1.71386390182395\n",
      "4 - 2.4285173348897\n",
      "4 - 2.66390310875648\n",
      "3 - 3.47920844394078\n",
      "3 - 2.96374897385863\n",
      "0 7 18 3 16 2 4 12 20\n",
      "1 13 19 22\n",
      "5 15 6 14\n",
      "8 10 17\n",
      "9 21 11\n",
      "Done.-7.99250936329588\t42\t0\n",
      "4.53058676654182\t42\t24.0149812734082\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.70162297128589\t42\t25.5355805243446\n",
      "5.97253433208489\t42\t25.8676654182272\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.55805243445693\t42\t26.4619225967541\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.13483146067416\t42\t26.916354556804\n",
      "14.6766541822722\t42\t27.3533083645443\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c7/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_c7/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c7.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 17786\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8396 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8396\t17786\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "763 hits go_terms_num = 17786\n",
      "cat 717, mi = 0.00307437, BYSTROEM_CORRELATED_WITH_IL5_DN\n",
      "cat 8001, mi = 0.00299551, GSE11961_FOLLICULAR_BCELL_VS_MEMORY_BCELL_DAY40_DN\n",
      "cat 8848, mi = 0.00283252, GSE17721_CTRL_VS_POLYIC_12H_BMDC_UP\n",
      "cat 9601, mi = 0.00282533, GSE22045_TREG_VS_TCONV_DN\n",
      "cat 16843, mi = 0.00282505, TONKS_TARGETS_OF_RUNX1_RUNX1T1_FUSION_HSC_DN\n",
      "cat 8444, mi = 0.00282476, GSE15330_HSC_VS_GRANULOCYTE_MONOCYTE_PROGENITOR_UP\n",
      "cat 14634, mi = 0.00278748, MORF_PHB\n",
      "cat 7383, mi = 0.00275579, GO_SPINDLE_POLE\n",
      "cat 9477, mi = 0.00270488, GSE21546_ELK1_KO_VS_SAP1A_KO_AND_ELK1_KO_DP_THYMOCYTES_DN\n",
      "cat 8496, mi = 0.00269350, GSE15659_NAIVE_CD4_TCELL_VS_RESTING_TREG_UP\n",
      "cat 10235, mi = 0.00268530, GSE2706_R848_VS_LPS_2H_STIM_DC_DN\n",
      "cat 15287, mi = 0.00268257, PID_PDGFRA_PATHWAY\n",
      "cat 10085, mi = 0.00267890, GSE25123_CTRL_VS_ROSIGLITAZONE_STIM_MACROPHAGE_DN\n",
      "cat 16504, mi = 0.00267248, SMID_BREAST_CANCER_ERBB2_UP\n",
      "cat 4362, mi = 0.00267137, GO_NEGATIVE_REGULATION_OF_CELL_DIVISION\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 717, mi = 0.003, pvalue = 0.000, BYSTROEM_CORRELATED_WITH_IL5_DN\n",
      "Passed: cat 8001, mi = 0.003, pvalue = 0.001, GSE11961_FOLLICULAR_BCELL_VS_MEMORY_BCELL_DAY40_DN\n",
      "Passed: cat 9601, mi = 0.003, pvalue = 0.000, GSE22045_TREG_VS_TCONV_DN\n",
      "Passed: cat 8444, mi = 0.003, pvalue = 0.001, GSE15330_HSC_VS_GRANULOCYTE_MONOCYTE_PROGENITOR_UP\n",
      "Passed: cat 7383, mi = 0.003, pvalue = 0.001, GO_SPINDLE_POLE\n",
      "Passed: cat 9477, mi = 0.003, pvalue = 0.001, GSE21546_ELK1_KO_VS_SAP1A_KO_AND_ELK1_KO_DP_THYMOCYTES_DN\n",
      "Passed: cat 10235, mi = 0.003, pvalue = 0.001, GSE2706_R848_VS_LPS_2H_STIM_DC_DN\n",
      "Passed: cat 16504, mi = 0.003, pvalue = 0.001, SMID_BREAST_CANCER_ERBB2_UP\n",
      "Passed: cat 441, mi = 0.003, pvalue = 0.001, BIOCARTA_GLEEVEC_PATHWAY\n",
      "Passed: cat 1828, mi = 0.003, pvalue = 0.000, GNF2_RPA1\n",
      "Passed: cat 11664, mi = 0.003, pvalue = 0.001, GSE40277_GATA1_AND_SATB1_TRANSDUCED_VS_CTRL_CD4_TCELL_UP\n",
      "Passed: cat 1662, mi = 0.003, pvalue = 0.002, GERHOLD_ADIPOGENESIS_DN\n",
      "Passed: cat 12675, mi = 0.003, pvalue = 0.001, GSE9960_GRAM_NEG_VS_GRAM_NEG_AND_POS_SEPSIS_PBMC_DN\n",
      "Passed: cat 1318, mi = 0.003, pvalue = 0.002, EVI1_05\n",
      "Passed: cat 9962, mi = 0.003, pvalue = 0.002, GSE24210_TCONV_VS_TREG_UP\n",
      "Passed: cat 9157, mi = 0.002, pvalue = 0.002, GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_DN\n",
      "Passed: cat 2710, mi = 0.002, pvalue = 0.003, GO_COATED_VESICLE_MEMBRANE\n",
      "Passed: cat 112, mi = 0.002, pvalue = 0.002, AKT_UP_MTOR_DN.V1_DN\n",
      "Passed: cat 2780, mi = 0.002, pvalue = 0.002, GO_CULLIN_RING_UBIQUITIN_LIGASE_COMPLEX\n",
      "Passed: cat 4014, mi = 0.002, pvalue = 0.003, GO_MESODERM_MORPHOGENESIS\n",
      "Passed: cat 8972, mi = 0.002, pvalue = 0.003, GSE17721_POLYIC_VS_CPG_12H_BMDC_UP\n",
      "Passed: cat 11230, mi = 0.002, pvalue = 0.003, GSE37301_MULTIPOTENT_PROGENITOR_VS_LYMPHOID_PRIMED_MPP_UP\n",
      "Passed: cat 12795, mi = 0.002, pvalue = 0.004, HALLMARK_FATTY_ACID_METABOLISM\n",
      "Passed: cat 11335, mi = 0.002, pvalue = 0.004, GSE37605_C57BL6_VS_NOD_FOXP3_IRES_GFP_TREG_DN\n",
      "Passed: cat 1904, mi = 0.002, pvalue = 0.004, GO_ACETYLTRANSFERASE_COMPLEX\n",
      "Passed: cat 12695, mi = 0.002, pvalue = 0.004, GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "Didn't pass (cutoff=0.0050): cat 17574, mi = 0.002, pvalue = 0.005, chr1p22\n",
      "Didn't pass (cutoff=0.0050): cat 16571, mi = 0.002, pvalue = 0.007, STAMBOLSKY_TARGETS_OF_MUTATED_TP53_DN\n",
      "Passed: cat 16004, mi = 0.002, pvalue = 0.005, REACTOME_RNA_POL_II_TRANSCRIPTION\n",
      "Passed: cat 9155, mi = 0.002, pvalue = 0.004, GSE18804_SPLEEN_MACROPHAGE_VS_BRAIN_TUMORAL_MACROPHAGE_DN\n",
      "Passed: cat 14550, mi = 0.002, pvalue = 0.004, MORF_CDK2\n",
      "Didn't pass (cutoff=0.0050): cat 14549, mi = 0.002, pvalue = 0.007, MORF_CDH4\n",
      "Didn't pass (cutoff=0.0050): cat 9134, mi = 0.002, pvalue = 0.006, GSE18791_CTRL_VS_NEWCASTLE_VIRUS_DC_1H_UP\n",
      "Didn't pass (cutoff=0.0050): cat 7640, mi = 0.002, pvalue = 0.008, GO_TRNA_BINDING\n",
      "Didn't pass (cutoff=0.0050): cat 9184, mi = 0.002, pvalue = 0.009, GSE19198_CTRL_VS_IL21_TREATED_TCELL_1H_UP\n",
      "Passed: cat 744, mi = 0.002, pvalue = 0.005, CAFFAREL_RESPONSE_TO_THC_24HR_5_UP\n",
      "Didn't pass (cutoff=0.0050): cat 17398, mi = 0.002, pvalue = 0.012, ZHAN_V2_LATE_DIFFERENTIATION_GENES\n",
      "Didn't pass (cutoff=0.0050): cat 6056, mi = 0.002, pvalue = 0.010, GO_REGULATION_OF_ADENYLATE_CYCLASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 6476, mi = 0.002, pvalue = 0.009, GO_REGULATION_OF_LYASE_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 12673, mi = 0.002, pvalue = 0.014, GSE9946_MATURE_STIMULATORY_VS_PROSTAGLANDINE2_TREATED_MATURE_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 12079, mi = 0.002, pvalue = 0.011, GSE46242_TH1_VS_ANERGIC_TH1_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3632, mi = 0.002, pvalue = 0.015, GO_INORGANIC_ANION_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 8218, mi = 0.002, pvalue = 0.011, GSE13522_WT_VS_IFNG_KO_SKIN_UP\n",
      "Didn't pass (cutoff=0.0050): cat 8263, mi = 0.002, pvalue = 0.012, GSE14000_TRANSLATED_RNA_VS_MRNA_4H_LPS_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 2638, mi = 0.002, pvalue = 0.015, GO_CHLORIDE_TRANSPORT\n",
      "Didn't pass (cutoff=0.0050): cat 7978, mi = 0.002, pvalue = 0.012, GSE11864_UNTREATED_VS_CSF1_IN_MAC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 10236, mi = 0.002, pvalue = 0.012, GSE2706_R848_VS_LPS_2H_STIM_DC_UP\n",
      "Didn't pass (cutoff=0.0050): cat 10781, mi = 0.002, pvalue = 0.013, GSE32533_MIR17_KO_VS_MIR17_OVEREXPRESS_ACT_CD4_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 12149, mi = 0.002, pvalue = 0.016, GSE4984_UNTREATED_VS_LPS_TREATED_DC_DN\n",
      "Didn't pass (cutoff=0.0050): cat 3631, mi = 0.002, pvalue = 0.019, GO_INORGANIC_ANION_TRANSMEMBRANE_TRANSPORTER_ACTIVITY\n",
      "Didn't pass (cutoff=0.0050): cat 11771, mi = 0.002, pvalue = 0.020, GSE41867_DAY6_VS_DAY15_LCMV_CLONE13_EFFECTOR_CD8_TCELL_DN\n",
      "Didn't pass (cutoff=0.0050): cat 8406, mi = 0.002, pvalue = 0.021, GSE14769_UNSTIM_VS_20MIN_LPS_BMDM_UP\n",
      "Didn't pass (cutoff=0.0050): cat 7710, mi = 0.002, pvalue = 0.016, GO_VACUOLAR_LUMEN\n",
      "Didn't pass (cutoff=0.0050): cat 5580, mi = 0.002, pvalue = 0.008, GO_POSITIVE_REGULATION_OF_OSTEOCLAST_DIFFERENTIATION\n",
      "Didn't pass (cutoff=0.0050): cat 4454, mi = 0.002, pvalue = 0.011, GO_NEGATIVE_REGULATION_OF_HISTONE_METHYLATION\n",
      "Didn't pass (cutoff=0.0050): cat 5826, mi = 0.002, pvalue = 0.012, GO_PROTEIN_DEPOLYMERIZATION\n",
      "Total number of categories evaluated = 313.\n",
      "Number of categories that passed the tests = 30\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Cluster rows .. best merge is 1, 23, 0.0613772540142111\n",
      "best merge is 8, 9, 0.202759336288934\n",
      "best merge is 8, 24, 0.218171174413954\n",
      "best merge is 4, 28, 0.224396188066859\n",
      "best merge is 22, 26, 0.321240326374184\n",
      "best merge is 7, 15, 0.342669870103567\n",
      "best merge is 0, 11, 0.345447789964843\n",
      "best merge is 17, 20, 0.356108139127671\n",
      "best merge is 6, 17, 0.408383664954716\n",
      "best merge is 10, 21, 0.425059983799119\n",
      "best merge is 8, 27, 0.449172581498358\n",
      "best merge is 4, 25, 0.455890562644585\n",
      "best merge is 3, 29, 0.457275813242259\n",
      "best merge is 12, 16, 0.499560363230778\n",
      "best merge is 0, 1, 0.527772325168454\n",
      "best merge is 5, 18, 0.574897194260299\n",
      "best merge is 3, 13, 0.601358373034695\n",
      "best merge is 7, 8, 0.650403020678836\n",
      "best merge is 2, 19, 0.653158146294409\n",
      "best merge is 0, 7, 0.708151765903412\n",
      "best merge is 4, 5, 0.713398841482064\n",
      "best merge is 12, 14, 0.760402145979446\n",
      "best merge is 3, 22, 0.786844228384451\n",
      "best merge is 0, 4, 0.887505983351766\n",
      "best merge is 6, 10, 0.915301807706165\n",
      "best merge is 0, 3, 0.944097791468013\n",
      "order classes\n",
      "15 - 1.3762282118325\n",
      "2 - 4.89868609720807\n",
      "5 - 3.21710316287642\n",
      "5 - 3.40307822355514\n",
      "3 - 4.48969923375482\n",
      "0 11 1 23 7 15 8 9 24 27 4 28 25 5 18\n",
      "2 19\n",
      "3 29 13 22 26\n",
      "6 17 20 10 21\n",
      "12 16 14\n",
      "Done.Start drawing\n",
      "9.06\t-14.97\n",
      "SMID_BREAST_CANCER_ERBB2_UP\n",
      "GSE37301_MULTIPOTENT_PROGENITOR_VS_LYMPHOID_PRIMED_MPP_UP\n",
      "HALLMARK_FATTY_ACID_METABOLISM\n",
      "REACTOME_RNA_POL_II_TRANSCRIPTION\n",
      "BIOCARTA_GLEEVEC_PATHWAY\n",
      "GNF2_RPA1\n",
      "GSE18804_SPLEEN_MACROPHAGE_VS_BRAIN_TUMORAL_MACROPHAGE_DN\n",
      "GSE15330_HSC_VS_GRANULOCYTE_MONOCYTE_PROGENITOR_UP\n",
      "GERHOLD_ADIPOGENESIS_DN\n",
      "GSE11961_FOLLICULAR_BCELL_VS_MEMORY_BCELL_DAY40_DN\n",
      "GSE18804_SPLEEN_MACROPHAGE_VS_COLON_TUMORAL_MACROPHAGE_DN\n",
      "GO_ACETYLTRANSFERASE_COMPLEX\n",
      "GSE37605_C57BL6_VS_NOD_FOXP3_IRES_GFP_TREG_DN\n",
      "GO_SPINDLE_POLE\n",
      "MORF_CDK2\n",
      "GSE9988_ANTI_TREM1_VS_CTRL_TREATED_MONOCYTES_DN\n",
      "GSE40277_GATA1_AND_SATB1_TRANSDUCED_VS_CTRL_CD4_TCELL_UP\n",
      "BYSTROEM_CORRELATED_WITH_IL5_DN\n",
      "CAFFAREL_RESPONSE_TO_THC_24HR_5_UP\n",
      "GSE2706_R848_VS_LPS_2H_STIM_DC_DN\n",
      "GSE21546_ELK1_KO_VS_SAP1A_KO_AND_ELK1_KO_DP_THYMOCYTES_DN\n",
      "GO_CULLIN_RING_UBIQUITIN_LIGASE_COMPLEX\n",
      "EVI1_05\n",
      "GSE9960_GRAM_NEG_VS_GRAM_NEG_AND_POS_SEPSIS_PBMC_DN\n",
      "GSE17721_POLYIC_VS_CPG_12H_BMDC_UP\n",
      "GSE24210_TCONV_VS_TREG_UP\n",
      "GO_COATED_VESICLE_MEMBRANE\n",
      "GSE22045_TREG_VS_TCONV_DN\n",
      "GO_MESODERM_MORPHOGENESIS\n",
      "AKT_UP_MTOR_DN.V1_DN\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "Cluster rows .. best merge is 1, 23, 0.0613772540142111\n",
      "best merge is 8, 9, 0.202759336288934\n",
      "best merge is 8, 24, 0.218171174413954\n",
      "best merge is 4, 28, 0.224396188066859\n",
      "best merge is 22, 26, 0.321240326374184\n",
      "best merge is 7, 15, 0.342669870103567\n",
      "best merge is 0, 11, 0.345447789964843\n",
      "best merge is 17, 20, 0.356108139127671\n",
      "best merge is 6, 17, 0.408383664954716\n",
      "best merge is 10, 21, 0.425059983799119\n",
      "best merge is 8, 27, 0.449172581498358\n",
      "best merge is 4, 25, 0.455890562644585\n",
      "best merge is 3, 29, 0.457275813242259\n",
      "best merge is 12, 16, 0.499560363230778\n",
      "best merge is 0, 1, 0.527772325168454\n",
      "best merge is 5, 18, 0.574897194260299\n",
      "best merge is 3, 13, 0.601358373034695\n",
      "best merge is 7, 8, 0.650403020678836\n",
      "best merge is 2, 19, 0.653158146294409\n",
      "best merge is 0, 7, 0.708151765903412\n",
      "best merge is 4, 5, 0.713398841482064\n",
      "best merge is 12, 14, 0.760402145979446\n",
      "best merge is 3, 22, 0.786844228384451\n",
      "best merge is 0, 4, 0.887505983351766\n",
      "best merge is 6, 10, 0.915301807706165\n",
      "best merge is 0, 3, 0.944097791468013\n",
      "order classes\n",
      "15 - 1.3762282118325\n",
      "2 - 4.89868609720807\n",
      "5 - 3.21710316287642\n",
      "5 - 3.40307822355514\n",
      "3 - 4.48969923375482\n",
      "0 11 1 23 7 15 8 9 24 27 4 28 25 5 18\n",
      "2 19\n",
      "3 29 13 22 26\n",
      "6 17 20 10 21\n",
      "12 16 14\n",
      "Done.-8.00124843945069\t42\t0\n",
      "4.52184769038701\t42\t23.9975031210986\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.69288389513109\t42\t25.5355805243446\n",
      "5.96379525593009\t42\t25.8501872659176\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.56679151061174\t42\t26.4619225967541\n",
      "6.7940074906367\t42\t26.6716604244694\n",
      "7.14357053682896\t42\t26.916354556804\n",
      "14.685393258427\t42\t27.3707865168539\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_full/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_full/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_full.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 50\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 1820 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 1820\t50\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "165 hits go_terms_num = 50\n",
      "cat 20, mi = 0.00895481, HALLMARK_MYC_TARGETS_V1\n",
      "cat 0, mi = 0.00879727, HALLMARK_FATTY_ACID_METABOLISM\n",
      "cat 47, mi = 0.00701340, HALLMARK_COMPLEMENT\n",
      "cat 29, mi = 0.00698062, HALLMARK_MYC_TARGETS_V2\n",
      "cat 12, mi = 0.00628367, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "cat 7, mi = 0.00619443, HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY\n",
      "cat 34, mi = 0.00609158, HALLMARK_ANDROGEN_RESPONSE\n",
      "cat 4, mi = 0.00605697, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "cat 16, mi = 0.00584621, HALLMARK_MTORC1_SIGNALING\n",
      "cat 17, mi = 0.00574767, HALLMARK_IL2_STAT5_SIGNALING\n",
      "cat 11, mi = 0.00569874, HALLMARK_ADIPOGENESIS\n",
      "cat 15, mi = 0.00549574, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "cat 14, mi = 0.00532185, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "cat 42, mi = 0.00512149, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "cat 9, mi = 0.00496901, HALLMARK_SPERMATOGENESIS\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 20, mi = 0.009, pvalue = 0.022, HALLMARK_MYC_TARGETS_V1\n",
      "Didn't pass (cutoff=0.0050): cat 0, mi = 0.009, pvalue = 0.024, HALLMARK_FATTY_ACID_METABOLISM\n",
      "Didn't pass (cutoff=0.0050): cat 47, mi = 0.007, pvalue = 0.078, HALLMARK_COMPLEMENT\n",
      "Didn't pass (cutoff=0.0050): cat 29, mi = 0.007, pvalue = 0.108, HALLMARK_MYC_TARGETS_V2\n",
      "Didn't pass (cutoff=0.0050): cat 12, mi = 0.006, pvalue = 0.181, HALLMARK_WNT_BETA_CATENIN_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 7, mi = 0.006, pvalue = 0.215, HALLMARK_REACTIVE_OXYGEN_SPECIES_PATHWAY\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.006, pvalue = 0.163, HALLMARK_ANDROGEN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 4, mi = 0.006, pvalue = 0.173, HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.006, pvalue = 0.159, HALLMARK_MTORC1_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 17, mi = 0.006, pvalue = 0.161, HALLMARK_IL2_STAT5_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 11, mi = 0.006, pvalue = 0.190, HALLMARK_ADIPOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 15, mi = 0.005, pvalue = 0.205, HALLMARK_UNFOLDED_PROTEIN_RESPONSE\n",
      "Didn't pass (cutoff=0.0050): cat 14, mi = 0.005, pvalue = 0.223, HALLMARK_ESTROGEN_RESPONSE_LATE\n",
      "Didn't pass (cutoff=0.0050): cat 42, mi = 0.005, pvalue = 0.289, HALLMARK_PI3K_AKT_MTOR_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.005, pvalue = 0.360, HALLMARK_SPERMATOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.005, pvalue = 0.401, HALLMARK_NOTCH_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 24, mi = 0.005, pvalue = 0.360, HALLMARK_IL6_JAK_STAT3_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 35, mi = 0.005, pvalue = 0.408, HALLMARK_ANGIOGENESIS\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.005, pvalue = 0.348, HALLMARK_TGF_BETA_SIGNALING\n",
      "Didn't pass (cutoff=0.0050): cat 36, mi = 0.005, pvalue = 0.342, HALLMARK_DNA_REPAIR\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "3.27\t-14.97\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_h/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_msigdb_h/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_h.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 9321 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 9321\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "847 hits go_terms_num = 790\n",
      "cat 676, mi = 0.00197794, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 657, mi = 0.00195680, TIA1_NA_POSTAR_CIMS\n",
      "cat 658, mi = 0.00195680, TIA1_NA_POSTAR_iCLIP\n",
      "cat 677, mi = 0.00189633, TNRC6C_HEK293_Plass\n",
      "cat 312, mi = 0.00183738, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "cat 321, mi = 0.00171814, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 672, mi = 0.00163405, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 569, mi = 0.00161163, SLBP_K562_POSTAR_eCLIP\n",
      "cat 668, mi = 0.00155590, TNRC6A_HEK293_Plass\n",
      "cat 300, mi = 0.00150436, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "cat 675, mi = 0.00140136, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "cat 770, mi = 0.00137254, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 275, mi = 0.00109756, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "cat 615, mi = 0.00104497, TAF15_Brain_POSTAR_CIMS\n",
      "cat 354, mi = 0.00103022, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.002, pvalue = 0.007, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.002, pvalue = 0.007, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.002, pvalue = 0.008, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 677, mi = 0.002, pvalue = 0.007, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.002, pvalue = 0.017, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.024, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.002, pvalue = 0.039, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 569, mi = 0.002, pvalue = 0.032, SLBP_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.002, pvalue = 0.033, TNRC6A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 300, mi = 0.002, pvalue = 0.033, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 675, mi = 0.001, pvalue = 0.066, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.001, pvalue = 0.073, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.001, pvalue = 0.236, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 615, mi = 0.001, pvalue = 0.241, TAF15_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.001, pvalue = 0.223, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.237, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.001, pvalue = 0.370, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.001, pvalue = 0.276, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.001, pvalue = 0.264, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 447, mi = 0.001, pvalue = 0.257, NPM1_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.06\t-14.97\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_ids/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_ids/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_ids.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_names/'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8587 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8587\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "781 hits go_terms_num = 790\n",
      "cat 677, mi = 0.00232668, TNRC6C_HEK293_Plass\n",
      "cat 674, mi = 0.00220793, TNRC6C_HEK293_POSTAR_PAR-CLIP\n",
      "cat 676, mi = 0.00215177, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00190344, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 657, mi = 0.00189051, TIA1_NA_POSTAR_CIMS\n",
      "cat 658, mi = 0.00189051, TIA1_NA_POSTAR_iCLIP\n",
      "cat 569, mi = 0.00163238, SLBP_K562_POSTAR_eCLIP\n",
      "cat 312, mi = 0.00153670, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "cat 668, mi = 0.00149003, TNRC6A_HEK293_Plass\n",
      "cat 672, mi = 0.00145890, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "cat 615, mi = 0.00140669, TAF15_Brain_POSTAR_CIMS\n",
      "cat 354, mi = 0.00139860, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 300, mi = 0.00139832, HNRNPF_HEK293T_POSTAR_CIMS\n",
      "cat 675, mi = 0.00134701, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "cat 770, mi = 0.00128441, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 677, mi = 0.002, pvalue = 0.003, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.021, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.002, pvalue = 0.023, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.002, pvalue = 0.021, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.002, pvalue = 0.074, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.113, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 615, mi = 0.001, pvalue = 0.115, TAF15_Brain_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.001, pvalue = 0.101, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 675, mi = 0.001, pvalue = 0.115, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 770, mi = 0.001, pvalue = 0.141, YTHDF3_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 667, mi = 0.001, pvalue = 0.135, TNRC6A_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 670, mi = 0.001, pvalue = 0.144, TNRC6B_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 629, mi = 0.001, pvalue = 0.164, TARBP2_MDA-LM2_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 275, mi = 0.001, pvalue = 0.240, HNRNPA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 749, mi = 0.001, pvalue = 0.221, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 326, mi = 0.001, pvalue = 0.201, HNRNPU_HeLa_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 31, mi = 0.001, pvalue = 0.415, AUH_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 635, mi = 0.001, pvalue = 0.297, TARDBP_H9_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 637, mi = 0.001, pvalue = 0.326, TARDBP_H9_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 671, mi = 0.001, pvalue = 0.311, TNRC6B_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 173, mi = 0.001, pvalue = 0.336, EIF4A3_HeLa_POSTAR_CIMS\n",
      "Total number of categories evaluated = 21.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.06\t-14.97\n",
      "TNRC6C_HEK293_Plass\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.99250936329588\t42\t0\n",
      "4.53058676654182\t42\t24.0149812734082\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.69288389513109\t42\t25.5355805243446\n",
      "5.96379525593009\t42\t25.8501872659176\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.55805243445693\t42\t26.4619225967541\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.14357053682896\t42\t26.916354556804\n",
      "14.685393258427\t42\t27.3707865168539\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8410 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8410\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "765 hits go_terms_num = 790\n",
      "cat 677, mi = 0.00235205, TNRC6C_HEK293_Plass\n",
      "cat 567, mi = 0.00229322, SFPQ_HepG2_POSTAR_eCLIP\n",
      "cat 701, mi = 0.00227242, UCHL5_HepG2_POSTAR_eCLIP\n",
      "cat 627, mi = 0.00200643, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "cat 626, mi = 0.00190005, TARBP2_HEK293_POSTAR_PAR-CLIP\n",
      "cat 301, mi = 0.00180240, HNRNPF_HEK293T_POSTAR_HITS-CLIP\n",
      "cat 302, mi = 0.00180240, HNRNPF_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 230, mi = 0.00177826, FUS_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 684, mi = 0.00174374, TROVE2_K562_POSTAR_eCLIP\n",
      "cat 396, mi = 0.00173327, LSM11_HepG2_POSTAR_eCLIP\n",
      "cat 765, mi = 0.00172022, YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "cat 433, mi = 0.00170068, NOL12_HepG2_POSTAR_eCLIP\n",
      "cat 616, mi = 0.00167924, TAF15_Brain_POSTAR_HITS-CLIP\n",
      "cat 732, mi = 0.00166760, XPO5_HepG2_POSTAR_eCLIP\n",
      "cat 380, mi = 0.00165813, LIN28A_H9_POSTAR_HITS-CLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 677, mi = 0.002, pvalue = 0.002, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.002, pvalue = 0.005, SFPQ_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 701, mi = 0.002, pvalue = 0.005, UCHL5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 627, mi = 0.002, pvalue = 0.012, TARBP2_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 626, mi = 0.002, pvalue = 0.018, TARBP2_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 684, mi = 0.002, pvalue = 0.037, TROVE2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 396, mi = 0.002, pvalue = 0.031, LSM11_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 765, mi = 0.002, pvalue = 0.033, YTHDF2_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 616, mi = 0.002, pvalue = 0.027, TAF15_Brain_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 732, mi = 0.002, pvalue = 0.045, XPO5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 380, mi = 0.002, pvalue = 0.051, LIN28A_H9_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 422, mi = 0.002, pvalue = 0.075, NCBP2_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 545, mi = 0.002, pvalue = 0.076, RPS5_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 653, mi = 0.002, pvalue = 0.068, TIA1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 66, mi = 0.002, pvalue = 0.071, CPSF2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 404, mi = 0.002, pvalue = 0.091, MBNL2_frontalcortex_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 758, mi = 0.002, pvalue = 0.060, YTHDF1_Huh7_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 757, mi = 0.002, pvalue = 0.062, YTHDF1_Huh7_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 86, mi = 0.002, pvalue = 0.067, CSTF2T_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 197, mi = 0.001, pvalue = 0.070, FAM120A_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 628, mi = 0.001, pvalue = 0.066, TARBP2_HEK293_POSTAR_Piranha_0.01\n",
      "Total number of categories evaluated = 16.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.06\t-14.97\n",
      "TNRC6C_HEK293_Plass\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.99250936329588\t42\t0\n",
      "4.53058676654182\t42\t24.0149812734082\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.69288389513109\t42\t25.5355805243446\n",
      "5.96379525593009\t42\t25.8501872659176\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.55805243445693\t42\t26.4619225967541\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.14357053682896\t42\t26.916354556804\n",
      "14.685393258427\t42\t27.3707865168539\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_3UTR/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_3UTR/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_3UTR.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8316 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8316\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "756 hits go_terms_num = 790\n",
      "cat 350, mi = 0.00221156, IGF2BP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 600, mi = 0.00214290, SSB_HEK293_POSTAR_PARalyzer\n",
      "cat 707, mi = 0.00204378, UPF1_HEK_293_FRT_POSTAR_HITS-CLIP\n",
      "cat 62, mi = 0.00198408, CPSF1_HEK293_POSTAR_PARalyzer\n",
      "cat 599, mi = 0.00198183, SSB_HEK293_POSTAR_PAR-CLIP\n",
      "cat 708, mi = 0.00188643, UPF1_HEK_293_FRT_POSTAR_Piranha_0.01\n",
      "cat 534, mi = 0.00186788, RBM5_HepG2_POSTAR_eCLIP\n",
      "cat 487, mi = 0.00186177, PTBP1_HepG2_POSTAR_eCLIP\n",
      "cat 185, mi = 0.00184198, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "cat 504, mi = 0.00179998, QKI_HepG2_POSTAR_eCLIP\n",
      "cat 167, mi = 0.00179907, EIF3G_HEK293T_POSTAR_PAR-CLIP\n",
      "cat 528, mi = 0.00179046, RBM22_K562_POSTAR_eCLIP\n",
      "cat 188, mi = 0.00178204, EWSR1_HEK293_POSTAR_Piranha_0.01\n",
      "cat 183, mi = 0.00174177, ELAVL1_HeLa_POSTAR_PAR-CLIP\n",
      "cat 273, mi = 0.00171740, HNRNPA1_HEK293T_POSTAR_HITS-CLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 350, mi = 0.002, pvalue = 0.002, IGF2BP3_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 600, mi = 0.002, pvalue = 0.009, SSB_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 707, mi = 0.002, pvalue = 0.013, UPF1_HEK_293_FRT_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 599, mi = 0.002, pvalue = 0.021, SSB_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 708, mi = 0.002, pvalue = 0.029, UPF1_HEK_293_FRT_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 534, mi = 0.002, pvalue = 0.030, RBM5_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 487, mi = 0.002, pvalue = 0.032, PTBP1_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 185, mi = 0.002, pvalue = 0.034, ELAVL1_HeLa_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 504, mi = 0.002, pvalue = 0.044, QKI_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 528, mi = 0.002, pvalue = 0.027, RBM22_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 183, mi = 0.002, pvalue = 0.052, ELAVL1_HeLa_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 273, mi = 0.002, pvalue = 0.029, HNRNPA1_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 274, mi = 0.002, pvalue = 0.030, HNRNPA1_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 567, mi = 0.002, pvalue = 0.067, SFPQ_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 577, mi = 0.002, pvalue = 0.079, SMNDC1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 50, mi = 0.002, pvalue = 0.066, CF_Im25_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 433, mi = 0.002, pvalue = 0.092, NOL12_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 315, mi = 0.002, pvalue = 0.086, HNRNPM_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 361, mi = 0.002, pvalue = 0.091, ILF3_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 288, mi = 0.002, pvalue = 0.111, HNRNPC_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 526, mi = 0.001, pvalue = 0.122, RBM22_HepG2_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 22.\n",
      "Number of categories that passed the tests = 1\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.06\t-14.97\n",
      "IGF2BP3_HEK293_POSTAR_Piranha_0.01\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.99250936329588\t42\t0\n",
      "4.53058676654182\t42\t24.0149812734082\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.69288389513109\t42\t25.5355805243446\n",
      "5.96379525593009\t42\t25.8501872659176\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.55805243445693\t42\t26.4619225967541\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.13483146067416\t42\t26.916354556804\n",
      "14.6766541822722\t42\t27.3533083645443\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_5UTR/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_5UTR/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_5UTR.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8422 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8422\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "766 hits go_terms_num = 790\n",
      "cat 45, mi = 0.00308456, CDC40_HepG2_POSTAR_eCLIP\n",
      "cat 141, mi = 0.00242199, DICER1_HEK293_POSTAR_PARalyzer\n",
      "cat 140, mi = 0.00239315, DICER1_HEK293_POSTAR_PAR-CLIP\n",
      "cat 424, mi = 0.00226353, NCBP2_K562_POSTAR_eCLIP\n",
      "cat 439, mi = 0.00221187, NOP56_HEK293_POSTAR_PAR-CLIP\n",
      "cat 440, mi = 0.00221187, NOP56_HEK293_POSTAR_PARalyzer\n",
      "cat 273, mi = 0.00219185, HNRNPA1_HEK293T_POSTAR_HITS-CLIP\n",
      "cat 274, mi = 0.00219185, HNRNPA1_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 319, mi = 0.00209003, HNRNPUL1_HepG2_POSTAR_eCLIP\n",
      "cat 28, mi = 0.00206511, ATXN2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 267, mi = 0.00205050, GTF2F1_K562_POSTAR_eCLIP\n",
      "cat 512, mi = 0.00200319, RBM10_HEK293T_POSTAR_CIMS\n",
      "cat 203, mi = 0.00191343, FASTKD2_K562_POSTAR_eCLIP\n",
      "cat 680, mi = 0.00191242, TRA2A_K562_POSTAR_eCLIP\n",
      "cat 506, mi = 0.00184374, QKI_K562_POSTAR_eCLIP\n",
      "\n",
      "RESULTS\n",
      "Passed: cat 45, mi = 0.003, pvalue = 0.000, CDC40_HepG2_POSTAR_eCLIP\n",
      "Passed: cat 141, mi = 0.002, pvalue = 0.002, DICER1_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 439, mi = 0.002, pvalue = 0.007, NOP56_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 440, mi = 0.002, pvalue = 0.006, NOP56_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 506, mi = 0.002, pvalue = 0.028, QKI_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.002, pvalue = 0.030, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 407, mi = 0.002, pvalue = 0.037, MBNL2_hippocampus_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.002, pvalue = 0.099, C17orf85_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 406, mi = 0.001, pvalue = 0.124, MBNL2_hippocampus_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 449, mi = 0.001, pvalue = 0.014, NSUN2_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.001, pvalue = 0.044, DCK1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 623, mi = 0.001, pvalue = 0.116, TAF15_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 382, mi = 0.001, pvalue = 0.143, LIN28A_HEK293_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 10, mi = 0.001, pvalue = 0.149, AGO3_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 58, mi = 0.001, pvalue = 0.122, CPSF-160_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.001, pvalue = 0.204, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 96, mi = 0.001, pvalue = 0.197, CstF-64tau_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 301, mi = 0.001, pvalue = 0.164, HNRNPF_HEK293T_POSTAR_HITS-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 302, mi = 0.001, pvalue = 0.168, HNRNPF_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 621, mi = 0.001, pvalue = 0.236, TAF15_HepG2_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 494, mi = 0.001, pvalue = 0.306, PUM2_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 366, mi = 0.001, pvalue = 0.139, KHDRBS1_K562_POSTAR_eCLIP\n",
      "Total number of categories evaluated = 74.\n",
      "Number of categories that passed the tests = 2\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.06\t-14.97\n",
      "CDC40_HepG2_POSTAR_eCLIP\n",
      "DICER1_HEK293_POSTAR_PARalyzer\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Done.\n",
      "-7.99250936329588\t42\t0\n",
      "4.53058676654182\t42\t24.0149812734082\n",
      "5.29088639200999\t42\t25.0461922596754\n",
      "5.69288389513109\t42\t25.5355805243446\n",
      "5.96379525593009\t42\t25.8501872659176\n",
      "6.17353308364544\t42\t26.0774032459426\n",
      "6.36579275905119\t42\t26.2696629213483\n",
      "6.55805243445693\t42\t26.4619225967541\n",
      "6.7852684144819\t42\t26.6541822721598\n",
      "7.13483146067416\t42\t26.916354556804\n",
      "14.6766541822722\t42\t27.3533083645443\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_coding_exons/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_coding_exons.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 790\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8225 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8225\t790\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "748 hits go_terms_num = 790\n",
      "cat 674, mi = 0.00216759, TNRC6C_HEK293_POSTAR_PAR-CLIP\n",
      "cat 428, mi = 0.00215423, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "cat 676, mi = 0.00193556, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "cat 312, mi = 0.00191598, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "cat 749, mi = 0.00177016, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "cat 657, mi = 0.00176987, TIA1_NA_POSTAR_CIMS\n",
      "cat 658, mi = 0.00176987, TIA1_NA_POSTAR_iCLIP\n",
      "cat 97, mi = 0.00164249, DCK1_HEK293_Plass\n",
      "cat 354, mi = 0.00153227, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "cat 321, mi = 0.00152280, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "cat 705, mi = 0.00150322, UPF1_HEK293_Plass\n",
      "cat 208, mi = 0.00149373, FBL_HEK293_Plass\n",
      "cat 39, mi = 0.00144929, C17orf85_HEK293_Plass\n",
      "cat 675, mi = 0.00140247, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "cat 677, mi = 0.00137701, TNRC6C_HEK293_Plass\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 674, mi = 0.002, pvalue = 0.009, TNRC6C_HEK293_POSTAR_PAR-CLIP\n",
      "Didn't pass (cutoff=0.0050): cat 428, mi = 0.002, pvalue = 0.008, NCBP3_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 676, mi = 0.002, pvalue = 0.025, TNRC6C_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 312, mi = 0.002, pvalue = 0.028, HNRNPM_HEK293T_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 749, mi = 0.002, pvalue = 0.046, YTHDC2_HEK293T_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 657, mi = 0.002, pvalue = 0.044, TIA1_NA_POSTAR_CIMS\n",
      "Didn't pass (cutoff=0.0050): cat 658, mi = 0.002, pvalue = 0.048, TIA1_NA_POSTAR_iCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 97, mi = 0.002, pvalue = 0.062, DCK1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 354, mi = 0.002, pvalue = 0.079, IGF2BP3_PL45_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 321, mi = 0.002, pvalue = 0.102, HNRNPUL1_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 705, mi = 0.002, pvalue = 0.100, UPF1_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 208, mi = 0.001, pvalue = 0.083, FBL_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 39, mi = 0.001, pvalue = 0.096, C17orf85_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 675, mi = 0.001, pvalue = 0.127, TNRC6C_HEK293_POSTAR_PARalyzer\n",
      "Didn't pass (cutoff=0.0050): cat 677, mi = 0.001, pvalue = 0.115, TNRC6C_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 775, mi = 0.001, pvalue = 0.130, YWHAG_K562_POSTAR_eCLIP\n",
      "Didn't pass (cutoff=0.0050): cat 387, mi = 0.001, pvalue = 0.135, LIN28A_HEK293_Plass\n",
      "Didn't pass (cutoff=0.0050): cat 672, mi = 0.001, pvalue = 0.215, TNRC6B_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 667, mi = 0.001, pvalue = 0.161, TNRC6A_HEK293_POSTAR_Piranha_0.01\n",
      "Didn't pass (cutoff=0.0050): cat 668, mi = 0.001, pvalue = 0.227, TNRC6A_HEK293_Plass\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.06\t-14.97\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_introns/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_introns/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_introns.pdf'\n",
      "Reading expression data ... 1 by 9366... Done.\n",
      "Creating hash ... Done\n",
      "Reading GO names table ... entering\n",
      "Done.\n",
      "Reading GO index table ... entering\n",
      "Done.\n",
      "go_terms_num: 101\n",
      "Creating hash table for gene names in gene ontology index file ... Done.\n",
      "gene num = 8225 out of 9366 \n",
      "Creating hash table for gene ontology names ... Done.\n",
      "Populating the GO profile table ... 8225\t101\n",
      "Done.\n",
      "Writing internal quantized groups ...Done\n",
      "748 hits go_terms_num = 101\n",
      "cat 88, mi = 0.00186845, SRSF9.1\n",
      "cat 34, mi = 0.00138271, KHDRBS3\n",
      "cat 69, mi = 0.00120125, RBMS3.1\n",
      "cat 21, mi = 0.00104800, HNRNPK\n",
      "cat 6, mi = 0.00103481, CPEB4\n",
      "cat 73, mi = 0.00092716, SNRNP70\n",
      "cat 68, mi = 0.00086228, RBMS3\n",
      "cat 16, mi = 0.00084150, HNRNPA1L2\n",
      "cat 9, mi = 0.00083507, ESRP2\n",
      "cat 62, mi = 0.00072638, RBM46\n",
      "cat 13, mi = 0.00069220, FXR2\n",
      "cat 56, mi = 0.00052536, RBM3\n",
      "cat 29, mi = 0.00042064, HuR.4\n",
      "cat 27, mi = 0.00022913, HuR.2\n",
      "cat 0, mi = 0.00000000, ANKHD1\n",
      "\n",
      "RESULTS\n",
      "Didn't pass (cutoff=0.0050): cat 88, mi = 0.002, pvalue = 0.035, SRSF9.1\n",
      "Didn't pass (cutoff=0.0050): cat 34, mi = 0.001, pvalue = 0.126, KHDRBS3\n",
      "Didn't pass (cutoff=0.0050): cat 69, mi = 0.001, pvalue = 0.253, RBMS3.1\n",
      "Didn't pass (cutoff=0.0050): cat 21, mi = 0.001, pvalue = 0.429, HNRNPK\n",
      "Didn't pass (cutoff=0.0050): cat 6, mi = 0.001, pvalue = 0.389, CPEB4\n",
      "Didn't pass (cutoff=0.0050): cat 73, mi = 0.001, pvalue = 0.257, SNRNP70\n",
      "Didn't pass (cutoff=0.0050): cat 68, mi = 0.001, pvalue = 0.498, RBMS3\n",
      "Didn't pass (cutoff=0.0050): cat 16, mi = 0.001, pvalue = 0.093, HNRNPA1L2\n",
      "Didn't pass (cutoff=0.0050): cat 9, mi = 0.001, pvalue = 0.719, ESRP2\n",
      "Didn't pass (cutoff=0.0050): cat 62, mi = 0.001, pvalue = 0.767, RBM46\n",
      "Didn't pass (cutoff=0.0050): cat 13, mi = 0.001, pvalue = 0.806, FXR2\n",
      "Didn't pass (cutoff=0.0050): cat 56, mi = 0.001, pvalue = 0.844, RBM3\n",
      "Didn't pass: cat 29, mi = 0.000, pvalue = 1.000, HuR.4\n",
      "Didn't pass (cutoff=0.0050): cat 27, mi = 0.000, pvalue = 0.991, HuR.2\n",
      "Didn't pass: cat 0, mi = 0.000, pvalue = 1.000, ANKHD1\n",
      "Didn't pass: cat 1, mi = 0.000, pvalue = 1.000, BRUNOL4\n",
      "Didn't pass: cat 2, mi = 0.000, pvalue = 1.000, BRUNOL5\n",
      "Didn't pass: cat 3, mi = 0.000, pvalue = 1.000, BRUNOL6\n",
      "Didn't pass: cat 4, mi = 0.000, pvalue = 1.000, CNOT4\n",
      "Didn't pass: cat 5, mi = 0.000, pvalue = 1.000, CPEB2\n",
      "Total number of categories evaluated = 10.\n",
      "Number of categories that passed the tests = 0\n",
      "Number of clusters: 11\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "Done\n",
      "Writing the p-value matrix to file...Done\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "Reading matrix ... Done.\n",
      "Start drawing\n",
      "9.06\t-14.97\n",
      "Outputing EPS file deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.eps\n",
      "Convert to PDF deseq/d_mtyl_T_vs_U.c.txt_PAGE/d_mtyl_T_vs_U.c.txt.summary.pdf\n",
      "Finished.\n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl  --pvaluematrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt --expfile=deseq/d_mtyl_T_vs_U.c.txt --order=1 --draw_sample_heatmap=false --min=-3 --max=3 --cluster=5 --quantized=0 \n",
      "perl /flash/bin/iPAGEv1.0/SCRIPTS/list_killed_cats.pl --pvmatrixfile=deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt\n",
      "DONE\n",
      "renamed 'deseq/d_mtyl_T_vs_U.c.txt_PAGE/' -> 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_DeepBind/'\n",
      "'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_DeepBind/d_mtyl_T_vs_U.c.txt.summary.pdf' -> 'plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_DeepBind.pdf'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "command failed: /flash/bin/iPAGEv1.0/PROGRAMS/page -expfile deseq/d_mtyl_T_vs_U.c.txt -goindexfile /flash/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_all_gene_names/human_ensembl_RBPs_all_gene_names_index.txt -gonamesfile /flash/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl_RBPs_all_gene_names/human_ensembl_RBPs_all_gene_names_names.txt -catmaxcount 200 -catmincount 0 -logfile deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt.log  -quantized 0 -pvaluematrixfile deseq/d_mtyl_T_vs_U.c.txt_PAGE/pvmatrix.txt -max_p 0.005  -F 1  -P 1  -C 1  -independence 1  -ebins 11  at /flash/bin/iPAGEv1.0/page.pl line 300.\n",
      "cp: cannot stat 'ipage/d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_names/d_mtyl_T_vs_U.c.txt.summary.pdf': No such file or directory\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/over.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `deseq/d_mtyl_T_vs_U.c.txt_PAGE//imgsrc/under.png' @ warning/png.c/MagickPNGWarningHandler/1748.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n",
      "nothing to output for html page at /flash/bin/iPAGEv1.0/SCRIPTS/mi_go_draw_matrix_html.pl line 79.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "mkdir -p ipage\n",
    "awk 'NR>1{split($1,id,\".\"); print id[1]\"\\t\"$3}' deseq/d_mtyl_T_vs_U.txt > deseq/d_mtyl_T_vs_U.c.txt\n",
    "\n",
    "for f in `ls -d /flash/bin/iPAGEv1.0/PAGE_DATA/ANNOTATIONS/human_ensembl*`; do\n",
    "    base=`basename \"$f\"`\n",
    "    perl $PAGEDIR/page.pl --expfile=deseq/d_mtyl_T_vs_U.c.txt \\\n",
    "    --species=$base --exptype=continuous --ebins=11 --nodups=1\n",
    "    mv -v deseq/d_mtyl_T_vs_U.c.txt_PAGE/ ipage/d_mtyl_T_vs_U_${base}/\n",
    "    cp -v ipage/d_mtyl_T_vs_U_${base}/d_mtyl_T_vs_U.c.txt.summary.pdf plots/ipage.d_mtyl_T_vs_U_${base}.pdf\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "### encode_tf\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_encode_tf.png\" title=\"encode_tf\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_c1\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c1.png\" title=\"msigdb_c1\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_c2\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c2.png\" title=\"msigdb_c2\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_c3\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c3.png\" title=\"msigdb_c3\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_c4\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c4.png\" title=\"msigdb_c4\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_c5\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c5.png\" title=\"msigdb_c5\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_c6\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c6.png\" title=\"msigdb_c6\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_c7\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c7.png\" title=\"msigdb_c7\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_full\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_full.png\" title=\"msigdb_full\" style=\"width:600px\">\n",
      "\n",
      "### msigdb_h\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_h.png\" title=\"msigdb_h\" style=\"width:600px\">\n",
      "\n",
      "### plots/ipage.d_mtyl_T_vs_U_human_ensembl\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl.png\" title=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl\" style=\"width:600px\">\n",
      "\n",
      "### RBPs_all_gene_ids\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_ids.png\" title=\"RBPs_all_gene_ids\" style=\"width:600px\">\n",
      "\n",
      "### RBPs_coding_gene_ids_by_3UTR\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_3UTR.png\" title=\"RBPs_coding_gene_ids_by_3UTR\" style=\"width:600px\">\n",
      "\n",
      "### RBPs_coding_gene_ids_by_5UTR\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_5UTR.png\" title=\"RBPs_coding_gene_ids_by_5UTR\" style=\"width:600px\">\n",
      "\n",
      "### RBPs_coding_gene_ids_by_coding_exons\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_coding_exons.png\" title=\"RBPs_coding_gene_ids_by_coding_exons\" style=\"width:600px\">\n",
      "\n",
      "### RBPs_coding_gene_ids_by_introns\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_introns.png\" title=\"RBPs_coding_gene_ids_by_introns\" style=\"width:600px\">\n",
      "\n",
      "### RBPs_coding_gene_ids\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids.png\" title=\"RBPs_coding_gene_ids\" style=\"width:600px\">\n",
      "\n",
      "### RBPs_DeepBind\n",
      "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_DeepBind.png\" title=\"RBPs_DeepBind\" style=\"width:600px\">\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "for i in `ls -d plots/ipage*pdf`; do\n",
    "    o=${i/pdf/png};\n",
    "    n=${i/plots\\/ipage.d_mtyl_T_vs_U_human_ensembl_/};\n",
    "    n=${n/.pdf/};\n",
    "#     magick -density 300 $i -flatten -quality 90 $o\n",
    "    echo '### '$n\n",
    "    echo '<img src=\"'$o'\" title=\"'$n'\" style=\"width:600px\">'\n",
    "    echo ''\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### encode_tf\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_encode_tf.png\" title=\"encode_tf\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_c1\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c1.png\" title=\"msigdb_c1\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_c2\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c2.png\" title=\"msigdb_c2\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_c3\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c3.png\" title=\"msigdb_c3\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_c4\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c4.png\" title=\"msigdb_c4\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_c5\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c5.png\" title=\"msigdb_c5\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_c6\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c6.png\" title=\"msigdb_c6\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_c7\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_c7.png\" title=\"msigdb_c7\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_full\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_full.png\" title=\"msigdb_full\" style=\"width:600px\">\n",
    "\n",
    "### msigdb_h\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_msigdb_h.png\" title=\"msigdb_h\" style=\"width:600px\">\n",
    "\n",
    "### plots/ipage.d_mtyl_T_vs_U_human_ensembl\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl.png\" title=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl\" style=\"width:600px\">\n",
    "\n",
    "### RBPs_all_gene_ids\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_all_gene_ids.png\" title=\"RBPs_all_gene_ids\" style=\"width:600px\">\n",
    "\n",
    "### RBPs_coding_gene_ids_by_3UTR\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_3UTR.png\" title=\"RBPs_coding_gene_ids_by_3UTR\" style=\"width:600px\">\n",
    "\n",
    "### RBPs_coding_gene_ids_by_5UTR\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_5UTR.png\" title=\"RBPs_coding_gene_ids_by_5UTR\" style=\"width:600px\">\n",
    "\n",
    "### RBPs_coding_gene_ids_by_coding_exons\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_coding_exons.png\" title=\"RBPs_coding_gene_ids_by_coding_exons\" style=\"width:600px\">\n",
    "\n",
    "### RBPs_coding_gene_ids_by_introns\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids_by_introns.png\" title=\"RBPs_coding_gene_ids_by_introns\" style=\"width:600px\">\n",
    "\n",
    "### RBPs_coding_gene_ids\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_coding_gene_ids.png\" title=\"RBPs_coding_gene_ids\" style=\"width:600px\">\n",
    "\n",
    "### RBPs_DeepBind\n",
    "<img src=\"plots/ipage.d_mtyl_T_vs_U_human_ensembl_RBPs_DeepBind.png\" title=\"RBPs_DeepBind\" style=\"width:600px\">"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
