{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "using [RADAR](https://scottzijiezhang.github.io/RADARmanual/workflow.html) for analysis. \n",
    "\n",
    "# Pre-processing and QC \n",
    "<img src=\"radar/boxPlot.png\" style=\"width:600px\">\n",
    "\n",
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"radar/PCAPlot.png\" style=\"width:400px\">\n",
    "    <td><img src=\"radar/peakDistribution.png\" style=\"width:400px\">\n",
    "  <tr>\n",
    "<table>\n",
    "\n",
    "\n",
    "\n",
    "# Differential methylation\n",
    "Global analysis show significant hyper-methylation camparing treated vs. untreated. \n",
    "<table>\n",
    "  <tr>\n",
    "    <td><img src=\"radar/peak_Volcanos.png\" style=\"width:500px\">\n",
    "    <td><img src=\"radar/peak_Histograms.png\" style=\"width:500px\">\n",
    "  <tr>\n",
    "<table>\n",
    "\n",
    "\n",
    "### exp. ZNF93 peak\n",
    "<img src=\"radar/top_peak.png\" style=\"width:400px\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# iPAGE\n",
    "\n",
    "### c1\n",
    "<img src=\"radar/plots/ipage.d_mtyl_T_vs_U_human_c1_gs.png\" title=\"c1\" style=\"width:600px\">\n",
    "\n",
    "### c2\n",
    "<img src=\"radar/plots/ipage.d_mtyl_T_vs_U_human_c2_gs.png\" title=\"c2\" style=\"width:600px\">\n",
    "\n",
    "### c3\n",
    "<img src=\"radar/plots/ipage.d_mtyl_T_vs_U_human_c3_gs.png\" title=\"c3\" style=\"width:600px\">\n",
    "\n",
    "### c4\n",
    "<img src=\"radar/plots/ipage.d_mtyl_T_vs_U_human_c4_gs.png\" title=\"c4\" style=\"width:600px\">\n",
    "\n",
    "### c5\n",
    "<img src=\"radar/plots/ipage.d_mtyl_T_vs_U_human_c5_gs.png\" title=\"c5\" style=\"width:600px\">\n",
    "\n",
    "### c7\n",
    "<img src=\"radar/plots/ipage.d_mtyl_T_vs_U_human_c7_gs.png\" title=\"c7\" style=\"width:600px\">\n",
    "\n",
    "### gs_RBPs_by_3UTR\n",
    "<img src=\"radar/plots/ipage.d_mtyl_T_vs_U_human_gs_RBPs_by_3UTR.png\" title=\"gs_RBPs_by_3UTR\" style=\"width:600px\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "library(RADAR)\r\n",
      "setwd('~/People/Gilbertlab/Decitabine_treatment/MeRIP-seq_treated/')\r\n",
      "# load bam files \r\n",
      "radar <- countReads(samplenames=c('T1','T2','U1','U2'), gtf = \"/rumi/shams/genomes/hg38/hg38_genes.gtf\",bamFolder='./bam/',modification = \"m6A\", strandToKeep = \"opposite\",outputDir='radar', threads = 18)\r\n",
      "\r\n",
      "saveRDS(radar, 'radar/raw.rds')\r\n",
      "\r\n",
      "pdf('radar/boxPlot.pdf')\r\n",
      "radar <- normalizeLibrary(radar)\r\n",
      "dev.off()\r\n",
      "png('radar/boxPlot.png')\r\n",
      "radar <- normalizeLibrary(radar)\r\n",
      "dev.off()\r\n",
      "\r\n",
      "radar <- adjustExprLevel(radar)\r\n",
      "\r\n",
      "variable(radar) <- data.frame( Group = c(rep(\"T\",2),rep(\"U\",2)) )\r\n",
      "\r\n",
      "radar <- filterBins(radar,minCountsCutOff = 15)\r\n",
      "radar <- diffIP_parallel(radar, thread = 8)\r\n",
      "top_bins <- extractIP(radar,filtered = T)[order(rowMeans( extractIP(radar,filtered = T) ),decreasing = T)[1:1000],]\r\n",
      "\r\n",
      "png('radar/PCAPlot.png')\r\n",
      "plotPCAfromMatrix(top_bins,group = unlist(variable(radar)) )\r\n",
      "dev.off()\r\n",
      "pdf('radar/PCAPlot.pdf')\r\n",
      "plotPCAfromMatrix(top_bins,group = unlist(variable(radar)) )\r\n",
      "plotPCAfromMatrix(top_bins,group = unlist(variable(radar)) )\r\n",
      "dev.off()\r\n",
      "\r\n",
      "radar <- reportResult(radar, cutoff = 0.1, Beta_cutoff = 0.5)\r\n",
      "saveRDS(radar, 'radar/final.rds')\r\n",
      "\r\n",
      "result <- results(radar)\r\n",
      "\r\n",
      "pdf('radar/HeatmapPlot.pdf')\r\n",
      "plotHeatMap(radar)\r\n",
      "dev.off()\r\n",
      "png('radar/HeatmapPlot.png')\r\n",
      "plotHeatMap(radar)\r\n",
      "dev.off()\r\n",
      "\r\n",
      "png('radar/peakDistribution.png')\r\n",
      "peakDistribution(radar)\r\n",
      "dev.off()\r\n",
      "\r\n",
      "res = results(radar)\r\n",
      "\r\n",
      "w <- wilcox.test( res$logFC, mu=0, alternative = \"greater\")\r\n",
      "t <- t.test(      res$logFC, mu=0, alternative = \"greater\")\r\n",
      "h = ggplot(res, aes(x=logFC)) + \r\n",
      "    ggtitle('Decitabine_treatment',\r\n",
      "sprintf(\"wilcox.test (p.value): %.5f \\t t.test (p.value):%.5f \\t [mu=0,alter=greater]\", w$p.value, t$p.value)) +geom_histogram(binwidth=0.1)\r\n",
      "ggsave(\"radar/peak_Histograms.png\", plot = h, device = 'png', dpi = 300)\r\n",
      "ggsave(\"radar/peak_Histograms.pdf\", plot = h, device = 'pdf', dpi = 300)\r\n",
      "\r\n",
      "\r\n",
      "v = res %>% ggplot(aes(x=logFC, y=-log10(p_value),colour=sig, fill=sig)) +geom_point(aes(color = sig),alpha = 1/10) +ggtitle('Decitabine_treatment') +\r\n",
      "geom_hline(yintercept=-log10(0.05), linetype=\"dashed\", alpha = 4/10) +\r\n",
      "geom_vline(xintercept=-3, linetype=\"dashed\", alpha = 4/10) +\r\n",
      "geom_vline(xintercept= 3, linetype=\"dashed\", alpha = 4/10) +\r\n",
      "scale_color_manual(values = c(\"grey\", \"red\")) +\r\n",
      "theme_bw() + theme(legend.position=\"none\") +\r\n",
      "geom_text_repel(\r\n",
      "data = subset(res[order(res$p_value),], sig == TRUE)[1:10,],aes(label = name),size = 5,box.padding = unit(0.35, \"lines\"),point.padding = unit(0.3, \"lines\"))\r\n",
      "ggsave(\"radar/peak_Volcanos.png\", plot = v, device = 'png', dpi = 300)\r\n",
      "ggsave(\"radar/peak_Volcanos.pdf\", plot = v, device = 'pdf', dpi = 300)\r\n",
      "\r\n",
      "radar <- PrepCoveragePlot(radar)\r\n",
      "\r\n",
      "png('radar/top_peak.png')\r\n",
      "plotGeneCov(radar,geneName = \"ZNF93\", center = median, libraryType = \"opposite\")\r\n",
      "dev.off()\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "cat radar/code.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat radar/d_mtyl_T_vs_U.txt | awk 'NR>1{print $5\"\\t\"$14}' > radar/d_mtyl_T_vs_U.c.txt\n",
    "# select hypo and hyper-methylation gene set"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
