{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "66822d2d-5a57-4e64-9527-000003b81651",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "74fe2e96-9581-40cc-bf3a-4ca441c52468",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -o gene2name\n",
    "suppressMessages(suppressWarnings(library(tidyverse)))\n",
    "\n",
    "GTF = '~/genomes/hg38/gencode.v34/gencode.v34.annotation.gtf'\n",
    "\n",
    "gtf <- rtracklayer::import(GTF)\n",
    "gene2name <- gtf[gtf$type == \"gene\"] %>% data.frame %>% column_to_rownames('gene_id') %>% dplyr::select('gene_name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "17778012-8380-4872-82e4-b24725e1f93f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43d809ad-ed96-4f94-a34a-ddd61ccc653b",
   "metadata": {},
   "source": [
    "> ### event-level analysis\n",
    "you can also do an event-level analysis\n",
    "where you use the exon-level results\n",
    "but as genefile use ChIP-seq data intersected with ex1-ex2-ex3 event (basically upstrem and downstream exon included with the intron in the middle)\n",
    "\n",
    "> for exon-level analysus\n",
    "`JunctionSeqallGenes.results.txt.gz`\n",
    "has the exon coordinates `chr\tstart\tend\tstrand`\n",
    "also has the transcripts that exon junction is in transcripts\n",
    "you can go to gff flatfile and find exons upstream and downstream of the featyre at hand"
   ]
  },
  {
   "cell_type": "raw",
   "id": "38f90b3e-eb70-4277-8904-2927e11a39b8",
   "metadata": {},
   "source": [
    "`JunctionSeqsigGenes.genewiseResults.txt.gz`\n",
    "genes listed here go to class ‘1’\n",
    "and all the other expressed genes class ‘0’"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "8a70a121-8e0c-4698-9086-3bc8bdea0247",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_results_table(PATH):\n",
    "    '''event-level analysis\n",
    "    '''\n",
    "    table = pd.read_csv(PATH,sep='\\t').drop(\"featureID\",axis=1).dropna(subset=['expr_dox']).dropna(axis=1)\n",
    "    # .drop(\"featureID\",axis=1)\n",
    "    table = pd.concat([\n",
    "        gene2name.loc[table.geneID,:].reset_index(drop=True),\n",
    "        table.reset_index(drop=True)\n",
    "    ],axis=1).set_index([\"geneID\",\"gene_name\",\"countbinID\"])\n",
    "    \n",
    "    table=table[[i for i in table if len(set(table[i]))>1]]\n",
    "\n",
    "    return table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "dc6a00f9-5ded-49f3-9a97-3a5d2b1b9073",
   "metadata": {},
   "outputs": [],
   "source": [
    "sigtable = read_results_table('jscs/sigGenes.results.txt.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "451554ff-6e66-4fec-82cb-aaf3d624bd9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>baseVar</th>\n",
       "      <th>dispBeforeSharing</th>\n",
       "      <th>dispFitted</th>\n",
       "      <th>dispersion</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padjust</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>featureType</th>\n",
       "      <th>padjust_noFilter</th>\n",
       "      <th>log2FC(dox/no-dox)</th>\n",
       "      <th>log2FCvst(dox/no-dox)</th>\n",
       "      <th>expr_no-dox</th>\n",
       "      <th>expr_dox</th>\n",
       "      <th>geneWisePadj</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneID</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>countbinID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">ENSG00000116288.13</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">PARK7</th>\n",
       "      <th>E003</th>\n",
       "      <td>89.500235</td>\n",
       "      <td>403.511739</td>\n",
       "      <td>0.042550</td>\n",
       "      <td>0.012923</td>\n",
       "      <td>0.033333</td>\n",
       "      <td>0.788027</td>\n",
       "      <td>0.988488</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961691</td>\n",
       "      <td>7961700</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000460192.5;ENST00000493678.5</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.992016</td>\n",
       "      <td>-0.137950</td>\n",
       "      <td>-0.068974</td>\n",
       "      <td>95.017271</td>\n",
       "      <td>84.126449</td>\n",
       "      <td>0.003250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E004</th>\n",
       "      <td>107.180418</td>\n",
       "      <td>981.856850</td>\n",
       "      <td>0.056942</td>\n",
       "      <td>0.011358</td>\n",
       "      <td>0.042257</td>\n",
       "      <td>0.554762</td>\n",
       "      <td>0.961012</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961700</td>\n",
       "      <td>7961710</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000460192.5;ENST00000493678.5;ENST000003...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.980430</td>\n",
       "      <td>-0.303329</td>\n",
       "      <td>-0.151660</td>\n",
       "      <td>119.706081</td>\n",
       "      <td>95.102792</td>\n",
       "      <td>0.003250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E005</th>\n",
       "      <td>195.340298</td>\n",
       "      <td>925.603565</td>\n",
       "      <td>0.016472</td>\n",
       "      <td>0.007782</td>\n",
       "      <td>0.013927</td>\n",
       "      <td>0.285620</td>\n",
       "      <td>0.878576</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961710</td>\n",
       "      <td>7961735</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000460192.5;ENST00000493678.5;ENST000003...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.980430</td>\n",
       "      <td>-0.282505</td>\n",
       "      <td>-0.141245</td>\n",
       "      <td>215.251825</td>\n",
       "      <td>174.646838</td>\n",
       "      <td>0.003250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E006</th>\n",
       "      <td>95.757555</td>\n",
       "      <td>353.996437</td>\n",
       "      <td>0.013577</td>\n",
       "      <td>0.012303</td>\n",
       "      <td>0.013202</td>\n",
       "      <td>0.139371</td>\n",
       "      <td>0.762814</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961735</td>\n",
       "      <td>7961747</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000338639.10</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.908534</td>\n",
       "      <td>-0.411065</td>\n",
       "      <td>-0.205527</td>\n",
       "      <td>109.522330</td>\n",
       "      <td>80.879329</td>\n",
       "      <td>0.003250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E007</th>\n",
       "      <td>227.881779</td>\n",
       "      <td>1930.523043</td>\n",
       "      <td>0.000395</td>\n",
       "      <td>0.007162</td>\n",
       "      <td>0.002474</td>\n",
       "      <td>0.005040</td>\n",
       "      <td>0.245486</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961747</td>\n",
       "      <td>7961793</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000338639.10;ENST00000465354.5</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.387210</td>\n",
       "      <td>-0.394167</td>\n",
       "      <td>-0.197070</td>\n",
       "      <td>260.050465</td>\n",
       "      <td>197.861783</td>\n",
       "      <td>0.003250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">ENSG00000147140.16</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">NONO</th>\n",
       "      <th>J066</th>\n",
       "      <td>652.632118</td>\n",
       "      <td>4788.507214</td>\n",
       "      <td>0.000830</td>\n",
       "      <td>0.003741</td>\n",
       "      <td>0.001521</td>\n",
       "      <td>0.071776</td>\n",
       "      <td>0.642267</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71297047</td>\n",
       "      <td>71297376</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000413858.5;ENST00000535149.5;ENST000002...</td>\n",
       "      <td>splice_site</td>\n",
       "      <td>0.795161</td>\n",
       "      <td>0.177730</td>\n",
       "      <td>0.088860</td>\n",
       "      <td>594.295157</td>\n",
       "      <td>708.977230</td>\n",
       "      <td>0.006783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>J067</th>\n",
       "      <td>609.522401</td>\n",
       "      <td>2370.891868</td>\n",
       "      <td>0.001440</td>\n",
       "      <td>0.003823</td>\n",
       "      <td>0.001992</td>\n",
       "      <td>0.828729</td>\n",
       "      <td>0.991763</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71297461</td>\n",
       "      <td>71297835</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000535149.5;ENST00000276079.13;ENST00000...</td>\n",
       "      <td>splice_site</td>\n",
       "      <td>0.996275</td>\n",
       "      <td>0.022242</td>\n",
       "      <td>0.011121</td>\n",
       "      <td>583.488486</td>\n",
       "      <td>631.329039</td>\n",
       "      <td>0.006783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>J068</th>\n",
       "      <td>683.650367</td>\n",
       "      <td>1177.148546</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.003688</td>\n",
       "      <td>0.000873</td>\n",
       "      <td>0.300250</td>\n",
       "      <td>0.887474</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71297938</td>\n",
       "      <td>71298468</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000535149.5;ENST00000276079.13;ENST00000...</td>\n",
       "      <td>splice_site</td>\n",
       "      <td>0.980430</td>\n",
       "      <td>-0.090327</td>\n",
       "      <td>-0.045160</td>\n",
       "      <td>678.933108</td>\n",
       "      <td>684.415709</td>\n",
       "      <td>0.006783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>J069</th>\n",
       "      <td>727.299294</td>\n",
       "      <td>52.267422</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.003622</td>\n",
       "      <td>0.000920</td>\n",
       "      <td>0.339330</td>\n",
       "      <td>0.904853</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71298508</td>\n",
       "      <td>71298706</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000535149.5;ENST00000276079.13;ENST00000...</td>\n",
       "      <td>splice_site</td>\n",
       "      <td>0.980430</td>\n",
       "      <td>-0.081709</td>\n",
       "      <td>-0.040851</td>\n",
       "      <td>722.213260</td>\n",
       "      <td>732.664002</td>\n",
       "      <td>0.006783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>J070</th>\n",
       "      <td>682.227425</td>\n",
       "      <td>662.603772</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.003690</td>\n",
       "      <td>0.000815</td>\n",
       "      <td>0.948915</td>\n",
       "      <td>0.998149</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71298816</td>\n",
       "      <td>71299941</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000535149.5;ENST00000276079.13;ENST00000...</td>\n",
       "      <td>splice_site</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.007272</td>\n",
       "      <td>-0.003636</td>\n",
       "      <td>662.075834</td>\n",
       "      <td>703.282210</td>\n",
       "      <td>0.006783</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2022 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           baseMean      baseVar  \\\n",
       "geneID             gene_name countbinID                            \n",
       "ENSG00000116288.13 PARK7     E003         89.500235   403.511739   \n",
       "                             E004        107.180418   981.856850   \n",
       "                             E005        195.340298   925.603565   \n",
       "                             E006         95.757555   353.996437   \n",
       "                             E007        227.881779  1930.523043   \n",
       "...                                             ...          ...   \n",
       "ENSG00000147140.16 NONO      J066        652.632118  4788.507214   \n",
       "                             J067        609.522401  2370.891868   \n",
       "                             J068        683.650367  1177.148546   \n",
       "                             J069        727.299294    52.267422   \n",
       "                             J070        682.227425   662.603772   \n",
       "\n",
       "                                         dispBeforeSharing  dispFitted  \\\n",
       "geneID             gene_name countbinID                                  \n",
       "ENSG00000116288.13 PARK7     E003                 0.042550    0.012923   \n",
       "                             E004                 0.056942    0.011358   \n",
       "                             E005                 0.016472    0.007782   \n",
       "                             E006                 0.013577    0.012303   \n",
       "                             E007                 0.000395    0.007162   \n",
       "...                                                    ...         ...   \n",
       "ENSG00000147140.16 NONO      J066                 0.000830    0.003741   \n",
       "                             J067                 0.001440    0.003823   \n",
       "                             J068                 0.000024    0.003688   \n",
       "                             J069                 0.000052    0.003622   \n",
       "                             J070                 0.000014    0.003690   \n",
       "\n",
       "                                         dispersion    pvalue   padjust   chr  \\\n",
       "geneID             gene_name countbinID                                         \n",
       "ENSG00000116288.13 PARK7     E003          0.033333  0.788027  0.988488  chr1   \n",
       "                             E004          0.042257  0.554762  0.961012  chr1   \n",
       "                             E005          0.013927  0.285620  0.878576  chr1   \n",
       "                             E006          0.013202  0.139371  0.762814  chr1   \n",
       "                             E007          0.002474  0.005040  0.245486  chr1   \n",
       "...                                             ...       ...       ...   ...   \n",
       "ENSG00000147140.16 NONO      J066          0.001521  0.071776  0.642267  chrX   \n",
       "                             J067          0.001992  0.828729  0.991763  chrX   \n",
       "                             J068          0.000873  0.300250  0.887474  chrX   \n",
       "                             J069          0.000920  0.339330  0.904853  chrX   \n",
       "                             J070          0.000815  0.948915  0.998149  chrX   \n",
       "\n",
       "                                            start       end strand  \\\n",
       "geneID             gene_name countbinID                              \n",
       "ENSG00000116288.13 PARK7     E003         7961691   7961700      +   \n",
       "                             E004         7961700   7961710      +   \n",
       "                             E005         7961710   7961735      +   \n",
       "                             E006         7961735   7961747      +   \n",
       "                             E007         7961747   7961793      +   \n",
       "...                                           ...       ...    ...   \n",
       "ENSG00000147140.16 NONO      J066        71297047  71297376      +   \n",
       "                             J067        71297461  71297835      +   \n",
       "                             J068        71297938  71298468      +   \n",
       "                             J069        71298508  71298706      +   \n",
       "                             J070        71298816  71299941      +   \n",
       "\n",
       "                                                                               transcripts  \\\n",
       "geneID             gene_name countbinID                                                      \n",
       "ENSG00000116288.13 PARK7     E003                      ENST00000460192.5;ENST00000493678.5   \n",
       "                             E004        ENST00000460192.5;ENST00000493678.5;ENST000003...   \n",
       "                             E005        ENST00000460192.5;ENST00000493678.5;ENST000003...   \n",
       "                             E006                                       ENST00000338639.10   \n",
       "                             E007                     ENST00000338639.10;ENST00000465354.5   \n",
       "...                                                                                    ...   \n",
       "ENSG00000147140.16 NONO      J066        ENST00000413858.5;ENST00000535149.5;ENST000002...   \n",
       "                             J067        ENST00000535149.5;ENST00000276079.13;ENST00000...   \n",
       "                             J068        ENST00000535149.5;ENST00000276079.13;ENST00000...   \n",
       "                             J069        ENST00000535149.5;ENST00000276079.13;ENST00000...   \n",
       "                             J070        ENST00000535149.5;ENST00000276079.13;ENST00000...   \n",
       "\n",
       "                                         featureType  padjust_noFilter  \\\n",
       "geneID             gene_name countbinID                                  \n",
       "ENSG00000116288.13 PARK7     E003        exonic_part          0.992016   \n",
       "                             E004        exonic_part          0.980430   \n",
       "                             E005        exonic_part          0.980430   \n",
       "                             E006        exonic_part          0.908534   \n",
       "                             E007        exonic_part          0.387210   \n",
       "...                                              ...               ...   \n",
       "ENSG00000147140.16 NONO      J066        splice_site          0.795161   \n",
       "                             J067        splice_site          0.996275   \n",
       "                             J068        splice_site          0.980430   \n",
       "                             J069        splice_site          0.980430   \n",
       "                             J070        splice_site          1.000000   \n",
       "\n",
       "                                         log2FC(dox/no-dox)  \\\n",
       "geneID             gene_name countbinID                       \n",
       "ENSG00000116288.13 PARK7     E003                 -0.137950   \n",
       "                             E004                 -0.303329   \n",
       "                             E005                 -0.282505   \n",
       "                             E006                 -0.411065   \n",
       "                             E007                 -0.394167   \n",
       "...                                                     ...   \n",
       "ENSG00000147140.16 NONO      J066                  0.177730   \n",
       "                             J067                  0.022242   \n",
       "                             J068                 -0.090327   \n",
       "                             J069                 -0.081709   \n",
       "                             J070                 -0.007272   \n",
       "\n",
       "                                         log2FCvst(dox/no-dox)  expr_no-dox  \\\n",
       "geneID             gene_name countbinID                                       \n",
       "ENSG00000116288.13 PARK7     E003                    -0.068974    95.017271   \n",
       "                             E004                    -0.151660   119.706081   \n",
       "                             E005                    -0.141245   215.251825   \n",
       "                             E006                    -0.205527   109.522330   \n",
       "                             E007                    -0.197070   260.050465   \n",
       "...                                                        ...          ...   \n",
       "ENSG00000147140.16 NONO      J066                     0.088860   594.295157   \n",
       "                             J067                     0.011121   583.488486   \n",
       "                             J068                    -0.045160   678.933108   \n",
       "                             J069                    -0.040851   722.213260   \n",
       "                             J070                    -0.003636   662.075834   \n",
       "\n",
       "                                           expr_dox  geneWisePadj  \n",
       "geneID             gene_name countbinID                            \n",
       "ENSG00000116288.13 PARK7     E003         84.126449      0.003250  \n",
       "                             E004         95.102792      0.003250  \n",
       "                             E005        174.646838      0.003250  \n",
       "                             E006         80.879329      0.003250  \n",
       "                             E007        197.861783      0.003250  \n",
       "...                                             ...           ...  \n",
       "ENSG00000147140.16 NONO      J066        708.977230      0.006783  \n",
       "                             J067        631.329039      0.006783  \n",
       "                             J068        684.415709      0.006783  \n",
       "                             J069        732.664002      0.006783  \n",
       "                             J070        703.282210      0.006783  \n",
       "\n",
       "[2022 rows x 19 columns]"
      ]
     },
     "execution_count": 269,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sigtable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "c09b2c55-55d5-424b-b7b1-40b9b040c1b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "71"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(sigtable.index.to_frame().geneID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "id": "4d488a23-8630-4f32-9904-a4730b9ef775",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>baseVar</th>\n",
       "      <th>dispBeforeSharing</th>\n",
       "      <th>dispFitted</th>\n",
       "      <th>dispersion</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padjust</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>featureType</th>\n",
       "      <th>padjust_noFilter</th>\n",
       "      <th>log2FC(dox/no-dox)</th>\n",
       "      <th>log2FCvst(dox/no-dox)</th>\n",
       "      <th>expr_no-dox</th>\n",
       "      <th>expr_dox</th>\n",
       "      <th>geneWisePadj</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneID</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>countbinID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">ENSG00000116288.13</th>\n",
       "      <th rowspan=\"5\" valign=\"top\">PARK7</th>\n",
       "      <th>E003</th>\n",
       "      <td>89.500235</td>\n",
       "      <td>403.511739</td>\n",
       "      <td>0.042550</td>\n",
       "      <td>0.012923</td>\n",
       "      <td>0.033333</td>\n",
       "      <td>0.788027</td>\n",
       "      <td>0.988488</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961691</td>\n",
       "      <td>7961700</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000460192.5;ENST00000493678.5</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.992016</td>\n",
       "      <td>-0.137950</td>\n",
       "      <td>-0.068974</td>\n",
       "      <td>95.017271</td>\n",
       "      <td>84.126449</td>\n",
       "      <td>0.00325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E004</th>\n",
       "      <td>107.180418</td>\n",
       "      <td>981.856850</td>\n",
       "      <td>0.056942</td>\n",
       "      <td>0.011358</td>\n",
       "      <td>0.042257</td>\n",
       "      <td>0.554762</td>\n",
       "      <td>0.961012</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961700</td>\n",
       "      <td>7961710</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000460192.5;ENST00000493678.5;ENST000003...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.980430</td>\n",
       "      <td>-0.303329</td>\n",
       "      <td>-0.151660</td>\n",
       "      <td>119.706081</td>\n",
       "      <td>95.102792</td>\n",
       "      <td>0.00325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E005</th>\n",
       "      <td>195.340298</td>\n",
       "      <td>925.603565</td>\n",
       "      <td>0.016472</td>\n",
       "      <td>0.007782</td>\n",
       "      <td>0.013927</td>\n",
       "      <td>0.285620</td>\n",
       "      <td>0.878576</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961710</td>\n",
       "      <td>7961735</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000460192.5;ENST00000493678.5;ENST000003...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.980430</td>\n",
       "      <td>-0.282505</td>\n",
       "      <td>-0.141245</td>\n",
       "      <td>215.251825</td>\n",
       "      <td>174.646838</td>\n",
       "      <td>0.00325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E006</th>\n",
       "      <td>95.757555</td>\n",
       "      <td>353.996437</td>\n",
       "      <td>0.013577</td>\n",
       "      <td>0.012303</td>\n",
       "      <td>0.013202</td>\n",
       "      <td>0.139371</td>\n",
       "      <td>0.762814</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961735</td>\n",
       "      <td>7961747</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000338639.10</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.908534</td>\n",
       "      <td>-0.411065</td>\n",
       "      <td>-0.205527</td>\n",
       "      <td>109.522330</td>\n",
       "      <td>80.879329</td>\n",
       "      <td>0.00325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E007</th>\n",
       "      <td>227.881779</td>\n",
       "      <td>1930.523043</td>\n",
       "      <td>0.000395</td>\n",
       "      <td>0.007162</td>\n",
       "      <td>0.002474</td>\n",
       "      <td>0.005040</td>\n",
       "      <td>0.245486</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961747</td>\n",
       "      <td>7961793</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000338639.10;ENST00000465354.5</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.387210</td>\n",
       "      <td>-0.394167</td>\n",
       "      <td>-0.197070</td>\n",
       "      <td>260.050465</td>\n",
       "      <td>197.861783</td>\n",
       "      <td>0.00325</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           baseMean      baseVar  \\\n",
       "geneID             gene_name countbinID                            \n",
       "ENSG00000116288.13 PARK7     E003         89.500235   403.511739   \n",
       "                             E004        107.180418   981.856850   \n",
       "                             E005        195.340298   925.603565   \n",
       "                             E006         95.757555   353.996437   \n",
       "                             E007        227.881779  1930.523043   \n",
       "\n",
       "                                         dispBeforeSharing  dispFitted  \\\n",
       "geneID             gene_name countbinID                                  \n",
       "ENSG00000116288.13 PARK7     E003                 0.042550    0.012923   \n",
       "                             E004                 0.056942    0.011358   \n",
       "                             E005                 0.016472    0.007782   \n",
       "                             E006                 0.013577    0.012303   \n",
       "                             E007                 0.000395    0.007162   \n",
       "\n",
       "                                         dispersion    pvalue   padjust   chr  \\\n",
       "geneID             gene_name countbinID                                         \n",
       "ENSG00000116288.13 PARK7     E003          0.033333  0.788027  0.988488  chr1   \n",
       "                             E004          0.042257  0.554762  0.961012  chr1   \n",
       "                             E005          0.013927  0.285620  0.878576  chr1   \n",
       "                             E006          0.013202  0.139371  0.762814  chr1   \n",
       "                             E007          0.002474  0.005040  0.245486  chr1   \n",
       "\n",
       "                                           start      end strand  \\\n",
       "geneID             gene_name countbinID                            \n",
       "ENSG00000116288.13 PARK7     E003        7961691  7961700      +   \n",
       "                             E004        7961700  7961710      +   \n",
       "                             E005        7961710  7961735      +   \n",
       "                             E006        7961735  7961747      +   \n",
       "                             E007        7961747  7961793      +   \n",
       "\n",
       "                                                                               transcripts  \\\n",
       "geneID             gene_name countbinID                                                      \n",
       "ENSG00000116288.13 PARK7     E003                      ENST00000460192.5;ENST00000493678.5   \n",
       "                             E004        ENST00000460192.5;ENST00000493678.5;ENST000003...   \n",
       "                             E005        ENST00000460192.5;ENST00000493678.5;ENST000003...   \n",
       "                             E006                                       ENST00000338639.10   \n",
       "                             E007                     ENST00000338639.10;ENST00000465354.5   \n",
       "\n",
       "                                         featureType  padjust_noFilter  \\\n",
       "geneID             gene_name countbinID                                  \n",
       "ENSG00000116288.13 PARK7     E003        exonic_part          0.992016   \n",
       "                             E004        exonic_part          0.980430   \n",
       "                             E005        exonic_part          0.980430   \n",
       "                             E006        exonic_part          0.908534   \n",
       "                             E007        exonic_part          0.387210   \n",
       "\n",
       "                                         log2FC(dox/no-dox)  \\\n",
       "geneID             gene_name countbinID                       \n",
       "ENSG00000116288.13 PARK7     E003                 -0.137950   \n",
       "                             E004                 -0.303329   \n",
       "                             E005                 -0.282505   \n",
       "                             E006                 -0.411065   \n",
       "                             E007                 -0.394167   \n",
       "\n",
       "                                         log2FCvst(dox/no-dox)  expr_no-dox  \\\n",
       "geneID             gene_name countbinID                                       \n",
       "ENSG00000116288.13 PARK7     E003                    -0.068974    95.017271   \n",
       "                             E004                    -0.151660   119.706081   \n",
       "                             E005                    -0.141245   215.251825   \n",
       "                             E006                    -0.205527   109.522330   \n",
       "                             E007                    -0.197070   260.050465   \n",
       "\n",
       "                                           expr_dox  geneWisePadj  \n",
       "geneID             gene_name countbinID                            \n",
       "ENSG00000116288.13 PARK7     E003         84.126449       0.00325  \n",
       "                             E004         95.102792       0.00325  \n",
       "                             E005        174.646838       0.00325  \n",
       "                             E006         80.879329       0.00325  \n",
       "                             E007        197.861783       0.00325  "
      ]
     },
     "execution_count": 270,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sigtable.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "69a4a82f-6596-4891-8aba-ffeb75ccf773",
   "metadata": {},
   "outputs": [],
   "source": [
    "alltable = read_results_table('jscs/allGenes.results.txt.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "ff3ccbf6-89bb-47f9-b787-1bfdf04cbd7f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8093"
      ]
     },
     "execution_count": 271,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(alltable.index.to_frame().geneID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "a7461b8b-91f0-4b7d-9744-0f302574418c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>baseVar</th>\n",
       "      <th>dispBeforeSharing</th>\n",
       "      <th>dispFitted</th>\n",
       "      <th>dispersion</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padjust</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcripts</th>\n",
       "      <th>featureType</th>\n",
       "      <th>padjust_noFilter</th>\n",
       "      <th>log2FC(dox/no-dox)</th>\n",
       "      <th>log2FCvst(dox/no-dox)</th>\n",
       "      <th>expr_no-dox</th>\n",
       "      <th>expr_dox</th>\n",
       "      <th>geneWisePadj</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneID</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>countbinID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">ENSG00000187634.12</th>\n",
       "      <th rowspan=\"4\" valign=\"top\">SAMD11</th>\n",
       "      <th>E036</th>\n",
       "      <td>93.110387</td>\n",
       "      <td>293.519556</td>\n",
       "      <td>2.222378e-03</td>\n",
       "      <td>0.012555</td>\n",
       "      <td>0.005819</td>\n",
       "      <td>0.094784</td>\n",
       "      <td>0.692075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>943907</td>\n",
       "      <td>944259</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000618779.4;ENST00000618323.4;ENST000006...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.840749</td>\n",
       "      <td>0.363475</td>\n",
       "      <td>0.181719</td>\n",
       "      <td>90.830636</td>\n",
       "      <td>94.467958</td>\n",
       "      <td>0.539425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E037</th>\n",
       "      <td>539.210683</td>\n",
       "      <td>5896.530559</td>\n",
       "      <td>8.370035e-03</td>\n",
       "      <td>0.005010</td>\n",
       "      <td>0.006941</td>\n",
       "      <td>0.034466</td>\n",
       "      <td>0.510059</td>\n",
       "      <td>chr1</td>\n",
       "      <td>944259</td>\n",
       "      <td>944574</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000618779.4;ENST00000618323.4;ENST000006...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.670369</td>\n",
       "      <td>0.500392</td>\n",
       "      <td>0.249254</td>\n",
       "      <td>573.202964</td>\n",
       "      <td>504.113400</td>\n",
       "      <td>0.539425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E038</th>\n",
       "      <td>351.463314</td>\n",
       "      <td>4337.431386</td>\n",
       "      <td>1.000001e-08</td>\n",
       "      <td>0.005852</td>\n",
       "      <td>0.005711</td>\n",
       "      <td>0.525253</td>\n",
       "      <td>0.955419</td>\n",
       "      <td>chr1</td>\n",
       "      <td>944574</td>\n",
       "      <td>944575</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000618779.4;ENST00000618323.4;ENST000006...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.980430</td>\n",
       "      <td>-0.134919</td>\n",
       "      <td>-0.067408</td>\n",
       "      <td>389.264320</td>\n",
       "      <td>313.465896</td>\n",
       "      <td>0.539425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E039</th>\n",
       "      <td>430.266075</td>\n",
       "      <td>7322.845950</td>\n",
       "      <td>4.485993e-03</td>\n",
       "      <td>0.005409</td>\n",
       "      <td>0.004843</td>\n",
       "      <td>0.008196</td>\n",
       "      <td>0.295338</td>\n",
       "      <td>chr1</td>\n",
       "      <td>944575</td>\n",
       "      <td>944581</td>\n",
       "      <td>+</td>\n",
       "      <td>ENST00000618779.4;ENST00000618323.4;ENST000006...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.447949</td>\n",
       "      <td>-0.504995</td>\n",
       "      <td>-0.252169</td>\n",
       "      <td>491.363974</td>\n",
       "      <td>369.123318</td>\n",
       "      <td>0.539425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000188976.11</th>\n",
       "      <th>NOC2L</th>\n",
       "      <th>E003</th>\n",
       "      <td>1005.274236</td>\n",
       "      <td>5695.877938</td>\n",
       "      <td>8.708665e-04</td>\n",
       "      <td>0.004280</td>\n",
       "      <td>0.001551</td>\n",
       "      <td>0.224609</td>\n",
       "      <td>0.839599</td>\n",
       "      <td>chr1</td>\n",
       "      <td>944204</td>\n",
       "      <td>944800</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000327044.7;ENST00000483767.5;ENST000004...</td>\n",
       "      <td>exonic_part</td>\n",
       "      <td>0.980430</td>\n",
       "      <td>0.117735</td>\n",
       "      <td>0.058857</td>\n",
       "      <td>1068.801971</td>\n",
       "      <td>940.612731</td>\n",
       "      <td>0.202226</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            baseMean      baseVar  \\\n",
       "geneID             gene_name countbinID                             \n",
       "ENSG00000187634.12 SAMD11    E036          93.110387   293.519556   \n",
       "                             E037         539.210683  5896.530559   \n",
       "                             E038         351.463314  4337.431386   \n",
       "                             E039         430.266075  7322.845950   \n",
       "ENSG00000188976.11 NOC2L     E003        1005.274236  5695.877938   \n",
       "\n",
       "                                         dispBeforeSharing  dispFitted  \\\n",
       "geneID             gene_name countbinID                                  \n",
       "ENSG00000187634.12 SAMD11    E036             2.222378e-03    0.012555   \n",
       "                             E037             8.370035e-03    0.005010   \n",
       "                             E038             1.000001e-08    0.005852   \n",
       "                             E039             4.485993e-03    0.005409   \n",
       "ENSG00000188976.11 NOC2L     E003             8.708665e-04    0.004280   \n",
       "\n",
       "                                         dispersion    pvalue   padjust   chr  \\\n",
       "geneID             gene_name countbinID                                         \n",
       "ENSG00000187634.12 SAMD11    E036          0.005819  0.094784  0.692075  chr1   \n",
       "                             E037          0.006941  0.034466  0.510059  chr1   \n",
       "                             E038          0.005711  0.525253  0.955419  chr1   \n",
       "                             E039          0.004843  0.008196  0.295338  chr1   \n",
       "ENSG00000188976.11 NOC2L     E003          0.001551  0.224609  0.839599  chr1   \n",
       "\n",
       "                                          start     end strand  \\\n",
       "geneID             gene_name countbinID                          \n",
       "ENSG00000187634.12 SAMD11    E036        943907  944259      +   \n",
       "                             E037        944259  944574      +   \n",
       "                             E038        944574  944575      +   \n",
       "                             E039        944575  944581      +   \n",
       "ENSG00000188976.11 NOC2L     E003        944204  944800      -   \n",
       "\n",
       "                                                                               transcripts  \\\n",
       "geneID             gene_name countbinID                                                      \n",
       "ENSG00000187634.12 SAMD11    E036        ENST00000618779.4;ENST00000618323.4;ENST000006...   \n",
       "                             E037        ENST00000618779.4;ENST00000618323.4;ENST000006...   \n",
       "                             E038        ENST00000618779.4;ENST00000618323.4;ENST000006...   \n",
       "                             E039        ENST00000618779.4;ENST00000618323.4;ENST000006...   \n",
       "ENSG00000188976.11 NOC2L     E003        ENST00000327044.7;ENST00000483767.5;ENST000004...   \n",
       "\n",
       "                                         featureType  padjust_noFilter  \\\n",
       "geneID             gene_name countbinID                                  \n",
       "ENSG00000187634.12 SAMD11    E036        exonic_part          0.840749   \n",
       "                             E037        exonic_part          0.670369   \n",
       "                             E038        exonic_part          0.980430   \n",
       "                             E039        exonic_part          0.447949   \n",
       "ENSG00000188976.11 NOC2L     E003        exonic_part          0.980430   \n",
       "\n",
       "                                         log2FC(dox/no-dox)  \\\n",
       "geneID             gene_name countbinID                       \n",
       "ENSG00000187634.12 SAMD11    E036                  0.363475   \n",
       "                             E037                  0.500392   \n",
       "                             E038                 -0.134919   \n",
       "                             E039                 -0.504995   \n",
       "ENSG00000188976.11 NOC2L     E003                  0.117735   \n",
       "\n",
       "                                         log2FCvst(dox/no-dox)  expr_no-dox  \\\n",
       "geneID             gene_name countbinID                                       \n",
       "ENSG00000187634.12 SAMD11    E036                     0.181719    90.830636   \n",
       "                             E037                     0.249254   573.202964   \n",
       "                             E038                    -0.067408   389.264320   \n",
       "                             E039                    -0.252169   491.363974   \n",
       "ENSG00000188976.11 NOC2L     E003                     0.058857  1068.801971   \n",
       "\n",
       "                                           expr_dox  geneWisePadj  \n",
       "geneID             gene_name countbinID                            \n",
       "ENSG00000187634.12 SAMD11    E036         94.467958      0.539425  \n",
       "                             E037        504.113400      0.539425  \n",
       "                             E038        313.465896      0.539425  \n",
       "                             E039        369.123318      0.539425  \n",
       "ENSG00000188976.11 NOC2L     E003        940.612731      0.202226  "
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alltable.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a321f6dd-41f5-45d3-80be-0a062d42d8e1",
   "metadata": {},
   "source": [
    "___\n",
    "> ### gives a gene-level file as well, you can use that. \n",
    "to run teiser\n",
    "with ChIP-seq data intersected with genes as genefile\n",
    "\n",
    "> this is the gene based file\n",
    "`JunctionSeqsigGenes.genewiseResults.txt.gz`\n",
    "genes listed here go to class ‘1’\n",
    "and all the other expressed genes class ‘0’\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "f20a2b9b-9401-4954-96c4-8a4973b7ffa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_genewiseResults_table(PATH):\n",
    "    '''event-level analysis\n",
    "    '''\n",
    "    table = pd.read_csv(PATH,sep='\\t')#.drop(\"featureID\",axis=1).dropna(subset=['expr_dox']).dropna(axis=1)\n",
    "    table = table.set_index([\"geneID\",\"geneName\"])\n",
    "\n",
    "    return table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "3a6eab6d-1f88-460e-ac4d-20037a3e82e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "genewise_table = read_genewiseResults_table(\"jscs/sigGenes.genewiseResults.txt.gz\")\n",
    "genewise_genes = set(genewise_table.index.get_level_values(1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "c61c9288-3097-4621-9c10-cc39c6400523",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>geneWisePadj</th>\n",
       "      <th>mostSigID</th>\n",
       "      <th>mostSigPadjust</th>\n",
       "      <th>numExons</th>\n",
       "      <th>numKnown</th>\n",
       "      <th>numNovel</th>\n",
       "      <th>exonsSig</th>\n",
       "      <th>knownSig</th>\n",
       "      <th>novelSig</th>\n",
       "      <th>numFeatures</th>\n",
       "      <th>numSig</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneID</th>\n",
       "      <th>geneName</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000116288.13</th>\n",
       "      <th>PARK7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>7954290</td>\n",
       "      <td>7985505</td>\n",
       "      <td>+</td>\n",
       "      <td>2286.2</td>\n",
       "      <td>0.003250</td>\n",
       "      <td>E011</td>\n",
       "      <td>0.001890</td>\n",
       "      <td>28</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28/15/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000179051.14</th>\n",
       "      <th>RCC2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17406759</td>\n",
       "      <td>17439677</td>\n",
       "      <td>-</td>\n",
       "      <td>4742.0</td>\n",
       "      <td>0.014966</td>\n",
       "      <td>J032</td>\n",
       "      <td>0.008920</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>19/13/0</td>\n",
       "      <td>0/1/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000142949.17</th>\n",
       "      <th>PTPRF</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43525186</td>\n",
       "      <td>43623666</td>\n",
       "      <td>+</td>\n",
       "      <td>4391.4</td>\n",
       "      <td>0.003330</td>\n",
       "      <td>E007</td>\n",
       "      <td>0.001930</td>\n",
       "      <td>69</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>69/47/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000117450.14</th>\n",
       "      <th>PRDX1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>45511035</td>\n",
       "      <td>45523047</td>\n",
       "      <td>-</td>\n",
       "      <td>8454.0</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>J019</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>15</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>15/9/0</td>\n",
       "      <td>0/1/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000135829.17</th>\n",
       "      <th>DHX9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>182839346</td>\n",
       "      <td>182887982</td>\n",
       "      <td>+</td>\n",
       "      <td>4657.1</td>\n",
       "      <td>0.014732</td>\n",
       "      <td>E003</td>\n",
       "      <td>0.008740</td>\n",
       "      <td>44</td>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>44/32/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000168453.15</th>\n",
       "      <th>HR</th>\n",
       "      <td>chr8</td>\n",
       "      <td>22114418</td>\n",
       "      <td>22133384</td>\n",
       "      <td>-</td>\n",
       "      <td>1094.6</td>\n",
       "      <td>0.011513</td>\n",
       "      <td>E033</td>\n",
       "      <td>0.006760</td>\n",
       "      <td>35</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35/22/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000147471.12</th>\n",
       "      <th>PLPBP</th>\n",
       "      <td>chr8</td>\n",
       "      <td>37762594</td>\n",
       "      <td>37779768</td>\n",
       "      <td>+</td>\n",
       "      <td>1159.0</td>\n",
       "      <td>0.013891</td>\n",
       "      <td>J027</td>\n",
       "      <td>0.008160</td>\n",
       "      <td>25</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>25/10/0</td>\n",
       "      <td>0/1/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000136942.15</th>\n",
       "      <th>RPL35</th>\n",
       "      <td>chr9</td>\n",
       "      <td>124857879</td>\n",
       "      <td>124861981</td>\n",
       "      <td>-</td>\n",
       "      <td>8494.2</td>\n",
       "      <td>0.000084</td>\n",
       "      <td>J021</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>15/6/0</td>\n",
       "      <td>2/1/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000130741.11</th>\n",
       "      <th>EIF2S3</th>\n",
       "      <td>chrX</td>\n",
       "      <td>24054945</td>\n",
       "      <td>24078810</td>\n",
       "      <td>+</td>\n",
       "      <td>3875.0</td>\n",
       "      <td>0.000309</td>\n",
       "      <td>E004</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>22/13/0</td>\n",
       "      <td>2/2/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000147140.16</th>\n",
       "      <th>NONO</th>\n",
       "      <td>chrX</td>\n",
       "      <td>71283191</td>\n",
       "      <td>71301168</td>\n",
       "      <td>+</td>\n",
       "      <td>6959.9</td>\n",
       "      <td>0.006783</td>\n",
       "      <td>E008</td>\n",
       "      <td>0.003940</td>\n",
       "      <td>46</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46/24/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              chr      start        end strand  baseMean  \\\n",
       "geneID             geneName                                                \n",
       "ENSG00000116288.13 PARK7     chr1    7954290    7985505      +    2286.2   \n",
       "ENSG00000179051.14 RCC2      chr1   17406759   17439677      -    4742.0   \n",
       "ENSG00000142949.17 PTPRF     chr1   43525186   43623666      +    4391.4   \n",
       "ENSG00000117450.14 PRDX1     chr1   45511035   45523047      -    8454.0   \n",
       "ENSG00000135829.17 DHX9      chr1  182839346  182887982      +    4657.1   \n",
       "...                           ...        ...        ...    ...       ...   \n",
       "ENSG00000168453.15 HR        chr8   22114418   22133384      -    1094.6   \n",
       "ENSG00000147471.12 PLPBP     chr8   37762594   37779768      +    1159.0   \n",
       "ENSG00000136942.15 RPL35     chr9  124857879  124861981      -    8494.2   \n",
       "ENSG00000130741.11 EIF2S3    chrX   24054945   24078810      +    3875.0   \n",
       "ENSG00000147140.16 NONO      chrX   71283191   71301168      +    6959.9   \n",
       "\n",
       "                             geneWisePadj mostSigID  mostSigPadjust  numExons  \\\n",
       "geneID             geneName                                                     \n",
       "ENSG00000116288.13 PARK7         0.003250      E011        0.001890        28   \n",
       "ENSG00000179051.14 RCC2          0.014966      J032        0.008920        19   \n",
       "ENSG00000142949.17 PTPRF         0.003330      E007        0.001930        69   \n",
       "ENSG00000117450.14 PRDX1         0.000073      J019        0.000046        15   \n",
       "ENSG00000135829.17 DHX9          0.014732      E003        0.008740        44   \n",
       "...                                   ...       ...             ...       ...   \n",
       "ENSG00000168453.15 HR            0.011513      E033        0.006760        35   \n",
       "ENSG00000147471.12 PLPBP         0.013891      J027        0.008160        25   \n",
       "ENSG00000136942.15 RPL35         0.000084      J021        0.000053        15   \n",
       "ENSG00000130741.11 EIF2S3        0.000309      E004        0.000196        22   \n",
       "ENSG00000147140.16 NONO          0.006783      E008        0.003940        46   \n",
       "\n",
       "                             numKnown  numNovel  exonsSig  knownSig  novelSig  \\\n",
       "geneID             geneName                                                     \n",
       "ENSG00000116288.13 PARK7           15         0         1         0         0   \n",
       "ENSG00000179051.14 RCC2            13         0         0         1         0   \n",
       "ENSG00000142949.17 PTPRF           47         0         1         0         0   \n",
       "ENSG00000117450.14 PRDX1            9         0         0         1         0   \n",
       "ENSG00000135829.17 DHX9            32         0         1         0         0   \n",
       "...                               ...       ...       ...       ...       ...   \n",
       "ENSG00000168453.15 HR              22         0         1         0         0   \n",
       "ENSG00000147471.12 PLPBP           10         0         0         1         0   \n",
       "ENSG00000136942.15 RPL35            6         0         2         1         0   \n",
       "ENSG00000130741.11 EIF2S3          13         0         2         2         0   \n",
       "ENSG00000147140.16 NONO            24         0         1         0         0   \n",
       "\n",
       "                            numFeatures numSig  \n",
       "geneID             geneName                     \n",
       "ENSG00000116288.13 PARK7        28/15/0  1/0/0  \n",
       "ENSG00000179051.14 RCC2         19/13/0  0/1/0  \n",
       "ENSG00000142949.17 PTPRF        69/47/0  1/0/0  \n",
       "ENSG00000117450.14 PRDX1         15/9/0  0/1/0  \n",
       "ENSG00000135829.17 DHX9         44/32/0  1/0/0  \n",
       "...                                 ...    ...  \n",
       "ENSG00000168453.15 HR           35/22/0  1/0/0  \n",
       "ENSG00000147471.12 PLPBP        25/10/0  0/1/0  \n",
       "ENSG00000136942.15 RPL35         15/6/0  2/1/0  \n",
       "ENSG00000130741.11 EIF2S3       22/13/0  2/2/0  \n",
       "ENSG00000147140.16 NONO         46/24/0  1/0/0  \n",
       "\n",
       "[71 rows x 16 columns]"
      ]
     },
     "execution_count": 258,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genewise_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "65cef25c-8041-48c7-9e89-0d0689556267",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_expression_table(PATH):\n",
    "    '''\n",
    "    '''\n",
    "    table = pd.read_csv(PATH,sep='\\t').dropna(subset=['expr_dox']).dropna(axis=1)\n",
    "    table[\"geneID\"] = table.featureID.str.split(':').str[0]\n",
    "    table[\"countbinID\"] = table.featureID.str.split(':').str[1]\n",
    "    \n",
    "    table = pd.concat([\n",
    "        gene2name.loc[table.geneID,:].reset_index(drop=True),\n",
    "        table.drop(\"featureID\",axis=1).reset_index(drop=True)\n",
    "    ],axis=1).set_index([\"geneID\",\"gene_name\",\"countbinID\"])\n",
    "    \n",
    "    return table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "4def3d55-d86b-47d6-b83d-bbf44ec57352",
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_table = read_expression_table('jscs/allGenes.expression.data.txt.gz')\n",
    "\n",
    "exp_table.index.get_level_values(1)\n",
    "exp_genes = set(exp_table.index.get_level_values(1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "ff4104a9-f50c-4a2a-8888-4c43acba8dd0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8093"
      ]
     },
     "execution_count": 260,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(exp_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "4bb190ec-0543-4eca-a661-3c040c2c2035",
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_binary_table = pd.DataFrame(\n",
    "    {\"spliced\":np.zeros(len(exp_genes))},\n",
    "    index=exp_genes, dtype=int\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "56e217e2-6728-4471-9792-bfb6dc183b6c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spliced</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TRAPPC12</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC6</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SEL1L3</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PDXK</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H1FX-AS1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC39A10</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SERP1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BET1L</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAPGEF1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FBXO28</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8093 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          spliced\n",
       "TRAPPC12        0\n",
       "CCDC6           0\n",
       "SEL1L3          0\n",
       "PDXK            0\n",
       "H1FX-AS1        0\n",
       "...           ...\n",
       "SLC39A10        0\n",
       "SERP1           0\n",
       "BET1L           0\n",
       "RAPGEF1         0\n",
       "FBXO28          0\n",
       "\n",
       "[8093 rows x 1 columns]"
      ]
     },
     "execution_count": 261,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "splice_binary_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "a1c8c08f-5a8f-492a-9fe1-9fb554d244ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_binary_table.loc[genewise_genes,:] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "40ced31e-c95c-4031-96e4-5149eacf87ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_binary_table.to_csv(\"splice_binary_table.txt\",sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "265df1ee-e64f-4b57-8d43-3591c3f3d4d4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "de4a3c5f-1675-42ce-bac5-dba239ff5bda",
   "metadata": {},
   "source": [
    "## ChIP-seq-data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "da0d9a09-cdf3-452b-a592-1cfda467ebd1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GFP_1_DOX_pos_IDX1.rep1_sorted_peaks.narrowPeak.bed\n",
      "GFP_2_DOX_pos_IDX2.rep1_sorted_peaks.narrowPeak.bed\n",
      "s303_1_DOX_neg_IDX13.rep1_sorted_peaks.narrowPeak.bed\n",
      "s303_1_DOX_pos_IDX7.rep1_sorted_peaks.narrowPeak.bed\n",
      "s303_2_DOX_neg_IDX14.rep1_sorted_peaks.narrowPeak.bed\n",
      "s303_2_DOX_pos_IDX8.rep1_sorted_peaks.narrowPeak.bed\n",
      "sox9_1_DOX_neg_IDX11.rep1_sorted_peaks.narrowPeak.bed\n",
      "SOX9_1_DOX_pos_IDX3.rep1_sorted_peaks.narrowPeak.bed\n",
      "sox9_2_DOX_neg_IDX12.rep1_sorted_peaks.narrowPeak.bed\n",
      "SOX9_2_DOX_pos_IDX4.rep1_sorted_peaks.narrowPeak.bed\n"
     ]
    }
   ],
   "source": [
    "ls ChIP-seq-data/bed_files/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "063972ca-ce54-4834-a687-ac6225474d1a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "================================================================\n",
      "ChIP-seq-data/bed_files/GFP_1_DOX_pos.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/GFP_2_DOX_pos.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/s303_1_DOX_neg.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/s303_1_DOX_pos.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/s303_2_DOX_neg.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/s303_2_DOX_pos.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/sox9_1_DOX_neg.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/SOX9_1_DOX_pos.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/sox9_2_DOX_neg.txt\n",
      "================================================================\n",
      "ChIP-seq-data/bed_files/SOX9_2_DOX_pos.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "GTF=~/genomes/hg38/gencode.v34/gencode.v34.annotation.gtf\n",
    "\n",
    "for f in ChIP-seq-data/bed_files/*.bed; do \n",
    "    o=${f/_IDX*/.txt}\n",
    "\n",
    "    echo \"================================================================\"\n",
    "    echo $o\n",
    "\n",
    "    cat $GTF \\\n",
    "    | awk -v FS='\\t' '$3==\"gene\"' \\\n",
    "    | bedtools intersect -wa -wb -a - -b $f \\\n",
    "    | cut -f 9,18 | uniq | awk -v FS='\\t' '{print $1,$2}' \\\n",
    "    | awk -v FS=\";\" -v OFS=\"\" '{gsub(\"\\\"\",\"\"); print $3}' \\\n",
    "    | awk -v FS=' ' -v OFS=\"\" '{print $2}' > $o\n",
    "\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "de6807b3-9fa9-425c-b7da-99d69f66a586",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/SOX9_1_DOX_pos.txt \\\n",
    "ChIP-seq-data/bed_files/SOX9_2_DOX_pos.txt \\\n",
    "| uniq > SOX9_DOX_pos.txt\n",
    "# ChIP-seq-data/bed_files/sox9_1_DOX_neg.txt \\\n",
    "# ChIP-seq-data/bed_files/sox9_2_DOX_neg.txt \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "id": "a2b2c200-a328-46c2-9110-50aec580ae5a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "POMK\n",
      "ZNF37BP\n",
      "LINC00662\n"
     ]
    }
   ],
   "source": [
    "cat ChIP-seq-data/bed_files/sox9_1_DOX_neg.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "6e19c672-3da1-4275-8b9e-15ae88f1a669",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cat SOX9_DOX_pos.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "id": "aea4253e-d803-4970-848c-bb1df2681d8f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Thu Jan 20 20:20:24 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 8093 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000616\n",
      "pass = 0.037600\n",
      "z = 3.717622\n",
      "freq 0: 0.047744\t383\t8022\t383\t8093\n",
      "freq 1: 0.000000\t0\t71\t383\t8093\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Option suffix requires an argument\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "export TEISERDIR='/data_gilbert/home/aarab/Workflows/TEISERv1.1'\n",
    "exp=\"splice_binary_table.txt\"\n",
    "\n",
    "    gene=\"SOX9_DOX_pos.txt\"\n",
    "\n",
    "    perl ${TEISERDIR}/run_mi_gene_list.pl \\\n",
    "        --expfile=$exp \\\n",
    "        --genefile=$gene \\\n",
    "        --exptype=discrete \\\n",
    "        --species=human \\\n",
    "        --doremovedups=0 \\\n",
    "        --doremoveextra=0 \n",
    "        # --ebins=11 \\\n",
    "\n",
    "    perl /data_gilbert/home/aarab/Workflows/TEISERv1.1/Scripts/teiser_draw_matrix.pl \\\n",
    "        --pvmatrixfile=splice_binary_table.txt_GENESET/splice_binary_table.txt.matrix \\\n",
    "        --summaryfile=splice_binary_table.txt_GENESET/splice_binary_table.txt.summary \\\n",
    "        --expfile=splice_binary_table.txt_GENESET/splice_binary_table.txt \\\n",
    "        --quantized=1 \\\n",
    "        --colmap=/data_gilbert/home/aarab/Workflows/TEISERv1.1/Scripts/HEATMAPS/cmap_1.txt --order=0 --min=-3 --max=3 --cluster=5 \n",
    "\n",
    "    out=${gene/.txt/}\n",
    "    mv ${exp}_GENESET $out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "9e019978-fb5c-4f0a-bdc0-768aac601236",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data_gilbert/home/aarab/iPAGE/run_mi_gene_list.pl\n"
     ]
    }
   ],
   "source": [
    "ls /data_gilbert/home/aarab/iPAGE/run_mi_gene_list.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "2b02c252-9211-4914-ab9f-07c569754ebe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "!echo ${TEISERDIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "afaa6fb6-1a66-4ef6-9517-b4060816ae5f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Jan 13 19:49:49 PST 2022\n"
     ]
    }
   ],
   "source": [
    "!date"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:gviz]",
   "language": "python",
   "name": "conda-env-gviz-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
