{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "66822d2d-5a57-4e64-9527-000003b81651",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "74fe2e96-9581-40cc-bf3a-4ca441c52468",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -o gene2name\n",
    "suppressMessages(suppressWarnings(library(tidyverse)))\n",
    "\n",
    "GTF = '~/genomes/hg38/gencode.v34/gencode.v34.annotation.gtf'\n",
    "\n",
    "gtf <- rtracklayer::import(GTF)\n",
    "gene2name <- gtf[gtf$type == \"gene\"] %>% data.frame %>% column_to_rownames('gene_id') %>% dplyr::select('gene_name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "17778012-8380-4872-82e4-b24725e1f93f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "70eaaec1-0d22-47ee-b549-abdb36293c54",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p enrichment/\n",
    "!mkdir -p enrichment/exon_level\n",
    "!mkdir -p enrichment/gene_level"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43d809ad-ed96-4f94-a34a-ddd61ccc653b",
   "metadata": {},
   "source": [
    "## event-level analysis\n",
    "> you can also do an event-level analysis\n",
    "where you use the exon-level results\n",
    "but as genefile use ChIP-seq data intersected with ex1-ex2-ex3 event (basically upstrem and downstream exon included with the intron in the middle)\n",
    "\n",
    "> for exon-level analysus\n",
    "`JunctionSeqallGenes.results.txt.gz`\n",
    "has the exon coordinates `chr\tstart\tend\tstrand`\n",
    "also has the transcripts that exon junction is in transcripts\n",
    "you can go to gff flatfile and find exons upstream and downstream of the featyre at hand"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f703df0-34f7-4cd4-b50c-4893e8930006",
   "metadata": {},
   "source": [
    "___\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "937f6715-28e9-46af-97bb-509a0398bdc6",
   "metadata": {},
   "source": [
    "__1) ex1-ex2-ex3 event__\n",
    "\n",
    "(basically upstrem and downstream exon included with the intron in the middle)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "36dd59da-e29b-432c-abc2-73b5c640873a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aggregate_gene\n",
      "exonic_part\n",
      "splice_site\n"
     ]
    }
   ],
   "source": [
    "!zcat annoFiles/JunctionSeq.flat.gff.gz | awk '{print $3}' | sort | uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "8d17b0f9-c71a-4fc7-a944-2de247df68bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "zcat annoFiles/JunctionSeq.flat.gff.gz \\\n",
    "| grep \"num 00[1-3];\" \\\n",
    "| awk -v FS='[\\t;]' -v OFS='\\t' '{print $1,$2,$3,$4,$5,$6,$7,$8,$10}' \\\n",
    "| sort -k9,9 \\\n",
    "| groupBy -g 1,9 -c 9,4,5,6,7,8 -o count,min,max,first,first,first \\\n",
    "| awk 'gsub(\" tx_set ENST\",\"ENST\")' \\\n",
    "| awk -v FS='\\t' -v OFS='\\t' '{print $1,$4,$5,$6,$7,$8,$2,$3}' \\\n",
    "| sort -k2,2 -k3,3 -n | sort -k1,1 -s \\\n",
    "> annoFiles/JunctionSeq.flat.exon1to3.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "44facfe0-f86c-4ca3-bacd-a08413a19fc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !cat ~/genomes/hg38/gencode.v34/gencode.v34.annotation.gtf | head "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "75aae9f9-5eed-4ea3-9ecd-b59b81c22f7a",
   "metadata": {},
   "source": [
    "__2) Process event level `JunctionSeq` results__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8a70a121-8e0c-4698-9086-3bc8bdea0247",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_results_table(PATH):\n",
    "    '''event-level analysis\n",
    "    '''\n",
    "    table = pd.read_csv(PATH,sep='\\t').drop(\"featureID\",axis=1).dropna(subset=['expr_case']).dropna(axis=1)\n",
    "    # .drop(\"featureID\",axis=1)\n",
    "    table = pd.concat([\n",
    "        gene2name.loc[table.geneID,:].reset_index(drop=True),\n",
    "        table.reset_index(drop=True)\n",
    "    ],axis=1).set_index([\"geneID\",\"gene_name\",\"countbinID\"])\n",
    "    \n",
    "    table=table[[i for i in table if len(set(table[i]))>1]]\n",
    "\n",
    "    return table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "dc6a00f9-5ded-49f3-9a97-3a5d2b1b9073",
   "metadata": {},
   "outputs": [],
   "source": [
    "sigtable = read_results_table('jscs/sigGenes.results.txt.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "451554ff-6e66-4fec-82cb-aaf3d624bd9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'exonic_part', 'splice_site'}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(sigtable.featureType)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c09b2c55-55d5-424b-b7b1-40b9b040c1b2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(sigtable.index.to_frame().geneID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "4d488a23-8630-4f32-9904-a4730b9ef775",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(1, 'featureID'),\n",
       " (2, 'geneID'),\n",
       " (3, 'countbinID'),\n",
       " (4, 'testable'),\n",
       " (5, 'status'),\n",
       " (6, 'allZero'),\n",
       " (7, 'baseMean'),\n",
       " (8, 'baseVar'),\n",
       " (9, 'dispBeforeSharing'),\n",
       " (10, 'dispFitted'),\n",
       " (11, 'dispersion'),\n",
       " (12, 'pvalue'),\n",
       " (13, 'padjust'),\n",
       " (14, 'chr'),\n",
       " (15, 'start'),\n",
       " (16, 'end'),\n",
       " (17, 'strand'),\n",
       " (18, 'transcripts'),\n",
       " (19, 'featureType'),\n",
       " (20, 'padjust_noFilter'),\n",
       " (21, 'log2FC(case/ctrl)'),\n",
       " (22, 'log2FCvst(case/ctrl)'),\n",
       " (23, 'expr_ctrl'),\n",
       " (24, 'expr_case'),\n",
       " (25, 'geneWisePadj')]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[(i+1,col) for i,col in enumerate(pd.read_csv('jscs/sigGenes.results.txt.gz',sep='\\t').columns)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "4f3f8431-9061-4091-9f40-d0180cb89e14",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "JunctionSeq.flat.exon1to3.bed  \u001b[0m\u001b[38;5;9mJunctionSeq.flat.gff.gz\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls annoFiles/JunctionSeq.flat.exon1to3.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "e34dc61a-6ec3-4dec-b971-ac77acb3b6e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "zcat jscs/sigGenes.results.txt.gz \\\n",
    "| awk -v FS='\\t' -v OFS='\\t' 'NR>1 {print $14,$15,$16,$17,$18}' \\\n",
    "| bedtools intersect -wa -c -a annoFiles/JunctionSeq.flat.exon1to3.bed -b - \\\n",
    "| sort -k9,9rn \\\n",
    "| awk -v FS='\\t' -v OFS='\\t' '{print $7,$($9==0)*$9+($9>=1)*1}' | sort -u -k1,1 \\\n",
    "> jscs/sigGenes.results.exon1to3.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1a13eb5-100e-43d7-90db-f152086b8fbe",
   "metadata": {},
   "source": [
    "### ChIP-seq data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "c686fa08-a381-41f3-9060-78db88842111",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GFP_1_DOX_pos_IDX1.rep1_sorted_peaks.narrowPeak.bed\n",
      "GFP_2_DOX_pos_IDX2.rep1_sorted_peaks.narrowPeak.bed\n",
      "s303_1_DOX_neg_IDX13.rep1_sorted_peaks.narrowPeak.bed\n",
      "s303_1_DOX_pos_IDX7.rep1_sorted_peaks.narrowPeak.bed\n",
      "s303_2_DOX_neg_IDX14.rep1_sorted_peaks.narrowPeak.bed\n",
      "s303_2_DOX_pos_IDX8.rep1_sorted_peaks.narrowPeak.bed\n",
      "sox9_1_DOX_neg_IDX11.rep1_sorted_peaks.narrowPeak.bed\n",
      "SOX9_1_DOX_pos_IDX3.rep1_sorted_peaks.narrowPeak.bed\n",
      "sox9_2_DOX_neg_IDX12.rep1_sorted_peaks.narrowPeak.bed\n",
      "SOX9_2_DOX_pos_IDX4.rep1_sorted_peaks.narrowPeak.bed\n"
     ]
    }
   ],
   "source": [
    "ls ChIP-seq-data/bed_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "id": "c204a10c-04ad-40e4-b191-d97c09fc3273",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/GFP_1_DOX_pos*.bed \\\n",
    "ChIP-seq-data/bed_files/GFP_2_DOX_pos*.bed \\\n",
    "| bedtools intersect -wa -c -a annoFiles/JunctionSeq.flat.exon1to3.bed -b - \\\n",
    "| awk '$9 >= 1{print $7}' | sort -u -k1,1 \\\n",
    "> enrichment/exon_level/GFP_DOX_pos.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "id": "9ab31603-73b2-46eb-b88a-7732567fb29b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash \n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/SOX9_1_DOX_pos*.bed \\\n",
    "ChIP-seq-data/bed_files/SOX9_2_DOX_pos*.bed \\\n",
    "| bedtools intersect -wa -c -a annoFiles/JunctionSeq.flat.exon1to3.bed -b - \\\n",
    "| awk '$9 >= 1{print $7}' | sort -u -k1,1 \\\n",
    "> enrichment/exon_level/SOX9_DOX_pos.txt\n",
    "\n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/sox9_1_DOX_neg*.bed \\\n",
    "ChIP-seq-data/bed_files/sox9_2_DOX_neg*.bed \\\n",
    "| bedtools intersect -wa -c -a annoFiles/JunctionSeq.flat.exon1to3.bed -b - \\\n",
    "| awk '$9 >= 1{print $7}' | sort -u -k1,1 \\\n",
    "> enrichment/exon_level/SOX9_DOX_neg.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "id": "cca1ca8b-3a10-4288-bc4f-a16509eb00c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/s303_1_DOX_neg*.bed \\\n",
    "ChIP-seq-data/bed_files/s303_2_DOX_neg*.bed \\\n",
    "| bedtools intersect -wa -c -a annoFiles/JunctionSeq.flat.exon1to3.bed -b - \\\n",
    "| awk '$9 >= 1{print $7}' | sort -u -k1,1  \\\n",
    "> enrichment/exon_level/S303_DOX_neg.txt\n",
    "\n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/s303_1_DOX_pos*.bed \\\n",
    "ChIP-seq-data/bed_files/s303_2_DOX_pos*.bed \\\n",
    "| bedtools intersect -wa -c -a annoFiles/JunctionSeq.flat.exon1to3.bed -b - \\\n",
    "| awk '$9 >= 1{print $7}' | sort -u -k1,1 \\\n",
    "> enrichment/exon_level/S303_DOX_pos.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "702b91a0-e03d-48e3-b949-b1af03e03ea9",
   "metadata": {},
   "source": [
    "`--exp` file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "d193b952-5a28-4206-87a6-97c16d611d8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "echo -e \"#event\\tspliced\" > enrichment/exon_level/splice_binary_table.txt\n",
    "cat jscs/sigGenes.results.exon1to3.txt >> enrichment/exon_level/splice_binary_table.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "69a4a82f-6596-4891-8aba-ffeb75ccf773",
   "metadata": {},
   "outputs": [],
   "source": [
    "# alltable = read_results_table('jscs/allGenes.results.txt.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ff3ccbf6-89bb-47f9-b787-1bfdf04cbd7f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# len(set(alltable.index.to_frame().geneID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a7461b8b-91f0-4b7d-9744-0f302574418c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# alltable.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "1837752b-e2d7-4995-9df2-90c8ab9e0f16",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 31 17:25:14 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 93265 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000000\n",
      "pass = 1.000000\n",
      "z = -0.103496\n",
      "freq 0: 0.000097\t9\t93144\t9\t93265\n",
      "freq 1: 0.000000\t0\t121\t9\t93265\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 31 17:25:26 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 93265 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000000\n",
      "pass = 1.000000\n",
      "z = -0.061769\n",
      "freq 0: 0.000032\t3\t93144\t3\t93265\n",
      "freq 1: 0.000000\t0\t121\t3\t93265\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 31 17:25:39 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 93265 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000005\n",
      "pass = 1.000000\n",
      "z = -0.235050\n",
      "freq 0: 0.002931\t273\t93144\t273\t93265\n",
      "freq 1: 0.000000\t0\t121\t273\t93265\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 31 17:25:51 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 93265 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000000\n",
      "pass = 1.000000\n",
      "z = -0.050065\n",
      "freq 0: 0.000021\t2\t93144\t2\t93265\n",
      "freq 1: 0.000000\t0\t121\t2\t93265\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 31 17:26:03 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 93265 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000003\n",
      "pass = 1.000000\n",
      "z = -0.351818\n",
      "freq 0: 0.001439\t134\t93144\t134\t93265\n",
      "freq 1: 0.000000\t0\t121\t134\t93265\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "export TEISERDIR='/data_gilbert/home/aarab/Workflows/TEISERv1.1'\n",
    "exp=\"splice_binary_table.txt\"\n",
    "\n",
    "cd enrichment/exon_level/\n",
    "for gene in *DOX*.txt; do\n",
    "\n",
    "    perl ${TEISERDIR}/run_mi_gene_list.pl \\\n",
    "        --expfile=$exp \\\n",
    "        --genefile=$gene \\\n",
    "        --exptype=discrete \\\n",
    "        --species=human \\\n",
    "        --doremovedups=0 \\\n",
    "        --doremoveextra=0 \n",
    "        # --ebins=11 \\\n",
    "\n",
    "    perl /data_gilbert/home/aarab/Workflows/TEISERv1.1/Scripts/teiser_draw_matrix.pl \\\n",
    "        --pvmatrixfile=splice_binary_table.txt_GENESET/splice_binary_table.txt.matrix \\\n",
    "        --summaryfile=splice_binary_table.txt_GENESET/splice_binary_table.txt.summary \\\n",
    "        --expfile=splice_binary_table.txt_GENESET/splice_binary_table.txt \\\n",
    "        --quantized=1 \\\n",
    "        --colmap=/data_gilbert/home/aarab/Workflows/TEISERv1.1/Scripts/HEATMAPS/cmap_1.txt --order=0 --min=-3 --max=3 --cluster=5 \n",
    "\n",
    "    out=${gene/.txt/}\n",
    "    rm -rf $out\n",
    "    mv ${exp}_GENESET $out\n",
    "    cp $out/${exp}.summary.pdf ${out}.pdf\n",
    "done\n",
    "cd ../../"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a321f6dd-41f5-45d3-80be-0a062d42d8e1",
   "metadata": {
    "tags": []
   },
   "source": [
    "___\n",
    "## gene-level analysis\n",
    "> to run teiser\n",
    "with ChIP-seq data intersected with genes as genefile\n",
    "\n",
    "> this is the gene based file\n",
    "`JunctionSeqsigGenes.genewiseResults.txt.gz`\n",
    "genes listed here go to class ‘1’\n",
    "and all the other expressed genes class ‘0’\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f20a2b9b-9401-4954-96c4-8a4973b7ffa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_genewiseResults_table(PATH):\n",
    "    '''event-level analysis\n",
    "    '''\n",
    "    table = pd.read_csv(PATH,sep='\\t')#.drop(\"featureID\",axis=1).dropna(subset=['expr_dox']).dropna(axis=1)\n",
    "    table = table.set_index([\"geneID\",\"geneName\"])\n",
    "\n",
    "    return table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "3a6eab6d-1f88-460e-ac4d-20037a3e82e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "genewise_table = read_genewiseResults_table(\"jscs/sigGenes.genewiseResults.txt.gz\")\n",
    "genewise_genes = set(genewise_table.index.get_level_values(1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c61c9288-3097-4621-9c10-cc39c6400523",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>baseMean</th>\n",
       "      <th>geneWisePadj</th>\n",
       "      <th>mostSigID</th>\n",
       "      <th>mostSigPadjust</th>\n",
       "      <th>numExons</th>\n",
       "      <th>numKnown</th>\n",
       "      <th>numNovel</th>\n",
       "      <th>exonsSig</th>\n",
       "      <th>knownSig</th>\n",
       "      <th>novelSig</th>\n",
       "      <th>numFeatures</th>\n",
       "      <th>numSig</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>geneID</th>\n",
       "      <th>geneName</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000179051.14</th>\n",
       "      <th>RCC2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17406759</td>\n",
       "      <td>17439677</td>\n",
       "      <td>-</td>\n",
       "      <td>4226.7</td>\n",
       "      <td>9.841564e-04</td>\n",
       "      <td>J032</td>\n",
       "      <td>5.370000e-04</td>\n",
       "      <td>19</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>19/13/0</td>\n",
       "      <td>1/1/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000053372.5</th>\n",
       "      <th>MRTO4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19251804</td>\n",
       "      <td>19260128</td>\n",
       "      <td>+</td>\n",
       "      <td>1654.2</td>\n",
       "      <td>5.966044e-03</td>\n",
       "      <td>E001</td>\n",
       "      <td>3.610000e-03</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13/8/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000163399.16</th>\n",
       "      <th>ATP1A1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>116372667</td>\n",
       "      <td>116410261</td>\n",
       "      <td>+</td>\n",
       "      <td>15877.3</td>\n",
       "      <td>1.311720e-02</td>\n",
       "      <td>E002</td>\n",
       "      <td>8.750000e-03</td>\n",
       "      <td>40</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>40/28/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000158716.9</th>\n",
       "      <th>DUSP23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>159780931</td>\n",
       "      <td>159782543</td>\n",
       "      <td>+</td>\n",
       "      <td>544.5</td>\n",
       "      <td>8.004604e-03</td>\n",
       "      <td>E006</td>\n",
       "      <td>5.340000e-03</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7/3/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000116750.13</th>\n",
       "      <th>UCHL5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>193012249</td>\n",
       "      <td>193060080</td>\n",
       "      <td>-</td>\n",
       "      <td>1049.6</td>\n",
       "      <td>8.947062e-03</td>\n",
       "      <td>E027</td>\n",
       "      <td>6.060000e-03</td>\n",
       "      <td>37</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>37/20/0</td>\n",
       "      <td>2/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000175348.11</th>\n",
       "      <th>TMEM9B</th>\n",
       "      <td>chr11</td>\n",
       "      <td>8947201</td>\n",
       "      <td>8965011</td>\n",
       "      <td>-</td>\n",
       "      <td>404.2</td>\n",
       "      <td>1.324392e-02</td>\n",
       "      <td>E007</td>\n",
       "      <td>9.050000e-03</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14/8/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000149136.9</th>\n",
       "      <th>SSRP1</th>\n",
       "      <td>chr11</td>\n",
       "      <td>57325985</td>\n",
       "      <td>57335892</td>\n",
       "      <td>-</td>\n",
       "      <td>5121.7</td>\n",
       "      <td>1.330073e-02</td>\n",
       "      <td>E032</td>\n",
       "      <td>8.980000e-03</td>\n",
       "      <td>33</td>\n",
       "      <td>20</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>33/20/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000167995.17</th>\n",
       "      <th>BEST1</th>\n",
       "      <td>chr11</td>\n",
       "      <td>61950062</td>\n",
       "      <td>61965515</td>\n",
       "      <td>+</td>\n",
       "      <td>49388.8</td>\n",
       "      <td>8.909568e-03</td>\n",
       "      <td>E030</td>\n",
       "      <td>6.060000e-03</td>\n",
       "      <td>30</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30/15/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000089597.18</th>\n",
       "      <th>GANAB</th>\n",
       "      <td>chr11</td>\n",
       "      <td>62624825</td>\n",
       "      <td>62646726</td>\n",
       "      <td>-</td>\n",
       "      <td>7773.9</td>\n",
       "      <td>4.982960e-03</td>\n",
       "      <td>E048</td>\n",
       "      <td>2.930000e-03</td>\n",
       "      <td>59</td>\n",
       "      <td>33</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>59/33/0</td>\n",
       "      <td>2/1/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000165526.9</th>\n",
       "      <th>RPUSD4</th>\n",
       "      <td>chr11</td>\n",
       "      <td>126202095</td>\n",
       "      <td>126211692</td>\n",
       "      <td>-</td>\n",
       "      <td>774.5</td>\n",
       "      <td>1.052282e-02</td>\n",
       "      <td>E008</td>\n",
       "      <td>7.020000e-03</td>\n",
       "      <td>30</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30/12/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000110955.9</th>\n",
       "      <th>ATP5F1B</th>\n",
       "      <td>chr12</td>\n",
       "      <td>56638174</td>\n",
       "      <td>56645984</td>\n",
       "      <td>-</td>\n",
       "      <td>16800.9</td>\n",
       "      <td>3.383969e-03</td>\n",
       "      <td>E034</td>\n",
       "      <td>1.800000e-03</td>\n",
       "      <td>34</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>34/18/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000100614.18</th>\n",
       "      <th>PPM1A</th>\n",
       "      <td>chr14</td>\n",
       "      <td>60245751</td>\n",
       "      <td>60299087</td>\n",
       "      <td>+</td>\n",
       "      <td>1052.1</td>\n",
       "      <td>4.766264e-03</td>\n",
       "      <td>E026</td>\n",
       "      <td>2.680000e-03</td>\n",
       "      <td>26</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>26/18/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000259768.6</th>\n",
       "      <th>AC004943.2</th>\n",
       "      <td>chr16</td>\n",
       "      <td>72665122</td>\n",
       "      <td>72822781</td>\n",
       "      <td>+</td>\n",
       "      <td>212.3</td>\n",
       "      <td>1.373038e-02</td>\n",
       "      <td>E009</td>\n",
       "      <td>8.750000e-03</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11/11/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000131149.19</th>\n",
       "      <th>GSE1</th>\n",
       "      <td>chr16</td>\n",
       "      <td>85169524</td>\n",
       "      <td>85676204</td>\n",
       "      <td>+</td>\n",
       "      <td>1888.3</td>\n",
       "      <td>9.742367e-03</td>\n",
       "      <td>E013</td>\n",
       "      <td>6.740000e-03</td>\n",
       "      <td>45</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>45/27/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000108298.11</th>\n",
       "      <th>RPL19</th>\n",
       "      <td>chr17</td>\n",
       "      <td>39200282</td>\n",
       "      <td>39204732</td>\n",
       "      <td>+</td>\n",
       "      <td>12380.4</td>\n",
       "      <td>1.035171e-02</td>\n",
       "      <td>E001</td>\n",
       "      <td>7.020000e-03</td>\n",
       "      <td>19</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19/8/0</td>\n",
       "      <td>2/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000131469.15</th>\n",
       "      <th>RPL27</th>\n",
       "      <td>chr17</td>\n",
       "      <td>42998272</td>\n",
       "      <td>43002959</td>\n",
       "      <td>+</td>\n",
       "      <td>5808.2</td>\n",
       "      <td>5.558194e-03</td>\n",
       "      <td>E002</td>\n",
       "      <td>3.610000e-03</td>\n",
       "      <td>20</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>20/9/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000125398.8</th>\n",
       "      <th>SOX9</th>\n",
       "      <td>chr17</td>\n",
       "      <td>72121019</td>\n",
       "      <td>72126416</td>\n",
       "      <td>+</td>\n",
       "      <td>5438.2</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>E002</td>\n",
       "      <td>1.460000e-24</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3/2/0</td>\n",
       "      <td>3/2/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000177885.15</th>\n",
       "      <th>GRB2</th>\n",
       "      <td>chr17</td>\n",
       "      <td>75318075</td>\n",
       "      <td>75405709</td>\n",
       "      <td>-</td>\n",
       "      <td>2198.5</td>\n",
       "      <td>1.397060e-02</td>\n",
       "      <td>E025</td>\n",
       "      <td>8.750000e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28/14/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000141551.14</th>\n",
       "      <th>CSNK1D</th>\n",
       "      <td>chr17</td>\n",
       "      <td>82239022</td>\n",
       "      <td>82273731</td>\n",
       "      <td>-</td>\n",
       "      <td>1921.0</td>\n",
       "      <td>1.179758e-02</td>\n",
       "      <td>E001</td>\n",
       "      <td>7.990000e-03</td>\n",
       "      <td>70</td>\n",
       "      <td>30</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>70/30/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115268.9</th>\n",
       "      <th>RPS15</th>\n",
       "      <td>chr19</td>\n",
       "      <td>1438357</td>\n",
       "      <td>1440494</td>\n",
       "      <td>+</td>\n",
       "      <td>6819.7</td>\n",
       "      <td>5.101297e-03</td>\n",
       "      <td>E010</td>\n",
       "      <td>3.120000e-03</td>\n",
       "      <td>30</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>30/13/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000169021.6</th>\n",
       "      <th>UQCRFS1</th>\n",
       "      <td>chr19</td>\n",
       "      <td>29205319</td>\n",
       "      <td>29213151</td>\n",
       "      <td>-</td>\n",
       "      <td>2029.3</td>\n",
       "      <td>7.833223e-04</td>\n",
       "      <td>J003</td>\n",
       "      <td>3.920000e-04</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2/1/0</td>\n",
       "      <td>0/1/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000144485.11</th>\n",
       "      <th>HES6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238238266</td>\n",
       "      <td>238240662</td>\n",
       "      <td>-</td>\n",
       "      <td>2325.4</td>\n",
       "      <td>1.312724e-02</td>\n",
       "      <td>E015</td>\n",
       "      <td>8.490000e-03</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24/8/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000174748.22</th>\n",
       "      <th>RPL15</th>\n",
       "      <td>chr3</td>\n",
       "      <td>23916590</td>\n",
       "      <td>23924374</td>\n",
       "      <td>+</td>\n",
       "      <td>14932.7</td>\n",
       "      <td>1.286613e-06</td>\n",
       "      <td>E008</td>\n",
       "      <td>8.580000e-07</td>\n",
       "      <td>39</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>39/15/0</td>\n",
       "      <td>2/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000181826.10</th>\n",
       "      <th>RELL1</th>\n",
       "      <td>chr4</td>\n",
       "      <td>37590799</td>\n",
       "      <td>37686376</td>\n",
       "      <td>-</td>\n",
       "      <td>305.7</td>\n",
       "      <td>8.783635e-08</td>\n",
       "      <td>E001</td>\n",
       "      <td>5.270000e-08</td>\n",
       "      <td>9</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>9/8/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000019582.15</th>\n",
       "      <th>CD74</th>\n",
       "      <td>chr5</td>\n",
       "      <td>150400040</td>\n",
       "      <td>150412929</td>\n",
       "      <td>-</td>\n",
       "      <td>251.1</td>\n",
       "      <td>2.558465e-11</td>\n",
       "      <td>E001</td>\n",
       "      <td>1.280000e-11</td>\n",
       "      <td>29</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>29/15/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000096063.16</th>\n",
       "      <th>SRPK1</th>\n",
       "      <td>chr6</td>\n",
       "      <td>35832965</td>\n",
       "      <td>35921342</td>\n",
       "      <td>-</td>\n",
       "      <td>5143.9</td>\n",
       "      <td>8.668981e-03</td>\n",
       "      <td>J069</td>\n",
       "      <td>5.910000e-03</td>\n",
       "      <td>44</td>\n",
       "      <td>25</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>44/25/0</td>\n",
       "      <td>2/1/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000272398.6</th>\n",
       "      <th>CD24</th>\n",
       "      <td>chr6</td>\n",
       "      <td>106969830</td>\n",
       "      <td>106975627</td>\n",
       "      <td>-</td>\n",
       "      <td>4149.0</td>\n",
       "      <td>1.389731e-03</td>\n",
       "      <td>E012</td>\n",
       "      <td>7.480000e-04</td>\n",
       "      <td>18</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>18/8/0</td>\n",
       "      <td>2/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000135272.11</th>\n",
       "      <th>MDFIC</th>\n",
       "      <td>chr7</td>\n",
       "      <td>114922153</td>\n",
       "      <td>115019202</td>\n",
       "      <td>+</td>\n",
       "      <td>492.6</td>\n",
       "      <td>1.342079e-02</td>\n",
       "      <td>E006</td>\n",
       "      <td>8.750000e-03</td>\n",
       "      <td>17</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17/9/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000164934.14</th>\n",
       "      <th>DCAF13</th>\n",
       "      <td>chr8</td>\n",
       "      <td>103414713</td>\n",
       "      <td>103443453</td>\n",
       "      <td>+</td>\n",
       "      <td>914.0</td>\n",
       "      <td>1.321837e-02</td>\n",
       "      <td>E001</td>\n",
       "      <td>9.130000e-03</td>\n",
       "      <td>28</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>28/13/0</td>\n",
       "      <td>1/0/0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 chr      start        end strand  baseMean  \\\n",
       "geneID             geneName                                                   \n",
       "ENSG00000179051.14 RCC2         chr1   17406759   17439677      -    4226.7   \n",
       "ENSG00000053372.5  MRTO4        chr1   19251804   19260128      +    1654.2   \n",
       "ENSG00000163399.16 ATP1A1       chr1  116372667  116410261      +   15877.3   \n",
       "ENSG00000158716.9  DUSP23       chr1  159780931  159782543      +     544.5   \n",
       "ENSG00000116750.13 UCHL5        chr1  193012249  193060080      -    1049.6   \n",
       "ENSG00000175348.11 TMEM9B      chr11    8947201    8965011      -     404.2   \n",
       "ENSG00000149136.9  SSRP1       chr11   57325985   57335892      -    5121.7   \n",
       "ENSG00000167995.17 BEST1       chr11   61950062   61965515      +   49388.8   \n",
       "ENSG00000089597.18 GANAB       chr11   62624825   62646726      -    7773.9   \n",
       "ENSG00000165526.9  RPUSD4      chr11  126202095  126211692      -     774.5   \n",
       "ENSG00000110955.9  ATP5F1B     chr12   56638174   56645984      -   16800.9   \n",
       "ENSG00000100614.18 PPM1A       chr14   60245751   60299087      +    1052.1   \n",
       "ENSG00000259768.6  AC004943.2  chr16   72665122   72822781      +     212.3   \n",
       "ENSG00000131149.19 GSE1        chr16   85169524   85676204      +    1888.3   \n",
       "ENSG00000108298.11 RPL19       chr17   39200282   39204732      +   12380.4   \n",
       "ENSG00000131469.15 RPL27       chr17   42998272   43002959      +    5808.2   \n",
       "ENSG00000125398.8  SOX9        chr17   72121019   72126416      +    5438.2   \n",
       "ENSG00000177885.15 GRB2        chr17   75318075   75405709      -    2198.5   \n",
       "ENSG00000141551.14 CSNK1D      chr17   82239022   82273731      -    1921.0   \n",
       "ENSG00000115268.9  RPS15       chr19    1438357    1440494      +    6819.7   \n",
       "ENSG00000169021.6  UQCRFS1     chr19   29205319   29213151      -    2029.3   \n",
       "ENSG00000144485.11 HES6         chr2  238238266  238240662      -    2325.4   \n",
       "ENSG00000174748.22 RPL15        chr3   23916590   23924374      +   14932.7   \n",
       "ENSG00000181826.10 RELL1        chr4   37590799   37686376      -     305.7   \n",
       "ENSG00000019582.15 CD74         chr5  150400040  150412929      -     251.1   \n",
       "ENSG00000096063.16 SRPK1        chr6   35832965   35921342      -    5143.9   \n",
       "ENSG00000272398.6  CD24         chr6  106969830  106975627      -    4149.0   \n",
       "ENSG00000135272.11 MDFIC        chr7  114922153  115019202      +     492.6   \n",
       "ENSG00000164934.14 DCAF13       chr8  103414713  103443453      +     914.0   \n",
       "\n",
       "                               geneWisePadj mostSigID  mostSigPadjust  \\\n",
       "geneID             geneName                                             \n",
       "ENSG00000179051.14 RCC2        9.841564e-04      J032    5.370000e-04   \n",
       "ENSG00000053372.5  MRTO4       5.966044e-03      E001    3.610000e-03   \n",
       "ENSG00000163399.16 ATP1A1      1.311720e-02      E002    8.750000e-03   \n",
       "ENSG00000158716.9  DUSP23      8.004604e-03      E006    5.340000e-03   \n",
       "ENSG00000116750.13 UCHL5       8.947062e-03      E027    6.060000e-03   \n",
       "ENSG00000175348.11 TMEM9B      1.324392e-02      E007    9.050000e-03   \n",
       "ENSG00000149136.9  SSRP1       1.330073e-02      E032    8.980000e-03   \n",
       "ENSG00000167995.17 BEST1       8.909568e-03      E030    6.060000e-03   \n",
       "ENSG00000089597.18 GANAB       4.982960e-03      E048    2.930000e-03   \n",
       "ENSG00000165526.9  RPUSD4      1.052282e-02      E008    7.020000e-03   \n",
       "ENSG00000110955.9  ATP5F1B     3.383969e-03      E034    1.800000e-03   \n",
       "ENSG00000100614.18 PPM1A       4.766264e-03      E026    2.680000e-03   \n",
       "ENSG00000259768.6  AC004943.2  1.373038e-02      E009    8.750000e-03   \n",
       "ENSG00000131149.19 GSE1        9.742367e-03      E013    6.740000e-03   \n",
       "ENSG00000108298.11 RPL19       1.035171e-02      E001    7.020000e-03   \n",
       "ENSG00000131469.15 RPL27       5.558194e-03      E002    3.610000e-03   \n",
       "ENSG00000125398.8  SOX9        0.000000e+00      E002    1.460000e-24   \n",
       "ENSG00000177885.15 GRB2        1.397060e-02      E025    8.750000e-03   \n",
       "ENSG00000141551.14 CSNK1D      1.179758e-02      E001    7.990000e-03   \n",
       "ENSG00000115268.9  RPS15       5.101297e-03      E010    3.120000e-03   \n",
       "ENSG00000169021.6  UQCRFS1     7.833223e-04      J003    3.920000e-04   \n",
       "ENSG00000144485.11 HES6        1.312724e-02      E015    8.490000e-03   \n",
       "ENSG00000174748.22 RPL15       1.286613e-06      E008    8.580000e-07   \n",
       "ENSG00000181826.10 RELL1       8.783635e-08      E001    5.270000e-08   \n",
       "ENSG00000019582.15 CD74        2.558465e-11      E001    1.280000e-11   \n",
       "ENSG00000096063.16 SRPK1       8.668981e-03      J069    5.910000e-03   \n",
       "ENSG00000272398.6  CD24        1.389731e-03      E012    7.480000e-04   \n",
       "ENSG00000135272.11 MDFIC       1.342079e-02      E006    8.750000e-03   \n",
       "ENSG00000164934.14 DCAF13      1.321837e-02      E001    9.130000e-03   \n",
       "\n",
       "                               numExons  numKnown  numNovel  exonsSig  \\\n",
       "geneID             geneName                                             \n",
       "ENSG00000179051.14 RCC2              19        13         0         1   \n",
       "ENSG00000053372.5  MRTO4             13         8         0         1   \n",
       "ENSG00000163399.16 ATP1A1            40        28         0         1   \n",
       "ENSG00000158716.9  DUSP23             7         3         0         1   \n",
       "ENSG00000116750.13 UCHL5             37        20         0         2   \n",
       "ENSG00000175348.11 TMEM9B            14         8         0         1   \n",
       "ENSG00000149136.9  SSRP1             33        20         0         1   \n",
       "ENSG00000167995.17 BEST1             30        15         0         1   \n",
       "ENSG00000089597.18 GANAB             59        33         0         2   \n",
       "ENSG00000165526.9  RPUSD4            30        12         0         1   \n",
       "ENSG00000110955.9  ATP5F1B           34        18         0         1   \n",
       "ENSG00000100614.18 PPM1A             26        18         0         1   \n",
       "ENSG00000259768.6  AC004943.2        11        11         0         1   \n",
       "ENSG00000131149.19 GSE1              45        27         0         1   \n",
       "ENSG00000108298.11 RPL19             19         8         0         2   \n",
       "ENSG00000131469.15 RPL27             20         9         0         1   \n",
       "ENSG00000125398.8  SOX9               3         2         0         3   \n",
       "ENSG00000177885.15 GRB2              28        14         0         1   \n",
       "ENSG00000141551.14 CSNK1D            70        30         0         1   \n",
       "ENSG00000115268.9  RPS15             30        13         0         1   \n",
       "ENSG00000169021.6  UQCRFS1            2         1         0         0   \n",
       "ENSG00000144485.11 HES6              24         8         0         1   \n",
       "ENSG00000174748.22 RPL15             39        15         0         2   \n",
       "ENSG00000181826.10 RELL1              9         8         0         1   \n",
       "ENSG00000019582.15 CD74              29        15         0         1   \n",
       "ENSG00000096063.16 SRPK1             44        25         0         2   \n",
       "ENSG00000272398.6  CD24              18         8         0         2   \n",
       "ENSG00000135272.11 MDFIC             17         9         0         1   \n",
       "ENSG00000164934.14 DCAF13            28        13         0         1   \n",
       "\n",
       "                               knownSig  novelSig numFeatures numSig  \n",
       "geneID             geneName                                           \n",
       "ENSG00000179051.14 RCC2               1         0     19/13/0  1/1/0  \n",
       "ENSG00000053372.5  MRTO4              0         0      13/8/0  1/0/0  \n",
       "ENSG00000163399.16 ATP1A1             0         0     40/28/0  1/0/0  \n",
       "ENSG00000158716.9  DUSP23             0         0       7/3/0  1/0/0  \n",
       "ENSG00000116750.13 UCHL5              0         0     37/20/0  2/0/0  \n",
       "ENSG00000175348.11 TMEM9B             0         0      14/8/0  1/0/0  \n",
       "ENSG00000149136.9  SSRP1              0         0     33/20/0  1/0/0  \n",
       "ENSG00000167995.17 BEST1              0         0     30/15/0  1/0/0  \n",
       "ENSG00000089597.18 GANAB              1         0     59/33/0  2/1/0  \n",
       "ENSG00000165526.9  RPUSD4             0         0     30/12/0  1/0/0  \n",
       "ENSG00000110955.9  ATP5F1B            0         0     34/18/0  1/0/0  \n",
       "ENSG00000100614.18 PPM1A              0         0     26/18/0  1/0/0  \n",
       "ENSG00000259768.6  AC004943.2         0         0     11/11/0  1/0/0  \n",
       "ENSG00000131149.19 GSE1               0         0     45/27/0  1/0/0  \n",
       "ENSG00000108298.11 RPL19              0         0      19/8/0  2/0/0  \n",
       "ENSG00000131469.15 RPL27              0         0      20/9/0  1/0/0  \n",
       "ENSG00000125398.8  SOX9               2         0       3/2/0  3/2/0  \n",
       "ENSG00000177885.15 GRB2               0         0     28/14/0  1/0/0  \n",
       "ENSG00000141551.14 CSNK1D             0         0     70/30/0  1/0/0  \n",
       "ENSG00000115268.9  RPS15              0         0     30/13/0  1/0/0  \n",
       "ENSG00000169021.6  UQCRFS1            1         0       2/1/0  0/1/0  \n",
       "ENSG00000144485.11 HES6               0         0      24/8/0  1/0/0  \n",
       "ENSG00000174748.22 RPL15              0         0     39/15/0  2/0/0  \n",
       "ENSG00000181826.10 RELL1              0         0       9/8/0  1/0/0  \n",
       "ENSG00000019582.15 CD74               0         0     29/15/0  1/0/0  \n",
       "ENSG00000096063.16 SRPK1              1         0     44/25/0  2/1/0  \n",
       "ENSG00000272398.6  CD24               0         0      18/8/0  2/0/0  \n",
       "ENSG00000135272.11 MDFIC              0         0      17/9/0  1/0/0  \n",
       "ENSG00000164934.14 DCAF13             0         0     28/13/0  1/0/0  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genewise_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "65cef25c-8041-48c7-9e89-0d0689556267",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_expression_table(PATH):\n",
    "    '''\n",
    "    '''\n",
    "    table = pd.read_csv(PATH,sep='\\t').dropna(subset=['expr_case']).dropna(axis=1)\n",
    "    table[\"geneID\"] = table.featureID.str.split(':').str[0]\n",
    "    table[\"countbinID\"] = table.featureID.str.split(':').str[1]\n",
    "    \n",
    "    table = pd.concat([\n",
    "        gene2name.loc[table.geneID,:].reset_index(drop=True),\n",
    "        table.drop(\"featureID\",axis=1).reset_index(drop=True)\n",
    "    ],axis=1).set_index([\"geneID\",\"gene_name\",\"countbinID\"])\n",
    "    \n",
    "    return table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "4def3d55-d86b-47d6-b83d-bbf44ec57352",
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_table = read_expression_table('jscs/allGenes.expression.data.txt.gz')\n",
    "\n",
    "exp_table.index.get_level_values(1)\n",
    "exp_genes = set(exp_table.index.get_level_values(1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "ff4104a9-f50c-4a2a-8888-4c43acba8dd0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8187"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(exp_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "4bb190ec-0543-4eca-a661-3c040c2c2035",
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_binary_table = pd.DataFrame(\n",
    "    {\"spliced\":np.zeros(len(exp_genes))},\n",
    "    index=exp_genes, dtype=int\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "56e217e2-6728-4471-9792-bfb6dc183b6c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spliced</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>APP</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PIK3R4</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZFP36</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SAV1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF16</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TSPAN1</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COX7A2L</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EOLA2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC079305.2</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF594</th>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8187 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            spliced\n",
       "APP               0\n",
       "PIK3R4            0\n",
       "ZFP36             0\n",
       "SAV1              0\n",
       "ZNF16             0\n",
       "...             ...\n",
       "TSPAN1            0\n",
       "COX7A2L           0\n",
       "EOLA2             0\n",
       "AC079305.2        0\n",
       "ZNF594            0\n",
       "\n",
       "[8187 rows x 1 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "splice_binary_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "a1c8c08f-5a8f-492a-9fe1-9fb554d244ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_binary_table.loc[genewise_genes,:] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "40ced31e-c95c-4031-96e4-5149eacf87ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "splice_binary_table.to_csv(\"ChIP-seq-data/splice_binary_table.txt\",sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de4a3c5f-1675-42ce-bac5-dba239ff5bda",
   "metadata": {},
   "source": [
    "### ChIP-seq-data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8558127a-fbfd-48c0-bac1-46541037d10a",
   "metadata": {},
   "source": [
    "    %%bash \n",
    "    GTF=~/genomes/hg38/gencode.v34/gencode.v34.annotation.gtf\n",
    "\n",
    "    for f in ChIP-seq-data/bed_files/*.bed; do \n",
    "        o=${f/_IDX*/.txt}\n",
    "\n",
    "        echo \"================================================================\"\n",
    "        echo $o\n",
    "\n",
    "        cat $GTF \\\n",
    "        | awk -v FS='\\t' '$3==\"gene\"' \\\n",
    "        | bedtools intersect -wa -wb -a - -b $f \\\n",
    "        | cut -f 9,18 | uniq | awk -v FS='\\t' '{print $1,$2}' \\\n",
    "        | awk -v FS=\";\" -v OFS=\"\" '{gsub(\"\\\"\",\"\"); print $3}' \\\n",
    "        | awk -v FS=' ' -v OFS=\"\" '{print $2}' > $o\n",
    "\n",
    "    done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "da0d9a09-cdf3-452b-a592-1cfda467ebd1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ChIP-seq-data/bed_files/GFP_1_DOX_pos.txt\n",
      "ChIP-seq-data/bed_files/GFP_2_DOX_pos.txt\n",
      "ChIP-seq-data/bed_files/s303_1_DOX_neg.txt\n",
      "ChIP-seq-data/bed_files/s303_1_DOX_pos.txt\n",
      "ChIP-seq-data/bed_files/s303_2_DOX_neg.txt\n",
      "ChIP-seq-data/bed_files/s303_2_DOX_pos.txt\n",
      "ChIP-seq-data/bed_files/sox9_1_DOX_neg.txt\n",
      "ChIP-seq-data/bed_files/SOX9_1_DOX_pos.txt\n",
      "ChIP-seq-data/bed_files/sox9_2_DOX_neg.txt\n",
      "ChIP-seq-data/bed_files/SOX9_2_DOX_pos.txt\n"
     ]
    }
   ],
   "source": [
    "ls ChIP-seq-data/bed_files/*.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "553b3ca6-eb27-4789-946f-acf018e000b3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash \n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/SOX9_1_DOX_pos.txt \\\n",
    "ChIP-seq-data/bed_files/SOX9_2_DOX_pos.txt \\\n",
    "| uniq > ChIP-seq-data/SOX9_DOX_pos.txt\n",
    "\n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/sox9_1_DOX_neg.txt \\\n",
    "ChIP-seq-data/bed_files/sox9_2_DOX_neg.txt \\\n",
    "| uniq > ChIP-seq-data/SOX9_DOX_neg.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "b63f80d7-ce49-494f-bd44-0a5938d8e173",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%%bash \n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/GFP_1_DOX_pos.txt \\\n",
    "ChIP-seq-data/bed_files/GFP_2_DOX_pos.txt \\\n",
    "| uniq > ChIP-seq-data/GFP_DOX_pos.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "e3bafcf7-0346-411c-bdb3-52b48521d801",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash \n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/s303_1_DOX_neg.txt \\\n",
    "ChIP-seq-data/bed_files/s303_2_DOX_neg.txt \\\n",
    "| uniq > ChIP-seq-data/S303_DOX_neg.txt\n",
    "\n",
    "cat \\\n",
    "ChIP-seq-data/bed_files/s303_1_DOX_pos.txt \\\n",
    "ChIP-seq-data/bed_files/s303_2_DOX_pos.txt \\\n",
    "| uniq > ChIP-seq-data/S303_DOX_pos.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "6e19c672-3da1-4275-8b9e-15ae88f1a669",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ChIP-seq-data/GFP_DOX_pos.txt   ChIP-seq-data/SOX9_DOX_neg.txt\n",
      "ChIP-seq-data/S303_DOX_neg.txt  ChIP-seq-data/SOX9_DOX_pos.txt\n",
      "ChIP-seq-data/S303_DOX_pos.txt\n"
     ]
    }
   ],
   "source": [
    "ls ChIP-seq-data/*DOX*.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "aea4253e-d803-4970-848c-bb1df2681d8f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 24 20:13:31 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 8187 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000001\n",
      "pass = 1.000000\n",
      "z = -0.078693\n",
      "freq 0: 0.000245\t2\t8158\t2\t8187\n",
      "freq 1: 0.000000\t0\t29\t2\t8187\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 24 20:13:34 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 8187 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000002\n",
      "pass = 1.000000\n",
      "z = -0.092512\n",
      "freq 0: 0.000368\t3\t8158\t3\t8187\n",
      "freq 1: 0.000000\t0\t29\t3\t8187\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 24 20:13:36 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 8187 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000007\n",
      "pass = 1.000000\n",
      "z = -0.680293\n",
      "freq 0: 0.082741\t675\t8158\t677\t8187\n",
      "freq 1: 0.068966\t2\t29\t677\t8187\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 24 20:13:38 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 8187 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000001\n",
      "pass = 1.000000\n",
      "z = -0.078693\n",
      "freq 0: 0.000245\t2\t8158\t2\t8187\n",
      "freq 1: 0.000000\t0\t29\t2\t8187\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "The TEISERDIR environment variable is /data_gilbert/home/aarab/Workflows/TEISERv1.1\n",
      "Mon Jan 24 20:13:40 PST 2022\n",
      "Remove duplicates, create splice_binary_table.txt_GENESET/splice_binary_table.txt\n",
      "step 2: seed optimization.\n",
      "Expfile loaded: 8187 values...\n",
      "Quantizing the input vector...Done\n",
      "Number of clusters: 2\n",
      "calculating the p-value matrix.\n",
      "Allocating memory ... Done\n",
      "doing stats: mi = 0.000011\n",
      "pass = 1.000000\n",
      "z = -0.763760\n",
      "freq 0: 0.047561\t388\t8158\t389\t8187\n",
      "freq 1: 0.034483\t1\t29\t389\t8187\n",
      "step 7: drawing matrix.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n",
      "Reading MI data ... Done.\n",
      "Start drawing\n",
      "0\n",
      "Outputing EPS file splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps\n",
      "Convert to PDF splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "ps2pdf -dEPSCrop -dAutoRotatePages=/None splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.eps splice_binary_table.txt_GENESET/splice_binary_table.txt.summary.pdf\n",
      "Finished.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n",
      "Option suffix requires an argument\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "export TEISERDIR='/data_gilbert/home/aarab/Workflows/TEISERv1.1'\n",
    "exp=\"splice_binary_table.txt\"\n",
    "\n",
    "cd ChIP-seq-data/\n",
    "for gene in *DOX*.txt; do\n",
    "\n",
    "    perl ${TEISERDIR}/run_mi_gene_list.pl \\\n",
    "        --expfile=$exp \\\n",
    "        --genefile=$gene \\\n",
    "        --exptype=discrete \\\n",
    "        --species=human \\\n",
    "        --doremovedups=0 \\\n",
    "        --doremoveextra=0 \n",
    "        # --ebins=11 \\\n",
    "\n",
    "    perl /data_gilbert/home/aarab/Workflows/TEISERv1.1/Scripts/teiser_draw_matrix.pl \\\n",
    "        --pvmatrixfile=splice_binary_table.txt_GENESET/splice_binary_table.txt.matrix \\\n",
    "        --summaryfile=splice_binary_table.txt_GENESET/splice_binary_table.txt.summary \\\n",
    "        --expfile=splice_binary_table.txt_GENESET/splice_binary_table.txt \\\n",
    "        --quantized=1 \\\n",
    "        --colmap=/data_gilbert/home/aarab/Workflows/TEISERv1.1/Scripts/HEATMAPS/cmap_1.txt --order=0 --min=-3 --max=3 --cluster=5 \n",
    "\n",
    "    out=${gene/.txt/}\n",
    "    rm -rf $out\n",
    "    mv ${exp}_GENESET $out\n",
    "    cp $out/${exp}.summary.pdf ${out}.pdf\n",
    "done\n",
    "cd ../"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "9e019978-fb5c-4f0a-bdc0-768aac601236",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data_gilbert/home/aarab/iPAGE/run_mi_gene_list.pl\n"
     ]
    }
   ],
   "source": [
    "ls /data_gilbert/home/aarab/iPAGE/run_mi_gene_list.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "2b02c252-9211-4914-ab9f-07c569754ebe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "!echo ${TEISERDIR}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "afaa6fb6-1a66-4ef6-9517-b4060816ae5f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Jan 13 19:49:49 PST 2022\n"
     ]
    }
   ],
   "source": [
    "!date"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:gviz]",
   "language": "python",
   "name": "conda-env-gviz-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
